diff --git a/./output/redis/440385de-cfdc800a-81008bf9/src@db.no_comments_mergegen.c b/./output/redis/440385de-cfdc800a-81008bf9/src@db.no_comments_merged.c
index 85f36f1..c489f0d 100644
--- a/./output/redis/440385de-cfdc800a-81008bf9/src@db.no_comments_mergegen.c
+++ b/./output/redis/440385de-cfdc800a-81008bf9/src@db.no_comments_merged.c
@@ -227 +227,2 @@ void flushallCommand(client *c) {
-closeChildInfoPipe(); updateDictResizePolicy();
+        closeChildInfoPipe();
+        updateDictResizePolicy();
