diff --git a/./output/redis/440385de-cfdc800a-81008bf9/src@aof.no_comments_mergegen.c b/./output/redis/440385de-cfdc800a-81008bf9/src@aof.no_comments_merged.c
index 8e869c5..0f12e9f 100644
--- a/./output/redis/440385de-cfdc800a-81008bf9/src@aof.no_comments_mergegen.c
+++ b/./output/redis/440385de-cfdc800a-81008bf9/src@aof.no_comments_merged.c
@@ -131,0 +132,2 @@ static void killAppendOnlyChild(void) {
+    closeChildInfoPipe();
+    updateDictResizePolicy();
