diff --git a/./output/redis/440385de-cfdc800a-81008bf9/src@replication.no_comments_mergegen.c b/./output/redis/440385de-cfdc800a-81008bf9/src@replication.no_comments_merged.c
index 17bc125..12946ff 100644
--- a/./output/redis/440385de-cfdc800a-81008bf9/src@replication.no_comments_mergegen.c
+++ b/./output/redis/440385de-cfdc800a-81008bf9/src@replication.no_comments_merged.c
@@ -763,0 +764 @@ closeChildInfoPipe();
+            updateDictResizePolicy();
