--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/87521214-1d4296f1-33f9b08b/server.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/87521214-1d4296f1-33f9b08b/server.no_comments_merged.c
@@ -8,2 +7,0 @@
-#include <event.h>
-#include <paths.h>
@@ -97 +112,0 @@ static void server_tidy_event(__unused int fd, __unused short events,
-<<<<<<< HEAD
@@ -103,6 +117,0 @@ static void server_tidy_event(__unused int fd, __unused short events,
-|||||||
-  log_debug("%s: took %llu milliseconds", __func__, get_timer() - t);
-=======
-  log_debug("%s: took %llu milliseconds", __func__,
-            (unsigned long long)(get_timer() - t));
->>>>>>> 1d4296f17fde47186e2d09752a912559af034e37
