diff --git a/./output/system_core/763fa64d-d7846a2c-1fa02519/fs_mgr@fs_mgr_avb.no_comments_conflict.cpp b/./output/system_core/763fa64d-d7846a2c-1fa02519/fs_mgr@fs_mgr_avb.no_comments_merged.cpp
index ae74129..5b98099 100644
--- a/./output/system_core/763fa64d-d7846a2c-1fa02519/fs_mgr@fs_mgr_avb.no_comments_conflict.cpp
+++ b/./output/system_core/763fa64d-d7846a2c-1fa02519/fs_mgr@fs_mgr_avb.no_comments_merged.cpp
@@ -1 +0,0 @@
-<<<<<<< HEAD
@@ -3,6 +1,0 @@
-||||||| 1fa025193
-#include <errno.h>
-=======
-#include "fs_mgr_avb.h"
-#include <errno.h>
->>>>>>> d7846a2c
@@ -374 +366,0 @@ FsManagerAvbUniquePtr FsManagerAvbHandle::DoOpen(FsManagerAvbOps* avb_ops) {
-<<<<<<< HEAD
@@ -380,8 +371,0 @@ FsManagerAvbUniquePtr FsManagerAvbHandle::DoOpen(FsManagerAvbOps* avb_ops) {
-||||||| 1fa025193
-    FsManagerAvbOps avb_ops(device_file_by_name_prefix);
-    AvbSlotVerifyResult verify_result = avb_ops.AvbSlotVerify(
-        fs_mgr_get_slot_suffix(), avb_verifier->IsDeviceUnlocked(), &avb_handle->avb_slot_data_);
-=======
-    AvbSlotVerifyResult verify_result = avb_ops->AvbSlotVerify(
-        fs_mgr_get_slot_suffix(), avb_verifier->IsDeviceUnlocked(), &avb_handle->avb_slot_data_);
->>>>>>> d7846a2c
