diff --git a/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_mergegen.cc b/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_merged.cc
index bbda78c..f12e77f 100644
--- a/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_mergegen.cc
+++ b/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_merged.cc
@@ -638,0 +639,2 @@ TraceWriter::TraceWriter(File* trace_file,
+  uint64_t start_time_monotonic = start_time_ + (MicroTime() - GetMicroTime(GetTimestamp()));
+  uint16_t trace_version = GetTraceVersion(clock_source_, trace_format_version_);
