--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_merged.hpp
@@ -3,2 +2,0 @@
-#include "utf8proc_wrapper.hpp"
-#include "duckdb/function/built_in_functions.hpp"
@@ -6 +4,5 @@ namespace duckdb {
-struct LowerFun {
+struct UpperFun {
+ static constexpr const char *Name = "upper";
+ static constexpr const char *Parameters = "string";
+ static constexpr const char *Description = "Convert string to upper case.";
+ static constexpr const char *Example = "upper('Hello')";
@@ -7,0 +10,6 @@ struct LowerFun {
+};
+struct UcaseFun {
+ using ALIAS = UpperFun;
+ static constexpr const char *Name = "ucase";
+};
+struct LowerFun {
@@ -11,0 +20 @@ struct LowerFun {
+ static ScalarFunction GetFunction();
@@ -24,18 +32,0 @@ struct ConcatWsFun {
-struct UpperFun {
- static constexpr const char *Name = "upper";
- static constexpr const char *Parameters = "string";
- static constexpr const char *Description = "Convert string to upper case.";
- static constexpr const char *Example = "upper('Hello')";
- static ScalarFunction GetFunction();
-};
-struct UcaseFun {
- using ALIAS = UpperFun;
- static constexpr const char *Name = "ucase";
-};
-struct StripAccentsFun {
- static ScalarFunction GetFunction();
- static constexpr const char *Name = "strip_accents";
- static constexpr const char *Parameters = "string";
- static constexpr const char *Description = "Strips accents from string.";
- static constexpr const char *Example = "strip_accents('mühleisen')";
-};
@@ -43 +33,0 @@ struct ConcatFun {
- static ScalarFunction GetFunction();
@@ -47,0 +38 @@ struct ConcatFun {
+ static ScalarFunction GetFunction();
@@ -74,0 +66,35 @@ struct ConcatOperatorFun {
+struct PrefixFun {
+ static constexpr const char *Name = "prefix";
+ static constexpr const char *Parameters = "";
+ static constexpr const char *Description = "";
+ static constexpr const char *Example = "";
+ static ScalarFunction GetFunction();
+};
+struct SuffixFun {
+ static constexpr const char *Name = "suffix";
+ static constexpr const char *Parameters = "";
+ static constexpr const char *Description = "";
+ static constexpr const char *Example = "";
+ static ScalarFunction GetFunction();
+};
+struct ContainsFun {
+ static constexpr const char *Name = "contains";
+ static constexpr const char *Parameters = "string,search_string";
+ static constexpr const char *Description = "Return true if search_string is found within string.";
+ static constexpr const char *Example = "contains('abc', 'a')";
+ static ScalarFunctionSet GetFunctions();
+};
+struct StripAccentsFun {
+ static constexpr const char *Name = "strip_accents";
+ static constexpr const char *Parameters = "string";
+ static constexpr const char *Description = "Strips accents from string.";
+ static constexpr const char *Example = "strip_accents('mühleisen')";
+ static ScalarFunction GetFunction();
+};
+struct NFCNormalizeFun {
+ static constexpr const char *Name = "nfc_normalize";
+ static constexpr const char *Parameters = "string";
+ static constexpr const char *Description = "Convert string to Unicode NFC normalized string. Useful for comparisons and ordering if text data is mixed between NFC normalized and not.";
+ static constexpr const char *Example = "nfc_normalize('ardèch')";
+ static ScalarFunction GetFunction();
+};
@@ -121,80 +146,0 @@ struct ArrayLengthFun {
-struct LikeFun {
- static constexpr const char *Name = "~~";
- static constexpr const char *Parameters = "";
- static constexpr const char *Description = "";
- static constexpr const char *Example = "";
- static ScalarFunction GetFunction();
-};
-struct NotLikeFun {
- static constexpr const char *Name = "!~~";
- static constexpr const char *Parameters = "";
- static constexpr const char *Description = "";
- static constexpr const char *Example = "";
- static ScalarFunction GetFunction();
-};
-struct GlobPatternFun {
- static constexpr const char *Name = "~~~";
- static constexpr const char *Parameters = "";
- static constexpr const char *Description = "";
- static constexpr const char *Example = "";
- static ScalarFunction GetFunction();
-};
-struct ILikeFun {
- static constexpr const char *Name = "~~*";
- static constexpr const char *Parameters = "";
- static constexpr const char *Description = "";
- static constexpr const char *Example = "";
- static ScalarFunction GetFunction();
-};
-struct NotILikeFun {
- static constexpr const char *Name = "!~~*";
- static constexpr const char *Parameters = "";
- static constexpr const char *Description = "";
- static constexpr const char *Example = "";
- static ScalarFunction GetFunction();
-};
-struct LikeEscapeFun {
- static constexpr const char *Name = "like_escape";
- static constexpr const char *Parameters = "string,like_specifier,escape_character";
- static constexpr const char *Description =
-     "Returns true if the string matches the like_specifier (see Pattern Matching) using case-sensitive matching. "
-     "escape_character is used to search for wildcard characters in the string.";
- static constexpr const char *Example = "like_escape('a%c', 'a$%c', '$')";
- static ScalarFunction GetFunction();
-};
-struct NotLikeEscapeFun {
- static constexpr const char *Name = "not_like_escape";
- static constexpr const char *Parameters = "string,like_specifier,escape_character";
- static constexpr const char *Description =
-     "Returns false if the string matches the like_specifier (see Pattern Matching) using case-sensitive matching. "
-     "escape_character is used to search for wildcard characters in the string.";
- static constexpr const char *Example = "not_like_escape('a%c', 'a$%c', '$')";
- static ScalarFunction GetFunction();
-};
-struct IlikeEscapeFun {
- static constexpr const char *Name = "ilike_escape";
- static constexpr const char *Parameters = "string,like_specifier,escape_character";
- static constexpr const char *Description =
-     "Returns true if the string matches the like_specifier (see Pattern Matching) using case-insensitive matching. "
-     "escape_character is used to search for wildcard characters in the string.";
- static constexpr const char *Example = "ilike_escape('A%c', 'a$%C', '$')";
- static ScalarFunction GetFunction();
-};
-struct NotIlikeEscapeFun {
- static constexpr const char *Name = "not_ilike_escape";
- static constexpr const char *Parameters = "string,like_specifier,escape_character";
- static constexpr const char *Description =
-     "Returns false if the string matches the like_specifier (see Pattern Matching) using case-insensitive "
-     "matching. escape_character is used to search for wildcard characters in the string.";
- static constexpr const char *Example = "not_ilike_escape('A%c', 'a$%C', '$')";
- static ScalarFunction GetFunction();
-};
-struct NFCNormalizeFun {
- static ScalarFunction GetFunction();
- static constexpr const char *Name = "nfc_normalize";
- static constexpr const char *Parameters = "string";
- static constexpr const char *Description =
-     "Convert string to Unicode NFC normalized string. Useful for comparisons and ordering if text data is mixed "
-     "between NFC normalized and not.";
- static constexpr const char *Example = "nfc_normalize('ardèch')";
-};
@@ -202 +147,0 @@ struct SubstringFun {
- static string_t SubstringASCII(Vector &result, string_t input, int64_t offset, int64_t length);
@@ -205,3 +150 @@ struct SubstringFun {
- static constexpr const char *Description =
-     "Extract substring of length characters starting from character start. Note that a start value of 1 refers to "
-     "the first character of the string.";
+ static constexpr const char *Description = "Extract substring of length characters starting from character start. Note that a start value of 1 refers to the first character of the string.";
@@ -218,3 +161 @@ struct SubstringGraphemeFun {
- static constexpr const char *Description =
-     "Extract substring of length grapheme clusters starting from character start. Note that a start value of 1 "
-     "refers to the first character of the string.";
+ static constexpr const char *Description = "Extract substring of length grapheme clusters starting from character start. Note that a start value of 1 refers to the first character of the string.";
@@ -224 +165,5 @@ struct SubstringGraphemeFun {
-struct PrefixFun {
+struct StringSplitFun {
+ static constexpr const char *Name = "string_split";
+ static constexpr const char *Parameters = "string,separator";
+ static constexpr const char *Description = "Splits the string along the separator";
+ static constexpr const char *Example = "string_split('hello-world', '-')";
@@ -226,4 +170,0 @@ struct PrefixFun {
- static constexpr const char *Name = "prefix";
- static constexpr const char *Parameters = "";
- static constexpr const char *Description = "";
- static constexpr const char *Example = "";
@@ -231,6 +172,3 @@ struct PrefixFun {
-struct SuffixFun {
- static ScalarFunction GetFunction();
- static constexpr const char *Name = "suffix";
- static constexpr const char *Parameters = "";
- static constexpr const char *Description = "";
- static constexpr const char *Example = "";
+struct StrSplitFun {
+ using ALIAS = StringSplitFun;
+ static constexpr const char *Name = "str_split";
@@ -238 +176,13 @@ struct SuffixFun {
-struct ContainsFun {
+struct StringToArrayFun {
+ using ALIAS = StringSplitFun;
+ static constexpr const char *Name = "string_to_array";
+};
+struct SplitFun {
+ using ALIAS = StringSplitFun;
+ static constexpr const char *Name = "split";
+};
+struct StringSplitRegexFun {
+ static constexpr const char *Name = "string_split_regex";
+ static constexpr const char *Parameters = "string,separator";
+ static constexpr const char *Description = "Splits the string along the regex";
+ static constexpr const char *Example = "string_split_regex('hello␣world; 42', ';?␣')";
@@ -240,4 +190,8 @@ struct ContainsFun {
- static constexpr const char *Name = "contains";
- static constexpr const char *Parameters = "string,search_string";
- static constexpr const char *Description = "Return true if search_string is found within string.";
- static constexpr const char *Example = "contains('abc', 'a')";
+};
+struct StrSplitRegexFun {
+ using ALIAS = StringSplitRegexFun;
+ static constexpr const char *Name = "str_split_regex";
+};
+struct RegexpSplitToArrayFun {
+ using ALIAS = StringSplitRegexFun;
+ static constexpr const char *Name = "regexp_split_to_array";
@@ -264,2 +216 @@ struct RegexpReplaceFun {
- static constexpr const char *Description = "If string contains the regexp pattern, replaces the matching part with "
-                                            "replacement. A set of optional options can be set.";
+ static constexpr const char *Description = "If string contains the regexp pattern, replaces the matching part with replacement. A set of optional options can be set.";
@@ -272,3 +223 @@ struct RegexpExtractFun {
- static constexpr const char *Description =
-     "If string contains the regexp pattern, returns the capturing group specified by optional parameter group. The "
-     "group must be a constant value. If no group is given, it defaults to 0. A set of optional options can be set.";
+ static constexpr const char *Description = "If string contains the regexp pattern, returns the capturing group specified by optional parameter group. The group must be a constant value. If no group is given, it defaults to 0. A set of optional options can be set.";
@@ -286 +234,5 @@ struct RegexpExtractAllFun {
-struct StringSplitFun {
+struct RegexpEscapeFun {
+ static constexpr const char *Name = "regexp_escape";
+ static constexpr const char *Parameters = "string";
+ static constexpr const char *Description = "Escapes all potentially meaningful regexp characters in the input string";
+ static constexpr const char *Example = "regexp_escape('https://duckdb.org')";
@@ -288,4 +239,0 @@ struct StringSplitFun {
- static constexpr const char *Name = "string_split";
- static constexpr const char *Parameters = "string,separator";
- static constexpr const char *Description = "Splits the string along the separator";
- static constexpr const char *Example = "string_split('hello-world', '-')";
@@ -293,3 +241,6 @@ struct StringSplitFun {
-struct StrSplitFun {
- using ALIAS = StringSplitFun;
- static constexpr const char *Name = "str_split";
+struct LikeFun {
+ static constexpr const char *Name = "~~";
+ static constexpr const char *Parameters = "";
+ static constexpr const char *Description = "";
+ static constexpr const char *Example = "";
+ static ScalarFunction GetFunction();
@@ -297,3 +248,6 @@ struct StrSplitFun {
-struct StringToArrayFun {
- using ALIAS = StringSplitFun;
- static constexpr const char *Name = "string_to_array";
+struct NotLikeFun {
+ static constexpr const char *Name = "!~~";
+ static constexpr const char *Parameters = "";
+ static constexpr const char *Description = "";
+ static constexpr const char *Example = "";
+ static ScalarFunction GetFunction();
@@ -301,3 +255,6 @@ struct StringToArrayFun {
-struct SplitFun {
- using ALIAS = StringSplitFun;
- static constexpr const char *Name = "split";
+struct GlobPatternFun {
+ static constexpr const char *Name = "~~~";
+ static constexpr const char *Parameters = "";
+ static constexpr const char *Description = "";
+ static constexpr const char *Example = "";
+ static ScalarFunction GetFunction();
@@ -305 +262,5 @@ struct SplitFun {
-struct RegexpEscapeFun {
+struct ILikeFun {
+ static constexpr const char *Name = "~~*";
+ static constexpr const char *Parameters = "";
+ static constexpr const char *Description = "";
+ static constexpr const char *Example = "";
@@ -307,5 +267,0 @@ struct RegexpEscapeFun {
- static constexpr const char *Name = "regexp_escape";
- static constexpr const char *Parameters = "string";
- static constexpr const char *Description =
-     "Escapes all potentially meaningful regexp characters in the input string";
- static constexpr const char *Example = "regexp_escape('https://duckdb.org')";
@@ -313,6 +269,6 @@ struct RegexpEscapeFun {
-struct StringSplitRegexFun {
- static ScalarFunctionSet GetFunctions();
- static constexpr const char *Name = "string_split_regex";
- static constexpr const char *Parameters = "string,separator";
- static constexpr const char *Description = "Splits the string along the regex";
- static constexpr const char *Example = "string_split_regex('hello␣world; 42', ';?␣')";
+struct NotILikeFun {
+ static constexpr const char *Name = "!~~*";
+ static constexpr const char *Parameters = "";
+ static constexpr const char *Description = "";
+ static constexpr const char *Example = "";
+ static ScalarFunction GetFunction();
@@ -320,3 +276,6 @@ struct StringSplitRegexFun {
-struct StrSplitRegexFun {
- using ALIAS = StringSplitRegexFun;
- static constexpr const char *Name = "str_split_regex";
+struct LikeEscapeFun {
+ static constexpr const char *Name = "like_escape";
+ static constexpr const char *Parameters = "string,like_specifier,escape_character";
+ static constexpr const char *Description = "Returns true if the string matches the like_specifier (see Pattern Matching) using case-sensitive matching. escape_character is used to search for wildcard characters in the string.";
+ static constexpr const char *Example = "like_escape('a%c', 'a$%c', '$')";
+ static ScalarFunction GetFunction();
@@ -324,3 +283,20 @@ struct StrSplitRegexFun {
-struct RegexpSplitToArrayFun {
- using ALIAS = StringSplitRegexFun;
- static constexpr const char *Name = "regexp_split_to_array";
+struct NotLikeEscapeFun {
+ static constexpr const char *Name = "not_like_escape";
+ static constexpr const char *Parameters = "string,like_specifier,escape_character";
+ static constexpr const char *Description = "Returns false if the string matches the like_specifier (see Pattern Matching) using case-sensitive matching. escape_character is used to search for wildcard characters in the string.";
+ static constexpr const char *Example = "not_like_escape('a%c', 'a$%c', '$')";
+ static ScalarFunction GetFunction();
+};
+struct IlikeEscapeFun {
+ static constexpr const char *Name = "ilike_escape";
+ static constexpr const char *Parameters = "string,like_specifier,escape_character";
+ static constexpr const char *Description = "Returns true if the string matches the like_specifier (see Pattern Matching) using case-insensitive matching. escape_character is used to search for wildcard characters in the string.";
+ static constexpr const char *Example = "ilike_escape('A%c', 'a$%C', '$')";
+ static ScalarFunction GetFunction();
+};
+struct NotIlikeEscapeFun {
+ static constexpr const char *Name = "not_ilike_escape";
+ static constexpr const char *Parameters = "string,like_specifier,escape_character";
+ static constexpr const char *Description = "Returns false if the string matches the like_specifier (see Pattern Matching) using case-insensitive matching. escape_character is used to search for wildcard characters in the string.";
+ static constexpr const char *Example = "not_ilike_escape('A%c', 'a$%C', '$')";
+ static ScalarFunction GetFunction();
@@ -329 +304,0 @@ struct MD5Fun {
- static ScalarFunctionSet GetFunctions();
@@ -333,0 +309 @@ struct MD5Fun {
+ static ScalarFunctionSet GetFunctions();
@@ -336 +311,0 @@ struct MD5NumberFun {
- static ScalarFunctionSet GetFunctions();
@@ -340,0 +316 @@ struct MD5NumberFun {
+ static ScalarFunctionSet GetFunctions();
@@ -343 +318,0 @@ struct SHA1Fun {
- static ScalarFunctionSet GetFunctions();
@@ -347,0 +323 @@ struct SHA1Fun {
+ static ScalarFunctionSet GetFunctions();
@@ -350 +325,0 @@ struct SHA256Fun {
- static ScalarFunctionSet GetFunctions();
@@ -354,0 +330 @@ struct SHA256Fun {
+ static ScalarFunctionSet GetFunctions();
