--- a/./output/system_core/f615d49b-5f4e8eac-d43b6152/init@ueventd.no_comments_conflict.cpp
+++ b/./output/system_core/f615d49b-5f4e8eac-d43b6152/init@ueventd.no_comments_merged.cpp
@@ -4,9 +3,0 @@
-<<<<<<< HEAD
-||||||| d43b61521
-#include <grp.h>
-#include <poll.h>
-#include <pwd.h>
-=======
-#include <grp.h>
-#include <pwd.h>
->>>>>>> 5f4e8eac
@@ -129 +119,0 @@ DeviceHandler CreateDeviceHandler() {
-<<<<<<< HEAD
@@ -143,17 +132,0 @@ int ueventd_main(int argc, char** argv) {
-||||||| d43b61521
-    pollfd ufd;
-    ufd.events = POLLIN;
-    ufd.fd = get_device_fd();
-    while (true) {
-        ufd.revents = 0;
-        int nr = poll(&ufd, 1, -1);
-        if (nr <= 0) {
-            continue;
-        }
-        if (ufd.revents & POLLIN) {
-            handle_device_fd();
-        }
-    }
-=======
-    device_poll();
->>>>>>> 5f4e8eac
