diff --git a/./output/system_core/f615d49b-5f4e8eac-d43b6152/init@ueventd.no_comments_mergegen.cpp b/./output/system_core/f615d49b-5f4e8eac-d43b6152/init@ueventd.no_comments_merged.cpp
index 7e6d512..39185f4 100644
--- a/./output/system_core/f615d49b-5f4e8eac-d43b6152/init@ueventd.no_comments_mergegen.cpp
+++ b/./output/system_core/f615d49b-5f4e8eac-d43b6152/init@ueventd.no_comments_merged.cpp
@@ -120 +120,13 @@ DeviceHandler CreateDeviceHandler() {
-int ueventd_main(int argc, char** argv) { umask(000); InitKernelLogging(argv); LOG(INFO) << "ueventd started!"; selinux_callback cb; cb.func_log = selinux_klog_callback; selinux_set_callback(SELINUX_CB_LOG, cb); DeviceHandler device_handler = CreateDeviceHandler(); UeventListener uevent_listener; if (access(COLDBOOT_DONE, F_OK) != 0) { ColdBoot cold_boot(uevent_listener, device_handler); cold_boot.Run(); }
+int ueventd_main(int argc, char** argv) {
+    umask(000);
+    InitKernelLogging(argv);
+    LOG(INFO) << "ueventd started!";
+    selinux_callback cb;
+    cb.func_log = selinux_klog_callback;
+    selinux_set_callback(SELINUX_CB_LOG, cb);
+    DeviceHandler device_handler = CreateDeviceHandler();
+    UeventListener uevent_listener;
+    if (access(COLDBOOT_DONE, F_OK) != 0) {
+        ColdBoot cold_boot(uevent_listener, device_handler);
+        cold_boot.Run();
+    }
