--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/34807388-141cd784-08be8832/server.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/34807388-141cd784-08be8832/server.no_comments_merged.c
@@ -8,2 +7,0 @@
-#include <event.h>
-#include <paths.h>
@@ -53,2 +60,8 @@ int server_is_marked(struct session *s, struct winlink *wl,
-int server_check_marked(void) { return (cmd_find_valid_state(&marked_pane)); }
-intserver_create_socket(int flags, char **cause, uint64_t flags) {
+int
+server_check_marked(void)
+{
+ return (cmd_find_valid_state(&marked_pane));
+}
+int
+server_create_socket(uint64_t flags, char **cause)
+{
