diff --git a/./output/tmux/34807388-141cd784-08be8832/server.no_comments_mergegen.c b/./output/tmux/34807388-141cd784-08be8832/server.no_comments_merged.c
index 0495265..a0a36af 100644
--- a/./output/tmux/34807388-141cd784-08be8832/server.no_comments_mergegen.c
+++ b/./output/tmux/34807388-141cd784-08be8832/server.no_comments_merged.c
@@ -65 +65,2 @@ server_check_marked(void)
-int server_create_socket(uint64_t flags, char **cause)
+int
+server_create_socket(uint64_t flags, char **cause)
