--- a/./output/grpc/664918f9-de6defa6-502c796a/src@compiler@ruby_generator_string-inl.no_comments_mergegen.h
+++ b/./output/grpc/664918f9-de6defa6-502c796a/src@compiler@ruby_generator_string-inl.no_comments_merged.h
@@ -76 +76 @@ inline std::string RubyPackage(const grpc::protobuf::FileDescriptor* file) {
-inline std::string RubyTypeOf(const grpc::protobuf::Descriptor* descriptor) { std::string proto_type = descriptor->full_name();
+inline std::string RubyTypeOf(const grpc::protobuf::Descriptor* descriptor) {
