diff --git a/./output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_mergegen.cc b/./output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_merged.cc
index a2643f1..ad9f773 100644
--- a/./output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_mergegen.cc
+++ b/./output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3 +3 @@ namespace grpc {
-namespace grpc {std::string Version() { return "1.31.2"; }
+std::string Version() { return "1.31.0-dev"; }
