--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_merged.cc
@@ -37 +37 @@ ServerBuilder::ServerBuilder()
-~ServerBuilder() {
+ServerBuilder::~ServerBuilder() {
@@ -179,5 +179 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
-  ChannelArguments args;
-  for (const auto& option : options_) {
-    option->UpdateArguments(&args);
-    option->UpdatePlugins(&plugins_);
-  }
+  grpc::ChannelArguments args;
@@ -185,8 +180,0 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
-    grpc_channel_args c_args = args.c_channel_args();
-    const grpc_arg* arg =
-        grpc_channel_args_find(&c_args, GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH);
-    if (arg != nullptr) {
-      gpr_log(
-          GPR_ERROR,
-          "gRPC ServerBuilder receives multiple max_receive_message_length");
-    }
@@ -195 +183 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
-  if (max_send_message_size_ >= 0) {
+  if (max_send_message_size_ >= -1) {
@@ -197,0 +186,4 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
+  for (const auto& option : options_) {
+    option->UpdateArguments(&args);
+    option->UpdatePlugins(&plugins_);
+  }
