--- a/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_mergegen.cc
+++ b/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_merged.cc
@@ -773 +773,2 @@ class StressTest {
-streamers[i] = db_->NewIterator(readoptionscopy[i], column_family); iters[i]->Seek(prefix_slices[i]);
+      iters[i] = db_->NewIterator(readoptionscopy[i], column_family);
+      iters[i]->Seek(prefix_slices[i]);
