diff --git a/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_conflict.cc b/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_merged.cc
index a0997df..fe60046 100644
--- a/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_conflict.cc
+++ b/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_merged.cc
@@ -773 +772,0 @@ class StressTest {
-<<<<<<< HEAD
@@ -775,6 +773,0 @@ class StressTest {
-      iters[i]->SeekToFirst();
-||||||| 2bad3cb0d
-      iters[i] = db_->NewIterator(readoptionscopy[i]);
-      iters[i]->SeekToFirst();
-=======
-      iters[i] = db_->NewIterator(readoptionscopy[i]);
@@ -782 +774,0 @@ class StressTest {
->>>>>>> 9b8a2b52
@@ -927,7 +918,0 @@ class StressTest {
-<<<<<<< HEAD
-          read_opts.prefix = &prefix;
-          Iterator* iter = db_->NewIterator(read_opts, column_family);
-||||||| 2bad3cb0d
-          read_opts.prefix = &prefix;
-          Iterator* iter = db_->NewIterator(read_opts);
-=======
@@ -935,2 +920 @@ class StressTest {
-          Iterator* iter = db_->NewIterator(read_opts);
->>>>>>> 9b8a2b52
+          Iterator* iter = db_->NewIterator(read_opts, column_family);
