diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/275832ae-ebaff6f9-83ab62e2/include@rocksdb@options.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/275832ae-ebaff6f9-83ab62e2/include@rocksdb@options.no_comments_merged.h
index bb247f3..fc2a75c 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/275832ae-ebaff6f9-83ab62e2/include@rocksdb@options.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/275832ae-ebaff6f9-83ab62e2/include@rocksdb@options.no_comments_merged.h
@@ -30,6 +30,2 @@ enum CompressionType : char {
-  kNoCompression = 0x0,
-  kSnappyCompression = 0x1,
-  kZlibCompression = 0x2,
-  kBZip2Compression = 0x3,
-  kLZ4Compression = 0x4,
-  kLZ4HCCompression = 0x5
+  kNoCompression = 0x0, kSnappyCompression = 0x1, kZlibCompression = 0x2,
+  kBZip2Compression = 0x3, kLZ4Compression = 0x4, kLZ4HCCompression = 0x5
@@ -59,0 +56 @@ struct ColumnFamilyOptions {
+  std::shared_ptr<CompactionFilterFactoryV2> compaction_filter_factory_v2;
@@ -113,0 +111 @@ struct ColumnFamilyOptions {
+  uint32_t min_partial_merge_operands;
@@ -151 +149,6 @@ struct DBOptions {
-  enum { NONE, NORMAL, SEQUENTIAL, WILLNEED } access_hint_on_compaction_start;
+  enum {
+    NONE,
+    NORMAL,
+    SEQUENTIAL,
+    WILLNEED
+  } access_hint_on_compaction_start;
@@ -160 +163,3 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
-  std::shared_ptr<CompactionFilterFactoryV2> compaction_filter_factory_v2;
+  Options() :
+    DBOptions(),
+    ColumnFamilyOptions() {}
@@ -166 +170,0 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
-  uint32_t min_partial_merge_operands;
