--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@client_context.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@client_context.no_comments_merged.h
@@ -20 +20 @@
-struct census_context
+struct census_context;
@@ -22 +22,16 @@ struct census_context
-namespace grpc_impl {
+namespace grpc {
+class ServerContext;
+class ServerContextBase;
+class CallbackServerContext;
+namespace internal {
+template <class InputMessage, class OutputMessage>
+class CallbackUnaryCallImpl;
+template <class Request, class Response>
+class ClientCallbackReaderWriterImpl;
+template <class Response>
+class ClientCallbackReaderImpl;
+template <class Request>
+class ClientCallbackWriterImpl;
+class ClientCallbackUnaryImpl;
+class ClientContextAccessor;
+}
@@ -37,5 +51,0 @@ class ClientAsyncResponseReader;
-}
-namespace grpc {
-class ServerContext;
-class ServerContextBase;
-class CallbackServerContext;
@@ -177,0 +190,24 @@ class ClientContext {
+  template <class R>
+  friend class ::grpc::ClientReader;
+  template <class W>
+  friend class ::grpc::ClientWriter;
+  template <class W, class R>
+  friend class ::grpc::ClientReaderWriter;
+  template <class R>
+  friend class ::grpc::ClientAsyncReader;
+  template <class W>
+  friend class ::grpc::ClientAsyncWriter;
+  template <class W, class R>
+  friend class ::grpc::ClientAsyncReaderWriter;
+  template <class R>
+  friend class ::grpc::ClientAsyncResponseReader;
+  template <class InputMessage, class OutputMessage>
+  friend class ::grpc::internal::BlockingUnaryCallImpl;
+  template <class InputMessage, class OutputMessage>
+  friend class ::grpc::internal::CallbackUnaryCallImpl;
+  template <class Request, class Response>
+  friend class ::grpc::internal::ClientCallbackReaderWriterImpl;
+  template <class Response>
+  friend class ::grpc::internal::ClientCallbackReaderImpl;
+  template <class Request>
+  friend class ::grpc::internal::ClientCallbackWriterImpl;
