diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_merged.h
index 83fe6fa..e078581 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_merged.h
@@ -11 +11 @@
-struct grpc_completion_queue
+struct grpc_completion_queue;
@@ -12,0 +13,3 @@ namespace grpc_impl {
+class ServerContextBase;
+}
+namespace grpc {
@@ -22,0 +26,13 @@ class ServerWriter;
+namespace internal {
+template <class W, class R>
+class ServerReaderWriterBody;
+template <class ServiceType, class RequestType, class ResponseType>
+class RpcMethodHandler;
+template <class ServiceType, class RequestType, class ResponseType>
+class ClientStreamingHandler;
+template <class ServiceType, class RequestType, class ResponseType>
+class ServerStreamingHandler;
+template <class Streamer, bool WriteNeeded>
+class TemplatedBidiStreamingHandler;
+template <::grpc::StatusCode code>
+class ErrorMethodHandler;
@@ -24 +39,0 @@ class ServerWriter;
-namespace grpc {
@@ -43 +58,3 @@ public:
-      : CompletionQueue(grpc_completion_queue_attributes{}
+      : CompletionQueue(grpc_completion_queue_attributes{
+            GRPC_CQ_CURRENT_VERSION, GRPC_CQ_NEXT, GRPC_CQ_DEFAULT_POLLING,
+            nullptr}) {}
@@ -85,0 +103,22 @@ private:
+  template <class R>
+  friend class ::grpc::ClientReader;
+  template <class W>
+  friend class ::grpc::ClientWriter;
+  template <class W, class R>
+  friend class ::grpc::ClientReaderWriter;
+  template <class R>
+  friend class ::grpc::ServerReader;
+  template <class W>
+  friend class ::grpc::ServerWriter;
+  template <class W, class R>
+  friend class ::grpc::internal::ServerReaderWriterBody;
+  template <class ServiceType, class RequestType, class ResponseType>
+  friend class ::grpc::internal::RpcMethodHandler;
+  template <class ServiceType, class RequestType, class ResponseType>
+  friend class ::grpc::internal::ClientStreamingHandler;
+  template <class ServiceType, class RequestType, class ResponseType>
+  friend class ::grpc::internal::ServerStreamingHandler;
+  template <class Streamer, bool WriteNeeded>
+  friend class ::grpc::internal::TemplatedBidiStreamingHandler;
+  template <::grpc::StatusCode code>
+  friend class ::grpc::internal::ErrorMethodHandler;
@@ -87,0 +127,2 @@ private:
+  template <class InputMessage, class OutputMessage>
+  friend class ::grpc::internal::BlockingUnaryCallImpl;
@@ -88,0 +130,2 @@ private:
+  template <class Op1, class Op2, class Op3, class Op4, class Op5, class Op6>
+  friend class ::grpc::internal::CallOpSet;
@@ -183 +226,4 @@ private:
-      : CompletionQueue(grpc_completion_queue_attributes{}
+      : CompletionQueue(grpc_completion_queue_attributes{
+            GRPC_CQ_CURRENT_VERSION, completion_type, polling_type,
+            shutdown_cb}),
+        polling_type_(polling_type) {}
