--- a/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@client_context.no_comments_conflict.h
+++ b/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@client_context.no_comments_merged.h
@@ -22 +22 @@ struct grpc_call;
-namespace grpc_impl {
+namespace grpc {
@@ -26,2 +25,0 @@ class CallbackServerContext;
-}
-namespace grpc {
@@ -54,13 +51,0 @@ class ClientAsyncResponseReader;
-<<<<<<< HEAD
-||||||| 249c26148e
-class ServerContextBase;
-class CallbackServerContext;
-}
-namespace grpc {
-=======
-}
-namespace grpc {
-class ServerContext;
-class ServerContextBase;
-class CallbackServerContext;
->>>>>>> 3c1cb4f3
