diff --git a/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_conflict.h b/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_merged.h
index 8a798d1..e078581 100644
--- a/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_conflict.h
+++ b/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_merged.h
@@ -124 +123,0 @@ class CompletionQueue : private ::grpc::GrpcLibraryCodegen {
-<<<<<<< HEAD
@@ -126,6 +124,0 @@ class CompletionQueue : private ::grpc::GrpcLibraryCodegen {
-  friend class ::grpc_impl::ServerContextBase;
-||||||| 249c26148e
-  friend class ::grpc_impl::internal::ErrorMethodHandler;
-  friend class ::grpc_impl::ServerContextBase;
-=======
-  friend class ::grpc_impl::internal::ErrorMethodHandler;
@@ -133 +125,0 @@ class CompletionQueue : private ::grpc::GrpcLibraryCodegen {
->>>>>>> 3c1cb4f3
