--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/src@compiler@cpp_generator.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/67667eb7-3c1cb4f3-249c2614/src@compiler@cpp_generator.no_comments_merged.cc
@@ -1250,7 +1249,0 @@ void PrintHeaderServerMethodStreamedUnary(
-<<<<<<< HEAD
-                   "        [this](::grpc_impl::ServerContext* context,\n"
-                   "               ::grpc::ServerUnaryStreamer<\n"
-|||||||
-                   "        [this](::grpc_impl::ServerContext* context,\n"
-                   "               ::grpc_impl::ServerUnaryStreamer<\n"
-=======
@@ -1258,2 +1251 @@ void PrintHeaderServerMethodStreamedUnary(
-                   "               ::grpc_impl::ServerUnaryStreamer<\n"
->>>>>>> 3c1cb4f3001f1a880401c72566bc6355bf5a5c30
+                   "               ::grpc::ServerUnaryStreamer<\n"
@@ -1311,7 +1302,0 @@ void PrintHeaderServerMethodSplitStreaming(
-<<<<<<< HEAD
-                   "        [this](::grpc_impl::ServerContext* context,\n"
-                   "               ::grpc::ServerSplitStreamer<\n"
-|||||||
-                   "        [this](::grpc_impl::ServerContext* context,\n"
-                   "               ::grpc_impl::ServerSplitStreamer<\n"
-=======
@@ -1319,2 +1304 @@ void PrintHeaderServerMethodSplitStreaming(
-                   "               ::grpc_impl::ServerSplitStreamer<\n"
->>>>>>> 3c1cb4f3001f1a880401c72566bc6355bf5a5c30
+                   "               ::grpc::ServerSplitStreamer<\n"
@@ -2106,7 +2089,0 @@ void PrintSourceService(grpc_generator::Printer* printer,
-<<<<<<< HEAD
-          "           ::grpc_impl::ServerContext* ctx,\n"
-          "           ::grpc::ServerReader<$Request$>* reader,\n"
-|||||||
-          "           ::grpc_impl::ServerContext* ctx,\n"
-          "           ::grpc_impl::ServerReader<$Request$>* reader,\n"
-=======
@@ -2114,2 +2091 @@ void PrintSourceService(grpc_generator::Printer* printer,
-          "           ::grpc_impl::ServerReader<$Request$>* reader,\n"
->>>>>>> 3c1cb4f3001f1a880401c72566bc6355bf5a5c30
+          "           ::grpc::ServerReader<$Request$>* reader,\n"
@@ -2141,7 +2116,0 @@ void PrintSourceService(grpc_generator::Printer* printer,
-<<<<<<< HEAD
-                     "           ::grpc_impl::ServerContext* ctx,\n"
-                     "           ::grpc::ServerReaderWriter<$Response$,\n"
-|||||||
-                     "           ::grpc_impl::ServerContext* ctx,\n"
-                     "           ::grpc_impl::ServerReaderWriter<$Response$,\n"
-=======
@@ -2149,2 +2118 @@ void PrintSourceService(grpc_generator::Printer* printer,
-                     "           ::grpc_impl::ServerReaderWriter<$Response$,\n"
->>>>>>> 3c1cb4f3001f1a880401c72566bc6355bf5a5c30
+                     "           ::grpc::ServerReaderWriter<$Response$,\n"
