--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/fd4d3e87-6f03e49e-bbfb44e9/utf8.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/fd4d3e87-6f03e49e-bbfb44e9/utf8.no_comments_merged.c
@@ -6 +5,0 @@
-#include <vis.h>
@@ -15,3 +14,8 @@ struct utf8_item {
-static intutf8_cmp(struct utf8_item *ui1, struct utf8_item *ui2) {
-  if (ui1->size < ui2->size) return (-1);
-  if (ui1->size > ui2->size) return (1);
+RB_HEAD(utf8_tree, utf8_item);
+static int
+utf8_cmp(struct utf8_item *ui1, struct utf8_item *ui2)
+{
+ if (ui1->size < ui2->size)
+  return (-1);
+ if (ui1->size > ui2->size)
+  return (1);
@@ -19,0 +24 @@ static intutf8_cmp(struct utf8_item *ui1, struct utf8_item *ui2) {
+RB_GENERATE_STATIC(utf8_tree, utf8_item, entry, utf8_cmp);
@@ -126,18 +172,0 @@ static enum utf8_state utf8_width(struct utf8_data *ud, int *width) {
-<<<<<<< HEAD
-#ifdef HAVE_UTF8PROC
-  width = utf8proc_wcwidth(wc);
-#else
-  width = wcwidth(wc);
-#endif
-  if (width < 0 || width > 0xff) {
-    log_debug("Unicode %04lx, wcwidth() %d", (long)wc, width);
-#ifndef __OpenBSD__
-    if (width < 0) return (1);
-#endif
-    return (-1);
-|||||||
-  width = wcwidth(wc);
-  if (width < 0 || width > 0xff) {
-    log_debug("Unicode %04lx, wcwidth() %d", (long)wc, width);
-    return (-1);
-=======
@@ -151 +180,0 @@ static enum utf8_state utf8_width(struct utf8_data *ud, int *width) {
->>>>>>> 6f03e49e68dfe0d9c0c7d49079c4383b26aca916
@@ -185 +225,26 @@ enum utf8_state utf8_append(struct utf8_data *ud, u_char ch) {
-int utf8_strvis(char *dst, const char *src, size_t len, int flag) {
+static int
+utf8_width(wchar_t wc)
+{
+ int width;
+#ifdef HAVE_UTF8PROC
+ width = utf8proc_wcwidth(wc);
+#else
+ width = wcwidth(wc);
+#endif
+ if (width < 0 || width > 0xff) {
+  log_debug("Unicode %04lx, wcwidth() %d", (long)wc, width);
+#ifndef __OpenBSD__
+  if (width < 0)
+   return (1);
+#endif
+  return (-1);
+ }
+ return (width);
+}
+ * Encode len characters from src into dst, which is guaranteed to have four
+ * bytes available for each character from src (for \abc or UTF-8) plus space
+ * for \0.
+ */
+int
+utf8_strvis(char *dst, const char *src, size_t len, int flag)
+{
