diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_merged.h
index 0ebd602..883aedf 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_merged.h
@@ -42,2 +42 @@
-#define REDIS_REQUEST_MAX_SIZE \
-  (1024 * 1024 * 256)
+#define REDIS_REQUEST_MAX_SIZE (1024*1024*256)
@@ -46,2 +45 @@
-#define REDIS_MAX_LOGMSG_LEN \
-  1024
+#define REDIS_MAX_LOGMSG_LEN 1024
@@ -83,8 +80,0 @@
-#define REDIS_IO_LOAD 1
-#define REDIS_IO_SAVE 2
-#define REDIS_IO_LOADINPROG 4
-#define REDIS_IO_SAVEINPROG 8
-#define REDIS_IO_ONLYLOADS 1
-#define REDIS_IO_ASAP 2
-#define REDIS_MAX_COMPLETED_JOBS_PROCESSED 1
-#define REDIS_THREAD_STACK_SIZE (1024 * 1024 * 4)
@@ -96 +85,0 @@
-#define REDIS_IO_WAIT 32
@@ -99,2 +88,2 @@
-#define REDIS_UNBLOCKED \
-  256
+#define REDIS_UNBLOCKED 256
+#define REDIS_LUA_CLIENT 512
@@ -109,4 +98,2 @@
-#define REDIS_REPL_WAIT_BGSAVE_START \
-  3
-#define REDIS_REPL_WAIT_BGSAVE_END \
-  4
+#define REDIS_REPL_WAIT_BGSAVE_START 3
+#define REDIS_REPL_WAIT_BGSAVE_END 4
@@ -125,0 +113 @@
+#define REDIS_NOTUSED(V) ((void) V)
@@ -147,0 +136,2 @@
+#define redisAssert(_e) ((_e)?(void)0 : (_redisAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define redisPanic(_e) _redisPanic(#_e,__FILE__,__LINE__),_exit(1)
@@ -167,0 +158,6 @@ typedef struct vmPointer {
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = REDIS_STRING; \
+    _var.encoding = REDIS_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0);
@@ -225,6 +221,8 @@ struct sharedObjectsStruct {
-      *colon, *nullbulk, *nullmultibulk, *queued, *emptymultibulk,
-      *wrongtypeerr, *nokeyerr, *syntaxerr, *sameobjecterr, *outofrangeerr,
-      *noscripterr, *loadingerr, *plus, *select0, *select1, *select2, *select3,
-      *select4, *select5, *select6, *select7, *select8, *select9, *messagebulk,
-      *pmessagebulk, *subscribebulk, *unsubscribebulk, *mbulk3, *mbulk4,
-      *psubscribebulk, *punsubscribebulk, *integers[REDIS_SHARED_INTEGERS];
+    *colon, *nullbulk, *nullmultibulk, *queued,
+    *emptymultibulk, *wrongtypeerr, *nokeyerr, *syntaxerr, *sameobjecterr,
+    *outofrangeerr, *noscripterr, *loadingerr, *plus,
+    *select0, *select1, *select2, *select3, *select4,
+    *select5, *select6, *select7, *select8, *select9,
+    *messagebulk, *pmessagebulk, *subscribebulk, *unsubscribebulk, *mbulk3,
+    *mbulk4, *psubscribebulk, *punsubscribebulk,
+    *integers[REDIS_SHARED_INTEGERS];
@@ -271,4 +269 @@ typedef struct clusterLink {
-#define REDIS_NODE_NULL_NAME \
-  "\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000" \
-  "\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000" \
-  "\000\000\000\000"
+#define REDIS_NODE_NULL_NAME "\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000"
@@ -464,2 +459 @@ typedef void redisCommandProc(redisClient *c);
-typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc,
-                              int *numkeys, int flags);
+typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc, int *numkeys, int flags);
@@ -570,2 +564 @@ unsigned long listTypeLength(robj *subject);
-listTypeIterator *listTypeInitIterator(robj *subject, int index,
-                                       unsigned char direction);
+listTypeIterator *listTypeInitIterator(robj *subject, int index, unsigned char direction);
@@ -610,2 +603 @@ robj *createZsetZiplistObject(void);
-int getLongFromObjectOrReply(redisClient *c, robj *o, long *target,
-                             const char *msg);
+int getLongFromObjectOrReply(redisClient *c, robj *o, long *target, const char *msg);
@@ -613,4 +605,2 @@ int checkType(redisClient *c, robj *o, int type);
-int getLongLongFromObjectOrReply(redisClient *c, robj *o, long long *target,
-                                 const char *msg);
-int getDoubleFromObjectOrReply(redisClient *c, robj *o, double *target,
-                               const char *msg);
+int getLongLongFromObjectOrReply(redisClient *c, robj *o, long long *target, const char *msg);
+int getDoubleFromObjectOrReply(redisClient *c, robj *o, double *target, const char *msg);
@@ -646,2 +636 @@ void backgroundSaveDoneHandler(int exitcode, int bysignal);
-int rdbSaveKeyValuePair(FILE *fp, robj *key, robj *val, time_t expireitme,
-                        time_t now);
+int rdbSaveKeyValuePair(FILE *fp, robj *key, robj *val, time_t expireitme, time_t now);
@@ -654,2 +643 @@ void flushAppendOnlyFile(void);
-void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv,
-                        int argc);
+void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv, int argc);
@@ -706,2 +694 @@ void hashTypeTryObjectEncoding(robj *subject, robj **o1, robj **o2);
-int hashTypeGet(robj *o, robj *key, robj **objval, unsigned char **v,
-                unsigned int *vlen);
+int hashTypeGet(robj *o, robj *key, robj **objval, unsigned char **v, unsigned int *vlen);
@@ -716,2 +703 @@ int hashTypeNext(hashTypeIterator *hi);
-int hashTypeCurrent(hashTypeIterator *hi, int what, robj **objval,
-                    unsigned char **v, unsigned int *vlen);
+int hashTypeCurrent(hashTypeIterator *hi, int what, robj **objval, unsigned char **v, unsigned int *vlen);
@@ -747,2 +733 @@ void signalFlushedDb(int dbid);
-unsigned int GetKeysInSlot(unsigned int hashslot, robj **keys,
-                           unsigned int count);
+unsigned int GetKeysInSlot(unsigned int hashslot, robj **keys, unsigned int count);
@@ -751,2 +736 @@ unsigned int GetKeysInSlot(unsigned int hashslot, robj **keys,
-int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc,
-                        int *numkeys, int flags);
+int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc, int *numkeys, int flags);
@@ -754,6 +738,3 @@ void getKeysFreeResult(int *result);
-int *noPreloadGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                      int *numkeys, int flags);
-int *renameGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                   int *numkeys, int flags);
-int *zunionInterGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                        int *numkeys, int flags);
+int *noPreloadGetKeys(struct redisCommand *cmd,robj **argv, int argc, int *numkeys, int flags);
+int *renameGetKeys(struct redisCommand *cmd,robj **argv, int argc, int *numkeys, int flags);
+int *zunionInterGetKeys(struct redisCommand *cmd,robj **argv, int argc, int *numkeys, int flags);
@@ -766,2 +747 @@ void clusterCron(void);
-clusterNode *getNodeByQuery(redisClient *c, struct redisCommand *cmd,
-                            robj **argv, int argc, int *hashslot, int *ask);
+clusterNode *getNodeByQuery(redisClient *c, struct redisCommand *cmd, robj **argv, int argc, int *hashslot, int *ask);
@@ -895,0 +876,6 @@ void evalShaCommand(redisClient *c);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
