--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/3cc82662-5ff8b6e5-9d37e2ab/libs@renderengine@skia@AutoBackendTexture.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/3cc82662-5ff8b6e5-9d37e2ab/libs@renderengine@skia@AutoBackendTexture.no_comments_merged.cpp
@@ -10,10 +9,0 @@
-#include <SkImage.h>
-#include <include/gpu/ganesh/SkImageGanesh.h>
-#include <include/gpu/ganesh/SkSurfaceGanesh.h>
-#include <include/gpu/ganesh/gl/GrGLBackendSurface.h>
-#include <include/gpu/ganesh/vk/GrVkBackendSurface.h>
-#include <include/gpu/vk/GrVkTypes.h>
-#include <android/hardware_buffer.h>
-#include "ColorSpaces.h"
-#include "log/log_main.h"
-#include "utils/Trace.h"
