diff --git a/./output/frameworks_native/3cc82662-5ff8b6e5-9d37e2ab/libs@renderengine@skia@AutoBackendTexture.no_comments_mergegen.cpp b/./output/frameworks_native/3cc82662-5ff8b6e5-9d37e2ab/libs@renderengine@skia@AutoBackendTexture.no_comments_merged.cpp
index 03a4818..81c51a5 100644
--- a/./output/frameworks_native/3cc82662-5ff8b6e5-9d37e2ab/libs@renderengine@skia@AutoBackendTexture.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/3cc82662-5ff8b6e5-9d37e2ab/libs@renderengine@skia@AutoBackendTexture.no_comments_merged.cpp
@@ -13 +13,3 @@ namespace skia {
-AutoBackendTexture::AutoBackendTexture(std::unique_ptr<SkiaBackendTexture> backendTexture, CleanupManager& cleanupMgr) : mCleanupMgr(cleanupMgr), mBackendTexture(std::move(backendTexture)) {}
+AutoBackendTexture::AutoBackendTexture(std::unique_ptr<SkiaBackendTexture> backendTexture,
+                                       CleanupManager& cleanupMgr)
+      : mCleanupMgr(cleanupMgr), mBackendTexture(std::move(backendTexture)) {}
