--- a/./output/duckdb/4776329b-f8ecea77-92cf6f1d/src@optimizer@optimizer.no_comments_mergegen.cpp
+++ b/./output/duckdb/4776329b-f8ecea77-92cf6f1d/src@optimizer@optimizer.no_comments_merged.cpp
@@ -11,0 +12 @@
+#include "duckdb/optimizer/empty_result_pullup.hpp"
@@ -15 +15,0 @@
-#include "duckdb/optimizer/empty_result_pullup.hpp"
@@ -26,0 +27 @@
+#include "duckdb/optimizer/sampling_pushdown.hpp"
@@ -31 +31,0 @@
-#include "duckdb/planner/binder.hpp"
