--- a/./output/duckdb/fec26383-30d444e4-5c0a1ed7/tools@pythonpkg@src@include@duckdb_python@pyfilesystem.no_comments_conflict.hpp
+++ b/./output/duckdb/fec26383-30d444e4-5c0a1ed7/tools@pythonpkg@src@include@duckdb_python@pyfilesystem.no_comments_merged.hpp
@@ -4 +3,0 @@
-<<<<<<< HEAD
@@ -8,9 +6,0 @@
-||||||| 5c0a1ed76e
-#include "duckdb_python/pybind_wrapper.hpp"
-#include "duckdb_python/python_object_container.hpp"
-#include <vector>
-=======
-#include "duckdb/common/vector.hpp"
-#include "duckdb_python/pybind_wrapper.hpp"
-#include "duckdb_python/python_object_container.hpp"
->>>>>>> 30d444e4
