diff --git a/./output/system_core/75175b60-796cd474-52aa2334/fastboot@device@commands.no_comments_mergegen.cpp b/./output/system_core/75175b60-796cd474-52aa2334/fastboot@device@commands.no_comments_merged.cpp
index c4b3bb7..8142af1 100644
--- a/./output/system_core/75175b60-796cd474-52aa2334/fastboot@device@commands.no_comments_mergegen.cpp
+++ b/./output/system_core/75175b60-796cd474-52aa2334/fastboot@device@commands.no_comments_merged.cpp
@@ -131 +131,19 @@ bool GetVarHandler(FastbootDevice* device, const std::vector<std::string>& args)
-bool OemPostWipeData(FastbootDevice* device) { auto fastboot_hal = device->fastboot_hal(); if (!fastboot_hal) { return false; } Result ret; auto ret_val = fastboot_hal->doOemSpecificErase([&](Result result) { ret = result; }); if (!ret_val.isOk()) { return false; } if (ret.status == Status::NOT_SUPPORTED) { return false; } else if (ret.status != Status::SUCCESS) { device->WriteStatus(FastbootResult::FAIL, ret.message); } else { device->WriteStatus(FastbootResult::OKAY, "Erasing succeeded"); } return true; }
+bool OemPostWipeData(FastbootDevice* device) {
+    auto fastboot_hal = device->fastboot_hal();
+    if (!fastboot_hal) {
+        return false;
+    }
+    Result ret;
+    auto ret_val = fastboot_hal->doOemSpecificErase([&](Result result) { ret = result; });
+    if (!ret_val.isOk()) {
+        return false;
+    }
+    if (ret.status == Status::NOT_SUPPORTED) {
+        return false;
+    } else if (ret.status != Status::SUCCESS) {
+        device->WriteStatus(FastbootResult::FAIL, ret.message);
+    } else {
+        device->WriteStatus(FastbootResult::OKAY, "Erasing succeeded");
+    }
+    return true;
+}
