diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_merged.h
index 686bdb5..bf58a51 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_merged.h
@@ -14 +14,8 @@ enum ValueType : unsigned char {
-kTypeDeletion = 0x0,kTypeValue = 0x1,kTypeMerge = 0x2,kTypeLogData = 0x3,kTypeColumnFamilyDeletion = 0x4,kTypeColumnFamilyValue = 0x5,kTypeColumnFamilyMerge = 0x6, kMaxValue = 0x7F
+  kTypeDeletion = 0x0,
+  kTypeValue = 0x1,
+  kTypeMerge = 0x2,
+  kTypeLogData = 0x3,
+  kTypeColumnFamilyDeletion = 0x4,
+  kTypeColumnFamilyValue = 0x5,
+  kTypeColumnFamilyMerge = 0x6,
+  kMaxValue = 0x7F
@@ -55 +62 @@ public:
-  ~InternalKeyComparator(){}
+  virtual ~InternalKeyComparator() {}
@@ -142 +148,0 @@ inline LookupKey::~LookupKey() {
-  if (start_ !{
