diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_conflict.h b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_merged.h
index 883b960..bf58a51 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_conflict.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_merged.h
@@ -17 +16,0 @@ enum ValueType : unsigned char {
-<<<<<<< HEAD
@@ -22,4 +20,0 @@ enum ValueType : unsigned char {
-||||||| 183ba01a0
-  kTypeLogData = 0x3
-=======
-  kTypeLogData = 0x3,
@@ -27 +21,0 @@ enum ValueType : unsigned char {
->>>>>>> 4564b2e8
