--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_mergegen.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_merged.h
@@ -182 +182,4 @@ new_superversion = create_superversion ? new SuperVersion() : nullptr;
-Status WriteLevel0TableForRecovery(ColumnFamilyData* cfd, MemTable* mem, VersionEdit* edit); Status WriteLevel0Table(ColumnFamilyData* cfd, std::vector<MemTable*>& mems, VersionEdit* edit, uint64_t* filenumber);
+  Status WriteLevel0TableForRecovery(ColumnFamilyData* cfd, MemTable* mem,
+                                     VersionEdit* edit);
+  Status WriteLevel0Table(ColumnFamilyData* cfd, autovector<MemTable*>& mems,
+                          VersionEdit* edit, uint64_t* filenumber);
