--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@dbformat.no_comments_merged.h
@@ -55 +62 @@ public:
-  ~InternalKeyComparator(){}
+  virtual ~InternalKeyComparator() {}
@@ -142 +148,0 @@ inline LookupKey::~LookupKey() {
-  if (start_ !{
