diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_iter.no_comments_conflict.cc b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_iter.no_comments_merged.cc
index 9e37df5..a73afe9 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_iter.no_comments_conflict.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_iter.no_comments_merged.cc
@@ -180,8 +179,0 @@ void DBIter::FindNextUserEntryInternal(bool skipping) {
-<<<<<<< HEAD
-          case kTypeColumnFamilyDeletion:
-          case kTypeColumnFamilyValue:
-          case kTypeColumnFamilyMerge:
-          case kTypeLogData:
-||||||| 183ba01a0
-          case kTypeLogData:
-=======
@@ -189 +180,0 @@ void DBIter::FindNextUserEntryInternal(bool skipping) {
->>>>>>> 4564b2e8
