diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_merged.cc
index 2efce99..39ac3a4 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_merged.cc
@@ -11 +10,0 @@
-#include "util/perf_context_imp.h"
@@ -13 +11,0 @@
-#include "util/mutexlock.h"
@@ -14,0 +13,2 @@
+#include "util/mutexlock.h"
+#include "util/perf_context_imp.h"
@@ -26,2 +26,15 @@ namespace rocksdb {
-MemTable::MemTable(const InternalKeyComparator& cmp, const ColumnFamilyOptions& options, const Options& options): comparator_(cmp), refs_(0), arena_(options.arena_block_size), arena_impl_(options.arena_block_size), &arena_)), table_(options.memtable_factory->CreateMemTableRep(comparator_, &arena_impl_)), flush_in_progress_(false), flush_completed_(false), file_number_(0), first_seqno_(0), mem_next_logfile_number_(0), mem_logfile_number_(0), locks_(options.inplace_update_support ? options.inplace_update_num_locks
-                                            : 0), prefix_extractor_(options.prefix_extractor) {
+MemTable::MemTable(const InternalKeyComparator& cmp,
+                   const ColumnFamilyOptions& options)
+    : comparator_(cmp),
+      refs_(0),
+      arena_(options.arena_block_size),
+      table_(options.memtable_factory->CreateMemTableRep(comparator_, &arena_)),
+      flush_in_progress_(false),
+      flush_completed_(false),
+      file_number_(0),
+      first_seqno_(0),
+      mem_next_logfile_number_(0),
+      mem_logfile_number_(0),
+      locks_(options.inplace_update_support ? options.inplace_update_num_locks
+                                            : 0),
+      prefix_extractor_(options.prefix_extractor) {
@@ -63 +76,2 @@ public:
-  MemTableIterator(const MemTable& mem, const ReadOptions& options): mem_(mem), iter_(), dynamic_prefix_seek_(false), valid_(false) {
+  MemTableIterator(const MemTable& mem, const ReadOptions& options)
+      : mem_(mem), iter_(), dynamic_prefix_seek_(false), valid_(false) {
@@ -235,8 +248,0 @@ bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
-<<<<<<< HEAD
-        case kTypeColumnFamilyDeletion:
-        case kTypeColumnFamilyValue:
-        case kTypeColumnFamilyMerge:
-        case kTypeLogData:
-||||||| 183ba01a0
-        case kTypeLogData:
-=======
@@ -244 +249,0 @@ bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
->>>>>>> 4564b2e8f9a2da07fa84fb5f3bba007469052dfb
@@ -298 +303,4 @@ void MemTable::Update(SequenceNumber seq,
-bool MemTable::UpdateCallback(SequenceNumber seq, const Slice& key, const Slice& delta, const Options& options) {
+bool MemTable::UpdateCallback(SequenceNumber seq,
+                              const Slice& key,
+                              const Slice& delta,
+                              const Options& options) {
