--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_mergegen.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_merged.cc
@@ -249 +249 @@ bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
-default: case kTypeColumnFamilyDeletion: case kTypeColumnFamilyValue: case kTypeColumnFamilyMerge: case kTypeLogData: assert(false);
+        default:
