--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@table_cache.no_comments_conflict.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@table_cache.no_comments_merged.h
@@ -45 +44,0 @@ class TableCache {
-<<<<<<< HEAD
@@ -47,11 +45,0 @@ class TableCache {
-  Status FindTable(const EnvOptions& toptions, uint64_t file_number,
-                   uint64_t file_size, Cache::Handle**, bool* table_io=nullptr,
-                   const bool no_io = false);
-||||||| 183ba01a0
-  std::shared_ptr<Cache> cache_;
-  Status FindTable(const EnvOptions& toptions, uint64_t file_number,
-                   uint64_t file_size, Cache::Handle**, bool* table_io=nullptr,
-                   const bool no_io = false);
-=======
-  std::shared_ptr<Cache> cache_;
->>>>>>> 4564b2e8
