diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_mergegen.cc b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_merged.cc
index c7eec76..4bd0df9 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_mergegen.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_merged.cc
@@ -49 +49 @@ static std::string PrintContents(WriteBatch* b) {
-state.append(")"); count++; break; case kTypeColumnFamilyDeletion: case kTypeColumnFamilyValue: case kTypeColumnFamilyMerge: default: assert(false);
+      default:
