--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch.no_comments_conflict.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch.no_comments_merged.cc
@@ -191 +190,0 @@ class MemTableInserter : public WriteBatch::Handler {
-<<<<<<< HEAD
@@ -203,2 +202,4 @@ class MemTableInserter : public WriteBatch::Handler {
-    if (options->inplace_update_support &&
-        mem->Update(sequence_, kTypeValue, key, value)) {
+    if (!options->inplace_update_support) {
+      mem->Add(sequence_, kTypeValue, key, value);
+    } else if (options->inplace_callback == nullptr) {
+      mem->Update(sequence_, key, value);
@@ -206,13 +206,0 @@ class MemTableInserter : public WriteBatch::Handler {
-||||||| 183ba01a0
-  virtual void Put(const Slice& key, const Slice& value) {
-    if (options_->inplace_update_support
-        && mem_->Update(sequence_, kTypeValue, key, value)) {
-      RecordTick(options_->statistics.get(), NUMBER_KEYS_UPDATED);
-=======
-  virtual void Put(const Slice& key, const Slice& value) {
-    if (!options_->inplace_update_support) {
-      mem_->Add(sequence_, kTypeValue, key, value);
-    } else if (options_->inplace_callback == nullptr) {
-      mem_->Update(sequence_, key, value);
-      RecordTick(options_->statistics.get(), NUMBER_KEYS_UPDATED);
->>>>>>> 4564b2e8
@@ -220,6 +208 @@ class MemTableInserter : public WriteBatch::Handler {
-<<<<<<< HEAD
-      mem->Add(sequence_, kTypeValue, key, value);
-||||||| 183ba01a0
-      mem_->Add(sequence_, kTypeValue, key, value);
-=======
-      if (mem_->UpdateCallback(sequence_, key, value, *options_)) {
+      if (mem->UpdateCallback(sequence_, key, value, *options)) {
@@ -236,2 +219 @@ class MemTableInserter : public WriteBatch::Handler {
-        auto status =
-          options_->inplace_callback(s.ok() ? prev_buffer: nullptr,
+        auto status = options->inplace_callback(s.ok() ? prev_buffer : nullptr,
@@ -241,2 +223,2 @@ class MemTableInserter : public WriteBatch::Handler {
-          mem_->Add(sequence_, kTypeValue, key, Slice(prev_buffer, prev_size));
-          RecordTick(options_->statistics.get(), NUMBER_KEYS_WRITTEN);
+          mem->Add(sequence_, kTypeValue, key, Slice(prev_buffer, prev_size));
+          RecordTick(options->statistics.get(), NUMBER_KEYS_WRITTEN);
@@ -244,2 +226,2 @@ class MemTableInserter : public WriteBatch::Handler {
-          mem_->Add(sequence_, kTypeValue, key, Slice(merged_value));
-          RecordTick(options_->statistics.get(), NUMBER_KEYS_WRITTEN);
+          mem->Add(sequence_, kTypeValue, key, Slice(merged_value));
+          RecordTick(options->statistics.get(), NUMBER_KEYS_WRITTEN);
@@ -248 +229,0 @@ class MemTableInserter : public WriteBatch::Handler {
->>>>>>> 4564b2e8
@@ -252 +232,0 @@ class MemTableInserter : public WriteBatch::Handler {
-<<<<<<< HEAD
@@ -261,5 +240,0 @@ class MemTableInserter : public WriteBatch::Handler {
-||||||| 183ba01a0
-  virtual void Merge(const Slice& key, const Slice& value) {
-=======
-  virtual void Merge(const Slice& key, const Slice& value) {
->>>>>>> 4564b2e8
