diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch.no_comments_mergegen.cc b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch.no_comments_merged.cc
index ea650e2..a6717f4 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch.no_comments_mergegen.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch.no_comments_merged.cc
@@ -191 +191,18 @@ class MemTableInserter : public WriteBatch::Handler {
-bool IgnoreUpdate() { return log_number_ != 0 && log_number_ < cf_mems_->GetLogNumber(); } virtual void PutCF(uint32_t column_family_id, const Slice& key, const Slice& value) { bool found = cf_mems_->Seek(column_family_id);
+  bool IgnoreUpdate() {
+    return log_number_ != 0 && log_number_ < cf_mems_->GetLogNumber();
+  }
+  virtual void PutCF(uint32_t column_family_id, const Slice& key,
+                     const Slice& value) {
+    bool found = cf_mems_->Seek(column_family_id);
+    if (!found || IgnoreUpdate()) {
+      return;
+    }
+    MemTable* mem = cf_mems_->GetMemTable();
+    const Options* options = cf_mems_->GetFullOptions();
+    if (!options->inplace_update_support) {
+      mem->Add(sequence_, kTypeValue, key, value);
+    } else if (options->inplace_callback == nullptr) {
+      mem->Update(sequence_, key, value);
+      RecordTick(options->statistics.get(), NUMBER_KEYS_UPDATED);
+    } else {
+      if (mem->UpdateCallback(sequence_, key, value, *options)) {
@@ -193 +210,20 @@ bool IgnoreUpdate() { return log_number_ != 0 && log_number_ < cf_mems_->GetLogN
-if (mem_->UpdateCallback(sequence_, key, value, *options_)) { } else {
+        SnapshotImpl read_from_snapshot;
+        read_from_snapshot.number_ = sequence_;
+        ReadOptions ropts;
+        ropts.snapshot = &read_from_snapshot;
+        std::string prev_value;
+        std::string merged_value;
+        Status s = db_->Get(ropts, key, &prev_value);
+        char* prev_buffer = const_cast<char*>(prev_value.c_str());
+        uint32_t prev_size = prev_value.size();
+        auto status = options->inplace_callback(s.ok() ? prev_buffer : nullptr,
+                                                s.ok() ? &prev_size : nullptr,
+                                                value, &merged_value);
+        if (status == UpdateStatus::UPDATED_INPLACE) {
+          mem->Add(sequence_, kTypeValue, key, Slice(prev_buffer, prev_size));
+          RecordTick(options->statistics.get(), NUMBER_KEYS_WRITTEN);
+        } else if (status == UpdateStatus::UPDATED) {
+          mem->Add(sequence_, kTypeValue, key, Slice(merged_value));
+          RecordTick(options->statistics.get(), NUMBER_KEYS_WRITTEN);
+        }
+      }
@@ -197 +233,8 @@ if (mem_->UpdateCallback(sequence_, key, value, *options_)) { } else {
-virtual void MergeCF(uint32_t column_family_id, const Slice& key, const Slice& value) { bool found = cf_mems_->Seek(column_family_id); if (!found || IgnoreUpdate()) { return; } MemTable* mem = cf_mems_->GetMemTable(); const Options* options = cf_mems_->GetFullOptions();
+  virtual void MergeCF(uint32_t column_family_id, const Slice& key,
+                       const Slice& value) {
+    bool found = cf_mems_->Seek(column_family_id);
+    if (!found || IgnoreUpdate()) {
+      return;
+    }
+    MemTable* mem = cf_mems_->GetMemTable();
+    const Options* options = cf_mems_->GetFullOptions();
