diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_merged.cc
index 812fe7e..4bd0df9 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_merged.cc
@@ -49,8 +48,0 @@ static std::string PrintContents(WriteBatch* b) {
-<<<<<<< HEAD
-      case kTypeColumnFamilyDeletion:
-      case kTypeColumnFamilyValue:
-      case kTypeColumnFamilyMerge:
-      case kTypeLogData:
-||||||| 183ba01a0
-      case kTypeLogData:
-=======
@@ -58 +49,0 @@ static std::string PrintContents(WriteBatch* b) {
->>>>>>> 4564b2e8f9a2da07fa84fb5f3bba007469052dfb
@@ -75 +66 @@ class WriteBatchTest {};
-TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
+TEST(WriteBatchTest, Empty) {
@@ -77,18 +68,58 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Put(2, Slice("twofoo"), Slice("bar2"));
-  batch.Put(8, Slice("eightfoo"), Slice("bar8"));
-  batch.Delete(8, Slice("eightfoo"));
-  batch.Merge(3, Slice("threethree"), Slice("3three"));
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Merge(Slice("omom"), Slice("nom"));
-  TestHandler handler;
-  batch.Iterate(&handler);
-  ASSERT_EQ(
-      "Put(foo, bar)"
-      "PutCF(2, twofoo, bar2)"
-      "PutCF(8, eightfoo, bar8)"
-      "DeleteCF(8, eightfoo)"
-      "MergeCF(3, threethree, 3three)"
-      "Put(foo, bar)"
-      "Merge(omom, nom)",
-      handler.seen);
+  ASSERT_EQ("", PrintContents(&batch));
+  ASSERT_EQ(0, WriteBatchInternal::Count(&batch));
+  ASSERT_EQ(0, batch.Count());
+}
+TEST(WriteBatchTest, Multiple) {
+  WriteBatch batch;
+  batch.Put(Slice("foo"), Slice("bar"));
+  batch.Delete(Slice("box"));
+  batch.Put(Slice("baz"), Slice("boo"));
+  WriteBatchInternal::SetSequence(&batch, 100);
+  ASSERT_EQ(100U, WriteBatchInternal::Sequence(&batch));
+  ASSERT_EQ(3, WriteBatchInternal::Count(&batch));
+  ASSERT_EQ("Put(baz, boo)@102"
+            "Delete(box)@101"
+            "Put(foo, bar)@100",
+            PrintContents(&batch));
+  ASSERT_EQ(3, batch.Count());
+}
+TEST(WriteBatchTest, Corruption) {
+  WriteBatch batch;
+  batch.Put(Slice("foo"), Slice("bar"));
+  batch.Delete(Slice("box"));
+  WriteBatchInternal::SetSequence(&batch, 200);
+  Slice contents = WriteBatchInternal::Contents(&batch);
+  WriteBatchInternal::SetContents(&batch,
+                                  Slice(contents.data(),contents.size()-1));
+  ASSERT_EQ("Put(foo, bar)@200"
+            "Corruption: bad WriteBatch Delete",
+            PrintContents(&batch));
+}
+TEST(WriteBatchTest, Append) {
+  WriteBatch b1, b2;
+  WriteBatchInternal::SetSequence(&b1, 200);
+  WriteBatchInternal::SetSequence(&b2, 300);
+  WriteBatchInternal::Append(&b1, &b2);
+  ASSERT_EQ("",
+            PrintContents(&b1));
+  ASSERT_EQ(0, b1.Count());
+  b2.Put("a", "va");
+  WriteBatchInternal::Append(&b1, &b2);
+  ASSERT_EQ("Put(a, va)@200",
+            PrintContents(&b1));
+  ASSERT_EQ(1, b1.Count());
+  b2.Clear();
+  b2.Put("b", "vb");
+  WriteBatchInternal::Append(&b1, &b2);
+  ASSERT_EQ("Put(a, va)@200"
+            "Put(b, vb)@201",
+            PrintContents(&b1));
+  ASSERT_EQ(2, b1.Count());
+  b2.Delete("foo");
+  WriteBatchInternal::Append(&b1, &b2);
+  ASSERT_EQ("Put(a, va)@200"
+            "Put(b, vb)@202"
+            "Put(b, vb)@201"
+            "Delete(foo)@203",
+            PrintContents(&b1));
+  ASSERT_EQ(4, b1.Count());
@@ -130,22 +161 @@ struct TestHandler : public WriteBatch::Handler {
-TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-  WriteBatch batch;
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Put(2, Slice("twofoo"), Slice("bar2"));
-  batch.Put(8, Slice("eightfoo"), Slice("bar8"));
-  batch.Delete(8, Slice("eightfoo"));
-  batch.Merge(3, Slice("threethree"), Slice("3three"));
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Merge(Slice("omom"), Slice("nom"));
-  TestHandler handler;
-  batch.Iterate(&handler);
-  ASSERT_EQ(
-      "Put(foo, bar)"
-      "PutCF(2, twofoo, bar2)"
-      "PutCF(8, eightfoo, bar8)"
-      "DeleteCF(8, eightfoo)"
-      "MergeCF(3, threethree, 3three)"
-      "Put(foo, bar)"
-      "Merge(omom, nom)",
-      handler.seen);
-}
-TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
+TEST(WriteBatchTest, Blob) {
@@ -153,7 +163,14 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Put(2, Slice("twofoo"), Slice("bar2"));
-  batch.Put(8, Slice("eightfoo"), Slice("bar8"));
-  batch.Delete(8, Slice("eightfoo"));
-  batch.Merge(3, Slice("threethree"), Slice("3three"));
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Merge(Slice("omom"), Slice("nom"));
+  batch.Put(Slice("k1"), Slice("v1"));
+  batch.Put(Slice("k2"), Slice("v2"));
+  batch.Put(Slice("k3"), Slice("v3"));
+  batch.PutLogData(Slice("blob1"));
+  batch.Delete(Slice("k2"));
+  batch.PutLogData(Slice("blob2"));
+  batch.Merge(Slice("foo"), Slice("bar"));
+  ASSERT_EQ(5, batch.Count());
+  ASSERT_EQ("Merge(foo, bar)@4"
+            "Put(k1, v1)@0"
+            "Delete(k2)@3"
+            "Put(k2, v2)@1"
+            "Put(k3, v3)@2",
+            PrintContents(&batch));
@@ -163,7 +180,7 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-      "Put(foo, bar)"
-      "PutCF(2, twofoo, bar2)"
-      "PutCF(8, eightfoo, bar8)"
-      "DeleteCF(8, eightfoo)"
-      "MergeCF(3, threethree, 3three)"
-      "Put(foo, bar)"
-      "Merge(omom, nom)",
+            "Put(k1, v1)"
+            "Put(k2, v2)"
+            "Put(k3, v3)"
+            "LogData(blob1)"
+            "Delete(k2)"
+            "LogData(blob2)"
+            "Merge(foo, bar)",
@@ -172 +189 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
+TEST(WriteBatchTest, Continue) {
@@ -174,24 +191,6 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Put(2, Slice("twofoo"), Slice("bar2"));
-  batch.Put(8, Slice("eightfoo"), Slice("bar8"));
-  batch.Delete(8, Slice("eightfoo"));
-  batch.Merge(3, Slice("threethree"), Slice("3three"));
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Merge(Slice("omom"), Slice("nom"));
-  TestHandler handler;
-  batch.Iterate(&handler);
-  ASSERT_EQ(
-      "Put(foo, bar)"
-      "PutCF(2, twofoo, bar2)"
-      "PutCF(8, eightfoo, bar8)"
-      "DeleteCF(8, eightfoo)"
-      "MergeCF(3, threethree, 3three)"
-      "Put(foo, bar)"
-      "Merge(omom, nom)",
-      handler.seen);
-}
-namespace {
-struct TestHandler : public WriteBatch::Handler {
-  std::string seen;
-  virtual void Put(const Slice& key, const Slice& value) {
-    seen += "Put(" + key.ToString() + ", " + value.ToString() + ")";
+  struct Handler : public TestHandler {
+    int num_seen = 0;
+    virtual void PutCF(uint32_t column_family_id, const Slice& key,
+                       const Slice& value) {
+      ++num_seen;
+      TestHandler::PutCF(column_family_id, key, value);
@@ -199,2 +198,4 @@ struct TestHandler : public WriteBatch::Handler {
-  virtual void Merge(const Slice& key, const Slice& value) {
-    seen += "Merge(" + key.ToString() + ", " + value.ToString() + ")";
+    virtual void MergeCF(uint32_t column_family_id, const Slice& key,
+                         const Slice& value) {
+      ++num_seen;
+      TestHandler::MergeCF(column_family_id, key, value);
@@ -203 +204,2 @@ struct TestHandler : public WriteBatch::Handler {
-    seen += "LogData(" + blob.ToString() + ")";
+      ++num_seen;
+      TestHandler::LogData(blob);
@@ -205,2 +207,3 @@ struct TestHandler : public WriteBatch::Handler {
-  virtual void Delete(const Slice& key) {
-    seen += "Delete(" + key.ToString() + ")";
+    virtual void DeleteCF(uint32_t column_family_id, const Slice& key) {
+      ++num_seen;
+      TestHandler::DeleteCF(column_family_id, key);
@@ -208 +211,2 @@ struct TestHandler : public WriteBatch::Handler {
-};
+    virtual bool Continue() override {
+      return num_seen < 3;
@@ -210,10 +214,6 @@ struct TestHandler : public WriteBatch::Handler {
-TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-  WriteBatch batch;
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Put(2, Slice("twofoo"), Slice("bar2"));
-  batch.Put(8, Slice("eightfoo"), Slice("bar8"));
-  batch.Delete(8, Slice("eightfoo"));
-  batch.Merge(3, Slice("threethree"), Slice("3three"));
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Merge(Slice("omom"), Slice("nom"));
-  TestHandler handler;
+  } handler;
+  batch.Put(Slice("k1"), Slice("v1"));
+  batch.PutLogData(Slice("blob1"));
+  batch.Delete(Slice("k1"));
+  batch.PutLogData(Slice("blob2"));
+  batch.Merge(Slice("foo"), Slice("bar"));
@@ -222,7 +222,3 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-      "Put(foo, bar)"
-      "PutCF(2, twofoo, bar2)"
-      "PutCF(8, eightfoo, bar8)"
-      "DeleteCF(8, eightfoo)"
-      "MergeCF(3, threethree, 3three)"
-      "Put(foo, bar)"
-      "Merge(omom, nom)",
+            "Put(k1, v1)"
+            "LogData(blob1)"
+            "Delete(k1)",
@@ -231 +227 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
+TEST(WriteBatchTest, PutGatherSlices) {
@@ -233,18 +229,19 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Put(2, Slice("twofoo"), Slice("bar2"));
-  batch.Put(8, Slice("eightfoo"), Slice("bar8"));
-  batch.Delete(8, Slice("eightfoo"));
-  batch.Merge(3, Slice("threethree"), Slice("3three"));
-  batch.Put(0, Slice("foo"), Slice("bar"));
-  batch.Merge(Slice("omom"), Slice("nom"));
-  TestHandler handler;
-  batch.Iterate(&handler);
-  ASSERT_EQ(
-      "Put(foo, bar)"
-      "PutCF(2, twofoo, bar2)"
-      "PutCF(8, eightfoo, bar8)"
-      "DeleteCF(8, eightfoo)"
-      "MergeCF(3, threethree, 3three)"
-      "Put(foo, bar)"
-      "Merge(omom, nom)",
-      handler.seen);
+  batch.Put(Slice("foo"), Slice("bar"));
+  {
+    Slice key_slice("baz");
+    Slice value_slices[2] = { Slice("header"), Slice("payload") };
+    batch.Put(SliceParts(&key_slice, 1),
+              SliceParts(value_slices, 2));
+  }
+  {
+    Slice key_slices[3] = { Slice("key"), Slice("part2"), Slice("part3") };
+    Slice value_slice("value");
+    batch.Put(SliceParts(key_slices, 3),
+              SliceParts(&value_slice, 1));
+  }
+  WriteBatchInternal::SetSequence(&batch, 100);
+  ASSERT_EQ("Put(baz, headerpayload)@101"
+            "Put(foo, bar)@100"
+            "Put(keypart2part3, value)@102",
+            PrintContents(&batch));
+  ASSERT_EQ(3, batch.Count());
@@ -274 +271,3 @@ TEST(WriteBatchTest, ColumnFamiliesBatchTest) {
-int main(int argc, char** argv) { return rocksdb::test::RunAllTests(); }
+int main(int argc, char** argv) {
+  return rocksdb::test::RunAllTests();
+}
