diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_merged.h
index 08e5336..7a6258c 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_merged.h
@@ -44,0 +45,5 @@ struct CompressionOptions {
+enum UpdateStatus {
+  UPDATE_FAILED = 0,
+  UPDATED_INPLACE = 1,
+  UPDATED = 2,
+};
@@ -92,2 +97,3 @@ struct ColumnFamilyOptions {
-  std::vector<std::shared_ptr<TablePropertiesCollector>>
-      table_properties_collectors;
+  typedef std::vector<std::shared_ptr<TablePropertiesCollector>>
+      TablePropertiesCollectors;
+  TablePropertiesCollectors table_properties_collectors;
@@ -95,0 +102,6 @@ struct ColumnFamilyOptions {
+  UpdateStatus (*inplace_callback)(char* existing_value,
+                                   uint32_t* existing_value_size,
+                                   Slice delta_value,
+                                   std::string* merged_value);
+  uint32_t memtable_prefix_bloom_bits;
+  uint32_t memtable_prefix_bloom_probes;
@@ -132 +144,6 @@ struct DBOptions {
-  enum { NONE, NORMAL, SEQUENTIAL, WILLNEED } access_hint_on_compaction_start;
+  enum {
+    NONE,
+    NORMAL,
+    SEQUENTIAL,
+    WILLNEED
+  } access_hint_on_compaction_start;
@@ -138,5 +154,0 @@ struct DBOptions {
-enum UpdateStatus {
-  UPDATE_FAILED = 0,
-  UPDATED_INPLACE = 1,
-  UPDATED = 2,
-};
@@ -144,11 +156,3 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
-  void Dump(Logger* log) const;
-  Options* PrepareForBulkLoad();
-  typedef std::vector<std::shared_ptr<TablePropertiesCollector>>
-      TablePropertiesCollectors;
-  UpdateStatus (*inplace_callback)(char* existing_value,
-                                   uint32_t* existing_value_size,
-                                   Slice delta_value,
-                                   std::string* merged_value);
-  uint32_t memtable_prefix_bloom_bits;
-  uint32_t memtable_prefix_bloom_probes;
-  Options() : DBOptions(), ColumnFamilyOptions() {}
+  Options() :
+    DBOptions(),
+    ColumnFamilyOptions() {}
@@ -157,0 +162,2 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
+  void Dump(Logger* log) const;
+  Options* PrepareForBulkLoad();
