diff --git a/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@db_impl.no_comments_mergegen.h b/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@db_impl.no_comments_merged.h
index 686ddaa..bf62c67 100644
--- a/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@db_impl.no_comments_mergegen.h
+++ b/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@db_impl.no_comments_merged.h
@@ -200 +200,2 @@ class DBImpl : public DB {
-Status MakeRoomForWrite(ColumnFamilyData* cfd, bool force );
+  Status MakeRoomForWrite(ColumnFamilyData* cfd,
+                          bool force );
