diff --git a/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@memtable_list.no_comments_mergegen.cc b/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@memtable_list.no_comments_merged.cc
index c4c7314..caeff8b 100644
--- a/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@memtable_list.no_comments_mergegen.cc
+++ b/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@memtable_list.no_comments_merged.cc
@@ -109 +109,4 @@ Status MemTableList::InstallMemtableFlushResults(
-uint64_t file_number, std::set<uint64_t>& pending_outputs, autovector<MemTable*>* to_delete, Directory* db_directory, LogBuffer* log_buffer) {
+    ColumnFamilyData* cfd, const autovector<MemTable*>& mems, VersionSet* vset,
+    port::Mutex* mu, Logger* info_log, uint64_t file_number,
+    std::set<uint64_t>& pending_outputs, autovector<MemTable*>* to_delete,
+    Directory* db_directory, LogBuffer* log_buffer) {
