--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@version_set.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@version_set.no_comments_merged.cc
@@ -119 +120,6 @@ class Version::LevelFileNumIterator : public Iterator {
-      : icmp_(icmp), flist_(flist), index_(flist->size()) {
+      : icmp_(icmp),
+        flist_(flist),
+        index_(flist->size()) {
+  }
+  virtual bool Valid() const {
+    return index_ < flist_->size();
@@ -121 +126,0 @@ class Version::LevelFileNumIterator : public Iterator {
-  virtual bool Valid() const { return index_ < flist_->size(); }
@@ -196,12 +200,0 @@ bool Version::PrefixMayMatch(const ReadOptions& options,
-<<<<<<< HEAD
-    const FileMetaData* meta_file =
-        reinterpret_cast<const FileMetaData*>(level_iter->value().data());
-    may_match = cfd_->table_cache()->PrefixMayMatch(
-        options, cfd_->internal_comparator(), *meta_file, internal_prefix,
-        nullptr);
-||||||| 8c4a3bfa5
-    const FileMetaData* meta_file =
-        reinterpret_cast<const FileMetaData*>(level_iter->value().data());
-    may_match = vset_->table_cache_->PrefixMayMatch(
-        options, vset_->icmp_, *meta_file, internal_prefix, nullptr);
-=======
@@ -213,3 +206,2 @@ bool Version::PrefixMayMatch(const ReadOptions& options,
-    may_match = vset_->table_cache_->PrefixMayMatch(options, vset_->icmp_, meta,
-                                                    internal_prefix, nullptr);
->>>>>>> 7efdd9ef4db90728b85085e05aa367a6ea727b0f
+    may_match = cfd_->table_cache()->PrefixMayMatch(
+        options, cfd_->internal_comparator(), meta, internal_prefix, nullptr);
@@ -1159,7 +1171,0 @@ VersionSet::~VersionSet() {
-{
-  column_family_set_.reset();
-  for (auto file : obsolete_files_) {
-    delete file;
-  }
-  obsolete_files_.clear();
-}
