diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/cmd-pipe-pane.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/cmd-pipe-pane.no_comments_merged.c
index 8bf018f..50df360 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/cmd-pipe-pane.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/cmd-pipe-pane.no_comments_merged.c
@@ -5 +4,0 @@
-#include <paths.h>
@@ -22,3 +21,5 @@ const struct cmd_entry cmd_pipe_pane_entry = {
-    .exec = cmd_pipe_pane_exec};
-static enum cmd_retval cmd_pipe_pane_exec(struct cmd *self,
-                                          struct cmdq_item *item) {
+ .exec = cmd_pipe_pane_exec
+};
+static enum cmd_retval
+cmd_pipe_pane_exec(struct cmd *self, struct cmdq_item *item)
+{
@@ -44,2 +45,4 @@ static enum cmd_retval cmd_pipe_pane_exec(struct cmd *self,
-  if (args->argc == 0 || *args->argv[0] == '\0') return (CMD_RETURN_NORMAL);
-  if (args_has(self->args, 'o') && old_fd != -1) return (CMD_RETURN_NORMAL);
+ if (args->argc == 0 || *args->argv[0] == '\0')
+  return (CMD_RETURN_NORMAL);
+ if (args_has(self->args, 'o') && old_fd != -1)
+  return (CMD_RETURN_NORMAL);
@@ -66,2 +69,4 @@ static enum cmd_retval cmd_pipe_pane_exec(struct cmd *self,
-      if (dup2(pipe_fd[1], STDIN_FILENO) == -1) _exit(1);
-      if (pipe_fd[1] != STDIN_FILENO) close(pipe_fd[1]);
+  if (dup2(pipe_fd[1], STDIN_FILENO) == -1)
+   _exit(1);
+  if (pipe_fd[1] != STDIN_FILENO)
+   close(pipe_fd[1]);
@@ -69,3 +74,6 @@ static enum cmd_retval cmd_pipe_pane_exec(struct cmd *self,
-      if (dup2(null_fd, STDOUT_FILENO) == -1) _exit(1);
-      if (dup2(null_fd, STDERR_FILENO) == -1) _exit(1);
-      if (null_fd != STDOUT_FILENO && null_fd != STDERR_FILENO) close(null_fd);
+  if (dup2(null_fd, STDOUT_FILENO) == -1)
+   _exit(1);
+  if (dup2(null_fd, STDERR_FILENO) == -1)
+   _exit(1);
+  if (null_fd != STDOUT_FILENO && null_fd != STDERR_FILENO)
+   close(null_fd);
@@ -80,3 +88,3 @@ static enum cmd_retval cmd_pipe_pane_exec(struct cmd *self,
-      wp->pipe_event =
-          bufferevent_new(wp->pipe_fd, NULL, cmd_pipe_pane_write_callback,
-                          cmd_pipe_pane_error_callback, wp);
+  wp->pipe_event = bufferevent_new(wp->pipe_fd, NULL,
+      cmd_pipe_pane_write_callback, cmd_pipe_pane_error_callback,
+      wp);
@@ -89,2 +97,3 @@ static enum cmd_retval cmd_pipe_pane_exec(struct cmd *self,
-static void cmd_pipe_pane_write_callback(__unused struct bufferevent *bufev,
-                                         void *data) {
+static void
+cmd_pipe_pane_write_callback(__unused struct bufferevent *bufev, void *data)
+{
@@ -93 +102,2 @@ static void cmd_pipe_pane_write_callback(__unused struct bufferevent *bufev,
-  if (window_pane_destroy_ready(wp)) server_destroy_pane(wp, 1);
+ if (window_pane_destroy_ready(wp))
+  server_destroy_pane(wp, 1);
@@ -95,2 +105,4 @@ static void cmd_pipe_pane_write_callback(__unused struct bufferevent *bufev,
-static void cmd_pipe_pane_error_callback(__unused struct bufferevent *bufev,
-                                         __unused short what, void *data) {
+static void
+cmd_pipe_pane_error_callback(__unused struct bufferevent *bufev,
+    __unused short what, void *data)
+{
@@ -102 +114,2 @@ static void cmd_pipe_pane_error_callback(__unused struct bufferevent *bufev,
-  if (window_pane_destroy_ready(wp)) server_destroy_pane(wp, 1);
+ if (window_pane_destroy_ready(wp))
+  server_destroy_pane(wp, 1);
