diff --git a/./output/tmux/fbbf5a10-51112221-58b79660/window.no_comments_mergegen.c b/./output/tmux/fbbf5a10-51112221-58b79660/window.no_comments_merged.c
index 8d7cddb..2d3fe93 100644
--- a/./output/tmux/fbbf5a10-51112221-58b79660/window.no_comments_mergegen.c
+++ b/./output/tmux/fbbf5a10-51112221-58b79660/window.no_comments_merged.c
@@ -766 +766,6 @@ window_pane_spawn(struct window_pane *wp, int argc, char **argv,
-#ifdef HAVE_UTEMPTER xsnprintf(s, sizeof s, "tmux(%lu).%%%u", (long) getpid(), wp->id); utempter_add_record(wp->fd, s); kill(getpid(), SIGCHLD); #endif
+#ifdef HAVE_UTEMPTER
+ xsnprintf(s, sizeof s, "tmux(%lu).%%%u", (long) getpid(), wp->id);
+ utempter_add_record(wp->fd, s);
+ kill(getpid(), SIGCHLD);
+#endif
+ sigprocmask(SIG_SETMASK, &oldset, NULL);
