--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/window.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/window.no_comments_merged.c
@@ -13,2 +12,0 @@
-#include <util.h>
-#include <vis.h>
@@ -30 +28,6 @@ static struct window_pane *window_pane_choose_best(struct window_pane **,
-int window_cmp(struct window *w1, struct window *w2) {
+RB_GENERATE(windows, window, entry, window_cmp);
+RB_GENERATE(winlinks, winlink, entry, winlink_cmp);
+RB_GENERATE(window_pane_tree, window_pane, tree_entry, window_pane_cmp);
+int
+window_cmp(struct window *w1, struct window *w2)
+{
@@ -622 +765,0 @@ int window_pane_spawn(struct window_pane *wp, int argc, char **argv,
-<<<<<<< HEAD
@@ -628,2 +770,0 @@ int window_pane_spawn(struct window_pane *wp, int argc, char **argv,
-  |||||||
-=======
@@ -631 +771,0 @@ int window_pane_spawn(struct window_pane *wp, int argc, char **argv,
->>>>>>> 51112221eeb31ced907f0dfcf077582996c20c07
