diff --git a/./output/redis/c33ef454-3f6893a4-f30b18f4/src@t_string.no_comments_conflict.c b/./output/redis/c33ef454-3f6893a4-f30b18f4/src@t_string.no_comments_merged.c
index 807e215..3b791e2 100644
--- a/./output/redis/c33ef454-3f6893a4-f30b18f4/src@t_string.no_comments_conflict.c
+++ b/./output/redis/c33ef454-3f6893a4-f30b18f4/src@t_string.no_comments_merged.c
@@ -217 +216,0 @@ void msetGenericCommand(client *c, int nx) {
-<<<<<<< HEAD
@@ -220,6 +218,0 @@ void msetGenericCommand(client *c, int nx) {
-||||||| f30b18f4d
-                busykeys++;
-=======
-                busykeys++;
-                break;
->>>>>>> 3f6893a4
