--- a/./output/redis/c33ef454-3f6893a4-f30b18f4/src@t_string.no_comments_mergegen.c
+++ b/./output/redis/c33ef454-3f6893a4-f30b18f4/src@t_string.no_comments_merged.c
@@ -217 +217,2 @@ void msetGenericCommand(client *c, int nx) {
-for (j = 1; j < c->argc; j += 2) { if (lookupKeyWrite(c->db,c->argv[j]) != NULL) { addReply(c, shared.czero); return; break;
+                addReply(c, shared.czero);
+                return;
