--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_merged.cc
@@ -5,0 +6 @@
+#include <grpc/support/port_platform.h>
@@ -516 +517,5 @@ grpc_chttp2_stream::grpc_chttp2_stream(grpc_chttp2_transport* t,
-    : t(t), refcount(refcount), reffer(this), metadata_buffer {
+    : t(t),
+      refcount(refcount),
+      reffer(this),
+      metadata_buffer{grpc_chttp2_incoming_metadata_buffer(arena),
+                      grpc_chttp2_incoming_metadata_buffer(arena)} {
@@ -1888,3 +1892,0 @@ void grpc_chttp2_mark_stream_closed(grpc_chttp2_transport* t,
-<<<<<<< HEAD
-|||||||
-=======
@@ -1895 +1896,0 @@ void grpc_chttp2_mark_stream_closed(grpc_chttp2_transport* t,
->>>>>>> 172f61a3a257257ce84394a87708b0d11aa7001b
