--- a/./output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_conflict.cc
+++ b/./output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_merged.cc
@@ -1893,3 +1892,0 @@ void grpc_chttp2_mark_stream_closed(grpc_chttp2_transport* t,
-<<<<<<< HEAD
-||||||| 18695e30ca
-=======
@@ -1900 +1896,0 @@ void grpc_chttp2_mark_stream_closed(grpc_chttp2_transport* t,
->>>>>>> 172f61a3
