diff --git a/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_conflict.cpp b/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_merged.cpp
index f366215..f3ec3aa 100644
--- a/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_conflict.cpp
+++ b/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -186 +185,0 @@ static std::string InitLlndkLibrariesVendor() {
-<<<<<<< HEAD
@@ -195,14 +193,0 @@ static std::string InitLlndkLibrariesVendor() {
-||||||| c968f45f41
-  std::string config_file = kLlndkLibrariesFile;
-  InsertVndkVersionStr(&config_file, false);
-  auto sonames = ReadConfig(config_file, always_true);
-=======
-  std::string config_file;
-  if (IsVendorVndkEnabled()) {
-    config_file = kLlndkLibrariesFile;
-    InsertVndkVersionStr(&config_file, false);
-  } else {
-    config_file = kLlndkLibrariesNoVndkFile;
-  }
-  auto sonames = ReadConfig(config_file, always_true);
->>>>>>> f701823d
@@ -222 +206,0 @@ static std::string InitLlndkLibrariesProduct() {
-<<<<<<< HEAD
@@ -231,14 +214,0 @@ static std::string InitLlndkLibrariesProduct() {
-||||||| c968f45f41
-  std::string config_file = kLlndkLibrariesFile;
-  InsertVndkVersionStr(&config_file, true);
-  auto sonames = ReadConfig(config_file, always_true);
-=======
-  std::string config_file;
-  if (IsProductVndkEnabled()) {
-    config_file = kLlndkLibrariesFile;
-    InsertVndkVersionStr(&config_file, true);
-  } else {
-    config_file = kLlndkLibrariesNoVndkFile;
-  }
-  auto sonames = ReadConfig(config_file, always_true);
->>>>>>> f701823d
