--- a/./output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_verity.no_comments_mergegen.cpp
+++ b/./output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_verity.no_comments_merged.cpp
@@ -723 +723,0 @@ int fs_mgr_setup_verity(struct fstab_rec *fstab)
-    bool verified_at_boot = false;
@@ -726,0 +727 @@ int fs_mgr_setup_verity(struct fstab_rec *fstab)
+    bool verified_at_boot = false;
