--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_fstab.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_fstab.no_comments_merged.c
@@ -56 +55,0 @@ static struct flag_list fs_mgr_flags[] = {
-<<<<<<< HEAD
@@ -58,2 +56,0 @@ static struct flag_list fs_mgr_flags[] = {
-|||||||
-=======
@@ -61 +57,0 @@ static struct flag_list fs_mgr_flags[] = {
->>>>>>> 34543c03e6319d102bc0b1e3ca54808ff398e676
