diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_fstab.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_fstab.no_comments_merged.c
index 69eb621..5c693f5 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_fstab.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_fstab.no_comments_merged.c
@@ -56 +55,0 @@ static struct flag_list fs_mgr_flags[] = {
-<<<<<<< HEAD
@@ -58,2 +56,0 @@ static struct flag_list fs_mgr_flags[] = {
-|||||||
-=======
@@ -61 +57,0 @@ static struct flag_list fs_mgr_flags[] = {
->>>>>>> 34543c03e6319d102bc0b1e3ca54808ff398e676
@@ -76,2 +72,6 @@ static struct flag_list encryption_modes[] = {
-    {"software", EM_SOFTWARE}, {"ice", EM_ICE}, {0, 0}};
-static uint64_t calculate_zram_size(unsigned int percentage) {
+    {"software", EM_SOFTWARE},
+    {"ice", EM_ICE},
+    {0, 0}
+};
+static uint64_t calculate_zram_size(unsigned int percentage)
+{
@@ -85 +85,2 @@ static uint64_t calculate_zram_size(unsigned int percentage) {
-static uint64_t parse_size(const char *arg) {
+static uint64_t parse_size(const char *arg)
+{
@@ -97,2 +98,3 @@ static int parse_flags(char *flags, struct flag_list *fl,
-                       struct fs_mgr_flag_values *flag_vals, char *fs_options,
-                       int fs_options_len) {
+                       struct fs_mgr_flag_values *flag_vals,
+                       char *fs_options, int fs_options_len)
+{
@@ -148,2 +150,2 @@ static int parse_flags(char *flags, struct flag_list *fl,
-            flag_vals->label =
-                strndup(label_start, (int)(label_end - label_start));
+                        flag_vals->label = strndup(label_start,
+                                                   (int) (label_end - label_start));
@@ -191 +193,2 @@ static int parse_flags(char *flags, struct flag_list *fl,
-struct fstab *fs_mgr_read_fstab_file(FILE *fstab_file) {
+struct fstab *fs_mgr_read_fstab_file(FILE *fstab_file)
+{
@@ -211 +214,2 @@ struct fstab *fs_mgr_read_fstab_file(FILE *fstab_file) {
-    if (*p == '#' || *p == '\0') continue;
+        if (*p == '#' || *p == '\0')
+            continue;
@@ -231 +235,2 @@ struct fstab *fs_mgr_read_fstab_file(FILE *fstab_file) {
-    if (*p == '#' || *p == '\0') continue;
+        if (*p == '#' || *p == '\0')
+            continue;
@@ -256,2 +261,2 @@ struct fstab *fs_mgr_read_fstab_file(FILE *fstab_file) {
-    fstab->recs[cnt].flags =
-        parse_flags(p, mount_flags, NULL, tmp_fs_options, FS_OPTIONS_LEN);
+        fstab->recs[cnt].flags = parse_flags(p, mount_flags, NULL,
+                                       tmp_fs_options, FS_OPTIONS_LEN);
@@ -267,2 +272,2 @@ struct fstab *fs_mgr_read_fstab_file(FILE *fstab_file) {
-    fstab->recs[cnt].fs_mgr_flags =
-        parse_flags(p, fs_mgr_flags, &flag_vals, NULL, 0);
+        fstab->recs[cnt].fs_mgr_flags = parse_flags(p, fs_mgr_flags,
+                                                    &flag_vals, NULL, 0);
@@ -289 +294,2 @@ err:
-  if (fstab) fs_mgr_free_fstab(fstab);
+    if (fstab)
+        fs_mgr_free_fstab(fstab);
@@ -292 +298,2 @@ err:
-struct fstab *fs_mgr_read_fstab(const char *fstab_path) {
+struct fstab *fs_mgr_read_fstab(const char *fstab_path)
+{
@@ -307 +314,2 @@ struct fstab *fs_mgr_read_fstab(const char *fstab_path) {
-void fs_mgr_free_fstab(struct fstab *fstab) {
+void fs_mgr_free_fstab(struct fstab *fstab)
+{
@@ -324,2 +332,4 @@ void fs_mgr_free_fstab(struct fstab *fstab) {
-int fs_mgr_add_entry(struct fstab *fstab, const char *mount_point,
-                     const char *fs_type, const char *blk_device) {
+int fs_mgr_add_entry(struct fstab *fstab,
+                     const char *mount_point, const char *fs_type,
+                     const char *blk_device)
+{
@@ -328,2 +338,2 @@ int fs_mgr_add_entry(struct fstab *fstab, const char *mount_point,
-  new_fstab_recs = (struct fstab_rec *)realloc(
-      fstab->recs, sizeof(struct fstab_rec) * (n + 1));
+    new_fstab_recs = (struct fstab_rec *)
+                     realloc(fstab->recs, sizeof(struct fstab_rec) * (n + 1));
@@ -342,2 +352,2 @@ int fs_mgr_add_entry(struct fstab *fstab, const char *mount_point,
-struct fstab_rec *fs_mgr_get_entry_for_mount_point_after(
-    struct fstab_rec *start_rec, struct fstab *fstab, const char *path) {
+struct fstab_rec *fs_mgr_get_entry_for_mount_point_after(struct fstab_rec *start_rec, struct fstab *fstab, const char *path)
+{
@@ -367,2 +377,2 @@ struct fstab_rec *fs_mgr_get_entry_for_mount_point_after(
-struct fstab_rec *fs_mgr_get_entry_for_mount_point(struct fstab *fstab,
-                                                   const char *path) {
+struct fstab_rec *fs_mgr_get_entry_for_mount_point(struct fstab *fstab, const char *path)
+{
@@ -371 +381,2 @@ struct fstab_rec *fs_mgr_get_entry_for_mount_point(struct fstab *fstab,
-int fs_mgr_is_voldmanaged(const struct fstab_rec *fstab) {
+int fs_mgr_is_voldmanaged(const struct fstab_rec *fstab)
+{
@@ -374 +385,2 @@ int fs_mgr_is_voldmanaged(const struct fstab_rec *fstab) {
-int fs_mgr_is_nonremovable(const struct fstab_rec *fstab) {
+int fs_mgr_is_nonremovable(const struct fstab_rec *fstab)
+{
@@ -377 +389,2 @@ int fs_mgr_is_nonremovable(const struct fstab_rec *fstab) {
-int fs_mgr_is_verified(const struct fstab_rec *fstab) {
+int fs_mgr_is_verified(const struct fstab_rec *fstab)
+{
@@ -380 +393,2 @@ int fs_mgr_is_verified(const struct fstab_rec *fstab) {
-int fs_mgr_is_encryptable(const struct fstab_rec *fstab) {
+int fs_mgr_is_encryptable(const struct fstab_rec *fstab)
+{
@@ -383 +397,2 @@ int fs_mgr_is_encryptable(const struct fstab_rec *fstab) {
-int fs_mgr_is_file_encrypted(const struct fstab_rec *fstab) {
+int fs_mgr_is_file_encrypted(const struct fstab_rec *fstab)
+{
@@ -386 +401,2 @@ int fs_mgr_is_file_encrypted(const struct fstab_rec *fstab) {
-const char *fs_mgr_get_file_encryption_mode(const struct fstab_rec *fstab) {
+const char* fs_mgr_get_file_encryption_mode(const struct fstab_rec *fstab)
+{
@@ -395 +411,2 @@ const char *fs_mgr_get_file_encryption_mode(const struct fstab_rec *fstab) {
-int fs_mgr_is_convertible_to_fbe(const struct fstab_rec *fstab) {
+int fs_mgr_is_convertible_to_fbe(const struct fstab_rec *fstab)
+{
@@ -398 +415,2 @@ int fs_mgr_is_convertible_to_fbe(const struct fstab_rec *fstab) {
-int fs_mgr_is_noemulatedsd(const struct fstab_rec *fstab) {
+int fs_mgr_is_noemulatedsd(const struct fstab_rec *fstab)
+{
@@ -401 +419,2 @@ int fs_mgr_is_noemulatedsd(const struct fstab_rec *fstab) {
-int fs_mgr_is_notrim(struct fstab_rec *fstab) {
+int fs_mgr_is_notrim(struct fstab_rec *fstab)
+{
@@ -404 +423,2 @@ int fs_mgr_is_notrim(struct fstab_rec *fstab) {
-int fs_mgr_is_formattable(struct fstab_rec *fstab) {
+int fs_mgr_is_formattable(struct fstab_rec *fstab)
+{
@@ -407 +427,2 @@ int fs_mgr_is_formattable(struct fstab_rec *fstab) {
-int fs_mgr_is_slotselect(struct fstab_rec *fstab) {
+int fs_mgr_is_slotselect(struct fstab_rec *fstab)
+{
@@ -410 +431,2 @@ int fs_mgr_is_slotselect(struct fstab_rec *fstab) {
-int fs_mgr_is_nofail(struct fstab_rec *fstab) {
+int fs_mgr_is_nofail(struct fstab_rec *fstab)
+{
@@ -413 +435,2 @@ int fs_mgr_is_nofail(struct fstab_rec *fstab) {
-int fs_mgr_is_latemount(struct fstab_rec *fstab) {
+int fs_mgr_is_latemount(struct fstab_rec *fstab)
+{
