diff --git a/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_mergegen.cpp b/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_merged.cpp
index 0cb4b02..f3ec3aa 100644
--- a/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_mergegen.cpp
+++ b/./output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -186 +186,8 @@ static std::string InitLlndkLibrariesVendor() {
-std::string InitLlndkLibrariesVendor() { std::string config_file; if (IsVendorVndkEnabled()) { config_file = kLlndkLibrariesFile; InsertVndkVersionStr(&config_file, false); } else { config_file = kLlndkLibrariesNoVndkFile; } Result<std::vector<std::string>> sonames = ReadConfig(config_file, always_true);
+  std::string config_file;
+  if (IsVendorVndkEnabled()) {
+    config_file = kLlndkLibrariesFile;
+    InsertVndkVersionStr(&config_file, false);
+  } else {
+    config_file = kLlndkLibrariesNoVndkFile;
+  }
+  Result<std::vector<std::string>> sonames = ReadConfig(config_file, always_true);
@@ -200 +207,8 @@ static std::string InitLlndkLibrariesProduct() {
-std::string config_file; if (IsProductVndkEnabled()) { config_file = kLlndkLibrariesFile; InsertVndkVersionStr(&config_file, true); } else { config_file = kLlndkLibrariesNoVndkFile; } Result<std::vector<std::string>> sonames = ReadConfig(config_file, always_true);
+  std::string config_file;
+  if (IsProductVndkEnabled()) {
+    config_file = kLlndkLibrariesFile;
+    InsertVndkVersionStr(&config_file, true);
+  } else {
+    config_file = kLlndkLibrariesNoVndkFile;
+  }
+  Result<std::vector<std::string>> sonames = ReadConfig(config_file, always_true);
