--- a/home/whalien/codebase/python/mergebot-eva/output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/art/04a110da-f701823d-c968f45f/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -23 +22,0 @@ using internal::ConfigEntry;
-using internal::ParseApexLibrariesConfig;
@@ -24,0 +24 @@ using internal::ParseConfig;
+using internal::ParseApexLibrariesConfig;
@@ -190 +189,0 @@ static std::string InitLlndkLibrariesVendor() {
-<<<<<<< HEAD
@@ -195,8 +193,0 @@ static std::string InitLlndkLibrariesVendor() {
-|||||||
-    auto sonames = ReadConfig(config_file, always_true);
-=======
-  } else {
-    config_file = kLlndkLibrariesNoVndkFile;
-  }
-  auto sonames = ReadConfig(config_file, always_true);
->>>>>>> f701823d2bc4911fc5eaa2d94799118751e8c637
@@ -220 +210,0 @@ static std::string InitLlndkLibrariesProduct() {
-<<<<<<< HEAD
@@ -225,8 +214,0 @@ static std::string InitLlndkLibrariesProduct() {
-|||||||
-      auto sonames = ReadConfig(config_file, always_true);
-=======
-  } else {
-    config_file = kLlndkLibrariesNoVndkFile;
-  }
-  auto sonames = ReadConfig(config_file, always_true);
->>>>>>> f701823d2bc4911fc5eaa2d94799118751e8c637
