diff --git a/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_mergegen.cpp b/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_merged.cpp
index f2a0456..42dcb72 100644
--- a/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_merged.cpp
@@ -130,2 +130,8 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-for (size_t i = 0; i < static_cast<size_t>(predictedR.size()) && predictionTime <= futureTime; ++i) { if (predictedR[i] < mModel->config().distanceNoiseFloor) {
-if confidence is < some threshold. const TfLiteMotionPredictorSample::Point predictedPoint = convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]); ALOGD_IF(isDebug(), "prediction %zu: %f, %f", i, predictedPoint.x, predictedPoint.y);
+    for (size_t i = 0; i < static_cast<size_t>(predictedR.size()) && predictionTime <= futureTime;
+         ++i) {
+        if (predictedR[i] < mModel->config().distanceNoiseFloor) {
+            break;
+        }
+        const TfLiteMotionPredictorSample::Point predictedPoint =
+                convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]);
+        ALOGD_IF(isDebug(), "prediction %zu: %f, %f", i, predictedPoint.x, predictedPoint.y);
