diff --git a/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_conflict.cpp b/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_merged.cpp
index 184482c..42dcb72 100644
--- a/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_conflict.cpp
+++ b/./output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_merged.cpp
@@ -130 +129,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-<<<<<<< HEAD
@@ -136,11 +134,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-||||||| 3881e39ce0
-    for (int i = 0; i < predictedR.size() && predictionTime <= futureTime; ++i) {
-        const TfLiteMotionPredictorSample::Point point =
-                convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]);
-=======
-    for (size_t i = 0; i < static_cast<size_t>(predictedR.size()) && predictionTime <= futureTime;
-         ++i) {
-        const TfLiteMotionPredictorSample::Point point =
-                convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]);
->>>>>>> 0b3dbcef
-<<<<<<< HEAD
@@ -150,5 +137,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-||||||| 3881e39ce0
-        ALOGD_IF(isDebug(), "prediction %d: %f, %f", i, point.x, point.y);
-=======
-        ALOGD_IF(isDebug(), "prediction %zu: %f, %f", i, point.x, point.y);
->>>>>>> 0b3dbcef
