--- a/./output/rocksdb/fb2346fc-45ad75db-d5de22dc/tools@db_stress.no_comments_mergegen.cc
+++ b/./output/rocksdb/fb2346fc-45ad75db-d5de22dc/tools@db_stress.no_comments_merged.cc
@@ -257,0 +258,3 @@ static bool ValidatePrefixSize(const char* flagname, int32_t value) {
+DEFINE_int32(prefix_size, 7, "Control the prefix size for HashSkipListRep");
+static const bool FLAGS_prefix_size_dummy =
+  google::RegisterFlagValidator(&FLAGS_prefix_size, &ValidatePrefixSize);
@@ -752,0 +756,3 @@ class StressTest {
+  Status MultiPrefixScan(ThreadState* thread, const ReadOptions& readoptions,
+                         ColumnFamilyHandle* column_family,
+                         const Slice& key) {
@@ -813,0 +820 @@ class StressTest {
+    unique_ptr<Iterator> iter(db_->NewIterator(readoptionscopy, column_family));
@@ -913 +920,2 @@ class StressTest {
-key_opts.prefix = &prefix; Iterator* iter = db_->NewIterator(read_opts, column_family); int64_t count = 0; for (iter->SeekToFirst();
+          Iterator* iter = db_->NewIterator(read_opts, column_family);
+          int64_t count = 0;
@@ -927 +935 @@ key_opts.prefix = &prefix; Iterator* iter = db_->NewIterator(read_opts, column_f
-<<<<<<< HEAD MultiPrefixScan(thread, read_opts, column_family, key); ======= MultiPrefixScan(thread, read_opts, key); >>>>>>> arrayapi
+          MultiPrefixScan(thread, read_opts, column_family, key);
