--- a/./output/libnativehelper/e597b7e0-ae7a6134-c3803b81/tests_mts@jni@libnativehelper_test.no_comments_conflict.cpp
+++ b/./output/libnativehelper/e597b7e0-ae7a6134-c3803b81/tests_mts@jni@libnativehelper_test.no_comments_merged.cpp
@@ -69 +68,0 @@ TEST_F(LibnativehelperTest, CreateUtfOrReturn) {
-<<<<<<< HEAD
@@ -99,2 +97,0 @@ TEST_F(LibnativehelperTest, CreateUtfOrReturnExprEvaluatesToValueBackedByTempora
-=======
->>>>>>> ae7a6134
