diff --git a/./output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_mergegen.h b/./output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_merged.h
index 07d48ba..02c26e5 100644
--- a/./output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_mergegen.h
+++ b/./output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_merged.h
@@ -81 +81,4 @@ class InternalStats {
-uint64_t GetBackgroundErrorCount() const { return bg_error_count_; } uint64_t BumpAndGetBackgroundErrorCount() { return ++bg_error_count_; } bool GetProperty(DBPropertyType property_type, const Slice& property, std::string* value, ColumnFamilyData* cfd);
+  uint64_t GetBackgroundErrorCount() const { return bg_error_count_; }
+  uint64_t BumpAndGetBackgroundErrorCount() { return ++bg_error_count_; }
+  bool GetProperty(DBPropertyType property_type, const Slice& property,
+                   std::string* value, ColumnFamilyData* cfd);
