diff --git a/./output/frameworks_av/e67b58ee-72e1cd48-6ad524ce/services@audioflinger@Effects.no_comments_conflict.h b/./output/frameworks_av/e67b58ee-72e1cd48-6ad524ce/services@audioflinger@Effects.no_comments_merged.h
index 24e4e86..f052dd8 100644
--- a/./output/frameworks_av/e67b58ee-72e1cd48-6ad524ce/services@audioflinger@Effects.no_comments_conflict.h
+++ b/./output/frameworks_av/e67b58ee-72e1cd48-6ad524ce/services@audioflinger@Effects.no_comments_merged.h
@@ -207 +206,0 @@ private:
-<<<<<<< HEAD
@@ -211,5 +209,0 @@ private:
-||||||| 6ad524cef7
-=======
-    std::optional<std::vector<uint32_t>> mVolume;
-    std::optional<std::vector<uint32_t>> mReturnedVolume;
->>>>>>> 72e1cd48
