diff --git a/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_mergegen.cc b/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_merged.cc
index 0476f5e..b2a630f 100644
--- a/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_mergegen.cc
+++ b/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_merged.cc
@@ -877 +877,6 @@ WARN_UNUSED bool OnDeviceRefresh::CheckBuildUserfaultFdGc() const {
-ART_FORMAT( "Userfaultfd GC check failed (build_enable_uffd_gc: {}, is_at_least_u: {}, " "kernel_supports_uffd: {}).", build_enable_uffd_gc, is_at_least_u, kernel_supports_uffd);
+    LOG(WARNING) << ART_FORMAT(
+        "Userfaultfd GC check failed (build_enable_uffd_gc: {}, is_at_least_u: {}, "
+        "kernel_supports_uffd: {}).",
+        build_enable_uffd_gc,
+        is_at_least_u,
+        kernel_supports_uffd);
@@ -1347 +1352,2 @@ WARN_UNUSED CompilationResult OnDeviceRefresh::RunDex2oat(
-std::string cmd_line = Join(args.Get(), ' '); LOG(INFO) << ART_FORMAT("{}: {} [timeout {}s]", debug_message, cmd_line, timeout);
+  std::string cmd_line = Join(args.Get(), ' ');
+  LOG(INFO) << ART_FORMAT("{}: {} [timeout {}s]", debug_message, cmd_line, timeout);
