diff --git a/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_conflict.cc b/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_merged.cc
index cd87ec4..b2a630f 100644
--- a/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_conflict.cc
+++ b/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_merged.cc
@@ -86 +85,0 @@ using ::android::modules::sdklevel::IsAtLeastU;
-<<<<<<< HEAD
@@ -88,4 +86,0 @@ using ::art::tools::CmdlineBuilder;
-||||||| 64a6fe7549
-using ::fmt::literals::operator""_format;
-=======
->>>>>>> 489049d1
@@ -882 +876,0 @@ WARN_UNUSED bool OnDeviceRefresh::CheckBuildUserfaultFdGc() const {
-<<<<<<< HEAD
@@ -889,8 +882,0 @@ WARN_UNUSED bool OnDeviceRefresh::CheckBuildUserfaultFdGc() const {
-||||||| 64a6fe7549
-    LOG(WARNING) << "Userfaultfd GC check failed (build-time: {}, runtime: {})."_format(
-        build_enable_uffd_gc, kernel_supports_uffd);
-=======
-    LOG(WARNING) << ART_FORMAT("Userfaultfd GC check failed (build-time: {}, runtime: {}).",
-                               build_enable_uffd_gc,
-                               kernel_supports_uffd);
->>>>>>> 489049d1
@@ -1366 +1351,0 @@ WARN_UNUSED CompilationResult OnDeviceRefresh::RunDex2oat(
-<<<<<<< HEAD
@@ -1369,7 +1353,0 @@ WARN_UNUSED CompilationResult OnDeviceRefresh::RunDex2oat(
-||||||| 64a6fe7549
-  std::string cmd_line = Join(args, ' ');
-  LOG(INFO) << "{}: {} [timeout {}s]"_format(debug_message, cmd_line, timeout);
-=======
-  std::string cmd_line = Join(args, ' ');
-  LOG(INFO) << ART_FORMAT("{}: {} [timeout {}s]", debug_message, cmd_line, timeout);
->>>>>>> 489049d1
