diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_merged.cc
index 90f6949..22a3f41 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_merged.cc
@@ -172,7 +171,0 @@ void HttpConnectHandshaker::DoHandshake(grpc_tcp_server_acceptor* ,
-<<<<<<< HEAD
-      gpr_mu_unlock(&mu_);
-      ExecCtx::Run(DEBUG_LOCATION, on_handshake_done, GRPC_ERROR_NONE);
-|||||||
-      gpr_mu_unlock(&mu_);
-      GRPC_CLOSURE_SCHED(on_handshake_done, GRPC_ERROR_NONE);
-=======
@@ -180,2 +173 @@ void HttpConnectHandshaker::DoHandshake(grpc_tcp_server_acceptor* ,
-    GRPC_CLOSURE_SCHED(on_handshake_done, GRPC_ERROR_NONE);
->>>>>>> a0f09cb540097369a495470da4c10fe62a6e9bd5
+    ExecCtx::Run(DEBUG_LOCATION, on_handshake_done, GRPC_ERROR_NONE);
@@ -240,6 +232,4 @@ void HttpConnectHandshaker::DoHandshake(grpc_tcp_server_acceptor* ,
-    GRPC_CLOSURE_INIT(&request_done_closure_,
-                      &HttpConnectHandshaker::OnWriteDone, this,
-                      grpc_schedule_on_exec_ctx);
-    GRPC_CLOSURE_INIT(&response_read_closure_,
-                      &HttpConnectHandshaker::OnReadDone, this,
-                      grpc_schedule_on_exec_ctx);
+  GRPC_CLOSURE_INIT(&request_done_closure_, &HttpConnectHandshaker::OnWriteDone,
+                    this, grpc_schedule_on_exec_ctx);
+  GRPC_CLOSURE_INIT(&response_read_closure_, &HttpConnectHandshaker::OnReadDone,
+                    this, grpc_schedule_on_exec_ctx);
@@ -255,2 +245 @@ void HttpConnectHandshaker::DoHandshake(grpc_tcp_server_acceptor* ,
-    () = delete;
-    () = delete;
+  ~HttpConnectHandshakerFactory() override = default;
