diff --git a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_conflict.cc b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_merged.cc
index 9638b17..01a2aad 100644
--- a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_conflict.cc
+++ b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_merged.cc
@@ -993,8 +992,0 @@ static void post_batch_completion(batch_control* bctl) {
-<<<<<<< HEAD
-    grpc_core::ExecCtx::Run(DEBUG_LOCATION,
-                            (grpc_closure*)bctl->completion_data.notify_tag.tag,
-                            error);
-||||||| ce842ecc31
-    GRPC_CLOSURE_SCHED((grpc_closure*)bctl->completion_data.notify_tag.tag,
-                       error);
-=======
@@ -1003 +994,0 @@ static void post_batch_completion(batch_control* bctl) {
->>>>>>> a0f09cb5
@@ -1301,6 +1291,0 @@ static grpc_call_error call_start_batch(grpc_call* call, const grpc_op* ops,
-<<<<<<< HEAD
-      grpc_core::ExecCtx::Run(DEBUG_LOCATION, (grpc_closure*)notify_tag,
-                              GRPC_ERROR_NONE);
-||||||| ce842ecc31
-      GRPC_CLOSURE_SCHED((grpc_closure*)notify_tag, GRPC_ERROR_NONE);
-=======
@@ -1308 +1292,0 @@ static grpc_call_error call_start_batch(grpc_call* call, const grpc_op* ops,
->>>>>>> a0f09cb5
