diff --git a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_mergegen.cc b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_merged.cc
index 5097181..01a2aad 100644
--- a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_mergegen.cc
+++ b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@surface@call.no_comments_merged.cc
@@ -993 +993,2 @@ static void post_batch_completion(batch_control* bctl) {
-grpc_core::ExecCtx::Run(DEBUG_LOCATION, (grpc_closure*)bctl->completion_data.notify_tag.tag, error);
+    GRPC_CLOSURE_RUN((grpc_closure*)bctl->completion_data.notify_tag.tag,
+                     error);
@@ -1291 +1292 @@ static grpc_call_error call_start_batch(grpc_call* call, const grpc_op* ops,
-grpc_core::ExecCtx::Run(DEBUG_LOCATION, (grpc_closure*)notify_tag, GRPC_ERROR_NONE);
+      GRPC_CLOSURE_RUN((grpc_closure*)notify_tag, GRPC_ERROR_NONE);
