--- a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_conflict.cc
+++ b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_merged.cc
@@ -169,11 +168,0 @@ void HttpConnectHandshaker::DoHandshake(grpc_tcp_server_acceptor* ,
-<<<<<<< HEAD
-    gpr_mu_lock(&mu_);
-    is_shutdown_ = true;
-    gpr_mu_unlock(&mu_);
-    ExecCtx::Run(DEBUG_LOCATION, on_handshake_done, GRPC_ERROR_NONE);
-||||||| ce842ecc31
-    gpr_mu_lock(&mu_);
-    is_shutdown_ = true;
-    gpr_mu_unlock(&mu_);
-    GRPC_CLOSURE_SCHED(on_handshake_done, GRPC_ERROR_NONE);
-=======
@@ -184,2 +173 @@ void HttpConnectHandshaker::DoHandshake(grpc_tcp_server_acceptor* ,
-    GRPC_CLOSURE_SCHED(on_handshake_done, GRPC_ERROR_NONE);
->>>>>>> a0f09cb5
+    ExecCtx::Run(DEBUG_LOCATION, on_handshake_done, GRPC_ERROR_NONE);
