--- a/./output/duckdb/71385710-31a2467e-07b75e3c/src@function@function_list.no_comments_conflict.cpp
+++ b/./output/duckdb/71385710-31a2467e-07b75e3c/src@function@function_list.no_comments_merged.cpp
@@ -34 +33,0 @@ static const StaticFunctionDefinition function[] = {
-<<<<<<< HEAD
@@ -60,21 +58,0 @@ static const StaticFunctionDefinition function[] = {
-||||||| 07b75e3c3a
-=======
- DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUbigintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUintegerFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUsmallintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUtinyintFun),
- DUCKDB_SCALAR_FUNCTION(InternalCompressStringHugeintFun),
- DUCKDB_SCALAR_FUNCTION(InternalCompressStringUbigintFun),
- DUCKDB_SCALAR_FUNCTION(InternalCompressStringUintegerFun),
- DUCKDB_SCALAR_FUNCTION(InternalCompressStringUsmallintFun),
- DUCKDB_SCALAR_FUNCTION(InternalCompressStringUtinyintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralBigintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralHugeintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralIntegerFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralSmallintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUbigintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUhugeintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUintegerFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUsmallintFun),
- DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressStringFun),
->>>>>>> 31a2467e
