diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@cpp@common@version_cc.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@cpp@common@version_cc.no_comments_merged.cc
index 745f23b..997a75b 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-grpc::string Version() < < < < < < < HEAD { return "1.30.0-dev"; }
-|||||||
-{ return "1.29.0-dev"; }
-=======
-{ return "1.29.1"; }
->>>>>>> 7d89dbb311f049b43bda7bbf6f7d7bf1b4c24419
+grpc::string Version() { return "1.30.0-dev"; }
