--- a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_conflict.cpp
+++ b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_merged.cpp
@@ -110 +109,0 @@ bool mediametrics_selfRecord(mediametrics_handle_t handle) {
-<<<<<<< HEAD
@@ -116,8 +114,0 @@ mediametrics_handle_t mediametrics_dup(mediametrics_handle_t handle) {
-||||||| faafe2989a
-=======
-mediametrics_handle_t mediametrics_dup(mediametrics_handle_t handle) {
-    android::MediaAnalyticsItem *item = (android::MediaAnalyticsItem *) handle;
-    if (item == NULL) return android::MediaAnalyticsItem::convert(item);
-    return android::MediaAnalyticsItem::convert(item->dup());
-}
->>>>>>> e23cbd26
