--- a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_mergegen.h
+++ b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_merged.h
@@ -227 +227,6 @@ private:
-mediametrics_handle_t item); void updateLowLatency(const sp<AMessage> &msg);
+    mediametrics_handle_t mMetricsHandle;
+    void initMediametrics();
+    void updateMediametrics();
+    void flushMediametrics();
+    void updateEphemeralMediametrics(mediametrics_handle_t item);
+    void updateLowLatency(const sp<AMessage> &msg);
