diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_merged.cpp
index 4212ea5..a279b2c 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_merged.cpp
@@ -6 +5,0 @@
-#include <media/MediaAnalyticsItem.h>
@@ -103 +102,3 @@ bool mediametrics_getCString(mediametrics_handle_t handle, attr_t attr,
-void mediametrics_freeCString(char *value) { free(value); }
+void mediametrics_freeCString(char *value) {
+    free(value);
+}
@@ -110,3 +111,3 @@ mediametrics_handle_t mediametrics_dup(mediametrics_handle_t handle) {
-  android::MediaAnalyticsItem *item = (android::MediaAnalyticsItem *)handle;
-  if (item == NULL) return android::MediaAnalyticsItem::convert(item);
-  return android::MediaAnalyticsItem::convert(item->dup());
+    Item *item = (Item *) handle;
+    if (item == NULL) return Item::convert(item);
+    return Item::convert(item->dup());
@@ -127,2 +128 @@ bool mediametrics_isEnabled() {
-bool mediametrics_getAttributes(mediametrics_handle_t handle, char **buffer,
-                                size_t *length) {
+bool mediametrics_getAttributes(mediametrics_handle_t handle, char **buffer, size_t *length) {
