--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@impl@codegen@client_context_impl.no_comments_conflict.h
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@impl@codegen@client_context_impl.no_comments_merged.h
@@ -248 +247,0 @@ class ClientContext {
-<<<<<<< HEAD
@@ -251,7 +249,0 @@ class ClientContext {
-||||||| b4031fa43d
-  grpc::string authority_;
-  std::shared_ptr<grpc_impl::CallCredentials> creds_;
-=======
-  std::string authority_;
-  std::shared_ptr<grpc_impl::CallCredentials> creds_;
->>>>>>> c6bd67c2
