--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@security@credentials.no_comments_mergegen.h
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@security@credentials.no_comments_merged.h
@@ -43,2 +43,60 @@ class ChannelCredentials : private grpc::GrpcLibraryCodegen {
-friend std::shared_ptr<grpc::Channel> CreateCustomChannel( const grpc::string& target, const std::shared_ptr<grpc::ChannelCredentials>& creds, const grpc::ChannelArguments& args); friend std::shared_ptr<grpc::Channel> grpc::experimental::CreateCustomChannelWithInterceptors( const grpc::string& target, const std::shared_ptr<grpc::ChannelCredentials>& creds, const grpc::ChannelArguments& args, std::vector<std::unique_ptr< grpc::experimental::ClientInterceptorFactoryInterface>> interceptor_creators); virtual std::shared_ptr<Channel> CreateChannelImpl( const std::string& target, const ChannelArguments& , std::vector<std::unique_ptr< grpc::experimental::ClientInterceptorFactoryInterface>> ) { return nullptr; } };
-std::shared_ptr<CallCredentials> GoogleIAMCredentials( const std::string& authorization_token, const std::string& authority_selector);
+ private:
+  friend std::shared_ptr<grpc::Channel> CreateCustomChannel(
+      const grpc::string& target,
+      const std::shared_ptr<grpc::ChannelCredentials>& creds,
+      const grpc::ChannelArguments& args);
+  friend std::shared_ptr<grpc::Channel>
+  grpc::experimental::CreateCustomChannelWithInterceptors(
+      const grpc::string& target,
+      const std::shared_ptr<grpc::ChannelCredentials>& creds,
+      const grpc::ChannelArguments& args,
+      std::vector<std::unique_ptr<
+          grpc::experimental::ClientInterceptorFactoryInterface>>
+          interceptor_creators);
+  virtual std::shared_ptr<Channel> CreateChannelImpl(
+      const grpc::string& target, const ChannelArguments& args) = 0;
+  virtual std::shared_ptr<Channel> CreateChannelWithInterceptors(
+      const grpc::string& , const ChannelArguments& ,
+      std::vector<std::unique_ptr<
+          grpc::experimental::ClientInterceptorFactoryInterface>>
+                              ) {
+    return nullptr;
+  }
+};
+class CallCredentials : private grpc::GrpcLibraryCodegen {
+ public:
+  CallCredentials();
+  ~CallCredentials();
+  virtual bool ApplyToCall(grpc_call* call) = 0;
+  virtual grpc::string DebugString() {
+    return "CallCredentials did not provide a debug string";
+  }
+ protected:
+  friend std::shared_ptr<ChannelCredentials> CompositeChannelCredentials(
+      const std::shared_ptr<ChannelCredentials>& channel_creds,
+      const std::shared_ptr<CallCredentials>& call_creds);
+  friend std::shared_ptr<CallCredentials> CompositeCallCredentials(
+      const std::shared_ptr<CallCredentials>& creds1,
+      const std::shared_ptr<CallCredentials>& creds2);
+  virtual SecureCallCredentials* AsSecureCredentials() = 0;
+};
+struct SslCredentialsOptions {
+  grpc::string pem_root_certs;
+  grpc::string pem_private_key;
+  grpc::string pem_cert_chain;
+};
+std::shared_ptr<ChannelCredentials> GoogleDefaultCredentials();
+std::shared_ptr<ChannelCredentials> SslCredentials(
+    const SslCredentialsOptions& options);
+std::shared_ptr<CallCredentials> GoogleComputeEngineCredentials();
+constexpr long kMaxAuthTokenLifetimeSecs = 3600;
+std::shared_ptr<CallCredentials> ServiceAccountJWTAccessCredentials(
+    const grpc::string& json_key,
+    long token_lifetime_seconds = kMaxAuthTokenLifetimeSecs);
+std::shared_ptr<CallCredentials> GoogleRefreshTokenCredentials(
+    const grpc::string& json_refresh_token);
+std::shared_ptr<CallCredentials> AccessTokenCredentials(
+    const grpc::string& access_token);
+std::shared_ptr<CallCredentials> GoogleIAMCredentials(
+    const grpc::string& authorization_token,
+    const grpc::string& authority_selector);
@@ -78,0 +137,5 @@ struct StsCredentialsOptions {
+static inline grpc::Status StsCredentialsOptionsFromJson(
+    const std::string& json_string, StsCredentialsOptions* options) {
+  return ::grpc::experimental::StsCredentialsOptionsFromJson(json_string,
+                                                                  options);
+}
