diff --git a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_mergegen.h b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_merged.h
index a830514..e2bb41c 100644
--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_mergegen.h
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_merged.h
@@ -10 +10,7 @@ namespace grpc {
-namespace grpc {
+std::shared_ptr<Channel> CreateChannel(
+    const grpc::string& target,
+    const std::shared_ptr<ChannelCredentials>& creds);
+std::shared_ptr<Channel> CreateCustomChannel(
+    const grpc::string& target,
+    const std::shared_ptr<ChannelCredentials>& creds,
+    const ChannelArguments& args);
@@ -11,0 +18,3 @@ namespace experimental {
+std::shared_ptr<Channel> CreateCustomChannelWithInterceptors(
+    const grpc::string& target,
+    const std::shared_ptr<ChannelCredentials>& creds,
