--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@security@credentials.no_comments_conflict.h
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@security@credentials.no_comments_merged.h
@@ -43 +42,0 @@ class ChannelCredentials : private grpc::GrpcLibraryCodegen {
-<<<<<<< HEAD
@@ -97,18 +95,0 @@ std::shared_ptr<CallCredentials> ServiceAccountJWTAccessCredentials(
-||||||| b4031fa43d
-static inline std::shared_ptr<grpc_impl::CallCredentials>
-ServiceAccountJWTAccessCredentials(
-    const grpc::string& json_key,
-    long token_lifetime_seconds = grpc::kMaxAuthTokenLifetimeSecs) {
-  return ::grpc_impl::ServiceAccountJWTAccessCredentials(
-      json_key, token_lifetime_seconds);
-}
-=======
-static inline std::shared_ptr<grpc_impl::CallCredentials>
-ServiceAccountJWTAccessCredentials(
-    const std::string& json_key,
-    long token_lifetime_seconds = grpc::kMaxAuthTokenLifetimeSecs) {
-  return ::grpc_impl::ServiceAccountJWTAccessCredentials(
-      json_key, token_lifetime_seconds);
-}
->>>>>>> c6bd67c2
-<<<<<<< HEAD
@@ -117,12 +97,0 @@ std::shared_ptr<CallCredentials> GoogleRefreshTokenCredentials(
-||||||| b4031fa43d
-static inline std::shared_ptr<grpc_impl::CallCredentials>
-GoogleRefreshTokenCredentials(const grpc::string& json_refresh_token) {
-  return ::grpc_impl::GoogleRefreshTokenCredentials(json_refresh_token);
-}
-=======
-static inline std::shared_ptr<grpc_impl::CallCredentials>
-GoogleRefreshTokenCredentials(const std::string& json_refresh_token) {
-  return ::grpc_impl::GoogleRefreshTokenCredentials(json_refresh_token);
-}
->>>>>>> c6bd67c2
-<<<<<<< HEAD
@@ -131,12 +99,0 @@ std::shared_ptr<CallCredentials> AccessTokenCredentials(
-||||||| b4031fa43d
-static inline std::shared_ptr<grpc_impl::CallCredentials>
-AccessTokenCredentials(const grpc::string& access_token) {
-  return ::grpc_impl::AccessTokenCredentials(access_token);
-}
-=======
-static inline std::shared_ptr<grpc_impl::CallCredentials>
-AccessTokenCredentials(const std::string& access_token) {
-  return ::grpc_impl::AccessTokenCredentials(access_token);
-}
->>>>>>> c6bd67c2
-<<<<<<< HEAD
@@ -146,15 +102,0 @@ std::shared_ptr<CallCredentials> GoogleIAMCredentials(
-||||||| b4031fa43d
-static inline std::shared_ptr<grpc_impl::CallCredentials> GoogleIAMCredentials(
-    const grpc::string& authorization_token,
-    const grpc::string& authority_selector) {
-  return ::grpc_impl::GoogleIAMCredentials(authorization_token,
-                                           authority_selector);
-}
-=======
-static inline std::shared_ptr<grpc_impl::CallCredentials> GoogleIAMCredentials(
-    const std::string& authorization_token,
-    const std::string& authority_selector) {
-  return ::grpc_impl::GoogleIAMCredentials(authorization_token,
-                                           authority_selector);
-}
->>>>>>> c6bd67c2
@@ -195,10 +136,0 @@ struct StsCredentialsOptions {
-<<<<<<< HEAD
-grpc::Status StsCredentialsOptionsFromJson(const grpc::string& json_string,
-                                           StsCredentialsOptions* options);
-||||||| b4031fa43d
-static inline grpc::Status StsCredentialsOptionsFromJson(
-    const grpc::string& json_string, StsCredentialsOptions* options) {
-  return ::grpc_impl::experimental::StsCredentialsOptionsFromJson(json_string,
-                                                                  options);
-}
-=======
@@ -207 +139 @@ static inline grpc::Status StsCredentialsOptionsFromJson(
-  return ::grpc_impl::experimental::StsCredentialsOptionsFromJson(json_string,
+  return ::grpc::experimental::StsCredentialsOptionsFromJson(json_string,
@@ -210 +141,0 @@ static inline grpc::Status StsCredentialsOptionsFromJson(
->>>>>>> c6bd67c2
