diff --git a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_conflict.h b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_merged.h
index 470b62d..e2bb41c 100644
--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_conflict.h
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_merged.h
@@ -10 +9,0 @@ namespace grpc {
-<<<<<<< HEAD
@@ -14,14 +12,0 @@ std::shared_ptr<Channel> CreateChannel(
-||||||| b4031fa43d
-static inline std::shared_ptr<::grpc::Channel> CreateChannel(
-    const grpc::string& target,
-    const std::shared_ptr<ChannelCredentials>& creds) {
-  return ::grpc_impl::CreateChannelImpl(target, creds);
-}
-=======
-static inline std::shared_ptr<::grpc::Channel> CreateChannel(
-    const std::string& target,
-    const std::shared_ptr<ChannelCredentials>& creds) {
-  return ::grpc_impl::CreateChannelImpl(target, creds);
-}
->>>>>>> c6bd67c2
-<<<<<<< HEAD
@@ -32,14 +16,0 @@ std::shared_ptr<Channel> CreateCustomChannel(
-||||||| b4031fa43d
-static inline std::shared_ptr<::grpc::Channel> CreateCustomChannel(
-    const grpc::string& target,
-    const std::shared_ptr<ChannelCredentials>& creds,
-    const ChannelArguments& args) {
-  return ::grpc_impl::CreateCustomChannelImpl(target, creds, args);
-}
-=======
-static inline std::shared_ptr<::grpc::Channel> CreateCustomChannel(
-    const std::string& target, const std::shared_ptr<ChannelCredentials>& creds,
-    const ChannelArguments& args) {
-  return ::grpc_impl::CreateCustomChannelImpl(target, creds, args);
-}
->>>>>>> c6bd67c2
@@ -47 +17,0 @@ namespace experimental {
-<<<<<<< HEAD
@@ -51,10 +20,0 @@ std::shared_ptr<Channel> CreateCustomChannelWithInterceptors(
-||||||| b4031fa43d
-static inline std::shared_ptr<::grpc::Channel>
-CreateCustomChannelWithInterceptors(
-    const grpc::string& target,
-    const std::shared_ptr<ChannelCredentials>& creds,
-=======
-static inline std::shared_ptr<::grpc::Channel>
-CreateCustomChannelWithInterceptors(
-    const std::string& target, const std::shared_ptr<ChannelCredentials>& creds,
->>>>>>> c6bd67c2
