diff --git a/./output/duckdb/32158ec8-39f4ff41-c0e7b60d/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_conflict.cpp b/./output/duckdb/32158ec8-39f4ff41-c0e7b60d/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_merged.cpp
index 3e0b9d7..b57da8d 100644
--- a/./output/duckdb/32158ec8-39f4ff41-c0e7b60d/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_conflict.cpp
+++ b/./output/duckdb/32158ec8-39f4ff41-c0e7b60d/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_merged.cpp
@@ -1329 +1328,0 @@ bool StringValueScanner::IsRowValid(CSVIterator &current_iterator) const {
-<<<<<<< HEAD
@@ -1331,6 +1330 @@ bool StringValueScanner::IsRowValid(CSVIterator &current_iterator) const {
- return (tuples.number_of_rows == 1 || tuples.first_line_is_comment) && tuples.borked_rows.empty();
-||||||| c0e7b60d7b
- return tuples.number_of_rows == 1 && tuples.borked_rows.empty();
-=======
- return tuples.number_of_rows == 1 && tuples.borked_rows.empty() && !tuples.current_errors.HasError();
->>>>>>> 39f4ff41
+ return (tuples.number_of_rows == 1 || tuples.first_line_is_comment) && tuples.borked_rows.empty() && !tuples.current_errors.HasError();;
@@ -1391,2 +1385 @@ void StringValueScanner::SetStart() {
-<<<<<<< HEAD
- if (!best_row.is_valid && state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0') {
+ if (!best_row.is_valid && state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0'&& state_machine->dialect_options.state_machine_options.quote.GetValue() != '\0') {
@@ -1401,10 +1393,0 @@ void StringValueScanner::SetStart() {
-||||||| c0e7b60d7b
- if (!any_valid_row && potential_start > next_new_line &&
-     state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0') {
-  TryRow(CSVState::ESCAPE, potential_start, largest_end_pos, any_valid_row);
-=======
- if (!any_valid_row && potential_start > next_new_line &&
-     state_machine->dialect_options.state_machine_options.quote.GetValue() != '\0' &&
-     state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0') {
-  TryRow(CSVState::ESCAPE, potential_start, largest_end_pos, any_valid_row);
->>>>>>> 39f4ff41
