--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/32158ec8-39f4ff41-c0e7b60d/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/32158ec8-39f4ff41-c0e7b60d/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_merged.cpp
@@ -763,0 +764,24 @@ bool StringValueResult::EmptyLine(StringValueResult &result, const idx_t buffer_
+ if (result.states.IsCarriageReturn() &&
+     result.state_machine.dialect_options.state_machine_options.new_line == NewLineIdentifier::CARRY_ON) {
+  result.last_position.buffer_pos++;
+ }
+ if (result.number_of_columns == 1) {
+  for (idx_t i = 0; i < result.null_str_count; i++) {
+   if (result.null_str_size[i] == 0) {
+    bool empty = false;
+    if (!result.state_machine.options.force_not_null.empty()) {
+     empty = result.state_machine.options.force_not_null[0];
+    }
+    if (empty) {
+     static_cast<string_t *>(result.vector_ptr[0])[result.number_of_rows] = string_t();
+    } else {
+     result.validity_mask[0]->SetInvalid(result.number_of_rows);
+    }
+    result.number_of_rows++;
+   }
+  }
+  if (result.number_of_rows >= result.result_size) {
+   return true;
+  }
+ }
+ return false;
@@ -1306 +1330 @@ bool StringValueScanner::IsRowValid(CSVIterator &current_iterator) const {
- return (tuples.number_of_rows == 1 || tuples.first_line_is_comment) && tuples.borked_rows.empty();
+ return (tuples.number_of_rows == 1 || tuples.first_line_is_comment) && tuples.borked_rows.empty() && !tuples.current_errors.HasError();;
@@ -1361,2 +1385 @@ void StringValueScanner::SetStart() {
-<<<<<<< HEAD
- if (!best_row.is_valid && state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0') {
+ if (!best_row.is_valid && state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0'&& state_machine->dialect_options.state_machine_options.quote.GetValue() != '\0') {
@@ -1369,10 +1391,0 @@ void StringValueScanner::SetStart() {
-|||||||
- if (!any_valid_row && potential_start > next_new_line &&
-     state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0') {
-  TryRow(CSVState::ESCAPE, potential_start, largest_end_pos, any_valid_row);
-=======
- if (!any_valid_row && potential_start > next_new_line &&
-     state_machine->dialect_options.state_machine_options.quote.GetValue() != '\0' &&
-     state_machine->dialect_options.state_machine_options.escape.GetValue() != '\0') {
-  TryRow(CSVState::ESCAPE, potential_start, largest_end_pos, any_valid_row);
->>>>>>> 39f4ff4108954c59650cbdc9538d52da47e5c28b
@@ -1468,0 +1482 @@ ValidatorLine StringValueScanner::GetValidationLine() {
+}
