diff --git a/./output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_mergegen.h b/./output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_merged.h
index 0579c18..e11affa 100644
--- a/./output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_mergegen.h
+++ b/./output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_merged.h
@@ -10 +9,0 @@
-#include <android/permission/PermissionChecker.h>
@@ -778,2 +776,0 @@ private:
-    std::unique_ptr<permission::PermissionChecker> mPermissionChecker =
-            std::make_unique<permission::PermissionChecker>();
