--- a/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@gc@collector@mark_compact.no_comments_mergegen.h
+++ b/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@gc@collector@mark_compact.no_comments_merged.h
@@ -48,2 +47,0 @@ class MarkCompact final : public GarbageCollector {
-  bool IsUsingSigbusFeature() const { return use_uffd_sigbus_; }
-  bool SigbusHandler(siginfo_t* info) REQUIRES(!lock_) NO_THREAD_SAFETY_ANALYSIS;
@@ -296 +294,6 @@ class MarkCompact final : public GarbageCollector {
-void ZeropageIoctl(void* addr, bool tolerate_eexist, bool tolerate_enoent); void CopyIoctl(void* dst, void* buffer);
+  void ZeropageIoctl(void* addr, bool tolerate_eexist, bool tolerate_enoent);
+  void CopyIoctl(void* dst, void* buffer);
+  void MapUpdatedLinearAllocPage(uint8_t* page,
+                                 uint8_t* shadow_page,
+                                 Atomic<PageState>& state,
+                                 bool page_touched);
@@ -298,0 +302 @@ void ZeropageIoctl(void* addr, bool tolerate_eexist, bool tolerate_enoent); void
+  Mutex lock_;
