diff --git a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_conflict.hpp b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_merged.hpp
index fcdffca..1b28665 100644
--- a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_conflict.hpp
+++ b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_merged.hpp
@@ -23,7 +23 @@ struct RowOperations {
-<<<<<<< HEAD
- static void DestroyStates(TupleDataLayout &layout, Vector &addresses, idx_t count);
-||||||| 5c0a1ed76e
- static void DestroyStates(RowLayout &layout, Vector &addresses, idx_t count);
-=======
- static void DestroyStates(RowOperationsState &state, RowLayout &layout, Vector &addresses, idx_t count);
->>>>>>> 30d444e4
+ static void DestroyStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, idx_t count);
@@ -34,6 +28 @@ struct RowOperations {
-<<<<<<< HEAD
- static void CombineStates(TupleDataLayout &layout, Vector &sources, Vector &targets, idx_t count);
-||||||| 5c0a1ed76e
- static void CombineStates(RowLayout &layout, Vector &sources, Vector &targets, idx_t count);
-=======
- static void CombineStates(RowOperationsState &state, RowLayout &layout, Vector &sources, Vector &targets,
+ static void CombineStates(RowOperationsState &state, TupleDataLayout &layout, Vector &sources, Vector &targets,
@@ -41,7 +30 @@ struct RowOperations {
->>>>>>> 30d444e4
-<<<<<<< HEAD
- static void FinalizeStates(TupleDataLayout &layout, Vector &addresses, DataChunk &result, idx_t aggr_idx);
-||||||| 5c0a1ed76e
- static void FinalizeStates(RowLayout &layout, Vector &addresses, DataChunk &result, idx_t aggr_idx);
-=======
- static void FinalizeStates(RowOperationsState &state, RowLayout &layout, Vector &addresses, DataChunk &result,
+ static void FinalizeStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, DataChunk &result,
@@ -49 +31,0 @@ struct RowOperations {
->>>>>>> 30d444e4
