diff --git a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@symbols.no_comments_mergegen.cpp b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@symbols.no_comments_merged.cpp
index 94f2438..562f166 100644
--- a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@symbols.no_comments_mergegen.cpp
+++ b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@symbols.no_comments_merged.cpp
@@ -36 +36,3 @@
-#include "duckdb/common/types/column/column_data_collection.hpp" #include "duckdb/common/types/column/column_data_allocator.hpp"
+#include "duckdb/common/types/column/column_data_collection.hpp"
+#include "duckdb/common/types/column/column_data_allocator.hpp"
+#include "duckdb/common/vector.hpp"
