diff --git a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@row_operations@row_aggregate.no_comments_conflict.cpp b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@row_operations@row_aggregate.no_comments_merged.cpp
index 5e5ee34..26670ef 100644
--- a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@row_operations@row_aggregate.no_comments_conflict.cpp
+++ b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@row_operations@row_aggregate.no_comments_merged.cpp
@@ -24,7 +24 @@ void RowOperations::InitializeStates(TupleDataLayout &layout, Vector &addresses,
-<<<<<<< HEAD
-void RowOperations::DestroyStates(TupleDataLayout &layout, Vector &addresses, idx_t count) {
-||||||| 5c0a1ed76e
-void RowOperations::DestroyStates(RowLayout &layout, Vector &addresses, idx_t count) {
-=======
-void RowOperations::DestroyStates(RowOperationsState &state, RowLayout &layout, Vector &addresses, idx_t count) {
->>>>>>> 30d444e4
+void RowOperations::DestroyStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, idx_t count) {
@@ -59,6 +53 @@ void RowOperations::UpdateFilteredStates(RowOperationsState &state, AggregateFil
-<<<<<<< HEAD
-void RowOperations::CombineStates(TupleDataLayout &layout, Vector &sources, Vector &targets, idx_t count) {
-||||||| 5c0a1ed76e
-void RowOperations::CombineStates(RowLayout &layout, Vector &sources, Vector &targets, idx_t count) {
-=======
-void RowOperations::CombineStates(RowOperationsState &state, RowLayout &layout, Vector &sources, Vector &targets,
+void RowOperations::CombineStates(RowOperationsState &state, TupleDataLayout &layout, Vector &sources, Vector &targets,
@@ -66 +54,0 @@ void RowOperations::CombineStates(RowOperationsState &state, RowLayout &layout,
->>>>>>> 30d444e4
@@ -80,8 +68,2 @@ void RowOperations::CombineStates(RowOperationsState &state, RowLayout &layout,
-<<<<<<< HEAD
-void RowOperations::FinalizeStates(TupleDataLayout &layout, Vector &addresses, DataChunk &result, idx_t aggr_idx) {
-||||||| 5c0a1ed76e
-void RowOperations::FinalizeStates(RowLayout &layout, Vector &addresses, DataChunk &result, idx_t aggr_idx) {
-=======
-void RowOperations::FinalizeStates(RowOperationsState &state, RowLayout &layout, Vector &addresses, DataChunk &result,
-                                   idx_t aggr_idx) {
->>>>>>> 30d444e4
+void RowOperations::FinalizeStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses,
+                                   DataChunk &result, idx_t aggr_idx) {
