diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_merged.h
index cbf2ced..e2bb41c 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@create_channel.no_comments_merged.h
@@ -3 +2,0 @@
-#include <grpcpp/create_channel_impl.h>
@@ -10,0 +10,7 @@ namespace grpc {
+std::shared_ptr<Channel> CreateChannel(
+    const grpc::string& target,
+    const std::shared_ptr<ChannelCredentials>& creds);
+std::shared_ptr<Channel> CreateCustomChannel(
+    const grpc::string& target,
+    const std::shared_ptr<ChannelCredentials>& creds,
+    const ChannelArguments& args);
@@ -20,7 +25,0 @@ std::shared_ptr<Channel> CreateCustomChannelWithInterceptors(
-std::shared_ptr<Channel> CreateChannel(
-    const grpc::string& target,
-    const std::shared_ptr<ChannelCredentials>& creds);
-std::shared_ptr<Channel> CreateCustomChannel(
-    const grpc::string& target,
-    const std::shared_ptr<ChannelCredentials>& creds,
-    const ChannelArguments& args);
