--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@impl@codegen@client_context_impl.no_comments_mergegen.h
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/include@grpcpp@impl@codegen@client_context_impl.no_comments_merged.h
@@ -248 +248,2 @@ class ClientContext {
-std::string authority_; std::shared_ptr<grpc::CallCredentials> creds_; mutable std::shared_ptr<const grpc::AuthContext> auth_context_;
+  grpc::string authority_;
+  std::shared_ptr<grpc::CallCredentials> creds_;
