diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_merged.cc
index ef122ff..1a3a273 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_merged.cc
@@ -32 +32 @@ std::shared_ptr<grpc::Channel> CreateCustomChannelWithInterceptors(
-    const grpc::string& target,
+    const std::string& target,
