--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_mergegen.cc
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_merged.cc
@@ -8 +8,3 @@
-namespace grpc { std::shared_ptr<grpc::Channel> CreateChannel( const std::string& target, const std::shared_ptr<grpc::ChannelCredentials>& creds) {
+namespace grpc {
+std::shared_ptr<grpc::Channel> CreateChannel(
+    const grpc::string& target,
@@ -12,0 +15 @@ std::shared_ptr<grpc::Channel> CreateCustomChannel(
+    const grpc::string& target,
