--- a/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_conflict.cc
+++ b/./output/grpc/fa0c7314-c6bd67c2-b4031fa4/src@cpp@client@create_channel.no_comments_merged.cc
@@ -8 +7,0 @@
-<<<<<<< HEAD
@@ -12,9 +10,0 @@ std::shared_ptr<grpc::Channel> CreateChannel(
-||||||| b4031fa43d
-namespace grpc_impl {
-std::shared_ptr<grpc::Channel> CreateChannelImpl(
-    const grpc::string& target,
-=======
-namespace grpc_impl {
-std::shared_ptr<grpc::Channel> CreateChannelImpl(
-    const std::string& target,
->>>>>>> c6bd67c2
@@ -24 +13,0 @@ std::shared_ptr<grpc::Channel> CreateChannelImpl(
-<<<<<<< HEAD
@@ -27,7 +15,0 @@ std::shared_ptr<grpc::Channel> CreateCustomChannel(
-||||||| b4031fa43d
-std::shared_ptr<grpc::Channel> CreateCustomChannelImpl(
-    const grpc::string& target,
-=======
-std::shared_ptr<grpc::Channel> CreateCustomChannelImpl(
-    const std::string& target,
->>>>>>> c6bd67c2
