diff --git a/./output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_mergegen.h b/./output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_merged.h
index 4083e9f..305613e 100644
--- a/./output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_mergegen.h
+++ b/./output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_merged.h
@@ -17,5 +16,0 @@ namespace aidl::android::os {
-#if defined(__ANDROID_VENDOR__)
-#define AT_LEAST_V_OR_202404 constexpr(__ANDROID_VENDOR_API__ >= 202404)
-#else
-#define AT_LEAST_V_OR_202404 (__builtin_available(android __ANDROID_API_FUTURE__, *))
-#endif
