--- a/./output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_conflict.h
+++ b/./output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_merged.h
@@ -17,5 +16,0 @@ namespace aidl::android::os {
-#if defined(__ANDROID_VENDOR__)
-#define AT_LEAST_V_OR_202404 constexpr(__ANDROID_VENDOR_API__ >= 202404)
-#else
-#define AT_LEAST_V_OR_202404 (__builtin_available(android __ANDROID_API_FUTURE__, *))
-#endif
@@ -25 +19,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -27,5 +20,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -38 +26,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -40,5 +27,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -49 +31,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -51,5 +32,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -63 +39,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -65,5 +40,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -79 +49,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -81,5 +50,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -93 +57,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -95,5 +58,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-            if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-            if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -107 +65,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -109,5 +66,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -137 +89,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -139,5 +90,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        } else if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        } else if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -153 +99,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -155,5 +100,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -166 +106,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -168,5 +107,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -179 +113,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -181,5 +114,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -190 +118,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -192,5 +119,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -201 +123,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -203,5 +124,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -212 +128,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -214,5 +129,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -223 +133,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -225,5 +134,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -234 +138,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -236,5 +139,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -255 +153,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -257,5 +154,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -269 +161,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -271,5 +162,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -283 +169,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -285,5 +170,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -297 +177,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -299,5 +178,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -318 +192,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -320,5 +193,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -329 +197,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -331,5 +198,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -342 +204,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -344,5 +205,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -355 +211,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -357,5 +212,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -368 +218,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -370,5 +219,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -384 +228,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -386,5 +229,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -407 +245,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -409,5 +246,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -433 +265,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -435,5 +266,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -446 +272,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -448,5 +273,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -459 +279,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -461,5 +280,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -472 +286,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -474,5 +287,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -499 +307,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -501,5 +308,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -521 +323,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -523,5 +324,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -556 +352,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -558,5 +353,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -569 +359,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -571,5 +360,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -582 +366,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -584,5 +367,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -595 +373,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -597,5 +374,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -608 +380,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -610,5 +381,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -621 +387,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -623,5 +388,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -634 +394,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -636,5 +395,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -647 +401,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -649,5 +402,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -660 +408,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -662,5 +409,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -673 +415,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -675,5 +416,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
@@ -686 +422,0 @@ class PersistableBundle {
-<<<<<<< HEAD
@@ -688,5 +423,0 @@ class PersistableBundle {
-||||||| eb83011e1d
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3a
