diff --git a/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@compaction_picker.no_comments_mergegen.cc b/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@compaction_picker.no_comments_merged.cc
index 432e894..3e4363b 100644
--- a/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@compaction_picker.no_comments_mergegen.cc
+++ b/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@compaction_picker.no_comments_merged.cc
@@ -2,0 +3,2 @@
+#include "util/log_buffer.h"
+#include "util/statistics.h"
