--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_merged.h
@@ -185 +186,2 @@ class DBImpl : public DB {
-                                   DeletionState& deletion_state);
+                                   DeletionState& deletion_state,
+                                   LogBuffer* log_buffer);
@@ -190 +191,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
@@ -192,8 +193,2 @@ class DBImpl : public DB {
-                          VersionEdit* edit, uint64_t* filenumber);
-||||||| e5fa4944f
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber);
-=======
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber, LogBuffer* log_buffer);
->>>>>>> d5de22dc099311b59a9e3d735702b6b9f18c7855
+                          VersionEdit* edit, uint64_t* filenumber,
+                          LogBuffer* log_buffer);
@@ -239,0 +237,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -303,4 +304,0 @@ class DBImpl : public DB {
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
