--- a/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@version_set.no_comments_mergegen.h
+++ b/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@version_set.no_comments_merged.h
@@ -25 +24,0 @@ class MemTable;
-class MergeContext; class TableCache;
@@ -28 +27,4 @@ class VersionSet;
-class MergeContext; class ColumnFamilyData; class ColumnFamilySet;
+class MergeContext;
+class ColumnFamilyData;
+class ColumnFamilySet;
+class TableCache;
