--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_merged.c
@@ -2 +1,0 @@
-#include <curses.h>
@@ -10 +8,0 @@
-#include <vis.h>
@@ -599,8 +647,3 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-                               int a, int b) {
-<<<<<<< HEAD
-  return (
-      tparm((char *)tty_term_string(term, code), a, b, 0, 0, 0, 0, 0, 0, 0));
-|||||||
-  return (tparm((char *)tty_term_string(term, code), a, b));
-=======
+const char *
+tty_term_string_ii(struct tty_term *term, enum tty_code_code code, int a, int b)
+{
@@ -611 +654,0 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
->>>>>>> 8f34504736cf3547992c4ba948c1e65f3813715c
@@ -627,8 +675,4 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-                               const char *a, const char *b) {
-<<<<<<< HEAD
-  return (tparm((char *)tty_term_string(term, code), (long)a, (long)b, 0, 0, 0,
-                0, 0, 0, 0));
-|||||||
-  return (tparm((char *)tty_term_string(term, code), a, b));
-=======
+const char *
+tty_term_string_ss(struct tty_term *term, enum tty_code_code code, const char *a,
+    const char *b)
+{
@@ -639 +683,0 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
->>>>>>> 8f34504736cf3547992c4ba948c1e65f3813715c
