diff --git a/./output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_mergegen.c b/./output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_merged.c
index 1c01bb5..6c3e6e0 100644
--- a/./output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_mergegen.c
+++ b/./output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_merged.c
@@ -641 +641,5 @@ tty_term_string_i(struct tty_term *term, enum tty_code_code code, int a)
-s = tparm((char *)x, a); if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name); return (s);
+ const char *x = tty_term_string(term, code), *s;
+ s = tparm((char *)x, a);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
+ return (s);
@@ -646 +650,5 @@ tty_term_string_ii(struct tty_term *term, enum tty_code_code code, int a, int b)
-s = tparm((char *)x, a, b); if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name); return (s);
+ const char *x = tty_term_string(term, code), *s;
+ s = tparm((char *)x, a, b);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
+ return (s);
@@ -652 +660,5 @@ tty_term_string_iii(struct tty_term *term, enum tty_code_code code, int a, int b
-s = tparm((char *)x, a, b, c); if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name); return (s);
+ const char *x = tty_term_string(term, code), *s;
+ s = tparm((char *)x, a, b, c);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
+ return (s);
@@ -657 +669,5 @@ tty_term_string_s(struct tty_term *term, enum tty_code_code code, const char *a)
-(long)a, 0, 0, 0, 0, 0, 0, 0, 0, 0)); *s; s = tparm((char *)x, (long)a); if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name); return (s);
+ const char *x = tty_term_string(term, code), *s;
+ s = tparm((char *)x, (long)a);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
+ return (s);
@@ -663 +679,5 @@ tty_term_string_ss(struct tty_term *term, enum tty_code_code code, const char *a
-const char *x = tty_term_string(term, code), (long)a, (long)b, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+ const char *x = tty_term_string(term, code), *s;
+ s = tparm((char *)x, (long)a, (long)b);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
+ return (s);
