diff --git a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread_list.no_comments_conflict.cc b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread_list.no_comments_merged.cc
index 8ef8ca0..e8d5ae0 100644
--- a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread_list.no_comments_conflict.cc
+++ b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread_list.no_comments_merged.cc
@@ -411,15 +410,0 @@ size_t ThreadList::FlipThreadRoots(Closure* thread_flip_visitor,
-<<<<<<< HEAD
-||||||| 4e97dd8703
-  Locks::mutator_lock_->ExclusiveUnlock(self);
-  collector->RegisterPause(NanoTime() - suspend_start_time);
-  if (pause_listener != nullptr) {
-    pause_listener->EndPause();
-  }
-=======
-  auto fake_mutator_lock_acquire = []() ACQUIRE(*Locks::mutator_lock_) NO_THREAD_SAFETY_ANALYSIS {};
-  auto fake_mutator_lock_release = []() RELEASE(*Locks::mutator_lock_) NO_THREAD_SAFETY_ANALYSIS {};
-  if (!gUseUserfaultfd) {
-    Locks::mutator_lock_->ExclusiveUnlock(self);
-    fake_mutator_lock_acquire();
-  }
->>>>>>> 9f9413b0
@@ -455,12 +439,0 @@ size_t ThreadList::FlipThreadRoots(Closure* thread_flip_visitor,
-<<<<<<< HEAD
-    for (Thread* thread : other_threads) {
-      thread->EnsureFlipFunctionStarted(self);
-      DCHECK(!thread->ReadFlag(ThreadFlag::kPendingFlipFunction));
-||||||| 4e97dd8703
-    ReaderMutexLock mu(self, *Locks::mutator_lock_);
-    for (Thread* thread : other_threads) {
-      thread->EnsureFlipFunctionStarted(self);
-      DCHECK(!thread->ReadFlag(ThreadFlag::kPendingFlipFunction));
-=======
-    if (gUseUserfaultfd) {
-      Locks::mutator_lock_->AssertExclusiveHeld(self);
@@ -473,12 +445,0 @@ size_t ThreadList::FlipThreadRoots(Closure* thread_flip_visitor,
-      Locks::mutator_lock_->ExclusiveUnlock(self);
-    } else {
-      fake_mutator_lock_release();
-      ReaderMutexLock mu(self, *Locks::mutator_lock_);
-      for (Thread* thread : other_threads) {
-        thread->EnsureFlipFunctionStarted(self);
-        DCHECK(!thread->ReadFlag(ThreadFlag::kPendingFlipFunction));
-      }
-      self->EnsureFlipFunctionStarted(self);
-      DCHECK(!self->ReadFlag(ThreadFlag::kPendingFlipFunction));
->>>>>>> 9f9413b0
-    }
