diff --git a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_conflict.h b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_merged.h
index 3d77125..033ea8a 100644
--- a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_conflict.h
+++ b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_merged.h
@@ -546 +545,0 @@ class Heap {
-<<<<<<< HEAD
@@ -550,7 +548,0 @@ class Heap {
-||||||| 4e97dd8703
-  CollectorType CurrentCollectorType() {
-=======
-  bool IsPerformingUffdCompaction() { return gUseUserfaultfd && mark_compact_->IsCompacting(); }
-  CollectorType CurrentCollectorType() {
-    DCHECK(!gUseUserfaultfd || collector_type_ == kCollectorTypeCMC);
->>>>>>> 9f9413b0
