diff --git a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_mergegen.h b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_merged.h
index a87b705..033ea8a 100644
--- a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_mergegen.h
+++ b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@gc@heap.no_comments_merged.h
@@ -546 +546,3 @@ class Heap {
-bool IsPerformingUffdCompaction() { return gUseUserfaultfd && mark_compact_->IsCompacting(); } CollectorType CurrentCollectorType() const { DCHECK(!gUseUserfaultfd || collector_type_ == kCollectorTypeCMC);
+  bool IsPerformingUffdCompaction() { return gUseUserfaultfd && mark_compact_->IsCompacting(); }
+  CollectorType CurrentCollectorType() const {
+    DCHECK(!gUseUserfaultfd || collector_type_ == kCollectorTypeCMC);
