diff --git a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread.no_comments_conflict.cc b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread.no_comments_merged.cc
index 9d0520e..68d10c5 100644
--- a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread.no_comments_conflict.cc
+++ b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@thread.no_comments_merged.cc
@@ -3523,4 +3523,2 @@ void Thread::VisitRoots(RootVisitor* visitor) {
-static void SweepCacheEntry(IsMarkedVisitor* visitor,
-                            const Instruction* inst,
-                            size_t* value,
-                            bool only_update_class) REQUIRES_SHARED(Locks::mutator_lock_) {
+static void SweepCacheEntry(IsMarkedVisitor* visitor, const Instruction* inst, size_t* value)
+    REQUIRES_SHARED(Locks::mutator_lock_) {
@@ -3538 +3535,0 @@ static void SweepCacheEntry(IsMarkedVisitor* visitor,
-<<<<<<< HEAD
@@ -3541,15 +3537,0 @@ static void SweepCacheEntry(IsMarkedVisitor* visitor,
-||||||| 4e97dd8703
-      mirror::Class* cls = reinterpret_cast<mirror::Class*>(*value);
-      if (cls == nullptr || cls == Runtime::GetWeakClassSentinel()) {
-=======
-      if (!only_update_class) {
-        mirror::Class* cls = reinterpret_cast<mirror::Class*>(*value);
-        if (cls == nullptr || cls == Runtime::GetWeakClassSentinel()) {
-          return;
-        }
-        Runtime::ProcessWeakClass(reinterpret_cast<GcRoot<mirror::Class>*>(value),
-                                  visitor,
-                                  Runtime::GetWeakClassSentinel());
-        return;
-      } else if (reinterpret_cast<mirror::Class*>(*value) == Runtime::GetWeakClassSentinel()) {
->>>>>>> 9f9413b0
@@ -3558 +3539,0 @@ static void SweepCacheEntry(IsMarkedVisitor* visitor,
-<<<<<<< HEAD
@@ -3566,9 +3546,0 @@ static void SweepCacheEntry(IsMarkedVisitor* visitor,
-||||||| 4e97dd8703
-      Runtime::ProcessWeakClass(
-          reinterpret_cast<GcRoot<mirror::Class>*>(value),
-          visitor,
-          Runtime::GetWeakClassSentinel());
-      return;
-=======
-      FALLTHROUGH_INTENDED;
->>>>>>> 9f9413b0
@@ -3598 +3569,0 @@ void Thread::SweepInterpreterCache(IsMarkedVisitor* visitor) {
-  bool only_update_class = Runtime::Current()->GetHeap()->IsPerformingUffdCompaction();
@@ -3600,4 +3571 @@ void Thread::SweepInterpreterCache(IsMarkedVisitor* visitor) {
-    SweepCacheEntry(visitor,
-                    reinterpret_cast<const Instruction*>(entry.first),
-                    &entry.second,
-                    only_update_class);
+    SweepCacheEntry(visitor, reinterpret_cast<const Instruction*>(entry.first), &entry.second);
