diff --git a/./output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_mergegen.cc b/./output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_merged.cc
index 27bcfcd..8474f01 100644
--- a/./output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_mergegen.cc
+++ b/./output/grpc/16fc98f5-172f61a3-18695e30/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_merged.cc
@@ -1893 +1893,4 @@ void grpc_chttp2_mark_stream_closed(grpc_chttp2_transport* t,
-if (s->read_closed && s->write_closed) {
+    grpc_error* overall_error = removal_error(error, s, "Stream removed");
+    if (overall_error != GRPC_ERROR_NONE) {
+      grpc_chttp2_fake_status(t, s, overall_error);
+    }
