diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/log.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/log.no_comments_merged.c
index eaa0085..f7f61a8 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/log.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/log.no_comments_merged.c
@@ -7 +6,0 @@
-#include <vis.h>
@@ -13 +12,3 @@ static void log_vwrite(const char *, va_list);
-static void log_event_cb(__unused int severity, const char *msg) {
+static void
+log_event_cb(__unused int severity, const char *msg)
+{
@@ -16,3 +17,13 @@ static void log_event_cb(__unused int severity, const char *msg) {
-voidlog_add_level(void) { log_level++; }
-intlog_get_level(void) { return (log_level); }
-void log_open(const char *name) {
+void
+log_add_level(void)
+{
+ log_level++;
+}
+int
+log_get_level(void)
+{
+ return (log_level);
+}
+void
+log_open(const char *name)
+{
@@ -20,2 +31,4 @@ void log_open(const char *name) {
-  if (log_level == 0) return;
-  if (log_file != NULL) fclose(log_file);
+ if (log_level == 0)
+  return;
+ if (log_file != NULL)
+  fclose(log_file);
@@ -25 +38,2 @@ void log_open(const char *name) {
-  if (log_file == NULL) return;
+ if (log_file == NULL)
+  return;
@@ -29,2 +43,5 @@ void log_open(const char *name) {
-void log_close(void) {
-  if (log_file != NULL) fclose(log_file);
+void
+log_close(void)
+{
+ if (log_file != NULL)
+  fclose(log_file);
@@ -34 +51,3 @@ void log_close(void) {
-static void log_vwrite(const char *msg, va_list ap) {
+static void
+log_vwrite(const char *msg, va_list ap)
+{
@@ -37,2 +56,4 @@ static void log_vwrite(const char *msg, va_list ap) {
-  if (log_file == NULL) return;
-  if (vasprintf(&fmt, msg, ap) == -1) exit(1);
+ if (log_file == NULL)
+  return;
+ if (vasprintf(&fmt, msg, ap) == -1)
+  exit(1);
@@ -42,2 +63,2 @@ static void log_vwrite(const char *msg, va_list ap) {
-  if (fprintf(log_file, "%lld.%06d %s\n", (long long)tv.tv_sec, (int)tv.tv_usec,
-              out) == -1)
+ if (fprintf(log_file, "%lld.%06d %s\n", (long long)tv.tv_sec,
+     (int)tv.tv_usec, out) == -1)
@@ -49 +70,3 @@ static void log_vwrite(const char *msg, va_list ap) {
-void log_debug(const char *msg, ...) {
+void
+log_debug(const char *msg, ...)
+{
@@ -55 +78,3 @@ void log_debug(const char *msg, ...) {
-__dead void fatal(const char *msg, ...) {
+__dead void
+fatal(const char *msg, ...)
+{
@@ -59 +84,2 @@ __dead void fatal(const char *msg, ...) {
-  if (asprintf(&fmt, "fatal: %s: %s", msg, strerror(errno)) == -1) exit(1);
+ if (asprintf(&fmt, "fatal: %s: %s", msg, strerror(errno)) == -1)
+  exit(1);
@@ -63 +89,3 @@ __dead void fatal(const char *msg, ...) {
-__dead void fatalx(const char *msg, ...) {
+__dead void
+fatalx(const char *msg, ...)
+{
@@ -67 +95,2 @@ __dead void fatalx(const char *msg, ...) {
-  if (asprintf(&fmt, "fatal: %s", msg) == -1) exit(1);
+ if (asprintf(&fmt, "fatal: %s", msg) == -1)
+  exit(1);
