diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_merged.c
index 4bb5874..0e07c81 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_merged.c
@@ -2,3 +1,0 @@
-#include <sys/queue.h>
-#include <sys/utsname.h>
-#include <imsg.h>
@@ -5,0 +3 @@
+#include <sys/utsname.h>
@@ -28 +26,3 @@ static void proc_update_event(struct tmuxpeer *);
-static void proc_event_cb(__unused int fd, short events, void *arg) {
+static void
+proc_event_cb(__unused int fd, short events, void *arg)
+{
@@ -42 +42,2 @@ static void proc_event_cb(__unused int fd, short events, void *arg) {
-      if (n == 0) break;
+   if (n == 0)
+    break;
@@ -45 +46,2 @@ static void proc_event_cb(__unused int fd, short events, void *arg) {
-        if (imsg.fd != -1) close(imsg.fd);
+    if (imsg.fd != -1)
+     close(imsg.fd);
@@ -65 +67,3 @@ static void proc_event_cb(__unused int fd, short events, void *arg) {
-static void proc_signal_cb(int signo, __unused short events, void *arg) {
+static void
+proc_signal_cb(int signo, __unused short events, void *arg)
+{
@@ -69 +73,3 @@ static void proc_signal_cb(int signo, __unused short events, void *arg) {
-static int peer_check_version(struct tmuxpeer *peer, struct imsg *imsg) {
+static int
+peer_check_version(struct tmuxpeer *peer, struct imsg *imsg)
+{
@@ -80 +86,3 @@ static int peer_check_version(struct tmuxpeer *peer, struct imsg *imsg) {
-static void proc_update_event(struct tmuxpeer *peer) {
+static void
+proc_update_event(struct tmuxpeer *peer)
+{
@@ -84 +92,2 @@ static void proc_update_event(struct tmuxpeer *peer) {
-  if (peer->ibuf.w.queued > 0) events |= EV_WRITE;
+ if (peer->ibuf.w.queued > 0)
+  events |= EV_WRITE;
@@ -88,2 +97,4 @@ static void proc_update_event(struct tmuxpeer *peer) {
-int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-              size_t len) {
+int
+proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
+    size_t len)
+{
@@ -93 +104,2 @@ int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-  if (peer->flags & PEER_BAD) return (-1);
+ if (peer->flags & PEER_BAD)
+  return (-1);
@@ -96 +108,2 @@ int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-  if (retval != 1) return (-1);
+ if (retval != 1)
+  return (-1);
@@ -100 +113,3 @@ int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-int proc_send_s(struct tmuxpeer *peer, enum msgtype type, const char *s) {
+int
+proc_send_s(struct tmuxpeer *peer, enum msgtype type, const char *s)
+{
@@ -103,2 +118,4 @@ int proc_send_s(struct tmuxpeer *peer, enum msgtype type, const char *s) {
-struct tmuxproc *proc_start(const char *name, struct event_base *base,
-                            int forkflag, void (*signalcb)(int)) {
+struct tmuxproc *
+proc_start(const char *name, struct event_base *base, int forkflag,
+    void (*signalcb)(int))
+{
@@ -116 +133,2 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-    if (daemon(1, 0) != 0) fatal("daemon failed");
+  if (daemon(1, 0) != 0)
+   fatal("daemon failed");
@@ -118 +136,2 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-    if (event_reinit(base) != 0) fatalx("event_reinit failed");
+  if (event_reinit(base) != 0)
+   fatalx("event_reinit failed");
@@ -120 +138,0 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-<<<<<<< HEAD
@@ -123,2 +140,0 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-  ||||||| logfile(name);
-=======
@@ -126 +141,0 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
->>>>>>> ac8678aefe157d7e40c5bcedd12333eaedf0df92
@@ -129,5 +144,6 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-  if (uname(&u) < 0) memset(&u, 0, sizeof u);
-  log_debug("%s started (%ld): socket %s, protocol %d", name, (long)getpid(),
-            socket_path, PROTOCOL_VERSION);
-  log_debug("on %s %s %s; libevent %s (%s)", u.sysname, u.release, u.version,
-            event_get_version(), event_get_method());
+ if (uname(&u) < 0)
+  memset(&u, 0, sizeof u);
+ log_debug("%s started (%ld): socket %s, protocol %d", name,
+     (long)getpid(), socket_path, PROTOCOL_VERSION);
+ log_debug("on %s %s %s; libevent %s (%s)", u.sysname, u.release,
+     u.version, event_get_version(), event_get_method());
@@ -140 +156,3 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-void proc_loop(struct tmuxproc *tp, int (*loopcb)(void)) {
+void
+proc_loop(struct tmuxproc *tp, int (*loopcb)(void))
+{
@@ -142 +160,2 @@ void proc_loop(struct tmuxproc *tp, int (*loopcb)(void)) {
-  do event_loop(EVLOOP_ONCE);
+ do
+  event_loop(EVLOOP_ONCE);
@@ -146,4 +165,9 @@ void proc_loop(struct tmuxproc *tp, int (*loopcb)(void)) {
-void proc_exit(struct tmuxproc *tp) { tp->exit = 1; }
-struct tmuxpeer *proc_add_peer(struct tmuxproc *tp, int fd,
-                               void (*dispatchcb)(struct imsg *, void *),
-                               void *arg) {
+void
+proc_exit(struct tmuxproc *tp)
+{
+ tp->exit = 1;
+}
+struct tmuxpeer *
+proc_add_peer(struct tmuxproc *tp, int fd,
+    void (*dispatchcb)(struct imsg *, void *), void *arg)
+{
@@ -161 +185,3 @@ struct tmuxpeer *proc_add_peer(struct tmuxproc *tp, int fd,
-void proc_remove_peer(struct tmuxpeer *peer) {
+void
+proc_remove_peer(struct tmuxpeer *peer)
+{
@@ -168 +194,5 @@ void proc_remove_peer(struct tmuxpeer *peer) {
-void proc_kill_peer(struct tmuxpeer *peer) { peer->flags |= PEER_BAD; }
+void
+proc_kill_peer(struct tmuxpeer *peer)
+{
+ peer->flags |= PEER_BAD;
+}
