--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_merged.c
@@ -7,2 +6,0 @@
-#include <getopt.h>
-#include <paths.h>
@@ -19,3 +16,0 @@ extern char *malloc_options;
-#ifdef DEBUG
-extern char *malloc_options;
-#endif
@@ -125 +146,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
@@ -129,6 +149,0 @@ int main(int argc, char **argv) {
-  |||||||
-#ifdef DEBUG
-      malloc_options = (char *)"AFGJPX";
-#endif
-  =======
->>>>>>> ac8678aefe157d7e40c5bcedd12333eaedf0df92
@@ -238,5 +258,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
-  exit(client_main(osdep_event_init(), argc, argv, flags));
-|||||||
-  exit(client_main(event_init(), argc, argv, flags));
-=======
@@ -244 +259,0 @@ int main(int argc, char **argv) {
->>>>>>> ac8678aefe157d7e40c5bcedd12333eaedf0df92
