diff --git a/./output/frameworks_native/e8caee22-c3e00913-0b3dbcef/libs@graphicsenv@GraphicsEnv.no_comments_mergegen.cpp b/./output/frameworks_native/e8caee22-c3e00913-0b3dbcef/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
index 49d3062..4a8d3c1 100644
--- a/./output/frameworks_native/e8caee22-c3e00913-0b3dbcef/libs@graphicsenv@GraphicsEnv.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/e8caee22-c3e00913-0b3dbcef/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
@@ -414 +414,2 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const bool shouldUseNati
-if (mAnglePath == "system") { mShouldUseSystemAngle = true;
+    if (mAnglePath == "system") {
+        mShouldUseSystemAngle = true;
@@ -427 +427,0 @@ const std::vector<std::string>& GraphicsEnv::getAngleEglFeatures() {
-bool GraphicsEnv::linkDriverNamespaceLocked(android_namespace_t* destNamespace, android_namespace_t* vndkNamespace, const std::string& sharedSphalLibraries) { const std::string llndkLibraries = getSystemNativeLibraries(NativeLibrary::LLNDK); if (llndkLibraries.empty()) { return false; } if (!android_link_namespaces(destNamespace, nullptr, llndkLibraries.c_str())) { ALOGE("Failed to link default namespace[%s]", dlerror()); return false; } const std::string vndkspLibraries = getSystemNativeLibraries(NativeLibrary::VNDKSP); if (vndkspLibraries.empty()) { return false; } if (!android_link_namespaces(destNamespace, vndkNamespace, vndkspLibraries.c_str())) { ALOGE("Failed to link vndk namespace[%s]", dlerror()); return false; } if (sharedSphalLibraries.empty()) { return true; }
@@ -478 +478,31 @@ bool GraphicsEnv::shouldUseSystemAngle() {
-bool GraphicsEnv::shouldUseNativeDriver() { return mShouldUseNativeDriver; } void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace, const std::string& layerPaths) { if (mLayerPaths.empty()) { mLayerPaths = layerPaths; mAppNamespace = appNamespace; } else { ALOGV("Vulkan layer search path already set, not clobbering with '%s' for namespace %p'", layerPaths.c_str(), appNamespace); } } NativeLoaderNamespace* GraphicsEnv::getAppNamespace() { return mAppNamespace; } const std::string& GraphicsEnv::getLayerPaths() { return mLayerPaths; } const std::string& GraphicsEnv::getDebugLayers() { return mDebugLayers; } const std::string& GraphicsEnv::getDebugLayersGLES() { return mDebugLayersGLES; } void GraphicsEnv::setDebugLayers(const std::string& layers) { mDebugLayers = layers; } void GraphicsEnv::setDebugLayersGLES(const std::string& layers) { mDebugLayersGLES = layers; }
+bool GraphicsEnv::shouldUseNativeDriver() {
+    return mShouldUseNativeDriver;
+}
+void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace,
+                                const std::string& layerPaths) {
+    if (mLayerPaths.empty()) {
+        mLayerPaths = layerPaths;
+        mAppNamespace = appNamespace;
+    } else {
+        ALOGV("Vulkan layer search path already set, not clobbering with '%s' for namespace %p'",
+              layerPaths.c_str(), appNamespace);
+    }
+}
+NativeLoaderNamespace* GraphicsEnv::getAppNamespace() {
+    return mAppNamespace;
+}
+const std::string& GraphicsEnv::getLayerPaths() {
+    return mLayerPaths;
+}
+const std::string& GraphicsEnv::getDebugLayers() {
+    return mDebugLayers;
+}
+const std::string& GraphicsEnv::getDebugLayersGLES() {
+    return mDebugLayersGLES;
+}
+void GraphicsEnv::setDebugLayers(const std::string& layers) {
+    mDebugLayers = layers;
+}
+void GraphicsEnv::setDebugLayersGLES(const std::string& layers) {
+    mDebugLayersGLES = layers;
+}
