--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/f615d49b-5f4e8eac-d43b6152/init@devices.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/f615d49b-5f4e8eac-d43b6152/init@devices.no_comments_merged.h
@@ -3 +2,0 @@
-#include <functional>
@@ -12,4 +10,0 @@
-#include <chrono>
-#include <optional>
-void device_poll(const coldboot_callback& callback = nullptr,
-                 const std::optional<std::chrono::milliseconds> relative_timeout = {});
@@ -70 +65 @@ class DeviceHandler {
-    ~DeviceHandler() {}
+    ~DeviceHandler(){};
