diff --git a/./output/art/fd522049-c15aee68-e82f25f3/artd@artd.no_comments_mergegen.h b/./output/art/fd522049-c15aee68-e82f25f3/artd@artd.no_comments_merged.h
index 8d786bf..bfe5a92 100644
--- a/./output/art/fd522049-c15aee68-e82f25f3/artd@artd.no_comments_mergegen.h
+++ b/./output/art/fd522049-c15aee68-e82f25f3/artd@artd.no_comments_merged.h
@@ -121 +121,6 @@ class Artd : public aidl::com::android::server::art::BnArtd {
-std::ScopedAStatus isInDalvikCache(const std::string& in_dexFile, bool* _aidl_return) override; ndk::ScopedAStatus validateDexPath(const std::string& in_dexPath, std::optional<std::string>* _aidl_return) override; ndk::ScopedAStatus validateClassLoaderContext(const std::string& in_dexPath, const std::string& in_classLoaderContext, std::optional<std::string>* _aidl_return) override;
+  ndk::ScopedAStatus isInDalvikCache(const std::string& in_dexFile, bool* _aidl_return) override;
+  ndk::ScopedAStatus validateDexPath(const std::string& in_dexPath,
+                                     std::optional<std::string>* _aidl_return) override;
+  ndk::ScopedAStatus validateClassLoaderContext(const std::string& in_dexPath,
+                                                const std::string& in_classLoaderContext,
+                                                std::optional<std::string>* _aidl_return) override;
