--- a/./output/art/52670375-d5a49147-bbe713b7/runtime@runtime.no_comments_conflict.cc
+++ b/./output/art/52670375-d5a49147-bbe713b7/runtime@runtime.no_comments_merged.cc
@@ -580 +579,0 @@ static void WaitUntilSingleThreaded() {
-<<<<<<< HEAD
@@ -587,8 +585,0 @@ static void WaitUntilSingleThreaded() {
-||||||| bbe713b72e
-=======
-  static constexpr ssize_t BUF_SIZE = 500;
-  static constexpr ssize_t BUF_PRINT_SIZE = 150;
-  static_assert(BUF_SIZE > BUF_PRINT_SIZE);
-  char buf[BUF_SIZE];
-  ssize_t bytes_read = -1;
->>>>>>> d5a49147
@@ -596 +586,0 @@ static void WaitUntilSingleThreaded() {
-<<<<<<< HEAD
@@ -600,19 +589,0 @@ static void WaitUntilSingleThreaded() {
-||||||| bbe713b72e
-    static constexpr int BUF_SIZE = 500;
-    char buf[BUF_SIZE];
-    int stat_fd = open("/proc/self/stat", O_RDONLY | O_CLOEXEC);
-    CHECK(stat_fd >= 0) << strerror(errno);
-    ssize_t bytes_read = TEMP_FAILURE_RETRY(read(stat_fd, buf, BUF_SIZE));
-    CHECK(bytes_read >= 0) << strerror(errno);
-    int ret = close(stat_fd);
-    DCHECK(ret == 0) << strerror(errno);
-    ssize_t pos = 0;
-=======
-    int stat_fd = open("/proc/self/stat", O_RDONLY | O_CLOEXEC);
-    CHECK(stat_fd >= 0) << strerror(errno);
-    bytes_read = TEMP_FAILURE_RETRY(read(stat_fd, buf, BUF_SIZE));
-    CHECK(bytes_read >= 0) << strerror(errno);
-    int ret = close(stat_fd);
-    DCHECK(ret == 0) << strerror(errno);
-    ssize_t pos = 0;
->>>>>>> d5a49147
@@ -637 +607,0 @@ static void WaitUntilSingleThreaded() {
-<<<<<<< HEAD
@@ -647,7 +616,0 @@ static void WaitUntilSingleThreaded() {
-||||||| bbe713b72e
-  LOG(FATAL) << "Failed to reach single-threaded state";
-=======
-  buf[std::min(BUF_PRINT_SIZE, bytes_read)] = '\0';
-  LOG(FATAL) << "Failed to reach single-threaded state: bytes_read = " << bytes_read
-             << " stat contents = \"" << buf << "...\"";
->>>>>>> d5a49147
