diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_merged.h
index 24a7c2a..019cc91 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_merged.h
@@ -7,4 +7,2 @@ namespace init {
-static constexpr const char kRestoreconProperty[] =
-    "selinux.restorecon_recursive";
-bool CanReadProperty(const std::string& source_context,
-                     const std::string& name);
+static constexpr const char kRestoreconProperty[] = "selinux.restorecon_recursive";
+bool CanReadProperty(const std::string& source_context, const std::string& name);
