diff --git a/./output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_conflict.h b/./output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_merged.h
index 6d4935e..019cc91 100644
--- a/./output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_conflict.h
+++ b/./output/system_core/e09432e1-c9d0e479-2707e2de/init@property_service.no_comments_merged.h
@@ -11 +10,0 @@ void StartPropertyService(int* epoll_socket);
-<<<<<<< HEAD
@@ -14,5 +12,0 @@ void StopSendingMessages();
-||||||| 2707e2de3
-void ResumePropertyService();
-void PausePropertyService();
-=======
->>>>>>> c9d0e479
