diff --git a/./output/frameworks_av/a54cf01e-1eeb251f-5c0ed421/services@camera@libcameraservice@CameraService.no_comments_conflict.cpp b/./output/frameworks_av/a54cf01e-1eeb251f-5c0ed421/services@camera@libcameraservice@CameraService.no_comments_merged.cpp
index d77bd60..382892d 100644
--- a/./output/frameworks_av/a54cf01e-1eeb251f-5c0ed421/services@camera@libcameraservice@CameraService.no_comments_conflict.cpp
+++ b/./output/frameworks_av/a54cf01e-1eeb251f-5c0ed421/services@camera@libcameraservice@CameraService.no_comments_merged.cpp
@@ -64 +63,0 @@
-<<<<<<< HEAD
@@ -66,5 +64,0 @@
-||||||| 5c0ed42195
-=======
-#include "utils/CameraThreadState.h"
-#include "utils/CameraTraces.h"
->>>>>>> 1eeb251f
