diff --git a/./output/art/9eadb229-b5a9e3a4-fbd7b4da/runtime@runtime.no_comments_conflict.h b/./output/art/9eadb229-b5a9e3a4-fbd7b4da/runtime@runtime.no_comments_merged.h
index 9de5427..7a7a633 100644
--- a/./output/art/9eadb229-b5a9e3a4-fbd7b4da/runtime@runtime.no_comments_conflict.h
+++ b/./output/art/9eadb229-b5a9e3a4-fbd7b4da/runtime@runtime.no_comments_merged.h
@@ -803 +802,0 @@ class Runtime {
-<<<<<<< HEAD
@@ -814,11 +812,0 @@ class Runtime {
-||||||| fbd7b4da9f
-=======
-  void SetOutOfMemoryErrorHook(void (*hook)()) {
-    out_of_memory_error_hook_ = hook;
-  }
-  void OutOfMemoryErrorHook() {
-    if (out_of_memory_error_hook_ != nullptr) {
-      out_of_memory_error_hook_();
-    }
-  }
->>>>>>> b5a9e3a4
