diff --git a/./output/redis/d8d528e9-c54afb6d-c086b85a/src@redis-cli.no_comments_mergegen.c b/./output/redis/d8d528e9-c54afb6d-c086b85a/src@redis-cli.no_comments_merged.c
index 976c1ca..4a9b894 100644
--- a/./output/redis/d8d528e9-c54afb6d-c086b85a/src@redis-cli.no_comments_mergegen.c
+++ b/./output/redis/d8d528e9-c54afb6d-c086b85a/src@redis-cli.no_comments_merged.c
@@ -414,0 +415 @@ static void repl() {
+    linenoiseSetCompletionCallback(completionCallback);
