--- a/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_conflict.cpp
+++ b/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
@@ -3709 +3708,0 @@ void SurfaceFlinger::setPowerModeInternal(const sp<DisplayDevice>& display, hal:
-<<<<<<< HEAD
@@ -3712,7 +3710,0 @@ void SurfaceFlinger::setPowerModeInternal(const sp<DisplayDevice>& display, hal:
-||||||| f2b19cdcde
-    const auto refreshRate = display->refreshRateConfigs().getActiveMode()->getFps();
-    if (currentMode == hal::PowerMode::OFF) {
-=======
-    const auto refreshRate = display->refreshRateConfigs().getActiveMode()->getFps();
-    if (*currentMode == hal::PowerMode::OFF) {
->>>>>>> 5c7935f8
@@ -3752 +3743,0 @@ void SurfaceFlinger::setPowerModeInternal(const sp<DisplayDevice>& display, hal:
-<<<<<<< HEAD
@@ -3757,5 +3747,0 @@ void SurfaceFlinger::setPowerModeInternal(const sp<DisplayDevice>& display, hal:
-||||||| f2b19cdcde
-        if (isDisplayActiveLocked(display) && currentMode == hal::PowerMode::DOZE_SUSPEND) {
-=======
-        if (isDisplayActiveLocked(display) && *currentMode == hal::PowerMode::DOZE_SUSPEND) {
->>>>>>> 5c7935f8
