--- a/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
@@ -3742,0 +3744,4 @@ const auto refreshRate = display->refreshRateConfigs().getActiveMode().getFps();
+        if (isDisplayActiveLocked(display) && *currentMode == hal::PowerMode::DOZE_SUSPEND) {
+            ALOGI("Force repainting for DOZE_SUSPEND -> DOZE or ON.");
+            mVisibleRegionsDirty = true;
+            scheduleRepaint();
