diff --git a/./output/frameworks_av/5b7f6327-b4efe510-99c77e68/services@audiopolicy@service@AudioPolicyEffects.no_comments_conflict.h b/./output/frameworks_av/5b7f6327-b4efe510-99c77e68/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
index a2da8f7..7b2201c 100644
--- a/./output/frameworks_av/5b7f6327-b4efe510-99c77e68/services@audiopolicy@service@AudioPolicyEffects.no_comments_conflict.h
+++ b/./output/frameworks_av/5b7f6327-b4efe510-99c77e68/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
@@ -19,6 +18,0 @@ public:
-<<<<<<< HEAD
-||||||| 99c77e683c
-    virtual ~AudioPolicyEffects();
-=======
-    ~AudioPolicyEffects() override;
->>>>>>> b4efe510
@@ -56 +49,0 @@ public:
-<<<<<<< HEAD
@@ -58,5 +50,0 @@ public:
-||||||| 99c77e683c
-    void setDefaultDeviceEffects();
-=======
-    void setDefaultDeviceEffects();
->>>>>>> b4efe510
@@ -124 +111,0 @@ private:
-<<<<<<< HEAD
@@ -132,12 +118,0 @@ private:
-||||||| 99c77e683c
-    static const char * const kInputSourceNames[AUDIO_SOURCE_CNT -1];
-=======
-    void initDefaultDeviceEffects() EXCLUDES(mDeviceEffectsMutex) EXCLUDES_EffectHandle_Mutex;
-    status_t loadAudioEffectConfig_ll(const sp<EffectsFactoryHalInterface>& effectsFactoryHal)
-            REQUIRES(mMutex, mDeviceEffectsMutex);
-    status_t loadAudioEffectConfigLegacy_l(const char* path) REQUIRES(mMutex);
-    status_t loadInputEffectConfigurations_l(
-            cnode* root, const Vector<EffectDesc*>& effects) REQUIRES(mMutex);
-    status_t loadStreamEffectConfigurations_l(
-            cnode* root, const Vector<EffectDesc*>& effects) REQUIRES(mMutex);
->>>>>>> b4efe510
@@ -146 +120,0 @@ private:
-<<<<<<< HEAD
@@ -151,12 +124,0 @@ private:
-||||||| 99c77e683c
-    status_t loadEffects(cnode *root, Vector <EffectDesc *>& effects);
-    EffectDesc *loadEffect(cnode *root);
-    status_t loadInputEffectConfigurations(cnode *root, const Vector <EffectDesc *>& effects);
-    status_t loadStreamEffectConfigurations(cnode *root, const Vector <EffectDesc *>& effects);
-    EffectDescVector *loadEffectConfig(cnode *root, const Vector <EffectDesc *>& effects);
-=======
-    static status_t loadEffects(cnode* root, Vector<EffectDesc*>& effects);
-    static EffectDesc* loadEffect(cnode* root);
-    static EffectDescVector *loadEffectConfig(cnode *root, const Vector <EffectDesc *>& effects);
->>>>>>> b4efe510
-<<<<<<< HEAD
@@ -167,9 +128,0 @@ private:
-||||||| 99c77e683c
-    void loadEffectParameters(cnode *root, Vector <effect_param_t *>& params);
-    effect_param_t *loadEffectParameter(cnode *root);
-    size_t readParamValue(cnode *node,
-=======
-    static void loadEffectParameters(cnode* root, Vector<effect_param_t*>& params);
-    static effect_param_t* loadEffectParameter(cnode* root);
-    static size_t readParamValue(cnode* node,
->>>>>>> b4efe510
@@ -183,4 +135,0 @@ private:
-<<<<<<< HEAD
-||||||| 99c77e683c
-=======
->>>>>>> b4efe510
@@ -188 +136,0 @@ private:
-<<<<<<< HEAD
@@ -193,8 +140,0 @@ private:
-||||||| 99c77e683c
-    KeyedVector< audio_source_t, EffectDescVector* > mInputSources;
-    KeyedVector< audio_session_t, EffectVector* > mInputSessions;
-=======
-    KeyedVector<audio_source_t, EffectDescVector*> mInputSources GUARDED_BY(mMutex);
-    KeyedVector<audio_session_t, EffectVector*> mInputSessions GUARDED_BY(mMutex);
->>>>>>> b4efe510
-<<<<<<< HEAD
@@ -205,16 +144,0 @@ private:
-||||||| 99c77e683c
-    KeyedVector< audio_stream_type_t, EffectDescVector* > mOutputStreams;
-    KeyedVector< audio_session_t, EffectVector* > mOutputSessions;
-=======
-    KeyedVector<audio_stream_type_t, EffectDescVector*> mOutputStreams GUARDED_BY(mMutex);
-    KeyedVector<audio_session_t, EffectVector*> mOutputSessions GUARDED_BY(mMutex);
->>>>>>> b4efe510
-<<<<<<< HEAD
-||||||| 99c77e683c
-    std::map<std::string, std::unique_ptr<DeviceEffects>> mDeviceEffects GUARDED_BY(mMutex);
-=======
-    std::mutex mDeviceEffectsMutex;
-    std::map<std::string, std::unique_ptr<DeviceEffects>> mDeviceEffects
-            GUARDED_BY(mDeviceEffectsMutex);
->>>>>>> b4efe510
-<<<<<<< HEAD
@@ -224,5 +147,0 @@ private:
-||||||| 99c77e683c
-    std::future<void> mDefaultDeviceEffectFuture;
-=======
-    std::future<void> mDefaultDeviceEffectFuture;
->>>>>>> b4efe510
