diff --git a/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_mergegen.cpp b/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_merged.cpp
index 1323792..de7a0e4 100644
--- a/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_merged.cpp
@@ -774 +774,8 @@ status_t AudioFlingerClientAdapter::getAudioMixPort(const struct audio_port_v7 *
-status_t AudioFlingerClientAdapter::setTracksInternalMute( const std::vector<media::TrackInternalMuteInfo>& tracksInternalMuted) { return statusTFromBinderStatus(mDelegate->setTracksInternalMute(tracksInternalMuted)); } status_t AudioFlingerClientAdapter::resetReferencesForTest() { RETURN_STATUS_IF_ERROR(statusTFromBinderStatus(mDelegate->resetReferencesForTest())); return OK; }
+status_t AudioFlingerClientAdapter::setTracksInternalMute(
+        const std::vector<media::TrackInternalMuteInfo>& tracksInternalMuted) {
+    return statusTFromBinderStatus(mDelegate->setTracksInternalMute(tracksInternalMuted));
+}
+status_t AudioFlingerClientAdapter::resetReferencesForTest() {
+    RETURN_STATUS_IF_ERROR(statusTFromBinderStatus(mDelegate->resetReferencesForTest()));
+    return OK;
+}
@@ -1252 +1259,8 @@ Status AudioFlingerServerAdapter::getAudioMixPort(const media::AudioPortFw &devi
-Status AudioFlingerServerAdapter::setTracksInternalMute( const std::vector<media::TrackInternalMuteInfo>& tracksInternalMute) { return Status::fromStatusT(mDelegate->setTracksInternalMute(tracksInternalMute)); } Status AudioFlingerServerAdapter::resetReferencesForTest() { RETURN_BINDER_IF_ERROR(mDelegate->resetReferencesForTest()); return Status::ok(); }
+Status AudioFlingerServerAdapter::setTracksInternalMute(
+        const std::vector<media::TrackInternalMuteInfo>& tracksInternalMute) {
+    return Status::fromStatusT(mDelegate->setTracksInternalMute(tracksInternalMute));
+}
+Status AudioFlingerServerAdapter::resetReferencesForTest() {
+    RETURN_BINDER_IF_ERROR(mDelegate->resetReferencesForTest());
+    return Status::ok();
+}
