--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_merged.h
@@ -50 +50 @@ class IAudioFlinger : public virtual RefBase {
-    virtual ~IAudioFlinger()
+    virtual ~IAudioFlinger() = default;
@@ -425,3 +426 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-<<<<<<< HEAD
-            SET_TRACKS_INTERNAL_MUTE =
-                    media::BnAudioFlingerService::TRANSACTION_setTracksInternalMute,
+            SET_TRACKS_INTERNAL_MUTE = media::BnAudioFlingerService::TRANSACTION_setTracksInternalMute,
@@ -430,5 +428,0 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-||||||| 7f0218eb8c
-=======
-            RESET_REFERENCES_FOR_TEST =
-                    media::BnAudioFlingerService::TRANSACTION_resetReferencesForTest,
->>>>>>> fd414a83bb92cf343b57eb7fbab48d56e2702007
@@ -557 +555 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-}
+};
