--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/services@audioflinger@AudioFlinger.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/services@audioflinger@AudioFlinger.no_comments_merged.cpp
@@ -42,3 +42 @@
-#define ALOGVV(a...) \
-    do { \
-    } while (0)
+#define ALOGVV(a...) do { } while(0)
@@ -48,2 +45,0 @@ using ::android::base::StringPrintf;
-using android::content::AttributionSourceState;
-using android::detail::AudioHalVersionInfo;
@@ -53,0 +50,2 @@ using media::audio::common::AudioMode;
+using android::content::AttributionSourceState;
+using android::detail::AudioHalVersionInfo;
@@ -145 +144 @@ static void sMediaLogInit() {
-    BINDER_METHOD_ENTRY(resetReferencesForTest)
+BINDER_METHOD_ENTRY(resetReferencesForTest) \
@@ -3796 +4020,0 @@ status_t AudioFlinger::getAudioMixPort(const struct audio_port_v7* devicePort,
-<<<<<<< HEAD
@@ -3818,10 +4042,2 @@ status_t AudioFlinger::resetReferencesForTest() {
-||||||| 7f0218eb8c
-=======
-status_t AudioFlinger::resetReferencesForTest() {
-    mDeviceEffectManager.clear();
-    mPatchPanel.clear();
-    mMelReporter->resetReferencesForTest();
-    return NO_ERROR;
-}
->>>>>>> fd414a83bb92cf343b57eb7fbab48d56e2702007
-status_t AudioFlinger::onTransactWrapper(TransactionCode code, [[maybe_unused]] const Parcel& data,
+status_t AudioFlinger::onTransactWrapper(TransactionCode code,
+                                         [[maybe_unused]] const Parcel& data,
@@ -3857 +4072,0 @@ status_t AudioFlinger::onTransactWrapper(TransactionCode code, [[maybe_unused]]
-<<<<<<< HEAD
@@ -3860,6 +4075,2 @@ status_t AudioFlinger::onTransactWrapper(TransactionCode code, [[maybe_unused]]
-||||||| 7f0218eb8c
-=======
-        case TransactionCode::RESET_REFERENCES_FOR_TEST:
->>>>>>> fd414a83bb92cf343b57eb7fbab48d56e2702007
-            ALOGW("%s: transaction %d received from PID %d", __func__, static_cast<int>(code),
-                  IPCThreadState::self()->getCallingPid());
+            ALOGW("%s: transaction %d received from PID %d",
+                  __func__, static_cast<int>(code), IPCThreadState::self()->getCallingPid());
@@ -3942 +4153,0 @@ status_t AudioFlinger::onTransactWrapper(TransactionCode code, [[maybe_unused]]
-}
