--- a/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/services@audioflinger@AudioFlinger.no_comments_conflict.cpp
+++ b/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/services@audioflinger@AudioFlinger.no_comments_merged.cpp
@@ -4021 +4020,0 @@ status_t AudioFlinger::getAudioMixPort(const struct audio_port_v7 *devicePort,
-<<<<<<< HEAD
@@ -4043,9 +4041,0 @@ status_t AudioFlinger::resetReferencesForTest() {
-||||||| 7f0218eb8c
-=======
-status_t AudioFlinger::resetReferencesForTest() {
-    mDeviceEffectManager.clear();
-    mPatchPanel.clear();
-    mMelReporter->resetReferencesForTest();
-    return NO_ERROR;
-}
->>>>>>> fd414a83
@@ -4083 +4072,0 @@ status_t AudioFlinger::onTransactWrapper(TransactionCode code,
-<<<<<<< HEAD
@@ -4086,4 +4074,0 @@ status_t AudioFlinger::onTransactWrapper(TransactionCode code,
-||||||| 7f0218eb8c
-=======
-        case TransactionCode::RESET_REFERENCES_FOR_TEST:
->>>>>>> fd414a83
