diff --git a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_mergegen.cc b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_merged.cc
index 4a59000..cc8edac 100644
--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_mergegen.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_merged.cc
@@ -299 +299,2 @@ void grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked(
-grpc_pollset_set* interested_parties, int query_timeout_ms, std::shared_ptr<grpc_core::WorkSerializer> work_serializer) {
+    int query_timeout_ms,
+    std::shared_ptr<grpc_core::WorkSerializer> work_serializer) {
@@ -556 +557,2 @@ static grpc_ares_request* grpc_dns_lookup_ares_locked_impl(
-grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked( r, dns_server, name, default_port, interested_parties, query_timeout_ms, std::move(work_serializer));
+      r, dns_server, name, default_port, interested_parties, query_timeout_ms,
+      std::move(work_serializer));
@@ -622,2 +624 @@ static void on_dns_lookup_done_locked(grpc_resolve_address_ares_request* r,
-  grpc_core::ExecCtx::Run(DEBUG_LOCATION, r->on_resolve_address_done,
-                          GRPC_ERROR_REF(error));
+  grpc_core::ExecCtx::Run(DEBUG_LOCATION, r->on_resolve_address_done, error);
@@ -625 +625,0 @@ static void on_dns_lookup_done_locked(grpc_resolve_address_ares_request* r,
-  GRPC_ERROR_UNREF(error);
@@ -641 +641,3 @@ static void grpc_resolve_address_invoke_dns_lookup_ares_locked(void* arg) {
-r->addresses, nullptr , nullptr , GRPC_DNS_ARES_DEFAULT_QUERY_TIMEOUT_MS, r->work_serializer);
+      &r->on_dns_lookup_done_locked, &r->addresses,
+      nullptr , nullptr ,
+      GRPC_DNS_ARES_DEFAULT_QUERY_TIMEOUT_MS, r->work_serializer);
