diff --git a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_mergegen.cc b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_merged.cc
index 2ef6761..34f35ad 100644
--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_mergegen.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_merged.cc
@@ -260,0 +261 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
+  if (addresses_ != nullptr || balancer_addresses_ != nullptr) {
@@ -262 +263,4 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
-if (r->addresses_ != nullptr) { result.addresses = std::move(*addresses_); } if (service_config_json_ != nullptr) {
+    if (addresses_ != nullptr) {
+      result.addresses = std::move(*addresses_);
+    }
+    if (service_config_json_ != nullptr) {
@@ -274 +278,10 @@ if (r->addresses_ != nullptr) { result.addresses = std::move(*addresses_); } if
- InlinedVector<grpc_arg, 1> new_args; if (r->balancer_addresses_ != nullptr) { new_args.push_back( CreateGrpclbBalancerAddressesArg(r->balancer_addresses_.get())); } result.args = grpc_channel_args_copy_and_add(_, new_args.data(), new_args.size()); grpc_channel_args_); result_handler()->ReturnResult(std::move(result)); addresses_.reset(); r->balancer_addresses_.reset();
+    InlinedVector<grpc_arg, 1> new_args;
+    if (balancer_addresses_ != nullptr) {
+      new_args.push_back(
+          CreateGrpclbBalancerAddressesArg(balancer_addresses_.get()));
+    }
+    result.args = grpc_channel_args_copy_and_add(channel_args_, new_args.data(),
+                                                 new_args.size());
+    result_handler()->ReturnResult(std::move(result));
+    addresses_.reset();
+    balancer_addresses_.reset();
