diff --git a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_conflict.cc b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_merged.cc
index 58d288e..34f35ad 100644
--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_conflict.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@dns_resolver_ares.no_comments_merged.cc
@@ -261,7 +261 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
-<<<<<<< HEAD
-  if (addresses_ != nullptr) {
-||||||| 1e6c9b4c6d
-  if (r->addresses_ != nullptr) {
-=======
-  if (r->addresses_ != nullptr || r->balancer_addresses_ != nullptr) {
->>>>>>> 615cd251
+  if (addresses_ != nullptr || balancer_addresses_ != nullptr) {
@@ -269 +263 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
-<<<<<<< HEAD
+    if (addresses_ != nullptr) {
@@ -270,0 +265 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
+    }
@@ -272,9 +266,0 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
-||||||| 1e6c9b4c6d
-    result.addresses = std::move(*r->addresses_);
-    if (r->service_config_json_ != nullptr) {
-=======
-    if (r->addresses_ != nullptr) {
-      result.addresses = std::move(*r->addresses_);
-    }
-    if (r->service_config_json_ != nullptr) {
->>>>>>> 615cd251
@@ -292,9 +277,0 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
-<<<<<<< HEAD
-    result.args = grpc_channel_args_copy(channel_args_);
-    result_handler()->ReturnResult(std::move(result));
-    addresses_.reset();
-||||||| 1e6c9b4c6d
-    result.args = grpc_channel_args_copy(r->channel_args_);
-    r->result_handler()->ReturnResult(std::move(result));
-    r->addresses_.reset();
-=======
@@ -302 +279 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
-    if (r->balancer_addresses_ != nullptr) {
+    if (balancer_addresses_ != nullptr) {
@@ -304,8 +281,7 @@ void AresDnsResolver::OnResolvedLocked(grpc_error* error) {
-          CreateGrpclbBalancerAddressesArg(r->balancer_addresses_.get()));
-    }
-    result.args = grpc_channel_args_copy_and_add(
-        r->channel_args_, new_args.data(), new_args.size());
-    r->result_handler()->ReturnResult(std::move(result));
-    r->addresses_.reset();
-    r->balancer_addresses_.reset();
->>>>>>> 615cd251
+          CreateGrpclbBalancerAddressesArg(balancer_addresses_.get()));
+    }
+    result.args = grpc_channel_args_copy_and_add(channel_args_, new_args.data(),
+                                                 new_args.size());
+    result_handler()->ReturnResult(std::move(result));
+    addresses_.reset();
+    balancer_addresses_.reset();
