--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_merged.cc
@@ -9 +8,0 @@
-#include "src/core/lib/iomgr/combiner.h"
@@ -65,3 +64,2 @@ static grpc_ares_request* test_dns_lookup_ares_locked(
-    bool check_grpclb, char** service_config_json, int query_timeout_ms,
-    std::shared_ptr<grpc_core::WorkSerializer> work_serializer,
-    grpc_core::Combiner* combiner) {
+    char** service_config_json, int query_timeout_ms,
+    std::shared_ptr<grpc_core::WorkSerializer> work_serializer) {
@@ -70,6 +67,0 @@ static grpc_ares_request* test_dns_lookup_ares_locked(
-<<<<<<< HEAD
-      addresses, check_grpclb, service_config_json, query_timeout_ms,
-      std::move(work_serializer));
-|||||||
-      addresses, check_grpclb, service_config_json, query_timeout_ms, combiner);
-=======
@@ -77,2 +69 @@ static grpc_ares_request* test_dns_lookup_ares_locked(
-      combiner);
->>>>>>> 615cd251b8de723a9ff75ac46db54f5de7bd2ec8
+      std::move(work_serializer));
@@ -149,9 +140,6 @@ class ResultHandler : public grpc_core::Resolver::ResultHandler {
-  struct ResolverOutput {
-    grpc_core::Resolver::Result result;
-    grpc_error* error = nullptr;
-    gpr_event ev;
-    ResolverOutput() { gpr_event_init(&ev); }
-    ~ResolverOutput() { GRPC_ERROR_UNREF(error); }
-  };
-  void SetOutput(ResolverOutput* output) {
-    gpr_atm_rel_store(&output_, reinterpret_cast<gpr_atm>(output));
+  using ResultCallback = void (*)(OnResolutionCallbackArg* state);
+  void SetCallback(ResultCallback result_cb, OnResolutionCallbackArg* state) {
+    GPR_ASSERT(result_cb_ == nullptr);
+    result_cb_ = result_cb;
+    GPR_ASSERT(state_ == nullptr);
+    state_ = state;
@@ -159,7 +147,8 @@ class ResultHandler : public grpc_core::Resolver::ResultHandler {
-  void ReturnResult(grpc_core::Resolver::Result result) override {
-    ResolverOutput* output =
-        reinterpret_cast<ResolverOutput*>(gpr_atm_acq_load(&output_));
-    GPR_ASSERT(output != nullptr);
-    output->result = std::move(result);
-    output->error = GRPC_ERROR_NONE;
-    gpr_event_set(&output->ev, (void*)1);
+  void ReturnResult(grpc_core::Resolver::Result ) override {
+    GPR_ASSERT(result_cb_ != nullptr);
+    GPR_ASSERT(state_ != nullptr);
+    ResultCallback cb = result_cb_;
+    OnResolutionCallbackArg* state = state_;
+    result_cb_ = nullptr;
+    state_ = nullptr;
+    cb(state);
@@ -168,5 +157,2 @@ class ResultHandler : public grpc_core::Resolver::ResultHandler {
-    ResolverOutput* output =
-        reinterpret_cast<ResolverOutput*>(gpr_atm_acq_load(&output_));
-    GPR_ASSERT(output != nullptr);
-    output->error = error;
-    gpr_event_set(&output->ev, (void*)1);
+    gpr_log(GPR_ERROR, "resolver returned error: %s", grpc_error_string(error));
+    GPR_ASSERT(false);
@@ -175 +161,2 @@ class ResultHandler : public grpc_core::Resolver::ResultHandler {
-  gpr_atm output_ = 0;
+  ResultCallback result_cb_ = nullptr;
+  OnResolutionCallbackArg* state_ = nullptr;
@@ -267 +253,0 @@ int main(int argc, char** argv) {
-  gpr_mu_init(&g_mu);
@@ -269,0 +256,3 @@ int main(int argc, char** argv) {
+  g_default_dns_lookup_ares_locked = grpc_dns_lookup_ares_locked;
+  grpc_dns_lookup_ares_locked = test_dns_lookup_ares_locked;
+  default_resolve_address = grpc_resolve_address_impl;
@@ -271,24 +260,4 @@ int main(int argc, char** argv) {
-  grpc_dns_lookup_ares_locked = my_dns_lookup_ares_locked;
-  grpc_cancel_ares_request_locked = my_cancel_ares_request_locked;
-  {
-    grpc_core::ExecCtx exec_ctx;
-    ResultHandler* result_handler = new ResultHandler();
-    grpc_core::OrphanablePtr<grpc_core::Resolver> resolver = create_resolver(
-        "dns:test",
-        std::unique_ptr<grpc_core::Resolver::ResultHandler>(result_handler));
-    ResultHandler::ResolverOutput output1;
-    result_handler->SetOutput(&output1);
-    resolver->StartLocked();
-    grpc_core::ExecCtx::Get()->Flush();
-    GPR_ASSERT(wait_loop(5, &output1.ev));
-    GPR_ASSERT(output1.result.addresses.empty());
-    GPR_ASSERT(output1.error != GRPC_ERROR_NONE);
-    ResultHandler::ResolverOutput output2;
-    result_handler->SetOutput(&output2);
-    grpc_core::ExecCtx::Get()->Flush();
-    GPR_ASSERT(wait_loop(30, &output2.ev));
-    GPR_ASSERT(!output2.result.addresses.empty());
-    GPR_ASSERT(output2.error == GRPC_ERROR_NONE);
-  }
-  grpc_shutdown();
-  gpr_mu_destroy(&g_mu);
+  test_cooldown();
+  grpc_shutdown_blocking();
+  GPR_ASSERT(g_all_callbacks_invoked);
+  return 0;
