diff --git a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_connectivity_test.no_comments_mergegen.cc b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_connectivity_test.no_comments_merged.cc
index 387087a..64d1f90 100644
--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_connectivity_test.no_comments_mergegen.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_connectivity_test.no_comments_merged.cc
@@ -43 +43,3 @@ static grpc_ares_request* my_dns_lookup_ares_locked(
-std::unique_ptr<grpc_core::ServerAddressList>* addresses, std::unique_ptr<grpc_core::ServerAddressList>* , char** , int , std::shared_ptr< grpc_core::WorkSerializer> ) {
+    std::unique_ptr<grpc_core::ServerAddressList>* ,
+    char** , int ,
+    std::shared_ptr<grpc_core::WorkSerializer> ) {
