--- a/./output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_conflict.c
+++ b/./output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_merged.c
@@ -167,7 +166,0 @@ void keysCommand(redisClient *c) {
-<<<<<<< HEAD
-||||||| 2f6b31c3b
-    addReply(c,lenobj);
-    decrRefCount(lenobj);
-=======
-    addReply(c,lenobj);
-    decrRefCount(lenobj);
@@ -175 +167,0 @@ void keysCommand(redisClient *c) {
->>>>>>> 556bdfba
