--- a/home/whalien/codebase/python/mergebot-eva/output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_merged.c
@@ -164,7 +166,0 @@ void keysCommand(redisClient *c) {
-<<<<<<< HEAD
-|||||||
-  addReply(c, lenobj);
-  decrRefCount(lenobj);
-=======
-  addReply(c, lenobj);
-  decrRefCount(lenobj);
@@ -172 +167,0 @@ void keysCommand(redisClient *c) {
->>>>>>> 556bdfbab9062d472e19d882ae045fece36a25ab
