--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/7909a740-0b3dbcef-3881e39c/libs@input@MotionPredictor.no_comments_merged.cpp
@@ -30,4 +30,4 @@ TfLiteMotionPredictorSample::Point convertPrediction(
-() = delete;
-{}
-() = delete;
-{}
+MotionPredictor::MotionPredictor(nsecs_t predictionTimestampOffsetNanos,
+                                 std::function<bool()> checkMotionPredictionEnabled)
+      : mPredictionTimestampOffsetNanos(predictionTimestampOffsetNanos),
+        mCheckMotionPredictionEnabled(std::move(checkMotionPredictionEnabled)) {}
@@ -130 +129,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-<<<<<<< HEAD
@@ -136,11 +134,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-|||||||
-    for (int i = 0; i < predictedR.size() && predictionTime <= futureTime; ++i) {
-        const TfLiteMotionPredictorSample::Point point =
-                convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]);
-=======
-    for (size_t i = 0; i < static_cast<size_t>(predictedR.size()) && predictionTime <= futureTime;
-         ++i) {
-        const TfLiteMotionPredictorSample::Point point =
-                convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]);
->>>>>>> 0b3dbcefb2d37c8c243b93eda498c5599ace25c6
-<<<<<<< HEAD
@@ -150,5 +137,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-|||||||
-        ALOGD_IF(isDebug(), "prediction %d: %f, %f", i, point.x, point.y);
-=======
-        ALOGD_IF(isDebug(), "prediction %zu: %f, %f", i, point.x, point.y);
->>>>>>> 0b3dbcefb2d37c8c243b93eda498c5599ace25c6
