diff --git a/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_mergegen.h b/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_merged.h
index bd779e2..04089fa 100644
--- a/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_mergegen.h
+++ b/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_merged.h
@@ -54 +54,2 @@ private:
-static EGLSurface createPlaceholderEglPbufferSurface( EGLDisplay display, EGLConfig config, int hwcFormat, Protection protection);
+    static EGLSurface createPlaceholderEglPbufferSurface(
+            EGLDisplay display, EGLConfig config, int hwcFormat, Protection protection);
