diff --git a/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_mergegen.h b/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_merged.h
index 33c41d4..fcbf3d6 100644
--- a/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_mergegen.h
+++ b/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_merged.h
@@ -79 +78,0 @@ public:
-using FrameRate = scheduler::LayerInfo::FrameRate;
@@ -239 +238 @@ using FrameRate = scheduler::LayerInfo::FrameRate;
-bool hasRoundedCorners() const { return getRoundedCornerState().hasRoundedCorners();
+    bool hasRoundedCorners() const { return getRoundedCornerState().hasRoundedCorners(); }
