diff --git a/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_conflict.h b/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_merged.h
index 5562cdd..fcbf3d6 100644
--- a/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_conflict.h
+++ b/./output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_merged.h
@@ -79,19 +78,0 @@ public:
-<<<<<<< HEAD
-||||||| b16c01a3e6
-    struct RoundedCornerState {
-        RoundedCornerState() = default;
-        RoundedCornerState(FloatRect cropRect, float radius)
-              : cropRect(cropRect), radius(radius) {}
-        FloatRect cropRect = FloatRect();
-        float radius = 0.0f;
-    };
-=======
-    struct RoundedCornerState {
-        RoundedCornerState() = default;
-        RoundedCornerState(const FloatRect& cropRect, const vec2& radius)
-              : cropRect(cropRect), radius(radius) {}
-        FloatRect cropRect = FloatRect();
-        vec2 radius;
-        bool hasRoundedCorners() const { return radius.x > 0.0f && radius.y > 0.0f; }
-    };
->>>>>>> b22850c1
@@ -257 +237,0 @@ public:
-<<<<<<< HEAD
@@ -259,5 +238,0 @@ public:
-||||||| b16c01a3e6
-    bool hasRoundedCorners() const override { return getRoundedCornerState().radius > .0f; }
-=======
-    bool hasRoundedCorners() const override { return getRoundedCornerState().hasRoundedCorners(); }
->>>>>>> b22850c1
