diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_merged.h
index 1a833f3..fcbf3d6 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_merged.h
@@ -5 +4,0 @@
-#include <gui/ISurfaceComposerClient.h>
@@ -31,2 +29,0 @@
-#include "ClientCache.h"
-#include "DisplayHardware/ComposerHal.h"
@@ -34,2 +30,0 @@
-#include "MonitoredProducer.h"
-#include "RenderArea.h"
@@ -53 +47,0 @@ class SurfaceFlinger;
-class LayerDebugInfo;
@@ -58 +52,3 @@ struct LayerFECompositionState;
-namespace gui {}
+namespace gui {
+class LayerDebugInfo;
+}
@@ -67,2 +62,0 @@ class Layer : public virtual RefBase {
-private:
-    friend class SlotGenerationTest;
@@ -162 +155,0 @@ public:
-    ;
@@ -179 +172 @@ public:
-    virtual bool isDimmingEnabled() const { return getDrawingState().dimmingEnabled; }
+    virtual bool isDimmingEnabled() const { return getDrawingState().dimmingEnabled; };
@@ -191 +183,0 @@ public:
-    ;
@@ -246,4 +238 @@ public:
-    bool hasRoundedCorners() const {
-        return getRoundedCornerState().hasRoundedCorners() {
-            return getRoundedCornerState().hasRoundedCorners();
-        }
+    bool hasRoundedCorners() const { return getRoundedCornerState().hasRoundedCorners(); }
@@ -251,4 +239,0 @@ public:
-        bool hasRoundedCorners() const override {
-            return getRoundedCornerState().hasRoundedCorners() {
-                return getRoundedCornerState().hasRoundedCorners();
-            }
@@ -289,3 +274 @@ public:
-            uint32_t getTransactionFlags() const {
-                return mTransactionFlags;
-            }
+    uint32_t getTransactionFlags() const { return mTransactionFlags; }
@@ -296,8 +279,3 @@ public:
-            void computeBounds(FloatRect parentBounds, ui::Transform parentTransform,
-                               float shadowRadius);
-            int32_t getSequence() const {
-                return sequence;
-            }
-            uint64_t getCurrentBufferId() const {
-                return getBuffer() ? getBuffer()->getId() : 0;
-            }
+    void computeBounds(FloatRect parentBounds, ui::Transform parentTransform, float shadowRadius);
+    int32_t getSequence() const { return sequence; }
+    uint64_t getCurrentBufferId() const { return getBuffer() ? getBuffer()->getId() : 0; }
@@ -315,3 +293 @@ public:
-            gui::WindowInfo::Type getWindowType() const {
-                return mWindowType;
-            }
+    gui::WindowInfo::Type getWindowType() const { return mWindowType; }
@@ -326,6 +302,2 @@ public:
-            inline const State& getDrawingState() const {
-                return mDrawingState;
-            }
-            inline State& getDrawingState() {
-                return mDrawingState;
-            }
+    inline const State& getDrawingState() const { return mDrawingState; }
+    inline State& getDrawingState() { return mDrawingState; }
@@ -345,3 +317 @@ public:
-            bool drawShadows() const {
-                return mEffectiveShadowRadius > 0.f;
-            };
+    bool drawShadows() const { return mEffectiveShadowRadius > 0.f; };
@@ -354,3 +324 @@ public:
-            LayerVector& getCurrentChildren() {
-                return mCurrentChildren;
-            }
+    LayerVector& getCurrentChildren() { return mCurrentChildren; }
@@ -359,12 +327,4 @@ public:
-            sp<Layer> getParent() const {
-                return mCurrentParent.promote();
-            }
-            bool isAtRoot() const {
-                return mIsAtRoot;
-            }
-            void setIsAtRoot(bool isAtRoot) {
-                mIsAtRoot = isAtRoot;
-            }
-            bool hasParent() const {
-                return getParent() != nullptr;
-            }
+    sp<Layer> getParent() const { return mCurrentParent.promote(); }
+    bool isAtRoot() const { return mIsAtRoot; }
+    void setIsAtRoot(bool isAtRoot) { mIsAtRoot = isAtRoot; }
+    bool hasParent() const { return getParent() != nullptr; }
@@ -380,2 +340 @@ public:
-            void setFrameTimelineVsyncForBufferTransaction(const FrameTimelineInfo& info,
-                                                           nsecs_t postTime);
+    void setFrameTimelineVsyncForBufferTransaction(const FrameTimelineInfo& info, nsecs_t postTime);
@@ -384,5 +343,4 @@ public:
-            void addSurfaceFrameDroppedForBuffer(std::shared_ptr<frametimeline::SurfaceFrame> &
-                                                 surfaceFrame);
-            void addSurfaceFramePresentedForBuffer(std::shared_ptr<frametimeline::SurfaceFrame> &
-                                                           surfaceFrame,
-                                                   nsecs_t acquireFenceTime,
+    void addSurfaceFrameDroppedForBuffer(
+            std::shared_ptr<frametimeline::SurfaceFrame>& surfaceFrame);
+    void addSurfaceFramePresentedForBuffer(
+            std::shared_ptr<frametimeline::SurfaceFrame>& surfaceFrame, nsecs_t acquireFenceTime,
@@ -390,5 +348,4 @@ public:
-            std::shared_ptr<frametimeline::SurfaceFrame>
-            createSurfaceFrameForTransaction(const FrameTimelineInfo& info, nsecs_t postTime);
-            std::shared_ptr<frametimeline::SurfaceFrame>
-            createSurfaceFrameForBuffer(const FrameTimelineInfo& info, nsecs_t queueTime,
-                                        std::string debugName);
+    std::shared_ptr<frametimeline::SurfaceFrame> createSurfaceFrameForTransaction(
+            const FrameTimelineInfo& info, nsecs_t postTime);
+    std::shared_ptr<frametimeline::SurfaceFrame> createSurfaceFrameForBuffer(
+            const FrameTimelineInfo& info, nsecs_t queueTime, std::string debugName);
@@ -396,3 +353 @@ public:
-            const std::string& getName() const {
-                return mName;
-            }
+    const std::string& getName() const { return mName; }
@@ -408,12 +363,4 @@ public:
-            void setGameMode(GameMode gameMode) {
-                mGameMode = gameMode;
-            }
-            GameMode getGameMode() const {
-                return mGameMode;
-            }
-            virtual uid_t getOwnerUid() const {
-                return mOwnerUid;
-            }
-            pid_t getOwnerPid() {
-                return mOwnerPid;
-            }
+    void setGameMode(GameMode gameMode) { mGameMode = gameMode; }
+    GameMode getGameMode() const { return mGameMode; }
+    virtual uid_t getOwnerUid() const { return mOwnerUid; }
+    pid_t getOwnerPid() { return mOwnerPid; }
@@ -439,6 +386,2 @@ public:
-            std::atomic<int32_t>* getPendingBufferCounter() {
-                return &mPendingBufferTransactions;
-            }
-            std::string getPendingBufferCounterName() {
-                return mBlastTransactionName;
-            }
+    std::atomic<int32_t>* getPendingBufferCounter() { return &mPendingBufferTransactions; }
+    std::string getPendingBufferCounterName() { return mBlastTransactionName; }
@@ -459,9 +401,0 @@ public:
-        private:
-            bool getAutoRefresh() const {
-                return mDrawingState.autoRefresh;
-            }
-            bool getSidebandStreamChanged() const {
-                return mSidebandStreamChanged;
-            }
-            std::atomic<bool> mSidebandStreamChanged{false};
-        protected:
@@ -475,9 +409,3 @@ public:
-            sp<Layer> getClonedFrom() {
-                return mClonedFrom != nullptr ? mClonedFrom.promote() : nullptr;
-            }
-            bool isClone() {
-                return mClonedFrom != nullptr;
-            }
-            bool isClonedFromAlive() {
-                return getClonedFrom() != nullptr;
-            }
+    sp<Layer> getClonedFrom() { return mClonedFrom != nullptr ? mClonedFrom.promote() : nullptr; }
+    bool isClone() { return mClonedFrom != nullptr; }
+    bool isClonedFromAlive() { return getClonedFrom() != nullptr; }
@@ -495,2 +423 @@ public:
-            void setupRoundedCornersCropCoordinates(Rect win, const FloatRect& roundedCornersCrop)
-                    const;
+    void setupRoundedCornersCropCoordinates(Rect win, const FloatRect& roundedCornersCrop) const;
@@ -534,3 +461,7 @@ public:
-            virtual bool canDrawShadows() const {
-                return true;
-            }
+    friend class SlotGenerationTest;
+    friend class TransactionFrameTracerTest;
+    friend class TransactionSurfaceFrameTest;
+    bool getAutoRefresh() const { return mDrawingState.autoRefresh; }
+    bool getSidebandStreamChanged() const { return mSidebandStreamChanged; }
+    std::atomic<bool> mSidebandStreamChanged{false};
+    virtual bool canDrawShadows() const { return true; }
@@ -568,3 +499 @@ public:
-            bool hasEffect() const {
-                return fillsColor() || drawShadows() || hasBlur();
-            }
+    bool hasEffect() const { return fillsColor() || drawShadows() || hasBlur(); }
@@ -575,5 +504 @@ public:
-                return ((mDrawingState.sidebandStream != nullptr) ||
-                        (mDrawingState.buffer != nullptr));
-            }
-            bool hasSomethingToDraw() const {
-                return hasEffect() || hasBufferOrSidebandStream();
+        return ((mDrawingState.sidebandStream != nullptr) || (mDrawingState.buffer != nullptr));
@@ -580,0 +506 @@ public:
+    bool hasSomethingToDraw() const { return hasEffect() || hasBufferOrSidebandStream(); }
@@ -606,2 +532 @@ public:
-            std::deque<std::shared_ptr<android::frametimeline::SurfaceFrame>>
-                    mPendingJankClassifications;
+    std::deque<std::shared_ptr<android::frametimeline::SurfaceFrame>> mPendingJankClassifications;
@@ -622 +547 @@ public:
-            LayerSnapshotGuard& operator=(const LayerSnapshotGuard&) = delete
+    LayerSnapshotGuard& operator=(const LayerSnapshotGuard&) = delete;
