diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_mergebot.hpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_merged.hpp
index 8008e73..5142084 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_merged.hpp
@@ -6,0 +7 @@ public:
+public:
@@ -18,0 +20 @@ public:
+public:
@@ -22,0 +25 @@ public:
+public:
