diff --git a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_mergegen.cpp b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_merged.cpp
index 3ba1c76..272fa14 100644
--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_mergegen.cpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_merged.cpp
@@ -22 +22,4 @@ public:
- SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, DataChunk &chunk, OperatorSinkInput &input) const { auto &state = input.local_state.Cast<BatchCollectorLocalState>(); state.data.Append(chunk, state.partition_info.batch_index.GetIndex());
+SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, DataChunk &chunk,
+                                            OperatorSinkInput &input) const {
+ auto &state = input.local_state.Cast<BatchCollectorLocalState>();
+ state.data.Append(chunk, state.partition_info.batch_index.GetIndex());
