--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_conflict.cpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_merged.cpp
@@ -22,11 +21,0 @@ public:
-<<<<<<< HEAD
-SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, GlobalSinkState &gstate,
-                                            LocalSinkState &lstate_p, DataChunk &input) const {
- auto &state = lstate_p.Cast<BatchCollectorLocalState>();
- state.data.Append(input, state.partition_info.batch_index.GetIndex());
-||||||| fc797c18cf
-SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, GlobalSinkState &gstate,
-                                            LocalSinkState &lstate_p, DataChunk &input) const {
- auto &state = lstate_p.Cast<BatchCollectorLocalState>();
- state.data.Append(input, state.batch_index);
-=======
@@ -36,2 +25 @@ SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, DataChunk
- state.data.Append(chunk, state.batch_index);
->>>>>>> 6255d307
+ state.data.Append(chunk, state.partition_info.batch_index.GetIndex());
