--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/921991c9-0084cbef-63b6eec2/arguments.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/921991c9-0084cbef-63b6eec2/arguments.no_comments_merged.c
@@ -4 +3,0 @@
-#include <vis.h>
@@ -7,0 +7 @@
+TAILQ_HEAD(args_values, args_value);
@@ -21 +21,4 @@ static int args_cmp(struct args_entry *, struct args_entry *);
-static int args_cmp(struct args_entry *a1, struct args_entry *a2) {
+RB_GENERATE_STATIC(args_tree, args_entry, entry, args_cmp);
+static int
+args_cmp(struct args_entry *a1, struct args_entry *a2)
+{
