--- a/./output/tmux/bc2e0cf7-392b381d-6f03e49e/server-client.no_comments_conflict.c
+++ b/./output/tmux/bc2e0cf7-392b381d-6f03e49e/server-client.no_comments_merged.c
@@ -1759 +1758,0 @@ server_client_dispatch_identify(struct client *c, struct imsg *imsg)
-<<<<<<< HEAD
@@ -1763,8 +1761,0 @@ server_client_dispatch_identify(struct client *c, struct imsg *imsg)
- if (c->flags & CLIENT_CONTROL) {
-  close(c->fd);
-  c->fd = -1;
-||||||| 6f03e49e
- if (c->flags & CLIENT_CONTROL) {
-  close(c->fd);
-  c->fd = -1;
-=======
@@ -1772 +1762,0 @@ server_client_dispatch_identify(struct client *c, struct imsg *imsg)
->>>>>>> 392b381d
