--- a/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/media@libaudioclient@tests@audio_aidl_legacy_conversion_tests.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/media@libaudioclient@tests@audio_aidl_legacy_conversion_tests.no_comments_merged.cpp
@@ -352 +352,41 @@ INSTANTIATE_TEST_SUITE_P(AudioDeviceDescriptionRoundTrip, AudioDeviceDescription
-class AudioDeviceRoundTripTest : public testing::TestWithParam<AudioDevice> {}; TEST_P(AudioDeviceRoundTripTest, Aidl2Legacy2Aidl) { const auto initial = GetParam(); audio_devices_t legacyType; String8 legacyAddress; status_t status = aidl2legacy_AudioDevice_audio_device(initial, &legacyType, &legacyAddress); ASSERT_EQ(OK, status); auto convBack = legacy2aidl_audio_device_AudioDevice(legacyType, legacyAddress); ASSERT_TRUE(convBack.ok()); EXPECT_EQ(initial, convBack.value()); } INSTANTIATE_TEST_SUITE_P( AudioDeviceRoundTrip, AudioDeviceRoundTripTest, testing::Values( make_AudioDevice(make_ADD_MicIn(), AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("bottom")), make_AudioDevice(make_ADD_RSubmixIn(), AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("")), make_AudioDevice(make_ADD_BtScoHeadset(), AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("")), make_AudioDevice(make_ADD_BtScoHeadset(), AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("1:2-in-3")),
+class AudioDeviceRoundTripTest : public testing::TestWithParam<AudioDevice> {};
+TEST_P(AudioDeviceRoundTripTest, Aidl2Legacy2Aidl) {
+    const auto initial = GetParam();
+    audio_devices_t legacyType;
+    String8 legacyAddress;
+    status_t status = aidl2legacy_AudioDevice_audio_device(initial, &legacyType, &legacyAddress);
+    ASSERT_EQ(OK, status);
+    auto convBack = legacy2aidl_audio_device_AudioDevice(legacyType, legacyAddress);
+    ASSERT_TRUE(convBack.ok());
+    EXPECT_EQ(initial, convBack.value());
+}
+INSTANTIATE_TEST_SUITE_P(
+        AudioDeviceRoundTrip, AudioDeviceRoundTripTest,
+        testing::Values(
+                make_AudioDevice(make_ADD_MicIn(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("bottom")),
+                make_AudioDevice(make_ADD_RSubmixIn(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("1:2-in-3")),
+                make_AudioDevice(make_ADD_BtScoHeadset(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("")),
+                make_AudioDevice(make_ADD_BtScoHeadset(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::mac>(
+                                         std::vector<uint8_t>{1, 2, 3, 4, 5, 6})),
+                make_AudioDevice(make_ADD_BtA2dpHeadphone(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("")),
+                make_AudioDevice(make_ADD_BtA2dpHeadphone(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::mac>(
+                                         std::vector<uint8_t>{1, 2, 3, 4, 5, 6})),
+                make_AudioDevice(make_ADD_BtLeHeadset(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::mac>(
+                                         std::vector<uint8_t>{1, 2, 3, 4, 5, 6})),
+                make_AudioDevice(make_ADD_BtLeBroadcast(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::id>("42")),
+                make_AudioDevice(make_ADD_IpV4Device(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::ipv4>(
+                                         std::vector<uint8_t>{192, 168, 0, 1})),
+                make_AudioDevice(make_ADD_UsbHeadset(),
+                                 AudioDeviceAddress::make<AudioDeviceAddress::Tag::alsa>(
+                                         std::vector<int32_t>{1, 2}))));
+class AudioFormatDescriptionRoundTripTest : public testing::TestWithParam<AudioFormatDescription> {
+};
