diff --git a/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/media@libaudioclient@tests@audio_aidl_legacy_conversion_tests.no_comments_conflict.cpp b/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/media@libaudioclient@tests@audio_aidl_legacy_conversion_tests.no_comments_merged.cpp
index 96f6959..c5213b6 100644
--- a/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/media@libaudioclient@tests@audio_aidl_legacy_conversion_tests.no_comments_conflict.cpp
+++ b/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/media@libaudioclient@tests@audio_aidl_legacy_conversion_tests.no_comments_merged.cpp
@@ -352 +351,0 @@ INSTANTIATE_TEST_SUITE_P(AudioDeviceDescriptionRoundTrip, AudioDeviceDescription
-<<<<<<< HEAD
@@ -394,7 +392,0 @@ class AudioFormatDescriptionRoundTripTest : public testing::TestWithParam<AudioF
-||||||| 0f5b5f3279
-class AudioFormatDescriptionRoundTripTest :
-        public testing::TestWithParam<AudioFormatDescription> {};
-=======
-class AudioFormatDescriptionRoundTripTest :
-        public testing::TestWithParam<AudioFormatDescription> {};
->>>>>>> aa7f1e5e
