--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@IOProfile.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@IOProfile.no_comments_merged.h
@@ -2 +1,0 @@
-#include "AudioPort.h"
@@ -13,2 +13 @@ class IOProfile : public AudioPort, public PolicyAudioPort {
-  IOProfile(const String8 &name, const std::string &name,
-            audio_port_role_t role)
+    IOProfile(const std::string &name, audio_port_role_t role)
@@ -20 +19 @@ class IOProfile : public AudioPort, public PolicyAudioPort {
-  ~IOProfile() = default;
+    virtual ~IOProfile() = default;
