diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
index dbb2ee2..53845cb 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
@@ -12,2 +12 @@
-    ({ \
-        auto _tmp = (x); \
+    ({ auto _tmp = (x); \
@@ -15,2 +14 @@
-        std::move(_tmp.value()); \
-    })
+       std::move(_tmp.value()); })
@@ -26 +23,0 @@ namespace android {
-using aidl_utils::binderStatusFromStatusT;
@@ -27,0 +25 @@ using binder::Status;
+using aidl_utils::binderStatusFromStatusT;
@@ -43,2 +41,6 @@ const std::vector<audio_usage_t>& SYSTEM_USAGES = {
-        AUDIO_USAGE_CALL_ASSISTANT, AUDIO_USAGE_EMERGENCY, AUDIO_USAGE_SAFETY,
-        AUDIO_USAGE_VEHICLE_STATUS, AUDIO_USAGE_ANNOUNCEMENT};
+    AUDIO_USAGE_CALL_ASSISTANT,
+    AUDIO_USAGE_EMERGENCY,
+    AUDIO_USAGE_SAFETY,
+    AUDIO_USAGE_VEHICLE_STATUS,
+    AUDIO_USAGE_ANNOUNCEMENT
+};
@@ -46,2 +48,2 @@ bool isSystemUsage(audio_usage_t usage) {
-    return std::find(std::begin(SYSTEM_USAGES), std::end(SYSTEM_USAGES), usage) !=
-           std::end(SYSTEM_USAGES);
+    return std::find(std::begin(SYSTEM_USAGES), std::end(SYSTEM_USAGES), usage)
+        != std::end(SYSTEM_USAGES);
@@ -50,2 +52,2 @@ bool AudioPolicyService::isSupportedSystemUsage(audio_usage_t usage) {
-    return std::find(std::begin(mSupportedSystemUsages), std::end(mSupportedSystemUsages), usage) !=
-           std::end(mSupportedSystemUsages);
+    return std::find(std::begin(mSupportedSystemUsages), std::end(mSupportedSystemUsages), usage)
+        != std::end(mSupportedSystemUsages);
@@ -60,2 +62,2 @@ status_t AudioPolicyService::validateUsage(const audio_attributes_t& attr,
-            if (attr.usage == AUDIO_USAGE_CALL_ASSISTANT &&
-                ((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0)) {
+            if (attr.usage == AUDIO_USAGE_CALL_ASSISTANT
+                    && ((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0)) {
@@ -68,2 +70,2 @@ status_t AudioPolicyService::validateUsage(const audio_attributes_t& attr,
-                ALOGE("%s: modify audio routing not allowed for attribution source: %s", __func__,
-                      attributionSource.toString().c_str());
+                ALOGE("%s: modify audio routing not allowed for attribution source: %s",
+                        __func__, attributionSource.toString().c_str());
@@ -78 +80,2 @@ status_t AudioPolicyService::validateUsage(const audio_attributes_t& attr,
-void AudioPolicyService::doOnNewAudioModulesAvailable() {
+void AudioPolicyService::doOnNewAudioModulesAvailable()
+{
@@ -105 +108,2 @@ Status AudioPolicyService::setDeviceConnectionState(
-    status_t status = mAudioPolicyManager->setDeviceConnectionState(state, port, encodedFormat);
+    status_t status = mAudioPolicyManager->setDeviceConnectionState(
+            state, port, encodedFormat);
@@ -127 +131,2 @@ Status AudioPolicyService::getDeviceConnectionState(const AudioDevice& deviceAid
-                    mAudioPolicyManager->getDeviceConnectionState(device, address.c_str())));
+                    mAudioPolicyManager->getDeviceConnectionState(
+                            device, address.c_str())));
@@ -131 +136,2 @@ Status AudioPolicyService::handleDeviceConfigChange(
-        const AudioDevice& deviceAidl, const std::string& deviceNameAidl,
+        const AudioDevice& deviceAidl,
+        const std::string& deviceNameAidl,
@@ -155,3 +161,4 @@ Status AudioPolicyService::handleDeviceConfigChange(
-Status AudioPolicyService::setPhoneState(AudioMode stateAidl, int32_t uidAidl) {
-    audio_mode_t state =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioMode_audio_mode_t(stateAidl));
+Status AudioPolicyService::setPhoneState(AudioMode stateAidl, int32_t uidAidl)
+{
+    audio_mode_t state = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioMode_audio_mode_t(stateAidl));
@@ -190 +197,2 @@ Status AudioPolicyService::setForceUse(media::AudioPolicyForceUse usageAidl,
-                                       media::AudioPolicyForcedConfig configAidl) {
+                                       media::AudioPolicyForcedConfig configAidl)
+{
@@ -223,2 +231 @@ Status AudioPolicyService::getForceUse(media::AudioPolicyForceUse usageAidl,
-                legacy2aidl_audio_policy_forced_cfg_t_AudioPolicyForcedConfig(
-                        AUDIO_POLICY_FORCE_NONE));
+            legacy2aidl_audio_policy_forced_cfg_t_AudioPolicyForcedConfig(AUDIO_POLICY_FORCE_NONE));
@@ -233 +240,2 @@ Status AudioPolicyService::getForceUse(media::AudioPolicyForceUse usageAidl,
-Status AudioPolicyService::getOutput(AudioStreamType streamAidl, int32_t* _aidl_return) {
+Status AudioPolicyService::getOutput(AudioStreamType streamAidl, int32_t* _aidl_return)
+{
@@ -236,2 +244,2 @@ Status AudioPolicyService::getOutput(AudioStreamType streamAidl, int32_t* _aidl_
-    if (uint32_t(stream) >= AUDIO_STREAM_PUBLIC_CNT && stream != AUDIO_STREAM_ASSISTANT &&
-        stream != AUDIO_STREAM_CALL_ASSISTANT) {
+    if (uint32_t(stream) >= AUDIO_STREAM_PUBLIC_CNT
+          && stream != AUDIO_STREAM_ASSISTANT && stream != AUDIO_STREAM_CALL_ASSISTANT) {
@@ -255 +263,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-                                            const AudioConfig& configAidl, int32_t flagsAidl,
+                                            const AudioConfig& configAidl,
+                                            int32_t flagsAidl,
@@ -257 +266,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-                                            media::GetOutputForAttrResponse* _aidl_return) {
+                                            media::GetOutputForAttrResponse* _aidl_return)
+{
@@ -260,2 +270,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-    audio_session_t session =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_session_t(sessionAidl));
+    audio_session_t session = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_session_t(sessionAidl));
@@ -265,2 +275,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-    audio_output_flags_t flags =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_output_flags_t_mask(flagsAidl));
+    audio_output_flags_t flags = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_output_flags_t_mask(flagsAidl));
@@ -280,2 +290,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-    if (!mPackageManager.allowPlaybackCapture(
-                VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_uid_t(attributionSource.uid)))) {
+    if (!mPackageManager.allowPlaybackCapture(VALUE_OR_RETURN_BINDER_STATUS(
+        aidl2legacy_int32_t_uid_t(attributionSource.uid)))) {
@@ -284,2 +294,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-    if (((attr.flags & (AUDIO_FLAG_BYPASS_INTERRUPTION_POLICY | AUDIO_FLAG_BYPASS_MUTE)) != 0) &&
-        !bypassInterruptionPolicyAllowed(attributionSource)) {
+    if (((attr.flags & (AUDIO_FLAG_BYPASS_INTERRUPTION_POLICY|AUDIO_FLAG_BYPASS_MUTE)) != 0)
+            && !bypassInterruptionPolicyAllowed(attributionSource)) {
@@ -291,2 +301,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-            ALOGE("%s: permission denied: ultrasound not allowed for uid %d pid %d", __func__,
-                  attributionSource.uid, attributionSource.pid);
+            ALOGE("%s: permission denied: ultrasound not allowed for uid %d pid %d",
+                    __func__, attributionSource.uid, attributionSource.pid);
@@ -300,3 +310,9 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-    status_t result = mAudioPolicyManager->getOutputForAttr(
-            &attr, &output, session, &stream, attributionSource, &config, &flags, &selectedDeviceId,
-            &portId, &secondaryOutputs, &outputType, &isSpatialized, &isBitPerfect);
+    status_t result = mAudioPolicyManager->getOutputForAttr(&attr, &output, session,
+                                                            &stream,
+                                                            attributionSource,
+                                                            &config,
+                                                            &flags, &selectedDeviceId, &portId,
+                                                            &secondaryOutputs,
+                                                            &outputType,
+                                                            &isSpatialized,
+                                                            &isBitPerfect);
@@ -308,2 +324,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-                if (((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0) &&
-                    !callAudioInterceptionAllowed(attributionSource)) {
+            if (((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0)
+                && !callAudioInterceptionAllowed(attributionSource)) {
@@ -328,2 +344,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-                LOG_ALWAYS_FATAL("%s() encountered an invalid output type %d", __func__,
-                                 (int)outputType);
+            LOG_ALWAYS_FATAL("%s() encountered an invalid output type %d",
+                __func__, (int)outputType);
@@ -336,2 +352,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-                new AudioPlaybackClient(attr, output, attributionSource, session, portId,
-                                        selectedDeviceId, stream, isSpatialized);
+                new AudioPlaybackClient(attr, output, attributionSource, session,
+                    portId, selectedDeviceId, stream, isSpatialized);
@@ -339,2 +355,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-        _aidl_return->output =
-                VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_io_handle_t_int32_t(output));
+        _aidl_return->output = VALUE_OR_RETURN_BINDER_STATUS(
+                legacy2aidl_audio_io_handle_t_int32_t(output));
@@ -345,5 +361,5 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-        _aidl_return->portId =
-                VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_handle_t_int32_t(portId));
-        _aidl_return->secondaryOutputs =
-                VALUE_OR_RETURN_BINDER_STATUS(convertContainer<std::vector<int32_t>>(
-                        secondaryOutputs, legacy2aidl_audio_io_handle_t_int32_t));
+        _aidl_return->portId = VALUE_OR_RETURN_BINDER_STATUS(
+                legacy2aidl_audio_port_handle_t_int32_t(portId));
+        _aidl_return->secondaryOutputs = VALUE_OR_RETURN_BINDER_STATUS(
+                convertContainer<std::vector<int32_t>>(secondaryOutputs,
+                                                       legacy2aidl_audio_io_handle_t_int32_t));
@@ -357,2 +373,2 @@ Status AudioPolicyService::getOutputForAttr(const media::AudioAttributesInternal
-        _aidl_return->configBase.channelMask =
-                VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_channel_mask_t_AudioChannelLayout(
+        _aidl_return->configBase.channelMask = VALUE_OR_RETURN_BINDER_STATUS(
+                legacy2aidl_audio_channel_mask_t_AudioChannelLayout(
@@ -367 +383,2 @@ void AudioPolicyService::getPlaybackClientAndEffects(audio_port_handle_t portId,
-                                                     const char* context) {
+                                                     const char *context)
+{
@@ -377,3 +394,4 @@ void AudioPolicyService::getPlaybackClientAndEffects(audio_port_handle_t portId,
-Status AudioPolicyService::startOutput(int32_t portIdAidl) {
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+Status AudioPolicyService::startOutput(int32_t portIdAidl)
+{
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -405,3 +423,4 @@ Status AudioPolicyService::startOutput(int32_t portIdAidl) {
-Status AudioPolicyService::stopOutput(int32_t portIdAidl) {
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+Status AudioPolicyService::stopOutput(int32_t portIdAidl)
+{
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -415 +434,2 @@ Status AudioPolicyService::stopOutput(int32_t portIdAidl) {
-status_t AudioPolicyService::doStopOutput(audio_port_handle_t portId) {
+status_t AudioPolicyService::doStopOutput(audio_port_handle_t portId)
+{
@@ -437,3 +457,4 @@ status_t AudioPolicyService::doStopOutput(audio_port_handle_t portId) {
-Status AudioPolicyService::releaseOutput(int32_t portIdAidl) {
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+Status AudioPolicyService::releaseOutput(int32_t portIdAidl)
+{
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -447 +468,2 @@ Status AudioPolicyService::releaseOutput(int32_t portIdAidl) {
-void AudioPolicyService::doReleaseOutput(audio_port_handle_t portId) {
+void AudioPolicyService::doReleaseOutput(audio_port_handle_t portId)
+{
@@ -453,2 +475,2 @@ void AudioPolicyService::doReleaseOutput(audio_port_handle_t portId) {
-        audioPolicyEffects->releaseOutputSessionEffects(client->io, client->stream,
-                                                        client->session);
+        audioPolicyEffects->releaseOutputSessionEffects(
+            client->io, client->stream, client->session);
@@ -464 +486,3 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-                                           int32_t inputAidl, int32_t riidAidl, int32_t sessionAidl,
+                                           int32_t inputAidl,
+                                           int32_t riidAidl,
+                                           int32_t sessionAidl,
@@ -466 +490,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-                                           const AudioConfigBase& configAidl, int32_t flagsAidl,
+                                           const AudioConfigBase& configAidl,
+                                           int32_t flagsAidl,
@@ -471,6 +496,6 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    audio_io_handle_t input =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_io_handle_t(inputAidl));
-    audio_unique_id_t riid =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_unique_id_t(riidAidl));
-    audio_session_t session =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_session_t(sessionAidl));
+    audio_io_handle_t input = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_io_handle_t(inputAidl));
+    audio_unique_id_t riid = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_unique_id_t(riidAidl));
+    audio_session_t session = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_session_t(sessionAidl));
@@ -479,2 +504,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    audio_input_flags_t flags =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_input_flags_t_mask(flagsAidl));
+    audio_input_flags_t flags = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_input_flags_t_mask(flagsAidl));
@@ -493,4 +518,6 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    if ((inputSource < AUDIO_SOURCE_DEFAULT) ||
-        (inputSource >= AUDIO_SOURCE_CNT && inputSource != AUDIO_SOURCE_HOTWORD &&
-         inputSource != AUDIO_SOURCE_FM_TUNER && inputSource != AUDIO_SOURCE_ECHO_REFERENCE &&
-         inputSource != AUDIO_SOURCE_ULTRASOUND)) {
+    if ((inputSource < AUDIO_SOURCE_DEFAULT)
+            || (inputSource >= AUDIO_SOURCE_CNT
+                && inputSource != AUDIO_SOURCE_HOTWORD
+                && inputSource != AUDIO_SOURCE_FM_TUNER
+                && inputSource != AUDIO_SOURCE_ECHO_REFERENCE
+                && inputSource != AUDIO_SOURCE_ULTRASOUND)) {
@@ -499,6 +526,8 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(validateUsage(attr, attributionSource)));
-    if (!(recordingAllowed(attributionSource, inputSource) ||
-          inputSource == AUDIO_SOURCE_FM_TUNER || inputSource == AUDIO_SOURCE_REMOTE_SUBMIX ||
-          inputSource == AUDIO_SOURCE_ECHO_REFERENCE)) {
-        ALOGE("%s permission denied: recording not allowed for %s", __func__,
-              attributionSource.toString().c_str());
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(validateUsage(attr,
+            attributionSource)));
+    if (!(recordingAllowed(attributionSource, inputSource)
+            || inputSource == AUDIO_SOURCE_FM_TUNER
+            || inputSource == AUDIO_SOURCE_REMOTE_SUBMIX
+            || inputSource == AUDIO_SOURCE_ECHO_REFERENCE)) {
+        ALOGE("%s permission denied: recording not allowed for %s",
+                __func__, attributionSource.toString().c_str());
@@ -509,3 +538,3 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    bool isCallAudioSource = inputSource == AUDIO_SOURCE_VOICE_UPLINK ||
-                             inputSource == AUDIO_SOURCE_VOICE_DOWNLINK ||
-                             inputSource == AUDIO_SOURCE_VOICE_CALL;
+    bool isCallAudioSource = inputSource == AUDIO_SOURCE_VOICE_UPLINK
+             || inputSource == AUDIO_SOURCE_VOICE_DOWNLINK
+             || inputSource == AUDIO_SOURCE_VOICE_CALL;
@@ -515 +544,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    if (inputSource == AUDIO_SOURCE_ECHO_REFERENCE && !canCaptureOutput) {
+    if (inputSource == AUDIO_SOURCE_ECHO_REFERENCE
+            && !canCaptureOutput) {
@@ -518,2 +548,3 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    if (inputSource == AUDIO_SOURCE_FM_TUNER && !canCaptureOutput &&
-        !captureTunerAudioInputAllowed(attributionSource)) {
+    if (inputSource == AUDIO_SOURCE_FM_TUNER
+        && !canCaptureOutput
+        && !captureTunerAudioInputAllowed(attributionSource)) {
@@ -526,3 +557,4 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    if (((flags & (AUDIO_INPUT_FLAG_HW_HOTWORD | AUDIO_INPUT_FLAG_HOTWORD_TAP |
-                   AUDIO_INPUT_FLAG_HW_LOOKBACK)) != 0) &&
-        !canCaptureHotword) {
+    if (((flags & (AUDIO_INPUT_FLAG_HW_HOTWORD |
+                        AUDIO_INPUT_FLAG_HOTWORD_TAP |
+                        AUDIO_INPUT_FLAG_HW_LOOKBACK)) != 0)
+            && !canCaptureHotword) {
@@ -530,2 +562 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-              " for uid %d pid %d",
-              __func__, attributionSource.uid, attributionSource.pid);
+              " for uid %d pid %d", __func__, attributionSource.uid, attributionSource.pid);
@@ -536,2 +567,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-            ALOGE("%s: permission denied: ultrasound not allowed for uid %d pid %d", __func__,
-                  attributionSource.uid, attributionSource.pid);
+            ALOGE("%s: permission denied: ultrasound not allowed for uid %d pid %d",
+                    __func__, attributionSource.uid, attributionSource.pid);
@@ -549,2 +580,3 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-                                                          attributionSource, &config, flags,
-                                                          &selectedDeviceId, &inputType, &portId);
+                                                          attributionSource, &config,
+                                                          flags, &selectedDeviceId,
+                                                          &inputType, &portId);
@@ -559 +591,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-                    if ((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0 && canInterceptCallAudio) {
+                if ((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0
+                        && canInterceptCallAudio) {
@@ -570,3 +603,3 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-                    if (!(modifyAudioRoutingAllowed(attributionSource) ||
-                          ((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0 &&
-                           canInterceptCallAudio))) {
+                if (!(modifyAudioRoutingAllowed(attributionSource)
+                        || ((attr.flags & AUDIO_FLAG_CALL_REDIRECTION) != 0
+                            && canInterceptCallAudio))) {
@@ -579,2 +612,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-                    LOG_ALWAYS_FATAL("%s encountered an invalid input type %d", __func__,
-                                     (int)inputType);
+                LOG_ALWAYS_FATAL("%s encountered an invalid input type %d",
+                        __func__, (int)inputType);
@@ -593,3 +626,4 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-        sp<AudioRecordClient> client = new AudioRecordClient(
-                attr, input, session, portId, selectedDeviceId, attributionSource, canCaptureOutput,
-                canCaptureHotword, mOutputCommandThread);
+        sp<AudioRecordClient> client = new AudioRecordClient(attr, input, session, portId,
+                                                             selectedDeviceId, attributionSource,
+                                                             canCaptureOutput, canCaptureHotword,
+                                                             mOutputCommandThread);
@@ -604,2 +638,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    _aidl_return->input =
-            VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_io_handle_t_int32_t(input));
+    _aidl_return->input = VALUE_OR_RETURN_BINDER_STATUS(
+            legacy2aidl_audio_io_handle_t_int32_t(input));
@@ -608,2 +642,2 @@ Status AudioPolicyService::getInputForAttr(const media::AudioAttributesInternal&
-    _aidl_return->portId =
-            VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_handle_t_int32_t(portId));
+    _aidl_return->portId = VALUE_OR_RETURN_BINDER_STATUS(
+            legacy2aidl_audio_port_handle_t_int32_t(portId));
@@ -621,3 +655,4 @@ std::string AudioPolicyService::getDeviceTypeStrForPortId(audio_port_handle_t po
-Status AudioPolicyService::startInput(int32_t portIdAidl) {
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+Status AudioPolicyService::startInput(int32_t portIdAidl)
+{
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -639,6 +674,6 @@ Status AudioPolicyService::startInput(int32_t portIdAidl) {
-                         client->attributes.source) ||
-          client->attributes.source == AUDIO_SOURCE_FM_TUNER ||
-          client->attributes.source == AUDIO_SOURCE_REMOTE_SUBMIX ||
-          client->attributes.source == AUDIO_SOURCE_ECHO_REFERENCE)) {
-        ALOGE("%s permission denied: recording not allowed for attribution source %s", __func__,
-              client->attributionSource.toString().c_str());
+                         client->attributes.source)
+            || client->attributes.source == AUDIO_SOURCE_FM_TUNER
+            || client->attributes.source == AUDIO_SOURCE_REMOTE_SUBMIX
+            || client->attributes.source == AUDIO_SOURCE_ECHO_REFERENCE)) {
+        ALOGE("%s permission denied: recording not allowed for attribution source %s",
+                __func__, client->attributionSource.toString().c_str());
@@ -649 +684,2 @@ Status AudioPolicyService::startInput(int32_t portIdAidl) {
-             portIdAidl, client->attributionSource.uid);
+            portIdAidl,
+            client->attributionSource.uid);
@@ -671 +707,2 @@ Status AudioPolicyService::startInput(int32_t portIdAidl) {
-        static constexpr char kAudioPolicyRqstDevice[] = "android.media.audiopolicy.rqst.device";
+        static constexpr char kAudioPolicyRqstDevice[] =
+                "android.media.audiopolicy.rqst.device";
@@ -681 +718,2 @@ Status AudioPolicyService::startInput(int32_t portIdAidl) {
-            item->setCString(kAudioPolicyRqstSrc, toString(client->attributes.source).c_str());
+            item->setCString(kAudioPolicyRqstSrc,
+                             toString(client->attributes.source).c_str());
@@ -691,2 +729,2 @@ Status AudioPolicyService::startInput(int32_t portIdAidl) {
-            item->setCString(kAudioPolicyRqstDevice,
-                             getDeviceTypeStrForPortId(client->deviceId).c_str());
+            item->setCString(
+                    kAudioPolicyRqstDevice, getDeviceTypeStrForPortId(client->deviceId).c_str());
@@ -708,2 +746,2 @@ Status AudioPolicyService::startInput(int32_t portIdAidl) {
-                        item->setCString(kAudioPolicyRqstPkg,
-                                         std::to_string(other->attributionSource.uid).c_str());
+                        item->setCString(kAudioPolicyRqstPkg, std::to_string(
+                            other->attributionSource.uid).c_str());
@@ -728,3 +766,4 @@ Status AudioPolicyService::startInput(int32_t portIdAidl) {
-Status AudioPolicyService::stopInput(int32_t portIdAidl) {
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+Status AudioPolicyService::stopInput(int32_t portIdAidl)
+{
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -747,3 +786,4 @@ Status AudioPolicyService::stopInput(int32_t portIdAidl) {
-Status AudioPolicyService::releaseInput(int32_t portIdAidl) {
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+Status AudioPolicyService::releaseInput(int32_t portIdAidl)
+{
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -787 +827,2 @@ Status AudioPolicyService::releaseInput(int32_t portIdAidl) {
-Status AudioPolicyService::initStreamVolume(AudioStreamType streamAidl, int32_t indexMinAidl,
+Status AudioPolicyService::initStreamVolume(AudioStreamType streamAidl,
+                                            int32_t indexMinAidl,
@@ -826,2 +867,3 @@ Status AudioPolicyService::setStreamVolumeIndex(AudioStreamType streamAidl,
-    return binderStatusFromStatusT(
-            mAudioPolicyManager->setStreamVolumeIndex(stream, index, device));
+    return binderStatusFromStatusT(mAudioPolicyManager->setStreamVolumeIndex(stream,
+                                                                             index,
+                                                                             device));
@@ -851,2 +893,2 @@ Status AudioPolicyService::setVolumeIndexForAttributes(
-        const media::AudioAttributesInternal& attrAidl, const AudioDeviceDescription& deviceAidl,
-        int32_t indexAidl) {
+        const media::AudioAttributesInternal& attrAidl,
+        const AudioDeviceDescription& deviceAidl, int32_t indexAidl) {
@@ -872,2 +914,2 @@ Status AudioPolicyService::getVolumeIndexForAttributes(
-        const media::AudioAttributesInternal& attrAidl, const AudioDeviceDescription& deviceAidl,
-        int32_t* _aidl_return) {
+        const media::AudioAttributesInternal& attrAidl,
+        const AudioDeviceDescription& deviceAidl, int32_t* _aidl_return) {
@@ -925 +967,2 @@ Status AudioPolicyService::getMaxVolumeIndexForAttributes(
-Status AudioPolicyService::getStrategyForStream(AudioStreamType streamAidl, int32_t* _aidl_return) {
+Status AudioPolicyService::getStrategyForStream(AudioStreamType streamAidl,
+                                                int32_t* _aidl_return) {
@@ -929,2 +972,2 @@ Status AudioPolicyService::getStrategyForStream(AudioStreamType streamAidl, int3
-        *_aidl_return =
-                VALUE_OR_RETURN_BINDER_STATUS(convertReinterpret<int32_t>(PRODUCT_STRATEGY_NONE));
+        *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+                convertReinterpret<int32_t>(PRODUCT_STRATEGY_NONE));
@@ -937 +980,2 @@ Status AudioPolicyService::getStrategyForStream(AudioStreamType streamAidl, int3
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_product_strategy_t_int32_t(
+    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+            legacy2aidl_product_strategy_t_int32_t(
@@ -943 +987,2 @@ Status AudioPolicyService::getDevicesForAttributes(const media::AudioAttributesI
-                                                   std::vector<AudioDevice>* _aidl_return) {
+                                                   std::vector<AudioDevice>* _aidl_return)
+{
@@ -954,2 +999,3 @@ Status AudioPolicyService::getDevicesForAttributes(const media::AudioAttributesI
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(convertContainer<std::vector<AudioDevice>>(
-            devices, legacy2aidl_AudioDeviceTypeAddress));
+    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<std::vector<AudioDevice>>(devices,
+                                                       legacy2aidl_AudioDeviceTypeAddress));
@@ -962,2 +1008,2 @@ Status AudioPolicyService::getOutputForEffect(const media::EffectDescriptor& des
-    RETURN_IF_BINDER_ERROR(
-            binderStatusFromStatusT(AudioValidator::validateEffectDescriptor(desc, "73126106")));
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            AudioValidator::validateEffectDescriptor(desc, "73126106")));
@@ -978,6 +1024,6 @@ Status AudioPolicyService::registerEffect(const media::EffectDescriptor& descAid
-    audio_io_handle_t io =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_io_handle_t(ioAidl));
-    product_strategy_t strategy =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_product_strategy_t(strategyAidl));
-    audio_session_t session =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_session_t(sessionAidl));
+    audio_io_handle_t io = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_io_handle_t(ioAidl));
+    product_strategy_t strategy = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_product_strategy_t(strategyAidl));
+    audio_session_t session = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_session_t(sessionAidl));
@@ -985,2 +1031,2 @@ Status AudioPolicyService::registerEffect(const media::EffectDescriptor& descAid
-    RETURN_IF_BINDER_ERROR(
-            binderStatusFromStatusT(AudioValidator::validateEffectDescriptor(desc, "73126106")));
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            AudioValidator::validateEffectDescriptor(desc, "73126106")));
@@ -995 +1041,2 @@ Status AudioPolicyService::registerEffect(const media::EffectDescriptor& descAid
-Status AudioPolicyService::unregisterEffect(int32_t idAidl) {
+Status AudioPolicyService::unregisterEffect(int32_t idAidl)
+{
@@ -1004 +1051,2 @@ Status AudioPolicyService::unregisterEffect(int32_t idAidl) {
-Status AudioPolicyService::setEffectEnabled(int32_t idAidl, bool enabled) {
+Status AudioPolicyService::setEffectEnabled(int32_t idAidl, bool enabled)
+{
@@ -1017,2 +1065,2 @@ Status AudioPolicyService::moveEffectsToIo(const std::vector<int32_t>& idsAidl,
-    audio_io_handle_t io =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_io_handle_t(ioAidl));
+    audio_io_handle_t io = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_io_handle_t(ioAidl));
@@ -1046 +1094,2 @@ Status AudioPolicyService::isStreamActive(AudioStreamType streamAidl, int32_t in
-Status AudioPolicyService::isStreamActiveRemotely(AudioStreamType streamAidl, int32_t inPastMsAidl,
+Status AudioPolicyService::isStreamActiveRemotely(AudioStreamType streamAidl,
+                                                  int32_t inPastMsAidl,
@@ -1064,2 +1113,2 @@ Status AudioPolicyService::isSourceActive(AudioSource sourceAidl, bool* _aidl_re
-    audio_source_t source =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioSource_audio_source_t(sourceAidl));
+    audio_source_t source = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioSource_audio_source_t(sourceAidl));
@@ -1074 +1123,2 @@ Status AudioPolicyService::isSourceActive(AudioSource sourceAidl, bool* _aidl_re
-status_t AudioPolicyService::getAudioPolicyEffects(sp<AudioPolicyEffects>& audioPolicyEffects) {
+status_t AudioPolicyService::getAudioPolicyEffects(sp<AudioPolicyEffects>& audioPolicyEffects)
+{
@@ -1088 +1138,2 @@ Status AudioPolicyService::queryDefaultPreProcessing(
-        int32_t audioSessionAidl, Int* countAidl,
+        int32_t audioSessionAidl,
+        Int* countAidl,
@@ -1090,2 +1141,2 @@ Status AudioPolicyService::queryDefaultPreProcessing(
-    audio_session_t audioSession =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_session_t(audioSessionAidl));
+    audio_session_t audioSession = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_session_t(audioSessionAidl));
@@ -1103,2 +1154,3 @@ Status AudioPolicyService::queryDefaultPreProcessing(
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(convertRange(
-            descriptors.get(), descriptors.get() + countReq, std::back_inserter(*_aidl_return),
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            convertRange(descriptors.get(), descriptors.get() + countReq,
+                         std::back_inserter(*_aidl_return),
@@ -1111,10 +1163,12 @@ Status AudioPolicyService::addSourceDefaultEffect(const AudioUuid& typeAidl,
-                                                  const AudioUuid& uuidAidl, int32_t priority,
-                                                  AudioSource sourceAidl, int32_t* _aidl_return) {
-    effect_uuid_t type =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioUuid_audio_uuid_t(typeAidl));
-    String16 opPackageName =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_string_view_String16(opPackageNameAidl));
-    effect_uuid_t uuid =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioUuid_audio_uuid_t(uuidAidl));
-    audio_source_t source =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioSource_audio_source_t(sourceAidl));
+                                                  const AudioUuid& uuidAidl,
+                                                  int32_t priority,
+                                                  AudioSource sourceAidl,
+                                                  int32_t* _aidl_return) {
+    effect_uuid_t type = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioUuid_audio_uuid_t(typeAidl));
+    String16 opPackageName = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_string_view_String16(opPackageNameAidl));
+    effect_uuid_t uuid = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioUuid_audio_uuid_t(uuidAidl));
+    audio_source_t source = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioSource_audio_source_t(sourceAidl));
@@ -1134,10 +1188,11 @@ Status AudioPolicyService::addStreamDefaultEffect(const AudioUuid& typeAidl,
-                                                  const AudioUuid& uuidAidl, int32_t priority,
-                                                  AudioUsage usageAidl, int32_t* _aidl_return) {
-    effect_uuid_t type =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioUuid_audio_uuid_t(typeAidl));
-    String16 opPackageName =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_string_view_String16(opPackageNameAidl));
-    effect_uuid_t uuid =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioUuid_audio_uuid_t(uuidAidl));
-    audio_usage_t usage =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioUsage_audio_usage_t(usageAidl));
+                                                  const AudioUuid& uuidAidl,
+                                                  int32_t priority, AudioUsage usageAidl,
+                                                  int32_t* _aidl_return) {
+    effect_uuid_t type = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioUuid_audio_uuid_t(typeAidl));
+    String16 opPackageName = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_string_view_String16(opPackageNameAidl));
+    effect_uuid_t uuid = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioUuid_audio_uuid_t(uuidAidl));
+    audio_usage_t usage = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioUsage_audio_usage_t(usageAidl));
@@ -1155,3 +1210,4 @@ Status AudioPolicyService::addStreamDefaultEffect(const AudioUuid& typeAidl,
-Status AudioPolicyService::removeSourceDefaultEffect(int32_t idAidl) {
-    audio_unique_id_t id =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_unique_id_t(idAidl));
+Status AudioPolicyService::removeSourceDefaultEffect(int32_t idAidl)
+{
+    audio_unique_id_t id = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_unique_id_t(idAidl));
@@ -1165,3 +1221,4 @@ Status AudioPolicyService::removeSourceDefaultEffect(int32_t idAidl) {
-Status AudioPolicyService::removeStreamDefaultEffect(int32_t idAidl) {
-    audio_unique_id_t id =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_unique_id_t(idAidl));
+Status AudioPolicyService::removeStreamDefaultEffect(int32_t idAidl)
+{
+    audio_unique_id_t id = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_unique_id_t(idAidl));
@@ -1224 +1281,2 @@ Status AudioPolicyService::isDirectOutputSupported(
-        const AudioConfigBase& configAidl, const media::AudioAttributesInternal& attributesAidl,
+        const AudioConfigBase& configAidl,
+        const media::AudioAttributesInternal& attributesAidl,
@@ -1245,6 +1303,6 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-    audio_port_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioPortRole_audio_port_role_t(roleAidl));
-    audio_port_type_t type =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioPortType_audio_port_type_t(typeAidl));
-    unsigned int num_ports =
-            VALUE_OR_RETURN_BINDER_STATUS(convertIntegral<unsigned int>(count->value));
+    audio_port_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioPortRole_audio_port_role_t(roleAidl));
+    audio_port_type_t type = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioPortType_audio_port_type_t(typeAidl));
+    unsigned int num_ports = VALUE_OR_RETURN_BINDER_STATUS(
+            convertIntegral<unsigned int>(count->value));
@@ -1279,2 +1337,2 @@ Status AudioPolicyService::listDeclaredDevicePorts(media::AudioPortRole role,
-    return binderStatusFromStatusT(
-            mAudioPolicyManager->listDeclaredDevicePorts(role, _aidl_return));
+    return binderStatusFromStatusT(mAudioPolicyManager->listDeclaredDevicePorts(
+                    role, _aidl_return));
@@ -1282 +1340,2 @@ Status AudioPolicyService::listDeclaredDevicePorts(media::AudioPortRole role,
-Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_return) {
+Status AudioPolicyService::getAudioPort(int portId,
+                                        media::AudioPortFw* _aidl_return) {
@@ -1290 +1348,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-<<<<<<< HEAD
@@ -1292,8 +1349,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-||||||| 0f5b5f3279
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        anonymizePortBluetoothAddress(&port);
-    }
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_v7_AudioPort(port));
-=======
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_v7_AudioPort(port));
->>>>>>> aa7f1e5e7a677a871df59c77d33dc38debfae3ca
@@ -1303,5 +1353,6 @@ Status AudioPolicyService::createAudioPatch(const media::AudioPatchFw& patchAidl
-                                            int32_t handleAidl, int32_t* _aidl_return) {
-    audio_patch patch =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioPatchFw_audio_patch(patchAidl));
-    audio_patch_handle_t handle =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(handleAidl));
+                                            int32_t handleAidl,
+                                            int32_t* _aidl_return) {
+    audio_patch patch = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioPatchFw_audio_patch(patchAidl));
+    audio_patch_handle_t handle = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(handleAidl));
@@ -1317,2 +1368,3 @@ Status AudioPolicyService::createAudioPatch(const media::AudioPatchFw& patchAidl
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(mAudioPolicyManager->createAudioPatch(
-            &patch, &handle, IPCThreadState::self()->getCallingUid())));
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            mAudioPolicyManager->createAudioPatch(&patch, &handle,
+                                                  IPCThreadState::self()->getCallingUid())));
@@ -1322,3 +1374,4 @@ Status AudioPolicyService::createAudioPatch(const media::AudioPatchFw& patchAidl
-Status AudioPolicyService::releaseAudioPatch(int32_t handleAidl) {
-    audio_patch_handle_t handle =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_patch_handle_t(handleAidl));
+Status AudioPolicyService::releaseAudioPatch(int32_t handleAidl)
+{
+    audio_patch_handle_t handle = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_patch_handle_t(handleAidl));
@@ -1333,2 +1386,3 @@ Status AudioPolicyService::releaseAudioPatch(int32_t handleAidl) {
-    return binderStatusFromStatusT(mAudioPolicyManager->releaseAudioPatch(
-            handle, IPCThreadState::self()->getCallingUid()));
+    return binderStatusFromStatusT(
+            mAudioPolicyManager->releaseAudioPatch(handle,
+                                                   IPCThreadState::self()->getCallingUid()));
@@ -1339,2 +1393,2 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-    unsigned int num_patches =
-            VALUE_OR_RETURN_BINDER_STATUS(convertIntegral<unsigned int>(count->value));
+    unsigned int num_patches = VALUE_OR_RETURN_BINDER_STATUS(
+            convertIntegral<unsigned int>(count->value));
@@ -1362 +1416,2 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-Status AudioPolicyService::setAudioPortConfig(const media::AudioPortConfigFw& configAidl) {
+Status AudioPolicyService::setAudioPortConfig(const media::AudioPortConfigFw& configAidl)
+{
@@ -1377 +1432,2 @@ Status AudioPolicyService::setAudioPortConfig(const media::AudioPortConfigFw& co
-Status AudioPolicyService::acquireSoundTriggerSession(media::SoundTriggerSession* _aidl_return) {
+Status AudioPolicyService::acquireSoundTriggerSession(media::SoundTriggerSession* _aidl_return)
+{
@@ -1390,4 +1446,4 @@ Status AudioPolicyService::acquireSoundTriggerSession(media::SoundTriggerSession
-    _aidl_return->session =
-            VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_session_t_int32_t(session));
-    _aidl_return->ioHandle =
-            VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_io_handle_t_int32_t(ioHandle));
+    _aidl_return->session = VALUE_OR_RETURN_BINDER_STATUS(
+            legacy2aidl_audio_session_t_int32_t(session));
+    _aidl_return->ioHandle = VALUE_OR_RETURN_BINDER_STATUS(
+            legacy2aidl_audio_io_handle_t_int32_t(ioHandle));
@@ -1398,3 +1454,4 @@ Status AudioPolicyService::acquireSoundTriggerSession(media::SoundTriggerSession
-Status AudioPolicyService::releaseSoundTriggerSession(int32_t sessionAidl) {
-    audio_session_t session =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_session_t(sessionAidl));
+Status AudioPolicyService::releaseSoundTriggerSession(int32_t sessionAidl)
+{
+    audio_session_t session = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_session_t(sessionAidl));
@@ -1415,3 +1472,3 @@ Status AudioPolicyService::registerPolicyMixes(const std::vector<media::AudioMix
-    RETURN_IF_BINDER_ERROR(
-            binderStatusFromStatusT(convertRange(mixesAidl.begin(), mixesAidl.begin() + size,
-                                                 std::back_inserter(mixes), aidl2legacy_AudioMix)));
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            convertRange(mixesAidl.begin(), mixesAidl.begin() + size, std::back_inserter(mixes),
+                         aidl2legacy_AudioMix)));
@@ -1420,2 +1477 @@ Status AudioPolicyService::registerPolicyMixes(const std::vector<media::AudioMix
-        return !is_mix_loopback_render(mix.mRouteFlags);
-    });
+            return !is_mix_loopback_render(mix.mRouteFlags); });
@@ -1426,2 +1482,2 @@ Status AudioPolicyService::registerPolicyMixes(const std::vector<media::AudioMix
-            std::any_of(mixes.begin(), mixes.end(),
-                        [](auto& mix) { return mix.mVoiceCommunicationCaptureAllowed; });
+        std::any_of(mixes.begin(), mixes.end(), [](auto& mix) {
+            return mix.mVoiceCommunicationCaptureAllowed; });
@@ -1429,2 +1485 @@ Status AudioPolicyService::registerPolicyMixes(const std::vector<media::AudioMix
-        return mix.mAllowPrivilegedMediaPlaybackCapture;
-    });
+            return mix.mAllowPrivilegedMediaPlaybackCapture; });
@@ -1449 +1504,2 @@ Status AudioPolicyService::registerPolicyMixes(const std::vector<media::AudioMix
-Status AudioPolicyService::setUidDeviceAffinities(int32_t uidAidl,
+Status AudioPolicyService::setUidDeviceAffinities(
+        int32_t uidAidl,
@@ -1452,3 +1508,3 @@ Status AudioPolicyService::setUidDeviceAffinities(int32_t uidAidl,
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -1477 +1533,2 @@ Status AudioPolicyService::removeUidDeviceAffinities(int32_t uidAidl) {
-Status AudioPolicyService::setUserIdDeviceAffinities(int32_t userIdAidl,
+Status AudioPolicyService::setUserIdDeviceAffinities(
+        int32_t userIdAidl,
@@ -1480,3 +1537,3 @@ Status AudioPolicyService::setUserIdDeviceAffinities(int32_t userIdAidl,
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -1513,2 +1570,2 @@ Status AudioPolicyService::startAudioSource(const media::AudioPortConfigFw& sour
-    RETURN_IF_BINDER_ERROR(
-            binderStatusFromStatusT(AudioValidator::validateAudioPortConfig(source)));
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            AudioValidator::validateAudioPortConfig(source)));
@@ -1529,3 +1586,4 @@ Status AudioPolicyService::startAudioSource(const media::AudioPortConfigFw& sour
-Status AudioPolicyService::stopAudioSource(int32_t portIdAidl) {
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+Status AudioPolicyService::stopAudioSource(int32_t portIdAidl)
+{
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -1539 +1597,2 @@ Status AudioPolicyService::stopAudioSource(int32_t portIdAidl) {
-Status AudioPolicyService::setMasterMono(bool mono) {
+Status AudioPolicyService::setMasterMono(bool mono)
+{
@@ -1550 +1609,2 @@ Status AudioPolicyService::setMasterMono(bool mono) {
-Status AudioPolicyService::getMasterMono(bool* _aidl_return) {
+Status AudioPolicyService::getMasterMono(bool* _aidl_return)
+{
@@ -1558,3 +1618,3 @@ Status AudioPolicyService::getMasterMono(bool* _aidl_return) {
-Status AudioPolicyService::getStreamVolumeDB(AudioStreamType streamAidl, int32_t indexAidl,
-                                             const AudioDeviceDescription& deviceAidl,
-                                             float* _aidl_return) {
+Status AudioPolicyService::getStreamVolumeDB(
+        AudioStreamType streamAidl, int32_t indexAidl,
+        const AudioDeviceDescription& deviceAidl, float* _aidl_return) {
@@ -1577,2 +1637,2 @@ Status AudioPolicyService::getSurroundFormats(Int* count,
-    unsigned int numSurroundFormats =
-            VALUE_OR_RETURN_BINDER_STATUS(convertIntegral<unsigned int>(count->value));
+    unsigned int numSurroundFormats = VALUE_OR_RETURN_BINDER_STATUS(
+            convertIntegral<unsigned int>(count->value));
@@ -1590,2 +1650,3 @@ Status AudioPolicyService::getSurroundFormats(Int* count,
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(mAudioPolicyManager->getSurroundFormats(
-            &numSurroundFormats, surroundFormats.get(), surroundFormatsEnabled.get())));
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            mAudioPolicyManager->getSurroundFormats(&numSurroundFormats, surroundFormats.get(),
+                                                    surroundFormatsEnabled.get())));
@@ -1593,4 +1654,7 @@ Status AudioPolicyService::getSurroundFormats(Int* count,
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(convertRange(
-            surroundFormats.get(), surroundFormats.get() + numSurroundFormatsReq,
-            std::back_inserter(*formats), legacy2aidl_audio_format_t_AudioFormatDescription)));
-    formatsEnabled->insert(formatsEnabled->begin(), surroundFormatsEnabled.get(),
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            convertRange(surroundFormats.get(), surroundFormats.get() + numSurroundFormatsReq,
+                         std::back_inserter(*formats),
+                         legacy2aidl_audio_format_t_AudioFormatDescription)));
+    formatsEnabled->insert(
+            formatsEnabled->begin(),
+            surroundFormatsEnabled.get(),
@@ -1603,2 +1667,2 @@ Status AudioPolicyService::getReportedSurroundFormats(
-    unsigned int numSurroundFormats =
-            VALUE_OR_RETURN_BINDER_STATUS(convertIntegral<unsigned int>(count->value));
+    unsigned int numSurroundFormats = VALUE_OR_RETURN_BINDER_STATUS(
+            convertIntegral<unsigned int>(count->value));
@@ -1615 +1679,2 @@ Status AudioPolicyService::getReportedSurroundFormats(
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(mAudioPolicyManager->getReportedSurroundFormats(
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            mAudioPolicyManager->getReportedSurroundFormats(
@@ -1618,3 +1683,4 @@ Status AudioPolicyService::getReportedSurroundFormats(
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(convertRange(
-            surroundFormats.get(), surroundFormats.get() + numSurroundFormatsReq,
-            std::back_inserter(*formats), legacy2aidl_audio_format_t_AudioFormatDescription)));
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            convertRange(surroundFormats.get(), surroundFormats.get() + numSurroundFormatsReq,
+                         std::back_inserter(*formats),
+                         legacy2aidl_audio_format_t_AudioFormatDescription)));
@@ -1637,3 +1703,4 @@ Status AudioPolicyService::getHwOffloadFormatsSupportedForBluetoothMedia(
-    *_aidl_return =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<std::vector<AudioFormatDescription>>(
-                    formats, legacy2aidl_audio_format_t_AudioFormatDescription));
+    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<std::vector<AudioFormatDescription>>(
+                    formats,
+                    legacy2aidl_audio_format_t_AudioFormatDescription));
@@ -1642,2 +1709,2 @@ Status AudioPolicyService::getHwOffloadFormatsSupportedForBluetoothMedia(
-Status AudioPolicyService::setSurroundFormatEnabled(const AudioFormatDescription& audioFormatAidl,
-                                                    bool enabled) {
+Status AudioPolicyService::setSurroundFormatEnabled(
+        const AudioFormatDescription& audioFormatAidl, bool enabled) {
@@ -1654,2 +1721,2 @@ Status AudioPolicyService::setSurroundFormatEnabled(const AudioFormatDescription
-Status convertInt32VectorToUidVectorWithLimit(const std::vector<int32_t>& uidsAidl,
-                                              std::vector<uid_t>& uids) {
+Status convertInt32VectorToUidVectorWithLimit(
+        const std::vector<int32_t>& uidsAidl, std::vector<uid_t>& uids) {
@@ -1657,2 +1724,5 @@ Status convertInt32VectorToUidVectorWithLimit(const std::vector<int32_t>& uidsAi
-            convertRangeWithLimit(uidsAidl.begin(), uidsAidl.end(), std::back_inserter(uids),
-                                  aidl2legacy_int32_t_uid_t, MAX_ITEMS_PER_LIST)));
+        convertRangeWithLimit(uidsAidl.begin(),
+            uidsAidl.end(),
+            std::back_inserter(uids),
+            aidl2legacy_int32_t_uid_t,
+            MAX_ITEMS_PER_LIST)));
@@ -1661 +1731,2 @@ Status convertInt32VectorToUidVectorWithLimit(const std::vector<int32_t>& uidsAi
-Status AudioPolicyService::setAssistantServicesUids(const std::vector<int32_t>& uidsAidl) {
+Status AudioPolicyService::setAssistantServicesUids(const std::vector<int32_t>& uidsAidl)
+{
@@ -1676 +1747,2 @@ Status AudioPolicyService::setActiveAssistantServicesUids(
-Status AudioPolicyService::setA11yServicesUids(const std::vector<int32_t>& uidsAidl) {
+Status AudioPolicyService::setA11yServicesUids(const std::vector<int32_t>& uidsAidl)
+{
@@ -1683 +1755,2 @@ Status AudioPolicyService::setA11yServicesUids(const std::vector<int32_t>& uidsA
-Status AudioPolicyService::setCurrentImeUid(int32_t uidAidl) {
+Status AudioPolicyService::setCurrentImeUid(int32_t uidAidl)
+{
@@ -1689 +1762,2 @@ Status AudioPolicyService::setCurrentImeUid(int32_t uidAidl) {
-Status AudioPolicyService::isHapticPlaybackSupported(bool* _aidl_return) {
+Status AudioPolicyService::isHapticPlaybackSupported(bool* _aidl_return)
+{
@@ -1698 +1772,2 @@ Status AudioPolicyService::isHapticPlaybackSupported(bool* _aidl_return) {
-Status AudioPolicyService::isUltrasoundSupported(bool* _aidl_return) {
+Status AudioPolicyService::isUltrasoundSupported(bool* _aidl_return)
+{
@@ -1707 +1782,2 @@ Status AudioPolicyService::isUltrasoundSupported(bool* _aidl_return) {
-Status AudioPolicyService::isHotwordStreamSupported(bool lookbackAudio, bool* _aidl_return) {
+Status AudioPolicyService::isHotwordStreamSupported(bool lookbackAudio, bool* _aidl_return)
+{
@@ -1727 +1803,2 @@ Status AudioPolicyService::listAudioProductStrategies(
-                    strategies, legacy2aidl_AudioProductStrategy));
+                    strategies,
+                    legacy2aidl_AudioProductStrategy));
@@ -1731,2 +1808,2 @@ Status AudioPolicyService::getProductStrategyFromAudioAttributes(
-        const media::AudioAttributesInternal& aaAidl, bool fallbackOnDefault,
-        int32_t* _aidl_return) {
+        const media::AudioAttributesInternal& aaAidl,
+        bool fallbackOnDefault, int32_t* _aidl_return) {
@@ -1740,2 +1817,2 @@ Status AudioPolicyService::getProductStrategyFromAudioAttributes(
-    RETURN_IF_BINDER_ERROR(
-            binderStatusFromStatusT(mAudioPolicyManager->getProductStrategyFromAudioAttributes(
+    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
+            mAudioPolicyManager->getProductStrategyFromAudioAttributes(
@@ -1743,2 +1820,2 @@ Status AudioPolicyService::getProductStrategyFromAudioAttributes(
-    *_aidl_return =
-            VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_product_strategy_t_int32_t(productStrategy));
+    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+            legacy2aidl_product_strategy_t_int32_t(productStrategy));
@@ -1747,2 +1824,2 @@ Status AudioPolicyService::getProductStrategyFromAudioAttributes(
-Status AudioPolicyService::listAudioVolumeGroups(
-        std::vector<media::AudioVolumeGroup>* _aidl_return) {
+Status AudioPolicyService::listAudioVolumeGroups(std::vector<media::AudioVolumeGroup>* _aidl_return)
+{
@@ -1756,3 +1833,3 @@ Status AudioPolicyService::listAudioVolumeGroups(
-    *_aidl_return =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<std::vector<media::AudioVolumeGroup>>(
-                    groups, legacy2aidl_AudioVolumeGroup));
+    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<std::vector<media::AudioVolumeGroup>>(groups,
+                                                                   legacy2aidl_AudioVolumeGroup));
@@ -1762,2 +1839,2 @@ Status AudioPolicyService::getVolumeGroupFromAudioAttributes(
-        const media::AudioAttributesInternal& aaAidl, bool fallbackOnDefault,
-        int32_t* _aidl_return) {
+        const media::AudioAttributesInternal& aaAidl,
+        bool fallbackOnDefault, int32_t* _aidl_return) {
@@ -1772 +1849,2 @@ Status AudioPolicyService::getVolumeGroupFromAudioAttributes(
-            binderStatusFromStatusT(mAudioPolicyManager->getVolumeGroupFromAudioAttributes(
+            binderStatusFromStatusT(
+                    mAudioPolicyManager->getVolumeGroupFromAudioAttributes(
@@ -1777 +1855,2 @@ Status AudioPolicyService::getVolumeGroupFromAudioAttributes(
-Status AudioPolicyService::setRttEnabled(bool enabled) {
+Status AudioPolicyService::setRttEnabled(bool enabled)
+{
@@ -1782 +1861,2 @@ Status AudioPolicyService::setRttEnabled(bool enabled) {
-Status AudioPolicyService::isCallScreenModeSupported(bool* _aidl_return) {
+Status AudioPolicyService::isCallScreenModeSupported(bool* _aidl_return)
+{
@@ -1791 +1871,2 @@ Status AudioPolicyService::isCallScreenModeSupported(bool* _aidl_return) {
-Status AudioPolicyService::setDevicesRoleForStrategy(int32_t strategyAidl,
+Status AudioPolicyService::setDevicesRoleForStrategy(
+        int32_t strategyAidl,
@@ -1794,7 +1875,7 @@ Status AudioPolicyService::setDevicesRoleForStrategy(int32_t strategyAidl,
-    product_strategy_t strategy =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_product_strategy_t(strategyAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    product_strategy_t strategy = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_product_strategy_t(strategyAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -1812 +1893,2 @@ Status AudioPolicyService::removeDevicesRoleForStrategy(
-        int32_t strategyAidl, media::DeviceRole roleAidl,
+        int32_t strategyAidl,
+        media::DeviceRole roleAidl,
@@ -1814,7 +1896,7 @@ Status AudioPolicyService::removeDevicesRoleForStrategy(
-    product_strategy_t strategy =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_product_strategy_t(strategyAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    product_strategy_t strategy = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_product_strategy_t(strategyAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -1833,4 +1915,4 @@ Status AudioPolicyService::clearDevicesRoleForStrategy(int32_t strategyAidl,
-    product_strategy_t strategy =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_product_strategy_t(strategyAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
+     product_strategy_t strategy = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_product_strategy_t(strategyAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
@@ -1847 +1929,2 @@ Status AudioPolicyService::clearDevicesRoleForStrategy(int32_t strategyAidl,
-Status AudioPolicyService::getDevicesForRoleAndStrategy(int32_t strategyAidl,
+Status AudioPolicyService::getDevicesForRoleAndStrategy(
+        int32_t strategyAidl,
@@ -1850,4 +1933,4 @@ Status AudioPolicyService::getDevicesForRoleAndStrategy(int32_t strategyAidl,
-    product_strategy_t strategy =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_product_strategy_t(strategyAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    product_strategy_t strategy = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_product_strategy_t(strategyAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
@@ -1861,2 +1944,3 @@ Status AudioPolicyService::getDevicesForRoleAndStrategy(int32_t strategyAidl,
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(convertContainer<std::vector<AudioDevice>>(
-            devices, legacy2aidl_AudioDeviceTypeAddress));
+    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<std::vector<AudioDevice>>(devices,
+                                                       legacy2aidl_AudioDeviceTypeAddress));
@@ -1871 +1955,2 @@ Status AudioPolicyService::setDevicesRoleForCapturePreset(
-        AudioSource audioSourceAidl, media::DeviceRole roleAidl,
+        AudioSource audioSourceAidl,
+        media::DeviceRole roleAidl,
@@ -1873,7 +1958,7 @@ Status AudioPolicyService::setDevicesRoleForCapturePreset(
-    audio_source_t audioSource =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    audio_source_t audioSource = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -1888 +1973,2 @@ Status AudioPolicyService::addDevicesRoleForCapturePreset(
-        AudioSource audioSourceAidl, media::DeviceRole roleAidl,
+        AudioSource audioSourceAidl,
+        media::DeviceRole roleAidl,
@@ -1890,7 +1976,7 @@ Status AudioPolicyService::addDevicesRoleForCapturePreset(
-    audio_source_t audioSource =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    audio_source_t audioSource = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -1905 +1991,2 @@ Status AudioPolicyService::removeDevicesRoleForCapturePreset(
-        AudioSource audioSourceAidl, media::DeviceRole roleAidl,
+        AudioSource audioSourceAidl,
+        media::DeviceRole roleAidl,
@@ -1907,7 +1994,7 @@ Status AudioPolicyService::removeDevicesRoleForCapturePreset(
-    audio_source_t audioSource =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    audio_source_t audioSource = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -1923,4 +2010,4 @@ Status AudioPolicyService::clearDevicesRoleForCapturePreset(AudioSource audioSou
-    audio_source_t audioSource =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    audio_source_t audioSource = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
@@ -1935 +2022,2 @@ Status AudioPolicyService::getDevicesForRoleAndCapturePreset(
-        AudioSource audioSourceAidl, media::DeviceRole roleAidl,
+        AudioSource audioSourceAidl,
+        media::DeviceRole roleAidl,
@@ -1937,4 +2025,4 @@ Status AudioPolicyService::getDevicesForRoleAndCapturePreset(
-    audio_source_t audioSource =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
-    device_role_t role =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_DeviceRole_device_role_t(roleAidl));
+    audio_source_t audioSource = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_AudioSource_audio_source_t(audioSourceAidl));
+    device_role_t role = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_DeviceRole_device_role_t(roleAidl));
@@ -1948,2 +2036,3 @@ Status AudioPolicyService::getDevicesForRoleAndCapturePreset(
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(convertContainer<std::vector<AudioDevice>>(
-            devices, legacy2aidl_AudioDeviceTypeAddress));
+    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<std::vector<AudioDevice>>(devices,
+                                                       legacy2aidl_AudioDeviceTypeAddress));
@@ -1952 +2041,2 @@ Status AudioPolicyService::getDevicesForRoleAndCapturePreset(
-Status AudioPolicyService::getSpatializer(const sp<media::INativeSpatializerCallback>& callback,
+Status AudioPolicyService::getSpatializer(
+        const sp<media::INativeSpatializerCallback>& callback,
@@ -1959 +2049,2 @@ Status AudioPolicyService::getSpatializer(const sp<media::INativeSpatializerCall
-        RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(mSpatializer->registerCallback(callback)));
+        RETURN_IF_BINDER_ERROR(
+                binderStatusFromStatusT(mSpatializer->registerCallback(callback)));
@@ -1966 +2057,2 @@ Status AudioPolicyService::canBeSpatialized(
-        const std::optional<AudioConfig>& configAidl, const std::vector<AudioDevice>& devicesAidl,
+        const std::optional<AudioConfig>& configAidl,
+        const std::vector<AudioDevice>& devicesAidl,
@@ -1979 +2071,2 @@ Status AudioPolicyService::canBeSpatialized(
-                aidl2legacy_AudioConfig_audio_config_t(configAidl.value(), false ));
+                                    aidl2legacy_AudioConfig_audio_config_t(configAidl.value(),
+                                    false ));
@@ -1981,3 +2074,3 @@ Status AudioPolicyService::canBeSpatialized(
-    AudioDeviceTypeAddrVector devices =
-            VALUE_OR_RETURN_BINDER_STATUS(convertContainer<AudioDeviceTypeAddrVector>(
-                    devicesAidl, aidl2legacy_AudioDeviceTypeAddress));
+    AudioDeviceTypeAddrVector devices = VALUE_OR_RETURN_BINDER_STATUS(
+            convertContainer<AudioDeviceTypeAddrVector>(devicesAidl,
+                                                        aidl2legacy_AudioDeviceTypeAddress));
@@ -2029,2 +2122,2 @@ Status AudioPolicyService::getSupportedMixerAttributes(
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -2033,2 +2126,3 @@ Status AudioPolicyService::getSupportedMixerAttributes(
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
-            mAudioPolicyManager->getSupportedMixerAttributes(portId, mixerAttrs)));
+    RETURN_IF_BINDER_ERROR(
+            binderStatusFromStatusT(mAudioPolicyManager->getSupportedMixerAttributes(
+                    portId, mixerAttrs)));
@@ -2037 +2131,2 @@ Status AudioPolicyService::getSupportedMixerAttributes(
-                    mixerAttrs, legacy2aidl_audio_mixer_attributes_t_AudioMixerAttributesInternal));
+                    mixerAttrs,
+                    legacy2aidl_audio_mixer_attributes_t_AudioMixerAttributesInternal));
@@ -2041 +2136,3 @@ Status AudioPolicyService::setPreferredMixerAttributes(
-        const media::AudioAttributesInternal& attrAidl, int32_t portIdAidl, int32_t uidAidl,
+        const media::AudioAttributesInternal& attrAidl,
+        int32_t portIdAidl,
+        int32_t uidAidl,
@@ -2051,2 +2148,2 @@ Status AudioPolicyService::setPreferredMixerAttributes(
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -2058 +2155,2 @@ Status AudioPolicyService::getPreferredMixerAttributes(
-        const media::AudioAttributesInternal& attrAidl, int32_t portIdAidl,
+        const media::AudioAttributesInternal& attrAidl,
+        int32_t portIdAidl,
@@ -2065,2 +2163,2 @@ Status AudioPolicyService::getPreferredMixerAttributes(
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
@@ -2069,2 +2167,3 @@ Status AudioPolicyService::getPreferredMixerAttributes(
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
-            mAudioPolicyManager->getPreferredMixerAttributes(&attr, portId, &mixerAttr)));
+    RETURN_IF_BINDER_ERROR(
+            binderStatusFromStatusT(mAudioPolicyManager->getPreferredMixerAttributes(
+                    &attr, portId, &mixerAttr)));
@@ -2076 +2175,3 @@ Status AudioPolicyService::clearPreferredMixerAttributes(
-        const media::AudioAttributesInternal& attrAidl, int32_t portIdAidl, int32_t uidAidl) {
+        const media::AudioAttributesInternal& attrAidl,
+        int32_t portIdAidl,
+        int32_t uidAidl) {
@@ -2083,2 +2184,2 @@ Status AudioPolicyService::clearPreferredMixerAttributes(
-    audio_port_handle_t portId =
-            VALUE_OR_RETURN_BINDER_STATUS(aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
+    audio_port_handle_t portId = VALUE_OR_RETURN_BINDER_STATUS(
+            aidl2legacy_int32_t_audio_port_handle_t(portIdAidl));
