--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
@@ -12,2 +12 @@
-    ({ \
-        auto _tmp = (x); \
+    ({ auto _tmp = (x); \
@@ -15,2 +14 @@
-        std::move(_tmp.value()); \
-    })
+       std::move(_tmp.value()); })
@@ -26 +23,0 @@ namespace android {
-using aidl_utils::binderStatusFromStatusT;
@@ -27,0 +25 @@ using binder::Status;
+using aidl_utils::binderStatusFromStatusT;
@@ -1290 +1348,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-<<<<<<< HEAD
@@ -1292,8 +1349,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-||||||| 0f5b5f3279
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        anonymizePortBluetoothAddress(&port);
-    }
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_v7_AudioPort(port));
-=======
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_v7_AudioPort(port));
->>>>>>> aa7f1e5e7a677a871df59c77d33dc38debfae3ca
