--- a/./output/frameworks_av/d730e941-41d44065-8a06b7a8/services@mediametrics@include@mediametricsservice@TimedAction.no_comments_mergegen.h
+++ b/./output/frameworks_av/d730e941-41d44065-8a06b7a8/services@mediametrics@include@mediametricsservice@TimedAction.no_comments_merged.h
@@ -8,0 +9,2 @@ class TimedAction {
+    using TimerClock = class std::chrono::system_clock;
+    static constexpr auto kWakeupInterval = std::chrono::minutes(3);
