--- a/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_conflict.cpp
+++ b/./output/frameworks_av/ac380723-aa7f1e5e-0f5b5f32/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
@@ -1349 +1348,0 @@ Status AudioPolicyService::getAudioPort(int portId,
-<<<<<<< HEAD
@@ -1351,8 +1349,0 @@ Status AudioPolicyService::getAudioPort(int portId,
-||||||| 0f5b5f3279
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        anonymizePortBluetoothAddress(&port);
-    }
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_v7_AudioPort(port));
-=======
-    *_aidl_return = VALUE_OR_RETURN_BINDER_STATUS(legacy2aidl_audio_port_v7_AudioPort(port));
->>>>>>> aa7f1e5e
