--- a/./output/grpc/a2357844-b5b4c9f1-ffb64347/include@grpcpp@server_impl.no_comments_conflict.h
+++ b/./output/grpc/a2357844-b5b4c9f1-ffb64347/include@grpcpp@server_impl.no_comments_merged.h
@@ -31,8 +30,0 @@ namespace grpc_impl {
-<<<<<<< HEAD
-class HealthCheckServiceInterface;
-||||||| ffb643479c
-class HealthCheckServiceInterface;
-class ServerContext;
-=======
-class ServerContext;
->>>>>>> b5b4c9f1
