--- a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@compaction_picker.no_comments_conflict.h
+++ b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@compaction_picker.no_comments_merged.h
@@ -54 +53,0 @@ class UniversalCompactionPicker : public CompactionPicker {
-<<<<<<< HEAD
@@ -58,10 +56,0 @@ class UniversalCompactionPicker : public CompactionPicker {
-  virtual Compaction* PickCompaction(Version* version) override;
-||||||| 8ca30bd51
-  UniversalCompactionPicker(const Options* options,
-                            const InternalKeyComparator* icmp)
-      : CompactionPicker(options, icmp) {}
-  virtual Compaction* PickCompaction(Version* version) override;
-=======
-  UniversalCompactionPicker(const Options* options,
-                            const InternalKeyComparator* icmp)
-      : CompactionPicker(options, icmp) {}
@@ -70 +58,0 @@ class UniversalCompactionPicker : public CompactionPicker {
->>>>>>> abeee9f2
@@ -81 +68,0 @@ class LevelCompactionPicker : public CompactionPicker {
-<<<<<<< HEAD
@@ -85,10 +71,0 @@ class LevelCompactionPicker : public CompactionPicker {
-  virtual Compaction* PickCompaction(Version* version) override;
-||||||| 8ca30bd51
-  LevelCompactionPicker(const Options* options,
-                        const InternalKeyComparator* icmp)
-      : CompactionPicker(options, icmp) {}
-  virtual Compaction* PickCompaction(Version* version) override;
-=======
-  LevelCompactionPicker(const Options* options,
-                        const InternalKeyComparator* icmp)
-      : CompactionPicker(options, icmp) {}
@@ -97 +73,0 @@ class LevelCompactionPicker : public CompactionPicker {
->>>>>>> abeee9f2
