--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@compaction_picker.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@compaction_picker.no_comments_merged.cc
@@ -3 +2,0 @@
-#include "util/statistics.h"
@@ -407 +406 @@ Compaction* UniversalCompactionPicker::PickCompaction(Version* version,
-        Log(options_->info_log, "Universal: compacting for file num\n");
+        LogToBuffer(log_buffer, "Universal: compacting for file num\n");
@@ -429,2 +427,0 @@ Compaction* UniversalCompactionPicker::PickCompaction(Version* version,
-  MeasureTime(options_->statistics.get(), NUM_FILES_IN_SINGLE_COMPACTION,
-              c->inputs_[0].size());
@@ -463,6 +459,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
-<<<<<<< HEAD
-      Log(logger_, "Universal: file %lu[%d] being compacted, skipping",
-||||||| 8ca30bd51
-      Log(options_->info_log,
-          "Universal: file %lu[%d] being compacted, skipping",
-=======
@@ -471 +461,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
@@ -477,5 +466,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
-<<<<<<< HEAD
-      Log(logger_, "Universal: Possible candidate file %lu[%d].",
-||||||| 8ca30bd51
-      Log(options_->info_log, "Universal: Possible candidate file %lu[%d].",
-=======
@@ -483 +467,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
@@ -519,8 +501,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
-<<<<<<< HEAD
-        Log(logger_, "Universal: Skipping file %lu[%d] with size %lu %d\n",
-            (unsigned long)f->number, i, (unsigned long)f->file_size,
-||||||| 8ca30bd51
-        Log(options_->info_log,
-            "Universal: Skipping file %lu[%d] with size %lu %d\n",
-            (unsigned long)f->number, i, (unsigned long)f->file_size,
-=======
@@ -530 +504,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
@@ -562,7 +535,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
-<<<<<<< HEAD
-    Log(logger_, "Universal: Picking file %lu[%d] with size %lu\n",
-        (unsigned long)f->number, i, (unsigned long)f->file_size);
-||||||| 8ca30bd51
-    Log(options_->info_log, "Universal: Picking file %lu[%d] with size %lu\n",
-        (unsigned long)f->number, i, (unsigned long)f->file_size);
-=======
@@ -571 +537,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalReadAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
@@ -593,7 +558,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
-<<<<<<< HEAD
-    Log(logger_, "Universal: skipping file %lu[%d] compacted %s",
-        (unsigned long)f->number, loop,
-||||||| 8ca30bd51
-    Log(options_->info_log, "Universal: skipping file %lu[%d] compacted %s",
-        (unsigned long)f->number, loop,
-=======
@@ -602 +560,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
@@ -609,7 +566,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
-<<<<<<< HEAD
-  Log(logger_, "Universal: First candidate file %lu[%d] %s",
-      (unsigned long)f->number, start_index, " to reduce size amp.\n");
-||||||| 8ca30bd51
-  Log(options_->info_log, "Universal: First candidate file %lu[%d] %s",
-      (unsigned long)f->number, start_index, " to reduce size amp.\n");
-=======
@@ -618 +568,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
@@ -624,7 +573,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
-<<<<<<< HEAD
-      Log(logger_, "Universal: Possible candidate file %lu[%d] %s.",
-          (unsigned long)f->number, loop,
-||||||| 8ca30bd51
-      Log(options_->info_log, "Universal: Possible candidate file %lu[%d] %s.",
-          (unsigned long)f->number, loop,
-=======
@@ -634 +576,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
@@ -647,8 +589 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
-<<<<<<< HEAD
-    Log(logger_,
-||||||| 8ca30bd51
-    Log(options_->info_log,
-=======
-    LogToBuffer(
-        log_buffer,
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
+    LogToBuffer(log_buffer,
@@ -660,8 +596 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
-<<<<<<< HEAD
-    Log(logger_,
-||||||| 8ca30bd51
-    Log(options_->info_log,
-=======
-    LogToBuffer(
-        log_buffer,
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
+    LogToBuffer(log_buffer,
@@ -681,8 +610,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
-<<<<<<< HEAD
-    Log(logger_, "Universal: size amp picking file %lu[%d] with size %lu",
-        (unsigned long)f->number, index, (unsigned long)f->file_size);
-||||||| 8ca30bd51
-    Log(options_->info_log,
-        "Universal: size amp picking file %lu[%d] with size %lu",
-        (unsigned long)f->number, index, (unsigned long)f->file_size);
-=======
@@ -692 +613,0 @@ Compaction* UniversalCompactionPicker::PickCompactionUniversalSizeAmp(
->>>>>>> abeee9f2cb6948ab95cc3645d613cc498ac3ff70
