diff --git a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_mergegen.h b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_merged.h
index 92a7c9b..4da7523 100644
--- a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_mergegen.h
+++ b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_merged.h
@@ -176,0 +177,9 @@ class VersionSet {
+  uint64_t MinLogNumber() const {
+    uint64_t min_log_num = std::numeric_limits<uint64_t>::max();
+    for (auto cfd : *column_family_set_) {
+      if (min_log_num > cfd->GetLogNumber()) {
+        min_log_num = cfd->GetLogNumber();
+      }
+    }
+    return min_log_num;
+  }
