--- a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_conflict.h
+++ b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_merged.h
@@ -177 +176,0 @@ class VersionSet {
-<<<<<<< HEAD
@@ -187,17 +185,0 @@ class VersionSet {
-||||||| 8ca30bd51
-  int NumberLevels() const { return num_levels_; }
-  Compaction* PickCompaction();
-  Compaction* CompactRange(int input_level,
-                           int output_level,
-                           const InternalKey* begin,
-                           const InternalKey* end,
-                           InternalKey** compaction_end);
-=======
-  int NumberLevels() const { return num_levels_; }
-  Compaction* PickCompaction(LogBuffer* log_buffer);
-  Compaction* CompactRange(int input_level,
-                           int output_level,
-                           const InternalKey* begin,
-                           const InternalKey* end,
-                           InternalKey** compaction_end);
->>>>>>> abeee9f2
