diff --git a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_mergegen.h b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_merged.h
index ab38285..afebc09 100644
--- a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_mergegen.h
+++ b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_merged.h
@@ -117,0 +118 @@ struct DBOptions {
+  InfoLogLevel info_log_level;
