--- a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_conflict.h
+++ b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_merged.h
@@ -55,15 +54,0 @@ struct ColumnFamilyOptions {
-<<<<<<< HEAD
-||||||| 8ca30bd51
-  bool create_if_missing;
-  bool error_if_exists;
-  bool paranoid_checks;
-  Env* env;
-  shared_ptr<Logger> info_log;
-=======
-  bool create_if_missing;
-  bool error_if_exists;
-  bool paranoid_checks;
-  Env* env;
-  shared_ptr<Logger> info_log;
-  InfoLogLevel info_log_level;
->>>>>>> abeee9f2
@@ -132,0 +118 @@ struct DBOptions {
+  InfoLogLevel info_log_level;
