diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_merged.h
index f8da674..afebc09 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_merged.h
@@ -29,6 +29,2 @@ enum CompressionType : char {
-  kNoCompression = 0x0,
-  kSnappyCompression = 0x1,
-  kZlibCompression = 0x2,
-  kBZip2Compression = 0x3,
-  kLZ4Compression = 0x4,
-  kLZ4HCCompression = 0x5
+  kNoCompression = 0x0, kSnappyCompression = 0x1, kZlibCompression = 0x2,
+  kBZip2Compression = 0x3, kLZ4Compression = 0x4, kLZ4HCCompression = 0x5
@@ -121,0 +118 @@ struct DBOptions {
+  InfoLogLevel info_log_level;
@@ -148 +145,6 @@ struct DBOptions {
-  enum { NONE, NORMAL, SEQUENTIAL, WILLNEED } access_hint_on_compaction_start;
+  enum {
+    NONE,
+    NORMAL,
+    SEQUENTIAL,
+    WILLNEED
+  } access_hint_on_compaction_start;
@@ -157 +159,3 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
-  InfoLogLevel info_log_level;
+  Options() :
+    DBOptions(),
+    ColumnFamilyOptions() {}
