--- a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/util@options.no_comments_conflict.cc
+++ b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/util@options.no_comments_merged.cc
@@ -23,15 +22,0 @@ ColumnFamilyOptions::ColumnFamilyOptions()
-<<<<<<< HEAD
-||||||| 8ca30bd51
-      create_if_missing(false),
-      error_if_exists(false),
-      paranoid_checks(false),
-      env(Env::Default()),
-      info_log(nullptr),
-=======
-      create_if_missing(false),
-      error_if_exists(false),
-      paranoid_checks(false),
-      env(Env::Default()),
-      info_log(nullptr),
-      info_log_level(INFO),
->>>>>>> abeee9f2
@@ -150,0 +136 @@ DBOptions::DBOptions()
+      info_log_level(INFO),
@@ -186,0 +173 @@ DBOptions::DBOptions(const Options& options)
+      info_log_level(options.info_log_level),
