--- a/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/db@version_set.no_comments_mergegen.cc
+++ b/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/db@version_set.no_comments_merged.cc
@@ -1183,0 +1184,16 @@ if (edit->IsColumnFamilyManipulation()) {
+    LogAndApplyCFHelper(edit);
+    batch_edits.push_back(edit);
+  } else {
+    v = new Version(column_family_data, this, current_version_number_++);
+    builder.reset(new Builder(column_family_data));
+    for (const auto& writer : manifest_writers_) {
+      if (writer->edit->IsColumnFamilyManipulation() ||
+          writer->cfd->GetID() != column_family_data->GetID()) {
+        break;
+      }
+      last_writer = writer;
+      LogAndApplyHelper(column_family_data, builder.get(), v, last_writer->edit,
+                        mu);
+      batch_edits.push_back(last_writer->edit);
+    }
+    builder->SaveTo(v);
@@ -1285,0 +1302,23 @@ if (edit->is_column_family_add_) {
+      assert(batch_edits.size() == 1);
+      assert(options != nullptr);
+      CreateColumnFamily(*options, edit);
+    } else if (edit->is_column_family_drop_) {
+      assert(batch_edits.size() == 1);
+      column_family_data->SetDropped();
+      if (column_family_data->Unref()) {
+        delete column_family_data;
+      }
+    } else {
+      uint64_t max_log_number_in_batch = 0;
+      for (auto& e : batch_edits) {
+        if (e->has_log_number_) {
+          max_log_number_in_batch =
+              std::max(max_log_number_in_batch, e->log_number_);
+        }
+      }
+      if (max_log_number_in_batch != 0) {
+        assert(column_family_data->GetLogNumber() < max_log_number_in_batch);
+        column_family_data->SetLogNumber(max_log_number_in_batch);
+      }
+      AppendVersion(column_family_data, v);
+    }
@@ -1860 +1913,0 @@ Status VersionSet::WriteSnapshot(log::Writer* log) {
-  edit.SetLogNumber(log_number_);
