diff --git a/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_mergegen.cc b/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_merged.cc
index 7376ba4..fe60046 100644
--- a/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_mergegen.cc
+++ b/./output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_merged.cc
@@ -773 +773,2 @@ class StressTest {
-streamers[i] = db_->NewIterator(readoptionscopy[i], column_family); iters[i]->Seek(prefix_slices[i]);
+      iters[i] = db_->NewIterator(readoptionscopy[i], column_family);
+      iters[i]->Seek(prefix_slices[i]);
@@ -918 +919,2 @@ streamers[i] = db_->NewIterator(readoptionscopy[i], column_family); iters[i]->Se
-read_opts.prefix_seek = true; Iterator* iter = db_->NewIterator(read_opts, column_family);
+          read_opts.prefix_seek = true;
+          Iterator* iter = db_->NewIterator(read_opts, column_family);
