diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_merged.cc
index 73af13e..fe60046 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a782bb98-9b8a2b52-2bad3cb0/tools@db_stress.no_comments_merged.cc
@@ -13,0 +14 @@
+#include "rocksdb/statistics.h"
@@ -28 +29,3 @@ static bool ValidateUint32Range(const char* flagname, uint64_t value) {
-    fprintf(stderr, "Invalid value for --%s: %lu, overflow\n", flagname,
+    fprintf(stderr,
+            "Invalid value for --%s: %lu, overflow\n",
+            flagname,
@@ -34,3 +37,84 @@ static bool ValidateUint32Range(const char* flagname, uint64_t value) {
-static const bool FLAGS_purge_redundant_percent_dummy __attribute__((unused)) =
-    google::RegisterFlagValidator(&FLAGS_purge_redundant_percent,
-                                  &ValidateInt32Percent);
+DEFINE_uint64(seed, 2341234, "Seed for PRNG");
+static const bool FLAGS_seed_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_seed, &ValidateUint32Range);
+DEFINE_int64(max_key, 1 * KB* KB,
+             "Max number of key/values to place in database");
+DEFINE_int32(column_families, 10, "Number of column families");
+DEFINE_bool(test_batches_snapshots, false,
+            "If set, the test uses MultiGet(), Multiut() and MultiDelete()"
+            " which read/write/delete multiple keys in a batch. In this mode,"
+            " we do not verify db content by comparing the content with the "
+            "pre-allocated array. Instead, we do partial verification inside"
+            " MultiGet() by checking various values in a batch. Benefit of"
+            " this mode:\n"
+            "\t(a) No need to acquire mutexes during writes (less cache "
+            "flushes in multi-core leading to speed up)\n"
+            "\t(b) No long validation at the end (more speed up)\n"
+            "\t(c) Test snapshot and atomicity of batch writes");
+DEFINE_int32(threads, 32, "Number of concurrent threads to run.");
+DEFINE_int32(ttl, -1,
+             "Opens the db with this ttl value if this is not -1. "
+             "Carefully specify a large value such that verifications on "
+             "deleted values don't fail");
+DEFINE_int32(value_size_mult, 8,
+             "Size of value will be this number times rand_int(1,3) bytes");
+DEFINE_bool(verify_before_write, false, "Verify before write");
+DEFINE_bool(histogram, false, "Print histogram of operation timings");
+DEFINE_bool(destroy_db_initially, true,
+            "Destroys the database dir before start if this is true");
+DEFINE_bool (verbose, false, "Verbose");
+DEFINE_int32(write_buffer_size, rocksdb::Options().write_buffer_size,
+             "Number of bytes to buffer in memtable before compacting");
+DEFINE_int32(max_write_buffer_number,
+             rocksdb::Options().max_write_buffer_number,
+             "The number of in-memory memtables. "
+             "Each memtable is of size FLAGS_write_buffer_size.");
+DEFINE_int32(min_write_buffer_number_to_merge,
+             rocksdb::Options().min_write_buffer_number_to_merge,
+             "The minimum number of write buffers that will be merged together "
+             "before writing to storage. This is cheap because it is an "
+             "in-memory merge. If this feature is not enabled, then all these "
+             "write buffers are flushed to L0 as separate files and this "
+             "increases read amplification because a get request has to check "
+             "in all of these files. Also, an in-memory merge may result in "
+             "writing less data to storage if there are duplicate records in"
+             " each of these individual write buffers.");
+DEFINE_int32(open_files, rocksdb::Options().max_open_files,
+             "Maximum number of files to keep open at the same time "
+             "(use default if == 0)");
+DEFINE_int64(compressed_cache_size, -1,
+             "Number of bytes to use as a cache of compressed data."
+             " Negative means use default settings.");
+DEFINE_int32(compaction_style, rocksdb::Options().compaction_style, "");
+DEFINE_int32(level0_file_num_compaction_trigger,
+             rocksdb::Options().level0_file_num_compaction_trigger,
+             "Level0 compaction start trigger");
+DEFINE_int32(level0_slowdown_writes_trigger,
+             rocksdb::Options().level0_slowdown_writes_trigger,
+             "Number of files in level-0 that will slow down writes");
+DEFINE_int32(level0_stop_writes_trigger,
+             rocksdb::Options().level0_stop_writes_trigger,
+             "Number of files in level-0 that will trigger put stop.");
+DEFINE_int32(block_size, rocksdb::Options().block_size,
+             "Number of bytes in a block.");
+DEFINE_int32(max_background_compactions,
+             rocksdb::Options().max_background_compactions,
+             "The maximum number of concurrent background compactions "
+             "that can occur in parallel.");
+DEFINE_int32(max_background_flushes, rocksdb::Options().max_background_flushes,
+             "The maximum number of concurrent background flushes "
+             "that can occur in parallel.");
+DEFINE_int32(universal_size_ratio, 0, "The ratio of file sizes that trigger"
+             " compaction in universal style");
+DEFINE_int32(universal_min_merge_width, 0, "The minimum number of files to "
+             "compact in universal style compaction");
+DEFINE_int32(universal_max_merge_width, 0, "The max number of files to compact"
+             " in universal style compaction");
+DEFINE_int32(universal_max_size_amplification_percent, 0,
+             "The max size amplification for universal style compaction");
+DEFINE_int32(clear_column_family_one_in, 1000000,
+             "With a chance of 1/N, delete a column family and then recreate "
+             "it again. If N == 0, never drop/create column families. "
+             "When test_batches_snapshots is true, this flag has no effect");
+DEFINE_int64(cache_size, 2 * KB * KB * KB,
+             "Number of bytes to use as a cache of uncompressed data.");
@@ -39,2 +123,2 @@ static bool ValidateInt32Positive(const char* flagname, int32_t value) {
-    fprintf(stderr, "Invalid value for --%s: %d, must be >=0\n", flagname,
-            value);
+    fprintf(stderr, "Invalid value for --%s: %d, must be >=0\n",
+            flagname, value);
@@ -44,0 +129,10 @@ static bool ValidateInt32Positive(const char* flagname, int32_t value) {
+DEFINE_int32(reopen, 10, "Number of times database reopens");
+static const bool FLAGS_reopen_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_reopen, &ValidateInt32Positive);
+DEFINE_int32(bloom_bits, 10, "Bloom filter bits per key. "
+             "Negative means use default settings.");
+DEFINE_string(db, "", "Use the db with the following name.");
+DEFINE_bool(verify_checksum, false,
+            "Verify checksum for every block read from storage");
+DEFINE_bool(mmap_read, rocksdb::EnvOptions().use_mmap_reads,
+            "Allow reads to occur via mmap-ing files");
@@ -45,0 +140,11 @@ static std::shared_ptr<rocksdb::Statistics> dbstats;
+DEFINE_bool(statistics, false, "Create database statistics");
+DEFINE_bool(sync, false, "Sync all writes to disk");
+DEFINE_bool(disable_data_sync, false,
+            "If true, do not wait until data is synced to disk.");
+DEFINE_bool(use_fsync, false, "If true, issue fsync instead of fdatasync");
+DEFINE_int32(kill_random_test, 0,
+             "If non-zero, kill at various points in source code with "
+             "probability 1/this");
+static const bool FLAGS_kill_random_test_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_kill_random_test,
+                                  &ValidateInt32Positive);
@@ -46,0 +152,8 @@ extern int rocksdb_kill_odds;
+DEFINE_bool(disable_wal, false, "If true, do not write WAL for write.");
+DEFINE_int32(target_file_size_base, 64 * KB,
+             "Target level-1 file size for compaction");
+DEFINE_int32(target_file_size_multiplier, 1,
+             "A multiplier to compute targe level-N file size (N >= 2)");
+DEFINE_uint64(max_bytes_for_level_base, 256 * KB, "Max bytes for level-1");
+DEFINE_int32(max_bytes_for_level_multiplier, 2,
+             "A multiplier to compute max bytes for level-N (N >= 2)");
@@ -49,2 +162,2 @@ static bool ValidateInt32Percent(const char* flagname, int32_t value) {
-    fprintf(stderr, "Invalid value for --%s: %d, 0<= pct <=100 \n", flagname,
-            value);
+    fprintf(stderr, "Invalid value for --%s: %d, 0<= pct <=100 \n",
+            flagname, value);
@@ -54,0 +168,26 @@ static bool ValidateInt32Percent(const char* flagname, int32_t value) {
+DEFINE_int32(readpercent, 10,
+             "Ratio of reads to total workload (expressed as a percentage)");
+static const bool FLAGS_readpercent_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_readpercent, &ValidateInt32Percent);
+DEFINE_int32(prefixpercent, 20,
+             "Ratio of prefix iterators to total workload (expressed as a"
+             " percentage)");
+static const bool FLAGS_prefixpercent_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_prefixpercent, &ValidateInt32Percent);
+DEFINE_int32(writepercent, 45,
+             " Ratio of deletes to total workload (expressed as a percentage)");
+static const bool FLAGS_writepercent_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_writepercent, &ValidateInt32Percent);
+DEFINE_int32(delpercent, 15,
+             "Ratio of deletes to total workload (expressed as a percentage)");
+static const bool FLAGS_delpercent_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_delpercent, &ValidateInt32Percent);
+DEFINE_int32(iterpercent, 10, "Ratio of iterations to total workload"
+             " (expressed as a percentage)");
+static const bool FLAGS_iterpercent_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_iterpercent, &ValidateInt32Percent);
+DEFINE_uint64(num_iterations, 10, "Number of iterations per MultiIterate run");
+static const bool FLAGS_num_iterations_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_num_iterations, &ValidateUint32Range);
+DEFINE_bool(disable_seek_compaction, false,
+            "Option to disable compation triggered by read.");
@@ -71,0 +211,2 @@ enum rocksdb::CompressionType StringToCompressionType(const char* ctype) {
+DEFINE_string(compression_type, "snappy",
+              "Algorithm to use to compress the database");
@@ -73,0 +215 @@ static enum rocksdb::CompressionType FLAGS_compression_type_e =
+DEFINE_string(hdfs, "", "Name of hdfs environment");
@@ -75 +217,20 @@ static rocksdb::Env* FLAGS_env = rocksdb::Env::Default();
-enum RepFactory { kSkipList, kHashSkipList, kVectorRep };
+DEFINE_uint64(ops_per_thread, 1200000, "Number of operations per thread.");
+static const bool FLAGS_ops_per_thread_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_ops_per_thread, &ValidateUint32Range);
+DEFINE_uint64(log2_keys_per_lock, 2, "Log2 of number of keys per lock");
+static const bool FLAGS_log2_keys_per_lock_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_log2_keys_per_lock,
+                                  &ValidateUint32Range);
+DEFINE_int32(purge_redundant_percent, 50,
+             "Percentage of times we want to purge redundant keys in memory "
+             "before flushing");
+static const bool FLAGS_purge_redundant_percent_dummy __attribute__((unused)) =
+    google::RegisterFlagValidator(&FLAGS_purge_redundant_percent,
+                                  &ValidateInt32Percent);
+DEFINE_bool(filter_deletes, false, "On true, deletes use KeyMayExist to drop"
+            " the delete if key not present");
+enum RepFactory {
+  kSkipList,
+  kHashSkipList,
+  kVectorRep
+};
@@ -87,0 +249 @@ static enum RepFactory FLAGS_rep_factory;
+DEFINE_string(memtablerep, "prefix_hash", "");
@@ -95,0 +258 @@ static bool ValidatePrefixSize(const char* flagname, int32_t value) {
+DEFINE_int32(prefix_size, 7, "Control the prefix size for HashSkipListRep");
@@ -97,0 +261,2 @@ static const bool FLAGS_prefix_size_dummy =
+DEFINE_bool(use_merge, false, "On true, replaces all writes with a Merge "
+            "that behaves like a Put");
@@ -182,14 +347,7 @@ class Stats {
-      if (next_report_ < 1000)
-        next_report_ += 100;
-      else if (next_report_ < 5000)
-        next_report_ += 500;
-      else if (next_report_ < 10000)
-        next_report_ += 1000;
-      else if (next_report_ < 50000)
-        next_report_ += 5000;
-      else if (next_report_ < 100000)
-        next_report_ += 10000;
-      else if (next_report_ < 500000)
-        next_report_ += 50000;
-      else
-        next_report_ += 100000;
+      if (next_report_ < 1000) next_report_ += 100;
+      else if (next_report_ < 5000) next_report_ += 500;
+      else if (next_report_ < 10000) next_report_ += 1000;
+      else if (next_report_ < 50000) next_report_ += 5000;
+      else if (next_report_ < 100000) next_report_ += 10000;
+      else if (next_report_ < 500000) next_report_ += 50000;
+      else next_report_ += 100000;
@@ -211,3 +369,9 @@ class Stats {
-  void AddIterations(int n) { iterations_ += n; }
-  void AddDeletes(int n) { deletes_ += n; }
-  void AddErrors(int n) { errors_ += n; }
+  void AddIterations(int n) {
+    iterations_ += n;
+  }
+  void AddDeletes(int n) {
+    deletes_ += n;
+  }
+  void AddErrors(int n) {
+    errors_ += n;
+  }
@@ -225,2 +389,2 @@ class Stats {
-    fprintf(stdout, "%.3f micros/op %ld ops/sec\n", seconds_ * 1e6 / done_,
-            (long)throughput);
+    fprintf(stdout, "%.3f micros/op %ld ops/sec\n",
+            seconds_ * 1e6 / done_, (long)throughput);
@@ -231,2 +395,2 @@ class Stats {
-    fprintf(stdout, "%-12s: %ld read and %ld found the key\n", "", gets_,
-            founds_);
+    fprintf(stdout, "%-12s: %ld read and %ld found the key\n", "",
+            gets_, founds_);
@@ -247,2 +411,2 @@ class SharedState {
-  explicit SharedState(StressTest* stress_test)
-      : cv_(&mu_),
+  explicit SharedState(StressTest* stress_test) :
+      cv_(&mu_),
@@ -279,17 +443,51 @@ class SharedState {
-  port::Mutex* GetMutex() { return &mu_; }
-  port::CondVar* GetCondVar() { return &cv_; }
-  StressTest* GetStressTest() const { return stress_test_; }
-  long GetMaxKey() const { return max_key_; }
-  uint32_t GetNumThreads() const { return num_threads_; }
-  void IncInitialized() { num_initialized_++; }
-  void IncOperated() { num_populated_++; }
-  void IncDone() { num_done_++; }
-  void IncVotedReopen() { vote_reopen_ = (vote_reopen_ + 1) % num_threads_; }
-  bool AllInitialized() const { return num_initialized_ >= num_threads_; }
-  bool AllOperated() const { return num_populated_ >= num_threads_; }
-  bool AllDone() const { return num_done_ >= num_threads_; }
-  bool AllVotedReopen() { return (vote_reopen_ == 0); }
-  void SetStart() { start_ = true; }
-  void SetStartVerify() { start_verify_ = true; }
-  bool Started() const { return start_; }
-  bool VerifyStarted() const { return start_verify_; }
+  port::Mutex* GetMutex() {
+    return &mu_;
+  }
+  port::CondVar* GetCondVar() {
+    return &cv_;
+  }
+  StressTest* GetStressTest() const {
+    return stress_test_;
+  }
+  long GetMaxKey() const {
+    return max_key_;
+  }
+  uint32_t GetNumThreads() const {
+    return num_threads_;
+  }
+  void IncInitialized() {
+    num_initialized_++;
+  }
+  void IncOperated() {
+    num_populated_++;
+  }
+  void IncDone() {
+    num_done_++;
+  }
+  void IncVotedReopen() {
+    vote_reopen_ = (vote_reopen_ + 1) % num_threads_;
+  }
+  bool AllInitialized() const {
+    return num_initialized_ >= num_threads_;
+  }
+  bool AllOperated() const {
+    return num_populated_ >= num_threads_;
+  }
+  bool AllDone() const {
+    return num_done_ >= num_threads_;
+  }
+  bool AllVotedReopen() {
+    return (vote_reopen_ == 0);
+  }
+  void SetStart() {
+    start_ = true;
+  }
+  void SetStartVerify() {
+    start_verify_ = true;
+  }
+  bool Started() const {
+    return start_;
+  }
+  bool VerifyStarted() const {
+    return start_verify_;
+  }
@@ -298,0 +497,13 @@ class SharedState {
+  void LockColumnFamily(int cf) {
+    for (auto& mutex : key_locks_[cf]) {
+      mutex.Lock();
+    }
+  }
+  void UnlockColumnFamily(int cf) {
+    for (auto& mutex : key_locks_[cf]) {
+      mutex.Unlock();
+    }
+  }
+  void ClearColumnFamily(int cf) {
+    std::fill(values_[cf].begin(), values_[cf].end(), SENTINEL);
+  }
@@ -321,14 +531,0 @@ class SharedState {
- public:
-  void LockColumnFamily(int cf) {
-    for (auto& mutex : key_locks_[cf]) {
-      mutex.Lock();
-    }
-  }
-  void UnlockColumnFamily(int cf) {
-    for (auto& mutex : key_locks_[cf]) {
-      mutex.Unlock();
-    }
-  }
-  void ClearColumnFamily(int cf) {
-    std::fill(values_[cf].begin(), values_[cf].end(), SENTINEL);
-  }
@@ -343 +540,4 @@ struct ThreadState {
-      : tid(index), rand(1000 + index + shared->GetSeed()), shared(shared) {}
+      : tid(index),
+        rand(1000 + index + shared->GetSeed()),
+        shared(shared) {
+  }
@@ -378,8 +577,0 @@ class StressTest {
-  {
-    for (auto cf : column_families_) {
-      delete cf;
-    }
-    column_families_.clear();
-    delete db_;
-    delete filter_policy_;
-  }
@@ -477,2 +669,4 @@ class StressTest {
-    std::string keys[10] = {"9", "8", "7", "6", "5", "4", "3", "2", "1", "0"};
-    std::string values[10] = {"9", "8", "7", "6", "5", "4", "3", "2", "1", "0"};
+    std::string keys[10] = {"9", "8", "7", "6", "5",
+                            "4", "3", "2", "1", "0"};
+    std::string values[10] = {"9", "8", "7", "6", "5",
+                              "4", "3", "2", "1", "0"};
@@ -501 +695 @@ class StressTest {
-  Status MultiPrefixScan(ThreadState* thread, const ReadOptions& readoptions,
+  Status MultiDelete(ThreadState* thread, const WriteOptions& writeoptions,
@@ -502,0 +697,62 @@ class StressTest {
+    std::string keys[10] = {"9", "7", "5", "3", "1",
+                            "8", "6", "4", "2", "0"};
+    WriteBatch batch;
+    Status s;
+    for (int i = 0; i < 10; i++) {
+      keys[i] += key.ToString();
+      batch.Delete(column_family, keys[i]);
+    }
+    s = db_->Write(writeoptions, &batch);
+    if (!s.ok()) {
+      fprintf(stderr, "multidelete error: %s\n", s.ToString().c_str());
+      thread->stats.AddErrors(1);
+    } else {
+      thread->stats.AddDeletes(10);
+    }
+    return s;
+  }
+  Status MultiGet(ThreadState* thread, const ReadOptions& readoptions,
+                  ColumnFamilyHandle* column_family, const Slice& key,
+                  std::string* value) {
+    std::string keys[10] = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"};
+    Slice key_slices[10];
+    std::string values[10];
+    ReadOptions readoptionscopy = readoptions;
+    readoptionscopy.snapshot = db_->GetSnapshot();
+    Status s;
+    for (int i = 0; i < 10; i++) {
+      keys[i] += key.ToString();
+      key_slices[i] = keys[i];
+      s = db_->Get(readoptionscopy, column_family, key_slices[i], value);
+      if (!s.ok() && !s.IsNotFound()) {
+        fprintf(stderr, "get error: %s\n", s.ToString().c_str());
+        values[i] = "";
+        thread->stats.AddErrors(1);
+      } else if (s.IsNotFound()) {
+        values[i] = "";
+        thread->stats.AddGets(1, 0);
+      } else {
+        values[i] = *value;
+        char expected_prefix = (keys[i])[0];
+        char actual_prefix = (values[i])[0];
+        if (actual_prefix != expected_prefix) {
+          fprintf(stderr, "error expected prefix = %c actual = %c\n",
+                  expected_prefix, actual_prefix);
+        }
+        (values[i])[0] = ' ';
+        thread->stats.AddGets(1, 1);
+      }
+    }
+    db_->ReleaseSnapshot(readoptionscopy.snapshot);
+    for (int i = 1; i < 10; i++) {
+      if (values[i] != values[0]) {
+        fprintf(stderr, "error : inconsistent values for key %s: %s, %s\n",
+                key.ToString().c_str(), values[0].c_str(),
+                values[i].c_str());
+      }
+    }
+    return s;
+  }
+  Status MultiPrefixScan(ThreadState* thread, const ReadOptions& readoptions,
+                         ColumnFamilyHandle* column_family,
+                         const Slice& key) {
@@ -517 +772,0 @@ class StressTest {
-<<<<<<< HEAD
@@ -519,6 +773,0 @@ class StressTest {
-      iters[i]->SeekToFirst();
-||||||| 2bad3cb0d
-      iters[i] = db_->NewIterator(readoptionscopy[i]);
-      iters[i]->SeekToFirst();
-=======
-      iters[i] = db_->NewIterator(readoptionscopy[i]);
@@ -526 +774,0 @@ class StressTest {
->>>>>>> 9b8a2b52d451dd1a162ef34c59df907f4b476b59
@@ -546 +794,2 @@ class StressTest {
-                  prefixes[i].c_str(), values[0].c_str(), values[i].c_str());
+                  prefixes[i].c_str(), values[0].c_str(),
+                  values[i].c_str());
@@ -611 +860,2 @@ class StressTest {
-          } else {
+          }
+          else {
@@ -669,7 +918,0 @@ class StressTest {
-<<<<<<< HEAD
-          read_opts.prefix = &prefix;
-          Iterator* iter = db_->NewIterator(read_opts, column_family);
-||||||| 2bad3cb0d
-          read_opts.prefix = &prefix;
-          Iterator* iter = db_->NewIterator(read_opts);
-=======
@@ -677,2 +920 @@ class StressTest {
-          Iterator* iter = db_->NewIterator(read_opts);
->>>>>>> 9b8a2b52d451dd1a162ef34c59df907f4b476b59
+          Iterator* iter = db_->NewIterator(read_opts, column_family);
@@ -818,0 +1061,21 @@ class StressTest {
+  static void PrintKeyValue(int cf, uint32_t key, const char* value,
+                            size_t sz) {
+    if (!FLAGS_verbose) {
+      return;
+    }
+    fprintf(stdout, "[CF %d] %u ==> (%u) ", cf, key, (unsigned int)sz);
+    for (size_t i = 0; i < sz; i++) {
+      fprintf(stdout, "%X", value[i]);
+    }
+    fprintf(stdout, "\n");
+  }
+  static size_t GenerateValue(uint32_t rand, char *v, size_t max_sz) {
+    size_t value_sz = ((rand % 3) + 1) * FLAGS_value_size_mult;
+    assert(value_sz <= max_sz && value_sz >= sizeof(uint32_t));
+    *((uint32_t*)v) = rand;
+    for (size_t i=sizeof(uint32_t); i < value_sz; i++) {
+      v[i] = (char)(rand ^ i);
+    }
+    v[value_sz] = '\0';
+    return value_sz;
+  }
@@ -828 +1091,2 @@ class StressTest {
-    fprintf(stdout, "Ops per thread      : %lu\n",
+    fprintf(stdout,
+            "Ops per thread      : %lu\n",
@@ -841 +1105,2 @@ class StressTest {
-    fprintf(stdout, "Iterations          : %lu\n",
+    fprintf(stdout,
+            "Iterations          : %lu\n",
@@ -843 +1108,2 @@ class StressTest {
-    fprintf(stdout, "Max key             : %lu\n",
+    fprintf(stdout,
+            "Max key             : %lu\n",
@@ -848 +1114,2 @@ class StressTest {
-    fprintf(stdout, "Batches/snapshots   : %d\n", FLAGS_test_batches_snapshots);
+    fprintf(stdout, "Batches/snapshots   : %d\n",
+            FLAGS_test_batches_snapshots);
@@ -851 +1118,2 @@ class StressTest {
-    fprintf(stdout, "Deletes use filter  : %d\n", FLAGS_filter_deletes);
+    fprintf(stdout, "Deletes use filter  : %d\n",
+            FLAGS_filter_deletes);
@@ -928,2 +1196 @@ class StressTest {
-      fprintf(
-          stderr,
+      fprintf(stderr,
@@ -1050,0 +1318 @@ class StressTest {
+ private:
@@ -1060,80 +1327,0 @@ class StressTest {
-  Status MultiDelete(ThreadState* thread, const WriteOptions& writeoptions,
-                     ColumnFamilyHandle* column_family, const Slice& key) {
-    std::string keys[10] = {"9", "7", "5", "3", "1", "8", "6", "4", "2", "0"};
-    WriteBatch batch;
-    Status s;
-    for (int i = 0; i < 10; i++) {
-      keys[i] += key.ToString();
-      batch.Delete(column_family, keys[i]);
-    }
-    s = db_->Write(writeoptions, &batch);
-    if (!s.ok()) {
-      fprintf(stderr, "multidelete error: %s\n", s.ToString().c_str());
-      thread->stats.AddErrors(1);
-    } else {
-      thread->stats.AddDeletes(10);
-    }
-    return s;
-  }
-  Status MultiGet(ThreadState* thread, const ReadOptions& readoptions,
-                  ColumnFamilyHandle* column_family, const Slice& key,
-                  std::string* value) {
-    std::string keys[10] = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"};
-    Slice key_slices[10];
-    std::string values[10];
-    ReadOptions readoptionscopy = readoptions;
-    readoptionscopy.snapshot = db_->GetSnapshot();
-    Status s;
-    for (int i = 0; i < 10; i++) {
-      keys[i] += key.ToString();
-      key_slices[i] = keys[i];
-      s = db_->Get(readoptionscopy, column_family, key_slices[i], value);
-      if (!s.ok() && !s.IsNotFound()) {
-        fprintf(stderr, "get error: %s\n", s.ToString().c_str());
-        values[i] = "";
-        thread->stats.AddErrors(1);
-      } else if (s.IsNotFound()) {
-        values[i] = "";
-        thread->stats.AddGets(1, 0);
-      } else {
-        values[i] = *value;
-        char expected_prefix = (keys[i])[0];
-        char actual_prefix = (values[i])[0];
-        if (actual_prefix != expected_prefix) {
-          fprintf(stderr, "error expected prefix = %c actual = %c\n",
-                  expected_prefix, actual_prefix);
-        }
-        (values[i])[0] = ' ';
-        thread->stats.AddGets(1, 1);
-      }
-    }
-    db_->ReleaseSnapshot(readoptionscopy.snapshot);
-    for (int i = 1; i < 10; i++) {
-      if (values[i] != values[0]) {
-        fprintf(stderr, "error : inconsistent values for key %s: %s, %s\n",
-                key.ToString().c_str(), values[0].c_str(), values[i].c_str());
-      }
-    }
-    return s;
-  }
-  static void PrintKeyValue(int cf, uint32_t key, const char* value,
-                            size_t sz) {
-    if (!FLAGS_verbose) {
-      return;
-    }
-    fprintf(stdout, "[CF %d] %u ==> (%u) ", cf, key, (unsigned int)sz);
-    for (size_t i = 0; i < sz; i++) {
-      fprintf(stdout, "%X", value[i]);
-    }
-    fprintf(stdout, "\n");
-  }
-  static size_t GenerateValue(uint32_t rand, char* v, size_t max_sz) {
-    size_t value_sz = ((rand % 3) + 1) * FLAGS_value_size_mult;
-    assert(value_sz <= max_sz && value_sz >= sizeof(uint32_t));
-    *((uint32_t*)v) = rand;
-    for (size_t i = sizeof(uint32_t); i < value_sz; i++) {
-      v[i] = (char)(rand ^ i);
-    }
-    v[value_sz] = '\0';
-    return value_sz;
-  }
@@ -1168,2 +1356,2 @@ int main(int argc, char** argv) {
-  if ((FLAGS_readpercent + FLAGS_prefixpercent + FLAGS_writepercent +
-       FLAGS_delpercent + FLAGS_iterpercent) != 100) {
+  if ((FLAGS_readpercent + FLAGS_prefixpercent +
+       FLAGS_writepercent + FLAGS_delpercent + FLAGS_iterpercent) != 100) {
@@ -1182 +1370,2 @@ int main(int argc, char** argv) {
-            FLAGS_reopen, (unsigned long)FLAGS_ops_per_thread);
+              FLAGS_reopen,
+              (unsigned long)FLAGS_ops_per_thread);
