diff --git a/./output/rocksdb/59a0c219-0be50ed1-9eaff629/db@db_test.no_comments_conflict.cc b/./output/rocksdb/59a0c219-0be50ed1-9eaff629/db@db_test.no_comments_merged.cc
index ca2aefa..24d7d30 100644
--- a/./output/rocksdb/59a0c219-0be50ed1-9eaff629/db@db_test.no_comments_conflict.cc
+++ b/./output/rocksdb/59a0c219-0be50ed1-9eaff629/db@db_test.no_comments_merged.cc
@@ -7698 +7697,0 @@ TEST_F(DBTest, PauseBackgroundWorkTest) {
-<<<<<<< HEAD
@@ -7831,2 +7829,0 @@ TEST_F(DBTest, WalFilterTest) {
-||||||| 9eaff629e
-=======
@@ -7910 +7906,0 @@ INSTANTIATE_TEST_CASE_P(BloomStatsTestWithParam, BloomStatsTestWithParam,
->>>>>>> 0be50ed1
