--- a/./output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_mergegen.h
+++ b/./output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_merged.h
@@ -1118 +1118,4 @@ void moduleFireServerEvent(uint64_t eid, int subid, void *data);
-int moduleTryServeClientBlockedOnKey(client *c, robj *key); void moduleUnblockClient(client *c); int moduleClientIsBlockedOnKeys(client *c);
+void processModuleLoadingProgressEvent(int is_aof);
+int moduleTryServeClientBlockedOnKey(client *c, robj *key);
+void moduleUnblockClient(client *c);
+int moduleClientIsBlockedOnKeys(client *c);
