--- a/./output/redis/59d52f7f-75fa7879-29252391/src@object.no_comments_mergegen.c
+++ b/./output/redis/59d52f7f-75fa7879-29252391/src@object.no_comments_merged.c
@@ -757 +757,9 @@ void objectCommand(client *c) {
-if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr,"help")) { const char *help[] = { "encoding <key> -- Return the kind of internal representation used in order to store the value associated with a key.", "freq <key> -- Return the access frequency index of the key. The returned integer is proportional to the logarithm of the recent access frequency of the key.", "idletime <key> -- Return the idle time of the key, that is the approximated number of seconds elapsed since the last access to the key.", "refcount <key> -- Return the number of references of the value associated with the specified key.", NULL }; addReplyHelp(c, help); } else if (!strcasecmp(c->argv[1]->ptr,"refcount") && c->argc == 3) {
+    if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr,"help")) {
+        const char *help[] = {
+            "encoding <key> -- Return the kind of internal representation used in order to store the value associated with a key.",
+            "freq <key> -- Return the access frequency index of the key. The returned integer is proportional to the logarithm of the recent access frequency of the key.",
+            "idletime <key> -- Return the idle time of the key, that is the approximated number of seconds elapsed since the last access to the key.",
+            "refcount <key> -- Return the number of references of the value associated with the specified key.",
+            NULL
+        };
+        addReplyHelp(c, help);
