diff --git a/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_conflict.hpp b/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_merged.hpp
index 3000bb9..26f83dd 100644
--- a/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_conflict.hpp
+++ b/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_merged.hpp
@@ -47 +46,0 @@ struct ListConcatFun {
-<<<<<<< HEAD
@@ -66,12 +64,0 @@ struct ConcatOperatorFun {
-||||||| 1fc514d508
-struct SubstringFun {
- static void RegisterFunction(BuiltinFunctions &set);
- static string_t SubstringUnicode(Vector &result, string_t input, int64_t offset, int64_t length);
- static string_t SubstringGrapheme(Vector &result, string_t input, int64_t offset, int64_t length);
-=======
-struct SubstringFun {
- static void RegisterFunction(BuiltinFunctions &set);
- static string_t SubstringASCII(Vector &result, string_t input, int64_t offset, int64_t length);
- static string_t SubstringUnicode(Vector &result, string_t input, int64_t offset, int64_t length);
- static string_t SubstringGrapheme(Vector &result, string_t input, int64_t offset, int64_t length);
->>>>>>> 58d5b727
