diff --git a/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_mergegen.hpp b/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_merged.hpp
index edb63b8..26f83dd 100644
--- a/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_mergegen.hpp
+++ b/./output/duckdb/5e67908d-58d5b727-1fc514d5/src@include@duckdb@function@scalar@string_functions.no_comments_merged.hpp
@@ -47 +47,18 @@ struct ListConcatFun {
-struct ListCatFun { using ALIAS = ListConcatFun; static constexpr const char *Name = "list_cat"; }; struct ArrayConcatFun { using ALIAS = ListConcatFun; static constexpr const char *Name = "array_concat"; }; struct ArrayCatFun { using ALIAS = ListConcatFun; static constexpr const char *Name = "array_cat"; }; struct ConcatOperatorFun { static constexpr const char *Name = "||"; static constexpr const char *Parameters = "list1,list2"; static constexpr const char *Description = "Concatenates two lists."; static constexpr const char *Example = "list_concat([2, 3], [4, 5, 6])"; static ScalarFunction GetFunction();
+struct ListCatFun {
+ using ALIAS = ListConcatFun;
+ static constexpr const char *Name = "list_cat";
+};
+struct ArrayConcatFun {
+ using ALIAS = ListConcatFun;
+ static constexpr const char *Name = "array_concat";
+};
+struct ArrayCatFun {
+ using ALIAS = ListConcatFun;
+ static constexpr const char *Name = "array_cat";
+};
+struct ConcatOperatorFun {
+ static constexpr const char *Name = "||";
+ static constexpr const char *Parameters = "list1,list2";
+ static constexpr const char *Description = "Concatenates two lists.";
+ static constexpr const char *Example = "list_concat([2, 3], [4, 5, 6])";
+ static ScalarFunction GetFunction();
