--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/c79657e5-10c42435-8d3f8f81/src@execution@window_executor.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/c79657e5-10c42435-8d3f8f81/src@execution@window_executor.no_comments_merged.cpp
@@ -5,0 +6,75 @@ namespace duckdb {
+WindowCollection::WindowCollection(BufferManager &buffer_manager, idx_t count, const vector<LogicalType> &types)
+    : all_valid(true), types(types), count(count), buffer_manager(buffer_manager) {
+ if (!types.empty()) {
+  inputs = make_uniq<ColumnDataCollection>(buffer_manager, types);
+ }
+}
+void WindowCollection::GetCollection(idx_t row_idx, ColumnDataCollectionSpec &spec) {
+ if (spec.second && row_idx == spec.first + spec.second->Count()) {
+  return;
+ }
+ lock_guard<mutex> collection_guard(lock);
+ auto collection = make_uniq<ColumnDataCollection>(buffer_manager, types);
+ spec = {row_idx, collection.get()};
+ Range probe {row_idx, collections.size()};
+ auto i = std::upper_bound(ranges.begin(), ranges.end(), probe);
+ ranges.insert(i, probe);
+ collections.emplace_back(std::move(collection));
+}
+void WindowCollection::Combine(bool build_validity) {
+ lock_guard<mutex> collection_guard(lock);
+ if (collections.empty()) {
+  return;
+ }
+ for (auto &range : ranges) {
+  inputs->Combine(*collections[range.second]);
+ }
+ collections.clear();
+ ranges.clear();
+ if (build_validity && !all_valid) {
+  D_ASSERT(inputs.get());
+  validity.Initialize(inputs->Count());
+  WindowCursor cursor(*this);
+  idx_t target_offset = 0;
+  while (cursor.Scan()) {
+   const auto count = cursor.chunk.size();
+   auto &other = FlatVector::Validity(cursor.chunk.data[0]);
+   validity.SliceInPlace(other, target_offset, 0, count);
+   target_offset += count;
+  }
+ }
+}
+WindowBuilder::WindowBuilder(WindowCollection &collection) : collection(collection) {
+}
+void WindowBuilder::Sink(DataChunk &chunk, idx_t input_idx) {
+ if (!sink.second || input_idx < sink.first || sink.first + sink.second->Count() < input_idx) {
+  collection.GetCollection(input_idx, sink);
+  D_ASSERT(sink.second);
+  sink.second->InitializeAppend(appender);
+ }
+ sink.second->Append(appender, chunk);
+ if (all_valid) {
+  UnifiedVectorFormat data;
+  chunk.data[0].ToUnifiedFormat(chunk.size(), data);
+  all_valid = data.validity.AllValid();
+  if (!all_valid) {
+   collection.all_valid = false;
+  }
+ }
+}
+WindowCursor::WindowCursor(const WindowCollection &paged) : paged(paged) {
+ if (paged.GetTypes().empty()) {
+  state.segment_index = 0;
+  state.chunk_index = 0;
+  state.current_row_index = 0;
+  state.next_row_index = paged.size();
+  state.properties = ColumnDataScanProperties::ALLOW_ZERO_COPY;
+  chunk.SetCardinality(state.next_row_index);
+  return;
+ } else if (chunk.data.empty()) {
+  auto &inputs = paged.inputs;
+  D_ASSERT(inputs.get());
+  inputs->InitializeScan(state);
+  inputs->InitializeScanChunk(state, chunk);
+ }
+}
@@ -496,2 +571,2 @@ public:
- () = delete;
- {}
+ ~WindowExecutorBoundsState() override {
+ }
@@ -499,2 +573,0 @@ public:
- () = delete;
- {}
@@ -782,0 +856 @@ public:
+public:
@@ -915,0 +990 @@ public:
+public:
@@ -1157,0 +1233 @@ public:
+public:
@@ -1214,2 +1289,0 @@ void WindowLeadLagExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-<<<<<<< HEAD
-    const auto target_limit = MinValue(partition_end[i], row_end) - row_idx;
@@ -1228,14 +1301,0 @@ void WindowLeadLagExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-|||||||
-    const auto target_limit = MinValue(partition_end[i], row_end) - row_idx;
-    const auto width = MinValue(source_limit, target_limit);
-    auto &source = payload_collection.data[0];
-    VectorOperations::Copy(source, result, index + width, index, i);
-    i += width;
-    row_idx += width;
-=======
-    const auto width = MinValue(source_limit, target_limit);
-    auto &source = payload_collection.data[0];
-    VectorOperations::Copy(source, result, index + width, index, i);
-    i += width;
-    row_idx += width;
->>>>>>> 10c42435f1805ee4415faa5d6da4943e8c98fa55
@@ -1362,75 +1421,0 @@ void WindowNthValueExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-WindowCollection::WindowCollection(BufferManager &buffer_manager, idx_t count, const vector<LogicalType> &types)
-    : all_valid(true), types(types), count(count), buffer_manager(buffer_manager) {
- if (!types.empty()) {
-  inputs = make_uniq<ColumnDataCollection>(buffer_manager, types);
- }
-}
-void WindowCollection::GetCollection(idx_t row_idx, ColumnDataCollectionSpec &spec) {
- if (spec.second && row_idx == spec.first + spec.second->Count()) {
-  return;
- }
- lock_guard<mutex> collection_guard(lock);
- auto collection = make_uniq<ColumnDataCollection>(buffer_manager, types);
- spec = {row_idx, collection.get()};
- Range probe {row_idx, collections.size()};
- auto i = std::upper_bound(ranges.begin(), ranges.end(), probe);
- ranges.insert(i, probe);
- collections.emplace_back(std::move(collection));
-}
-void WindowCollection::Combine(bool build_validity) {
- lock_guard<mutex> collection_guard(lock);
- if (collections.empty()) {
-  return;
- }
- for (auto &range : ranges) {
-  inputs->Combine(*collections[range.second]);
- }
- collections.clear();
- ranges.clear();
- if (build_validity && !all_valid) {
-  D_ASSERT(inputs.get());
-  validity.Initialize(inputs->Count());
-  WindowCursor cursor(*this);
-  idx_t target_offset = 0;
-  while (cursor.Scan()) {
-   const auto count = cursor.chunk.size();
-   auto &other = FlatVector::Validity(cursor.chunk.data[0]);
-   validity.SliceInPlace(other, target_offset, 0, count);
-   target_offset += count;
-  }
- }
-}
-WindowBuilder::WindowBuilder(WindowCollection &collection) : collection(collection) {
-}
-void WindowBuilder::Sink(DataChunk &chunk, idx_t input_idx) {
- if (!sink.second || input_idx < sink.first || sink.first + sink.second->Count() < input_idx) {
-  collection.GetCollection(input_idx, sink);
-  D_ASSERT(sink.second);
-  sink.second->InitializeAppend(appender);
- }
- sink.second->Append(appender, chunk);
- if (all_valid) {
-  UnifiedVectorFormat data;
-  chunk.data[0].ToUnifiedFormat(chunk.size(), data);
-  all_valid = data.validity.AllValid();
-  if (!all_valid) {
-   collection.all_valid = false;
-  }
- }
-}
-WindowCursor::WindowCursor(const WindowCollection &paged) : paged(paged) {
- if (paged.GetTypes().empty()) {
-  state.segment_index = 0;
-  state.chunk_index = 0;
-  state.current_row_index = 0;
-  state.next_row_index = paged.size();
-  state.properties = ColumnDataScanProperties::ALLOW_ZERO_COPY;
-  chunk.SetCardinality(state.next_row_index);
-  return;
- } else if (chunk.data.empty()) {
-  auto &inputs = paged.inputs;
-  D_ASSERT(inputs.get());
-  inputs->InitializeScan(state);
-  inputs->InitializeScanChunk(state, chunk);
- }
-}
