diff --git a/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_conflict.h b/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
index 4634cc7..7b2201c 100644
--- a/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_conflict.h
+++ b/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
@@ -137 +136,0 @@ private:
-<<<<<<< HEAD
@@ -140,7 +138,0 @@ private:
-||||||| c91de17e04
-    KeyedVector<audio_source_t, EffectDescVector*> mInputSources GUARDED_BY(mMutex);
-=======
-    KeyedVector<audio_source_t, std::shared_ptr<EffectDescVector>> mInputSources
-            GUARDED_BY(mMutex);
->>>>>>> 8a06b7a8
-<<<<<<< HEAD
@@ -149,6 +140,0 @@ private:
-||||||| c91de17e04
-    KeyedVector<audio_session_t, EffectVector*> mInputSessions GUARDED_BY(mMutex);
-=======
-    KeyedVector<audio_session_t, std::shared_ptr<EffectVector>> mInputSessions GUARDED_BY(mMutex);
->>>>>>> 8a06b7a8
-<<<<<<< HEAD
@@ -157,7 +142,0 @@ private:
-||||||| c91de17e04
-    KeyedVector<audio_stream_type_t, EffectDescVector*> mOutputStreams GUARDED_BY(mMutex);
-=======
-    KeyedVector<audio_stream_type_t, std::shared_ptr<EffectDescVector>> mOutputStreams
-            GUARDED_BY(mMutex);
->>>>>>> 8a06b7a8
-<<<<<<< HEAD
@@ -166,6 +144,0 @@ private:
-||||||| c91de17e04
-    KeyedVector<audio_session_t, EffectVector*> mOutputSessions GUARDED_BY(mMutex);
-=======
-    KeyedVector<audio_session_t, std::shared_ptr<EffectVector>> mOutputSessions
-            GUARDED_BY(mMutex);
->>>>>>> 8a06b7a8
