diff --git a/./output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_conflict.h b/./output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_merged.h
index 04ef067..6fb7ca2 100644
--- a/./output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_conflict.h
+++ b/./output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_merged.h
@@ -21 +20,0 @@ class CgroupController {
-<<<<<<< HEAD
@@ -23,41 +21,0 @@ class CgroupController {
-||||||| 6dc78dc56
-    static constexpr size_t CGROUP_NAME_BUF_SZ = 16;
-    static constexpr size_t CGROUP_PATH_BUF_SZ = 32;
-    uint32_t version_;
-    char name_[CGROUP_NAME_BUF_SZ];
-    char path_[CGROUP_PATH_BUF_SZ];
-};
-class CgroupDescriptor {
-  public:
-    CgroupDescriptor(uint32_t version, const std::string& name, const std::string& path,
-                     mode_t mode, const std::string& uid, const std::string& gid);
-    const CgroupController* controller() const { return &controller_; }
-    mode_t mode() const { return mode_; }
-    std::string uid() const { return uid_; }
-    std::string gid() const { return gid_; }
-  private:
-    CgroupController controller_;
-    mode_t mode_;
-    std::string uid_;
-    std::string gid_;
-};
-struct CgroupFile {
-    static constexpr uint32_t FILE_VERSION_1 = 1;
-    static constexpr uint32_t FILE_CURR_VERSION = FILE_VERSION_1;
-    uint32_t version_;
-    uint32_t controller_count_;
-    CgroupController controllers_[];
-=======
-    static constexpr size_t CGROUP_NAME_BUF_SZ = 16;
-    static constexpr size_t CGROUP_PATH_BUF_SZ = 32;
-    uint32_t version_;
-    char name_[CGROUP_NAME_BUF_SZ];
-    char path_[CGROUP_PATH_BUF_SZ];
-};
-struct CgroupFile {
-    static constexpr uint32_t FILE_VERSION_1 = 1;
-    static constexpr uint32_t FILE_CURR_VERSION = FILE_VERSION_1;
-    uint32_t version_;
-    uint32_t controller_count_;
-    CgroupController controllers_[];
->>>>>>> e7f9de2c
