--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_merged.c
@@ -2,3 +1,0 @@
-#include <sys/queue.h>
-#include <sys/utsname.h>
-#include <imsg.h>
@@ -5,0 +3 @@
+#include <sys/utsname.h>
@@ -120 +138,0 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-<<<<<<< HEAD
@@ -123,2 +140,0 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
-  ||||||| logfile(name);
-=======
@@ -126 +141,0 @@ struct tmuxproc *proc_start(const char *name, struct event_base *base,
->>>>>>> ac8678aefe157d7e40c5bcedd12333eaedf0df92
