diff --git a/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_mergegen.c b/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_merged.c
index 5e5e9b1..0e07c81 100644
--- a/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_mergegen.c
+++ b/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/proc.no_comments_merged.c
@@ -139 +139,3 @@ proc_start(const char *name, struct event_base *base, int forkflag,
-log_open(name); #ifdef HAVE_SETPROCTITLE setproctitle("%s (%s)", name, socket_path);
+ logfile(name);
+#ifdef HAVE_SETPROCTITLE
+ log_open(name);
