diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_merged.c
index d80c678..44f55b2 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_merged.c
@@ -7,2 +6,0 @@
-#include <getopt.h>
-#include <paths.h>
@@ -19,3 +16,0 @@ extern char *malloc_options;
-#ifdef DEBUG
-extern char *malloc_options;
-#endif
@@ -33 +28,3 @@ char *__progname = (char *)"tmux";
-__dead void usage(void) {
+__dead void
+usage(void)
+{
@@ -40 +37,3 @@ __dead void usage(void) {
-const char *getshell(void) {
+const char *
+getshell(void)
+{
@@ -44 +43,2 @@ const char *getshell(void) {
-  if (checkshell(shell)) return (shell);
+ if (checkshell(shell))
+  return (shell);
@@ -46 +46,2 @@ const char *getshell(void) {
-  if (pw != NULL && checkshell(pw->pw_shell)) return (pw->pw_shell);
+ if (pw != NULL && checkshell(pw->pw_shell))
+  return (pw->pw_shell);
@@ -49,4 +50,9 @@ const char *getshell(void) {
-int checkshell(const char *shell) {
-  if (shell == NULL || *shell == '\0' || *shell != '/') return (0);
-  if (areshell(shell)) return (0);
-  if (access(shell, X_OK) != 0) return (0);
+int
+checkshell(const char *shell)
+{
+ if (shell == NULL || *shell == '\0' || *shell != '/')
+  return (0);
+ if (areshell(shell))
+  return (0);
+ if (access(shell, X_OK) != 0)
+  return (0);
@@ -55 +61,3 @@ int checkshell(const char *shell) {
-int areshell(const char *shell) {
+int
+areshell(const char *shell)
+{
@@ -62,2 +70,4 @@ int areshell(const char *shell) {
-  if (*progname == '-') progname++;
-  if (strcmp(ptr, progname) == 0) return (1);
+ if (*progname == '-')
+  progname++;
+ if (strcmp(ptr, progname) == 0)
+  return (1);
@@ -66 +76,3 @@ int areshell(const char *shell) {
-static char *make_label(const char *label) {
+static char *
+make_label(const char *label)
+{
@@ -71 +83,2 @@ static char *make_label(const char *label) {
-  if (label == NULL) label = "default";
+ if (label == NULL)
+  label = "default";
@@ -77,2 +90,4 @@ static char *make_label(const char *label) {
-  if (mkdir(base, S_IRWXU) != 0 && errno != EEXIST) goto fail;
-  if (lstat(base, &sb) != 0) goto fail;
+ if (mkdir(base, S_IRWXU) != 0 && errno != EEXIST)
+  goto fail;
+ if (lstat(base, &sb) != 0)
+  goto fail;
@@ -97 +112,3 @@ fail:
-void setblocking(int fd, int state) {
+void
+setblocking(int fd, int state)
+{
@@ -107 +124,3 @@ void setblocking(int fd, int state) {
-const char *find_home(void) {
+const char *
+find_home(void)
+{
@@ -110 +129,2 @@ const char *find_home(void) {
-  if (home != NULL) return (home);
+ if (home != NULL)
+  return (home);
@@ -121 +141,3 @@ const char *find_home(void) {
-int main(int argc, char **argv) {
+int
+main(int argc, char **argv)
+{
@@ -125 +146,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
@@ -129,6 +149,0 @@ int main(int argc, char **argv) {
-  |||||||
-#ifdef DEBUG
-      malloc_options = (char *)"AFGJPX";
-#endif
-  =======
->>>>>>> ac8678aefe157d7e40c5bcedd12333eaedf0df92
@@ -188 +203,2 @@ int main(int argc, char **argv) {
-  if (shellcmd != NULL && argc != 0) usage();
+ if (shellcmd != NULL && argc != 0)
+  usage();
@@ -191,2 +207 @@ int main(int argc, char **argv) {
-             "recvfd proc exec tty ps",
-             NULL) != 0)
+     "recvfd proc exec tty ps", NULL) != 0)
@@ -199,4 +214,8 @@ int main(int argc, char **argv) {
-    if (s == NULL || *s == '\0') s = getenv("LC_CTYPE");
-    if (s == NULL || *s == '\0') s = getenv("LANG");
-    if (s == NULL || *s == '\0') s = "";
-    if (strcasestr(s, "UTF-8") != NULL || strcasestr(s, "UTF8") != NULL)
+  if (s == NULL || *s == '\0')
+   s = getenv("LC_CTYPE");
+  if (s == NULL || *s == '\0')
+   s = getenv("LANG");
+  if (s == NULL || *s == '\0')
+   s = "";
+  if (strcasestr(s, "UTF-8") != NULL ||
+      strcasestr(s, "UTF8") != NULL)
@@ -206 +225,2 @@ int main(int argc, char **argv) {
-  for (var = environ; *var != NULL; var++) environ_put(global_environ, *var);
+ for (var = environ; *var != NULL; var++)
+  environ_put(global_environ, *var);
@@ -217 +237,2 @@ int main(int argc, char **argv) {
-    if (strrchr(s, '/') != NULL) s = strrchr(s, '/') + 1;
+  if (strrchr(s, '/') != NULL)
+   s = strrchr(s, '/') + 1;
@@ -238,5 +258,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
-  exit(client_main(osdep_event_init(), argc, argv, flags));
-|||||||
-  exit(client_main(event_init(), argc, argv, flags));
-=======
@@ -244 +259,0 @@ int main(int argc, char **argv) {
->>>>>>> ac8678aefe157d7e40c5bcedd12333eaedf0df92
