diff --git a/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_mergegen.c b/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_merged.c
index fc068ff..44f55b2 100644
--- a/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_mergegen.c
+++ b/./output/tmux/0a2ef2b9-ac8678ae-b32ce34c/tmux.no_comments_merged.c
@@ -14 +14,3 @@
-#if defined(DEBUG) && defined(__OpenBSD__) extern char *malloc_options; #endif
+#if defined(DEBUG) && defined(__OpenBSD__)
+extern char *malloc_options;
+#endif
@@ -145,0 +148,2 @@ main(int argc, char **argv)
+ malloc_options = (char *) "AFGJPX";
+#endif
@@ -255 +259 @@ main(int argc, char **argv)
- exit(client_main(osdep_event_init(), argc, argv, flags, shellcmd));
+ exit(client_main(event_init(), argc, argv, flags, shellcmd));
