--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_mergegen.h
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_merged.h
@@ -3 +3,3 @@
-#include <aidl/android/media/IResourceManagerClient.h> #include <aidl/android/media/IResourceManagerService.h> #include <android/binder/IBinder.h> #include <media/IResourceManagerService.h>
+#include <aidl/android/media/IResourceManagerClient.h>
+#include <aidl/android/media/IResourceManagerService.h>
+#include <android/binder_auto_utils.h>
@@ -12 +17,2 @@ class DrmSessionManagerTest;
-class IResourceManagerClient; using aidl::android::media::IResourceManagerService;
+using aidl::android::media::IResourceManagerClient;
+using aidl::android::media::IResourceManagerService;
@@ -20 +26 @@ typedef std::map<std::vector<uint8_t>, SessionInfo> SessionInfoMap;
-struct DrmSessionManager : public IBinder::DeathRecipient {
+struct DrmSessionManager : public RefBase {
@@ -27,0 +36,3 @@ void addSession(int pid, const std::shared_ptr<IResourceManagerClient>& drm, con
+    size_t getSessionCount() const;
+    bool containsSession(const Vector<uint8_t>& sessionId) const;
+    void binderDied();
@@ -34 +45,3 @@ std::shared_ptr<IResourceManagerService> mService;
-bool mInitialized; ::ndk::ScopedAIBinder_DeathRecipient mDeathRecipient;
+    SessionInfoMap mSessionMap;
+    bool mInitialized;
+    ::ndk::ScopedAIBinder_DeathRecipient mDeathRecipient;
