--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_conflict.h
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_merged.h
@@ -3 +2,0 @@
-<<<<<<< HEAD
@@ -7,5 +5,0 @@
-||||||| 35b28e5f80
-=======
-#include <binder/IBinder.h>
-#include <media/IResourceManagerService.h>
->>>>>>> 77b8c802
@@ -17 +10,0 @@
-<<<<<<< HEAD
@@ -22,6 +14,0 @@
-||||||| 35b28e5f80
-=======
-#include <map>
-#include <utility>
-#include <vector>
->>>>>>> 77b8c802
@@ -30 +16,0 @@ class DrmSessionManagerTest;
-<<<<<<< HEAD
@@ -33,6 +18,0 @@ using aidl::android::media::IResourceManagerService;
-||||||| 35b28e5f80
-struct DrmSessionClientInterface;
-struct ProcessInfoInterface;
-=======
-class IResourceManagerClient;
->>>>>>> 77b8c802
@@ -46 +26 @@ typedef std::map<std::vector<uint8_t>, SessionInfo> SessionInfoMap;
-struct DrmSessionManager : public IBinder::DeathRecipient {
+struct DrmSessionManager : public RefBase {
@@ -49 +28,0 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-<<<<<<< HEAD
@@ -51,6 +29,0 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-||||||| 35b28e5f80
-    explicit DrmSessionManager(sp<ProcessInfoInterface> processInfo);
-=======
-    explicit DrmSessionManager(const sp<IResourceManagerService> &service);
->>>>>>> 77b8c802
-<<<<<<< HEAD
@@ -60,5 +32,0 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-||||||| 35b28e5f80
-    void addSession(int pid, const sp<DrmSessionClientInterface>& drm, const Vector<uint8_t>& sessionId);
-=======
-    void addSession(int pid, const sp<IResourceManagerClient>& drm, const Vector<uint8_t>& sessionId);
->>>>>>> 77b8c802
@@ -68 +35,0 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-<<<<<<< HEAD
@@ -72,6 +38,0 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-||||||| 35b28e5f80
-=======
-    size_t getSessionCount() const;
-    bool containsSession(const Vector<uint8_t>& sessionId) const;
-    virtual void binderDied(const wp<IBinder>& );
->>>>>>> 77b8c802
@@ -82 +42,0 @@ private:
-<<<<<<< HEAD
@@ -84,9 +43,0 @@ private:
-||||||| 35b28e5f80
-    int64_t getTime_l();
-    bool getLowestPriority_l(int* lowestPriorityPid, int* lowestPriority);
-    bool getLeastUsedSession_l(
-            int pid, sp<DrmSessionClientInterface>* drm, Vector<uint8_t>* sessionId);
-    sp<ProcessInfoInterface> mProcessInfo;
-=======
-    sp<IResourceManagerService> mService;
->>>>>>> 77b8c802
@@ -94 +44,0 @@ private:
-<<<<<<< HEAD
@@ -98,7 +47,0 @@ private:
-||||||| 35b28e5f80
-    PidSessionInfosMap mSessionMap;
-    int64_t mTime;
-=======
-    SessionInfoMap mSessionMap;
-    bool mInitialized;
->>>>>>> 77b8c802
