--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_conflict.h
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_merged.h
@@ -10,7 +9,0 @@
-<<<<<<< HEAD
-||||||| 35b28e5f80
-#include <media/MediaAnalyticsItem.h>
-=======
-#include <media/IResourceManagerService.h>
-#include <media/MediaAnalyticsItem.h>
->>>>>>> 77b8c802
@@ -44 +36,0 @@ struct DrmHal : public IDrm,
-<<<<<<< HEAD
@@ -46,16 +37,0 @@ struct DrmHal : public IDrm,
-||||||| 35b28e5f80
-=======
-    struct DrmSessionClient : public BnResourceManagerClient {
-        explicit DrmSessionClient(DrmHal* drm, const Vector<uint8_t>& sessionId)
-          : mSessionId(sessionId),
-            mDrm(drm) {}
-        virtual bool reclaimResource();
-        virtual String8 getName();
-        const Vector<uint8_t> mSessionId;
-    protected:
-        virtual ~DrmSessionClient();
-    private:
-        wp<DrmHal> mDrm;
-        DISALLOW_EVIL_CONSTRUCTORS(DrmSessionClient);
-    };
->>>>>>> 77b8c802
@@ -169 +144,0 @@ private:
-<<<<<<< HEAD
@@ -171,5 +145,0 @@ private:
-||||||| 35b28e5f80
-    Vector<Vector<uint8_t>> mOpenSessions;
-=======
-    Vector<sp<DrmSessionClient>> mOpenSessions;
->>>>>>> 77b8c802
