--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_merged.h
@@ -10,2 +9,0 @@
-#include <media/IResourceManagerService.h>
-#include <media/MediaAnalyticsItem.h>
@@ -18,3 +16 @@
-using ::android::hardware::hidl_vec;
-using ::android::hardware::Return;
-using ::android::hardware::Void;
+namespace drm = ::android::hardware::drm;
@@ -27,0 +24,3 @@ using drm::V1_2::OfflineLicenseState;
+using ::android::hardware::hidl_vec;
+using ::android::hardware::Return;
+using ::android::hardware::Void;
@@ -36,13 +35,3 @@ inline bool operator==(const Vector<uint8_t> &l, const Vector<uint8_t> &r) {
-struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  struct DrmSessionClient : public BnResourceManagerClient {
-    explicit DrmSessionClient(DrmHal *drm, const Vector<uint8_t> &sessionId)
-        : mSessionId(sessionId), mDrm(drm) {}
-    virtual bool reclaimResource();
-    virtual String8 getName();
-    const Vector<uint8_t> mSessionId;
-   protected:
-    virtual ~DrmSessionClient();
-   private:
-    wp<DrmHal> mDrm;
-    DISALLOW_EVIL_CONSTRUCTORS(DrmSessionClient);
-  };
+struct DrmHal : public IDrm,
+                public IDrmPluginListener_V1_2 {
+    struct DrmSessionClient;
@@ -159 +144,0 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-<<<<<<< HEAD
@@ -161,5 +145,0 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-||||||| 35b28e5f80
-  Vector<Vector<uint8_t>> mOpenSessions;
-=======
-  Vector<sp<DrmSessionClient>> mOpenSessions;
->>>>>>> 77b8c802dbf3d63e19ff44e7bb2fe47b37e8d6b2
@@ -184 +163 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  Mutex DrmHal::mLock;
+    DISALLOW_EVIL_CONSTRUCTORS(DrmHal);
