diff --git a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@include@media@MediaResource.no_comments_mergegen.h b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@include@media@MediaResource.no_comments_merged.h
index baf5ffd..4638470 100644
--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@include@media@MediaResource.no_comments_mergegen.h
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@include@media@MediaResource.no_comments_merged.h
@@ -5 +4,0 @@
-#include <vector>
@@ -12 +11,2 @@ public:
-using Type = MediaResourceType; using SubType = MediaResourceSubType;
+    using Type = MediaResourceType;
+    using SubType = MediaResourceSubType;
@@ -17 +17,5 @@ using Type = MediaResourceType; using SubType = MediaResourceSubType;
-static MediaResource CpuBoostResource(); static MediaResource VideoBatteryResource(); static MediaResource DrmSessionResource(const std::vector<int8_t> &id, int64_t value);
+    static MediaResource CodecResource(bool secure, bool video);
+    static MediaResource GraphicMemoryResource(int64_t value);
+    static MediaResource CpuBoostResource();
+    static MediaResource VideoBatteryResource();
+    static MediaResource DrmSessionResource(const std::vector<int8_t> &id, int64_t value);
@@ -21 +25,8 @@ inline static const char *asString(MediaResource::Type i, const char *def = "??"
-switch (i) { case MediaResource::Type::kUnspecified: return "unspecified"; case MediaResource::Type::kSecureCodec: return "secure-codec"; case MediaResource::Type::kNonSecureCodec: return "non-secure-codec"; case MediaResource::Type::kGraphicMemory: return "graphic-memory"; case MediaResource::Type::kCpuBoost: return "cpu-boost"; case MediaResource::Type::kBattery: return "battery"; case MediaResource::Type::kDrmSession: return "drm-session";
+        case MediaResource::Type::kUnspecified: return "unspecified";
+        case MediaResource::Type::kSecureCodec: return "secure-codec";
+        case MediaResource::Type::kNonSecureCodec: return "non-secure-codec";
+        case MediaResource::Type::kGraphicMemory: return "graphic-memory";
+        case MediaResource::Type::kCpuBoost: return "cpu-boost";
+        case MediaResource::Type::kBattery: return "battery";
+        case MediaResource::Type::kDrmSession: return "drm-session";
+        default: return def;
