--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmediaplayerservice@tests@DrmSessionManager_test.no_comments_conflict.cpp
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmediaplayerservice@tests@DrmSessionManager_test.no_comments_merged.cpp
@@ -5 +4,0 @@
-<<<<<<< HEAD
@@ -8,5 +6,0 @@
-||||||| 35b28e5f80
-=======
-#include <media/IResourceManagerService.h>
-#include <media/IResourceManagerClient.h>
->>>>>>> 77b8c802
@@ -16 +9,0 @@
-<<<<<<< HEAD
@@ -21,6 +13,0 @@
-||||||| 35b28e5f80
-=======
-#include <algorithm>
-#include <vector>
-#include "ResourceManagerService.h"
->>>>>>> 77b8c802
@@ -28 +14,0 @@ namespace android {
-<<<<<<< HEAD
@@ -41,10 +26,0 @@ static Vector<uint8_t> toAndroidVector(const std::vector<uint8_t> &vec) {
-||||||| 35b28e5f80
-=======
-static Vector<uint8_t> toAndroidVector(const std::vector<uint8_t> &vec) {
-    Vector<uint8_t> aVec;
-    for (auto b : vec) {
-        aVec.push_back(b);
-    }
-    return aVec;
-}
->>>>>>> 77b8c802
@@ -64 +39,0 @@ private:
-<<<<<<< HEAD
@@ -70,13 +44,0 @@ struct FakeDrm : public BnResourceManagerClient {
-||||||| 35b28e5f80
-struct FakeDrm : public DrmSessionClientInterface {
-    FakeDrm() {}
-    virtual ~FakeDrm() {}
-=======
-struct FakeDrm : public BnResourceManagerClient {
-    FakeDrm(const std::vector<uint8_t>& sessionId, const sp<DrmSessionManager>& manager)
-        : mSessionId(toAndroidVector(sessionId)),
-          mReclaimed(false),
-          mDrmSessionManager(manager) {}
-    virtual ~FakeDrm() {}
->>>>>>> 77b8c802
-<<<<<<< HEAD
@@ -88,10 +49,0 @@ struct FakeDrm : public BnResourceManagerClient {
-||||||| 35b28e5f80
-    virtual bool reclaimSession(const Vector<uint8_t>& sessionId) {
-        mReclaimedSessions.push_back(sessionId);
-        return true;
-=======
-    virtual bool reclaimResource() {
-        mReclaimed = true;
-        mDrmSessionManager->removeSession(mSessionId);
-        return true;
->>>>>>> 77b8c802
@@ -99 +50,0 @@ struct FakeDrm : public BnResourceManagerClient {
-<<<<<<< HEAD
@@ -108,12 +58,0 @@ struct FakeDrm : public BnResourceManagerClient {
-||||||| 35b28e5f80
-    const Vector<Vector<uint8_t> >& reclaimedSessions() const {
-        return mReclaimedSessions;
-=======
-    virtual String8 getName() {
-        String8 name("FakeDrm[");
-        for (size_t i = 0; i < mSessionId.size(); ++i) {
-            name.appendFormat("%02x", mSessionId[i]);
-        }
-        name.append("]");
-        return name;
->>>>>>> 77b8c802
@@ -130 +68,0 @@ private:
-<<<<<<< HEAD
@@ -145,18 +82,0 @@ private:
-||||||| 35b28e5f80
-=======
-struct FakeSystemCallback :
-        public ResourceManagerService::SystemCallbackInterface {
-    FakeSystemCallback() {}
-    virtual void noteStartVideo(int ) override {}
-    virtual void noteStopVideo(int ) override {}
-    virtual void noteResetVideo() override {}
-    virtual bool requestCpusetBoost(
-            bool , const sp<IInterface> & ) override {
-        return true;
-    }
-protected:
-    virtual ~FakeSystemCallback() {}
-private:
-    DISALLOW_EVIL_CONSTRUCTORS(FakeSystemCallback);
-};
->>>>>>> 77b8c802
@@ -171 +90,0 @@ public:
-<<<<<<< HEAD
@@ -181,16 +99,0 @@ public:
-||||||| 35b28e5f80
-        : mDrmSessionManager(new DrmSessionManager(new FakeProcessInfo())),
-          mTestDrm1(new FakeDrm()),
-          mTestDrm2(new FakeDrm()) {
-        GetSessionId(kTestSessionId1, ARRAY_SIZE(kTestSessionId1), &mSessionId1);
-        GetSessionId(kTestSessionId2, ARRAY_SIZE(kTestSessionId2), &mSessionId2);
-        GetSessionId(kTestSessionId3, ARRAY_SIZE(kTestSessionId3), &mSessionId3);
-=======
-        : mService(new ResourceManagerService(new FakeProcessInfo(), new FakeSystemCallback())),
-          mDrmSessionManager(new DrmSessionManager(mService)),
-          mTestDrm1(new FakeDrm(kTestSessionId1, mDrmSessionManager)),
-          mTestDrm2(new FakeDrm(kTestSessionId2, mDrmSessionManager)),
-          mTestDrm3(new FakeDrm(kTestSessionId3, mDrmSessionManager)) {
-        DrmSessionManager *ptr = new DrmSessionManager(mService);
-        EXPECT_NE(ptr, nullptr);
->>>>>>> 77b8c802
@@ -204 +106,0 @@ protected:
-<<<<<<< HEAD
@@ -206,4 +107,0 @@ protected:
-||||||| 35b28e5f80
-=======
-    sp<IResourceManagerService> mService;
->>>>>>> 77b8c802
@@ -211 +108,0 @@ protected:
-<<<<<<< HEAD
@@ -215,11 +111,0 @@ protected:
-||||||| 35b28e5f80
-    sp<FakeDrm> mTestDrm1;
-    sp<FakeDrm> mTestDrm2;
-    Vector<uint8_t> mSessionId1;
-    Vector<uint8_t> mSessionId2;
-    Vector<uint8_t> mSessionId3;
-=======
-    sp<FakeDrm> mTestDrm1;
-    sp<FakeDrm> mTestDrm2;
-    sp<FakeDrm> mTestDrm3;
->>>>>>> 77b8c802
@@ -257 +142,0 @@ TEST_F(DrmSessionManagerTest, reclaimSession) {
-<<<<<<< HEAD
@@ -262,11 +146,0 @@ TEST_F(DrmSessionManagerTest, reclaimSession) {
-||||||| 35b28e5f80
-    sp<FakeDrm> drm = new FakeDrm;
-    const uint8_t ids[] = {1, 3, 5};
-    Vector<uint8_t> sessionId;
-    GetSessionId(ids, ARRAY_SIZE(ids), &sessionId);
-    mDrmSessionManager->addSession(15, drm, sessionId);
-=======
-    const std::vector<uint8_t> sid{1, 3, 5};
-    sp<FakeDrm> drm = new FakeDrm(sid, mDrmSessionManager);
-    mDrmSessionManager->addSession(15, drm, drm->mSessionId);
->>>>>>> 77b8c802
