--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/acbda983-2d73a175-8cd6ee62/src@core@lib@security@credentials@google_default@google_default_credentials.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/acbda983-2d73a175-8cd6ee62/src@core@lib@security@credentials@google_default@google_default_credentials.no_comments_merged.cc
@@ -214,22 +214,2 @@ end:
-grpc_channel_credentials* grpc_google_default_credentials_create(
-    grpc_call_credentials* call_credentials) {
-  grpc_channel_credentials* result = nullptr;
-  grpc_core::RefCountedPtr<grpc_call_credentials> call_creds(call_credentials);
-  grpc_error* error = GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-      "Failed to create Google credentials");
-  grpc_core::ExecCtx exec_ctx;
-  GRPC_API_TRACE("grpc_google_default_credentials_create(%p)", 1,
-                 (call_credentials));
-  update_tenancy();
-<<<<<<< HEAD
-  if (call_credentials == nullptr) {
-    default_call_creds(&call_creds, error);
-|||||||
-  err = create_default_creds_from_path(
-      gpr_getenv(GRPC_GOOGLE_CREDENTIALS_ENV_VAR), &call_creds);
-  if (err == GRPC_ERROR_NONE) goto end;
-  error = grpc_error_add_child(error, err);
-  err = create_default_creds_from_path(
-      grpc_get_well_known_google_credentials_file_path(), &call_creds);
-  if (err == GRPC_ERROR_NONE) goto end;
-  error = grpc_error_add_child(error, err);
+static void update_tenancy() {
+  gpr_once_init(&g_once, init_default_credentials);
@@ -245,7 +224,0 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-  if (g_metadata_server_available) {
-    call_creds = grpc_core::RefCountedPtr<grpc_call_credentials>(
-        grpc_google_compute_engine_credentials_create(nullptr));
-    if (call_creds == nullptr) {
-      error = grpc_error_add_child(
-          error, GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                     "Failed to get credentials from network"));
@@ -253 +226,4 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-=======
+static void default_call_creds(
+    grpc_core::RefCountedPtr<grpc_call_credentials>* call_creds,
+    grpc_error* error) {
+  grpc_error* err;
@@ -256 +232 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-    err = create_default_creds_from_path(path_from_env, &call_creds);
+    err = create_default_creds_from_path(path_from_env, call_creds);
@@ -258 +234 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-    if (err == GRPC_ERROR_NONE) goto end;
+    if (err == GRPC_ERROR_NONE) return;
@@ -262,2 +238,2 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-      grpc_get_well_known_google_credentials_file_path(), &call_creds);
-  if (err == GRPC_ERROR_NONE) goto end;
+      grpc_get_well_known_google_credentials_file_path(), call_creds);
+  if (err == GRPC_ERROR_NONE) return;
@@ -265,9 +240,0 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-  gpr_mu_lock(&g_state_mu);
-  if (!g_metadata_server_available) {
-    g_is_on_gce = g_gce_tenancy_checker();
-    g_metadata_server_available = g_is_on_gce;
-  }
-  if (!g_metadata_server_available) {
-    g_metadata_server_available = is_metadata_server_reachable();
-  }
-  gpr_mu_unlock(&g_state_mu);
@@ -275 +242 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-    call_creds = grpc_core::RefCountedPtr<grpc_call_credentials>(
+    *call_creds = grpc_core::RefCountedPtr<grpc_call_credentials>(
@@ -277 +244 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
-    if (call_creds == nullptr) {
+    if (*call_creds == nullptr) {
@@ -282 +249,14 @@ grpc_channel_credentials* grpc_google_default_credentials_create(
->>>>>>> 2d73a17587193380686ae36fc170962f286591f9
+  }
+}
+grpc_channel_credentials* grpc_google_default_credentials_create(
+    grpc_call_credentials* call_credentials) {
+  grpc_channel_credentials* result = nullptr;
+  grpc_core::RefCountedPtr<grpc_call_credentials> call_creds(call_credentials);
+  grpc_error* error = GRPC_ERROR_CREATE_FROM_STATIC_STRING(
+      "Failed to create Google credentials");
+  grpc_core::ExecCtx exec_ctx;
+  GRPC_API_TRACE("grpc_google_default_credentials_create(%p)", 1,
+                 (call_credentials));
+  update_tenancy();
+  if (call_credentials == nullptr) {
+    default_call_creds(&call_creds, error);
@@ -330,34 +309,0 @@ void grpc_override_well_known_credentials_path_getter(
-static void update_tenancy() {
-  gpr_once_init(&g_once, init_default_credentials);
-  gpr_mu_lock(&g_state_mu);
-  if (!g_metadata_server_available) {
-    g_is_on_gce = g_gce_tenancy_checker();
-    g_metadata_server_available = g_is_on_gce;
-  }
-  if (!g_metadata_server_available) {
-    g_metadata_server_available = is_metadata_server_reachable();
-  }
-  gpr_mu_unlock(&g_state_mu);
-}
-static void default_call_creds(
-    grpc_core::RefCountedPtr<grpc_call_credentials>* call_creds,
-    grpc_error* error) {
-  grpc_error* err;
-  err = create_default_creds_from_path(
-      gpr_getenv(GRPC_GOOGLE_CREDENTIALS_ENV_VAR), call_creds);
-  if (err == GRPC_ERROR_NONE) return;
-  error = grpc_error_add_child(error, err);
-  err = create_default_creds_from_path(
-      grpc_get_well_known_google_credentials_file_path(), call_creds);
-  if (err == GRPC_ERROR_NONE) return;
-  error = grpc_error_add_child(error, err);
-  if (g_metadata_server_available) {
-    *call_creds = grpc_core::RefCountedPtr<grpc_call_credentials>(
-        grpc_google_compute_engine_credentials_create(nullptr));
-    if (*call_creds == nullptr) {
-      error = grpc_error_add_child(
-          error, GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                     "Failed to get credentials from network"));
-    }
-  }
-}
