--- a/./output/tmux/2a32565e-7c0789d2-09afc6c8/utf8.no_comments_conflict.c
+++ b/./output/tmux/2a32565e-7c0789d2-09afc6c8/utf8.no_comments_merged.c
@@ -168 +167,0 @@ utf8_width(struct utf8_data *ud, int *width)
-<<<<<<< HEAD
@@ -169,0 +169 @@ utf8_width(struct utf8_data *ud, int *width)
+<<<<<<< HEAD
@@ -174 +173,0 @@ utf8_width(struct utf8_data *ud, int *width)
-||||||| 09afc6c8
@@ -176,2 +175 @@ utf8_width(struct utf8_data *ud, int *width)
- log_debug("UTF-8 %.*s is %08X", (int)ud->size, ud->data, (u_int)wc);
->>>>>>> 7c0789d2
+>>>>>>> obsd-master
@@ -179 +176,0 @@ utf8_width(struct utf8_data *ud, int *width)
-<<<<<<< HEAD
@@ -183,0 +181 @@ utf8_width(struct utf8_data *ud, int *width)
+<<<<<<< HEAD
@@ -185,3 +182,0 @@ utf8_width(struct utf8_data *ud, int *width)
-||||||| 09afc6c8
- log_debug("UTF-8 %.*s %#x, wcwidth() %d", (int)ud->size, ud->data,
-     (u_int)wc, *width);
@@ -189,5 +184 @@ utf8_width(struct utf8_data *ud, int *width)
- log_debug("wcwidth(%08X) returned %d", (u_int)wc, *width);
- if (*width < 0) {
-  *width = (wc >= 0x80 && wc <= 0x9f) ? 0 : 1;
- }
->>>>>>> 7c0789d2
+>>>>>>> obsd-master
