--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/92b9379d-7f55d0c6-c84b6df7/src@core@controller.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/92b9379d-7f55d0c6-c84b6df7/src@core@controller.no_comments_merged.cpp
@@ -2 +1,0 @@
-#include "src/config/configwindow.h"
@@ -3,0 +3 @@
+#include "src/config/configwindow.h"
@@ -45,2 +44,0 @@ Controller::Controller()
-{
-    m_appLatestVersion
@@ -91,4 +86,0 @@ Controller::Controller()
-    {
-        delete m_history;
-        delete m_trayIconMenu;
-    }
@@ -303,5 +290,0 @@ Controller::Controller()
-        connect(configAction,
-                &QAction::triggered,
-                this,
-                &Controller::openConfigWindow);
-        QAction* infoAction = new QAction(tr("&About"), this);
@@ -309 +292,3 @@ Controller::Controller()
-          infoAction, &QAction::triggered, this, &Controller::openInfoWindow);
+      configAction, &QAction::triggered, this, &Controller::openConfigWindow);
+    QAction* infoAction = new QAction(tr("&About"), this);
+    connect(infoAction, &QAction::triggered, this, &Controller::openInfoWindow);
