diff --git a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_conflict.h b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_merged.h
index b9fcf88..76c89dc 100644
--- a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_conflict.h
+++ b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_merged.h
@@ -8 +7,0 @@
-<<<<<<< HEAD
@@ -10,7 +8,0 @@
-||||||| faafe2989a
-#include <media/MediaResource.h>
-#include <media/MediaAnalyticsItem.h>
-=======
-#include <media/MediaResource.h>
-#include <media/MediaMetrics.h>
->>>>>>> e23cbd26
@@ -235 +226,0 @@ private:
-<<<<<<< HEAD
@@ -242,13 +232,0 @@ private:
-||||||| faafe2989a
-    MediaAnalyticsItem *mAnalyticsItem;
-    void initAnalyticsItem();
-    void updateAnalyticsItem();
-    void flushAnalyticsItem();
-    void updateEphemeralAnalytics(MediaAnalyticsItem *item);
-=======
-    mediametrics_handle_t mMetricsHandle;
-    void initMediametrics();
-    void updateMediametrics();
-    void flushMediametrics();
-    void updateEphemeralMediametrics(mediametrics_handle_t item);
->>>>>>> e23cbd26
