--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_merged.h
@@ -8,2 +7,0 @@
-#include <media/MediaResource.h>
-#include <media/MediaAnalyticsItem.h>
@@ -77,2 +69,10 @@ struct MediaCodec : public AHandler {
-  status_t configure(const sp<AMessage> &format,
-                     const sp<Surface> &nativeWindow, const sp<ICrypto> &crypto,
+    status_t configure(
+            const sp<AMessage> &format,
+            const sp<Surface> &nativeWindow,
+            const sp<ICrypto> &crypto,
+            uint32_t flags);
+    status_t configure(
+            const sp<AMessage> &format,
+            const sp<Surface> &nativeWindow,
+            const sp<ICrypto> &crypto,
+            const sp<IDescrambler> &descrambler,
@@ -80,3 +79,0 @@ struct MediaCodec : public AHandler {
-  status_t configure(const sp<AMessage> &format,
-                     const sp<Surface> &nativeWindow, const sp<ICrypto> &crypto,
-                     const sp<IDescrambler> &descrambler, uint32_t flags);
@@ -132,0 +143 @@ struct MediaCodec : public AHandler {
+private:
@@ -235,2 +246,2 @@ struct MediaCodec : public AHandler {
-  DISALLOW_EVIL_CONSTRUCTORS(MediaCodec);
-  DISALLOW_EVIL_CONSTRUCTORS(MediaCodec);
+    List<size_t> mAvailPortBuffers[2];
+    std::vector<BufferInfo> mPortBuffers[2];
