diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_merged.h
index a4face9..76c89dc 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@include@media@stagefright@MediaCodec.no_comments_merged.h
@@ -8,2 +7,0 @@
-#include <media/MediaResource.h>
-#include <media/MediaAnalyticsItem.h>
@@ -41,4 +39 @@ struct IDescrambler;
-}
-}
-}
-}
+}}}}
@@ -67,9 +62,6 @@ struct MediaCodec : public AHandler {
-  static sp<MediaCodec> CreateByType(const sp<ALooper> &looper,
-                                     const AString &mime, bool encoder,
-                                     status_t *err = NULL, pid_t pid = kNoPid,
-                                     uid_t uid = kNoUid);
-  static sp<MediaCodec> CreateByComponentName(const sp<ALooper> &looper,
-                                              const AString &name,
-                                              status_t *err = NULL,
-                                              pid_t pid = kNoPid,
-                                              uid_t uid = kNoUid);
+    static sp<MediaCodec> CreateByType(
+            const sp<ALooper> &looper, const AString &mime, bool encoder, status_t *err = NULL,
+            pid_t pid = kNoPid, uid_t uid = kNoUid);
+    static sp<MediaCodec> CreateByComponentName(
+            const sp<ALooper> &looper, const AString &name, status_t *err = NULL,
+            pid_t pid = kNoPid, uid_t uid = kNoUid);
@@ -77,2 +69,10 @@ struct MediaCodec : public AHandler {
-  status_t configure(const sp<AMessage> &format,
-                     const sp<Surface> &nativeWindow, const sp<ICrypto> &crypto,
+    status_t configure(
+            const sp<AMessage> &format,
+            const sp<Surface> &nativeWindow,
+            const sp<ICrypto> &crypto,
+            uint32_t flags);
+    status_t configure(
+            const sp<AMessage> &format,
+            const sp<Surface> &nativeWindow,
+            const sp<ICrypto> &crypto,
+            const sp<IDescrambler> &descrambler,
@@ -80,3 +79,0 @@ struct MediaCodec : public AHandler {
-  status_t configure(const sp<AMessage> &format,
-                     const sp<Surface> &nativeWindow, const sp<ICrypto> &crypto,
-                     const sp<IDescrambler> &descrambler, uint32_t flags);
@@ -93,2 +90,6 @@ struct MediaCodec : public AHandler {
-  status_t queueInputBuffer(size_t index, size_t offset, size_t size,
-                            int64_t presentationTimeUs, uint32_t flags,
+    status_t queueInputBuffer(
+            size_t index,
+            size_t offset,
+            size_t size,
+            int64_t presentationTimeUs,
+            uint32_t flags,
@@ -96 +97,3 @@ struct MediaCodec : public AHandler {
-  status_t queueSecureInputBuffer(size_t index, size_t offset,
+    status_t queueSecureInputBuffer(
+            size_t index,
+            size_t offset,
@@ -98,2 +101,4 @@ struct MediaCodec : public AHandler {
-                                  size_t numSubSamples, const uint8_t key[16],
-                                  const uint8_t iv[16], CryptoPlugin::Mode mode,
+            size_t numSubSamples,
+            const uint8_t key[16],
+            const uint8_t iv[16],
+            CryptoPlugin::Mode mode,
@@ -101 +106,2 @@ struct MediaCodec : public AHandler {
-                                  int64_t presentationTimeUs, uint32_t flags,
+            int64_t presentationTimeUs,
+            uint32_t flags,
@@ -104,2 +110,6 @@ struct MediaCodec : public AHandler {
-  status_t dequeueOutputBuffer(size_t *index, size_t *offset, size_t *size,
-                               int64_t *presentationTimeUs, uint32_t *flags,
+    status_t dequeueOutputBuffer(
+            size_t *index,
+            size_t *offset,
+            size_t *size,
+            int64_t *presentationTimeUs,
+            uint32_t *flags,
@@ -132,0 +143 @@ struct MediaCodec : public AHandler {
+private:
@@ -235,2 +246,2 @@ struct MediaCodec : public AHandler {
-  DISALLOW_EVIL_CONSTRUCTORS(MediaCodec);
-  DISALLOW_EVIL_CONSTRUCTORS(MediaCodec);
+    List<size_t> mAvailPortBuffers[2];
+    std::vector<BufferInfo> mPortBuffers[2];
@@ -250,4 +261,3 @@ struct MediaCodec : public AHandler {
-  static sp<CodecBase> GetCodecBase(const AString &name,
-                                    const char *owner = nullptr);
-  static status_t PostAndAwaitResponse(const sp<AMessage> &msg,
-                                       sp<AMessage> *response);
+    static sp<CodecBase> GetCodecBase(const AString &name, const char *owner = nullptr);
+    static status_t PostAndAwaitResponse(
+            const sp<AMessage> &msg, sp<AMessage> *response);
@@ -263,7 +273,5 @@ struct MediaCodec : public AHandler {
-  status_t getBufferAndFormat(size_t portIndex, size_t index,
-                              sp<MediaCodecBuffer> *buffer,
-                              sp<AMessage> *format);
-  bool handleDequeueInputBuffer(const sp<AReplyToken> &replyID,
-                                bool newRequest = false);
-  bool handleDequeueOutputBuffer(const sp<AReplyToken> &replyID,
-                                 bool newRequest = false);
+    status_t getBufferAndFormat(
+            size_t portIndex, size_t index,
+            sp<MediaCodecBuffer> *buffer, sp<AMessage> *format);
+    bool handleDequeueInputBuffer(const sp<AReplyToken> &replyID, bool newRequest = false);
+    bool handleDequeueOutputBuffer(const sp<AReplyToken> &replyID, bool newRequest = false);
@@ -285,2 +293 @@ struct MediaCodec : public AHandler {
-  status_t amendOutputFormatWithCodecSpecificData(
-      const sp<MediaCodecBuffer> &buffer);
+    status_t amendOutputFormatWithCodecSpecificData(const sp<MediaCodecBuffer> &buffer);
@@ -311,2 +318 @@ struct MediaCodec : public AHandler {
-  int64_t
-      mInputBufferCounter;
+    int64_t mInputBufferCounter;
@@ -328,11 +334,3 @@ struct MediaCodec : public AHandler {
-    Histogram()
-        : mFloor(0),
-          mWidth(0),
-          mBelow(0),
-          mAbove(0),
-          mMin(INT64_MAX),
-          mMax(INT64_MIN),
-          mSum(0),
-          mCount(0),
-          mBucketCount(0),
-          mBuckets(NULL){};
+        Histogram() : mFloor(0), mWidth(0), mBelow(0), mAbove(0),
+                      mMin(INT64_MAX), mMax(INT64_MIN), mSum(0), mCount(0),
+                      mBucketCount(0), mBuckets(NULL) {};
@@ -340,4 +338 @@ struct MediaCodec : public AHandler {
-    void clear() {
-      if (mBuckets != NULL) free(mBuckets);
-      mBuckets = NULL;
-    };
+        void clear() { if (mBuckets != NULL) free(mBuckets); mBuckets = NULL; };
