diff --git a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_mergegen.cpp b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_merged.cpp
index 93cfc08..a279b2c 100644
--- a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libmediametrics@MediaMetrics.no_comments_merged.cpp
@@ -110 +110,5 @@ bool mediametrics_selfRecord(mediametrics_handle_t handle) {
-mediametrics_handle_t mediametrics_dup(mediametrics_handle_t handle) { Item *item = (Item *) handle; if (item == NULL) return Item::convert(item); return Item::convert(item->dup()); }
+mediametrics_handle_t mediametrics_dup(mediametrics_handle_t handle) {
+    Item *item = (Item *) handle;
+    if (item == NULL) return Item::convert(item);
+    return Item::convert(item->dup());
+}
