diff --git a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@MediaCodec.no_comments_conflict.cpp b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@MediaCodec.no_comments_merged.cpp
index cad21d5..149d733 100644
--- a/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@MediaCodec.no_comments_conflict.cpp
+++ b/./output/frameworks_av/9468f951-e23cbd26-faafe298/media@libstagefright@MediaCodec.no_comments_merged.cpp
@@ -566 +565,0 @@ void MediaCodec::updateEphemeralMediametrics(mediametrics_handle_t item) {
-<<<<<<< HEAD
@@ -587,13 +585,0 @@ void MediaCodec::updateLowLatency(const sp<AMessage> &msg) {
-||||||| faafe2989a
-void MediaCodec::flushAnalyticsItem() {
-    updateAnalyticsItem();
-    if (mAnalyticsItem != NULL) {
-        if (mAnalyticsItem->count() > 0) {
-            mAnalyticsItem->selfrecord();
-=======
-void MediaCodec::flushMediametrics() {
-    updateMediametrics();
-    if (mMetricsHandle != 0) {
-        if (mediametrics_count(mMetricsHandle) > 0) {
-            mediametrics_selfRecord(mMetricsHandle);
->>>>>>> e23cbd26
@@ -601,8 +586,0 @@ void MediaCodec::flushMediametrics() {
-<<<<<<< HEAD
-||||||| faafe2989a
-        delete mAnalyticsItem;
-        mAnalyticsItem = NULL;
-=======
-        mediametrics_delete(mMetricsHandle);
-        mMetricsHandle = 0;
->>>>>>> e23cbd26
@@ -1690 +1667,0 @@ void MediaCodec::onMessageReceived(const sp<AMessage> &msg) {
-<<<<<<< HEAD
@@ -1692,7 +1668,0 @@ void MediaCodec::onMessageReceived(const sp<AMessage> &msg) {
-||||||| faafe2989a
-                        resourceType = MediaResource::kSecureCodec;
-                        mAnalyticsItem->setInt32(kCodecSecure, 1);
-=======
-                        resourceType = MediaResource::kSecureCodec;
-                        mediametrics_setInt32(mMetricsHandle, kCodecSecure, 1);
->>>>>>> e23cbd26
@@ -1701,7 +1670,0 @@ void MediaCodec::onMessageReceived(const sp<AMessage> &msg) {
-<<<<<<< HEAD
-                        mediametrics_setInt32(mMetricsHandle, kCodecSecure, 0);
-||||||| faafe2989a
-                        resourceType = MediaResource::kNonSecureCodec;
-                        mAnalyticsItem->setInt32(kCodecSecure, 0);
-=======
-                        resourceType = MediaResource::kNonSecureCodec;
@@ -1709 +1671,0 @@ void MediaCodec::onMessageReceived(const sp<AMessage> &msg) {
->>>>>>> e23cbd26
