diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/redis/4339706e-f42846e8-3f1c8475/src@server.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/redis/4339706e-f42846e8-3f1c8475/src@server.no_comments_merged.h
index c83aa62..2b6d100 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/4339706e-f42846e8-3f1c8475/src@server.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/4339706e-f42846e8-3f1c8475/src@server.no_comments_merged.h
@@ -127,6 +127,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER \
-  10
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER \
-  100
-#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES \
-  (100 << 20)
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER 10
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER 100
+#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES (100<<20)
@@ -134,8 +131,4 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX \
-  75
-#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS \
-  1000
-#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN \
-  (512ll * 1024 * 1024)
-#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL \
-  10
+#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX 75
+#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS 1000
+#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN (512ll*1024*1024)
+#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL 10
@@ -144,2 +137 @@ typedef long long mstime_t;
-#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC \
-  25
+#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC 25
@@ -205,2 +197 @@ typedef long long mstime_t;
-#define CLIENT_MONITOR \
-  (1 << 2)
+#define CLIENT_MONITOR (1<<2)
@@ -208,2 +199 @@ typedef long long mstime_t;
-#define CLIENT_BLOCKED \
-  (1 << 4)
+#define CLIENT_BLOCKED (1<<4)
@@ -211,4 +201,3 @@ typedef long long mstime_t;
-#define CLIENT_CLOSE_AFTER_REPLY \
-  (1 << 6)
-#define CLIENT_UNBLOCKED \
-  (1 << 7)
+#define CLIENT_CLOSE_AFTER_REPLY (1<<6)
+#define CLIENT_UNBLOCKED (1<<7)
+#define CLIENT_LUA (1<<8)
@@ -217,6 +206,3 @@ typedef long long mstime_t;
-#define CLIENT_UNIX_SOCKET \
-  (1 << 11)
-#define CLIENT_DIRTY_EXEC \
-  (1 << 12)
-#define CLIENT_MASTER_FORCE_REPLY \
-  (1 << 13)
+#define CLIENT_UNIX_SOCKET (1<<11)
+#define CLIENT_DIRTY_EXEC (1<<12)
+#define CLIENT_MASTER_FORCE_REPLY (1<<13)
@@ -231,4 +217,3 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_WRITE \
-  (1 << 21)
-#define CLIENT_REPLY_SKIP_NEXT \
-  (1 << 23)
+#define CLIENT_PENDING_WRITE (1<<21)
+#define CLIENT_REPLY_OFF (1<<22)
+#define CLIENT_REPLY_SKIP_NEXT (1<<23)
@@ -240,6 +225,4 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_READ \
-  (1 << 29)
-#define CLIENT_PENDING_COMMAND \
-  (1 << 30)
-#define CLIENT_TRACKING \
-  (1 << 31)
+#define CLIENT_PENDING_READ (1<<29)
+#define CLIENT_PENDING_COMMAND (1<<30)
+#define CLIENT_TRACKING (1<<31)
+#define CLIENT_TRACKING_BROKEN_REDIR (1ULL<<32)
@@ -259,2 +242 @@ typedef long long mstime_t;
-#define CLIENT_TYPE_OBUF_COUNT \
-  3
+#define CLIENT_TYPE_OBUF_COUNT 3
@@ -299,0 +282 @@ typedef long long mstime_t;
+#define UNUSED(V) ((void) V)
@@ -332,4 +315,2 @@ typedef long long mstime_t;
-#define MAXMEMORY_ALLKEYS_LRU \
-  ((4 << 8) | MAXMEMORY_FLAG_LRU | MAXMEMORY_FLAG_ALLKEYS)
-#define MAXMEMORY_ALLKEYS_LFU \
-  ((5 << 8) | MAXMEMORY_FLAG_LFU | MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LRU ((4<<8)|MAXMEMORY_FLAG_LRU|MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LFU ((5<<8)|MAXMEMORY_FLAG_LFU|MAXMEMORY_FLAG_ALLKEYS)
@@ -343,2 +324,2 @@ typedef long long mstime_t;
-#define SHUTDOWN_SAVE \
-  1
+#define SHUTDOWN_SAVE 1
+#define SHUTDOWN_NOSAVE 2
@@ -370,4 +351 @@ typedef long long mstime_t;
-#define NOTIFY_ALL \
-  (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | \
-   NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | \
-   NOTIFY_KEY_MISS)
+#define NOTIFY_ALL (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | NOTIFY_KEY_MISS)
@@ -374,0 +353,4 @@ typedef long long mstime_t;
+#define run_with_period(_ms_) if ((_ms_ <= 1000/server.hz) || !(server.cronloops%((_ms_)/(1000/server.hz))))
+#define serverAssertWithInfo(_c,_o,_e) ((_e)?(void)0 : (_serverAssertWithInfo(_c,_o,#_e,__FILE__,__LINE__),_exit(1)))
+#define serverAssert(_e) ((_e)?(void)0 : (_serverAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define serverPanic(...) _serverPanic(__FILE__,__LINE__,__VA_ARGS__),_exit(1)
@@ -383,0 +366,2 @@ typedef long long mstime_t;
+#define REDISMODULE_TYPE_ENCVER(id) (id & REDISMODULE_TYPE_ENCVER_MASK)
+#define REDISMODULE_TYPE_SIGN(id) ((id & ~((uint64_t)REDISMODULE_TYPE_ENCVER_MASK)) >>REDISMODULE_TYPE_ENCVER_BITS)
@@ -393,2 +377 @@ typedef void (*moduleTypeSaveFunc)(struct RedisModuleIO *io, void *value);
-typedef int (*moduleTypeAuxLoadFunc)(struct RedisModuleIO *rdb, int encver,
-                                     int when);
+typedef int (*moduleTypeAuxLoadFunc)(struct RedisModuleIO *rdb, int encver, int when);
@@ -396,4 +379,2 @@ typedef void (*moduleTypeAuxSaveFunc)(struct RedisModuleIO *rdb, int when);
-typedef void (*moduleTypeRewriteFunc)(struct RedisModuleIO *io,
-                                      struct redisObject *key, void *value);
-typedef void (*moduleTypeDigestFunc)(struct RedisModuleDigest *digest,
-                                     void *value);
+typedef void (*moduleTypeRewriteFunc)(struct RedisModuleIO *io, struct redisObject *key, void *value);
+typedef void (*moduleTypeDigestFunc)(struct RedisModuleDigest *digest, void *value);
@@ -428,0 +410,9 @@ typedef struct RedisModuleIO {
+#define moduleInitIOContext(iovar,mtype,rioptr,keyptr) do { \
+    iovar.rio = rioptr; \
+    iovar.type = mtype; \
+    iovar.bytes = 0; \
+    iovar.error = 0; \
+    iovar.ver = 0; \
+    iovar.key = keyptr; \
+    iovar.ctx = NULL; \
+} while(0);
@@ -432,0 +423,4 @@ typedef struct RedisModuleDigest {
+#define moduleInitDigestContext(mdvar) do { \
+    memset(mdvar.o,0,sizeof(mdvar.o)); \
+    memset(mdvar.x,0,sizeof(mdvar.x)); \
+} while(0);
@@ -455,0 +450,6 @@ char *getObjectTypeName(robj *);
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = OBJ_STRING; \
+    _var.encoding = OBJ_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0)
@@ -469,2 +469 @@ typedef struct redisDb {
-  list *
-      defrag_later;
+    list *defrag_later;
@@ -501,2 +500 @@ typedef struct readyList {
-#define USER_COMMAND_BITS_COUNT \
-  1024
+#define USER_COMMAND_BITS_COUNT 1024
@@ -507,2 +505 @@ typedef struct readyList {
-#define USER_FLAG_NOPASS \
-  (1 << 4)
+#define USER_FLAG_NOPASS (1<<4)
@@ -515,0 +513 @@ typedef struct user {
+} user;
@@ -552,2 +550 @@ typedef struct user {
-    char
-        replid[CONFIG_RUN_ID_SIZE + 1];
+    char replid[CONFIG_RUN_ID_SIZE+1];
@@ -580,8 +577,9 @@ typedef struct user {
-    robj *crlf, *ok, *err, *emptybulk, *czero, *cone, *pong, *space, *colon,
-        *queued, *null[4], *nullarray[4], *emptyarray, *wrongtypeerr, *nokeyerr,
-        *syntaxerr, *sameobjecterr, *outofrangeerr, *noscripterr, *loadingerr,
-        *slowscripterr, *bgsaveerr, *masterdownerr, *roslaveerr, *execaborterr,
-        *noautherr, *noreplicaserr, *busykeyerr, *oomerr, *plus, *messagebulk,
-        *pmessagebulk, *subscribebulk, *unsubscribebulk, *psubscribebulk,
-        *punsubscribebulk, *del, *unlink, *rpop, *lpop, *lpush, *rpoplpush,
-        *zpopmin, *zpopmax, *emptyscan, *select[PROTO_SHARED_SELECT_CMDS],
+    robj *crlf, *ok, *err, *emptybulk, *czero, *cone, *pong, *space,
+    *colon, *queued, *null[4], *nullarray[4],
+    *emptyarray, *wrongtypeerr, *nokeyerr, *syntaxerr, *sameobjecterr,
+    *outofrangeerr, *noscripterr, *loadingerr, *slowscripterr, *bgsaveerr,
+    *masterdownerr, *roslaveerr, *execaborterr, *noautherr, *noreplicaserr,
+    *busykeyerr, *oomerr, *plus, *messagebulk, *pmessagebulk, *subscribebulk,
+    *unsubscribebulk, *psubscribebulk, *punsubscribebulk, *del, *unlink,
+    *rpop, *lpop, *lpush, *rpoplpush, *zpopmin, *zpopmax, *emptyscan,
+    *select[PROTO_SHARED_SELECT_CMDS],
@@ -616,2 +614 @@ typedef struct user {
-  extern clientBufferLimitsConfig
-      clientBufferLimitsDefaults[CLIENT_TYPE_OBUF_COUNT];
+extern clientBufferLimitsConfig clientBufferLimitsDefaults[CLIENT_TYPE_OBUF_COUNT];
@@ -663,2 +660 @@ typedef struct user {
-#define RDB_SAVE_INFO_INIT \
-  { -1, 0, "000000000000000000000000000000", -1 }
+#define RDB_SAVE_INFO_INIT {-1,0,"000000000000000000000000000000",-1}
@@ -738,3 +734,4 @@ typedef struct user {
-    struct redisCommand *delCommand, *multiCommand, *lpushCommand, *lpopCommand,
-        *rpopCommand, *zpopminCommand, *zpopmaxCommand, *sremCommand,
-        *execCommand, *expireCommand, *pexpireCommand, *xclaimCommand,
+    struct redisCommand *delCommand, *multiCommand, *lpushCommand,
+                        *lpopCommand, *rpopCommand, *zpopminCommand,
+                        *zpopmaxCommand, *sremCommand, *execCommand,
+                        *expireCommand, *pexpireCommand, *xclaimCommand,
@@ -747,2 +744 @@ typedef struct user {
-    long long
-        stat_expired_time_cap_reached_count;
+    long long stat_expired_time_cap_reached_count;
@@ -753,6 +749,3 @@ typedef struct user {
-    long long
-        stat_active_defrag_misses;
-    long long
-        stat_active_defrag_key_hits;
-    long long
-        stat_active_defrag_key_misses;
+    long long stat_active_defrag_misses;
+    long long stat_active_defrag_key_hits;
+    long long stat_active_defrag_key_misses;
@@ -787,2 +780 @@ typedef struct user {
-    size_t
-        active_defrag_ignore_bytes;
+    size_t active_defrag_ignore_bytes;
@@ -793,4 +785,2 @@ typedef struct user {
-    unsigned long
-        active_defrag_max_scan_fields;
-    _Atomic size_t
-        client_max_querybuf_len;
+    unsigned long active_defrag_max_scan_fields;
+    _Atomic size_t client_max_querybuf_len;
@@ -998,2 +988 @@ typedef struct user {
-  typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc,
-                                int *numkeys);
+typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1050,0 +1040 @@ typedef struct user {
+#include "stream.h"
@@ -1071,2 +1061 @@ typedef struct user {
-  int *moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv,
-                                  int argc, int *numkeys);
+int *moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1079,2 +1068 @@ typedef struct user {
-  void moduleBlockedClientPipeReadable(aeEventLoop *el, int fd, void *privdata,
-                                       int mask);
+void moduleBlockedClientPipeReadable(aeEventLoop *el, int fd, void *privdata, int mask);
@@ -1084,2 +1072 @@ typedef struct user {
-  void moduleNotifyKeyspaceEvent(int type, const char *event, robj *key,
-                                 int dbid);
+void moduleNotifyKeyspaceEvent(int type, const char *event, robj *key, int dbid);
@@ -1087 +1073,0 @@ typedef struct user {
-  int moduleAllDatatypesHandleErrors();
@@ -1088,0 +1075 @@ typedef struct user {
+int moduleAllDatatypesHandleErrors();
@@ -1199,2 +1186 @@ typedef struct user {
-  listTypeIterator *listTypeInitIterator(robj *subject, long index,
-                                         unsigned char direction);
+listTypeIterator *listTypeInitIterator(robj *subject, long index, unsigned char direction);
@@ -1251,2 +1237 @@ typedef struct user {
-  int getLongFromObjectOrReply(client *c, robj *o, long *target,
-                               const char *msg);
+int getLongFromObjectOrReply(client *c, robj *o, long *target, const char *msg);
@@ -1254,4 +1239,2 @@ typedef struct user {
-  int getLongLongFromObjectOrReply(client *c, robj *o, long long *target,
-                                   const char *msg);
-  int getDoubleFromObjectOrReply(client *c, robj *o, double *target,
-                                 const char *msg);
+int getLongLongFromObjectOrReply(client *c, robj *o, long long *target, const char *msg);
+int getDoubleFromObjectOrReply(client *c, robj *o, double *target, const char *msg);
@@ -1261,2 +1244 @@ typedef struct user {
-  int getLongDoubleFromObjectOrReply(client *c, robj *o, long double *target,
-                                     const char *msg);
+int getLongDoubleFromObjectOrReply(client *c, robj *o, long double *target, const char *msg);
@@ -1268,0 +1251 @@ typedef struct user {
+#define sdsEncodedObject(objptr) (objptr->encoding == OBJ_ENCODING_RAW || objptr->encoding == OBJ_ENCODING_EMBSTR)
@@ -1273,4 +1256,2 @@ typedef struct user {
-  void replicationFeedSlavesFromMasterStream(list *slaves, char *buf,
-                                             size_t buflen);
-  void replicationFeedMonitors(client *c, list *monitors, int dictid,
-                               robj **argv, int argc);
+void replicationFeedSlavesFromMasterStream(list *slaves, char *buf, size_t buflen);
+void replicationFeedMonitors(client *c, list *monitors, int dictid, robj **argv, int argc);
@@ -1309,0 +1291 @@ typedef struct user {
+#include "rdb.h"
@@ -1313,2 +1295 @@ typedef struct user {
-  void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv,
-                          int argc);
+void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv, int argc);
@@ -1386,2 +1367 @@ typedef struct user {
-  void genericZpopCommand(client *c, robj **keyv, int keyc, int where,
-                          int emitkey, robj *countarg);
+void genericZpopCommand(client *c, robj **keyv, int keyc, int where, int emitkey, robj *countarg);
@@ -1401,2 +1381 @@ typedef struct user {
-  int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree,
-                        float *level);
+int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *level);
@@ -1412,4 +1391,2 @@ typedef struct user {
-  void propagate(struct redisCommand *cmd, int dbid, robj **argv, int argc,
-                 int flags);
-  void alsoPropagate(struct redisCommand *cmd, int dbid, robj **argv, int argc,
-                     int target);
+void propagate(struct redisCommand *cmd, int dbid, robj **argv, int argc, int flags);
+void alsoPropagate(struct redisCommand *cmd, int dbid, robj **argv, int argc, int target);
@@ -1446,2 +1423 @@ typedef struct user {
-#define RESTART_SERVER_CONFIG_REWRITE \
-  (1 << 1)
+#define RESTART_SERVER_CONFIG_REWRITE (1<<1)
@@ -1458,2 +1434 @@ typedef struct user {
-  unsigned long setTypeRandomElements(robj *set, unsigned long count,
-                                      robj *aux_set);
+unsigned long setTypeRandomElements(robj *set, unsigned long count, robj *aux_set);
@@ -1474 +1449,2 @@ typedef struct user {
-                                  unsigned char **vstr, unsigned int *vlen,
+                                unsigned char **vstr,
+                                unsigned int *vlen,
@@ -1477,3 +1453 @@ typedef struct user {
-  void hashTypeCurrentObject(hashTypeIterator *hi, int what,
-                             unsigned char **vstr, unsigned int *vlen,
-                             long long *vll);
+void hashTypeCurrentObject(hashTypeIterator *hi, int what, unsigned char **vstr, unsigned int *vlen, long long *vll);
@@ -1497,2 +1471 @@ typedef struct user {
-  void rewriteConfigRewriteLine(struct rewriteConfigState *state,
-                                const char *option, sds line, int force);
+void rewriteConfigRewriteLine(struct rewriteConfigState *state, const char *option, sds line, int force);
@@ -1528,2 +1501 @@ typedef struct user {
-  long long emptyDbGeneric(redisDb *dbarray, int dbnum, int flags,
-                           void(callback)(void *));
+long long emptyDbGeneric(redisDb *dbarray, int dbnum, int flags, void(callback)(void*));
@@ -1534,2 +1506 @@ typedef struct user {
-  unsigned int getKeysInSlot(unsigned int hashslot, robj **keys,
-                             unsigned int count);
+unsigned int getKeysInSlot(unsigned int hashslot, robj **keys, unsigned int count);
@@ -1549,2 +1520 @@ typedef struct user {
-  int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc,
-                          int *numkeys);
+int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1552,12 +1522,6 @@ typedef struct user {
-  int *zunionInterGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                          int *numkeys);
-  int *evalGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                   int *numkeys);
-  int *sortGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                   int *numkeys);
-  int *migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                      int *numkeys);
-  int *georadiusGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                        int *numkeys);
-  int *xreadGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                    int *numkeys);
+int *zunionInterGetKeys(struct redisCommand *cmd,robj **argv, int argc, int *numkeys);
+int *evalGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *sortGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *georadiusGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *xreadGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1571,3 +1535 @@ typedef struct user {
-  int clusterSendModuleMessageToTarget(const char *target, uint64_t module_id,
-                                       uint8_t type, unsigned char *payload,
-                                       uint32_t len);
+int clusterSendModuleMessageToTarget(const char *target, uint64_t module_id, uint8_t type, unsigned char *payload, uint32_t len);
@@ -1592,2 +1554 @@ typedef struct user {
-  int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout,
-                                  int unit);
+int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int unit);
@@ -1597,2 +1558 @@ typedef struct user {
-  void blockForKeys(client *c, int btype, robj **keys, int numkeys,
-                    mstime_t timeout, robj *target, streamID *ids);
+void blockForKeys(client *c, int btype, robj **keys, int numkeys, mstime_t timeout, robj *target, streamID *ids);
@@ -1811,2 +1771,7 @@ typedef struct user {
-  void _serverAssertWithInfo(const client *c, const robj *o, const char *estr,
-                             const char *file, int line);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
+void _serverAssertWithInfo(const client *c, const robj *o, const char *estr, const char *file, int line);
@@ -1826,0 +1792,4 @@ typedef struct user {
+#define redisDebug(fmt,...) \
+    printf("DEBUG %s:%d > " fmt "\n", __FILE__, __LINE__, __VA_ARGS__)
+#define redisDebugMark() \
+    printf("-- MARK %s:%d --\n", __FILE__, __LINE__)
