--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@db_impl.no_comments_merged.h
@@ -245,0 +255,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -302,0 +317,4 @@ class DBImpl : public DB {
+  using DB::GetPropertiesOfAllTables;
+  virtual Status GetPropertiesOfAllTables(ColumnFamilyHandle* column_family,
+                                          TablePropertiesCollection* props)
+      override;
@@ -306,8 +323,0 @@ class DBImpl : public DB {
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
-  using DB::GetPropertiesOfAllTables;
-  virtual Status GetPropertiesOfAllTables(
-      ColumnFamilyHandle* column_family,
-      TablePropertiesCollection* props) override;
