--- a/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@version_set.no_comments_mergegen.cc
+++ b/./output/rocksdb/8555ce2d-7efdd9ef-8c4a3bfa/db@version_set.no_comments_merged.cc
@@ -201 +201,7 @@ bool Version::PrefixMayMatch(const ReadOptions& options,
-const EncodedFileMetaData* encoded_meta = reinterpret_cast<const EncodedFileMetaData*>( level_iter->value().data()); FileMetaData meta(encoded_meta->number, encoded_meta->file_size); meta.table_reader_handle = encoded_meta->table_reader_handle; may_match = cfd_->table_cache()->PrefixMayMatch(options, cfd_->internal_comparator(), *meta_file, internal_prefix, nullptr);
+    const EncodedFileMetaData* encoded_meta =
+        reinterpret_cast<const EncodedFileMetaData*>(
+            level_iter->value().data());
+    FileMetaData meta(encoded_meta->number, encoded_meta->file_size);
+    meta.table_reader_handle = encoded_meta->table_reader_handle;
+    may_match = cfd_->table_cache()->PrefixMayMatch(
+        options, cfd_->internal_comparator(), meta, internal_prefix, nullptr);
