--- a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@AudioContainers.no_comments_mergegen.h
+++ b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@AudioContainers.no_comments_merged.h
@@ -32 +47,44 @@ static inline ChannelMaskSet asOutMask(const ChannelMaskSet& channelMasks) {
-static inline bool isSingleDeviceType(const DeviceTypeSet& deviceTypes, audio_devices_t deviceType) { return deviceTypes.size() == 1 && *(deviceTypes.begin()) == deviceType; } typedef bool (*DeviceTypeUnaryPredicate)(audio_devices_t); static inline bool isSingleDeviceType(const DeviceTypeSet& deviceTypes, DeviceTypeUnaryPredicate p) { return deviceTypes.size() == 1 && p(*(deviceTypes.begin())); } static inline bool areAllOfSameDeviceType(const DeviceTypeSet& deviceTypes, std::function<bool(audio_devices_t)> p) { return std::all_of(deviceTypes.begin(), deviceTypes.end(), p); } static inline void resetDeviceTypes(DeviceTypeSet& deviceTypes, audio_devices_t typeToAdd) { deviceTypes.clear(); deviceTypes.insert(typeToAdd); }
+static inline bool isSingleDeviceType(const DeviceTypeSet& deviceTypes,
+                                      audio_devices_t deviceType) {
+    return deviceTypes.size() == 1 && *(deviceTypes.begin()) == deviceType;
+}
+typedef bool (*DeviceTypeUnaryPredicate)(audio_devices_t);
+static inline bool isSingleDeviceType(const DeviceTypeSet& deviceTypes,
+                                      DeviceTypeUnaryPredicate p) {
+    return deviceTypes.size() == 1 && p(*(deviceTypes.begin()));
+}
+static inline bool areAllOfSameDeviceType(const DeviceTypeSet& deviceTypes,
+                                          std::function<bool(audio_devices_t)> p) {
+    return std::all_of(deviceTypes.begin(), deviceTypes.end(), p);
+}
+static inline void resetDeviceTypes(DeviceTypeSet& deviceTypes, audio_devices_t typeToAdd) {
+    deviceTypes.clear();
+    deviceTypes.insert(typeToAdd);
+}
+static inline audio_devices_t deviceTypesToBitMask(const DeviceTypeSet& deviceTypes) {
+    audio_devices_t types = AUDIO_DEVICE_NONE;
+    for (auto deviceType : deviceTypes) {
+        types |= deviceType;
+    }
+    return types;
+}
+static inline DeviceTypeSet deviceTypesFromBitMask(audio_devices_t types) {
+    DeviceTypeSet deviceTypes;
+    if ((types & AUDIO_DEVICE_BIT_IN) == 0) {
+        for (auto deviceType : AUDIO_DEVICE_OUT_ALL_ARRAY) {
+            if ((types & deviceType) == deviceType) {
+                deviceTypes.insert(deviceType);
+            }
+        }
+    } else {
+        for (auto deviceType : AUDIO_DEVICE_IN_ALL_ARRAY) {
+            if ((types & deviceType) == deviceType) {
+                deviceTypes.insert(deviceType);
+            }
+        }
+    }
+    return deviceTypes;
+}
+bool deviceTypesToString(const DeviceTypeSet& deviceTypes, std::string &str);
+std::string dumpDeviceTypes(const DeviceTypeSet& deviceTypes);
+std::string toString(const DeviceTypeSet& deviceTypes);
