diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@AudioContainers.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@AudioContainers.no_comments_merged.h
index e2722de..11eb08e 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@AudioContainers.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@AudioContainers.no_comments_merged.h
@@ -24 +24,2 @@ static std::vector<T> Intersection(const std::set<T>& a, const std::set<T>& b) {
-  std::set_intersection(a.begin(), a.end(), b.begin(), b.end(),
+    std::set_intersection(a.begin(), a.end(),
+                          b.begin(), b.end(),
@@ -55,2 +56,2 @@ static inline bool isSingleDeviceType(const DeviceTypeSet& deviceTypes,
-static inline bool areAllOfSameDeviceType(
-    const DeviceTypeSet& deviceTypes, std::function<bool(audio_devices_t)> p) {
+static inline bool areAllOfSameDeviceType(const DeviceTypeSet& deviceTypes,
+                                          std::function<bool(audio_devices_t)> p) {
@@ -59,2 +60 @@ static inline bool areAllOfSameDeviceType(
-static inline void resetDeviceTypes(DeviceTypeSet& deviceTypes,
-                                    audio_devices_t typeToAdd) {
+static inline void resetDeviceTypes(DeviceTypeSet& deviceTypes, audio_devices_t typeToAdd) {
@@ -64,2 +64 @@ static inline void resetDeviceTypes(DeviceTypeSet& deviceTypes,
-static inline audio_devices_t deviceTypesToBitMask(
-    const DeviceTypeSet& deviceTypes) {
+static inline audio_devices_t deviceTypesToBitMask(const DeviceTypeSet& deviceTypes) {
