--- a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_conflict.cpp
+++ b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_merged.cpp
@@ -383 +382,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-<<<<<<< HEAD
@@ -389,7 +387,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-||||||| 3944a1b7f7
-    const String8& address = devices.getFirstValidAddress();
-    audio_devices_t device = devices.types();
-=======
-    const String8& address = devices.getFirstValidAddress();
-    DeviceTypeSet deviceTypes = devices.types();
->>>>>>> cb0fcde9
@@ -420 +411,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-    audio_devices_t deviceType = deviceTypesToBitMask(deviceTypes);
@@ -424 +414,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-<<<<<<< HEAD
@@ -426,7 +415,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-||||||| 3944a1b7f7
-                                                   &device,
-                                                   address,
-=======
-                                                   &deviceType,
-                                                   address,
->>>>>>> cb0fcde9
@@ -435,12 +417,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-<<<<<<< HEAD
-||||||| 3944a1b7f7
-    LOG_ALWAYS_FATAL_IF(mDevices.types() != device,
-                        "%s openOutput returned device %08x when given device %08x",
-                        __FUNCTION__, mDevices.types(), device);
-=======
-    deviceTypes = deviceTypesFromBitMask(deviceType);
-    LOG_ALWAYS_FATAL_IF(mDevices.types() != deviceTypes,
-                        "%s openOutput returned device %s when given device %s",
-                        __FUNCTION__, dumpDeviceTypes(mDevices.types()).c_str(),
-                        dumpDeviceTypes(deviceTypes).c_str());
->>>>>>> cb0fcde9
@@ -449 +419,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-<<<<<<< HEAD
@@ -454,7 +423,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-||||||| 3944a1b7f7
-                            "%s openOutput returned output handle %d for device %08x",
-                            __FUNCTION__, *output, device);
-=======
-                            "%s openOutput returned output handle %d for device %s",
-                            __FUNCTION__, *output, dumpDeviceTypes(deviceTypes).c_str());
->>>>>>> cb0fcde9
