--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_merged.cpp
@@ -12 +11,0 @@
-#define APM_AUDIO_OUT_DEVICE_REMOTE_ALL AUDIO_DEVICE_OUT_REMOTE_SUBMIX
@@ -114 +112,0 @@ bool AudioOutputDescriptor::setVolume(float volumeDb, VolumeSource volumeSource,
-                                      audio_devices_t ,
@@ -331,3 +331,2 @@ void SwAudioOutputDescriptor::toAudioPort(struct audio_port *port) const {
-bool SwAudioOutputDescriptor::setVolume(float volumeDb, VolumeSource vs,
-                                        const StreamTypeVector &streamTypes,
-                                        audio_devices_t device,
+bool SwAudioOutputDescriptor::setVolume(float volumeDb,
+                                        VolumeSource vs, const StreamTypeVector &streamTypes,
@@ -387 +382,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-<<<<<<< HEAD
@@ -393,7 +387,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-||||||| 3944a1b7f7
-  const String8 &address = devices.getFirstValidAddress();
-  audio_devices_t device = devices.types();
-=======
-  const String8 &address = devices.getFirstValidAddress();
-  DeviceTypeSet deviceTypes = devices.types();
->>>>>>> cb0fcde90788a378765af0f9fd010c7de886bff5
@@ -423,3 +411 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-        mDevices.toString().c_str(), mProfile.get(),
-        mProfile->getName().c_str());
-  audio_devices_t deviceType = deviceTypesToBitMask(deviceTypes);
+          mDevices.toString().c_str(), mProfile.get(), mProfile->getName().c_str());
@@ -427,2 +413,2 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-                                                 output, &lConfig,
-<<<<<<< HEAD
+                                                   output,
+                                                   &lConfig,
@@ -430,19 +416,2 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-||||||| 3944a1b7f7
-                                                 &device, address,
-=======
-                                                 &deviceType, address,
->>>>>>> cb0fcde90788a378765af0f9fd010c7de886bff5
-                                                 &mLatency, mFlags);
-<<<<<<< HEAD
-||||||| 3944a1b7f7
-  LOG_ALWAYS_FATAL_IF(
-      mDevices.types() != device,
-      "%s openOutput returned device %08x when given device %08x", __FUNCTION__,
-      mDevices.types(), device);
-=======
-  deviceTypes = deviceTypesFromBitMask(deviceType);
-  LOG_ALWAYS_FATAL_IF(mDevices.types() != deviceTypes,
-                      "%s openOutput returned device %s when given device %s",
-                      __FUNCTION__, dumpDeviceTypes(mDevices.types()).c_str(),
-                      dumpDeviceTypes(deviceTypes).c_str());
->>>>>>> cb0fcde90788a378765af0f9fd010c7de886bff5
+                                                   &mLatency,
+                                                   mFlags);
@@ -450,3 +419 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-    LOG_ALWAYS_FATAL_IF(
-        *output == AUDIO_IO_HANDLE_NONE,
-<<<<<<< HEAD
+        LOG_ALWAYS_FATAL_IF(*output == AUDIO_IO_HANDLE_NONE,
@@ -457,7 +423,0 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-||||||| 3944a1b7f7
-        "%s openOutput returned output handle %d for device %08x", __FUNCTION__,
-        *output, device);
-=======
-        "%s openOutput returned output handle %d for device %s", __FUNCTION__,
-        *output, dumpDeviceTypes(deviceTypes).c_str());
->>>>>>> cb0fcde90788a378765af0f9fd010c7de886bff5
@@ -719 +693 @@ void HwAudioOutputCollection::dump(String8 *dst) const {
-}
+};
