--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@include@Volume.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@include@Volume.no_comments_merged.h
@@ -36,3 +37,2 @@ class Volume {
-  static audio_devices_t getDeviceForVolume(
-      const android::DeviceTypeSet& deviceTypes) {
-    audio_devices_t deviceType = AUDIO_DEVICE_NONE;
+    static audio_devices_t getDeviceForVolume(const android::DeviceTypeSet& deviceTypes)
+    {
@@ -40,25 +40 @@ class Volume {
-      deviceType = AUDIO_DEVICE_OUT_SPEAKER;
-    } else if (deviceTypes.size() > 1) {
-      if (deviceTypes.count(AUDIO_DEVICE_OUT_SPEAKER) != 0) {
-        deviceType = AUDIO_DEVICE_OUT_SPEAKER;
-      } else if (deviceTypes.count(AUDIO_DEVICE_OUT_SPEAKER_SAFE) != 0) {
-        deviceType = AUDIO_DEVICE_OUT_SPEAKER_SAFE;
-      } else if (deviceTypes.count(AUDIO_DEVICE_OUT_HDMI_ARC) != 0) {
-        deviceType = AUDIO_DEVICE_OUT_HDMI_ARC;
-      } else if (deviceTypes.count(AUDIO_DEVICE_OUT_AUX_LINE) != 0) {
-        deviceType = AUDIO_DEVICE_OUT_AUX_LINE;
-      } else if (deviceTypes.count(AUDIO_DEVICE_OUT_SPDIF) != 0) {
-        deviceType = AUDIO_DEVICE_OUT_SPDIF;
-      } else {
-        std::vector<audio_devices_t> a2dpDevices = android::Intersection(
-            deviceTypes, android::getAudioDeviceOutAllA2dpSet());
-        if (a2dpDevices.size() > 1) {
-          ALOGW("getDeviceForVolume() invalid device combination: %s",
-                android::dumpDeviceTypes(deviceTypes).c_str());
-        }
-        if (!a2dpDevices.empty()) {
-          deviceType = a2dpDevices[0];
-        }
-      }
-    } else {
-      deviceType = *(deviceTypes.begin());
+            return AUDIO_DEVICE_OUT_SPEAKER;
@@ -65,0 +42 @@ class Volume {
+        audio_devices_t deviceType = apm_extract_one_audio_device(deviceTypes);
@@ -70,2 +47 @@ class Volume {
-             "getDeviceForVolume() invalid device combination: %s, returning "
-             "AUDIO_DEVICE_NONE",
+                 "getDeviceForVolume() invalid device combination: %s, returning AUDIO_DEVICE_NONE",
