diff --git a/./output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_mergegen.cc b/./output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_merged.cc
index 83ef13a..539f939 100644
--- a/./output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_mergegen.cc
+++ b/./output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3 +3 @@ namespace grpc {
-namespace grpc {std::string Version() { return "1.32.0-dev"; }
+std::string Version() { return "1.32.0-dev"; }
