diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/0526d074-d768fc25-32186950/proc.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/0526d074-d768fc25-32186950/proc.no_comments_merged.c
index b16d5cd..e4057a0 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/0526d074-d768fc25-32186950/proc.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/0526d074-d768fc25-32186950/proc.no_comments_merged.c
@@ -4,2 +3,0 @@
-#include <event.h>
-#include <imsg.h>
@@ -43 +41,3 @@ static void proc_update_event(struct tmuxpeer *);
-static void proc_event_cb(__unused int fd, short events, void *arg) {
+static void
+proc_event_cb(__unused int fd, short events, void *arg)
+{
@@ -48 +48,2 @@ static void proc_event_cb(__unused int fd, short events, void *arg) {
-    if (((n = imsg_read(&peer->ibuf)) == -1 && errno != EAGAIN) || n == 0) {
+  if (((n = imsg_read(&peer->ibuf)) == -1 && errno != EAGAIN) ||
+      n == 0) {
@@ -57 +58,2 @@ static void proc_event_cb(__unused int fd, short events, void *arg) {
-      if (n == 0) break;
+   if (n == 0)
+    break;
@@ -60 +62,2 @@ static void proc_event_cb(__unused int fd, short events, void *arg) {
-        if (imsg.fd != -1) close(imsg.fd);
+    if (imsg.fd != -1)
+     close(imsg.fd);
@@ -80 +83,3 @@ static void proc_event_cb(__unused int fd, short events, void *arg) {
-static void proc_signal_cb(int signo, __unused short events, void *arg) {
+static void
+proc_signal_cb(int signo, __unused short events, void *arg)
+{
@@ -84 +89,3 @@ static void proc_signal_cb(int signo, __unused short events, void *arg) {
-static int peer_check_version(struct tmuxpeer *peer, struct imsg *imsg) {
+static int
+peer_check_version(struct tmuxpeer *peer, struct imsg *imsg)
+{
@@ -95 +102,3 @@ static int peer_check_version(struct tmuxpeer *peer, struct imsg *imsg) {
-static void proc_update_event(struct tmuxpeer *peer) {
+static void
+proc_update_event(struct tmuxpeer *peer)
+{
@@ -99 +108,2 @@ static void proc_update_event(struct tmuxpeer *peer) {
-  if (peer->ibuf.w.queued > 0) events |= EV_WRITE;
+ if (peer->ibuf.w.queued > 0)
+  events |= EV_WRITE;
@@ -103,2 +113,4 @@ static void proc_update_event(struct tmuxpeer *peer) {
-int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-              size_t len) {
+int
+proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
+    size_t len)
+{
@@ -108 +120,2 @@ int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-  if (peer->flags & PEER_BAD) return (-1);
+ if (peer->flags & PEER_BAD)
+  return (-1);
@@ -111 +124,2 @@ int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-  if (retval != 1) return (-1);
+ if (retval != 1)
+  return (-1);
@@ -115 +129,3 @@ int proc_send(struct tmuxpeer *peer, enum msgtype type, int fd, const void *buf,
-struct tmuxproc *proc_start(const char *name) {
+struct tmuxproc *
+proc_start(const char *name)
+{
@@ -120 +136,2 @@ struct tmuxproc *proc_start(const char *name) {
-  if (uname(&u) < 0) memset(&u, 0, sizeof u);
+ if (uname(&u) < 0)
+  memset(&u, 0, sizeof u);
@@ -124,2 +141 @@ struct tmuxproc *proc_start(const char *name) {
-  log_debug(
-      "using libevent %s (%s)"
+ log_debug("using libevent %s (%s)"
@@ -132,2 +148 @@ struct tmuxproc *proc_start(const char *name) {
-      ,
-      event_get_version(), event_get_method()
+     , event_get_version(), event_get_method()
@@ -135,2 +150 @@ struct tmuxproc *proc_start(const char *name) {
-                               ,
-      utf8proc_version()
+     , utf8proc_version ()
@@ -144 +158,3 @@ struct tmuxproc *proc_start(const char *name) {
-void proc_loop(struct tmuxproc *tp, int (*loopcb)(void)) {
+void
+proc_loop(struct tmuxproc *tp, int (*loopcb)(void))
+{
@@ -146 +162,2 @@ void proc_loop(struct tmuxproc *tp, int (*loopcb)(void)) {
-  do event_loop(EVLOOP_ONCE);
+ do
+  event_loop(EVLOOP_ONCE);
@@ -150 +167,3 @@ void proc_loop(struct tmuxproc *tp, int (*loopcb)(void)) {
-void proc_exit(struct tmuxproc *tp) {
+void
+proc_exit(struct tmuxproc *tp)
+{
@@ -156 +175,3 @@ void proc_exit(struct tmuxproc *tp) {
-void proc_set_signals(struct tmuxproc *tp, void (*signalcb)(int)) {
+void
+proc_set_signals(struct tmuxproc *tp, void (*signalcb)(int))
+{
@@ -185 +206,3 @@ void proc_set_signals(struct tmuxproc *tp, void (*signalcb)(int)) {
-void proc_clear_signals(struct tmuxproc *tp, int defaults) {
+void
+proc_clear_signals(struct tmuxproc *tp, int defaults)
+{
@@ -213,3 +236,4 @@ void proc_clear_signals(struct tmuxproc *tp, int defaults) {
-struct tmuxpeer *proc_add_peer(struct tmuxproc *tp, int fd,
-                               void (*dispatchcb)(struct imsg *, void *),
-                               void *arg) {
+struct tmuxpeer *
+proc_add_peer(struct tmuxproc *tp, int fd,
+    void (*dispatchcb)(struct imsg *, void *), void *arg)
+{
@@ -228 +252,3 @@ struct tmuxpeer *proc_add_peer(struct tmuxproc *tp, int fd,
-void proc_remove_peer(struct tmuxpeer *peer) {
+void
+proc_remove_peer(struct tmuxpeer *peer)
+{
@@ -236,3 +262,13 @@ void proc_remove_peer(struct tmuxpeer *peer) {
-void proc_kill_peer(struct tmuxpeer *peer) { peer->flags |= PEER_BAD; }
-void proc_toggle_log(struct tmuxproc *tp) { log_toggle(tp->name); }
-pid_tproc_fork_and_daemon(int *fd) {
+void
+proc_kill_peer(struct tmuxpeer *peer)
+{
+ peer->flags |= PEER_BAD;
+}
+void
+proc_toggle_log(struct tmuxproc *tp)
+{
+ log_toggle(tp->name);
+}
+pid_t
+proc_fork_and_daemon(int *fd)
+{
@@ -249 +285,2 @@ pid_tproc_fork_and_daemon(int *fd) {
-      if (daemon(1, 0) != 0) fatal("daemon failed");
+  if (daemon(1, 0) != 0)
+   fatal("daemon failed");
