diff --git a/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_mergegen.h b/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_merged.h
index cd03e35..eee842f 100644
--- a/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_mergegen.h
+++ b/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_merged.h
@@ -172 +172,10 @@ class SnapshotManager final : public ISnapshotManager {
-enum class SnapshotDriver { DM_SNAPSHOT, DM_USER, };
+    enum class SnapshotDriver {
+        DM_SNAPSHOT,
+        DM_USER,
+    };
+    using MergeConsistencyChecker =
+            std::function<MergeFailureCode(const std::string& name, const SnapshotStatus& status)>;
+    void set_merge_consistency_checker(MergeConsistencyChecker checker) {
+        merge_consistency_checker_ = checker;
+    }
+    MergeConsistencyChecker merge_consistency_checker() const { return merge_consistency_checker_; }
@@ -383 +392,2 @@ enum class SnapshotDriver { DM_SNAPSHOT, DM_USER, };
-std::optional<bool> is_snapshot_userspace_; MergeConsistencyChecker merge_consistency_checker_;
+    std::optional<bool> is_snapshot_userspace_;
+    MergeConsistencyChecker merge_consistency_checker_;
