--- a/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@snapshot.no_comments_conflict.cpp
+++ b/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@snapshot.no_comments_merged.cpp
@@ -88 +87,0 @@ std::unique_ptr<SnapshotManager> SnapshotManager::NewForFirstStageMount(IDeviceI
-<<<<<<< HEAD
@@ -92,8 +90,0 @@ SnapshotManager::SnapshotManager(IDeviceInfo* device)
-||||||| 6834fe66d
-SnapshotManager::SnapshotManager(IDeviceInfo* device) : device_(device) {
-    metadata_dir_ = device_->GetMetadataDir();
-=======
-SnapshotManager::SnapshotManager(IDeviceInfo* device) : device_(device) {
-    metadata_dir_ = device_->GetMetadataDir();
-    merge_consistency_checker_ = android::snapshot::CheckMergeConsistency;
->>>>>>> 739f4f5f
