diff --git a/./output/frameworks_av/e52f5c58-94c719f4-38b856f7/services@camera@libcameraservice@CameraService.no_comments_mergegen.h b/./output/frameworks_av/e52f5c58-94c719f4-38b856f7/services@camera@libcameraservice@CameraService.no_comments_merged.h
index 3cc8c7f..fe3079a 100644
--- a/./output/frameworks_av/e52f5c58-94c719f4-38b856f7/services@camera@libcameraservice@CameraService.no_comments_mergegen.h
+++ b/./output/frameworks_av/e52f5c58-94c719f4-38b856f7/services@camera@libcameraservice@CameraService.no_comments_merged.h
@@ -33,0 +34 @@
+#include "utils/AttributionAndPermissionUtils.h"
@@ -75 +76,3 @@ public:
-                                cameraServiceProxyWrapper = nullptr);
+                                cameraServiceProxyWrapper = nullptr,
+                                std::shared_ptr<AttributionAndPermissionUtils>
+                                attributionAndPermissionUtils = nullptr);
@@ -237,0 +241 @@ public:
+                std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
@@ -313,0 +318 @@ public:
+                std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
@@ -367 +371,0 @@ private:
-    bool isAutomotiveExteriorSystemCamera(const std::string& cameraId) const;
@@ -471 +475,5 @@ private:
-explicit SensorPrivacyPolicy(wp<CameraService> service, std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils) : AttributionAndPermissionUtilsEncapsulator(attributionAndPermissionUtils), mService(service), mSensorPrivacyEnabled(false),
+            explicit SensorPrivacyPolicy(wp<CameraService> service,
+                    std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils)
+                    : AttributionAndPermissionUtilsEncapsulator(attributionAndPermissionUtils),
+                      mService(service),
+                      mSensorPrivacyEnabled(false),
