--- a/./output/tmux/9cd45dda-8a4a2153-40ad11b2/tty-term.no_comments_conflict.c
+++ b/./output/tmux/9cd45dda-8a4a2153-40ad11b2/tty-term.no_comments_merged.c
@@ -470 +469,0 @@ tty_term_create(struct tty *tty, char *name, char **caps, u_int ncaps,
-<<<<<<< HEAD
@@ -475,4 +473,0 @@ tty_term_create(struct tty *tty, char *name, char **caps, u_int ncaps,
-||||||| 40ad11b2
- del_curterm(cur_term);
-=======
->>>>>>> 8a4a2153
