diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/9cd45dda-8a4a2153-40ad11b2/tty-term.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/9cd45dda-8a4a2153-40ad11b2/tty-term.no_comments_merged.c
index 002f7de..f0eeee1 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/9cd45dda-8a4a2153-40ad11b2/tty-term.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/9cd45dda-8a4a2153-40ad11b2/tty-term.no_comments_merged.c
@@ -2 +1,0 @@
-#include <curses.h>
@@ -10 +8,0 @@
-#include <vis.h>
@@ -260,3 +258,10 @@ static const struct tty_term_code_entry tty_term_codes[] = {
-    [TTYC_XT] = {TTYCODE_FLAG, "XT"}};
-u_int tty_term_ncodes(void) { return (nitems(tty_term_codes)); }
-static char *tty_term_strip(const char *s) {
+ [TTYC_XT] = { TTYCODE_FLAG, "XT" }
+};
+u_int
+tty_term_ncodes(void)
+{
+ return (nitems(tty_term_codes));
+}
+static char *
+tty_term_strip(const char *s)
+{
@@ -266 +271,2 @@ static char *tty_term_strip(const char *s) {
-  if (strchr(s, '$') == NULL) return (xstrdup(s));
+ if (strchr(s, '$') == NULL)
+  return (xstrdup(s));
@@ -270,3 +276,6 @@ static char *tty_term_strip(const char *s) {
-      while (*ptr != '\0' && *ptr != '>') ptr++;
-      if (*ptr == '>') ptr++;
-      if (*ptr == '\0') break;
+   while (*ptr != '\0' && *ptr != '>')
+    ptr++;
+   if (*ptr == '>')
+    ptr++;
+   if (*ptr == '\0')
+    break;
@@ -275 +284,2 @@ static char *tty_term_strip(const char *s) {
-    if (len == (sizeof buf) - 1) break;
+  if (len == (sizeof buf) - 1)
+   break;
@@ -280 +290,3 @@ static char *tty_term_strip(const char *s) {
-static char *tty_term_override_next(const char *s, size_t *offset) {
+static char *
+tty_term_override_next(const char *s, size_t *offset)
+{
@@ -283 +295,2 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-  if (s[at] == '\0') return (NULL);
+ if (s[at] == '\0')
+  return (NULL);
@@ -295 +308,2 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-    if (n == (sizeof value) - 1) return (NULL);
+  if (n == (sizeof value) - 1)
+   return (NULL);
@@ -304,2 +318,3 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-void tty_term_apply(struct tty_term *term, const char *capabilities,
-                    int quiet) {
+void
+tty_term_apply(struct tty_term *term, const char *capabilities, int quiet)
+{
@@ -314 +329,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-    if (*s == '\0') continue;
+  if (*s == '\0')
+   continue;
@@ -339 +355,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-      if (strcmp(s, ent->name) != 0) continue;
+   if (strcmp(s, ent->name) != 0)
+    continue;
@@ -349 +366,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-          if (code->type == TTYCODE_STRING) free(code->value.string);
+    if (code->type == TTYCODE_STRING)
+     free(code->value.string);
@@ -355 +373,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-          if (errstr != NULL) break;
+    if (errstr != NULL)
+     break;
@@ -368 +387,3 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-void tty_term_apply_overrides(struct tty_term *term) {
+void
+tty_term_apply_overrides(struct tty_term *term)
+{
@@ -387,2 +408,4 @@ void tty_term_apply_overrides(struct tty_term *term) {
-struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-                                 u_int ncaps, int *feat, char **cause) {
+struct tty_term *
+tty_term_create(struct tty *tty, char *name, char **caps, u_int ncaps,
+    int *feat, char **cause)
+{
@@ -407 +430,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-    if (namelen == 0) continue;
+  if (namelen == 0)
+   continue;
@@ -411,2 +435,4 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-      if (strncmp(ent->name, caps[i], namelen) != 0) continue;
-      if (ent->name[namelen] != '\0') continue;
+   if (strncmp(ent->name, caps[i], namelen) != 0)
+    continue;
+   if (ent->name[namelen] != '\0')
+    continue;
@@ -444 +469,0 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-<<<<<<< HEAD
@@ -449,4 +473,0 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-  |||||||
-      del_curterm(cur_term);
-=======
->>>>>>> 8a4a2153fdec71397fe555073d4d77ee68947fff
@@ -468 +489,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-      (!tty_term_has(term, TTYC_SETRGBF) || !tty_term_has(term, TTYC_SETRGBB)))
+     (!tty_term_has(term, TTYC_SETRGBF) ||
+     !tty_term_has(term, TTYC_SETRGBB)))
@@ -470 +492,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-  if (tty_term_has(term, TTYC_SETRGBF) && tty_term_has(term, TTYC_SETRGBB))
+ if (tty_term_has(term, TTYC_SETRGBF) &&
+     tty_term_has(term, TTYC_SETRGBB))
@@ -474 +497,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-  if (!tty_term_flag(term, TTYC_AM)) term->flags |= TERM_NOAM;
+ if (!tty_term_flag(term, TTYC_AM))
+  term->flags |= TERM_NOAM;
@@ -489 +513,3 @@ error:
-void tty_term_free(struct tty_term *term) {
+void
+tty_term_free(struct tty_term *term)
+{
@@ -501,2 +527,4 @@ void tty_term_free(struct tty_term *term) {
-int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-                       char **cause) {
+int
+tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
+    char **cause)
+{
@@ -511 +539,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        xasprintf(cause, "can't use hardcopy terminal: %s", name);
+   xasprintf(cause, "can't use hardcopy terminal: %s",
+       name);
@@ -514 +543,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        xasprintf(cause, "missing or unsuitable terminal: %s", name);
+   xasprintf(cause, "missing or unsuitable terminal: %s",
+       name);
@@ -534 +564,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (s == NULL || s == (char *)-1) continue;
+   if (s == NULL || s == (char *)-1)
+    continue;
@@ -538 +569,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (n == -1 || n == -2) continue;
+   if (n == -1 || n == -2)
+    continue;
@@ -544 +576,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (n == -1) continue;
+   if (n == -1)
+    continue;
@@ -561 +594,3 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-void tty_term_free_list(char **caps, u_int ncaps) {
+void
+tty_term_free_list(char **caps, u_int ncaps)
+{
@@ -563 +598,2 @@ void tty_term_free_list(char **caps, u_int ncaps) {
-  for (i = 0; i < ncaps; i++) free(caps[i]);
+ for (i = 0; i < ncaps; i++)
+  free(caps[i]);
@@ -566 +602,3 @@ void tty_term_free_list(char **caps, u_int ncaps) {
-int tty_term_has(struct tty_term *term, enum tty_code_code code) {
+int
+tty_term_has(struct tty_term *term, enum tty_code_code code)
+{
@@ -569,2 +607,5 @@ int tty_term_has(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_string(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return ("");
+const char *
+tty_term_string(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return ("");
@@ -575,27 +616,32 @@ const char *tty_term_string(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_string1(struct tty_term *term, enum tty_code_code code,
-                             int a) {
-  return (
-      tparm((char *)tty_term_string(term, code), a, 0, 0, 0, 0, 0, 0, 0, 0));
-}
-const char *tty_term_string2(struct tty_term *term, enum tty_code_code code,
-                             int a, int b) {
-  return (
-      tparm((char *)tty_term_string(term, code), a, b, 0, 0, 0, 0, 0, 0, 0));
-}
-const char *tty_term_string3(struct tty_term *term, enum tty_code_code code,
-                             int a, int b, int c) {
-  return (
-      tparm((char *)tty_term_string(term, code), a, b, c, 0, 0, 0, 0, 0, 0));
-}
-const char *tty_term_ptr1(struct tty_term *term, enum tty_code_code code,
-                          const void *a) {
-  return (tparm((char *)tty_term_string(term, code), (long)a, 0, 0, 0, 0, 0, 0,
-                0, 0));
-}
-const char *tty_term_ptr2(struct tty_term *term, enum tty_code_code code,
-                          const void *a, const void *b) {
-  return (tparm((char *)tty_term_string(term, code), (long)a, (long)b, 0, 0, 0,
-                0, 0, 0, 0));
-}
-int tty_term_number(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return (0);
+const char *
+tty_term_string1(struct tty_term *term, enum tty_code_code code, int a)
+{
+ return (tparm((char *) tty_term_string(term, code), a, 0, 0, 0, 0, 0, 0, 0, 0));
+}
+const char *
+tty_term_string2(struct tty_term *term, enum tty_code_code code, int a, int b)
+{
+ return (tparm((char *) tty_term_string(term, code), a, b, 0, 0, 0, 0, 0, 0, 0));
+}
+const char *
+tty_term_string3(struct tty_term *term, enum tty_code_code code, int a, int b,
+    int c)
+{
+ return (tparm((char *) tty_term_string(term, code), a, b, c, 0, 0, 0, 0, 0, 0));
+}
+const char *
+tty_term_ptr1(struct tty_term *term, enum tty_code_code code, const void *a)
+{
+ return (tparm((char *) tty_term_string(term, code), (long)a, 0, 0, 0, 0, 0, 0, 0, 0));
+}
+const char *
+tty_term_ptr2(struct tty_term *term, enum tty_code_code code, const void *a,
+    const void *b)
+{
+ return (tparm((char *) tty_term_string(term, code), (long)a, (long)b, 0, 0, 0, 0, 0, 0, 0));
+}
+int
+tty_term_number(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return (0);
@@ -606,3 +652,7 @@ int tty_term_number(struct tty_term *term, enum tty_code_code code) {
-int tty_term_flag(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return (0);
-  if (term->codes[code].type != TTYCODE_FLAG) fatalx("not a flag: %d", code);
+int
+tty_term_flag(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return (0);
+ if (term->codes[code].type != TTYCODE_FLAG)
+  fatalx("not a flag: %d", code);
@@ -611 +661,3 @@ int tty_term_flag(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
+const char *
+tty_term_describe(struct tty_term *term, enum tty_code_code code)
+{
@@ -616,2 +668,2 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: [missing]", code,
-                tty_term_codes[code].name);
+  xsnprintf(s, sizeof s, "%4u: %s: [missing]",
+      code, tty_term_codes[code].name);
@@ -622,2 +674,3 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (string) %s", code,
-                tty_term_codes[code].name, out);
+  xsnprintf(s, sizeof s, "%4u: %s: (string) %s",
+      code, tty_term_codes[code].name,
+      out);
@@ -626,2 +679,3 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (number) %d", code,
-                tty_term_codes[code].name, term->codes[code].value.number);
+  xsnprintf(s, sizeof s, "%4u: %s: (number) %d",
+      code, tty_term_codes[code].name,
+      term->codes[code].value.number);
@@ -630,2 +684,2 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (flag) %s", code,
-                tty_term_codes[code].name,
+  xsnprintf(s, sizeof s, "%4u: %s: (flag) %s",
+      code, tty_term_codes[code].name,
