--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c7939031-458346ed-d06e71f0/services@audiopolicy@tests@AudioPolicyManagerTestClient.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c7939031-458346ed-d06e71f0/services@audiopolicy@tests@AudioPolicyManagerTestClient.no_comments_merged.h
@@ -99 +109 @@ class AudioPolicyManagerTestClient : public AudioPolicyTestClient {
-    }
+    };
