--- a/./output/frameworks_av/c7939031-458346ed-d06e71f0/services@audiopolicy@tests@AudioPolicyManagerTestClient.no_comments_conflict.h
+++ b/./output/frameworks_av/c7939031-458346ed-d06e71f0/services@audiopolicy@tests@AudioPolicyManagerTestClient.no_comments_merged.h
@@ -223 +222,0 @@ private:
-<<<<<<< HEAD
@@ -226,4 +224,0 @@ private:
-||||||| d06e71f07f
-=======
-    std::set<audio_io_handle_t> mOpenedInputs;
->>>>>>> 458346ed
