diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/92b9379d-7f55d0c6-c84b6df7/src@core@controller.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/92b9379d-7f55d0c6-c84b6df7/src@core@controller.no_comments_merged.cpp
index 8c99b02..3bb300e 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/92b9379d-7f55d0c6-c84b6df7/src@core@controller.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/92b9379d-7f55d0c6-c84b6df7/src@core@controller.no_comments_merged.cpp
@@ -2 +1,0 @@
-#include "src/config/configwindow.h"
@@ -3,0 +3 @@
+#include "src/config/configwindow.h"
@@ -45,2 +44,0 @@ Controller::Controller()
-{
-    m_appLatestVersion
@@ -59,4 +57,2 @@ Controller::Controller()
-        connect(
-          nativeFilter, &GlobalShortcutFilter::printPressed, this, [this]() {
-              this->requestCapture(
-                CaptureRequest(CaptureRequest::GRAPHICAL_MODE));
+    connect(nativeFilter, &GlobalShortcutFilter::printPressed, this, [this]() {
+        this->requestCapture(CaptureRequest(CaptureRequest::GRAPHICAL_MODE));
@@ -91,4 +86,0 @@ Controller::Controller()
-    {
-        delete m_history;
-        delete m_trayIconMenu;
-    }
@@ -102,8 +94,4 @@ Controller::Controller()
-        connect(this,
-                &Controller::captureTaken,
-                this,
-                &Controller::handleCaptureTaken);
-        connect(this,
-                &Controller::captureFailed,
-                this,
-                &Controller::handleCaptureFailed);
+    connect(
+      this, &Controller::captureTaken, this, &Controller::handleCaptureTaken);
+    connect(
+      this, &Controller::captureFailed, this, &Controller::handleCaptureFailed);
@@ -303,5 +290,0 @@ Controller::Controller()
-        connect(configAction,
-                &QAction::triggered,
-                this,
-                &Controller::openConfigWindow);
-        QAction* infoAction = new QAction(tr("&About"), this);
@@ -309 +292,3 @@ Controller::Controller()
-          infoAction, &QAction::triggered, this, &Controller::openInfoWindow);
+      configAction, &QAction::triggered, this, &Controller::openConfigWindow);
+    QAction* infoAction = new QAction(tr("&About"), this);
+    connect(infoAction, &QAction::triggered, this, &Controller::openInfoWindow);
@@ -311,2 +296 @@ Controller::Controller()
-        connect(
-          m_appUpdates, &QAction::triggered, this, &Controller::appUpdates);
+    connect(m_appUpdates, &QAction::triggered, this, &Controller::appUpdates);
@@ -316,4 +300,2 @@ Controller::Controller()
-        connect(recentAction,
-                SIGNAL(triggered()),
-                this,
-                SLOT(showRecentScreenshots()));
+    connect(
+      recentAction, SIGNAL(triggered()), this, SLOT(showRecentScreenshots()));
@@ -348,2 +330 @@ Controller::Controller()
-            auto trayIconActivated =
-              [this](QSystemTrayIcon::ActivationReason r) {
+        auto trayIconActivated = [this](QSystemTrayIcon::ActivationReason r) {
@@ -365,2 +346 @@ Controller::Controller()
-        connect(
-          m_trayIcon, &QSystemTrayIcon::activated, this, trayIconActivated);
+    connect(m_trayIcon, &QSystemTrayIcon::activated, this, trayIconActivated);
@@ -369,4 +349,2 @@ Controller::Controller()
-        connect(qApp,
-                &QCoreApplication::aboutToQuit,
-                m_trayIcon,
-                &QSystemTrayIcon::hide);
+    connect(
+      qApp, &QCoreApplication::aboutToQuit, m_trayIcon, &QSystemTrayIcon::hide);
@@ -385,2 +363,3 @@ Controller::Controller()
-    void Controller::sendTrayNotification(
-      const QString& text, const QString& title, const int timeout)
+void Controller::sendTrayNotification(const QString& text,
+                                      const QString& title,
+                                      const int timeout)
