diff --git a/./output/art/7919e470-f5107e92-438e0372/runtime@oat_file_assistant.no_comments_mergegen.cc b/./output/art/7919e470-f5107e92-438e0372/runtime@oat_file_assistant.no_comments_merged.cc
index e31c5c5..630eeeb 100644
--- a/./output/art/7919e470-f5107e92-438e0372/runtime@oat_file_assistant.no_comments_mergegen.cc
+++ b/./output/art/7919e470-f5107e92-438e0372/runtime@oat_file_assistant.no_comments_merged.cc
@@ -8 +8,2 @@
-#include "android-base/logging.h" #include "android-base/properties.h"
+#include "android-base/logging.h"
+#include "android-base/properties.h"
@@ -247,0 +249,7 @@ if (info.CheckDisableCompactDexExperiment()) {
+    return kDex2OatFromScratch;
+  }
+  DexOptNeeded dexopt_needed = info.GetDexOptNeeded(
+      target_compiler_filter, GetDexOptTrigger(target_compiler_filter, profile_changed, downgrade));
+  if (dexopt_needed != kNoDexOptNeeded && (&info == &dm_for_oat_ || &info == &dm_for_odex_)) {
+    return kDex2OatFromScratch;
+  }
