--- a/./output/duckdb/683c18a9-564eb25e-8ce36237/src@storage@serialization@serialize_nodes.no_comments_conflict.cpp
+++ b/./output/duckdb/683c18a9-564eb25e-8ce36237/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
@@ -185,4 +184,0 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
-<<<<<<< HEAD
- serializer.WriteProperty<CSVOption<bool>>(139, "dialect_options.state_machine_options.rfc_4180", dialect_options.state_machine_options.rfc_4180);
-||||||| 8ce3623758
-=======
@@ -190 +186 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
->>>>>>> 564eb25e
+ serializer.WriteProperty<CSVOption<bool>>(140, "dialect_options.state_machine_options.rfc_4180", dialect_options.state_machine_options.rfc_4180);
@@ -233,4 +228,0 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
-<<<<<<< HEAD
- deserializer.ReadProperty<CSVOption<bool>>(139, "dialect_options.state_machine_options.rfc_4180", result.dialect_options.state_machine_options.rfc_4180);
-||||||| 8ce3623758
-=======
@@ -238 +230 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
->>>>>>> 564eb25e
+ deserializer.ReadProperty<CSVOption<bool>>(140, "dialect_options.state_machine_options.rfc_4180", result.dialect_options.state_machine_options.rfc_4180);
