--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/56d7d9e8-ffb64347-0649026e/include@grpcpp@impl@codegen@async_unary_call_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/56d7d9e8-ffb64347-0649026e/include@grpcpp@impl@codegen@async_unary_call_impl.no_comments_merged.h
@@ -6,2 +5,0 @@
-#include <grpcpp/impl/codegen/server_context_impl.h>
-#include <grpcpp/impl/codegen/client_context_impl.h>
