--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_merged.cpp
@@ -148 +148,2 @@ static auto& getIAudioPolicyServiceStatistics() {
-#define BINDER_METHOD_ENTRY(ENTRY) {(Code)media::BnAudioPolicyService::TRANSACTION_ ##ENTRY, #ENTRY},
+#define BINDER_METHOD_ENTRY(ENTRY) \
+        {(Code)media::BnAudioPolicyService::TRANSACTION_##ENTRY, #ENTRY},
@@ -891 +971,3 @@ void AudioPolicyService::setAppState_l(sp<AudioRecordClient> client, app_state_t
-NO_THREAD_SAFETY_ANALYSIS() {
+status_t AudioPolicyService::dump(int fd, const Vector<String16>& args __unused)
+NO_THREAD_SAFETY_ANALYSIS
+{
@@ -1749 +1846,3 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-NO_THREAD_SAFETY_ANALYSIS() {
+status_t AudioPolicyService::AudioCommandThread::dump(int fd)
+NO_THREAD_SAFETY_ANALYSIS
+{
@@ -1988,9 +2113,2 @@ status_t AudioPolicyService::AudioCommandThread::sendCommand(sp<AudioCommand>& c
-<<<<<<< HEAD
-        if (command->mCond.wait_for(ul, std::chrono::nanoseconds(timeOutNs), getTid()) ==
-            std::cv_status::timeout) {
-||||||| 47af5d30a6
-        if (command->mCond.waitRelative(command->mLock, timeOutNs) != NO_ERROR) {
-=======
-        if (command->mCond.wait_for(ul, std::chrono::nanoseconds(timeOutNs)) ==
-            std::cv_status::timeout) {
->>>>>>> 43915f776988367ca9317b75225cd0291c0fd430
+        if (command->mCond.wait_for(
+                ul, std::chrono::nanoseconds(timeOutNs), getTid()) == std::cv_status::timeout) {
@@ -2204,2 +2352,6 @@ int aps_move_effects(void* service __unused, audio_session_t session, audio_io_h
-char* aps_get_parameters(void* service __unused, audio_io_handle_t io_handle, const char* keys);
-void aps_set_parameters(void* service, audio_io_handle_t io_handle, const char* kv_pairs,
+char * aps_get_parameters(void *service __unused, audio_io_handle_t io_handle,
+                                     const char *keys);
+void aps_set_parameters(void *service, audio_io_handle_t io_handle,
+                                   const char *kv_pairs, int delay_ms);
+int aps_set_stream_volume(void *service, audio_stream_type_t stream,
+                                     float volume, audio_io_handle_t output,
@@ -2207,2 +2358,0 @@ void aps_set_parameters(void* service, audio_io_handle_t io_handle, const char*
-int aps_set_stream_volume(void* service, audio_stream_type_t stream, float volume,
-                          audio_io_handle_t output, int delay_ms);
@@ -2210 +2360 @@ int aps_set_voice_volume(void* service, float volume, int delay_ms);
-}
+};
