diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_merged.cpp
index a703075..ad5967f 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_merged.cpp
@@ -148 +148,2 @@ static auto& getIAudioPolicyServiceStatistics() {
-#define BINDER_METHOD_ENTRY(ENTRY) {(Code)media::BnAudioPolicyService::TRANSACTION_ ##ENTRY, #ENTRY},
+#define BINDER_METHOD_ENTRY(ENTRY) \
+        {(Code)media::BnAudioPolicyService::TRANSACTION_##ENTRY, #ENTRY},
@@ -150 +151,3 @@ static auto& getIAudioPolicyServiceStatistics() {
-            IAUDIOPOLICYSERVICE_BINDER_METHOD_MACRO_LIST METHOD_STATISTICS_BINDER_CODE_NAMES(Code)};
+        IAUDIOPOLICYSERVICE_BINDER_METHOD_MACRO_LIST
+        METHOD_STATISTICS_BINDER_CODE_NAMES(Code)
+    };
@@ -155 +158,2 @@ static auto& getIAudioPolicyServiceStatistics() {
-static AudioPolicyInterface* createAudioPolicyManager(AudioPolicyClientInterface* clientInterface) {
+static AudioPolicyInterface* createAudioPolicyManager(AudioPolicyClientInterface *clientInterface)
+{
@@ -163,4 +167,3 @@ static AudioPolicyInterface* createAudioPolicyManager(AudioPolicyClientInterface
-        apm = new AudioPolicyManager(
-                config,
-                loadApmEngineLibraryAndCreateEngine(config->getEngineLibraryNameSuffix(),
-                                                    apmConfig.engineConfig),
+        apm = new AudioPolicyManager(config,
+                loadApmEngineLibraryAndCreateEngine(
+                        config->getEngineLibraryNameSuffix(), apmConfig.engineConfig),
@@ -170,2 +173,2 @@ static AudioPolicyInterface* createAudioPolicyManager(AudioPolicyClientInterface
-        apm = new AudioPolicyManager(
-                config, loadApmEngineLibraryAndCreateEngine(config->getEngineLibraryNameSuffix()),
+        apm = new AudioPolicyManager(config,
+                loadApmEngineLibraryAndCreateEngine(config->getEngineLibraryNameSuffix()),
@@ -181 +184,2 @@ static AudioPolicyInterface* createAudioPolicyManager(AudioPolicyClientInterface
-static void destroyAudioPolicyManager(AudioPolicyInterface* interface) {
+static void destroyAudioPolicyManager(AudioPolicyInterface *interface)
+{
@@ -195 +199,2 @@ AudioPolicyService::AudioPolicyService()
-void AudioPolicyService::loadAudioPolicyManager() {
+void AudioPolicyService::loadAudioPolicyManager()
+{
@@ -199,2 +204,2 @@ void AudioPolicyService::loadAudioPolicyManager() {
-        mCreateAudioPolicyManager = reinterpret_cast<CreateAudioPolicyManagerInstance>(
-                dlsym(mLibraryHandle, "createAudioPolicyManager"));
+        mCreateAudioPolicyManager = reinterpret_cast<CreateAudioPolicyManagerInstance>
+                                            (dlsym(mLibraryHandle, "createAudioPolicyManager"));
@@ -215 +220,2 @@ void AudioPolicyService::loadAudioPolicyManager() {
-void AudioPolicyService::onFirstRef() {
+void AudioPolicyService::onFirstRef()
+{
@@ -220,2 +226 @@ void AudioPolicyService::onFirstRef() {
-                .record();
-    });
+            .record(); });
@@ -267 +272,2 @@ void AudioPolicyService::onAudioSystemReady() {
-void AudioPolicyService::unloadAudioPolicyManager() {
+void AudioPolicyService::unloadAudioPolicyManager()
+{
@@ -276 +282,2 @@ void AudioPolicyService::unloadAudioPolicyManager() {
-AudioPolicyService::~AudioPolicyService() {
+AudioPolicyService::~AudioPolicyService()
+{
@@ -289 +296,2 @@ AudioPolicyService::~AudioPolicyService() {
-Status AudioPolicyService::registerClient(const sp<media::IAudioPolicyServiceClient>& client) {
+Status AudioPolicyService::registerClient(const sp<media::IAudioPolicyServiceClient>& client)
+{
@@ -299 +307,4 @@ Status AudioPolicyService::registerClient(const sp<media::IAudioPolicyServiceCli
-        sp<NotificationClient> notificationClient = new NotificationClient(this, client, uid, pid);
+        sp<NotificationClient> notificationClient = new NotificationClient(this,
+                                                                           client,
+                                                                           uid,
+                                                                           pid);
@@ -307 +318,2 @@ Status AudioPolicyService::registerClient(const sp<media::IAudioPolicyServiceCli
-Status AudioPolicyService::setAudioPortCallbacksEnabled(bool enabled) {
+Status AudioPolicyService::setAudioPortCallbacksEnabled(bool enabled)
+{
@@ -318 +330,2 @@ Status AudioPolicyService::setAudioPortCallbacksEnabled(bool enabled) {
-Status AudioPolicyService::setAudioVolumeGroupCallbacksEnabled(bool enabled) {
+Status AudioPolicyService::setAudioVolumeGroupCallbacksEnabled(bool enabled)
+{
@@ -329 +342,2 @@ Status AudioPolicyService::setAudioVolumeGroupCallbacksEnabled(bool enabled) {
-void AudioPolicyService::removeNotificationClient(uid_t uid, pid_t pid) {
+void AudioPolicyService::removeNotificationClient(uid_t uid, pid_t pid)
+{
@@ -349 +363,2 @@ void AudioPolicyService::removeNotificationClient(uid_t uid, pid_t pid) {
-void AudioPolicyService::onAudioPortListUpdate() {
+void AudioPolicyService::onAudioPortListUpdate()
+{
@@ -352 +367,2 @@ void AudioPolicyService::onAudioPortListUpdate() {
-void AudioPolicyService::doOnAudioPortListUpdate() {
+void AudioPolicyService::doOnAudioPortListUpdate()
+{
@@ -358 +374,2 @@ void AudioPolicyService::doOnAudioPortListUpdate() {
-void AudioPolicyService::onAudioPatchListUpdate() {
+void AudioPolicyService::onAudioPatchListUpdate()
+{
@@ -361 +378,2 @@ void AudioPolicyService::onAudioPatchListUpdate() {
-void AudioPolicyService::doOnAudioPatchListUpdate() {
+void AudioPolicyService::doOnAudioPatchListUpdate()
+{
@@ -367 +385,2 @@ void AudioPolicyService::doOnAudioPatchListUpdate() {
-void AudioPolicyService::onAudioVolumeGroupChanged(volume_group_t group, int flags) {
+void AudioPolicyService::onAudioVolumeGroupChanged(volume_group_t group, int flags)
+{
@@ -370 +389,2 @@ void AudioPolicyService::onAudioVolumeGroupChanged(volume_group_t group, int fla
-void AudioPolicyService::doOnAudioVolumeGroupChanged(volume_group_t group, int flags) {
+void AudioPolicyService::doOnAudioVolumeGroupChanged(volume_group_t group, int flags)
+{
@@ -376,2 +396,4 @@ void AudioPolicyService::doOnAudioVolumeGroupChanged(volume_group_t group, int f
-void AudioPolicyService::onDynamicPolicyMixStateUpdate(const String8& regId, int32_t state) {
-    ALOGV("AudioPolicyService::onDynamicPolicyMixStateUpdate(%s, %d)", regId.c_str(), state);
+void AudioPolicyService::onDynamicPolicyMixStateUpdate(const String8& regId, int32_t state)
+{
+    ALOGV("AudioPolicyService::onDynamicPolicyMixStateUpdate(%s, %d)",
+            regId.c_str(), state);
@@ -380 +402,2 @@ void AudioPolicyService::onDynamicPolicyMixStateUpdate(const String8& regId, int
-void AudioPolicyService::doOnDynamicPolicyMixStateUpdate(const String8& regId, int32_t state) {
+void AudioPolicyService::doOnDynamicPolicyMixStateUpdate(const String8& regId, int32_t state)
+{
@@ -387,7 +410,11 @@ void AudioPolicyService::onRecordingConfigurationUpdate(
-        int event, const record_client_info_t* clientInfo, const audio_config_base_t* clientConfig,
-        std::vector<effect_descriptor_t> clientEffects, const audio_config_base_t* deviceConfig,
-        std::vector<effect_descriptor_t> effects, audio_patch_handle_t patchHandle,
-        audio_source_t source) {
-    mOutputCommandThread->recordingConfigurationUpdateCommand(event, clientInfo, clientConfig,
-                                                              clientEffects, deviceConfig, effects,
-                                                              patchHandle, source);
+                                                    int event,
+                                                    const record_client_info_t *clientInfo,
+                                                    const audio_config_base_t *clientConfig,
+                                                    std::vector<effect_descriptor_t> clientEffects,
+                                                    const audio_config_base_t *deviceConfig,
+                                                    std::vector<effect_descriptor_t> effects,
+                                                    audio_patch_handle_t patchHandle,
+                                                    audio_source_t source)
+{
+    mOutputCommandThread->recordingConfigurationUpdateCommand(event, clientInfo,
+            clientConfig, clientEffects, deviceConfig, effects, patchHandle, source);
@@ -396,4 +423,9 @@ void AudioPolicyService::doOnRecordingConfigurationUpdate(
-        int event, const record_client_info_t* clientInfo, const audio_config_base_t* clientConfig,
-        std::vector<effect_descriptor_t> clientEffects, const audio_config_base_t* deviceConfig,
-        std::vector<effect_descriptor_t> effects, audio_patch_handle_t patchHandle,
-        audio_source_t source) {
+                                                  int event,
+                                                  const record_client_info_t *clientInfo,
+                                                  const audio_config_base_t *clientConfig,
+                                                  std::vector<effect_descriptor_t> clientEffects,
+                                                  const audio_config_base_t *deviceConfig,
+                                                  std::vector<effect_descriptor_t> effects,
+                                                  audio_patch_handle_t patchHandle,
+                                                  audio_source_t source)
+{
@@ -402,3 +434,2 @@ void AudioPolicyService::doOnRecordingConfigurationUpdate(
-        mNotificationClients.valueAt(i)->onRecordingConfigurationUpdate(
-                event, clientInfo, clientConfig, clientEffects, deviceConfig, effects, patchHandle,
-                source);
+        mNotificationClients.valueAt(i)->onRecordingConfigurationUpdate(event, clientInfo,
+                clientConfig, clientEffects, deviceConfig, effects, patchHandle, source);
@@ -407 +438,2 @@ void AudioPolicyService::doOnRecordingConfigurationUpdate(
-void AudioPolicyService::onRoutingUpdated() {
+void AudioPolicyService::onRoutingUpdated()
+{
@@ -410 +442,2 @@ void AudioPolicyService::onRoutingUpdated() {
-void AudioPolicyService::doOnRoutingUpdated() {
+void AudioPolicyService::doOnRoutingUpdated()
+{
@@ -416 +449,2 @@ void AudioPolicyService::doOnRoutingUpdated() {
-void AudioPolicyService::onVolumeRangeInitRequest() {
+void AudioPolicyService::onVolumeRangeInitRequest()
+{
@@ -419 +453,2 @@ void AudioPolicyService::onVolumeRangeInitRequest() {
-void AudioPolicyService::doOnVolumeRangeInitRequest() {
+void AudioPolicyService::doOnVolumeRangeInitRequest()
+{
@@ -425 +460,2 @@ void AudioPolicyService::doOnVolumeRangeInitRequest() {
-void AudioPolicyService::onCheckSpatializer() {
+void AudioPolicyService::onCheckSpatializer()
+{
@@ -429 +465,2 @@ void AudioPolicyService::onCheckSpatializer() {
-void AudioPolicyService::onCheckSpatializer_l() {
+void AudioPolicyService::onCheckSpatializer_l()
+{
@@ -434,3 +471,4 @@ void AudioPolicyService::onCheckSpatializer_l() {
-void AudioPolicyService::doOnCheckSpatializer() {
-    ALOGV("%s mSpatializer %p level %d", __func__, mSpatializer.get(),
-          (int)mSpatializer->getLevel());
+void AudioPolicyService::doOnCheckSpatializer()
+{
+    ALOGV("%s mSpatializer %p level %d",
+        __func__, mSpatializer.get(), (int)mSpatializer->getLevel());
@@ -444,3 +482,4 @@ void AudioPolicyService::doOnCheckSpatializer() {
-            status_t status = mAudioPolicyManager->getSpatializerOutput(&config, &attr, &newOutput);
-            ALOGV("%s currentOutput %d newOutput %d channel_mask %#x", __func__, currentOutput,
-                  newOutput, config.channel_mask);
+            status_t status =
+                    mAudioPolicyManager->getSpatializerOutput(&config, &attr, &newOutput);
+            ALOGV("%s currentOutput %d newOutput %d channel_mask %#x",
+                    __func__, currentOutput, newOutput, config.channel_mask);
@@ -470,2 +509,2 @@ void AudioPolicyService::doOnCheckSpatializer() {
-size_t AudioPolicyService::countActiveClientsOnOutput_l(audio_io_handle_t output,
-                                                        bool spatializedOnly) {
+size_t AudioPolicyService::countActiveClientsOnOutput_l(
+        audio_io_handle_t output, bool spatializedOnly) {
@@ -475 +514,2 @@ size_t AudioPolicyService::countActiveClientsOnOutput_l(audio_io_handle_t output
-        if (client->io == output && client->active && (!spatializedOnly || client->isSpatialized)) {
+        if (client->io == output && client->active
+                && (!spatializedOnly || client->isSpatialized)) {
@@ -487 +527,2 @@ void AudioPolicyService::onUpdateActiveSpatializerTracks_l() {
-void AudioPolicyService::doOnUpdateActiveSpatializerTracks() {
+void AudioPolicyService::doOnUpdateActiveSpatializerTracks()
+{
@@ -500 +541,3 @@ status_t AudioPolicyService::clientCreateAudioPatch(const struct audio_patch* pa
-                                                    audio_patch_handle_t* handle, int delayMs) {
+                                                audio_patch_handle_t *handle,
+                                                int delayMs)
+{
@@ -503 +546,3 @@ status_t AudioPolicyService::clientCreateAudioPatch(const struct audio_patch* pa
-status_t AudioPolicyService::clientReleaseAudioPatch(audio_patch_handle_t handle, int delayMs) {
+status_t AudioPolicyService::clientReleaseAudioPatch(audio_patch_handle_t handle,
+                                                 int delayMs)
+{
@@ -507 +552,2 @@ status_t AudioPolicyService::clientSetAudioPortConfig(const struct audio_port_co
-                                                      int delayMs) {
+                                                      int delayMs)
+{
@@ -511,10 +557,13 @@ AudioPolicyService::NotificationClient::NotificationClient(
-        const sp<AudioPolicyService>& service, const sp<media::IAudioPolicyServiceClient>& client,
-        uid_t uid, pid_t pid)
-    : mService(service),
-      mUid(uid),
-      mPid(pid),
-      mAudioPolicyServiceClient(client),
-      mAudioPortCallbacksEnabled(false),
-      mAudioVolumeGroupCallbacksEnabled(false) {}
-AudioPolicyService::NotificationClient::~NotificationClient() {}
-void AudioPolicyService::NotificationClient::binderDied(const wp<IBinder>& who __unused) {
+        const sp<AudioPolicyService>& service,
+        const sp<media::IAudioPolicyServiceClient>& client,
+        uid_t uid,
+        pid_t pid)
+    : mService(service), mUid(uid), mPid(pid), mAudioPolicyServiceClient(client),
+      mAudioPortCallbacksEnabled(false), mAudioVolumeGroupCallbacksEnabled(false)
+{
+}
+AudioPolicyService::NotificationClient::~NotificationClient()
+{
+}
+void AudioPolicyService::NotificationClient::binderDied(const wp<IBinder>& who __unused)
+{
@@ -527 +576,2 @@ void AudioPolicyService::NotificationClient::binderDied(const wp<IBinder>& who _
-void AudioPolicyService::NotificationClient::onAudioPortListUpdate() {
+void AudioPolicyService::NotificationClient::onAudioPortListUpdate()
+{
@@ -532 +582,2 @@ void AudioPolicyService::NotificationClient::onAudioPortListUpdate() {
-void AudioPolicyService::NotificationClient::onAudioPatchListUpdate() {
+void AudioPolicyService::NotificationClient::onAudioPatchListUpdate()
+{
@@ -538 +589,2 @@ void AudioPolicyService::NotificationClient::onAudioVolumeGroupChanged(volume_gr
-                                                                       int flags) {
+                                                                      int flags)
+{
@@ -543,2 +595,3 @@ void AudioPolicyService::NotificationClient::onAudioVolumeGroupChanged(volume_gr
-void AudioPolicyService::NotificationClient::onDynamicPolicyMixStateUpdate(const String8& regId,
-                                                                           int32_t state) {
+void AudioPolicyService::NotificationClient::onDynamicPolicyMixStateUpdate(
+        const String8& regId, int32_t state)
+{
@@ -551,4 +604,9 @@ void AudioPolicyService::NotificationClient::onRecordingConfigurationUpdate(
-        int event, const record_client_info_t* clientInfo, const audio_config_base_t* clientConfig,
-        std::vector<effect_descriptor_t> clientEffects, const audio_config_base_t* deviceConfig,
-        std::vector<effect_descriptor_t> effects, audio_patch_handle_t patchHandle,
-        audio_source_t source) {
+                                            int event,
+                                            const record_client_info_t *clientInfo,
+                                            const audio_config_base_t *clientConfig,
+                                            std::vector<effect_descriptor_t> clientEffects,
+                                            const audio_config_base_t *deviceConfig,
+                                            std::vector<effect_descriptor_t> effects,
+                                            audio_patch_handle_t patchHandle,
+                                            audio_source_t source)
+{
@@ -560,2 +618,2 @@ void AudioPolicyService::NotificationClient::onRecordingConfigurationUpdate(
-            AudioConfigBase clientConfigAidl =
-                    VALUE_OR_RETURN_STATUS(legacy2aidl_audio_config_base_t_AudioConfigBase(
+            AudioConfigBase clientConfigAidl = VALUE_OR_RETURN_STATUS(
+                    legacy2aidl_audio_config_base_t_AudioConfigBase(
@@ -563,5 +621,6 @@ void AudioPolicyService::NotificationClient::onRecordingConfigurationUpdate(
-            std::vector<media::EffectDescriptor> clientEffectsAidl =
-                    VALUE_OR_RETURN_STATUS(convertContainer<std::vector<media::EffectDescriptor>>(
-                            clientEffects, legacy2aidl_effect_descriptor_t_EffectDescriptor));
-            AudioConfigBase deviceConfigAidl =
-                    VALUE_OR_RETURN_STATUS(legacy2aidl_audio_config_base_t_AudioConfigBase(
+            std::vector<media::EffectDescriptor> clientEffectsAidl = VALUE_OR_RETURN_STATUS(
+                    convertContainer<std::vector<media::EffectDescriptor>>(
+                            clientEffects,
+                            legacy2aidl_effect_descriptor_t_EffectDescriptor));
+            AudioConfigBase deviceConfigAidl = VALUE_OR_RETURN_STATUS(
+                    legacy2aidl_audio_config_base_t_AudioConfigBase(
@@ -569,7 +628,8 @@ void AudioPolicyService::NotificationClient::onRecordingConfigurationUpdate(
-            std::vector<media::EffectDescriptor> effectsAidl =
-                    VALUE_OR_RETURN_STATUS(convertContainer<std::vector<media::EffectDescriptor>>(
-                            effects, legacy2aidl_effect_descriptor_t_EffectDescriptor));
-            int32_t patchHandleAidl =
-                    VALUE_OR_RETURN_STATUS(legacy2aidl_audio_patch_handle_t_int32_t(patchHandle));
-            media::audio::common::AudioSource sourceAidl =
-                    VALUE_OR_RETURN_STATUS(legacy2aidl_audio_source_t_AudioSource(source));
+            std::vector<media::EffectDescriptor> effectsAidl = VALUE_OR_RETURN_STATUS(
+                    convertContainer<std::vector<media::EffectDescriptor>>(
+                            effects,
+                            legacy2aidl_effect_descriptor_t_EffectDescriptor));
+            int32_t patchHandleAidl = VALUE_OR_RETURN_STATUS(
+                    legacy2aidl_audio_patch_handle_t_int32_t(patchHandle));
+            media::audio::common::AudioSource sourceAidl = VALUE_OR_RETURN_STATUS(
+                    legacy2aidl_audio_source_t_AudioSource(source));
@@ -577,3 +637,8 @@ void AudioPolicyService::NotificationClient::onRecordingConfigurationUpdate(
-                    mAudioPolicyServiceClient->onRecordingConfigurationUpdate(
-                            eventAidl, clientInfoAidl, clientConfigAidl, clientEffectsAidl,
-                            deviceConfigAidl, effectsAidl, patchHandleAidl, sourceAidl));
+                    mAudioPolicyServiceClient->onRecordingConfigurationUpdate(eventAidl,
+                                                                              clientInfoAidl,
+                                                                              clientConfigAidl,
+                                                                              clientEffectsAidl,
+                                                                              deviceConfigAidl,
+                                                                              effectsAidl,
+                                                                              patchHandleAidl,
+                                                                              sourceAidl));
@@ -584 +649,2 @@ void AudioPolicyService::NotificationClient::onRecordingConfigurationUpdate(
-void AudioPolicyService::NotificationClient::setAudioPortCallbacksEnabled(bool enabled) {
+void AudioPolicyService::NotificationClient::setAudioPortCallbacksEnabled(bool enabled)
+{
@@ -587 +653,2 @@ void AudioPolicyService::NotificationClient::setAudioPortCallbacksEnabled(bool e
-void AudioPolicyService::NotificationClient::setAudioVolumeGroupCallbacksEnabled(bool enabled) {
+void AudioPolicyService::NotificationClient::setAudioVolumeGroupCallbacksEnabled(bool enabled)
+{
@@ -590 +657,2 @@ void AudioPolicyService::NotificationClient::setAudioVolumeGroupCallbacksEnabled
-void AudioPolicyService::NotificationClient::onRoutingUpdated() {
+void AudioPolicyService::NotificationClient::onRoutingUpdated()
+{
@@ -595 +663,2 @@ void AudioPolicyService::NotificationClient::onRoutingUpdated() {
-void AudioPolicyService::NotificationClient::onVolumeRangeInitRequest() {
+void AudioPolicyService::NotificationClient::onVolumeRangeInitRequest()
+{
@@ -605 +674,2 @@ static void dumpReleaseLock(audio_utils::mutex& mutex, bool locked)
-        RELEASE(mutex) NO_THREAD_SAFETY_ANALYSIS {
+        RELEASE(mutex) NO_THREAD_SAFETY_ANALYSIS
+{
@@ -608 +678,2 @@ static void dumpReleaseLock(audio_utils::mutex& mutex, bool locked)
-status_t AudioPolicyService::dumpInternals(int fd) {
+status_t AudioPolicyService::dumpInternals(int fd)
+{
@@ -633 +704,2 @@ status_t AudioPolicyService::dumpInternals(int fd) {
-void AudioPolicyService::updateUidStates() {
+void AudioPolicyService::updateUidStates()
+{
@@ -637 +709,2 @@ void AudioPolicyService::updateUidStates() {
-void AudioPolicyService::updateUidStates_l() {
+void AudioPolicyService::updateUidStates_l()
+{
@@ -654 +727,2 @@ void AudioPolicyService::updateUidStates_l() {
-    bool rttCallActive = (isInCall || isInCommunication) && mUidPolicy->isRttEnabled();
+    bool rttCallActive = (isInCall || isInCommunication)
+            && mUidPolicy->isRttEnabled();
@@ -663,2 +737,2 @@ void AudioPolicyService::updateUidStates_l() {
-        uid_t currentUid =
-                VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(current->attributionSource.uid));
+        uid_t currentUid = VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(
+                current->attributionSource.uid));
@@ -676 +750,2 @@ void AudioPolicyService::updateUidStates_l() {
-            bool isPrivacySensitive = (current->attributes.flags & AUDIO_FLAG_CAPTURE_PRIVATE) != 0;
+            bool isPrivacySensitive =
+                    (current->attributes.flags & AUDIO_FLAG_CAPTURE_PRIVATE) != 0;
@@ -701,2 +776,2 @@ void AudioPolicyService::updateUidStates_l() {
-            if (!(current->attributes.source == AUDIO_SOURCE_HOTWORD ||
-                  ((isA11yOnTop || rttCallActive) && isAssistant))) {
+            if (!(current->attributes.source == AUDIO_SOURCE_HOTWORD
+                    || ((isA11yOnTop || rttCallActive) && isAssistant))) {
@@ -704,6 +779,6 @@ void AudioPolicyService::updateUidStates_l() {
-                    if (current->startTimeNs > latestSensitiveStartNs ||
-                        (isInCommunication && currentUid == mPhoneStateOwnerUid)) {
-                        if (!isInCommunication || latestSensitiveActiveOrComm == nullptr ||
-                            VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(
-                                    latestSensitiveActiveOrComm->attributionSource.uid)) !=
-                                    mPhoneStateOwnerUid) {
+                    if (current->startTimeNs > latestSensitiveStartNs
+                            || (isInCommunication && currentUid == mPhoneStateOwnerUid)) {
+                        if (!isInCommunication || latestSensitiveActiveOrComm == nullptr
+                                || VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(
+                                    latestSensitiveActiveOrComm->attributionSource.uid))
+                                        != mPhoneStateOwnerUid) {
@@ -727 +802,2 @@ void AudioPolicyService::updateUidStates_l() {
-        if (currentUid == mPhoneStateOwnerUid && !isVirtualSource(current->attributes.source)) {
+        if (currentUid == mPhoneStateOwnerUid &&
+                !isVirtualSource(current->attributes.source)) {
@@ -746 +822,2 @@ void AudioPolicyService::updateUidStates_l() {
-    if (topActive != nullptr && topSensitiveActive != nullptr && !topActive->canCaptureOutput) {
+    if (topActive != nullptr && topSensitiveActive != nullptr
+            && !topActive->canCaptureOutput) {
@@ -751,2 +828,2 @@ void AudioPolicyService::updateUidStates_l() {
-        uid_t currentUid =
-                VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(current->attributionSource.uid));
+        uid_t currentUid = VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(
+            current->attributionSource.uid));
@@ -757,12 +834,6 @@ void AudioPolicyService::updateUidStates_l() {
-        bool isTopOrLatestActive = topActive == nullptr ? false
-                                                        : current->attributionSource.uid ==
-                                                                  topActive->attributionSource.uid;
-        bool isTopOrLatestSensitive = topSensitiveActive == nullptr
-                                              ? false
-                                              : current->attributionSource.uid ==
-                                                        topSensitiveActive->attributionSource.uid;
-        bool isTopOrLatestAssistant =
-                latestActiveAssistant == nullptr
-                        ? false
-                        : current->attributionSource.uid ==
-                                  latestActiveAssistant->attributionSource.uid;
+        bool isTopOrLatestActive = topActive == nullptr ? false :
+            current->attributionSource.uid == topActive->attributionSource.uid;
+        bool isTopOrLatestSensitive = topSensitiveActive == nullptr ? false :
+            current->attributionSource.uid == topSensitiveActive->attributionSource.uid;
+        bool isTopOrLatestAssistant = latestActiveAssistant == nullptr ? false :
+            current->attributionSource.uid == latestActiveAssistant->attributionSource.uid;
@@ -770,2 +841,2 @@ void AudioPolicyService::updateUidStates_l() {
-            uid_t recordUid =
-                    VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(recordClient->attributionSource.uid));
+            uid_t recordUid = VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(
+                recordClient->attributionSource.uid));
@@ -773,5 +844,5 @@ void AudioPolicyService::updateUidStates_l() {
-            bool canCaptureCommunication = recordClient->canCaptureOutput ||
-                                           !isPhoneStateOwnerActive ||
-                                           recordUid == mPhoneStateOwnerUid;
-            return !(isInCall && !canCaptureCall) &&
-                   !(isInCommunication && !canCaptureCommunication);
+            bool canCaptureCommunication = recordClient->canCaptureOutput
+                || !isPhoneStateOwnerActive
+                || recordUid == mPhoneStateOwnerUid;
+            return !(isInCall && !canCaptureCall)
+                && !(isInCommunication && !canCaptureCommunication);
@@ -781,2 +852,4 @@ void AudioPolicyService::updateUidStates_l() {
-        bool allowCapture = !isAssistantOnTop && (isTopOrLatestActive || isTopOrLatestSensitive) &&
-                            allowSensitiveCapture && canCaptureIfInCallOrCommunication(current);
+        bool allowCapture = !isAssistantOnTop
+                && (isTopOrLatestActive || isTopOrLatestSensitive)
+                && allowSensitiveCapture
+                && canCaptureIfInCallOrCommunication(current);
@@ -792,3 +865,5 @@ void AudioPolicyService::updateUidStates_l() {
-            } else if (allowSensitiveCapture && canCaptureIfInCallOrCommunication(current)) {
-                if (isTopOrLatestAssistant &&
-                    (source == AUDIO_SOURCE_VOICE_RECOGNITION || source == AUDIO_SOURCE_HOTWORD)) {
+            } else if (allowSensitiveCapture
+                    && canCaptureIfInCallOrCommunication(current)) {
+                if (isTopOrLatestAssistant
+                    && (source == AUDIO_SOURCE_VOICE_RECOGNITION
+                        || source == AUDIO_SOURCE_HOTWORD)) {
@@ -805,3 +880,4 @@ void AudioPolicyService::updateUidStates_l() {
-            } else if (allowSensitiveCapture && canCaptureIfInCallOrCommunication(current)) {
-                if ((source == AUDIO_SOURCE_VOICE_RECOGNITION) ||
-                    (source == AUDIO_SOURCE_HOTWORD)) {
+            } else if (allowSensitiveCapture
+                        && canCaptureIfInCallOrCommunication(current)) {
+                if ((source == AUDIO_SOURCE_VOICE_RECOGNITION) || (source == AUDIO_SOURCE_HOTWORD))
+                {
@@ -812,2 +888,3 @@ void AudioPolicyService::updateUidStates_l() {
-            if (!isAssistantOnTop && allowSensitiveCapture &&
-                canCaptureIfInCallOrCommunication(current)) {
+            if (!isAssistantOnTop
+                    && allowSensitiveCapture
+                    && canCaptureIfInCallOrCommunication(current)) {
@@ -822 +899,2 @@ void AudioPolicyService::updateUidStates_l() {
-            if (onlyHotwordActive && canCaptureIfInCallOrCommunication(current)) {
+            if (onlyHotwordActive
+                    && canCaptureIfInCallOrCommunication(current)) {
@@ -830,3 +908,3 @@ void AudioPolicyService::updateUidStates_l() {
-        setAppState_l(current, allowCapture
-                                       ? apmStatFromAmState(mUidPolicy->getUidState(currentUid))
-                                       : APP_STATE_IDLE);
+        setAppState_l(current,
+                      allowCapture ? apmStatFromAmState(mUidPolicy->getUidState(currentUid)) :
+                                APP_STATE_IDLE);
@@ -851 +929,2 @@ app_state_t AudioPolicyService::apmStatFromAmState(int amState) {
-bool AudioPolicyService::isVirtualSource(audio_source_t source) {
+bool AudioPolicyService::isVirtualSource(audio_source_t source)
+{
@@ -865 +944,2 @@ bool AudioPolicyService::isVirtualSource(audio_source_t source) {
-void AudioPolicyService::setAppState_l(sp<AudioRecordClient> client, app_state_t state) {
+void AudioPolicyService::setAppState_l(sp<AudioRecordClient> client, app_state_t state)
+{
@@ -891 +971,3 @@ void AudioPolicyService::setAppState_l(sp<AudioRecordClient> client, app_state_t
-NO_THREAD_SAFETY_ANALYSIS() {
+status_t AudioPolicyService::dump(int fd, const Vector<String16>& args __unused)
+NO_THREAD_SAFETY_ANALYSIS
+{
@@ -934 +1016,2 @@ NO_THREAD_SAFETY_ANALYSIS() {
-status_t AudioPolicyService::dumpPermissionDenial(int fd) {
+status_t AudioPolicyService::dumpPermissionDenial(int fd)
+{
@@ -938,2 +1021 @@ status_t AudioPolicyService::dumpPermissionDenial(int fd) {
-    snprintf(buffer, SIZE,
-             "Permission Denial: "
+    snprintf(buffer, SIZE, "Permission Denial: "
@@ -941 +1023,2 @@ status_t AudioPolicyService::dumpPermissionDenial(int fd) {
-             IPCThreadState::self()->getCallingPid(), IPCThreadState::self()->getCallingUid());
+            IPCThreadState::self()->getCallingPid(),
+            IPCThreadState::self()->getCallingUid());
@@ -946,2 +1029,2 @@ status_t AudioPolicyService::dumpPermissionDenial(int fd) {
-status_t AudioPolicyService::onTransact(uint32_t code, const Parcel& data, Parcel* reply,
-                                        uint32_t flags) {
+status_t AudioPolicyService::onTransact(
+        uint32_t code, const Parcel& data, Parcel* reply, uint32_t flags) {
@@ -963,2 +1046,2 @@ status_t AudioPolicyService::onTransact(uint32_t code, const Parcel& data, Parce
-            ALOGW("%s: transaction %d received from PID %d", __func__, code,
-                  IPCThreadState::self()->getCallingPid());
+            ALOGW("%s: transaction %d received from PID %d",
+                  __func__, code, IPCThreadState::self()->getCallingPid());
@@ -1024,2 +1107,2 @@ status_t AudioPolicyService::onTransact(uint32_t code, const Parcel& data, Parce
-                ALOGW("%s: transaction %d received from PID %d unauthorized UID %d", __func__, code,
-                      IPCThreadState::self()->getCallingPid(),
+                ALOGW("%s: transaction %d received from PID %d unauthorized UID %d",
+                      __func__, code, IPCThreadState::self()->getCallingPid(),
@@ -1046,3 +1129,3 @@ status_t AudioPolicyService::onTransact(uint32_t code, const Parcel& data, Parce
-            },
-            mediautils::TimeCheck::kDefaultTimeoutDuration,
-            mediautils::TimeCheck::kDefaultSecondChanceDuration, true );
+    }, mediautils::TimeCheck::kDefaultTimeoutDuration,
+    mediautils::TimeCheck::kDefaultSecondChanceDuration,
+    true );
@@ -1195,3 +1278 @@ status_t AudioPolicyService::printHelp(int out) {
-    return dprintf(
-            out,
-            "Audio policy service commands:\n"
+    return dprintf(out, "Audio policy service commands:\n"
@@ -1213,5 +1294,6 @@ void AudioPolicyService::UidPolicy::registerSelf() {
-    mAm.registerUidObserver(
-            this,
-            ActivityManager::UID_OBSERVER_GONE | ActivityManager::UID_OBSERVER_IDLE |
-                    ActivityManager::UID_OBSERVER_ACTIVE | ActivityManager::UID_OBSERVER_PROCSTATE,
-            ActivityManager::PROCESS_STATE_UNKNOWN, String16("audioserver"));
+    mAm.registerUidObserver(this, ActivityManager::UID_OBSERVER_GONE
+            | ActivityManager::UID_OBSERVER_IDLE
+            | ActivityManager::UID_OBSERVER_ACTIVE
+            | ActivityManager::UID_OBSERVER_PROCSTATE,
+            ActivityManager::PROCESS_STATE_UNKNOWN,
+            String16("audioserver"));
@@ -1266,2 +1348,3 @@ bool AudioPolicyService::UidPolicy::isUidActive(uid_t uid) {
-        mCachedUids.insert(std::pair<uid_t, std::pair<bool, int>>(
-                uid, std::pair<bool, int>(active, ActivityManager::PROCESS_STATE_UNKNOWN)));
+        mCachedUids.insert(std::pair<uid_t,
+                           std::pair<bool, int>>(uid, std::pair<bool, int>(active,
+                                                      ActivityManager::PROCESS_STATE_UNKNOWN)));
@@ -1312,2 +1395,2 @@ int AudioPolicyService::UidPolicy::getUidState(uid_t uid) {
-        mCachedUids.insert(
-                std::pair<uid_t, std::pair<bool, int>>(uid, std::pair<bool, int>(active, state)));
+        mCachedUids.insert(std::pair<uid_t,
+                           std::pair<bool, int>>(uid, std::pair<bool, int>(active, state)));
@@ -1326 +1409,2 @@ void AudioPolicyService::UidPolicy::onUidIdle(uid_t uid, __unused bool disabled)
-void AudioPolicyService::UidPolicy::onUidStateChanged(uid_t uid, int32_t procState,
+void AudioPolicyService::UidPolicy::onUidStateChanged(uid_t uid,
+                                                      int32_t procState,
@@ -1333 +1417,2 @@ void AudioPolicyService::UidPolicy::onUidStateChanged(uid_t uid, int32_t procSta
-void AudioPolicyService::UidPolicy::onUidProcAdjChanged(uid_t uid __unused, int32_t adj __unused) {}
+void AudioPolicyService::UidPolicy::onUidProcAdjChanged(uid_t uid __unused, int32_t adj __unused) {
+}
@@ -1343,2 +1428,6 @@ void AudioPolicyService::UidPolicy::notifyService() {
-void AudioPolicyService::UidPolicy::updateUid(std::unordered_map<uid_t, std::pair<bool, int>>* uids,
-                                              uid_t uid, bool active, int state, bool insert) {
+void AudioPolicyService::UidPolicy::updateUid(std::unordered_map<uid_t,
+                                              std::pair<bool, int>> *uids,
+                                              uid_t uid,
+                                              bool active,
+                                              int state,
+                                              bool insert) {
@@ -1358,2 +1447,5 @@ void AudioPolicyService::UidPolicy::updateUid(std::unordered_map<uid_t, std::pai
-void AudioPolicyService::UidPolicy::updateUidLocked(
-        std::unordered_map<uid_t, std::pair<bool, int>>* uids, uid_t uid, bool active, int state,
+void AudioPolicyService::UidPolicy::updateUidLocked(std::unordered_map<uid_t,
+                                                    std::pair<bool, int>> *uids,
+                                                    uid_t uid,
+                                                    bool active,
+                                                    int state,
@@ -1376,2 +1468,2 @@ void AudioPolicyService::UidPolicy::updateUidLocked(
-        uids->insert(
-                std::pair<uid_t, std::pair<bool, int>>(uid, std::pair<bool, int>(active, state)));
+        uids->insert(std::pair<uid_t, std::pair<bool, int>>(uid,
+                                      std::pair<bool, int>(active, state)));
@@ -1386,2 +1478,2 @@ bool AudioPolicyService::UidPolicy::isA11yOnTop() {
-        if (uid.second.second >= ActivityManager::PROCESS_STATE_TOP &&
-            uid.second.second <= ActivityManager::PROCESS_STATE_BOUND_FOREGROUND_SERVICE) {
+        if (uid.second.second >= ActivityManager::PROCESS_STATE_TOP
+                && uid.second.second <= ActivityManager::PROCESS_STATE_BOUND_FOREGROUND_SERVICE) {
@@ -1393 +1485,2 @@ bool AudioPolicyService::UidPolicy::isA11yOnTop() {
-bool AudioPolicyService::UidPolicy::isA11yUid(uid_t uid) {
+bool AudioPolicyService::UidPolicy::isA11yUid(uid_t uid)
+{
@@ -1401 +1494,2 @@ void AudioPolicyService::UidPolicy::setAssistantUids(const std::vector<uid_t>& u
-bool AudioPolicyService::UidPolicy::isAssistantUid(uid_t uid) {
+bool AudioPolicyService::UidPolicy::isAssistantUid(uid_t uid)
+{
@@ -1408,3 +1502,4 @@ void AudioPolicyService::UidPolicy::setActiveAssistantUids(const std::vector<uid
-bool AudioPolicyService::UidPolicy::isActiveAssistantUid(uid_t uid) {
-    std::vector<uid_t>::iterator it =
-            find(mActiveAssistantUids.begin(), mActiveAssistantUids.end(), uid);
+bool AudioPolicyService::UidPolicy::isActiveAssistantUid(uid_t uid)
+{
+    std::vector<uid_t>::iterator it = find(mActiveAssistantUids.begin(),
+            mActiveAssistantUids.end(), uid);
@@ -1467,2 +1562,5 @@ AudioPolicyService::AudioCommandThread::AudioCommandThread(String8 name,
-    : Thread(false), mName(name), mService(service) {}
-AudioPolicyService::AudioCommandThread::~AudioCommandThread() {
+    : Thread(false), mName(name), mService(service)
+{
+}
+AudioPolicyService::AudioCommandThread::~AudioCommandThread()
+{
@@ -1474 +1572,2 @@ AudioPolicyService::AudioCommandThread::~AudioCommandThread() {
-void AudioPolicyService::AudioCommandThread::onFirstRef() {
+void AudioPolicyService::AudioCommandThread::onFirstRef()
+{
@@ -1477 +1576,2 @@ void AudioPolicyService::AudioCommandThread::onFirstRef() {
-bool AudioPolicyService::AudioCommandThread::threadLoop() {
+bool AudioPolicyService::AudioCommandThread::threadLoop()
+{
@@ -1480 +1580,2 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-    while (!exitPending()) {
+    while (!exitPending())
+    {
@@ -1496,2 +1597 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-                            volume %f, output %d",
-                              data->mStream, data->mVolume, data->mIO);
+                            volume %f, output %d", data->mStream, data->mVolume, data->mIO);
@@ -1500 +1600,2 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-                                                                        data->mVolume, data->mIO);
+                                                                    data->mVolume,
+                                                                    data->mIO);
@@ -1508,2 +1609 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-                        command->mStatus =
-                                AudioSystem::setParameters(data->mIO, data->mKeyValuePairs);
+                    command->mStatus = AudioSystem::setParameters(data->mIO, data->mKeyValuePairs);
@@ -1601,2 +1701 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-                        SetAudioPortConfigData* data =
-                                (SetAudioPortConfigData*)command->mParam.get();
+                    SetAudioPortConfigData *data = (SetAudioPortConfigData *)command->mParam.get();
@@ -1635,3 +1734,3 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-                        svc->doOnRecordingConfigurationUpdate(
-                                data->mEvent, &data->mClientInfo, &data->mClientConfig,
-                                data->mClientEffects, &data->mDeviceConfig, data->mEffects,
+                    svc->doOnRecordingConfigurationUpdate(data->mEvent, &data->mClientInfo,
+                            &data->mClientConfig, data->mClientEffects,
+                            &data->mDeviceConfig, data->mEffects,
@@ -1642,2 +1741 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-                        SetEffectSuspendedData* data =
-                                (SetEffectSuspendedData*)command->mParam.get();
+                    SetEffectSuspendedData *data = (SetEffectSuspendedData *)command->mParam.get();
@@ -1648,2 +1746 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-                            af->setEffectSuspended(data->mEffectId, data->mSessionId,
-                                                   data->mSuspended);
+                        af->setEffectSuspended(data->mEffectId, data->mSessionId, data->mSuspended);
@@ -1749 +1846,3 @@ bool AudioPolicyService::AudioCommandThread::threadLoop() {
-NO_THREAD_SAFETY_ANALYSIS() {
+status_t AudioPolicyService::AudioCommandThread::dump(int fd)
+NO_THREAD_SAFETY_ANALYSIS
+{
@@ -1779 +1878,2 @@ status_t AudioPolicyService::AudioCommandThread::volumeCommand(audio_stream_type
-                                                               int delayMs) {
+                                                               int delayMs)
+{
@@ -1788,2 +1888,2 @@ status_t AudioPolicyService::AudioCommandThread::volumeCommand(audio_stream_type
-    ALOGV("AudioCommandThread() adding set volume stream %d, volume %f, output %d", stream, volume,
-          output);
+    ALOGV("AudioCommandThread() adding set volume stream %d, volume %f, output %d",
+            stream, volume, output);
@@ -1794 +1894,2 @@ status_t AudioPolicyService::AudioCommandThread::parametersCommand(audio_io_hand
-                                                                   int delayMs) {
+                                                                   int delayMs)
+{
@@ -1802,2 +1903,2 @@ status_t AudioPolicyService::AudioCommandThread::parametersCommand(audio_io_hand
-    ALOGV("AudioCommandThread() adding set parameter string %s, io %d ,delay %d", keyValuePairs,
-          ioHandle, delayMs);
+    ALOGV("AudioCommandThread() adding set parameter string %s, io %d ,delay %d",
+            keyValuePairs, ioHandle, delayMs);
@@ -1806 +1907,2 @@ status_t AudioPolicyService::AudioCommandThread::parametersCommand(audio_io_hand
-status_t AudioPolicyService::AudioCommandThread::voiceVolumeCommand(float volume, int delayMs) {
+status_t AudioPolicyService::AudioCommandThread::voiceVolumeCommand(float volume, int delayMs)
+{
@@ -1818 +1920,2 @@ void AudioPolicyService::AudioCommandThread::setEffectSuspendedCommand(int effec
-                                                                       bool suspended) {
+                                                                       bool suspended)
+{
@@ -1830 +1933,2 @@ void AudioPolicyService::AudioCommandThread::setEffectSuspendedCommand(int effec
-void AudioPolicyService::AudioCommandThread::stopOutputCommand(audio_port_handle_t portId) {
+void AudioPolicyService::AudioCommandThread::stopOutputCommand(audio_port_handle_t portId)
+{
@@ -1839 +1943,2 @@ void AudioPolicyService::AudioCommandThread::stopOutputCommand(audio_port_handle
-void AudioPolicyService::AudioCommandThread::releaseOutputCommand(audio_port_handle_t portId) {
+void AudioPolicyService::AudioCommandThread::releaseOutputCommand(audio_port_handle_t portId)
+{
@@ -1849 +1954,4 @@ status_t AudioPolicyService::AudioCommandThread::createAudioPatchCommand(
-        const struct audio_patch* patch, audio_patch_handle_t* handle, int delayMs) {
+                                                const struct audio_patch *patch,
+                                                audio_patch_handle_t *handle,
+                                                int delayMs)
+{
@@ -1865,2 +1973,3 @@ status_t AudioPolicyService::AudioCommandThread::createAudioPatchCommand(
-status_t AudioPolicyService::AudioCommandThread::releaseAudioPatchCommand(
-        audio_patch_handle_t handle, int delayMs) {
+status_t AudioPolicyService::AudioCommandThread::releaseAudioPatchCommand(audio_patch_handle_t handle,
+                                                 int delayMs)
+{
@@ -1876 +1985,2 @@ status_t AudioPolicyService::AudioCommandThread::releaseAudioPatchCommand(
-void AudioPolicyService::AudioCommandThread::updateAudioPortListCommand() {
+void AudioPolicyService::AudioCommandThread::updateAudioPortListCommand()
+{
@@ -1882 +1992,2 @@ void AudioPolicyService::AudioCommandThread::updateAudioPortListCommand() {
-void AudioPolicyService::AudioCommandThread::updateUidStatesCommand() {
+void AudioPolicyService::AudioCommandThread::updateUidStatesCommand()
+{
@@ -1888 +1999,2 @@ void AudioPolicyService::AudioCommandThread::updateUidStatesCommand() {
-void AudioPolicyService::AudioCommandThread::updateAudioPatchListCommand() {
+void AudioPolicyService::AudioCommandThread::updateAudioPatchListCommand()
+{
@@ -1895 +2007,2 @@ void AudioPolicyService::AudioCommandThread::changeAudioVolumeGroupCommand(volum
-                                                                           int flags) {
+                                                                           int flags)
+{
@@ -1906 +2019,2 @@ status_t AudioPolicyService::AudioCommandThread::setAudioPortConfigCommand(
-        const struct audio_port_config* config, int delayMs) {
+                                            const struct audio_port_config *config, int delayMs)
+{
@@ -1917 +2031,2 @@ void AudioPolicyService::AudioCommandThread::dynamicPolicyMixStateUpdateCommand(
-        const String8& regId, int32_t state) {
+        const String8& regId, int32_t state)
+{
@@ -1929,4 +2044,9 @@ void AudioPolicyService::AudioCommandThread::recordingConfigurationUpdateCommand
-        int event, const record_client_info_t* clientInfo, const audio_config_base_t* clientConfig,
-        std::vector<effect_descriptor_t> clientEffects, const audio_config_base_t* deviceConfig,
-        std::vector<effect_descriptor_t> effects, audio_patch_handle_t patchHandle,
-        audio_source_t source) {
+                                                int event,
+                                                const record_client_info_t *clientInfo,
+                                                const audio_config_base_t *clientConfig,
+                                                std::vector<effect_descriptor_t> clientEffects,
+                                                const audio_config_base_t *deviceConfig,
+                                                std::vector<effect_descriptor_t> effects,
+                                                audio_patch_handle_t patchHandle,
+                                                audio_source_t source)
+{
@@ -1949 +2069,2 @@ void AudioPolicyService::AudioCommandThread::recordingConfigurationUpdateCommand
-void AudioPolicyService::AudioCommandThread::audioModulesUpdateCommand() {
+void AudioPolicyService::AudioCommandThread::audioModulesUpdateCommand()
+{
@@ -1954 +2075,2 @@ void AudioPolicyService::AudioCommandThread::audioModulesUpdateCommand() {
-void AudioPolicyService::AudioCommandThread::routingChangedCommand() {
+void AudioPolicyService::AudioCommandThread::routingChangedCommand()
+{
@@ -1960 +2082,2 @@ void AudioPolicyService::AudioCommandThread::routingChangedCommand() {
-void AudioPolicyService::AudioCommandThread::checkSpatializerCommand() {
+void AudioPolicyService::AudioCommandThread::checkSpatializerCommand()
+{
@@ -1966 +2089,2 @@ void AudioPolicyService::AudioCommandThread::checkSpatializerCommand() {
-void AudioPolicyService::AudioCommandThread::updateActiveSpatializerTracksCommand() {
+void AudioPolicyService::AudioCommandThread::updateActiveSpatializerTracksCommand()
+{
@@ -1972 +2096,2 @@ void AudioPolicyService::AudioCommandThread::updateActiveSpatializerTracksComman
-void AudioPolicyService::AudioCommandThread::volRangeInitReqCommand() {
+void AudioPolicyService::AudioCommandThread::volRangeInitReqCommand()
+{
@@ -1978,2 +2103,2 @@ void AudioPolicyService::AudioCommandThread::volRangeInitReqCommand() {
-status_t AudioPolicyService::AudioCommandThread::sendCommand(sp<AudioCommand>& command,
-                                                             int delayMs) {
+status_t AudioPolicyService::AudioCommandThread::sendCommand(sp<AudioCommand>& command, int delayMs)
+{
@@ -1988,9 +2113,2 @@ status_t AudioPolicyService::AudioCommandThread::sendCommand(sp<AudioCommand>& c
-<<<<<<< HEAD
-        if (command->mCond.wait_for(ul, std::chrono::nanoseconds(timeOutNs), getTid()) ==
-            std::cv_status::timeout) {
-||||||| 47af5d30a6
-        if (command->mCond.waitRelative(command->mLock, timeOutNs) != NO_ERROR) {
-=======
-        if (command->mCond.wait_for(ul, std::chrono::nanoseconds(timeOutNs)) ==
-            std::cv_status::timeout) {
->>>>>>> 43915f776988367ca9317b75225cd0291c0fd430
+        if (command->mCond.wait_for(
+                ul, std::chrono::nanoseconds(timeOutNs), getTid()) == std::cv_status::timeout) {
@@ -2003,2 +2121,2 @@ status_t AudioPolicyService::AudioCommandThread::sendCommand(sp<AudioCommand>& c
-void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& command,
-                                                             int delayMs) {
+void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& command, int delayMs)
+{
@@ -2020,2 +2138 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& c
-        } else if (command2->mCommand != command->mCommand)
-            continue;
+        } else if (command2->mCommand != command->mCommand) continue;
@@ -2059,2 +2176,2 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& c
-                ALOGV("Filtering out volume command on output %d for stream %d", data->mIO,
-                      data->mStream);
+            ALOGV("Filtering out volume command on output %d for stream %d",
+                    data->mIO, data->mStream);
@@ -2068,2 +2185,2 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& c
-                ALOGV("Filtering out voice volume command value %f replaced by %f", data2->mVolume,
-                      data->mVolume);
+            ALOGV("Filtering out voice volume command value %f replaced by %f",
+                  data2->mVolume, data->mVolume);
@@ -2105 +2222,2 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& c
-                        if (isOutputDiff) break;
+                    if (isOutputDiff)
+                       break;
@@ -2139,2 +2257,2 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& c
-    ALOGV("inserting command: %d at index %zd, num commands %zu", command->mCommand, i + 1,
-          mAudioCommands.size());
+    ALOGV("inserting command: %d at index %zd, num commands %zu",
+            command->mCommand, i+1, mAudioCommands.size());
@@ -2143 +2261,2 @@ void AudioPolicyService::AudioCommandThread::insertCommand_l(sp<AudioCommand>& c
-void AudioPolicyService::AudioCommandThread::exit() {
+void AudioPolicyService::AudioCommandThread::exit()
+{
@@ -2152,7 +2271,15 @@ void AudioPolicyService::AudioCommandThread::exit() {
-void AudioPolicyService::AudioCommandThread::AudioCommand::dump(char* buffer, size_t size) {
-    snprintf(buffer, size, "   %02d      %06d.%03d  %01u    %p\n", mCommand, (int)ns2s(mTime),
-             (int)ns2ms(mTime) % 1000, mWaitStatus, mParam.get());
-}
-void AudioPolicyService::setParameters(audio_io_handle_t ioHandle, const char* keyValuePairs,
-                                       int delayMs) {
-    mAudioCommandThread->parametersCommand(ioHandle, keyValuePairs, delayMs);
+void AudioPolicyService::AudioCommandThread::AudioCommand::dump(char* buffer, size_t size)
+{
+    snprintf(buffer, size, "   %02d      %06d.%03d  %01u    %p\n",
+            mCommand,
+            (int)ns2s(mTime),
+            (int)ns2ms(mTime)%1000,
+            mWaitStatus,
+            mParam.get());
+}
+void AudioPolicyService::setParameters(audio_io_handle_t ioHandle,
+                                       const char *keyValuePairs,
+                                       int delayMs)
+{
+    mAudioCommandThread->parametersCommand(ioHandle, keyValuePairs,
+                                           delayMs);
@@ -2160,3 +2287,7 @@ void AudioPolicyService::setParameters(audio_io_handle_t ioHandle, const char* k
-int AudioPolicyService::setStreamVolume(audio_stream_type_t stream, float volume,
-                                        audio_io_handle_t output, int delayMs) {
-    return (int)mAudioCommandThread->volumeCommand(stream, volume, output, delayMs);
+int AudioPolicyService::setStreamVolume(audio_stream_type_t stream,
+                                        float volume,
+                                        audio_io_handle_t output,
+                                        int delayMs)
+{
+    return (int)mAudioCommandThread->volumeCommand(stream, volume,
+                                                   output, delayMs);
@@ -2164 +2295,2 @@ int AudioPolicyService::setStreamVolume(audio_stream_type_t stream, float volume
-int AudioPolicyService::setVoiceVolume(float volume, int delayMs) {
+int AudioPolicyService::setVoiceVolume(float volume, int delayMs)
+{
@@ -2167,2 +2299,4 @@ int AudioPolicyService::setVoiceVolume(float volume, int delayMs) {
-void AudioPolicyService::setEffectSuspended(int effectId, audio_session_t sessionId,
-                                            bool suspended) {
+void AudioPolicyService::setEffectSuspended(int effectId,
+                                            audio_session_t sessionId,
+                                            bool suspended)
+{
@@ -2171 +2305,2 @@ void AudioPolicyService::setEffectSuspended(int effectId, audio_session_t sessio
-Status AudioPolicyService::onNewAudioModulesAvailable() {
+Status AudioPolicyService::onNewAudioModulesAvailable()
+{
@@ -2176,4 +2311,8 @@ extern "C" {
-audio_module_handle_t aps_load_hw_module(void* service __unused, const char* name);
-audio_io_handle_t aps_open_output(void* service __unused, audio_devices_t* pDevices,
-                                  uint32_t* pSamplingRate, audio_format_t* pFormat,
-                                  audio_channel_mask_t* pChannelMask, uint32_t* pLatencyMs,
+audio_module_handle_t aps_load_hw_module(void *service __unused,
+                                             const char *name);
+audio_io_handle_t aps_open_output(void *service __unused,
+                                         audio_devices_t *pDevices,
+                                         uint32_t *pSamplingRate,
+                                         audio_format_t *pFormat,
+                                         audio_channel_mask_t *pChannelMask,
+                                         uint32_t *pLatencyMs,
@@ -2181,2 +2320,4 @@ audio_io_handle_t aps_open_output(void* service __unused, audio_devices_t* pDevi
-audio_io_handle_t aps_open_output_on_module(void* service __unused, audio_module_handle_t module,
-                                            audio_devices_t* pDevices, uint32_t* pSamplingRate,
+audio_io_handle_t aps_open_output_on_module(void *service __unused,
+                                                   audio_module_handle_t module,
+                                                   audio_devices_t *pDevices,
+                                                   uint32_t *pSamplingRate,
@@ -2185 +2326,2 @@ audio_io_handle_t aps_open_output_on_module(void* service __unused, audio_module
-                                            uint32_t* pLatencyMs, audio_output_flags_t flags,
+                                                   uint32_t *pLatencyMs,
+                                                   audio_output_flags_t flags,
@@ -2187 +2329,2 @@ audio_io_handle_t aps_open_output_on_module(void* service __unused, audio_module
-audio_io_handle_t aps_open_dup_output(void* service __unused, audio_io_handle_t output1,
+audio_io_handle_t aps_open_dup_output(void *service __unused,
+                                                 audio_io_handle_t output1,
@@ -2192,2 +2335,4 @@ int aps_restore_output(void* service __unused, audio_io_handle_t output);
-audio_io_handle_t aps_open_input(void* service __unused, audio_devices_t* pDevices,
-                                 uint32_t* pSamplingRate, audio_format_t* pFormat,
+audio_io_handle_t aps_open_input(void *service __unused,
+                                        audio_devices_t *pDevices,
+                                        uint32_t *pSamplingRate,
+                                        audio_format_t *pFormat,
@@ -2196,2 +2341,4 @@ audio_io_handle_t aps_open_input(void* service __unused, audio_devices_t* pDevic
-audio_io_handle_t aps_open_input_on_module(void* service __unused, audio_module_handle_t module,
-                                           audio_devices_t* pDevices, uint32_t* pSamplingRate,
+audio_io_handle_t aps_open_input_on_module(void *service __unused,
+                                                  audio_module_handle_t module,
+                                                  audio_devices_t *pDevices,
+                                                  uint32_t *pSamplingRate,
@@ -2202 +2349,2 @@ int aps_invalidate_stream(void* service __unused, audio_stream_type_t stream);
-int aps_move_effects(void* service __unused, audio_session_t session, audio_io_handle_t src_output,
+int aps_move_effects(void *service __unused, audio_session_t session,
+                                audio_io_handle_t src_output,
@@ -2204,2 +2352,6 @@ int aps_move_effects(void* service __unused, audio_session_t session, audio_io_h
-char* aps_get_parameters(void* service __unused, audio_io_handle_t io_handle, const char* keys);
-void aps_set_parameters(void* service, audio_io_handle_t io_handle, const char* kv_pairs,
+char * aps_get_parameters(void *service __unused, audio_io_handle_t io_handle,
+                                     const char *keys);
+void aps_set_parameters(void *service, audio_io_handle_t io_handle,
+                                   const char *kv_pairs, int delay_ms);
+int aps_set_stream_volume(void *service, audio_stream_type_t stream,
+                                     float volume, audio_io_handle_t output,
@@ -2207,2 +2358,0 @@ void aps_set_parameters(void* service, audio_io_handle_t io_handle, const char*
-int aps_set_stream_volume(void* service, audio_stream_type_t stream, float volume,
-                          audio_io_handle_t output, int delay_ms);
@@ -2210 +2360 @@ int aps_set_voice_volume(void* service, float volume, int delay_ms);
-}
+};
