diff --git a/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyEffects.no_comments_conflict.h b/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
index 8a79b8e..7b2201c 100644
--- a/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyEffects.no_comments_conflict.h
+++ b/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
@@ -95 +94,0 @@ private:
-<<<<<<< HEAD
@@ -98,7 +96,0 @@ private:
-||||||| 47af5d30a6
-        int mRefCount;
-        Vector< sp<AudioEffect> >mEffects;
-=======
-        int mRefCount;
-        Vector< sp<AudioEffect> >mEffects;
->>>>>>> 43915f77
@@ -153,6 +144,0 @@ private:
-<<<<<<< HEAD
-||||||| 47af5d30a6
-    std::map<std::string, std::unique_ptr<DeviceEffects>> mDeviceEffects GUARDED_BY(mLock);
-=======
-    std::map<std::string, std::unique_ptr<DeviceEffects>> mDeviceEffects GUARDED_BY(mMutex);
->>>>>>> 43915f77
