diff --git a/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_conflict.cpp b/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_merged.cpp
index 641181a..ad5967f 100644
--- a/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_conflict.cpp
+++ b/./output/frameworks_av/91dfd997-43915f77-47af5d30/services@audiopolicy@service@AudioPolicyService.no_comments_merged.cpp
@@ -2113 +2112,0 @@ status_t AudioPolicyService::AudioCommandThread::sendCommand(sp<AudioCommand>& c
-<<<<<<< HEAD
@@ -2116,6 +2114,0 @@ status_t AudioPolicyService::AudioCommandThread::sendCommand(sp<AudioCommand>& c
-||||||| 47af5d30a6
-        if (command->mCond.waitRelative(command->mLock, timeOutNs) != NO_ERROR) {
-=======
-        if (command->mCond.wait_for(
-                    ul, std::chrono::nanoseconds(timeOutNs)) == std::cv_status::timeout) {
->>>>>>> 43915f77
