--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/61853ad8-a9628142-bc5cb168/src@module.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/61853ad8-a9628142-bc5cb168/src@module.no_comments_merged.c
@@ -4 +3,0 @@
-#include <sys/wait.h>
@@ -5,0 +5 @@
+#include <sys/wait.h>
@@ -28 +26,0 @@ struct RedisModule {
-  RedisModuleInfoFunc info_cb;
@@ -29,0 +28 @@ struct RedisModule {
+    RedisModuleInfoFunc info_cb;
@@ -73,5 +72 @@ typedef struct RedisModuleCtx RedisModuleCtx;
-#define REDISMODULE_CTX_INIT \
-  { \
-    (void *)(unsigned long)&RM_GetApi, NULL, NULL, NULL, NULL, 0, 0, 0, NULL, \
-        0, NULL, NULL, 0, NULL \
-  }
+#define REDISMODULE_CTX_INIT {(void*)(unsigned long)&RM_GetApi, NULL, NULL, NULL, NULL, 0, 0, 0, NULL, 0, NULL, NULL, 0, NULL}
@@ -117,2 +111 @@ typedef struct RedisModuleCommandProxy RedisModuleCommandProxy;
-#define REDISMODULE_REPLYFLAG_NESTED \
-  (1 << 1)
+#define REDISMODULE_REPLYFLAG_NESTED (1<<1)
@@ -2889,0 +2833,2 @@ int moduleRegisterApi(const char *funcname, void *funcptr) {
+#define REGISTER_API(name) \
+    moduleRegisterApi("RedisModule_" #name, (void *)(unsigned long)RM_ ## name)
@@ -3297 +3245,3 @@ void moduleRegisterCoreAPI(void) {
-<<<<<<< HEAD
+    REGISTER_API(Fork);
+    REGISTER_API(ExitFromChild);
+    REGISTER_API(KillForkChild);
@@ -3307,6 +3256,0 @@ void moduleRegisterCoreAPI(void) {
-|||||||
-=======
-  REGISTER_API(Fork);
-  REGISTER_API(ExitFromChild);
-  REGISTER_API(KillForkChild);
->>>>>>> a9628142fd3ce5e3d221ba4b60bcb33c394140e1
