diff --git a/./output/tmux/6a2f32b4-ff135b34-1c8f7c1f/tmux.no_comments_mergegen.h b/./output/tmux/6a2f32b4-ff135b34-1c8f7c1f/tmux.no_comments_merged.h
index 6bbdd63..214d417 100644
--- a/./output/tmux/6a2f32b4-ff135b34-1c8f7c1f/tmux.no_comments_mergegen.h
+++ b/./output/tmux/6a2f32b4-ff135b34-1c8f7c1f/tmux.no_comments_merged.h
@@ -2206 +2206,3 @@ int utf8_cstrhas(const char *, const struct utf8_data *);
-char *osdep_get_name(int, char *); char *osdep_get_cwd(int); struct event_base *osdep_event_init(void);
+char *osdep_get_name(int, char *);
+char *osdep_get_cwd(int);
+struct event_base *osdep_event_init(void);
