diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/136e4efb-1a8abe72-161d9e58/java@rocksjni@portal.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/136e4efb-1a8abe72-161d9e58/java@rocksjni@portal.no_comments_merged.h
index 7a50704..449d78b 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/136e4efb-1a8abe72-161d9e58/java@rocksjni@portal.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/136e4efb-1a8abe72-161d9e58/java@rocksjni@portal.no_comments_merged.h
@@ -10 +10 @@ class RocksDBJni {
-    static jclass jclazz = env->FindClass("org/rocksdb/WriteBatch");
+    static jclass jclazz = env->FindClass("org/rocksdb/RocksDB");
@@ -14,2 +14,5 @@ class RocksDBJni {
-  static void setHandle(JNIEnv* env, jobject jdb, rocksdb::DB* db) {
-    env->SetLongField(jdb, getHandleFieldID(env), reinterpret_cast<jlong>(db));
+  static jfieldID getHandleFieldID(JNIEnv* env) {
+    static jfieldID fid = env->GetFieldID(
+        getJClass(env), "nativeHandle_", "J");
+    assert(fid != nullptr);
+    return fid;
@@ -17,2 +20,3 @@ class RocksDBJni {
-  static void setHandle(JNIEnv* env, jobject jdb, rocksdb::DB* db) {
-    env->SetLongField(jdb, getHandleFieldID(env), reinterpret_cast<jlong>(db));
+  static rocksdb::DB* getHandle(JNIEnv* env, jobject jdb) {
+    return reinterpret_cast<rocksdb::DB*>(
+        env->GetLongField(jdb, getHandleFieldID(env)));
@@ -21 +25,3 @@ class RocksDBJni {
-    env->SetLongField(jdb, getHandleFieldID(env), reinterpret_cast<jlong>(db));
+    env->SetLongField(
+        jdb, getHandleFieldID(env),
+        reinterpret_cast<jlong>(db));
@@ -27 +33 @@ class RocksDBExceptionJni {
-    static jclass jclazz = env->FindClass("org/rocksdb/WriteBatch");
+    static jclass jclazz = env->FindClass("org/rocksdb/RocksDBException");
@@ -36,2 +42,2 @@ class RocksDBExceptionJni {
-    static jmethodID mid =
-        env->GetMethodID(getJClass(env), "<init>", "(Ljava/lang/String;)V");
+    static jmethodID mid = env->GetMethodID(
+        getJClass(env), "<init>", "(Ljava/lang/String;)V");
@@ -45 +51 @@ class OptionsJni {
-    static jclass jclazz = env->FindClass("org/rocksdb/WriteBatch");
+    static jclass jclazz = env->FindClass("org/rocksdb/Options");
@@ -49,2 +55,5 @@ class OptionsJni {
-  static void setHandle(JNIEnv* env, jobject jobj, rocksdb::Options* op) {
-    env->SetLongField(jobj, getHandleFieldID(env), reinterpret_cast<jlong>(op));
+  static jfieldID getHandleFieldID(JNIEnv* env) {
+    static jfieldID fid = env->GetFieldID(
+        getJClass(env), "nativeHandle_", "J");
+    assert(fid != nullptr);
+    return fid;
@@ -52,2 +61,3 @@ class OptionsJni {
-  static void setHandle(JNIEnv* env, jobject jobj, rocksdb::Options* op) {
-    env->SetLongField(jobj, getHandleFieldID(env), reinterpret_cast<jlong>(op));
+  static rocksdb::Options* getHandle(JNIEnv* env, jobject jobj) {
+    return reinterpret_cast<rocksdb::Options*>(
+        env->GetLongField(jobj, getHandleFieldID(env)));
@@ -56 +66,3 @@ class OptionsJni {
-    env->SetLongField(jobj, getHandleFieldID(env), reinterpret_cast<jlong>(op));
+    env->SetLongField(
+        jobj, getHandleFieldID(env),
+        reinterpret_cast<jlong>(op));
@@ -62 +74 @@ class WriteOptionsJni {
-    static jclass jclazz = env->FindClass("org/rocksdb/WriteBatch");
+    static jclass jclazz = env->FindClass("org/rocksdb/WriteOptions");
@@ -66,2 +78,5 @@ class WriteOptionsJni {
-  static void setHandle(JNIEnv* env, jobject jobj, rocksdb::WriteOptions* op) {
-    env->SetLongField(jobj, getHandleFieldID(env), reinterpret_cast<jlong>(op));
+  static jfieldID getHandleFieldID(JNIEnv* env) {
+    static jfieldID fid = env->GetFieldID(
+        getJClass(env), "nativeHandle_", "J");
+    assert(fid != nullptr);
+    return fid;
@@ -69,2 +84,3 @@ class WriteOptionsJni {
-  static void setHandle(JNIEnv* env, jobject jobj, rocksdb::WriteOptions* op) {
-    env->SetLongField(jobj, getHandleFieldID(env), reinterpret_cast<jlong>(op));
+  static rocksdb::WriteOptions* getHandle(JNIEnv* env, jobject jobj) {
+    return reinterpret_cast<rocksdb::WriteOptions*>(
+        env->GetLongField(jobj, getHandleFieldID(env)));
@@ -73 +89,3 @@ class WriteOptionsJni {
-    env->SetLongField(jobj, getHandleFieldID(env), reinterpret_cast<jlong>(op));
+    env->SetLongField(
+        jobj, getHandleFieldID(env),
+        reinterpret_cast<jlong>(op));
@@ -83,2 +101,5 @@ class WriteBatchJni {
-  static void setHandle(JNIEnv* env, jobject jwb, rocksdb::WriteBatch* wb) {
-    env->SetLongField(jwb, getHandleFieldID(env), reinterpret_cast<jlong>(wb));
+  static jfieldID getHandleFieldID(JNIEnv* env) {
+    static jfieldID fid = env->GetFieldID(
+        getJClass(env), "nativeHandle_", "J");
+    assert(fid != nullptr);
+    return fid;
@@ -86,2 +107,3 @@ class WriteBatchJni {
-  static void setHandle(JNIEnv* env, jobject jwb, rocksdb::WriteBatch* wb) {
-    env->SetLongField(jwb, getHandleFieldID(env), reinterpret_cast<jlong>(wb));
+  static rocksdb::WriteBatch* getHandle(JNIEnv* env, jobject jwb) {
+    return reinterpret_cast<rocksdb::WriteBatch*>(
+        env->GetLongField(jwb, getHandleFieldID(env)));
@@ -90 +112,3 @@ class WriteBatchJni {
-    env->SetLongField(jwb, getHandleFieldID(env), reinterpret_cast<jlong>(wb));
+    env->SetLongField(
+        jwb, getHandleFieldID(env),
+        reinterpret_cast<jlong>(wb));
@@ -92,0 +117 @@ class WriteBatchJni {
+<<<<<<< HEAD
@@ -96 +121,2 @@ class HistogramDataJni {
-    static jmethodID mid = env->GetMethodID(jclazz, "<init>", "(DDDDD)V");
+    static jmethodID mid = env->GetMethodID(
+        jclazz, "<init>", "(DDDDD)V");
@@ -99,2 +125 @@ class HistogramDataJni {
-  }
-};
+=======
@@ -109 +134,2 @@ class BackupableDBOptionsJni {
-    static jfieldID fid = env->GetFieldID(getJClass(env), "nativeHandle_", "J");
+    static jfieldID fid = env->GetFieldID(
+        getJClass(env), "nativeHandle_", "J");
@@ -117,3 +143,6 @@ class BackupableDBOptionsJni {
-  static void setHandle(JNIEnv* env, jobject jobj,
-                        rocksdb::BackupableDBOptions* op) {
-    env->SetLongField(jobj, getHandleFieldID(env), reinterpret_cast<jlong>(op));
+  static void setHandle(
+      JNIEnv* env, jobject jobj, rocksdb::BackupableDBOptions* op) {
+    env->SetLongField(
+        jobj, getHandleFieldID(env),
+        reinterpret_cast<jlong>(op));
+>>>>>>> 1a8abe72768b2b5cea800aa390c28e5ace6a552e
