--- a/./output/system_core/9c0d2e12-5f4e8eac-d43b6152/init@devices.no_comments_conflict.h
+++ b/./output/system_core/9c0d2e12-5f4e8eac-d43b6152/init@devices.no_comments_merged.h
@@ -5 +4,0 @@
-<<<<<<< HEAD
@@ -28,16 +26,0 @@ class Permissions {
-||||||| d43b61521
-enum coldboot_action_t {
-    COLDBOOT_CONTINUE = 0,
-    COLDBOOT_CREATE,
-    COLDBOOT_STOP,
-    COLDBOOT_FINISH
-=======
-#include <chrono>
-#include <functional>
-#include <optional>
-enum coldboot_action_t {
-    COLDBOOT_CONTINUE = 0,
-    COLDBOOT_CREATE,
-    COLDBOOT_STOP,
-    COLDBOOT_FINISH
->>>>>>> 5f4e8eac
@@ -58 +40,0 @@ class Subsystem {
-<<<<<<< HEAD
@@ -60,12 +41,0 @@ class Subsystem {
-||||||| d43b61521
-extern int add_dev_perms(const char *name, const char *attr,
-                         mode_t perm, unsigned int uid,
-                         unsigned int gid, unsigned short prefix,
-                         unsigned short wildcard);
-int get_device_fd();
-=======
-extern int add_dev_perms(const char *name, const char *attr,
-                         mode_t perm, unsigned int uid,
-                         unsigned int gid, unsigned short prefix,
-                         unsigned short wildcard);
->>>>>>> 5f4e8eac
@@ -76 +45,0 @@ extern int add_dev_perms(const char *name, const char *attr,
-<<<<<<< HEAD
@@ -121,7 +89,0 @@ void SanitizePartitionName(std::string* string);
-||||||| d43b61521
-#endif
-=======
-void device_poll(const coldboot_callback& callback = nullptr,
-                 const std::optional<std::chrono::milliseconds> relative_timeout = {});
-#endif
->>>>>>> 5f4e8eac
