--- a/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@main@attached_database.no_comments_mergegen.cpp
+++ b/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@main@attached_database.no_comments_merged.cpp
@@ -41 +41,5 @@ AttachOptions::AttachOptions(const unique_ptr<AttachInfo> &info, const AccessMod
-if (entry.first == "default_table") { default_table = QualifiedName::Parse(StringValue::Get(entry.second.DefaultCastAs(LogicalType::VARCHAR))); continue; }
+  if (entry.first == "default_table") {
+   default_table = QualifiedName::Parse(StringValue::Get(entry.second.DefaultCastAs(LogicalType::VARCHAR)));
+   continue;
+  }
+  options[entry.first] = entry.second;
