--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/ddf8a49a-56a1d037-0a75499c/src@main@attached_database.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/ddf8a49a-56a1d037-0a75499c/src@main@attached_database.no_comments_merged.cpp
@@ -41 +40,0 @@ AttachOptions::AttachOptions(const unique_ptr<AttachInfo> &info, const AccessMod
-<<<<<<< HEAD
@@ -46,8 +44,0 @@ AttachOptions::AttachOptions(const unique_ptr<AttachInfo> &info, const AccessMod
-  if (unrecognized_option.empty()) {
-   unrecognized_option = entry.first;
-  }
-|||||||
-  if (unrecognized_option.empty()) {
-   unrecognized_option = entry.first;
-  }
-=======
@@ -55 +45,0 @@ AttachOptions::AttachOptions(const unique_ptr<AttachInfo> &info, const AccessMod
->>>>>>> 56a1d0376de5d227b5c48dcb7ad8dbb6866fece5
