--- a/./output/rocksdb/7ac437c6-758fa8c3-63cef900/db@version_set.no_comments_conflict.cc
+++ b/./output/rocksdb/7ac437c6-758fa8c3-63cef900/db@version_set.no_comments_merged.cc
@@ -373 +372,0 @@ Version::Version(ColumnFamilyData* cfd, VersionSet* vset,
-<<<<<<< HEAD
@@ -375,4 +373,0 @@ Version::Version(ColumnFamilyData* cfd, VersionSet* vset,
-||||||| 63cef9007
-      num_levels_(vset->num_levels_),
-=======
-      num_levels_(vset->num_levels_),
@@ -380 +374,0 @@ Version::Version(ColumnFamilyData* cfd, VersionSet* vset,
->>>>>>> 758fa8c3
