--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/440385de-cfdc800a-81008bf9/src@db.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/440385de-cfdc800a-81008bf9/src@db.no_comments_merged.c
@@ -216 +226,0 @@ void flushallCommand(client *c) {
-<<<<<<< HEAD
@@ -218,2 +227,0 @@ void flushallCommand(client *c) {
-|||||||
-=======
@@ -221 +228,0 @@ void flushallCommand(client *c) {
->>>>>>> cfdc800a5ff5a2bb02ccd1e21c1c36e6cb5a474d
@@ -714,3 +729 @@ int *getKeysUsingCommandTable(struct redisCommand *cmd, robj **argv, int argc,
-        serverPanic(
-            "Redis built-in command declared keys positions not matching the "
-            "arity requirements.");
+                serverPanic("Redis built-in command declared keys positions not matching the arity requirements.");
