--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/8c3eed81-43a0b172-3e96c44e/libprocessgroup@sched_policy.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/8c3eed81-43a0b172-3e96c44e/libprocessgroup@sched_policy.no_comments_merged.cpp
@@ -4,6 +3,0 @@
-#include <fcntl.h>
-#include <android-base/macros.h>
-#include <log/log.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
@@ -18,0 +13 @@ static inline SchedPolicy _policy(SchedPolicy p) {
+#if defined(__ANDROID__)
@@ -170,0 +153,9 @@ int get_sched_policy(int tid, SchedPolicy* policy) {
+#else
+int set_sched_policy(int, SchedPolicy) {
+    return 0;
+}
+int get_sched_policy(int, SchedPolicy* policy) {
+    *policy = SP_SYSTEM_DEFAULT;
+    return 0;
+}
+#endif
