--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/3898d4e7-135bb1ed-86862c97/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/3898d4e7-135bb1ed-86862c97/tmux.no_comments_merged.c
@@ -4 +3,0 @@
-#include <err.h>
@@ -8 +6,0 @@
-#include <paths.h>
@@ -13 +10,0 @@
-#include <util.h>
@@ -267,5 +314,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
-  exit(client_main(osdep_event_init(), argc, argv, flags));
-|||||||
-  exit(client_main(event_init(), argc, argv, flags));
-=======
@@ -273 +315,0 @@ int main(int argc, char **argv) {
->>>>>>> 135bb1edeeab3faae8001100aa7c173be9aa91e1
