--- a/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@include@duckdb_python@connection_wrapper.no_comments_conflict.hpp
+++ b/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@include@duckdb_python@connection_wrapper.no_comments_merged.hpp
@@ -23,5 +22,0 @@ public:
-<<<<<<< HEAD
- static shared_ptr<DuckDBPyConnection> Append(const string &name, PandasDataFrame value,
-||||||| 754ff32005
- static shared_ptr<DuckDBPyConnection> Append(const string &name, DataFrame value,
-=======
@@ -41,2 +36 @@ public:
- static shared_ptr<DuckDBPyConnection> Append(const string &name, DataFrame value,
->>>>>>> cc616059
+ static shared_ptr<DuckDBPyConnection> Append(const string &name, PandasDataFrame value,
