diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@include@duckdb_python@connection_wrapper.no_comments_mergebot.hpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@include@duckdb_python@connection_wrapper.no_comments_merged.hpp
index 1b1983d..758d539 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@include@duckdb_python@connection_wrapper.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@include@duckdb_python@connection_wrapper.no_comments_merged.hpp
@@ -2 +1,0 @@
-#include "duckdb_python/pyconnection.hpp"
@@ -9 +8,2 @@ public:
- PyConnectionWrapper()
+ PyConnectionWrapper() = delete;
+public:
