--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@duckdb_python.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@duckdb_python.no_comments_merged.cpp
@@ -1 +0,0 @@
-#include "duckdb_python/pybind_wrapper.hpp"
@@ -6 +4,0 @@
-#include "duckdb_python/pyconnection.hpp"
@@ -8,2 +5,0 @@
-#include "duckdb_python/exceptions.hpp"
-#include "duckdb_python/conversions/pyconnection_default.hpp"
@@ -20,0 +17 @@
+namespace py = pybind11;
