--- a/./output/frameworks_native/4df90e51-468b8318-7110dc0c/services@surfaceflinger@SurfaceFlinger.no_comments_conflict.h
+++ b/./output/frameworks_native/4df90e51-468b8318-7110dc0c/services@surfaceflinger@SurfaceFlinger.no_comments_merged.h
@@ -723,6 +722,0 @@ private:
-<<<<<<< HEAD
-||||||| 7110dc0c92
-    std::array<FenceWithFenceTime, 2> mPreviousPresentFences;
-=======
-    std::array<FenceWithFenceTime, 5> mPreviousPresentFences;
->>>>>>> 468b8318
