diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_merged.c
index a87399c..5b3cc6a 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/49128f0b-556bdfba-2f6b31c3/src@db.no_comments_merged.c
@@ -9,2 +9,4 @@ robj *lookupKey(redisDb *db, robj *key) {
-          val->storage == REDIS_VM_SWAPPING) {
-        if (val->storage == REDIS_VM_SWAPPING) vmCancelThreadedIOJob(val);
+                val->storage == REDIS_VM_SWAPPING)
+            {
+                if (val->storage == REDIS_VM_SWAPPING)
+                    vmCancelThreadedIOJob(val);
@@ -98 +100,2 @@ int selectDb(redisClient *c, int id) {
-  if (id < 0 || id >= server.dbnum) return REDIS_ERR;
+    if (id < 0 || id >= server.dbnum)
+        return REDIS_ERR;
@@ -164,7 +166,0 @@ void keysCommand(redisClient *c) {
-<<<<<<< HEAD
-|||||||
-  addReply(c, lenobj);
-  decrRefCount(lenobj);
-=======
-  addReply(c, lenobj);
-  decrRefCount(lenobj);
@@ -172 +167,0 @@ void keysCommand(redisClient *c) {
->>>>>>> 556bdfbab9062d472e19d882ae045fece36a25ab
@@ -191 +186,3 @@ void dbsizeCommand(redisClient *c) {
-void lastsaveCommand(redisClient *c) { addReplyLongLong(c, server.lastsave); }
+void lastsaveCommand(redisClient *c) {
+    addReplyLongLong(c,server.lastsave);
+}
@@ -200,18 +197,6 @@ void typeCommand(redisClient *c) {
-      case REDIS_STRING:
-        type = "string";
-        break;
-      case REDIS_LIST:
-        type = "list";
-        break;
-      case REDIS_SET:
-        type = "set";
-        break;
-      case REDIS_ZSET:
-        type = "zset";
-        break;
-      case REDIS_HASH:
-        type = "hash";
-        break;
-      default:
-        type = "unknown";
-        break;
+        case REDIS_STRING: type = "string"; break;
+        case REDIS_LIST: type = "list"; break;
+        case REDIS_SET: type = "set"; break;
+        case REDIS_ZSET: type = "zset"; break;
+        case REDIS_HASH: type = "hash"; break;
+        default: type = "unknown"; break;
@@ -245 +230,2 @@ void shutdownCommand(redisClient *c) {
-  if (prepareForShutdown() == REDIS_OK) exit(0);
+    if (prepareForShutdown() == REDIS_OK)
+        exit(0);
@@ -271,2 +257,6 @@ void renameGenericCommand(redisClient *c, int nx) {
-void renameCommand(redisClient *c) { renameGenericCommand(c, 0); }
-void renamenxCommand(redisClient *c) { renameGenericCommand(c, 1); }
+void renameCommand(redisClient *c) {
+    renameGenericCommand(c,0);
+}
+void renamenxCommand(redisClient *c) {
+    renameGenericCommand(c,1);
+}
@@ -316,2 +306 @@ time_t getExpire(redisDb *db, robj *key) {
-      (de = dictFind(db->expires, key->ptr)) == NULL)
-    return -1;
+       (de = dictFind(db->expires,key->ptr)) == NULL) return -1;
@@ -328 +317,2 @@ void propagateExpire(redisDb *db, robj *key) {
-  if (server.appendonly) feedAppendOnlyFile(cmd, db->id, argv, 2);
+    if (server.appendonly)
+        feedAppendOnlyFile(cmd,db->id,argv,2);
