--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
@@ -50,3 +49,0 @@ using namespace std::chrono_literals;
-using ::aidl::android::media::IAidlBufferSource;
-using ::aidl::android::media::IAidlNode;
-using android::base::StringPrintf;
@@ -53,0 +51 @@ using ::android::hardware::graphics::bufferqueue::V1_0::utils::H2BGraphicBufferP
+using android::base::StringPrintf;
@@ -54,0 +53,2 @@ using ::android::hardware::media::c2::V1_0::IInputSurface;
+using ::aidl::android::media::IAidlBufferSource;
+using ::aidl::android::media::IAidlNode;
@@ -78 +78 @@ class CCodecWatchdog : public AHandler {
-    ~CCodecWatchdog()
+    ~CCodecWatchdog() = default;
@@ -125,3 +125,5 @@ class C2InputSurfaceWrapper : public InputSurfaceWrapper {
-    explicit C2InputSurfaceWrapper(const std::shared_ptr<Codec2Client::InputSurface>& surface)
-        : mSurface(surface){}
-          () = delete;
+    explicit C2InputSurfaceWrapper(
+            const std::shared_ptr<Codec2Client::InputSurface> &surface) :
+        mSurface(surface) {
+    }
+    ~C2InputSurfaceWrapper() override = default;
@@ -168 +173 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    () = delete;
+    ~HGraphicBufferSourceWrapper() override = default;
@@ -397 +408 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    () = delete;
+    ~AGraphicBufferSourceWrapper() override = default;
@@ -409,10 +420,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-<<<<<<< HEAD
-                (*node),
-                static_cast<::aidl::android::hardware::graphics::common::Dataspace>(dataspace)));
-||||||| 63728a5cdd
-                mNode,
-                static_cast<::aidl::android::hardware::graphics::common::Dataspace>(mDataSpace)));
-=======
-                (*node),
-                static_cast<::aidl::android::hardware::graphics::common::Dataspace>(mDataSpace)));
->>>>>>> 4aa3f70e45fd04e9a77762dd991d3f31ec0a66eb
+                (*node), static_cast<::aidl::android::hardware::graphics::common::Dataspace>(
+                        dataspace)));
@@ -604,4 +607,5 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    Codec2ClientInterfaceWrapper(std::shared_ptr<Codec2Client> client) : mClient(client) {}
-    virtual ~Codec2ClientInterfaceWrapper()
-            virtual c2_status_t
-            config_sm(const std::vector<C2Param*>& params,
+    Codec2ClientInterfaceWrapper(std::shared_ptr<Codec2Client> client)
+        : mClient(client) { }
+    virtual ~Codec2ClientInterfaceWrapper() = default;
+    virtual c2_status_t config_sm(
+            const std::vector<C2Param *> &params,
@@ -610,2 +614,3 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    }
-    virtual c2_status_t copyBuffer(std::shared_ptr<C2GraphicBuffer>,
+    };
+    virtual c2_status_t copyBuffer(
+            std::shared_ptr<C2GraphicBuffer>,
@@ -761,2 +780,2 @@ class CCodecCallbackImpl : public CCodecCallback {
-    explicit CCodecCallbackImpl(CCodec* codec) : mCodec(codec){}() = delete;
-    () = delete;
+    explicit CCodecCallbackImpl(CCodec *codec) : mCodec(codec) {}
+    ~CCodecCallbackImpl() override = default;
@@ -2537 +2619 @@ class IntfCache {
-    IntfCache()
+    IntfCache() = default;
