diff --git a/./output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_conflict.cpp b/./output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
index cb093fe..49e3bdc 100644
--- a/./output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_conflict.cpp
+++ b/./output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
@@ -420 +419,0 @@ public:
-<<<<<<< HEAD
@@ -423,7 +421,0 @@ public:
-||||||| 63728a5cdd
-                mNode, static_cast<::aidl::android::hardware::graphics::common::Dataspace>(
-                        mDataSpace)));
-=======
-                (*node), static_cast<::aidl::android::hardware::graphics::common::Dataspace>(
-                        mDataSpace)));
->>>>>>> 4aa3f70e
