--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/c469f6ad-b8e02f2b-f4d37173/src@rdb.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/c469f6ad-b8e02f2b-f4d37173/src@rdb.no_comments_merged.c
@@ -37,4 +36 @@ void rdbReportError(int corruption_error, int linenum, char *reason, ...) {
-      serverLog(LL_WARNING,
-                "%s. Failure loading rdb format from socket, assuming "
-                "connection error, resuming operation.",
-                msg);
+            serverLog(LL_WARNING, "%s. Failure loading rdb format from socket, assuming connection error, resuming operation.", msg);
@@ -1323,4 +1335 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, robj *key) {
-      serverLog(LL_WARNING,
-                "The RDB file contains module data I can't load: no matching "
-                "module '%s'",
-                name);
+            serverLog(LL_WARNING,"The RDB file contains module data I can't load: no matching module '%s'", name);
@@ -1346,4 +1354 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, robj *key) {
-        serverLog(LL_WARNING,
-                  "The RDB file contains module data for the module '%s' that "
-                  "is not terminated by the proper module value EOF marker",
-                  name);
+                serverLog(LL_WARNING,"The RDB file contains module data for the module '%s' that is not terminated by the proper module value EOF marker", name);
@@ -1355,5 +1360 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, robj *key) {
-      serverLog(LL_WARNING,
-                "The RDB file contains module data for the module type '%s', "
-                "that the responsible module is not able to load. Check for "
-                "modules log above for additional clues.",
-                name);
+            serverLog(LL_WARNING,"The RDB file contains module data for the module type '%s', that the responsible module is not able to load. Check for modules log above for additional clues.", name);
@@ -1524,4 +1528 @@ int rdbLoadRio(rio *rdb, rdbSaveInfo *rsi, int loading_aof) {
-        serverLog(LL_WARNING,
-                  "The RDB file contains AUX module data I can't load: no "
-                  "matching module '%s'",
-                  name);
+                serverLog(LL_WARNING,"The RDB file contains AUX module data I can't load: no matching module '%s'", name);
@@ -1531,4 +1532 @@ int rdbLoadRio(rio *rdb, rdbSaveInfo *rsi, int loading_aof) {
-          serverLog(LL_WARNING,
-                    "The RDB file contains module AUX data, but the module "
-                    "'%s' doesn't seem to support it.",
-                    name);
+                    serverLog(LL_WARNING,"The RDB file contains module AUX data, but the module '%s' doesn't seem to support it.", name);
@@ -1542,5 +1540 @@ int rdbLoadRio(rio *rdb, rdbSaveInfo *rsi, int loading_aof) {
-          serverLog(LL_WARNING,
-                    "The RDB file contains module AUX data for the module type "
-                    "'%s', that the responsible module is not able to load. "
-                    "Check for modules log above for additional clues.",
-                    name);
+                    serverLog(LL_WARNING,"The RDB file contains module AUX data for the module type '%s', that the responsible module is not able to load. Check for modules log above for additional clues.", name);
@@ -1555,5 +1549 @@ int rdbLoadRio(rio *rdb, rdbSaveInfo *rsi, int loading_aof) {
-          serverLog(
-              LL_WARNING,
-              "The RDB file contains module AUX data for the module '%s' that "
-              "is not terminated by the proper module value EOF marker",
-              name);
+                    serverLog(LL_WARNING,"The RDB file contains module AUX data for the module '%s' that is not terminated by the proper module value EOF marker", name);
@@ -1712,25 +1700,0 @@ int rdbSaveToSlavesSockets(rdbSaveInfo *rsi) {
-<<<<<<< HEAD
-      size_t private_dirty = zmalloc_get_private_dirty(-1);
-      if (private_dirty) {
-        serverLog(LL_NOTICE, "RDB: %zu MB of memory used by copy-on-write",
-                  private_dirty / (1024 * 1024));
-      }
-      server.child_info_data.cow_size = private_dirty;
-      sendChildInfo(CHILD_INFO_TYPE_RDB);
-|||||||
-      size_t private_dirty = zmalloc_get_private_dirty(-1);
-      if (private_dirty) {
-        serverLog(LL_NOTICE, "RDB: %zu MB of memory used by copy-on-write",
-                  private_dirty / (1024 * 1024));
-      }
-      server.child_info_data.cow_size = private_dirty;
-      sendChildInfo(CHILD_INFO_TYPE_RDB);
-      void *msg = zmalloc(sizeof(uint64_t) * (1 + 2 * numfds));
-      uint64_t *len = msg;
-      uint64_t *ids = len + 1;
-      int j, msglen;
-      *len = numfds;
-      for (j = 0; j < numfds; j++) {
-        *ids++ = clientids[j];
-        *ids++ = slave_sockets.io.fdset.state[j];
-=======
@@ -1738,9 +1701,0 @@ int rdbSaveToSlavesSockets(rdbSaveInfo *rsi) {
-      void *msg = zmalloc(sizeof(uint64_t) * (1 + 2 * numfds));
-      uint64_t *len = msg;
-      uint64_t *ids = len + 1;
-      int j, msglen;
-      *len = numfds;
-      for (j = 0; j < numfds; j++) {
-        *ids++ = clientids[j];
-        *ids++ = slave_sockets.io.fdset.state[j];
->>>>>>> b8e02f2b4005febbdaa11ff978c4f98b664464c9
@@ -1775,8 +1730,3 @@ int rdbSaveToSlavesSockets(rdbSaveInfo *rsi) {
-<<<<<<< HEAD
-      close(
-          server
-              .rdb_pipe_write);
-      if (aeCreateFileEvent(server.el, server.rdb_pipe_read, AE_READABLE,
-                            rdbPipeReadHandler, NULL) == AE_ERR) {
-        serverPanic(
-            "Unrecoverable error creating server.rdb_pipe_read file event.");
+            close(server.rdb_pipe_write);
+            if (aeCreateFileEvent(server.el, server.rdb_pipe_read, AE_READABLE, rdbPipeReadHandler,NULL) == AE_ERR) {
+                serverPanic("Unrecoverable error creating server.rdb_pipe_read file event.");
@@ -1784,5 +1733,0 @@ int rdbSaveToSlavesSockets(rdbSaveInfo *rsi) {
-      updateDictResizePolicy();
-|||||||
-        updateDictResizePolicy();
-=======
->>>>>>> b8e02f2b4005febbdaa11ff978c4f98b664464c9
