--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/c469f6ad-b8e02f2b-f4d37173/src@cluster.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/c469f6ad-b8e02f2b-f4d37173/src@cluster.no_comments_merged.c
@@ -721,2 +737 @@ void clearNodeFailureIfNeeded(clusterNode *node) {
-              "Clear FAIL state for node %.40s: is reachable again and nobody "
-              "is serving its slots after some time.",
+            "Clear FAIL state for node %.40s: is reachable again and nobody is serving its slots after some time.",
@@ -1044,5 +1090,6 @@ int clusterProcessPacket(clusterLink *link) {
-      } else if (memcmp(link->node->name, hdr->sender, CLUSTER_NAMELEN) != 0) {
-        serverLog(LL_DEBUG,
-                  "PONG contains mismatching sender ID. About node %.40s added "
-                  "%d ms ago, having flags %d",
-                  link->node->name, (int)(mstime() - (link->node->ctime)),
+            } else if (memcmp(link->node->name,hdr->sender,
+                        CLUSTER_NAMELEN) != 0)
+            {
+                serverLog(LL_DEBUG,"PONG contains mismatching sender ID. About node %.40s added %d ms ago, having flags %d",
+                    link->node->name,
+                    (int)(mstime()-(link->node->ctime)),
@@ -2403,2 +2495 @@ void clusterCommand(client *c) {
-        "COUNT-failure-reports <node-id> -- Return number of failure reports "
-        "for <node-id>.",
+"COUNT-failure-reports <node-id> -- Return number of failure reports for <node-id>.",
@@ -2406,4 +2497,2 @@ void clusterCommand(client *c) {
-        "DELSLOTS <slot> [slot ...] -- Delete slots information from current "
-        "node.",
-        "FAILOVER [force|takeover] -- Promote current replica node to being a "
-        "master.",
+"DELSLOTS <slot> [slot ...] -- Delete slots information from current node.",
+"FAILOVER [force|takeover] -- Promote current replica node to being a master.",
@@ -2411,2 +2500 @@ void clusterCommand(client *c) {
-        "GETKEYSINSLOT <slot> <count> -- Return key names stored by current "
-        "node in a slot.",
+"GETKEYSINSLOT <slot> <count> -- Return key names stored by current node in a slot.",
@@ -2419,4 +2507,2 @@ void clusterCommand(client *c) {
-        "    <id> <ip:port> <flags> <master> <pings> <pongs> <epoch> <link> "
-        "<slot> ... <slot>",
-        "REPLICATE <node-id> -- Configure current node as replica to "
-        "<node-id>.",
+"    <id> <ip:port> <flags> <master> <pings> <pongs> <epoch> <link> <slot> ... <slot>",
+"REPLICATE <node-id> -- Configure current node as replica to <node-id>.",
@@ -2425,2 +2511 @@ void clusterCommand(client *c) {
-        "SETSLOT <slot> (importing|migrating|stable|node <node-id>) -- Set "
-        "slot state.",
+"SETSLOT <slot> (importing|migrating|stable|node <node-id>) -- Set slot state.",
@@ -2428,2 +2513 @@ void clusterCommand(client *c) {
-        "SLOTS -- Return information about slots range mappings. Each range is "
-        "made of:",
+"SLOTS -- Return information about slots range mappings. Each range is made of:",
@@ -2574,2 +2664 @@ void clusterCommand(client *c) {
-                    "Invalid CLUSTER SETSLOT action or number of arguments. "
-                    "Try CLUSTER HELP");
+                "Invalid CLUSTER SETSLOT action or number of arguments. Try CLUSTER HELP");
