diff --git a/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@cluster.no_comments_mergegen.c b/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@cluster.no_comments_merged.c
index 8f29852..ff146c7 100644
--- a/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@cluster.no_comments_mergegen.c
+++ b/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@cluster.no_comments_merged.c
@@ -1312 +1312,2 @@ void clusterLinkConnectHandler(connection *conn) {
-void clusterReadHandler(connection *conn) { clusterMsg buf[1];
+void clusterReadHandler(connection *conn) {
+    clusterMsg buf[1];
