--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server-client.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server-client.no_comments_merged.c
@@ -6,2 +5,0 @@
-#include <imsg.h>
-#include <paths.h>
@@ -743,0 +889,4 @@ voidserver_client_dispatch_identify(struct client *c, struct imsg *imsg) {
+#ifdef __CYGWIN__
+ c->fd = open(c->ttyname, O_RDWR|O_NOCTTY);
+ c->cwd = open(".", O_RDONLY);
+#endif
