diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server.no_comments_merged.c
index 188ee88..3c22c45 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server.no_comments_merged.c
@@ -10 +9,0 @@
-#include <paths.h>
@@ -38,2 +37,3 @@ void server_child_stopped(pid_t, int);
-void server_set_marked(struct session *s, struct winlink *wl,
-                       struct window_pane *wp) {
+void
+server_set_marked(struct session *s, struct winlink *wl, struct window_pane *wp)
+{
@@ -46 +46,3 @@ void server_set_marked(struct session *s, struct winlink *wl,
-void server_clear_marked(void) {
+void
+server_clear_marked(void)
+{
@@ -53,5 +55,9 @@ void server_clear_marked(void) {
-int server_is_marked(struct session *s, struct winlink *wl,
-                     struct window_pane *wp) {
-  if (s == NULL || wl == NULL || wp == NULL) return (0);
-  if (marked_session != s || marked_winlink != wl) return (0);
-  if (marked_window_pane != wp) return (0);
+int
+server_is_marked(struct session *s, struct winlink *wl, struct window_pane *wp)
+{
+ if (s == NULL || wl == NULL || wp == NULL)
+  return (0);
+ if (marked_session != s || marked_winlink != wl)
+  return (0);
+ if (marked_window_pane != wp)
+  return (0);
@@ -60 +66,3 @@ int server_is_marked(struct session *s, struct winlink *wl,
-int server_check_marked(void) {
+int
+server_check_marked(void)
+{
@@ -62,3 +70,6 @@ int server_check_marked(void) {
-  if (marked_window_pane == NULL) return (0);
-  if (marked_layout_cell != marked_window_pane->layout_cell) return (0);
-  if (!session_alive(marked_session)) return (0);
+ if (marked_window_pane == NULL)
+  return (0);
+ if (marked_layout_cell != marked_window_pane->layout_cell)
+  return (0);
+ if (!session_alive(marked_session))
+  return (0);
@@ -66 +77,2 @@ int server_check_marked(void) {
-    if (wl->window == marked_window && wl == marked_winlink) break;
+  if (wl->window == marked_window && wl == marked_winlink)
+   break;
@@ -68,2 +80,4 @@ int server_check_marked(void) {
-  if (wl == NULL) return (0);
-  if (!window_has_pane(marked_window, marked_window_pane)) return (0);
+ if (wl == NULL)
+  return (0);
+ if (!window_has_pane(marked_window, marked_window_pane))
+  return (0);
@@ -72 +86,3 @@ int server_check_marked(void) {
-int server_create_socket(void) {
+int
+server_create_socket(void)
+{
@@ -85 +101,2 @@ int server_create_socket(void) {
-  if ((fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1) return (-1);
+ if ((fd = socket(AF_UNIX, SOCK_STREAM, 0)) == -1)
+  return (-1);
@@ -87 +104,2 @@ int server_create_socket(void) {
-  if (bind(fd, (struct sockaddr *)&sa, sizeof(sa)) == -1) return (-1);
+ if (bind(fd, (struct sockaddr *) &sa, sizeof(sa)) == -1)
+  return (-1);
@@ -89 +107,2 @@ int server_create_socket(void) {
-  if (listen(fd, 16) == -1) return (-1);
+ if (listen(fd, 16) == -1)
+  return (-1);
@@ -93 +112,3 @@ int server_create_socket(void) {
-int server_start(struct event_base *base, int lockfd, char *lockfile) {
+int
+server_start(struct event_base *base, int lockfd, char *lockfile)
+{
@@ -105,2 +126 @@ int server_start(struct event_base *base, int lockfd, char *lockfile) {
-             "ps",
-             NULL) != 0)
+     "ps", NULL) != 0)
@@ -118,9 +137,0 @@ int server_start(struct event_base *base, int lockfd, char *lockfile) {
-<<<<<<< HEAD
-  log_debug("socket path %s", socket_path);
-#ifdef HAVE_SETPROCTITLE
-  setproctitle("server (%s)", socket_path);
-#endif
-  ||||||| log_debug("socket path %s", socket_path);
-  setproctitle("server (%s)", socket_path);
-=======
->>>>>>> 44657bf932b068aff5ce1019a4e8a2e7b00b5321
@@ -128 +139,2 @@ int server_start(struct event_base *base, int lockfd, char *lockfile) {
-  if (server_fd == -1) fatal("couldn't create socket");
+ if (server_fd == -1)
+  fatal("couldn't create socket");
@@ -141 +153,3 @@ int server_start(struct event_base *base, int lockfd, char *lockfile) {
-int server_loop(void) {
+int
+server_loop(void)
+{
@@ -145 +159,2 @@ int server_loop(void) {
-    if (!RB_EMPTY(&sessions)) return (0);
+  if (!RB_EMPTY(&sessions))
+   return (0);
@@ -148 +163,2 @@ int server_loop(void) {
-    if (c->session != NULL) return (0);
+  if (c->session != NULL)
+   return (0);
@@ -151 +167,2 @@ int server_loop(void) {
-  if (!TAILQ_EMPTY(&clients)) return (0);
+ if (!TAILQ_EMPTY(&clients))
+  return (0);
@@ -154 +171,3 @@ int server_loop(void) {
-void server_send_exit(void) {
+void
+server_send_exit(void)
+{
@@ -168 +187,3 @@ void server_send_exit(void) {
-void server_update_socket(void) {
+void
+server_update_socket(void)
+{
@@ -182 +203,2 @@ void server_update_socket(void) {
-    if (stat(socket_path, &sb) != 0) return;
+  if (stat(socket_path, &sb) != 0)
+   return;
@@ -185,3 +207,6 @@ void server_update_socket(void) {
-      if (mode & S_IRUSR) mode |= S_IXUSR;
-      if (mode & S_IRGRP) mode |= S_IXGRP;
-      if (mode & S_IROTH) mode |= S_IXOTH;
+   if (mode & S_IRUSR)
+    mode |= S_IXUSR;
+   if (mode & S_IRGRP)
+    mode |= S_IXGRP;
+   if (mode & S_IROTH)
+    mode |= S_IXOTH;
@@ -193 +218,3 @@ void server_update_socket(void) {
-void server_accept(int fd, short events, unused void *data) {
+void
+server_accept(int fd, short events, unused void *data)
+{
@@ -198 +225,2 @@ void server_accept(int fd, short events, unused void *data) {
-  if (!(events & EV_READ)) return;
+ if (!(events & EV_READ))
+  return;
@@ -201 +229,2 @@ void server_accept(int fd, short events, unused void *data) {
-    if (errno == EAGAIN || errno == EINTR || errno == ECONNABORTED) return;
+  if (errno == EAGAIN || errno == EINTR || errno == ECONNABORTED)
+   return;
@@ -214 +243,3 @@ void server_accept(int fd, short events, unused void *data) {
-void server_add_accept(int timeout) {
+void
+server_add_accept(int timeout)
+{
@@ -216 +247,2 @@ void server_add_accept(int timeout) {
-  if (event_initialized(&server_ev_accept)) event_del(&server_ev_accept);
+ if (event_initialized(&server_ev_accept))
+  event_del(&server_ev_accept);
@@ -218 +250,2 @@ void server_add_accept(int timeout) {
-    event_set(&server_ev_accept, server_fd, EV_READ, server_accept, NULL);
+  event_set(&server_ev_accept, server_fd, EV_READ, server_accept,
+      NULL);
@@ -221 +254,2 @@ void server_add_accept(int timeout) {
-    event_set(&server_ev_accept, server_fd, EV_TIMEOUT, server_accept, NULL);
+  event_set(&server_ev_accept, server_fd, EV_TIMEOUT,
+      server_accept, NULL);
@@ -225 +259,3 @@ void server_add_accept(int timeout) {
-voidserver_signal(int sig) {
+void
+server_signal(int sig)
+{
@@ -247 +283,3 @@ voidserver_signal(int sig) {
-void server_child_signal(void) {
+void
+server_child_signal(void)
+{
@@ -253 +291,2 @@ void server_child_signal(void) {
-        if (errno == ECHILD) return;
+   if (errno == ECHILD)
+    return;
@@ -264 +303,3 @@ void server_child_signal(void) {
-void server_child_exited(pid_t pid, int status) {
+void
+server_child_exited(pid_t pid, int status)
+{
@@ -284 +325,3 @@ void server_child_exited(pid_t pid, int status) {
-void server_child_stopped(pid_t pid, int status) {
+void
+server_child_stopped(pid_t pid, int status)
+{
@@ -287 +330,2 @@ void server_child_stopped(pid_t pid, int status) {
-  if (WSTOPSIG(status) == SIGTTIN || WSTOPSIG(status) == SIGTTOU) return;
+ if (WSTOPSIG(status) == SIGTTIN || WSTOPSIG(status) == SIGTTOU)
+  return;
@@ -291 +335,2 @@ void server_child_stopped(pid_t pid, int status) {
-        if (killpg(pid, SIGCONT) != 0) kill(pid, SIGCONT);
+    if (killpg(pid, SIGCONT) != 0)
+     kill(pid, SIGCONT);
