diff --git a/./output/duckdb/4776329b-f8ecea77-92cf6f1d/src@optimizer@optimizer.no_comments_conflict.cpp b/./output/duckdb/4776329b-f8ecea77-92cf6f1d/src@optimizer@optimizer.no_comments_merged.cpp
index b7607b7..bf4ce42 100644
--- a/./output/duckdb/4776329b-f8ecea77-92cf6f1d/src@optimizer@optimizer.no_comments_conflict.cpp
+++ b/./output/duckdb/4776329b-f8ecea77-92cf6f1d/src@optimizer@optimizer.no_comments_merged.cpp
@@ -11,0 +12 @@
+#include "duckdb/optimizer/empty_result_pullup.hpp"
@@ -15 +15,0 @@
-#include "duckdb/optimizer/empty_result_pullup.hpp"
@@ -26,0 +27 @@
+#include "duckdb/optimizer/sampling_pushdown.hpp"
@@ -30,7 +30,0 @@
-<<<<<<< HEAD
-||||||| 92cf6f1dd7
-#include "duckdb/optimizer/join_filter_pushdown_optimizer.hpp"
-=======
-#include "duckdb/optimizer/sampling_pushdown.hpp"
-#include "duckdb/optimizer/join_filter_pushdown_optimizer.hpp"
->>>>>>> f8ecea77
