diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_merged.h
index 78b7f1f..e5f8d39 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_merged.h
@@ -5,4 +5,2 @@
-class FastbootDevice
-        bool
-        GetVersion(FastbootDevice* device, const std::vector<std::string>& args,
-                   std::string* message);
+class FastbootDevice;
+bool GetVersion(FastbootDevice* device, const std::vector<std::string>& args, std::string* message);
