--- a/./output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_conflict.h
+++ b/./output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_merged.h
@@ -37 +36,0 @@ bool GetHardwareRevision(FastbootDevice* device, const std::vector<std::string>&
-<<<<<<< HEAD
@@ -41,4 +39,0 @@ bool GetOffModeChargeState(FastbootDevice* device, const std::vector<std::string
-||||||| e80c90f0f
-=======
-bool GetVariant(FastbootDevice* device, const std::vector<std::string>& args, std::string* message);
->>>>>>> e97756b4
