diff --git a/./output/system_core/9abdef0c-ef52ae14-79af3c6d/init@util.no_comments_mergegen.h b/./output/system_core/9abdef0c-ef52ae14-79af3c6d/init@util.no_comments_merged.h
index b1445b7..83c1705 100644
--- a/./output/system_core/9abdef0c-ef52ae14-79af3c6d/init@util.no_comments_mergegen.h
+++ b/./output/system_core/9abdef0c-ef52ae14-79af3c6d/init@util.no_comments_merged.h
@@ -15 +15,2 @@ int create_socket(const char *name, int type, mode_t perm,
-bool read_file(const std::string& path, std::string* content); bool write_file(const std::string& path, const std::string& content);
+bool read_file(const std::string& path, std::string* content);
+bool write_file(const std::string& path, const std::string& content);
