--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -7 +7 @@
-struct android_namespace_t
+struct android_namespace_t;
@@ -11,4 +10,0 @@ struct android_namespace_t
-    private:
-        bool linkDriverNamespaceLocked(android_namespace_t* destNamespace,
-                                       android_namespace_t* vndkNamespace,
-                                       const std::string& sharedSphalLibraries);
@@ -37 +31,0 @@ struct android_namespace_t
-<<<<<<< HEAD
@@ -39,9 +33 @@ struct android_namespace_t
-                          const std::string& packageName,
-                          const std::vector<std::string> eglFeatures);
-|||||||
-        void setAngleInfo(const std::string path, const std::string appName, std::string devOptIn,
-                          const std::vector<std::string> eglFeatures);
-=======
-        void setAngleInfo(const std::string& path, const std::string& packageName,
-                          const std::string& devOptIn, const std::vector<std::string> eglFeatures);
->>>>>>> e20d85480a8bf5f62954cede949426ae5cdc6420
+                      const std::string& packageName, const std::vector<std::string> eglFeatures);
@@ -62 +48,3 @@ struct android_namespace_t
-        void updateShouldUseAngle();
+    bool linkDriverNamespaceLocked(android_namespace_t* destNamespace,
+                                   android_namespace_t* vndkNamespace,
+                                   const std::string& sharedSphalLibraries);
@@ -64,3 +52,2 @@ struct android_namespace_t
-        void sendGpuStatsLocked(GpuStatsInfo::Api api, bool isDriverLoaded,
-                                int64_t driverLoadingTime);
-        GraphicsEnv()
+    void sendGpuStatsLocked(GpuStatsInfo::Api api, bool isDriverLoaded, int64_t driverLoadingTime);
+    GraphicsEnv() = default;
@@ -74 +61 @@ struct android_namespace_t
-        UseAngle mShouldUseAngle = UNKNOWN;
+    bool mShouldUseAngle = false;
