diff --git a/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_conflict.h b/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
index d52929b..df48b44 100644
--- a/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_conflict.h
+++ b/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -32 +31,0 @@ public:
-<<<<<<< HEAD
@@ -35,7 +33,0 @@ public:
-||||||| e631a0223f
-    void setAngleInfo(const std::string path, const std::string appName, std::string devOptIn,
-                      const std::vector<std::string> eglFeatures);
-=======
-    void setAngleInfo(const std::string& path, const std::string& packageName,
-                      const std::string& devOptIn, const std::vector<std::string> eglFeatures);
->>>>>>> e20d8548
@@ -45 +36,0 @@ public:
-<<<<<<< HEAD
@@ -49,4 +39,0 @@ public:
-||||||| e631a0223f
-=======
-    void nativeToggleAngleAsSystemDriver(bool enabled);
->>>>>>> e20d8548
@@ -61,9 +47,0 @@ private:
-<<<<<<< HEAD
-||||||| e631a0223f
-    enum UseAngle { UNKNOWN, YES, NO };
-    void* loadLibrary(std::string name);
-    void updateUseAngle();
-=======
-    enum UseAngle { UNKNOWN, YES, NO };
-    void updateShouldUseAngle();
->>>>>>> e20d8548
@@ -79 +56,0 @@ private:
-<<<<<<< HEAD
@@ -88,10 +64,0 @@ private:
-||||||| e631a0223f
-=======
-    android_namespace_t* mDriverNamespace = nullptr;
-    std::string mAnglePath;
-    std::string mPackageName;
-    std::string mAngleDeveloperOptIn;
-    std::vector<std::string> mAngleEglFeatures;
-    UseAngle mShouldUseAngle = UNKNOWN;
-    android_namespace_t* mAngleNamespace = nullptr;
->>>>>>> e20d8548
