--- a/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@GraphicsEnv.no_comments_conflict.cpp
+++ b/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
@@ -120 +119,0 @@ bool GraphicsEnv::isDebuggable() {
-<<<<<<< HEAD
@@ -123,7 +121,0 @@ void GraphicsEnv::setDriverPathAndSphalLibraries(const std::string& path,
-||||||| e631a0223f
-void GraphicsEnv::setDriverPathAndSphalLibraries(const std::string path,
-                                                 const std::string sphalLibraries) {
-=======
-void GraphicsEnv::setDriverPathAndSphalLibraries(const std::string& path,
-                                                 const std::string& sphalLibraries) {
->>>>>>> e20d8548
@@ -403,27 +394,0 @@ bool GraphicsEnv::setInjectLayersPrSetDumpable() {
-<<<<<<< HEAD
-||||||| e631a0223f
-void* GraphicsEnv::loadLibrary(std::string name) {
-    const android_dlextinfo dlextinfo = {
-            .flags = ANDROID_DLEXT_USE_NAMESPACE,
-            .library_namespace = getAngleNamespace(),
-    };
-    std::string libName = std::string("lib") + name + "_angle.so";
-    void* so = android_dlopen_ext(libName.c_str(), RTLD_LOCAL | RTLD_NOW, &dlextinfo);
-    if (so) {
-        ALOGD("dlopen_ext from APK (%s) success at %p", libName.c_str(), so);
-        return so;
-    } else {
-        ALOGE("dlopen_ext(\"%s\") failed: %s", libName.c_str(), dlerror());
-    }
-    return nullptr;
-}
-bool GraphicsEnv::shouldUseAngle(std::string appName) {
-    if (appName != mAngleAppName) {
-        ALOGE("App name does not match: expected '%s', got '%s'", mAngleAppName.c_str(),
-              appName.c_str());
-        return false;
-    }
-    return shouldUseAngle();
-}
-=======
->>>>>>> e20d8548
@@ -435 +399,0 @@ bool GraphicsEnv::shouldUseAngle() {
-<<<<<<< HEAD
@@ -437,5 +400,0 @@ bool GraphicsEnv::shouldUseAngle() {
-||||||| e631a0223f
-    return (mUseAngle == YES) ? true : false;
-=======
-    return (mShouldUseAngle == YES) ? true : false;
->>>>>>> e20d8548
@@ -443 +401,0 @@ bool GraphicsEnv::shouldUseAngle() {
-<<<<<<< HEAD
@@ -446,33 +403,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const bool shouldUseNati
-||||||| e631a0223f
-void GraphicsEnv::updateUseAngle() {
-    const char* ANGLE_PREFER_ANGLE = "angle";
-    const char* ANGLE_PREFER_NATIVE = "native";
-    mUseAngle = NO;
-    if (mAngleDeveloperOptIn == ANGLE_PREFER_ANGLE) {
-        ALOGV("User set \"Developer Options\" to force the use of ANGLE");
-        mUseAngle = YES;
-    } else if (mAngleDeveloperOptIn == ANGLE_PREFER_NATIVE) {
-        ALOGV("User set \"Developer Options\" to force the use of Native");
-    } else {
-        ALOGV("User set invalid \"Developer Options\": '%s'", mAngleDeveloperOptIn.c_str());
-    }
-}
-void GraphicsEnv::setAngleInfo(const std::string path, const std::string appName,
-                               const std::string developerOptIn,
-=======
-void GraphicsEnv::updateShouldUseAngle() {
-    const char* ANGLE_PREFER_ANGLE = "angle";
-    const char* ANGLE_PREFER_NATIVE = "native";
-    mShouldUseAngle = NO;
-    if (mAngleDeveloperOptIn == ANGLE_PREFER_ANGLE) {
-        ALOGV("User set \"Developer Options\" to force the use of ANGLE");
-        mShouldUseAngle = YES;
-    } else if (mAngleDeveloperOptIn == ANGLE_PREFER_NATIVE) {
-        ALOGV("User set \"Developer Options\" to force the use of Native");
-    } else {
-        ALOGV("User set invalid \"Developer Options\": '%s'", mAngleDeveloperOptIn.c_str());
-    }
-}
-void GraphicsEnv::setAngleInfo(const std::string& path, const std::string& packageName,
-                               const std::string& developerOptIn,
->>>>>>> e20d8548
@@ -480 +404,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const std::string& packa
-<<<<<<< HEAD
@@ -483,9 +406,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const std::string& packa
-||||||| e631a0223f
-    if (mUseAngle != UNKNOWN) {
-        ALOGV("Already evaluated the rules file for '%s': use ANGLE = %s", appName.c_str(),
-              (mUseAngle == YES) ? "true" : "false");
-=======
-    if (mShouldUseAngle != UNKNOWN) {
-        ALOGV("Already evaluated the rules file for '%s': use ANGLE = %s", packageName.c_str(),
-              (mShouldUseAngle == YES) ? "true" : "false");
->>>>>>> e20d8548
@@ -496 +410,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const std::string& packa
-<<<<<<< HEAD
@@ -507,15 +420,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const std::string& packa
-||||||| e631a0223f
-    mAnglePath = path;
-    ALOGV("setting ANGLE app name to '%s'", appName.c_str());
-    mAngleAppName = appName;
-    ALOGV("setting ANGLE application opt-in to '%s'", developerOptIn.c_str());
-    mAngleDeveloperOptIn = developerOptIn;
-    updateUseAngle();
-=======
-    mAnglePath = path;
-    ALOGV("setting app package name to '%s'", packageName.c_str());
-    mPackageName = packageName;
-    ALOGV("setting ANGLE application opt-in to '%s'", developerOptIn.c_str());
-    mAngleDeveloperOptIn = developerOptIn;
-    updateShouldUseAngle();
->>>>>>> e20d8548
@@ -523 +421,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const std::string& packa
-<<<<<<< HEAD
@@ -585,6 +482,0 @@ void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace,
-||||||| e631a0223f
-void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace, const std::string layerPaths) {
-=======
-void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace,
-                                const std::string& layerPaths) {
->>>>>>> e20d8548
@@ -602,16 +493,0 @@ NativeLoaderNamespace* GraphicsEnv::getAppNamespace() {
-<<<<<<< HEAD
-||||||| e631a0223f
-std::string& GraphicsEnv::getAngleAppName() {
-    return mAngleAppName;
-}
-const std::vector<std::string>& GraphicsEnv::getAngleEglFeatures() {
-    return mAngleEglFeatures;
-}
-=======
-std::string& GraphicsEnv::getPackageName() {
-    return mPackageName;
-}
-const std::vector<std::string>& GraphicsEnv::getAngleEglFeatures() {
-    return mAngleEglFeatures;
-}
->>>>>>> e20d8548
