diff --git a/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergegen.h b/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
index 09ad53f..df48b44 100644
--- a/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergegen.h
+++ b/./output/frameworks_native/56d743ea-e20d8548-e631a022/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -31,0 +32,2 @@ public:
+    void setAngleInfo(const std::string& path, const bool shouldUseNativeDriver,
+                      const std::string& packageName, const std::vector<std::string> eglFeatures);
@@ -34,0 +37,3 @@ public:
+    void nativeToggleAngleAsSystemDriver(bool enabled);
+    bool shouldUseSystemAngle();
+    bool shouldUseNativeDriver();
@@ -51,0 +57,8 @@ private:
+    android_namespace_t* mDriverNamespace = nullptr;
+    std::string mAnglePath;
+    std::string mPackageName;
+    std::vector<std::string> mAngleEglFeatures;
+    bool mShouldUseAngle = false;
+    bool mShouldUseSystemAngle = false;
+    bool mShouldUseNativeDriver = false;
+    android_namespace_t* mAngleNamespace = nullptr;
