--- a/./output/frameworks_av/578a017a-eda0965e-b0b34366/services@audioflinger@Effects.no_comments_conflict.cpp
+++ b/./output/frameworks_av/578a017a-eda0965e-b0b34366/services@audioflinger@Effects.no_comments_merged.cpp
@@ -1154 +1153,0 @@ status_t EffectModule::setVolumeInternal(
-<<<<<<< HEAD
@@ -1156,4 +1154,0 @@ status_t EffectModule::setVolumeInternal(
-||||||| b0b3436619
-        mVolume = {*left, *right};
-=======
->>>>>>> eda0965e
