--- a/home/whalien/codebase/python/mergebot-eva/output/redis/eaeba1b2-39c70e72-cf76af6b/src@config.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/eaeba1b2-39c70e72-cf76af6b/src@config.no_comments_merged.c
@@ -1 +1 @@
-#include "redis.h"
+#include "server.h"
@@ -5 +4,0 @@
-#include "server.h"
@@ -223,3 +220 @@ void loadServerConfigFromString(char *config) {
-        err =
-            "Invalid log facility. Must be one of USER or between "
-            "LOCAL0-LOCAL7";
+                err = "Invalid log facility. Must be one of USER or between LOCAL0-LOCAL7";
@@ -741,3 +718,24 @@ void loadServerConfig(char *filename, char *options) {
-#define config_set_else \
-  } \
-  else
+#define config_set_bool_field(_name,_var) \
+    } else if (!strcasecmp(c->argv[2]->ptr,_name)) { \
+        int yn = yesnotoi(o->ptr); \
+        if (yn == -1) goto badfmt; \
+        _var = yn;
+#define config_set_numerical_field(_name,_var,min,max) \
+    } else if (!strcasecmp(c->argv[2]->ptr,_name)) { \
+        if (getLongLongFromObject(o,&ll) == C_ERR) goto badfmt; \
+        if (min != LLONG_MIN && ll < min) goto badfmt; \
+        if (max != LLONG_MAX && ll > max) goto badfmt; \
+        _var = ll;
+#define config_set_memory_field(_name,_var) \
+    } else if (!strcasecmp(c->argv[2]->ptr,_name)) { \
+        ll = memtoll(o->ptr,&err); \
+        if (err || ll < 0) goto badfmt; \
+        _var = ll;
+#define config_set_enum_field(_name,_var,_enumvar) \
+    } else if (!strcasecmp(c->argv[2]->ptr,_name)) { \
+        int enumval = configEnumGetValue(_enumvar,o->ptr); \
+        if (enumval == INT_MIN) goto badfmt; \
+        _var = enumval;
+#define config_set_special_field(_name) \
+    } else if (!strcasecmp(c->argv[2]->ptr,_name)) {
+#define config_set_else } else
@@ -768,21 +764,3 @@ void configSetCommand(client *c) {
-<<<<<<< HEAD
-    }
-    else if (!strcasecmp(c->argv[2]->ptr, "maxmemory")) {
-      if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll < 0) goto badfmt;
-      server.maxmemory = ll;
-      if (server.maxmemory) {
-        if (server.maxmemory < zmalloc_used_memory()) {
-          redisLog(REDIS_WARNING,
-                   "WARNING: the new maxmemory value set via CONFIG SET is "
-                   "smaller than the current memory usage. This will result in "
-                   "key eviction and/or the inability to accept new write "
-                   "commands depending on the maxmemory-policy.");
-        }
-        freeMemoryIfNeeded();
-      }
-    }
-    else if (!strcasecmp(c->argv[2]->ptr, "maxclients")) {
-|||||||
-    }
-    else if (!strcasecmp(c->argv[2]->ptr, "maxmemory")) {
-      if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll < 0) goto badfmt;
+    } else if (!strcasecmp(c->argv[2]->ptr,"maxmemory")) {
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
+            ll < 0) goto badfmt;
@@ -792,5 +770 @@ void configSetCommand(client *c) {
-          redisLog(REDIS_WARNING,
-                   "WARNING: the new maxmemory value set via CONFIG SET is "
-                   "smaller than the current memory usage. This will result in "
-                   "keys eviction and/or inability to accept new write "
-                   "commands depending on the maxmemory-policy.");
+                redisLog(REDIS_WARNING,"WARNING: the new maxmemory value set via CONFIG SET is smaller than the current memory usage. This will result in key eviction and/or the inability to accept new write commands depending on the maxmemory-policy.");
@@ -800,5 +774,2 @@ void configSetCommand(client *c) {
-    }
-    else if (!strcasecmp(c->argv[2]->ptr, "maxclients")){
-=======
-    }
-    config_set_special_field("cluster-announce-ip") {
+    } else if (!strcasecmp(c->argv[2]->ptr,"maxclients")) {
+    } config_set_special_field("cluster-announce-ip") {
@@ -807,3 +778 @@ void configSetCommand(client *c) {
-    }
-    config_set_special_field("maxclients") {
->>>>>>> 39c70e728b5af0c50989ffbc05e568099f3e081b
+    } config_set_special_field("maxclients") {
@@ -816,4 +785 @@ void configSetCommand(client *c) {
-          addReplyErrorFormat(c,
-                              "The operating system is not able to handle the "
-                              "specified number of clients, try with %d",
-                              server.maxclients);
+                addReplyErrorFormat(c,"The operating system is not able to handle the specified number of clients, try with %d", server.maxclients);
@@ -824,6 +790,6 @@ void configSetCommand(client *c) {
-            server.maxclients + CONFIG_FDSET_INCR) {
-          if (aeResizeSetSize(
-                  server.el, server.maxclients + CONFIG_FDSET_INCR) == AE_ERR) {
-            addReplyError(c,
-                          "The event loop API used by Redis is not able to "
-                          "handle the specified number of clients");
+                server.maxclients + CONFIG_FDSET_INCR)
+            {
+                if (aeResizeSetSize(server.el,
+                    server.maxclients + CONFIG_FDSET_INCR) == AE_ERR)
+                {
+                    addReplyError(c,"The event loop API used by Redis is not able to handle the specified number of clients");
@@ -1073,5 +1035 @@ void configSetCommand(client *c) {
-          serverLog(LL_WARNING,
-                    "WARNING: the new maxmemory value set via CONFIG SET is "
-                    "smaller than the current memory usage. This will result "
-                    "in keys eviction and/or inability to accept new write "
-                    "commands depending on the maxmemory-policy.");
+                serverLog(LL_WARNING,"WARNING: the new maxmemory value set via CONFIG SET is smaller than the current memory usage. This will result in keys eviction and/or inability to accept new write commands depending on the maxmemory-policy.");
@@ -1106,0 +1065,29 @@ void configSetCommand(client *c) {
+#define config_get_string_field(_name,_var) do { \
+    if (stringmatch(pattern,_name,1)) { \
+        addReplyBulkCString(c,_name); \
+        addReplyBulkCString(c,_var ? _var : ""); \
+        matches++; \
+    } \
+} while(0);
+#define config_get_bool_field(_name,_var) do { \
+    if (stringmatch(pattern,_name,1)) { \
+        addReplyBulkCString(c,_name); \
+        addReplyBulkCString(c,_var ? "yes" : "no"); \
+        matches++; \
+    } \
+} while(0);
+#define config_get_numerical_field(_name,_var) do { \
+    if (stringmatch(pattern,_name,1)) { \
+        ll2string(buf,sizeof(buf),_var); \
+        addReplyBulkCString(c,_name); \
+        addReplyBulkCString(c,buf); \
+        matches++; \
+    } \
+} while(0);
+#define config_get_enum_field(_name,_var,_enumvar) do { \
+    if (stringmatch(pattern,_name,1)) { \
+        addReplyBulkCString(c,_name); \
+        addReplyBulkCString(c,configEnumGetNameOrUnknown(_enumvar,_var)); \
+        matches++; \
+    } \
+} while(0);
@@ -1919,2 +1748 @@ void configSetCommand(client *c) {
-          "GET <pattern> -- Return parameters matching the glob-like <pattern> "
-          "and their values.",
+"GET <pattern> -- Return parameters matching the glob-like <pattern> and their values.",
