--- a/home/whalien/codebase/python/mergebot-eva/output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_merged.c
@@ -238,23 +235,0 @@ void zaddGenericCommand(redisClient *c, robj *key, robj *ele, double score,
-<<<<<<< HEAD
-    de = dictFind(zs->dict, ele);
-    if (de) {
-      double *oldscore = dictGetEntryVal(de);
-      *score = *oldscore + scoreval;
-    } else {
-      *score = scoreval;
-    }
-    if (isnan(*score)) {
-      addReplyError(c, "resulting score is not a number (NaN)");
-      zfree(score);
-|||||||
-    de = dictFind(zs->dict, ele);
-    if (de) {
-      double *oldscore = dictGetEntryVal(de);
-      *score = *oldscore + scoreval;
-    } else {
-      *score = scoreval;
-    }
-    if (isnan(*score)) {
-      addReplySds(c, sdsnew("-ERR resulting score is not a number (NaN)\r\n"));
-      zfree(score);
-=======
@@ -264,2 +240 @@ void zaddGenericCommand(redisClient *c, robj *key, robj *ele, double score,
-      addReplySds(c, sdsnew("-ERR resulting score is not a number (NaN)\r\n"));
->>>>>>> 69ef89f2cf5a699d97475ff8e7c3ce714c6947cf
+            addReplyError(c,"resulting score is not a number (NaN)");
@@ -408,2 +376,2 @@ int qsortCompareZsetopsrcByCardinality(const void *s1, const void *s2) {
-inline static void zunionInterAggregate(double *target, double val,
-                                        int aggregate) {
+#define zunionInterDictValue(_e) (dictGetEntryVal(_e) == NULL ? 1.0 : *(double*)dictGetEntryVal(_e))
+inline static void zunionInterAggregate(double *target, double val, int aggregate) {
