diff --git a/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_mergegen.c b/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_merged.c
index 8a640d8..5f1b40c 100644
--- a/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_mergegen.c
+++ b/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_merged.c
@@ -236 +236,5 @@ void zaddGenericCommand(redisClient *c, robj *key, robj *ele, double score, int
-dictEntry *de = dictFind(zs->dict,ele); if (de != NULL) score += *(double*)dictGetEntryVal(de); if (isnan(score)) { addReplyError(c,"resulting score is not a number (NaN)");
+        dictEntry *de = dictFind(zs->dict,ele);
+        if (de != NULL)
+            score += *(double*)dictGetEntryVal(de);
+        if (isnan(score)) {
+            addReplyError(c,"resulting score is not a number (NaN)");
