--- a/./output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_mergegen.cpp
+++ b/./output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_merged.cpp
@@ -234 +234,2 @@ void ReadCSVTableFunction::ReadCSVAddNamedParameters(TableFunction &table_functi
-table_function.named_parameters["rfc_4180"] = LogicalType::BOOLEAN; table_function.named_parameters["encoding"] = LogicalType::VARCHAR;
+ table_function.named_parameters["encoding"] = LogicalType::VARCHAR;
+ table_function.named_parameters["rfc_4180"] = LogicalType::BOOLEAN;
