--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/683c18a9-564eb25e-8ce36237/src@storage@serialization@serialize_nodes.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/683c18a9-564eb25e-8ce36237/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
@@ -196,10 +183,2 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment",
-                                                      dialect_options.state_machine_options.comment,
-                                                      CSVOption<char>('\0'));
- serializer.WritePropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header",
-                                            dialect_options.rows_until_header);
-<<<<<<< HEAD
- serializer.WriteProperty<CSVOption<bool>>(139, "dialect_options.state_machine_options.rfc_4180",
-                                           dialect_options.state_machine_options.rfc_4180);
-|||||||
-=======
+ serializer.WritePropertyWithDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment", dialect_options.state_machine_options.comment, CSVOption<char>('\0'));
+ serializer.WritePropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header", dialect_options.rows_until_header);
@@ -207 +186 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
->>>>>>> 564eb25ef0ec700c8a289fe852c16629a19650bd
+ serializer.WriteProperty<CSVOption<bool>>(140, "dialect_options.state_machine_options.rfc_4180", dialect_options.state_machine_options.rfc_4180);
@@ -257,10 +227,2 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadPropertyWithExplicitDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment",
-                                                               result.dialect_options.state_machine_options.comment,
-                                                               CSVOption<char>('\0'));
- deserializer.ReadPropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header",
-                                             result.dialect_options.rows_until_header);
-<<<<<<< HEAD
- deserializer.ReadProperty<CSVOption<bool>>(139, "dialect_options.state_machine_options.rfc_4180",
-                                            result.dialect_options.state_machine_options.rfc_4180);
-|||||||
-=======
+ deserializer.ReadPropertyWithExplicitDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment", result.dialect_options.state_machine_options.comment, CSVOption<char>('\0'));
+ deserializer.ReadPropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header", result.dialect_options.rows_until_header);
@@ -268 +230 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
->>>>>>> 564eb25ef0ec700c8a289fe852c16629a19650bd
+ deserializer.ReadProperty<CSVOption<bool>>(140, "dialect_options.state_machine_options.rfc_4180", result.dialect_options.state_machine_options.rfc_4180);
