--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_merged.cpp
@@ -234,4 +233,0 @@ void ReadCSVTableFunction::ReadCSVAddNamedParameters(TableFunction &table_functi
-<<<<<<< HEAD
- table_function.named_parameters["rfc_4180"] = LogicalType::BOOLEAN;
-|||||||
-=======
@@ -239 +235 @@ void ReadCSVTableFunction::ReadCSVAddNamedParameters(TableFunction &table_functi
->>>>>>> 564eb25ef0ec700c8a289fe852c16629a19650bd
+ table_function.named_parameters["rfc_4180"] = LogicalType::BOOLEAN;
