diff --git a/./output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_conflict.cpp b/./output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_merged.cpp
index c891aa7..0ca647e 100644
--- a/./output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_conflict.cpp
+++ b/./output/duckdb/683c18a9-564eb25e-8ce36237/src@function@table@read_csv.no_comments_merged.cpp
@@ -234,4 +233,0 @@ void ReadCSVTableFunction::ReadCSVAddNamedParameters(TableFunction &table_functi
-<<<<<<< HEAD
- table_function.named_parameters["rfc_4180"] = LogicalType::BOOLEAN;
-||||||| 8ce3623758
-=======
@@ -239 +235 @@ void ReadCSVTableFunction::ReadCSVAddNamedParameters(TableFunction &table_functi
->>>>>>> 564eb25e
+ table_function.named_parameters["rfc_4180"] = LogicalType::BOOLEAN;
