--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_merged.h
@@ -145 +164,3 @@ class OdrConfig final {
-  OdrConfig() OdrConfig(const OdrConfig&) OdrConfig& operator=(const OdrConfig&) = delete
+  OdrConfig() = delete;
+  OdrConfig(const OdrConfig&) = delete;
+  OdrConfig& operator=(const OdrConfig&) = delete;
