--- a/./output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_conflict.h
+++ b/./output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_merged.h
@@ -28 +27,0 @@ class OdrConfig final {
-<<<<<<< HEAD
@@ -31,4 +29,0 @@ class OdrConfig final {
-||||||| c4be0fcab2
-=======
-  std::optional<bool> partial_compilation_;
->>>>>>> 2c4b444b
@@ -97 +91,0 @@ class OdrConfig final {
-<<<<<<< HEAD
@@ -118,10 +111,0 @@ class OdrConfig final {
-||||||| c4be0fcab2
-  const std::string& GetSystemServerClasspath() const { return system_server_classpath_; }
-  const std::string& GetUpdatableBcpPackagesFile() const { return updatable_bcp_packages_file_; }
-=======
-  bool GetPartialCompilation() const {
-    return partial_compilation_.value_or(true);
-  }
-  const std::string& GetSystemServerClasspath() const { return system_server_classpath_; }
-  const std::string& GetUpdatableBcpPackagesFile() const { return updatable_bcp_packages_file_; }
->>>>>>> 2c4b444b
@@ -137 +120,0 @@ class OdrConfig final {
-<<<<<<< HEAD
@@ -144,6 +126,0 @@ class OdrConfig final {
-||||||| c4be0fcab2
-=======
-  void SetPartialCompilation(bool value) {
-    partial_compilation_ = value;
-  }
->>>>>>> 2c4b444b
