--- a/./output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_mergegen.cc
+++ b/./output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_merged.cc
@@ -6765,0 +6766 @@ TEST_F(DBTest, TransactionLogIterator) {
+#ifndef NDEBUG
@@ -6808,0 +6810 @@ TEST_F(DBTest, TransactionLogIteratorRace) {
+#endif
@@ -6928,0 +6931,3 @@ class RecoveryTestHelper {
+#ifdef OS_WIN
+    test->Close();
+#endif
