diff --git a/./output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_mergegen.h b/./output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_merged.h
index d530faf..e5f8d39 100644
--- a/./output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_mergegen.h
+++ b/./output/system_core/f2edb84d-e97756b4-e80c90f0/fastboot@device@variables.no_comments_merged.h
@@ -37 +37,3 @@ bool GetHardwareRevision(FastbootDevice* device, const std::vector<std::string>&
-bool GetVariant(FastbootDevice* device, const std::vector<std::string>& args, std::string* message); bool GetOffModeChargeState(FastbootDevice* device, const std::vector<std::string>& args, std::string* message);
+bool GetVariant(FastbootDevice* device, const std::vector<std::string>& args, std::string* message);
+bool GetOffModeChargeState(FastbootDevice* device, const std::vector<std::string>& args,
+                           std::string* message);
