diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@include@media@AudioProfile.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@include@media@AudioProfile.no_comments_merged.h
index bb0033d..62885cf 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@include@media@AudioProfile.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@include@media@AudioProfile.no_comments_merged.h
@@ -10 +10,2 @@ namespace android {
-class AudioProfile final : public RefBase, public Parcelable {
+class AudioProfile final : public RefBase, public Parcelable
+{
@@ -12,5 +13,4 @@ class AudioProfile final : public RefBase, public Parcelable {
-  static sp<AudioProfile> createFullDynamic(
-      audio_format_t dynamicFormat = AUDIO_FORMAT_DEFAULT);
-  AudioProfile(audio_format_t format, audio_channel_mask_t channelMasks,
-               uint32_t samplingRate);
-  AudioProfile(audio_format_t format, const ChannelMaskSet &channelMasks,
+    static sp<AudioProfile> createFullDynamic(audio_format_t dynamicFormat = AUDIO_FORMAT_DEFAULT);
+    AudioProfile(audio_format_t format, audio_channel_mask_t channelMasks, uint32_t samplingRate);
+    AudioProfile(audio_format_t format,
+                 const ChannelMaskSet &channelMasks,
@@ -24,4 +24,3 @@ class AudioProfile final : public RefBase, public Parcelable {
-  bool isValid() const {
-    return hasValidFormat() && hasValidRates() && hasValidChannels();
-  }
-  bool supportsChannels(audio_channel_mask_t channels) const {
+    bool isValid() const { return hasValidFormat() && hasValidRates() && hasValidChannels(); }
+    bool supportsChannels(audio_channel_mask_t channels) const
+    {
@@ -30,3 +29 @@ class AudioProfile final : public RefBase, public Parcelable {
-  bool supportsRate(uint32_t rate) const {
-    return mSamplingRates.count(rate) != 0;
-  }
+    bool supportsRate(uint32_t rate) const { return mSamplingRates.count(rate) != 0; }
@@ -42,3 +39 @@ class AudioProfile final : public RefBase, public Parcelable {
-  bool isDynamic() {
-    return mIsDynamicFormat || mIsDynamicChannels || mIsDynamicRate;
-  }
+    bool isDynamic() { return mIsDynamicFormat || mIsDynamicChannels || mIsDynamicRate; }
@@ -51,2 +46 @@ class AudioProfile final : public RefBase, public Parcelable {
-  audio_format_t
-      mFormat;
+    audio_format_t mFormat;
@@ -59,8 +53 @@ class AudioProfile final : public RefBase, public Parcelable {
-<<<<<<< HEAD
-class AudioProfileVector : public std::vector<sp<AudioProfile>>,
-                           public Parcelable
-||||||| 5f90ad4290
-class AudioProfileVectorBase : public std::vector<sp<AudioProfile>>
-=======
-class AudioProfileVector : public std::vector<sp<AudioProfile>>
->>>>>>> 80ffad94f2b41098bad4f35f2399eb05e8c5ee34
+class AudioProfileVector : public std::vector<sp<AudioProfile>>, public Parcelable
@@ -69 +56 @@ class AudioProfileVector : public std::vector<sp<AudioProfile>>
-  ~AudioProfileVector() = default;
+    virtual ~AudioProfileVector() = default;
