--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@include@media@AudioProfile.no_comments_conflict.h
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@include@media@AudioProfile.no_comments_merged.h
@@ -53 +52,0 @@ private:
-<<<<<<< HEAD
@@ -55,5 +53,0 @@ class AudioProfileVector : public std::vector<sp<AudioProfile>>, public Parcelab
-||||||| 5f90ad4290
-class AudioProfileVectorBase : public std::vector<sp<AudioProfile> >
-=======
-class AudioProfileVector : public std::vector<sp<AudioProfile>>
->>>>>>> 80ffad94
