--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@Serializer.no_comments_conflict.cpp
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@Serializer.no_comments_merged.cpp
@@ -401 +400,0 @@ Return<DevicePortTraits::Element> DevicePortTraits::deserialize(const xmlNode *c
-<<<<<<< HEAD
@@ -403,11 +401,0 @@ Return<DevicePortTraits::Element> DevicePortTraits::deserialize(const xmlNode *c
-||||||| 5f90ad4290
-    if (!address.empty()) {
-        ALOGV("%s: address=%s for %s", __func__, address.c_str(), name.c_str());
-        deviceDesc->setAddress(String8(address.c_str()));
-    }
-=======
-    if (!address.empty()) {
-        ALOGV("%s: address=%s for %s", __func__, address.c_str(), name.c_str());
-        deviceDesc->setAddress(address);
-    }
->>>>>>> 80ffad94
