diff --git a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_conflict.h b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
index 5a612fa..4fdab07 100644
--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_conflict.h
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
@@ -2 +1,0 @@
-<<<<<<< HEAD
@@ -6,6 +4,0 @@
-||||||| 5f90ad4290
-#include "AudioPort.h"
-=======
-#include "PolicyAudioPort.h"
-#include <media/DeviceDescriptorBase.h>
->>>>>>> 80ffad94
