--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_merged.cpp
@@ -28 +28 @@ status_t HwModule::addOutputProfile(const std::string& name, const audio_config_
-sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device, "" , address.string()); addDynamicDevice(devDesc);
+    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device, "" , address.string());
@@ -85 +85 @@ status_t HwModule::addInputProfile(const std::string& name, const audio_config_t
-sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device, "" , address.string()); addDynamicDevice(devDesc);
+    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device, "" , address.string());
