diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
index 75156e4..4fdab07 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
@@ -2 +1,0 @@
-#include "AudioPort.h"
@@ -14,2 +13,2 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-                         public PolicyAudioPort,
-                         public PolicyAudioPortConfig {
+                         public PolicyAudioPort, public PolicyAudioPortConfig
+{
@@ -19,0 +19,4 @@ class DeviceDescriptor : public DeviceDescriptorBase,
+    DeviceDescriptor(audio_devices_t type, const std::string &tagName,
+            const std::string &address, const FormatVector &encodedFormats = FormatVector{});
+    DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr, const std::string &tagName = "",
+            const FormatVector &encodedFormats = FormatVector{});
@@ -36,2 +39 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-  virtual status_t applyAudioPortConfig(
-      const struct audio_port_config *config,
+    virtual status_t applyAudioPortConfig(const struct audio_port_config *config,
@@ -39,2 +41 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-  virtual void toAudioPortConfig(
-      struct audio_port_config *dstConfig,
+    virtual void toAudioPortConfig(struct audio_port_config *dstConfig,
@@ -52,2 +53 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-  std::string
-      mTagName;
+    std::string mTagName;
@@ -56,7 +55,0 @@ class DeviceDescriptor : public DeviceDescriptorBase,
- public:
-  DeviceDescriptor(audio_devices_t type, const std::string &tagName,
-                   const std::string &address,
-                   const FormatVector &encodedFormats = FormatVector{});
-  DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr,
-                   const std::string &tagName = "",
-                   const FormatVector &encodedFormats = FormatVector{});
@@ -64 +57,2 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
+class DeviceVector : public SortedVector<sp<DeviceDescriptor> >
+{
@@ -67 +61,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  explicit DeviceVector(const sp<DeviceDescriptor> &item) : DeviceVector() {
+    explicit DeviceVector(const sp<DeviceDescriptor>& item) : DeviceVector()
+    {
@@ -85,4 +80,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  DeviceVector getFirstDevicesFromTypes(
-      std::vector<audio_devices_t> orderedTypes) const;
-  sp<DeviceDescriptor> getFirstExistingDevice(
-      std::vector<audio_devices_t> orderedTypes) const;
+    DeviceVector getFirstDevicesFromTypes(std::vector<audio_devices_t> orderedTypes) const;
+    sp<DeviceDescriptor> getFirstExistingDevice(std::vector<audio_devices_t> orderedTypes) const;
@@ -90,2 +83 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  void replaceDevicesByType(audio_devices_t typeToRemove,
-                            const DeviceVector &devicesToAdd);
+    void replaceDevicesByType(audio_devices_t typeToRemove, const DeviceVector &devicesToAdd);
@@ -101,3 +93 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  bool contains(const sp<DeviceDescriptor> &item) const {
-    return indexOf(item) >= 0;
-  }
+    bool contains(const sp<DeviceDescriptor>& item) const { return indexOf(item) >= 0; }
@@ -108 +98,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  ssize_t merge(const DeviceVector &devices) {
+    ssize_t merge(const DeviceVector &devices)
+    {
@@ -115 +106,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  bool operator==(const DeviceVector &right) const {
+    bool operator==(const DeviceVector &right) const
+    {
@@ -126 +118,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  bool operator!=(const DeviceVector &right) const {
+    bool operator!=(const DeviceVector &right) const
+    {
@@ -129 +122,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  String8 getFirstValidAddress() const {
+    String8 getFirstValidAddress() const
+    {
@@ -138,2 +132 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  void dump(String8 *dst, const String8 &tag, int spaces = 0,
-            bool verbose = true) const;
+    void dump(String8 *dst, const String8 &tag, int spaces = 0, bool verbose = true) const;
