--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_conflict.cpp
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_merged.cpp
@@ -13 +12,0 @@ namespace android {
-<<<<<<< HEAD
@@ -16,5 +14,0 @@ AudioOutputDescriptor::AudioOutputDescriptor(const sp<PolicyAudioPort>& policyAu
-||||||| 5f90ad4290
-AudioOutputDescriptor::AudioOutputDescriptor(const sp<AudioPort>& port,
-=======
-AudioOutputDescriptor::AudioOutputDescriptor(const sp<PolicyAudioPort>& policyAudioPort,
->>>>>>> 80ffad94
