--- a/./output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl_readonly.no_comments_conflict.cc
+++ b/./output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl_readonly.no_comments_merged.cc
@@ -44,8 +43,0 @@ Status DBImplReadOnly::Get(const ReadOptions& options,
-<<<<<<< HEAD
-  MemTable* mem = GetDefaultColumnFamily()->mem();
-  Version* current = GetDefaultColumnFamily()->current();
-||||||| 30a700657
-  MemTable* mem = GetMemTable();
-  Version* current = versions_->current();
-=======
->>>>>>> 5b3b6549
@@ -53 +45 @@ Status DBImplReadOnly::Get(const ReadOptions& options,
-  SuperVersion* super_version = GetSuperVersion();
+  SuperVersion* super_version = GetDefaultColumnFamily()->GetSuperVersion();
@@ -78 +69,0 @@ Status DB::OpenForReadOnly(const Options& options, const std::string& dbname,
-<<<<<<< HEAD
@@ -86,4 +76,0 @@ Status DB::OpenForReadOnly(const Options& options, const std::string& dbname,
-||||||| 30a700657
-  Status s = impl->Recover(true , error_if_log_file_exist);
-=======
-  Status s = impl->Recover(true , error_if_log_file_exist);
@@ -91 +78,3 @@ Status DB::OpenForReadOnly(const Options& options, const std::string& dbname,
-    delete impl->InstallSuperVersion(new DBImpl::SuperVersion());
+    for (auto cfd : *impl->versions_->GetColumnFamilySet()) {
+      delete cfd->InstallSuperVersion(new SuperVersion());
+    }
@@ -93 +81,0 @@ Status DB::OpenForReadOnly(const Options& options, const std::string& dbname,
->>>>>>> 5b3b6549
