--- a/./output/frameworks_native/03a8ff5c-3b468eca-b00cdc55/services@surfaceflinger@SurfaceFlinger.no_comments_conflict.cpp
+++ b/./output/frameworks_native/03a8ff5c-3b468eca-b00cdc55/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
@@ -1950 +1949,0 @@ bool SurfaceFlinger::updateLayerSnapshots(VsyncId vsyncId, nsecs_t frameTimeNs,
-<<<<<<< HEAD
@@ -1956,15 +1954,0 @@ bool SurfaceFlinger::updateLayerSnapshots(VsyncId vsyncId, nsecs_t frameTimeNs,
-||||||| b00cdc5556
-    if (!mLegacyFrontEndEnabled) {
-        ATRACE_NAME("DisplayCallbackAndStatsUpdates");
-        mustComposite |= applyTransactions(update.transactions, vsyncId);
-        traverseLegacyLayers([&](Layer* layer) { layer->commitTransaction(); });
-        const nsecs_t latchTime = systemTime();
-        bool unused = false;
-=======
-    if (!mLegacyFrontEndEnabled) {
-        ATRACE_NAME("DisplayCallbackAndStatsUpdates");
-        mustComposite |= applyTransactionsLocked(update.transactions, vsyncId);
-        traverseLegacyLayers([&](Layer* layer) { layer->commitTransaction(); });
-        const nsecs_t latchTime = systemTime();
-        bool unused = false;
->>>>>>> 3b468eca
@@ -4871 +4854,0 @@ void SurfaceFlinger::initializeDisplays() {
-<<<<<<< HEAD
@@ -4875,12 +4857,0 @@ void SurfaceFlinger::initializeDisplays() {
-||||||| b00cdc5556
-    if (mLegacyFrontEndEnabled) {
-        applyTransactions(transactions, VsyncId{0});
-    } else {
-        applyAndCommitDisplayTransactionStates(transactions);
-=======
-    if (mLegacyFrontEndEnabled) {
-        applyTransactions(transactions, VsyncId{0});
-    } else {
-        Mutex::Autolock lock(mStateLock);
-        applyAndCommitDisplayTransactionStatesLocked(transactions);
->>>>>>> 3b468eca
