diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_merged.h
index ebcfdf8..d733eee 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_merged.h
@@ -108 +108,3 @@ class DBImpl : public DB {
-  Status TEST_CompactRange(int level, const Slice* begin, const Slice* end);
+  Status TEST_CompactRange(int level,
+                           const Slice* begin,
+                           const Slice* end);
@@ -118,4 +120,3 @@ class DBImpl : public DB {
-  void TEST_SetDefaultTimeToCheck(
-      uint64_t default_interval_to_delete_obsolete_WAL) {
-    default_interval_to_delete_obsolete_WAL_ =
-        default_interval_to_delete_obsolete_WAL;
+  void TEST_SetDefaultTimeToCheck(uint64_t default_interval_to_delete_obsolete_WAL)
+  {
+    default_interval_to_delete_obsolete_WAL_ = default_interval_to_delete_obsolete_WAL;
@@ -126 +127,2 @@ class DBImpl : public DB {
-      return candidate_files.size() || sst_delete_files.size() ||
+      return candidate_files.size() ||
+        sst_delete_files.size() ||
@@ -152 +154,2 @@ class DBImpl : public DB {
-  void FindObsoleteFiles(DeletionState& deletion_state, bool force,
+  void FindObsoleteFiles(DeletionState& deletion_state,
+                         bool force,
@@ -197,2 +200 @@ class DBImpl : public DB {
-  Status BackgroundCompaction(bool* madeProgress,
-                              DeletionState& deletion_state);
+  Status BackgroundCompaction(bool* madeProgress,DeletionState& deletion_state);
@@ -210 +212,2 @@ class DBImpl : public DB {
-                                VectorLogPtr& log_files, WalFileType type);
+                                VectorLogPtr& log_files,
+                                WalFileType type);
@@ -220,0 +224,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -271 +278,2 @@ class DBImpl : public DB {
-      SequenceNumber in, std::vector<SequenceNumber>& snapshots,
+    SequenceNumber in,
+    std::vector<SequenceNumber>& snapshots,
@@ -278,4 +285,0 @@ class DBImpl : public DB {
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
