--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_merged.h
@@ -220,0 +224,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -278,4 +285,0 @@ class DBImpl : public DB {
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
