diff --git a/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_conflict.h b/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_merged.h
index 4852f76..d733eee 100644
--- a/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_conflict.h
+++ b/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@db_impl.no_comments_merged.h
@@ -124,17 +123,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
-||||||| 4159a284c
-  struct SuperVersion {
-    MemTable* mem;
-    MemTableListVersion* imm;
-    Version* current;
-    std::atomic<uint32_t> refs;
-    autovector<MemTable*> to_delete;
-    SuperVersion() = default;
-    ~SuperVersion();
-    SuperVersion* Ref();
-    bool Unref();
-    void Cleanup();
-    void Init(MemTable* new_mem, MemTableListVersion* new_imm,
-              Version* new_current);
-  };
-=======
@@ -142,15 +124,0 @@ class DBImpl : public DB {
-  struct SuperVersion {
-    MemTable* mem;
-    MemTableListVersion* imm;
-    Version* current;
-    std::atomic<uint32_t> refs;
-    autovector<MemTable*> to_delete;
-    SuperVersion() = default;
-    ~SuperVersion();
-    SuperVersion* Ref();
-    bool Unref();
-    void Cleanup();
-    void Init(MemTable* new_mem, MemTableListVersion* new_imm,
-              Version* new_current);
-  };
->>>>>>> e6b3e3b4
