--- a/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@memtable_list.no_comments_mergegen.h
+++ b/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@memtable_list.no_comments_merged.h
@@ -60 +60,7 @@ class MemTableList {
-const autovector<MemTable*>& m, VersionSet* vset, port::Mutex* mu, Logger* info_log, uint64_t file_number, std::set<uint64_t>& pending_outputs, autovector<MemTable*>* to_delete, Directory* db_directory);
+  Status InstallMemtableFlushResults(ColumnFamilyData* cfd,
+                                     const autovector<MemTable*>& m,
+                                     VersionSet* vset, port::Mutex* mu,
+                                     Logger* info_log, uint64_t file_number,
+                                     std::set<uint64_t>& pending_outputs,
+                                     autovector<MemTable*>* to_delete,
+                                     Directory* db_directory);
