--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@memtable_list.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@memtable_list.no_comments_merged.h
@@ -12,0 +13,4 @@
+#include "db/memtable.h"
+#include "rocksdb/db.h"
+#include "rocksdb/iterator.h"
+#include "rocksdb/options.h"
@@ -56,16 +60,2 @@ class MemTableList {
-<<<<<<< HEAD
-  Status InstallMemtableFlushResults(
-      ColumnFamilyData* cfd, const autovector<MemTable*>& m, VersionSet* vset,
-      Status flushStatus, port::Mutex* mu, Logger* info_log,
-      uint64_t file_number, std::set<uint64_t>& pending_outputs,
-      autovector<MemTable*>* to_delete, Directory* db_directory);
-||||||| 4159a284c
-  Status InstallMemtableFlushResults(const autovector<MemTable*>& m,
-                                     VersionSet* vset, Status flushStatus,
-                                     port::Mutex* mu, Logger* info_log,
-                                     uint64_t file_number,
-                                     std::set<uint64_t>& pending_outputs,
-                                     autovector<MemTable*>* to_delete,
-                                     Directory* db_directory);
-=======
-  Status InstallMemtableFlushResults(const autovector<MemTable*>& m,
+  Status InstallMemtableFlushResults(ColumnFamilyData* cfd,
+                                     const autovector<MemTable*>& m,
@@ -77 +66,0 @@ class MemTableList {
->>>>>>> e6b3e3b4db380092807b1dfc3c08e37f9d62ec73
