--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_merged.h
@@ -24 +21,0 @@ class MemTable;
-class TableCache;
@@ -149,0 +150,3 @@ class VersionSet {
+  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
+  static Status ListColumnFamilies(std::vector<std::string>* column_families,
+                                   const std::string& dbname, Env* env);
@@ -196 +199,0 @@ class VersionSet {
-  friend class Compaction;
@@ -226,4 +228,0 @@ class VersionSet {
- public:
-  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
-  static Status ListColumnFamilies(std::vector<std::string>* column_families,
-                                   const std::string& dbname, Env* env);
