diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_merged.h
index 27cdcc3..fa0e320 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_merged.h
@@ -17,3 +17 @@ namespace rocksdb {
-namespace log {
-class Writer;
-}
+namespace log { class Writer; }
@@ -24 +21,0 @@ class MemTable;
-class TableCache;
@@ -32,2 +29,4 @@ extern int FindFile(const InternalKeyComparator& icmp,
-                    const std::vector<FileMetaData*>& files, const Slice& key);
-extern bool SomeFileOverlapsRange(const InternalKeyComparator& icmp,
+                    const std::vector<FileMetaData*>& files,
+                    const Slice& key);
+extern bool SomeFileOverlapsRange(
+    const InternalKeyComparator& icmp,
@@ -47,2 +46,3 @@ class Version {
-           Status* status, MergeContext* merge_context, GetStats* stats,
-           const Options& db_option, bool* value_found = nullptr);
+           Status* status, MergeContext* merge_context,
+           GetStats* stats, const Options& db_option,
+           bool* value_found = nullptr);
@@ -75,3 +75,3 @@ class Version {
-      unsigned int
-          index);
-  bool OverlapInLevel(int level, const Slice* smallest_user_key,
+      unsigned int index);
+  bool OverlapInLevel(int level,
+                      const Slice* smallest_user_key,
@@ -149,0 +150,3 @@ class VersionSet {
+  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
+  static Status ListColumnFamilies(std::vector<std::string>* column_families,
+                                   const std::string& dbname, Env* env);
@@ -188 +191,2 @@ class VersionSet {
-  void GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata);
+  void GetLiveFilesMetaData(
+    std::vector<LiveFileMetaData> *metadata);
@@ -196 +199,0 @@ class VersionSet {
-  friend class Compaction;
@@ -226,4 +228,0 @@ class VersionSet {
- public:
-  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
-  static Status ListColumnFamilies(std::vector<std::string>* column_families,
-                                   const std::string& dbname, Env* env);
