diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_merged.h
index 0c0626a..02c26e5 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@internal_stats.no_comments_merged.h
@@ -7 +7 @@
-class ColumnFamilyData
+class ColumnFamilyData;
@@ -84,2 +84 @@ class ColumnFamilyData
-                     std::string* value, VersionSet* version_set,
-                     const MemTableList& imm);
+                   std::string* value, ColumnFamilyData* cfd);
