--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@version_set.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/159928df-fcd5c5e8-758fa8c3/db@version_set.no_comments_merged.cc
@@ -112 +113,6 @@ class Version::LevelFileNumIterator : public Iterator {
-      : icmp_(icmp), flist_(flist), index_(flist->size()) {
+      : icmp_(icmp),
+        flist_(flist),
+        index_(flist->size()) {
+  }
+  virtual bool Valid() const {
+    return index_ < flist_->size();
@@ -114 +119,0 @@ class Version::LevelFileNumIterator : public Iterator {
-  virtual bool Valid() const { return index_ < flist_->size(); }
@@ -368 +372,0 @@ Version::Version(ColumnFamilyData* cfd, VersionSet* vset,
-      num_levels_(vset->num_levels_),
@@ -370 +373,0 @@ Version::Version(ColumnFamilyData* cfd, VersionSet* vset,
-      finalized_(false),
@@ -501,18 +507,0 @@ void Version::ComputeCompactionScore(
-<<<<<<< HEAD
-  assert(!finalized_);
-  finalized_ = true;
-  if (cfd_->options()->compaction_style == kCompactionStyleUniversal) {
-    std::sort(files_[0].begin(), files_[0].end(), NewestFirstBySeqNo);
-  } else {
-    std::sort(files_[0].begin(), files_[0].end(), NewestFirst);
-  }
-||||||| 758fa8c35
-  assert(!finalized_);
-  finalized_ = true;
-  if (vset_->options_->compaction_style == kCompactionStyleUniversal) {
-    std::sort(files_[0].begin(), files_[0].end(), NewestFirstBySeqNo);
-  } else {
-    std::sort(files_[0].begin(), files_[0].end(), NewestFirst);
-  }
-=======
->>>>>>> fcd5c5e8289d88f1100d24e7331020ee3d280464
@@ -1078 +1084 @@ class VersionSet::Builder {
-    if (v->vset_->options_->compaction_style == kCompactionStyleUniversal) {
+    if (cfd_->options()->compaction_style == kCompactionStyleUniversal) {
@@ -1133,7 +1138,0 @@ VersionSet::~VersionSet() {
-{
-  column_family_set_.reset();
-  for (auto file : obsolete_files_) {
-    delete file;
-  }
-  obsolete_files_.clear();
-}
@@ -1241 +1239,0 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-<<<<<<< HEAD
@@ -1243,3 +1240,0 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-||||||| 758fa8c35
-=======
->>>>>>> fcd5c5e8289d88f1100d24e7331020ee3d280464
@@ -1593 +1589,0 @@ Status VersionSet::Recover(
-<<<<<<< HEAD
@@ -1595 +1591 @@ Status VersionSet::Recover(
-      v->Finalize(size_being_compacted);
+      v->ComputeCompactionScore(size_being_compacted);
@@ -1599,7 +1594,0 @@ Status VersionSet::Recover(
-||||||| 758fa8c35
-      compaction_picker_->SizeBeingCompacted(size_being_compacted);
-      v->Finalize(size_being_compacted);
-=======
-      compaction_picker_->SizeBeingCompacted(size_being_compacted);
-      v->ComputeCompactionScore(size_being_compacted);
->>>>>>> fcd5c5e8289d88f1100d24e7331020ee3d280464
@@ -1876 +1866,0 @@ Status VersionSet::DumpManifest(Options& options, std::string& dscname,
-<<<<<<< HEAD
@@ -1878 +1868 @@ Status VersionSet::DumpManifest(Options& options, std::string& dscname,
-      v->Finalize(size_being_compacted);
+      v->ComputeCompactionScore(size_being_compacted);
@@ -1893,7 +1882,0 @@ Status VersionSet::DumpManifest(Options& options, std::string& dscname,
-||||||| 758fa8c35
-        compaction_picker_->SizeBeingCompacted(size_being_compacted);
-        v->Finalize(size_being_compacted);
-=======
-        compaction_picker_->SizeBeingCompacted(size_being_compacted);
-        v->ComputeCompactionScore(size_being_compacted);
->>>>>>> fcd5c5e8289d88f1100d24e7331020ee3d280464
@@ -2155,2 +2143,0 @@ ColumnFamilyData* VersionSet::CreateColumnFamily(
-  std::vector<uint64_t> size_being_compacted(options.num_levels - 1, 0);
-  v->Finalize(size_being_compacted);
