diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_merged.cpp
index 395147c..2f1e7e0 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_merged.cpp
@@ -12 +11,0 @@
-#include "src/widgets/capture/capturetoolbutton.h"
@@ -14 +12,0 @@
-#include <QLabel>
@@ -22,9 +19,0 @@ ConfigWindow::ConfigWindow(QWidget* parent)
-<<<<<<< HEAD
-    setMinimumSize(GlobalValues::buttonBaseSize() * 14,
-                   GlobalValues::buttonBaseSize() * 12);
-    setWindowIcon(QIcon(":img/app/flameshot.svg"));
-|||||||
-    const int size = GlobalValues::buttonBaseSize() * 12;
-    setMinimumSize(size, size);
-    setWindowIcon(QIcon(":img/app/flameshot.svg"));
-=======
@@ -34 +22,0 @@ ConfigWindow::ConfigWindow(QWidget* parent)
->>>>>>> ac7142c4df5cf218ddf5b032d80ac340639f99d0
