diff --git a/./output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_mergegen.h b/./output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_merged.h
index 96c9030..5eb84a2 100644
--- a/./output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_mergegen.h
+++ b/./output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_merged.h
@@ -142 +142,2 @@ typedef long long mstime_t;
-#define REDIS_AUTOSYNC_BYTES (1024*1024*32) #define LIMIT_PENDING_QUERYBUF (4*1024*1024)
+#define REDIS_AUTOSYNC_BYTES (1024*1024*32)
+#define LIMIT_PENDING_QUERYBUF (4*1024*1024)
