diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_merged.c
index 536ed74..4ac6d29 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_merged.c
@@ -16,2 +16 @@ void listTypePush(robj *subject, robj *value, int where) {
-    subject->ptr =
-        ziplistPush(subject->ptr, value->ptr, sdslen(value->ptr), pos);
+        subject->ptr = ziplistPush(subject->ptr,value->ptr,sdslen(value->ptr),pos);
@@ -74,2 +73 @@ unsigned long listTypeLength(robj *subject) {
-listTypeIterator *listTypeInitIterator(robj *subject, int index,
-                                       unsigned char direction) {
+listTypeIterator *listTypeInitIterator(robj *subject, int index, unsigned char direction) {
@@ -89 +87,3 @@ listTypeIterator *listTypeInitIterator(robj *subject, int index,
-void listTypeReleaseIterator(listTypeIterator *li) { zfree(li); }
+void listTypeReleaseIterator(listTypeIterator *li) {
+    zfree(li);
+}
@@ -147,2 +147 @@ void listTypeInsert(listTypeEntry *entry, robj *value, int where) {
-        subject->ptr = ziplistPush(subject->ptr, value->ptr, sdslen(value->ptr),
-                                   REDIS_TAIL);
+                subject->ptr = ziplistPush(subject->ptr,value->ptr,sdslen(value->ptr),REDIS_TAIL);
@@ -150,2 +149 @@ void listTypeInsert(listTypeEntry *entry, robj *value, int where) {
-        subject->ptr =
-            ziplistInsert(subject->ptr, next, value->ptr, sdslen(value->ptr));
+                subject->ptr = ziplistInsert(subject->ptr,next,value->ptr,sdslen(value->ptr));
@@ -154,2 +152 @@ void listTypeInsert(listTypeEntry *entry, robj *value, int where) {
-      subject->ptr = ziplistInsert(subject->ptr, entry->zi, value->ptr,
-                                   sdslen(value->ptr));
+            subject->ptr = ziplistInsert(subject->ptr,entry->zi,value->ptr,sdslen(value->ptr));
@@ -243,2 +240,6 @@ void pushGenericCommand(redisClient *c, int where) {
-void lpushCommand(redisClient *c) { pushGenericCommand(c, REDIS_HEAD); }
-void rpushCommand(redisClient *c) { pushGenericCommand(c, REDIS_TAIL); }
+void lpushCommand(redisClient *c) {
+    pushGenericCommand(c,REDIS_HEAD);
+}
+void rpushCommand(redisClient *c) {
+    pushGenericCommand(c,REDIS_TAIL);
+}
@@ -251,2 +252 @@ void pushxGenericCommand(redisClient *c, robj *refval, robj *val, int where) {
-      checkType(c, subject, REDIS_LIST))
-    return;
+        checkType(c,subject,REDIS_LIST)) return;
@@ -386,2 +386,6 @@ void popGenericCommand(redisClient *c, int where) {
-void lpopCommand(redisClient *c) { popGenericCommand(c, REDIS_HEAD); }
-void rpopCommand(redisClient *c) { popGenericCommand(c, REDIS_TAIL); }
+void lpopCommand(redisClient *c) {
+    popGenericCommand(c,REDIS_HEAD);
+}
+void rpopCommand(redisClient *c) {
+    popGenericCommand(c,REDIS_TAIL);
+}
@@ -395,4 +399,2 @@ void lrangeCommand(redisClient *c) {
-  if ((o = lookupKeyReadOrReply(c, c->argv[1], shared.emptymultibulk)) ==
-          NULL ||
-      checkType(c, o, REDIS_LIST))
-    return;
+    if ((o = lookupKeyReadOrReply(c,c->argv[1],shared.emptymultibulk)) == NULL
+         || checkType(c,o,REDIS_LIST)) return;
@@ -428,2 +430 @@ void ltrimCommand(redisClient *c) {
-      checkType(c, o, REDIS_LIST))
-    return;
+        checkType(c,o,REDIS_LIST)) return;
@@ -470 +471,2 @@ void lremCommand(redisClient *c) {
-  if (subject->encoding == REDIS_ENCODING_ZIPLIST) obj = getDecodedObject(obj);
+    if (subject->encoding == REDIS_ENCODING_ZIPLIST)
+        obj = getDecodedObject(obj);
@@ -487 +489,2 @@ void lremCommand(redisClient *c) {
-  if (subject->encoding == REDIS_ENCODING_ZIPLIST) decrRefCount(obj);
+    if (subject->encoding == REDIS_ENCODING_ZIPLIST)
+        decrRefCount(obj);
@@ -495,2 +498 @@ void rpoplpushcommand(redisClient *c) {
-      checkType(c, sobj, REDIS_LIST))
-    return;
+        checkType(c,sobj,REDIS_LIST)) return;
@@ -585,2 +587 @@ void blockingPopGenericCommand(redisClient *c, int where) {
-                                   "timeout is not an integer") != REDIS_OK)
-    return;
+            "timeout is not an integer") != REDIS_OK) return;
@@ -616 +616,0 @@ void blockingPopGenericCommand(redisClient *c, int where) {
-<<<<<<< HEAD
@@ -621,3 +620,0 @@ void blockingPopGenericCommand(redisClient *c, int where) {
-|||||||
-=======
->>>>>>> 94364d53b4746e8cd9e3da633162cb1e34f0bdb6
@@ -628,2 +625,6 @@ void blockingPopGenericCommand(redisClient *c, int where) {
-void blpopCommand(redisClient *c) { blockingPopGenericCommand(c, REDIS_HEAD); }
-void brpopCommand(redisClient *c) { blockingPopGenericCommand(c, REDIS_TAIL); }
+void blpopCommand(redisClient *c) {
+    blockingPopGenericCommand(c,REDIS_HEAD);
+}
+void brpopCommand(redisClient *c) {
+    blockingPopGenericCommand(c,REDIS_TAIL);
+}
