diff --git a/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_conflict.c b/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_merged.c
index ec4a91f..4ac6d29 100644
--- a/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_conflict.c
+++ b/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_merged.c
@@ -617 +616,0 @@ void blockingPopGenericCommand(redisClient *c, int where) {
-<<<<<<< HEAD
@@ -622,3 +620,0 @@ void blockingPopGenericCommand(redisClient *c, int where) {
-||||||| bad7d097e
-=======
->>>>>>> 94364d53
