diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
index cd23e56..86b80eb 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
@@ -34,2 +34 @@ void BlockingSample::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<unique_ptr<BaseReservoirSampling>>(100, "base_reservoir_sample",
-                                                                        base_reservoir_sample);
+ serializer.WritePropertyWithDefault<unique_ptr<BaseReservoirSampling>>(100, "base_reservoir_sample", base_reservoir_sample);
@@ -40,2 +39 @@ unique_ptr<BlockingSample> BlockingSample::Deserialize(Deserializer &deserialize
- auto base_reservoir_sample =
-     deserializer.ReadPropertyWithDefault<unique_ptr<BaseReservoirSampling>>(100, "base_reservoir_sample");
+ auto base_reservoir_sample = deserializer.ReadPropertyWithDefault<unique_ptr<BaseReservoirSampling>>(100, "base_reservoir_sample");
@@ -63,2 +61 @@ void BaseReservoirSampling::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<idx_t>(103, "num_entries_to_skip_b4_next_sample",
-                                            num_entries_to_skip_b4_next_sample);
+ serializer.WritePropertyWithDefault<idx_t>(103, "num_entries_to_skip_b4_next_sample", num_entries_to_skip_b4_next_sample);
@@ -66,2 +63 @@ void BaseReservoirSampling::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<std::priority_queue<std::pair<double, idx_t>>>(105, "reservoir_weights",
-                                                                                    reservoir_weights);
+ serializer.WritePropertyWithDefault<std::priority_queue<std::pair<double, idx_t>>>(105, "reservoir_weights", reservoir_weights);
@@ -74,2 +70 @@ unique_ptr<BaseReservoirSampling> BaseReservoirSampling::Deserialize(Deserialize
- deserializer.ReadPropertyWithDefault<idx_t>(103, "num_entries_to_skip_b4_next_sample",
-                                             result->num_entries_to_skip_b4_next_sample);
+ deserializer.ReadPropertyWithDefault<idx_t>(103, "num_entries_to_skip_b4_next_sample", result->num_entries_to_skip_b4_next_sample);
@@ -77,2 +72 @@ unique_ptr<BaseReservoirSampling> BaseReservoirSampling::Deserialize(Deserialize
- deserializer.ReadPropertyWithDefault<std::priority_queue<std::pair<double, idx_t>>>(105, "reservoir_weights",
-                                                                                     result->reservoir_weights);
+ deserializer.ReadPropertyWithDefault<std::priority_queue<std::pair<double, idx_t>>>(105, "reservoir_weights", result->reservoir_weights);
@@ -173,6 +167,3 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WriteProperty<CSVOption<char>>(121, "dialect_options.state_machine_options.delimiter",
-                                           dialect_options.state_machine_options.delimiter);
- serializer.WriteProperty<CSVOption<char>>(122, "dialect_options.state_machine_options.quote",
-                                           dialect_options.state_machine_options.quote);
- serializer.WriteProperty<CSVOption<char>>(123, "dialect_options.state_machine_options.escape",
-                                           dialect_options.state_machine_options.escape);
+ serializer.WriteProperty<CSVOption<char>>(121, "dialect_options.state_machine_options.delimiter", dialect_options.state_machine_options.delimiter);
+ serializer.WriteProperty<CSVOption<char>>(122, "dialect_options.state_machine_options.quote", dialect_options.state_machine_options.quote);
+ serializer.WriteProperty<CSVOption<char>>(123, "dialect_options.state_machine_options.escape", dialect_options.state_machine_options.escape);
@@ -181,2 +172 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WriteProperty<CSVOption<NewLineIdentifier>>(126, "dialect_options.state_machine_options.new_line",
-                                                        dialect_options.state_machine_options.new_line);
+ serializer.WriteProperty<CSVOption<NewLineIdentifier>>(126, "dialect_options.state_machine_options.new_line", dialect_options.state_machine_options.new_line);
@@ -184,2 +174 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WriteProperty<map<LogicalTypeId, CSVOption<StrpTimeFormat>>>(128, "dialect_options.date_format",
-                                                                         dialect_options.date_format);
+ serializer.WriteProperty<map<LogicalTypeId, CSVOption<StrpTimeFormat>>>(128, "dialect_options.date_format", dialect_options.date_format);
@@ -189,2 +178 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<CSVOption<string>>(132, "rejects_scan_name", rejects_scan_name,
-                                                        {"reject_scans"});
+ serializer.WritePropertyWithDefault<CSVOption<string>>(132, "rejects_scan_name", rejects_scan_name, {"reject_scans"});
@@ -193,2 +181 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<case_insensitive_map_t<idx_t>>(135, "sql_types_per_column",
-                                                                    sql_types_per_column);
+ serializer.WritePropertyWithDefault<case_insensitive_map_t<idx_t>>(135, "sql_types_per_column", sql_types_per_column);
@@ -196,5 +183,2 @@ void CSVReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment",
-                                                      dialect_options.state_machine_options.comment,
-                                                      CSVOption<char>('\0'));
- serializer.WritePropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header",
-                                            dialect_options.rows_until_header);
+ serializer.WritePropertyWithDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment", dialect_options.state_machine_options.comment, CSVOption<char>('\0'));
+ serializer.WritePropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header", dialect_options.rows_until_header);
@@ -221,2 +205 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadPropertyWithExplicitDefault<string>(117, "rejects_table_name", result.rejects_table_name,
-                                                      "reject_errors");
+ deserializer.ReadPropertyWithExplicitDefault<string>(117, "rejects_table_name", result.rejects_table_name, "reject_errors");
@@ -226,6 +209,3 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadProperty<CSVOption<char>>(121, "dialect_options.state_machine_options.delimiter",
-                                            result.dialect_options.state_machine_options.delimiter);
- deserializer.ReadProperty<CSVOption<char>>(122, "dialect_options.state_machine_options.quote",
-                                            result.dialect_options.state_machine_options.quote);
- deserializer.ReadProperty<CSVOption<char>>(123, "dialect_options.state_machine_options.escape",
-                                            result.dialect_options.state_machine_options.escape);
+ deserializer.ReadProperty<CSVOption<char>>(121, "dialect_options.state_machine_options.delimiter", result.dialect_options.state_machine_options.delimiter);
+ deserializer.ReadProperty<CSVOption<char>>(122, "dialect_options.state_machine_options.quote", result.dialect_options.state_machine_options.quote);
+ deserializer.ReadProperty<CSVOption<char>>(123, "dialect_options.state_machine_options.escape", result.dialect_options.state_machine_options.escape);
@@ -234,2 +214 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadProperty<CSVOption<NewLineIdentifier>>(126, "dialect_options.state_machine_options.new_line",
-                                                         result.dialect_options.state_machine_options.new_line);
+ deserializer.ReadProperty<CSVOption<NewLineIdentifier>>(126, "dialect_options.state_machine_options.new_line", result.dialect_options.state_machine_options.new_line);
@@ -237,4 +216,2 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadProperty<map<LogicalTypeId, CSVOption<StrpTimeFormat>>>(128, "dialect_options.date_format",
-                                                                          result.dialect_options.date_format);
- deserializer.ReadPropertyWithDefault<string>(129, "sniffer_user_mismatch_error",
-                                              result.sniffer_user_mismatch_error);
+ deserializer.ReadProperty<map<LogicalTypeId, CSVOption<StrpTimeFormat>>>(128, "dialect_options.date_format", result.dialect_options.date_format);
+ deserializer.ReadPropertyWithDefault<string>(129, "sniffer_user_mismatch_error", result.sniffer_user_mismatch_error);
@@ -243,2 +220 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadPropertyWithExplicitDefault<CSVOption<string>>(132, "rejects_scan_name", result.rejects_scan_name,
-                                                                 {"reject_scans"});
+ deserializer.ReadPropertyWithExplicitDefault<CSVOption<string>>(132, "rejects_scan_name", result.rejects_scan_name, {"reject_scans"});
@@ -247,2 +223 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadPropertyWithDefault<case_insensitive_map_t<idx_t>>(135, "sql_types_per_column",
-                                                                     result.sql_types_per_column);
+ deserializer.ReadPropertyWithDefault<case_insensitive_map_t<idx_t>>(135, "sql_types_per_column", result.sql_types_per_column);
@@ -250,5 +225,2 @@ CSVReaderOptions CSVReaderOptions::Deserialize(Deserializer &deserializer) {
- deserializer.ReadPropertyWithExplicitDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment",
-                                                               result.dialect_options.state_machine_options.comment,
-                                                               CSVOption<char>('\0'));
- deserializer.ReadPropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header",
-                                             result.dialect_options.rows_until_header);
+ deserializer.ReadPropertyWithExplicitDefault<CSVOption<char>>(137, "dialect_options.state_machine_options.comment", result.dialect_options.state_machine_options.comment, CSVOption<char>('\0'));
+ deserializer.ReadPropertyWithDefault<idx_t>(138, "dialect_options.rows_until_header", result.dialect_options.rows_until_header);
@@ -284,2 +256 @@ void ColumnDefinition::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<unordered_map<string, string>>(106, "tags", tags,
-                                                                    unordered_map<string, string>());
+ serializer.WritePropertyWithDefault<unordered_map<string, string>>(106, "tags", tags, unordered_map<string, string>());
@@ -295,2 +266 @@ ColumnDefinition ColumnDefinition::Deserialize(Deserializer &deserializer) {
- deserializer.ReadPropertyWithExplicitDefault<unordered_map<string, string>>(106, "tags", result.tags,
-                                                                             unordered_map<string, string>());
+ deserializer.ReadPropertyWithExplicitDefault<unordered_map<string, string>>(106, "tags", result.tags, unordered_map<string, string>());
@@ -330,2 +300 @@ void CommonTableExpressionMap::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<InsertionOrderPreservingMap<unique_ptr<CommonTableExpressionInfo>>>(100, "map",
-                                                                                                         map);
+ serializer.WritePropertyWithDefault<InsertionOrderPreservingMap<unique_ptr<CommonTableExpressionInfo>>>(100, "map", map);
@@ -335,2 +304 @@ CommonTableExpressionMap CommonTableExpressionMap::Deserialize(Deserializer &des
- deserializer.ReadPropertyWithDefault<InsertionOrderPreservingMap<unique_ptr<CommonTableExpressionInfo>>>(
-     100, "map", result.map);
+ deserializer.ReadPropertyWithDefault<InsertionOrderPreservingMap<unique_ptr<CommonTableExpressionInfo>>>(100, "map", result.map);
@@ -397,2 +365 @@ void MultiFileReaderBindData::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<vector<HivePartitioningIndex>>(101, "hive_partitioning_indexes",
-                                                                    hive_partitioning_indexes);
+ serializer.WritePropertyWithDefault<vector<HivePartitioningIndex>>(101, "hive_partitioning_indexes", hive_partitioning_indexes);
@@ -403,2 +370 @@ MultiFileReaderBindData MultiFileReaderBindData::Deserialize(Deserializer &deser
- deserializer.ReadPropertyWithDefault<vector<HivePartitioningIndex>>(101, "hive_partitioning_indexes",
-                                                                     result.hive_partitioning_indexes);
+ deserializer.ReadPropertyWithDefault<vector<HivePartitioningIndex>>(101, "hive_partitioning_indexes", result.hive_partitioning_indexes);
@@ -413,4 +379,2 @@ void MultiFileReaderOptions::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<case_insensitive_map_t<LogicalType>>(105, "hive_types_schema",
-                                                                          hive_types_schema);
- serializer.WritePropertyWithDefault<string>(106, "filename_column", filename_column,
-                                             MultiFileReaderOptions::DEFAULT_FILENAME_COLUMN);
+ serializer.WritePropertyWithDefault<case_insensitive_map_t<LogicalType>>(105, "hive_types_schema", hive_types_schema);
+ serializer.WritePropertyWithDefault<string>(106, "filename_column", filename_column, MultiFileReaderOptions::DEFAULT_FILENAME_COLUMN);
@@ -422,2 +386 @@ MultiFileReaderOptions MultiFileReaderOptions::Deserialize(Deserializer &deseria
- deserializer.ReadPropertyWithDefault<bool>(102, "auto_detect_hive_partitioning",
-                                            result.auto_detect_hive_partitioning);
+ deserializer.ReadPropertyWithDefault<bool>(102, "auto_detect_hive_partitioning", result.auto_detect_hive_partitioning);
@@ -426,4 +389,2 @@ MultiFileReaderOptions MultiFileReaderOptions::Deserialize(Deserializer &deseria
- deserializer.ReadPropertyWithDefault<case_insensitive_map_t<LogicalType>>(105, "hive_types_schema",
-                                                                           result.hive_types_schema);
- deserializer.ReadPropertyWithExplicitDefault<string>(106, "filename_column", result.filename_column,
-                                                      MultiFileReaderOptions::DEFAULT_FILENAME_COLUMN);
+ deserializer.ReadPropertyWithDefault<case_insensitive_map_t<LogicalType>>(105, "hive_types_schema", result.hive_types_schema);
+ deserializer.ReadPropertyWithExplicitDefault<string>(106, "filename_column", result.filename_column, MultiFileReaderOptions::DEFAULT_FILENAME_COLUMN);
@@ -445,2 +406 @@ void PivotColumn::Serialize(Serializer &serializer) const {
- serializer.WritePropertyWithDefault<vector<unique_ptr<ParsedExpression>>>(100, "pivot_expressions",
-                                                                           pivot_expressions);
+ serializer.WritePropertyWithDefault<vector<unique_ptr<ParsedExpression>>>(100, "pivot_expressions", pivot_expressions);
@@ -453,2 +413 @@ PivotColumn PivotColumn::Deserialize(Deserializer &deserializer) {
- deserializer.ReadPropertyWithDefault<vector<unique_ptr<ParsedExpression>>>(100, "pivot_expressions",
-                                                                            result.pivot_expressions);
+ deserializer.ReadPropertyWithDefault<vector<unique_ptr<ParsedExpression>>>(100, "pivot_expressions", result.pivot_expressions);
