diff --git a/./output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_conflict.cpp b/./output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_merged.cpp
index 64663d8..a3136c6 100644
--- a/./output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_conflict.cpp
+++ b/./output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_merged.cpp
@@ -18 +17,0 @@ using std::stringstream;
-void ReorderTableEntries(catalog_entry_vector_t &tables);
@@ -196 +194,0 @@ SourceResultType PhysicalExport::GetData(ExecutionContext &context, DataChunk &c
-<<<<<<< HEAD
@@ -203,13 +200,0 @@ SourceResultType PhysicalExport::GetData(ExecutionContext &context, DataChunk &c
-||||||| c220f7bc2f
- auto schema_list = Catalog::GetSchemas(ccontext, info->catalog);
- ExtractEntries(context.client, schema_list, entries);
- entries.tables.clear();
- for (idx_t i = 0; i < exported_tables.data.size(); i++) {
-  entries.tables.push_back(exported_tables.data[i].entry);
-=======
- auto schema_list = Catalog::GetSchemas(ccontext, info->catalog);
- ExtractEntries(context.client, schema_list, entries);
- entries.tables.clear();
- for (idx_t i = 0; i < exported_tables->data.size(); i++) {
-  entries.tables.push_back(exported_tables->data[i].entry);
->>>>>>> 18132018
