--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7cdf5ee9-f165221d-24350879/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7cdf5ee9-f165221d-24350879/tmux.no_comments_merged.c
@@ -4 +3,0 @@
-#include <err.h>
@@ -7 +5,0 @@
-#include <paths.h>
@@ -11 +8,0 @@
-#include <util.h>
@@ -161,5 +194,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
-  while ((opt = getopt(argc, argv, "2c:Cdf:lL:qS:uUVv")) != -1) {
-|||||||
-  while ((opt = getopt(argc, argv, "2c:Cdf:lL:qS:uUv")) != -1) {
-=======
@@ -167 +195,0 @@ int main(int argc, char **argv) {
->>>>>>> f165221dc4641837ee9f589bb666d310a495904c
