--- a/./output/frameworks_native/ddfd0eff-c84d6d7d-fe42efcc/services@inputflinger@dispatcher@InputDispatcher.no_comments_conflict.h
+++ b/./output/frameworks_native/ddfd0eff-c84d6d7d-fe42efcc/services@inputflinger@dispatcher@InputDispatcher.no_comments_merged.h
@@ -139 +138,0 @@ private:
-<<<<<<< HEAD
@@ -143,9 +141,0 @@ private:
-||||||| fe42efccdf
-    sp<android::gui::WindowInfoHandle> findTouchedWindowAtLocked(
-            int32_t displayId, int32_t x, int32_t y, TouchState* touchState, bool isStylus = false,
-            bool addOutsideTargets = false, bool ignoreDragWindow = false) REQUIRES(mLock);
-=======
-    sp<android::gui::WindowInfoHandle> findTouchedWindowAtLocked(
-            int32_t displayId, float x, float y, TouchState* touchState, bool isStylus = false,
-            bool addOutsideTargets = false, bool ignoreDragWindow = false) REQUIRES(mLock);
->>>>>>> c84d6d7d
