diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/a2357844-b5b4c9f1-ffb64347/include@grpcpp@server_impl.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/grpc/a2357844-b5b4c9f1-ffb64347/include@grpcpp@server_impl.no_comments_merged.h
index 56e38a9..fbae0cf 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/a2357844-b5b4c9f1-ffb64347/include@grpcpp@server_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/a2357844-b5b4c9f1-ffb64347/include@grpcpp@server_impl.no_comments_merged.h
@@ -22 +22 @@
-struct grpc_server
+struct grpc_server;
@@ -24,0 +25 @@ struct grpc_server
+class ServerContext;
@@ -30 +30,0 @@ namespace grpc_impl {
-class ServerContext;
@@ -84,0 +85 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
+ protected:
@@ -93,2 +94,3 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-  std::unique_ptr < grpc::experimental::ServerInterceptorFactoryInterface >>
-      *interceptor_creators() override {
+  std::vector<
+      std::unique_ptr<grpc::experimental::ServerInterceptorFactoryInterface>>*
+  interceptor_creators() override {
