diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_merged.h
index 214bb7a..e8f285a 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_merged.h
@@ -132,6 +132,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER \
-  10
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER \
-  100
-#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES \
-  (100 << 20)
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER 10
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER 100
+#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES (100<<20)
@@ -139,8 +136,4 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX \
-  75
-#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS \
-  1000
-#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN \
-  (512ll * 1024 * 1024)
-#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL \
-  10
+#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX 75
+#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS 1000
+#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN (512ll*1024*1024)
+#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL 10
@@ -149,2 +142 @@ typedef long long mstime_t;
-#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC \
-  25
+#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC 25
@@ -212,2 +204 @@ typedef long long mstime_t;
-#define CLIENT_MONITOR \
-  (1 << 2)
+#define CLIENT_MONITOR (1<<2)
@@ -215,2 +206 @@ typedef long long mstime_t;
-#define CLIENT_BLOCKED \
-  (1 << 4)
+#define CLIENT_BLOCKED (1<<4)
@@ -218,4 +208,3 @@ typedef long long mstime_t;
-#define CLIENT_CLOSE_AFTER_REPLY \
-  (1 << 6)
-#define CLIENT_UNBLOCKED \
-  (1 << 7)
+#define CLIENT_CLOSE_AFTER_REPLY (1<<6)
+#define CLIENT_UNBLOCKED (1<<7)
+#define CLIENT_LUA (1<<8)
@@ -224,6 +213,3 @@ typedef long long mstime_t;
-#define CLIENT_UNIX_SOCKET \
-  (1 << 11)
-#define CLIENT_DIRTY_EXEC \
-  (1 << 12)
-#define CLIENT_MASTER_FORCE_REPLY \
-  (1 << 13)
+#define CLIENT_UNIX_SOCKET (1<<11)
+#define CLIENT_DIRTY_EXEC (1<<12)
+#define CLIENT_MASTER_FORCE_REPLY (1<<13)
@@ -238,4 +224,3 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_WRITE \
-  (1 << 21)
-#define CLIENT_REPLY_SKIP_NEXT \
-  (1 << 23)
+#define CLIENT_PENDING_WRITE (1<<21)
+#define CLIENT_REPLY_OFF (1<<22)
+#define CLIENT_REPLY_SKIP_NEXT (1<<23)
@@ -247,6 +232,4 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_READ \
-  (1 << 29)
-#define CLIENT_PENDING_COMMAND \
-  (1 << 30)
-#define CLIENT_TRACKING \
-  (1 << 31)
+#define CLIENT_PENDING_READ (1<<29)
+#define CLIENT_PENDING_COMMAND (1<<30)
+#define CLIENT_TRACKING (1<<31)
+#define CLIENT_TRACKING_BROKEN_REDIR (1ULL<<32)
@@ -266,2 +249 @@ typedef long long mstime_t;
-#define CLIENT_TYPE_OBUF_COUNT \
-  3
+#define CLIENT_TYPE_OBUF_COUNT 3
@@ -306,0 +289 @@ typedef long long mstime_t;
+#define UNUSED(V) ((void) V)
@@ -339,4 +322,2 @@ typedef long long mstime_t;
-#define MAXMEMORY_ALLKEYS_LRU \
-  ((4 << 8) | MAXMEMORY_FLAG_LRU | MAXMEMORY_FLAG_ALLKEYS)
-#define MAXMEMORY_ALLKEYS_LFU \
-  ((5 << 8) | MAXMEMORY_FLAG_LFU | MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LRU ((4<<8)|MAXMEMORY_FLAG_LRU|MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LFU ((5<<8)|MAXMEMORY_FLAG_LFU|MAXMEMORY_FLAG_ALLKEYS)
@@ -350,2 +331,2 @@ typedef long long mstime_t;
-#define SHUTDOWN_SAVE \
-  1
+#define SHUTDOWN_SAVE 1
+#define SHUTDOWN_NOSAVE 2
@@ -377,4 +358 @@ typedef long long mstime_t;
-#define NOTIFY_ALL \
-  (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | \
-   NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | \
-   NOTIFY_KEY_MISS)
+#define NOTIFY_ALL (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | NOTIFY_KEY_MISS)
@@ -381,0 +360,4 @@ typedef long long mstime_t;
+#define run_with_period(_ms_) if ((_ms_ <= 1000/server.hz) || !(server.cronloops%((_ms_)/(1000/server.hz))))
+#define serverAssertWithInfo(_c,_o,_e) ((_e)?(void)0 : (_serverAssertWithInfo(_c,_o,#_e,__FILE__,__LINE__),_exit(1)))
+#define serverAssert(_e) ((_e)?(void)0 : (_serverAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define serverPanic(...) _serverPanic(__FILE__,__LINE__,__VA_ARGS__),_exit(1)
@@ -390,0 +373,2 @@ typedef long long mstime_t;
+#define REDISMODULE_TYPE_ENCVER(id) (id & REDISMODULE_TYPE_ENCVER_MASK)
+#define REDISMODULE_TYPE_SIGN(id) ((id & ~((uint64_t)REDISMODULE_TYPE_ENCVER_MASK)) >>REDISMODULE_TYPE_ENCVER_BITS)
@@ -400,2 +384 @@ typedef void (*moduleTypeSaveFunc)(struct RedisModuleIO *io, void *value);
-typedef int (*moduleTypeAuxLoadFunc)(struct RedisModuleIO *rdb, int encver,
-                                     int when);
+typedef int (*moduleTypeAuxLoadFunc)(struct RedisModuleIO *rdb, int encver, int when);
@@ -403,4 +386,2 @@ typedef void (*moduleTypeAuxSaveFunc)(struct RedisModuleIO *rdb, int when);
-typedef void (*moduleTypeRewriteFunc)(struct RedisModuleIO *io,
-                                      struct redisObject *key, void *value);
-typedef void (*moduleTypeDigestFunc)(struct RedisModuleDigest *digest,
-                                     void *value);
+typedef void (*moduleTypeRewriteFunc)(struct RedisModuleIO *io, struct redisObject *key, void *value);
+typedef void (*moduleTypeDigestFunc)(struct RedisModuleDigest *digest, void *value);
@@ -435,0 +417,9 @@ typedef struct RedisModuleIO {
+#define moduleInitIOContext(iovar,mtype,rioptr,keyptr) do { \
+    iovar.rio = rioptr; \
+    iovar.type = mtype; \
+    iovar.bytes = 0; \
+    iovar.error = 0; \
+    iovar.ver = 0; \
+    iovar.key = keyptr; \
+    iovar.ctx = NULL; \
+} while(0);
@@ -439,0 +430,4 @@ typedef struct RedisModuleDigest {
+#define moduleInitDigestContext(mdvar) do { \
+    memset(mdvar.o,0,sizeof(mdvar.o)); \
+    memset(mdvar.x,0,sizeof(mdvar.x)); \
+} while(0);
@@ -462,0 +457,6 @@ char *getObjectTypeName(robj *);
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = OBJ_STRING; \
+    _var.encoding = OBJ_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0)
@@ -476,2 +476 @@ typedef struct redisDb {
-  list *
-      defrag_later;
+    list *defrag_later;
@@ -508,2 +507 @@ typedef struct readyList {
-#define USER_COMMAND_BITS_COUNT \
-  1024
+#define USER_COMMAND_BITS_COUNT 1024
@@ -514,2 +512 @@ typedef struct readyList {
-#define USER_FLAG_NOPASS \
-  (1 << 4)
+#define USER_FLAG_NOPASS (1<<4)
@@ -523,2 +520,4 @@ typedef struct user {
-#define CLIENT_ID_AOF \
-  (UINT64_MAX)
+} user;
+#define CLIENT_ID_AOF (UINT64_MAX)
+typedef struct client {
+    uint64_t id;
@@ -574,0 +574 @@ typedef struct user {
+} client;
@@ -585,2 +585,2 @@ typedef struct user {
-    robj *crlf, *ok, *err, *emptybulk, *czero, *cone, *pong, *space, *colon,
-        *queued, *null[4], *nullarray[4], *emptymap[4], *emptyset[4],
+    robj *crlf, *ok, *err, *emptybulk, *czero, *cone, *pong, *space,
+    *colon, *queued, *null[4], *nullarray[4], *emptymap[4], *emptyset[4],
@@ -590,4 +590,4 @@ typedef struct user {
-        *busykeyerr, *oomerr, *plus, *messagebulk, *pmessagebulk,
-        *subscribebulk, *unsubscribebulk, *psubscribebulk, *punsubscribebulk,
-        *del, *unlink, *rpop, *lpop, *lpush, *rpoplpush, *zpopmin, *zpopmax,
-        *emptyscan, *select[PROTO_SHARED_SELECT_CMDS],
+    *busykeyerr, *oomerr, *plus, *messagebulk, *pmessagebulk, *subscribebulk,
+    *unsubscribebulk, *psubscribebulk, *punsubscribebulk, *del, *unlink,
+    *rpop, *lpop, *lpush, *rpoplpush, *zpopmin, *zpopmax, *emptyscan,
+    *select[PROTO_SHARED_SELECT_CMDS],
@@ -622,2 +622 @@ typedef struct user {
-  extern clientBufferLimitsConfig
-      clientBufferLimitsDefaults[CLIENT_TYPE_OBUF_COUNT];
+extern clientBufferLimitsConfig clientBufferLimitsDefaults[CLIENT_TYPE_OBUF_COUNT];
@@ -669,2 +668 @@ typedef struct user {
-#define RDB_SAVE_INFO_INIT \
-  { -1, 0, "000000000000000000000000000000", -1 }
+#define RDB_SAVE_INFO_INIT {-1,0,"000000000000000000000000000000",-1}
@@ -760,3 +758,4 @@ typedef struct user {
-    struct redisCommand *delCommand, *multiCommand, *lpushCommand, *lpopCommand,
-        *rpopCommand, *zpopminCommand, *zpopmaxCommand, *sremCommand,
-        *execCommand, *expireCommand, *pexpireCommand, *xclaimCommand,
+    struct redisCommand *delCommand, *multiCommand, *lpushCommand,
+                        *lpopCommand, *rpopCommand, *zpopminCommand,
+                        *zpopmaxCommand, *sremCommand, *execCommand,
+                        *expireCommand, *pexpireCommand, *xclaimCommand,
@@ -769,2 +768 @@ typedef struct user {
-    long long
-        stat_expired_time_cap_reached_count;
+    long long stat_expired_time_cap_reached_count;
@@ -775,6 +773,3 @@ typedef struct user {
-    long long
-        stat_active_defrag_misses;
-    long long
-        stat_active_defrag_key_hits;
-    long long
-        stat_active_defrag_key_misses;
+    long long stat_active_defrag_misses;
+    long long stat_active_defrag_key_hits;
+    long long stat_active_defrag_key_misses;
@@ -811,2 +806 @@ typedef struct user {
-    size_t
-        active_defrag_ignore_bytes;
+    size_t active_defrag_ignore_bytes;
@@ -817,4 +811,2 @@ typedef struct user {
-    unsigned long
-        active_defrag_max_scan_fields;
-    _Atomic size_t
-        client_max_querybuf_len;
+    unsigned long active_defrag_max_scan_fields;
+    _Atomic size_t client_max_querybuf_len;
@@ -1034,2 +1026 @@ typedef struct user {
-  typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc,
-                                int *numkeys);
+typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1086,0 +1078 @@ typedef struct user {
+#include "stream.h"
@@ -1107,2 +1099 @@ typedef struct user {
-  int *moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv,
-                                  int argc, int *numkeys);
+int *moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1115,2 +1106 @@ typedef struct user {
-  void moduleBlockedClientPipeReadable(aeEventLoop *el, int fd, void *privdata,
-                                       int mask);
+void moduleBlockedClientPipeReadable(aeEventLoop *el, int fd, void *privdata, int mask);
@@ -1120,2 +1110 @@ typedef struct user {
-  void moduleNotifyKeyspaceEvent(int type, const char *event, robj *key,
-                                 int dbid);
+void moduleNotifyKeyspaceEvent(int type, const char *event, robj *key, int dbid);
@@ -1127,2 +1116 @@ typedef struct user {
-  sds modulesCollectInfo(sds info, sds section, int for_crash_report,
-                         int sections);
+sds modulesCollectInfo(sds info, sds section, int for_crash_report, int sections);
@@ -1245,2 +1233 @@ typedef struct user {
-  listTypeIterator *listTypeInitIterator(robj *subject, long index,
-                                         unsigned char direction);
+listTypeIterator *listTypeInitIterator(robj *subject, long index, unsigned char direction);
@@ -1297,2 +1284 @@ typedef struct user {
-  int getLongFromObjectOrReply(client *c, robj *o, long *target,
-                               const char *msg);
+int getLongFromObjectOrReply(client *c, robj *o, long *target, const char *msg);
@@ -1300,4 +1286,2 @@ typedef struct user {
-  int getLongLongFromObjectOrReply(client *c, robj *o, long long *target,
-                                   const char *msg);
-  int getDoubleFromObjectOrReply(client *c, robj *o, double *target,
-                                 const char *msg);
+int getLongLongFromObjectOrReply(client *c, robj *o, long long *target, const char *msg);
+int getDoubleFromObjectOrReply(client *c, robj *o, double *target, const char *msg);
@@ -1307,2 +1291 @@ typedef struct user {
-  int getLongDoubleFromObjectOrReply(client *c, robj *o, long double *target,
-                                     const char *msg);
+int getLongDoubleFromObjectOrReply(client *c, robj *o, long double *target, const char *msg);
@@ -1314,0 +1298 @@ typedef struct user {
+#define sdsEncodedObject(objptr) (objptr->encoding == OBJ_ENCODING_RAW || objptr->encoding == OBJ_ENCODING_EMBSTR)
@@ -1319,4 +1303,2 @@ typedef struct user {
-  void replicationFeedSlavesFromMasterStream(list *slaves, char *buf,
-                                             size_t buflen);
-  void replicationFeedMonitors(client *c, list *monitors, int dictid,
-                               robj **argv, int argc);
+void replicationFeedSlavesFromMasterStream(list *slaves, char *buf, size_t buflen);
+void replicationFeedMonitors(client *c, list *monitors, int dictid, robj **argv, int argc);
@@ -1348,2 +1330 @@ typedef struct user {
-  void rdbPipeReadHandler(struct aeEventLoop *eventLoop, int fd,
-                          void *clientData, int mask);
+void rdbPipeReadHandler(struct aeEventLoop *eventLoop, int fd, void *clientData, int mask);
@@ -1360,0 +1342 @@ typedef struct user {
+#include "rdb.h"
@@ -1363,2 +1345 @@ typedef struct user {
-  void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv,
-                          int argc);
+void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv, int argc);
@@ -1439,2 +1420 @@ typedef struct user {
-  void genericZpopCommand(client *c, robj **keyv, int keyc, int where,
-                          int emitkey, robj *countarg);
+void genericZpopCommand(client *c, robj **keyv, int keyc, int where, int emitkey, robj *countarg);
@@ -1454,2 +1434 @@ typedef struct user {
-  int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree,
-                        float *level);
+int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *level);
@@ -1465,4 +1444,2 @@ typedef struct user {
-  void propagate(struct redisCommand *cmd, int dbid, robj **argv, int argc,
-                 int flags);
-  void alsoPropagate(struct redisCommand *cmd, int dbid, robj **argv, int argc,
-                     int target);
+void propagate(struct redisCommand *cmd, int dbid, robj **argv, int argc, int flags);
+void alsoPropagate(struct redisCommand *cmd, int dbid, robj **argv, int argc, int target);
@@ -1502,2 +1479 @@ typedef struct user {
-#define RESTART_SERVER_CONFIG_REWRITE \
-  (1 << 1)
+#define RESTART_SERVER_CONFIG_REWRITE (1<<1)
@@ -1514,2 +1490 @@ typedef struct user {
-  unsigned long setTypeRandomElements(robj *set, unsigned long count,
-                                      robj *aux_set);
+unsigned long setTypeRandomElements(robj *set, unsigned long count, robj *aux_set);
@@ -1530 +1505,2 @@ typedef struct user {
-                                  unsigned char **vstr, unsigned int *vlen,
+                                unsigned char **vstr,
+                                unsigned int *vlen,
@@ -1533,3 +1509 @@ typedef struct user {
-  void hashTypeCurrentObject(hashTypeIterator *hi, int what,
-                             unsigned char **vstr, unsigned int *vlen,
-                             long long *vll);
+void hashTypeCurrentObject(hashTypeIterator *hi, int what, unsigned char **vstr, unsigned int *vlen, long long *vll);
@@ -1553,2 +1527 @@ typedef struct user {
-  void rewriteConfigRewriteLine(struct rewriteConfigState *state,
-                                const char *option, sds line, int force);
+void rewriteConfigRewriteLine(struct rewriteConfigState *state, const char *option, sds line, int force);
@@ -1584,2 +1557 @@ typedef struct user {
-  long long emptyDbGeneric(redisDb *dbarray, int dbnum, int flags,
-                           void(callback)(void *));
+long long emptyDbGeneric(redisDb *dbarray, int dbnum, int flags, void(callback)(void*));
@@ -1590,2 +1562 @@ typedef struct user {
-  unsigned int getKeysInSlot(unsigned int hashslot, robj **keys,
-                             unsigned int count);
+unsigned int getKeysInSlot(unsigned int hashslot, robj **keys, unsigned int count);
@@ -1605,2 +1576 @@ typedef struct user {
-  int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc,
-                          int *numkeys);
+int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1608,12 +1578,6 @@ typedef struct user {
-  int *zunionInterGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                          int *numkeys);
-  int *evalGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                   int *numkeys);
-  int *sortGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                   int *numkeys);
-  int *migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                      int *numkeys);
-  int *georadiusGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                        int *numkeys);
-  int *xreadGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                    int *numkeys);
+int *zunionInterGetKeys(struct redisCommand *cmd,robj **argv, int argc, int *numkeys);
+int *evalGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *sortGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *georadiusGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *xreadGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1627,3 +1591 @@ typedef struct user {
-  int clusterSendModuleMessageToTarget(const char *target, uint64_t module_id,
-                                       uint8_t type, unsigned char *payload,
-                                       uint32_t len);
+int clusterSendModuleMessageToTarget(const char *target, uint64_t module_id, uint8_t type, unsigned char *payload, uint32_t len);
@@ -1648,2 +1610 @@ typedef struct user {
-  int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout,
-                                  int unit);
+int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int unit);
@@ -1653,2 +1614 @@ typedef struct user {
-  void blockForKeys(client *c, int btype, robj **keys, int numkeys,
-                    mstime_t timeout, robj *target, streamID *ids);
+void blockForKeys(client *c, int btype, robj **keys, int numkeys, mstime_t timeout, robj *target, streamID *ids);
@@ -1868,2 +1828,7 @@ typedef struct user {
-  void _serverAssertWithInfo(const client *c, const robj *o, const char *estr,
-                             const char *file, int line);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
+void _serverAssertWithInfo(const client *c, const robj *o, const char *estr, const char *file, int line);
@@ -1886,0 +1852,4 @@ typedef struct user {
+#define redisDebug(fmt,...) \
+    printf("DEBUG %s:%d > " fmt "\n", __FILE__, __LINE__, __VA_ARGS__)
+#define redisDebugMark() \
+    printf("-- MARK %s:%d --\n", __FILE__, __LINE__)
