--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_merged.h
@@ -132,6 +132,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER \
-  10
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER \
-  100
-#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES \
-  (100 << 20)
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER 10
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER 100
+#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES (100<<20)
@@ -139,8 +136,4 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX \
-  75
-#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS \
-  1000
-#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN \
-  (512ll * 1024 * 1024)
-#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL \
-  10
+#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX 75
+#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS 1000
+#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN (512ll*1024*1024)
+#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL 10
@@ -149,2 +142 @@ typedef long long mstime_t;
-#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC \
-  25
+#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC 25
@@ -212,2 +204 @@ typedef long long mstime_t;
-#define CLIENT_MONITOR \
-  (1 << 2)
+#define CLIENT_MONITOR (1<<2)
@@ -215,2 +206 @@ typedef long long mstime_t;
-#define CLIENT_BLOCKED \
-  (1 << 4)
+#define CLIENT_BLOCKED (1<<4)
@@ -218,4 +208,3 @@ typedef long long mstime_t;
-#define CLIENT_CLOSE_AFTER_REPLY \
-  (1 << 6)
-#define CLIENT_UNBLOCKED \
-  (1 << 7)
+#define CLIENT_CLOSE_AFTER_REPLY (1<<6)
+#define CLIENT_UNBLOCKED (1<<7)
+#define CLIENT_LUA (1<<8)
@@ -224,6 +213,3 @@ typedef long long mstime_t;
-#define CLIENT_UNIX_SOCKET \
-  (1 << 11)
-#define CLIENT_DIRTY_EXEC \
-  (1 << 12)
-#define CLIENT_MASTER_FORCE_REPLY \
-  (1 << 13)
+#define CLIENT_UNIX_SOCKET (1<<11)
+#define CLIENT_DIRTY_EXEC (1<<12)
+#define CLIENT_MASTER_FORCE_REPLY (1<<13)
@@ -238,4 +224,3 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_WRITE \
-  (1 << 21)
-#define CLIENT_REPLY_SKIP_NEXT \
-  (1 << 23)
+#define CLIENT_PENDING_WRITE (1<<21)
+#define CLIENT_REPLY_OFF (1<<22)
+#define CLIENT_REPLY_SKIP_NEXT (1<<23)
@@ -247,6 +232,4 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_READ \
-  (1 << 29)
-#define CLIENT_PENDING_COMMAND \
-  (1 << 30)
-#define CLIENT_TRACKING \
-  (1 << 31)
+#define CLIENT_PENDING_READ (1<<29)
+#define CLIENT_PENDING_COMMAND (1<<30)
+#define CLIENT_TRACKING (1<<31)
+#define CLIENT_TRACKING_BROKEN_REDIR (1ULL<<32)
@@ -266,2 +249 @@ typedef long long mstime_t;
-#define CLIENT_TYPE_OBUF_COUNT \
-  3
+#define CLIENT_TYPE_OBUF_COUNT 3
@@ -306,0 +289 @@ typedef long long mstime_t;
+#define UNUSED(V) ((void) V)
@@ -339,4 +322,2 @@ typedef long long mstime_t;
-#define MAXMEMORY_ALLKEYS_LRU \
-  ((4 << 8) | MAXMEMORY_FLAG_LRU | MAXMEMORY_FLAG_ALLKEYS)
-#define MAXMEMORY_ALLKEYS_LFU \
-  ((5 << 8) | MAXMEMORY_FLAG_LFU | MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LRU ((4<<8)|MAXMEMORY_FLAG_LRU|MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LFU ((5<<8)|MAXMEMORY_FLAG_LFU|MAXMEMORY_FLAG_ALLKEYS)
@@ -350,2 +331,2 @@ typedef long long mstime_t;
-#define SHUTDOWN_SAVE \
-  1
+#define SHUTDOWN_SAVE 1
+#define SHUTDOWN_NOSAVE 2
@@ -377,4 +358 @@ typedef long long mstime_t;
-#define NOTIFY_ALL \
-  (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | \
-   NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | \
-   NOTIFY_KEY_MISS)
+#define NOTIFY_ALL (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | NOTIFY_KEY_MISS)
@@ -381,0 +360,4 @@ typedef long long mstime_t;
+#define run_with_period(_ms_) if ((_ms_ <= 1000/server.hz) || !(server.cronloops%((_ms_)/(1000/server.hz))))
+#define serverAssertWithInfo(_c,_o,_e) ((_e)?(void)0 : (_serverAssertWithInfo(_c,_o,#_e,__FILE__,__LINE__),_exit(1)))
+#define serverAssert(_e) ((_e)?(void)0 : (_serverAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define serverPanic(...) _serverPanic(__FILE__,__LINE__,__VA_ARGS__),_exit(1)
@@ -390,0 +373,2 @@ typedef long long mstime_t;
+#define REDISMODULE_TYPE_ENCVER(id) (id & REDISMODULE_TYPE_ENCVER_MASK)
+#define REDISMODULE_TYPE_SIGN(id) ((id & ~((uint64_t)REDISMODULE_TYPE_ENCVER_MASK)) >>REDISMODULE_TYPE_ENCVER_BITS)
@@ -435,0 +417,9 @@ typedef struct RedisModuleIO {
+#define moduleInitIOContext(iovar,mtype,rioptr,keyptr) do { \
+    iovar.rio = rioptr; \
+    iovar.type = mtype; \
+    iovar.bytes = 0; \
+    iovar.error = 0; \
+    iovar.ver = 0; \
+    iovar.key = keyptr; \
+    iovar.ctx = NULL; \
+} while(0);
@@ -439,0 +430,4 @@ typedef struct RedisModuleDigest {
+#define moduleInitDigestContext(mdvar) do { \
+    memset(mdvar.o,0,sizeof(mdvar.o)); \
+    memset(mdvar.x,0,sizeof(mdvar.x)); \
+} while(0);
@@ -462,0 +457,6 @@ char *getObjectTypeName(robj *);
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = OBJ_STRING; \
+    _var.encoding = OBJ_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0)
@@ -508,2 +507 @@ typedef struct readyList {
-#define USER_COMMAND_BITS_COUNT \
-  1024
+#define USER_COMMAND_BITS_COUNT 1024
@@ -514,2 +512 @@ typedef struct readyList {
-#define USER_FLAG_NOPASS \
-  (1 << 4)
+#define USER_FLAG_NOPASS (1<<4)
@@ -523,2 +520,4 @@ typedef struct user {
-#define CLIENT_ID_AOF \
-  (UINT64_MAX)
+} user;
+#define CLIENT_ID_AOF (UINT64_MAX)
+typedef struct client {
+    uint64_t id;
@@ -574,0 +574 @@ typedef struct user {
+} client;
@@ -669,2 +668 @@ typedef struct user {
-#define RDB_SAVE_INFO_INIT \
-  { -1, 0, "000000000000000000000000000000", -1 }
+#define RDB_SAVE_INFO_INIT {-1,0,"000000000000000000000000000000",-1}
@@ -1086,0 +1078 @@ typedef struct user {
+#include "stream.h"
@@ -1314,0 +1298 @@ typedef struct user {
+#define sdsEncodedObject(objptr) (objptr->encoding == OBJ_ENCODING_RAW || objptr->encoding == OBJ_ENCODING_EMBSTR)
@@ -1360,0 +1342 @@ typedef struct user {
+#include "rdb.h"
@@ -1502,2 +1479 @@ typedef struct user {
-#define RESTART_SERVER_CONFIG_REWRITE \
-  (1 << 1)
+#define RESTART_SERVER_CONFIG_REWRITE (1<<1)
@@ -1868,2 +1828,7 @@ typedef struct user {
-  void _serverAssertWithInfo(const client *c, const robj *o, const char *estr,
-                             const char *file, int line);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
+void _serverAssertWithInfo(const client *c, const robj *o, const char *estr, const char *file, int line);
@@ -1886,0 +1852,4 @@ typedef struct user {
+#define redisDebug(fmt,...) \
+    printf("DEBUG %s:%d > " fmt "\n", __FILE__, __LINE__, __VA_ARGS__)
+#define redisDebugMark() \
+    printf("-- MARK %s:%d --\n", __FILE__, __LINE__)
