--- a/./output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_conflict.h
+++ b/./output/redis/51c3ff8d-fdaea2a7-764b420f/src@server.no_comments_merged.h
@@ -1118 +1117,0 @@ void moduleFireServerEvent(uint64_t eid, int subid, void *data);
-<<<<<<< HEAD
@@ -1120,2 +1118,0 @@ void processModuleLoadingProgressEvent(int is_aof);
-||||||| 764b420f5
-=======
@@ -1125 +1121,0 @@ int moduleClientIsBlockedOnKeys(client *c);
->>>>>>> fdaea2a7
