diff --git a/./output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_conflict.h b/./output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_merged.h
index 7c29cb6..e0b5c70 100644
--- a/./output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_conflict.h
+++ b/./output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_merged.h
@@ -142 +141,0 @@ private:
-<<<<<<< HEAD
@@ -163,21 +161,0 @@ private:
-||||||| 3be023edb8
-=======
-    class Spatializer {
-    public:
-        explicit Spatializer(AudioAnalytics &audioAnalytics)
-            : mAudioAnalytics(audioAnalytics) {}
-        void onEvent(const std::shared_ptr<const android::mediametrics::Item> &item);
-        std::pair<std::string, int32_t> dump(
-                int32_t lines = INT32_MAX, const char *prefix = nullptr) const;
-    private:
-        struct DeviceState {
-            std::string enabled;
-            std::string hasHeadTracker;
-            std::string headTrackerEnabled;
-        };
-        AudioAnalytics& mAudioAnalytics;
-        mutable std::mutex mLock;
-        std::map<std::string, DeviceState> mDeviceStateMap GUARDED_BY(mLock);
-        SimpleLog mSimpleLog GUARDED_BY(mLock) {64};
-    } mSpatializer{*this};
->>>>>>> 4591aab2
