diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_merged.h
index 54c4f6a..e0b5c70 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_merged.h
@@ -11 +11,2 @@ namespace android::mediametrics {
-class AudioAnalytics {
+class AudioAnalytics
+{
@@ -17,2 +18,2 @@ class AudioAnalytics {
-    std::pair<std::string, int32_t> dump(int32_t lines = INT32_MAX, int64_t sinceNs = 0,
-                                         const char* prefix = nullptr) const;
+    std::pair<std::string, int32_t> dump(
+            int32_t lines = INT32_MAX, int64_t sinceNs = 0, const char *prefix = nullptr) const;
@@ -37 +38,2 @@ class AudioAnalytics {
-    bool reportAudioRecordStatus(const std::shared_ptr<const mediametrics::Item>& item,
+    bool reportAudioRecordStatus(
+            const std::shared_ptr<const mediametrics::Item>& item,
@@ -39,3 +41,4 @@ class AudioAnalytics {
-                                 const std::string& statusString, uid_t uid,
-                                 const std::string& message, int32_t subCode) const;
-    bool reportAudioTrackStatus(const std::shared_ptr<const mediametrics::Item>& item,
+            const std::string& statusString, uid_t uid, const std::string& message,
+            int32_t subCode) const;
+    bool reportAudioTrackStatus(
+            const std::shared_ptr<const mediametrics::Item>& item,
@@ -43,2 +46,2 @@ class AudioAnalytics {
-                                const std::string& statusString, uid_t uid,
-                                const std::string& message, int32_t subCode) const;
+            const std::string& statusString, uid_t uid, const std::string& message,
+            int32_t subCode) const;
@@ -62 +65,2 @@ class AudioAnalytics {
-        void endAudioIntervalGroup(const std::shared_ptr<const android::mediametrics::Item>& item,
+        void endAudioIntervalGroup(
+                const std::shared_ptr<const android::mediametrics::Item> &item,
@@ -71,2 +75,4 @@ class AudioAnalytics {
-        void a2dpConnected(const std::shared_ptr<const android::mediametrics::Item>& item);
-        void createPatch(const std::shared_ptr<const android::mediametrics::Item>& item);
+        void a2dpConnected(
+                const std::shared_ptr<const android::mediametrics::Item> &item);
+        void createPatch(
+                const std::shared_ptr<const android::mediametrics::Item> &item);
@@ -97 +103,2 @@ class AudioAnalytics {
-        void endAAudioStream(const std::shared_ptr<const android::mediametrics::Item>& item,
+        void endAAudioStream(
+                const std::shared_ptr<const android::mediametrics::Item> &item,
@@ -105 +112,2 @@ class AudioAnalytics {
-        explicit Health(AudioAnalytics& audioAnalytics) : mAudioAnalytics(audioAnalytics) {}
+        explicit Health(AudioAnalytics &audioAnalytics)
+            : mAudioAnalytics(audioAnalytics) {}
@@ -112,4 +120,2 @@ class AudioAnalytics {
-                case Module::AUDIOFLINGER:
-                    return "AudioFlinger";
-                case Module::AUDIOPOLICY:
-                    return "AudioPolicy";
+                case Module::AUDIOFLINGER: return "AudioFlinger";
+                case Module::AUDIOPOLICY: return "AudioPolicy";
@@ -123,2 +129,2 @@ class AudioAnalytics {
-        std::pair<std::string, int32_t> dump(int32_t lines = INT32_MAX,
-                                             const char* prefix = nullptr) const;
+        std::pair<std::string, int32_t> dump(
+                int32_t lines = INT32_MAX, const char *prefix = nullptr) const;
@@ -138 +144,2 @@ class AudioAnalytics {
-        explicit Spatializer(AudioAnalytics& audioAnalytics) : mAudioAnalytics(audioAnalytics) {}
+        explicit Spatializer(AudioAnalytics &audioAnalytics)
+            : mAudioAnalytics(audioAnalytics) {}
@@ -140,2 +147,2 @@ class AudioAnalytics {
-        std::pair<std::string, int32_t> dump(int32_t lines = INT32_MAX,
-                                             const char* prefix = nullptr) const;
+        std::pair<std::string, int32_t> dump(
+                int32_t lines = INT32_MAX, const char *prefix = nullptr) const;
@@ -148,0 +156 @@ class AudioAnalytics {
+        static constexpr int64_t kBootDurationThreshold = 120 * 1e9;
@@ -149,0 +158 @@ class AudioAnalytics {
+        int64_t mFirstCreateTimeNs GUARDED_BY(mLock) = 0;
