diff --git a/./output/grpc/664918f9-de6defa6-502c796a/src@core@ext@filters@client_channel@client_channel.no_comments_mergegen.cc b/./output/grpc/664918f9-de6defa6-502c796a/src@core@ext@filters@client_channel@client_channel.no_comments_merged.cc
index 87c8d61..be64a59 100644
--- a/./output/grpc/664918f9-de6defa6-502c796a/src@core@ext@filters@client_channel@client_channel.no_comments_mergegen.cc
+++ b/./output/grpc/664918f9-de6defa6-502c796a/src@core@ext@filters@client_channel@client_channel.no_comments_merged.cc
@@ -116 +116,2 @@ class ChannelData {
- ExternalConnectivityWatcher::RemoveWatcherFromExternalWatchersMap( this, on_complete, cancel);
+    ExternalConnectivityWatcher::RemoveWatcherFromExternalWatchersMap(
+        this, on_complete, cancel);
