--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/664918f9-de6defa6-502c796a/src@compiler@ruby_generator_string-inl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/664918f9-de6defa6-502c796a/src@compiler@ruby_generator_string-inl.no_comments_merged.h
@@ -76 +76 @@ inline std::string RubyPackage(const grpc::protobuf::FileDescriptor* file) {
-inline grpc::string RubyTypeOf(const grpc::protobuf::Descriptor* descriptor) {
+inline std::string RubyTypeOf(const grpc::protobuf::Descriptor* descriptor) {
@@ -83,2 +83,2 @@ inline grpc::string RubyTypeOf(const grpc::protobuf::Descriptor* descriptor) {
-  grpc::string res(proto_type);
-  if (res.find('.') == grpc::string::npos) {
+  std::string res(proto_type);
+  if (res.find('.') == std::string::npos) {
@@ -87 +87 @@ inline grpc::string RubyTypeOf(const grpc::protobuf::Descriptor* descriptor) {
-    std::vector<grpc::string> prefixes_and_type = Split(res, '.');
+    std::vector<std::string> prefixes_and_type = Split(res, '.');
