diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_merged.cc
index 62d28f7..ad9f773 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/664918f9-de6defa6-502c796a/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-std::string Version() < < < < < < < HEAD { return "1.31.0-dev"; }
-|||||||
-{ return "1.30.0-dev"; }
-=======
-{ return "1.30.2"; }
->>>>>>> de6defa6fff08de20e36f9168f5b277e292daf46
+std::string Version() { return "1.31.0-dev"; }
