--- a/./output/duckdb/ea6567fe-d84e329b-9de6e816/tools@pythonpkg@src@pyconnection.no_comments_conflict.cpp
+++ b/./output/duckdb/ea6567fe-d84e329b-9de6e816/tools@pythonpkg@src@pyconnection.no_comments_merged.cpp
@@ -390 +390 @@ void DuckDBPyConnection::RegisterArrowObject(const py::object &arrow_object, con
-     make_unique<PythonTableArrowArrayStreamFactory>(arrow_object.ptr(), connection->context->config);
+     make_uniq<PythonTableArrowArrayStreamFactory>(arrow_object.ptr(), connection->context->config);
@@ -404 +404 @@ void DuckDBPyConnection::RegisterArrowObject(const py::object &arrow_object, con
-     make_shared<PythonDependencies>(make_unique<RegisteredArrow>(std::move(stream_factory), arrow_object)));
+     make_shared<PythonDependencies>(make_uniq<RegisteredArrow>(std::move(stream_factory), arrow_object)));
@@ -424,12 +423,0 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
-<<<<<<< HEAD
-   vector<shared_ptr<ExternalDependency>> dependencies;
-   dependencies.push_back(make_shared<PythonDependencies>(make_unique<RegisteredObject>(python_object),
-                                                          make_unique<RegisteredObject>(new_df)));
-   connection->context->external_dependencies[name] = std::move(dependencies);
-  }
-||||||| 9de6e81659
-  vector<shared_ptr<ExternalDependency>> dependencies;
-  dependencies.push_back(make_shared<PythonDependencies>(make_unique<RegisteredObject>(python_object),
-                                                         make_unique<RegisteredObject>(new_df)));
-  connection->context->external_dependencies[name] = std::move(dependencies);
-=======
@@ -440 +428 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
->>>>>>> d84e329b
+  }
@@ -455 +442,0 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
-<<<<<<< HEAD
@@ -457,37 +443,0 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
-||||||| 9de6e81659
-  auto stream_factory =
-      make_unique<PythonTableArrowArrayStreamFactory>(arrow_object.ptr(), connection->context->config);
-  auto stream_factory_produce = PythonTableArrowArrayStreamFactory::Produce;
-  auto stream_factory_get_schema = PythonTableArrowArrayStreamFactory::GetSchema;
-  {
-   py::gil_scoped_release release;
-   temporary_views[name] =
-       connection
-           ->TableFunction("arrow_scan", {Value::POINTER((uintptr_t)stream_factory.get()),
-                                          Value::POINTER((uintptr_t)stream_factory_produce),
-                                          Value::POINTER((uintptr_t)stream_factory_get_schema)})
-           ->CreateView(name, true, true);
-  }
-  vector<shared_ptr<ExternalDependency>> dependencies;
-  dependencies.push_back(
-      make_shared<PythonDependencies>(make_unique<RegisteredArrow>(std::move(stream_factory), arrow_object)));
-  connection->context->external_dependencies[name] = std::move(dependencies);
-=======
-  auto stream_factory =
-      make_uniq<PythonTableArrowArrayStreamFactory>(arrow_object.ptr(), connection->context->config);
-  auto stream_factory_produce = PythonTableArrowArrayStreamFactory::Produce;
-  auto stream_factory_get_schema = PythonTableArrowArrayStreamFactory::GetSchema;
-  {
-   py::gil_scoped_release release;
-   temporary_views[name] =
-       connection
-           ->TableFunction("arrow_scan", {Value::POINTER((uintptr_t)stream_factory.get()),
-                                          Value::POINTER((uintptr_t)stream_factory_produce),
-                                          Value::POINTER((uintptr_t)stream_factory_get_schema)})
-           ->CreateView(name, true, true);
-  }
-  vector<shared_ptr<ExternalDependency>> dependencies;
-  dependencies.push_back(
-      make_shared<PythonDependencies>(make_uniq<RegisteredArrow>(std::move(stream_factory), arrow_object)));
-  connection->context->external_dependencies[name] = std::move(dependencies);
->>>>>>> d84e329b
@@ -1081 +1030,0 @@ static unique_ptr<TableRef> TryReplacement(py::dict &dict, py::str &table_name,
-<<<<<<< HEAD
@@ -1086,15 +1034,0 @@ static unique_ptr<TableRef> TryReplacement(py::dict &dict, py::str &table_name,
-   string name = "df_" + StringUtil::GenerateRandomName();
-   auto new_df = PandasScanFunction::PandasReplaceCopiedNames(entry);
-   children.push_back(make_unique<ConstantExpression>(Value::POINTER((uintptr_t)new_df.ptr())));
-   table_function->function = make_unique<FunctionExpression>("pandas_scan", std::move(children));
-   table_function->external_dependency = make_unique<PythonDependencies>(
-       make_unique<RegisteredObject>(entry), make_unique<RegisteredObject>(new_df));
-  }
-||||||| 9de6e81659
-  string name = "df_" + StringUtil::GenerateRandomName();
-  auto new_df = PandasScanFunction::PandasReplaceCopiedNames(entry);
-  children.push_back(make_unique<ConstantExpression>(Value::POINTER((uintptr_t)new_df.ptr())));
-  table_function->function = make_unique<FunctionExpression>("pandas_scan", std::move(children));
-  table_function->external_dependency = make_unique<PythonDependencies>(make_unique<RegisteredObject>(entry),
-                                                                        make_unique<RegisteredObject>(new_df));
-=======
@@ -1107 +1041 @@ static unique_ptr<TableRef> TryReplacement(py::dict &dict, py::str &table_name,
->>>>>>> d84e329b
+  }
