diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/250e7f69-d3ea4c86-2ebd2720/src@config.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/250e7f69-d3ea4c86-2ebd2720/src@config.no_comments_merged.c
index f6fa39a..52b77b9 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/250e7f69-d3ea4c86-2ebd2720/src@config.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/250e7f69-d3ea4c86-2ebd2720/src@config.no_comments_merged.c
@@ -3,6 +3,3 @@ int yesnotoi(char *s) {
-  if (!strcasecmp(s, "yes"))
-    return 1;
-  else if (!strcasecmp(s, "no"))
-    return 0;
-  else
-    return -1;
+    if (!strcasecmp(s,"yes")) return 1;
+    else if (!strcasecmp(s,"no")) return 0;
+    else return -1;
@@ -11,2 +8 @@ void appendServerSaveParams(time_t seconds, int changes) {
-  server.saveparams = zrealloc(
-      server.saveparams, sizeof(struct saveparam) * (server.saveparamslen + 1));
+    server.saveparams = zrealloc(server.saveparams,sizeof(struct saveparam)*(server.saveparamslen+1));
@@ -38,2 +34 @@ void loadServerConfigFromString(char *config) {
-        err = "Invalid timeout value";
-        goto loaderr;
+                err = "Invalid timeout value"; goto loaderr;
@@ -44,2 +39 @@ void loadServerConfigFromString(char *config) {
-        err = "Invalid port";
-        goto loaderr;
+                err = "Invalid port"; goto loaderr;
@@ -55,2 +49 @@ void loadServerConfigFromString(char *config) {
-        err = "Invalid socket file permissions";
-        goto loaderr;
+                err = "Invalid socket file permissions"; goto loaderr;
@@ -63,2 +56 @@ void loadServerConfigFromString(char *config) {
-          err = "Invalid save parameters";
-          goto loaderr;
+                    err = "Invalid save parameters"; goto loaderr;
@@ -72,2 +64,2 @@ void loadServerConfigFromString(char *config) {
-        redisLog(REDIS_WARNING, "Can't chdir to '%s': %s", argv[1],
-                 strerror(errno));
+                redisLog(REDIS_WARNING,"Can't chdir to '%s': %s",
+                    argv[1], strerror(errno));
@@ -77,8 +69,4 @@ void loadServerConfigFromString(char *config) {
-      if (!strcasecmp(argv[1], "debug"))
-        server.verbosity = REDIS_DEBUG;
-      else if (!strcasecmp(argv[1], "verbose"))
-        server.verbosity = REDIS_VERBOSE;
-      else if (!strcasecmp(argv[1], "notice"))
-        server.verbosity = REDIS_NOTICE;
-      else if (!strcasecmp(argv[1], "warning"))
-        server.verbosity = REDIS_WARNING;
+            if (!strcasecmp(argv[1],"debug")) server.verbosity = REDIS_DEBUG;
+            else if (!strcasecmp(argv[1],"verbose")) server.verbosity = REDIS_VERBOSE;
+            else if (!strcasecmp(argv[1],"notice")) server.verbosity = REDIS_NOTICE;
+            else if (!strcasecmp(argv[1],"warning")) server.verbosity = REDIS_WARNING;
@@ -99,2 +87,2 @@ void loadServerConfigFromString(char *config) {
-          err = sdscatprintf(sdsempty(), "Can't open the log file: %s",
-                             strerror(errno));
+                    err = sdscatprintf(sdsempty(),
+                        "Can't open the log file: %s", strerror(errno));
@@ -107,2 +95 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -118,5 +105,11 @@ void loadServerConfigFromString(char *config) {
-          {"user", LOG_USER}, {"local0", LOG_LOCAL0},
-          {"local1", LOG_LOCAL1}, {"local2", LOG_LOCAL2},
-          {"local3", LOG_LOCAL3}, {"local4", LOG_LOCAL4},
-          {"local5", LOG_LOCAL5}, {"local6", LOG_LOCAL6},
-          {"local7", LOG_LOCAL7}, {NULL, 0}};
+                {"user", LOG_USER},
+                {"local0", LOG_LOCAL0},
+                {"local1", LOG_LOCAL1},
+                {"local2", LOG_LOCAL2},
+                {"local3", LOG_LOCAL3},
+                {"local4", LOG_LOCAL4},
+                {"local5", LOG_LOCAL5},
+                {"local6", LOG_LOCAL6},
+                {"local7", LOG_LOCAL7},
+                {NULL, 0}
+            };
@@ -131,3 +124 @@ void loadServerConfigFromString(char *config) {
-        err =
-            "Invalid log facility. Must be one of USER or between "
-            "LOCAL0-LOCAL7";
+                err = "Invalid log facility. Must be one of USER or between LOCAL0-LOCAL7";
@@ -139,2 +130 @@ void loadServerConfigFromString(char *config) {
-        err = "Invalid number of databases";
-        goto loaderr;
+                err = "Invalid number of databases"; goto loaderr;
@@ -191,2 +181 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -195,2 +184 @@ void loadServerConfigFromString(char *config) {
-      redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"",
-               argv[0]);
+            redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"", argv[0]);
@@ -199,2 +187 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -204,2 +191 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -209,2 +195 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -215,2 +200 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -222 +206,2 @@ void loadServerConfigFromString(char *config) {
-    } else if (!strcasecmp(argv[0], "no-appendfsync-on-rewrite") && argc == 2) {
+        } else if (!strcasecmp(argv[0],"no-appendfsync-on-rewrite")
+                   && argc == 2) {
@@ -224,2 +209 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -239 +223,2 @@ void loadServerConfigFromString(char *config) {
-               argc == 2) {
+                   argc == 2)
+        {
@@ -245 +230,3 @@ void loadServerConfigFromString(char *config) {
-    } else if (!strcasecmp(argv[0], "auto-aof-rewrite-min-size") && argc == 2) {
+        } else if (!strcasecmp(argv[0],"auto-aof-rewrite-min-size") &&
+                   argc == 2)
+        {
@@ -256,2 +243 @@ void loadServerConfigFromString(char *config) {
-      redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"",
-               argv[0]);
+            redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"", argv[0]);
@@ -260,2 +246 @@ void loadServerConfigFromString(char *config) {
-      redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"",
-               argv[0]);
+            redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"", argv[0]);
@@ -291,2 +276 @@ void loadServerConfigFromString(char *config) {
-          err = "Target command name already exists";
-          goto loaderr;
+                    err = "Target command name already exists"; goto loaderr;
@@ -297,2 +281 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -305 +288,3 @@ void loadServerConfigFromString(char *config) {
-    } else if (!strcasecmp(argv[0], "slowlog-log-slower-than") && argc == 2) {
+        } else if (!strcasecmp(argv[0],"slowlog-log-slower-than") &&
+                   argc == 2)
+        {
@@ -310 +295,2 @@ void loadServerConfigFromString(char *config) {
-               argc == 5) {
+                   argc == 5)
+        {
@@ -331,2 +317 @@ void loadServerConfigFromString(char *config) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -335,2 +320 @@ void loadServerConfigFromString(char *config) {
-      err = "Bad directive or wrong number of arguments";
-      goto loaderr;
+            err = "Bad directive or wrong number of arguments"; goto loaderr;
@@ -358,2 +342,2 @@ void loadServerConfig(char *filename, char *options) {
-        redisLog(REDIS_WARNING, "Fatal error, can't open config file '%s'",
-                 filename);
+                redisLog(REDIS_WARNING,
+                    "Fatal error, can't open config file '%s'", filename);
@@ -377,4 +361,2 @@ void configSetCommand(redisClient *c) {
-  redisAssertWithInfo(c, c->argv[2],
-                      c->argv[2]->encoding == REDIS_ENCODING_RAW);
-  redisAssertWithInfo(c, c->argv[2],
-                      c->argv[3]->encoding == REDIS_ENCODING_RAW);
+    redisAssertWithInfo(c,c->argv[2],c->argv[2]->encoding == REDIS_ENCODING_RAW);
+    redisAssertWithInfo(c,c->argv[2],c->argv[3]->encoding == REDIS_ENCODING_RAW);
@@ -392 +374,2 @@ void configSetCommand(redisClient *c) {
-    if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll < 0) goto badfmt;
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
+            ll < 0) goto badfmt;
@@ -412 +395,2 @@ void configSetCommand(redisClient *c) {
-    if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll <= 0) goto badfmt;
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
+            ll <= 0) goto badfmt;
@@ -415,2 +399,2 @@ void configSetCommand(redisClient *c) {
-    if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll < 0 || ll > LONG_MAX)
-      goto badfmt;
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
+            ll < 0 || ll > LONG_MAX) goto badfmt;
@@ -439 +423,2 @@ void configSetCommand(redisClient *c) {
-        addReplyError(c, "Unable to turn on AOF. Check server logs.");
+                addReplyError(c,
+                    "Unable to turn on AOF. Check server logs.");
@@ -460 +445,2 @@ void configSetCommand(redisClient *c) {
-      if (eptr[0] != '\0' || ((j & 1) == 0 && val < 1) ||
+            if (eptr[0] != '\0' ||
+                ((j & 1) == 0 && val < 1) ||
@@ -581 +567,2 @@ badfmt:
-                      (char *)o->ptr, (char *)c->argv[2]->ptr);
+            (char*)o->ptr,
+            (char*)c->argv[2]->ptr);
@@ -582,0 +570,22 @@ badfmt:
+#define config_get_string_field(_name,_var) do { \
+    if (stringmatch(pattern,_name,0)) { \
+        addReplyBulkCString(c,_name); \
+        addReplyBulkCString(c,_var ? _var : ""); \
+        matches++; \
+    } \
+} while(0);
+#define config_get_bool_field(_name,_var) do { \
+    if (stringmatch(pattern,_name,0)) { \
+        addReplyBulkCString(c,_name); \
+        addReplyBulkCString(c,_var ? "yes" : "no"); \
+        matches++; \
+    } \
+} while(0);
+#define config_get_numerical_field(_name,_var) do { \
+    if (stringmatch(pattern,_name,0)) { \
+        ll2string(buf,sizeof(buf),_var); \
+        addReplyBulkCString(c,_name); \
+        addReplyBulkCString(c,buf); \
+        matches++; \
+    } \
+} while(0);
@@ -604,4 +613,4 @@ void configGetCommand(redisClient *c) {
-  config_get_numerical_field("hash-max-zipmap-entries",
-                             server.hash_max_zipmap_entries);
-  config_get_numerical_field("hash-max-zipmap-value",
-                             server.hash_max_zipmap_value);
+    config_get_numerical_field("hash-max-ziplist-entries",
+            server.hash_max_ziplist_entries);
+    config_get_numerical_field("hash-max-ziplist-value",
+            server.hash_max_ziplist_value);
@@ -621 +630,2 @@ void configGetCommand(redisClient *c) {
-  config_get_numerical_field("slowlog-max-len", server.slowlog_max_len);
+    config_get_numerical_field("slowlog-max-len",
+            server.slowlog_max_len);
@@ -624,2 +634 @@ void configGetCommand(redisClient *c) {
-  config_get_numerical_field("repl-ping-slave-period",
-                             server.repl_ping_slave_period);
+    config_get_numerical_field("repl-ping-slave-period",server.repl_ping_slave_period);
@@ -630 +639,2 @@ void configGetCommand(redisClient *c) {
-  config_get_bool_field("slave-serve-stale-data", server.repl_serve_stale_data);
+    config_get_bool_field("slave-serve-stale-data",
+            server.repl_serve_stale_data);
@@ -643 +653,2 @@ void configGetCommand(redisClient *c) {
-    if (getcwd(buf, sizeof(buf)) == NULL) buf[0] = '\0';
+        if (getcwd(buf,sizeof(buf)) == NULL)
+            buf[0] = '\0';
@@ -651,21 +662,7 @@ void configGetCommand(redisClient *c) {
-      case REDIS_MAXMEMORY_VOLATILE_LRU:
-        s = "volatile-lru";
-        break;
-      case REDIS_MAXMEMORY_VOLATILE_TTL:
-        s = "volatile-ttl";
-        break;
-      case REDIS_MAXMEMORY_VOLATILE_RANDOM:
-        s = "volatile-random";
-        break;
-      case REDIS_MAXMEMORY_ALLKEYS_LRU:
-        s = "allkeys-lru";
-        break;
-      case REDIS_MAXMEMORY_ALLKEYS_RANDOM:
-        s = "allkeys-random";
-        break;
-      case REDIS_MAXMEMORY_NO_EVICTION:
-        s = "noeviction";
-        break;
-      default:
-        s = "unknown";
-        break;
+        case REDIS_MAXMEMORY_VOLATILE_LRU: s = "volatile-lru"; break;
+        case REDIS_MAXMEMORY_VOLATILE_TTL: s = "volatile-ttl"; break;
+        case REDIS_MAXMEMORY_VOLATILE_RANDOM: s = "volatile-random"; break;
+        case REDIS_MAXMEMORY_ALLKEYS_LRU: s = "allkeys-lru"; break;
+        case REDIS_MAXMEMORY_ALLKEYS_RANDOM: s = "allkeys-random"; break;
+        case REDIS_MAXMEMORY_NO_EVICTION: s = "noeviction"; break;
+        default: s = "unknown"; break;
@@ -680,12 +677,4 @@ void configGetCommand(redisClient *c) {
-      case AOF_FSYNC_NO:
-        policy = "no";
-        break;
-      case AOF_FSYNC_EVERYSEC:
-        policy = "everysec";
-        break;
-      case AOF_FSYNC_ALWAYS:
-        policy = "always";
-        break;
-      default:
-        policy = "unknown";
-        break;
+        case AOF_FSYNC_NO: policy = "no"; break;
+        case AOF_FSYNC_EVERYSEC: policy = "everysec"; break;
+        case AOF_FSYNC_ALWAYS: policy = "always"; break;
+        default: policy = "unknown"; break;
@@ -701 +690,2 @@ void configGetCommand(redisClient *c) {
-      buf = sdscatprintf(buf, "%ld %d", server.saveparams[j].seconds,
+            buf = sdscatprintf(buf,"%ld %d",
+                    server.saveparams[j].seconds,
@@ -703 +693,2 @@ void configGetCommand(redisClient *c) {
-      if (j != server.saveparamslen - 1) buf = sdscatlen(buf, " ", 1);
+            if (j != server.saveparamslen-1)
+                buf = sdscatlen(buf," ",1);
@@ -710,134 +700,0 @@ void configGetCommand(redisClient *c) {
-<<<<<<< HEAD
-|||||||
-  if (stringmatch(pattern, "auto-aof-rewrite-percentage", 0)) {
-    addReplyBulkCString(c, "auto-aof-rewrite-percentage");
-    addReplyBulkLongLong(c, server.aof_rewrite_perc);
-    matches++;
-  }
-  if (stringmatch(pattern, "auto-aof-rewrite-min-size", 0)) {
-    addReplyBulkCString(c, "auto-aof-rewrite-min-size");
-    addReplyBulkLongLong(c, server.aof_rewrite_min_size);
-    matches++;
-  }
-  if (stringmatch(pattern, "slave-serve-stale-data", 0)) {
-    addReplyBulkCString(c, "slave-serve-stale-data");
-    addReplyBulkCString(c, server.repl_serve_stale_data ? "yes" : "no");
-    matches++;
-  }
-  if (stringmatch(pattern, "hash-max-zipmap-entries", 0)) {
-    addReplyBulkCString(c, "hash-max-zipmap-entries");
-    addReplyBulkLongLong(c, server.hash_max_zipmap_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "hash-max-zipmap-value", 0)) {
-    addReplyBulkCString(c, "hash-max-zipmap-value");
-    addReplyBulkLongLong(c, server.hash_max_zipmap_value);
-    matches++;
-  }
-  if (stringmatch(pattern, "list-max-ziplist-entries", 0)) {
-    addReplyBulkCString(c, "list-max-ziplist-entries");
-    addReplyBulkLongLong(c, server.list_max_ziplist_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "list-max-ziplist-value", 0)) {
-    addReplyBulkCString(c, "list-max-ziplist-value");
-    addReplyBulkLongLong(c, server.list_max_ziplist_value);
-    matches++;
-  }
-  if (stringmatch(pattern, "set-max-intset-entries", 0)) {
-    addReplyBulkCString(c, "set-max-intset-entries");
-    addReplyBulkLongLong(c, server.set_max_intset_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "zset-max-ziplist-entries", 0)) {
-    addReplyBulkCString(c, "zset-max-ziplist-entries");
-    addReplyBulkLongLong(c, server.zset_max_ziplist_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "zset-max-ziplist-value", 0)) {
-    addReplyBulkCString(c, "zset-max-ziplist-value");
-    addReplyBulkLongLong(c, server.zset_max_ziplist_value);
-    matches++;
-  }
-  if (stringmatch(pattern, "lua-time-limit", 0)) {
-    addReplyBulkCString(c, "lua-time-limit");
-    addReplyBulkLongLong(c, server.lua_time_limit);
-    matches++;
-  }
-  if (stringmatch(pattern, "slowlog-log-slower-than", 0)) {
-    addReplyBulkCString(c, "slowlog-log-slower-than");
-    addReplyBulkLongLong(c, server.slowlog_log_slower_than);
-    matches++;
-  }
-  if (stringmatch(pattern, "slowlog-max-len", 0)) {
-    addReplyBulkCString(c, "slowlog-max-len");
-    addReplyBulkLongLong(c, server.slowlog_max_len);
-    matches++;
-  }
-=======
-  if (stringmatch(pattern, "auto-aof-rewrite-percentage", 0)) {
-    addReplyBulkCString(c, "auto-aof-rewrite-percentage");
-    addReplyBulkLongLong(c, server.aof_rewrite_perc);
-    matches++;
-  }
-  if (stringmatch(pattern, "auto-aof-rewrite-min-size", 0)) {
-    addReplyBulkCString(c, "auto-aof-rewrite-min-size");
-    addReplyBulkLongLong(c, server.aof_rewrite_min_size);
-    matches++;
-  }
-  if (stringmatch(pattern, "slave-serve-stale-data", 0)) {
-    addReplyBulkCString(c, "slave-serve-stale-data");
-    addReplyBulkCString(c, server.repl_serve_stale_data ? "yes" : "no");
-    matches++;
-  }
-  if (stringmatch(pattern, "hash-max-ziplist-entries", 0)) {
-    addReplyBulkCString(c, "hash-max-ziplist-entries");
-    addReplyBulkLongLong(c, server.hash_max_ziplist_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "hash-max-ziplist-value", 0)) {
-    addReplyBulkCString(c, "hash-max-ziplist-value");
-    addReplyBulkLongLong(c, server.hash_max_ziplist_value);
-    matches++;
-  }
-  if (stringmatch(pattern, "list-max-ziplist-entries", 0)) {
-    addReplyBulkCString(c, "list-max-ziplist-entries");
-    addReplyBulkLongLong(c, server.list_max_ziplist_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "list-max-ziplist-value", 0)) {
-    addReplyBulkCString(c, "list-max-ziplist-value");
-    addReplyBulkLongLong(c, server.list_max_ziplist_value);
-    matches++;
-  }
-  if (stringmatch(pattern, "set-max-intset-entries", 0)) {
-    addReplyBulkCString(c, "set-max-intset-entries");
-    addReplyBulkLongLong(c, server.set_max_intset_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "zset-max-ziplist-entries", 0)) {
-    addReplyBulkCString(c, "zset-max-ziplist-entries");
-    addReplyBulkLongLong(c, server.zset_max_ziplist_entries);
-    matches++;
-  }
-  if (stringmatch(pattern, "zset-max-ziplist-value", 0)) {
-    addReplyBulkCString(c, "zset-max-ziplist-value");
-    addReplyBulkLongLong(c, server.zset_max_ziplist_value);
-    matches++;
-  }
-  if (stringmatch(pattern, "lua-time-limit", 0)) {
-    addReplyBulkCString(c, "lua-time-limit");
-    addReplyBulkLongLong(c, server.lua_time_limit);
-    matches++;
-  }
-  if (stringmatch(pattern, "slowlog-log-slower-than", 0)) {
-    addReplyBulkCString(c, "slowlog-log-slower-than");
-    addReplyBulkLongLong(c, server.slowlog_log_slower_than);
-    matches++;
-  }
-  if (stringmatch(pattern, "slowlog-max-len", 0)) {
-    addReplyBulkCString(c, "slowlog-max-len");
-    addReplyBulkLongLong(c, server.slowlog_max_len);
-    matches++;
-  }
->>>>>>> d3ea4c86a8c8efb2ab922db89af915833e437e7e
@@ -847,15 +704,5 @@ void configGetCommand(redisClient *c) {
-      case REDIS_WARNING:
-        s = "warning";
-        break;
-      case REDIS_VERBOSE:
-        s = "verbose";
-        break;
-      case REDIS_NOTICE:
-        s = "notice";
-        break;
-      case REDIS_DEBUG:
-        s = "debug";
-        break;
-      default:
-        s = "unknown";
-        break;
+        case REDIS_WARNING: s = "warning"; break;
+        case REDIS_VERBOSE: s = "verbose"; break;
+        case REDIS_NOTICE: s = "notice"; break;
+        case REDIS_DEBUG: s = "debug"; break;
+        default: s = "unknown"; break;
@@ -871 +718,2 @@ void configGetCommand(redisClient *c) {
-      buf = sdscatprintf(buf, "%s %llu %llu %ld", getClientLimitClassName(j),
+            buf = sdscatprintf(buf,"%s %llu %llu %ld",
+                    getClientLimitClassName(j),
@@ -875 +723,2 @@ void configGetCommand(redisClient *c) {
-      if (j != REDIS_CLIENT_LIMIT_NUM_CLASSES - 1) buf = sdscatlen(buf, " ", 1);
+            if (j != REDIS_CLIENT_LIMIT_NUM_CLASSES-1)
+                buf = sdscatlen(buf," ",1);
@@ -893 +742,2 @@ void configGetCommand(redisClient *c) {
-      snprintf(buf, sizeof(buf), "%s %d", server.masterhost, server.masterport);
+            snprintf(buf,sizeof(buf),"%s %d",
+                server.masterhost, server.masterport);
@@ -918 +768,2 @@ void configCommand(redisClient *c) {
-    addReplyError(c, "CONFIG subcommand must be one of GET, SET, RESETSTAT");
+        addReplyError(c,
+            "CONFIG subcommand must be one of GET, SET, RESETSTAT");
