diff --git a/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_conflict.c b/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_merged.c
index a7a210c..5f1b40c 100644
--- a/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_conflict.c
+++ b/./output/redis/2b00385d-69ef89f2-68254919/src@t_zset.no_comments_merged.c
@@ -236,24 +235,0 @@ void zaddGenericCommand(redisClient *c, robj *key, robj *ele, double score, int
-<<<<<<< HEAD
-        de = dictFind(zs->dict,ele);
-        if (de) {
-            double *oldscore = dictGetEntryVal(de);
-            *score = *oldscore + scoreval;
-        } else {
-            *score = scoreval;
-        }
-        if (isnan(*score)) {
-            addReplyError(c,"resulting score is not a number (NaN)");
-            zfree(score);
-||||||| 682549192
-        de = dictFind(zs->dict,ele);
-        if (de) {
-            double *oldscore = dictGetEntryVal(de);
-            *score = *oldscore + scoreval;
-        } else {
-            *score = scoreval;
-        }
-        if (isnan(*score)) {
-            addReplySds(c,
-                sdsnew("-ERR resulting score is not a number (NaN)\r\n"));
-            zfree(score);
-=======
@@ -264,3 +240 @@ void zaddGenericCommand(redisClient *c, robj *key, robj *ele, double score, int
-            addReplySds(c,
-                sdsnew("-ERR resulting score is not a number (NaN)\r\n"));
->>>>>>> 69ef89f2
+            addReplyError(c,"resulting score is not a number (NaN)");
