diff --git a/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_conflict.h b/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_merged.h
index 2086d75..075777f 100644
--- a/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_conflict.h
+++ b/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_merged.h
@@ -68 +67,0 @@ class Artd : public aidl::com::android::server::art::BnArtd {
-<<<<<<< HEAD
@@ -74,5 +72,0 @@ class Artd : public aidl::com::android::server::art::BnArtd {
-||||||| 0b3fa1ed20
-      bool* _aidl_return) override;
-=======
-      aidl::com::android::server::art::CopyAndRewriteProfileResult* _aidl_return) override;
->>>>>>> 085665c2
