diff --git a/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_mergegen.h b/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_merged.h
index 82ceee1..075777f 100644
--- a/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_mergegen.h
+++ b/./output/art/dd9ce37d-085665c2-0b3fa1ed/artd@artd.no_comments_merged.h
@@ -68,0 +69,4 @@ aidl::com::android::server::art::CopyAndRewriteProfileResult* _aidl_return) over
+  ndk::ScopedAStatus copyAndRewriteEmbeddedProfile(
+      aidl::com::android::server::art::OutputProfile* in_dst,
+      const std::string& in_dexFile,
+      aidl::com::android::server::art::CopyAndRewriteProfileResult* _aidl_return) override;
