diff --git a/./output/tmux/4c07367b-63aa9686-e2f6f58f/tmux.no_comments_mergegen.h b/./output/tmux/4c07367b-63aa9686-e2f6f58f/tmux.no_comments_merged.h
index 87b15f7..21d327d 100644
--- a/./output/tmux/4c07367b-63aa9686-e2f6f58f/tmux.no_comments_mergegen.h
+++ b/./output/tmux/4c07367b-63aa9686-e2f6f58f/tmux.no_comments_merged.h
@@ -9 +9,5 @@
-#include "compat.h" #include "tmux-protocol.h"
+#ifdef HAVE_UTEMPTER
+#include <utempter.h>
+#endif
+#include "compat.h"
+#include "tmux-protocol.h"
