diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_merged.c
index d37e002..bb113e2 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_merged.c
@@ -3 +2,0 @@
-#include <paths.h>
@@ -6 +4,0 @@
-#include <util.h>
@@ -10 +8,3 @@
-static void spawn_log(const char *from, struct spawn_context *sc) {
+static void
+spawn_log(const char *from, struct spawn_context *sc)
+{
@@ -28 +28,3 @@ static void spawn_log(const char *from, struct spawn_context *sc) {
-struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
+struct winlink *
+spawn_window(struct spawn_context *sc, char **cause)
+{
@@ -42 +44,2 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-        if (wp->fd != -1) break;
+    if (wp->fd != -1)
+     break;
@@ -45 +48,2 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-        xasprintf(cause, "window %s:%d still active", s->name, sc->wl->idx);
+    xasprintf(cause, "window %s:%d still active",
+        s->name, sc->wl->idx);
@@ -76 +80,2 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-    if (idx == -1) idx = -1 - options_get_number(s->options, "base-index");
+  if (idx == -1)
+   idx = -1 - options_get_number(s->options, "base-index");
@@ -81 +86,2 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-    default_window_size(sc->tc, s, NULL, &sx, &sy, &xpixel, &ypixel, -1);
+  default_window_size(sc->tc, s, NULL, &sx, &sy, &xpixel, &ypixel,
+      -1);
@@ -87 +93,2 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-    if (s->curw == NULL) s->curw = sc->wl;
+  if (s->curw == NULL)
+   s->curw = sc->wl;
@@ -96 +103,2 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-    if (~sc->flags & SPAWN_RESPAWN) winlink_remove(&s->windows, sc->wl);
+  if (~sc->flags & SPAWN_RESPAWN)
+   winlink_remove(&s->windows, sc->wl);
@@ -101 +109,2 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-      w->name = format_single(item, sc->name, c, s, NULL, NULL);
+   w->name = format_single(item, sc->name, c, s, NULL,
+       NULL);
@@ -106,2 +115,4 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-  if (~sc->flags & SPAWN_DETACHED) session_select(s, sc->wl->idx);
-  if (~sc->flags & SPAWN_RESPAWN) notify_session_window("window-linked", s, w);
+ if (~sc->flags & SPAWN_DETACHED)
+  session_select(s, sc->wl->idx);
+ if (~sc->flags & SPAWN_RESPAWN)
+  notify_session_window("window-linked", s, w);
@@ -111 +122,3 @@ struct winlink *spawn_window(struct spawn_context *sc, char **cause) {
-struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
+struct window_pane *
+spawn_pane(struct spawn_context *sc, char **cause)
+{
@@ -140 +153,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-      xasprintf(cause, "pane %s:%d.%u still active", s->name, sc->wl->idx, idx);
+   xasprintf(cause, "pane %s:%d.%u still active",
+       s->name, sc->wl->idx, idx);
@@ -184 +198,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-  if (sc->environ != NULL) environ_copy(sc->environ, child);
+ if (sc->environ != NULL)
+  environ_copy(sc->environ, child);
@@ -188 +203,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-    if (ee != NULL) environ_set(child, "PATH", 0, "%s", ee->value);
+  if (ee != NULL)
+   environ_set(child, "PATH", 0, "%s", ee->value);
@@ -194 +210,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-    if (!checkshell(tmp)) tmp = _PATH_BSHELL;
+  if (!checkshell(tmp))
+   tmp = _PATH_BSHELL;
@@ -205 +222,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-  if (cwd != NULL) log_debug("%s: cwd=%s", __func__, cwd);
+ if (cwd != NULL)
+  log_debug("%s: cwd=%s", __func__, cwd);
@@ -234 +252,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-  if (new_wp->pid != 0) goto complete;
+ if (new_wp->pid != 0)
+  goto complete;
@@ -236 +255,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-    if ((tmp = find_home()) == NULL || chdir(tmp) != 0) chdir("/");
+  if ((tmp = find_home()) == NULL || chdir(tmp) != 0)
+   chdir("/");
@@ -238,2 +258,4 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-  if (tcgetattr(STDIN_FILENO, &now) != 0) _exit(1);
-  if (s->tio != NULL) memcpy(now.c_cc, s->tio->c_cc, sizeof now.c_cc);
+ if (tcgetattr(STDIN_FILENO, &now) != 0)
+  _exit(1);
+ if (s->tio != NULL)
+  memcpy(now.c_cc, s->tio->c_cc, sizeof now.c_cc);
@@ -245 +267,2 @@ struct window_pane *spawn_pane(struct spawn_context *sc, char **cause) {
-  if (tcsetattr(STDIN_FILENO, TCSANOW, &now) != 0) _exit(1);
+ if (tcsetattr(STDIN_FILENO, TCSANOW, &now) != 0)
+  _exit(1);
@@ -273 +295,0 @@ complete:
-<<<<<<< HEAD
@@ -283,4 +304,0 @@ complete:
-|||||||
-  new_wp->pipe_off = 0;
-=======
->>>>>>> 31e3f2d530090793815d145a16a1ce3b469c4266
@@ -291 +309,2 @@ complete:
-  if (sc->flags & SPAWN_RESPAWN) return (new_wp);
+ if (sc->flags & SPAWN_RESPAWN)
+  return (new_wp);
@@ -298 +317,2 @@ complete:
-  if (~sc->flags & SPAWN_NONOTIFY) notify_window("window-layout-changed", w);
+ if (~sc->flags & SPAWN_NONOTIFY)
+  notify_window("window-layout-changed", w);
