--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_merged.c
@@ -3 +2,0 @@
-#include <paths.h>
@@ -6 +4,0 @@
-#include <util.h>
@@ -273 +295,0 @@ complete:
-<<<<<<< HEAD
@@ -283,4 +304,0 @@ complete:
-|||||||
-  new_wp->pipe_off = 0;
-=======
->>>>>>> 31e3f2d530090793815d145a16a1ce3b469c4266
