--- a/home/whalien/codebase/python/mergebot-eva/output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_merged.h
@@ -119,6 +119,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER \
-  10
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER \
-  100
-#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES \
-  (100 << 20)
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER 10
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER 100
+#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES (100<<20)
@@ -126,6 +123,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX \
-  75
-#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS \
-  1000
-#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN \
-  (512ll * 1024 * 1024)
+#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX 75
+#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS 1000
+#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN (512ll*1024*1024)
@@ -134,2 +128 @@ typedef long long mstime_t;
-#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC \
-  25
+#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC 25
@@ -148 +140,0 @@ typedef long long mstime_t;
-#define REDIS_AUTOSYNC_BYTES (1024 * 1024 * 32)
@@ -150 +142 @@ typedef long long mstime_t;
-#define AOF_AUTOSYNC_BYTES (1024 * 1024 * 32)
+#define REDIS_AUTOSYNC_BYTES (1024*1024*32)
@@ -175,2 +167 @@ typedef long long mstime_t;
-#define CLIENT_MONITOR \
-  (1 << 2)
+#define CLIENT_MONITOR (1<<2)
@@ -178,2 +169 @@ typedef long long mstime_t;
-#define CLIENT_BLOCKED \
-  (1 << 4)
+#define CLIENT_BLOCKED (1<<4)
@@ -181,4 +171,3 @@ typedef long long mstime_t;
-#define CLIENT_CLOSE_AFTER_REPLY \
-  (1 << 6)
-#define CLIENT_UNBLOCKED \
-  (1 << 7)
+#define CLIENT_CLOSE_AFTER_REPLY (1<<6)
+#define CLIENT_UNBLOCKED (1<<7)
+#define CLIENT_LUA (1<<8)
@@ -187,6 +176,3 @@ typedef long long mstime_t;
-#define CLIENT_UNIX_SOCKET \
-  (1 << 11)
-#define CLIENT_DIRTY_EXEC \
-  (1 << 12)
-#define CLIENT_MASTER_FORCE_REPLY \
-  (1 << 13)
+#define CLIENT_UNIX_SOCKET (1<<11)
+#define CLIENT_DIRTY_EXEC (1<<12)
+#define CLIENT_MASTER_FORCE_REPLY (1<<13)
@@ -201,4 +187,3 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_WRITE \
-  (1 << 21)
-#define CLIENT_REPLY_SKIP_NEXT \
-  (1 << 23)
+#define CLIENT_PENDING_WRITE (1<<21)
+#define CLIENT_REPLY_OFF (1<<22)
+#define CLIENT_REPLY_SKIP_NEXT (1<<23)
@@ -222,2 +207 @@ typedef long long mstime_t;
-#define CLIENT_TYPE_OBUF_COUNT \
-  3
+#define CLIENT_TYPE_OBUF_COUNT 3
@@ -262,0 +247 @@ typedef long long mstime_t;
+#define UNUSED(V) ((void) V)
@@ -291,4 +276,2 @@ typedef long long mstime_t;
-#define MAXMEMORY_ALLKEYS_LRU \
-  ((4 << 8) | MAXMEMORY_FLAG_LRU | MAXMEMORY_FLAG_ALLKEYS)
-#define MAXMEMORY_ALLKEYS_LFU \
-  ((5 << 8) | MAXMEMORY_FLAG_LFU | MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LRU ((4<<8)|MAXMEMORY_FLAG_LRU|MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LFU ((5<<8)|MAXMEMORY_FLAG_LFU|MAXMEMORY_FLAG_ALLKEYS)
@@ -302,2 +285,2 @@ typedef long long mstime_t;
-#define SHUTDOWN_SAVE \
-  1
+#define SHUTDOWN_SAVE 1
+#define SHUTDOWN_NOSAVE 2
@@ -328,3 +311 @@ typedef long long mstime_t;
-#define NOTIFY_ALL \
-  (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | \
-   NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM)
+#define NOTIFY_ALL (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM)
@@ -331,0 +313,4 @@ typedef long long mstime_t;
+#define run_with_period(_ms_) if ((_ms_ <= 1000/server.hz) || !(server.cronloops%((_ms_)/(1000/server.hz))))
+#define serverAssertWithInfo(_c,_o,_e) ((_e)?(void)0 : (_serverAssertWithInfo(_c,_o,#_e,__FILE__,__LINE__),_exit(1)))
+#define serverAssert(_e) ((_e)?(void)0 : (_serverAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define serverPanic(...) _serverPanic(__FILE__,__LINE__,__VA_ARGS__),_exit(1)
@@ -340,0 +326,2 @@ typedef long long mstime_t;
+#define REDISMODULE_TYPE_ENCVER(id) (id & REDISMODULE_TYPE_ENCVER_MASK)
+#define REDISMODULE_TYPE_SIGN(id) ((id & ~((uint64_t)REDISMODULE_TYPE_ENCVER_MASK)) >>REDISMODULE_TYPE_ENCVER_BITS)
@@ -376,0 +362,8 @@ typedef struct RedisModuleIO {
+#define moduleInitIOContext(iovar,mtype,rioptr) do { \
+    iovar.rio = rioptr; \
+    iovar.type = mtype; \
+    iovar.bytes = 0; \
+    iovar.error = 0; \
+    iovar.ver = 0; \
+    iovar.ctx = NULL; \
+} while(0);
@@ -380,0 +374,4 @@ typedef struct RedisModuleDigest {
+#define moduleInitDigestContext(mdvar) do { \
+    memset(mdvar.o,0,sizeof(mdvar.o)); \
+    memset(mdvar.x,0,sizeof(mdvar.x)); \
+} while(0);
@@ -402,0 +400,6 @@ typedef struct redisObject {
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = OBJ_STRING; \
+    _var.encoding = OBJ_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0)
@@ -583,2 +585 @@ typedef struct rdbSaveInfo {
-#define RDB_SAVE_INFO_INIT \
-  { -1, 0, "000000000000000000000000000000", -1 }
+#define RDB_SAVE_INFO_INIT {-1,0,"000000000000000000000000000000",-1}
@@ -950,0 +947 @@ typedef struct {
+#include "stream.h"
@@ -1132,0 +1123 @@ unsigned long long estimateObjectIdleTime(robj *o);
+#define sdsEncodedObject(objptr) (objptr->encoding == OBJ_ENCODING_RAW || objptr->encoding == OBJ_ENCODING_EMBSTR)
@@ -1168,0 +1158 @@ void stopLoading(void);
+#include "rdb.h"
@@ -1281,2 +1266 @@ void freeMemoryOverheadData(struct redisMemOverhead *mh);
-#define RESTART_SERVER_CONFIG_REWRITE \
-  (1 << 1)
+#define RESTART_SERVER_CONFIG_REWRITE (1<<1)
@@ -1634,2 +1606,7 @@ void xtrimCommand(client *c);
-void _serverAssertWithInfo(const client *c, const robj *o, const char *estr,
-                           const char *file, int line);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
+void _serverAssertWithInfo(const client *c, const robj *o, const char *estr, const char *file, int line);
@@ -1648,0 +1626,4 @@ void xorDigest(unsigned char *digest, void *ptr, size_t len);
+#define redisDebug(fmt,...) \
+    printf("DEBUG %s:%d > " fmt "\n", __FILE__, __LINE__, __VA_ARGS__)
+#define redisDebugMark() \
+    printf("-- MARK %s:%d --\n", __FILE__, __LINE__)
