diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_merged.h
index 5b3dd1f..5eb84a2 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/7de1ada0-2edcafb3-3d5e2c62/src@server.no_comments_merged.h
@@ -119,6 +119,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER \
-  10
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER \
-  100
-#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES \
-  (100 << 20)
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER 10
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER 100
+#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES (100<<20)
@@ -126,6 +123,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX \
-  75
-#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS \
-  1000
-#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN \
-  (512ll * 1024 * 1024)
+#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX 75
+#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS 1000
+#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN (512ll*1024*1024)
@@ -134,2 +128 @@ typedef long long mstime_t;
-#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC \
-  25
+#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC 25
@@ -148 +140,0 @@ typedef long long mstime_t;
-#define REDIS_AUTOSYNC_BYTES (1024 * 1024 * 32)
@@ -150 +142 @@ typedef long long mstime_t;
-#define AOF_AUTOSYNC_BYTES (1024 * 1024 * 32)
+#define REDIS_AUTOSYNC_BYTES (1024*1024*32)
@@ -175,2 +167 @@ typedef long long mstime_t;
-#define CLIENT_MONITOR \
-  (1 << 2)
+#define CLIENT_MONITOR (1<<2)
@@ -178,2 +169 @@ typedef long long mstime_t;
-#define CLIENT_BLOCKED \
-  (1 << 4)
+#define CLIENT_BLOCKED (1<<4)
@@ -181,4 +171,3 @@ typedef long long mstime_t;
-#define CLIENT_CLOSE_AFTER_REPLY \
-  (1 << 6)
-#define CLIENT_UNBLOCKED \
-  (1 << 7)
+#define CLIENT_CLOSE_AFTER_REPLY (1<<6)
+#define CLIENT_UNBLOCKED (1<<7)
+#define CLIENT_LUA (1<<8)
@@ -187,6 +176,3 @@ typedef long long mstime_t;
-#define CLIENT_UNIX_SOCKET \
-  (1 << 11)
-#define CLIENT_DIRTY_EXEC \
-  (1 << 12)
-#define CLIENT_MASTER_FORCE_REPLY \
-  (1 << 13)
+#define CLIENT_UNIX_SOCKET (1<<11)
+#define CLIENT_DIRTY_EXEC (1<<12)
+#define CLIENT_MASTER_FORCE_REPLY (1<<13)
@@ -201,4 +187,3 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_WRITE \
-  (1 << 21)
-#define CLIENT_REPLY_SKIP_NEXT \
-  (1 << 23)
+#define CLIENT_PENDING_WRITE (1<<21)
+#define CLIENT_REPLY_OFF (1<<22)
+#define CLIENT_REPLY_SKIP_NEXT (1<<23)
@@ -222,2 +207 @@ typedef long long mstime_t;
-#define CLIENT_TYPE_OBUF_COUNT \
-  3
+#define CLIENT_TYPE_OBUF_COUNT 3
@@ -262,0 +247 @@ typedef long long mstime_t;
+#define UNUSED(V) ((void) V)
@@ -291,4 +276,2 @@ typedef long long mstime_t;
-#define MAXMEMORY_ALLKEYS_LRU \
-  ((4 << 8) | MAXMEMORY_FLAG_LRU | MAXMEMORY_FLAG_ALLKEYS)
-#define MAXMEMORY_ALLKEYS_LFU \
-  ((5 << 8) | MAXMEMORY_FLAG_LFU | MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LRU ((4<<8)|MAXMEMORY_FLAG_LRU|MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LFU ((5<<8)|MAXMEMORY_FLAG_LFU|MAXMEMORY_FLAG_ALLKEYS)
@@ -302,2 +285,2 @@ typedef long long mstime_t;
-#define SHUTDOWN_SAVE \
-  1
+#define SHUTDOWN_SAVE 1
+#define SHUTDOWN_NOSAVE 2
@@ -328,3 +311 @@ typedef long long mstime_t;
-#define NOTIFY_ALL \
-  (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | \
-   NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM)
+#define NOTIFY_ALL (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM)
@@ -331,0 +313,4 @@ typedef long long mstime_t;
+#define run_with_period(_ms_) if ((_ms_ <= 1000/server.hz) || !(server.cronloops%((_ms_)/(1000/server.hz))))
+#define serverAssertWithInfo(_c,_o,_e) ((_e)?(void)0 : (_serverAssertWithInfo(_c,_o,#_e,__FILE__,__LINE__),_exit(1)))
+#define serverAssert(_e) ((_e)?(void)0 : (_serverAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define serverPanic(...) _serverPanic(__FILE__,__LINE__,__VA_ARGS__),_exit(1)
@@ -340,0 +326,2 @@ typedef long long mstime_t;
+#define REDISMODULE_TYPE_ENCVER(id) (id & REDISMODULE_TYPE_ENCVER_MASK)
+#define REDISMODULE_TYPE_SIGN(id) ((id & ~((uint64_t)REDISMODULE_TYPE_ENCVER_MASK)) >>REDISMODULE_TYPE_ENCVER_BITS)
@@ -348,4 +335,2 @@ typedef void (*moduleTypeSaveFunc)(struct RedisModuleIO *io, void *value);
-typedef void (*moduleTypeRewriteFunc)(struct RedisModuleIO *io,
-                                      struct redisObject *key, void *value);
-typedef void (*moduleTypeDigestFunc)(struct RedisModuleDigest *digest,
-                                     void *value);
+typedef void (*moduleTypeRewriteFunc)(struct RedisModuleIO *io, struct redisObject *key, void *value);
+typedef void (*moduleTypeDigestFunc)(struct RedisModuleDigest *digest, void *value);
@@ -376,0 +362,8 @@ typedef struct RedisModuleIO {
+#define moduleInitIOContext(iovar,mtype,rioptr) do { \
+    iovar.rio = rioptr; \
+    iovar.type = mtype; \
+    iovar.bytes = 0; \
+    iovar.error = 0; \
+    iovar.ver = 0; \
+    iovar.ctx = NULL; \
+} while(0);
@@ -380,0 +374,4 @@ typedef struct RedisModuleDigest {
+#define moduleInitDigestContext(mdvar) do { \
+    memset(mdvar.o,0,sizeof(mdvar.o)); \
+    memset(mdvar.x,0,sizeof(mdvar.x)); \
+} while(0);
@@ -402,0 +400,6 @@ typedef struct redisObject {
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = OBJ_STRING; \
+    _var.encoding = OBJ_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0)
@@ -412,2 +415 @@ typedef struct redisDb {
-  list *
-      defrag_later;
+    list *defrag_later;
@@ -502,5 +504,5 @@ struct sharedObjectsStruct {
-      *colon, *nullbulk, *nullmultibulk, *queued, *emptymultibulk,
-      *wrongtypeerr, *nokeyerr, *syntaxerr, *sameobjecterr, *outofrangeerr,
-      *noscripterr, *loadingerr, *slowscripterr, *bgsaveerr, *masterdownerr,
-      *roslaveerr, *execaborterr, *noautherr, *noreplicaserr, *busykeyerr,
-      *oomerr, *plus, *messagebulk, *pmessagebulk, *subscribebulk,
+    *colon, *nullbulk, *nullmultibulk, *queued,
+    *emptymultibulk, *wrongtypeerr, *nokeyerr, *syntaxerr, *sameobjecterr,
+    *outofrangeerr, *noscripterr, *loadingerr, *slowscripterr, *bgsaveerr,
+    *masterdownerr, *roslaveerr, *execaborterr, *noautherr, *noreplicaserr,
+    *busykeyerr, *oomerr, *plus, *messagebulk, *pmessagebulk, *subscribebulk,
@@ -509 +511,2 @@ struct sharedObjectsStruct {
-      *select[PROTO_SHARED_SELECT_CMDS], *integers[OBJ_SHARED_INTEGERS],
+    *select[PROTO_SHARED_SELECT_CMDS],
+    *integers[OBJ_SHARED_INTEGERS],
@@ -537,2 +540 @@ typedef struct clientBufferLimitsConfig {
-extern clientBufferLimitsConfig
-    clientBufferLimitsDefaults[CLIENT_TYPE_OBUF_COUNT];
+extern clientBufferLimitsConfig clientBufferLimitsDefaults[CLIENT_TYPE_OBUF_COUNT];
@@ -583,2 +585 @@ typedef struct rdbSaveInfo {
-#define RDB_SAVE_INFO_INIT \
-  { -1, 0, "000000000000000000000000000000", -1 }
+#define RDB_SAVE_INFO_INIT {-1,0,"000000000000000000000000000000",-1}
@@ -652,3 +653,4 @@ struct redisServer {
-  struct redisCommand *delCommand, *multiCommand, *lpushCommand, *lpopCommand,
-      *rpopCommand, *zpopminCommand, *zpopmaxCommand, *sremCommand,
-      *execCommand, *expireCommand, *pexpireCommand, *xclaimCommand;
+    struct redisCommand *delCommand, *multiCommand, *lpushCommand,
+                        *lpopCommand, *rpopCommand, *zpopminCommand,
+                        *zpopmaxCommand, *sremCommand, *execCommand,
+                        *expireCommand, *pexpireCommand, *xclaimCommand;
@@ -665,6 +667,3 @@ struct redisServer {
-  long long
-      stat_active_defrag_misses;
-  long long
-      stat_active_defrag_key_hits;
-  long long
-      stat_active_defrag_key_misses;
+    long long stat_active_defrag_misses;
+    long long stat_active_defrag_key_hits;
+    long long stat_active_defrag_key_misses;
@@ -699,2 +698 @@ struct redisServer {
-  size_t
-      active_defrag_ignore_bytes;
+    size_t active_defrag_ignore_bytes;
@@ -705,2 +703 @@ struct redisServer {
-  unsigned long
-      active_defrag_max_scan_fields;
+    unsigned long active_defrag_max_scan_fields;
@@ -899,2 +896 @@ typedef void redisCommandProc(client *c);
-typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc,
-                              int *numkeys);
+typedef int *redisGetKeysProc(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -950,0 +947 @@ typedef struct {
+#include "stream.h"
@@ -971,2 +968 @@ void moduleLoadFromQueue(void);
-int *moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv, int argc,
-                                int *numkeys);
+int *moduleGetCommandKeysViaAPI(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -979,2 +975 @@ void moduleBlockedClientTimedOut(client *c);
-void moduleBlockedClientPipeReadable(aeEventLoop *el, int fd, void *privdata,
-                                     int mask);
+void moduleBlockedClientPipeReadable(aeEventLoop *el, int fd, void *privdata, int mask);
@@ -984,2 +979 @@ void moduleReleaseGIL(void);
-void moduleNotifyKeyspaceEvent(int type, const char *event, robj *key,
-                               int dbid);
+void moduleNotifyKeyspaceEvent(int type, const char *event, robj *key, int dbid);
@@ -1065,2 +1059 @@ unsigned long listTypeLength(const robj *subject);
-listTypeIterator *listTypeInitIterator(robj *subject, long index,
-                                       unsigned char direction);
+listTypeIterator *listTypeInitIterator(robj *subject, long index, unsigned char direction);
@@ -1119,4 +1112,2 @@ int checkType(client *c, robj *o, int type);
-int getLongLongFromObjectOrReply(client *c, robj *o, long long *target,
-                                 const char *msg);
-int getDoubleFromObjectOrReply(client *c, robj *o, double *target,
-                               const char *msg);
+int getLongLongFromObjectOrReply(client *c, robj *o, long long *target, const char *msg);
+int getDoubleFromObjectOrReply(client *c, robj *o, double *target, const char *msg);
@@ -1126,2 +1117 @@ int getLongDoubleFromObject(robj *o, long double *target);
-int getLongDoubleFromObjectOrReply(client *c, robj *o, long double *target,
-                                   const char *msg);
+int getLongDoubleFromObjectOrReply(client *c, robj *o, long double *target, const char *msg);
@@ -1132,0 +1123 @@ unsigned long long estimateObjectIdleTime(robj *o);
+#define sdsEncodedObject(objptr) (objptr->encoding == OBJ_ENCODING_RAW || objptr->encoding == OBJ_ENCODING_EMBSTR)
@@ -1137,4 +1128,2 @@ void replicationFeedSlaves(list *slaves, int dictid, robj **argv, int argc);
-void replicationFeedSlavesFromMasterStream(list *slaves, char *buf,
-                                           size_t buflen);
-void replicationFeedMonitors(client *c, list *monitors, int dictid, robj **argv,
-                             int argc);
+void replicationFeedSlavesFromMasterStream(list *slaves, char *buf, size_t buflen);
+void replicationFeedMonitors(client *c, list *monitors, int dictid, robj **argv, int argc);
@@ -1168,0 +1158 @@ void stopLoading(void);
+#include "rdb.h"
@@ -1171,2 +1161 @@ void flushAppendOnlyFile(int force);
-void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv,
-                        int argc);
+void feedAppendOnlyFile(struct redisCommand *cmd, int dictid, robj **argv, int argc);
@@ -1223,2 +1212 @@ int zsetDel(robj *zobj, sds ele);
-void genericZpopCommand(client *c, robj **keyv, int keyc, int where,
-                        int emitkey, robj *countarg);
+void genericZpopCommand(client *c, robj **keyv, int keyc, int where, int emitkey, robj *countarg);
@@ -1238,2 +1226 @@ int zslLexValueLteMax(sds value, zlexrangespec *spec);
-int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree,
-                      float *level);
+int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *level);
@@ -1247,4 +1234,2 @@ void call(client *c, int flags);
-void propagate(struct redisCommand *cmd, int dbid, robj **argv, int argc,
-               int flags);
-void alsoPropagate(struct redisCommand *cmd, int dbid, robj **argv, int argc,
-                   int target);
+void propagate(struct redisCommand *cmd, int dbid, robj **argv, int argc, int flags);
+void alsoPropagate(struct redisCommand *cmd, int dbid, robj **argv, int argc, int target);
@@ -1281,2 +1266 @@ void freeMemoryOverheadData(struct redisMemOverhead *mh);
-#define RESTART_SERVER_CONFIG_REWRITE \
-  (1 << 1)
+#define RESTART_SERVER_CONFIG_REWRITE (1<<1)
@@ -1293,2 +1277 @@ int setTypeRandomElement(robj *setobj, sds *sdsele, int64_t *llele);
-unsigned long setTypeRandomElements(robj *set, unsigned long count,
-                                    robj *aux_set);
+unsigned long setTypeRandomElements(robj *set, unsigned long count, robj *aux_set);
@@ -1310 +1293,2 @@ void hashTypeCurrentFromZiplist(hashTypeIterator *hi, int what,
-                                unsigned char **vstr, unsigned int *vlen,
+                                unsigned char **vstr,
+                                unsigned int *vlen,
@@ -1313,2 +1297 @@ sds hashTypeCurrentFromHashTable(hashTypeIterator *hi, int what);
-void hashTypeCurrentObject(hashTypeIterator *hi, int what, unsigned char **vstr,
-                           unsigned int *vlen, long long *vll);
+void hashTypeCurrentObject(hashTypeIterator *hi, int what, unsigned char **vstr, unsigned int *vlen, long long *vll);
@@ -1331,2 +1314 @@ struct rewriteConfigState;
-void rewriteConfigRewriteLine(struct rewriteConfigState *state,
-                              const char *option, sds line, int force);
+void rewriteConfigRewriteLine(struct rewriteConfigState *state, const char *option, sds line, int force);
@@ -1365,2 +1347 @@ void signalFlushedDb(int dbid);
-unsigned int getKeysInSlot(unsigned int hashslot, robj **keys,
-                           unsigned int count);
+unsigned int getKeysInSlot(unsigned int hashslot, robj **keys, unsigned int count);
@@ -1379,2 +1360 @@ size_t lazyfreeGetPendingObjectsCount(void);
-int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc,
-                        int *numkeys);
+int *getKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1382,2 +1362 @@ void getKeysFreeResult(int *result);
-int *zunionInterGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                        int *numkeys);
+int *zunionInterGetKeys(struct redisCommand *cmd,robj **argv, int argc, int *numkeys);
@@ -1386,6 +1365,3 @@ int *sortGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
-int *migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                    int *numkeys);
-int *georadiusGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                      int *numkeys);
-int *xreadGetKeys(struct redisCommand *cmd, robj **argv, int argc,
-                  int *numkeys);
+int *migrateGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *georadiusGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
+int *xreadGetKeys(struct redisCommand *cmd, robj **argv, int argc, int *numkeys);
@@ -1399,3 +1375 @@ void clusterBeforeSleep(void);
-int clusterSendModuleMessageToTarget(const char *target, uint64_t module_id,
-                                     uint8_t type, unsigned char *payload,
-                                     uint32_t len);
+int clusterSendModuleMessageToTarget(const char *target, uint64_t module_id, uint8_t type, unsigned char *payload, uint32_t len);
@@ -1419,2 +1393 @@ void replyToBlockedClientTimedOut(client *c);
-int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout,
-                                int unit);
+int getTimeoutFromObjectOrReply(client *c, robj *object, mstime_t *timeout, int unit);
@@ -1424,2 +1397 @@ void signalKeyAsReady(redisDb *db, robj *key);
-void blockForKeys(client *c, int btype, robj **keys, int numkeys,
-                  mstime_t timeout, robj *target, streamID *ids);
+void blockForKeys(client *c, int btype, robj **keys, int numkeys, mstime_t timeout, robj *target, streamID *ids);
@@ -1634,2 +1606,7 @@ void xtrimCommand(client *c);
-void _serverAssertWithInfo(const client *c, const robj *o, const char *estr,
-                           const char *file, int line);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
+void _serverAssertWithInfo(const client *c, const robj *o, const char *estr, const char *file, int line);
@@ -1648,0 +1626,4 @@ void xorDigest(unsigned char *digest, void *ptr, size_t len);
+#define redisDebug(fmt,...) \
+    printf("DEBUG %s:%d > " fmt "\n", __FILE__, __LINE__, __VA_ARGS__)
+#define redisDebugMark() \
+    printf("-- MARK %s:%d --\n", __FILE__, __LINE__)
