diff --git a/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_mergegen.c b/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_merged.c
index a785a48..4ac6d29 100644
--- a/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_mergegen.c
+++ b/./output/redis/fc09b9f4-94364d53-bad7d097/src@t_list.no_comments_merged.c
@@ -617 +617,4 @@ void blockingPopGenericCommand(redisClient *c, int where) {
-                                                                                                                                        if (c->flags & REDIS_MULTI) { addReply(c,shared.nullmultibulk); return; }
+    if (c->flags & REDIS_MULTI) {
+        addReply(c,shared.nullmultibulk);
+        return;
+    }
