diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_merged.h
index ad63845..be23597 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_merged.h
@@ -47,4 +47,4 @@ typedef struct record_client_info record_client_info_t;
-ConversionResult<record_client_info_t> aidl2legacy_RecordClientInfo_record_client_info_t(
-        const media::RecordClientInfo& aidl);
-ConversionResult<media::RecordClientInfo> legacy2aidl_record_client_info_t_RecordClientInfo(
-        const record_client_info_t& legacy);
+ConversionResult<record_client_info_t>
+aidl2legacy_RecordClientInfo_record_client_info_t(const media::RecordClientInfo& aidl);
+ConversionResult<media::RecordClientInfo>
+legacy2aidl_record_client_info_t_RecordClientInfo(const record_client_info_t& legacy);
@@ -53 +53,2 @@ typedef void (*dynamic_policy_callback)(int event, String8 regId, int val);
-typedef void (*record_config_callback)(int event, const record_client_info_t* clientInfo,
+typedef void (*record_config_callback)(int event,
+                                       const record_client_info_t *clientInfo,
@@ -58 +59,2 @@ typedef void (*record_config_callback)(int event, const record_client_info_t* cl
-                                       audio_patch_handle_t patchHandle, audio_source_t source);
+                                       audio_patch_handle_t patchHandle,
+                                       audio_source_t source);
@@ -67 +69,2 @@ class IAudioPolicyService;
-class AudioSystem {
+class AudioSystem
+{
@@ -70,0 +74,3 @@ class AudioSystem {
+    template <typename ServiceInterface, typename Client, typename AidlInterface,
+            typename ServiceTraits>
+    friend class ServiceHandler;
@@ -106,9 +112,15 @@ class AudioSystem {
-    static size_t calculateMinFrameCount(uint32_t afLatencyMs, uint32_t afFrameCount,
-                                         uint32_t afSampleRate, uint32_t sampleRate,
-                                         float speed );
-    static status_t getOutputSamplingRate(uint32_t* samplingRate, audio_stream_type_t stream);
-    static status_t getOutputFrameCount(size_t* frameCount, audio_stream_type_t stream);
-    static status_t getOutputLatency(uint32_t* latency, audio_stream_type_t stream);
-    static status_t getSamplingRate(audio_io_handle_t ioHandle, uint32_t* samplingRate);
-    static status_t getFrameCount(audio_io_handle_t ioHandle, size_t* frameCount);
-    static status_t getLatency(audio_io_handle_t output, uint32_t* latency);
+    static size_t calculateMinFrameCount(
+            uint32_t afLatencyMs, uint32_t afFrameCount, uint32_t afSampleRate,
+            uint32_t sampleRate, float speed );
+    static status_t getOutputSamplingRate(uint32_t* samplingRate,
+            audio_stream_type_t stream);
+    static status_t getOutputFrameCount(size_t* frameCount,
+            audio_stream_type_t stream);
+    static status_t getOutputLatency(uint32_t* latency,
+            audio_stream_type_t stream);
+    static status_t getSamplingRate(audio_io_handle_t ioHandle,
+                                          uint32_t* samplingRate);
+    static status_t getFrameCount(audio_io_handle_t ioHandle,
+                                  size_t* frameCount);
+    static status_t getLatency(audio_io_handle_t output,
+                               uint32_t* latency);
@@ -118 +130,2 @@ class AudioSystem {
-    static status_t getRenderPosition(audio_io_handle_t output, uint32_t* halFrames,
+    static status_t getRenderPosition(audio_io_handle_t output,
+                                      uint32_t *halFrames,
@@ -127 +140,2 @@ class AudioSystem {
-    static status_t getFrameCountHAL(audio_io_handle_t ioHandle, size_t* frameCount);
+    static status_t getFrameCountHAL(audio_io_handle_t ioHandle,
+                                     size_t* frameCount);
@@ -141,2 +155,4 @@ class AudioSystem {
-    static status_t handleDeviceConfigChange(audio_devices_t device, const char* device_address,
-                                             const char* device_name, audio_format_t encodedFormat);
+    static status_t handleDeviceConfigChange(audio_devices_t device,
+                                             const char *device_address,
+                                             const char *device_name,
+                                             audio_format_t encodedFormat);
@@ -146,2 +162,4 @@ class AudioSystem {
-    static status_t getOutputForAttr(audio_attributes_t* attr, audio_io_handle_t* output,
-                                     audio_session_t session, audio_stream_type_t* stream,
+    static status_t getOutputForAttr(audio_attributes_t *attr,
+                                     audio_io_handle_t *output,
+                                     audio_session_t session,
+                                     audio_stream_type_t *stream,
@@ -149 +167,2 @@ class AudioSystem {
-                                     audio_config_t* config, audio_output_flags_t flags,
+                                     audio_config_t *config,
+                                     audio_output_flags_t flags,
@@ -153 +172,2 @@ class AudioSystem {
-                                     bool* isSpatialized, bool* isBitPerfect);
+                                     bool *isSpatialized,
+                                     bool *isBitPerfect);
@@ -157,2 +177,4 @@ class AudioSystem {
-    static status_t getInputForAttr(const audio_attributes_t* attr, audio_io_handle_t* input,
-                                    audio_unique_id_t riid, audio_session_t session,
+    static status_t getInputForAttr(const audio_attributes_t *attr,
+                                    audio_io_handle_t *input,
+                                    audio_unique_id_t riid,
+                                    audio_session_t session,
@@ -160 +182,2 @@ class AudioSystem {
-                                    audio_config_base_t* config, audio_input_flags_t flags,
+                                    audio_config_base_t *config,
+                                    audio_input_flags_t flags,
@@ -166,2 +189,5 @@ class AudioSystem {
-    static status_t initStreamVolume(audio_stream_type_t stream, int indexMin, int indexMax);
-    static status_t setStreamVolumeIndex(audio_stream_type_t stream, int index,
+    static status_t initStreamVolume(audio_stream_type_t stream,
+                                      int indexMin,
+                                      int indexMax);
+    static status_t setStreamVolumeIndex(audio_stream_type_t stream,
+                                         int index,
@@ -169 +195,2 @@ class AudioSystem {
-    static status_t getStreamVolumeIndex(audio_stream_type_t stream, int* index,
+    static status_t getStreamVolumeIndex(audio_stream_type_t stream,
+                                         int *index,
@@ -171 +198,2 @@ class AudioSystem {
-    static status_t setVolumeIndexForAttributes(const audio_attributes_t& attr, int index,
+    static status_t setVolumeIndexForAttributes(const audio_attributes_t &attr,
+                                                int index,
@@ -173 +201,2 @@ class AudioSystem {
-    static status_t getVolumeIndexForAttributes(const audio_attributes_t& attr, int& index,
+    static status_t getVolumeIndexForAttributes(const audio_attributes_t &attr,
+                                                int &index,
@@ -179 +208,2 @@ class AudioSystem {
-                                            AudioDeviceTypeAddrVector* devices, bool forVolume);
+                                            AudioDeviceTypeAddrVector *devices,
+                                            bool forVolume);
@@ -181,2 +211,5 @@ class AudioSystem {
-    static status_t registerEffect(const effect_descriptor_t* desc, audio_io_handle_t io,
-                                   product_strategy_t strategy, audio_session_t session, int id);
+    static status_t registerEffect(const effect_descriptor_t *desc,
+                                    audio_io_handle_t io,
+                                    product_strategy_t strategy,
+                                    audio_session_t session,
+                                    int id);
@@ -197,2 +230,4 @@ class AudioSystem {
-    static status_t listAudioPorts(audio_port_role_t role, audio_port_type_t type,
-                                   unsigned int* num_ports, struct audio_port_v7* ports,
+    static status_t listAudioPorts(audio_port_role_t role,
+                                   audio_port_type_t type,
+                                   unsigned int *num_ports,
+                                   struct audio_port_v7 *ports,
@@ -203 +238,2 @@ class AudioSystem {
-    static status_t createAudioPatch(const struct audio_patch* patch, audio_patch_handle_t* handle);
+    static status_t createAudioPatch(const struct audio_patch *patch,
+                                       audio_patch_handle_t *handle);
@@ -205 +241,2 @@ class AudioSystem {
-    static status_t listAudioPatches(unsigned int* num_patches, struct audio_patch* patches,
+    static status_t listAudioPatches(unsigned int *num_patches,
+                                      struct audio_patch *patches,
@@ -216,2 +253,2 @@ class AudioSystem {
-            const std::vector<std::pair<AudioMix, std::vector<AudioMixMatchCriterion>>>&
-                    mixesWithUpdates);
+        const std::vector<
+                std::pair<AudioMix, std::vector<AudioMixMatchCriterion>>>& mixesWithUpdates);
@@ -230 +267,2 @@ class AudioSystem {
-    static float getStreamVolumeDB(audio_stream_type_t stream, int index, audio_devices_t device);
+    static float getStreamVolumeDB(
+            audio_stream_type_t stream, int index, audio_devices_t device);
@@ -247,2 +285,2 @@ class AudioSystem {
-    static status_t getProductStrategyFromAudioAttributes(const audio_attributes_t& aa,
-                                                          product_strategy_t& productStrategy,
+    static status_t getProductStrategyFromAudioAttributes(
+            const audio_attributes_t &aa, product_strategy_t &productStrategy,
@@ -253,2 +291,2 @@ class AudioSystem {
-    static status_t getVolumeGroupFromAudioAttributes(const audio_attributes_t& aa,
-                                                      volume_group_t& volumeGroup,
+    static status_t getVolumeGroupFromAudioAttributes(
+            const audio_attributes_t &aa, volume_group_t &volumeGroup,
@@ -259,15 +297,5 @@ class AudioSystem {
-    static status_t setDevicesRoleForStrategy(product_strategy_t strategy, device_role_t role,
-                                              const AudioDeviceTypeAddrVector& devices);
-    static status_t removeDevicesRoleForStrategy(product_strategy_t strategy, device_role_t role,
-                                                 const AudioDeviceTypeAddrVector& devices);
-    static status_t clearDevicesRoleForStrategy(product_strategy_t strategy, device_role_t role);
-    static status_t getDevicesForRoleAndStrategy(product_strategy_t strategy, device_role_t role,
-                                                 AudioDeviceTypeAddrVector& devices);
-    static status_t setDevicesRoleForCapturePreset(audio_source_t audioSource, device_role_t role,
-                                                   const AudioDeviceTypeAddrVector& devices);
-    static status_t addDevicesRoleForCapturePreset(audio_source_t audioSource, device_role_t role,
-                                                   const AudioDeviceTypeAddrVector& devices);
-    static status_t removeDevicesRoleForCapturePreset(audio_source_t audioSource,
-                                                      device_role_t role,
-                                                      const AudioDeviceTypeAddrVector& devices);
-    static status_t clearDevicesRoleForCapturePreset(audio_source_t audioSource,
+    static status_t setDevicesRoleForStrategy(product_strategy_t strategy,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t removeDevicesRoleForStrategy(product_strategy_t strategy,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t clearDevicesRoleForStrategy(product_strategy_t strategy,
@@ -274,0 +303,11 @@ class AudioSystem {
+    static status_t getDevicesForRoleAndStrategy(product_strategy_t strategy,
+            device_role_t role, AudioDeviceTypeAddrVector &devices);
+    static status_t setDevicesRoleForCapturePreset(audio_source_t audioSource,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t addDevicesRoleForCapturePreset(audio_source_t audioSource,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t removeDevicesRoleForCapturePreset(
+            audio_source_t audioSource, device_role_t role,
+            const AudioDeviceTypeAddrVector& devices);
+    static status_t clearDevicesRoleForCapturePreset(
+            audio_source_t audioSource, device_role_t role);
@@ -276,3 +315,3 @@ class AudioSystem {
-                                                      device_role_t role,
-                                                      AudioDeviceTypeAddrVector& devices);
-    static status_t getDeviceForStrategy(product_strategy_t strategy, AudioDeviceTypeAddr& device);
+            device_role_t role, AudioDeviceTypeAddrVector &devices);
+    static status_t getDeviceForStrategy(product_strategy_t strategy,
+            AudioDeviceTypeAddr &device);
@@ -281 +320,2 @@ class AudioSystem {
-    static status_t canBeSpatialized(const audio_attributes_t* attr, const audio_config_t* config,
+    static status_t canBeSpatialized(const audio_attributes_t *attr,
+                                     const audio_config_t *config,
@@ -291 +331,2 @@ class AudioSystem {
-    static status_t setRequestedLatencyMode(audio_io_handle_t output, audio_latency_mode_t mode);
+    static status_t setRequestedLatencyMode(
+            audio_io_handle_t output, audio_latency_mode_t mode);
@@ -300 +341,2 @@ class AudioSystem {
-                                                audio_port_handle_t portId, uid_t uid,
+                                                audio_port_handle_t portId,
+                                                uid_t uid,
@@ -306 +348,2 @@ class AudioSystem {
-                                                  audio_port_handle_t portId, uid_t uid);
+                                                  audio_port_handle_t portId,
+                                                  uid_t uid);
@@ -316 +359,2 @@ class AudioSystem {
-    class AudioVolumeGroupCallback : public virtual RefBase {
+    class AudioVolumeGroupCallback : public virtual RefBase
+    {
@@ -325 +369,2 @@ class AudioSystem {
-    class AudioPortCallback : public virtual RefBase {
+    class AudioPortCallback : public virtual RefBase
+    {
@@ -335 +380,2 @@ class AudioSystem {
-    class AudioDeviceCallback : public virtual RefBase {
+    class AudioDeviceCallback : public virtual RefBase
+    {
@@ -343 +389,2 @@ class AudioSystem {
-                                           audio_io_handle_t audioIo, audio_port_handle_t portId);
+                                           audio_io_handle_t audioIo,
+                                           audio_port_handle_t portId);
@@ -347 +394,2 @@ class AudioSystem {
-    class SupportedLatencyModesCallback : public virtual RefBase {
+    class SupportedLatencyModesCallback : public virtual RefBase
+    {
@@ -365 +413,2 @@ class AudioSystem {
-    class AudioFlingerClient : public IBinder::DeathRecipient, public media::BnAudioFlingerClient {
+    class AudioFlingerClient: public IBinder::DeathRecipient, public media::BnAudioFlingerClient
+    {
@@ -370,2 +419 @@ class AudioSystem {
-                                    audio_channel_mask_t channelMask, size_t* buffSize)
-                EXCLUDES(mMutex);
+                audio_channel_mask_t channelMask, size_t* buffSize) EXCLUDES(mMutex);
@@ -374,3 +422,3 @@ class AudioSystem {
-        binder::Status ioConfigChanged(media::AudioIoConfigEvent event,
-                                       const media::AudioIoDescriptor& ioDesc) final
-                EXCLUDES(mMutex);
+        binder::Status ioConfigChanged(
+                media::AudioIoConfigEvent event,
+                const media::AudioIoDescriptor& ioDesc) final EXCLUDES(mMutex);
@@ -379,2 +427,2 @@ class AudioSystem {
-                const std::vector<media::audio::common::AudioLatencyMode>& latencyModes) final
-                EXCLUDES(mMutex);
+                const std::vector<media::audio::common::AudioLatencyMode>& latencyModes)
+                final EXCLUDES(mMutex);
@@ -382,2 +430 @@ class AudioSystem {
-                                        audio_io_handle_t audioIo, audio_port_handle_t portId)
-                EXCLUDES(mMutex);
+                audio_io_handle_t audioIo, audio_port_handle_t portId) EXCLUDES(mMutex);
@@ -385,4 +432,3 @@ class AudioSystem {
-                                           audio_io_handle_t audioIo, audio_port_handle_t portId)
-                EXCLUDES(mMutex);
-        status_t addSupportedLatencyModesCallback(const sp<SupportedLatencyModesCallback>& callback)
-                EXCLUDES(mMutex);
+                audio_io_handle_t audioIo, audio_port_handle_t portId) EXCLUDES(mMutex);
+        status_t addSupportedLatencyModesCallback(
+                const sp<SupportedLatencyModesCallback>& callback) EXCLUDES(mMutex);
@@ -397,2 +443,2 @@ class AudioSystem {
-        std::vector<wp<SupportedLatencyModesCallback>> mSupportedLatencyModesCallbacks
-                GUARDED_BY(mMutex);
+        std::vector<wp<SupportedLatencyModesCallback>>
+                mSupportedLatencyModesCallbacks GUARDED_BY(mMutex);
@@ -415,4 +461,4 @@ class AudioSystem {
-        int addAudioVolumeGroupCallback(const sp<AudioVolumeGroupCallback>& callback)
-                EXCLUDES(mMutex);
-        int removeAudioVolumeGroupCallback(const sp<AudioVolumeGroupCallback>& callback)
-                EXCLUDES(mMutex);
+        int addAudioVolumeGroupCallback(
+                const sp<AudioVolumeGroupCallback>& callback) EXCLUDES(mMutex);
+        int removeAudioVolumeGroupCallback(
+                const sp<AudioVolumeGroupCallback>& callback) EXCLUDES(mMutex);
@@ -430 +476,2 @@ class AudioSystem {
-                int32_t event, const media::RecordClientInfo& clientInfo,
+                int32_t event,
+                const media::RecordClientInfo& clientInfo,
@@ -434 +481,2 @@ class AudioSystem {
-                const std::vector<media::EffectDescriptor>& effects, int32_t patchHandle,
+                const std::vector<media::EffectDescriptor>& effects,
+                int32_t patchHandle,
@@ -449,8 +497,4 @@ class AudioSystem {
-    static dynamic_policy_callback gDynPolicyCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
-    static record_config_callback gRecordConfigCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
-    static routing_callback gRoutingCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
-    static vol_range_init_req_callback gVolRangeInitReqCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
+    static dynamic_policy_callback gDynPolicyCallback GUARDED_BY(gMutex);
+    static record_config_callback gRecordConfigCallback GUARDED_BY(gMutex);
+    static routing_callback gRoutingCallback GUARDED_BY(gMutex);
+    static vol_range_init_req_callback gVolRangeInitReqCallback GUARDED_BY(gMutex);
@@ -460,2 +504 @@ class AudioSystem {
-            sp<CaptureStateListenerImpl>
-                    AudioSystem::gSoundTriggerCaptureStateListener;
+            GUARDED_BY(gErrorCallbacksMutex);
@@ -464,2 +507 @@ class AudioSystem {
-            sp<CaptureStateListenerImpl>
-                    AudioSystem::gSoundTriggerCaptureStateListener;
+            GUARDED_BY(gSoundTriggerMutex);
