diff --git a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@AudioSystem.no_comments_mergegen.cpp b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@AudioSystem.no_comments_merged.cpp
index b53e5ee..c97b0e8 100644
--- a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@AudioSystem.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@AudioSystem.no_comments_merged.cpp
@@ -52 +52,8 @@ sp<CaptureStateListenerImpl> AudioSystem::gSoundTriggerCaptureStateListener;
-template <typename ServiceInterface, typename Client, typename AidlInterface, typename ServiceTraits> class ServiceHandler { public: sp<ServiceInterface> getService() EXCLUDES(mMutex) NO_THREAD_SAFETY_ANALYSIS {
+template <typename ServiceInterface, typename Client, typename AidlInterface,
+        typename ServiceTraits>
+class ServiceHandler {
+public:
+    sp<ServiceInterface> getService()
+            EXCLUDES(mMutex) NO_THREAD_SAFETY_ANALYSIS {
+        sp<ServiceInterface> service;
+        sp<Client> client;
@@ -96,0 +104,3 @@ if (mCvGetter) mCvGetter.reset();
+        client = mClient;
+        service = mService;
+        if (mCanStartThreadPool) {
@@ -105 +115,19 @@ if (mCvGetter) mCvGetter.reset();
-status_t setLocalService(const sp<ServiceInterface>& service) EXCLUDES(mMutex) { std::lock_guard _l(mMutex);
+    status_t setLocalService(const sp<ServiceInterface>& service) EXCLUDES(mMutex) {
+        std::lock_guard _l(mMutex);
+        if (mService != nullptr && service != nullptr) return INVALID_OPERATION;
+        mLocalService = service;
+        if (mCvGetter) mCvGetter->notify_one();
+        return OK;
+    }
+    sp<Client> getClient() EXCLUDES(mMutex) {
+        const auto service = getService();
+        if (service == nullptr) return nullptr;
+        std::lock_guard _l(mMutex);
+        return mClient;
+    }
+    void setBinder(const sp<IBinder>& binder) EXCLUDES(mMutex) {
+        std::lock_guard _l(mMutex);
+        if (mService != nullptr) {
+            ALOGW("%s: ignoring; %s connection already established.",
+                    __func__, ServiceTraits::SERVICE_NAME);
+            return;
@@ -107 +135,2 @@ status_t setLocalService(const sp<ServiceInterface>& service) EXCLUDES(mMutex) {
-sp<AudioSystem::AudioFlingerClient> AudioSystem::getAudioFlingerClient() { return gAudioFlingerServiceHandler.getClient();
+        mBinder = binder;
+        if (mCvGetter) mCvGetter->notify_one();
@@ -109 +138,24 @@ sp<AudioSystem::AudioFlingerClient> AudioSystem::getAudioFlingerClient() { retur
-void AudioSystem::setAudioFlingerBinder(const sp<IBinder>& audioFlinger) { if (audioFlinger->getInterfaceDescriptor() != media::IAudioFlingerService::descriptor) { ALOGE("%s: received a binder of type %s", __func__, String8(audioFlinger->getInterfaceDescriptor()).c_str()); return; } gAudioFlingerServiceHandler.setBinder(audioFlinger);
+    void clearService() EXCLUDES(mMutex) {
+        std::lock_guard _l(mMutex);
+        mService.clear();
+        if (mClient) ServiceTraits::onClearService(mClient);
+    }
+    void disableThreadPool() {
+        mCanStartThreadPool = false;
+    }
+private:
+    std::mutex mSingleGetter;
+    std::mutex mMutex;
+    std::shared_ptr<std::condition_variable> mCvGetter GUARDED_BY(mMutex);
+    sp<IBinder> mBinder GUARDED_BY(mMutex);
+    sp<ServiceInterface> mLocalService GUARDED_BY(mMutex);
+    sp<ServiceInterface> mService GUARDED_BY(mMutex);
+    sp<Client> mClient GUARDED_BY(mMutex);
+    std::atomic<bool> mCanStartThreadPool = true;
+};
+struct AudioFlingerTraits {
+    static void onServiceCreate(
+            const sp<IAudioFlinger>& af, const sp<AudioSystem::AudioFlingerClient>& afc) {
+        const int64_t token = IPCThreadState::self()->clearCallingIdentity();
+        af->registerClient(afc);
+        IPCThreadState::self()->restoreCallingIdentity(token);
@@ -111 +163,28 @@ void AudioSystem::setAudioFlingerBinder(const sp<IBinder>& audioFlinger) { if (a
-status_t AudioSystem::setLocalAudioFlinger(const sp<IAudioFlinger>& af) { return gAudioFlingerServiceHandler.setLocalService(af);
+    static sp<IAudioFlinger> createServiceAdapter(
+            const sp<media::IAudioFlingerService>& aidlInterface) {
+        return sp<AudioFlingerClientAdapter>::make(aidlInterface);
+    }
+    static void onClearService(const sp<AudioSystem::AudioFlingerClient>& afc) {
+        afc->clearIoCache();
+    }
+    static constexpr const char* SERVICE_NAME = IAudioFlinger::DEFAULT_SERVICE_NAME;
+};
+[[clang::no_destroy]] static constinit ServiceHandler<IAudioFlinger,
+        AudioSystem::AudioFlingerClient, media::IAudioFlingerService,
+        AudioFlingerTraits> gAudioFlingerServiceHandler;
+sp<IAudioFlinger> AudioSystem::get_audio_flinger() {
+    return gAudioFlingerServiceHandler.getService();
+}
+sp<AudioSystem::AudioFlingerClient> AudioSystem::getAudioFlingerClient() {
+    return gAudioFlingerServiceHandler.getClient();
+}
+void AudioSystem::setAudioFlingerBinder(const sp<IBinder>& audioFlinger) {
+    if (audioFlinger->getInterfaceDescriptor() != media::IAudioFlingerService::descriptor) {
+        ALOGE("%s: received a binder of type %s",
+                __func__, String8(audioFlinger->getInterfaceDescriptor()).c_str());
+        return;
+    }
+    gAudioFlingerServiceHandler.setBinder(audioFlinger);
+}
+status_t AudioSystem::setLocalAudioFlinger(const sp<IAudioFlinger>& af) {
+    return gAudioFlingerServiceHandler.setLocalService(af);
@@ -694 +773,5 @@ void AudioSystem::clearAudioPolicyService() {
-gAudioPolicyServiceHandler.clearService(); } void AudioSystem::disableThreadPool() { gAudioFlingerServiceHandler.disableThreadPool(); gAudioPolicyServiceHandler.disableThreadPool();
+    gAudioPolicyServiceHandler.clearService();
+}
+void AudioSystem::disableThreadPool() {
+    gAudioFlingerServiceHandler.disableThreadPool();
+    gAudioPolicyServiceHandler.disableThreadPool();
