diff --git a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_conflict.h b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_merged.h
index f8dafb9..be23597 100644
--- a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_conflict.h
+++ b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_merged.h
@@ -108 +107,0 @@ public:
-<<<<<<< HEAD
@@ -111,7 +109,0 @@ public:
-||||||| a77d2bd563
-    static const sp<IAudioFlinger> get_audio_flinger();
-    static const sp<IAudioFlinger> get_audio_flinger_for_fuzzer();
-=======
-    static sp<IAudioFlinger> get_audio_flinger();
-    static sp<IAudioFlinger> get_audio_flinger_for_fuzzer();
->>>>>>> 38b856f7
