diff --git a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_mergegen.h b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_merged.h
index f4d0652..302239b 100644
--- a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_mergegen.h
+++ b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_merged.h
@@ -114 +114,4 @@ void filterParameters(const CameraMetadata& src, const CameraMetadata& deviceInf
-metadata_vendor_id_t vendorTagId, CameraMetadata& dst); template <typename T> bool contains(std::set<T> container, T value) { return container.find(value) != container.end(); }
+        metadata_vendor_id_t vendorTagId, CameraMetadata& dst);
+template <typename T> bool contains(std::set<T> container, T value) {
+    return container.find(value) != container.end();
+}
