--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_merged.cc
@@ -4 +3,0 @@
-#include "src/core/lib/gprpp/string_view.h"
@@ -10 +9 @@ template <typename EntryType>
-std::map<StringView, double, StringLess> ParseMap(
+std::map<absl::string_view, double, StringLess> ParseMap(
@@ -16 +15 @@ std::map<StringView, double, StringLess> ParseMap(
-  std::map<StringView, double, StringLess> result;
+  std::map<absl::string_view, double, StringLess> result;
@@ -22 +21 @@ std::map<StringView, double, StringLess> ParseMap(
-    char* key = static_cast<char*>(arena->Alloc(key_view.size + 1));
+    char* key = static_cast<char*>(arena->Alloc(key_view.size));
@@ -24 +23 @@ std::map<StringView, double, StringLess> ParseMap(
-    result[StringView(key, key_view.size)] = value_func(entry);
+    result[absl::string_view(key, key_view.size)] = value_func(entry);
