diff --git a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_conflict.cc b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_merged.cc
index 5100195..a44a257 100644
--- a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_conflict.cc
+++ b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_merged.cc
@@ -15,2 +15 @@ std::map<absl::string_view, double, StringLess> ParseMap(
-<<<<<<< HEAD
-  std::map<StringView, double, StringLess> result;
+  std::map<absl::string_view, double, StringLess> result;
@@ -22,14 +21 @@ std::map<absl::string_view, double, StringLess> ParseMap(
-||||||| 694f491e06
-  std::map<StringView, double, StringLess> result;
-  size_t size;
-  const auto* const* entries = entry_func(msg, &size);
-  for (size_t i = 0; i < size; ++i) {
-    upb_strview key_view = key_func(entries[i]);
-=======
-  std::map<absl::string_view, double, StringLess> result;
-  size_t size;
-  const auto* const* entries = entry_func(msg, &size);
-  for (size_t i = 0; i < size; ++i) {
-    upb_strview key_view = key_func(entries[i]);
->>>>>>> 729af3a4
-    char* key = static_cast<char*>(arena->Alloc(key_view.size + 1));
+    char* key = static_cast<char*>(arena->Alloc(key_view.size));
@@ -37,7 +23 @@ std::map<absl::string_view, double, StringLess> ParseMap(
-<<<<<<< HEAD
-    result[StringView(key, key_view.size)] = value_func(entry);
-||||||| 694f491e06
-    result[StringView(key, key_view.size)] = value_func(entries[i]);
-=======
-    result[absl::string_view(key, key_view.size)] = value_func(entries[i]);
->>>>>>> 729af3a4
+    result[absl::string_view(key, key_view.size)] = value_func(entry);
