--- a/./output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_conflict.cpp
+++ b/./output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_merged.cpp
@@ -156 +155,0 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
-<<<<<<< HEAD
@@ -160,17 +158,0 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
-||||||| 3fcfeaceae
-                                        int fenceFd, int32_t* outBytesPerPixel,
-                                        int32_t* outBytesPerStride) {
-    ATRACE_CALL();
-    const uint64_t usage = static_cast<uint64_t>(
-            android_convertGralloc1To0Usage(producerUsage, consumerUsage));
-    return mMapper->lock(handle, usage, bounds, fenceFd, vaddr, outBytesPerPixel,
-                         outBytesPerStride);
-=======
-                                        int fenceFd, int32_t* outBytesPerPixel,
-                                        int32_t* outBytesPerStride) {
-    ATRACE_CALL();
-    const uint64_t usage = static_cast<uint64_t>(ANDROID_NATIVE_UNSIGNED_CAST(
-            android_convertGralloc1To0Usage(producerUsage, consumerUsage)));
-    return mMapper->lock(handle, usage, bounds, fenceFd, vaddr, outBytesPerPixel,
-                         outBytesPerStride);
->>>>>>> 7f1bf8c6
