--- a/./output/art/48fe6c57-c1b8344e-9da5bba7/libnativeloader@native_loader_test.no_comments_conflict.cpp
+++ b/./output/art/48fe6c57-c1b8344e-9da5bba7/libnativeloader@native_loader_test.no_comments_merged.cpp
@@ -296 +295,0 @@ TEST_P(NativeLoaderTest_Create, UnbundledProductApp) {
-<<<<<<< HEAD
@@ -309,20 +307,0 @@ TEST_P(NativeLoaderTest_Create, UnbundledProductApp) {
-||||||| 9da5bba7cf
-  expected_namespace_prefix = "product-clns";
-  expected_library_path = expected_library_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR;
-  expected_permitted_path =
-      expected_permitted_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR;
-  expected_shared_libs_to_platform_ns =
-      append_extended_libraries(default_public_libraries() + ":" + llndk_libraries_product());
-  expected_link_with_vndk_product_ns = true;
-=======
-  if (is_product_treblelized()) {
-    expected_namespace_prefix = "product-clns";
-    expected_library_path =
-        expected_library_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR;
-    expected_permitted_path =
-        expected_permitted_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR;
-    expected_shared_libs_to_platform_ns =
-        append_extended_libraries(default_public_libraries() + ":" + llndk_libraries_product());
-    expected_link_with_vndk_product_ns = true;
-  }
->>>>>>> c1b8344e
