diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/071849f8-71d453f1-8636848e/input.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/071849f8-71d453f1-8636848e/input.no_comments_merged.c
index ed1dafa..9742c44 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/071849f8-71d453f1-8636848e/input.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/071849f8-71d453f1-8636848e/input.no_comments_merged.c
@@ -16 +16,5 @@ struct input_param {
-  enum { INPUT_MISSING, INPUT_NUMBER, INPUT_STRING } type;
+ enum {
+  INPUT_MISSING,
+  INPUT_NUMBER,
+  INPUT_STRING
+ } type;
@@ -41 +45,4 @@ struct input_ctx {
-  enum { INPUT_END_ST, INPUT_END_BEL } input_end;
+ enum {
+  INPUT_END_ST,
+  INPUT_END_BEL
+ } input_end;
@@ -123,7 +130,14 @@ static const struct input_table_entry input_esc_table[] = {
-    {'0', "(", INPUT_ESC_SCSG0_ON}, {'0', ")", INPUT_ESC_SCSG1_ON},
-    {'7', "", INPUT_ESC_DECSC}, {'8', "", INPUT_ESC_DECRC},
-    {'8', "#", INPUT_ESC_DECALN}, {'=', "", INPUT_ESC_DECKPAM},
-    {'>', "", INPUT_ESC_DECKPNM}, {'B', "(", INPUT_ESC_SCSG0_OFF},
-    {'B', ")", INPUT_ESC_SCSG1_OFF}, {'D', "", INPUT_ESC_IND},
-    {'E', "", INPUT_ESC_NEL}, {'H', "", INPUT_ESC_HTS},
-    {'M', "", INPUT_ESC_RI}, {'\\', "", INPUT_ESC_ST},
+ { '0', "(", INPUT_ESC_SCSG0_ON },
+ { '0', ")", INPUT_ESC_SCSG1_ON },
+ { '7', "", INPUT_ESC_DECSC },
+ { '8', "", INPUT_ESC_DECRC },
+ { '8', "#", INPUT_ESC_DECALN },
+ { '=', "", INPUT_ESC_DECKPAM },
+ { '>', "", INPUT_ESC_DECKPNM },
+ { 'B', "(", INPUT_ESC_SCSG0_OFF },
+ { 'B', ")", INPUT_ESC_SCSG1_OFF },
+ { 'D', "", INPUT_ESC_IND },
+ { 'E', "", INPUT_ESC_NEL },
+ { 'H', "", INPUT_ESC_HTS },
+ { 'M', "", INPUT_ESC_RI },
+ { '\\', "", INPUT_ESC_ST },
@@ -172,19 +186,38 @@ static const struct input_table_entry input_csi_table[] = {
-    {'@', "", INPUT_CSI_ICH}, {'A', "", INPUT_CSI_CUU},
-    {'B', "", INPUT_CSI_CUD}, {'C', "", INPUT_CSI_CUF},
-    {'D', "", INPUT_CSI_CUB}, {'E', "", INPUT_CSI_CNL},
-    {'F', "", INPUT_CSI_CPL}, {'G', "", INPUT_CSI_HPA},
-    {'H', "", INPUT_CSI_CUP}, {'J', "", INPUT_CSI_ED},
-    {'K', "", INPUT_CSI_EL}, {'L', "", INPUT_CSI_IL},
-    {'M', "", INPUT_CSI_DL}, {'P', "", INPUT_CSI_DCH},
-    {'S', "", INPUT_CSI_SU}, {'T', "", INPUT_CSI_SD},
-    {'X', "", INPUT_CSI_ECH}, {'Z', "", INPUT_CSI_CBT},
-    {'`', "", INPUT_CSI_HPA}, {'b', "", INPUT_CSI_REP},
-    {'c', "", INPUT_CSI_DA}, {'c', ">", INPUT_CSI_DA_TWO},
-    {'d', "", INPUT_CSI_VPA}, {'f', "", INPUT_CSI_CUP},
-    {'g', "", INPUT_CSI_TBC}, {'h', "", INPUT_CSI_SM},
-    {'h', "?", INPUT_CSI_SM_PRIVATE}, {'l', "", INPUT_CSI_RM},
-    {'l', "?", INPUT_CSI_RM_PRIVATE}, {'m', "", INPUT_CSI_SGR},
-    {'m', ">", INPUT_CSI_MODSET}, {'n', "", INPUT_CSI_DSR},
-    {'n', ">", INPUT_CSI_MODOFF}, {'q', " ", INPUT_CSI_DECSCUSR},
-    {'q', ">", INPUT_CSI_XDA}, {'r', "", INPUT_CSI_DECSTBM},
-    {'s', "", INPUT_CSI_SCP}, {'t', "", INPUT_CSI_WINOPS},
+ { '@', "", INPUT_CSI_ICH },
+ { 'A', "", INPUT_CSI_CUU },
+ { 'B', "", INPUT_CSI_CUD },
+ { 'C', "", INPUT_CSI_CUF },
+ { 'D', "", INPUT_CSI_CUB },
+ { 'E', "", INPUT_CSI_CNL },
+ { 'F', "", INPUT_CSI_CPL },
+ { 'G', "", INPUT_CSI_HPA },
+ { 'H', "", INPUT_CSI_CUP },
+ { 'J', "", INPUT_CSI_ED },
+ { 'K', "", INPUT_CSI_EL },
+ { 'L', "", INPUT_CSI_IL },
+ { 'M', "", INPUT_CSI_DL },
+ { 'P', "", INPUT_CSI_DCH },
+ { 'S', "", INPUT_CSI_SU },
+ { 'T', "", INPUT_CSI_SD },
+ { 'X', "", INPUT_CSI_ECH },
+ { 'Z', "", INPUT_CSI_CBT },
+ { '`', "", INPUT_CSI_HPA },
+ { 'b', "", INPUT_CSI_REP },
+ { 'c', "", INPUT_CSI_DA },
+ { 'c', ">", INPUT_CSI_DA_TWO },
+ { 'd', "", INPUT_CSI_VPA },
+ { 'f', "", INPUT_CSI_CUP },
+ { 'g', "", INPUT_CSI_TBC },
+ { 'h', "", INPUT_CSI_SM },
+ { 'h', "?", INPUT_CSI_SM_PRIVATE },
+ { 'l', "", INPUT_CSI_RM },
+ { 'l', "?", INPUT_CSI_RM_PRIVATE },
+ { 'm', "", INPUT_CSI_SGR },
+ { 'm', ">", INPUT_CSI_MODSET },
+ { 'n', "", INPUT_CSI_DSR },
+ { 'n', ">", INPUT_CSI_MODOFF },
+ { 'q', " ", INPUT_CSI_DECSCUSR },
+ { 'q', ">", INPUT_CSI_XDA },
+ { 'r', "", INPUT_CSI_DECSTBM },
+ { 's', "", INPUT_CSI_SCP },
+ { 't', "", INPUT_CSI_WINOPS },
@@ -207,160 +240,19 @@ struct input_state {
-      {0x1a, 0x1a, input_c0_dispatch, &input_state_ground}, { \
-    0x1b, 0x1b, NULL, &input_state_esc_enter \
-  }
-static const struct input_transition input_state_ground_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x7e, input_print, NULL},
-    {0x7f, 0x7f, NULL, NULL},
-    {0x80, 0xff, input_top_bit_set, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_esc_enter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_esc_intermediate},
-    {0x30, 0x4f, input_esc_dispatch, &input_state_ground},
-    {0x50, 0x50, NULL, &input_state_dcs_enter},
-    {0x51, 0x57, input_esc_dispatch, &input_state_ground},
-    {0x58, 0x58, NULL, &input_state_consume_st},
-    {0x59, 0x59, input_esc_dispatch, &input_state_ground},
-    {0x5a, 0x5a, input_esc_dispatch, &input_state_ground},
-    {0x5b, 0x5b, NULL, &input_state_csi_enter},
-    {0x5c, 0x5c, input_esc_dispatch, &input_state_ground},
-    {0x5d, 0x5d, NULL, &input_state_osc_string},
-    {0x5e, 0x5e, NULL, &input_state_consume_st},
-    {0x5f, 0x5f, NULL, &input_state_apc_string},
-    {0x60, 0x6a, input_esc_dispatch, &input_state_ground},
-    {0x6b, 0x6b, NULL, &input_state_rename_string},
-    {0x6c, 0x7e, input_esc_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_esc_intermediate_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, NULL},
-    {0x30, 0x7e, input_esc_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_enter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_csi_intermediate},
-    {0x30, 0x39, input_parameter, &input_state_csi_parameter},
-    {0x3a, 0x3a, input_parameter, &input_state_csi_parameter},
-    {0x3b, 0x3b, input_parameter, &input_state_csi_parameter},
-    {0x3c, 0x3f, input_intermediate, &input_state_csi_parameter},
-    {0x40, 0x7e, input_csi_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_parameter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_csi_intermediate},
-    {0x30, 0x39, input_parameter, NULL},
-    {0x3a, 0x3a, input_parameter, NULL},
-    {0x3b, 0x3b, input_parameter, NULL},
-    {0x3c, 0x3f, NULL, &input_state_csi_ignore},
-    {0x40, 0x7e, input_csi_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_intermediate_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, NULL},
-    {0x30, 0x3f, NULL, &input_state_csi_ignore},
-    {0x40, 0x7e, input_csi_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_ignore_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x3f, NULL, NULL},
-    {0x40, 0x7e, NULL, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_enter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL},
-    {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_dcs_intermediate},
-    {0x30, 0x39, input_parameter, &input_state_dcs_parameter},
-    {0x3a, 0x3a, NULL, &input_state_dcs_ignore},
-    {0x3b, 0x3b, input_parameter, &input_state_dcs_parameter},
-    {0x3c, 0x3f, input_intermediate, &input_state_dcs_parameter},
-    {0x40, 0x7e, input_input, &input_state_dcs_handler},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_parameter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL},
-    {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_dcs_intermediate},
-    {0x30, 0x39, input_parameter, NULL},
-    {0x3a, 0x3a, NULL, &input_state_dcs_ignore},
-    {0x3b, 0x3b, input_parameter, NULL},
-    {0x3c, 0x3f, NULL, &input_state_dcs_ignore},
-    {0x40, 0x7e, input_input, &input_state_dcs_handler},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_intermediate_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL},
-    {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL},
-    {0x20, 0x2f, input_intermediate, NULL},
-    {0x30, 0x3f, NULL, &input_state_dcs_ignore},
-    {0x40, 0x7e, input_input, &input_state_dcs_handler},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_handler_table[] = {
-    {0x00, 0x1a, input_input, NULL},
-    {0x1b, 0x1b, NULL, &input_state_dcs_escape},
-    {0x1c, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_escape_table[] = {
-    {0x00, 0x5b, input_input, &input_state_dcs_handler},
-    {0x5c, 0x5c, input_dcs_dispatch, &input_state_ground},
-    {0x5d, 0xff, input_input, &input_state_dcs_handler},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_ignore_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_osc_string_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x06, NULL, NULL}, {0x07, 0x07, input_end_bel, &input_state_ground},
-    {0x08, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_apc_string_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_rename_string_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_consume_st_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
+ { 0x1a, 0x1a, input_c0_dispatch, &input_state_ground }, \
+ { 0x1b, 0x1b, NULL, &input_state_esc_enter }
+static const struct input_transition input_state_ground_table[];
+static const struct input_transition input_state_esc_enter_table[];
+static const struct input_transition input_state_esc_intermediate_table[];
+static const struct input_transition input_state_csi_enter_table[];
+static const struct input_transition input_state_csi_parameter_table[];
+static const struct input_transition input_state_csi_intermediate_table[];
+static const struct input_transition input_state_csi_ignore_table[];
+static const struct input_transition input_state_dcs_enter_table[];
+static const struct input_transition input_state_dcs_parameter_table[];
+static const struct input_transition input_state_dcs_intermediate_table[];
+static const struct input_transition input_state_dcs_handler_table[];
+static const struct input_transition input_state_dcs_escape_table[];
+static const struct input_transition input_state_dcs_ignore_table[];
+static const struct input_transition input_state_osc_string_table[];
+static const struct input_transition input_state_apc_string_table[];
+static const struct input_transition input_state_rename_string_table[];
+static const struct input_transition input_state_consume_st_table[];
@@ -368 +260,4 @@ static const struct input_state input_state_ground = {
-    "ground", input_ground, NULL, input_state_ground_table};
+ "ground",
+ input_ground, NULL,
+ input_state_ground_table
+};
@@ -370 +265,4 @@ static const struct input_state input_state_esc_enter = {
-    "esc_enter", input_clear, NULL, input_state_esc_enter_table};
+ "esc_enter",
+ input_clear, NULL,
+ input_state_esc_enter_table
+};
@@ -372 +270,4 @@ static const struct input_state input_state_esc_intermediate = {
-    "esc_intermediate", NULL, NULL, input_state_esc_intermediate_table};
+ "esc_intermediate",
+ NULL, NULL,
+ input_state_esc_intermediate_table
+};
@@ -374 +275,4 @@ static const struct input_state input_state_csi_enter = {
-    "csi_enter", input_clear, NULL, input_state_csi_enter_table};
+ "csi_enter",
+ input_clear, NULL,
+ input_state_csi_enter_table
+};
@@ -376 +280,4 @@ static const struct input_state input_state_csi_parameter = {
-    "csi_parameter", NULL, NULL, input_state_csi_parameter_table};
+ "csi_parameter",
+ NULL, NULL,
+ input_state_csi_parameter_table
+};
@@ -378 +285,4 @@ static const struct input_state input_state_csi_intermediate = {
-    "csi_intermediate", NULL, NULL, input_state_csi_intermediate_table};
+ "csi_intermediate",
+ NULL, NULL,
+ input_state_csi_intermediate_table
+};
@@ -380 +290,4 @@ static const struct input_state input_state_csi_ignore = {
-    "csi_ignore", NULL, NULL, input_state_csi_ignore_table};
+ "csi_ignore",
+ NULL, NULL,
+ input_state_csi_ignore_table
+};
@@ -382 +295,4 @@ static const struct input_state input_state_dcs_enter = {
-    "dcs_enter", input_enter_dcs, NULL, input_state_dcs_enter_table};
+ "dcs_enter",
+ input_enter_dcs, NULL,
+ input_state_dcs_enter_table
+};
@@ -384 +300,4 @@ static const struct input_state input_state_dcs_parameter = {
-    "dcs_parameter", NULL, NULL, input_state_dcs_parameter_table};
+ "dcs_parameter",
+ NULL, NULL,
+ input_state_dcs_parameter_table
+};
@@ -386 +305,4 @@ static const struct input_state input_state_dcs_intermediate = {
-    "dcs_intermediate", NULL, NULL, input_state_dcs_intermediate_table};
+ "dcs_intermediate",
+ NULL, NULL,
+ input_state_dcs_intermediate_table
+};
@@ -388 +310,4 @@ static const struct input_state input_state_dcs_handler = {
-    "dcs_handler", NULL, NULL, input_state_dcs_handler_table};
+ "dcs_handler",
+ NULL, NULL,
+ input_state_dcs_handler_table
+};
@@ -390 +315,4 @@ static const struct input_state input_state_dcs_escape = {
-    "dcs_escape", NULL, NULL, input_state_dcs_escape_table};
+ "dcs_escape",
+ NULL, NULL,
+ input_state_dcs_escape_table
+};
@@ -392 +320,4 @@ static const struct input_state input_state_dcs_ignore = {
-    "dcs_ignore", NULL, NULL, input_state_dcs_ignore_table};
+ "dcs_ignore",
+ NULL, NULL,
+ input_state_dcs_ignore_table
+};
@@ -394,2 +325,4 @@ static const struct input_state input_state_osc_string = {
-    "osc_string", input_enter_osc, input_exit_osc,
-    input_state_osc_string_table};
+ "osc_string",
+ input_enter_osc, input_exit_osc,
+ input_state_osc_string_table
+};
@@ -397,2 +330,4 @@ static const struct input_state input_state_apc_string = {
-    "apc_string", input_enter_apc, input_exit_apc,
-    input_state_apc_string_table};
+ "apc_string",
+ input_enter_apc, input_exit_apc,
+ input_state_apc_string_table
+};
@@ -400,2 +335,4 @@ static const struct input_state input_state_rename_string = {
-    "rename_string", input_enter_rename, input_exit_rename,
-    input_state_rename_string_table};
+ "rename_string",
+ input_enter_rename, input_exit_rename,
+ input_state_rename_string_table
+};
@@ -403,2 +340,4 @@ static const struct input_state input_state_consume_st = {
-    "consume_st", input_enter_rename, NULL,
-    input_state_consume_st_table};
+ "consume_st",
+ input_enter_rename, NULL,
+ input_state_consume_st_table
+};
@@ -413 +352,2 @@ static const struct input_transition input_state_ground_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -435 +375,2 @@ static const struct input_transition input_state_esc_enter_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -444 +385,2 @@ static const struct input_transition input_state_esc_intermediate_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -457 +399,2 @@ static const struct input_transition input_state_csi_enter_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -470 +413,2 @@ static const struct input_transition input_state_csi_parameter_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -480 +424,2 @@ static const struct input_transition input_state_csi_intermediate_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -489 +434,2 @@ static const struct input_transition input_state_csi_ignore_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -502 +448,2 @@ static const struct input_transition input_state_dcs_enter_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -515 +462,2 @@ static const struct input_transition input_state_dcs_parameter_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -525 +473,2 @@ static const struct input_transition input_state_dcs_intermediate_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -530 +479,2 @@ static const struct input_transition input_state_dcs_handler_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -535 +485,2 @@ static const struct input_transition input_state_dcs_escape_table[] = {
-    {-1, -1, NULL, NULL}};
+ { -1, -1, NULL, NULL }
+};
@@ -538,3 +489,6 @@ static const struct input_transition input_state_dcs_ignore_table[] = {
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
+ { 0x00, 0x17, NULL, NULL },
+ { 0x19, 0x19, NULL, NULL },
+ { 0x1c, 0x1f, NULL, NULL },
+ { 0x20, 0xff, NULL, NULL },
+ { -1, -1, NULL, NULL }
+};
@@ -543,4 +497,8 @@ static const struct input_transition input_state_osc_string_table[] = {
-    {0x00, 0x06, NULL, NULL}, {0x07, 0x07, input_end_bel, &input_state_ground},
-    {0x08, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
+ { 0x00, 0x06, NULL, NULL },
+ { 0x07, 0x07, input_end_bel, &input_state_ground },
+ { 0x08, 0x17, NULL, NULL },
+ { 0x19, 0x19, NULL, NULL },
+ { 0x1c, 0x1f, NULL, NULL },
+ { 0x20, 0xff, input_input, NULL },
+ { -1, -1, NULL, NULL }
+};
@@ -549,3 +507,6 @@ static const struct input_transition input_state_apc_string_table[] = {
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
+ { 0x00, 0x17, NULL, NULL },
+ { 0x19, 0x19, NULL, NULL },
+ { 0x1c, 0x1f, NULL, NULL },
+ { 0x20, 0xff, input_input, NULL },
+ { -1, -1, NULL, NULL }
+};
@@ -554,3 +515,6 @@ static const struct input_transition input_state_rename_string_table[] = {
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
+ { 0x00, 0x17, NULL, NULL },
+ { 0x19, 0x19, NULL, NULL },
+ { 0x1c, 0x1f, NULL, NULL },
+ { 0x20, 0xff, input_input, NULL },
+ { -1, -1, NULL, NULL }
+};
@@ -559,4 +523,9 @@ static const struct input_transition input_state_consume_st_table[] = {
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static int input_table_compare(const void *key, const void *value) {
+ { 0x00, 0x17, NULL, NULL },
+ { 0x19, 0x19, NULL, NULL },
+ { 0x1c, 0x1f, NULL, NULL },
+ { 0x20, 0xff, NULL, NULL },
+ { -1, -1, NULL, NULL }
+};
+static int
+input_table_compare(const void *key, const void *value)
+{
@@ -565 +534,2 @@ static int input_table_compare(const void *key, const void *value) {
-  if (ictx->ch != entry->ch) return (ictx->ch - entry->ch);
+ if (ictx->ch != entry->ch)
+  return (ictx->ch - entry->ch);
@@ -568,2 +538,3 @@ static int input_table_compare(const void *key, const void *value) {
-static void input_timer_callback(__unused int fd, __unused short events,
-                                 void *arg) {
+static void
+input_timer_callback(__unused int fd, __unused short events, void *arg)
+{
@@ -574 +545,3 @@ static void input_timer_callback(__unused int fd, __unused short events,
-static void input_start_timer(struct input_ctx *ictx) {
+static void
+input_start_timer(struct input_ctx *ictx)
+{
@@ -579 +552,3 @@ static void input_start_timer(struct input_ctx *ictx) {
-static void input_reset_cell(struct input_ctx *ictx) {
+static void
+input_reset_cell(struct input_ctx *ictx)
+{
@@ -587 +562,3 @@ static void input_reset_cell(struct input_ctx *ictx) {
-static void input_save_state(struct input_ctx *ictx) {
+static void
+input_save_state(struct input_ctx *ictx)
+{
@@ -595 +572,3 @@ static void input_save_state(struct input_ctx *ictx) {
-static void input_restore_state(struct input_ctx *ictx) {
+static void
+input_restore_state(struct input_ctx *ictx)
+{
@@ -604,2 +583,4 @@ static void input_restore_state(struct input_ctx *ictx) {
-struct input_ctx *input_init(struct window_pane *wp, struct bufferevent *bev,
-                             struct colour_palette *palette) {
+struct input_ctx *
+input_init(struct window_pane *wp, struct bufferevent *bev,
+    struct colour_palette *palette)
+{
@@ -614 +595,2 @@ struct input_ctx *input_init(struct window_pane *wp, struct bufferevent *bev,
-  if (ictx->since_ground == NULL) fatalx("out of memory");
+ if (ictx->since_ground == NULL)
+  fatalx("out of memory");
@@ -619 +601,3 @@ struct input_ctx *input_init(struct window_pane *wp, struct bufferevent *bev,
-void input_free(struct input_ctx *ictx) {
+void
+input_free(struct input_ctx *ictx)
+{
@@ -622 +606,2 @@ void input_free(struct input_ctx *ictx) {
-    if (ictx->param_list[i].type == INPUT_STRING) free(ictx->param_list[i].str);
+  if (ictx->param_list[i].type == INPUT_STRING)
+   free(ictx->param_list[i].str);
@@ -629 +614,3 @@ void input_free(struct input_ctx *ictx) {
-void input_reset(struct input_ctx *ictx, int clear) {
+void
+input_reset(struct input_ctx *ictx, int clear)
+{
@@ -646 +633,3 @@ void input_reset(struct input_ctx *ictx, int clear) {
-struct evbuffer *input_pending(struct input_ctx *ictx) {
+struct evbuffer *
+input_pending(struct input_ctx *ictx)
+{
@@ -649,3 +638,5 @@ struct evbuffer *input_pending(struct input_ctx *ictx) {
-static void input_set_state(struct input_ctx *ictx,
-                            const struct input_transition *itr) {
-  if (ictx->state->exit != NULL) ictx->state->exit(ictx);
+static void
+input_set_state(struct input_ctx *ictx, const struct input_transition *itr)
+{
+ if (ictx->state->exit != NULL)
+  ictx->state->exit(ictx);
@@ -653 +644,2 @@ static void input_set_state(struct input_ctx *ictx,
-  if (ictx->state->enter != NULL) ictx->state->enter(ictx);
+ if (ictx->state->enter != NULL)
+  ictx->state->enter(ictx);
@@ -655 +647,3 @@ static void input_set_state(struct input_ctx *ictx,
-static void input_parse(struct input_ctx *ictx, u_char *buf, size_t len) {
+static void
+input_parse(struct input_ctx *ictx, u_char *buf, size_t len)
+{
@@ -662 +656,3 @@ static void input_parse(struct input_ctx *ictx, u_char *buf, size_t len) {
-    if (ictx->state != state || itr == NULL || ictx->ch < itr->first ||
+  if (ictx->state != state ||
+      itr == NULL ||
+      ictx->ch < itr->first ||
@@ -666 +662,3 @@ static void input_parse(struct input_ctx *ictx, u_char *buf, size_t len) {
-        if (ictx->ch >= itr->first && ictx->ch <= itr->last) break;
+    if (ictx->ch >= itr->first &&
+        ictx->ch <= itr->last)
+     break;
@@ -674,3 +672,6 @@ static void input_parse(struct input_ctx *ictx, u_char *buf, size_t len) {
-    if (itr->handler != input_print) screen_write_collect_end(sctx);
-    if (itr->handler != NULL && itr->handler(ictx) != 0) continue;
-    if (itr->state != NULL) input_set_state(ictx, itr);
+  if (itr->handler != input_print)
+   screen_write_collect_end(sctx);
+  if (itr->handler != NULL && itr->handler(ictx) != 0)
+   continue;
+  if (itr->state != NULL)
+   input_set_state(ictx, itr);
@@ -681 +682,3 @@ static void input_parse(struct input_ctx *ictx, u_char *buf, size_t len) {
-void input_parse_pane(struct window_pane *wp) {
+void
+input_parse_pane(struct window_pane *wp)
+{
@@ -688 +691,3 @@ void input_parse_pane(struct window_pane *wp) {
-void input_parse_buffer(struct window_pane *wp, u_char *buf, size_t len) {
+void
+input_parse_buffer(struct window_pane *wp, u_char *buf, size_t len)
+{
@@ -691 +696,2 @@ void input_parse_buffer(struct window_pane *wp, u_char *buf, size_t len) {
-  if (len == 0) return;
+ if (len == 0)
+  return;
@@ -694 +700,2 @@ void input_parse_buffer(struct window_pane *wp, u_char *buf, size_t len) {
-  if (!TAILQ_EMPTY(&wp->modes)) wp->flags |= PANE_UNSEENCHANGES;
+ if (!TAILQ_EMPTY(&wp->modes))
+  wp->flags |= PANE_UNSEENCHANGES;
@@ -699,2 +706,2 @@ void input_parse_buffer(struct window_pane *wp, u_char *buf, size_t len) {
-  log_debug("%s: %%%u %s, %zu bytes: %.*s", __func__, wp->id, ictx->state->name,
-            len, (int)len, buf);
+ log_debug("%s: %%%u %s, %zu bytes: %.*s", __func__, wp->id,
+     ictx->state->name, len, (int)len, buf);
@@ -704,3 +711,4 @@ void input_parse_buffer(struct window_pane *wp, u_char *buf, size_t len) {
-void input_parse_screen(struct input_ctx *ictx, struct screen *s,
-                        screen_write_init_ctx_cb cb, void *arg, u_char *buf,
-                        size_t len) {
+void
+input_parse_screen(struct input_ctx *ictx, struct screen *s,
+    screen_write_init_ctx_cb cb, void *arg, u_char *buf, size_t len)
+{
@@ -708 +716,2 @@ void input_parse_screen(struct input_ctx *ictx, struct screen *s,
-  if (len == 0) return;
+ if (len == 0)
+  return;
@@ -713 +722,3 @@ void input_parse_screen(struct input_ctx *ictx, struct screen *s,
-static int input_split(struct input_ctx *ictx) {
+static int
+input_split(struct input_ctx *ictx)
+{
@@ -719 +730,2 @@ static int input_split(struct input_ctx *ictx) {
-    if (ictx->param_list[i].type == INPUT_STRING) free(ictx->param_list[i].str);
+  if (ictx->param_list[i].type == INPUT_STRING)
+   free(ictx->param_list[i].str);
@@ -722 +734,2 @@ static int input_split(struct input_ctx *ictx) {
-  if (ictx->param_len == 0) return (0);
+ if (ictx->param_len == 0)
+  return (0);
@@ -735 +748,2 @@ static int input_split(struct input_ctx *ictx) {
-        if (errstr != NULL) return (-1);
+    if (errstr != NULL)
+     return (-1);
@@ -739 +753,2 @@ static int input_split(struct input_ctx *ictx) {
-    if (ictx->param_list_len == nitems(ictx->param_list)) return (-1);
+  if (ictx->param_list_len == nitems(ictx->param_list))
+   return (-1);
@@ -752,2 +767,3 @@ static int input_split(struct input_ctx *ictx) {
-static int input_get(struct input_ctx *ictx, u_int validx, int minval,
-                     int defval) {
+static int
+input_get(struct input_ctx *ictx, u_int validx, int minval, int defval)
+{
@@ -756 +772,2 @@ static int input_get(struct input_ctx *ictx, u_int validx, int minval,
-  if (validx >= ictx->param_list_len) return (defval);
+ if (validx >= ictx->param_list_len)
+     return (defval);
@@ -758,2 +775,4 @@ static int input_get(struct input_ctx *ictx, u_int validx, int minval,
-  if (ip->type == INPUT_MISSING) return (defval);
-  if (ip->type == INPUT_STRING) return (-1);
+ if (ip->type == INPUT_MISSING)
+  return (defval);
+ if (ip->type == INPUT_STRING)
+  return (-1);
@@ -761 +780,2 @@ static int input_get(struct input_ctx *ictx, u_int validx, int minval,
-  if (retval < minval) return (minval);
+ if (retval < minval)
+  return (minval);
@@ -764 +784,3 @@ static int input_get(struct input_ctx *ictx, u_int validx, int minval,
-static void input_reply(struct input_ctx *ictx, const char *fmt, ...) {
+static void
+input_reply(struct input_ctx *ictx, const char *fmt, ...)
+{
@@ -768 +790,2 @@ static void input_reply(struct input_ctx *ictx, const char *fmt, ...) {
-  if (bev == NULL) return;
+ if (bev == NULL)
+  return;
@@ -776 +799,3 @@ static void input_reply(struct input_ctx *ictx, const char *fmt, ...) {
-static void input_clear(struct input_ctx *ictx) {
+static void
+input_clear(struct input_ctx *ictx)
+{
@@ -787 +812,3 @@ static void input_clear(struct input_ctx *ictx) {
-static void input_ground(struct input_ctx *ictx) {
+static void
+input_ground(struct input_ctx *ictx)
+{
@@ -795 +822,3 @@ static void input_ground(struct input_ctx *ictx) {
-static int input_print(struct input_ctx *ictx) {
+static int
+input_print(struct input_ctx *ictx)
+{
@@ -810 +839,3 @@ static int input_print(struct input_ctx *ictx) {
-static int input_intermediate(struct input_ctx *ictx) {
+static int
+input_intermediate(struct input_ctx *ictx)
+{
@@ -819 +850,3 @@ static int input_intermediate(struct input_ctx *ictx) {
-static int input_parameter(struct input_ctx *ictx) {
+static int
+input_parameter(struct input_ctx *ictx)
+{
@@ -828 +861,3 @@ static int input_parameter(struct input_ctx *ictx) {
-static int input_input(struct input_ctx *ictx) {
+static int
+input_input(struct input_ctx *ictx)
+{
@@ -844 +879,3 @@ static int input_input(struct input_ctx *ictx) {
-static int input_c0_dispatch(struct input_ctx *ictx) {
+static int
+input_c0_dispatch(struct input_ctx *ictx)
+{
@@ -854 +891,2 @@ static int input_c0_dispatch(struct input_ctx *ictx) {
-      if (wp != NULL) alerts_queue(wp->window, WINDOW_BELL);
+  if (wp != NULL)
+   alerts_queue(wp->window, WINDOW_BELL);
@@ -860 +898,2 @@ static int input_c0_dispatch(struct input_ctx *ictx) {
-      if (s->cx >= screen_size_x(s) - 1) break;
+  if (s->cx >= screen_size_x(s) - 1)
+   break;
@@ -863 +902,2 @@ static int input_c0_dispatch(struct input_ctx *ictx) {
-        if (bit_test(s->tabs, s->cx)) break;
+   if (bit_test(s->tabs, s->cx))
+    break;
@@ -870 +910,2 @@ static int input_c0_dispatch(struct input_ctx *ictx) {
-      if (s->mode & MODE_CRLF) screen_write_carriagereturn(sctx);
+  if (s->mode & MODE_CRLF)
+   screen_write_carriagereturn(sctx);
@@ -888 +929,3 @@ static int input_c0_dispatch(struct input_ctx *ictx) {
-static int input_esc_dispatch(struct input_ctx *ictx) {
+static int
+input_esc_dispatch(struct input_ctx *ictx)
+{
@@ -892 +935,2 @@ static int input_esc_dispatch(struct input_ctx *ictx) {
-  if (ictx->flags & INPUT_DISCARD) return (0);
+ if (ictx->flags & INPUT_DISCARD)
+  return (0);
@@ -915 +959,2 @@ static int input_esc_dispatch(struct input_ctx *ictx) {
-      if (s->cx < screen_size_x(s)) bit_set(s->tabs, s->cx);
+  if (s->cx < screen_size_x(s))
+   bit_set(s->tabs, s->cx);
@@ -953 +998,3 @@ static int input_esc_dispatch(struct input_ctx *ictx) {
-static int input_csi_dispatch(struct input_ctx *ictx) {
+static int
+input_csi_dispatch(struct input_ctx *ictx)
+{
@@ -959,4 +1006,6 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-  if (ictx->flags & INPUT_DISCARD) return (0);
-  log_debug("%s: '%c' \"%s\" \"%s\"", __func__, ictx->ch, ictx->interm_buf,
-            ictx->param_buf);
-  if (input_split(ictx) != 0) return (0);
+ if (ictx->flags & INPUT_DISCARD)
+  return (0);
+ log_debug("%s: '%c' \"%s\" \"%s\"", __func__, ictx->ch,
+     ictx->interm_buf, ictx->param_buf);
+ if (input_split(ictx) != 0)
+  return (0);
@@ -972 +1021,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (cx > screen_size_x(s) - 1) cx = screen_size_x(s) - 1;
+  if (cx > screen_size_x(s) - 1)
+   cx = screen_size_x(s) - 1;
@@ -974 +1024,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n == -1) break;
+  if (n == -1)
+   break;
@@ -976 +1027,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-        do cx--;
+   do
+    cx--;
@@ -983 +1035,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_cursorleft(sctx, n);
+  if (n != -1)
+   screen_write_cursorleft(sctx, n);
@@ -987 +1040,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_cursordown(sctx, n);
+  if (n != -1)
+   screen_write_cursordown(sctx, n);
@@ -991 +1045,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_cursorright(sctx, n);
+  if (n != -1)
+   screen_write_cursorright(sctx, n);
@@ -996 +1051,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1 && m != -1) screen_write_cursormove(sctx, m - 1, n - 1, 1);
+  if (n != -1 && m != -1)
+   screen_write_cursormove(sctx, m - 1, n - 1, 1);
@@ -1001 +1057,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (options_get_number(global_options, "extended-keys") == 2) break;
+  if (options_get_number(global_options, "extended-keys") == 2)
+   break;
@@ -1009 +1066,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n == 4) screen_write_mode_clear(sctx, MODE_KEXTENDED);
+  if (n == 4)
+   screen_write_mode_clear(sctx, MODE_KEXTENDED);
@@ -1016 +1074,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_cursorup(sctx, n);
+  if (n != -1)
+   screen_write_cursorup(sctx, n);
@@ -1062 +1121,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_clearcharacter(sctx, n, bg);
+  if (n != -1)
+   screen_write_clearcharacter(sctx, n, bg);
@@ -1066 +1126,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_deletecharacter(sctx, n, bg);
+  if (n != -1)
+   screen_write_deletecharacter(sctx, n, bg);
@@ -1071 +1132,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1 && m != -1) screen_write_scrollregion(sctx, n - 1, m - 1);
+  if (n != -1 && m != -1)
+   screen_write_scrollregion(sctx, n - 1, m - 1);
@@ -1075 +1137,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_deleteline(sctx, n, bg);
+  if (n != -1)
+   screen_write_deleteline(sctx, n, bg);
@@ -1135 +1198,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_cursormove(sctx, n - 1, -1, 1);
+  if (n != -1)
+   screen_write_cursormove(sctx, n - 1, -1, 1);
@@ -1139 +1203,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_insertcharacter(sctx, n, bg);
+  if (n != -1)
+   screen_write_insertcharacter(sctx, n, bg);
@@ -1143 +1208,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_insertline(sctx, n, bg);
+  if (n != -1)
+   screen_write_insertline(sctx, n, bg);
@@ -1147 +1213,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n == -1) break;
+  if (n == -1)
+   break;
@@ -1149,2 +1216,4 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n > m) n = m;
-      if (ictx->last == -1) break;
+  if (n > m)
+   n = m;
+  if (ictx->last == -1)
+   break;
@@ -1152 +1221,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      for (i = 0; i < n; i++) input_print(ictx);
+  for (i = 0; i < n; i++)
+   input_print(ictx);
@@ -1177 +1247,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_scrollup(sctx, n, bg);
+  if (n != -1)
+   screen_write_scrollup(sctx, n, bg);
@@ -1181 +1252,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_scrolldown(sctx, n, bg);
+  if (n != -1)
+   screen_write_scrolldown(sctx, n, bg);
@@ -1188 +1260,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-          if (s->cx < screen_size_x(s)) bit_clear(s->tabs, s->cx);
+   if (s->cx < screen_size_x(s))
+    bit_clear(s->tabs, s->cx);
@@ -1200 +1273,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_write_cursormove(sctx, -1, n - 1, 1);
+  if (n != -1)
+   screen_write_cursormove(sctx, -1, n - 1, 1);
@@ -1204 +1278,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n != -1) screen_set_cursor_style(n, &s->cstyle, &s->mode);
+  if (n != -1)
+   screen_set_cursor_style(n, &s->cstyle, &s->mode);
@@ -1208 +1283,2 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-      if (n == 0) input_reply(ictx, "\033P>|tmux %s\033\\", getversion());
+  if (n == 0)
+   input_reply(ictx, "\033P>|tmux %s\033\\", getversion());
@@ -1214 +1290,3 @@ static int input_csi_dispatch(struct input_ctx *ictx) {
-static void input_csi_dispatch_rm(struct input_ctx *ictx) {
+static void
+input_csi_dispatch_rm(struct input_ctx *ictx)
+{
@@ -1233 +1311,3 @@ static void input_csi_dispatch_rm(struct input_ctx *ictx) {
-static void input_csi_dispatch_rm_private(struct input_ctx *ictx) {
+static void
+input_csi_dispatch_rm_private(struct input_ctx *ictx)
+{
@@ -1293 +1373,3 @@ static void input_csi_dispatch_rm_private(struct input_ctx *ictx) {
-static void input_csi_dispatch_sm(struct input_ctx *ictx) {
+static void
+input_csi_dispatch_sm(struct input_ctx *ictx)
+{
@@ -1312 +1394,3 @@ static void input_csi_dispatch_sm(struct input_ctx *ictx) {
-static void input_csi_dispatch_sm_private(struct input_ctx *ictx) {
+static void
+input_csi_dispatch_sm_private(struct input_ctx *ictx)
+{
@@ -1378 +1462,3 @@ static void input_csi_dispatch_sm_private(struct input_ctx *ictx) {
-static void input_csi_dispatch_winops(struct input_ctx *ictx) {
+static void
+input_csi_dispatch_winops(struct input_ctx *ictx)
+{
@@ -1404 +1490,2 @@ static void input_csi_dispatch_winops(struct input_ctx *ictx) {
-        if (input_get(ictx, m, 0, -1) == -1) return;
+   if (input_get(ictx, m, 0, -1) == -1)
+    return;
@@ -1408 +1495,2 @@ static void input_csi_dispatch_winops(struct input_ctx *ictx) {
-        if (input_get(ictx, m, 0, -1) == -1) return;
+   if (input_get(ictx, m, 0, -1) == -1)
+    return;
@@ -1429 +1517,2 @@ static void input_csi_dispatch_winops(struct input_ctx *ictx) {
-            if (wp == NULL) break;
+    if (wp == NULL)
+     break;
@@ -1446,2 +1535,3 @@ static void input_csi_dispatch_winops(struct input_ctx *ictx) {
-static int input_csi_dispatch_sgr_256_do(struct input_ctx *ictx, int fgbg,
-                                         int c) {
+static int
+input_csi_dispatch_sgr_256_do(struct input_ctx *ictx, int fgbg, int c)
+{
@@ -1464,2 +1554,3 @@ static int input_csi_dispatch_sgr_256_do(struct input_ctx *ictx, int fgbg,
-static void input_csi_dispatch_sgr_256(struct input_ctx *ictx, int fgbg,
-                                       u_int *i) {
+static void
+input_csi_dispatch_sgr_256(struct input_ctx *ictx, int fgbg, u_int *i)
+{
@@ -1468 +1559,2 @@ static void input_csi_dispatch_sgr_256(struct input_ctx *ictx, int fgbg,
-  if (input_csi_dispatch_sgr_256_do(ictx, fgbg, c)) (*i)++;
+ if (input_csi_dispatch_sgr_256_do(ictx, fgbg, c))
+  (*i)++;
@@ -1470,2 +1562,4 @@ static void input_csi_dispatch_sgr_256(struct input_ctx *ictx, int fgbg,
-static int input_csi_dispatch_sgr_rgb_do(struct input_ctx *ictx, int fgbg,
-                                         int r, int g, int b) {
+static int
+input_csi_dispatch_sgr_rgb_do(struct input_ctx *ictx, int fgbg, int r, int g,
+    int b)
+{
@@ -1473,3 +1567,6 @@ static int input_csi_dispatch_sgr_rgb_do(struct input_ctx *ictx, int fgbg,
-  if (r == -1 || r > 255) return (0);
-  if (g == -1 || g > 255) return (0);
-  if (b == -1 || b > 255) return (0);
+ if (r == -1 || r > 255)
+  return (0);
+ if (g == -1 || g > 255)
+  return (0);
+ if (b == -1 || b > 255)
+  return (0);
@@ -1484,2 +1581,3 @@ static int input_csi_dispatch_sgr_rgb_do(struct input_ctx *ictx, int fgbg,
-static void input_csi_dispatch_sgr_rgb(struct input_ctx *ictx, int fgbg,
-                                       u_int *i) {
+static void
+input_csi_dispatch_sgr_rgb(struct input_ctx *ictx, int fgbg, u_int *i)
+{
@@ -1490 +1588,2 @@ static void input_csi_dispatch_sgr_rgb(struct input_ctx *ictx, int fgbg,
-  if (input_csi_dispatch_sgr_rgb_do(ictx, fgbg, r, g, b)) (*i) += 3;
+ if (input_csi_dispatch_sgr_rgb_do(ictx, fgbg, r, g, b))
+  (*i) += 3;
@@ -1492 +1591,3 @@ static void input_csi_dispatch_sgr_rgb(struct input_ctx *ictx, int fgbg,
-static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
+static void
+input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i)
+{
@@ -1498 +1599,2 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-  for (n = 0; n < nitems(p); n++) p[n] = -1;
+ for (n = 0; n < nitems(p); n++)
+  p[n] = -1;
@@ -1518 +1620,2 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-  if (n == 0) return;
+ if (n == 0)
+  return;
@@ -1520 +1623,2 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-    if (n != 2) return;
+  if (n != 2)
+   return;
@@ -1548 +1652,2 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-  if (n < 2 || (p[0] != 38 && p[0] != 48 && p[0] != 58)) return;
+ if (n < 2 || (p[0] != 38 && p[0] != 48 && p[0] != 58))
+  return;
@@ -1551 +1656,2 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-      if (n < 3) break;
+  if (n < 3)
+   break;
@@ -1556,2 +1662,4 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-      if (n < i + 3) break;
-      input_csi_dispatch_sgr_rgb_do(ictx, p[0], p[i], p[i + 1], p[i + 2]);
+  if (n < i + 3)
+   break;
+  input_csi_dispatch_sgr_rgb_do(ictx, p[0], p[i], p[i + 1],
+      p[i + 2]);
@@ -1560 +1668,2 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-      if (n < 3) break;
+  if (n < 3)
+   break;
@@ -1565 +1674,3 @@ static void input_csi_dispatch_sgr_colon(struct input_ctx *ictx, u_int i) {
-static void input_csi_dispatch_sgr(struct input_ctx *ictx) {
+static void
+input_csi_dispatch_sgr(struct input_ctx *ictx)
+{
@@ -1579 +1690,2 @@ static void input_csi_dispatch_sgr(struct input_ctx *ictx) {
-    if (n == -1) continue;
+  if (n == -1)
+   continue;
@@ -1707 +1819,3 @@ static void input_csi_dispatch_sgr(struct input_ctx *ictx) {
-static int input_end_bel(struct input_ctx *ictx) {
+static int
+input_end_bel(struct input_ctx *ictx)
+{
@@ -1712 +1826,3 @@ static int input_end_bel(struct input_ctx *ictx) {
-static void input_enter_dcs(struct input_ctx *ictx) {
+static void
+input_enter_dcs(struct input_ctx *ictx)
+{
@@ -1718 +1834,3 @@ static void input_enter_dcs(struct input_ctx *ictx) {
-static int input_dcs_dispatch(struct input_ctx *ictx) {
+static int
+input_dcs_dispatch(struct input_ctx *ictx)
+{
@@ -1730 +1848,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  if (wp == NULL) return (0);
+ if (wp == NULL)
+  return (0);
@@ -1734 +1852,0 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-<<<<<<< HEAD
@@ -1738 +1856,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (si != NULL) screen_write_sixelimage(sctx, si, ictx->cell.cell.bg);
+  if (si != NULL)
+   screen_write_sixelimage(sctx, si, ictx->cell.cell.bg);
@@ -1742,8 +1861,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-|||||||
-    allow_passthrough = options_get_number(wp->options, "allow-passthrough");
-=======
-  }
-  log_debug("%s: %zu bytes", __func__, len);
-  allow_passthrough = options_get_number(wp->options, "allow-passthrough");
->>>>>>> 71d453f169ffac345db3e4c82f4409fc41091002
-    if (!allow_passthrough) return (0);
+ if (!allow_passthrough)
+  return (0);
@@ -1757 +1870,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_enter_osc(struct input_ctx * ictx) {
+static void
+input_enter_osc(struct input_ctx *ictx)
+{
@@ -1763 +1878,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_exit_osc(struct input_ctx * ictx) {
+static void
+input_exit_osc(struct input_ctx *ictx)
+{
@@ -1768,2 +1885,4 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (ictx->flags & INPUT_DISCARD) return;
-    if (ictx->input_len < 1 || *p < '0' || *p > '9') return;
+ if (ictx->flags & INPUT_DISCARD)
+  return;
+ if (ictx->input_len < 1 || *p < '0' || *p > '9')
+  return;
@@ -1773,3 +1892,6 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    while (*p >= '0' && *p <= '9') option = option * 10 + *p++ - '0';
-    if (*p != ';' && *p != '\0') return;
-    if (*p == ';') p++;
+ while (*p >= '0' && *p <= '9')
+  option = option * 10 + *p++ - '0';
+ if (*p != ';' && *p != '\0')
+  return;
+ if (*p == ';')
+  p++;
@@ -1832 +1954,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_enter_apc(struct input_ctx * ictx) {
+static void
+input_enter_apc(struct input_ctx *ictx)
+{
@@ -1838 +1962,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_exit_apc(struct input_ctx * ictx) {
+static void
+input_exit_apc(struct input_ctx *ictx)
+{
@@ -1841 +1967,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (ictx->flags & INPUT_DISCARD) return;
+ if (ictx->flags & INPUT_DISCARD)
+  return;
@@ -1849 +1976,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_enter_rename(struct input_ctx * ictx) {
+static void
+input_enter_rename(struct input_ctx *ictx)
+{
@@ -1855 +1984,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_exit_rename(struct input_ctx * ictx) {
+static void
+input_exit_rename(struct input_ctx *ictx)
+{
@@ -1859,3 +1990,6 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (wp == NULL) return;
-    if (ictx->flags & INPUT_DISCARD) return;
-    if (!options_get_number(ictx->wp->options, "allow-rename")) return;
+ if (wp == NULL)
+  return;
+ if (ictx->flags & INPUT_DISCARD)
+  return;
+ if (!options_get_number(ictx->wp->options, "allow-rename"))
+  return;
@@ -1863 +1997,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (!utf8_isvalid(ictx->input_buf)) return;
+ if (!utf8_isvalid(ictx->input_buf))
+  return;
@@ -1867 +2002,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (o != NULL) options_remove_or_default(o, -1, NULL);
+  if (o != NULL)
+   options_remove_or_default(o, -1, NULL);
@@ -1877 +2013,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static int input_top_bit_set(struct input_ctx * ictx) {
+static int
+input_top_bit_set(struct input_ctx *ictx)
+{
@@ -1882 +2020,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (utf8_open(ud, ictx->ch) != UTF8_MORE) return (0);
+  if (utf8_open(ud, ictx->ch) != UTF8_MORE)
+   return (0);
@@ -1896,2 +2035,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    log_debug("%s %hhu '%*s' (width %hhu)", __func__, ud->size, (int)ud->size,
-              ud->data, ud->width);
+ log_debug("%s %hhu '%*s' (width %hhu)", __func__, ud->size,
+     (int)ud->size, ud->data, ud->width);
@@ -1902 +2041,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_colour_reply(struct input_ctx * ictx, u_int n, int c) {
+static void
+input_osc_colour_reply(struct input_ctx *ictx, u_int n, int c)
+{
@@ -1905,2 +2046,4 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (c != -1) c = colour_force_rgb(c);
-    if (c == -1) return;
+    if (c != -1)
+     c = colour_force_rgb(c);
+    if (c == -1)
+     return;
@@ -1912,2 +2055,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    input_reply(ictx, "\033]%u;rgb:%02hhx%02hhx/%02hhx%02hhx/%02hhx%02hhx%s", n,
-                r, r, g, g, b, b, end);
+    input_reply(ictx, "\033]%u;rgb:%02hhx%02hhx/%02hhx%02hhx/%02hhx%02hhx%s",
+ n, r, r, g, g, b, b, end);
@@ -1915 +2058,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_4(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_4(struct input_ctx *ictx, const char *p)
+{
@@ -1933 +2078,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-        if (c != -1) input_osc_colour_reply(ictx, 4, c);
+   if (c != -1)
+    input_osc_colour_reply(ictx, 4, c);
@@ -1940 +2086,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (colour_palette_set(ictx->palette, idx, c)) redraw = 1;
+  if (colour_palette_set(ictx->palette, idx, c))
+   redraw = 1;
@@ -1943,2 +2090,4 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (bad) log_debug("bad OSC 4: %s", p);
-    if (redraw) screen_write_fullredraw(&ictx->ctx);
+ if (bad)
+  log_debug("bad OSC 4: %s", p);
+ if (redraw)
+  screen_write_fullredraw(&ictx->ctx);
@@ -1947 +2096,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_8(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_8(struct input_ctx *ictx, const char *p)
+{
@@ -1954 +2105,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-        if (id != NULL) goto bad;
+   if (id != NULL)
+    goto bad;
@@ -1957 +2109,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (*end == ';') break;
+  if (*end == ';')
+   break;
@@ -1959 +2112,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (end == NULL || *end != ';') goto bad;
+ if (end == NULL || *end != ';')
+  goto bad;
@@ -1977 +2131,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static int input_get_fg_client(struct window_pane * wp) {
+static int
+input_get_fg_client(struct window_pane *wp)
+{
@@ -1981,3 +2137,6 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (loop->flags & CLIENT_UNATTACHEDFLAGS) continue;
-      if (loop->session == NULL || !session_has(loop->session, w)) continue;
-      if (loop->tty.fg == -1) continue;
+  if (loop->flags & CLIENT_UNATTACHEDFLAGS)
+   continue;
+  if (loop->session == NULL || !session_has(loop->session, w))
+   continue;
+  if (loop->tty.fg == -1)
+   continue;
@@ -1988 +2147,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static int input_get_bg_client(struct window_pane * wp) {
+static int
+input_get_bg_client(struct window_pane *wp)
+{
@@ -1992,3 +2153,6 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (loop->flags & CLIENT_UNATTACHEDFLAGS) continue;
-      if (loop->session == NULL || !session_has(loop->session, w)) continue;
-      if (loop->tty.bg == -1) continue;
+  if (loop->flags & CLIENT_UNATTACHEDFLAGS)
+   continue;
+  if (loop->session == NULL || !session_has(loop->session, w))
+   continue;
+  if (loop->tty.bg == -1)
+   continue;
@@ -1999 +2163,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_10(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_10(struct input_ctx *ictx, const char *p)
+{
@@ -2004 +2170,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (wp == NULL) return;
+  if (wp == NULL)
+   return;
@@ -2019 +2186,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (wp != NULL) wp->flags |= PANE_STYLECHANGED;
+  if (wp != NULL)
+   wp->flags |= PANE_STYLECHANGED;
@@ -2023 +2191,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_110(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_110(struct input_ctx *ictx, const char *p)
+{
@@ -2025 +2195,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (*p != '\0') return;
+ if (*p != '\0')
+  return;
@@ -2028 +2199,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (wp != NULL) wp->flags |= PANE_STYLECHANGED;
+  if (wp != NULL)
+   wp->flags |= PANE_STYLECHANGED;
@@ -2032 +2204,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_11(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_11(struct input_ctx *ictx, const char *p)
+{
@@ -2037 +2211,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (wp == NULL) return;
+  if (wp == NULL)
+   return;
@@ -2052 +2227,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (wp != NULL) wp->flags |= PANE_STYLECHANGED;
+  if (wp != NULL)
+   wp->flags |= PANE_STYLECHANGED;
@@ -2056 +2232,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_111(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_111(struct input_ctx *ictx, const char *p)
+{
@@ -2058 +2236,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (*p != '\0') return;
+ if (*p != '\0')
+  return;
@@ -2061 +2240,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (wp != NULL) wp->flags |= PANE_STYLECHANGED;
+  if (wp != NULL)
+   wp->flags |= PANE_STYLECHANGED;
@@ -2065 +2245,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_12(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_12(struct input_ctx *ictx, const char *p)
+{
@@ -2071 +2253,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-        if (c == -1) c = ictx->ctx.s->default_ccolour;
+   if (c == -1)
+    c = ictx->ctx.s->default_ccolour;
@@ -2082 +2265,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_112(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_112(struct input_ctx *ictx, const char *p)
+{
@@ -2086 +2271,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_133(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_133(struct input_ctx *ictx, const char *p)
+{
@@ -2090 +2277,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (line > gd->hsize + gd->sy - 1) return;
+ if (line > gd->hsize + gd->sy - 1)
+  return;
@@ -2101 +2289,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_52(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_52(struct input_ctx *ictx, const char *p)
+{
@@ -2113 +2303,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (wp == NULL) return;
+ if (wp == NULL)
+  return;
@@ -2115,2 +2306,4 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (state != 2) return;
-    if ((end = strchr(p, ';')) == NULL) return;
+ if (state != 2)
+  return;
+ if ((end = strchr(p, ';')) == NULL)
+  return;
@@ -2118 +2311,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (*end == '\0') return;
+ if (*end == '\0')
+  return;
@@ -2126 +2320,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if ((pb = paste_get_top(NULL)) != NULL) buf = paste_buffer_data(pb, &len);
+  if ((pb = paste_get_top(NULL)) != NULL)
+   buf = paste_buffer_data(pb, &len);
@@ -2134 +2329,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (len == 0) return;
+ if (len == 0)
+  return;
@@ -2146 +2342,3 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  static void input_osc_104(struct input_ctx * ictx, const char *p) {
+static void
+input_osc_104(struct input_ctx *ictx, const char *p)
+{
@@ -2166,2 +2364,4 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (colour_palette_set(ictx->palette, idx, -1)) redraw = 1;
-      if (*s == ';') s++;
+  if (colour_palette_set(ictx->palette, idx, -1))
+   redraw = 1;
+  if (*s == ';')
+   s++;
@@ -2169,2 +2369,4 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (bad) log_debug("bad OSC 104: %s", p);
-    if (redraw) screen_write_fullredraw(&ictx->ctx);
+ if (bad)
+  log_debug("bad OSC 104: %s", p);
+ if (redraw)
+  screen_write_fullredraw(&ictx->ctx);
@@ -2173,2 +2375,4 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-  void input_reply_clipboard(struct bufferevent * bev, const char *buf,
-                             size_t len, const char *end) {
+void
+input_reply_clipboard(struct bufferevent *bev, const char *buf, size_t len,
+    const char *end)
+{
@@ -2178 +2382,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-      if (len >= ((size_t)INT_MAX * 3 / 4) - 1) return;
+  if (len >= ((size_t)INT_MAX * 3 / 4) - 1)
+   return;
@@ -2187 +2392,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-    if (outlen != 0) bufferevent_write(bev, out, outlen);
+ if (outlen != 0)
+  bufferevent_write(bev, out, outlen);
