--- a/./output/tmux/071849f8-71d453f1-8636848e/input.no_comments_mergegen.c
+++ b/./output/tmux/071849f8-71d453f1-8636848e/input.no_comments_merged.c
@@ -1853 +1853,8 @@ input_dcs_dispatch(struct input_ctx *ictx)
-#ifdef ENABLE_SIXEL if (buf[0] == 'q') { si = sixel_parse(buf, len, w->xpixel, w->ypixel); if (si != NULL) screen_write_sixelimage(sctx, si, ictx->cell.cell.bg); } #endif
+#ifdef ENABLE_SIXEL
+ if (buf[0] == 'q') {
+  si = sixel_parse(buf, len, w->xpixel, w->ypixel);
+  if (si != NULL)
+   screen_write_sixelimage(sctx, si, ictx->cell.cell.bg);
+ }
+#endif
+ allow_passthrough = options_get_number(wp->options, "allow-passthrough");
