diff --git a/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@duckdb_python.no_comments_mergegen.cpp b/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@duckdb_python.no_comments_merged.cpp
index f488f53..9cc445d 100644
--- a/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@duckdb_python.no_comments_mergegen.cpp
+++ b/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@duckdb_python.no_comments_merged.cpp
@@ -9,0 +10 @@
+#include "duckdb_python/typing.hpp"
