--- a/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@python_import_cache.no_comments_mergegen.cpp
+++ b/./output/duckdb/b2a4c3bb-cc616059-754ff320/tools@pythonpkg@src@python_import_cache.no_comments_merged.cpp
@@ -33 +33,6 @@ void PythonImportCacheItem::LoadAttribute(const string &name, PythonImportCache
-auto source_object = source(); if (py::hasattr(source_object, name.c_str())) { object = AddCache(cache, std::move(source_object.attr(name.c_str()))); } else { object = nullptr; return; }
+ if (py::hasattr(source_object, name.c_str())) {
+  object = AddCache(cache, std::move(source_object.attr(name.c_str())));
+ } else {
+  object = nullptr;
+  return;
+ }
