--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/ee799cc9-d4433288-7869c733/include@android@surface_control.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/ee799cc9-d4433288-7869c733/include@android@surface_control.no_comments_merged.h
@@ -12,4 +12,9 @@ typedef struct ASurfaceControl ASurfaceControl;
-void ASurfaceControl_acquire(
-        ASurfaceControl* _Nonnull surface_control)
-        void ASurfaceControl_release(
-                ASurfaceControl* _Nonnull surface_control) struct ASurfaceTransaction;
+ASurfaceControl* _Nullable ASurfaceControl_createFromWindow(ANativeWindow* _Nonnull parent,
+                                                            const char* _Nonnull debug_name)
+        __INTRODUCED_IN(29);
+ASurfaceControl* _Nullable ASurfaceControl_create(ASurfaceControl* _Nonnull parent,
+                                                  const char* _Nonnull debug_name)
+        __INTRODUCED_IN(29);
+void ASurfaceControl_acquire(ASurfaceControl* _Nonnull surface_control) __INTRODUCED_IN(31);
+void ASurfaceControl_release(ASurfaceControl* _Nonnull surface_control) __INTRODUCED_IN(29);
+struct ASurfaceTransaction;
@@ -17,4 +22,3 @@ typedef struct ASurfaceTransaction ASurfaceTransaction;
-void ASurfaceTransaction_delete(
-        ASurfaceTransaction* _Nullable transaction)
-        void ASurfaceTransaction_apply(
-                ASurfaceTransaction* _Nonnull transaction)
+ASurfaceTransaction* _Nonnull ASurfaceTransaction_create() __INTRODUCED_IN(29);
+void ASurfaceTransaction_delete(ASurfaceTransaction* _Nullable transaction) __INTRODUCED_IN(29);
+void ASurfaceTransaction_apply(ASurfaceTransaction* _Nonnull transaction) __INTRODUCED_IN(29);
@@ -23,0 +28 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
+        __INTRODUCED_IN(29);
@@ -25,0 +31 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
+        __INTRODUCED_IN(31);
@@ -27 +33 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-                ASurfaceTransactionStats* _Nonnull surface_transaction_stats)
+        ASurfaceTransactionStats* _Nonnull surface_transaction_stats) __INTRODUCED_IN(29);
@@ -29 +35 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-                ASurfaceTransactionStats* _Nonnull surface_transaction_stats)
+        ASurfaceTransactionStats* _Nonnull surface_transaction_stats) __INTRODUCED_IN(29);
@@ -33 +39 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-                size_t* _Nonnull outASurfaceControlsSize)
+        size_t* _Nonnull outASurfaceControlsSize) __INTRODUCED_IN(29);
@@ -35 +41 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-                ASurfaceControl* _Nonnull* _Nonnull surface_controls)
+        ASurfaceControl* _Nonnull* _Nonnull surface_controls) __INTRODUCED_IN(29);
@@ -38 +44 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-                ASurfaceControl* _Nonnull surface_control)
+        ASurfaceControl* _Nonnull surface_control) __INTRODUCED_IN(29);
@@ -41 +47 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-                ASurfaceControl* _Nonnull surface_control)
+        ASurfaceControl* _Nonnull surface_control) __INTRODUCED_IN(29);
@@ -44,0 +51 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
+        __INTRODUCED_IN(29);
@@ -48,2 +55,2 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-        void ASurfaceTransaction_reparent(
-                ASurfaceTransaction* _Nonnull transaction,
+        __INTRODUCED_IN(31);
+void ASurfaceTransaction_reparent(ASurfaceTransaction* _Nonnull transaction,
@@ -51 +58 @@ typedef void (*ASurfaceTransaction_OnComplete)(void* _Null_unspecified context,
-                ASurfaceControl* _Nullable new_parent)
+                                  ASurfaceControl* _Nullable new_parent) __INTRODUCED_IN(29);
@@ -58,0 +66 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
+        __INTRODUCED_IN(29);
@@ -60,2 +68,2 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
-                                           ASurfaceControl* _Nonnull surface_control,
-                                           int32_t z_order)
+                                   ASurfaceControl* _Nonnull surface_control, int32_t z_order)
+        __INTRODUCED_IN(29);
@@ -64,0 +73 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
+        __INTRODUCED_IN(29);
@@ -66,2 +75,3 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
-                                          ASurfaceControl* _Nonnull surface_control, float r,
-                                          float g, float b, float alpha, enum ADataSpace dataspace)
+                                  ASurfaceControl* _Nonnull surface_control, float r, float g,
+                                  float b, float alpha, enum ADataSpace dataspace)
+        __INTRODUCED_IN(29);
@@ -69,3 +79,3 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
-                                             ASurfaceControl* _Nonnull surface_control,
-                                             const ARect& source, const ARect& destination,
-                                             int32_t transform)
+                                     ASurfaceControl* _Nonnull surface_control, const ARect& source,
+                                     const ARect& destination, int32_t transform)
+        __INTRODUCED_IN(29);
@@ -73,4 +83,3 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
-                                         ASurfaceControl* _Nonnull surface_control,
-                                         const ARect& crop)
-        void ASurfaceTransaction_setPosition(
-                ASurfaceTransaction* _Nonnull transaction,
+                                 ASurfaceControl* _Nonnull surface_control, const ARect& crop)
+        __INTRODUCED_IN(31);
+void ASurfaceTransaction_setPosition(ASurfaceTransaction* _Nonnull transaction,
@@ -78,3 +87,2 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
-                int32_t y)
-        void ASurfaceTransaction_setBufferTransform(
-                ASurfaceTransaction* _Nonnull transaction,
+                                     int32_t y) __INTRODUCED_IN(31);
+void ASurfaceTransaction_setBufferTransform(ASurfaceTransaction* _Nonnull transaction,
@@ -82,3 +90,2 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
-                int32_t transform)
-        void ASurfaceTransaction_setScale(
-                ASurfaceTransaction* _Nonnull transaction,
+                                            int32_t transform) __INTRODUCED_IN(31);
+void ASurfaceTransaction_setScale(ASurfaceTransaction* _Nonnull transaction,
@@ -86 +93 @@ void ASurfaceTransaction_setVisibility(ASurfaceTransaction* _Nonnull transaction
-                float yScale)
+                                  float yScale) __INTRODUCED_IN(31);
@@ -94,0 +102 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
+                                               __INTRODUCED_IN(29);
@@ -98,3 +106,3 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-        void ASurfaceTransaction_setDesiredPresentTime(
-                ASurfaceTransaction* _Nonnull transaction,
-                int64_t desiredPresentTime)
+                                         __INTRODUCED_IN(29);
+void ASurfaceTransaction_setDesiredPresentTime(ASurfaceTransaction* _Nonnull transaction,
+                                               int64_t desiredPresentTime) __INTRODUCED_IN(29);
@@ -101,0 +110,3 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
+                                        ASurfaceControl* _Nonnull surface_control, float alpha)
+                                        __INTRODUCED_IN(29);
+void ASurfaceTransaction_setBufferDataSpace(ASurfaceTransaction* _Nonnull transaction,
@@ -103,8 +114,2 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-                                                float alpha)
-        void ASurfaceTransaction_setBufferDataSpace(
-                ASurfaceTransaction* _Nonnull transaction,
-                ASurfaceControl* _Nonnull surface_control,
-                enum ADataSpace
-                        data_space)
-        void ASurfaceTransaction_setHdrMetadata_smpte2086(
-                ASurfaceTransaction* _Nonnull transaction,
+                                            enum ADataSpace data_space) __INTRODUCED_IN(29);
+void ASurfaceTransaction_setHdrMetadata_smpte2086(ASurfaceTransaction* _Nonnull transaction,
@@ -113,2 +118,2 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-        void ASurfaceTransaction_setHdrMetadata_cta861_3(
-                ASurfaceTransaction* _Nonnull transaction,
+                                                  __INTRODUCED_IN(29);
+void ASurfaceTransaction_setHdrMetadata_cta861_3(ASurfaceTransaction* _Nonnull transaction,
@@ -117,4 +122,5 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-        void ASurfaceTransaction_setExtendedRangeBrightness(
-                ASurfaceTransaction* _Nonnull transaction,
-                ASurfaceControl* _Nonnull surface_control, float currentBufferRatio,
-                float desiredRatio)
+                                                 __INTRODUCED_IN(29);
+void ASurfaceTransaction_setExtendedRangeBrightness(ASurfaceTransaction* _Nonnull transaction,
+                                                    ASurfaceControl* _Nonnull surface_control,
+                                                    float currentBufferRatio, float desiredRatio)
+                                                    __INTRODUCED_IN(__ANDROID_API_U__);
@@ -124,2 +130,2 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-        void ASurfaceTransaction_setFrameRate(
-                ASurfaceTransaction* _Nonnull transaction,
+        __INTRODUCED_IN(__ANDROID_API_V__);
+void ASurfaceTransaction_setFrameRate(ASurfaceTransaction* _Nonnull transaction,
@@ -127,4 +133,4 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-                int8_t compatibility)
-        void ASurfaceTransaction_setFrameRateWithChangeStrategy(
-                ASurfaceTransaction* _Nonnull transaction,
-                ASurfaceControl* _Nonnull surface_control, float frameRate, int8_t compatibility,
+                                      int8_t compatibility) __INTRODUCED_IN(30);
+void ASurfaceTransaction_setFrameRateWithChangeStrategy(ASurfaceTransaction* _Nonnull transaction,
+                                                        ASurfaceControl* _Nonnull surface_control,
+                                                        float frameRate, int8_t compatibility,
@@ -131,0 +138 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
+                                                        __INTRODUCED_IN(31);
@@ -134,2 +141,2 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-        void ASurfaceTransaction_setEnableBackPressure(
-                ASurfaceTransaction* _Nonnull transaction,
+                                        __INTRODUCED_IN(__ANDROID_API_U__);
+void ASurfaceTransaction_setEnableBackPressure(ASurfaceTransaction* _Nonnull transaction,
@@ -137,4 +144,5 @@ void ASurfaceTransaction_setBufferTransparency(ASurfaceTransaction* _Nonnull tra
-                bool enableBackPressure)
-        void ASurfaceTransaction_setFrameTimeline(
-                ASurfaceTransaction* _Nonnull transaction,
-                AVsyncId vsyncId) #endif
+                                               bool enableBackPressure) __INTRODUCED_IN(31);
+void ASurfaceTransaction_setFrameTimeline(ASurfaceTransaction* _Nonnull transaction,
+                                          AVsyncId vsyncId) __INTRODUCED_IN(33);
+__END_DECLS
+#endif
