--- a/./output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_conflict.cpp
+++ b/./output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_merged.cpp
@@ -51 +50,0 @@ MemoryHeapBase::MemoryHeapBase(size_t size, uint32_t flags, char const * name)
-<<<<<<< HEAD
@@ -54,7 +52,0 @@ MemoryHeapBase::MemoryHeapBase(size_t size, uint32_t flags, char const * name)
-||||||| 35388a2d76
-        const int SEAL_FLAGS = ((mFlags & READ_ONLY) ? F_SEAL_FUTURE_WRITE : 0) |
-                ((mFlags & MEMFD_ALLOW_SEALING) ? 0 : F_SEAL_SEAL);
-=======
-        const int SEAL_FLAGS = ((mFlags & READ_ONLY) ? F_SEAL_FUTURE_WRITE : 0) | F_SEAL_GROW |
-                F_SEAL_SHRINK | ((mFlags & MEMFD_ALLOW_SEALING) ? 0 : F_SEAL_SEAL);
->>>>>>> 99785d04
