diff --git a/./output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_mergegen.c b/./output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_merged.c
index 0a7d5ba..8988cdf 100644
--- a/./output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_mergegen.c
+++ b/./output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_merged.c
@@ -296,3 +296 @@ format_window_pane(struct format_tree *ft, struct window_pane *wp)
-=======
- format_add(ft, "pane_current_path", "%s", osdep_get_cwd(wp->pid));
->>>>>>> 253f1395
+ format_add(ft, "pane_current_path", "%s", osdep_get_cwd(wp->fd));
