--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@GraphicsEnv.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
@@ -44,9 +43,0 @@ static bool isVndkEnabled() {
-namespace {
-static bool isVndkEnabled() {
-#ifdef __BIONIC__
-    static bool isVndkEnabled = android::base::GetProperty("ro.vndk.version", "") != "";
-    return isVndkEnabled;
-#endif
-    return false;
-}
-}
@@ -104,0 +96,8 @@ static const std::string getSystemNativeLibraries(NativeLibrary type) {
+}
+static sp<IGpuService> getGpuService() {
+    static const sp<IBinder> binder = defaultServiceManager()->checkService(String16("gpu"));
+    if (!binder) {
+        ALOGE("Failed to get gpu service");
+        return nullptr;
+    }
+    return interface_cast<IGpuService>(binder);
@@ -133,0 +133,73 @@ void GraphicsEnv::setDriverPathAndSphalLibraries(const std::string& path,
+bool GraphicsEnv::linkDriverNamespaceLocked(android_namespace_t* destNamespace,
+                                            android_namespace_t* vndkNamespace,
+                                            const std::string& sharedSphalLibraries) {
+    const std::string llndkLibraries = getSystemNativeLibraries(NativeLibrary::LLNDK);
+    if (llndkLibraries.empty()) {
+        return false;
+    }
+    if (!android_link_namespaces(destNamespace, nullptr, llndkLibraries.c_str())) {
+        ALOGE("Failed to link default namespace[%s]", dlerror());
+        return false;
+    }
+    const std::string vndkspLibraries = getSystemNativeLibraries(NativeLibrary::VNDKSP);
+    if (vndkspLibraries.empty()) {
+        return false;
+    }
+    if (!android_link_namespaces(destNamespace, vndkNamespace, vndkspLibraries.c_str())) {
+        ALOGE("Failed to link vndk namespace[%s]", dlerror());
+        return false;
+    }
+    if (sharedSphalLibraries.empty()) {
+        return true;
+    }
+    auto sphalNamespace = android_get_exported_namespace("sphal");
+    if (!sphalNamespace) {
+        ALOGE("Depend on these libraries[%s] in sphal, but failed to get sphal namespace",
+              sharedSphalLibraries.c_str());
+        return false;
+    }
+    if (!android_link_namespaces(destNamespace, sphalNamespace, sharedSphalLibraries.c_str())) {
+        ALOGE("Failed to link sphal namespace[%s]", dlerror());
+        return false;
+    }
+    return true;
+}
+android_namespace_t* GraphicsEnv::getDriverNamespace() {
+    std::lock_guard<std::mutex> lock(mNamespaceMutex);
+    if (mDriverNamespace) {
+        return mDriverNamespace;
+    }
+    if (mDriverPath.empty()) {
+        const char* id = getenv("UPDATABLE_GFX_DRIVER");
+        if (id == nullptr || std::strcmp(id, "1") != 0) {
+            return nullptr;
+        }
+        const sp<IGpuService> gpuService = getGpuService();
+        if (!gpuService) {
+            return nullptr;
+        }
+        mDriverPath = gpuService->getUpdatableDriverPath();
+        if (mDriverPath.empty()) {
+            return nullptr;
+        }
+        mDriverPath.append(UPDATABLE_DRIVER_ABI);
+        ALOGI("Driver path is setup via UPDATABLE_GFX_DRIVER: %s", mDriverPath.c_str());
+    }
+    auto vndkNamespace = android_get_exported_namespace("vndk");
+    if (!vndkNamespace) {
+        return nullptr;
+    }
+    mDriverNamespace = android_create_namespace("updatable gfx driver",
+                                                mDriverPath.c_str(),
+                                                mDriverPath.c_str(),
+                                                ANDROID_NAMESPACE_TYPE_ISOLATED,
+                                                nullptr,
+                                                nullptr);
+    if (!linkDriverNamespaceLocked(mDriverNamespace, vndkNamespace, mSphalLibraries)) {
+        mDriverNamespace = nullptr;
+    }
+    return mDriverNamespace;
+}
+std::string GraphicsEnv::getDriverPath() const {
+    return mDriverPath;
+}
@@ -261,10 +333,2 @@ void GraphicsEnv::setVulkanDeviceExtensions(uint32_t enabledExtensionCount,
-    setTargetStatsArray(android::GpuStatsInfo::Stats::VULKAN_DEVICE_EXTENSION, extensionHashes,
-                        numStats);
-}
-static sp<IGpuService> getGpuService() {
-    static const sp<IBinder> binder = defaultServiceManager()->checkService(String16("gpu"));
-    if (!binder) {
-        ALOGE("Failed to get gpu service");
-        return nullptr;
-    }
-    return interface_cast<IGpuService>(binder);
+    setTargetStatsArray(android::GpuStatsInfo::Stats::VULKAN_DEVICE_EXTENSION,
+                        extensionHashes, numStats);
@@ -338,3 +402,2 @@ bool GraphicsEnv::shouldUseAngle() {
-void GraphicsEnv::setAngleInfo(const std::string& path, const bool useSystemAngle,
-                               const bool shouldUseNativeDriver, const std::string& packageName,
-                               const std::string& developerOptIn,
+void GraphicsEnv::setAngleInfo(const std::string& path, const bool shouldUseNativeDriver,
+                               const std::string& packageName,
@@ -350 +412,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const bool useSystemAngl
-<<<<<<< HEAD
@@ -359,23 +420,0 @@ void GraphicsEnv::setAngleInfo(const std::string& path, const bool useSystemAngl
-|||||||
-    mPackageName = packageName;
-    ALOGV("setting ANGLE application opt-in to '%s'", developerOptIn.c_str());
-    mAngleDeveloperOptIn = developerOptIn;
-    updateShouldUseAngle();
-=======
-    mPackageName = std::move(packageName);
-    mShouldUseAngle = true;
-    mUseSystemAngle = useSystemAngle;
->>>>>>> 58e0e1422d717b7317d6ef18f2508f1c7a01d214
-}
-void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace,
-                                const std::string& layerPaths) {
-    if (mLayerPaths.empty()) {
-        mLayerPaths = layerPaths;
-        mAppNamespace = appNamespace;
-    } else {
-        ALOGV("Vulkan layer search path already set, not clobbering with '%s' for namespace %p'",
-              layerPaths.c_str(), appNamespace);
-    }
-}
-NativeLoaderNamespace* GraphicsEnv::getAppNamespace() {
-    return mAppNamespace;
@@ -389,88 +427,0 @@ const std::vector<std::string>& GraphicsEnv::getAngleEglFeatures() {
-const std::string& GraphicsEnv::getLayerPaths() {
-    return mLayerPaths;
-}
-const std::string& GraphicsEnv::getDebugLayers() {
-    return mDebugLayers;
-}
-const std::string& GraphicsEnv::getDebugLayersGLES() {
-    return mDebugLayersGLES;
-}
-void GraphicsEnv::setDebugLayers(const std::string& layers) {
-    mDebugLayers = layers;
-}
-void GraphicsEnv::setDebugLayersGLES(const std::string& layers) {
-    mDebugLayersGLES = layers;
-}
-bool GraphicsEnv::linkDriverNamespaceLocked(android_namespace_t* destNamespace,
-                                            android_namespace_t* vndkNamespace,
-                                            const std::string& sharedSphalLibraries) {
-    const std::string llndkLibraries = getSystemNativeLibraries(NativeLibrary::LLNDK);
-    if (llndkLibraries.empty()) {
-        return false;
-    }
-    if (!android_link_namespaces(destNamespace, nullptr, llndkLibraries.c_str())) {
-        ALOGE("Failed to link default namespace[%s]", dlerror());
-        return false;
-    }
-    const std::string vndkspLibraries = getSystemNativeLibraries(NativeLibrary::VNDKSP);
-    if (vndkspLibraries.empty()) {
-        return false;
-    }
-    if (!android_link_namespaces(destNamespace, vndkNamespace, vndkspLibraries.c_str())) {
-        ALOGE("Failed to link vndk namespace[%s]", dlerror());
-        return false;
-    }
-    if (sharedSphalLibraries.empty()) {
-        return true;
-    }
-    auto sphalNamespace = android_get_exported_namespace("sphal");
-    if (!sphalNamespace) {
-        ALOGE("Depend on these libraries[%s] in sphal, but failed to get sphal namespace",
-              sharedSphalLibraries.c_str());
-        return false;
-    }
-    if (!android_link_namespaces(destNamespace, sphalNamespace, sharedSphalLibraries.c_str())) {
-        ALOGE("Failed to link sphal namespace[%s]", dlerror());
-        return false;
-    }
-    return true;
-}
-android_namespace_t* GraphicsEnv::getDriverNamespace() {
-    std::lock_guard<std::mutex> lock(mNamespaceMutex);
-    if (mDriverNamespace) {
-        return mDriverNamespace;
-    }
-    if (mDriverPath.empty()) {
-        const char* id = getenv("UPDATABLE_GFX_DRIVER");
-        if (id == nullptr || std::strcmp(id, "1") != 0) {
-            return nullptr;
-        }
-        const sp<IGpuService> gpuService = getGpuService();
-        if (!gpuService) {
-            return nullptr;
-        }
-        mDriverPath = gpuService->getUpdatableDriverPath();
-        if (mDriverPath.empty()) {
-            return nullptr;
-        }
-        mDriverPath.append(UPDATABLE_DRIVER_ABI);
-        ALOGI("Driver path is setup via UPDATABLE_GFX_DRIVER: %s", mDriverPath.c_str());
-    }
-    auto vndkNamespace = android_get_exported_namespace("vndk");
-    if (!vndkNamespace) {
-        return nullptr;
-    }
-    mDriverNamespace = android_create_namespace("updatable gfx driver",
-                                                mDriverPath.c_str(),
-                                                mDriverPath.c_str(),
-                                                ANDROID_NAMESPACE_TYPE_ISOLATED,
-                                                nullptr,
-                                                nullptr);
-    if (!linkDriverNamespaceLocked(mDriverNamespace, vndkNamespace, mSphalLibraries)) {
-        mDriverNamespace = nullptr;
-    }
-    return mDriverNamespace;
-}
-std::string GraphicsEnv::getDriverPath() const {
-    return mDriverPath;
-}
@@ -482 +432,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-<<<<<<< HEAD
@@ -485,7 +434,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-|||||||
-    if (mAnglePath.empty()) {
-        ALOGV("mAnglePath is empty, not creating ANGLE namespace");
-=======
-    if (mAnglePath.empty() && !mUseSystemAngle) {
-        ALOGV("mAnglePath is empty and not using system ANGLE, abort creating ANGLE namespace");
->>>>>>> 58e0e1422d717b7317d6ef18f2508f1c7a01d214
@@ -494 +436,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-<<<<<<< HEAD
@@ -508,18 +449,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-|||||||
-    mAngleNamespace = android_create_namespace("ANGLE",
-                                               nullptr,
-                                               mAnglePath.c_str(),
-=======
-    const char* const defaultLibraryPaths =
-#if defined(__LP64__)
-            "/vendor/lib64/egl:/system/lib64/egl";
-#else
-            "/vendor/lib/egl:/system/lib/egl";
-#endif
-    mAngleNamespace =
-            android_create_namespace("ANGLE",
-                                     mUseSystemAngle ? defaultLibraryPaths
-                                                     : mAnglePath.c_str(),
-                                     mUseSystemAngle ? defaultLibraryPaths
-                                                     : mAnglePath.c_str(),
->>>>>>> 58e0e1422d717b7317d6ef18f2508f1c7a01d214
@@ -528 +451,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-<<<<<<< HEAD
@@ -531,6 +453,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-|||||||
-                                               nullptr);
-=======
-                                     mUseSystemAngle ? android_get_exported_namespace("sphal")
-                                                     : nullptr);
->>>>>>> 58e0e1422d717b7317d6ef18f2508f1c7a01d214
@@ -538 +454,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-<<<<<<< HEAD
@@ -549,13 +464,0 @@ android_namespace_t* GraphicsEnv::getAngleNamespace() {
-|||||||
-=======
-    if (!mUseSystemAngle) {
-        return mAngleNamespace;
-    }
-    auto vndkNamespace = android_get_exported_namespace("vndk");
-    if (!vndkNamespace) {
-        return nullptr;
-    }
-    if (!linkDriverNamespaceLocked(mAngleNamespace, vndkNamespace, "")) {
-        mAngleNamespace = nullptr;
-    }
->>>>>>> 58e0e1422d717b7317d6ef18f2508f1c7a01d214
@@ -577,0 +481,28 @@ bool GraphicsEnv::shouldUseNativeDriver() {
+void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace,
+                                const std::string& layerPaths) {
+    if (mLayerPaths.empty()) {
+        mLayerPaths = layerPaths;
+        mAppNamespace = appNamespace;
+    } else {
+        ALOGV("Vulkan layer search path already set, not clobbering with '%s' for namespace %p'",
+              layerPaths.c_str(), appNamespace);
+    }
+}
+NativeLoaderNamespace* GraphicsEnv::getAppNamespace() {
+    return mAppNamespace;
+}
+const std::string& GraphicsEnv::getLayerPaths() {
+    return mLayerPaths;
+}
+const std::string& GraphicsEnv::getDebugLayers() {
+    return mDebugLayers;
+}
+const std::string& GraphicsEnv::getDebugLayersGLES() {
+    return mDebugLayersGLES;
+}
+void GraphicsEnv::setDebugLayers(const std::string& layers) {
+    mDebugLayers = layers;
+}
+void GraphicsEnv::setDebugLayersGLES(const std::string& layers) {
+    mDebugLayersGLES = layers;
+}
