--- a/./output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_conflict.cc
+++ b/./output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_merged.cc
@@ -1098,9 +1097,0 @@ grpc_error* RdsResponseParse(
-<<<<<<< HEAD
-    XdsApi::RdsUpdate local_rds_update;
-    grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config,
-                                         &local_rds_update);
-||||||| 25518e5b08
-    XdsApi::RdsUpdate local_rds_update;
-    grpc_error* error =
-        RouteConfigParse(client, tracer, route_config, &local_rds_update);
-=======
@@ -1110,2 +1101 @@ grpc_error* RdsResponseParse(
-        RouteConfigParse(client, tracer, route_config, &rds_update);
->>>>>>> c6947b74
+        RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1402 +1391,0 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-<<<<<<< HEAD
@@ -1404,8 +1393,2 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-                         expected_server_name, &result.lds_update, arena.ptr());
-||||||| 25518e5b08
-        LdsResponseParse(client_, tracer_, response, expected_server_name,
-                         &result.lds_update, arena.ptr());
-=======
-        LdsResponseParse(client_, tracer_, response, expected_listener_names,
-                         &result.lds_update_map, arena.ptr());
->>>>>>> c6947b74
+                         expected_listener_names, &result.lds_update_map,
+                         arena.ptr());
@@ -1413,10 +1396,2 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-<<<<<<< HEAD
-    result.parse_error = RdsResponseParse(
-        client_, tracer_, symtab_.ptr(), response,
-        expected_route_configuration_names, &result.rds_update, arena.ptr());
-||||||| 25518e5b08
-    result.parse_error = RdsResponseParse(client_, tracer_, response,
-                                          expected_route_configuration_names,
-                                          &result.rds_update, arena.ptr());
-=======
-    result.parse_error = RdsResponseParse(client_, tracer_, response,
+    result.parse_error =
+        RdsResponseParse(client_, tracer_, symtab_.ptr(), response,
@@ -1425 +1399,0 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
->>>>>>> c6947b74
