diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_merged.cc
index 61de797..e8b5ef8 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_merged.cc
@@ -67,2 +67,2 @@ XdsApi::Route::Matchers::PathMatcher::PathMatcher(const PathMatcher& other)
-XdsApi::Route::Matchers::PathMatcher&
-XdsApi::Route::Matchers::PathMatcher::operator=(const PathMatcher& other) {
+XdsApi::Route::Matchers::PathMatcher& XdsApi::Route::Matchers::PathMatcher::
+operator=(const PathMatcher& other) {
@@ -111,3 +110,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-  switch (type) {
-    case HeaderMatcherType::REGEX:
-      regex_match {
@@ -129,3 +126,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      XdsApi::Route::Matchers::HeaderMatcher&
-      XdsApi::Route::Matchers::HeaderMatcher::operator=(
-          const HeaderMatcher& other) {
+XdsApi::Route::Matchers::HeaderMatcher& XdsApi::Route::Matchers::HeaderMatcher::
+operator=(const HeaderMatcher& other) {
@@ -160,2 +156 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return range_start != other.range_start &&
-                   range_end != other.range_end;
+      return range_start != other.range_start && range_end != other.range_end;
@@ -172,2 +167 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   string_matcher);
+                             invert_match ? " not" : "", name, string_matcher);
@@ -180,2 +174,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   range_start, range_end);
+                             invert_match ? " not" : "", name, range_start,
+                             range_end);
@@ -188,2 +182 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   string_matcher);
+                             invert_match ? " not" : "", name, string_matcher);
@@ -192,2 +185 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   string_matcher);
+                             invert_match ? " not" : "", name, string_matcher);
@@ -284,2 +276 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        if (domain_pattern[domain_pattern.size() - 1] == '*')
-          return PREFIX_MATCH;
+  if (domain_pattern[domain_pattern.size() - 1] == '*') return PREFIX_MATCH;
@@ -289,2 +280,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      XdsApi::RdsUpdate::VirtualHost*
-      XdsApi::RdsUpdate::FindVirtualHostForDomain(const std::string& domain) {
+XdsApi::RdsUpdate::VirtualHost* XdsApi::RdsUpdate::FindVirtualHostForDomain(
+    const std::string& domain) {
@@ -322,2 +313 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      bool XdsApi::EdsUpdate::Priority::operator==(const Priority& other)
-          const {
+bool XdsApi::EdsUpdate::Priority::operator==(const Priority& other) const {
@@ -370,2 +360 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        return absl::StrCat("priorities=[",
-                            absl::StrJoin(priority_strings, ", "),
+  return absl::StrCat("priorities=[", absl::StrJoin(priority_strings, ", "),
@@ -374,6 +363,6 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
+const char* XdsApi::kLdsTypeUrl =
+    "type.googleapis.com/envoy.config.listener.v3.Listener";
+const char* XdsApi::kRdsTypeUrl =
+    "type.googleapis.com/envoy.config.route.v3.RouteConfiguration";
+const char* XdsApi::kCdsTypeUrl =
+    "type.googleapis.com/envoy.config.cluster.v3.Cluster";
@@ -408,4 +397,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            build_version_(absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING,
-                                        " ", grpc_version_string())),
-            user_agent_name_(
-                absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING)) {}
+      build_version_(absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING, " ",
+                                  grpc_version_string())),
+      user_agent_name_(absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING)) {}
@@ -417,2 +405 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateMetadataValue(upb_arena* arena,
-                                 google_protobuf_Value* value_pb,
+void PopulateMetadataValue(upb_arena* arena, google_protobuf_Value* value_pb,
@@ -420,2 +407 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateListValue(upb_arena* arena,
-                             google_protobuf_ListValue* list_value,
+void PopulateListValue(upb_arena* arena, google_protobuf_ListValue* list_value,
@@ -424,2 +410 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          auto* value_pb =
-              google_protobuf_ListValue_add_values(list_value, arena);
+    auto* value_pb = google_protobuf_ListValue_add_values(list_value, arena);
@@ -429,2 +414 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateMetadata(upb_arena* arena,
-                            google_protobuf_Struct* metadata_pb,
+void PopulateMetadata(upb_arena* arena, google_protobuf_Struct* metadata_pb,
@@ -439,2 +423 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateMetadataValue(upb_arena* arena,
-                                 google_protobuf_Value* value_pb,
+void PopulateMetadataValue(upb_arena* arena, google_protobuf_Value* value_pb,
@@ -487,2 +470 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      std::string EncodeStringField(uint32_t field_number,
-                                    const std::string& str) {
+std::string EncodeStringField(uint32_t field_number, const std::string& str) {
@@ -493,2 +475 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateBuildVersion(upb_arena* arena,
-                                envoy_config_core_v3_Node* node_msg,
+void PopulateBuildVersion(upb_arena* arena, envoy_config_core_v3_Node* node_msg,
@@ -545,2 +526 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            node_msg,
-            upb_strview_makez("envoy.lb.does_not_support_overprovisioning"),
+      node_msg, upb_strview_makez("envoy.lb.does_not_support_overprovisioning"),
@@ -564,2 +544,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_log(GPR_DEBUG, "[xds_client %p] constructed ADS request: %s",
-                  client, buf);
+    gpr_log(GPR_DEBUG, "[xds_client %p] constructed ADS request: %s", client,
+            buf);
@@ -569,2 +549 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          upb_arena* arena,
-          envoy_service_discovery_v3_DiscoveryRequest* request) {
+    upb_arena* arena, envoy_service_discovery_v3_DiscoveryRequest* request) {
@@ -598,2 +577,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const std::string& version, const std::string& nonce,
-          grpc_error* error, bool populate_node) {
+    const std::string& version, const std::string& nonce, grpc_error* error,
+    bool populate_node) {
@@ -619,2 +598,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          upb_strview error_description_strview = upb_strview_make(
-              reinterpret_cast<const char*>(
+    upb_strview error_description_strview =
+        upb_strview_make(reinterpret_cast<const char*>(
@@ -626,2 +605 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          google_rpc_Status_set_message(error_detail,
-                                        error_description_strview);
+    google_rpc_Status_set_message(error_detail, error_description_strview);
@@ -632,4 +610,4 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_service_discovery_v3_DiscoveryRequest_mutable_node(
-                  request, arena.ptr());
-          PopulateNode(arena.ptr(), bootstrap_, build_version_,
-                       user_agent_name_, node_msg);
+        envoy_service_discovery_v3_DiscoveryRequest_mutable_node(request,
+                                                                 arena.ptr());
+    PopulateNode(arena.ptr(), bootstrap_, build_version_, user_agent_name_,
+                 node_msg);
@@ -654,2 +632 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_log(GPR_DEBUG, "[xds_client %p] received response: %s", client,
-                  buf);
+    gpr_log(GPR_DEBUG, "[xds_client %p] received response: %s", client, buf);
@@ -667,2 +644 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_log(GPR_DEBUG, "[xds_client %p] RouteConfiguration: %s", client,
-                  buf);
+    gpr_log(GPR_DEBUG, "[xds_client %p] RouteConfiguration: %s", client, buf);
@@ -671,2 +647 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void MaybeLogCluster(XdsClient* client, TraceFlag* tracer,
-                           upb_symtab* symtab,
+void MaybeLogCluster(XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -692,2 +667,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_log(GPR_DEBUG, "[xds_client %p] ClusterLoadAssignment: %s",
-                  client, buf);
+    gpr_log(GPR_DEBUG, "[xds_client %p] ClusterLoadAssignment: %s", client,
+            buf);
@@ -696,3 +671,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      grpc_error* RoutePathMatchParse(
-          const envoy_config_route_v3_RouteMatch* match, XdsApi::Route* route,
-          bool* ignore_route) {
+grpc_error* RoutePathMatchParse(const envoy_config_route_v3_RouteMatch* match,
+                                XdsApi::Route* route, bool* ignore_route) {
@@ -712,2 +686 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            } else if (prefix_elements.size() == 2 &&
-                       prefix_elements[0].empty()) {
+      } else if (prefix_elements.size() == 2 && prefix_elements[0].empty()) {
@@ -753,2 +726 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::unique_ptr<RE2> regex =
-              absl::make_unique<RE2>(std::move(matcher));
+    std::unique_ptr<RE2> regex = absl::make_unique<RE2>(std::move(matcher));
@@ -776,2 +748,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          header_matcher.name = UpbStringToStdString(
-              envoy_config_route_v3_HeaderMatcher_name(header));
+    header_matcher.name =
+        UpbStringToStdString(envoy_config_route_v3_HeaderMatcher_name(header));
@@ -779,2 +751,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::EXACT;
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::EXACT;
@@ -795,2 +767,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::REGEX;
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::REGEX;
@@ -798,4 +770,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_range_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::RANGE;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_range_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::RANGE;
@@ -806,2 +777 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            header_matcher.range_end =
-                envoy_type_v3_Int64Range_end(range_matcher);
+      header_matcher.range_end = envoy_type_v3_Int64Range_end(range_matcher);
@@ -813,4 +783,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_present_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::PRESENT;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_present_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::PRESENT;
@@ -819,4 +788,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_prefix_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::PREFIX;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_prefix_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::PREFIX;
@@ -825,4 +793,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_suffix_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::SUFFIX;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_suffix_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::SUFFIX;
@@ -837,2 +804 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          route->matchers.header_matchers.emplace_back(
-              std::move(header_matcher));
+    route->matchers.header_matchers.emplace_back(std::move(header_matcher));
@@ -851,2 +817 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            uint32_t numerator =
-                envoy_type_v3_FractionalPercent_numerator(fraction);
+      uint32_t numerator = envoy_type_v3_FractionalPercent_numerator(fraction);
@@ -895,2 +860 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_route_v3_WeightedCluster_total_weight(
-                  weighted_cluster);
+        envoy_config_route_v3_WeightedCluster_total_weight(weighted_cluster);
@@ -901,3 +865,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_config_route_v3_WeightedCluster_ClusterWeight* const*
-              clusters = envoy_config_route_v3_WeightedCluster_clusters(
-                  weighted_cluster, &clusters_size);
+    const envoy_config_route_v3_WeightedCluster_ClusterWeight* const* clusters =
+        envoy_config_route_v3_WeightedCluster_clusters(weighted_cluster,
+                                                       &clusters_size);
@@ -934,2 +898 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "RouteAction weighted_cluster has no valid clusters "
-                "specified.");
+          "RouteAction weighted_cluster has no valid clusters specified.");
@@ -954,2 +917 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsApi::RdsUpdate::VirtualHost& vhost =
-              rds_update->virtual_hosts.back();
+    XdsApi::RdsUpdate::VirtualHost& vhost = rds_update->virtual_hosts.back();
@@ -957,3 +919,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          upb_strview const* domains =
-              envoy_config_route_v3_VirtualHost_domains(virtual_hosts[i],
-                                                        &domain_size);
+    upb_strview const* domains = envoy_config_route_v3_VirtualHost_domains(
+        virtual_hosts[i], &domain_size);
@@ -964,2 +925,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                  "Invalid domain pattern.");
+        return GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+            absl::StrCat("Invalid domain pattern \"", domain_pattern, "\".")
+                .c_str());
@@ -970,2 +932 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "VirtualHost has no domains");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("VirtualHost has no domains");
@@ -975,2 +936 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_route_v3_VirtualHost_routes(virtual_hosts[i],
-                                                       &num_routes);
+        envoy_config_route_v3_VirtualHost_routes(virtual_hosts[i], &num_routes);
@@ -992,2 +952 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            grpc_error* error =
-                RoutePathMatchParse(match, &route, &ignore_route);
+      grpc_error* error = RoutePathMatchParse(match, &route, &ignore_route);
@@ -1013,2 +972 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "No valid routes specified.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("No valid routes specified.");
@@ -1022,2 +980,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const std::string& expected_server_name,
-          absl::optional<XdsApi::LdsUpdate>* lds_update, upb_arena* arena) {
+    const std::set<absl::string_view>& expected_listener_names,
+    XdsApi::LdsUpdateMap* lds_update_map, upb_arena* arena) {
@@ -1026,2 +984 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1037,2 +994,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_listener_v3_Listener_parse(
-                  encoded_listener.data, encoded_listener.size, arena);
+        envoy_config_listener_v3_Listener_parse(encoded_listener.data,
+                                                encoded_listener.size, arena);
@@ -1040,2 +997,12 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Can't decode listener.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode listener.");
+    }
+    std::string listener_name =
+        UpbStringToStdString(envoy_config_listener_v3_Listener_name(listener));
+    if (expected_listener_names.find(listener_name) ==
+        expected_listener_names.end()) {
+      continue;
+    }
+    if (lds_update_map->find(listener_name) != lds_update_map->end()) {
+      return GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+          absl::StrCat("duplicate listener name \"", listener_name, "\"")
+              .c_str());
@@ -1043,3 +1010 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          absl::string_view name =
-              UpbStringToAbsl(envoy_config_listener_v3_Listener_name(listener));
-          if (name != expected_server_name) continue;
+    XdsApi::LdsUpdate& lds_update = (*lds_update_map)[listener_name];
@@ -1057,2 +1022 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                      encoded_api_listener.data, encoded_api_listener.size,
-                      arena);
+                encoded_api_listener.data, encoded_api_listener.size, arena);
@@ -1061,2 +1025 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "Could not parse HttpConnectionManager config from "
-                "ApiListener");
+          "Could not parse HttpConnectionManager config from ApiListener");
@@ -1070,2 +1033,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            grpc_error* error = RouteConfigParse(client, tracer, symtab,
-                                                 route_config, &rds_update);
+      grpc_error* error =
+          RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1073,3 +1036,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            lds_update->emplace();
-            (*lds_update)->rds_update = std::move(rds_update);
-            return GRPC_ERROR_NONE;
+      lds_update.rds_update = std::move(rds_update);
+      continue;
@@ -1080,2 +1042 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "HttpConnectionManager neither has inlined route_config nor "
-                "RDS.");
+          "HttpConnectionManager neither has inlined route_config nor RDS.");
@@ -1083,2 +1044 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_extensions_filters_network_http_connection_manager_v3_Rds*
-              rds =
+    const envoy_extensions_filters_network_http_connection_manager_v3_Rds* rds =
@@ -1096,2 +1056 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "HttpConnectionManager ConfigSource for RDS does not specify "
-                "ADS.");
+          "HttpConnectionManager ConfigSource for RDS does not specify ADS.");
@@ -1099,2 +1058 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          lds_update->emplace();
-          (*lds_update)->route_config_name = UpbStringToStdString(
+    lds_update.route_config_name = UpbStringToStdString(
@@ -1103 +1060,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          return GRPC_ERROR_NONE;
@@ -1111 +1068 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          absl::optional<XdsApi::RdsUpdate>* rds_update, upb_arena* arena) {
+    XdsApi::RdsUpdateMap* rds_update_map, upb_arena* arena) {
@@ -1114,2 +1071 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1128,2 +1084 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Can't decode route_config.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode route_config.");
@@ -1131 +1086 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          absl::string_view route_config_name = UpbStringToAbsl(
+    std::string route_config_name = UpbStringToStdString(
@@ -1137,3 +1092,10 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsApi::RdsUpdate local_rds_update;
-          grpc_error* error = RouteConfigParse(client, tracer, symtab,
-                                               route_config, &local_rds_update);
+    if (rds_update_map->find(route_config_name) != rds_update_map->end()) {
+      return GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+          absl::StrCat("duplicate route config name \"", route_config_name,
+                       "\"")
+              .c_str());
+    }
+    XdsApi::RdsUpdate& rds_update =
+        (*rds_update_map)[std::move(route_config_name)];
+    grpc_error* error =
+        RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1141,2 +1102,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          rds_update->emplace(std::move(local_rds_update));
-          return GRPC_ERROR_NONE;
@@ -1153,2 +1113 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1156 +1114,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsApi::CdsUpdate cds_update;
@@ -1162,2 +1120 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const upb_strview encoded_cluster =
-              google_protobuf_Any_value(resources[i]);
+    const upb_strview encoded_cluster = google_protobuf_Any_value(resources[i]);
@@ -1165,2 +1122,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_cluster_v3_Cluster_parse(
-                  encoded_cluster.data, encoded_cluster.size, arena);
+        envoy_config_cluster_v3_Cluster_parse(encoded_cluster.data,
+                                              encoded_cluster.size, arena);
@@ -1168,2 +1125 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Can't decode cluster.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode cluster.");
@@ -1172,2 +1128,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::string cluster_name = UpbStringToStdString(
-              envoy_config_cluster_v3_Cluster_name(cluster));
+    std::string cluster_name =
+        UpbStringToStdString(envoy_config_cluster_v3_Cluster_name(cluster));
@@ -1182,0 +1139 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
+    XdsApi::CdsUpdate& cds_update = (*cds_update_map)[std::move(cluster_name)];
@@ -1184,2 +1141 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "DiscoveryType not found.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("DiscoveryType not found.");
@@ -1189,2 +1145 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "DiscoveryType is not EDS.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("DiscoveryType is not EDS.");
@@ -1192,2 +1147 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_config_cluster_v3_Cluster_EdsClusterConfig*
-              eds_cluster_config =
+    const envoy_config_cluster_v3_Cluster_EdsClusterConfig* eds_cluster_config =
@@ -1222,2 +1175,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          cds_update_map->emplace(std::move(cluster_name),
-                                  std::move(cds_update));
@@ -1244,2 +1196 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        uint32_t port =
-            envoy_config_core_v3_SocketAddress_port_value(socket_address);
+  uint32_t port = envoy_config_core_v3_SocketAddress_port_value(socket_address);
@@ -1255,3 +1206,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_config_endpoint_v3_LocalityLbEndpoints*
-              locality_lb_endpoints,
-          XdsApi::PriorityListUpdate::LocalityMap::Locality* output_locality) {
+    const envoy_config_endpoint_v3_LocalityLbEndpoints* locality_lb_endpoints,
+    XdsApi::EdsUpdate::Priority::Locality* output_locality, size_t* priority) {
@@ -1262,2 +1212 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            lb_weight != nullptr ? google_protobuf_UInt32Value_value(lb_weight)
-                                 : 0;
+      lb_weight != nullptr ? google_protobuf_UInt32Value_value(lb_weight) : 0;
@@ -1268,6 +1217,6 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        std::string region = UpbStringToStdString(
-            envoy_config_core_v3_Locality_region(locality));
-        std::string zone = UpbStringToStdString(
-            envoy_config_core_v3_Locality_region(locality));
-        std::string sub_zone = UpbStringToStdString(
-            envoy_config_core_v3_Locality_sub_zone(locality));
+  std::string region =
+      UpbStringToStdString(envoy_config_core_v3_Locality_region(locality));
+  std::string zone =
+      UpbStringToStdString(envoy_config_core_v3_Locality_region(locality));
+  std::string sub_zone =
+      UpbStringToStdString(envoy_config_core_v3_Locality_sub_zone(locality));
@@ -1282 +1231 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              lb_endpoints[i], &output_locality->serverlist);
+        lb_endpoints[i], &output_locality->endpoints);
@@ -1285,2 +1234 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        output_locality->priority =
-            envoy_config_endpoint_v3_LocalityLbEndpoints_priority(
+  *priority = envoy_config_endpoint_v3_LocalityLbEndpoints_priority(
@@ -1293 +1241 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsApi::DropConfig* drop_config) {
+    XdsApi::EdsUpdate::DropConfig* drop_config) {
@@ -1298,2 +1246 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-              "Empty drop category name");
+    return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Empty drop category name");
@@ -1319,2 +1266 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Unknown denominator type");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Unknown denominator type");
@@ -1333,2 +1279 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1336 +1280,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsApi::EdsUpdate eds_update;
@@ -1344,2 +1288 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          envoy_config_endpoint_v3_ClusterLoadAssignment*
-              cluster_load_assignment =
+    envoy_config_endpoint_v3_ClusterLoadAssignment* cluster_load_assignment =
@@ -1364,2 +1307 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                absl::StrCat("duplicate resource name \"", eds_service_name,
-                             "\"")
+          absl::StrCat("duplicate resource name \"", eds_service_name, "\"")
@@ -1367,0 +1310,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
+    XdsApi::EdsUpdate& eds_update =
+        (*eds_update_map)[std::move(eds_service_name)];
@@ -1373,2 +1317,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            XdsApi::PriorityListUpdate::LocalityMap::Locality locality;
-            grpc_error* error = LocalityParse(endpoints[j], &locality);
+      size_t priority;
+      XdsApi::EdsUpdate::Priority::Locality locality;
+      grpc_error* error = LocalityParse(endpoints[j], &locality, &priority);
@@ -1377 +1322,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            eds_update.priority_list_update.Add(locality);
+      while (eds_update.priorities.size() < priority + 1) {
+        eds_update.priorities.emplace_back();
@@ -1379,4 +1325,5 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          for (uint32_t priority = 0;
-               priority < eds_update.priority_list_update.size(); ++priority) {
-            auto* locality_map = eds_update.priority_list_update.Find(priority);
-            if (locality_map == nullptr || locality_map->size() == 0) {
+      eds_update.priorities[priority].localities.emplace(locality.name.get(),
+                                                         std::move(locality));
+    }
+    for (const auto& priority : eds_update.priorities) {
+      if (priority.localities.empty()) {
@@ -1387 +1334 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          eds_update.drop_config = MakeRefCounted<XdsApi::DropConfig>();
+    eds_update.drop_config = MakeRefCounted<XdsApi::EdsUpdate::DropConfig>();
@@ -1398,2 +1345,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              grpc_error* error = DropParseAndAppend(
-                  drop_overload[j], eds_update.drop_config.get());
+        grpc_error* error =
+            DropParseAndAppend(drop_overload[j], eds_update.drop_config.get());
@@ -1403,2 +1349,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          eds_update_map->emplace(std::move(eds_service_name),
-                                  std::move(eds_update));
@@ -1431,2 +1376 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                reinterpret_cast<const char*>(
-                    GRPC_SLICE_START_PTR(encoded_response)),
+          reinterpret_cast<const char*>(GRPC_SLICE_START_PTR(encoded_response)),
@@ -1435,2 +1379,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-              "Can't decode DiscoveryResponse.");
+    result.parse_error =
+        GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode DiscoveryResponse.");
@@ -1443,2 +1387 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_version_info(
-                response));
+      envoy_service_discovery_v3_DiscoveryResponse_version_info(response));
@@ -1449 +1391,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-<<<<<<< HEAD
@@ -1451,7 +1392,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                               expected_server_name, &result.lds_update,
-                               arena.ptr());
-|||||||
-              LdsResponseParse(client_, tracer_, response, expected_server_name,
-                               &result.lds_update, arena.ptr());
-=======
-              LdsResponseParse(client_, tracer_, response,
@@ -1460 +1394,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
->>>>>>> c6947b749a69f51f60ce5ea8887efdd58462591b
@@ -1462 +1395,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-<<<<<<< HEAD
@@ -1466,8 +1398,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                               &result.rds_update, arena.ptr());
-|||||||
-          result.parse_error = RdsResponseParse(
-              client_, tracer_, response, expected_route_configuration_names,
-              &result.rds_update, arena.ptr());
-=======
-          result.parse_error = RdsResponseParse(
-              client_, tracer_, response, expected_route_configuration_names,
@@ -1475 +1399,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
->>>>>>> c6947b749a69f51f60ce5ea8887efdd58462591b
@@ -1477,2 +1401,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = CdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response, expected_cluster_names,
+    result.parse_error = CdsResponseParse(client_, tracer_, symtab_.ptr(),
+                                          response, expected_cluster_names,
@@ -1481,3 +1405,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = EdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response,
-              expected_eds_service_names, &result.eds_update_map, arena.ptr());
+    result.parse_error = EdsResponseParse(client_, tracer_, symtab_.ptr(),
+                                          response, expected_eds_service_names,
+                                          &result.eds_update_map, arena.ptr());
@@ -1487,0 +1412,13 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
+void MaybeLogLrsRequest(
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
+    const envoy_service_load_stats_v3_LoadStatsRequest* request) {
+  if (GRPC_TRACE_FLAG_ENABLED(*tracer) &&
+      gpr_should_log(GPR_LOG_SEVERITY_DEBUG)) {
+    const upb_msgdef* msg_type =
+        envoy_service_load_stats_v3_LoadStatsRequest_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(request, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] constructed LRS request: %s", client,
+            buf);
+  }
+}
@@ -1502,2 +1439,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_load_stats_v3_LoadStatsRequest_mutable_node(
-                request, arena.ptr());
+      envoy_service_load_stats_v3_LoadStatsRequest_mutable_node(request,
+                                                                arena.ptr());
@@ -1509 +1446 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        MaybeLogLrsRequest(client_, tracer_, request, build_version_);
+  MaybeLogLrsRequest(client_, tracer_, symtab_.ptr(), request);
@@ -1518,2 +1455,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_config_endpoint_v3_UpstreamLocalityStats_mutable_locality(
-                output, arena);
+      envoy_config_endpoint_v3_UpstreamLocalityStats_mutable_locality(output,
+                                                                      arena);
@@ -1554,13 +1490,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void MaybeLogLrsRequest(
-          XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
-          const envoy_service_load_stats_v3_LoadStatsRequest* request) {
-        if (GRPC_TRACE_FLAG_ENABLED(*tracer) &&
-            gpr_should_log(GPR_LOG_SEVERITY_DEBUG)) {
-          const upb_msgdef* msg_type =
-              envoy_service_load_stats_v3_LoadStatsRequest_getmsgdef(symtab);
-          char buf[10240];
-          upb_text_encode(request, msg_type, nullptr, 0, buf, sizeof(buf));
-          gpr_log(GPR_DEBUG, "[xds_client %p] constructed LRS request: %s",
-                  client, buf);
-        }
-      }
@@ -1599,2 +1523 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_config_endpoint_v3_ClusterStats_DroppedRequests*
-                dropped_requests =
+      envoy_config_endpoint_v3_ClusterStats_DroppedRequests* dropped_requests =
@@ -1609,2 +1532 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          total_dropped_requests +=
-              load_report.dropped_requests.uncategorized_drops;
+    total_dropped_requests += load_report.dropped_requests.uncategorized_drops;
@@ -1613,2 +1535,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_timespec timespec = grpc_millis_to_timespec(
-              load_report.load_report_interval, GPR_TIMESPAN);
+    gpr_timespec timespec =
+        grpc_millis_to_timespec(load_report.load_report_interval, GPR_TIMESPAN);
@@ -1618,4 +1540,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          google_protobuf_Duration_set_seconds(load_report_interval,
-                                               timespec.tv_sec);
-          google_protobuf_Duration_set_nanos(load_report_interval,
-                                             timespec.tv_nsec);
+    google_protobuf_Duration_set_seconds(load_report_interval, timespec.tv_sec);
+    google_protobuf_Duration_set_nanos(load_report_interval, timespec.tv_nsec);
@@ -1626,2 +1546,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      grpc_error* XdsApi::ParseLrsResponse(
-          const grpc_slice& encoded_response, bool* send_all_clusters,
+grpc_error* XdsApi::ParseLrsResponse(const grpc_slice& encoded_response,
+                                     bool* send_all_clusters,
@@ -1633,2 +1553 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                reinterpret_cast<const char*>(
-                    GRPC_SLICE_START_PTR(encoded_response)),
+          reinterpret_cast<const char*>(GRPC_SLICE_START_PTR(encoded_response)),
@@ -1645,2 +1564,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_service_load_stats_v3_LoadStatsResponse_clusters(
-                  decoded_response, &size);
+        envoy_service_load_stats_v3_LoadStatsResponse_clusters(decoded_response,
+                                                               &size);
