diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/include@nativehelper@JniConstants.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/include@nativehelper@JniConstants.no_comments_merged.h
index 309d122..adbfd04 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/include@nativehelper@JniConstants.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/include@nativehelper@JniConstants.no_comments_merged.h
@@ -60,0 +61,2 @@ struct JniConstants {
+#define NATIVE_METHOD(className,functionName,signature) \
+    { #functionName, signature, reinterpret_cast<void*>(className ## _ ## functionName) }
