diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_merged.cpp
index f3f68cf..6562701 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_merged.cpp
@@ -87,2 +87 @@ void JniConstants::init(JNIEnv* env) {
-  inetAddressHolderClass =
-      findClass(env, "java/net/InetAddress$InetAddressHolder");
+    inetAddressHolderClass = findClass(env, "java/net/InetAddress$InetAddressHolder");
@@ -90,8 +89 @@ void JniConstants::init(JNIEnv* env) {
-<<<<<<< /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/dc8d6a-800dca/ours/JniConstants.cpp
-||||||| /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/dc8d6a-800dca/base/JniConstants.cpp
-  inetUnixAddressClass = findClass(env, "java/net/InetUnixAddress");
-=======
-  inetSocketAddressHolderClass =
-      findClass(env, "java/net/InetSocketAddress$InetSocketAddressHolder");
-  inetUnixAddressClass = findClass(env, "java/net/InetUnixAddress");
->>>>>>> /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/dc8d6a-800dca/theirs/JniConstants.cpp
+    inetSocketAddressHolderClass = findClass(env, "java/net/InetSocketAddress$InetSocketAddressHolder");
@@ -106,2 +98 @@ void JniConstants::init(JNIEnv* env) {
-  netlinkSocketAddressClass =
-      findClass(env, "android/system/NetlinkSocketAddress");
+    netlinkSocketAddressClass = findClass(env, "android/system/NetlinkSocketAddress");
@@ -111,2 +102 @@ void JniConstants::init(JNIEnv* env) {
-  packetSocketAddressClass =
-      findClass(env, "android/system/PacketSocketAddress");
+    packetSocketAddressClass = findClass(env, "android/system/PacketSocketAddress");
@@ -114,2 +104 @@ void JniConstants::init(JNIEnv* env) {
-  patternSyntaxExceptionClass =
-      findClass(env, "java/util/regex/PatternSyntaxException");
+    patternSyntaxExceptionClass = findClass(env, "java/util/regex/PatternSyntaxException");
@@ -124,2 +113 @@ void JniConstants::init(JNIEnv* env) {
-  structGroupSourceReqClass =
-      findClass(env, "android/system/StructGroupSourceReq");
+    structGroupSourceReqClass = findClass(env, "android/system/StructGroupSourceReq");
