--- a/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_merged.cpp
@@ -90,8 +89 @@ void JniConstants::init(JNIEnv* env) {
-<<<<<<< /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/dc8d6a-800dca/ours/JniConstants.cpp
-||||||| /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/dc8d6a-800dca/base/JniConstants.cpp
-  inetUnixAddressClass = findClass(env, "java/net/InetUnixAddress");
-=======
-  inetSocketAddressHolderClass =
-      findClass(env, "java/net/InetSocketAddress$InetSocketAddressHolder");
-  inetUnixAddressClass = findClass(env, "java/net/InetUnixAddress");
->>>>>>> /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/dc8d6a-800dca/theirs/JniConstants.cpp
+    inetSocketAddressHolderClass = findClass(env, "java/net/InetSocketAddress$InetSocketAddressHolder");
