diff --git a/./output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_conflict.cpp b/./output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_merged.cpp
index 271a330..6562701 100644
--- a/./output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_conflict.cpp
+++ b/./output/libnativehelper/dc8d6a70-800dca96-b7a25286/JniConstants.no_comments_merged.cpp
@@ -25,2 +24,0 @@ jclass JniConstants::inetSocketAddressClass;
-<<<<<<< HEAD
-=======
@@ -28,2 +25,0 @@ jclass JniConstants::inetSocketAddressHolderClass;
-jclass JniConstants::inetUnixAddressClass;
->>>>>>> 800dca96
@@ -93,2 +88,0 @@ void JniConstants::init(JNIEnv* env) {
-<<<<<<< HEAD
-=======
@@ -96,2 +89,0 @@ void JniConstants::init(JNIEnv* env) {
-    inetUnixAddressClass = findClass(env, "java/net/InetUnixAddress");
->>>>>>> 800dca96
