--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/8776c85f-7f36ff03-615cd251/src@core@ext@filters@client_channel@lb_policy@xds@eds.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/8776c85f-7f36ff03-615cd251/src@core@ext@filters@client_channel@lb_policy@xds@eds.no_comments_merged.cc
@@ -20 +19,0 @@
-#include "src/core/lib/iomgr/combiner.h"
@@ -45 +44 @@ class EdsLbConfig : public LoadBalancingPolicy::Config {
-  }
+  };
