--- a/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_merged.c
@@ -22,0 +23 @@
+#include <sys/resource.h>
@@ -161,9 +152,0 @@ struct redisCommand redisCommandTable[] = {
-<<<<<<< HEAD
-    {"client", clientCommand, -2, 0, NULL, 0, 0, 0, 0, 0},
-    {"eval", evalCommand, -3, REDIS_CMD_DENYOOM, zunionInterGetKeys, 0, 0, 0, 0,
-     0},
-    {"evalsha", evalShaCommand, -3, REDIS_CMD_DENYOOM, zunionInterGetKeys, 0, 0,
-     0, 0, 0}
-|||||||
-    {"client", clientCommand, -2, 0, NULL, 0, 0, 0, 0, 0}
-=======
@@ -170,0 +154,2 @@ struct redisCommand redisCommandTable[] = {
+    {"eval",evalCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
+    {"evalsha",evalShaCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
@@ -172 +156,0 @@ struct redisCommand redisCommandTable[] = {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -428,3 +427 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-    redisLog(REDIS_WARNING,
-             "SIGTERM received but errors trying to shut down the server, "
-             "check the logs for more information");
+        redisLog(REDIS_WARNING,"SIGTERM received but errors trying to shut down the server, check the logs for more information");
@@ -730 +719,0 @@ void initServer() {
-<<<<<<< HEAD
@@ -732,2 +720,0 @@ void initServer() {
-|||||||
-=======
@@ -735 +721,0 @@ void initServer() {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -1230,6 +1245 @@ void linuxOvercommitMemoryWarning(void) {
-    redisLog(
-        REDIS_WARNING,
-        "WARNING overcommit_memory is set to 0! Background save may fail under "
-        "low memory condition. To fix this issue add 'vm.overcommit_memory = "
-        "1' to /etc/sysctl.conf and then reboot or run the command 'sysctl "
-        "vm.overcommit_memory=1' for this to take effect.");
+        redisLog(REDIS_WARNING,"WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.");
@@ -1290,4 +1304 @@ int main(int argc, char **argv) {
-    redisLog(
-        REDIS_WARNING,
-        "Warning: no config file specified, using the default config. In order "
-        "to specify a config file use 'redis-server /path/to/redis.conf'");
+        redisLog(REDIS_WARNING,"Warning: no config file specified, using the default config. In order to specify a config file use 'redis-server /path/to/redis.conf'");
