diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_merged.c
index 6d2a301..8eccee4 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_merged.c
@@ -22,0 +23 @@
+#include <sys/resource.h>
@@ -30,4 +31,2 @@ struct redisCommand redisCommandTable[] = {
-    {"setnx", setnxCommand, 3, REDIS_CMD_DENYOOM, noPreloadGetKeys, 1, 1, 1, 0,
-     0},
-    {"setex", setexCommand, 4, REDIS_CMD_DENYOOM, noPreloadGetKeys, 2, 2, 1, 0,
-     0},
+    {"setnx",setnxCommand,3,REDIS_CMD_DENYOOM,noPreloadGetKeys,1,1,1,0,0},
+    {"setex",setexCommand,4,REDIS_CMD_DENYOOM,noPreloadGetKeys,2,2,1,0,0},
@@ -54,2 +53 @@ struct redisCommand redisCommandTable[] = {
-    {"brpoplpush", brpoplpushCommand, 4, REDIS_CMD_DENYOOM, NULL, 1, 2, 1, 0,
-     0},
+    {"brpoplpush",brpoplpushCommand,4,REDIS_CMD_DENYOOM,NULL,1,2,1,0,0},
@@ -72,2 +70 @@ struct redisCommand redisCommandTable[] = {
-    {"sinterstore", sinterstoreCommand, -3, REDIS_CMD_DENYOOM, NULL, 2, -1, 1,
-     0, 0},
+    {"sinterstore",sinterstoreCommand,-3,REDIS_CMD_DENYOOM,NULL,2,-1,1,0,0},
@@ -75,2 +72 @@ struct redisCommand redisCommandTable[] = {
-    {"sunionstore", sunionstoreCommand, -3, REDIS_CMD_DENYOOM, NULL, 2, -1, 1,
-     0, 0},
+    {"sunionstore",sunionstoreCommand,-3,REDIS_CMD_DENYOOM,NULL,2,-1,1,0,0},
@@ -78,2 +74 @@ struct redisCommand redisCommandTable[] = {
-    {"sdiffstore", sdiffstoreCommand, -3, REDIS_CMD_DENYOOM, NULL, 2, -1, 1, 0,
-     0},
+    {"sdiffstore",sdiffstoreCommand,-3,REDIS_CMD_DENYOOM,NULL,2,-1,1,0,0},
@@ -86,4 +81,2 @@ struct redisCommand redisCommandTable[] = {
-    {"zunionstore", zunionstoreCommand, -4, REDIS_CMD_DENYOOM,
-     zunionInterGetKeys, 0, 0, 0, 0, 0},
-    {"zinterstore", zinterstoreCommand, -4, REDIS_CMD_DENYOOM,
-     zunionInterGetKeys, 0, 0, 0, 0, 0},
+    {"zunionstore",zunionstoreCommand,-4,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
+    {"zinterstore",zinterstoreCommand,-4,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
@@ -152,2 +145 @@ struct redisCommand redisCommandTable[] = {
-    {"publish", publishCommand, 3, REDIS_CMD_FORCE_REPLICATION, NULL, 0, 0, 0,
-     0, 0},
+    {"publish",publishCommand,3,REDIS_CMD_FORCE_REPLICATION,NULL,0,0,0,0,0},
@@ -161,9 +152,0 @@ struct redisCommand redisCommandTable[] = {
-<<<<<<< HEAD
-    {"client", clientCommand, -2, 0, NULL, 0, 0, 0, 0, 0},
-    {"eval", evalCommand, -3, REDIS_CMD_DENYOOM, zunionInterGetKeys, 0, 0, 0, 0,
-     0},
-    {"evalsha", evalShaCommand, -3, REDIS_CMD_DENYOOM, zunionInterGetKeys, 0, 0,
-     0, 0, 0}
-|||||||
-    {"client", clientCommand, -2, 0, NULL, 0, 0, 0, 0, 0}
-=======
@@ -170,0 +154,2 @@ struct redisCommand redisCommandTable[] = {
+    {"eval",evalCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
+    {"evalsha",evalShaCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
@@ -172 +156,0 @@ struct redisCommand redisCommandTable[] = {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -217 +201,2 @@ long long ustime(void) {
-void dictVanillaFree(void *privdata, void *val) {
+void dictVanillaFree(void *privdata, void *val)
+{
@@ -221 +206,2 @@ void dictVanillaFree(void *privdata, void *val) {
-void dictListDestructor(void *privdata, void *val) {
+void dictListDestructor(void *privdata, void *val)
+{
@@ -225 +211,3 @@ void dictListDestructor(void *privdata, void *val) {
-int dictSdsKeyCompare(void *privdata, const void *key1, const void *key2) {
+int dictSdsKeyCompare(void *privdata, const void *key1,
+        const void *key2)
+{
@@ -233 +221,3 @@ int dictSdsKeyCompare(void *privdata, const void *key1, const void *key2) {
-int dictSdsKeyCaseCompare(void *privdata, const void *key1, const void *key2) {
+int dictSdsKeyCaseCompare(void *privdata, const void *key1,
+        const void *key2)
+{
@@ -237 +227,2 @@ int dictSdsKeyCaseCompare(void *privdata, const void *key1, const void *key2) {
-void dictRedisObjectDestructor(void *privdata, void *val) {
+void dictRedisObjectDestructor(void *privdata, void *val)
+{
@@ -242 +233,2 @@ void dictRedisObjectDestructor(void *privdata, void *val) {
-void dictSdsDestructor(void *privdata, void *val) {
+void dictSdsDestructor(void *privdata, void *val)
+{
@@ -246 +238,3 @@ void dictSdsDestructor(void *privdata, void *val) {
-int dictObjKeyCompare(void *privdata, const void *key1, const void *key2) {
+int dictObjKeyCompare(void *privdata, const void *key1,
+        const void *key2)
+{
@@ -260 +254,3 @@ unsigned int dictSdsCaseHash(const void *key) {
-int dictEncObjKeyCompare(void *privdata, const void *key1, const void *key2) {
+int dictEncObjKeyCompare(void *privdata, const void *key1,
+        const void *key2)
+{
@@ -263 +259,2 @@ int dictEncObjKeyCompare(void *privdata, const void *key1, const void *key2) {
-  if (o1->encoding == REDIS_ENCODING_INT && o2->encoding == REDIS_ENCODING_INT)
+    if (o1->encoding == REDIS_ENCODING_INT &&
+        o2->encoding == REDIS_ENCODING_INT)
@@ -365,2 +362,4 @@ void tryResizeHashTables(void) {
-    if (htNeedsResize(server.db[j].dict)) dictResize(server.db[j].dict);
-    if (htNeedsResize(server.db[j].expires)) dictResize(server.db[j].expires);
+        if (htNeedsResize(server.db[j].dict))
+            dictResize(server.db[j].dict);
+        if (htNeedsResize(server.db[j].expires))
+            dictResize(server.db[j].expires);
@@ -414,2 +413,2 @@ void updateLRUClock(void) {
-  server.lruclock =
-      (time(NULL) / REDIS_LRU_CLOCK_RESOLUTION) & REDIS_LRU_CLOCK_MAX;
+    server.lruclock = (time(NULL)/REDIS_LRU_CLOCK_RESOLUTION) &
+                                                REDIS_LRU_CLOCK_MAX;
@@ -428,3 +427 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-    redisLog(REDIS_WARNING,
-             "SIGTERM received but errors trying to shut down the server, "
-             "check the logs for more information");
+        redisLog(REDIS_WARNING,"SIGTERM received but errors trying to shut down the server, check the logs for more information");
@@ -438,3 +435 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-      redisLog(REDIS_VERBOSE,
-               "DB %d: %lld keys (%lld volatile) in %lld slots HT.", j, used,
-               vkeys, size);
+            redisLog(REDIS_VERBOSE,"DB %d: %lld keys (%lld volatile) in %lld slots HT.",j,used,vkeys,size);
@@ -448,2 +443 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-    redisLog(REDIS_VERBOSE,
-             "%d clients connected (%d slaves), %zu bytes in use",
+        redisLog(REDIS_VERBOSE,"%d clients connected (%d slaves), %zu bytes in use",
@@ -451 +445,2 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-             listLength(server.slaves), zmalloc_used_memory());
+            listLength(server.slaves),
+            zmalloc_used_memory());
@@ -456 +451,2 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-      server.aofrewrite_scheduled) {
+        server.aofrewrite_scheduled)
+    {
@@ -477 +473,2 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-      if (server.dirty >= sp->changes && now - server.lastsave > sp->seconds) {
+            if (server.dirty >= sp->changes &&
+                now-server.lastsave > sp->seconds) {
@@ -484 +481,2 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-    if (server.bgsavechildpid == -1 && server.bgrewritechildpid == -1 &&
+         if (server.bgsavechildpid == -1 &&
+             server.bgrewritechildpid == -1 &&
@@ -486,4 +484,4 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-        server.appendonly_current_size > server.auto_aofrewrite_min_size) {
-      int base = server.auto_aofrewrite_base_size
-                     ? server.auto_aofrewrite_base_size
-                     : 1;
+             server.appendonly_current_size > server.auto_aofrewrite_min_size)
+         {
+            int base = server.auto_aofrewrite_base_size ?
+                            server.auto_aofrewrite_base_size : 1;
@@ -492,3 +490 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-        redisLog(REDIS_NOTICE,
-                 "Starting automatic rewriting of AOF on %lld%% growth",
-                 growth);
+                redisLog(REDIS_NOTICE,"Starting automatic rewriting of AOF on %lld%% growth",growth);
@@ -515 +511,2 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
-    if (c->querybuf && sdslen(c->querybuf) > 0) processInputBuffer(c);
+        if (c->querybuf && sdslen(c->querybuf) > 0)
+            processInputBuffer(c);
@@ -533,3 +530 @@ void createSharedObjects(void) {
-  shared.wrongtypeerr = createObject(
-      REDIS_STRING,
-      sdsnew(
+    shared.wrongtypeerr = createObject(REDIS_STRING,sdsnew(
@@ -537,14 +532,12 @@ void createSharedObjects(void) {
-  shared.nokeyerr = createObject(REDIS_STRING, sdsnew("-ERR no such key\r\n"));
-  shared.syntaxerr =
-      createObject(REDIS_STRING, sdsnew("-ERR syntax error\r\n"));
-  shared.sameobjecterr = createObject(
-      REDIS_STRING,
-      sdsnew("-ERR source and destination objects are the same\r\n"));
-  shared.outofrangeerr =
-      createObject(REDIS_STRING, sdsnew("-ERR index out of range\r\n"));
-  shared.noscripterr = createObject(
-      REDIS_STRING,
-      sdsnew("-NOSCRIPT No matching script. Please use EVAL.\r\n"));
-  shared.loadingerr = createObject(
-      REDIS_STRING,
-      sdsnew("-LOADING Redis is loading the dataset in memory\r\n"));
+    shared.nokeyerr = createObject(REDIS_STRING,sdsnew(
+        "-ERR no such key\r\n"));
+    shared.syntaxerr = createObject(REDIS_STRING,sdsnew(
+        "-ERR syntax error\r\n"));
+    shared.sameobjecterr = createObject(REDIS_STRING,sdsnew(
+        "-ERR source and destination objects are the same\r\n"));
+    shared.outofrangeerr = createObject(REDIS_STRING,sdsnew(
+        "-ERR index out of range\r\n"));
+    shared.noscripterr = createObject(REDIS_STRING,sdsnew(
+        "-NOSCRIPT No matching script. Please use EVAL.\r\n"));
+    shared.loadingerr = createObject(REDIS_STRING,sdsnew(
+        "-LOADING Redis is loading the dataset in memory\r\n"));
@@ -715,2 +708 @@ void initServer() {
-                                           acceptTcpHandler, NULL) == AE_ERR)
-    oom("creating file event");
+        acceptTcpHandler,NULL) == AE_ERR) oom("creating file event");
@@ -718,2 +710 @@ void initServer() {
-                                           acceptUnixHandler, NULL) == AE_ERR)
-    oom("creating file event");
+        acceptUnixHandler,NULL) == AE_ERR) oom("creating file event");
@@ -721,2 +712 @@ void initServer() {
-    server.appendfd =
-        open(server.appendfilename, O_WRONLY | O_APPEND | O_CREAT, 0644);
+        server.appendfd = open(server.appendfilename,O_WRONLY|O_APPEND|O_CREAT,0644);
@@ -730 +719,0 @@ void initServer() {
-<<<<<<< HEAD
@@ -732,2 +720,0 @@ void initServer() {
-|||||||
-=======
@@ -735 +721,0 @@ void initServer() {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -793 +779,2 @@ int processCommand(redisClient *c) {
-    addReplyErrorFormat(c, "unknown command '%s'", (char *)c->argv[0]->ptr);
+        addReplyErrorFormat(c,"unknown command '%s'",
+            (char*)c->argv[0]->ptr);
@@ -801 +788,2 @@ int processCommand(redisClient *c) {
-  if (server.requirepass && !c->authenticated && c->cmd->proc != authCommand) {
+    if (server.requirepass && !c->authenticated && c->cmd->proc != authCommand)
+    {
@@ -809,3 +797 @@ int processCommand(redisClient *c) {
-      addReplyError(
-          c,
-          "The cluster is down. Check with CLUSTER INFO for more information");
+            addReplyError(c,"The cluster is down. Check with CLUSTER INFO for more information");
@@ -815,2 +801 @@ int processCommand(redisClient *c) {
-      clusterNode *n =
-          getNodeByQuery(c, c->cmd, c->argv, c->argc, &hashslot, &ask);
+            clusterNode *n = getNodeByQuery(c,c->cmd,c->argv,c->argc,&hashslot,&ask);
@@ -821,3 +806,3 @@ int processCommand(redisClient *c) {
-        addReplySds(
-            c, sdscatprintf(sdsempty(), "-%s %d %s:%d\r\n",
-                            ask ? "ASK" : "MOVED", hashslot, n->ip, n->port));
+                addReplySds(c,sdscatprintf(sdsempty(),
+                    "-%s %d %s:%d\r\n", ask ? "ASK" : "MOVED",
+                    hashslot,n->ip,n->port));
@@ -830 +815,2 @@ int processCommand(redisClient *c) {
-      zmalloc_used_memory() > server.maxmemory) {
+        zmalloc_used_memory() > server.maxmemory)
+    {
@@ -834,3 +820,4 @@ int processCommand(redisClient *c) {
-  if ((dictSize(c->pubsub_channels) > 0 ||
-       listLength(c->pubsub_patterns) > 0) &&
-      c->cmd->proc != subscribeCommand && c->cmd->proc != unsubscribeCommand &&
+    if ((dictSize(c->pubsub_channels) > 0 || listLength(c->pubsub_patterns) > 0)
+        &&
+        c->cmd->proc != subscribeCommand &&
+        c->cmd->proc != unsubscribeCommand &&
@@ -839,2 +826 @@ int processCommand(redisClient *c) {
-    addReplyError(
-        c, "only (P)SUBSCRIBE / (P)UNSUBSCRIBE / QUIT allowed in this context");
+        addReplyError(c,"only (P)SUBSCRIBE / (P)UNSUBSCRIBE / QUIT allowed in this context");
@@ -844,4 +830,5 @@ int processCommand(redisClient *c) {
-      server.repl_serve_stale_data == 0 && c->cmd->proc != infoCommand &&
-      c->cmd->proc != slaveofCommand) {
-    addReplyError(
-        c, "link with MASTER is down and slave-serve-stale-data is set to no");
+        server.repl_serve_stale_data == 0 &&
+        c->cmd->proc != infoCommand && c->cmd->proc != slaveofCommand)
+    {
+        addReplyError(c,
+            "link with MASTER is down and slave-serve-stale-data is set to no");
@@ -854,3 +841,4 @@ int processCommand(redisClient *c) {
-  if (c->flags & REDIS_MULTI && c->cmd->proc != execCommand &&
-      c->cmd->proc != discardCommand && c->cmd->proc != multiCommand &&
-      c->cmd->proc != watchCommand) {
+    if (c->flags & REDIS_MULTI &&
+        c->cmd->proc != execCommand && c->cmd->proc != discardCommand &&
+        c->cmd->proc != multiCommand && c->cmd->proc != watchCommand)
+    {
@@ -894,2 +882,6 @@ void authCommand(redisClient *c) {
-void pingCommand(redisClient *c) { addReply(c, shared.pong); }
-void echoCommand(redisClient *c) { addReplyBulk(c, c->argv[1]); }
+void pingCommand(redisClient *c) {
+    addReply(c,shared.pong);
+}
+void echoCommand(redisClient *c) {
+    addReplyBulk(c,c->argv[1]);
+}
@@ -941 +933,2 @@ sds genRedisInfoString(char *section) {
-                        REDIS_VERSION, redisGitSHA1(),
+            REDIS_VERSION,
+            redisGitSHA1(),
@@ -943,3 +936,7 @@ sds genRedisInfoString(char *section) {
-                        (sizeof(long) == 8) ? "64" : "32", aeGetApiName(),
-                        (long)getpid(), server.port, uptime,
-                        uptime / (3600 * 24), (unsigned long)server.lruclock);
+            (sizeof(long) == 8) ? "64" : "32",
+            aeGetApiName(),
+            (long) getpid(),
+            server.port,
+            uptime,
+            uptime/(3600*24),
+            (unsigned long) server.lruclock);
@@ -956 +953,2 @@ sds genRedisInfoString(char *section) {
-                        lol, bib, server.bpop_blocked_clients);
+            lol, bib,
+            server.bpop_blocked_clients);
@@ -964,2 +962 @@ sds genRedisInfoString(char *section) {
-    info = sdscatprintf(
-        info,
+        info = sdscatprintf(info,
@@ -975,3 +972,9 @@ sds genRedisInfoString(char *section) {
-        zmalloc_used_memory(), hmem, zmalloc_get_rss(), server.stat_peak_memory,
-        peak_hmem, ((long long)lua_gc(server.lua, LUA_GCCOUNT, 0)) * 1024LL,
-        zmalloc_get_fragmentation_ratio(), ZMALLOC_LIB);
+            zmalloc_used_memory(),
+            hmem,
+            zmalloc_get_rss(),
+            server.stat_peak_memory,
+            peak_hmem,
+            ((long long)lua_gc(server.lua,LUA_GCCOUNT,0))*1024LL,
+            zmalloc_get_fragmentation_ratio(),
+            ZMALLOC_LIB
+            );
@@ -989,2 +992,5 @@ sds genRedisInfoString(char *section) {
-                        server.loading, server.appendonly, server.dirty,
-                        server.bgsavechildpid != -1, server.lastsave,
+            server.loading,
+            server.appendonly,
+            server.dirty,
+            server.bgsavechildpid != -1,
+            server.lastsave,
@@ -1004,5 +1010,4 @@ sds genRedisInfoString(char *section) {
-      off_t remaining_bytes =
-          server.loading_total_bytes - server.loading_loaded_bytes;
-      perc =
-          ((double)server.loading_loaded_bytes / server.loading_total_bytes) *
-          100;
+            off_t remaining_bytes = server.loading_total_bytes-
+                                    server.loading_loaded_bytes;
+            perc = ((double)server.loading_loaded_bytes /
+                   server.loading_total_bytes) * 100;
@@ -1020,2 +1025,2 @@ sds genRedisInfoString(char *section) {
-                          "loading_eta_seconds:%ld\r\n",
-                          (unsigned long)server.loading_start_time,
+                "loading_eta_seconds:%ld\r\n"
+                ,(unsigned long) server.loading_start_time,
@@ -1023,2 +1028,4 @@ sds genRedisInfoString(char *section) {
-                          (unsigned long long)server.loading_loaded_bytes, perc,
-                          eta);
+                (unsigned long long) server.loading_loaded_bytes,
+                perc,
+                eta
+            );
@@ -1029,2 +1036 @@ sds genRedisInfoString(char *section) {
-    info =
-        sdscatprintf(info,
+        info = sdscatprintf(info,
@@ -1041,3 +1047,6 @@ sds genRedisInfoString(char *section) {
-                     server.stat_numconnections, server.stat_numcommands,
-                     server.stat_expiredkeys, server.stat_evictedkeys,
-                     server.stat_keyspace_hits, server.stat_keyspace_misses,
+            server.stat_numconnections,
+            server.stat_numcommands,
+            server.stat_expiredkeys,
+            server.stat_evictedkeys,
+            server.stat_keyspace_hits,
+            server.stat_keyspace_misses,
@@ -1045 +1054,2 @@ sds genRedisInfoString(char *section) {
-                     listLength(server.pubsub_patterns), server.stat_fork_time);
+            listLength(server.pubsub_patterns),
+            server.stat_fork_time);
@@ -1054,2 +1064 @@ sds genRedisInfoString(char *section) {
-      info = sdscatprintf(
-          info,
+            info = sdscatprintf(info,
@@ -1060,6 +1069,9 @@ sds genRedisInfoString(char *section) {
-          "master_sync_in_progress:%d\r\n",
-          server.masterhost, server.masterport,
-          (server.replstate == REDIS_REPL_CONNECTED) ? "up" : "down",
-          server.master ? ((int)(time(NULL) - server.master->lastinteraction))
-                        : -1,
-          server.replstate == REDIS_REPL_TRANSFER);
+                "master_sync_in_progress:%d\r\n"
+                ,server.masterhost,
+                server.masterport,
+                (server.replstate == REDIS_REPL_CONNECTED) ?
+                    "up" : "down",
+                server.master ?
+                ((int)(time(NULL)-server.master->lastinteraction)) : -1,
+                server.replstate == REDIS_REPL_TRANSFER
+            );
@@ -1069,3 +1081,4 @@ sds genRedisInfoString(char *section) {
-                            "master_sync_last_io_seconds_ago:%d\r\n",
-                            (long)server.repl_transfer_left,
-                            (int)(time(NULL) - server.repl_transfer_lastio));
+                    "master_sync_last_io_seconds_ago:%d\r\n"
+                    ,(long)server.repl_transfer_left,
+                    (int)(time(NULL)-server.repl_transfer_lastio)
+                );
@@ -1074 +1087,2 @@ sds genRedisInfoString(char *section) {
-        info = sdscatprintf(info, "master_link_down_since_seconds:%ld\r\n",
+                info = sdscatprintf(info,
+                    "master_link_down_since_seconds:%ld\r\n",
@@ -1078 +1092,2 @@ sds genRedisInfoString(char *section) {
-    info = sdscatprintf(info, "connected_slaves:%d\r\n",
+        info = sdscatprintf(info,
+            "connected_slaves:%d\r\n",
@@ -1083,2 +1098 @@ sds genRedisInfoString(char *section) {
-    info = sdscatprintf(
-        info,
+        info = sdscatprintf(info,
@@ -1090,4 +1104,2 @@ sds genRedisInfoString(char *section) {
-        (float)self_ru.ru_utime.tv_sec +
-            (float)self_ru.ru_utime.tv_usec / 1000000,
-        (float)self_ru.ru_stime.tv_sec +
-            (float)self_ru.ru_stime.tv_usec / 1000000,
+        (float)self_ru.ru_utime.tv_sec+(float)self_ru.ru_utime.tv_usec/1000000,
+        (float)self_ru.ru_stime.tv_sec+(float)self_ru.ru_stime.tv_usec/1000000,
@@ -1104,2 +1116,2 @@ sds genRedisInfoString(char *section) {
-      info = sdscatprintf(
-          info, "cmdstat_%s:calls=%lld,usec=%lld,usec_per_call=%.2f\r\n",
+            info = sdscatprintf(info,
+                "cmdstat_%s:calls=%lld,usec=%lld,usec_per_call=%.2f\r\n",
@@ -1125,2 +1137,2 @@ sds genRedisInfoString(char *section) {
-        info = sdscatprintf(info, "db%d:keys=%lld,expires=%lld\r\n", j, keys,
-                            vkeys);
+                info = sdscatprintf(info, "db%d:keys=%lld,expires=%lld\r\n",
+                    j, keys, vkeys);
@@ -1139,2 +1151,2 @@ void infoCommand(redisClient *c) {
-  addReplySds(
-      c, sdscatprintf(sdsempty(), "$%lu\r\n", (unsigned long)sdslen(info)));
+    addReplySds(c,sdscatprintf(sdsempty(),"$%lu\r\n",
+        (unsigned long)sdslen(info)));
@@ -1162 +1174,2 @@ void freeMemoryIfNeeded(void) {
-          server.maxmemory_policy == REDIS_MAXMEMORY_ALLKEYS_RANDOM) {
+                server.maxmemory_policy == REDIS_MAXMEMORY_ALLKEYS_RANDOM)
+            {
@@ -1169 +1182,2 @@ void freeMemoryIfNeeded(void) {
-          server.maxmemory_policy == REDIS_MAXMEMORY_VOLATILE_RANDOM) {
+                server.maxmemory_policy == REDIS_MAXMEMORY_VOLATILE_RANDOM)
+            {
@@ -1174 +1188,2 @@ void freeMemoryIfNeeded(void) {
-               server.maxmemory_policy == REDIS_MAXMEMORY_VOLATILE_LRU) {
+                server.maxmemory_policy == REDIS_MAXMEMORY_VOLATILE_LRU)
+            {
@@ -1230,6 +1245 @@ void linuxOvercommitMemoryWarning(void) {
-    redisLog(
-        REDIS_WARNING,
-        "WARNING overcommit_memory is set to 0! Background save may fail under "
-        "low memory condition. To fix this issue add 'vm.overcommit_memory = "
-        "1' to /etc/sysctl.conf and then reboot or run the command 'sysctl "
-        "vm.overcommit_memory=1' for this to take effect.");
+        redisLog(REDIS_WARNING,"WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.");
@@ -1258,2 +1268,2 @@ void version() {
-  printf("Redis server version %s (%s:%d)\n", REDIS_VERSION, redisGitSHA1(),
-         atoi(redisGitDirty()) > 0);
+    printf("Redis server version %s (%s:%d)\n", REDIS_VERSION,
+        redisGitSHA1(), atoi(redisGitDirty()) > 0);
@@ -1270 +1280,3 @@ void redisAsciiArt(void) {
-  snprintf(buf, 1024 * 16, ascii_logo, REDIS_VERSION, redisGitSHA1(),
+    snprintf(buf,1024*16,ascii_logo,
+        REDIS_VERSION,
+        redisGitSHA1(),
@@ -1273,2 +1285,4 @@ void redisAsciiArt(void) {
-           server.cluster_enabled ? "cluster" : "stand alone", server.port,
-           (long)getpid());
+        server.cluster_enabled ? "cluster" : "stand alone",
+        server.port,
+        (long) getpid()
+    );
@@ -1282,2 +1296,2 @@ int main(int argc, char **argv) {
-    if (strcmp(argv[1], "-v") == 0 || strcmp(argv[1], "--version") == 0)
-      version();
+        if (strcmp(argv[1], "-v") == 0 ||
+            strcmp(argv[1], "--version") == 0) version();
@@ -1290,4 +1304 @@ int main(int argc, char **argv) {
-    redisLog(
-        REDIS_WARNING,
-        "Warning: no config file specified, using the default config. In order "
-        "to specify a config file use 'redis-server /path/to/redis.conf'");
+        redisLog(REDIS_WARNING,"Warning: no config file specified, using the default config. In order to specify a config file use 'redis-server /path/to/redis.conf'");
@@ -1306,2 +1317 @@ int main(int argc, char **argv) {
-      redisLog(REDIS_NOTICE, "DB loaded from append only file: %.3f seconds",
-               (float)(ustime() - start) / 1000000);
+            redisLog(REDIS_NOTICE,"DB loaded from append only file: %.3f seconds",(float)(ustime()-start)/1000000);
@@ -1310,2 +1320 @@ int main(int argc, char **argv) {
-      redisLog(REDIS_NOTICE, "DB loaded from disk: %.3f seconds",
-               (float)(ustime() - start) / 1000000);
+            redisLog(REDIS_NOTICE,"DB loaded from disk: %.3f seconds",(float)(ustime()-start)/1000000);
@@ -1314,3 +1323 @@ int main(int argc, char **argv) {
-    redisLog(REDIS_NOTICE,
-             "The server is now ready to accept connections on port %d",
-             server.port);
+        redisLog(REDIS_NOTICE,"The server is now ready to accept connections on port %d", server.port);
@@ -1318,3 +1325 @@ int main(int argc, char **argv) {
-    redisLog(REDIS_NOTICE,
-             "The server is now ready to accept connections at %s",
-             server.unixsocket);
+        redisLog(REDIS_NOTICE,"The server is now ready to accept connections at %s", server.unixsocket);
@@ -1363,2 +1368 @@ static void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-           "======= Ooops! Redis %s got signal: -%d- =======", REDIS_VERSION,
-           sig);
+        "======= Ooops! Redis %s got signal: -%d- =======", REDIS_VERSION, sig);
@@ -1372 +1376,2 @@ static void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-  for (i = 1; i < trace_size; ++i) redisLog(REDIS_WARNING, "%s", messages[i]);
+    for (i=1; i<trace_size; ++i)
+        redisLog(REDIS_WARNING,"%s", messages[i]);
