--- a/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@config.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@config.no_comments_merged.c
@@ -133,3 +126 @@ void loadServerConfig(char *filename) {
-        err =
-            "Invalid log facility. Must be one of USER or between "
-            "LOCAL0-LOCAL7";
+                err = "Invalid log facility. Must be one of USER or between LOCAL0-LOCAL7";
@@ -283 +267,0 @@ void loadServerConfig(char *filename) {
-<<<<<<< HEAD
@@ -286,3 +270,3 @@ void loadServerConfig(char *filename) {
-|||||||
-=======
-    } else if (!strcasecmp(argv[0], "slowlog-log-slower-than") && argc == 2) {
+        } else if (!strcasecmp(argv[0],"slowlog-log-slower-than") &&
+                   argc == 2)
+        {
@@ -292 +275,0 @@ void loadServerConfig(char *filename) {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -442 +428,0 @@ void configSetCommand(redisClient *c) {
-<<<<<<< HEAD
@@ -446,2 +431,0 @@ void configSetCommand(redisClient *c) {
-|||||||
-=======
@@ -454 +437,0 @@ void configSetCommand(redisClient *c) {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -638 +601,0 @@ void configGetCommand(redisClient *c) {
-<<<<<<< HEAD
@@ -642,4 +604,0 @@ void configGetCommand(redisClient *c) {
-    matches++;
-  }
-|||||||
-=======
@@ -656 +614,0 @@ void configGetCommand(redisClient *c) {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
