diff --git a/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_conflict.hpp b/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_merged.hpp
index 97abc3b..87765b5 100644
--- a/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_conflict.hpp
+++ b/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_merged.hpp
@@ -64 +63,0 @@ public:
-<<<<<<< HEAD
@@ -70,8 +68,0 @@ public:
-||||||| 41104b611e
- idx_t Size() {
-  return entries;
-=======
- static idx_t InitialCapacity();
- idx_t Size() {
-  return entries;
->>>>>>> 4be6bdb5
@@ -78,0 +70 @@ public:
+ static idx_t InitialCapacity();
@@ -94 +86 @@ private:
- TupleDataAppendState append_state;
+ TupleDataAppendState td_append_state;
@@ -112 +103,0 @@ private:
-<<<<<<< HEAD
@@ -118,6 +109 @@ private:
-||||||| 41104b611e
- idx_t FindOrCreateGroupsInternal(DataChunk &groups, Vector &group_hashes, Vector &addresses,
-                                  SelectionVector &new_groups);
- template <class FUNC = std::function<void(idx_t, idx_t, data_ptr_t)>>
- void PayloadApply(FUNC fun);
-=======
+ template <class ENTRY>
@@ -126,3 +111,0 @@ private:
- template <class FUNC = std::function<void(idx_t, idx_t, data_ptr_t)>>
- void PayloadApply(FUNC fun);
->>>>>>> 4be6bdb5
