diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_mergebot.hpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_merged.hpp
index 88bfc9c..87765b5 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_merged.hpp
@@ -39,0 +40 @@ public:
+public:
@@ -49,0 +51 @@ public:
+public:
@@ -79,0 +82 @@ private:
+ idx_t capacity;
@@ -83,2 +86 @@ private:
- TupleDataAppendState append_state;
- idx_t capacity;
+ TupleDataAppendState td_append_state;
@@ -93 +95,2 @@ private:
- GroupedAggregateHashTable(const GroupedAggregateHashTable &)
+private:
+ GroupedAggregateHashTable(const GroupedAggregateHashTable &) = delete;
@@ -95,0 +99,9 @@ private:
+ template <class ENTRY>
+ void VerifyInternal();
+ template <class ENTRY>
+ void Resize(idx_t size);
+ template <class ENTRY>
+ void InitializeHashes();
+ template <class ENTRY>
+ idx_t FindOrCreateGroupsInternal(DataChunk &groups, Vector &group_hashes_v, Vector &addresses_v,
+                                  SelectionVector &new_groups);
@@ -96,0 +109,3 @@ private:
+ template <class ENTRY>
+ idx_t FindOrCreateGroupsInternal(AggregateHTAppendState &state, DataChunk &groups, Vector &group_hashes,
+                                  Vector &addresses, SelectionVector &new_groups);
