diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_merged.cpp
index 2e515da..0deb876 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_merged.cpp
@@ -41,6 +41 @@ idx_t PartitionableHashTable::ListAddChunk(HashTableList &list, DataChunk &group
-<<<<<<< HEAD
- if (list.empty() || list.back()->Count() + groups.size() > list.back()->MaxCapacity()) {
-|||||||
- if (list.empty() || list.back()->Size() + groups.size() > list.back()->MaxCapacity()) {
-=======
- if (list.empty() || list.back()->Size() + groups.size() >= list.back()->MaxCapacity()) {
+ if (list.empty() || list.back()->Count() + groups.size() >= list.back()->MaxCapacity()) {
@@ -48 +42,0 @@ idx_t PartitionableHashTable::ListAddChunk(HashTableList &list, DataChunk &group
->>>>>>> 4be6bdb565f5cdd5618fc012b77a3bd579105255
