diff --git a/./output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_mergegen.c b/./output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_merged.c
index 889fe8d..025d63e 100644
--- a/./output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_mergegen.c
+++ b/./output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_merged.c
@@ -138 +138,3 @@ static int cliReadMultiBulkReply(int fd) {
-if (config.tty) printf("%d. ", c); if (cliReadReply(fd)) retval = 1; if (elements) printf("%c",config.mb_sep);
+        if (config.tty) printf("%d. ", c);
+        if (cliReadReply(fd)) retval = 1;
+        if (elements) printf("%c",config.mb_sep);
