diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_merged.c
index ed81eac..025d63e 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_merged.c
@@ -42,2 +42 @@ static int cliConnect(int force) {
-      fprintf(stderr, "Could not connect to Redis at %s:%d: %s", config.hostip,
-              config.hostport, err);
+            fprintf(stderr, "Could not connect to Redis at %s:%d: %s", config.hostip, config.hostport, err);
@@ -70 +69,2 @@ static int cliReadSingleLineReply(int fd, int quiet) {
-  if (!quiet) printf("%s", reply);
+    if (!quiet)
+        printf("%s", reply);
@@ -82,15 +82,5 @@ static void printStringRepr(char *s, int len) {
-      case '\n':
-        printf("\\n");
-        break;
-      case '\r':
-        printf("\\r");
-        break;
-      case '\t':
-        printf("\\t");
-        break;
-      case '\a':
-        printf("\\a");
-        break;
-      case '\b':
-        printf("\\b");
-        break;
+        case '\n': printf("\\n"); break;
+        case '\r': printf("\\r"); break;
+        case '\t': printf("\\t"); break;
+        case '\a': printf("\\a"); break;
+        case '\b': printf("\\b"); break;
@@ -148 +137,0 @@ static int cliReadMultiBulkReply(int fd) {
-<<<<<<< HEAD
@@ -150,7 +138,0 @@ static int cliReadMultiBulkReply(int fd) {
-    if (cliReadReply(fd)) return 1;
-    if (elements) printf("%c", config.mb_sep);
-|||||||
-    printf("%d. ", c);
-    if (cliReadReply(fd)) return 1;
-=======
-    printf("%d. ", c);
@@ -158 +140 @@ static int cliReadMultiBulkReply(int fd) {
->>>>>>> c0b3d42372dbe67c6ef096372869e2b60d4a1cdc
+        if (elements) printf("%c",config.mb_sep);
@@ -169 +151,2 @@ static int cliReadReply(int fd) {
-        (nread == 0 || (nread == -1 && errno == ECONNRESET))) {
+            (nread == 0 || (nread == -1 && errno == ECONNRESET)))
+        {
@@ -199 +182,2 @@ static int selectDb(int fd) {
-  if (config.dbnum == 0) return 0;
+    if (config.dbnum == 0)
+        return 0;
@@ -218,2 +202,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-  if (!strcasecmp(command, "subscribe") || !strcasecmp(command, "psubscribe"))
-    config.pubsub_mode = 1;
+    if (!strcasecmp(command,"subscribe") ||
+        !strcasecmp(command,"psubscribe")) config.pubsub_mode = 1;
@@ -228 +212,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-    cmd = sdscatprintf(cmd, "$%lu\r\n", (unsigned long)sdslen(argv[j]));
+        cmd = sdscatprintf(cmd,"$%lu\r\n",
+            (unsigned long)sdslen(argv[j]));
@@ -282,4 +267,3 @@ static int parseOptions(int argc, char **argv) {
-              "Starting interactive mode using -i is deprecated. Interactive "
-              "mode is started\n"
-              "by default when redis-cli is executed without a command to "
-              "execute.\n");
+"Starting interactive mode using -i is deprecated. Interactive mode is started\n"
+"by default when redis-cli is executed without a command to execute.\n"
+            );
@@ -302,2 +286 @@ static sds readArgFromStdin(void) {
-    if (nread == 0)
-      break;
+        if (nread == 0) break;
@@ -313,9 +296,3 @@ static void usage() {
-  fprintf(stderr,
-          "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r "
-          "repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-a authpw] "
-          "[-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
-  fprintf(stderr,
-          "\nIf a pipe from standard input is detected this data is used as "
-          "last argument.\n\n");
+    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
+    fprintf(stderr, "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
+    fprintf(stderr, "\nIf a pipe from standard input is detected this data is used as last argument.\n\n");
@@ -325,3 +302 @@ static void usage() {
-  fprintf(stderr,
-          "\nRun in interactive mode: redis-cli -i or just don't pass any "
-          "command\n");
+    fprintf(stderr, "\nRun in interactive mode: redis-cli -i or just don't pass any command\n");
@@ -333 +308,2 @@ static char **convertToSds(int count, char **args) {
-  for (j = 0; j < count; j++) sds[j] = sdsnew(args[j]);
+  for(j = 0; j < count; j++)
+    sds[j] = sdsnew(args[j]);
@@ -351 +327,2 @@ static void repl() {
-            strcasecmp(argv[0], "exit") == 0) {
+                    strcasecmp(argv[0],"exit") == 0)
+                {
@@ -366 +343,2 @@ static void repl() {
-      for (j = 0; j < argc; j++) sdsfree(argv[j]);
+            for (j = 0; j < argc; j++)
+                sdsfree(argv[j]);
@@ -403 +380,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
@@ -405,6 +381,0 @@ int main(int argc, char **argv) {
-|||||||
-  if (argc == 0 || config.interactive == 1) repl();
-=======
-  if (argc == 0) config.interactive = 1;
-  if (config.interactive) repl();
->>>>>>> c0b3d42372dbe67c6ef096372869e2b60d4a1cdc
