--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7cdf5ee9-f165221d-24350879/format.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7cdf5ee9-f165221d-24350879/format.no_comments_merged.c
@@ -14 +14 @@
-struct format_entry
+struct format_entry;
@@ -80,0 +87 @@ static int format_entry_cmp(struct format_entry *, struct format_entry *);
+RB_GENERATE_STATIC(format_entry_tree, format_entry, entry, format_entry_cmp);
@@ -165 +176,4 @@ static void printflike(3, 4)
-static void format_job_update(struct job *job) {
+#define format_log(ft,fmt,...) format_log1(ft, __func__, fmt, ##__VA_ARGS__)
+static void
+format_job_update(struct job *job)
+{
@@ -748,4 +885,0 @@ struct format_tree *format_create(struct client *c, struct cmdq_item *item,
-<<<<<<< HEAD
-  format_add(ft, "version", "%s", VERSION);
-|||||||
-=======
@@ -753 +886,0 @@ struct format_tree *format_create(struct client *c, struct cmdq_item *item,
->>>>>>> f165221dc4641837ee9f589bb666d310a495904c
