--- a/./output/tmux/7cdf5ee9-f165221d-24350879/proc.no_comments_mergegen.c
+++ b/./output/tmux/7cdf5ee9-f165221d-24350879/proc.no_comments_merged.c
@@ -131 +131,2 @@ proc_start(const char *name)
-log_debug("%s started (%ld): version %s, socket %s, protocol %d", name, (long)getpid(), VERSION, socket_path, PROTOCOL_VERSION); log_debug("on %s %s %s; libevent %s (%s)", u.sysname, u.release,
+ log_debug("%s started (%ld): version %s, socket %s, protocol %d", name,
+     (long)getpid(), getversion(), socket_path, PROTOCOL_VERSION);
