--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7cdf5ee9-f165221d-24350879/proc.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7cdf5ee9-f165221d-24350879/proc.no_comments_merged.c
@@ -2 +1,0 @@
-#include <sys/queue.h>
@@ -5 +3,0 @@
-#include <imsg.h>
@@ -113,8 +129,2 @@ struct tmuxproc *proc_start(const char *name) {
-  if (uname(&u) < 0) memset(&u, 0, sizeof u);
-<<<<<<< HEAD
-  log_debug("%s started (%ld): version %s, socket %s, protocol %d", name,
-            (long)getpid(), VERSION, socket_path, PROTOCOL_VERSION);
-|||||||
-  log_debug("%s started (%ld): socket %s, protocol %d", name, (long)getpid(),
-            socket_path, PROTOCOL_VERSION);
-=======
+ if (uname(&u) < 0)
+  memset(&u, 0, sizeof u);
@@ -123,3 +133,2 @@ struct tmuxproc *proc_start(const char *name) {
->>>>>>> f165221dc4641837ee9f589bb666d310a495904c
-  log_debug("on %s %s %s; libevent %s (%s)", u.sysname, u.release, u.version,
-            event_get_version(), event_get_method());
+ log_debug("on %s %s %s; libevent %s (%s)", u.sysname, u.release,
+     u.version, event_get_version(), event_get_method());
