diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_conflict.h b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_merged.h
index 23de43d..1551b4e 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_conflict.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_merged.h
@@ -127,32 +126,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
-||||||| 183ba01a0
-  struct SuperVersion {
-    MemTable* mem;
-    MemTableListVersion* imm;
-    Version* current;
-    std::atomic<uint32_t> refs;
-    std::vector<MemTable*> to_delete;
-    explicit SuperVersion(const int num_memtables = 0);
-    ~SuperVersion();
-    SuperVersion* Ref();
-    bool Unref();
-    void Cleanup();
-    void Init(MemTable* new_mem, MemTableListVersion* new_imm,
-              Version* new_current);
-  };
-=======
-  struct SuperVersion {
-    MemTable* mem;
-    MemTableListVersion* imm;
-    Version* current;
-    std::atomic<uint32_t> refs;
-    autovector<MemTable*> to_delete;
-    SuperVersion() = default;
-    ~SuperVersion();
-    SuperVersion* Ref();
-    bool Unref();
-    void Cleanup();
-    void Init(MemTable* new_mem, MemTableListVersion* new_imm,
-              Version* new_current);
-  };
->>>>>>> 4564b2e8
@@ -178 +145,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
@@ -180,7 +146,0 @@ class DBImpl : public DB {
-||||||| 183ba01a0
-      new_superversion =
-          create_superversion ? new SuperVersion(num_memtables) : nullptr;
-=======
-      new_superversion =
-          create_superversion ? new SuperVersion() : nullptr;
->>>>>>> 4564b2e8
@@ -222 +181,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
@@ -225 +184 @@ class DBImpl : public DB {
-  Status WriteLevel0Table(ColumnFamilyData* cfd, std::vector<MemTable*>& mems,
+  Status WriteLevel0Table(ColumnFamilyData* cfd, autovector<MemTable*>& mems,
@@ -227,9 +185,0 @@ class DBImpl : public DB {
-||||||| 183ba01a0
-  Status WriteLevel0TableForRecovery(MemTable* mem, VersionEdit* edit);
-  Status WriteLevel0Table(std::vector<MemTable*> &mems, VersionEdit* edit,
-                                uint64_t* filenumber);
-=======
-  Status WriteLevel0TableForRecovery(MemTable* mem, VersionEdit* edit);
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                                uint64_t* filenumber);
->>>>>>> 4564b2e8
