--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_impl.no_comments_merged.h
@@ -12,0 +13 @@
+#include "port/port.h"
@@ -17,3 +17,0 @@
-#include "port/port.h"
-#include "util/stats_logger.h"
-#include "memtablelist.h"
@@ -20,0 +19 @@
+#include "util/stats_logger.h"
@@ -145,6 +145,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
-      new_superversion = create_superversion ? new SuperVersion() : nullptr;
-||||||| 183ba01a0
-      new_superversion =
-          create_superversion ? new SuperVersion(num_memtables) : nullptr;
-=======
@@ -152 +146,0 @@ class DBImpl : public DB {
->>>>>>> 4564b2e8f9a2da07fa84fb5f3bba007469052dfb
@@ -189,2 +184 @@ class DBImpl : public DB {
-<<<<<<< HEAD
-  Status WriteLevel0Table(ColumnFamilyData* cfd, std::vector<MemTable*>& mems,
+  Status WriteLevel0Table(ColumnFamilyData* cfd, autovector<MemTable*>& mems,
@@ -192,7 +185,0 @@ class DBImpl : public DB {
-||||||| 183ba01a0
-  Status WriteLevel0Table(std::vector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber);
-=======
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber);
->>>>>>> 4564b2e8f9a2da07fa84fb5f3bba007469052dfb
@@ -237,0 +225,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -295,4 +286,0 @@ class DBImpl : public DB {
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
