--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_test.no_comments_conflict.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@db_test.no_comments_merged.cc
@@ -484,8 +483,0 @@ class DBTest {
-<<<<<<< HEAD
-            case kTypeColumnFamilyDeletion:
-            case kTypeColumnFamilyValue:
-            case kTypeColumnFamilyMerge:
-            case kTypeLogData:
-||||||| 183ba01a0
-            case kTypeLogData:
-=======
@@ -493 +484,0 @@ class DBTest {
->>>>>>> 4564b2e8
