diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_conflict.cc b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_merged.cc
index cc82d32..39ac3a4 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_conflict.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@memtable.no_comments_merged.cc
@@ -249,8 +248,0 @@ bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
-<<<<<<< HEAD
-        case kTypeColumnFamilyDeletion:
-        case kTypeColumnFamilyValue:
-        case kTypeColumnFamilyMerge:
-        case kTypeLogData:
-||||||| 183ba01a0
-        case kTypeLogData:
-=======
@@ -258 +249,0 @@ bool MemTable::Get(const LookupKey& key, std::string* value, Status* s,
->>>>>>> 4564b2e8
