diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_merged.h
index 90de240..ec3569f 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_merged.h
@@ -17,3 +17 @@ namespace rocksdb {
-namespace log {
-class Writer;
-}
+namespace log { class Writer; }
@@ -24 +21,0 @@ class MemTable;
-class TableCache;
@@ -31,2 +28,4 @@ extern int FindFile(const InternalKeyComparator& icmp,
-                    const std::vector<FileMetaData*>& files, const Slice& key);
-extern bool SomeFileOverlapsRange(const InternalKeyComparator& icmp,
+                    const std::vector<FileMetaData*>& files,
+                    const Slice& key);
+extern bool SomeFileOverlapsRange(
+    const InternalKeyComparator& icmp,
@@ -46,2 +45,3 @@ class Version {
-           Status* status, MergeContext* merge_context, GetStats* stats,
-           const Options& db_option, bool* value_found = nullptr);
+           Status* status, MergeContext* merge_context,
+           GetStats* stats, const Options& db_option,
+           bool* value_found = nullptr);
@@ -74,3 +74,3 @@ class Version {
-      unsigned int
-          index);
-  bool OverlapInLevel(int level, const Slice* smallest_user_key,
+      unsigned int index);
+  bool OverlapInLevel(int level,
+                      const Slice* smallest_user_key,
@@ -148,0 +149,3 @@ class VersionSet {
+  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
+  static Status ListColumnFamilies(std::vector<std::string>* column_families,
+                                   const std::string& dbname, Env* env);
@@ -186,4 +189,3 @@ class VersionSet {
-                            FileMetaData* metadata, ColumnFamilyData** cfd);
-  Status GetMetadataForFile(uint64_t number, int* filelevel,
-                            FileMetaData** metadata);
-  void GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata);
+                            FileMetaData** metadata, ColumnFamilyData** cfd);
+  void GetLiveFilesMetaData(
+    std::vector<LiveFileMetaData> *metadata);
@@ -198 +199,0 @@ class VersionSet {
-  friend class Compaction;
@@ -228,4 +228,0 @@ class VersionSet {
- public:
-  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
-  static Status ListColumnFamilies(std::vector<std::string>* column_families,
-                                   const std::string& dbname, Env* env);
