--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_conflict.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_merged.h
@@ -5 +4,0 @@
-<<<<<<< HEAD
@@ -7,2 +5,0 @@
-||||||| 183ba01a0
-=======
@@ -10 +6,0 @@
->>>>>>> 4564b2e8
@@ -24,7 +19,0 @@ struct FileMetaData {
-<<<<<<< HEAD
-  FileMetaData() : refs(0), allowed_seeks(1 << 30), file_size(0),
-                   being_compacted(false) {}
-||||||| 183ba01a0
-  FileMetaData() : refs(0), allowed_seeks(1 << 30), file_size(0),
-                   being_compacted(false) { }
-=======
@@ -37 +28,0 @@ struct FileMetaData {
->>>>>>> 4564b2e8
