--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_mergegen.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_merged.h
@@ -188 +188,2 @@ class VersionSet {
-Status GetMetadataForFile(uint64_t number, int* filelevel, FileMetaData* metadata, ColumnFamilyData** cfd);
+  Status GetMetadataForFile(uint64_t number, int* filelevel,
+                            FileMetaData** metadata, ColumnFamilyData** cfd);
