--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_merged.h
@@ -24 +21,0 @@ class MemTable;
-class TableCache;
@@ -148,0 +149,3 @@ class VersionSet {
+  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
+  static Status ListColumnFamilies(std::vector<std::string>* column_families,
+                                   const std::string& dbname, Env* env);
@@ -186,4 +189,3 @@ class VersionSet {
-                            FileMetaData* metadata, ColumnFamilyData** cfd);
-  Status GetMetadataForFile(uint64_t number, int* filelevel,
-                            FileMetaData** metadata);
-  void GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata);
+                            FileMetaData** metadata, ColumnFamilyData** cfd);
+  void GetLiveFilesMetaData(
+    std::vector<LiveFileMetaData> *metadata);
@@ -198 +199,0 @@ class VersionSet {
-  friend class Compaction;
@@ -228,4 +228,0 @@ class VersionSet {
- public:
-  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
-  static Status ListColumnFamilies(std::vector<std::string>* column_families,
-                                   const std::string& dbname, Env* env);
