--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_mergegen.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_merged.h
@@ -19 +20,9 @@ struct FileMetaData {
-FileMetaData(uint64_t number, uint64_t file_size) : refs(0), allowed_seeks(1 << 30), number(number), file_size(file_size), being_compacted(false), table_reader_handle(nullptr) { } FileMetaData() : FileMetaData(0, 0) {}
+  Cache::Handle* table_reader_handle;
+  FileMetaData(uint64_t number, uint64_t file_size)
+      : refs(0),
+        allowed_seeks(1 << 30),
+        number(number),
+        file_size(file_size),
+        being_compacted(false),
+        table_reader_handle(nullptr) {}
+  FileMetaData() : FileMetaData(0, 0) {}
