diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_conflict.h b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_merged.h
index 019e9d0..ec3569f 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_conflict.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_set.no_comments_merged.h
@@ -188 +187,0 @@ class VersionSet {
-<<<<<<< HEAD
@@ -190,14 +189 @@ class VersionSet {
-                            FileMetaData* metadata, ColumnFamilyData** cfd);
-||||||| 183ba01a0
-  double MaxBytesForLevel(int level);
-  uint64_t MaxFileSizeForLevel(int level);
-  void ReleaseCompactionFiles(Compaction* c, Status status);
-  Status GetMetadataForFile(
-    uint64_t number, int *filelevel, FileMetaData *metadata);
-=======
-  double MaxBytesForLevel(int level);
-  uint64_t MaxFileSizeForLevel(int level);
-  void ReleaseCompactionFiles(Compaction* c, Status status);
-  Status GetMetadataForFile(
-    uint64_t number, int *filelevel, FileMetaData **metadata);
->>>>>>> 4564b2e8
+                            FileMetaData** metadata, ColumnFamilyData** cfd);
