diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_merged.h
index 1472452..7feb3c6 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@version_edit.no_comments_merged.h
@@ -5 +4,0 @@
-#include "rocksdb/cache.h"
@@ -6,0 +6 @@
+#include "rocksdb/cache.h"
@@ -55,2 +55,4 @@ class VersionEdit {
-  void AddFile(int level, uint64_t file, uint64_t file_size,
-               const InternalKey& smallest, const InternalKey& largest,
+  void AddFile(int level, uint64_t file,
+               uint64_t file_size,
+               const InternalKey& smallest,
+               const InternalKey& largest,
@@ -72 +74,3 @@ class VersionEdit {
-  int NumEntries() { return new_files_.size() + deleted_files_.size(); }
+  int NumEntries() {
+    return new_files_.size() + deleted_files_.size();
+  }
