diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_conflict.cc b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_merged.cc
index 372b677..4bd0df9 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_conflict.cc
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/db@write_batch_test.no_comments_merged.cc
@@ -49,8 +48,0 @@ static std::string PrintContents(WriteBatch* b) {
-<<<<<<< HEAD
-      case kTypeColumnFamilyDeletion:
-      case kTypeColumnFamilyValue:
-      case kTypeColumnFamilyMerge:
-      case kTypeLogData:
-||||||| 183ba01a0
-      case kTypeLogData:
-=======
@@ -58 +49,0 @@ static std::string PrintContents(WriteBatch* b) {
->>>>>>> 4564b2e8
