diff --git a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_conflict.h b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_merged.h
index d884b35..7a6258c 100644
--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_conflict.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_merged.h
@@ -45,6 +44,0 @@ struct CompressionOptions {
-<<<<<<< HEAD
-struct Options;
-struct ColumnFamilyOptions {
-||||||| 183ba01a0
-struct Options {
-=======
@@ -56,2 +50,2 @@ enum UpdateStatus {
-struct Options {
->>>>>>> 4564b2e8
+struct Options;
+struct ColumnFamilyOptions {
@@ -62,14 +55,0 @@ struct Options {
-<<<<<<< HEAD
-||||||| 183ba01a0
-  bool create_if_missing;
-  bool error_if_exists;
-  bool paranoid_checks;
-  Env* env;
-  shared_ptr<Logger> info_log;
-=======
-  bool create_if_missing;
-  bool error_if_exists;
-  bool paranoid_checks;
-  Env* env;
-  shared_ptr<Logger> info_log;
->>>>>>> 4564b2e8
@@ -79,6 +58,0 @@ struct Options {
-<<<<<<< HEAD
-||||||| 183ba01a0
-  int max_open_files;
-=======
-  int max_open_files;
->>>>>>> 4564b2e8
@@ -123,2 +97,3 @@ struct Options {
-  std::vector<std::shared_ptr<TablePropertiesCollector>>
-    table_properties_collectors;
+  typedef std::vector<std::shared_ptr<TablePropertiesCollector>>
+      TablePropertiesCollectors;
+  TablePropertiesCollectors table_properties_collectors;
@@ -126,0 +102,6 @@ struct Options {
+  UpdateStatus (*inplace_callback)(char* existing_value,
+                                   uint32_t* existing_value_size,
+                                   Slice delta_value,
+                                   std::string* merged_value);
+  uint32_t memtable_prefix_bloom_bits;
+  uint32_t memtable_prefix_bloom_probes;
@@ -182,23 +162,0 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
-<<<<<<< HEAD
-||||||| 183ba01a0
-  std::shared_ptr<MemTableRepFactory> memtable_factory;
-  std::shared_ptr<TableFactory> table_factory;
-  std::vector<std::shared_ptr<TablePropertiesCollector>>
-  table_properties_collectors;
-  bool inplace_update_support;
-  size_t inplace_update_num_locks;
-=======
-  std::shared_ptr<MemTableRepFactory> memtable_factory;
-  std::shared_ptr<TableFactory> table_factory;
-  typedef std::vector<std::shared_ptr<TablePropertiesCollector>>
-          TablePropertiesCollectors;
-  TablePropertiesCollectors table_properties_collectors;
-  bool inplace_update_support;
-  size_t inplace_update_num_locks;
-  UpdateStatus (*inplace_callback)(char* existing_value,
-                                   uint32_t* existing_value_size,
-                                   Slice delta_value,
-                                   std::string* merged_value);
-  uint32_t memtable_prefix_bloom_bits;
-  uint32_t memtable_prefix_bloom_probes;
->>>>>>> 4564b2e8
