--- a/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_mergegen.h
+++ b/./output/rocksdb/0b4ccf76-4564b2e8-183ba01a/include@rocksdb@options.no_comments_merged.h
@@ -45,0 +46,6 @@ enum UpdateStatus {
+  UPDATE_FAILED = 0,
+  UPDATED_INPLACE = 1,
+  UPDATED = 2,
+};
+struct Options;
+struct ColumnFamilyOptions {
@@ -50 +55,0 @@ enum UpdateStatus {
-std::shared_ptr<CompactionFilterFactory> compaction_filter_factory;
@@ -92,2 +97,3 @@ std::shared_ptr<CompactionFilterFactory> compaction_filter_factory;
-  std::vector<std::shared_ptr<TablePropertiesCollector>>
-    table_properties_collectors;
+  typedef std::vector<std::shared_ptr<TablePropertiesCollector>>
+      TablePropertiesCollectors;
+  TablePropertiesCollectors table_properties_collectors;
@@ -95,0 +102,6 @@ std::shared_ptr<CompactionFilterFactory> compaction_filter_factory;
+  UpdateStatus (*inplace_callback)(char* existing_value,
+                                   uint32_t* existing_value_size,
+                                   Slice delta_value,
+                                   std::string* merged_value);
+  uint32_t memtable_prefix_bloom_bits;
+  uint32_t memtable_prefix_bloom_probes;
