diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_merged.h
index 058f4c1..6fb7ca2 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/8298fc61-e7f9de2c-6dc78dc5/libprocessgroup@cgroup_map.no_comments_merged.h
@@ -11,2 +10,0 @@ class CgroupController {
- private:
-  const ACgroupController* controller_ = nullptr;
@@ -14,2 +12 @@ class CgroupController {
-  explicitCgroupController(const ACgroupController* controller)
-      : controller_(controller) {}
+    explicit CgroupController(const ACgroupController* controller) : controller_(controller) {}
@@ -20,12 +16,0 @@ class CgroupController {
-  uint32_t CgroupController::version() const {
-    CHECK(HasValue());
-    return ACgroupController_getVersion(controller_);
-  }
-  const char* CgroupController::name() const {
-    CHECK(HasValue());
-    return ACgroupController_getName(controller_);
-  }
-  const char* CgroupController::path() const {
-    CHECK(HasValue());
-    return ACgroupController_getPath(controller_);
-  }
@@ -33,2 +18 @@ class CgroupController {
-  std::string GetProcsFilePath(const std::string& path, uid_t uid,
-                               pid_t pid) const;
+    std::string GetProcsFilePath(const std::string& path, uid_t uid, pid_t pid) const;
@@ -35,0 +20,2 @@ class CgroupController {
+  private:
+    const ACgroupController* controller_ = nullptr;
@@ -38,2 +23,0 @@ class CgroupMap {
- private:
-  bool loaded_ = false;
@@ -44,0 +29 @@ class CgroupMap {
+    bool loaded_ = false;
