--- a/./output/tmux/168eab11-41b318ac-8f345047/tty-term.no_comments_mergegen.c
+++ b/./output/tmux/168eab11-41b318ac-8f345047/tty-term.no_comments_merged.c
@@ -642 +642,11 @@ tty_term_string_i(struct tty_term *term, enum tty_code_code code, int a)
-#if defined(HAVE_TIPARM_S) s = tiparm_s(1, 0, x, a); #elif defined(HAVE_TIPARM) s = tiparm(x, a); #else s = tparm((char *)x, a, 0, 0, 0, 0, 0, 0); #endif if (s == NULL) { log_debug("could not expand %s", tty_term_codes[code].name); return (""); }
+#if defined(HAVE_TIPARM_S)
+ s = tiparm_s(1, 0, x, a);
+#elif defined(HAVE_TIPARM)
+ s = tiparm(x, a);
+#else
+ s = tparm((char *)x, a, 0, 0, 0, 0, 0, 0, 0, 0);
+#endif
+ if (s == NULL) {
+  log_debug("could not expand %s", tty_term_codes[code].name);
+  return ("");
+ }
@@ -649 +659,11 @@ tty_term_string_ii(struct tty_term *term, enum tty_code_code code, int a, int b)
-#if defined(HAVE_TIPARM_S) s = tiparm_s(2, 0, x, a, b); #elif defined(HAVE_TIPARM) s = tiparm(x, a, b); #else s = tparm((char *)x, a, b, 0, 0, 0, 0, 0); #endif if (s == NULL) { log_debug("could not expand %s", tty_term_codes[code].name); return (""); } return (s);
+#if defined(HAVE_TIPARM_S)
+ s = tiparm_s(2, 0, x, a, b);
+#elif defined(HAVE_TIPARM)
+ s = tiparm(x, a, b);
+#else
+ s = tparm((char *)x, a, b, 0, 0, 0, 0, 0, 0, 0);
+#endif
+ if (s == NULL) {
+  log_debug("could not expand %s", tty_term_codes[code].name);
+  return ("");
+ }
@@ -657 +677,11 @@ tty_term_string_iii(struct tty_term *term, enum tty_code_code code, int a,
-#if defined(HAVE_TIPARM_S) s = tiparm_s(3, 0, x, a, b, c); #elif defined(HAVE_TIPARM) s = tiparm(x, a, b, c); #else s = tparm((char *)x, a, b, c, 0, 0, 0, 0, 0); #endif if (s == NULL) { log_debug("could not expand %s", tty_term_codes[code].name); return (""); } return (s);
+#if defined(HAVE_TIPARM_S)
+ s = tiparm_s(3, 0, x, a, b, c);
+#elif defined(HAVE_TIPARM)
+ s = tiparm(x, a, b, c);
+#else
+ s = tparm((char *)x, a, b, c, 0, 0, 0, 0, 0, 0);
+#endif
+ if (s == NULL) {
+  log_debug("could not expand %s", tty_term_codes[code].name);
+  return ("");
+ }
@@ -664 +694,11 @@ tty_term_string_s(struct tty_term *term, enum tty_code_code code, const char *a)
-#if defined(HAVE_TIPARM_S) s = tiparm_s(1, 1, x, a); #elif defined(HAVE_TIPARM) s = tiparm(x, a); #else s = tparm((char *)x, (long)a, 0, 0, 0, 0, 0, 0); #endif if (s == NULL) { log_debug("could not expand %s", tty_term_codes[code].name); return (""); } return (s);
+#if defined(HAVE_TIPARM_S)
+ s = tiparm_s(1, 1, x, a);
+#elif defined(HAVE_TIPARM)
+ s = tiparm(x, a);
+#else
+ s = tparm((char *)x, (long)a, 0, 0, 0, 0, 0, 0, 0, 0);
+#endif
+ if (s == NULL) {
+  log_debug("could not expand %s", tty_term_codes[code].name);
+  return ("");
+ }
@@ -672 +712,11 @@ tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-#if defined(HAVE_TIPARM_S) s = tiparm_s(2, 3, x, a, b); #elif defined(HAVE_TIPARM) s = tiparm(x, a, b); #else s = tparm((char *)x, (long)a, (long)b, 0, 0, 0, 0, 0, 0); #endif if (s == NULL) { log_debug("could not expand %s", tty_term_codes[code].name); return (""); } return (s);
+#if defined(HAVE_TIPARM_S)
+ s = tiparm_s(2, 3, x, a, b);
+#elif defined(HAVE_TIPARM)
+ s = tiparm(x, a, b);
+#else
+ s = tparm((char *)x, (long)a, (long)b, 0, 0, 0, 0, 0, 0, 0);
+#endif
+ if (s == NULL) {
+  log_debug("could not expand %s", tty_term_codes[code].name);
+  return ("");
+ }
