--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/168eab11-41b318ac-8f345047/tty-term.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/168eab11-41b318ac-8f345047/tty-term.no_comments_merged.c
@@ -2 +1,0 @@
-#include <curses.h>
@@ -10 +8,0 @@
-#include <vis.h>
@@ -595 +641,0 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -603,6 +648,0 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-|||||||
-  s = tparm((char *)x, a);
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-=======
-  s = tparm((char *)x, a);
@@ -613 +652,0 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
->>>>>>> 41b318ac7c881fc4cdf834ea5e14a7e8646605aa
@@ -619 +658,0 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -627,6 +665,0 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-|||||||
-  s = tparm((char *)x, a, b);
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-=======
-  s = tparm((char *)x, a, b);
@@ -637 +669,0 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
->>>>>>> 41b318ac7c881fc4cdf834ea5e14a7e8646605aa
@@ -643 +676,0 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -651,6 +683,0 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-|||||||
-  s = tparm((char *)x, a, b, c);
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-=======
-  s = tparm((char *)x, a, b, c);
@@ -661 +687,0 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
->>>>>>> 41b318ac7c881fc4cdf834ea5e14a7e8646605aa
@@ -667 +693,0 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -675,6 +700,0 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-|||||||
-  s = tparm((char *)x, (long)a);
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-=======
-  s = tparm((char *)x, (long)a);
@@ -685 +704,0 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
->>>>>>> 41b318ac7c881fc4cdf834ea5e14a7e8646605aa
@@ -691 +711,0 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -699,6 +718,0 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-|||||||
-  s = tparm((char *)x, (long)a, (long)b);
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
-=======
-  s = tparm((char *)x, (long)a, (long)b);
@@ -709 +722,0 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
->>>>>>> 41b318ac7c881fc4cdf834ea5e14a7e8646605aa
