diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/tmux.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/tmux.no_comments_merged.c
index ba0c6a8..63e6a01 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/tmux.no_comments_merged.c
@@ -3 +2,0 @@
-#include <err.h>
@@ -7 +5,0 @@
-#include <paths.h>
@@ -11 +8,0 @@
-#include <util.h>
@@ -30 +27,3 @@ static int checkshell(const char *);
-static __dead void usage(void) {
+static __dead void
+usage(void)
+{
@@ -37 +36,3 @@ static __dead void usage(void) {
-static const char *getshell(void) {
+static const char *
+getshell(void)
+{
@@ -41 +42,2 @@ static const char *getshell(void) {
-  if (checkshell(shell)) return (shell);
+ if (checkshell(shell))
+  return (shell);
@@ -43 +45,2 @@ static const char *getshell(void) {
-  if (pw != NULL && checkshell(pw->pw_shell)) return (pw->pw_shell);
+ if (pw != NULL && checkshell(pw->pw_shell))
+  return (pw->pw_shell);
@@ -46,4 +49,9 @@ static const char *getshell(void) {
-static int checkshell(const char *shell) {
-  if (shell == NULL || *shell != '/') return (0);
-  if (areshell(shell)) return (0);
-  if (access(shell, X_OK) != 0) return (0);
+static int
+checkshell(const char *shell)
+{
+ if (shell == NULL || *shell != '/')
+  return (0);
+ if (areshell(shell))
+  return (0);
+ if (access(shell, X_OK) != 0)
+  return (0);
@@ -52 +60,3 @@ static int checkshell(const char *shell) {
-int areshell(const char *shell) {
+int
+areshell(const char *shell)
+{
@@ -59,2 +69,4 @@ int areshell(const char *shell) {
-  if (*progname == '-') progname++;
-  if (strcmp(ptr, progname) == 0) return (1);
+ if (*progname == '-')
+  progname++;
+ if (strcmp(ptr, progname) == 0)
+  return (1);
@@ -63 +75,3 @@ int areshell(const char *shell) {
-static char *make_label(const char *label) {
+static char *
+make_label(const char *label)
+{
@@ -68 +82,2 @@ static char *make_label(const char *label) {
-  if (label == NULL) label = "default";
+ if (label == NULL)
+  label = "default";
@@ -74,2 +89,4 @@ static char *make_label(const char *label) {
-  if (mkdir(base, S_IRWXU) != 0 && errno != EEXIST) goto fail;
-  if (lstat(base, &sb) != 0) goto fail;
+ if (mkdir(base, S_IRWXU) != 0 && errno != EEXIST)
+  goto fail;
+ if (lstat(base, &sb) != 0)
+  goto fail;
@@ -95 +112,3 @@ fail:
-void setblocking(int fd, int state) {
+void
+setblocking(int fd, int state)
+{
@@ -105 +124,3 @@ void setblocking(int fd, int state) {
-const char *find_home(void) {
+const char *
+find_home(void)
+{
@@ -108 +129,2 @@ const char *find_home(void) {
-  if (home != NULL) return (home);
+ if (home != NULL)
+  return (home);
@@ -119 +141,3 @@ const char *find_home(void) {
-int main(int argc, char **argv) {
+int
+main(int argc, char **argv)
+{
@@ -185,2 +209,4 @@ int main(int argc, char **argv) {
-  if (shell_command != NULL && argc != 0) usage();
-  if ((ptm_fd = getptmfd()) == -1) err(1, "getptmfd");
+ if (shell_command != NULL && argc != 0)
+  usage();
+ if ((ptm_fd = getptmfd()) == -1)
+  err(1, "getptmfd");
@@ -188,2 +214 @@ int main(int argc, char **argv) {
-             "recvfd proc exec tty ps",
-             NULL) != 0)
+     "recvfd proc exec tty ps", NULL) != 0)
@@ -195,4 +220,8 @@ int main(int argc, char **argv) {
-    if (s == NULL || *s == '\0') s = getenv("LC_CTYPE");
-    if (s == NULL || *s == '\0') s = getenv("LANG");
-    if (s == NULL || *s == '\0') s = "";
-    if (strcasestr(s, "UTF-8") != NULL || strcasestr(s, "UTF8") != NULL)
+  if (s == NULL || *s == '\0')
+   s = getenv("LC_CTYPE");
+  if (s == NULL || *s == '\0')
+   s = getenv("LANG");
+  if (s == NULL || *s == '\0')
+   s = "";
+  if (strcasestr(s, "UTF-8") != NULL ||
+      strcasestr(s, "UTF8") != NULL)
@@ -203 +232,2 @@ int main(int argc, char **argv) {
-  for (var = environ; *var != NULL; var++) environ_put(global_environ, *var);
+ for (var = environ; *var != NULL; var++)
+  environ_put(global_environ, *var);
@@ -210 +240,2 @@ int main(int argc, char **argv) {
-    if (oe->scope == OPTIONS_TABLE_SERVER) options_default(global_options, oe);
+  if (oe->scope == OPTIONS_TABLE_SERVER)
+   options_default(global_options, oe);
@@ -219 +250,2 @@ int main(int argc, char **argv) {
-    if (strrchr(s, '/') != NULL) s = strrchr(s, '/') + 1;
+  if (strrchr(s, '/') != NULL)
+   s = strrchr(s, '/') + 1;
@@ -240,7 +272 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
-  exit(client_main(osdep_event_init(), argc, argv, flags, shellcmd));
-|||||||
-  exit(client_main(event_init(), argc, argv, flags, shellcmd));
-=======
-  exit(client_main(event_init(), argc, argv, flags));
->>>>>>> 51112221eeb31ced907f0dfcf077582996c20c07
+ exit(client_main(osdep_event_init(), argc, argv, flags));
