--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/fbbf5a10-51112221-58b79660/tmux.no_comments_merged.c
@@ -3 +2,0 @@
-#include <err.h>
@@ -7 +5,0 @@
-#include <paths.h>
@@ -11 +8,0 @@
-#include <util.h>
@@ -240,7 +272 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
-  exit(client_main(osdep_event_init(), argc, argv, flags, shellcmd));
-|||||||
-  exit(client_main(event_init(), argc, argv, flags, shellcmd));
-=======
-  exit(client_main(event_init(), argc, argv, flags));
->>>>>>> 51112221eeb31ced907f0dfcf077582996c20c07
+ exit(client_main(osdep_event_init(), argc, argv, flags));
