diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@statsd.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@statsd.no_comments_merged.cc
index 5aa47b4..00362f5 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@statsd.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@statsd.no_comments_merged.cc
@@ -100,2 +100 @@ constexpr std::optional<int32_t> EncodeDatumId(DatumId datum_id) {
-          statsd::
-              ART_DATUM_REPORTED__KIND__ART_DATUM_GC_YOUNG_GENERATION_COLLECTION_THROUGHPUT_AVG_MB_PER_SEC);
+          statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_YOUNG_GENERATION_COLLECTION_THROUGHPUT_AVG_MB_PER_SEC);
@@ -104,2 +103 @@ constexpr std::optional<int32_t> EncodeDatumId(DatumId datum_id) {
-          statsd::
-              ART_DATUM_REPORTED__KIND__ART_DATUM_GC_FULL_HEAP_COLLECTION_THROUGHPUT_AVG_MB_PER_SEC);
+          statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_FULL_HEAP_COLLECTION_THROUGHPUT_AVG_MB_PER_SEC);
@@ -108,2 +106 @@ constexpr std::optional<int32_t> EncodeDatumId(DatumId datum_id) {
-          statsd::
-              ART_DATUM_REPORTED__KIND__ART_DATUM_GC_YOUNG_GENERATION_TRACING_THROUGHPUT_AVG_MB_PER_SEC);
+          statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_YOUNG_GENERATION_TRACING_THROUGHPUT_AVG_MB_PER_SEC);
@@ -112,2 +109 @@ constexpr std::optional<int32_t> EncodeDatumId(DatumId datum_id) {
-          statsd::
-              ART_DATUM_REPORTED__KIND__ART_DATUM_GC_FULL_HEAP_TRACING_THROUGHPUT_AVG_MB_PER_SEC);
+          statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_FULL_HEAP_TRACING_THROUGHPUT_AVG_MB_PER_SEC);
@@ -115 +111,2 @@ constexpr std::optional<int32_t> EncodeDatumId(DatumId datum_id) {
-      return std::make_optional(statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_WORLD_STOP_TIME_US);
+      return std::make_optional(
+          statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_WORLD_STOP_TIME_US);
@@ -120 +117,2 @@ constexpr std::optional<int32_t> EncodeDatumId(DatumId datum_id) {
-      return std::make_optional(statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_WORLD_STOP_COUNT);
+      return std::make_optional(
+          statsd::ART_DATUM_REPORTED__KIND__ART_DATUM_GC_WORLD_STOP_COUNT);
@@ -185 +183,2 @@ constexpr int32_t EncodeCompileFilter(CompilerFilterReporting filter) {
-      return statsd::ART_DATUM_REPORTED__COMPILE_FILTER__ART_COMPILATION_FILTER_EVERYTHING_PROFILE;
+      return statsd::
+          ART_DATUM_REPORTED__COMPILE_FILTER__ART_COMPILATION_FILTER_EVERYTHING_PROFILE;
@@ -195,2 +194 @@ constexpr int32_t EncodeCompileFilter(CompilerFilterReporting filter) {
-      return statsd::
-          ART_DATUM_REPORTED__COMPILE_FILTER__ART_COMPILATION_FILTER_FAKE_RUN_FROM_APK_FALLBACK;
+       return statsd::ART_DATUM_REPORTED__COMPILE_FILTER__ART_COMPILATION_FILTER_FAKE_RUN_FROM_APK_FALLBACK;
@@ -343 +340,0 @@ class StatsdBackend : public MetricsBackend {
-<<<<<<< HEAD
@@ -348,8 +344,0 @@ class StatsdBackend : public MetricsBackend {
-|||||||
-        EncodeInstructionSet(kRuntimeISA));
-  }
-}
-=======
-        EncodeInstructionSet(kRuntimeISA));
-  }
->>>>>>> 37c19554816823543e0d80d0b3ba4318ab8ec3ab
@@ -383,0 +373 @@ void ReportDeviceMetrics() {
+#pragma clang diagnostic pop
