--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@reporter.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@reporter.no_comments_merged.cc
@@ -19 +19,5 @@ std::unique_ptr<MetricsReporter> MetricsReporter::Create(const ReportingConfig&
-MetricsReporter::MetricsReporter(const ReportingConfig& config, Runtime* runtime) : config_ {}
+MetricsReporter::MetricsReporter(const ReportingConfig& config, Runtime* runtime)
+    : config_{config},
+      runtime_{runtime},
+      startup_reported_{false},
+      report_interval_index_{0} {}
@@ -150 +155,0 @@ void MetricsReporter::ReportMetrics() {
-<<<<<<< HEAD
@@ -152,7 +156,0 @@ void MetricsReporter::ReportMetrics() {
-|||||||
-  for (auto& backend : backends_) {
-    metrics->ReportAllMetrics(backend.get());
-  }
-=======
-  metrics->ReportAllMetricsAndResetValueMetrics(ToRawPointers(backends_));
->>>>>>> 37c19554816823543e0d80d0b3ba4318ab8ec3ab
@@ -254,0 +259 @@ std::optional<ReportingPeriodSpec> ReportingPeriodSpec::Parse(const std::string&
+#pragma clang diagnostic pop
