--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@statsd.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@statsd.no_comments_merged.cc
@@ -343 +340,0 @@ class StatsdBackend : public MetricsBackend {
-<<<<<<< HEAD
@@ -348,8 +344,0 @@ class StatsdBackend : public MetricsBackend {
-|||||||
-        EncodeInstructionSet(kRuntimeISA));
-  }
-}
-=======
-        EncodeInstructionSet(kRuntimeISA));
-  }
->>>>>>> 37c19554816823543e0d80d0b3ba4318ab8ec3ab
@@ -383,0 +373 @@ void ReportDeviceMetrics() {
+#pragma clang diagnostic pop
