diff --git a/./output/redis/4379b8b4-dfb12f06-3a5bf75e/src@pubsub.no_comments_mergegen.c b/./output/redis/4379b8b4-dfb12f06-3a5bf75e/src@pubsub.no_comments_merged.c
index f6ddd06..41883ad 100644
--- a/./output/redis/4379b8b4-dfb12f06-3a5bf75e/src@pubsub.no_comments_mergegen.c
+++ b/./output/redis/4379b8b4-dfb12f06-3a5bf75e/src@pubsub.no_comments_merged.c
@@ -225 +225,5 @@ int pubsubPublishMessage(robj *channel, robj *message) {
-sdslen(channel->ptr),0)) continue; listRewind(clients,&li); while ((ln = listNext(&li)) != NULL) { client *c = listNodeValue(ln); addReplyPubsubPatMessage(c,pattern,channel,message);
+                                sdslen(channel->ptr),0)) continue;
+            listRewind(clients,&li);
+            while ((ln = listNext(&li)) != NULL) {
+                client *c = listNodeValue(ln);
+                addReplyPubsubPatMessage(c,pattern,channel,message);
