diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/5817f628-bca2a30e-018580fb/src@objective-c@GRPCClient@version.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/5817f628-bca2a30e-018580fb/src@objective-c@GRPCClient@version.no_comments_merged.h
index f8db7cd..b2fe861 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/5817f628-bca2a30e-018580fb/src@objective-c@GRPCClient@version.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/5817f628-bca2a30e-018580fb/src@objective-c@GRPCClient@version.no_comments_merged.h
@@ -2,2 +1,0 @@
-#define GRPC_OBJC_VERSION_STRING @"1.25.0-dev"
-#define GRPC_OBJC_VERSION_STRING @"1.25.0"
