diff --git a/./output/grpc/5817f628-bca2a30e-018580fb/src@cpp@common@version_cc.no_comments_mergegen.cc b/./output/grpc/5817f628-bca2a30e-018580fb/src@cpp@common@version_cc.no_comments_merged.cc
index f67413a..40e2329 100644
--- a/./output/grpc/5817f628-bca2a30e-018580fb/src@cpp@common@version_cc.no_comments_mergegen.cc
+++ b/./output/grpc/5817f628-bca2a30e-018580fb/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3 +3 @@ namespace grpc {
-namespace grpc {grpc::string Version() { return "1.26.0-dev"; }
+grpc::string Version() { return "1.26.0-dev"; }
