--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_merged.cpp
@@ -53 +54,10 @@ PointerChoreographer::PointerChoreographer(InputListenerInterface& listener,
-      : mTouchControllerConstructor([this]() {}
+      : mTouchControllerConstructor([this]() {
+            return mPolicy.createPointerController(
+                    PointerControllerInterface::ControllerType::TOUCH);
+        }),
+        mNextListener(listener),
+        mPolicy(policy),
+        mDefaultMouseDisplayId(ADISPLAY_ID_DEFAULT),
+        mNotifiedPointerDisplayId(ADISPLAY_ID_NONE),
+        mShowTouchesEnabled(false),
+        mStylusPointerIconEnabled(false) {}
@@ -320 +329,0 @@ PointerChoreographer::PointerDisplayChange PointerChoreographer::updatePointerCo
-<<<<<<< HEAD
@@ -323,6 +331,0 @@ PointerChoreographer::PointerDisplayChange PointerChoreographer::updatePointerCo
-|||||||
-        if (isFromSource(sources, AINPUT_SOURCE_MOUSE) ||
-            isFromSource(sources, AINPUT_SOURCE_MOUSE_RELATIVE)) {
-=======
-        if (isMouseOrTouchpad(sources)) {
->>>>>>> b5f3b9b846e90b10177f4c2b78fa361a93e934b1
