diff --git a/./output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_mergegen.cpp b/./output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_merged.cpp
index 074de4e..609d3c3 100644
--- a/./output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_merged.cpp
@@ -330 +330,2 @@ PointerChoreographer::PointerDisplayChange PointerChoreographer::updatePointerCo
-const bool isKnownMouse = mMouseDevices.count(info.getId()) != 0; if (isMouseOrTouchpad(sources) || isKnownMouse) {
+        const bool isKnownMouse = mMouseDevices.count(info.getId()) != 0;
+        if (isMouseOrTouchpad(sources) || isKnownMouse) {
