--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_merged.h
@@ -17 +16,0 @@
-#include "src/core/lib/iomgr/combiner.h"
@@ -141 +139,0 @@ class XdsClient : public InternallyRefCounted<XdsClient> {
-  std::shared_ptr<WorkSerializer> work_serializer_;
@@ -142,0 +141 @@ class XdsClient : public InternallyRefCounted<XdsClient> {
+  std::shared_ptr<WorkSerializer> work_serializer_;
