diff --git a/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_conflict.h b/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_merged.h
index 734432f..285f887 100644
--- a/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_conflict.h
+++ b/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_merged.h
@@ -140,5 +139,0 @@ class XdsClient : public InternallyRefCounted<XdsClient> {
-<<<<<<< HEAD
-  std::shared_ptr<WorkSerializer> work_serializer_;
-||||||| 7f36ff039f
-  Combiner* combiner_;
-=======
@@ -146,2 +141 @@ class XdsClient : public InternallyRefCounted<XdsClient> {
-  Combiner* combiner_;
->>>>>>> 006c8158
+  std::shared_ptr<WorkSerializer> work_serializer_;
