--- a/./output/frameworks_av/4376579d-083f262d-b6e50060/media@module@libmediatranscoding@TranscodingResourcePolicy.no_comments_conflict.cpp
+++ b/./output/frameworks_av/4376579d-083f262d-b6e50060/media@module@libmediatranscoding@TranscodingResourcePolicy.no_comments_merged.cpp
@@ -69 +68,0 @@ TranscodingResourcePolicy::~TranscodingResourcePolicy() {
-<<<<<<< HEAD
@@ -82,16 +80,0 @@ TranscodingResourcePolicy::~TranscodingResourcePolicy() {
-||||||| b6e500602a
-=======
-    {
-        std::lock_guard<std::mutex> guard(sCookiesMutex);
-        for (auto it = sCookies.begin(); it != sCookies.end();) {
-            const uintptr_t key = it->first;
-            std::lock_guard guard(mCookieKeysLock);
-            if (mCookieKeys.find(key) != mCookieKeys.end()) {
-                mCookieKeys.erase(key);
-                it = sCookies.erase(it);
-            } else {
-                it++;
-            }
-        }
-    }
->>>>>>> 083f262d
@@ -123 +105,0 @@ void TranscodingResourcePolicy::registerSelf() {
-<<<<<<< HEAD
@@ -134,13 +115,0 @@ void TranscodingResourcePolicy::registerSelf() {
-||||||| b6e500602a
-    AIBinder_linkToDeath(binder.get(), mDeathRecipient.get(), reinterpret_cast<void*>(this));
-=======
-    std::unique_ptr<TranscodingResourcePolicyCookie> cookie =
-            std::make_unique<TranscodingResourcePolicyCookie>(this);
-    uintptr_t cookieKey = sCookieKeyCounter++;
-    sCookies.emplace(cookieKey, std::move(cookie));
-    {
-        std::lock_guard guard(mCookieKeysLock);
-        mCookieKeys.insert(cookieKey);
-    }
-    AIBinder_linkToDeath(binder.get(), mDeathRecipient.get(), reinterpret_cast<void*>(cookieKey));
->>>>>>> 083f262d
