--- a/./output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_mergegen.c
+++ b/./output/redis/15bc1cc1-34a8b517-d5b36c51/src@redis.no_comments_merged.c
@@ -153 +153,4 @@ struct redisCommand redisCommandTable[] = {
-{"client",clientCommand,-2,0,NULL,0,0,0,0,0}, {"eval",evalCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0}, {"evalsha",evalShaCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0}, {"slowlog",slowlogCommand,-2,0,NULL,0,0,0,0}
+    {"client",clientCommand,-2,0,NULL,0,0,0,0,0},
+    {"eval",evalCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
+    {"evalsha",evalShaCommand,-3,REDIS_CMD_DENYOOM,zunionInterGetKeys,0,0,0,0,0},
+    {"slowlog",slowlogCommand,-2,0,NULL,0,0,0,0,0}
