diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@config.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@config.no_comments_merged.c
index 2c787de..f152d89 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@config.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/15bc1cc1-34a8b517-d5b36c51/src@config.no_comments_merged.c
@@ -3,6 +3,3 @@ int yesnotoi(char *s) {
-  if (!strcasecmp(s, "yes"))
-    return 1;
-  else if (!strcasecmp(s, "no"))
-    return 0;
-  else
-    return -1;
+    if (!strcasecmp(s,"yes")) return 1;
+    else if (!strcasecmp(s,"no")) return 0;
+    else return -1;
@@ -11,2 +8 @@ void appendServerSaveParams(time_t seconds, int changes) {
-  server.saveparams = zrealloc(
-      server.saveparams, sizeof(struct saveparam) * (server.saveparamslen + 1));
+    server.saveparams = zrealloc(server.saveparams,sizeof(struct saveparam)*(server.saveparamslen+1));
@@ -31,2 +27 @@ void loadServerConfig(char *filename) {
-      redisLog(REDIS_WARNING, "Fatal error, can't open config file '%s'",
-               filename);
+            redisLog(REDIS_WARNING, "Fatal error, can't open config file '%s'", filename);
@@ -51,2 +46 @@ void loadServerConfig(char *filename) {
-        err = "Invalid timeout value";
-        goto loaderr;
+                err = "Invalid timeout value"; goto loaderr;
@@ -57,2 +51 @@ void loadServerConfig(char *filename) {
-        err = "Invalid port";
-        goto loaderr;
+                err = "Invalid port"; goto loaderr;
@@ -68,2 +61 @@ void loadServerConfig(char *filename) {
-        err = "Invalid save parameters";
-        goto loaderr;
+                err = "Invalid save parameters"; goto loaderr;
@@ -74,2 +66,2 @@ void loadServerConfig(char *filename) {
-        redisLog(REDIS_WARNING, "Can't chdir to '%s': %s", argv[1],
-                 strerror(errno));
+                redisLog(REDIS_WARNING,"Can't chdir to '%s': %s",
+                    argv[1], strerror(errno));
@@ -79,8 +71,4 @@ void loadServerConfig(char *filename) {
-      if (!strcasecmp(argv[1], "debug"))
-        server.verbosity = REDIS_DEBUG;
-      else if (!strcasecmp(argv[1], "verbose"))
-        server.verbosity = REDIS_VERBOSE;
-      else if (!strcasecmp(argv[1], "notice"))
-        server.verbosity = REDIS_NOTICE;
-      else if (!strcasecmp(argv[1], "warning"))
-        server.verbosity = REDIS_WARNING;
+            if (!strcasecmp(argv[1],"debug")) server.verbosity = REDIS_DEBUG;
+            else if (!strcasecmp(argv[1],"verbose")) server.verbosity = REDIS_VERBOSE;
+            else if (!strcasecmp(argv[1],"notice")) server.verbosity = REDIS_NOTICE;
+            else if (!strcasecmp(argv[1],"warning")) server.verbosity = REDIS_WARNING;
@@ -101,2 +89,2 @@ void loadServerConfig(char *filename) {
-          err = sdscatprintf(sdsempty(), "Can't open the log file: %s",
-                             strerror(errno));
+                    err = sdscatprintf(sdsempty(),
+                        "Can't open the log file: %s", strerror(errno));
@@ -109,2 +97 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -120,5 +107,11 @@ void loadServerConfig(char *filename) {
-          {"user", LOG_USER}, {"local0", LOG_LOCAL0},
-          {"local1", LOG_LOCAL1}, {"local2", LOG_LOCAL2},
-          {"local3", LOG_LOCAL3}, {"local4", LOG_LOCAL4},
-          {"local5", LOG_LOCAL5}, {"local6", LOG_LOCAL6},
-          {"local7", LOG_LOCAL7}, {NULL, 0}};
+                {"user", LOG_USER},
+                {"local0", LOG_LOCAL0},
+                {"local1", LOG_LOCAL1},
+                {"local2", LOG_LOCAL2},
+                {"local3", LOG_LOCAL3},
+                {"local4", LOG_LOCAL4},
+                {"local5", LOG_LOCAL5},
+                {"local6", LOG_LOCAL6},
+                {"local7", LOG_LOCAL7},
+                {NULL, 0}
+            };
@@ -133,3 +126 @@ void loadServerConfig(char *filename) {
-        err =
-            "Invalid log facility. Must be one of USER or between "
-            "LOCAL0-LOCAL7";
+                err = "Invalid log facility. Must be one of USER or between LOCAL0-LOCAL7";
@@ -141,2 +132 @@ void loadServerConfig(char *filename) {
-        err = "Invalid number of databases";
-        goto loaderr;
+                err = "Invalid number of databases"; goto loaderr;
@@ -181,2 +171 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -185,2 +174 @@ void loadServerConfig(char *filename) {
-      redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"",
-               argv[0]);
+            redisLog(REDIS_WARNING, "Deprecated configuration directive: \"%s\"", argv[0]);
@@ -189,2 +177 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -194,2 +181 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -199,2 +185 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -204,2 +189 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -210 +194,2 @@ void loadServerConfig(char *filename) {
-    } else if (!strcasecmp(argv[0], "no-appendfsync-on-rewrite") && argc == 2) {
+        } else if (!strcasecmp(argv[0],"no-appendfsync-on-rewrite")
+                   && argc == 2) {
@@ -212,2 +197 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -227 +211,2 @@ void loadServerConfig(char *filename) {
-               argc == 2) {
+                   argc == 2)
+        {
@@ -233 +218,3 @@ void loadServerConfig(char *filename) {
-    } else if (!strcasecmp(argv[0], "auto-aof-rewrite-min-size") && argc == 2) {
+        } else if (!strcasecmp(argv[0],"auto-aof-rewrite-min-size") &&
+                   argc == 2)
+        {
@@ -271,2 +258 @@ void loadServerConfig(char *filename) {
-          err = "Target command name already exists";
-          goto loaderr;
+                    err = "Target command name already exists"; goto loaderr;
@@ -277,2 +263 @@ void loadServerConfig(char *filename) {
-        err = "argument must be 'yes' or 'no'";
-        goto loaderr;
+                err = "argument must be 'yes' or 'no'"; goto loaderr;
@@ -283 +267,0 @@ void loadServerConfig(char *filename) {
-<<<<<<< HEAD
@@ -286,3 +270,3 @@ void loadServerConfig(char *filename) {
-|||||||
-=======
-    } else if (!strcasecmp(argv[0], "slowlog-log-slower-than") && argc == 2) {
+        } else if (!strcasecmp(argv[0],"slowlog-log-slower-than") &&
+                   argc == 2)
+        {
@@ -292 +275,0 @@ void loadServerConfig(char *filename) {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -294,2 +277 @@ void loadServerConfig(char *filename) {
-      err = "Bad directive or wrong number of arguments";
-      goto loaderr;
+            err = "Bad directive or wrong number of arguments"; goto loaderr;
@@ -297 +279,2 @@ void loadServerConfig(char *filename) {
-    for (j = 0; j < argc; j++) sdsfree(argv[j]);
+        for (j = 0; j < argc; j++)
+            sdsfree(argv[j]);
@@ -326 +309,2 @@ void configSetCommand(redisClient *c) {
-    if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll < 0) goto badfmt;
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
+            ll < 0) goto badfmt;
@@ -346 +330,2 @@ void configSetCommand(redisClient *c) {
-    if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll <= 0) goto badfmt;
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
+            ll <= 0) goto badfmt;
@@ -349,2 +334,2 @@ void configSetCommand(redisClient *c) {
-    if (getLongLongFromObject(o, &ll) == REDIS_ERR || ll < 0 || ll > LONG_MAX)
-      goto badfmt;
+        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
+            ll < 0 || ll > LONG_MAX) goto badfmt;
@@ -375 +360,2 @@ void configSetCommand(redisClient *c) {
-          addReplyError(c, "Unable to turn on AOF. Check server logs.");
+                    addReplyError(c,
+                        "Unable to turn on AOF. Check server logs.");
@@ -397 +383,2 @@ void configSetCommand(redisClient *c) {
-      if (eptr[0] != '\0' || ((j & 1) == 0 && val < 1) ||
+            if (eptr[0] != '\0' ||
+                ((j & 1) == 0 && val < 1) ||
@@ -442 +428,0 @@ void configSetCommand(redisClient *c) {
-<<<<<<< HEAD
@@ -446,2 +431,0 @@ void configSetCommand(redisClient *c) {
-|||||||
-=======
@@ -454 +437,0 @@ void configSetCommand(redisClient *c) {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -464 +447,2 @@ badfmt:
-                      (char *)o->ptr, (char *)c->argv[2]->ptr);
+            (char*)o->ptr,
+            (char*)c->argv[2]->ptr);
@@ -507,21 +491,7 @@ void configGetCommand(redisClient *c) {
-      case REDIS_MAXMEMORY_VOLATILE_LRU:
-        s = "volatile-lru";
-        break;
-      case REDIS_MAXMEMORY_VOLATILE_TTL:
-        s = "volatile-ttl";
-        break;
-      case REDIS_MAXMEMORY_VOLATILE_RANDOM:
-        s = "volatile-random";
-        break;
-      case REDIS_MAXMEMORY_ALLKEYS_LRU:
-        s = "allkeys-lru";
-        break;
-      case REDIS_MAXMEMORY_ALLKEYS_RANDOM:
-        s = "allkeys-random";
-        break;
-      case REDIS_MAXMEMORY_NO_EVICTION:
-        s = "noeviction";
-        break;
-      default:
-        s = "unknown";
-        break;
+        case REDIS_MAXMEMORY_VOLATILE_LRU: s = "volatile-lru"; break;
+        case REDIS_MAXMEMORY_VOLATILE_TTL: s = "volatile-ttl"; break;
+        case REDIS_MAXMEMORY_VOLATILE_RANDOM: s = "volatile-random"; break;
+        case REDIS_MAXMEMORY_ALLKEYS_LRU: s = "allkeys-lru"; break;
+        case REDIS_MAXMEMORY_ALLKEYS_RANDOM: s = "allkeys-random"; break;
+        case REDIS_MAXMEMORY_NO_EVICTION: s = "noeviction"; break;
+        default: s = "unknown"; break;
@@ -558,12 +528,4 @@ void configGetCommand(redisClient *c) {
-      case APPENDFSYNC_NO:
-        policy = "no";
-        break;
-      case APPENDFSYNC_EVERYSEC:
-        policy = "everysec";
-        break;
-      case APPENDFSYNC_ALWAYS:
-        policy = "always";
-        break;
-      default:
-        policy = "unknown";
-        break;
+        case APPENDFSYNC_NO: policy = "no"; break;
+        case APPENDFSYNC_EVERYSEC: policy = "everysec"; break;
+        case APPENDFSYNC_ALWAYS: policy = "always"; break;
+        default: policy = "unknown"; break;
@@ -579 +541,2 @@ void configGetCommand(redisClient *c) {
-      buf = sdscatprintf(buf, "%ld %d", server.saveparams[j].seconds,
+            buf = sdscatprintf(buf,"%ld %d",
+                    server.saveparams[j].seconds,
@@ -581 +544,2 @@ void configGetCommand(redisClient *c) {
-      if (j != server.saveparamslen - 1) buf = sdscatlen(buf, " ", 1);
+            if (j != server.saveparamslen-1)
+                buf = sdscatlen(buf," ",1);
@@ -638 +601,0 @@ void configGetCommand(redisClient *c) {
-<<<<<<< HEAD
@@ -642,4 +604,0 @@ void configGetCommand(redisClient *c) {
-    matches++;
-  }
-|||||||
-=======
@@ -656 +614,0 @@ void configGetCommand(redisClient *c) {
->>>>>>> 34a8b51768a2579010e128b05e001bf1d8f99995
@@ -676 +634,2 @@ void configCommand(redisClient *c) {
-    addReplyError(c, "CONFIG subcommand must be one of GET, SET, RESETSTAT");
+        addReplyError(c,
+            "CONFIG subcommand must be one of GET, SET, RESETSTAT");
