--- a/./output/grpc/acbda983-2d73a175-8cd6ee62/src@core@lib@security@credentials@google_default@google_default_credentials.no_comments_mergegen.cc
+++ b/./output/grpc/acbda983-2d73a175-8cd6ee62/src@core@lib@security@credentials@google_default@google_default_credentials.no_comments_merged.cc
@@ -230,2 +230,4 @@ static void default_call_creds(
-  err = create_default_creds_from_path(
-      gpr_getenv(GRPC_GOOGLE_CREDENTIALS_ENV_VAR), call_creds);
+  char* path_from_env = gpr_getenv(GRPC_GOOGLE_CREDENTIALS_ENV_VAR);
+  if (path_from_env != nullptr) {
+    err = create_default_creds_from_path(path_from_env, call_creds);
+    gpr_free(path_from_env);
@@ -233,0 +236 @@ static void default_call_creds(
+  }
