--- a/./output/art/0fb2fde6-86309e74-344f4e7e/runtime@thread.no_comments_mergegen.cc
+++ b/./output/art/0fb2fde6-86309e74-344f4e7e/runtime@thread.no_comments_merged.cc
@@ -1363 +1363,7 @@ bool Thread::RequestSynchronousCheckpoint(Closure* function, ThreadState suspend
-if (gUseUserfaultfd) {
+      DCHECK_EQ(self->GetState(), ThreadState::kRunnable);
+      DCHECK(
+          !self->GetStateAndFlags(std::memory_order_relaxed).IsAnyOfFlagsSet(FlipFunctionFlags()));
+      EnsureFlipFunctionStarted(self);
+      while (GetStateAndFlags(std::memory_order_acquire).IsAnyOfFlagsSet(FlipFunctionFlags())) {
+        sched_yield();
+      }
