diff --git a/./output/tmux/c489bf0a-aa7dccf8-048f1ff1/file.no_comments_conflict.c b/./output/tmux/c489bf0a-aa7dccf8-048f1ff1/file.no_comments_merged.c
index 16f209a..79fc08f 100644
--- a/./output/tmux/c489bf0a-aa7dccf8-048f1ff1/file.no_comments_conflict.c
+++ b/./output/tmux/c489bf0a-aa7dccf8-048f1ff1/file.no_comments_merged.c
@@ -2,7 +1,0 @@
-<<<<<<< HEAD
-||||||| 048f1ff1
-#include <sys/queue.h>
-=======
-#include <sys/queue.h>
-#include <sys/uio.h>
->>>>>>> aa7dccf8
