--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/c489bf0a-aa7dccf8-048f1ff1/file.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/c489bf0a-aa7dccf8-048f1ff1/file.no_comments_merged.c
@@ -2,3 +1,0 @@
-#include <sys/queue.h>
-#include <sys/uio.h>
-#include <imsg.h>
@@ -13 +10,4 @@ static int file_next_stream = 3;
-static char *file_get_path(struct client *c, const char *file) {
+RB_GENERATE(client_files, client_file, entry, file_cmp);
+static char *
+file_get_path(struct client *c, const char *file)
+{
