diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_merged.cpp
index 9f42e3c..764bfa8 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_merged.cpp
@@ -20,0 +21 @@ using LockResult = GraphicBufferMapper::LockResult;
+ANDROID_SINGLETON_STATIC_INSTANCE( GraphicBufferMapper )
@@ -84,2 +85,3 @@ status_t GraphicBufferMapper::importBufferNoValidate(const native_handle_t* rawH
-void GraphicBufferMapper::getTransportSize(buffer_handle_t handle, uint32_t* outTransportNumFds,
-                                           uint32_t* outTransportNumInts) {
+void GraphicBufferMapper::getTransportSize(buffer_handle_t handle,
+            uint32_t* outTransportNumFds, uint32_t* outTransportNumInts)
+{
@@ -88 +90,2 @@ void GraphicBufferMapper::getTransportSize(buffer_handle_t handle, uint32_t* out
-status_t GraphicBufferMapper::freeBuffer(buffer_handle_t handle) {
+status_t GraphicBufferMapper::freeBuffer(buffer_handle_t handle)
+{
@@ -154 +156,0 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
-<<<<<<< HEAD
@@ -157,13 +158,0 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
-|||||||
-    ATRACE_CALL();
-    const uint64_t usage =
-            static_cast<uint64_t>(android_convertGralloc1To0Usage(producerUsage, consumerUsage));
-    return mMapper->lock(handle, usage, bounds, fenceFd, vaddr, outBytesPerPixel,
-                         outBytesPerStride);
-=======
-    ATRACE_CALL();
-    const uint64_t usage = static_cast<uint64_t>(ANDROID_NATIVE_UNSIGNED_CAST(
-            android_convertGralloc1To0Usage(producerUsage, consumerUsage)));
-    return mMapper->lock(handle, usage, bounds, fenceFd, vaddr, outBytesPerPixel,
-                         outBytesPerStride);
->>>>>>> 7f1bf8c6158a565d200edf59bb546739e390af44
@@ -306 +295 @@ status_t GraphicBufferMapper::setSmpte2094_10(buffer_handle_t bufferHandle,
-}
+};
