--- a/./output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/c1698103-7f1bf8c6-3fcfeace/libs@ui@GraphicBufferMapper.no_comments_merged.cpp
@@ -156 +156,3 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
-return lockAsync(handle, android_convertGralloc1To0Usage(producerUsage, consumerUsage), bounds, vaddr, fenceFd);
+                                        int fenceFd) {
+    return lockAsync(handle, android_convertGralloc1To0Usage(producerUsage, consumerUsage), bounds,
+                     vaddr, fenceFd);
