diff --git a/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_mergegen.h b/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_merged.h
index efd1110..004825b 100644
--- a/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_mergegen.h
+++ b/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_merged.h
@@ -16 +16,10 @@ private slots:
-void showHelpChanged(bool checked); void showDesktopNotificationChanged(bool checked); void showTrayIconChanged(bool checked); void autostartChanged(bool checked); void closeAfterCaptureChanged(bool checked); void saveAfterCopyChanged(bool checked); void changeSavePath(); void importConfiguration(); void exportFileConfiguration(); void resetConfiguration(); void showDesktopNotificationChanged(bool checked); void showTrayIconChanged(bool checked); void autostartChanged(bool checked); void closeAfterCaptureChanged(bool checked); void saveAfterCopyChanged(bool checked); void changeSavePath(); void importConfiguration(); void exportFileConfiguration(); void resetConfiguration(); void showDesktopNotificationChanged(bool checked); void showTrayIconChanged(bool checked);
+  void showHelpChanged(bool checked);
+  void showDesktopNotificationChanged(bool checked);
+  void showTrayIconChanged(bool checked);
+  void autostartChanged(bool checked);
+  void closeAfterCaptureChanged(bool checked);
+  void saveAfterCopyChanged(bool checked);
+  void changeSavePath();
+  void importConfiguration();
+  void exportFileConfiguration();
+  void resetConfiguration();
@@ -18,2 +27,21 @@ private:
-QVBoxLayout* m_layout; QCheckBox* m_sysNotifications; QCheckBox* m_showTray; QCheckBox* m_helpMessage; QCheckBox* m_autostart; QCheckBox* m_closeAfterCapture; QCheckBox* m_copyAndCloseAfterUpload; QPushButton* m_importButton; QPushButton* m_exportButton; QPushButton* m_resetButton; QCheckBox *m_saveAfterCopy; QLineEdit *m_savePath; QPushButton *m_changeSaveButton;
-void initShowHelp(); void initShowDesktopNotification(); void initShowTrayIcon(); void initConfingButtons(); void initAutostart(); void initCloseAfterCapture(); void initCopyAndCloseAfterUpload(); void initSaveAfterCopy();
+  QVBoxLayout* m_layout;
+  QCheckBox* m_sysNotifications;
+  QCheckBox* m_showTray;
+  QCheckBox* m_helpMessage;
+  QCheckBox* m_autostart;
+  QCheckBox* m_closeAfterCapture;
+  QCheckBox* m_copyAndCloseAfterUpload;
+  QPushButton* m_importButton;
+  QPushButton* m_exportButton;
+  QPushButton* m_resetButton;
+  QCheckBox* m_saveAfterCopy;
+  QLineEdit* m_savePath;
+  QPushButton* m_changeSaveButton;
+  void initShowHelp();
+  void initShowDesktopNotification();
+  void initShowTrayIcon();
+  void initConfingButtons();
+  void initAutostart();
+  void initCloseAfterCapture();
+  void initCopyAndCloseAfterUpload();
+  void initSaveAfterCopy();
