diff --git a/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@screenshotsaver.no_comments_mergegen.cpp b/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@screenshotsaver.no_comments_merged.cpp
index d8e528e..2386a17 100644
--- a/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@screenshotsaver.no_comments_mergegen.cpp
+++ b/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@screenshotsaver.no_comments_merged.cpp
@@ -11 +11,2 @@ ScreenshotSaver::ScreenshotSaver() {}
-void ScreenshotSaver::saveToClipboard(const QPixmap& capture)
+void
+ScreenshotSaver::saveToClipboard(const QPixmap& capture)
@@ -12,0 +14,5 @@ void ScreenshotSaver::saveToClipboard(const QPixmap& capture)
+  if (ConfigHandler().saveAfterCopyValue()) {
+    if (!ConfigHandler().saveAfterCopyPathValue().isEmpty()) {
+      saveToFilesystem(capture, ConfigHandler().saveAfterCopyPathValue());
+    }
+  } else {
@@ -13,0 +20 @@ void ScreenshotSaver::saveToClipboard(const QPixmap& capture)
+  }
