diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_merged.h
index 701784e..004825b 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_merged.h
@@ -3 +3 @@
-class QVBoxLayout
+class QVBoxLayout;
@@ -13 +13 @@ public:
-public:
+public slots:
@@ -15 +15 @@ public:
-private:
+private slots:
@@ -25,0 +26 @@ private:
+private:
