--- a/./output/frameworks_av/793c8f6c-9cd52ba9-f17bebe8/services@audiopolicy@service@Spatializer.no_comments_conflict.h
+++ b/./output/frameworks_av/793c8f6c-9cd52ba9-f17bebe8/services@audiopolicy@service@Spatializer.no_comments_merged.h
@@ -203 +202,0 @@ private:
-<<<<<<< HEAD
@@ -206,8 +204,0 @@ private:
-||||||| f17bebe88c
-    HeadToStagePoseRecorder mPoseRecorder GUARDED_BY(mLock) =
-            HeadToStagePoseRecorder(std::chrono::seconds(1), mMaxLocalLogLine);
-=======
-    media::VectorRecorder mPoseRecorder GUARDED_BY(mLock) {
-        6 , std::chrono::seconds(1), mMaxLocalLogLine };
->>>>>>> 9cd52ba9
-<<<<<<< HEAD
@@ -216,7 +206,0 @@ private:
-||||||| f17bebe88c
-    HeadToStagePoseRecorder mPoseDurableRecorder GUARDED_BY(mLock) =
-            HeadToStagePoseRecorder(std::chrono::minutes(1), mMaxLocalLogLine);
-=======
-    media::VectorRecorder mPoseDurableRecorder GUARDED_BY(mLock) {
-        6 , std::chrono::minutes(1), mMaxLocalLogLine };
->>>>>>> 9cd52ba9
