diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_merged.cpp
index 37ac9a6..a29c204 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_merged.cpp
@@ -35 +35,4 @@ MemoryHeapBase::MemoryHeapBase()
-      : mFD(-1), mSize(0), mBase(MAP_FAILED), mDevice(nullptr), mNeedUnmap(false), mOffset(0) {}
+    : mFD(-1), mSize(0), mBase(MAP_FAILED),
+      mDevice(nullptr), mNeedUnmap(false), mOffset(0)
+{
+}
@@ -37,7 +40,3 @@ MemoryHeapBase::MemoryHeapBase(size_t size, uint32_t flags, char const* name)
-      : mFD(-1),
-        mSize(0),
-        mBase(MAP_FAILED),
-        mFlags(flags),
-        mDevice(nullptr),
-        mNeedUnmap(false),
-        mOffset(0) {
+    : mFD(-1), mSize(0), mBase(MAP_FAILED), mFlags(flags),
+      mDevice(nullptr), mNeedUnmap(false), mOffset(0)
+{
@@ -52 +50,0 @@ MemoryHeapBase::MemoryHeapBase(size_t size, uint32_t flags, char const* name)
-<<<<<<< HEAD
@@ -55,7 +52,0 @@ MemoryHeapBase::MemoryHeapBase(size_t size, uint32_t flags, char const* name)
-|||||||
-        const int SEAL_FLAGS = ((mFlags & READ_ONLY) ? F_SEAL_FUTURE_WRITE : 0) |
-                ((mFlags & MEMFD_ALLOW_SEALING) ? 0 : F_SEAL_SEAL);
-=======
-        const int SEAL_FLAGS = ((mFlags & READ_ONLY) ? F_SEAL_FUTURE_WRITE : 0) | F_SEAL_GROW |
-                F_SEAL_SHRINK | ((mFlags & MEMFD_ALLOW_SEALING) ? 0 : F_SEAL_SEAL);
->>>>>>> 99785d0473407e38e18084e2bbf040d12e57beee
@@ -83,7 +74,3 @@ MemoryHeapBase::MemoryHeapBase(const char* device, size_t size, uint32_t flags)
-      : mFD(-1),
-        mSize(0),
-        mBase(MAP_FAILED),
-        mFlags(flags),
-        mDevice(nullptr),
-        mNeedUnmap(false),
-        mOffset(0) {
+    : mFD(-1), mSize(0), mBase(MAP_FAILED), mFlags(flags),
+      mDevice(nullptr), mNeedUnmap(false), mOffset(0)
+{
@@ -94 +81,2 @@ MemoryHeapBase::MemoryHeapBase(const char* device, size_t size, uint32_t flags)
-    if (flags & NO_CACHING) open_flags |= O_SYNC;
+    if (flags & NO_CACHING)
+        open_flags |= O_SYNC;
@@ -106,7 +94,3 @@ MemoryHeapBase::MemoryHeapBase(int fd, size_t size, uint32_t flags, off_t offset
-      : mFD(-1),
-        mSize(0),
-        mBase(MAP_FAILED),
-        mFlags(flags),
-        mDevice(nullptr),
-        mNeedUnmap(false),
-        mOffset(0) {
+    : mFD(-1), mSize(0), mBase(MAP_FAILED), mFlags(flags),
+      mDevice(nullptr), mNeedUnmap(false), mOffset(0)
+{
@@ -120 +104,2 @@ MemoryHeapBase::MemoryHeapBase(int fd, size_t size, uint32_t flags, off_t offset
-status_t MemoryHeapBase::init(int fd, void* base, size_t size, int flags, const char* device) {
+status_t MemoryHeapBase::init(int fd, void *base, size_t size, int flags, const char* device)
+{
@@ -131 +116,2 @@ status_t MemoryHeapBase::init(int fd, void* base, size_t size, int flags, const
-status_t MemoryHeapBase::mapfd(int fd, bool writeableByCaller, size_t size, off_t offset) {
+status_t MemoryHeapBase::mapfd(int fd, bool writeableByCaller, size_t size, off_t offset)
+{
@@ -137,2 +123,2 @@ status_t MemoryHeapBase::mapfd(int fd, bool writeableByCaller, size_t size, off_
-                ALOGE("%s: size of file %lld cannot fit in memory", __func__,
-                      (long long)sb.st_size);
+                ALOGE("%s: size of file %lld cannot fit in memory",
+                        __func__, (long long)sb.st_size);
@@ -148 +134,2 @@ status_t MemoryHeapBase::mapfd(int fd, bool writeableByCaller, size_t size, off_
-        void* base = (uint8_t*)mmap(nullptr, size, prot, MAP_SHARED, fd, offset);
+        void* base = (uint8_t*)mmap(nullptr, size,
+                prot, MAP_SHARED, fd, offset);
@@ -150 +137,2 @@ status_t MemoryHeapBase::mapfd(int fd, bool writeableByCaller, size_t size, off_
-            ALOGE("mmap(fd=%d, size=%zu) failed (%s)", fd, size, strerror(errno));
+            ALOGE("mmap(fd=%d, size=%zu) failed (%s)",
+                    fd, size, strerror(errno));
@@ -165 +153,2 @@ status_t MemoryHeapBase::mapfd(int fd, bool writeableByCaller, size_t size, off_
-MemoryHeapBase::~MemoryHeapBase() {
+MemoryHeapBase::~MemoryHeapBase()
+{
@@ -168 +157,2 @@ MemoryHeapBase::~MemoryHeapBase() {
-void MemoryHeapBase::dispose() {
+void MemoryHeapBase::dispose()
+{
