diff --git a/./output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_mergegen.cpp b/./output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_merged.cpp
index ee40466..a29c204 100644
--- a/./output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/ec8148a0-99785d04-35388a2d/libs@binder@MemoryHeapBase.no_comments_merged.cpp
@@ -51 +51,2 @@ MemoryHeapBase::MemoryHeapBase(size_t size, uint32_t flags, char const * name)
-const int SEAL_FLAGS = ((mFlags & READ_ONLY) ? F_SEAL_FUTURE_WRITE : 0) | F_SEAL_GROW | F_SEAL_SHRINK | ((mFlags & MEMFD_ALLOW_SEALING_FLAG) ? 0 : F_SEAL_SEAL);
+        const int SEAL_FLAGS = ((mFlags & READ_ONLY) ? F_SEAL_FUTURE_WRITE : 0) | F_SEAL_GROW |
+                F_SEAL_SHRINK | ((mFlags & MEMFD_ALLOW_SEALING_FLAG) ? 0 : F_SEAL_SEAL);
