--- a/./output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_conflict.h
+++ b/./output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_merged.h
@@ -52 +51,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-<<<<<<< HEAD
@@ -54,5 +52,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-||||||| 7be8e6a84b
-  std::shared_ptr<Channel> InProcessChannel(const ChannelArguments& args);
-=======
-  std::shared_ptr<Channel> InProcessChannel(const grpc::ChannelArguments& args);
->>>>>>> c8b95f97
@@ -62 +55,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-<<<<<<< HEAD
@@ -65,7 +57,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-||||||| 7be8e6a84b
-    std::shared_ptr<Channel> InProcessChannelWithInterceptors(
-        const ChannelArguments& args,
-=======
-    std::shared_ptr<Channel> InProcessChannelWithInterceptors(
-        const grpc::ChannelArguments& args,
->>>>>>> c8b95f97
@@ -84 +69,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-<<<<<<< HEAD
@@ -98,27 +82,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-||||||| 7be8e6a84b
-  Server(ChannelArguments* args,
-         std::shared_ptr<std::vector<std::unique_ptr<ServerCompletionQueue>>>
-             sync_server_cqs,
-         int min_pollers, int max_pollers, int sync_cq_timeout_msec,
-         std::vector<
-             std::shared_ptr<grpc::internal::ExternalConnectionAcceptorImpl>>
-             acceptors,
-         grpc_resource_quota* server_rq = nullptr,
-         std::vector<std::unique_ptr<
-             grpc::experimental::ServerInterceptorFactoryInterface>>
-             interceptor_creators = std::vector<std::unique_ptr<
-                 grpc::experimental::ServerInterceptorFactoryInterface>>());
-=======
-  Server(grpc::ChannelArguments* args,
-         std::shared_ptr<std::vector<std::unique_ptr<ServerCompletionQueue>>>
-             sync_server_cqs,
-         int min_pollers, int max_pollers, int sync_cq_timeout_msec,
-         std::vector<
-             std::shared_ptr<grpc::internal::ExternalConnectionAcceptorImpl>>
-             acceptors,
-         grpc_resource_quota* server_rq = nullptr,
-         std::vector<std::unique_ptr<
-             grpc::experimental::ServerInterceptorFactoryInterface>>
-             interceptor_creators = std::vector<std::unique_ptr<
-                 grpc::experimental::ServerInterceptorFactoryInterface>>());
->>>>>>> c8b95f97
