diff --git a/./output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_mergegen.h b/./output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_merged.h
index ed4d6a5..9e27378 100644
--- a/./output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_mergegen.h
+++ b/./output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_merged.h
@@ -56 +56,2 @@ std::shared_ptr<grpc::Channel> InProcessChannel(const ChannelArguments& args);
-std::shared_ptr<grpc::Channel> InProcessChannelWithInterceptors( const ChannelArguments& args,
+    std::shared_ptr<grpc::Channel> InProcessChannelWithInterceptors(
+        const ChannelArguments& args,
@@ -69 +70,13 @@ std::shared_ptr<grpc::Channel> InProcessChannelWithInterceptors( const ChannelAr
-Server( ChannelArguments* args, std::shared_ptr<std::vector<std::unique_ptr<grpc::ServerCompletionQueue>>> sync_server_cqs, int min_pollers, int max_pollers, int sync_cq_timeout_msec, std::vector< std::shared_ptr<grpc::internal::ExternalConnectionAcceptorImpl>> acceptors, grpc_resource_quota* server_rq = nullptr, std::vector<std::unique_ptr< grpc::experimental::ServerInterceptorFactoryInterface>> interceptor_creators = std::vector<std::unique_ptr< grpc::experimental::ServerInterceptorFactoryInterface>>());
+  Server(
+      ChannelArguments* args,
+      std::shared_ptr<std::vector<std::unique_ptr<grpc::ServerCompletionQueue>>>
+          sync_server_cqs,
+      int min_pollers, int max_pollers, int sync_cq_timeout_msec,
+      std::vector<
+          std::shared_ptr<grpc::internal::ExternalConnectionAcceptorImpl>>
+          acceptors,
+      grpc_resource_quota* server_rq = nullptr,
+      std::vector<std::unique_ptr<
+          grpc::experimental::ServerInterceptorFactoryInterface>>
+          interceptor_creators = std::vector<std::unique_ptr<
+              grpc::experimental::ServerInterceptorFactoryInterface>>());
