--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/ab775de1-c8b95f97-7be8e6a8/include@grpcpp@server_impl.no_comments_merged.h
@@ -9,2 +8,0 @@
-#include <grpcpp/channel_impl.h>
-#include <grpcpp/completion_queue_impl.h>
@@ -12 +9,0 @@
-#include <grpcpp/impl/codegen/completion_queue_impl.h>
@@ -16 +12,0 @@
-#include <grpcpp/support/channel_arguments.h>
@@ -23 +19 @@
-#include <grpcpp/support/channel_arguments_impl.h>
+#include <grpcpp/support/channel_arguments.h>
@@ -26 +22 @@
-struct grpc_server
+struct grpc_server;
@@ -56 +52 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-  std::shared_ptr<Channel> InProcessChannel(const grpc::ChannelArguments& args);
+  std::shared_ptr<grpc::Channel> InProcessChannel(const ChannelArguments& args);
@@ -60 +55,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-<<<<<<< HEAD
@@ -63,7 +57,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-|||||||
-    std::shared_ptr<Channel> InProcessChannelWithInterceptors(
-        const ChannelArguments& args,
-=======
-    std::shared_ptr<Channel> InProcessChannelWithInterceptors(
-        const grpc::ChannelArguments& args,
->>>>>>> c8b95f97daf8c2649de8d7bf85edf89566a35b9f
@@ -82 +69,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-<<<<<<< HEAD
@@ -86,7 +72,0 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-|||||||
-  Server(ChannelArguments* args,
-         std::shared_ptr<std::vector<std::unique_ptr<ServerCompletionQueue>>>
-=======
-  Server(grpc::ChannelArguments* args,
-         std::shared_ptr<std::vector<std::unique_ptr<ServerCompletionQueue>>>
->>>>>>> c8b95f97daf8c2649de8d7bf85edf89566a35b9f
@@ -104,0 +85 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
+ protected:
@@ -113,2 +94,3 @@ class Server : public grpc::ServerInterface, private grpc::GrpcLibraryCodegen {
-  std::unique_ptr < grpc::experimental::ServerInterceptorFactoryInterface >>
-      *interceptor_creators() override {
+  std::vector<
+      std::unique_ptr<grpc::experimental::ServerInterceptorFactoryInterface>>*
+  interceptor_creators() override {
