--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_merged.c
@@ -6,2 +5,0 @@
-#define REDISMODULE_CORE 1
-#include "redismodule.h"
@@ -79,7 +76 @@ typedef struct RedisModuleCtx RedisModuleCtx;
-#define REDISMODULE_CTX_INIT \
-  { \
-    (void *)(unsigned long)&RM_GetApi, NULL, NULL, NULL, NULL, 0, 0, 0, NULL, \
-        0, NULL, NULL, NULL, 0, NULL, { \
-      0 \
-    } \
-  }
+#define REDISMODULE_CTX_INIT {(void*)(unsigned long)&RM_GetApi, NULL, NULL, NULL, NULL, 0, 0, 0, NULL, 0, NULL, NULL, NULL, 0, NULL, {0}}
@@ -126,2 +116 @@ typedef struct RedisModuleCommandProxy RedisModuleCommandProxy;
-#define REDISMODULE_REPLYFLAG_NESTED \
-  (1 << 1)
+#define REDISMODULE_REPLYFLAG_NESTED (1<<1)
@@ -199,0 +184,2 @@ typedef struct RedisModuleServerInfoData {
+#define SHOULD_SIGNAL_MODIFIED_KEYS(ctx) \
+    ctx->module? !(ctx->module->options & REDISMODULE_OPTION_NO_IMPLICIT_SIGNAL_MODIFIED) : 1
@@ -3574,0 +3496,2 @@ int moduleRegisterApi(const char *funcname, void *funcptr) {
+#define REGISTER_API(name) \
+    moduleRegisterApi("RedisModule_" #name, (void *)(unsigned long)RM_ ## name)
