diff --git a/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_conflict.c b/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_merged.c
index db53f14..efcedff 100644
--- a/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_conflict.c
+++ b/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_merged.c
@@ -5 +4,0 @@
-<<<<<<< HEAD
@@ -7,6 +5,0 @@
-#define REDISMODULE_CORE 1
-#include "redismodule.h"
-||||||| 5f450e492
-#define REDISMODULE_CORE 1
-#include "redismodule.h"
-=======
@@ -14 +6,0 @@
->>>>>>> c6fb9d09
