--- a/./output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_mergegen.c
+++ b/./output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_merged.c
@@ -1790 +1790,4 @@ server_client_dispatch_identify(struct client *c, struct imsg *imsg)
-#ifdef __CYGWIN__ c->fd = open(c->ttyname, O_RDWR|O_NOCTTY); #endif
+#ifdef __CYGWIN__
+ c->fd = open(c->ttyname, O_RDWR|O_NOCTTY);
+#endif
+ if (c->flags & CLIENT_CONTROL)
