--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_merged.c
@@ -7,2 +6,0 @@
-#include <imsg.h>
-#include <paths.h>
@@ -40 +45,4 @@ static int server_client_window_cmp(struct client_window *cw1,
-u_int server_client_how_many(void) {
+RB_GENERATE(client_windows, client_window, entry, server_client_window_cmp);
+u_int
+server_client_how_many(void)
+{
