--- a/./output/duckdb/484ac52a-7bdaddf8-30d444e4/tools@pythonpkg@src@pandas@scan.no_comments_conflict.cpp
+++ b/./output/duckdb/484ac52a-7bdaddf8-30d444e4/tools@pythonpkg@src@pandas@scan.no_comments_merged.cpp
@@ -6,0 +7 @@
+#include "duckdb_python/numpy/numpy_bind.hpp"
@@ -65,7 +65,0 @@ unique_ptr<FunctionData> PandasScanFunction::PandasScanBind(ClientContext &conte
-<<<<<<< HEAD
- Pandas::Bind(context, df, pandas_bind_data, return_types, names);
- auto df_columns = py::list(df.attr("columns"));
-||||||| 30d444e45d
- VectorConversion::BindPandas(DBConfig::GetConfig(context), df, pandas_bind_data, return_types, names);
- auto df_columns = py::list(df.attr("columns"));
-=======
@@ -74 +68 @@ unique_ptr<FunctionData> PandasScanFunction::PandasScanBind(ClientContext &conte
-  VectorConversion::BindNumpy(DBConfig::GetConfig(context), df, pandas_bind_data, return_types, names);
+  NumpyBind::Bind(context, df, pandas_bind_data, return_types, names);
@@ -76 +70 @@ unique_ptr<FunctionData> PandasScanFunction::PandasScanBind(ClientContext &conte
-  VectorConversion::BindPandas(DBConfig::GetConfig(context), df, pandas_bind_data, return_types, names);
+  Pandas::Bind(context, df, pandas_bind_data, return_types, names);
@@ -79 +72,0 @@ unique_ptr<FunctionData> PandasScanFunction::PandasScanBind(ClientContext &conte
->>>>>>> 7bdaddf8
@@ -139 +132 @@ void PandasScanFunction::PandasBackendScanSwitch(PandasColumnBindData &bind_data
-  Numpy::Scan(bind_data, count, offset, out);
+  NumpyScan::Scan(bind_data, count, offset, out);
