diff --git a/./output/duckdb/484ac52a-7bdaddf8-30d444e4/tools@pythonpkg@src@pandas@scan.no_comments_mergegen.cpp b/./output/duckdb/484ac52a-7bdaddf8-30d444e4/tools@pythonpkg@src@pandas@scan.no_comments_merged.cpp
index f6ac9aa..7171b79 100644
--- a/./output/duckdb/484ac52a-7bdaddf8-30d444e4/tools@pythonpkg@src@pandas@scan.no_comments_mergegen.cpp
+++ b/./output/duckdb/484ac52a-7bdaddf8-30d444e4/tools@pythonpkg@src@pandas@scan.no_comments_merged.cpp
@@ -6,0 +7 @@
+#include "duckdb_python/numpy/numpy_bind.hpp"
@@ -65 +66,7 @@ unique_ptr<FunctionData> PandasScanFunction::PandasScanBind(ClientContext &conte
-Pandas::Bind(context, df, pandas_bind_data, return_types, names); } auto df_columns = py::list(df.attr("keys")());
+ auto is_py_dict = py::isinstance<py::dict>(df);
+ if (is_py_dict) {
+  NumpyBind::Bind(context, df, pandas_bind_data, return_types, names);
+ } else {
+  Pandas::Bind(context, df, pandas_bind_data, return_types, names);
+ }
+ auto df_columns = py::list(df.attr("keys")());
@@ -125 +132 @@ void PandasScanFunction::PandasBackendScanSwitch(PandasColumnBindData &bind_data
-  Numpy::Scan(bind_data, count, offset, out);
+  NumpyScan::Scan(bind_data, count, offset, out);
