diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/e0c3d791-d7227f1c-10d0f99e/media@libmediaplayerservice@nuplayer@NuPlayerDriver.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/e0c3d791-d7227f1c-10d0f99e/media@libmediaplayerservice@nuplayer@NuPlayerDriver.no_comments_merged.h
index a0f79c3..4888fa4 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/e0c3d791-d7227f1c-10d0f99e/media@libmediaplayerservice@nuplayer@NuPlayerDriver.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/e0c3d791-d7227f1c-10d0f99e/media@libmediaplayerservice@nuplayer@NuPlayerDriver.no_comments_merged.h
@@ -2 +1,0 @@
-#include <media/MediaAnalyticsItem.h>
@@ -13 +12,2 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-  virtual status_t setDataSource(const sp<IMediaHTTPService> &httpService,
+    virtual status_t setDataSource(
+            const sp<IMediaHTTPService> &httpService,
@@ -23,2 +23 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-  virtual status_t setBufferingSettings(
-      const BufferingSettings &buffering) override;
+    virtual status_t setBufferingSettings(const BufferingSettings& buffering) override;
@@ -33,2 +32 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-  virtual status_t setSyncSettings(const AVSyncSettings &sync,
-                                   float videoFpsHint);
+    virtual status_t setSyncSettings(const AVSyncSettings &sync, float videoFpsHint);
@@ -37,2 +35 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-      int msec,
-      MediaPlayerSeekMode mode = MediaPlayerSeekMode::SEEK_PREVIOUS_SYNC);
+            int msec, MediaPlayerSeekMode mode = MediaPlayerSeekMode::SEEK_PREVIOUS_SYNC);
@@ -49,2 +46,2 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-  virtual status_t getMetadata(const media::Metadata::Filter &ids,
-                               Parcel *records);
+    virtual status_t getMetadata(
+            const media::Metadata::Filter& ids, Parcel *records);
@@ -62,2 +59 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-  void notifyListener(int msg, int ext1 = 0, int ext2 = 0,
-                      const Parcel *in = NULL);
+    void notifyListener(int msg, int ext1 = 0, int ext2 = 0, const Parcel *in = NULL);
@@ -65,2 +61 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-  virtual status_t prepareDrm(const uint8_t uuid[16],
-                              const Vector<uint8_t> &drmSessionId);
+    virtual status_t prepareDrm(const uint8_t uuid[16], const Vector<uint8_t> &drmSessionId);
@@ -113,2 +108 @@ struct NuPlayerDriver : public MediaPlayerInterface {
-  void notifyListener_l(int msg, int ext1 = 0, int ext2 = 0,
-                        const Parcel *in = NULL);
+    void notifyListener_l(int msg, int ext1 = 0, int ext2 = 0, const Parcel *in = NULL);
