--- a/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/a7de6785-b50e3d9d-dd7513f9/JNIHelp.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/libnativehelper/a7de6785-b50e3d9d-dd7513f9/JNIHelp.no_comments_merged.cpp
@@ -7 +6,0 @@
-#include <string>
@@ -9,0 +9 @@
+#include <string>
@@ -231,12 +219,2 @@ jobject jniCreateFileDescriptor(C_JNIEnv* env, int fd) {
-<<<<<<< /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/ours/JNIHelp.cpp
-  jobject fileDescriptor = (*env)->NewObject(e, gFileDescriptorClassInfo.clazz,
-                                             gFileDescriptorClassInfo.ctor);
-||||||| /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/base/JNIHelp.cpp
-  jobject fileDescriptor = (*env)->NewObject(
-      e, gCachedFields.fileDescriptorClass, gCachedFields.fileDescriptorCtor);
-=======
-  static jmethodID ctor =
-      e->GetMethodID(JniConstants::fileDescriptorClass, "<init>", "()V");
-  jobject fileDescriptor =
-      (*env)->NewObject(e, JniConstants::fileDescriptorClass, ctor);
->>>>>>> /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/theirs/JNIHelp.cpp
+    static jmethodID ctor = e->GetMethodID(JniConstants::fileDescriptorClass, "<init>", "()V");
+    jobject fileDescriptor = (*env)->NewObject(e, JniConstants::fileDescriptorClass, ctor);
@@ -248,8 +226 @@ int jniGetFDFromFileDescriptor(C_JNIEnv* env, jobject fileDescriptor) {
-<<<<<<< /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/ours/JNIHelp.cpp
-  return (*env)->GetIntField(e, fileDescriptor,
-                             gFileDescriptorClassInfo.descriptor);
-||||||| /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/base/JNIHelp.cpp
-  return (*env)->GetIntField(e, fileDescriptor, gCachedFields.descriptorField);
-=======
-  static jfieldID fid =
-      e->GetFieldID(JniConstants::fileDescriptorClass, "descriptor", "I");
+    static jfieldID fid = e->GetFieldID(JniConstants::fileDescriptorClass, "descriptor", "I");
@@ -257 +227,0 @@ int jniGetFDFromFileDescriptor(C_JNIEnv* env, jobject fileDescriptor) {
->>>>>>> /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/theirs/JNIHelp.cpp
@@ -262,8 +231 @@ void jniSetFileDescriptorOfFD(C_JNIEnv* env, jobject fileDescriptor,
-<<<<<<< /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/ours/JNIHelp.cpp
-  (*env)->SetIntField(e, fileDescriptor, gFileDescriptorClassInfo.descriptor,
-                      value);
-||||||| /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/base/JNIHelp.cpp
-  (*env)->SetIntField(e, fileDescriptor, gCachedFields.descriptorField, value);
-=======
-  static jfieldID fid =
-      e->GetFieldID(JniConstants::fileDescriptorClass, "descriptor", "I");
+    static jfieldID fid = e->GetFieldID(JniConstants::fileDescriptorClass, "descriptor", "I");
@@ -271,5 +232,0 @@ void jniSetFileDescriptorOfFD(C_JNIEnv* env, jobject fileDescriptor,
->>>>>>> /tmp/.mergebot/47060b8d762c346617290efed906fb91073cf64c/a7de67-b50e3d/theirs/JNIHelp.cpp
-}
-jobject jniGetReferent(C_JNIEnv* env, jobject ref) {
-  JNIEnv* e = reinterpret_cast<JNIEnv*>(env);
-  return (*env)->CallObjectMethod(e, ref, gReferenceClassInfo.get);
@@ -289,8 +242,0 @@ extern "C" jbyte* jniGetNonMovableArrayElements(C_JNIEnv* env,
-static struct {
-  jclass clazz;
-  jmethodID ctor;
-  jfieldID descriptor;
-} gFileDescriptorClassInfo;
-static struct {
-  jmethodID get;
-} gReferenceClassInfo;
