--- a/./output/libnativehelper/a7de6785-b50e3d9d-dd7513f9/JNIHelp.no_comments_conflict.cpp
+++ b/./output/libnativehelper/a7de6785-b50e3d9d-dd7513f9/JNIHelp.no_comments_merged.cpp
@@ -217,45 +216,0 @@ const char* jniStrError(int errnum, char* buf, size_t buflen) {
-<<<<<<< HEAD
-static struct {
-    jclass clazz;
-    jmethodID ctor;
-    jfieldID descriptor;
-} gFileDescriptorClassInfo;
-static struct {
-    jmethodID get;
-} gReferenceClassInfo;
-jint JNI_OnLoad(JavaVM* vm, void*) {
-    JNIEnv* env;
-    if (vm->GetEnv(reinterpret_cast<void**>(&env), JNI_VERSION_1_6) != JNI_OK) {
-        ALOGE("JavaVM::GetEnv() failed");
-        abort();
-    }
-    gFileDescriptorClassInfo.clazz =
-            reinterpret_cast<jclass>(env->NewGlobalRef(env->FindClass("java/io/FileDescriptor")));
-    if (gFileDescriptorClassInfo.clazz == NULL) {
-        abort();
-    }
-    gFileDescriptorClassInfo.ctor =
-            env->GetMethodID(gFileDescriptorClassInfo.clazz, "<init>", "()V");
-    if (gFileDescriptorClassInfo.ctor == NULL) {
-        abort();
-    }
-    gFileDescriptorClassInfo.descriptor =
-            env->GetFieldID(gFileDescriptorClassInfo.clazz, "descriptor", "I");
-    if (gFileDescriptorClassInfo.descriptor == NULL) {
-        abort();
-    }
-    jclass clazz = reinterpret_cast<jclass>(env->FindClass("java/lang/ref/Reference"));
-    if (clazz == NULL) {
-        abort();
-    }
-    gReferenceClassInfo.get = env->GetMethodID(clazz, "get", "()Ljava/lang/Object;");
-    if (gReferenceClassInfo.get == NULL) {
-        abort();
-    }
-    return JNI_VERSION_1_6;
-}
-jobject jniCreateFileDescriptor(C_JNIEnv* env, int fd) {
-    JNIEnv* e = reinterpret_cast<JNIEnv*>(env);
-    jobject fileDescriptor = (*env)->NewObject(e,
-            gFileDescriptorClassInfo.clazz, gFileDescriptorClassInfo.ctor);
-=======
@@ -266 +220,0 @@ jobject jniCreateFileDescriptor(C_JNIEnv* env, int fd) {
->>>>>>> b50e3d9d
@@ -272,3 +225,0 @@ int jniGetFDFromFileDescriptor(C_JNIEnv* env, jobject fileDescriptor) {
-<<<<<<< HEAD
-    return (*env)->GetIntField(e, fileDescriptor, gFileDescriptorClassInfo.descriptor);
-=======
@@ -277 +227,0 @@ int jniGetFDFromFileDescriptor(C_JNIEnv* env, jobject fileDescriptor) {
->>>>>>> b50e3d9d
@@ -281,7 +230,0 @@ void jniSetFileDescriptorOfFD(C_JNIEnv* env, jobject fileDescriptor, int value)
-<<<<<<< HEAD
-    (*env)->SetIntField(e, fileDescriptor, gFileDescriptorClassInfo.descriptor, value);
-}
-jobject jniGetReferent(C_JNIEnv* env, jobject ref) {
-    JNIEnv* e = reinterpret_cast<JNIEnv*>(env);
-    return (*env)->CallObjectMethod(e, ref, gReferenceClassInfo.get);
-=======
@@ -290 +232,0 @@ jobject jniGetReferent(C_JNIEnv* env, jobject ref) {
->>>>>>> b50e3d9d
