--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_merged.cpp
@@ -5,0 +6 @@
+#include "duckdb/common/http_stats.hpp"
@@ -8 +8,0 @@
-#include "duckdb/common/progress_bar.hpp"
@@ -16,0 +17 @@
+#include "duckdb/main/attached_database.hpp"
@@ -19,0 +21 @@
+#include "duckdb/main/database_manager.hpp"
@@ -36 +37,0 @@
-#include "duckdb/common/types/column_data_collection.hpp"
@@ -44,0 +46 @@
+#include "duckdb/transaction/meta_transaction.hpp"
@@ -47,4 +48,0 @@
-#include "duckdb/main/database_manager.hpp"
-#include "duckdb/transaction/meta_transaction.hpp"
-#include "duckdb/common/http_stats.hpp"
-#include "duckdb/main/attached_database.hpp"
