--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_merged.cc
@@ -7 +6,0 @@
-#include "absl/strings/numbers.h"
@@ -20 +18,0 @@
-#include "src/core/lib/gprpp/host_port.h"
@@ -113,3 +110,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-  switch (type) {
-    case HeaderMatcherType::REGEX:
-      regex_match {
@@ -358,6 +344,6 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
+const char* XdsApi::kLdsTypeUrl =
+    "type.googleapis.com/envoy.config.listener.v3.Listener";
+const char* XdsApi::kRdsTypeUrl =
+    "type.googleapis.com/envoy.config.route.v3.RouteConfiguration";
+const char* XdsApi::kCdsTypeUrl =
+    "type.googleapis.com/envoy.config.cluster.v3.Cluster";
@@ -749 +716 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -753,21 +720,5 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField(
-              "version_info",
-              envoy_service_discovery_v3_DiscoveryResponse_version_info(
-                  response),
-              &fields);
-          size_t num_resources;
-          envoy_service_discovery_v3_DiscoveryResponse_resources(
-              response, &num_resources);
-          fields.emplace_back(
-              absl::StrCat("resources: <", num_resources, " element(s)>"));
-          AddStringField(
-              "type_url",
-              envoy_service_discovery_v3_DiscoveryResponse_type_url(response),
-              &fields);
-          AddStringField(
-              "nonce",
-              envoy_service_discovery_v3_DiscoveryResponse_nonce(response),
-              &fields);
-          gpr_log(GPR_DEBUG, "[xds_client %p] received response: %s", client,
-                  absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_service_discovery_v3_DiscoveryResponse_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(response, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] received response: %s", client, buf);
@@ -777 +728 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -781,125 +732,5 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField(
-              "name",
-              envoy_config_route_v3_RouteConfiguration_name(route_config),
-              &fields);
-          size_t num_virtual_hosts;
-          const envoy_config_route_v3_VirtualHost* const* virtual_hosts =
-              envoy_config_route_v3_RouteConfiguration_virtual_hosts(
-                  route_config, &num_virtual_hosts);
-          for (size_t i = 0; i < num_virtual_hosts; ++i) {
-            const auto* virtual_host = virtual_hosts[i];
-            fields.push_back("virtual_hosts {");
-            AddStringField("  name",
-                           envoy_config_route_v3_VirtualHost_name(virtual_host),
-                           &fields);
-            size_t num_domains;
-            const upb_strview* const domains =
-                envoy_config_route_v3_VirtualHost_domains(virtual_host,
-                                                          &num_domains);
-            for (size_t j = 0; j < num_domains; ++j) {
-              AddStringField("  domains", domains[j], &fields);
-            }
-            size_t num_routes;
-            const envoy_config_route_v3_Route* const* routes =
-                envoy_config_route_v3_VirtualHost_routes(virtual_host,
-                                                         &num_routes);
-            for (size_t j = 0; j < num_routes; ++j) {
-              const auto* route = routes[j];
-              fields.push_back("  route {");
-              AddStringField("    name",
-                             envoy_config_route_v3_Route_name(route), &fields);
-              const envoy_config_route_v3_RouteMatch* match =
-                  envoy_config_route_v3_Route_match(route);
-              if (match != nullptr) {
-                fields.emplace_back("    match {");
-                if (envoy_config_route_v3_RouteMatch_has_prefix(match)) {
-                  AddStringField("      prefix",
-                                 envoy_config_route_v3_RouteMatch_prefix(match),
-                                 &fields,
-                                                  true);
-                } else if (envoy_config_route_v3_RouteMatch_has_path(match)) {
-                  AddStringField("      path",
-                                 envoy_config_route_v3_RouteMatch_path(match),
-                                 &fields,
-                                                  true);
-                } else if (envoy_config_route_v3_RouteMatch_has_safe_regex(
-                               match)) {
-                  fields.emplace_back("      safe_regex: <not printed>");
-                } else {
-                  fields.emplace_back("      <unknown path matching type>");
-                }
-                size_t num_headers;
-                envoy_config_route_v3_RouteMatch_headers(match, &num_headers);
-                if (num_headers > 0) {
-                  fields.emplace_back(absl::StrCat(
-                      "      headers: <", num_headers, " element(s)>"));
-                }
-                fields.emplace_back("    }");
-              }
-              if (envoy_config_route_v3_Route_has_route(route)) {
-                const envoy_config_route_v3_RouteAction* action =
-                    envoy_config_route_v3_Route_route(route);
-                fields.emplace_back("    route {");
-                if (envoy_config_route_v3_RouteAction_has_cluster(action)) {
-                  AddStringField(
-                      "      cluster",
-                      envoy_config_route_v3_RouteAction_cluster(action),
-                      &fields);
-                } else if (envoy_config_route_v3_RouteAction_has_cluster_header(
-                               action)) {
-                  AddStringField(
-                      "      cluster_header",
-                      envoy_config_route_v3_RouteAction_cluster_header(action),
-                      &fields);
-                } else if (
-                    envoy_config_route_v3_RouteAction_has_weighted_clusters(
-                        action)) {
-                  const envoy_config_route_v3_WeightedCluster*
-                      weighted_clusters =
-                          envoy_config_route_v3_RouteAction_weighted_clusters(
-                              action);
-                  fields.emplace_back("      weighted_clusters {");
-                  size_t num_cluster_weights;
-                  const envoy_config_route_v3_WeightedCluster_ClusterWeight* const*
-                      cluster_weights =
-                          envoy_config_route_v3_WeightedCluster_clusters(
-                              weighted_clusters, &num_cluster_weights);
-                  for (size_t i = 0; i < num_cluster_weights; ++i) {
-                    const envoy_config_route_v3_WeightedCluster_ClusterWeight*
-                        cluster_weight = cluster_weights[i];
-                    fields.emplace_back("        clusters {");
-                    AddStringField(
-                        "          name",
-                        envoy_config_route_v3_WeightedCluster_ClusterWeight_name(
-                            cluster_weight),
-                        &fields);
-                    AddUInt32ValueField(
-                        "          weight",
-                        envoy_config_route_v3_WeightedCluster_ClusterWeight_weight(
-                            cluster_weight),
-                        &fields);
-                    fields.emplace_back("        }");
-                  }
-                  AddUInt32ValueField(
-                      "        total_weight",
-                      envoy_config_route_v3_WeightedCluster_total_weight(
-                          weighted_clusters),
-                      &fields);
-                  fields.emplace_back("      }");
-                }
-                fields.emplace_back("    }");
-              } else if (envoy_config_route_v3_Route_has_redirect(route)) {
-                fields.emplace_back("    redirect: <not printed>");
-              } else if (envoy_config_route_v3_Route_has_direct_response(
-                             route)) {
-                fields.emplace_back("    direct_response: <not printed>");
-              } else if (envoy_config_route_v3_Route_has_filter_action(route)) {
-                fields.emplace_back("    filter_action: <not printed>");
-              }
-              fields.push_back("  }");
-            }
-            fields.push_back("}");
-          }
-          gpr_log(GPR_DEBUG, "[xds_client %p] RouteConfiguration: %s", client,
-                  absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_config_route_v3_RouteConfiguration_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(route_config, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] RouteConfiguration: %s", client, buf);
@@ -908 +739 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void MaybeLogCluster(XdsClient* client, TraceFlag* tracer,
+void MaybeLogCluster(XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -912,48 +743,5 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField("name", envoy_config_cluster_v3_Cluster_name(cluster),
-                         &fields);
-          if (envoy_config_cluster_v3_Cluster_has_type(cluster)) {
-            fields.emplace_back(absl::StrCat(
-                "type: ", envoy_config_cluster_v3_Cluster_type(cluster)));
-          } else if (envoy_config_cluster_v3_Cluster_has_cluster_type(
-                         cluster)) {
-            fields.emplace_back("cluster_type: <not printed>");
-          } else {
-            fields.emplace_back("<unknown type>");
-          }
-          const envoy_config_cluster_v3_Cluster_EdsClusterConfig*
-              eds_cluster_config =
-                  envoy_config_cluster_v3_Cluster_eds_cluster_config(cluster);
-          if (eds_cluster_config != nullptr) {
-            fields.emplace_back("eds_cluster_config {");
-            const struct envoy_config_core_v3_ConfigSource* eds_config =
-                envoy_config_cluster_v3_Cluster_EdsClusterConfig_eds_config(
-                    eds_cluster_config);
-            if (eds_config != nullptr) {
-              if (envoy_config_core_v3_ConfigSource_has_ads(eds_config)) {
-                fields.emplace_back("  eds_config { ads {} }");
-              } else {
-                fields.emplace_back("  eds_config: <non-ADS type>");
-              }
-            }
-            AddStringField(
-                "  service_name",
-                envoy_config_cluster_v3_Cluster_EdsClusterConfig_service_name(
-                    eds_cluster_config),
-                &fields);
-            fields.emplace_back("}");
-          }
-          fields.emplace_back(
-              absl::StrCat("lb_policy: ",
-                           envoy_config_cluster_v3_Cluster_lb_policy(cluster)));
-          const envoy_config_core_v3_ConfigSource* lrs_server =
-              envoy_config_cluster_v3_Cluster_lrs_server(cluster);
-          if (lrs_server != nullptr) {
-            if (envoy_config_core_v3_ConfigSource_has_self(lrs_server)) {
-              fields.emplace_back("lrs_server { self {} }");
-            } else {
-              fields.emplace_back("lrs_server: <non-self type>");
-            }
-          }
-          gpr_log(GPR_DEBUG, "[xds_client %p] Cluster: %s", client,
-                  absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_config_cluster_v3_Cluster_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(cluster, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] Cluster: %s", client, buf);
@@ -963 +751 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -967,121 +755,6 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField(
-              "cluster_name",
-              envoy_config_endpoint_v3_ClusterLoadAssignment_cluster_name(cla),
-              &fields);
-          size_t num_localities;
-          const struct envoy_config_endpoint_v3_LocalityLbEndpoints* const*
-              locality_endpoints =
-                  envoy_config_endpoint_v3_ClusterLoadAssignment_endpoints(
-                      cla, &num_localities);
-          for (size_t i = 0; i < num_localities; ++i) {
-            const auto* locality_endpoint = locality_endpoints[i];
-            fields.emplace_back("endpoints {");
-            const auto* locality =
-                envoy_config_endpoint_v3_LocalityLbEndpoints_locality(
-                    locality_endpoint);
-            if (locality != nullptr) {
-              fields.emplace_back("  locality {");
-              AddLocalityField(2, locality, &fields);
-              fields.emplace_back("  }");
-            }
-            size_t num_lb_endpoints;
-            const envoy_config_endpoint_v3_LbEndpoint* const* lb_endpoints =
-                envoy_config_endpoint_v3_LocalityLbEndpoints_lb_endpoints(
-                    locality_endpoint, &num_lb_endpoints);
-            for (size_t j = 0; j < num_lb_endpoints; ++j) {
-              const auto* lb_endpoint = lb_endpoints[j];
-              fields.emplace_back("  lb_endpoints {");
-              uint32_t health_status =
-                  envoy_config_endpoint_v3_LbEndpoint_health_status(
-                      lb_endpoint);
-              if (health_status > 0) {
-                fields.emplace_back(
-                    absl::StrCat("    health_status: ", health_status));
-              }
-              const envoy_config_endpoint_v3_Endpoint* endpoint =
-                  envoy_config_endpoint_v3_LbEndpoint_endpoint(lb_endpoint);
-              if (endpoint != nullptr) {
-                fields.emplace_back("    endpoint {");
-                const auto* address =
-                    envoy_config_endpoint_v3_Endpoint_address(endpoint);
-                if (address != nullptr) {
-                  fields.emplace_back("      address {");
-                  const auto* socket_address =
-                      envoy_config_core_v3_Address_socket_address(address);
-                  if (socket_address != nullptr) {
-                    fields.emplace_back("        socket_address {");
-                    AddStringField("          address",
-                                   envoy_config_core_v3_SocketAddress_address(
-                                       socket_address),
-                                   &fields);
-                    if (envoy_config_core_v3_SocketAddress_has_port_value(
-                            socket_address)) {
-                      fields.emplace_back(absl::StrCat(
-                          "          port_value: ",
-                          envoy_config_core_v3_SocketAddress_port_value(
-                              socket_address)));
-                    } else {
-                      fields.emplace_back("        <non-numeric port>");
-                    }
-                    fields.emplace_back("        }");
-                  } else {
-                    fields.emplace_back("        <non-socket address>");
-                  }
-                  fields.emplace_back("      }");
-                }
-                fields.emplace_back("    }");
-              }
-              fields.emplace_back("  }");
-            }
-            AddUInt32ValueField(
-                "  load_balancing_weight",
-                envoy_config_endpoint_v3_LocalityLbEndpoints_load_balancing_weight(
-                    locality_endpoint),
-                &fields);
-            uint32_t priority =
-                envoy_config_endpoint_v3_LocalityLbEndpoints_priority(
-                    locality_endpoint);
-            if (priority > 0) {
-              fields.emplace_back(absl::StrCat("  priority: ", priority));
-            }
-            fields.emplace_back("}");
-          }
-          const envoy_config_endpoint_v3_ClusterLoadAssignment_Policy* policy =
-              envoy_config_endpoint_v3_ClusterLoadAssignment_policy(cla);
-          if (policy != nullptr) {
-            fields.emplace_back("policy {");
-            size_t num_drop_overloads;
-            const envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_DropOverload* const*
-                drop_overloads =
-                    envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_drop_overloads(
-                        policy, &num_drop_overloads);
-            for (size_t i = 0; i < num_drop_overloads; ++i) {
-              auto* drop_overload = drop_overloads[i];
-              fields.emplace_back("  drop_overloads {");
-              AddStringField(
-                  "    category",
-                  envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_DropOverload_category(
-                      drop_overload),
-                  &fields);
-              const auto* drop_percentage =
-                  envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_DropOverload_drop_percentage(
-                      drop_overload);
-              if (drop_percentage != nullptr) {
-                fields.emplace_back("    drop_percentage {");
-                fields.emplace_back(
-                    absl::StrCat("      numerator: ",
-                                 envoy_type_v3_FractionalPercent_numerator(
-                                     drop_percentage)));
-                fields.emplace_back(
-                    absl::StrCat("      denominator: ",
-                                 envoy_type_v3_FractionalPercent_denominator(
-                                     drop_percentage)));
-                fields.emplace_back("    }");
-              }
-              fields.emplace_back("  }");
-            }
-            fields.emplace_back("}");
-          }
-          gpr_log(GPR_DEBUG, "[xds_client %p] ClusterLoadAssignment: %s",
-                  client, absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_config_endpoint_v3_ClusterLoadAssignment_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(cla, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] ClusterLoadAssignment: %s", client,
+            buf);
@@ -1328,2 +990 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "RouteAction weighted_cluster has no valid clusters "
-                "specified.");
+          "RouteAction weighted_cluster has no valid clusters specified.");
@@ -1338 +999 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1341 +1002 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        MaybeLogRouteConfiguration(client, tracer, route_config);
+  MaybeLogRouteConfiguration(client, tracer, symtab, route_config);
@@ -1414 +1068 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1455,2 +1106 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "Could not parse HttpConnectionManager config from "
-                "ApiListener");
+          "Could not parse HttpConnectionManager config from ApiListener");
@@ -1465 +1115 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                RouteConfigParse(client, tracer, route_config, &rds_update);
+          RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1474,2 +1124 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "HttpConnectionManager neither has inlined route_config nor "
-                "RDS.");
+          "HttpConnectionManager neither has inlined route_config nor RDS.");
@@ -1490,2 +1138 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "HttpConnectionManager ConfigSource for RDS does not specify "
-                "ADS.");
+          "HttpConnectionManager ConfigSource for RDS does not specify ADS.");
@@ -1502 +1149 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1532,2 +1177,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          grpc_error* error =
-              RouteConfigParse(client, tracer, route_config, &local_rds_update);
+    grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config,
+                                         &local_rds_update);
@@ -1541 +1186 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1565,3 +1207,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          MaybeLogCluster(client, tracer, cluster);
-          std::string cluster_name = UpbStringToStdString(
-              envoy_config_cluster_v3_Cluster_name(cluster));
+    MaybeLogCluster(client, tracer, symtab, cluster);
+    std::string cluster_name =
+        UpbStringToStdString(envoy_config_cluster_v3_Cluster_name(cluster));
@@ -1721 +1354 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1747 +1378 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          MaybeLogClusterLoadAssignment(client, tracer,
+    MaybeLogClusterLoadAssignment(client, tracer, symtab,
@@ -1842,10 +1467,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = LdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response, expected_server_name,
-              &result.lds_update, arena.ptr());
-        } else if (IsRds(result.type_url)) {
-<<<<<<< HEAD
-          result.parse_error = RdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response, expected_server_name,
-              expected_route_configuration_names, &result.rds_update,
-              arena.ptr());
-|||||||
@@ -1853,4 +1469,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              RdsResponseParse(client_, tracer_, response, expected_server_name,
-                               expected_route_configuration_names,
-                               &result.rds_update, arena.ptr());
-=======
+        LdsResponseParse(client_, tracer_, symtab_.ptr(), response,
+                         expected_server_name, &result.lds_update, arena.ptr());
+  } else if (IsRds(result.type_url)) {
@@ -1858,3 +1473,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              client_, tracer_, response, expected_route_configuration_names,
-              &result.rds_update, arena.ptr());
->>>>>>> d109f4b4064982d6770cd8776a1dd58f4c3935a4
+        client_, tracer_, symtab_.ptr(), response,
+        expected_route_configuration_names, &result.rds_update, arena.ptr());
