diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_merged.cc
index 2e42e3f..db0b30c 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_merged.cc
@@ -7 +6,0 @@
-#include "absl/strings/numbers.h"
@@ -20 +18,0 @@
-#include "src/core/lib/gprpp/host_port.h"
@@ -69,2 +67,2 @@ XdsApi::Route::Matchers::PathMatcher::PathMatcher(const PathMatcher& other)
-XdsApi::Route::Matchers::PathMatcher&
-XdsApi::Route::Matchers::PathMatcher::operator=(const PathMatcher& other) {
+XdsApi::Route::Matchers::PathMatcher& XdsApi::Route::Matchers::PathMatcher::
+operator=(const PathMatcher& other) {
@@ -113,3 +110,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-  switch (type) {
-    case HeaderMatcherType::REGEX:
-      regex_match {
@@ -131,3 +126,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      XdsApi::Route::Matchers::HeaderMatcher&
-      XdsApi::Route::Matchers::HeaderMatcher::operator=(
-          const HeaderMatcher& other) {
+XdsApi::Route::Matchers::HeaderMatcher& XdsApi::Route::Matchers::HeaderMatcher::
+operator=(const HeaderMatcher& other) {
@@ -162,2 +156 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return range_start != other.range_start &&
-                   range_end != other.range_end;
+      return range_start != other.range_start && range_end != other.range_end;
@@ -174,2 +167 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   string_matcher);
+                             invert_match ? " not" : "", name, string_matcher);
@@ -182,2 +174,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   range_start, range_end);
+                             invert_match ? " not" : "", name, range_start,
+                             range_end);
@@ -190,2 +182 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   string_matcher);
+                             invert_match ? " not" : "", name, string_matcher);
@@ -194,2 +185 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                                   invert_match ? " not" : "", name,
-                                   string_matcher);
+                             invert_match ? " not" : "", name, string_matcher);
@@ -286,2 +276 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        if (domain_pattern[domain_pattern.size() - 1] == '*')
-          return PREFIX_MATCH;
+  if (domain_pattern[domain_pattern.size() - 1] == '*') return PREFIX_MATCH;
@@ -292,2 +281 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      XdsApi::RdsUpdate::FindVirtualHostForDomain(const std::string& domain)
-          const {
+XdsApi::RdsUpdate::FindVirtualHostForDomain(const std::string& domain) const {
@@ -328,2 +316 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        if (!Contains(locality.priority))
-          priorities_.resize(locality.priority + 1);
+  if (!Contains(locality.priority)) priorities_.resize(locality.priority + 1);
@@ -333,2 +320,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      const XdsApi::PriorityListUpdate::LocalityMap*
-      XdsApi::PriorityListUpdate::Find(uint32_t priority) const {
+const XdsApi::PriorityListUpdate::LocalityMap* XdsApi::PriorityListUpdate::Find(
+    uint32_t priority) const {
@@ -346,2 +333 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      bool XdsApi::DropConfig::ShouldDrop(const std::string** category_name)
-          const {
+bool XdsApi::DropConfig::ShouldDrop(const std::string** category_name) const {
@@ -358,6 +344,6 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
-      const char* XdsApi::kEdsTypeUrl =
-          "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
+const char* XdsApi::kLdsTypeUrl =
+    "type.googleapis.com/envoy.config.listener.v3.Listener";
+const char* XdsApi::kRdsTypeUrl =
+    "type.googleapis.com/envoy.config.route.v3.RouteConfiguration";
+const char* XdsApi::kCdsTypeUrl =
+    "type.googleapis.com/envoy.config.cluster.v3.Cluster";
@@ -392,4 +378,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            build_version_(absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING,
-                                        " ", grpc_version_string())),
-            user_agent_name_(
-                absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING)) {}
+      build_version_(absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING, " ",
+                                  grpc_version_string())),
+      user_agent_name_(absl::StrCat("gRPC C-core ", GPR_PLATFORM_STRING)) {}
@@ -401,2 +386 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateMetadataValue(upb_arena* arena,
-                                 google_protobuf_Value* value_pb,
+void PopulateMetadataValue(upb_arena* arena, google_protobuf_Value* value_pb,
@@ -404,2 +388 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateListValue(upb_arena* arena,
-                             google_protobuf_ListValue* list_value,
+void PopulateListValue(upb_arena* arena, google_protobuf_ListValue* list_value,
@@ -408,2 +391 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          auto* value_pb =
-              google_protobuf_ListValue_add_values(list_value, arena);
+    auto* value_pb = google_protobuf_ListValue_add_values(list_value, arena);
@@ -413,2 +395 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateMetadata(upb_arena* arena,
-                            google_protobuf_Struct* metadata_pb,
+void PopulateMetadata(upb_arena* arena, google_protobuf_Struct* metadata_pb,
@@ -423,2 +404 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateMetadataValue(upb_arena* arena,
-                                 google_protobuf_Value* value_pb,
+void PopulateMetadataValue(upb_arena* arena, google_protobuf_Value* value_pb,
@@ -471,2 +451 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      std::string EncodeStringField(uint32_t field_number,
-                                    const std::string& str) {
+std::string EncodeStringField(uint32_t field_number, const std::string& str) {
@@ -477,2 +456 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void PopulateBuildVersion(upb_arena* arena,
-                                envoy_config_core_v3_Node* node_msg,
+void PopulateBuildVersion(upb_arena* arena, envoy_config_core_v3_Node* node_msg,
@@ -508,2 +486,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            google_protobuf_Value_set_string_value(
-                value, StdStringToUpbString(server_name));
+      google_protobuf_Value_set_string_value(value,
+                                             StdStringToUpbString(server_name));
@@ -540,2 +518 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            node_msg,
-            upb_strview_makez("envoy.lb.does_not_support_overprovisioning"),
+      node_msg, upb_strview_makez("envoy.lb.does_not_support_overprovisioning"),
@@ -563,2 +540 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              name, " { value: ", google_protobuf_UInt32Value_value(value),
-              " }"));
+        name, " { value: ", google_protobuf_UInt32Value_value(value), " }"));
@@ -567,2 +543,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      inline void AddLocalityField(
-          int indent_level, const envoy_config_core_v3_Locality* locality,
+inline void AddLocalityField(int indent_level,
+                             const envoy_config_core_v3_Locality* locality,
@@ -573,2 +549,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        AddStringField(field.c_str(),
-                       envoy_config_core_v3_Locality_region(locality), fields);
+  AddStringField(field.c_str(), envoy_config_core_v3_Locality_region(locality),
+                 fields);
@@ -576,2 +552,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        AddStringField(field.c_str(),
-                       envoy_config_core_v3_Locality_zone(locality), fields);
+  AddStringField(field.c_str(), envoy_config_core_v3_Locality_zone(locality),
+                 fields);
@@ -580,2 +556 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                       envoy_config_core_v3_Locality_sub_zone(locality),
-                       fields);
+                 envoy_config_core_v3_Locality_sub_zone(locality), fields);
@@ -598,2 +573 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            AddStringField("      key",
-                           google_protobuf_Struct_FieldsEntry_key(entry),
+      AddStringField("      key", google_protobuf_Struct_FieldsEntry_key(entry),
@@ -608,2 +582 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                    UpbStringToAbsl(google_protobuf_Value_string_value(value)),
-                    "\"");
+              UpbStringToAbsl(google_protobuf_Value_string_value(value)), "\"");
@@ -613,2 +586 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                value_str =
-                    absl::StrCat("double_value: ",
+          value_str = absl::StrCat("double_value: ",
@@ -617,2 +589,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                value_str = absl::StrCat(
-                    "bool_value: ", google_protobuf_Value_bool_value(value));
+          value_str = absl::StrCat("bool_value: ",
+                                   google_protobuf_Value_bool_value(value));
@@ -626,2 +598 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              fields->emplace_back(
-                  absl::StrCat("      value { ", value_str, " }"));
+        fields->emplace_back(absl::StrCat("      value { ", value_str, " }"));
@@ -647,2 +618 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                       envoy_config_core_v3_Node_user_agent_version(node),
-                       fields);
+                 envoy_config_core_v3_Node_user_agent_version(node), fields);
@@ -651,2 +621 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_config_core_v3_Node_client_features(node,
-                                                      &num_client_features);
+      envoy_config_core_v3_Node_client_features(node, &num_client_features);
@@ -667,2 +636,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_log(GPR_DEBUG, "[xds_client %p] constructed ADS request: %s",
-                  client, buf);
+    gpr_log(GPR_DEBUG, "[xds_client %p] constructed ADS request: %s", client,
+            buf);
@@ -672,2 +641 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          upb_arena* arena,
-          envoy_service_discovery_v3_DiscoveryRequest* request) {
+    upb_arena* arena, envoy_service_discovery_v3_DiscoveryRequest* request) {
@@ -701,2 +669,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const std::string& version, const std::string& nonce,
-          grpc_error* error, bool populate_node) {
+    const std::string& version, const std::string& nonce, grpc_error* error,
+    bool populate_node) {
@@ -722,2 +690,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          upb_strview error_description_strview = upb_strview_make(
-              reinterpret_cast<const char*>(
+    upb_strview error_description_strview =
+        upb_strview_make(reinterpret_cast<const char*>(
@@ -729,2 +697 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          google_rpc_Status_set_message(error_detail,
-                                        error_description_strview);
+    google_rpc_Status_set_message(error_detail, error_description_strview);
@@ -735,4 +702,4 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_service_discovery_v3_DiscoveryRequest_mutable_node(
-                  request, arena.ptr());
-          PopulateNode(arena.ptr(), bootstrap_, build_version_,
-                       user_agent_name_, "", node_msg);
+        envoy_service_discovery_v3_DiscoveryRequest_mutable_node(request,
+                                                                 arena.ptr());
+    PopulateNode(arena.ptr(), bootstrap_, build_version_, user_agent_name_, "",
+                 node_msg);
@@ -749 +716 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -753,21 +720,5 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField(
-              "version_info",
-              envoy_service_discovery_v3_DiscoveryResponse_version_info(
-                  response),
-              &fields);
-          size_t num_resources;
-          envoy_service_discovery_v3_DiscoveryResponse_resources(
-              response, &num_resources);
-          fields.emplace_back(
-              absl::StrCat("resources: <", num_resources, " element(s)>"));
-          AddStringField(
-              "type_url",
-              envoy_service_discovery_v3_DiscoveryResponse_type_url(response),
-              &fields);
-          AddStringField(
-              "nonce",
-              envoy_service_discovery_v3_DiscoveryResponse_nonce(response),
-              &fields);
-          gpr_log(GPR_DEBUG, "[xds_client %p] received response: %s", client,
-                  absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_service_discovery_v3_DiscoveryResponse_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(response, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] received response: %s", client, buf);
@@ -777 +728 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -781,125 +732,5 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField(
-              "name",
-              envoy_config_route_v3_RouteConfiguration_name(route_config),
-              &fields);
-          size_t num_virtual_hosts;
-          const envoy_config_route_v3_VirtualHost* const* virtual_hosts =
-              envoy_config_route_v3_RouteConfiguration_virtual_hosts(
-                  route_config, &num_virtual_hosts);
-          for (size_t i = 0; i < num_virtual_hosts; ++i) {
-            const auto* virtual_host = virtual_hosts[i];
-            fields.push_back("virtual_hosts {");
-            AddStringField("  name",
-                           envoy_config_route_v3_VirtualHost_name(virtual_host),
-                           &fields);
-            size_t num_domains;
-            const upb_strview* const domains =
-                envoy_config_route_v3_VirtualHost_domains(virtual_host,
-                                                          &num_domains);
-            for (size_t j = 0; j < num_domains; ++j) {
-              AddStringField("  domains", domains[j], &fields);
-            }
-            size_t num_routes;
-            const envoy_config_route_v3_Route* const* routes =
-                envoy_config_route_v3_VirtualHost_routes(virtual_host,
-                                                         &num_routes);
-            for (size_t j = 0; j < num_routes; ++j) {
-              const auto* route = routes[j];
-              fields.push_back("  route {");
-              AddStringField("    name",
-                             envoy_config_route_v3_Route_name(route), &fields);
-              const envoy_config_route_v3_RouteMatch* match =
-                  envoy_config_route_v3_Route_match(route);
-              if (match != nullptr) {
-                fields.emplace_back("    match {");
-                if (envoy_config_route_v3_RouteMatch_has_prefix(match)) {
-                  AddStringField("      prefix",
-                                 envoy_config_route_v3_RouteMatch_prefix(match),
-                                 &fields,
-                                                  true);
-                } else if (envoy_config_route_v3_RouteMatch_has_path(match)) {
-                  AddStringField("      path",
-                                 envoy_config_route_v3_RouteMatch_path(match),
-                                 &fields,
-                                                  true);
-                } else if (envoy_config_route_v3_RouteMatch_has_safe_regex(
-                               match)) {
-                  fields.emplace_back("      safe_regex: <not printed>");
-                } else {
-                  fields.emplace_back("      <unknown path matching type>");
-                }
-                size_t num_headers;
-                envoy_config_route_v3_RouteMatch_headers(match, &num_headers);
-                if (num_headers > 0) {
-                  fields.emplace_back(absl::StrCat(
-                      "      headers: <", num_headers, " element(s)>"));
-                }
-                fields.emplace_back("    }");
-              }
-              if (envoy_config_route_v3_Route_has_route(route)) {
-                const envoy_config_route_v3_RouteAction* action =
-                    envoy_config_route_v3_Route_route(route);
-                fields.emplace_back("    route {");
-                if (envoy_config_route_v3_RouteAction_has_cluster(action)) {
-                  AddStringField(
-                      "      cluster",
-                      envoy_config_route_v3_RouteAction_cluster(action),
-                      &fields);
-                } else if (envoy_config_route_v3_RouteAction_has_cluster_header(
-                               action)) {
-                  AddStringField(
-                      "      cluster_header",
-                      envoy_config_route_v3_RouteAction_cluster_header(action),
-                      &fields);
-                } else if (
-                    envoy_config_route_v3_RouteAction_has_weighted_clusters(
-                        action)) {
-                  const envoy_config_route_v3_WeightedCluster*
-                      weighted_clusters =
-                          envoy_config_route_v3_RouteAction_weighted_clusters(
-                              action);
-                  fields.emplace_back("      weighted_clusters {");
-                  size_t num_cluster_weights;
-                  const envoy_config_route_v3_WeightedCluster_ClusterWeight* const*
-                      cluster_weights =
-                          envoy_config_route_v3_WeightedCluster_clusters(
-                              weighted_clusters, &num_cluster_weights);
-                  for (size_t i = 0; i < num_cluster_weights; ++i) {
-                    const envoy_config_route_v3_WeightedCluster_ClusterWeight*
-                        cluster_weight = cluster_weights[i];
-                    fields.emplace_back("        clusters {");
-                    AddStringField(
-                        "          name",
-                        envoy_config_route_v3_WeightedCluster_ClusterWeight_name(
-                            cluster_weight),
-                        &fields);
-                    AddUInt32ValueField(
-                        "          weight",
-                        envoy_config_route_v3_WeightedCluster_ClusterWeight_weight(
-                            cluster_weight),
-                        &fields);
-                    fields.emplace_back("        }");
-                  }
-                  AddUInt32ValueField(
-                      "        total_weight",
-                      envoy_config_route_v3_WeightedCluster_total_weight(
-                          weighted_clusters),
-                      &fields);
-                  fields.emplace_back("      }");
-                }
-                fields.emplace_back("    }");
-              } else if (envoy_config_route_v3_Route_has_redirect(route)) {
-                fields.emplace_back("    redirect: <not printed>");
-              } else if (envoy_config_route_v3_Route_has_direct_response(
-                             route)) {
-                fields.emplace_back("    direct_response: <not printed>");
-              } else if (envoy_config_route_v3_Route_has_filter_action(route)) {
-                fields.emplace_back("    filter_action: <not printed>");
-              }
-              fields.push_back("  }");
-            }
-            fields.push_back("}");
-          }
-          gpr_log(GPR_DEBUG, "[xds_client %p] RouteConfiguration: %s", client,
-                  absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_config_route_v3_RouteConfiguration_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(route_config, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] RouteConfiguration: %s", client, buf);
@@ -908 +739 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      void MaybeLogCluster(XdsClient* client, TraceFlag* tracer,
+void MaybeLogCluster(XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -912,48 +743,5 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField("name", envoy_config_cluster_v3_Cluster_name(cluster),
-                         &fields);
-          if (envoy_config_cluster_v3_Cluster_has_type(cluster)) {
-            fields.emplace_back(absl::StrCat(
-                "type: ", envoy_config_cluster_v3_Cluster_type(cluster)));
-          } else if (envoy_config_cluster_v3_Cluster_has_cluster_type(
-                         cluster)) {
-            fields.emplace_back("cluster_type: <not printed>");
-          } else {
-            fields.emplace_back("<unknown type>");
-          }
-          const envoy_config_cluster_v3_Cluster_EdsClusterConfig*
-              eds_cluster_config =
-                  envoy_config_cluster_v3_Cluster_eds_cluster_config(cluster);
-          if (eds_cluster_config != nullptr) {
-            fields.emplace_back("eds_cluster_config {");
-            const struct envoy_config_core_v3_ConfigSource* eds_config =
-                envoy_config_cluster_v3_Cluster_EdsClusterConfig_eds_config(
-                    eds_cluster_config);
-            if (eds_config != nullptr) {
-              if (envoy_config_core_v3_ConfigSource_has_ads(eds_config)) {
-                fields.emplace_back("  eds_config { ads {} }");
-              } else {
-                fields.emplace_back("  eds_config: <non-ADS type>");
-              }
-            }
-            AddStringField(
-                "  service_name",
-                envoy_config_cluster_v3_Cluster_EdsClusterConfig_service_name(
-                    eds_cluster_config),
-                &fields);
-            fields.emplace_back("}");
-          }
-          fields.emplace_back(
-              absl::StrCat("lb_policy: ",
-                           envoy_config_cluster_v3_Cluster_lb_policy(cluster)));
-          const envoy_config_core_v3_ConfigSource* lrs_server =
-              envoy_config_cluster_v3_Cluster_lrs_server(cluster);
-          if (lrs_server != nullptr) {
-            if (envoy_config_core_v3_ConfigSource_has_self(lrs_server)) {
-              fields.emplace_back("lrs_server { self {} }");
-            } else {
-              fields.emplace_back("lrs_server: <non-self type>");
-            }
-          }
-          gpr_log(GPR_DEBUG, "[xds_client %p] Cluster: %s", client,
-                  absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_config_cluster_v3_Cluster_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(cluster, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] Cluster: %s", client, buf);
@@ -963 +751 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -967,121 +755,6 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::vector<std::string> fields;
-          AddStringField(
-              "cluster_name",
-              envoy_config_endpoint_v3_ClusterLoadAssignment_cluster_name(cla),
-              &fields);
-          size_t num_localities;
-          const struct envoy_config_endpoint_v3_LocalityLbEndpoints* const*
-              locality_endpoints =
-                  envoy_config_endpoint_v3_ClusterLoadAssignment_endpoints(
-                      cla, &num_localities);
-          for (size_t i = 0; i < num_localities; ++i) {
-            const auto* locality_endpoint = locality_endpoints[i];
-            fields.emplace_back("endpoints {");
-            const auto* locality =
-                envoy_config_endpoint_v3_LocalityLbEndpoints_locality(
-                    locality_endpoint);
-            if (locality != nullptr) {
-              fields.emplace_back("  locality {");
-              AddLocalityField(2, locality, &fields);
-              fields.emplace_back("  }");
-            }
-            size_t num_lb_endpoints;
-            const envoy_config_endpoint_v3_LbEndpoint* const* lb_endpoints =
-                envoy_config_endpoint_v3_LocalityLbEndpoints_lb_endpoints(
-                    locality_endpoint, &num_lb_endpoints);
-            for (size_t j = 0; j < num_lb_endpoints; ++j) {
-              const auto* lb_endpoint = lb_endpoints[j];
-              fields.emplace_back("  lb_endpoints {");
-              uint32_t health_status =
-                  envoy_config_endpoint_v3_LbEndpoint_health_status(
-                      lb_endpoint);
-              if (health_status > 0) {
-                fields.emplace_back(
-                    absl::StrCat("    health_status: ", health_status));
-              }
-              const envoy_config_endpoint_v3_Endpoint* endpoint =
-                  envoy_config_endpoint_v3_LbEndpoint_endpoint(lb_endpoint);
-              if (endpoint != nullptr) {
-                fields.emplace_back("    endpoint {");
-                const auto* address =
-                    envoy_config_endpoint_v3_Endpoint_address(endpoint);
-                if (address != nullptr) {
-                  fields.emplace_back("      address {");
-                  const auto* socket_address =
-                      envoy_config_core_v3_Address_socket_address(address);
-                  if (socket_address != nullptr) {
-                    fields.emplace_back("        socket_address {");
-                    AddStringField("          address",
-                                   envoy_config_core_v3_SocketAddress_address(
-                                       socket_address),
-                                   &fields);
-                    if (envoy_config_core_v3_SocketAddress_has_port_value(
-                            socket_address)) {
-                      fields.emplace_back(absl::StrCat(
-                          "          port_value: ",
-                          envoy_config_core_v3_SocketAddress_port_value(
-                              socket_address)));
-                    } else {
-                      fields.emplace_back("        <non-numeric port>");
-                    }
-                    fields.emplace_back("        }");
-                  } else {
-                    fields.emplace_back("        <non-socket address>");
-                  }
-                  fields.emplace_back("      }");
-                }
-                fields.emplace_back("    }");
-              }
-              fields.emplace_back("  }");
-            }
-            AddUInt32ValueField(
-                "  load_balancing_weight",
-                envoy_config_endpoint_v3_LocalityLbEndpoints_load_balancing_weight(
-                    locality_endpoint),
-                &fields);
-            uint32_t priority =
-                envoy_config_endpoint_v3_LocalityLbEndpoints_priority(
-                    locality_endpoint);
-            if (priority > 0) {
-              fields.emplace_back(absl::StrCat("  priority: ", priority));
-            }
-            fields.emplace_back("}");
-          }
-          const envoy_config_endpoint_v3_ClusterLoadAssignment_Policy* policy =
-              envoy_config_endpoint_v3_ClusterLoadAssignment_policy(cla);
-          if (policy != nullptr) {
-            fields.emplace_back("policy {");
-            size_t num_drop_overloads;
-            const envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_DropOverload* const*
-                drop_overloads =
-                    envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_drop_overloads(
-                        policy, &num_drop_overloads);
-            for (size_t i = 0; i < num_drop_overloads; ++i) {
-              auto* drop_overload = drop_overloads[i];
-              fields.emplace_back("  drop_overloads {");
-              AddStringField(
-                  "    category",
-                  envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_DropOverload_category(
-                      drop_overload),
-                  &fields);
-              const auto* drop_percentage =
-                  envoy_config_endpoint_v3_ClusterLoadAssignment_Policy_DropOverload_drop_percentage(
-                      drop_overload);
-              if (drop_percentage != nullptr) {
-                fields.emplace_back("    drop_percentage {");
-                fields.emplace_back(
-                    absl::StrCat("      numerator: ",
-                                 envoy_type_v3_FractionalPercent_numerator(
-                                     drop_percentage)));
-                fields.emplace_back(
-                    absl::StrCat("      denominator: ",
-                                 envoy_type_v3_FractionalPercent_denominator(
-                                     drop_percentage)));
-                fields.emplace_back("    }");
-              }
-              fields.emplace_back("  }");
-            }
-            fields.emplace_back("}");
-          }
-          gpr_log(GPR_DEBUG, "[xds_client %p] ClusterLoadAssignment: %s",
-                  client, absl::StrJoin(fields, "\n").c_str());
+    const upb_msgdef* msg_type =
+        envoy_config_endpoint_v3_ClusterLoadAssignment_getmsgdef(symtab);
+    char buf[10240];
+    upb_text_encode(cla, msg_type, nullptr, 0, buf, sizeof(buf));
+    gpr_log(GPR_DEBUG, "[xds_client %p] ClusterLoadAssignment: %s", client,
+            buf);
@@ -1090,3 +763,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      grpc_error* RoutePathMatchParse(
-          const envoy_config_route_v3_RouteMatch* match, XdsApi::Route* route,
-          bool* ignore_route) {
+grpc_error* RoutePathMatchParse(const envoy_config_route_v3_RouteMatch* match,
+                                XdsApi::Route* route, bool* ignore_route) {
@@ -1106,2 +778 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            } else if (prefix_elements.size() == 2 &&
-                       prefix_elements[0].empty()) {
+      } else if (prefix_elements.size() == 2 && prefix_elements[0].empty()) {
@@ -1147,2 +818 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          std::unique_ptr<RE2> regex =
-              absl::make_unique<RE2>(std::move(matcher));
+    std::unique_ptr<RE2> regex = absl::make_unique<RE2>(std::move(matcher));
@@ -1170,2 +840,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          header_matcher.name = UpbStringToStdString(
-              envoy_config_route_v3_HeaderMatcher_name(header));
+    header_matcher.name =
+        UpbStringToStdString(envoy_config_route_v3_HeaderMatcher_name(header));
@@ -1173,2 +843,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::EXACT;
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::EXACT;
@@ -1189,2 +859,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::REGEX;
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::REGEX;
@@ -1192,4 +862,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_range_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::RANGE;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_range_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::RANGE;
@@ -1200,2 +869 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            header_matcher.range_end =
-                envoy_type_v3_Int64Range_end(range_matcher);
+      header_matcher.range_end = envoy_type_v3_Int64Range_end(range_matcher);
@@ -1207,4 +875,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_present_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::PRESENT;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_present_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::PRESENT;
@@ -1213,4 +880,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_prefix_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::PREFIX;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_prefix_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::PREFIX;
@@ -1219,4 +885,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          } else if (envoy_config_route_v3_HeaderMatcher_has_suffix_match(
-                         header)) {
-            header_matcher.type = XdsApi::Route::Matchers::HeaderMatcher::
-                HeaderMatcherType::SUFFIX;
+    } else if (envoy_config_route_v3_HeaderMatcher_has_suffix_match(header)) {
+      header_matcher.type =
+          XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcherType::SUFFIX;
@@ -1231,2 +896 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          route->matchers.header_matchers.emplace_back(
-              std::move(header_matcher));
+    route->matchers.header_matchers.emplace_back(std::move(header_matcher));
@@ -1245,2 +909 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            uint32_t numerator =
-                envoy_type_v3_FractionalPercent_numerator(fraction);
+      uint32_t numerator = envoy_type_v3_FractionalPercent_numerator(fraction);
@@ -1289,2 +952 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_route_v3_WeightedCluster_total_weight(
-                  weighted_cluster);
+        envoy_config_route_v3_WeightedCluster_total_weight(weighted_cluster);
@@ -1295,3 +957,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_config_route_v3_WeightedCluster_ClusterWeight* const*
-              clusters = envoy_config_route_v3_WeightedCluster_clusters(
-                  weighted_cluster, &clusters_size);
+    const envoy_config_route_v3_WeightedCluster_ClusterWeight* const* clusters =
+        envoy_config_route_v3_WeightedCluster_clusters(weighted_cluster,
+                                                       &clusters_size);
@@ -1328,2 +990 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "RouteAction weighted_cluster has no valid clusters "
-                "specified.");
+          "RouteAction weighted_cluster has no valid clusters specified.");
@@ -1338 +999 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1341 +1002 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        MaybeLogRouteConfiguration(client, tracer, route_config);
+  MaybeLogRouteConfiguration(client, tracer, symtab, route_config);
@@ -1348,2 +1009 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsApi::RdsUpdate::VirtualHost& vhost =
-              rds_update->virtual_hosts.back();
+    XdsApi::RdsUpdate::VirtualHost& vhost = rds_update->virtual_hosts.back();
@@ -1351,3 +1011,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          upb_strview const* domains =
-              envoy_config_route_v3_VirtualHost_domains(virtual_hosts[i],
-                                                        &domain_size);
+    upb_strview const* domains = envoy_config_route_v3_VirtualHost_domains(
+        virtual_hosts[i], &domain_size);
@@ -1358,2 +1017 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                  "Invalid domain pattern.");
+        return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Invalid domain pattern.");
@@ -1364,2 +1022 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "VirtualHost has no domains");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("VirtualHost has no domains");
@@ -1369,2 +1026 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_route_v3_VirtualHost_routes(virtual_hosts[i],
-                                                       &num_routes);
+        envoy_config_route_v3_VirtualHost_routes(virtual_hosts[i], &num_routes);
@@ -1386,2 +1042 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            grpc_error* error =
-                RoutePathMatchParse(match, &route, &ignore_route);
+      grpc_error* error = RoutePathMatchParse(match, &route, &ignore_route);
@@ -1407,2 +1062 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "No valid routes specified.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("No valid routes specified.");
@@ -1414 +1068 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1420,2 +1074 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1431,2 +1084,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_listener_v3_Listener_parse(
-                  encoded_listener.data, encoded_listener.size, arena);
+        envoy_config_listener_v3_Listener_parse(encoded_listener.data,
+                                                encoded_listener.size, arena);
@@ -1434,2 +1087 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Can't decode listener.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode listener.");
@@ -1451,2 +1103 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                      encoded_api_listener.data, encoded_api_listener.size,
-                      arena);
+                encoded_api_listener.data, encoded_api_listener.size, arena);
@@ -1455,2 +1106 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "Could not parse HttpConnectionManager config from "
-                "ApiListener");
+          "Could not parse HttpConnectionManager config from ApiListener");
@@ -1465 +1115 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                RouteConfigParse(client, tracer, route_config, &rds_update);
+          RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1474,2 +1124 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "HttpConnectionManager neither has inlined route_config nor "
-                "RDS.");
+          "HttpConnectionManager neither has inlined route_config nor RDS.");
@@ -1477,2 +1126 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_extensions_filters_network_http_connection_manager_v3_Rds*
-              rds =
+    const envoy_extensions_filters_network_http_connection_manager_v3_Rds* rds =
@@ -1490,2 +1138 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                "HttpConnectionManager ConfigSource for RDS does not specify "
-                "ADS.");
+          "HttpConnectionManager ConfigSource for RDS does not specify ADS.");
@@ -1502 +1149 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1508,2 +1155 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1522,2 +1168 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Can't decode route_config.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode route_config.");
@@ -1532,2 +1177,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          grpc_error* error =
-              RouteConfigParse(client, tracer, route_config, &local_rds_update);
+    grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config,
+                                         &local_rds_update);
@@ -1541 +1186 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1547,2 +1192 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1556,2 +1200 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const upb_strview encoded_cluster =
-              google_protobuf_Any_value(resources[i]);
+    const upb_strview encoded_cluster = google_protobuf_Any_value(resources[i]);
@@ -1559,2 +1202,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_config_cluster_v3_Cluster_parse(
-                  encoded_cluster.data, encoded_cluster.size, arena);
+        envoy_config_cluster_v3_Cluster_parse(encoded_cluster.data,
+                                              encoded_cluster.size, arena);
@@ -1562,2 +1205 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Can't decode cluster.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode cluster.");
@@ -1565,3 +1207,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          MaybeLogCluster(client, tracer, cluster);
-          std::string cluster_name = UpbStringToStdString(
-              envoy_config_cluster_v3_Cluster_name(cluster));
+    MaybeLogCluster(client, tracer, symtab, cluster);
+    std::string cluster_name =
+        UpbStringToStdString(envoy_config_cluster_v3_Cluster_name(cluster));
@@ -1578,2 +1220 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "DiscoveryType not found.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("DiscoveryType not found.");
@@ -1583,2 +1224 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "DiscoveryType is not EDS.");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("DiscoveryType is not EDS.");
@@ -1586,2 +1226 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_config_cluster_v3_Cluster_EdsClusterConfig*
-              eds_cluster_config =
+    const envoy_config_cluster_v3_Cluster_EdsClusterConfig* eds_cluster_config =
@@ -1616,2 +1255 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          cds_update_map->emplace(std::move(cluster_name),
-                                  std::move(cds_update));
+    cds_update_map->emplace(std::move(cluster_name), std::move(cds_update));
@@ -1638,2 +1276 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        uint32_t port =
-            envoy_config_core_v3_SocketAddress_port_value(socket_address);
+  uint32_t port = envoy_config_core_v3_SocketAddress_port_value(socket_address);
@@ -1649,2 +1286 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const envoy_config_endpoint_v3_LocalityLbEndpoints*
-              locality_lb_endpoints,
+    const envoy_config_endpoint_v3_LocalityLbEndpoints* locality_lb_endpoints,
@@ -1656,2 +1292 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            lb_weight != nullptr ? google_protobuf_UInt32Value_value(lb_weight)
-                                 : 0;
+      lb_weight != nullptr ? google_protobuf_UInt32Value_value(lb_weight) : 0;
@@ -1662,6 +1297,6 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-        std::string region = UpbStringToStdString(
-            envoy_config_core_v3_Locality_region(locality));
-        std::string zone = UpbStringToStdString(
-            envoy_config_core_v3_Locality_region(locality));
-        std::string sub_zone = UpbStringToStdString(
-            envoy_config_core_v3_Locality_sub_zone(locality));
+  std::string region =
+      UpbStringToStdString(envoy_config_core_v3_Locality_region(locality));
+  std::string zone =
+      UpbStringToStdString(envoy_config_core_v3_Locality_region(locality));
+  std::string sub_zone =
+      UpbStringToStdString(envoy_config_core_v3_Locality_sub_zone(locality));
@@ -1692,2 +1327 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-              "Empty drop category name");
+    return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Empty drop category name");
@@ -1713,2 +1347 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-                "Unknown denominator type");
+      return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Unknown denominator type");
@@ -1721 +1354 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          XdsClient* client, TraceFlag* tracer,
+    XdsClient* client, TraceFlag* tracer, upb_symtab* symtab,
@@ -1727,2 +1360 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_resources(response,
-                                                                   &size);
+      envoy_service_discovery_v3_DiscoveryResponse_resources(response, &size);
@@ -1738,2 +1370 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          envoy_config_endpoint_v3_ClusterLoadAssignment*
-              cluster_load_assignment =
+    envoy_config_endpoint_v3_ClusterLoadAssignment* cluster_load_assignment =
@@ -1747 +1378 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          MaybeLogClusterLoadAssignment(client, tracer,
+    MaybeLogClusterLoadAssignment(client, tracer, symtab,
@@ -1758,2 +1389 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                absl::StrCat("duplicate resource name \"", eds_service_name,
-                             "\"")
+          absl::StrCat("duplicate resource name \"", eds_service_name, "\"")
@@ -1792,2 +1422,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              grpc_error* error = DropParseAndAppend(
-                  drop_overload[j], eds_update.drop_config.get());
+        grpc_error* error =
+            DropParseAndAppend(drop_overload[j], eds_update.drop_config.get());
@@ -1797,2 +1427 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          eds_update_map->emplace(std::move(eds_service_name),
-                                  std::move(eds_update));
+    eds_update_map->emplace(std::move(eds_service_name), std::move(eds_update));
@@ -1816,2 +1445 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          const grpc_slice& encoded_response,
-          const std::string& expected_server_name,
+    const grpc_slice& encoded_response, const std::string& expected_server_name,
@@ -1825,2 +1453 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                reinterpret_cast<const char*>(
-                    GRPC_SLICE_START_PTR(encoded_response)),
+          reinterpret_cast<const char*>(GRPC_SLICE_START_PTR(encoded_response)),
@@ -1829,2 +1456,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = GRPC_ERROR_CREATE_FROM_STATIC_STRING(
-              "Can't decode DiscoveryResponse.");
+    result.parse_error =
+        GRPC_ERROR_CREATE_FROM_STATIC_STRING("Can't decode DiscoveryResponse.");
@@ -1837,2 +1464 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_discovery_v3_DiscoveryResponse_version_info(
-                response));
+      envoy_service_discovery_v3_DiscoveryResponse_version_info(response));
@@ -1842,10 +1467,0 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = LdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response, expected_server_name,
-              &result.lds_update, arena.ptr());
-        } else if (IsRds(result.type_url)) {
-<<<<<<< HEAD
-          result.parse_error = RdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response, expected_server_name,
-              expected_route_configuration_names, &result.rds_update,
-              arena.ptr());
-|||||||
@@ -1853,4 +1469,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              RdsResponseParse(client_, tracer_, response, expected_server_name,
-                               expected_route_configuration_names,
-                               &result.rds_update, arena.ptr());
-=======
+        LdsResponseParse(client_, tracer_, symtab_.ptr(), response,
+                         expected_server_name, &result.lds_update, arena.ptr());
+  } else if (IsRds(result.type_url)) {
@@ -1858,3 +1473,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              client_, tracer_, response, expected_route_configuration_names,
-              &result.rds_update, arena.ptr());
->>>>>>> d109f4b4064982d6770cd8776a1dd58f4c3935a4
+        client_, tracer_, symtab_.ptr(), response,
+        expected_route_configuration_names, &result.rds_update, arena.ptr());
@@ -1862,2 +1476,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = CdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response, expected_cluster_names,
+    result.parse_error = CdsResponseParse(client_, tracer_, symtab_.ptr(),
+                                          response, expected_cluster_names,
@@ -1866,3 +1480,3 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          result.parse_error = EdsResponseParse(
-              client_, tracer_, symtab_.ptr(), response,
-              expected_eds_service_names, &result.eds_update_map, arena.ptr());
+    result.parse_error = EdsResponseParse(client_, tracer_, symtab_.ptr(),
+                                          response, expected_eds_service_names,
+                                          &result.eds_update_map, arena.ptr());
@@ -1882,2 +1496,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_log(GPR_DEBUG, "[xds_client %p] constructed LRS request: %s",
-                  client, buf);
+    gpr_log(GPR_DEBUG, "[xds_client %p] constructed LRS request: %s", client,
+            buf);
@@ -1895,2 +1509 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      grpc_slice XdsApi::CreateLrsInitialRequest(
-          const std::string& server_name) {
+grpc_slice XdsApi::CreateLrsInitialRequest(const std::string& server_name) {
@@ -1901,2 +1514,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_service_load_stats_v3_LoadStatsRequest_mutable_node(
-                request, arena.ptr());
+      envoy_service_load_stats_v3_LoadStatsRequest_mutable_node(request,
+                                                                arena.ptr());
@@ -1917,2 +1530,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_config_endpoint_v3_UpstreamLocalityStats_mutable_locality(
-                output, arena);
+      envoy_config_endpoint_v3_UpstreamLocalityStats_mutable_locality(output,
+                                                                      arena);
@@ -1985,2 +1598 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-            envoy_config_endpoint_v3_ClusterStats_DroppedRequests*
-                dropped_requests =
+      envoy_config_endpoint_v3_ClusterStats_DroppedRequests* dropped_requests =
@@ -1997,2 +1609,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          gpr_timespec timespec = grpc_millis_to_timespec(
-              load_report.load_report_interval, GPR_TIMESPAN);
+    gpr_timespec timespec =
+        grpc_millis_to_timespec(load_report.load_report_interval, GPR_TIMESPAN);
@@ -2002,4 +1614,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-          google_protobuf_Duration_set_seconds(load_report_interval,
-                                               timespec.tv_sec);
-          google_protobuf_Duration_set_nanos(load_report_interval,
-                                             timespec.tv_nsec);
+    google_protobuf_Duration_set_seconds(load_report_interval, timespec.tv_sec);
+    google_protobuf_Duration_set_nanos(load_report_interval, timespec.tv_nsec);
@@ -2010,2 +1620,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-      grpc_error* XdsApi::ParseLrsResponse(
-          const grpc_slice& encoded_response, bool* send_all_clusters,
+grpc_error* XdsApi::ParseLrsResponse(const grpc_slice& encoded_response,
+                                     bool* send_all_clusters,
@@ -2017,2 +1627 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-                reinterpret_cast<const char*>(
-                    GRPC_SLICE_START_PTR(encoded_response)),
+          reinterpret_cast<const char*>(GRPC_SLICE_START_PTR(encoded_response)),
@@ -2029,2 +1638,2 @@ XdsApi::Route::Matchers::HeaderMatcher::HeaderMatcher(
-              envoy_service_load_stats_v3_LoadStatsResponse_clusters(
-                  decoded_response, &size);
+        envoy_service_load_stats_v3_LoadStatsResponse_clusters(decoded_response,
+                                                               &size);
