diff --git a/./output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_mergegen.cc b/./output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_merged.cc
index b17c988..db0b30c 100644
--- a/./output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_mergegen.cc
+++ b/./output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_merged.cc
@@ -1001 +1001,2 @@ grpc_error* RouteConfigParse(
-size_t size; envoy_config_route_v3_RouteConfiguration* route_config, XdsApi::RdsUpdate* rds_update) { MaybeLogRouteConfiguration(client, tracer, symtab, route_config);
+    XdsApi::RdsUpdate* rds_update) {
+  MaybeLogRouteConfiguration(client, tracer, symtab, route_config);
@@ -1113 +1114,2 @@ grpc_error* LdsResponseParse(
-grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config, &rds_update); if (error != GRPC_ERROR_NONE) return error;
+      grpc_error* error =
+          RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1175 +1177,2 @@ grpc_error* RdsResponseParse(
-grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config, &local_rds_update); if (error != GRPC_ERROR_NONE) return error;
+    grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config,
+                                         &local_rds_update);
@@ -1375 +1378,3 @@ grpc_error* EdsResponseParse(
-classLogClusterLoadAssignment(client, tracer, symtab, cluster_load_assignment);
+    MaybeLogClusterLoadAssignment(client, tracer, symtab,
+                                  cluster_load_assignment);
+    std::string eds_service_name = UpbStringToStdString(
@@ -1467 +1472,3 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-} else if (IsCds(result.type_url)) {
+    result.parse_error = RdsResponseParse(
+        client_, tracer_, symtab_.ptr(), response,
+        expected_route_configuration_names, &result.rds_update, arena.ptr());
