diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odr_config.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odr_config.no_comments_merged.h
index 220f479..00df6c1 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odr_config.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odr_config.no_comments_merged.h
@@ -38 +37,0 @@ const android::base::NoDestructor<std::vector<SystemPropertyConfig>> kSystemProp
-<<<<<<< HEAD
@@ -43,9 +41,0 @@ const android::base::NoDestructor<std::vector<SystemPropertyConfig>> kSystemProp
-|||||||
-                          .default_value = "false"},
-     SystemPropertyConfig{.name = kPhDisableCompactDex, .default_value = "false"}}};
-=======
-                          .default_value = "false"},
-     SystemPropertyConfig{.name = kPhDisableCompactDex, .default_value = "false"},
-     SystemPropertyConfig{.name = kSystemPropertySystemServerCompilerFilterOverride,
-                          .default_value = ""}}};
->>>>>>> af3127b30b1b93a728eb1461ce697d182f8a731a
@@ -85 +75,2 @@ class OdrConfig final {
-        artifact_dir_(GetApexDataDalvikCacheDirectory(InstructionSet::kNone)) {}
+      artifact_dir_(GetApexDataDalvikCacheDirectory(InstructionSet::kNone)) {
+  }
@@ -130,5 +121,15 @@ class OdrConfig final {
-  bool HasPartialCompilation() const { return partial_compilation_.has_value(); }
-  bool GetPartialCompilation() const { return partial_compilation_.value_or(true); }
-  bool GetRefresh() const { return refresh_.value_or(true); }
-  const std::string& GetSystemServerClasspath() const { return system_server_classpath_; }
-  const std::string& GetBootImageCompilerFilter() const { return boot_image_compiler_filter_; }
+  bool HasPartialCompilation() const {
+    return partial_compilation_.has_value();
+  }
+  bool GetPartialCompilation() const {
+    return partial_compilation_.value_or(true);
+  }
+  bool GetRefresh() const {
+    return refresh_.value_or(true);
+  }
+  const std::string& GetSystemServerClasspath() const {
+    return system_server_classpath_;
+  }
+  const std::string& GetBootImageCompilerFilter() const {
+    return boot_image_compiler_filter_;
+  }
@@ -138 +139,3 @@ class OdrConfig final {
-  const std::string& GetStagingDir() const { return staging_dir_; }
+  const std::string& GetStagingDir() const {
+    return staging_dir_;
+  }
@@ -149 +152,3 @@ class OdrConfig final {
-  void SetArtifactDirectory(const std::string& artifact_dir) { artifact_dir_ = artifact_dir; }
+  void SetArtifactDirectory(const std::string& artifact_dir) {
+    artifact_dir_ = artifact_dir;
+  }
@@ -151,2 +156,6 @@ class OdrConfig final {
-  void SetPartialCompilation(bool value) { partial_compilation_ = value; }
-  void SetRefresh(bool value) { refresh_ = value; }
+  void SetPartialCompilation(bool value) {
+    partial_compilation_ = value;
+  }
+  void SetRefresh(bool value) {
+    refresh_ = value;
+  }
@@ -166 +175,3 @@ class OdrConfig final {
-  void SetStagingDir(const std::string& staging_dir) { staging_dir_ = staging_dir; }
+  void SetStagingDir(const std::string& staging_dir) {
+    staging_dir_ = staging_dir;
+  }
@@ -195 +206,3 @@ class OdrConfig final {
-  OdrConfig() OdrConfig(const OdrConfig&) OdrConfig& operator=(const OdrConfig&) = delete
+  OdrConfig() = delete;
+  OdrConfig(const OdrConfig&) = delete;
+  OdrConfig& operator=(const OdrConfig&) = delete;
