diff --git a/./output/frameworks_native/acb53f92-c0ed405f-1ce0f4aa/services@inputflinger@PointerChoreographer.no_comments_mergegen.cpp b/./output/frameworks_native/acb53f92-c0ed405f-1ce0f4aa/services@inputflinger@PointerChoreographer.no_comments_merged.cpp
index 1b40b07..b874430 100644
--- a/./output/frameworks_native/acb53f92-c0ed405f-1ce0f4aa/services@inputflinger@PointerChoreographer.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/acb53f92-c0ed405f-1ce0f4aa/services@inputflinger@PointerChoreographer.no_comments_merged.cpp
@@ -82 +82,3 @@ NotifyMotionArgs PointerChoreographer::processMouseEventLocked(const NotifyMotio
-auto [displayId, pc] = ensureMouseControllerLocked(args.displayId); NotifyMotionArgs newArgs(args); newArgs.displayId = displayId;
+    auto [displayId, pc] = ensureMouseControllerLocked(args.displayId);
+    NotifyMotionArgs newArgs(args);
+    newArgs.displayId = displayId;
