--- a/./output/frameworks_native/acb53f92-c0ed405f-1ce0f4aa/services@inputflinger@PointerChoreographer.no_comments_conflict.cpp
+++ b/./output/frameworks_native/acb53f92-c0ed405f-1ce0f4aa/services@inputflinger@PointerChoreographer.no_comments_merged.cpp
@@ -82 +81,0 @@ NotifyMotionArgs PointerChoreographer::processMouseEventLocked(const NotifyMotio
-<<<<<<< HEAD
@@ -86,5 +84,0 @@ NotifyMotionArgs PointerChoreographer::processMouseEventLocked(const NotifyMotio
-||||||| 1ce0f4aab8
-    auto [displayId, pc] = getDisplayIdAndMouseControllerLocked(args.displayId);
-=======
-    auto [displayId, pc] = ensureMouseControllerLocked(args.displayId);
->>>>>>> c0ed405f
