--- a/./output/frameworks_native/03a8ff5c-3b468eca-b00cdc55/services@surfaceflinger@SurfaceFlinger.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/03a8ff5c-3b468eca-b00cdc55/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
@@ -1950 +1950,5 @@ bool SurfaceFlinger::updateLayerSnapshots(VsyncId vsyncId, nsecs_t frameTimeNs,
-ATRACE_NAME("DisplayCallbackAndStatsUpdates"); mustComposite |= applyTransactionsLocked(update.transactions, vsyncId); traverseLegacyLayers([&](Layer* layer) { layer->commitTransaction(); });
+    ATRACE_NAME("DisplayCallbackAndStatsUpdates");
+    mustComposite |= applyTransactionsLocked(update.transactions, vsyncId);
+    traverseLegacyLayers([&](Layer* layer) { layer->commitTransaction(); });
+    const nsecs_t latchTime = systemTime();
+    bool unused = false;
@@ -4851 +4855,3 @@ void SurfaceFlinger::initializeDisplays() {
-{ Mutex::Autolock lock(mStateLock);
+    {
+        Mutex::Autolock lock(mStateLock);
+        applyAndCommitDisplayTransactionStatesLocked(transactions);
