diff --git a/./output/system_core/151677dc-ba36715a-83cda925/lmkd@include@lmkd.no_comments_conflict.h b/./output/system_core/151677dc-ba36715a-83cda925/lmkd@include@lmkd.no_comments_merged.h
index b750745..f3b07ca 100644
--- a/./output/system_core/151677dc-ba36715a-83cda925/lmkd@include@lmkd.no_comments_conflict.h
+++ b/./output/system_core/151677dc-ba36715a-83cda925/lmkd@include@lmkd.no_comments_merged.h
@@ -11 +10,0 @@ enum lmk_cmd {
-<<<<<<< HEAD
@@ -14,4 +12,0 @@ enum lmk_cmd {
-||||||| 83cda925e
-=======
-    LMK_PROCPURGE,
->>>>>>> ba36715a
@@ -79 +73,0 @@ inline size_t lmkd_pack_set_procremove(LMKD_CTRL_PACKET packet,
-<<<<<<< HEAD
@@ -105,7 +98,0 @@ inline size_t lmkd_pack_set_getkillcnt_repl(LMKD_CTRL_PACKET packet, int kill_cn
-||||||| 83cda925e
-=======
-inline size_t lmkd_pack_set_procpurge(LMKD_CTRL_PACKET packet) {
-    packet[0] = htonl(LMK_PROCPURGE);
-    return sizeof(int);
-}
->>>>>>> ba36715a
