--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/151677dc-ba36715a-83cda925/lmkd@include@lmkd.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/151677dc-ba36715a-83cda925/lmkd@include@lmkd.no_comments_merged.h
@@ -11 +10,0 @@ enum lmk_cmd {
-<<<<<<< HEAD
@@ -14,4 +12,0 @@ enum lmk_cmd {
-|||||||
-=======
-  LMK_PROCPURGE,
->>>>>>> ba36715afc3f79ce9f954486373e7c4da32bae19
@@ -104,0 +99 @@ inline size_t lmkd_pack_set_getkillcnt_repl(LMKD_CTRL_PACKET packet,
+__END_DECLS
