--- a/./output/grpc/8e5d8e64-59dc7fb2-e758ccaf/src@cpp@server@server_builder.no_comments_conflict.cc
+++ b/./output/grpc/8e5d8e64-59dc7fb2-e758ccaf/src@cpp@server@server_builder.no_comments_merged.cc
@@ -156,5 +155,0 @@ ServerBuilder& ServerBuilder::SetResourceQuota(
-<<<<<<< HEAD
-    const ResourceQuota& resource_quota) {
-||||||| e758ccaff0
-    const grpc_impl::ResourceQuota& resource_quota) {
-=======
@@ -162 +156,0 @@ ServerBuilder& ServerBuilder::SetResourceQuota(
->>>>>>> 59dc7fb2
