--- a/home/whalien/codebase/python/mergebot-eva/output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_merged.c
@@ -10 +9,0 @@
-#include "anet.h"
@@ -12 +10,0 @@
-#include "adlist.h"
@@ -14 +11,0 @@
-#include "help.h"
@@ -17,0 +15 @@
+#include "help.h"
@@ -181,7 +183,0 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-<<<<<<< HEAD
-    showInteractiveHelp();
-    return REDIS_OK;
-|||||||
-    showInteractiveHelp();
-    return 0;
-=======
@@ -189,2 +185 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-    return 0;
->>>>>>> 5397f2b596b5189edbed3e45a42d18f3c99341d6
+        return REDIS_OK;
@@ -242,4 +239,3 @@ static int parseOptions(int argc, char **argv) {
-              "Starting interactive mode using -i is deprecated. Interactive "
-              "mode is started\n"
-              "by default when redis-cli is executed without a command to "
-              "execute.\n");
+"Starting interactive mode using -i is deprecated. Interactive mode is started\n"
+"by default when redis-cli is executed without a command to execute.\n"
+            );
@@ -278,7 +272,2 @@ static void usage() {
-  fprintf(
-      stderr,
-      "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a "
-      "authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... "
-          "arg(N-1)\n\n");
+    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
+    fprintf(stderr, "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... arg(N-1)\n\n");
@@ -288,3 +277 @@ static void usage() {
-  fprintf(stderr,
-          "\nRun in interactive mode: redis-cli -i or just don't pass any "
-          "command\n");
+    fprintf(stderr, "\nRun in interactive mode: redis-cli -i or just don't pass any command\n");
