diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/d9319aaa-b16ce8d6-f316a028/src@core@lib@gpr@string.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/grpc/d9319aaa-b16ce8d6-f316a028/src@core@lib@gpr@string.no_comments_merged.h
index 72f117d..90bf9ce 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/d9319aaa-b16ce8d6-f316a028/src@core@lib@gpr@string.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/d9319aaa-b16ce8d6-f316a028/src@core@lib@gpr@string.no_comments_merged.h
@@ -27,6 +26,0 @@ char* gpr_format_timespec(gpr_timespec);
-struct gpr_strvec {
-  char** strs;
-  size_t count;
-  size_t capacity;
-};
-void gpr_strvec_init(gpr_strvec* strs);
@@ -37 +30,0 @@ bool gpr_parse_bool_value(const char* value, bool* dst);
-void gpr_strvec_add(gpr_strvec* strs, char* add);
