diff --git a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_mergegen.cpp b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_merged.cpp
index 2cdd36a..7e54dc5 100644
--- a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_mergegen.cpp
+++ b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_merged.cpp
@@ -1042 +1042,3 @@ unique_ptr<GlobalSourceState> PhysicalWindow::GetGlobalSourceState(ClientContext
-jResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const { auto &lsource = input.local_state.Cast<WindowLocalSourceState>(); auto &gsource = input.global_state.Cast<WindowGlobalSourceState>();
+SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const {
+ auto &lsource = input.local_state.Cast<WindowLocalSourceState>();
+ auto &gsource = input.global_state.Cast<WindowGlobalSourceState>();
