diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_mergebot.hpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_merged.hpp
index ef817a0..e481974 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_merged.hpp
@@ -36 +35,0 @@ struct ClientConfig {
- bool force_async_pipelines = true;
@@ -37,0 +37 @@ struct ClientConfig {
+ bool force_async_pipelines = true;
