--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_merged.cpp
@@ -25 +23,0 @@ SourceResultType PhysicalColumnDataScan::GetData(ExecutionContext &context, Data
- D_ASSERT(collection);
