diff --git a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_conflict.cpp b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_merged.cpp
index 5fe4a7e..7e54dc5 100644
--- a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_conflict.cpp
+++ b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_merged.cpp
@@ -1042 +1041,0 @@ unique_ptr<GlobalSourceState> PhysicalWindow::GetGlobalSourceState(ClientContext
-<<<<<<< HEAD
@@ -1045 +1043,0 @@ SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &c
- auto &lpsource = lsource.partition_source;
@@ -1047,14 +1044,0 @@ SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &c
- auto &gpsource = gsource.partition_source;
-||||||| e8610c85fb
-void PhysicalWindow::GetData(ExecutionContext &context, DataChunk &chunk, GlobalSourceState &gstate_p,
-                             LocalSourceState &lstate_p) const {
- auto &lsource = lstate_p.Cast<WindowLocalSourceState>();
- auto &lpsource = lsource.partition_source;
- auto &gsource = gstate_p.Cast<WindowGlobalSourceState>();
- auto &gpsource = gsource.partition_source;
-=======
-void PhysicalWindow::GetData(ExecutionContext &context, DataChunk &chunk, GlobalSourceState &gstate_p,
-                             LocalSourceState &lstate_p) const {
- auto &lsource = lstate_p.Cast<WindowLocalSourceState>();
- auto &gsource = gstate_p.Cast<WindowGlobalSourceState>();
->>>>>>> 65cfcd58
