diff --git a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_conflict.hpp b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_merged.hpp
index df1dfa7..e481974 100644
--- a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_conflict.hpp
+++ b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_merged.hpp
@@ -36,4 +35,0 @@ struct ClientConfig {
-<<<<<<< HEAD
- bool force_async_pipelines = true;
-||||||| e8610c85fb
-=======
@@ -41 +37 @@ struct ClientConfig {
->>>>>>> 65cfcd58
+ bool force_async_pipelines = true;
