diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_merged.cpp
index 39ae75e..86f2d7e 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_merged.cpp
@@ -22,2 +22 @@ unique_ptr<GlobalSourceState> PhysicalColumnDataScan::GetGlobalSourceState(Clien
-SourceResultType PhysicalColumnDataScan::GetData(ExecutionContext &context, DataChunk &chunk,
-                                                 OperatorSourceInput &input) const {
+SourceResultType PhysicalColumnDataScan::GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const {
@@ -25 +23,0 @@ SourceResultType PhysicalColumnDataScan::GetData(ExecutionContext &context, Data
- D_ASSERT(collection);
