--- a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@common@enum_util.no_comments_mergegen.cpp
+++ b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@common@enum_util.no_comments_merged.cpp
@@ -500 +500,3 @@ const StringUtil::EnumStringLiteral *GetCSVStateValues() {
-  { static_cast<uint32_t>(CSVState::STANDARD_NEWLINE), "STANDARD_NEWLINE" }
+  { static_cast<uint32_t>(CSVState::STANDARD_NEWLINE), "STANDARD_NEWLINE" },
+    { static_cast<uint32_t>(CSVState::UNQUOTED_ESCAPE), "UNQUOTED_ESCAPE" },
+    { static_cast<uint32_t>(CSVState::ESCAPED_RETURN), "ESCAPED_RETURN" }
@@ -510 +512,9 @@ CSVState EnumUtil::FromString<CSVState>(const char *value) {
-const StringUtil::EnumStringLiteral *GetCTEMaterializeValues() { static constexpr StringUtil::EnumStringLiteral values[] { { static_cast<uint32_t>(CTEMaterialize::CTE_MATERIALIZE_DEFAULT), "CTE_MATERIALIZE_DEFAULT" }, { static_cast<uint32_t>(CTEMaterialize::CTE_MATERIALIZE_ALWAYS), "CTE_MATERIALIZE_ALWAYS" }, { static_cast<uint32_t>(CTEMaterialize::CTE_MATERIALIZE_NEVER), "CTE_MATERIALIZE_NEVER" } }; return CSVState::COMMENT; } if (StringUtil::Equals(value, "UNQUOTED_ESCAPE")) { return CSVState::UNQUOTED_ESCAPE; } if (StringUtil::Equals(value, "ESCAPED_RETURN")) { return CSVState::ESCAPED_RETURN; } throw NotImplementedException(StringUtil::Format("Enum value: '%s' not implemented in FromString<CSVState>", value));
+ return static_cast<CSVState>(StringUtil::StringToEnum(GetCSVStateValues(), 13, "CSVState", value));
+}
+const StringUtil::EnumStringLiteral *GetCTEMaterializeValues() {
+ static constexpr StringUtil::EnumStringLiteral values[] {
+  { static_cast<uint32_t>(CTEMaterialize::CTE_MATERIALIZE_DEFAULT), "CTE_MATERIALIZE_DEFAULT" },
+  { static_cast<uint32_t>(CTEMaterialize::CTE_MATERIALIZE_ALWAYS), "CTE_MATERIALIZE_ALWAYS" },
+  { static_cast<uint32_t>(CTEMaterialize::CTE_MATERIALIZE_NEVER), "CTE_MATERIALIZE_NEVER" }
+ };
+ return values;
