diff --git a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state_machine.no_comments_mergegen.hpp b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state_machine.no_comments_merged.hpp
index 3bcd317..4e473e1 100644
--- a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state_machine.no_comments_mergegen.hpp
+++ b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state_machine.no_comments_merged.hpp
@@ -49 +49,11 @@ struct CSVStates {
-inline bool IsEscaped() const { switch (states[1]) { case CSVState::ESCAPE: case CSVState::UNQUOTED_ESCAPE: case CSVState::ESCAPED_RETURN: return true; case CSVState::QUOTED: return states[0] == CSVState::UNQUOTED; default: return false; } }
+ inline bool IsEscaped() const {
+  switch (states[1]) {
+  case CSVState::ESCAPE:
+  case CSVState::UNQUOTED_ESCAPE:
+  case CSVState::ESCAPED_RETURN:
+   return true;
+  case CSVState::QUOTED:
+   return states[0] == CSVState::UNQUOTED;
+  default:
+   return false;
+  }
