--- a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state_machine.no_comments_conflict.hpp
+++ b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state_machine.no_comments_merged.hpp
@@ -49,2 +49 @@ struct CSVStates {
-<<<<<<< HEAD
- inline bool IsEscaped() {
+ inline bool IsEscaped() const {
@@ -61,7 +59,0 @@ struct CSVStates {
-||||||| d4c7e729ac
- inline bool IsEscaped() {
-  return states[1] == CSVState::ESCAPE || (states[0] == CSVState::UNQUOTED && states[1] == CSVState::QUOTED);
-=======
- inline bool IsEscaped() const {
-  return states[1] == CSVState::ESCAPE || (states[0] == CSVState::UNQUOTED && states[1] == CSVState::QUOTED);
->>>>>>> 8ce36237
