diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/57fe03dc-574a9e4b-aa7dccf8/window-buffer.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/57fe03dc-574a9e4b-aa7dccf8/window-buffer.no_comments_merged.c
index c10c72a..f2b2fbf 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/57fe03dc-574a9e4b-aa7dccf8/window-buffer.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/57fe03dc-574a9e4b-aa7dccf8/window-buffer.no_comments_merged.c
@@ -6 +5,0 @@
-#include <vis.h>
@@ -10,2 +9 @@ static struct screen *window_buffer_init(struct window_mode_entry *,
-                                         struct cmd_find_state *,
-                                         struct args *);
+        struct cmd_find_state *, struct args *);
@@ -13,4 +11,5 @@ static void window_buffer_free(struct window_mode_entry *);
-static void window_buffer_resize(struct window_mode_entry *, u_int, u_int);
-static void window_buffer_key(struct window_mode_entry *, struct client *,
-                              struct session *, struct winlink *, key_code,
-                              struct mouse_event *);
+static void window_buffer_resize(struct window_mode_entry *, u_int,
+        u_int);
+static void window_buffer_key(struct window_mode_entry *,
+        struct client *, struct session *,
+        struct winlink *, key_code, struct mouse_event *);
@@ -18 +17,2 @@ static void window_buffer_key(struct window_mode_entry *, struct client *,
-#define WINDOW_BUFFER_DEFAULT_FORMAT "#{t/p:buffer_created}: #{buffer_sample}"
+#define WINDOW_BUFFER_DEFAULT_FORMAT \
+ "#{t/p:buffer_created}: #{buffer_sample}"
@@ -20,5 +20,10 @@ static const struct menu_item window_buffer_menu_items[] = {
-    {"Paste", 'p', NULL}, {"Paste Tagged", 'P', NULL},
-    {"", KEYC_NONE, NULL}, {"Tag", 't', NULL},
-    {"Tag All", '\024', NULL}, {"Tag None", 'T', NULL},
-    {"", KEYC_NONE, NULL}, {"Delete", 'd', NULL},
-    {"Delete Tagged", 'D', NULL}, {"", KEYC_NONE, NULL},
+ { "Paste", 'p', NULL },
+ { "Paste Tagged", 'P', NULL },
+ { "", KEYC_NONE, NULL },
+ { "Tag", 't', NULL },
+ { "Tag All", '\024', NULL },
+ { "Tag None", 'T', NULL },
+ { "", KEYC_NONE, NULL },
+ { "Delete", 'd', NULL },
+ { "Delete Tagged", 'D', NULL },
+ { "", KEYC_NONE, NULL },
@@ -26 +31,2 @@ static const struct menu_item window_buffer_menu_items[] = {
-    {NULL, KEYC_NONE, NULL}};
+ { NULL, KEYC_NONE, NULL }
+};
@@ -40 +46,5 @@ enum window_buffer_sort_type {
-static const char *window_buffer_sort_list[] = {"time", "name", "size"};
+static const char *window_buffer_sort_list[] = {
+ "time",
+ "name",
+ "size"
+};
@@ -61,2 +71,3 @@ struct window_buffer_editdata {
-static struct window_buffer_itemdata *window_buffer_add_item(
-    struct window_buffer_modedata *data) {
+static struct window_buffer_itemdata *
+window_buffer_add_item(struct window_buffer_modedata *data)
+{
@@ -69 +80,3 @@ static struct window_buffer_itemdata *window_buffer_add_item(
-static void window_buffer_free_item(struct window_buffer_itemdata *item) {
+static void
+window_buffer_free_item(struct window_buffer_itemdata *item)
+{
@@ -73 +86,3 @@ static void window_buffer_free_item(struct window_buffer_itemdata *item) {
-static int window_buffer_cmp(const void *a0, const void *b0) {
+static int
+window_buffer_cmp(const void *a0, const void *b0)
+{
@@ -81,2 +96,4 @@ static int window_buffer_cmp(const void *a0, const void *b0) {
-  if (result == 0) result = strcmp((*a)->name, (*b)->name);
-  if (window_buffer_sort->reversed) result = -result;
+ if (result == 0)
+  result = strcmp((*a)->name, (*b)->name);
+ if (window_buffer_sort->reversed)
+  result = -result;
@@ -85,3 +102,4 @@ static int window_buffer_cmp(const void *a0, const void *b0) {
-static void window_buffer_build(void *modedata,
-                                struct mode_tree_sort_criteria *sort_crit,
-                                __unused uint64_t *tag, const char *filter) {
+static void
+window_buffer_build(void *modedata, struct mode_tree_sort_criteria *sort_crit,
+    __unused uint64_t *tag, const char *filter)
+{
@@ -120 +138,2 @@ static void window_buffer_build(void *modedata,
-    if (pb == NULL) continue;
+  if (pb == NULL)
+   continue;
@@ -134 +153,2 @@ static void window_buffer_build(void *modedata,
-    mode_tree_add(data->data, NULL, item, item->order, item->name, text, -1);
+  mode_tree_add(data->data, NULL, item, item->order, item->name,
+      text, -1);
@@ -139,3 +159,4 @@ static void window_buffer_build(void *modedata,
-static void window_buffer_draw(__unused void *modedata, void *itemdata,
-                               struct screen_write_ctx *ctx, u_int sx,
-                               u_int sy) {
+static void
+window_buffer_draw(__unused void *modedata, void *itemdata,
+    struct screen_write_ctx *ctx, u_int sx, u_int sy)
+{
@@ -149 +170,2 @@ static void window_buffer_draw(__unused void *modedata, void *itemdata,
-  if (pb == NULL) return;
+ if (pb == NULL)
+  return;
@@ -153 +175,2 @@ static void window_buffer_draw(__unused void *modedata, void *itemdata,
-    while (end != pdata + psize && *end != '\n') end++;
+  while (end != pdata + psize && *end != '\n')
+   end++;
@@ -158 +181,2 @@ static void window_buffer_draw(__unused void *modedata, void *itemdata,
-      screen_write_nputs(ctx, sx, &grid_default_cell, "%s", buf);
+   screen_write_nputs(ctx, sx, &grid_default_cell, "%s",
+       buf);
@@ -160 +184,2 @@ static void window_buffer_draw(__unused void *modedata, void *itemdata,
-    if (end == pdata + psize) break;
+  if (end == pdata + psize)
+   break;
@@ -165,2 +190,3 @@ static void window_buffer_draw(__unused void *modedata, void *itemdata,
-static int window_buffer_search(__unused void *modedata, void *itemdata,
-                                const char *ss) {
+static int
+window_buffer_search(__unused void *modedata, void *itemdata, const char *ss)
+{
@@ -171,2 +197,4 @@ static int window_buffer_search(__unused void *modedata, void *itemdata,
-  if ((pb = paste_get_name(item->name)) == NULL) return (0);
-  if (strstr(item->name, ss) != NULL) return (1);
+ if ((pb = paste_get_name(item->name)) == NULL)
+  return (0);
+ if (strstr(item->name, ss) != NULL)
+  return (1);
@@ -176 +204,3 @@ static int window_buffer_search(__unused void *modedata, void *itemdata,
-static void window_buffer_menu(void *modedata, struct client *c, key_code key) {
+static void
+window_buffer_menu(void *modedata, struct client *c, key_code key)
+{
@@ -181 +211,2 @@ static void window_buffer_menu(void *modedata, struct client *c, key_code key) {
-  if (wme == NULL || wme->data != modedata) return;
+ if (wme == NULL || wme->data != modedata)
+  return;
@@ -184,3 +215,4 @@ static void window_buffer_menu(void *modedata, struct client *c, key_code key) {
-static struct screen *window_buffer_init(struct window_mode_entry *wme,
-                                         struct cmd_find_state *fs,
-                                         struct args *args) {
+static struct screen *
+window_buffer_init(struct window_mode_entry *wme, struct cmd_find_state *fs,
+    struct args *args)
+{
@@ -201,4 +233,4 @@ static struct screen *window_buffer_init(struct window_mode_entry *wme,
-  data->data = mode_tree_start(
-      wp, args, window_buffer_build, window_buffer_draw, window_buffer_search,
-      window_buffer_menu, NULL, data, window_buffer_menu_items,
-      window_buffer_sort_list, nitems(window_buffer_sort_list), &s);
+ data->data = mode_tree_start(wp, args, window_buffer_build,
+     window_buffer_draw, window_buffer_search, window_buffer_menu, NULL,
+     data, window_buffer_menu_items, window_buffer_sort_list,
+     nitems(window_buffer_sort_list), &s);
@@ -210 +242,3 @@ static struct screen *window_buffer_init(struct window_mode_entry *wme,
-static void window_buffer_free(struct window_mode_entry *wme) {
+static void
+window_buffer_free(struct window_mode_entry *wme)
+{
@@ -213 +247,2 @@ static void window_buffer_free(struct window_mode_entry *wme) {
-  if (data == NULL) return;
+ if (data == NULL)
+  return;
@@ -222,2 +257,3 @@ static void window_buffer_free(struct window_mode_entry *wme) {
-static void window_buffer_resize(struct window_mode_entry *wme, u_int sx,
-                                 u_int sy) {
+static void
+window_buffer_resize(struct window_mode_entry *wme, u_int sx, u_int sy)
+{
@@ -227,3 +263,4 @@ static void window_buffer_resize(struct window_mode_entry *wme, u_int sx,
-static void window_buffer_do_delete(void *modedata, void *itemdata,
-                                    __unused struct client *c,
-                                    __unused key_code key) {
+static void
+window_buffer_do_delete(void *modedata, void *itemdata,
+    __unused struct client *c, __unused key_code key)
+{
@@ -233,2 +270,4 @@ static void window_buffer_do_delete(void *modedata, void *itemdata,
-  if (item == mode_tree_get_current(data->data)) mode_tree_down(data->data, 0);
-  if ((pb = paste_get_name(item->name)) != NULL) paste_free(pb);
+ if (item == mode_tree_get_current(data->data))
+  mode_tree_down(data->data, 0);
+ if ((pb = paste_get_name(item->name)) != NULL)
+  paste_free(pb);
@@ -236,2 +275,4 @@ static void window_buffer_do_delete(void *modedata, void *itemdata,
-static void window_buffer_do_paste(void *modedata, void *itemdata,
-                                   struct client *c, __unused key_code key) {
+static void
+window_buffer_do_paste(void *modedata, void *itemdata, struct client *c,
+    __unused key_code key)
+{
@@ -243 +284,3 @@ static void window_buffer_do_paste(void *modedata, void *itemdata,
-static voidwindow_buffer_finish_edit(struct window_buffer_editdata *ed) {
+static void
+window_buffer_finish_edit(struct window_buffer_editdata *ed)
+{
@@ -247 +290,3 @@ static voidwindow_buffer_finish_edit(struct window_buffer_editdata *ed) {
-static voidwindow_buffer_edit_close_cb(char *buf, size_t len, void *arg) {
+static void
+window_buffer_edit_close_cb(char *buf, size_t len, void *arg)
+{
@@ -265 +310,3 @@ static voidwindow_buffer_edit_close_cb(char *buf, size_t len, void *arg) {
-  if (oldlen != '\0' && oldbuf[oldlen - 1] != '\n' && buf[len - 1] == '\n')
+ if (oldlen != '\0' &&
+     oldbuf[oldlen - 1] != '\n' &&
+     buf[len - 1] == '\n')
@@ -267 +314,2 @@ static voidwindow_buffer_edit_close_cb(char *buf, size_t len, void *arg) {
-  if (len != 0) paste_replace(pb, buf, len);
+ if (len != 0)
+  paste_replace(pb, buf, len);
@@ -280,3 +328,4 @@ static voidwindow_buffer_edit_close_cb(char *buf, size_t len, void *arg) {
-static voidwindow_buffer_start_edit(struct window_buffer_modedata *data,
-                                    struct window_buffer_itemdata *item,
-                                    struct client *c) {
+static void
+window_buffer_start_edit(struct window_buffer_modedata *data,
+    struct window_buffer_itemdata *item, struct client *c)
+{
@@ -287 +336,2 @@ static voidwindow_buffer_start_edit(struct window_buffer_modedata *data,
-  if ((pb = paste_get_name(item->name)) == NULL) return;
+ if ((pb = paste_get_name(item->name)) == NULL)
+  return;
@@ -296,4 +346,5 @@ static voidwindow_buffer_start_edit(struct window_buffer_modedata *data,
-static void window_buffer_key(struct window_mode_entry *wme, struct client *c,
-                              __unused struct session *s,
-                              __unused struct winlink *wl, key_code key,
-                              struct mouse_event *m) {
+static void
+window_buffer_key(struct window_mode_entry *wme, struct client *c,
+    __unused struct session *s, __unused struct winlink *wl, key_code key,
+    struct mouse_event *m)
+{
