--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/57fe03dc-574a9e4b-aa7dccf8/window-buffer.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/57fe03dc-574a9e4b-aa7dccf8/window-buffer.no_comments_merged.c
@@ -6 +5,0 @@
-#include <vis.h>
@@ -18 +17,2 @@ static void window_buffer_key(struct window_mode_entry *, struct client *,
-#define WINDOW_BUFFER_DEFAULT_FORMAT "#{t/p:buffer_created}: #{buffer_sample}"
+#define WINDOW_BUFFER_DEFAULT_FORMAT \
+ "#{t/p:buffer_created}: #{buffer_sample}"
