--- a/./output/redis/d433ebc6-b4f2e412-2b00385d/src@t_zset.no_comments_mergegen.c
+++ b/./output/redis/d433ebc6-b4f2e412-2b00385d/src@t_zset.no_comments_merged.c
@@ -502 +502 @@ double score, value;
-if (other) { value = src[j].weight * zunionInterDictValue(other); zunionInterAggregate(&score,value,aggregate);
+                        zunionInterAggregate(&score,value,aggregate);
@@ -522,0 +523,2 @@ if (other) { value = src[j].weight * zunionInterDictValue(other); zunionInterAgg
+                if (dictFind(dstzset->dict,dictGetEntryKey(de)) != NULL)
+                    continue;
@@ -528 +530 @@ if (other) { value = src[j].weight * zunionInterDictValue(other); zunionInterAgg
-if (other) { value = src[j].weight * zunionInterDictValue(other); zunionInterAggregate(&score,value,aggregate);
+                        zunionInterAggregate(&score,value,aggregate);
