--- a/home/whalien/codebase/python/mergebot-eva/output/redis/d433ebc6-b4f2e412-2b00385d/src@t_zset.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/d433ebc6-b4f2e412-2b00385d/src@t_zset.no_comments_merged.c
@@ -249,2 +248 @@ int zslParseRange(robj *min, robj *max, zrangespec *spec) {
-void zaddGenericCommand(redisClient *c, robj *key, robj *ele, double score,
-                        int incr, double scoreval, int doincrement) {
+void zaddGenericCommand(redisClient *c, robj *key, robj *ele, double score, int incr) {
@@ -412,2 +404,2 @@ int qsortCompareZsetopsrcByCardinality(const void *s1, const void *s2) {
-inline static void zunionInterAggregate(double *target, double val,
-                                        int aggregate) {
+#define zunionInterDictValue(_e) (dictGetEntryVal(_e) == NULL ? 1.0 : *(double*)dictGetEntryVal(_e))
+inline static void zunionInterAggregate(double *target, double val, int aggregate) {
@@ -508 +495,0 @@ void zunionInterGenericCommand(redisClient *c, robj *dstkey, int op) {
-<<<<<<< HEAD
@@ -511,7 +497,0 @@ void zunionInterGenericCommand(redisClient *c, robj *dstkey, int op) {
-|||||||
-        double *score = zmalloc(sizeof(double)), value;
-        *score = src[0].weight * zunionInterDictValue(de);
-=======
-        double score, value;
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
-        score = src[0].weight * zunionInterDictValue(de);
@@ -527 +506,0 @@ void zunionInterGenericCommand(redisClient *c, robj *dstkey, int op) {
-<<<<<<< HEAD
@@ -529,7 +507,0 @@ void zunionInterGenericCommand(redisClient *c, robj *dstkey, int op) {
-|||||||
-        if (j != setnum) {
-          zfree(score);
-        } else {
-=======
-        if (j == setnum) {
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
@@ -551,9 +523,2 @@ void zunionInterGenericCommand(redisClient *c, robj *dstkey, int op) {
-<<<<<<< HEAD
-        if (dictFind(dstzset->dict, dictGetEntryKey(de)) != NULL) continue;
-        score = src[i].weight * zunionInterDictValue(de);
-|||||||
-        if (dictFind(dstzset->dict, dictGetEntryKey(de)) != NULL) continue;
-        double *score = zmalloc(sizeof(double)), value;
-        *score = src[i].weight * zunionInterDictValue(de);
-=======
-        if (dictFind(dstzset->dict, dictGetEntryKey(de)) != NULL) continue;
+                if (dictFind(dstzset->dict,dictGetEntryKey(de)) != NULL)
+                    continue;
@@ -561 +525,0 @@ void zunionInterGenericCommand(redisClient *c, robj *dstkey, int op) {
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
@@ -702 +664,0 @@ void genericZrangebyscoreCommand(redisClient *c, int reverse, int justcount) {
-<<<<<<< HEAD
@@ -707,19 +668,0 @@ void genericZrangebyscoreCommand(redisClient *c, int reverse, int justcount) {
-|||||||
-    zset *zsetobj = o->ptr;
-    zskiplist *zsl = zsetobj->zsl;
-    zskiplistNode *ln;
-    robj *ele;
-    void *replylen = NULL;
-    unsigned long rangelen = 0;
-    ln = zslFirstWithScore(zsl, min);
-    while (minex && ln && ln->score == min) ln = ln->forward[0];
-=======
-    zset *zsetobj = o->ptr;
-    zskiplist *zsl = zsetobj->zsl;
-    zskiplistNode *ln;
-    robj *ele;
-    void *replylen = NULL;
-    unsigned long rangelen = 0;
-    ln = zslFirstWithScore(zsl, min);
-    while (minex && ln && ln->score == min) ln = ln->level[0].forward;
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
@@ -730,2 +673,2 @@ void genericZrangebyscoreCommand(redisClient *c, int reverse, int justcount) {
-  if (!justcount) replylen = addDeferredMultiBulkLength(c);
-<<<<<<< HEAD
+    if (!justcount)
+        replylen = addDeferredMultiBulkLength(c);
@@ -737,13 +679,0 @@ void genericZrangebyscoreCommand(redisClient *c, int reverse, int justcount) {
-|||||||
-    while (ln && (maxex ? (ln->score < max) : (ln->score <= max))) {
-      if (offset) {
-        offset--;
-        ln = ln->forward[0];
-        continue;
-=======
-    while (ln && (maxex ? (ln->score < max) : (ln->score <= max))) {
-      if (offset) {
-        offset--;
-        ln = ln->level[0].forward;
-        continue;
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
@@ -770 +700,0 @@ void genericZrangebyscoreCommand(redisClient *c, int reverse, int justcount) {
-<<<<<<< HEAD
@@ -775,9 +704,0 @@ void genericZrangebyscoreCommand(redisClient *c, int reverse, int justcount) {
-|||||||
-        ln = ln->forward[0];
-        rangelen++;
-        if (limit > 0) limit--;
-=======
-        ln = ln->level[0].forward;
-        rangelen++;
-        if (limit > 0) limit--;
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
