diff --git a/./output/tmux/921991c9-0084cbef-63b6eec2/arguments.no_comments_conflict.c b/./output/tmux/921991c9-0084cbef-63b6eec2/arguments.no_comments_merged.c
index 73c6483..2905c69 100644
--- a/./output/tmux/921991c9-0084cbef-63b6eec2/arguments.no_comments_conflict.c
+++ b/./output/tmux/921991c9-0084cbef-63b6eec2/arguments.no_comments_merged.c
@@ -5 +4,0 @@
-<<<<<<< HEAD
@@ -7,6 +5,0 @@
-||||||| 63b6eec2
-#include <unistd.h>
-#include <vis.h>
-=======
-#include <vis.h>
->>>>>>> 0084cbef
