--- a/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_merged.cpp
@@ -25,0 +26 @@ Camera2ClientBase<TClientBase>::Camera2ClientBase(
+        std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
