--- a/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@CameraService.no_comments_conflict.h
+++ b/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@CameraService.no_comments_merged.h
@@ -34 +33,0 @@
-<<<<<<< HEAD
@@ -37,4 +35,0 @@
-||||||| b97e7d6371
-#include "utils/AttributionAndPermissionUtils.h"
-=======
->>>>>>> ab2b3e66
@@ -83 +78,3 @@ public:
-                                cameraServiceProxyWrapper = nullptr);
+                                cameraServiceProxyWrapper = nullptr,
+                                std::shared_ptr<AttributionAndPermissionUtils>
+                                attributionAndPermissionUtils = nullptr);
@@ -200,5 +196,0 @@ public:
-<<<<<<< HEAD
-    bool isAutomotiveExteriorSystemCamera(const std::string& cameraId) const;
-||||||| b97e7d6371
-    bool isAutomotiveDevice() const;
-    bool isAutomotivePrivilegedClient(int32_t uid) const;
@@ -206,2 +197,0 @@ public:
-=======
->>>>>>> ab2b3e66
@@ -258,0 +249 @@ public:
+                std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
@@ -334,0 +326 @@ public:
+                std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
@@ -388 +379,0 @@ private:
-    bool isAutomotiveExteriorSystemCamera(const std::string& cameraId) const;
@@ -400 +390,0 @@ private:
-<<<<<<< HEAD
@@ -404,8 +393,0 @@ private:
-||||||| b97e7d6371
-    bool hasPermissionsForCamera(int callingPid, int callingUid) const;
-=======
-    bool checkPermission(const std::string& cameraId, const std::string& permission,
-            const content::AttributionSourceState& attributionSource, const std::string& message,
-            int32_t attributedOpCode) const;
->>>>>>> ab2b3e66
-<<<<<<< HEAD
@@ -425,9 +406,0 @@ private:
-||||||| b97e7d6371
-    bool hasPermissionsForCamera(const std::string& cameraId, int callingPid, int callingUid) const;
-    bool hasPermissionsForSystemCamera(const std::string& cameraId, int callingPid, int callingUid,
-            bool checkCameraPermissions = true) const;
-=======
-    bool hasPermissionsForSystemCamera(const std::string& cameraId, int callingPid, int callingUid)
-            const;
->>>>>>> ab2b3e66
-<<<<<<< HEAD
@@ -526 +498,0 @@ private:
-<<<<<<< HEAD
@@ -532,10 +503,0 @@ private:
-||||||| b97e7d6371
-            explicit SensorPrivacyPolicy(wp<CameraService> service,
-                    std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils)
-                    : mService(service),
-                      mAttributionAndPermissionUtils(attributionAndPermissionUtils),
-                      mSensorPrivacyEnabled(false),
-=======
-            explicit SensorPrivacyPolicy(wp<CameraService> service)
-                    : mService(service), mSensorPrivacyEnabled(false),
->>>>>>> ab2b3e66
