diff --git a/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_conflict.cpp b/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_merged.cpp
index aacbdd7..e695ee2 100644
--- a/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_conflict.cpp
+++ b/./output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_merged.cpp
@@ -25,0 +26 @@ Camera2ClientBase<TClientBase>::Camera2ClientBase(
+        std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
@@ -39 +39,0 @@ Camera2ClientBase<TClientBase>::Camera2ClientBase(
-<<<<<<< HEAD
@@ -43,9 +42,0 @@ Camera2ClientBase<TClientBase>::Camera2ClientBase(
-||||||| b97e7d6371
-        TClientBase(cameraService, remoteCallback, attributionAndPermissionUtils, clientPackageName,
-                systemNativeClient, clientFeatureId, cameraId, api1CameraId, cameraFacing,
-                sensorOrientation, clientPid, clientUid, servicePid, overrideToPortrait),
-=======
-        TClientBase(cameraService, remoteCallback, clientPackageName, systemNativeClient,
-                clientFeatureId, cameraId, api1CameraId, cameraFacing, sensorOrientation, clientPid,
-                clientUid, servicePid, overrideToPortrait),
->>>>>>> ab2b3e66
