--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/8bb0e69f-ab2b3e66-b97e7d63/services@camera@libcameraservice@common@Camera2ClientBase.no_comments_merged.cpp
@@ -7,0 +8 @@
+#include <gui/Surface.h>
@@ -17 +17,0 @@
-#include "utils/CameraThreadState.h"
@@ -19,0 +20 @@ using namespace camera2;
+namespace wm_flags = com::android::window::flags;
@@ -43,22 +54,2 @@ template <typename TClientBase>
-Camera2ClientBase<TClientBase>::Camera2ClientBase(
-        const sp<CameraService>& cameraService, const sp<TCamCallbacks>& remoteCallback,
-        std::shared_ptr<CameraServiceProxyWrapper> cameraServiceProxyWrapper,
-        const std::string& clientPackageName, bool systemNativeClient,
-        const std::optional<std::string>& clientFeatureId, const std::string& cameraId,
-        int api1CameraId, int cameraFacing, int sensorOrientation, int clientPid, uid_t clientUid,
-        int servicePid, bool overrideForPerfClass, bool overrideToPortrait, bool legacyClient)
-    : TClientBase(cameraService, remoteCallback, clientPackageName, systemNativeClient,
-                  clientFeatureId, cameraId, api1CameraId, cameraFacing, sensorOrientation,
-                  clientPid, clientUid, servicePid, overrideToPortrait),
-      mSharedCameraCallbacks(remoteCallback),
-      mCameraServiceProxyWrapper(cameraServiceProxyWrapper),
-      mDeviceActive(false),
-      mApi1CameraId(api1CameraId) {
-    ALOGI("Camera %s: Opened. Client: %s (PID %d, UID %d)", cameraId.c_str(),
-          clientPackageName.c_str(), clientPid, clientUid);
-    mInitialClientPid = clientPid;
-    mOverrideForPerfClass = overrideForPerfClass;
-    mLegacyClient = legacyClient;
-}
-template <typename TClientBase>
-status_t Camera2ClientBase<TClientBase>::checkPid(const char* checkLocation) const {
+status_t Camera2ClientBase<TClientBase>::checkPid(const char* checkLocation)
+        const {
@@ -77,0 +68,60 @@ template <typename TClientBase>
+template <typename TProviderPtr>
+status_t Camera2ClientBase<TClientBase>::initializeImpl(TProviderPtr providerPtr,
+        const std::string& monitorTags) {
+    ATRACE_CALL();
+    ALOGV("%s: Initializing client for camera %s", __FUNCTION__,
+          TClientBase::mCameraIdStr.c_str());
+    status_t res;
+    IPCTransport providerTransport = IPCTransport::INVALID;
+    res = providerPtr->getCameraIdIPCTransport(TClientBase::mCameraIdStr,
+            &providerTransport);
+    if (res != OK) {
+        return res;
+    }
+    switch (providerTransport) {
+        case IPCTransport::HIDL:
+            mDevice =
+                    new HidlCamera3Device(mCameraServiceProxyWrapper,
+                            TClientBase::mAttributionAndPermissionUtils,
+                            TClientBase::mCameraIdStr, mOverrideForPerfClass,
+                            TClientBase::mRotationOverride, mLegacyClient);
+            break;
+        case IPCTransport::AIDL:
+            mDevice =
+                    new AidlCamera3Device(mCameraServiceProxyWrapper,
+                            TClientBase::mAttributionAndPermissionUtils,
+                            TClientBase::mCameraIdStr, mOverrideForPerfClass,
+                            TClientBase::mRotationOverride, mLegacyClient);
+             break;
+        default:
+            ALOGE("%s Invalid transport for camera id %s", __FUNCTION__,
+                    TClientBase::mCameraIdStr.c_str());
+            return NO_INIT;
+    }
+    if (mDevice == NULL) {
+        ALOGE("%s: Camera %s: No device connected",
+                __FUNCTION__, TClientBase::mCameraIdStr.c_str());
+        return NO_INIT;
+    }
+    res = TClientBase::startCameraOps();
+    if (res != OK) {
+        TClientBase::finishCameraOps();
+        return res;
+    }
+    res = mDevice->initialize(providerPtr, monitorTags);
+    if (res != OK) {
+        ALOGE("%s: Camera %s: unable to initialize device: %s (%d)",
+                __FUNCTION__, TClientBase::mCameraIdStr.c_str(), strerror(-res), res);
+        TClientBase::finishCameraOps();
+        return res;
+    }
+    wp<NotificationListener> weakThis(this);
+    res = mDevice->setNotifyCallback(weakThis);
+    if (res != OK) {
+        ALOGE("%s: Camera %s: Unable to set notify callback: %s (%d)",
+                __FUNCTION__, TClientBase::mCameraIdStr.c_str(), strerror(-res), res);
+        return res;
+    }
+    return OK;
+}
+template <typename TClientBase>
@@ -342,0 +410,2 @@ status_t Camera2ClientBase<TClientBase>::injectSessionParams(const CameraMetadat
+template class Camera2ClientBase<CameraService::Client>;
+template class Camera2ClientBase<CameraDeviceClientBase>;
