diff --git a/./output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_mergegen.cpp b/./output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_merged.cpp
index 4691080..0bc1f1c 100644
--- a/./output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_mergegen.cpp
+++ b/./output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_merged.cpp
@@ -6 +6,4 @@
-#include "src/capture/screengrabber.h" #include <src/utils/confighandler.h>
+#include "src/capture/screengrabber.h"
+#include "src/utils/confighandler.h"
+#include <QScreen>
+#include <QWindow>
