--- a/./output/libnativehelper/076ee54f-3db6f943-f39979f9/JniConstants.no_comments_conflict.cpp
+++ b/./output/libnativehelper/076ee54f-3db6f943-f39979f9/JniConstants.no_comments_merged.cpp
@@ -6 +5,0 @@
-<<<<<<< HEAD
@@ -11,3 +9,0 @@ static std::mutex g_constants_mutex;
-jclass JniConstants::bigDecimalClass;
-=======
->>>>>>> 3db6f943
@@ -62 +57,0 @@ void JniConstants::init(JNIEnv* env) {
-<<<<<<< HEAD
@@ -70,3 +64,0 @@ void JniConstants::init(JNIEnv* env) {
-    bigDecimalClass = findClass(env, "java/math/BigDecimal");
-=======
->>>>>>> 3db6f943
