diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_merged.cpp
index 25f3c86..ea0645f 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_merged.cpp
@@ -13,4 +12,0 @@
-#include <vintf/RuntimeInfo.h>
-#include <vintf/VintfObject.h>
-#endif
-#if defined(__ANDROID__)
@@ -68 +64 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-    <<<<<<< HEAD if (configs.empty()) {
+    if (configs.empty()) {
@@ -74,10 +69,0 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-|||||||
-    std::shared_ptr<const android::vintf::RuntimeInfo> runtimeInfo =
-            android::vintf::VintfObject::GetInstance()->getRuntimeInfo(
-                    vintf::RuntimeInfo::FetchFlag::CONFIG_GZ);
-    LOG_ALWAYS_FATAL_IF(runtimeInfo == nullptr, "Kernel configs could not be fetched");
-=======
-    std::map<std::string, std::string> kernelConfigs;
-    const status_t result = android::kernelconfigs::LoadKernelConfigs(&kernelConfigs);
-    LOG_ALWAYS_FATAL_IF(result != OK, "Kernel configs could not be fetched");
->>>>>>> bc7936f870b420a5368cac869b129d60254e31fb
@@ -104,3 +90,4 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-KeyLayoutMap::KeyLayoutMap() KeyLayoutMap::~KeyLayoutMap()
-        base::Result<std::shared_ptr<KeyLayoutMap>> KeyLayoutMap::loadContents(
-                const std::string& filename, const char* contents) {
+KeyLayoutMap::KeyLayoutMap() = default;
+KeyLayoutMap::~KeyLayoutMap() = default;
+base::Result<std::shared_ptr<KeyLayoutMap>> KeyLayoutMap::loadContents(const std::string& filename,
+                                                                       const char* contents) {
@@ -160,2 +147,2 @@ base::Result<std::shared_ptr<KeyLayoutMap>> KeyLayoutMap::load(Tokenizer* tokeni
-status_t KeyLayoutMap::mapKey(int32_t scanCode, int32_t usageCode, int32_t* outKeyCode,
-                              uint32_t* outFlags) const {
+status_t KeyLayoutMap::mapKey(int32_t scanCode, int32_t usageCode,
+        int32_t* outKeyCode, uint32_t* outFlags) const {
@@ -256,3 +243,5 @@ std::optional<int32_t> KeyLayoutMap::findUsageCodeForLed(int32_t ledCode) const
-KeyLayoutMap::Parser::Parser(KeyLayoutMap* map, Tokenizer* tokenizer)
-      : mMap(map), mTokenizer(tokenizer) {}
-KeyLayoutMap::Parser::~Parser() {}
+KeyLayoutMap::Parser::Parser(KeyLayoutMap* map, Tokenizer* tokenizer) :
+        mMap(map), mTokenizer(tokenizer) {
+}
+KeyLayoutMap::Parser::~Parser() {
+}
@@ -379,2 +368,2 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-            ALOGE("%s: Expected inverted axis label, got '%s'.", mTokenizer->getLocation().c_str(),
-                  axisToken.c_str());
+            ALOGE("%s: Expected inverted axis label, got '%s'.",
+                    mTokenizer->getLocation().c_str(), axisToken.c_str());
@@ -390,2 +379,2 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-            ALOGE("%s: Expected split value, got '%s'.", mTokenizer->getLocation().c_str(),
-                  splitToken.c_str());
+            ALOGE("%s: Expected split value, got '%s'.",
+                    mTokenizer->getLocation().c_str(), splitToken.c_str());
@@ -399,2 +388,2 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-            ALOGE("%s: Expected low axis label, got '%s'.", mTokenizer->getLocation().c_str(),
-                  lowAxisToken.c_str());
+            ALOGE("%s: Expected low axis label, got '%s'.",
+                    mTokenizer->getLocation().c_str(), lowAxisToken.c_str());
@@ -408,2 +397,2 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-            ALOGE("%s: Expected high axis label, got '%s'.", mTokenizer->getLocation().c_str(),
-                  highAxisToken.c_str());
+            ALOGE("%s: Expected high axis label, got '%s'.",
+                    mTokenizer->getLocation().c_str(), highAxisToken.c_str());
@@ -433,2 +422,2 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-                ALOGE("%s: Expected flat value, got '%s'.", mTokenizer->getLocation().c_str(),
-                      flatToken.c_str());
+                ALOGE("%s: Expected flat value, got '%s'.",
+                        mTokenizer->getLocation().c_str(), flatToken.c_str());
