diff --git a/./output/frameworks_av/763e86dc-67110b51-c7caded3/services@camera@libcameraservice@CameraService.no_comments_conflict.h b/./output/frameworks_av/763e86dc-67110b51-c7caded3/services@camera@libcameraservice@CameraService.no_comments_merged.h
index b770b95..a03d129 100644
--- a/./output/frameworks_av/763e86dc-67110b51-c7caded3/services@camera@libcameraservice@CameraService.no_comments_conflict.h
+++ b/./output/frameworks_av/763e86dc-67110b51-c7caded3/services@camera@libcameraservice@CameraService.no_comments_merged.h
@@ -435 +434,0 @@ private:
-<<<<<<< HEAD
@@ -440,7 +438,0 @@ private:
-||||||| c7caded36e
-                    int uid) : mParent(parent), mListener(listener), mListenerUid(uid) {}
-=======
-                    int uid, int pid, bool openCloseCallbackAllowed) : mParent(parent),
-                    mListener(listener), mListenerUid(uid), mListenerPid(pid),
-                    mOpenCloseCallbackAllowed(openCloseCallbackAllowed) {}
->>>>>>> 67110b51
@@ -459 +450,0 @@ private:
-<<<<<<< HEAD
@@ -462,4 +452,0 @@ private:
-||||||| c7caded36e
-=======
-            bool isOpenCloseCallbackAllowed() { return mOpenCloseCallbackAllowed; }
->>>>>>> 67110b51
@@ -469 +455,0 @@ private:
-<<<<<<< HEAD
@@ -474,7 +459,0 @@ private:
-||||||| c7caded36e
-            int mListenerUid;
-=======
-            int mListenerUid;
-            int mListenerPid;
-            bool mOpenCloseCallbackAllowed = false;
->>>>>>> 67110b51
