diff --git a/./output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_mergegen.c b/./output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_merged.c
index 574e17e..dadf4fa 100644
--- a/./output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_mergegen.c
+++ b/./output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_merged.c
@@ -183,0 +184,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
+        output_help(--argc, ++argv);
+        return REDIS_OK;
