diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_merged.c
index 64609b7..dadf4fa 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_merged.c
@@ -10 +9,0 @@
-#include "anet.h"
@@ -12 +10,0 @@
-#include "adlist.h"
@@ -14 +11,0 @@
-#include "help.h"
@@ -17,0 +15 @@
+#include "help.h"
@@ -71 +69,2 @@ static int cliConnect(int force) {
-    if (context != NULL) redisFree(context);
+        if (context != NULL)
+            redisFree(context);
@@ -80,2 +79 @@ static int cliConnect(int force) {
-        fprintf(stderr, "%s:%d: %s\n", config.hostip, config.hostport,
-                context->errstr);
+                fprintf(stderr,"%s:%d: %s\n",config.hostip,config.hostport,context->errstr);
@@ -88,2 +86,4 @@ static int cliConnect(int force) {
-    if (cliAuth() != REDIS_OK) return REDIS_ERR;
-    if (cliSelect() != REDIS_OK) return REDIS_ERR;
+        if (cliAuth() != REDIS_OK)
+            return REDIS_ERR;
+        if (cliSelect() != REDIS_OK)
+            return REDIS_ERR;
@@ -161 +161,2 @@ static int cliReadReply() {
-    if (config.shutdown) return REDIS_OK;
+        if (config.shutdown)
+            return REDIS_OK;
@@ -163,2 +164,4 @@ static int cliReadReply() {
-      if (context->err == REDIS_ERR_IO && errno == ECONNRESET) return REDIS_ERR;
-      if (context->err == REDIS_ERR_EOF) return REDIS_ERR;
+            if (context->err == REDIS_ERR_IO && errno == ECONNRESET)
+                return REDIS_ERR;
+            if (context->err == REDIS_ERR_EOF)
+                return REDIS_ERR;
@@ -181,7 +183,0 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-<<<<<<< HEAD
-    showInteractiveHelp();
-    return REDIS_OK;
-|||||||
-    showInteractiveHelp();
-    return 0;
-=======
@@ -189,2 +185 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-    return 0;
->>>>>>> 5397f2b596b5189edbed3e45a42d18f3c99341d6
+        return REDIS_OK;
@@ -194,2 +189,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-  if (!strcasecmp(command, "subscribe") || !strcasecmp(command, "psubscribe"))
-    config.pubsub_mode = 1;
+    if (!strcasecmp(command,"subscribe") ||
+        !strcasecmp(command,"psubscribe")) config.pubsub_mode = 1;
@@ -197 +192,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-  for (j = 0; j < argc; j++) argvlen[j] = sdslen(argv[j]);
+    for (j = 0; j < argc; j++)
+        argvlen[j] = sdslen(argv[j]);
@@ -210 +206,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-    if (cliReadReply() != REDIS_OK) return REDIS_ERR;
+        if (cliReadReply() != REDIS_OK)
+            return REDIS_ERR;
@@ -242,4 +239,3 @@ static int parseOptions(int argc, char **argv) {
-              "Starting interactive mode using -i is deprecated. Interactive "
-              "mode is started\n"
-              "by default when redis-cli is executed without a command to "
-              "execute.\n");
+"Starting interactive mode using -i is deprecated. Interactive mode is started\n"
+"by default when redis-cli is executed without a command to execute.\n"
+            );
@@ -247,2 +243 @@ static int parseOptions(int argc, char **argv) {
-      fprintf(
-          stderr,
+            fprintf(stderr,
@@ -251 +246,2 @@ static int parseOptions(int argc, char **argv) {
-          "automatically used as last argument.\n");
+"automatically used as last argument.\n"
+            );
@@ -253,2 +249 @@ static int parseOptions(int argc, char **argv) {
-      printf("redis-cli shipped with Redis version %s (%s)\n", REDIS_VERSION,
-             redisGitSHA1());
+            printf("redis-cli shipped with Redis version %s (%s)\n", REDIS_VERSION, redisGitSHA1());
@@ -267,2 +262 @@ static sds readArgFromStdin(void) {
-    if (nread == 0)
-      break;
+        if (nread == 0) break;
@@ -278,7 +272,2 @@ static void usage() {
-  fprintf(
-      stderr,
-      "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a "
-      "authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... "
-          "arg(N-1)\n\n");
+    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
+    fprintf(stderr, "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... arg(N-1)\n\n");
@@ -288,3 +277 @@ static void usage() {
-  fprintf(stderr,
-          "\nRun in interactive mode: redis-cli -i or just don't pass any "
-          "command\n");
+    fprintf(stderr, "\nRun in interactive mode: redis-cli -i or just don't pass any command\n");
@@ -296 +283,2 @@ static char **convertToSds(int count, char **args) {
-  for (j = 0; j < count; j++) sds[j] = sdsnew(args[j]);
+  for(j = 0; j < count; j++)
+    sds[j] = sdsnew(args[j]);
@@ -315 +303,2 @@ static void repl() {
-            strcasecmp(argv[0], "exit") == 0) {
+                    strcasecmp(argv[0],"exit") == 0)
+                {
@@ -333 +322,2 @@ static void repl() {
-      for (j = 0; j < argc; j++) sdsfree(argv[j]);
+            for (j = 0; j < argc; j++)
+                sdsfree(argv[j]);
