--- a/./output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_conflict.c
+++ b/./output/redis/c086b85a-5397f2b5-21dbc649/src@redis-cli.no_comments_merged.c
@@ -184,7 +183,0 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-<<<<<<< HEAD
-        showInteractiveHelp();
-        return REDIS_OK;
-||||||| 21dbc6499
-        showInteractiveHelp();
-        return 0;
-=======
@@ -192,2 +185 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-        return 0;
->>>>>>> 5397f2b5
+        return REDIS_OK;
