--- a/./output/rocksdb/9625acbf-8ca30bd5-a1d56e73/db@db_impl.no_comments_conflict.cc
+++ b/./output/rocksdb/9625acbf-8ca30bd5-a1d56e73/db@db_impl.no_comments_merged.cc
@@ -2043,7 +2042,0 @@ static void CleanupIteratorState(void* arg1, void* arg2) {
-<<<<<<< HEAD
-  DBImpl::DeletionState deletion_state;
-||||||| a1d56e73e
-  DBImpl::DeletionState deletion_state(state->db->GetOptions().
-                                       max_write_buffer_number);
-=======
->>>>>>> 8ca30bd5
