--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/a4f14d91-caf7750f-9cadfde2/libnativebridge@native_bridge.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/a4f14d91-caf7750f-9cadfde2/libnativebridge@native_bridge.no_comments_merged.cc
@@ -152,0 +153,15 @@ bool LoadNativeBridge(const char* nb_library_filename,
+#if defined(__arm__)
+static const char* kRuntimeISA = "arm";
+#elif defined(__aarch64__)
+static const char* kRuntimeISA = "arm64";
+#elif defined(__mips__) && !defined(__LP64__)
+static const char* kRuntimeISA = "mips";
+#elif defined(__mips__) && defined(__LP64__)
+static const char* kRuntimeISA = "mips64";
+#elif defined(__i386__)
+static const char* kRuntimeISA = "x86";
+#elif defined(__x86_64__)
+static const char* kRuntimeISA = "x86_64";
+#else
+static const char* kRuntimeISA = "unknown";
+#endif
@@ -385,2 +396 @@ const char* NativeBridgeGetError() {
-            return "native bridge implementation is not compatible with version 3, cannot get "
-                   "message";
+      return "native bridge implementation is not compatible with version 3, cannot get message";
@@ -402 +411,2 @@ bool NativeBridgeIsPathSupported(const char* path) {
-bool NativeBridgeInitNamespace(const char* public_ns_sonames, const char* anon_ns_library_path) {
+bool NativeBridgeInitAnonymousNamespace(const char* public_ns_sonames,
+                                        const char* anon_ns_library_path) {
@@ -450 +463 @@ void* NativeBridgeLoadLibraryExt(const char* libpath, int flag, native_bridge_na
-}
+};
