diff --git a/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_mergegen.c b/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_merged.c
index 9ddfe8f..efcedff 100644
--- a/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_mergegen.c
+++ b/./output/redis/ff682d79-c6fb9d09-5f450e49/src@module.no_comments_merged.c
@@ -5 +5,2 @@
-#include <sys/stat.h> #define REDISMODULE_CORE 1 #include <sys/wait.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
