--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_merged.cpp
@@ -4 +3,0 @@
-#include <paths.h>
@@ -8 +6,0 @@
-#include <sys/epoll.h>
@@ -10 +7,0 @@
-#include <sys/sysmacros.h>
@@ -25 +21,0 @@
-#include <private/android_filesystem_config.h>
@@ -35,2 +30,0 @@
-#include "init_first_stage.h"
-#include "log.h"
@@ -44,2 +37,0 @@
-#include "ueventd.h"
-#include "watchdogd.h"
