diff --git a/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_mergegen.cpp b/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_merged.cpp
index c2a4d02..b8df972 100644
--- a/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_mergegen.cpp
+++ b/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_merged.cpp
@@ -36 +36,4 @@
-#include "boringssl_self_test.h" #include "epoll.h" #include "first_stage_init.h" #include "first_stage_mount.h"
+#include "boringssl_self_test.h"
+#include "epoll.h"
+#include "first_stage_init.h"
+#include "first_stage_mount.h"
