--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/796162d7-50e44460-5d1d32dc/lmkd@lmkd.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/796162d7-50e44460-5d1d32dc/lmkd@lmkd.no_comments_merged.c
@@ -224,0 +251,2 @@ static struct proc *pidhash[PIDHASH_SZ];
+#define pid_hashfn(x) ((((x) >> 8) ^ (x)) & (PIDHASH_SZ - 1))
+#define ADJTOSLOT(adj) ((adj) + -OOM_SCORE_ADJ_MIN)
@@ -366,0 +400,5 @@ static bool writefilestring(const char *path, const char *s,
+static inline long get_time_diff_ms(struct timespec *from,
+                                    struct timespec *to) {
+    return (to->tv_sec - from->tv_sec) * (long)MS_PER_SEC +
+           (to->tv_nsec - from->tv_nsec) / (long)NS_PER_MS;
+}
@@ -999 +1043,0 @@ static int kill_one_process(struct proc *procp, int min_oom_score) {
-<<<<<<< HEAD
@@ -1002,6 +1046,2 @@ static int kill_one_process(struct proc *procp, int min_oom_score) {
-|||||||
-=======
-  set_process_group_and_prio(pid, SP_FOREGROUND, ANDROID_PRIORITY_HIGHEST);
->>>>>>> 50e44460e76c950df2ae58ada53a86ba74f1fdc9
-  ALOGI("Kill '%s' (%d), uid %d, oom_adj %d to free %ldkB", taskname, pid, uid,
-        procp->oomadj, tasksize * page_k);
+    ALOGI("Kill '%s' (%d), uid %d, oom_adj %d to free %ldkB",
+        taskname, pid, uid, procp->oomadj, tasksize * page_k);
@@ -1110,7 +1150,2 @@ enum vmpressure_level downgrade_level(enum vmpressure_level level) {
-  return (enum vmpressure_level)((level > VMPRESS_LEVEL_LOW) ? level - 1
-                                                             : level);
-}
-static inline long get_time_diff_ms(struct timespec *from,
-                                    struct timespec *to) {
-  return (to->tv_sec - from->tv_sec) * (long)MS_PER_SEC +
-         (to->tv_nsec - from->tv_nsec) / (long)NS_PER_MS;
+    return (enum vmpressure_level)((level > VMPRESS_LEVEL_LOW) ?
+        level - 1 : level);
@@ -1512,3 +1540 @@ static int init(void) {
-      ALOGE(
-          "Kernel does not support memory pressure events or in-kernel low "
-          "memory killer");
+            ALOGE("Kernel does not support memory pressure events or in-kernel low memory killer");
