diff --git a/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_conflict.cc b/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_merged.cc
index f239c22..9313530 100644
--- a/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_conflict.cc
+++ b/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_merged.cc
@@ -32 +31,0 @@ using ::android::fs_mgr::ReadFstabFromProcMounts;
-<<<<<<< HEAD
@@ -36,4 +34,0 @@ using ::art::service::ValidatePathElementSubstring;
-||||||| 64a6fe7549
-using ::fmt::literals::operator""_format;
-=======
->>>>>>> 489049d1
