diff --git a/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_mergegen.cc b/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_merged.cc
index d73459f..9313530 100644
--- a/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_mergegen.cc
+++ b/./output/art/b90c0b34-489049d1-64a6fe75/artd@path_utils.no_comments_merged.cc
@@ -32 +32,3 @@ using ::android::fs_mgr::ReadFstabFromProcMounts;
-using ::art::service::ValidateDexPath; using ::art::service::ValidatePathElement; using ::art::service::ValidatePathElementSubstring;
+using ::art::service::ValidateDexPath;
+using ::art::service::ValidatePathElement;
+using ::art::service::ValidatePathElementSubstring;
