--- a/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_mergegen.cc
+++ b/./output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_merged.cc
@@ -877 +877,6 @@ WARN_UNUSED bool OnDeviceRefresh::CheckBuildUserfaultFdGc() const {
-ART_FORMAT( "Userfaultfd GC check failed (build_enable_uffd_gc: {}, is_at_least_u: {}, " "kernel_supports_uffd: {}).", build_enable_uffd_gc, is_at_least_u, kernel_supports_uffd);
+    LOG(WARNING) << ART_FORMAT(
+        "Userfaultfd GC check failed (build_enable_uffd_gc: {}, is_at_least_u: {}, "
+        "kernel_supports_uffd: {}).",
+        build_enable_uffd_gc,
+        is_at_least_u,
+        kernel_supports_uffd);
