diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_merged.cc
index 8ba50bf..b2a630f 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/b90c0b34-489049d1-64a6fe75/odrefresh@odrefresh.no_comments_merged.cc
@@ -42 +41,0 @@
-#include "android-base/parsebool.h"
@@ -48 +46,0 @@
-#include "android/log.h"
@@ -50 +47,0 @@
-#include "base/globals.h"
@@ -52 +48,0 @@
-#include "base/string_view_cpp20.h"
@@ -56,2 +51,0 @@
-#include "dexoptanalyzer.h"
-#include "log/log.h"
@@ -59 +52,0 @@
-#include "odr_compilation_log.h"
@@ -62 +54,0 @@
-#include "fmt/format.h"
@@ -78 +69,0 @@
-#include "read_barrier_config.h"
@@ -81,0 +73,2 @@ namespace {
+namespace apex = com::android::apex;
+namespace art_apex = com::android::art;
@@ -92,0 +86 @@ using ::android::modules::sdklevel::IsAtLeastU;
+using ::art::tools::CmdlineBuilder;
@@ -187 +181 @@ Result<void> CheckComponents(
-        [](const T&, const T&) -> Result<void> {
+        [](const T&, const T&) -> Result<void> { return {}; }) {
@@ -312 +306,3 @@ bool IsCpuSetSpecValid(const std::string& cpu_set) {
-Result<void> AddDex2OatConcurrencyArguments( CmdlineBuilder& args, bool is_compilation_os, const OdrSystemProperties& system_properties) {
+Result<void> AddDex2OatConcurrencyArguments( CmdlineBuilder& args,
+                                            bool is_compilation_os,
+                                            const OdrSystemProperties& system_properties) {
@@ -340 +336,3 @@ void AddDex2OatDebugInfo( CmdlineBuilder& args) {
-void AddDex2OatInstructionSet( CmdlineBuilder& args, InstructionSet isa, const OdrSystemProperties& system_properties) {
+void AddDex2OatInstructionSet( CmdlineBuilder& args,
+                              InstructionSet isa,
+                              const OdrSystemProperties& system_properties) {
@@ -522,0 +521,8 @@ OnDeviceRefresh::OnDeviceRefresh(const OdrConfig& config)
+OnDeviceRefresh::OnDeviceRefresh(const OdrConfig& config,
+                                 const std::string& cache_info_filename,
+                                 std::unique_ptr<ExecUtils> exec_utils,
+                                 android::base::function_ref<bool()> check_compilation_space)
+    : config_(config),
+      cache_info_filename_(cache_info_filename),
+      start_time_(time(nullptr)),
+      exec_utils_(std::move(exec_utils)),
@@ -871 +876,0 @@ WARN_UNUSED bool OnDeviceRefresh::CheckBuildUserfaultFdGc() const {
-<<<<<<< HEAD
@@ -878,8 +882,0 @@ WARN_UNUSED bool OnDeviceRefresh::CheckBuildUserfaultFdGc() const {
-|||||||
-    LOG(WARNING) << "Userfaultfd GC check failed (build-time: {}, runtime: {})."_format(
-        build_enable_uffd_gc, kernel_supports_uffd);
-=======
-    LOG(WARNING) << ART_FORMAT("Userfaultfd GC check failed (build-time: {}, runtime: {}).",
-                               build_enable_uffd_gc,
-                               kernel_supports_uffd);
->>>>>>> 489049d1c7c655461981e07d32e5ff0975e4d3d9
@@ -1355 +1351,0 @@ WARN_UNUSED CompilationResult OnDeviceRefresh::RunDex2oat(
-<<<<<<< HEAD
@@ -1358,7 +1353,0 @@ WARN_UNUSED CompilationResult OnDeviceRefresh::RunDex2oat(
-|||||||
-  std::string cmd_line = Join(args, ' ');
-  LOG(INFO) << "{}: {} [timeout {}s]"_format(debug_message, cmd_line, timeout);
-=======
-  std::string cmd_line = Join(args, ' ');
-  LOG(INFO) << ART_FORMAT("{}: {} [timeout {}s]", debug_message, cmd_line, timeout);
->>>>>>> 489049d1c7c655461981e07d32e5ff0975e4d3d9
