--- a/./output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_conflict.cc
+++ b/./output/grpc/478ea19b-d109f4b4-cfd05080/src@core@ext@xds@xds_api.no_comments_merged.cc
@@ -1001,7 +1000,0 @@ grpc_error* RouteConfigParse(
-<<<<<<< HEAD
-    const std::string& expected_server_name, XdsApi::RdsUpdate* rds_update) {
-  MaybeLogRouteConfiguration(client, tracer, symtab, route_config);
-||||||| cfd0508009
-    const std::string& expected_server_name, XdsApi::RdsUpdate* rds_update) {
-  MaybeLogRouteConfiguration(client, tracer, route_config);
-=======
@@ -1009,2 +1002 @@ grpc_error* RouteConfigParse(
-  MaybeLogRouteConfiguration(client, tracer, route_config);
->>>>>>> d109f4b4
+  MaybeLogRouteConfiguration(client, tracer, symtab, route_config);
@@ -1122,7 +1113,0 @@ grpc_error* LdsResponseParse(
-<<<<<<< HEAD
-      grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config,
-                                           expected_server_name, &rds_update);
-||||||| cfd0508009
-      grpc_error* error = RouteConfigParse(client, tracer, route_config,
-                                           expected_server_name, &rds_update);
-=======
@@ -1130,2 +1115 @@ grpc_error* LdsResponseParse(
-          RouteConfigParse(client, tracer, route_config, &rds_update);
->>>>>>> d109f4b4
+          RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1193,11 +1177,2 @@ grpc_error* RdsResponseParse(
-<<<<<<< HEAD
-    grpc_error* error =
-        RouteConfigParse(client, tracer, symtab, route_config,
-                         expected_server_name, &local_rds_update);
-||||||| cfd0508009
-    grpc_error* error = RouteConfigParse(
-        client, tracer, route_config, expected_server_name, &local_rds_update);
-=======
-    grpc_error* error =
-        RouteConfigParse(client, tracer, route_config, &local_rds_update);
->>>>>>> d109f4b4
+    grpc_error* error = RouteConfigParse(client, tracer, symtab, route_config,
+                                         &local_rds_update);
@@ -1403 +1377,0 @@ grpc_error* EdsResponseParse(
-<<<<<<< HEAD
@@ -1406,6 +1379,0 @@ grpc_error* EdsResponseParse(
-    upb_strview cluster_name =
-||||||| cfd0508009
-    MaybeLogClusterLoadAssignment(client, tracer, cluster_load_assignment);
-    upb_strview cluster_name =
-=======
-    MaybeLogClusterLoadAssignment(client, tracer, cluster_load_assignment);
@@ -1413 +1380,0 @@ grpc_error* EdsResponseParse(
->>>>>>> d109f4b4
@@ -1505,5 +1471,0 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-<<<<<<< HEAD
-    result.parse_error = RdsResponseParse(
-        client_, tracer_, symtab_.ptr(), response, expected_server_name,
-        expected_route_configuration_names, &result.rds_update, arena.ptr());
-||||||| cfd0508009
@@ -1511 +1473 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-        client_, tracer_, response, expected_server_name,
+        client_, tracer_, symtab_.ptr(), response,
@@ -1513,5 +1474,0 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-=======
-    result.parse_error = RdsResponseParse(client_, tracer_, response,
-                                          expected_route_configuration_names,
-                                          &result.rds_update, arena.ptr());
->>>>>>> d109f4b4
