diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@GRPCClient@version.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@GRPCClient@version.no_comments_merged.h
index 326a957..43c9b67 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@GRPCClient@version.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@GRPCClient@version.no_comments_merged.h
@@ -2,2 +1,0 @@
-#define GRPC_OBJC_VERSION_STRING @"1.35.0-dev"
-#define GRPC_OBJC_VERSION_STRING @"1.35.0-pre1"
