diff --git a/./output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@tests@version.no_comments_conflict.h b/./output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@tests@version.no_comments_merged.h
index ab2718c..8922761 100644
--- a/./output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@tests@version.no_comments_conflict.h
+++ b/./output/grpc/be34f568-5a00d68e-46075a4c/src@objective-c@tests@version.no_comments_merged.h
@@ -1 +0,0 @@
-<<<<<<< HEAD
@@ -4,7 +2,0 @@
-||||||| 46075a4c81
-#define GRPC_OBJC_VERSION_STRING @"1.35.0-dev"
-#define GRPC_C_VERSION_STRING @"14.0.0"
-=======
-#define GRPC_OBJC_VERSION_STRING @"1.35.0-pre1"
-#define GRPC_C_VERSION_STRING @"14.0.0"
->>>>>>> 5a00d68e
