--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/be34f568-5a00d68e-46075a4c/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/be34f568-5a00d68e-46075a4c/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-std::string Version() < < < < < < < HEAD { return "1.36.0-dev"; }
-|||||||
-{ return "1.35.0-dev"; }
-=======
-{ return "1.35.0-pre1"; }
->>>>>>> 5a00d68e9af264c59b24ae24d003ab80c3a639fd
+std::string Version() { return "1.36.0-dev"; }
