--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/client.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/client.no_comments_merged.c
@@ -246,9 +267,2 @@ void client_send_identify(int flags) {
-  if ((fd = dup(STDIN_FILENO)) == -1) fatal("dup failed");
-<<<<<<< HEAD
-#endif
-  imsg_compose(&client_ibuf, MSG_IDENTIFY, PROTOCOL_VERSION, -1, fd, &data,
-               sizeof data);
-|||||||
-  imsg_compose(&client_ibuf, MSG_IDENTIFY, PROTOCOL_VERSION, -1, fd, &data,
-               sizeof data);
-=======
+ if ((fd = dup(STDIN_FILENO)) == -1)
+  fatal("dup failed");
@@ -259 +272,0 @@ void client_send_identify(int flags) {
->>>>>>> 4901d9ddc8d8c33ecdca363dcb67e66482745fa5
