--- a/./output/tmux/c190c732-4901d9dd-a9ebb62d/client.no_comments_mergegen.c
+++ b/./output/tmux/c190c732-4901d9dd-a9ebb62d/client.no_comments_merged.c
@@ -269 +269,4 @@ client_send_identify(int flags)
-#endif main_write_one(MSG_IDENTIFY_STDIN, fd, NULL, 0); for (ss = environ; *ss != NULL; ss++) client_write_one(MSG_IDENTIFY_ENVIRON, -1, *ss, strlen(*ss) + 1); client_write_one(MSG_IDENTIFY_DONE, -1, NULL, 0);
+ client_write_one(MSG_IDENTIFY_STDIN, fd, NULL, 0);
+ for (ss = environ; *ss != NULL; ss++)
+  client_write_one(MSG_IDENTIFY_ENVIRON, -1, *ss, strlen(*ss) + 1);
+ client_write_one(MSG_IDENTIFY_DONE, -1, NULL, 0);
