diff --git a/./output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_mergegen.c b/./output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_merged.c
index 9f09e46..3563d0e 100644
--- a/./output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_mergegen.c
+++ b/./output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_merged.c
@@ -1,0 +2,3 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <paths.h>
