diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-save-buffer.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-save-buffer.no_comments_merged.c
index 1f6469f..7516359 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-save-buffer.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-save-buffer.no_comments_merged.c
@@ -5,0 +6 @@
+#include <string.h>
@@ -8 +8,0 @@
-#include <string.h>
@@ -11,7 +11,4 @@ enum cmd_retval cmd_save_buffer_exec(struct cmd *, struct cmd_q *);
-const struct cmd_entry cmd_save_buffer_entry = {"save-buffer",
-                                                "saveb",
-                                                "ab:",
-                                                1,
-                                                1,
-                                                "[-a] " CMD_BUFFER_USAGE
-                                                " path",
+const struct cmd_entry cmd_save_buffer_entry = {
+ "save-buffer", "saveb",
+ "ab:", 1, 1,
+ "[-a] " CMD_BUFFER_USAGE " path",
@@ -20 +17,2 @@ const struct cmd_entry cmd_save_buffer_entry = {"save-buffer",
-                                                cmd_save_buffer_exec};
+ cmd_save_buffer_exec
+};
@@ -22,3 +20,10 @@ const struct cmd_entry cmd_show_buffer_entry = {
-    "show-buffer", "showb", "b:", 0, 0, CMD_BUFFER_USAGE, 0, NULL,
-    cmd_save_buffer_exec};
-enum cmd_retval cmd_save_buffer_exec(struct cmd *self, struct cmd_q *cmdq) {
+ "show-buffer", "showb",
+ "b:", 0, 0,
+ CMD_BUFFER_USAGE,
+ 0,
+ NULL,
+ cmd_save_buffer_exec
+};
+enum cmd_retval
+cmd_save_buffer_exec(struct cmd *self, struct cmd_q *cmdq)
+{
@@ -61 +66,2 @@ enum cmd_retval cmd_save_buffer_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if (c->session == NULL || (c->flags & CLIENT_CONTROL)) goto do_stdout;
+  if (c->session == NULL || (c->flags & CLIENT_CONTROL))
+   goto do_stdout;
@@ -73 +79,2 @@ enum cmd_retval cmd_save_buffer_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if (fd != -1) f = fdopen(fd, "ab");
+  if (fd != -1)
+   f = fdopen(fd, "ab");
@@ -76 +83,2 @@ enum cmd_retval cmd_save_buffer_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if (fd != -1) f = fdopen(fd, "wb");
+  if (fd != -1)
+   f = fdopen(fd, "wb");
@@ -79 +87,2 @@ enum cmd_retval cmd_save_buffer_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if (fd != -1) close(fd);
+  if (fd != -1)
+   close(fd);
