--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/format.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/format.no_comments_merged.c
@@ -15 +15,4 @@ void format_window_pane_tabs(struct format_tree *, struct window_pane *);
-int format_cmp(struct format_entry *fe1, struct format_entry *fe2) {
+RB_GENERATE(format_tree, format_entry, entry, format_cmp);
+int
+format_cmp(struct format_entry *fe1, struct format_entry *fe2)
+{
@@ -381,12 +432,2 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
-  if (wp->cmd != NULL) format_add(ft, "pane_start_command", "%s", wp->cmd);
-<<<<<<< HEAD
-  if (wp->cwd != NULL) format_add(ft, "pane_start_path", "%s", wp->cwd);
-  if ((cwd = osdep_get_cwd(wp->fd)) != NULL)
-    format_add(ft, "pane_current_path", "%s", cwd);
-  if ((cmd = osdep_get_name(wp->fd, wp->tty)) != NULL) {
-|||||||
-  if (wp->cwd != NULL) format_add(ft, "pane_start_path", "%s", wp->cwd);
-  if ((cwd = get_proc_cwd(wp->fd)) != NULL)
-    format_add(ft, "pane_current_path", "%s", cwd);
-  if ((cmd = get_proc_name(wp->fd, wp->tty)) != NULL) {
-=======
+ if (wp->cmd != NULL)
+  format_add(ft, "pane_start_command", "%s", wp->cmd);
@@ -394 +434,0 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
->>>>>>> 4901d9ddc8d8c33ecdca363dcb67e66482745fa5
