--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/server-client.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/server-client.no_comments_merged.c
@@ -6 +5,0 @@
-#include <paths.h>
@@ -502,24 +588,0 @@ int server_client_msg_dispatch(struct client *c) {
-<<<<<<< HEAD
-        if (datalen != sizeof commanddata) fatalx("bad MSG_COMMAND size");
-        memcpy(&commanddata, imsg.data, sizeof commanddata);
-        server_client_msg_command(c, &commanddata);
-        break;
-      case MSG_IDENTIFY:
-        if (datalen != sizeof identifydata) fatalx("bad MSG_IDENTIFY size");
-        memcpy(&identifydata, imsg.data, sizeof identifydata);
-#ifdef __CYGWIN__
-        imsg.fd = open(identifydata.ttyname, O_RDWR | O_NOCTTY);
-#endif
-        if (imsg.fd == -1) fatalx("MSG_IDENTIFY missing fd");
-        server_client_msg_identify(c, &identifydata, imsg.fd);
-|||||||
-        if (datalen != sizeof commanddata) fatalx("bad MSG_COMMAND size");
-        memcpy(&commanddata, imsg.data, sizeof commanddata);
-        server_client_msg_command(c, &commanddata);
-        break;
-      case MSG_IDENTIFY:
-        if (datalen != sizeof identifydata) fatalx("bad MSG_IDENTIFY size");
-        if (imsg.fd == -1) fatalx("MSG_IDENTIFY missing fd");
-        memcpy(&identifydata, imsg.data, sizeof identifydata);
-        server_client_msg_identify(c, &identifydata, imsg.fd);
-=======
@@ -527 +589,0 @@ int server_client_msg_dispatch(struct client *c) {
->>>>>>> 4901d9ddc8d8c33ecdca363dcb67e66482745fa5
