--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_merged.cpp
@@ -19,0 +20,11 @@ using media::audio::common::AudioUuid;
+#define VALUE_OR_RETURN_BINDER(x) \
+    ({ \
+       auto _tmp = (x); \
+       if (!_tmp.ok()) return Status::fromStatusT(_tmp.error()); \
+       std::move(_tmp.value()); \
+     })
+#define RETURN_BINDER_IF_ERROR(x) \
+    { \
+       auto _tmp = (x); \
+       if (_tmp != OK) return Status::fromStatusT(_tmp); \
+    }
