diff --git a/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_conflict.cpp b/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_merged.cpp
index 72a0740..de7a0e4 100644
--- a/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_conflict.cpp
+++ b/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@IAudioFlinger.no_comments_merged.cpp
@@ -774 +773,0 @@ status_t AudioFlingerClientAdapter::getAudioMixPort(const struct audio_port_v7 *
-<<<<<<< HEAD
@@ -783,7 +781,0 @@ status_t AudioFlingerClientAdapter::resetReferencesForTest() {
-||||||| 7f0218eb8c
-=======
-status_t AudioFlingerClientAdapter::resetReferencesForTest() {
-    RETURN_STATUS_IF_ERROR(statusTFromBinderStatus(mDelegate->resetReferencesForTest()));
-    return OK;
-}
->>>>>>> fd414a83
@@ -1267 +1258,0 @@ Status AudioFlingerServerAdapter::getAudioMixPort(const media::AudioPortFw &devi
-<<<<<<< HEAD
@@ -1276,7 +1266,0 @@ Status AudioFlingerServerAdapter::resetReferencesForTest() {
-||||||| 7f0218eb8c
-=======
-Status AudioFlingerServerAdapter::resetReferencesForTest() {
-    RETURN_BINDER_IF_ERROR(mDelegate->resetReferencesForTest());
-    return Status::ok();
-}
->>>>>>> fd414a83
