diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_merged.h
index aaf39b2..6c00640 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_merged.h
@@ -50 +50 @@ class IAudioFlinger : public virtual RefBase {
-    virtual ~IAudioFlinger()
+    virtual ~IAudioFlinger() = default;
@@ -120,2 +120,2 @@ class IAudioFlinger : public virtual RefBase {
-        static ConversionResult<CreateRecordOutput> fromAidl(
-                const media::CreateRecordResponse& aidl);
+        static ConversionResult<CreateRecordOutput>
+        fromAidl(const media::CreateRecordResponse& aidl);
@@ -140 +140,2 @@ class IAudioFlinger : public virtual RefBase {
-    virtual float streamVolume(audio_stream_type_t stream, audio_io_handle_t output) const = 0;
+    virtual float streamVolume(audio_stream_type_t stream,
+                                    audio_io_handle_t output) const = 0;
@@ -146,2 +147,4 @@ class IAudioFlinger : public virtual RefBase {
-    virtual status_t setParameters(audio_io_handle_t ioHandle, const String8& keyValuePairs) = 0;
-    virtual String8 getParameters(audio_io_handle_t ioHandle, const String8& keys) const = 0;
+    virtual status_t setParameters(audio_io_handle_t ioHandle,
+                                    const String8& keyValuePairs) = 0;
+    virtual String8 getParameters(audio_io_handle_t ioHandle, const String8& keys)
+                                    const = 0;
@@ -171 +174,2 @@ class IAudioFlinger : public virtual RefBase {
-                                         const effect_uuid_t* pTypeUUID, uint32_t preferredTypeFlag,
+                                         const effect_uuid_t *pTypeUUID,
+                                         uint32_t preferredTypeFlag,
@@ -177 +181,3 @@ class IAudioFlinger : public virtual RefBase {
-    virtual void setEffectSuspended(int effectId, audio_session_t sessionId, bool suspended) = 0;
+    virtual void setEffectSuspended(int effectId,
+                                    audio_session_t sessionId,
+                                    bool suspended) = 0;
@@ -207,2 +213,2 @@ class IAudioFlinger : public virtual RefBase {
-    virtual status_t setRequestedLatencyMode(audio_io_handle_t output,
-                                             audio_latency_mode_t mode) = 0;
+    virtual status_t setRequestedLatencyMode(
+            audio_io_handle_t output, audio_latency_mode_t mode) = 0;
@@ -244 +250,2 @@ class AudioFlingerClientAdapter : public IAudioFlinger {
-    float streamVolume(audio_stream_type_t stream, audio_io_handle_t output) const override;
+    float streamVolume(audio_stream_type_t stream,
+                       audio_io_handle_t output) const override;
@@ -250,2 +257,4 @@ class AudioFlingerClientAdapter : public IAudioFlinger {
-    status_t setParameters(audio_io_handle_t ioHandle, const String8& keyValuePairs) override;
-    String8 getParameters(audio_io_handle_t ioHandle, const String8& keys) const override;
+    status_t setParameters(audio_io_handle_t ioHandle,
+                           const String8& keyValuePairs) override;
+    String8 getParameters(audio_io_handle_t ioHandle, const String8& keys)
+    const override;
@@ -274 +283,2 @@ class AudioFlingerClientAdapter : public IAudioFlinger {
-    status_t getEffectDescriptor(const effect_uuid_t* pEffectUUID, const effect_uuid_t* pTypeUUID,
+    status_t getEffectDescriptor(const effect_uuid_t* pEffectUUID,
+                                 const effect_uuid_t* pTypeUUID,
@@ -281 +291,3 @@ class AudioFlingerClientAdapter : public IAudioFlinger {
-    void setEffectSuspended(int effectId, audio_session_t sessionId, bool suspended) override;
+    void setEffectSuspended(int effectId,
+                            audio_session_t sessionId,
+                            bool suspended) override;
@@ -300 +312,2 @@ class AudioFlingerClientAdapter : public IAudioFlinger {
-    status_t updateSecondaryOutputs(const TrackSecondaryOutputsMap& trackSecondaryOutputs) override;
+    status_t updateSecondaryOutputs(
+            const TrackSecondaryOutputsMap& trackSecondaryOutputs) override;
@@ -309,3 +322,4 @@ class AudioFlingerClientAdapter : public IAudioFlinger {
-    status_t setRequestedLatencyMode(audio_io_handle_t output, audio_latency_mode_t mode) override;
-    status_t getSupportedLatencyModes(audio_io_handle_t output,
-                                      std::vector<audio_latency_mode_t>* modes) const override;
+    status_t setRequestedLatencyMode(audio_io_handle_t output,
+            audio_latency_mode_t mode) override;
+    status_t getSupportedLatencyModes(
+            audio_io_handle_t output, std::vector<audio_latency_mode_t>* modes) const override;
@@ -367,4 +381,2 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-            ACQUIRE_AUDIO_SESSION_ID =
-                    media::BnAudioFlingerService::TRANSACTION_acquireAudioSessionId,
-            RELEASE_AUDIO_SESSION_ID =
-                    media::BnAudioFlingerService::TRANSACTION_releaseAudioSessionId,
+            ACQUIRE_AUDIO_SESSION_ID = media::BnAudioFlingerService::TRANSACTION_acquireAudioSessionId,
+            RELEASE_AUDIO_SESSION_ID = media::BnAudioFlingerService::TRANSACTION_releaseAudioSessionId,
@@ -377,4 +389,2 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-            GET_PRIMARY_OUTPUT_SAMPLING_RATE =
-                    media::BnAudioFlingerService::TRANSACTION_getPrimaryOutputSamplingRate,
-            GET_PRIMARY_OUTPUT_FRAME_COUNT =
-                    media::BnAudioFlingerService::TRANSACTION_getPrimaryOutputFrameCount,
+            GET_PRIMARY_OUTPUT_SAMPLING_RATE = media::BnAudioFlingerService::TRANSACTION_getPrimaryOutputSamplingRate,
+            GET_PRIMARY_OUTPUT_FRAME_COUNT = media::BnAudioFlingerService::TRANSACTION_getPrimaryOutputFrameCount,
@@ -387,2 +397 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-            GET_AUDIO_HW_SYNC_FOR_SESSION =
-                    media::BnAudioFlingerService::TRANSACTION_getAudioHwSyncForSession,
+            GET_AUDIO_HW_SYNC_FOR_SESSION = media::BnAudioFlingerService::TRANSACTION_getAudioHwSyncForSession,
@@ -398,2 +407 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-            UPDATE_SECONDARY_OUTPUTS =
-                    media::BnAudioFlingerService::TRANSACTION_updateSecondaryOutputs,
+            UPDATE_SECONDARY_OUTPUTS = media::BnAudioFlingerService::TRANSACTION_updateSecondaryOutputs,
@@ -401,12 +409,6 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-            GET_AAUDIO_MIXER_BURST_COUNT =
-                    media::BnAudioFlingerService::TRANSACTION_getAAudioMixerBurstCount,
-            GET_AAUDIO_HARDWARE_BURST_MIN_USEC =
-                    media::BnAudioFlingerService::TRANSACTION_getAAudioHardwareBurstMinUsec,
-            SET_DEVICE_CONNECTED_STATE =
-                    media::BnAudioFlingerService::TRANSACTION_setDeviceConnectedState,
-            SET_SIMULATE_DEVICE_CONNECTIONS =
-                    media::BnAudioFlingerService::TRANSACTION_setSimulateDeviceConnections,
-            SET_REQUESTED_LATENCY_MODE =
-                    media::BnAudioFlingerService::TRANSACTION_setRequestedLatencyMode,
-            GET_SUPPORTED_LATENCY_MODES =
-                    media::BnAudioFlingerService::TRANSACTION_getSupportedLatencyModes,
+            GET_AAUDIO_MIXER_BURST_COUNT = media::BnAudioFlingerService::TRANSACTION_getAAudioMixerBurstCount,
+            GET_AAUDIO_HARDWARE_BURST_MIN_USEC = media::BnAudioFlingerService::TRANSACTION_getAAudioHardwareBurstMinUsec,
+            SET_DEVICE_CONNECTED_STATE = media::BnAudioFlingerService::TRANSACTION_setDeviceConnectedState,
+            SET_SIMULATE_DEVICE_CONNECTIONS = media::BnAudioFlingerService::TRANSACTION_setSimulateDeviceConnections,
+            SET_REQUESTED_LATENCY_MODE = media::BnAudioFlingerService::TRANSACTION_setRequestedLatencyMode,
+            GET_SUPPORTED_LATENCY_MODES = media::BnAudioFlingerService::TRANSACTION_getSupportedLatencyModes,
@@ -419,2 +421 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-            GET_SOUND_DOSE_INTERFACE =
-                    media::BnAudioFlingerService::TRANSACTION_getSoundDoseInterface,
+            GET_SOUND_DOSE_INTERFACE = media::BnAudioFlingerService::TRANSACTION_getSoundDoseInterface,
@@ -425,3 +426 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-<<<<<<< HEAD
-            SET_TRACKS_INTERNAL_MUTE =
-                    media::BnAudioFlingerService::TRANSACTION_setTracksInternalMute,
+            SET_TRACKS_INTERNAL_MUTE = media::BnAudioFlingerService::TRANSACTION_setTracksInternalMute,
@@ -430,5 +428,0 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-||||||| 7f0218eb8c
-=======
-            RESET_REFERENCES_FOR_TEST =
-                    media::BnAudioFlingerService::TRANSACTION_resetReferencesForTest,
->>>>>>> fd414a83bb92cf343b57eb7fbab48d56e2702007
@@ -437 +431,3 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-        virtual status_t onTransactWrapper(TransactionCode code, const Parcel& data, uint32_t flags,
+        virtual status_t onTransactWrapper(TransactionCode code,
+                                           const Parcel& data,
+                                           uint32_t flags,
@@ -450 +446,2 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-    explicit AudioFlingerServerAdapter(const sp<AudioFlingerServerAdapter::Delegate>& delegate);
+    explicit AudioFlingerServerAdapter(
+            const sp<AudioFlingerServerAdapter::Delegate>& delegate);
@@ -468,2 +465,2 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-    Status setStreamVolume(media::audio::common::AudioStreamType stream, float value,
-                           int32_t output) override;
+    Status setStreamVolume(media::audio::common::AudioStreamType stream,
+                           float value, int32_t output) override;
@@ -471,2 +468,2 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-    Status streamVolume(media::audio::common::AudioStreamType stream, int32_t output,
-                        float* _aidl_return) override;
+    Status streamVolume(media::audio::common::AudioStreamType stream,
+                        int32_t output, float* _aidl_return) override;
@@ -479,2 +476,2 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-    Status getParameters(int32_t ioHandle, const std::string& keys,
-                         std::string* _aidl_return) override;
+    Status
+    getParameters(int32_t ioHandle, const std::string& keys, std::string* _aidl_return) override;
@@ -538,4 +535,4 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-    Status setRequestedLatencyMode(int output,
-                                   media::audio::common::AudioLatencyMode mode) override;
-    Status getSupportedLatencyModes(
-            int output, std::vector<media::audio::common::AudioLatencyMode>* _aidl_return) override;
+    Status setRequestedLatencyMode(
+            int output, media::audio::common::AudioLatencyMode mode) override;
+    Status getSupportedLatencyModes(int output,
+            std::vector<media::audio::common::AudioLatencyMode>* _aidl_return) override;
@@ -549 +546,2 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-    Status getAudioMixPort(const media::AudioPortFw& devicePort, const media::AudioPortFw& mixPort,
+    Status getAudioMixPort(const media::AudioPortFw& devicePort,
+                           const media::AudioPortFw& mixPort,
@@ -557 +555 @@ class AudioFlingerServerAdapter : public media::BnAudioFlingerService {
-}
+};
