--- a/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_conflict.h
+++ b/./output/frameworks_av/83c7bb9b-fd414a83-7f0218eb/media@libaudioclient@include@media@IAudioFlinger.no_comments_merged.h
@@ -226 +225,0 @@ public:
-<<<<<<< HEAD
@@ -230,4 +228,0 @@ public:
-||||||| 7f0218eb8c
-=======
-    virtual status_t resetReferencesForTest() = 0;
->>>>>>> fd414a83
@@ -340 +334,0 @@ public:
-<<<<<<< HEAD
@@ -344,4 +337,0 @@ public:
-||||||| 7f0218eb8c
-=======
-    status_t resetReferencesForTest() override;
->>>>>>> fd414a83
@@ -436 +425,0 @@ public:
-<<<<<<< HEAD
@@ -440,5 +428,0 @@ public:
-||||||| 7f0218eb8c
-=======
-            RESET_REFERENCES_FOR_TEST =
-                    media::BnAudioFlingerService::TRANSACTION_resetReferencesForTest,
->>>>>>> fd414a83
@@ -565 +548,0 @@ public:
-<<<<<<< HEAD
@@ -569,4 +551,0 @@ public:
-||||||| 7f0218eb8c
-=======
-    Status resetReferencesForTest() override;
->>>>>>> fd414a83
