--- a/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@Surface.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@Surface.no_comments_merged.cpp
@@ -1530 +1530,5 @@ int Surface::dispatchSetFrameTimelineInfo(va_list args) {
-<<<<<<< HEAD ftlInfo; ftlInfo.vsyncId = frameTimelineVsyncId; ftlInfo.inputEventId = inputEventId; ftlInfo.startTimeNanos = startTimeNanos; return setFrameTimelineInfo(frameNumber, ftlInfo);
+    FrameTimelineInfo ftlInfo;
+    ftlInfo.vsyncId = frameTimelineVsyncId;
+    ftlInfo.inputEventId = inputEventId;
+    ftlInfo.startTimeNanos = startTimeNanos;
+    return setFrameTimelineInfo(frameNumber, ftlInfo);
@@ -2166 +2170,3 @@ void Surface::ProducerListenerProxy::onBuffersDiscarded(const std::vector<int32_
-status_t Surface::setFrameTimelineInfo(uint64_t , const FrameTimelineInfo& ) {
+status_t Surface::setFrameTimelineInfo(uint64_t ,
+                                       const FrameTimelineInfo& ) {
+    return BAD_VALUE;
