--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_merged.h
@@ -17,5 +16,0 @@
-#define ATRACE_FORMAT(fmt,...) \
-    TraceUtils::TraceEnder __traceEnder = \
-            (TraceUtils::atraceFormatBegin(fmt, ##__VA_ARGS__), TraceUtils::TraceEnder())
-#define ATRACE_FORMAT_BEGIN(fmt,...) TraceUtils::atraceFormatBegin(fmt, ##__VA_ARGS__)
-#define ATRACE_FORMAT_INSTANT(fmt,...) TraceUtils::intantFormat(fmt, ##__VA_ARGS__)
@@ -47,10 +41,0 @@ public:
-    static void intantFormat(const char* fmt, ...) {
-        if (CC_LIKELY(!ATRACE_ENABLED())) return;
-        const int BUFFER_SIZE = 256;
-        va_list ap;
-        char buf[BUFFER_SIZE];
-        va_start(ap, fmt);
-        vsnprintf(buf, BUFFER_SIZE, fmt, ap);
-        va_end(ap);
-        ATRACE_INSTANT(buf);
-    }
