diff --git a/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_mergegen.h b/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_merged.h
index c0de9d1..c67af3b 100644
--- a/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_mergegen.h
+++ b/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_merged.h
@@ -17 +16,0 @@
-#define ATRACE_FORMAT_INSTANT(fmt,...) TraceUtils::intantFormat(fmt, ##__VA_ARGS__)
@@ -34 +33,4 @@ public:
-static void instantFormat(const char* fmt, ...) { const int BUFFER_SIZE = 256; va_list ap; char buf[BUFFER_SIZE]; static void intantFormat(const char* fmt, ...) { if (CC_LIKELY(!ATRACE_ENABLED())) return; const int BUFFER_SIZE = 256; va_list ap; char buf[BUFFER_SIZE]; va_start(ap, fmt); vsnprintf(buf, BUFFER_SIZE, fmt, ap); va_end(ap); ATRACE_INSTANT(buf); } static void instantFormat(const char* fmt, ...) { if (CC_LIKELY(!ATRACE_ENABLED())) return; const int BUFFER_SIZE = 256; va_list ap; char buf[BUFFER_SIZE];
+    static void instantFormat(const char* fmt, ...) {
+        const int BUFFER_SIZE = 256;
+        va_list ap;
+        char buf[BUFFER_SIZE];
