diff --git a/./output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_mergegen.cpp b/./output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_merged.cpp
index 5ce4429..2f1e7e0 100644
--- a/./output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_mergegen.cpp
+++ b/./output/flameshot/a4726482-ac7142c4-2bc3d826/src@config@configwindow.no_comments_merged.cpp
@@ -20 +20,3 @@ ConfigWindow::ConfigWindow(QWidget* parent)
-setMinimumSize(GlobalValues::buttonBaseSize() * 14, GlobalValues::buttonBaseSize() * 12);
+    const int size = GlobalValues::buttonBaseSize() * 12;
+    setMinimumSize(size, size);
+    setWindowIcon(QIcon(":img/app/org.flameshot.Flameshot.svg"));
