--- a/./output/redis/4339706e-f42846e8-3f1c8475/src@server.no_comments_conflict.h
+++ b/./output/redis/4339706e-f42846e8-3f1c8475/src@server.no_comments_merged.h
@@ -1074,4 +1073,0 @@ void moduleCallCommandFilters(client *c);
-<<<<<<< HEAD
-int moduleAllDatatypesHandleErrors();
-||||||| 3f1c84751
-=======
@@ -1079 +1075 @@ ssize_t rdbSaveModulesAux(rio *rdb, int when);
->>>>>>> f42846e8
+int moduleAllDatatypesHandleErrors();
