diff --git a/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@rdb.no_comments_mergegen.c b/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@rdb.no_comments_merged.c
index d73f7e1..215a0fd 100644
--- a/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@rdb.no_comments_mergegen.c
+++ b/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@rdb.no_comments_merged.c
@@ -1701 +1701 @@ int rdbSaveToSlavesSockets(rdbSaveInfo *rsi) {
-if (retval == C_OK) { sendChildCOWInfo(CHILD_INFO_TYPE_RDB);
+            sendChildCOWInfo(CHILD_INFO_TYPE_RDB, "RDB");
@@ -1730 +1730,4 @@ if (retval == C_OK) { sendChildCOWInfo(CHILD_INFO_TYPE_RDB);
-close(server.rdb_pipe_write); if (aeCreateFileEvent(server.el, server.rdb_pipe_read, AE_READABLE, rdbPipeReadHandler,NULL) == AE_ERR) { serverPanic("Unrecoverable error creating server.rdb_pipe_read file event."); }
+            close(server.rdb_pipe_write);
+            if (aeCreateFileEvent(server.el, server.rdb_pipe_read, AE_READABLE, rdbPipeReadHandler,NULL) == AE_ERR) {
+                serverPanic("Unrecoverable error creating server.rdb_pipe_read file event.");
+            }
