--- a/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@server.no_comments_mergegen.c
+++ b/./output/redis/c469f6ad-b8e02f2b-f4d37173/src@server.no_comments_merged.c
@@ -1123,0 +1124,4 @@ void checkChildrenDone(void) {
+        if (exitcode == SERVER_CHILD_NOERROR_RETVAL) {
+            bysignal = SIGUSR1;
+            exitcode = 1;
+        }
@@ -1126 +1130 @@ void checkChildrenDone(void) {
-                "rdb_child_pid = %d, aof_child_pid = %d",
+                "rdb_child_pid = %d, aof_child_pid = %d, module_child_pid = %d",
@@ -1129 +1133,2 @@ void checkChildrenDone(void) {
-                (int) server.aof_child_pid);
+                (int) server.aof_child_pid,
+                (int) server.module_child_pid);
@@ -1135,0 +1141,3 @@ void checkChildrenDone(void) {
+        } else if (pid == server.module_child_pid) {
+            ModuleForkDoneHandler(exitcode,bysignal);
+            if (!bysignal && exitcode == 0) receiveChildInfo();
@@ -1225 +1233 @@ int serverCron(struct aeEventLoop *eventLoop, long long id, void *clientData) {
-outChildren()) { checkChildren()) { backgroundSaveDoneHandler(exitcode,bysignal); if (!bysignal && exitcode == 0) receiveChildInfo(); } else if (pid == server.rdb_child_pid) { backgroundRewrite } if (pid == -1) { serverLog(LL_WARNING,"wait3() returned an error: %s. " "rdb_child_pid = %d, aof_child_pid, (int) server.module_child_pid);
+        checkChildrenDone();
@@ -2651 +2659,3 @@ sds genRedisInfoString(char *section) {
-(int64_t)uptime, (int64_t)(uptime/(3600*24)),
+            server.port ? server.port : server.tls_port,
+            (int64_t)uptime,
+            (int64_t)(uptime/(3600*24)),
