--- a/home/whalien/codebase/python/mergebot-eva/output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@screenshotsaver.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@screenshotsaver.no_comments_merged.cpp
@@ -14,5 +13,0 @@ ScreenshotSaver::saveToClipboard(const QPixmap& capture)
-<<<<<<< HEAD
-  SystemNotification().sendMessage(QObject::tr("Capture saved to clipboard"));
-|||||||
-  SystemNotification().sendMessage(QObject::tr("Capture saved to clipboard"));
-=======
@@ -26 +20,0 @@ ScreenshotSaver::saveToClipboard(const QPixmap& capture)
->>>>>>> dfde97e88e616995ea4b16950ec55db43ce93904
