--- a/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@confighandler.no_comments_mergegen.h
+++ b/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@utils@confighandler.no_comments_merged.h
@@ -42 +42,6 @@ public:
-bool saveAfterCopyValue(); void setSaveAfterCopy(const bool); QString saveAfterCopyPathValue(); void setSaveAfterCopyPath(const QString &);
+  bool saveAfterCopyValue();
+  void setSaveAfterCopy(const bool);
+  QString saveAfterCopyPathValue();
+  void setSaveAfterCopyPath(const QString&);
+  void setDefaults();
+  void setAllTheButtons();
