--- a/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_mergegen.h
+++ b/./output/flameshot/1662744a-dfde97e8-91ba28c3/src@config@geneneralconf.no_comments_merged.h
@@ -16 +16,10 @@ private slots:
-void showHelpChanged(bool checked); void showDesktopNotificationChanged(bool checked); void showTrayIconChanged(bool checked); void autostartChanged(bool checked); void closeAfterCaptureChanged(bool checked); void saveAfterCopyChanged(bool checked); void changeSavePath(); void importConfiguration(); void exportFileConfiguration(); void resetConfiguration(); void showDesktopNotificationChanged(bool checked); void showTrayIconChanged(bool checked); void autostartChanged(bool checked); void closeAfterCaptureChanged(bool checked); void saveAfterCopyChanged(bool checked); void changeSavePath(); void importConfiguration(); void exportFileConfiguration(); void resetConfiguration(); void showDesktopNotificationChanged(bool checked); void showTrayIconChanged(bool checked);
+  void showHelpChanged(bool checked);
+  void showDesktopNotificationChanged(bool checked);
+  void showTrayIconChanged(bool checked);
+  void autostartChanged(bool checked);
+  void closeAfterCaptureChanged(bool checked);
+  void saveAfterCopyChanged(bool checked);
+  void changeSavePath();
+  void importConfiguration();
+  void exportFileConfiguration();
+  void resetConfiguration();
