diff --git a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_conflict.cc b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_merged.cc
index a67da62..cc8edac 100644
--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_conflict.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/src@core@ext@filters@client_channel@resolver@dns@c_ares@grpc_ares_wrapper.no_comments_merged.cc
@@ -299,2 +299 @@ void grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked(
-<<<<<<< HEAD
-    bool check_grpclb, int query_timeout_ms,
+    int query_timeout_ms,
@@ -302,5 +300,0 @@ void grpc_dns_lookup_ares_continue_after_check_localhost_and_ip_literals_locked(
-||||||| 1e6c9b4c6d
-    bool check_grpclb, int query_timeout_ms, grpc_core::Combiner* combiner) {
-=======
-    int query_timeout_ms, grpc_core::Combiner* combiner) {
->>>>>>> 615cd251
@@ -563,7 +556,0 @@ static grpc_ares_request* grpc_dns_lookup_ares_locked_impl(
-<<<<<<< HEAD
-      r, dns_server, name, default_port, interested_parties, check_grpclb,
-      query_timeout_ms, std::move(work_serializer));
-||||||| 1e6c9b4c6d
-      r, dns_server, name, default_port, interested_parties, check_grpclb,
-      query_timeout_ms, combiner);
-=======
@@ -571,2 +558 @@ static grpc_ares_request* grpc_dns_lookup_ares_locked_impl(
-      combiner);
->>>>>>> 615cd251
+      std::move(work_serializer));
@@ -638,2 +624 @@ static void on_dns_lookup_done_locked(grpc_resolve_address_ares_request* r,
-  grpc_core::ExecCtx::Run(DEBUG_LOCATION, r->on_resolve_address_done,
-                          GRPC_ERROR_REF(error));
+  grpc_core::ExecCtx::Run(DEBUG_LOCATION, r->on_resolve_address_done, error);
@@ -641 +625,0 @@ static void on_dns_lookup_done_locked(grpc_resolve_address_ares_request* r,
-  GRPC_ERROR_UNREF(error);
@@ -657,9 +640,0 @@ static void grpc_resolve_address_invoke_dns_lookup_ares_locked(void* arg) {
-<<<<<<< HEAD
-      &r->on_dns_lookup_done_locked, &r->addresses, false ,
-      nullptr , GRPC_DNS_ARES_DEFAULT_QUERY_TIMEOUT_MS,
-      r->work_serializer);
-||||||| 1e6c9b4c6d
-      &r->on_dns_lookup_done_locked, &r->addresses, false ,
-      nullptr , GRPC_DNS_ARES_DEFAULT_QUERY_TIMEOUT_MS,
-      r->combiner);
-=======
@@ -668,2 +643 @@ static void grpc_resolve_address_invoke_dns_lookup_ares_locked(void* arg) {
-      GRPC_DNS_ARES_DEFAULT_QUERY_TIMEOUT_MS, r->combiner);
->>>>>>> 615cd251
+      GRPC_DNS_ARES_DEFAULT_QUERY_TIMEOUT_MS, r->work_serializer);
