--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@end2end@goaway_server_test.no_comments_conflict.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@end2end@goaway_server_test.no_comments_merged.cc
@@ -77 +77 @@ static grpc_ares_request* my_dns_lookup_ares_locked(
-    std::shared_ptr<grpc_core::WorkSerializer> combiner) {
+    std::shared_ptr<grpc_core::WorkSerializer> work_serializer) {
@@ -79,6 +78,0 @@ static grpc_ares_request* my_dns_lookup_ares_locked(
-<<<<<<< HEAD
-    return iomgr_dns_lookup_ares_locked(dns_server, addr, default_port,
-                                        interested_parties, on_done, addresses,
-                                        check_grpclb, service_config_json,
-                                        query_timeout_ms, std::move(combiner));
-||||||| 1e6c9b4c6d
@@ -87,6 +81,2 @@ static grpc_ares_request* my_dns_lookup_ares_locked(
-        check_grpclb, service_config_json, query_timeout_ms, combiner);
-=======
-    return iomgr_dns_lookup_ares_locked(
-        dns_server, addr, default_port, interested_parties, on_done, addresses,
-        balancer_addresses, service_config_json, query_timeout_ms, combiner);
->>>>>>> 615cd251
+        balancer_addresses, service_config_json, query_timeout_ms,
+        std::move(work_serializer));
