diff --git a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_conflict.cc b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_merged.cc
index 37b709d..44d2649 100644
--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_conflict.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_merged.cc
@@ -68,6 +67,0 @@ static grpc_ares_request* test_dns_lookup_ares_locked(
-<<<<<<< HEAD
-      addresses, check_grpclb, service_config_json, query_timeout_ms,
-      std::move(work_serializer));
-||||||| 1e6c9b4c6d
-      addresses, check_grpclb, service_config_json, query_timeout_ms, combiner);
-=======
@@ -75,2 +69 @@ static grpc_ares_request* test_dns_lookup_ares_locked(
-      combiner);
->>>>>>> 615cd251
+      std::move(work_serializer));
