--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@end2end@goaway_server_test.no_comments_mergegen.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@end2end@goaway_server_test.no_comments_merged.cc
@@ -77 +77 @@ static grpc_ares_request* my_dns_lookup_ares_locked(
-    std::shared_ptr<grpc_core::WorkSerializer> combiner) {
+    std::shared_ptr<grpc_core::WorkSerializer> work_serializer) {
@@ -79 +79,4 @@ static grpc_ares_request* my_dns_lookup_ares_locked(
-return iomgr_dns_lookup_ares_locked(dns_server, addr, default_port, interested_parties, on_done, addresses, balancer_addresses, service_config_json, query_timeout_ms, std::move(combiner));
+    return iomgr_dns_lookup_ares_locked(
+        dns_server, addr, default_port, interested_parties, on_done, addresses,
+        balancer_addresses, service_config_json, query_timeout_ms,
+        std::move(work_serializer));
