--- a/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_mergegen.cc
+++ b/./output/grpc/c80fdb09-615cd251-1e6c9b4c/test@core@client_channel@resolvers@dns_resolver_cooldown_test.no_comments_merged.cc
@@ -68 +68,2 @@ static grpc_ares_request* test_dns_lookup_ares_locked(
-grpc_ares_request* result = g_default_dns_lookup_ares_locked( dns_server, name, default_port, g_iomgr_args.pollset_set, on_done, addresses, balancer_addresses, service_config_json, query_timeout_ms, std::move(work_serializer)); ++g_resolution_count;
+      addresses, balancer_addresses, service_config_json, query_timeout_ms,
+      std::move(work_serializer));
