diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8e0660dc-8b1d65e8-8bf4487a/libs@binder@ndk@include_ndk@android@persistable_bundle.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8e0660dc-8b1d65e8-8bf4487a/libs@binder@ndk@include_ndk@android@persistable_bundle.no_comments_merged.h
index b01d404..99898c9 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8e0660dc-8b1d65e8-8bf4487a/libs@binder@ndk@include_ndk@android@persistable_bundle.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8e0660dc-8b1d65e8-8bf4487a/libs@binder@ndk@include_ndk@android@persistable_bundle.no_comments_merged.h
@@ -6,5 +5,0 @@
-#define __INTRODUCED_IN_LLNDK(x) 
-#endif
-#if defined(__ANDROID_VENDOR__)
-#include <android/llndk-versioning.h>
-#else
@@ -27,0 +23,4 @@ typedef char* _Nullable (*_Nonnull APersistableBundle_stringAllocator)(int32_t s
+APersistableBundle* _Nullable APersistableBundle_new() __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+APersistableBundle* _Nullable APersistableBundle_dup(const APersistableBundle* _Nonnull pBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -28,0 +28 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -31,3 +31,4 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        binder_status_t
-        APersistableBundle_readFromParcel(const AParcel* _Nonnull parcel,
-                                          APersistableBundle* _Nullable* _Nonnull outPBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+binder_status_t APersistableBundle_readFromParcel(
+        const AParcel* _Nonnull parcel, APersistableBundle* _Nullable* _Nonnull outPBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -35,0 +37 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -37,32 +39,33 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_erase(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key)
-        void APersistableBundle_putBoolean(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                bool val)
-        void APersistableBundle_putInt(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                int32_t val)
-        void APersistableBundle_putLong(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                int64_t val)
-        void APersistableBundle_putDouble(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                double val)
-        void APersistableBundle_putString(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                const char* _Nonnull val)
-        void APersistableBundle_putBooleanVector(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                const bool* _Nonnull vec,
-                int32_t num)
-        void APersistableBundle_putIntVector(APersistableBundle* _Nonnull pBundle,
-                                             const char* _Nonnull key, const int32_t* _Nonnull vec,
-                                             int32_t num)
-        void APersistableBundle_putLongVector(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                const int64_t* _Nonnull vec,
-                int32_t num)
-        void APersistableBundle_putDoubleVector(
-                APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                const double* _Nonnull vec,
-                int32_t num)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_erase(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putBoolean(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                                   bool val) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putInt(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                               int32_t val) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putLong(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                                int64_t val) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putDouble(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                                  double val) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putString(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                                  const char* _Nonnull val) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putBooleanVector(APersistableBundle* _Nonnull pBundle,
+                                         const char* _Nonnull key, const bool* _Nonnull vec,
+                                         int32_t num) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putIntVector(APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                                     const int32_t* _Nonnull vec, int32_t num)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putLongVector(APersistableBundle* _Nonnull pBundle,
+                                      const char* _Nonnull key, const int64_t* _Nonnull vec,
+                                      int32_t num) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+void APersistableBundle_putDoubleVector(APersistableBundle* _Nonnull pBundle,
+                                        const char* _Nonnull key, const double* _Nonnull vec,
+                                        int32_t num) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
@@ -71,2 +74,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-                                                const char* _Nullable const* _Nullable vec,
-                                                int32_t num)
+                                        const char* _Nullable const* _Nullable vec, int32_t num)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -75,0 +79 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -78,3 +82,4 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        bool APersistableBundle_getInt(
-                const APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                int32_t* _Nonnull val)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+bool APersistableBundle_getInt(const APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                               int32_t* _Nonnull val) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
@@ -82,0 +88 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -85,4 +91,6 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t APersistableBundle_getString(
-                const APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                char* _Nullable* _Nonnull val, APersistableBundle_stringAllocator stringAllocator,
-                void* _Nullable context)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getString(const APersistableBundle* _Nonnull pBundle,
+                                     const char* _Nonnull key, char* _Nullable* _Nonnull val,
+                                     APersistableBundle_stringAllocator stringAllocator,
+                                     void* _Nullable context) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
@@ -90,9 +98 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-                                                    const char* _Nonnull key,
-                                                    bool* _Nullable buffer, int32_t bufferSizeBytes)
-        int32_t APersistableBundle_getIntVector(
-                const APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                int32_t* _Nullable buffer,
-                int32_t bufferSizeBytes)
-        int32_t APersistableBundle_getLongVector(
-                const APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
-                int64_t* _Nullable buffer,
+                                            const char* _Nonnull key, bool* _Nullable buffer,
@@ -100,2 +100,10 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_getDoubleVector(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getIntVector(const APersistableBundle* _Nonnull pBundle,
+                                        const char* _Nonnull key, int32_t* _Nullable buffer,
+                                        int32_t bufferSizeBytes) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getLongVector(const APersistableBundle* _Nonnull pBundle,
+                                         const char* _Nonnull key, int64_t* _Nullable buffer,
+                                         int32_t bufferSizeBytes) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getDoubleVector(const APersistableBundle* _Nonnull pBundle,
@@ -104,2 +112,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_getStringVector(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getStringVector(const APersistableBundle* _Nonnull pBundle,
@@ -111,2 +119,3 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        bool APersistableBundle_getPersistableBundle(
-                const APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+bool APersistableBundle_getPersistableBundle(const APersistableBundle* _Nonnull pBundle,
+                                             const char* _Nonnull key,
@@ -114,2 +123,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_getBooleanKeys(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getBooleanKeys(const APersistableBundle* _Nonnull pBundle,
@@ -120,18 +129,24 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t APersistableBundle_getIntKeys(
-                const APersistableBundle* _Nonnull pBundle, char* _Nullable* _Nullable outKeys,
-                int32_t bufferSizeBytes, APersistableBundle_stringAllocator stringAllocator,
-                void* _Nullable context)
-        int32_t APersistableBundle_getLongKeys(
-                const APersistableBundle* _Nonnull pBundle, char* _Nullable* _Nullable outKeys,
-                int32_t bufferSizeBytes, APersistableBundle_stringAllocator stringAllocator,
-                void* _Nullable context)
-        int32_t APersistableBundle_getDoubleKeys(
-                const APersistableBundle* _Nonnull pBundle, char* _Nullable* _Nullable outKeys,
-                int32_t bufferSizeBytes, APersistableBundle_stringAllocator stringAllocator,
-                void* _Nullable context)
-        int32_t APersistableBundle_getStringKeys(
-                const APersistableBundle* _Nonnull pBundle, char* _Nullable* _Nullable outKeys,
-                int32_t bufferSizeBytes, APersistableBundle_stringAllocator stringAllocator,
-                void* _Nullable context)
-        int32_t
-        APersistableBundle_getBooleanVectorKeys(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getIntKeys(const APersistableBundle* _Nonnull pBundle,
+                                      char* _Nullable* _Nullable outKeys, int32_t bufferSizeBytes,
+                                      APersistableBundle_stringAllocator stringAllocator,
+                                      void* _Nullable context) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getLongKeys(const APersistableBundle* _Nonnull pBundle,
+                                       char* _Nullable* _Nullable outKeys, int32_t bufferSizeBytes,
+                                       APersistableBundle_stringAllocator stringAllocator,
+                                       void* _Nullable context) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getDoubleKeys(const APersistableBundle* _Nonnull pBundle,
+                                         char* _Nullable* _Nullable outKeys,
+                                         int32_t bufferSizeBytes,
+                                         APersistableBundle_stringAllocator stringAllocator,
+                                         void* _Nullable context) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getStringKeys(const APersistableBundle* _Nonnull pBundle,
+                                         char* _Nullable* _Nullable outKeys,
+                                         int32_t bufferSizeBytes,
+                                         APersistableBundle_stringAllocator stringAllocator,
+                                         void* _Nullable context) __INTRODUCED_IN(__ANDROID_API_V__)
+        __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getBooleanVectorKeys(const APersistableBundle* _Nonnull pBundle,
@@ -142,2 +157,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_getIntVectorKeys(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getIntVectorKeys(const APersistableBundle* _Nonnull pBundle,
@@ -148,2 +163,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_getLongVectorKeys(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getLongVectorKeys(const APersistableBundle* _Nonnull pBundle,
@@ -154,2 +169,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_getDoubleVectorKeys(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getDoubleVectorKeys(const APersistableBundle* _Nonnull pBundle,
@@ -160,2 +175,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-        int32_t
-        APersistableBundle_getStringVectorKeys(const APersistableBundle* _Nonnull pBundle,
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+int32_t APersistableBundle_getStringVectorKeys(const APersistableBundle* _Nonnull pBundle,
@@ -165,0 +181 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
+        __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
@@ -169 +185,2 @@ void APersistableBundle_delete(APersistableBundle* _Nullable pBundle)
-                void* _Nullable context)
+        void* _Nullable context) __INTRODUCED_IN(__ANDROID_API_V__) __INTRODUCED_IN_LLNDK(202404);
+__END_DECLS
