diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_merged.h
index 5a86b2e..69665be 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_merged.h
@@ -2 +1,0 @@
-#include <android-base/thread_annotations.h>
@@ -51,6 +50 @@ class VisualizerContext final : public EffectContext {
-    < < < < < < < HEAD int32_t mCaptureSamples = kMaxCaptureBufSize;
-||||||| a77d2bd563
-    uint32_t mCaptureSamples
-=======
-            int32_t mCaptureSamples
->>>>>>> 38b856f7cfafe276a72c6dd61891235cd065763b
+    int32_t mCaptureSamples = kMaxCaptureBufSize;
@@ -58 +52,2 @@ class VisualizerContext final : public EffectContext {
-    Visualizer::MeasurementMode mMeasurementMode = Visualizer::MeasurementMode::NONE;
+    Visualizer::MeasurementMode mMeasurementMode =
+            Visualizer::MeasurementMode::NONE;
@@ -63,2 +58 @@ class VisualizerContext final : public EffectContext {
-    REQUIRES(mMutex);
-    REQUIRES(mMutex);
+    uint32_t getDeltaTimeMsFromUpdatedTime_l();
