diff --git a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_mergegen.h b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_merged.h
index 970d811..69665be 100644
--- a/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_mergegen.h
+++ b/./output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libeffects@visualizer@aidl@VisualizerContext.no_comments_merged.h
@@ -48 +48,3 @@ class VisualizerContext final : public EffectContext {
-std::array<uint8_t, kMaxCaptureBufSize> mCaptureBuf; uint32_t mDownstreamLatency = 0; int32_t mCaptureSamples = kMaxCaptureBufSize;
+    std::array<uint8_t, kMaxCaptureBufSize> mCaptureBuf;
+    uint32_t mDownstreamLatency = 0;
+    int32_t mCaptureSamples = kMaxCaptureBufSize;
