--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/media@libaudioclient@include@media@AudioSystem.no_comments_merged.h
@@ -70,0 +74,3 @@ class AudioSystem {
+    template <typename ServiceInterface, typename Client, typename AidlInterface,
+            typename ServiceTraits>
+    friend class ServiceHandler;
@@ -259,15 +297,5 @@ class AudioSystem {
-    static status_t setDevicesRoleForStrategy(product_strategy_t strategy, device_role_t role,
-                                              const AudioDeviceTypeAddrVector& devices);
-    static status_t removeDevicesRoleForStrategy(product_strategy_t strategy, device_role_t role,
-                                                 const AudioDeviceTypeAddrVector& devices);
-    static status_t clearDevicesRoleForStrategy(product_strategy_t strategy, device_role_t role);
-    static status_t getDevicesForRoleAndStrategy(product_strategy_t strategy, device_role_t role,
-                                                 AudioDeviceTypeAddrVector& devices);
-    static status_t setDevicesRoleForCapturePreset(audio_source_t audioSource, device_role_t role,
-                                                   const AudioDeviceTypeAddrVector& devices);
-    static status_t addDevicesRoleForCapturePreset(audio_source_t audioSource, device_role_t role,
-                                                   const AudioDeviceTypeAddrVector& devices);
-    static status_t removeDevicesRoleForCapturePreset(audio_source_t audioSource,
-                                                      device_role_t role,
-                                                      const AudioDeviceTypeAddrVector& devices);
-    static status_t clearDevicesRoleForCapturePreset(audio_source_t audioSource,
+    static status_t setDevicesRoleForStrategy(product_strategy_t strategy,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t removeDevicesRoleForStrategy(product_strategy_t strategy,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t clearDevicesRoleForStrategy(product_strategy_t strategy,
@@ -274,0 +303,11 @@ class AudioSystem {
+    static status_t getDevicesForRoleAndStrategy(product_strategy_t strategy,
+            device_role_t role, AudioDeviceTypeAddrVector &devices);
+    static status_t setDevicesRoleForCapturePreset(audio_source_t audioSource,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t addDevicesRoleForCapturePreset(audio_source_t audioSource,
+            device_role_t role, const AudioDeviceTypeAddrVector &devices);
+    static status_t removeDevicesRoleForCapturePreset(
+            audio_source_t audioSource, device_role_t role,
+            const AudioDeviceTypeAddrVector& devices);
+    static status_t clearDevicesRoleForCapturePreset(
+            audio_source_t audioSource, device_role_t role);
@@ -449,8 +497,4 @@ class AudioSystem {
-    static dynamic_policy_callback gDynPolicyCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
-    static record_config_callback gRecordConfigCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
-    static routing_callback gRoutingCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
-    static vol_range_init_req_callback gVolRangeInitReqCallbacksp<CaptureStateListenerImpl>
-            AudioSystem::gSoundTriggerCaptureStateListener;
+    static dynamic_policy_callback gDynPolicyCallback GUARDED_BY(gMutex);
+    static record_config_callback gRecordConfigCallback GUARDED_BY(gMutex);
+    static routing_callback gRoutingCallback GUARDED_BY(gMutex);
+    static vol_range_init_req_callback gVolRangeInitReqCallback GUARDED_BY(gMutex);
@@ -460,2 +504 @@ class AudioSystem {
-            sp<CaptureStateListenerImpl>
-                    AudioSystem::gSoundTriggerCaptureStateListener;
+            GUARDED_BY(gErrorCallbacksMutex);
@@ -464,2 +507 @@ class AudioSystem {
-            sp<CaptureStateListenerImpl>
-                    AudioSystem::gSoundTriggerCaptureStateListener;
+            GUARDED_BY(gSoundTriggerMutex);
