diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_merged.h
index a3fa707..302239b 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c5e03bef-38b856f7-a77d2bd5/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_merged.h
@@ -18,2 +18,2 @@
-    binder::Status::fromServiceSpecificError( \
-            errorCode, fmt::sprintf("%s:%d: %s", __FUNCTION__, __LINE__, errorString).c_str())
+    binder::Status::fromServiceSpecificError(errorCode, \
+            fmt::sprintf("%s:%d: %s", __FUNCTION__, __LINE__, errorString).c_str())
@@ -21,3 +21,3 @@
-    binder::Status::fromServiceSpecificError( \
-            errorCode, \
-            fmt::sprintf("%s:%d: " errorString, __FUNCTION__, __LINE__, __VA_ARGS__).c_str())
+    binder::Status::fromServiceSpecificError(errorCode, \
+            fmt::sprintf("%s:%d: " errorString, __FUNCTION__, __LINE__, \
+                    __VA_ARGS__).c_str())
@@ -36 +36,2 @@ typedef enum camera_request_template {
-typedef std::function<CameraMetadata(const std::string&, bool overrideForPerfClass)> metadataGetter;
+typedef std::function<CameraMetadata (const std::string &, bool overrideForPerfClass)>
+        metadataGetter;
@@ -57,3 +58,4 @@ namespace SessionConfigurationUtils {
-camera3::Size getMaxJpegResolution(const CameraMetadata& metadata, bool ultraHighResolution);
-size_t getUHRMaxJpegBufferSize(camera3::Size uhrMaxJpegSize, camera3::Size defaultMaxJpegSize,
-                               size_t defaultMaxJpegBufferSize);
+camera3::Size getMaxJpegResolution(const CameraMetadata &metadata,
+        bool ultraHighResolution);
+size_t getUHRMaxJpegBufferSize(camera3::Size uhrMaxJpegSize,
+        camera3::Size defaultMaxJpegSize, size_t defaultMaxJpegBufferSize);
@@ -62,2 +64 @@ bool roundBufferDimensionNearest(int32_t width, int32_t height, int32_t format,
-                                 android_dataspace dataSpace, const CameraMetadata& info,
-                                 bool maxResolution,
+        android_dataspace dataSpace, const CameraMetadata& info, bool maxResolution,
@@ -66 +67,2 @@ bool isPublicFormat(int32_t format);
-binder::Status createSurfaceFromGbp(camera3::OutputStreamInfo& streamInfo, bool isStreamInfoValid,
+binder::Status createSurfaceFromGbp(
+        camera3::OutputStreamInfo& streamInfo, bool isStreamInfoValid,
@@ -68,5 +70,4 @@ binder::Status createSurfaceFromGbp(camera3::OutputStreamInfo& streamInfo, bool
-                                    const std::string& logicalCameraId,
-                                    const CameraMetadata& physicalCameraMetadata,
-                                    const std::vector<int32_t>& sensorPixelModesUsed,
-                                    int64_t dynamicRangeProfile, int64_t streamUseCase,
-                                    int timestampBase, int mirrorMode, int32_t colorSpace);
+        const std::string &logicalCameraId, const CameraMetadata &physicalCameraMetadata,
+        const std::vector<int32_t> &sensorPixelModesUsed, int64_t dynamicRangeProfile,
+        int64_t streamUseCase, int timestampBase, int mirrorMode,
+        int32_t colorSpace);
@@ -81,5 +82,5 @@ bool isStreamUseCaseSupported(int64_t streamUseCase, const CameraMetadata& devic
-void mapStreamInfo(const OutputStreamInfo& streamInfo, camera3::camera_stream_rotation_t rotation,
-                   const std::string& physicalId, int32_t groupId,
-                   aidl::android::hardware::camera::device::Stream* stream );
-binder::Status checkPhysicalCameraId(const std::vector<std::string>& physicalCameraIds,
-                                     const std::string& physicalCameraId,
+void mapStreamInfo(const OutputStreamInfo &streamInfo,
+        camera3::camera_stream_rotation_t rotation, const std::string &physicalId,
+        int32_t groupId, aidl::android::hardware::camera::device::Stream *stream );
+binder::Status checkPhysicalCameraId(
+const std::vector<std::string> &physicalCameraIds, const std::string &physicalCameraId,
@@ -87,6 +88,9 @@ binder::Status checkPhysicalCameraId(const std::vector<std::string>& physicalCam
-binder::Status checkSurfaceType(size_t numBufferProducers, bool deferredConsumer, int surfaceType);
-binder::Status checkOperatingMode(int operatingMode, const CameraMetadata& staticInfo,
-                                  const std::string& cameraId);
-binder::Status convertToHALStreamCombination(
-        const SessionConfiguration& sessionConfiguration, const std::string& logicalCameraId,
-        const CameraMetadata& deviceInfo, bool isCompositeJpegRDisabled, metadataGetter getMetadata,
+binder::Status checkSurfaceType(size_t numBufferProducers,
+bool deferredConsumer, int surfaceType);
+binder::Status checkOperatingMode(int operatingMode,
+const CameraMetadata &staticInfo, const std::string &cameraId);
+binder::Status
+convertToHALStreamCombination(
+    const SessionConfiguration& sessionConfiguration,
+    const std::string &logicalCameraId, const CameraMetadata &deviceInfo,
+    bool isCompositeJpegRDisabled, metadataGetter getMetadata,
@@ -95,2 +99,2 @@ binder::Status convertToHALStreamCombination(
-        bool overrideForPerfClass, metadata_vendor_id_t vendorTagId, bool checkSessionParams,
-        bool* earlyExit);
+    bool overrideForPerfClass, metadata_vendor_id_t vendorTagId,
+    bool checkSessionParams, bool *earlyExit);
@@ -102,2 +106,3 @@ status_t checkAndOverrideSensorPixelModesUsed(
-bool targetPerfClassPrimaryCamera(const std::set<std::string>& perfClassPrimaryCameraIds,
-                                  const std::string& cameraId, int32_t targetSdkVersion);
+bool targetPerfClassPrimaryCamera(
+        const std::set<std::string>& perfClassPrimaryCameraIds, const std::string& cameraId,
+        int32_t targetSdkVersion);
@@ -106,2 +111 @@ binder::Status mapRequestTemplateFromClient(const std::string& cameraId, int tem
-status_t mapRequestTemplateToAidl(
-        camera_request_template_t templateId,
+status_t mapRequestTemplateToAidl(camera_request_template_t templateId,
@@ -111,2 +115 @@ void filterParameters(const CameraMetadata& src, const CameraMetadata& deviceInf
-template <typename T>
-bool contains(std::set<T> container, T value) {
+template <typename T> bool contains(std::set<T> container, T value) {
