diff --git a/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@db_impl.no_comments_mergegen.cc b/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@db_impl.no_comments_merged.cc
index 6cc1cd9..4315747 100644
--- a/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@db_impl.no_comments_mergegen.cc
+++ b/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@db_impl.no_comments_merged.cc
@@ -261 +261 @@ Status DBImpl::NewDB() {
-new_db.SetVersionNumber(); new_db.SetComparatorName(user_comparator()->Name());
+  new_db.SetVersionNumber();
@@ -343 +343,3 @@ void DBImpl::FindObsoleteFiles(DeletionState& deletion_state,
-{"_state.log_number = versions_->MinLogNumber();
+  deletion_state.pending_manifest_file_number =
+      versions_->PendingManifestFileNumber();
+  deletion_state.log_number = versions_->MinLogNumber();
