--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_set.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_set.no_comments_merged.cc
@@ -1 +0,0 @@
-#include "db/version_set.h"
@@ -2,0 +2 @@
+#include "db/version_set.h"
@@ -112 +113,6 @@ class Version::LevelFileNumIterator : public Iterator {
-      : icmp_(icmp), flist_(flist), index_(flist->size()) {
+      : icmp_(icmp),
+        flist_(flist),
+        index_(flist->size()) {
+  }
+  virtual bool Valid() const {
+    return index_ < flist_->size();
@@ -114 +119,0 @@ class Version::LevelFileNumIterator : public Iterator {
-  virtual bool Valid() const { return index_ < flist_->size(); }
@@ -1089,7 +1115,4 @@ class VersionSet::Builder {
-VersionSet::VersionSet(const std::string& dbname, const Options* options,
-                       const DBOptions* options,
-                       const EnvOptions& storage_options, Cache* table_cache,
-                       TableCache* table_cache,
-                       const InternalKeyComparator* cmp)
-    : column_family_set_(
-          new ColumnFamilySet(dbname, options, storage_options, table_cache)),
+VersionSet::VersionSet(const std::string& dbname, const DBOptions* options,
+                       const EnvOptions& storage_options, Cache* table_cache)
+    : column_family_set_(new ColumnFamilySet(dbname, options, storage_options,
+                                             table_cache)),
@@ -1099,9 +1121,0 @@ VersionSet::VersionSet(const std::string& dbname, const Options* options,
-      table_cache_(table_cache),
-      icmp_(*cmp),
-      pending_manifest_file_number_(0),
-      last_sequence_(0),
-      num_levels_(options_->num_levels),
-      dummy_versions_(this),
-      current_(nullptr),
-      need_slowdown_for_num_level0_files_(false),
-      log_number_(0),
@@ -1109,0 +1124 @@ VersionSet::VersionSet(const std::string& dbname, const Options* options,
+      pending_manifest_file_number_(0),
@@ -1123,7 +1137,0 @@ VersionSet::~VersionSet() {
-{
-  column_family_set_.reset();
-  for (auto file : obsolete_files_) {
-    delete file;
-  }
-  obsolete_files_.clear();
-}
@@ -1202,2 +1210,2 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-<<<<<<< HEAD
-    manifest_file_number_ = NewFileNumber();
+  } else {
+    pending_manifest_file_number_ = manifest_file_number_;
@@ -1206,2 +1213,0 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-    new_manifest_filename = DescriptorFileName(dbname_, manifest_file_number_);
-    edit->SetNextFile(next_file_number_);
@@ -1211,10 +1216,0 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-||||||| 5601bc461
-    manifest_file_number_ = NewFileNumber();
-  }
-  if (new_descriptor_log) {
-    new_manifest_filename = DescriptorFileName(dbname_, manifest_file_number_);
-    edit->SetNextFile(next_file_number_);
-=======
-  } else {
-    pending_manifest_file_number_ = manifest_file_number_;
->>>>>>> 63cef90078a324233686ce0d9bb5c255545f2d3d
@@ -1306 +1301,0 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-<<<<<<< HEAD
@@ -1325,8 +1319,0 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-||||||| 5601bc461
-    manifest_file_size_ = new_manifest_file_size;
-    AppendVersion(v);
-=======
-    manifest_file_number_ = pending_manifest_file_number_;
-    manifest_file_size_ = new_manifest_file_size;
-    AppendVersion(v);
->>>>>>> 63cef90078a324233686ce0d9bb5c255545f2d3d
@@ -1338,0 +1326 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
+    manifest_file_number_ = pending_manifest_file_number_;
@@ -1520 +1509,0 @@ Status VersionSet::Recover(
-<<<<<<< HEAD
@@ -1522,8 +1510,0 @@ Status VersionSet::Recover(
-||||||| 5601bc461
-      builder.Apply(&edit);
-=======
-      builder.Apply(&edit);
-      if (edit.has_version_number_) {
-        have_version_number = true;
-      }
->>>>>>> 63cef90078a324233686ce0d9bb5c255545f2d3d
@@ -1531 +1511,0 @@ Status VersionSet::Recover(
-<<<<<<< HEAD
@@ -1533,12 +1512,0 @@ Status VersionSet::Recover(
-            s = Status::Corruption(
-                "Log Numbers in MANIFEST are not always increasing");
-          }
-          cfd->SetLogNumber(edit.log_number_);
-||||||| 5601bc461
-        if (have_log_number && log_number > edit.log_number_) {
-          s = Status::Corruption("log number decreases");
-          break;
-        }
-        log_number = edit.log_number_;
-=======
-        if (have_log_number && log_number >= edit.log_number_) {
@@ -1547,2 +1515 @@ Status VersionSet::Recover(
-          log_number = edit.log_number_;
->>>>>>> 63cef90078a324233686ce0d9bb5c255545f2d3d
+            cfd->SetLogNumber(edit.log_number_);
@@ -1551 +1518 @@ Status VersionSet::Recover(
-<<<<<<< HEAD
+        }
@@ -1560,2 +1527,2 @@ Status VersionSet::Recover(
-||||||| 5601bc461
-=======
+      if (edit.has_version_number_) {
+        have_version_number = true;
@@ -1563 +1529,0 @@ Status VersionSet::Recover(
->>>>>>> 63cef90078a324233686ce0d9bb5c255545f2d3d
@@ -1582 +1548,3 @@ Status VersionSet::Recover(
-        s = Status::Corruption("log number decreases");
+        s = Status::Corruption(
+            "MANIFEST corruption - Log numbers in records NOT "
+            "monotonically increasing");
@@ -1585,2 +1553,2 @@ Status VersionSet::Recover(
-            "decreasing of log_number is detected "
-            "in MANIFEST\n");
+            "MANIFEST corruption detected, but ignored - Log numbers in "
+            "records NOT monotonically increasing");
@@ -1929,0 +1900 @@ Status VersionSet::WriteSnapshot(log::Writer* log) {
+  bool first_record = false;
@@ -1933 +1904,4 @@ Status VersionSet::WriteSnapshot(log::Writer* log) {
-<<<<<<< HEAD
+      if (first_record) {
+        edit.SetVersionNumber();
+        first_record = false;
+      }
@@ -1947,6 +1920,0 @@ Status VersionSet::WriteSnapshot(log::Writer* log) {
-||||||| 5601bc461
-      edit.SetComparatorName(icmp_.user_comparator()->Name());
-=======
-      edit.SetVersionNumber();
-      edit.SetComparatorName(icmp_.user_comparator()->Name());
->>>>>>> 63cef90078a324233686ce0d9bb5c255545f2d3d
