diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_merged.h
index 6f1ed52..59166be 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_merged.h
@@ -63,2 +63,4 @@ class VersionEdit {
-  void AddFile(int level, uint64_t file, uint64_t file_size,
-               const InternalKey& smallest, const InternalKey& largest,
+  void AddFile(int level, uint64_t file,
+               uint64_t file_size,
+               const InternalKey& smallest,
+               const InternalKey& largest,
@@ -80 +82,3 @@ class VersionEdit {
-  int NumEntries() { return new_files_.size() + deleted_files_.size(); }
+  int NumEntries() {
+    return new_files_.size() + deleted_files_.size();
+  }
@@ -128 +132,3 @@ class VersionEdit {
-  enum { kManifestVersion = 1 };
+  enum {
+    kManifestVersion = 1
+  };
