diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/18cc2f6f-b4031fa4-37d5d939/include@grpcpp@impl@codegen@client_context_impl.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/grpc/18cc2f6f-b4031fa4-37d5d939/include@grpcpp@impl@codegen@client_context_impl.no_comments_merged.h
index 8cf0525..a8c8f33 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/18cc2f6f-b4031fa4-37d5d939/include@grpcpp@impl@codegen@client_context_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/18cc2f6f-b4031fa4-37d5d939/include@grpcpp@impl@codegen@client_context_impl.no_comments_merged.h
@@ -20 +20 @@
-struct census_context
+struct census_context;
@@ -22,0 +23 @@ namespace grpc {
+class CallCredentials;
@@ -49 +49,0 @@ class ClientContextAccessor;
-class CallCredentials;
@@ -163 +163,3 @@ class ClientContext {
-  struct census_context* census_context() const { return census_context_; }
+  struct census_context* census_context() const {
+    return census_context_;
+  }
@@ -180,0 +183,24 @@ class ClientContext {
+  template <class R>
+  friend class ::grpc_impl::ClientReader;
+  template <class W>
+  friend class ::grpc_impl::ClientWriter;
+  template <class W, class R>
+  friend class ::grpc_impl::ClientReaderWriter;
+  template <class R>
+  friend class ::grpc_impl::ClientAsyncReader;
+  template <class W>
+  friend class ::grpc_impl::ClientAsyncWriter;
+  template <class W, class R>
+  friend class ::grpc_impl::ClientAsyncReaderWriter;
+  template <class R>
+  friend class ::grpc_impl::ClientAsyncResponseReader;
+  template <class InputMessage, class OutputMessage>
+  friend class ::grpc::internal::BlockingUnaryCallImpl;
+  template <class InputMessage, class OutputMessage>
+  friend class ::grpc_impl::internal::CallbackUnaryCallImpl;
+  template <class Request, class Response>
+  friend class ::grpc_impl::internal::ClientCallbackReaderWriterImpl;
+  template <class Response>
+  friend class ::grpc_impl::internal::ClientCallbackReaderImpl;
+  template <class Request>
+  friend class ::grpc_impl::internal::ClientCallbackWriterImpl;
