--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/b923f115-b3d475f8-74ee2f75/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/b923f115-b3d475f8-74ee2f75/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_merged.cpp
@@ -191,0 +192,3 @@ std::string PrintToStringParamName(const ::testing::TestParamInfo<std::filesyste
+INSTANTIATE_TEST_CASE_P(TransactionTraceTestSuites, TransactionTraceTestSuite,
+                        testing::ValuesIn(TransactionTraceTestSuite::sTransactionTraces),
+                        PrintToStringParamName);
