diff --git a/./output/frameworks_native/b923f115-b3d475f8-74ee2f75/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_mergegen.cpp b/./output/frameworks_native/b923f115-b3d475f8-74ee2f75/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_merged.cpp
index 1cab2ad..6164ac3 100644
--- a/./output/frameworks_native/b923f115-b3d475f8-74ee2f75/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/b923f115-b3d475f8-74ee2f75/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_merged.cpp
@@ -89 +89,3 @@ inline void PrintTo(const LayerInfo& info, ::std::ostream* os) {
-struct find_id { uint64_t id; find_id(uint64_t id) : id(id) {}
+struct find_id {
+    uint64_t id;
+    find_id(uint64_t id) : id(id) {}
