diff --git a/./output/system_core/c51ed4c5-3f96053b-5658f346/fs_mgr@libsnapshot@snapshot_test.no_comments_mergegen.cpp b/./output/system_core/c51ed4c5-3f96053b-5658f346/fs_mgr@libsnapshot@snapshot_test.no_comments_merged.cpp
index a50a984..cc7fe29 100644
--- a/./output/system_core/c51ed4c5-3f96053b-5658f346/fs_mgr@libsnapshot@snapshot_test.no_comments_mergegen.cpp
+++ b/./output/system_core/c51ed4c5-3f96053b-5658f346/fs_mgr@libsnapshot@snapshot_test.no_comments_merged.cpp
@@ -2144 +2144 @@ void KillSnapuserd() {
- snapuserd_client->DetachSnapuserdd();
+    snapuserd_client->DetachSnapuserd();
