--- a/./output/system_core/c51ed4c5-3f96053b-5658f346/fs_mgr@libsnapshot@utility.no_comments_mergegen.cpp
+++ b/./output/system_core/c51ed4c5-3f96053b-5658f346/fs_mgr@libsnapshot@utility.no_comments_merged.cpp
@@ -13 +13 @@
-#include <libdm/dm.h> #include <liblp/property_fetcher.h>
+#include <liblp/property_fetcher.h>
@@ -221,4 +220,0 @@ bool KernelSupportsCompressedSnapshots() {
-bool KernelSupportsCompressedSnapshots() {
-    auto& dm = DeviceMapper::Instance();
-    return dm.GetTargetByName("user", nullptr);
-}
