diff --git a/./output/frameworks_native/b3d475f8-56d743ea-f14baedf/libs@input@MotionPredictor.no_comments_conflict.cpp b/./output/frameworks_native/b3d475f8-56d743ea-f14baedf/libs@input@MotionPredictor.no_comments_merged.cpp
index f33bb02..42dcb72 100644
--- a/./output/frameworks_native/b3d475f8-56d743ea-f14baedf/libs@input@MotionPredictor.no_comments_conflict.cpp
+++ b/./output/frameworks_native/b3d475f8-56d743ea-f14baedf/libs@input@MotionPredictor.no_comments_merged.cpp
@@ -130 +129,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-<<<<<<< HEAD
@@ -133,8 +131,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-        const TfLiteMotionPredictorSample::Point point =
-                convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]);
-||||||| f14baedf9e
-    for (int i = 0; i < predictedR.size() && predictionTime <= futureTime; ++i) {
-        const TfLiteMotionPredictorSample::Point point =
-                convertPrediction(axisFrom, axisTo, predictedR[i], predictedPhi[i]);
-=======
-    for (int i = 0; i < predictedR.size() && predictionTime <= futureTime; ++i) {
@@ -144,6 +134,0 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
->>>>>>> 56d743ea
-<<<<<<< HEAD
-        ALOGD_IF(isDebug(), "prediction %zu: %f, %f", i, point.x, point.y);
-||||||| f14baedf9e
-        ALOGD_IF(isDebug(), "prediction %d: %f, %f", i, point.x, point.y);
-=======
@@ -152,2 +137 @@ std::unique_ptr<MotionEvent> MotionPredictor::predict(nsecs_t timestamp) {
-        ALOGD_IF(isDebug(), "prediction %d: %f, %f", i, predictedPoint.x, predictedPoint.y);
->>>>>>> 56d743ea
+        ALOGD_IF(isDebug(), "prediction %zu: %f, %f", i, predictedPoint.x, predictedPoint.y);
