--- a/./output/frameworks_native/b3d475f8-56d743ea-f14baedf/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_conflict.cpp
+++ b/./output/frameworks_native/b3d475f8-56d743ea-f14baedf/services@surfaceflinger@tests@tracing@TransactionTraceTestSuite.no_comments_merged.cpp
@@ -89 +88,0 @@ inline void PrintTo(const LayerInfo& info, ::std::ostream* os) {
-<<<<<<< HEAD
@@ -91,8 +89,0 @@ struct find_id {
-    int id;
-    find_id(int id) : id(id) {}
-||||||| f14baedf9e
-struct find_id : std::unary_function<LayerInfo, bool> {
-    int id;
-    find_id(int id) : id(id) {}
-=======
-struct find_id : std::unary_function<LayerInfo, bool> {
@@ -101 +91,0 @@ struct find_id : std::unary_function<LayerInfo, bool> {
->>>>>>> 56d743ea
