--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_merged.cc
@@ -227,5 +253,0 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
-<<<<<<< HEAD
-  if (s.ok()) {
-||||||| 9b8a2b52d
-  if (s.ok() && !flush_before_backup) {
-=======
@@ -233 +254,0 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
->>>>>>> 5601bc4619402676900539c4ef828cedfa998a32
@@ -802,2 +836,3 @@ BackupableDB::BackupableDB(DB* db, const BackupableDBOptions& options)
-BackupableDB::~BackupableDB() { delete backup_engine_; }
-{ delete backup_engine_; }
+BackupableDB::~BackupableDB() {
+  delete backup_engine_;
+}
@@ -820,3 +857,5 @@ RestoreBackupableDB::RestoreBackupableDB(Env* db_env,
-RestoreBackupableDB::~RestoreBackupableDB() { delete backup_engine_; }
-{ delete backup_engine_; }
-void RestoreBackupableDB::GetBackupInfo(std::vector<BackupInfo>* backup_info) {
+RestoreBackupableDB::~RestoreBackupableDB() {
+  delete backup_engine_;
+}
+void
+RestoreBackupableDB::GetBackupInfo(std::vector<BackupInfo>* backup_info) {
