--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/40c2e498-29579cc5-38b68d85/libnativeloader@public_libraries.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/40c2e498-29579cc5-38b68d85/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -23 +22,0 @@ using internal::ConfigEntry;
-using internal::ParseApexLibrariesConfig;
@@ -24,0 +24 @@ using internal::ParseConfig;
+using internal::ParseApexLibrariesConfig;
@@ -156 +153,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -158,4 +154,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-  if (is_product_vndk_version_defined()) {
-=======
->>>>>>> 29579cc5f300a30eafd2b0ea39b7181b5cf31dce
@@ -162,0 +156 @@ static std::string InitProductPublicLibraries() {
+  }
@@ -171 +164,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -175,6 +167,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-    if (!is_product_vndk_version_defined()) {
-      ReadExtensionLibraries("/product/etc", &sonames);
-    }
-=======
->>>>>>> 29579cc5f300a30eafd2b0ea39b7181b5cf31dce
@@ -217 +203,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -222,7 +207,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-    if (!is_product_vndk_version_defined()) {
-      ALOGD("InitLlndkLibrariesProduct: No product VNDK version defined");
-      return "";
-    }
-=======
->>>>>>> 29579cc5f300a30eafd2b0ea39b7181b5cf31dce
