--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_merged.h
@@ -6,2 +5,0 @@
-#include <utils/KeyedVector.h>
-#include <utils/RefBase.h>
@@ -34,0 +34,2 @@ public:
+    std::optional<int32_t> findUsageCodeForLed(int32_t ledCode) const;
+    std::optional<AxisInfo> mapAxis(int32_t scanCode) const;
@@ -75,3 +75,0 @@ private:
-public:
-    std::optional<int32_t> findUsageCodeForLed(int32_t ledCode) const;
-    std::optional<AxisInfo> mapAxis(int32_t scanCode) const;
