diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_merged.h
index 98b4382..9c144ab 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_merged.h
@@ -6,2 +5,0 @@
-#include <utils/KeyedVector.h>
-#include <utils/RefBase.h>
@@ -23 +21,2 @@ struct AxisInfo {
-    AxisInfo() : mode(MODE_NORMAL), axis(-1), highAxis(-1), splitValue(0), flatOverride(-1) {}
+    AxisInfo() : mode(MODE_NORMAL), axis(-1), highAxis(-1), splitValue(0), flatOverride(-1) {
+    }
@@ -31,2 +30,2 @@ public:
-    status_t mapKey(int32_t scanCode, int32_t usageCode, int32_t* outKeyCode,
-                    uint32_t* outFlags) const;
+    status_t mapKey(int32_t scanCode, int32_t usageCode,
+            int32_t* outKeyCode, uint32_t* outFlags) const;
@@ -34,0 +34,2 @@ public:
+    std::optional<int32_t> findUsageCodeForLed(int32_t ledCode) const;
+    std::optional<AxisInfo> mapAxis(int32_t scanCode) const;
@@ -75,3 +75,0 @@ private:
-public:
-    std::optional<int32_t> findUsageCodeForLed(int32_t ledCode) const;
-    std::optional<AxisInfo> mapAxis(int32_t scanCode) const;
