--- a/./output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_conflict.h
+++ b/./output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/include@input@KeyLayoutMap.no_comments_merged.h
@@ -6,7 +5,0 @@
-<<<<<<< HEAD
-||||||| 0d4ac562fb
-#include <utils/KeyedVector.h>
-#include <utils/RefBase.h>
-=======
-#include <utils/KeyedVector.h>
->>>>>>> ef1a90de
@@ -33 +25,0 @@ public:
-<<<<<<< HEAD
@@ -36,6 +27,0 @@ public:
-||||||| 0d4ac562fb
-    static base::Result<std::shared_ptr<KeyLayoutMap>> load(const std::string& filename);
-    static base::Result<std::shared_ptr<KeyLayoutMap>> load(Tokenizer* tokenizer);
-=======
-    static base::Result<std::shared_ptr<KeyLayoutMap>> load(const std::string& filename);
->>>>>>> ef1a90de
