--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/libs@input@KeyLayoutMap.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/libs@input@KeyLayoutMap.no_comments_merged.cpp
@@ -9 +8,0 @@
-#include <utils/Log.h>
@@ -19,3 +17,0 @@
-#define DEBUG_PARSER 0
-#define DEBUG_PARSER_PERFORMANCE 0
-#define DEBUG_MAPPING 0
@@ -79,2 +75,2 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-KeyLayoutMap::KeyLayoutMap()
-KeyLayoutMap::~KeyLayoutMap()
+KeyLayoutMap::KeyLayoutMap() = default;
+KeyLayoutMap::~KeyLayoutMap() = default;
@@ -411 +407,0 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-<<<<<<< HEAD
@@ -415,11 +410,0 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-|||||||
-            scanCode,
-            axisInfo.mode, axisInfo.axis, axisInfo.highAxis,
-            axisInfo.splitValue, axisInfo.flatOverride);
-#endif
-    mMap->mAxes.add(scanCode, axisInfo);
-=======
-             scanCode, axisInfo.mode, axisInfo.axis, axisInfo.highAxis, axisInfo.splitValue,
-             axisInfo.flatOverride);
-    mMap->mAxes.add(scanCode, axisInfo);
->>>>>>> ef1a90de7aa6e77ce90b8015011628014ede9f2c
