diff --git a/./output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_mergegen.h b/./output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_merged.h
index df763d2..710959c 100644
--- a/./output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_mergegen.h
+++ b/./output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_merged.h
@@ -11 +11,2 @@
-const std::string kAndroidDtDir("/proc/device-tree/firmware/android/"); using android::base::boot_clock;
+const std::string kAndroidDtDir("/proc/device-tree/firmware/android/");
+using android::base::boot_clock;
