--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c986050f-be2bb331-c5e03bef/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c986050f-be2bb331-c5e03bef/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_merged.h
@@ -18,2 +18,2 @@
-    binder::Status::fromServiceSpecificError( \
-            errorCode, fmt::sprintf("%s:%d: %s", __FUNCTION__, __LINE__, errorString).c_str())
+    binder::Status::fromServiceSpecificError(errorCode, \
+            fmt::sprintf("%s:%d: %s", __FUNCTION__, __LINE__, errorString).c_str())
@@ -21,3 +21,3 @@
-    binder::Status::fromServiceSpecificError( \
-            errorCode, \
-            fmt::sprintf("%s:%d: " errorString, __FUNCTION__, __LINE__, __VA_ARGS__).c_str())
+    binder::Status::fromServiceSpecificError(errorCode, \
+            fmt::sprintf("%s:%d: " errorString, __FUNCTION__, __LINE__, \
+                    __VA_ARGS__).c_str())
