diff --git a/./output/frameworks_av/d730e941-41d44065-8a06b7a8/services@mediametrics@include@mediametricsservice@TimedAction.no_comments_conflict.h b/./output/frameworks_av/d730e941-41d44065-8a06b7a8/services@mediametrics@include@mediametricsservice@TimedAction.no_comments_merged.h
index 8583b4e..aabdd4a 100644
--- a/./output/frameworks_av/d730e941-41d44065-8a06b7a8/services@mediametrics@include@mediametricsservice@TimedAction.no_comments_conflict.h
+++ b/./output/frameworks_av/d730e941-41d44065-8a06b7a8/services@mediametrics@include@mediametricsservice@TimedAction.no_comments_merged.h
@@ -9 +8,0 @@ class TimedAction {
-<<<<<<< HEAD
@@ -12,5 +10,0 @@ class TimedAction {
-||||||| 8a06b7a825
-=======
-    using TimerClock = class std::chrono::system_clock;
-    static constexpr auto kWakeupInterval = std::chrono::minutes(15);
->>>>>>> 41d44065
