diff --git a/./output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@HeadTrackingProcessor.no_comments_conflict.cpp b/./output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@HeadTrackingProcessor.no_comments_merged.cpp
index 1b90cd8..724afc6 100644
--- a/./output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@HeadTrackingProcessor.no_comments_conflict.cpp
+++ b/./output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@HeadTrackingProcessor.no_comments_merged.cpp
@@ -8 +7,0 @@
-<<<<<<< HEAD
@@ -10,4 +8,0 @@
-||||||| 6eb188e8ba
-#include "QuaternionUtil.h"
-=======
->>>>>>> f17bebe8
