diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@Twist.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@Twist.no_comments_merged.cpp
index a76063f..7f66bd3 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@Twist.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/fdb3ca4f-f17bebe8-6eb188e8/media@libheadtracking@Twist.no_comments_merged.cpp
@@ -4 +3,0 @@
-#include "QuaternionUtil.h"
@@ -24,3 +23,2 @@ std::string Twist3f::toString() const {
-                              mTranslationalVelocity[0], mTranslationalVelocity[1],
-                              mTranslationalVelocity[2], mRotationalVelocity[0],
-                              mRotationalVelocity[1], mRotationalVelocity[2]);
+        mTranslationalVelocity[0], mTranslationalVelocity[1], mTranslationalVelocity[2],
+        mRotationalVelocity[0], mRotationalVelocity[1], mRotationalVelocity[2]);
