diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/0c34baf8-ac380be0-2396c225/odrefresh@odrefresh.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/art/0c34baf8-ac380be0-2396c225/odrefresh@odrefresh.no_comments_merged.h
index 1cf638c..c276cc7 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/0c34baf8-ac380be0-2396c225/odrefresh@odrefresh.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/0c34baf8-ac380be0-2396c225/odrefresh@odrefresh.no_comments_merged.h
@@ -22 +21,0 @@ namespace odrefresh {
-void TestOnlyEnableMainlineExtension();
@@ -28,0 +28 @@ struct BootImages {
+  OdrMetrics::BcpCompilationType GetTypeForMetrics() const;
@@ -66 +66 @@ struct CompilationResult {
-  CompilationResult()
+  CompilationResult() = default;
@@ -69,9 +68,0 @@ class PreconditionCheckResult {
- private:
-  PreconditionCheckResult(std::optional<OdrMetrics::Trigger> trigger,
-                          bool primary_boot_image_ok,
-                          bool boot_image_mainline_extension_ok,
-                          bool system_server_ok)
-      : trigger_(trigger),
-        primary_boot_image_ok_(primary_boot_image_ok),
-        boot_image_mainline_extension_ok_(boot_image_mainline_extension_ok),
-        system_server_ok_(system_server_ok) {}
@@ -108,0 +100,8 @@ class PreconditionCheckResult {
+  PreconditionCheckResult(std::optional<OdrMetrics::Trigger> trigger,
+                          bool primary_boot_image_ok,
+                          bool boot_image_mainline_extension_ok,
+                          bool system_server_ok)
+      : trigger_(trigger),
+        primary_boot_image_ok_(primary_boot_image_ok),
+        boot_image_mainline_extension_ok_(boot_image_mainline_extension_ok),
+        system_server_ok_(system_server_ok) {}
