--- a/./output/frameworks_native/a99e1cd6-8bf4487a-7c46c7a7/libs@ui@GraphicBufferMapper.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/a99e1cd6-8bf4487a-7c46c7a7/libs@ui@GraphicBufferMapper.no_comments_merged.cpp
@@ -156 +156,3 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
-return lockAsync(handle, android_convertGralloc1To0Usage(producerUsage, consumerUsage))); return mMapper->lock(handle, usage, bounds, vaddr, fenceFd);
+                                        int fenceFd) {
+    return lockAsync(handle, android_convertGralloc1To0Usage(producerUsage, consumerUsage), bounds,
+                     vaddr, fenceFd);
