--- a/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_conflict.h
+++ b/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@include@gui@TraceUtils.no_comments_merged.h
@@ -17 +16,0 @@
-#define ATRACE_FORMAT_INSTANT(fmt,...) TraceUtils::intantFormat(fmt, ##__VA_ARGS__)
@@ -34 +32,0 @@ public:
-<<<<<<< HEAD
@@ -39,15 +36,0 @@ public:
-||||||| cc7f34be98
-};
-=======
-    static void intantFormat(const char* fmt, ...) {
-        if (CC_LIKELY(!ATRACE_ENABLED())) return;
-        const int BUFFER_SIZE = 256;
-        va_list ap;
-        char buf[BUFFER_SIZE];
-        va_start(ap, fmt);
-        vsnprintf(buf, BUFFER_SIZE, fmt, ap);
-        va_end(ap);
-        ATRACE_INSTANT(buf);
-    }
-};
->>>>>>> d49f15fb
