--- a/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@Surface.no_comments_conflict.cpp
+++ b/./output/frameworks_native/65d6f39b-d49f15fb-cc7f34be/libs@gui@Surface.no_comments_merged.cpp
@@ -1530 +1529,0 @@ int Surface::dispatchSetFrameTimelineInfo(va_list args) {
-<<<<<<< HEAD
@@ -1536,5 +1534,0 @@ int Surface::dispatchSetFrameTimelineInfo(va_list args) {
-||||||| cc7f34be98
-    return setFrameTimelineInfo({frameTimelineVsyncId, inputEventId, startTimeNanos});
-=======
-    return setFrameTimelineInfo(frameNumber, {frameTimelineVsyncId, inputEventId, startTimeNanos});
->>>>>>> d49f15fb
@@ -2176 +2169,0 @@ void Surface::ProducerListenerProxy::onBuffersDiscarded(const std::vector<int32_
-<<<<<<< HEAD
@@ -2180,8 +2172,0 @@ status_t Surface::setFrameTimelineInfo(uint64_t ,
-||||||| cc7f34be98
-status_t Surface::setFrameTimelineInfo(const FrameTimelineInfo& frameTimelineInfo) {
-    return composerService()->setFrameTimelineInfo(mGraphicBufferProducer, frameTimelineInfo);
-=======
-status_t Surface::setFrameTimelineInfo(uint64_t ,
-                                       const FrameTimelineInfo& frameTimelineInfo) {
-    return composerService()->setFrameTimelineInfo(mGraphicBufferProducer, frameTimelineInfo);
->>>>>>> d49f15fb
