diff --git a/./output/frameworks_av/793c8f6c-9cd52ba9-f17bebe8/services@audiopolicy@service@Spatializer.no_comments_mergegen.h b/./output/frameworks_av/793c8f6c-9cd52ba9-f17bebe8/services@audiopolicy@service@Spatializer.no_comments_merged.h
index 8bd2a8d..7205450 100644
--- a/./output/frameworks_av/793c8f6c-9cd52ba9-f17bebe8/services@audiopolicy@service@Spatializer.no_comments_mergegen.h
+++ b/./output/frameworks_av/793c8f6c-9cd52ba9-f17bebe8/services@audiopolicy@service@Spatializer.no_comments_merged.h
@@ -203 +203,4 @@ private:
-media::VectorRecorder mPoseDurableRecorder GUARDED_BY(mLock) { 6 , std::chrono::minutes(1), mMaxLocalLogLine, { 3 }
+    media::VectorRecorder mPoseRecorder GUARDED_BY(mLock) {
+        6 , std::chrono::seconds(1), mMaxLocalLogLine, { 3 } };
+    media::VectorRecorder mPoseDurableRecorder GUARDED_BY(mLock) {
+        6 , std::chrono::minutes(1), mMaxLocalLogLine, { 3 } };
