diff --git a/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_mergegen.cpp b/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_merged.cpp
index 5a68ea9..f3ec3aa 100644
--- a/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_mergegen.cpp
+++ b/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -153 +153,3 @@ static std::string InitProductPublicLibraries() {
-if (is_product_treblelized()) { ReadExtensionLibraries("/product/etc", &sonames); }
+  if (is_product_treblelized()) {
+    ReadExtensionLibraries("/product/etc", &sonames);
+  }
@@ -162 +164,3 @@ static std::string InitExtendedPublicLibraries() {
-if (!is_product_treblelized()) { ReadExtensionLibraries("/product/etc", &sonames); }
+  if (!is_product_treblelized()) {
+    ReadExtensionLibraries("/product/etc", &sonames);
+  }
@@ -199 +203,4 @@ static std::string InitLlndkLibrariesProduct() {
-if (!is_product_treblelized()) { ALOGD("InitLlndkLibrariesProduct: Product is not treblelized"); return ""; }
+  if (!is_product_treblelized()) {
+    ALOGD("InitLlndkLibrariesProduct: Product is not treblelized");
+    return "";
+  }
@@ -339 +346,10 @@ const std::map<std::string, std::string>& apex_public_libraries() {
-bool is_product_treblelized() { #if defined(ART_TARGET_ANDROID)
+bool is_product_treblelized() {
+#if defined(ART_TARGET_ANDROID)
+  static bool product_treblelized =
+      !(android::base::GetIntProperty("ro.product.first_api_level", 0) < __ANDROID_API_R__ &&
+        !android::sysprop::VndkProperties::product_vndk_version().has_value());
+  return product_treblelized;
+#else
+  return false;
+#endif
+}
