--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@checkpoint_manager.no_comments_conflict.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@checkpoint_manager.no_comments_merged.cpp
@@ -280,7 +279,0 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
-<<<<<<< HEAD
-  auto art = make_unique<ART>(info->column_ids, TableIOManager::Get(storage), std::move(unbound_expressions),
-                              info->constraint_type, storage.db, root_block_id, root_offset);
-||||||| 3dcba6d0b5
-  auto art = make_unique<ART>(info->column_ids, TableIOManager::Get(storage), std::move(unbound_expressions),
-                              info->constraint_type, storage.db, true, root_block_id, root_offset);
-=======
@@ -288,2 +281 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
-                            info->constraint_type, storage.db, true, root_block_id, root_offset);
->>>>>>> d84e329b
+                              info->constraint_type, storage.db, root_block_id, root_offset);
