--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@catalog@catalog_entry@duck_table_entry.no_comments_mergegen.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@catalog@catalog_entry@duck_table_entry.no_comments_merged.cpp
@@ -36,0 +37,2 @@ void AddDataTableIndex(DataTable *storage, const ColumnList &columns, const vect
+  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
+                         constraint_type, storage->db, index_block->block_id, index_block->offset);
@@ -38 +40,2 @@ void AddDataTableIndex(DataTable *storage, const ColumnList &columns, const vect
-} else { art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions), constraint_type, storage->db);
+  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
+                         constraint_type, storage->db);
