--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@checkpoint_manager.no_comments_mergegen.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@checkpoint_manager.no_comments_merged.cpp
@@ -280 +280,2 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
-auto &storage = table_catalog->GetStorage(); auto art = make_uniq<ART>(info->column_ids, TableIOManager::Get(storage), std::move(unbound_expressions), info->constraint_type, storage.db, root_block_id, root_offset);
+  auto art = make_uniq<ART>(info->column_ids, TableIOManager::Get(storage), std::move(unbound_expressions),
+                              info->constraint_type, storage.db, root_block_id, root_offset);
