--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_conflict.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_merged.cpp
@@ -318,7 +317,0 @@ void ReplayState::ReplayCreateIndex() {
-<<<<<<< HEAD
-  index = make_unique<ART>(info->column_ids, TableIOManager::Get(data_table), expressions, info->constraint_type,
-                           data_table.db);
-||||||| 3dcba6d0b5
-  index = make_unique<ART>(info->column_ids, TableIOManager::Get(data_table), expressions, info->constraint_type,
-                           data_table.db, true);
-=======
@@ -326,2 +319 @@ void ReplayState::ReplayCreateIndex() {
-                         data_table.db, true);
->>>>>>> d84e329b
+                           data_table.db);
