--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@catalog@catalog_entry@duck_table_entry.no_comments_conflict.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@catalog@catalog_entry@duck_table_entry.no_comments_merged.cpp
@@ -37,2 +37 @@ void AddDataTableIndex(DataTable *storage, const ColumnList &columns, const vect
-<<<<<<< HEAD
-  art = make_unique<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
+  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
@@ -40,7 +38,0 @@ void AddDataTableIndex(DataTable *storage, const ColumnList &columns, const vect
-||||||| 3dcba6d0b5
-  art = make_unique<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
-                         constraint_type, storage->db, true, index_block->block_id, index_block->offset);
-=======
-  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions), constraint_type,
-                       storage->db, true, index_block->block_id, index_block->offset);
->>>>>>> d84e329b
@@ -48,2 +40 @@ void AddDataTableIndex(DataTable *storage, const ColumnList &columns, const vect
-<<<<<<< HEAD
-  art = make_unique<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
+  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
@@ -51,7 +41,0 @@ void AddDataTableIndex(DataTable *storage, const ColumnList &columns, const vect
-||||||| 3dcba6d0b5
-  art = make_unique<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions),
-                         constraint_type, storage->db, true);
-=======
-  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions), constraint_type,
-                       storage->db, true);
->>>>>>> d84e329b
