--- a/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_mergegen.hpp
+++ b/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@include@duckdb@execution@aggregate_hashtable.no_comments_merged.hpp
@@ -64 +64,2 @@ public:
-<<<<<<< HEAD static idx_t InitialCapacity(); TupleDataCollection &GetDataCollection() { return *data_collection; } static idx_t Count() {
+ TupleDataCollection &GetDataCollection() {
+  return *data_collection;
@@ -65,0 +67,4 @@ public:
+ idx_t Count() {
+  return data_collection->Count();
+ }
+ static idx_t InitialCapacity();
@@ -81 +86 @@ private:
- TupleDataAppendState append_state;
+ TupleDataAppendState td_append_state;
@@ -99 +104,8 @@ private:
-class ENTRY> idx_t FindOrCreateGroupsInternal(AggregateHTAppendState &state, DataChunk &groups, Vector &group_hashes_v, Vector &addresses_v, SelectionVector &new_groups);
+ void InitializeHashes();
+ template <class ENTRY>
+ idx_t FindOrCreateGroupsInternal(DataChunk &groups, Vector &group_hashes_v, Vector &addresses_v,
+                                  SelectionVector &new_groups);
+ void UpdateBlockPointers();
+ template <class ENTRY>
+ idx_t FindOrCreateGroupsInternal(AggregateHTAppendState &state, DataChunk &groups, Vector &group_hashes,
+                                  Vector &addresses, SelectionVector &new_groups);
