diff --git a/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_mergegen.cpp b/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_merged.cpp
index 8bb026c..0deb876 100644
--- a/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_mergegen.cpp
+++ b/./output/duckdb/62d1116a-4be6bdb5-41104b61/src@execution@partitionable_hashtable.no_comments_merged.cpp
@@ -40,0 +41,2 @@ idx_t PartitionableHashTable::ListAddChunk(HashTableList &list, DataChunk &group
+ if (list.empty() || list.back()->Count() + groups.size() >= list.back()->MaxCapacity()) {
+  idx_t new_capacity = GroupedAggregateHashTable::InitialCapacity();
