diff --git a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_mergegen.cpp b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_merged.cpp
index 0328f21..3e4be52 100644
--- a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_mergegen.cpp
+++ b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_merged.cpp
@@ -914,2 +914,2 @@ void RowGroupCollection::VerifyNewConstraint(DataTable &parent, const BoundConst
- vector<column_t> column_ids;
- column_ids.push_back(physical_index);
+ vector<StorageIndex> column_ids;
+ column_ids.emplace_back(physical_index);
@@ -917 +917,3 @@ void RowGroupCollection::VerifyNewConstraint(DataTable &parent, const BoundConst
-std::vector<StorageIndex> cids; cids.emplace_SCAN_COMMITTED_ROWS_OMIT_PERMANENTLY_DELETED; state.Initialize(column_ids, nullptr); InitializeScan(state.table_state, column_ids, nullptr);
+ auto scan_type = TableScanType::TABLE_SCAN_COMMITTED_ROWS_OMIT_PERMANENTLY_DELETED;
+ state.Initialize(column_ids, nullptr);
+ InitializeScan(state.table_state, column_ids, nullptr);
