diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_merged.cpp
index 762a814..ab3f876 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_merged.cpp
@@ -2 +1,0 @@
-#include "duckdb/catalog/catalog_search_path.hpp"
@@ -6 +5,5 @@
-#include "duckdb/main/secret/secret_manager.hpp"
+#include "duckdb/catalog/catalog_search_path.hpp"
+#include "duckdb/catalog/duck_catalog.hpp"
+#include "duckdb/function/scalar_macro_function.hpp"
+#include "duckdb/function/table/table_scan.hpp"
+#include "duckdb/main/attached_database.hpp"
@@ -7,0 +11 @@
+#include "duckdb/main/client_data.hpp"
@@ -8,0 +13,5 @@
+#include "duckdb/main/database_manager.hpp"
+#include "duckdb/main/secret/secret_manager.hpp"
+#include "duckdb/parser/constraints/foreign_key_constraint.hpp"
+#include "duckdb/parser/constraints/list.hpp"
+#include "duckdb/parser/constraints/unique_constraint.hpp"
@@ -12,2 +20,0 @@
-#include "duckdb/planner/expression/bound_cast_expression.hpp"
-#include "duckdb/planner/expression/bound_columnref_expression.hpp"
@@ -16,2 +22,0 @@
-#include "duckdb/parser/parsed_data/create_view_info.hpp"
-#include "duckdb/parser/tableref/table_function_ref.hpp"
@@ -18,0 +24 @@
+#include "duckdb/parser/parsed_data/create_view_info.hpp"
@@ -20,0 +27,2 @@
+#include "duckdb/parser/tableref/basetableref.hpp"
+#include "duckdb/parser/tableref/table_function_ref.hpp"
@@ -22,0 +31,2 @@
+#include "duckdb/planner/expression/bound_cast_expression.hpp"
+#include "duckdb/planner/expression/bound_columnref_expression.hpp"
@@ -26 +35,0 @@
-#include "duckdb/planner/operator/logical_create_index.hpp"
@@ -33,3 +41,0 @@
-#include "duckdb/parser/constraints/foreign_key_constraint.hpp"
-#include "duckdb/function/scalar_macro_function.hpp"
-#include "duckdb/storage/data_table.hpp"
@@ -37,9 +42,0 @@
-#include "duckdb/main/client_data.hpp"
-#include "duckdb/parser/constraints/unique_constraint.hpp"
-#include "duckdb/parser/constraints/list.hpp"
-#include "duckdb/main/database_manager.hpp"
-#include "duckdb/main/attached_database.hpp"
-#include "duckdb/catalog/duck_catalog.hpp"
-#include "duckdb/function/table/table_scan.hpp"
-#include "duckdb/parser/tableref/basetableref.hpp"
-#include "duckdb/planner/expression_binder/select_bind_state.hpp"
@@ -496,38 +492,0 @@ unique_ptr<LogicalOperator> DuckCatalog::BindCreateIndex(Binder &binder, CreateS
-<<<<<<< HEAD
- auto &column_ids = get.GetColumnIds();
- for (auto &column_id : column_ids) {
-  if (column_id.IsRowIdColumn()) {
-   throw BinderException("Cannot create an index on the rowid!");
-  }
-  auto col_id = column_id.GetPrimaryIndex();
-  create_index_info->column_ids.push_back(col_id);
-  create_index_info->scan_types.push_back(get.returned_types[col_id]);
- }
- create_index_info->scan_types.emplace_back(LogicalType::ROW_TYPE);
- create_index_info->names = get.names;
- create_index_info->schema = table.schema.name;
- auto &bind_data = get.bind_data->Cast<TableScanBindData>();
- bind_data.is_create_index = true;
- get.AddColumnId(COLUMN_IDENTIFIER_ROW_ID);
- auto result = make_uniq<LogicalCreateIndex>(std::move(create_index_info), std::move(expressions), table);
- result->children.push_back(std::move(plan));
- return std::move(result);
-|||||||
- auto &column_ids = get.GetColumnIds();
- for (auto &column_id : column_ids) {
-  if (column_id == COLUMN_IDENTIFIER_ROW_ID) {
-   throw BinderException("Cannot create an index on the rowid!");
-  }
-  create_index_info->scan_types.push_back(get.returned_types[column_id]);
- }
- create_index_info->scan_types.emplace_back(LogicalType::ROW_TYPE);
- create_index_info->names = get.names;
- create_index_info->column_ids = column_ids;
- create_index_info->schema = table.schema.name;
- auto &bind_data = get.bind_data->Cast<TableScanBindData>();
- bind_data.is_create_index = true;
- get.AddColumnId(COLUMN_IDENTIFIER_ROW_ID);
- auto result = make_uniq<LogicalCreateIndex>(std::move(create_index_info), std::move(expressions), table);
- result->children.push_back(std::move(plan));
- return std::move(result);
-=======
@@ -536 +494,0 @@ unique_ptr<LogicalOperator> DuckCatalog::BindCreateIndex(Binder &binder, CreateS
->>>>>>> eebe9bceb34a3ed24f02abd9fdec23408122c0a6
