diff --git a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_conflict.cpp b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_merged.cpp
index 9457182..ab3f876 100644
--- a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_conflict.cpp
+++ b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_merged.cpp
@@ -493,38 +492,0 @@ unique_ptr<LogicalOperator> DuckCatalog::BindCreateIndex(Binder &binder, CreateS
-<<<<<<< HEAD
- auto &column_ids = get.GetColumnIds();
- for (auto &column_id : column_ids) {
-  if (column_id.IsRowIdColumn()) {
-   throw BinderException("Cannot create an index on the rowid!");
-  }
-  auto col_id = column_id.GetPrimaryIndex();
-  create_index_info->column_ids.push_back(col_id);
-  create_index_info->scan_types.push_back(get.returned_types[col_id]);
- }
- create_index_info->scan_types.emplace_back(LogicalType::ROW_TYPE);
- create_index_info->names = get.names;
- create_index_info->schema = table.schema.name;
- auto &bind_data = get.bind_data->Cast<TableScanBindData>();
- bind_data.is_create_index = true;
- get.AddColumnId(COLUMN_IDENTIFIER_ROW_ID);
- auto result = make_uniq<LogicalCreateIndex>(std::move(create_index_info), std::move(expressions), table);
- result->children.push_back(std::move(plan));
- return std::move(result);
-||||||| 9afef29d90
- auto &column_ids = get.GetColumnIds();
- for (auto &column_id : column_ids) {
-  if (column_id == COLUMN_IDENTIFIER_ROW_ID) {
-   throw BinderException("Cannot create an index on the rowid!");
-  }
-  create_index_info->scan_types.push_back(get.returned_types[column_id]);
- }
- create_index_info->scan_types.emplace_back(LogicalType::ROW_TYPE);
- create_index_info->names = get.names;
- create_index_info->column_ids = column_ids;
- create_index_info->schema = table.schema.name;
- auto &bind_data = get.bind_data->Cast<TableScanBindData>();
- bind_data.is_create_index = true;
- get.AddColumnId(COLUMN_IDENTIFIER_ROW_ID);
- auto result = make_uniq<LogicalCreateIndex>(std::move(create_index_info), std::move(expressions), table);
- result->children.push_back(std::move(plan));
- return std::move(result);
-=======
@@ -533 +494,0 @@ unique_ptr<LogicalOperator> DuckCatalog::BindCreateIndex(Binder &binder, CreateS
->>>>>>> eebe9bce
