diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_merged.cpp
index 60c566c..46725d9 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_merged.cpp
@@ -503 +502,0 @@ void WriteAheadLogDeserializer::ReplayCreateIndex() {
-<<<<<<< HEAD
@@ -505,5 +503,0 @@ void WriteAheadLogDeserializer::ReplayCreateIndex() {
-|||||||
- vector<column_t> column_ids;
-=======
- vector<column_t> column_ids;
->>>>>>> eebe9bceb34a3ed24f02abd9fdec23408122c0a6
