diff --git a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_mergegen.cpp b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_merged.cpp
index e3ce097..46725d9 100644
--- a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_mergegen.cpp
+++ b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_merged.cpp
@@ -502,0 +503 @@ void WriteAheadLogDeserializer::ReplayCreateIndex() {
+ vector<ColumnIndex> column_ids;
