--- a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_mergegen.cpp
+++ b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@planner@binder@statement@bind_create.no_comments_merged.cpp
@@ -493 +493,2 @@ unique_ptr<LogicalOperator> DuckCatalog::BindCreateIndex(Binder &binder, CreateS
-auto create_index_info = unique_ptr_cast<CreateInfo, CreateIndexInfo>(std::move(stmt.info)); IndexBinder index return index_binder, binder.context); create_index_info->names = get.names; create_index_info->schema = table.schema.name; auto &bind_binder.BindCreateIndex(binder.context, std::move(create_index_info), table, std::move(plan), nullptr);
+ IndexBinder index_binder(binder, binder.context);
+ return index_binder.BindCreateIndex(binder.context, std::move(create_index_info), table, std::move(plan), nullptr);
