--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/7c748c06-041d8e2a-70bc5f77/src@aof.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/7c748c06-041d8e2a-70bc5f77/src@aof.no_comments_merged.c
@@ -2,0 +3 @@
+#include "rio.h"
@@ -10 +11,4 @@
-#include "rio.h"
+void aofUpdateCurrentSize(void);
+void aof_background_fsync(int fd) {
+    bioCreateBackgroundJob(REDIS_BIO_AOF_FSYNC,(void*)(long)fd,NULL,NULL);
+}
@@ -32,4 +36 @@ int startAppendOnly(void) {
-    redisLog(REDIS_WARNING,
-             "Used tried to switch on AOF via CONFIG, but I can't open the AOF "
-             "file: %s",
-             strerror(errno));
+        redisLog(REDIS_WARNING,"Used tried to switch on AOF via CONFIG, but I can't open the AOF file: %s",strerror(errno));
@@ -41,5 +42 @@ int startAppendOnly(void) {
-    redisLog(REDIS_WARNING,
-             "Used tried to switch on AOF via CONFIG, I can't trigger a "
-             "background AOF rewrite operation. Check the above logs for more "
-             "info about the error.",
-             strerror(errno));
+        redisLog(REDIS_WARNING,"Used tried to switch on AOF via CONFIG, I can't trigger a background AOF rewrite operation. Check the above logs for more info about the error.",strerror(errno));
@@ -64,4 +61 @@ void flushAppendOnlyFile(int force) {
-      redisLog(REDIS_NOTICE,
-               "Asynchronous AOF fsync is taking too long (disk is busy?). "
-               "Writing the AOF buffer without waiting for fsync to complete, "
-               "this may slow down Redis.");
+            redisLog(REDIS_NOTICE,"Asynchronous AOF fsync is taking too long (disk is busy?). Writing the AOF buffer without waiting for fsync to complete, this may slow down Redis.");
@@ -277,3 +260 @@ fmterr:
-  redisLog(REDIS_WARNING,
-           "Bad file format reading the append only file: make a backup of "
-           "your AOF file, then use ./redis-check-aof --fix <filename>");
+    redisLog(REDIS_WARNING,"Bad file format reading the append only file: make a backup of your AOF file, then use ./redis-check-aof --fix <filename>");
@@ -572,3 +551 @@ void backgroundRewriteDoneHandler(int exitcode, int bysignal) {
-                 "Short write trying to flush the parent diff to the rewritten "
-                 "AOF: %s",
-                 strerror(errno));
+                    "Short write trying to flush the parent diff to the rewritten AOF: %s", strerror(errno));
@@ -628,4 +601,0 @@ cleanup:
-void aofUpdateCurrentSize(void);
-void aof_background_fsync(int fd) {
-  bioCreateBackgroundJob(REDIS_BIO_AOF_FSYNC, (void *)(long)fd, NULL, NULL);
-}
