--- a/./output/art/0c34baf8-ac380be0-2396c225/odrefresh@odrefresh.no_comments_conflict.h
+++ b/./output/art/0c34baf8-ac380be0-2396c225/odrefresh@odrefresh.no_comments_merged.h
@@ -22 +21,0 @@ namespace odrefresh {
-<<<<<<< HEAD
@@ -31,11 +29,0 @@ struct BootImages {
-||||||| 2396c22559
-=======
-void TestOnlyEnableMainlineExtension();
-class OnDeviceRefresh;
-struct BootImages {
-  static constexpr int kMaxCount = 2;
-  bool primary_boot_image : 1;
-  bool boot_image_mainline_extension : 1;
-  int Count() const;
-};
->>>>>>> ac380be0
