diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_merged.cc
index f8728dc..eefa794 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_merged.cc
@@ -19 +19 @@ void grpc_lb_policy_cds_shutdown(void);
-void grpc_lb_policy_xds_init(void);
+void grpc_lb_policy_eds_init(void);
@@ -26,0 +27,2 @@ void grpc_lb_policy_weighted_target_shutdown(void);
+void grpc_lb_policy_xds_routing_init(void);
+void grpc_lb_policy_xds_routing_shutdown(void);
@@ -50,2 +52,4 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_http_filters_init, grpc_http_filters_shutdown);
-  grpc_register_plugin(grpc_chttp2_plugin_init, grpc_chttp2_plugin_shutdown);
+  grpc_register_plugin(grpc_http_filters_init,
+                       grpc_http_filters_shutdown);
+  grpc_register_plugin(grpc_chttp2_plugin_init,
+                       grpc_chttp2_plugin_shutdown);
@@ -54,3 +58,6 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_client_channel_init, grpc_client_channel_shutdown);
-  grpc_register_plugin(grpc_inproc_plugin_init, grpc_inproc_plugin_shutdown);
-  grpc_register_plugin(grpc_resolver_fake_init, grpc_resolver_fake_shutdown);
+  grpc_register_plugin(grpc_client_channel_init,
+                       grpc_client_channel_shutdown);
+  grpc_register_plugin(grpc_inproc_plugin_init,
+                       grpc_inproc_plugin_shutdown);
+  grpc_register_plugin(grpc_resolver_fake_init,
+                       grpc_resolver_fake_shutdown);
@@ -59,23 +66,6 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_lb_policy_cds_init, grpc_lb_policy_cds_shutdown);
-<<<<<<< HEAD
-  grpc_register_plugin(grpc_lb_policy_xds_init, grpc_lb_policy_xds_shutdown);
-  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
-                       grpc_lb_policy_xds_routing_shutdown);
-|||||||
-<<<<<<< HEAD
-  grpc_register_plugin(grpc_lb_policy_xds_init, grpc_lb_policy_xds_shutdown);
-  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
-                       grpc_lb_policy_xds_routing_shutdown);
-|||||||
-  grpc_register_plugin(grpc_lb_policy_xds_init, grpc_lb_policy_xds_shutdown);
-=======
-  grpc_register_plugin(grpc_lb_policy_eds_init, grpc_lb_policy_eds_shutdown);
-  grpc_register_plugin(grpc_lb_policy_lrs_init, grpc_lb_policy_lrs_shutdown);
-  grpc_register_plugin(grpc_lb_policy_priority_init,
-                       grpc_lb_policy_priority_shutdown);
-  grpc_register_plugin(grpc_lb_policy_weighted_target_init,
-                       grpc_lb_policy_weighted_target_shutdown);
->>>>>>> 9e82d59cd770e2c5c2b899bf44d943bb6b81cfbd
-=======
-  grpc_register_plugin(grpc_lb_policy_eds_init, grpc_lb_policy_eds_shutdown);
-  grpc_register_plugin(grpc_lb_policy_lrs_init, grpc_lb_policy_lrs_shutdown);
+  grpc_register_plugin(grpc_lb_policy_cds_init,
+                       grpc_lb_policy_cds_shutdown);
+  grpc_register_plugin(grpc_lb_policy_eds_init,
+                       grpc_lb_policy_eds_shutdown);
+  grpc_register_plugin(grpc_lb_policy_lrs_init,
+                       grpc_lb_policy_lrs_shutdown);
@@ -86 +76,2 @@ void grpc_register_built_in_plugins(void) {
->>>>>>> 9e82d59cd770e2c5c2b899bf44d943bb6b81cfbd
+  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
+                       grpc_lb_policy_xds_routing_shutdown);
@@ -97 +88,2 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_resolver_xds_init, grpc_resolver_xds_shutdown);
+  grpc_register_plugin(grpc_resolver_xds_init,
+                       grpc_resolver_xds_shutdown);
@@ -100 +92,2 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_max_age_filter_init, grpc_max_age_filter_shutdown);
+  grpc_register_plugin(grpc_max_age_filter_init,
+                       grpc_max_age_filter_shutdown);
@@ -108,2 +100,0 @@ void grpc_register_built_in_plugins(void) {
-void grpc_lb_policy_xds_routing_init(void);
-void grpc_lb_policy_xds_routing_shutdown(void);
