--- a/./output/duckdb/e64412da-8656b2cc-fb51e97b/src@include@duckdb@common@string_util.no_comments_mergegen.hpp
+++ b/./output/duckdb/e64412da-8656b2cc-fb51e97b/src@include@duckdb@common@string_util.no_comments_merged.hpp
@@ -178 +178,10 @@ public:
-struct EnumStringLiteral { uint32_t number; const char *string; }; DUCKDB_API static uint32_t StringToEnum(const EnumStringLiteral enum_list[], idx_t enum_count, const char *enum_name, const char *str_value); DUCKDB_API static const char *EnumToString(const EnumStringLiteral enum_list[], idx_t enum_count, const char *enum_name, uint32_t enum_value);
+ struct EnumStringLiteral {
+  uint32_t number;
+  const char *string;
+ };
+ DUCKDB_API static uint32_t StringToEnum(const EnumStringLiteral enum_list[], idx_t enum_count,
+                                         const char *enum_name, const char *str_value);
+ DUCKDB_API static const char *EnumToString(const EnumStringLiteral enum_list[], idx_t enum_count,
+                                            const char *enum_name, uint32_t enum_value);
+ DUCKDB_API static const uint8_t ASCII_TO_LOWER_MAP[];
+ DUCKDB_API static const uint8_t ASCII_TO_UPPER_MAP[];
