--- a/./output/frameworks_av/763e86dc-67110b51-c7caded3/services@camera@libcameraservice@CameraService.no_comments_mergegen.h
+++ b/./output/frameworks_av/763e86dc-67110b51-c7caded3/services@camera@libcameraservice@CameraService.no_comments_merged.h
@@ -435 +435,4 @@ private:
-int uid, int pid, bool isVendorClient, bool openCloseCallbackAllowed) : mParent(parent), mListener(listener), mListenerUid(uid), mListenerPid(pid), mIsVendorListener(isVendorClient), mOpenCloseCallbackAllowed(openCloseCallbackAllowed) {
+                    int uid, int pid, bool isVendorClient, bool openCloseCallbackAllowed)
+                    : mParent(parent), mListener(listener), mListenerUid(uid), mListenerPid(pid),
+                      mIsVendorListener(isVendorClient),
+                      mOpenCloseCallbackAllowed(openCloseCallbackAllowed) { }
@@ -452 +456,4 @@ bool isVendorListener() { return mIsVendorListener; } bool isOpenCloseCallbackAl
-int mListenerUid = -1; int mListenerPid = -1; bool mIsVendorListener = false; bool mOpenCloseCallbackAllowed = false; int mListenerPid;
+            int mListenerUid = -1;
+            int mListenerPid = -1;
+            bool mIsVendorListener = false;
+            bool mOpenCloseCallbackAllowed = false;
