diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/3898d4e7-135bb1ed-86862c97/tmux.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/3898d4e7-135bb1ed-86862c97/tmux.no_comments_merged.c
index add75ad..cdcf703 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/3898d4e7-135bb1ed-86862c97/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/3898d4e7-135bb1ed-86862c97/tmux.no_comments_merged.c
@@ -4 +3,0 @@
-#include <err.h>
@@ -8 +6,0 @@
-#include <paths.h>
@@ -13 +10,0 @@
-#include <util.h>
@@ -31 +28,3 @@ static const char *getshell(void);
-static __dead void usage(void) {
+static __dead void
+usage(void)
+{
@@ -38 +37,3 @@ static __dead void usage(void) {
-static const char *getshell(void) {
+static const char *
+getshell(void)
+{
@@ -42 +43,2 @@ static const char *getshell(void) {
-  if (checkshell(shell)) return (shell);
+ if (checkshell(shell))
+  return (shell);
@@ -44 +46,2 @@ static const char *getshell(void) {
-  if (pw != NULL && checkshell(pw->pw_shell)) return (pw->pw_shell);
+ if (pw != NULL && checkshell(pw->pw_shell))
+  return (pw->pw_shell);
@@ -47,4 +50,9 @@ static const char *getshell(void) {
-int checkshell(const char *shell) {
-  if (shell == NULL || *shell != '/') return (0);
-  if (areshell(shell)) return (0);
-  if (access(shell, X_OK) != 0) return (0);
+int
+checkshell(const char *shell)
+{
+ if (shell == NULL || *shell != '/')
+  return (0);
+ if (areshell(shell))
+  return (0);
+ if (access(shell, X_OK) != 0)
+  return (0);
@@ -53 +61,3 @@ int checkshell(const char *shell) {
-static int areshell(const char *shell) {
+static int
+areshell(const char *shell)
+{
@@ -60,2 +70,4 @@ static int areshell(const char *shell) {
-  if (*progname == '-') progname++;
-  if (strcmp(ptr, progname) == 0) return (1);
+ if (*progname == '-')
+  progname++;
+ if (strcmp(ptr, progname) == 0)
+  return (1);
@@ -64 +76,3 @@ static int areshell(const char *shell) {
-static char *make_label(const char *label, char **cause) {
+static char *
+make_label(const char *label, char **cause)
+{
@@ -69 +83,2 @@ static char *make_label(const char *label, char **cause) {
-  if (label == NULL) label = "default";
+ if (label == NULL)
+  label = "default";
@@ -82,2 +97,4 @@ static char *make_label(const char *label, char **cause) {
-  if (mkdir(resolved, S_IRWXU) != 0 && errno != EEXIST) goto fail;
-  if (lstat(resolved, &sb) != 0) goto fail;
+ if (mkdir(resolved, S_IRWXU) != 0 && errno != EEXIST)
+  goto fail;
+ if (lstat(resolved, &sb) != 0)
+  goto fail;
@@ -98 +115,3 @@ fail:
-void setblocking(int fd, int state) {
+void
+setblocking(int fd, int state)
+{
@@ -108 +127,3 @@ void setblocking(int fd, int state) {
-const char *sig2name(int signo) {
+const char *
+sig2name(int signo)
+{
@@ -111 +132,2 @@ const char *sig2name(int signo) {
-  if (signo > 0 && signo < NSIG) return (sys_signame[signo]);
+     if (signo > 0 && signo < NSIG)
+      return (sys_signame[signo]);
@@ -116 +138,3 @@ const char *sig2name(int signo) {
-const char *find_cwd(void) {
+const char *
+find_cwd(void)
+{
@@ -120,5 +144,10 @@ const char *find_cwd(void) {
-  if (getcwd(cwd, sizeof cwd) == NULL) return (NULL);
-  if ((pwd = getenv("PWD")) == NULL || *pwd == '\0') return (cwd);
-  if (realpath(pwd, resolved1) == NULL) return (cwd);
-  if (realpath(cwd, resolved2) == NULL) return (cwd);
-  if (strcmp(resolved1, resolved2) != 0) return (cwd);
+ if (getcwd(cwd, sizeof cwd) == NULL)
+  return (NULL);
+ if ((pwd = getenv("PWD")) == NULL || *pwd == '\0')
+  return (cwd);
+ if (realpath(pwd, resolved1) == NULL)
+  return (cwd);
+ if (realpath(cwd, resolved2) == NULL)
+  return (cwd);
+ if (strcmp(resolved1, resolved2) != 0)
+  return (cwd);
@@ -127 +156,3 @@ const char *find_cwd(void) {
-const char *find_home(void) {
+const char *
+find_home(void)
+{
@@ -130 +161,2 @@ const char *find_home(void) {
-  if (home != NULL) return (home);
+ if (home != NULL)
+  return (home);
@@ -141,2 +173,8 @@ const char *find_home(void) {
-const char *getversion(void) { return TMUX_VERSION; }
-int main(int argc, char **argv) {
+const char *
+getversion(void)
+{
+ return TMUX_VERSION;
+}
+int
+main(int argc, char **argv)
+{
@@ -159 +197,2 @@ int main(int argc, char **argv) {
-  if (**argv == '-') flags = CLIENT_LOGIN;
+ if (**argv == '-')
+  flags = CLIENT_LOGIN;
@@ -208,2 +247,4 @@ int main(int argc, char **argv) {
-  if (shell_command != NULL && argc != 0) usage();
-  if ((ptm_fd = getptmfd()) == -1) err(1, "getptmfd");
+ if (shell_command != NULL && argc != 0)
+  usage();
+ if ((ptm_fd = getptmfd()) == -1)
+  err(1, "getptmfd");
@@ -211,2 +252 @@ int main(int argc, char **argv) {
-             "recvfd proc exec tty ps",
-             NULL) != 0)
+     "recvfd proc exec tty ps", NULL) != 0)
@@ -218,4 +258,8 @@ int main(int argc, char **argv) {
-    if (s == NULL || *s == '\0') s = getenv("LC_CTYPE");
-    if (s == NULL || *s == '\0') s = getenv("LANG");
-    if (s == NULL || *s == '\0') s = "";
-    if (strcasestr(s, "UTF-8") != NULL || strcasestr(s, "UTF8") != NULL)
+  if (s == NULL || *s == '\0')
+   s = getenv("LC_CTYPE");
+  if (s == NULL || *s == '\0')
+   s = getenv("LANG");
+  if (s == NULL || *s == '\0')
+   s = "";
+  if (strcasestr(s, "UTF-8") != NULL ||
+      strcasestr(s, "UTF8") != NULL)
@@ -225 +269,2 @@ int main(int argc, char **argv) {
-  for (var = environ; *var != NULL; var++) environ_put(global_environ, *var, 0);
+ for (var = environ; *var != NULL; var++)
+  environ_put(global_environ, *var, 0);
@@ -232 +277,2 @@ int main(int argc, char **argv) {
-    if (oe->scope & OPTIONS_TABLE_SERVER) options_default(global_options, oe);
+  if (oe->scope & OPTIONS_TABLE_SERVER)
+   options_default(global_options, oe);
@@ -235 +281,2 @@ int main(int argc, char **argv) {
-    if (oe->scope & OPTIONS_TABLE_WINDOW) options_default(global_w_options, oe);
+  if (oe->scope & OPTIONS_TABLE_WINDOW)
+   options_default(global_w_options, oe);
@@ -240 +287,2 @@ int main(int argc, char **argv) {
-    if (strrchr(s, '/') != NULL) s = strrchr(s, '/') + 1;
+  if (strrchr(s, '/') != NULL)
+   s = strrchr(s, '/') + 1;
@@ -267,5 +314,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
-  exit(client_main(osdep_event_init(), argc, argv, flags));
-|||||||
-  exit(client_main(event_init(), argc, argv, flags));
-=======
@@ -273 +315,0 @@ int main(int argc, char **argv) {
->>>>>>> 135bb1edeeab3faae8001100aa7c173be9aa91e1
