--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@AudioProfile.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@AudioProfile.no_comments_merged.cpp
@@ -204 +218,2 @@ bool AudioProfileVector::hasDynamicRateFor(audio_format_t format) const {
-void AudioProfileVectorBase::dump(std::string *dst, int spaces) const {
+void AudioProfileVector::dump(std::string *dst, int spaces) const
+{
