diff --git a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@AudioProfile.no_comments_conflict.cpp b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@AudioProfile.no_comments_merged.cpp
index 7ce13db..b980c87 100644
--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@AudioProfile.no_comments_conflict.cpp
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/media@libaudiofoundation@AudioProfile.no_comments_merged.cpp
@@ -90 +89,0 @@ void AudioProfile::dump(std::string *dst, int spaces) const
-<<<<<<< HEAD
@@ -139,5 +137,0 @@ ssize_t AudioProfileVector::add(const sp<AudioProfile> &profile)
-||||||| 5f90ad4290
-ssize_t AudioProfileVectorBase::add(const sp<AudioProfile> &profile)
-=======
-ssize_t AudioProfileVector::add(const sp<AudioProfile> &profile)
->>>>>>> 80ffad94
