--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_merged.cpp
@@ -12 +11,0 @@
-#define APM_AUDIO_OUT_DEVICE_REMOTE_ALL AUDIO_DEVICE_OUT_REMOTE_SUBMIX
@@ -14,2 +13,2 @@ namespace android {
-AudioOutputDescriptor::AudioOutputDescriptor(
-    const sp<AudioPort> &port, const sp<PolicyAudioPort> &policyAudioPort,
+DeviceTypeSet APM_AUDIO_OUT_DEVICE_REMOTE_ALL = {AUDIO_DEVICE_OUT_REMOTE_SUBMIX};
+AudioOutputDescriptor::AudioOutputDescriptor(const sp<PolicyAudioPort>& policyAudioPort,
@@ -17,3 +16,2 @@ AudioOutputDescriptor::AudioOutputDescriptor(
-    : mPort(port),
-      mPolicyAudioPort(policyAudioPort),
-      mClientInterface(clientInterface) {
+    : mPolicyAudioPort(policyAudioPort), mClientInterface(clientInterface)
+{
@@ -679,3 +693 @@ void HwAudioOutputCollection::dump(String8 *dst) const {
-DeviceTypeSet APM_AUDIO_OUT_DEVICE_REMOTE_ALL = {
-    AUDIO_DEVICE_OUT_REMOTE_SUBMIX};
-}
+};
