--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_merged.cpp
@@ -25,12 +26,3 @@ status_t HwModule::addOutputProfile(const std::string &name,
-  profile->addAudioProfile(new AudioProfile(
-      config->format, config->channel_mask, config->sample_rate));
-<<<<<<< HEAD
-  sp<DeviceDescriptor> devDesc =
-      new DeviceDescriptor(device, "" , address.string());
-||||||| 5f90ad4290
-  sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-  devDesc->setAddress(address);
-=======
-  sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-  devDesc->setAddress(address.string());
->>>>>>> 80ffad94f2b41098bad4f35f2399eb05e8c5ee34
+    profile->addAudioProfile(new AudioProfile(config->format, config->channel_mask,
+                                              config->sample_rate));
+    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device, "" , address.string());
@@ -87,12 +83,3 @@ status_t HwModule::addInputProfile(const std::string &name,
-  profile->addAudioProfile(new AudioProfile(
-      config->format, config->channel_mask, config->sample_rate));
-<<<<<<< HEAD
-  sp<DeviceDescriptor> devDesc =
-      new DeviceDescriptor(device, "" , address.string());
-||||||| 5f90ad4290
-  sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-  devDesc->setAddress(address);
-=======
-  sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-  devDesc->setAddress(address.string());
->>>>>>> 80ffad94f2b41098bad4f35f2399eb05e8c5ee34
+    profile->addAudioProfile(new AudioProfile(config->format, config->channel_mask,
+                                              config->sample_rate));
+    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device, "" , address.string());
@@ -198,3 +183 @@ bool HwModule::supportsPatch(const sp<PolicyAudioPort> &srcPort,
-                             const sp<PolicyAudioPort> &dstPort,
-                             const sp<AudioPort> &srcPort,
-                             const sp<AudioPort> &dstPort) const {
+                             const sp<PolicyAudioPort> &dstPort) const {
@@ -318,6 +305,0 @@ sp<DeviceDescriptor> HwModuleCollection::createDevice(
-<<<<<<< HEAD
-||||||| 5f90ad4290
-  device->setAddress(String8(address));
-=======
-  device->setAddress(address);
->>>>>>> 80ffad94f2b41098bad4f35f2399eb05e8c5ee34
