--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
@@ -2 +1,0 @@
-#include "AudioPort.h"
@@ -19,0 +19,4 @@ class DeviceDescriptor : public DeviceDescriptorBase,
+    DeviceDescriptor(audio_devices_t type, const std::string &tagName,
+            const std::string &address, const FormatVector &encodedFormats = FormatVector{});
+    DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr, const std::string &tagName = "",
+            const FormatVector &encodedFormats = FormatVector{});
@@ -56,7 +55,0 @@ class DeviceDescriptor : public DeviceDescriptorBase,
- public:
-  DeviceDescriptor(audio_devices_t type, const std::string &tagName,
-                   const std::string &address,
-                   const FormatVector &encodedFormats = FormatVector{});
-  DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr,
-                   const std::string &tagName = "",
-                   const FormatVector &encodedFormats = FormatVector{});
