diff --git a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_conflict.cpp b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_merged.cpp
index 6096b8e..1520f81 100644
--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_conflict.cpp
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@src@HwModule.no_comments_merged.cpp
@@ -28 +27,0 @@ status_t HwModule::addOutputProfile(const std::string& name, const audio_config_
-<<<<<<< HEAD
@@ -30,7 +28,0 @@ status_t HwModule::addOutputProfile(const std::string& name, const audio_config_
-||||||| 5f90ad4290
-    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-    devDesc->setAddress(address);
-=======
-    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-    devDesc->setAddress(address.string());
->>>>>>> 80ffad94
@@ -93 +84,0 @@ status_t HwModule::addInputProfile(const std::string& name, const audio_config_t
-<<<<<<< HEAD
@@ -95,7 +85,0 @@ status_t HwModule::addInputProfile(const std::string& name, const audio_config_t
-||||||| 5f90ad4290
-    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-    devDesc->setAddress(address);
-=======
-    sp<DeviceDescriptor> devDesc = new DeviceDescriptor(device);
-    devDesc->setAddress(address.string());
->>>>>>> 80ffad94
@@ -322,6 +305,0 @@ sp<DeviceDescriptor> HwModuleCollection::createDevice(const audio_devices_t type
-<<<<<<< HEAD
-||||||| 5f90ad4290
-    device->setAddress(String8(address));
-=======
-    device->setAddress(address);
->>>>>>> 80ffad94
