diff --git a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergegen.h b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
index 14c3e0b..4fdab07 100644
--- a/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergegen.h
+++ b/./output/frameworks_av/1e2bdbb4-80ffad94-5f90ad42/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
@@ -2 +2,3 @@
-#include "PolicyAudioPort.h" #include <media/AudioContainers.h> #include <media/DeviceDescriptorBase.h>
+#include "PolicyAudioPort.h"
+#include <media/AudioContainers.h>
+#include <media/DeviceDescriptorBase.h>
