--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/83e87bac-fddc4757-4b2579a0/src@server.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/83e87bac-fddc4757-4b2579a0/src@server.no_comments_merged.h
@@ -127,6 +127,3 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER \
-  10
-#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER \
-  100
-#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES \
-  (100 << 20)
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_LOWER 10
+#define CONFIG_DEFAULT_DEFRAG_THRESHOLD_UPPER 100
+#define CONFIG_DEFAULT_DEFRAG_IGNORE_BYTES (100<<20)
@@ -134,8 +131,4 @@ typedef long long mstime_t;
-#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX \
-  75
-#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS \
-  1000
-#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN \
-  (512ll * 1024 * 1024)
-#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL \
-  10
+#define CONFIG_DEFAULT_DEFRAG_CYCLE_MAX 75
+#define CONFIG_DEFAULT_DEFRAG_MAX_SCAN_FIELDS 1000
+#define CONFIG_DEFAULT_PROTO_MAX_BULK_LEN (512ll*1024*1024)
+#define CONFIG_DEFAULT_TRACKING_TABLE_MAX_FILL 10
@@ -144,2 +137 @@ typedef long long mstime_t;
-#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC \
-  25
+#define ACTIVE_EXPIRE_CYCLE_SLOW_TIME_PERC 25
@@ -173,0 +166 @@ typedef long long mstime_t;
+#define CMD_SKIP_MONITOR (1ULL<<11)
@@ -179 +171,0 @@ typedef long long mstime_t;
-#define CMD_SKIP_MONITOR (1ULL << 11)
@@ -201,25 +192,0 @@ typedef long long mstime_t;
-#define CMD_ASKING (1ULL << 12)
-#define CMD_FAST (1ULL << 13)
-#define CMD_MODULE_GETKEYS (1ULL << 14)
-#define CMD_MODULE_NO_CLUSTER (1ULL << 15)
-#define CMD_CATEGORY_KEYSPACE (1ULL << 16)
-#define CMD_CATEGORY_READ (1ULL << 17)
-#define CMD_CATEGORY_WRITE (1ULL << 18)
-#define CMD_CATEGORY_SET (1ULL << 19)
-#define CMD_CATEGORY_SORTEDSET (1ULL << 20)
-#define CMD_CATEGORY_LIST (1ULL << 21)
-#define CMD_CATEGORY_HASH (1ULL << 22)
-#define CMD_CATEGORY_STRING (1ULL << 23)
-#define CMD_CATEGORY_BITMAP (1ULL << 24)
-#define CMD_CATEGORY_HYPERLOGLOG (1ULL << 25)
-#define CMD_CATEGORY_GEO (1ULL << 26)
-#define CMD_CATEGORY_STREAM (1ULL << 27)
-#define CMD_CATEGORY_PUBSUB (1ULL << 28)
-#define CMD_CATEGORY_ADMIN (1ULL << 29)
-#define CMD_CATEGORY_FAST (1ULL << 30)
-#define CMD_CATEGORY_SLOW (1ULL << 31)
-#define CMD_CATEGORY_BLOCKING (1ULL << 32)
-#define CMD_CATEGORY_DANGEROUS (1ULL << 33)
-#define CMD_CATEGORY_CONNECTION (1ULL << 34)
-#define CMD_CATEGORY_TRANSACTION (1ULL << 35)
-#define CMD_CATEGORY_SCRIPTING (1ULL << 36)
@@ -231,2 +198 @@ typedef long long mstime_t;
-#define CLIENT_MONITOR \
-  (1 << 2)
+#define CLIENT_MONITOR (1<<2)
@@ -234,2 +200 @@ typedef long long mstime_t;
-#define CLIENT_BLOCKED \
-  (1 << 4)
+#define CLIENT_BLOCKED (1<<4)
@@ -237,4 +202,3 @@ typedef long long mstime_t;
-#define CLIENT_CLOSE_AFTER_REPLY \
-  (1 << 6)
-#define CLIENT_UNBLOCKED \
-  (1 << 7)
+#define CLIENT_CLOSE_AFTER_REPLY (1<<6)
+#define CLIENT_UNBLOCKED (1<<7)
+#define CLIENT_LUA (1<<8)
@@ -243,6 +207,3 @@ typedef long long mstime_t;
-#define CLIENT_UNIX_SOCKET \
-  (1 << 11)
-#define CLIENT_DIRTY_EXEC \
-  (1 << 12)
-#define CLIENT_MASTER_FORCE_REPLY \
-  (1 << 13)
+#define CLIENT_UNIX_SOCKET (1<<11)
+#define CLIENT_DIRTY_EXEC (1<<12)
+#define CLIENT_MASTER_FORCE_REPLY (1<<13)
@@ -257,4 +218,3 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_WRITE \
-  (1 << 21)
-#define CLIENT_REPLY_SKIP_NEXT \
-  (1 << 23)
+#define CLIENT_PENDING_WRITE (1<<21)
+#define CLIENT_REPLY_OFF (1<<22)
+#define CLIENT_REPLY_SKIP_NEXT (1<<23)
@@ -266,6 +226,4 @@ typedef long long mstime_t;
-#define CLIENT_PENDING_READ \
-  (1 << 29)
-#define CLIENT_PENDING_COMMAND \
-  (1 << 30)
-#define CLIENT_TRACKING \
-  (1 << 31)
+#define CLIENT_PENDING_READ (1<<29)
+#define CLIENT_PENDING_COMMAND (1<<30)
+#define CLIENT_TRACKING (1<<31)
+#define CLIENT_TRACKING_BROKEN_REDIR (1ULL<<32)
@@ -285,2 +243 @@ typedef long long mstime_t;
-#define CLIENT_TYPE_OBUF_COUNT \
-  3
+#define CLIENT_TYPE_OBUF_COUNT 3
@@ -325,0 +283 @@ typedef long long mstime_t;
+#define UNUSED(V) ((void) V)
@@ -358,4 +316,2 @@ typedef long long mstime_t;
-#define MAXMEMORY_ALLKEYS_LRU \
-  ((4 << 8) | MAXMEMORY_FLAG_LRU | MAXMEMORY_FLAG_ALLKEYS)
-#define MAXMEMORY_ALLKEYS_LFU \
-  ((5 << 8) | MAXMEMORY_FLAG_LFU | MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LRU ((4<<8)|MAXMEMORY_FLAG_LRU|MAXMEMORY_FLAG_ALLKEYS)
+#define MAXMEMORY_ALLKEYS_LFU ((5<<8)|MAXMEMORY_FLAG_LFU|MAXMEMORY_FLAG_ALLKEYS)
@@ -369,2 +325,2 @@ typedef long long mstime_t;
-#define SHUTDOWN_SAVE \
-  1
+#define SHUTDOWN_SAVE 1
+#define SHUTDOWN_NOSAVE 2
@@ -396,4 +352 @@ typedef long long mstime_t;
-#define NOTIFY_ALL \
-  (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | \
-   NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | \
-   NOTIFY_KEY_MISS)
+#define NOTIFY_ALL (NOTIFY_GENERIC | NOTIFY_STRING | NOTIFY_LIST | NOTIFY_SET | NOTIFY_HASH | NOTIFY_ZSET | NOTIFY_EXPIRED | NOTIFY_EVICTED | NOTIFY_STREAM | NOTIFY_KEY_MISS)
@@ -400,0 +354,4 @@ typedef long long mstime_t;
+#define run_with_period(_ms_) if ((_ms_ <= 1000/server.hz) || !(server.cronloops%((_ms_)/(1000/server.hz))))
+#define serverAssertWithInfo(_c,_o,_e) ((_e)?(void)0 : (_serverAssertWithInfo(_c,_o,#_e,__FILE__,__LINE__),_exit(1)))
+#define serverAssert(_e) ((_e)?(void)0 : (_serverAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define serverPanic(...) _serverPanic(__FILE__,__LINE__,__VA_ARGS__),_exit(1)
@@ -409,0 +367,2 @@ typedef long long mstime_t;
+#define REDISMODULE_TYPE_ENCVER(id) (id & REDISMODULE_TYPE_ENCVER_MASK)
+#define REDISMODULE_TYPE_SIGN(id) ((id & ~((uint64_t)REDISMODULE_TYPE_ENCVER_MASK)) >>REDISMODULE_TYPE_ENCVER_BITS)
@@ -454,0 +411,9 @@ typedef struct RedisModuleIO {
+#define moduleInitIOContext(iovar,mtype,rioptr,keyptr) do { \
+    iovar.rio = rioptr; \
+    iovar.type = mtype; \
+    iovar.bytes = 0; \
+    iovar.error = 0; \
+    iovar.ver = 0; \
+    iovar.key = keyptr; \
+    iovar.ctx = NULL; \
+} while(0);
@@ -458,0 +424,4 @@ typedef struct RedisModuleDigest {
+#define moduleInitDigestContext(mdvar) do { \
+    memset(mdvar.o,0,sizeof(mdvar.o)); \
+    memset(mdvar.x,0,sizeof(mdvar.x)); \
+} while(0);
@@ -481,0 +451,6 @@ char *getObjectTypeName(robj *);
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = OBJ_STRING; \
+    _var.encoding = OBJ_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0)
@@ -527,2 +501 @@ typedef struct readyList {
-#define USER_COMMAND_BITS_COUNT \
-  1024
+#define USER_COMMAND_BITS_COUNT 1024
@@ -533,2 +506 @@ typedef struct readyList {
-#define USER_FLAG_NOPASS \
-  (1 << 4)
+#define USER_FLAG_NOPASS (1<<4)
@@ -541,0 +514 @@ typedef struct user {
+} user;
@@ -690,2 +661 @@ typedef struct user {
-#define RDB_SAVE_INFO_INIT \
-  { -1, 0, "000000000000000000000000000000", -1 }
+#define RDB_SAVE_INFO_INIT {-1,0,"000000000000000000000000000000",-1}
@@ -1080,0 +1044 @@ typedef struct user {
+#include "stream.h"
@@ -1300,0 +1257 @@ typedef struct user {
+#define sdsEncodedObject(objptr) (objptr->encoding == OBJ_ENCODING_RAW || objptr->encoding == OBJ_ENCODING_EMBSTR)
@@ -1341,0 +1297 @@ typedef struct user {
+#include "rdb.h"
@@ -1481,2 +1432 @@ typedef struct user {
-#define RESTART_SERVER_CONFIG_REWRITE \
-  (1 << 1)
+#define RESTART_SERVER_CONFIG_REWRITE (1<<1)
@@ -1846,2 +1780,7 @@ typedef struct user {
-  void _serverAssertWithInfo(const client *c, const robj *o, const char *estr,
-                             const char *file, int line);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
+void _serverAssertWithInfo(const client *c, const robj *o, const char *estr, const char *file, int line);
@@ -1862,0 +1802,4 @@ typedef struct user {
+#define redisDebug(fmt,...) \
+    printf("DEBUG %s:%d > " fmt "\n", __FILE__, __LINE__, __VA_ARGS__)
+#define redisDebugMark() \
+    printf("-- MARK %s:%d --\n", __FILE__, __LINE__)
