diff --git a/./output/grpc/5817f628-bca2a30e-018580fb/src@php@ext@grpc@version.no_comments_conflict.h b/./output/grpc/5817f628-bca2a30e-018580fb/src@php@ext@grpc@version.no_comments_merged.h
index b95bfc4..8d8a519 100644
--- a/./output/grpc/5817f628-bca2a30e-018580fb/src@php@ext@grpc@version.no_comments_conflict.h
+++ b/./output/grpc/5817f628-bca2a30e-018580fb/src@php@ext@grpc@version.no_comments_merged.h
@@ -3 +2,0 @@
-<<<<<<< HEAD
@@ -5,5 +3,0 @@
-||||||| 018580fb89
-#define PHP_GRPC_VERSION "1.25.0dev"
-=======
-#define PHP_GRPC_VERSION "1.25.0"
->>>>>>> bca2a30e
