--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/5817f628-bca2a30e-018580fb/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/5817f628-bca2a30e-018580fb/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-grpc::string Version() < < < < < < < HEAD { return "1.26.0-dev"; }
-|||||||
-{ return "1.25.0-dev"; }
-=======
-{ return "1.25.0"; }
->>>>>>> bca2a30e97f46e7eaec83da0e13c9313aca1ce53
+grpc::string Version() { return "1.26.0-dev"; }
