diff --git a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@iomgr@tcp_posix.no_comments_mergegen.cc b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@iomgr@tcp_posix.no_comments_merged.cc
index b5e0f0b..158238a 100644
--- a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@iomgr@tcp_posix.no_comments_mergegen.cc
+++ b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@lib@iomgr@tcp_posix.no_comments_merged.cc
@@ -317 +317 @@ static void call_read_cb(grpc_tcp* tcp, grpc_error* error) {
-grpc_core::ExecCtx::Run(DEBUG_LOCATION, cb, error);
+  GRPC_CLOSURE_RUN(cb, error);
@@ -499 +499 @@ static void tcp_read(grpc_endpoint* ep, grpc_slice_buffer* incoming_buffer,
-grpc_core::ExecCtx::Run(DEBUG_LOCATION, &tcp->read_done_closure, GRPC_ERROR_NONE);
+    GRPC_CLOSURE_RUN(&tcp->read_done_closure, GRPC_ERROR_NONE);
@@ -799 +799 @@ static void tcp_handle_write(void* arg , grpc_error* error) {
-grpc_core::ExecCtx::Run(DEBUG_LOCATION, cb,
+    GRPC_CLOSURE_RUN(cb, GRPC_ERROR_REF(error));
@@ -840 +840,5 @@ static void tcp_write(grpc_endpoint* ep, grpc_slice_buffer* buf,
-grpc_core::ExecCtx::Run( DEBUG_LOCATION,cb, grpc_fd_is_shutdown(tcp->em_fd) ? tcp_annotate_error(GRPC_ERROR_CREATE_FROM_STATIC_STRING("EOF"), tcp) : GRPC_ERROR_NONE);
+    GRPC_CLOSURE_RUN(cb,
+                     grpc_fd_is_shutdown(tcp->em_fd)
+                         ? tcp_annotate_error(
+                               GRPC_ERROR_CREATE_FROM_STATIC_STRING("EOF"), tcp)
+                         : GRPC_ERROR_NONE);
@@ -861 +865 @@ grpc_core::ExecCtx::Run( DEBUG_LOCATION,cb, grpc_fd_is_shutdown(tcp->em_fd) ? tc
-grpc_core::ExecCtx::Run(DEBUG_LOCATION, cb, error);
+    GRPC_CLOSURE_RUN(cb, error);
