--- a/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_mergegen.cc
+++ b/./output/grpc/57c0ac01-a0f09cb5-ce842ecc/src@core@ext@filters@client_channel@http_connect_handshaker.no_comments_merged.cc
@@ -169 +169,5 @@ void HttpConnectHandshaker::DoHandshake(grpc_tcp_server_acceptor* ,
-MutexLock lock(&mu_); is_shutdown_ = true; } GRPC_CLOSURE_SCHED(DEBUG_LOCATION, on_handshake_done, GRPC_ERROR_NONE);
+    {
+      MutexLock lock(&mu_);
+      is_shutdown_ = true;
+    }
+    ExecCtx::Run(DEBUG_LOCATION, on_handshake_done, GRPC_ERROR_NONE);
