diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/a5545dbc-5aadee6d-347cd0b5/tty-term.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/a5545dbc-5aadee6d-347cd0b5/tty-term.no_comments_merged.c
index 1cb9fed..4180006 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/a5545dbc-5aadee6d-347cd0b5/tty-term.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/a5545dbc-5aadee6d-347cd0b5/tty-term.no_comments_merged.c
@@ -2 +1,0 @@
-#include <curses.h>
@@ -10 +8,0 @@
-#include <vis.h>
@@ -266,3 +264,10 @@ static const struct tty_term_code_entry tty_term_codes[] = {
-    [TTYC_XT] = {TTYCODE_FLAG, "XT"}};
-u_int tty_term_ncodes(void) { return (nitems(tty_term_codes)); }
-static char *tty_term_strip(const char *s) {
+ [TTYC_XT] = { TTYCODE_FLAG, "XT" }
+};
+u_int
+tty_term_ncodes(void)
+{
+ return (nitems(tty_term_codes));
+}
+static char *
+tty_term_strip(const char *s)
+{
@@ -272 +277,2 @@ static char *tty_term_strip(const char *s) {
-  if (strchr(s, '$') == NULL) return (xstrdup(s));
+ if (strchr(s, '$') == NULL)
+  return (xstrdup(s));
@@ -276,3 +282,6 @@ static char *tty_term_strip(const char *s) {
-      while (*ptr != '\0' && *ptr != '>') ptr++;
-      if (*ptr == '>') ptr++;
-      if (*ptr == '\0') break;
+   while (*ptr != '\0' && *ptr != '>')
+    ptr++;
+   if (*ptr == '>')
+    ptr++;
+   if (*ptr == '\0')
+    break;
@@ -281 +290,2 @@ static char *tty_term_strip(const char *s) {
-    if (len == (sizeof buf) - 1) break;
+  if (len == (sizeof buf) - 1)
+   break;
@@ -286 +296,3 @@ static char *tty_term_strip(const char *s) {
-static char *tty_term_override_next(const char *s, size_t *offset) {
+static char *
+tty_term_override_next(const char *s, size_t *offset)
+{
@@ -289 +301,2 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-  if (s[at] == '\0') return (NULL);
+ if (s[at] == '\0')
+  return (NULL);
@@ -301 +314,2 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-    if (n == (sizeof value) - 1) return (NULL);
+  if (n == (sizeof value) - 1)
+   return (NULL);
@@ -310,2 +324,3 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-void tty_term_apply(struct tty_term *term, const char *capabilities,
-                    int quiet) {
+void
+tty_term_apply(struct tty_term *term, const char *capabilities, int quiet)
+{
@@ -320 +335,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-    if (*s == '\0') continue;
+  if (*s == '\0')
+   continue;
@@ -345 +361,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-      if (strcmp(s, ent->name) != 0) continue;
+   if (strcmp(s, ent->name) != 0)
+    continue;
@@ -355 +372,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-          if (code->type == TTYCODE_STRING) free(code->value.string);
+    if (code->type == TTYCODE_STRING)
+     free(code->value.string);
@@ -361 +379,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-          if (errstr != NULL) break;
+    if (errstr != NULL)
+     break;
@@ -374 +393,3 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-void tty_term_apply_overrides(struct tty_term *term) {
+void
+tty_term_apply_overrides(struct tty_term *term)
+{
@@ -393 +414,2 @@ void tty_term_apply_overrides(struct tty_term *term) {
-  if (tty_term_has(term, TTYC_SETRGBF) && tty_term_has(term, TTYC_SETRGBB))
+ if (tty_term_has(term, TTYC_SETRGBF) &&
+     tty_term_has(term, TTYC_SETRGBB))
@@ -421,2 +443,4 @@ void tty_term_apply_overrides(struct tty_term *term) {
-struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-                                 u_int ncaps, int *feat, char **cause) {
+struct tty_term *
+tty_term_create(struct tty *tty, char *name, char **caps, u_int ncaps,
+    int *feat, char **cause)
+{
@@ -441 +465,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-    if (namelen == 0) continue;
+  if (namelen == 0)
+   continue;
@@ -445,2 +470,4 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-      if (strncmp(ent->name, caps[i], namelen) != 0) continue;
-      if (ent->name[namelen] != '\0') continue;
+   if (strncmp(ent->name, caps[i], namelen) != 0)
+    continue;
+   if (ent->name[namelen] != '\0')
+    continue;
@@ -497 +524,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-      (!tty_term_has(term, TTYC_SETRGBF) || !tty_term_has(term, TTYC_SETRGBB)))
+     (!tty_term_has(term, TTYC_SETRGBF) ||
+     !tty_term_has(term, TTYC_SETRGBB)))
@@ -499 +527,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-  if (tty_apply_features(term, *feat)) tty_term_apply_overrides(term);
+ if (tty_apply_features(term, *feat))
+  tty_term_apply_overrides(term);
@@ -507 +536,3 @@ error:
-void tty_term_free(struct tty_term *term) {
+void
+tty_term_free(struct tty_term *term)
+{
@@ -519,2 +550,4 @@ void tty_term_free(struct tty_term *term) {
-int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-                       char **cause) {
+int
+tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
+    char **cause)
+{
@@ -529 +562,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        xasprintf(cause, "can't use hardcopy terminal: %s", name);
+   xasprintf(cause, "can't use hardcopy terminal: %s",
+       name);
@@ -532 +566,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        xasprintf(cause, "missing or unsuitable terminal: %s", name);
+   xasprintf(cause, "missing or unsuitable terminal: %s",
+       name);
@@ -552 +587,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (s == NULL || s == (char *)-1) continue;
+   if (s == NULL || s == (char *)-1)
+    continue;
@@ -556 +592,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (n == -1 || n == -2) continue;
+   if (n == -1 || n == -2)
+    continue;
@@ -562 +599,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (n == -1) continue;
+   if (n == -1)
+    continue;
@@ -581 +619,3 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-void tty_term_free_list(char **caps, u_int ncaps) {
+void
+tty_term_free_list(char **caps, u_int ncaps)
+{
@@ -583 +623,2 @@ void tty_term_free_list(char **caps, u_int ncaps) {
-  for (i = 0; i < ncaps; i++) free(caps[i]);
+ for (i = 0; i < ncaps; i++)
+  free(caps[i]);
@@ -586 +627,3 @@ void tty_term_free_list(char **caps, u_int ncaps) {
-int tty_term_has(struct tty_term *term, enum tty_code_code code) {
+int
+tty_term_has(struct tty_term *term, enum tty_code_code code)
+{
@@ -589,2 +632,5 @@ int tty_term_has(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_string(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return ("");
+const char *
+tty_term_string(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return ("");
@@ -595,2 +641,3 @@ const char *tty_term_string(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-                              int a) {
+const char *
+tty_term_string_i(struct tty_term *term, enum tty_code_code code, int a)
+{
@@ -598 +644,0 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -606,4 +651,0 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-  ||||||| s = tparm((char *)x, a);
-=======
-  s = tiparm_s(1, 0, x, a);
->>>>>>> 5aadee6df4750a4c428fc7e7edc9e8e315c3f610
@@ -616,2 +658,3 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-                               int a, int b) {
+const char *
+tty_term_string_ii(struct tty_term *term, enum tty_code_code code, int a, int b)
+{
@@ -619 +661,0 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -627,4 +668,0 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-  ||||||| s = tparm((char *)x, a, b);
-=======
-  s = tiparm_s(2, 0, x, a, b);
->>>>>>> 5aadee6df4750a4c428fc7e7edc9e8e315c3f610
@@ -637,2 +675,4 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-                                int a, int b, int c) {
+const char *
+tty_term_string_iii(struct tty_term *term, enum tty_code_code code, int a,
+    int b, int c)
+{
@@ -640 +679,0 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -648,4 +686,0 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-  ||||||| s = tparm((char *)x, a, b, c);
-=======
-  s = tiparm_s(3, 0, x, a, b, c);
->>>>>>> 5aadee6df4750a4c428fc7e7edc9e8e315c3f610
@@ -658,2 +693,3 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-                              const char *a) {
+const char *
+tty_term_string_s(struct tty_term *term, enum tty_code_code code, const char *a)
+{
@@ -661 +696,0 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -669,4 +703,0 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-  ||||||| s = tparm((char *)x, (long)a);
-=======
-  s = tiparm_s(1, 1, x, a);
->>>>>>> 5aadee6df4750a4c428fc7e7edc9e8e315c3f610
@@ -679,2 +710,4 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-                               const char *a, const char *b) {
+const char *
+tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
+    const char *a, const char *b)
+{
@@ -682 +714,0 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-<<<<<<< HEAD
@@ -690,4 +721,0 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-  ||||||| s = tparm((char *)x, (long)a, (long)b);
-=======
-  s = tiparm_s(2, 3, x, a, b);
->>>>>>> 5aadee6df4750a4c428fc7e7edc9e8e315c3f610
@@ -700,2 +728,5 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-int tty_term_number(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return (0);
+int
+tty_term_number(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return (0);
@@ -706,3 +737,7 @@ int tty_term_number(struct tty_term *term, enum tty_code_code code) {
-int tty_term_flag(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return (0);
-  if (term->codes[code].type != TTYCODE_FLAG) fatalx("not a flag: %d", code);
+int
+tty_term_flag(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return (0);
+ if (term->codes[code].type != TTYCODE_FLAG)
+  fatalx("not a flag: %d", code);
@@ -711 +746,3 @@ int tty_term_flag(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
+const char *
+tty_term_describe(struct tty_term *term, enum tty_code_code code)
+{
@@ -716,2 +753,2 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: [missing]", code,
-                tty_term_codes[code].name);
+  xsnprintf(s, sizeof s, "%4u: %s: [missing]",
+      code, tty_term_codes[code].name);
@@ -722,2 +759,3 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (string) %s", code,
-                tty_term_codes[code].name, out);
+  xsnprintf(s, sizeof s, "%4u: %s: (string) %s",
+      code, tty_term_codes[code].name,
+      out);
@@ -726,2 +764,3 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (number) %d", code,
-                tty_term_codes[code].name, term->codes[code].value.number);
+  xsnprintf(s, sizeof s, "%4u: %s: (number) %d",
+      code, tty_term_codes[code].name,
+      term->codes[code].value.number);
@@ -730,2 +769,2 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (flag) %s", code,
-                tty_term_codes[code].name,
+  xsnprintf(s, sizeof s, "%4u: %s: (flag) %s",
+      code, tty_term_codes[code].name,
