diff --git a/./output/system_core/0bb51f76-8cbbc2dc-d61ce22b/fs_mgr@libsnapshot@snapuserd@user-space-merge@snapuserd_server.no_comments_mergegen.h b/./output/system_core/0bb51f76-8cbbc2dc-d61ce22b/fs_mgr@libsnapshot@snapuserd@user-space-merge@snapuserd_server.no_comments_merged.h
index bffbb12..85f292b 100644
--- a/./output/system_core/0bb51f76-8cbbc2dc-d61ce22b/fs_mgr@libsnapshot@snapuserd@user-space-merge@snapuserd_server.no_comments_mergegen.h
+++ b/./output/system_core/0bb51f76-8cbbc2dc-d61ce22b/fs_mgr@libsnapshot@snapuserd@user-space-merge@snapuserd_server.no_comments_merged.h
@@ -17,0 +18 @@
+#include "handler_manager.h"
@@ -23,34 +23,0 @@ static constexpr uint8_t kMaxMergeThreads = 2;
-enum class DaemonOps {
-    INIT,
-    START,
-    QUERY,
-    STOP,
-    DELETE,
-    DETACH,
-    SUPPORTS,
-    INITIATE,
-    PERCENTAGE,
-    GETSTATUS,
-    UPDATE_VERIFY,
-    INVALID,
-};
-class HandlerThread {
-  public:
-    explicit HandlerThread(std::shared_ptr<SnapshotHandler> snapuserd);
-    void FreeResources() {
-        if (snapuserd_) {
-            snapuserd_->FreeResources();
-            snapuserd_ = nullptr;
-        }
-    }
-    const std::shared_ptr<SnapshotHandler>& snapuserd() const { return snapuserd_; }
-    std::thread& thread() { return thread_; }
-    const std::string& misc_name() const { return misc_name_; }
-    bool ThreadTerminated() { return thread_terminated_; }
-    void SetThreadTerminated() { thread_terminated_ = true; }
-  private:
-    std::thread thread_;
-    std::shared_ptr<SnapshotHandler> snapuserd_;
-    std::string misc_name_;
-    bool thread_terminated_ = false;
-};
@@ -64,3 +30,0 @@ class UserSnapshotServer {
-    int num_partitions_merge_complete_ = 0;
-    int active_merge_threads_ = 0;
-    bool stop_monitor_merge_thread_ = false;
@@ -69 +33,2 @@ class UserSnapshotServer {
-std::optional<bool> is_merge_monitor_started_; android::base::unique_fd monitor_merge_event_fd_;
+    std::unique_ptr<ISnapshotHandlerManager> handlers_;
+    std::unique_ptr<IBlockServerFactory> block_server_factory_;
@@ -71,3 +35,0 @@ std::optional<bool> is_merge_monitor_started_; android::base::unique_fd monitor_
-    using HandlerList = std::vector<std::shared_ptr<HandlerThread>>;
-    HandlerList dm_users_;
-    std::queue<std::shared_ptr<HandlerThread>> merge_handlers_;
@@ -81,2 +42,0 @@ std::optional<bool> is_merge_monitor_started_; android::base::unique_fd monitor_
-    bool RemoveAndJoinHandler(const std::string& control_device);
-    DaemonOps Resolveop(std::string& input);
@@ -86,2 +45,0 @@ std::optional<bool> is_merge_monitor_started_; android::base::unique_fd monitor_
-    void RunThread(std::shared_ptr<HandlerThread> handler);
-    void MonitorMerge();
@@ -90,5 +47,0 @@ std::optional<bool> is_merge_monitor_started_; android::base::unique_fd monitor_
-    HandlerList::iterator FindHandler(std::lock_guard<std::mutex>* proof_of_lock,
-                                      const std::string& misc_name);
-    double GetMergePercentage(std::lock_guard<std::mutex>* proof_of_lock);
-    void TerminateMergeThreads(std::lock_guard<std::mutex>* proof_of_lock);
-    bool UpdateVerification(std::lock_guard<std::mutex>* proof_of_lock);
@@ -107,5 +60 @@ std::optional<bool> is_merge_monitor_started_; android::base::unique_fd monitor_
-    bool StartHandler(const std::shared_ptr<HandlerThread>& handler);
-    bool StartMerge(std::lock_guard<std::mutex>* proof_of_lock,
-                    const std::shared_ptr<HandlerThread>& handler);
-    std::string GetMergeStatus(const std::shared_ptr<HandlerThread>& handler);
-    void WakeupMonitorMergeThread();
+    bool StartHandler(const std::string& misc_name);
