diff --git a/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_conflict.cpp b/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_merged.cpp
index b47f4da..f3ec3aa 100644
--- a/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_conflict.cpp
+++ b/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -153 +152,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -157,7 +155,0 @@ static std::string InitProductPublicLibraries() {
-||||||| f701823d2b
-  if (is_product_vndk_version_defined()) {
-    ReadExtensionLibraries("/product/etc", &sonames);
-  }
-=======
-  ReadExtensionLibraries("/product/etc", &sonames);
->>>>>>> 96d09a10
@@ -172 +163,0 @@ static std::string InitExtendedPublicLibraries() {
-<<<<<<< HEAD
@@ -176,6 +166,0 @@ static std::string InitExtendedPublicLibraries() {
-||||||| f701823d2b
-  if (!is_product_vndk_version_defined()) {
-    ReadExtensionLibraries("/product/etc", &sonames);
-  }
-=======
->>>>>>> 96d09a10
@@ -218 +202,0 @@ static std::string InitLlndkLibrariesProduct() {
-<<<<<<< HEAD
@@ -223,7 +206,0 @@ static std::string InitLlndkLibrariesProduct() {
-||||||| f701823d2b
-  if (!is_product_vndk_version_defined()) {
-    ALOGD("InitLlndkLibrariesProduct: No product VNDK version defined");
-    return "";
-  }
-=======
->>>>>>> 96d09a10
@@ -369 +345,0 @@ const std::map<std::string, std::string>& apex_public_libraries() {
-<<<<<<< HEAD
@@ -380,10 +355,0 @@ bool is_product_treblelized() {
-||||||| f701823d2b
-bool is_product_vndk_version_defined() {
-#if defined(ART_TARGET_ANDROID)
-  return android::sysprop::VndkProperties::product_vndk_version().has_value();
-#else
-  return false;
-#endif
-}
-=======
->>>>>>> 96d09a10
