diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_merged.cpp
index da59c6a..f3ec3aa 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -23 +22,0 @@ using internal::ConfigEntry;
-using internal::ParseApexLibrariesConfig;
@@ -24,0 +24 @@ using internal::ParseConfig;
+using internal::ParseApexLibrariesConfig;
@@ -153 +152,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -155,4 +153,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-  if (is_product_vndk_version_defined()) {
-=======
->>>>>>> 96d09a108d67d8cc07fed6c499e7fa3901c57446
@@ -159,0 +155 @@ static std::string InitProductPublicLibraries() {
+  }
@@ -168 +163,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -172,6 +166,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-    if (!is_product_vndk_version_defined()) {
-      ReadExtensionLibraries("/product/etc", &sonames);
-    }
-=======
->>>>>>> 96d09a108d67d8cc07fed6c499e7fa3901c57446
@@ -214 +202,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -219,7 +206,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-    if (!is_product_vndk_version_defined()) {
-      ALOGD("InitLlndkLibrariesProduct: No product VNDK version defined");
-      return "";
-    }
-=======
->>>>>>> 96d09a108d67d8cc07fed6c499e7fa3901c57446
@@ -318,2 +299 @@ static std::string InitProductPublicLibraries() {
-    return ApexLibrariesConfigLine{
-        std::move(tokens[0]), std::move(tokens[1]), std::move(tokens[2])};
+  return ApexLibrariesConfigLine{std::move(tokens[0]), std::move(tokens[1]), std::move(tokens[2])};
@@ -422,2 +402 @@ Result<std::vector<std::string>> ParseConfig(
-    if (entry.bitness == ONLY_32)
-      continue;
+    if (entry.bitness == ONLY_32) continue;
@@ -425,2 +404 @@ Result<std::vector<std::string>> ParseConfig(
-    if (entry.bitness == ONLY_64)
-      continue;
+    if (entry.bitness == ONLY_64) continue;
@@ -429,2 +407 @@ Result<std::vector<std::string>> ParseConfig(
-    if (entry.soname == "libRS.so")
-      continue;
+    if (entry.soname == "libRS.so") continue;
@@ -442,2 +419 @@ Result<std::vector<std::string>> ParseConfig(
-Result<std::map<std::string, std::string>> ParseApexLibrariesConfig(const std::string& file_content,
-                                                                    const std::string& tag) {
+Result<std::map<std::string, std::string>> ParseApexLibrariesConfig(const std::string& file_content, const std::string& tag) {
