diff --git a/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@library_namespaces.no_comments_mergegen.cpp b/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@library_namespaces.no_comments_merged.cpp
index 35ac217..28a0dfa 100644
--- a/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@library_namespaces.no_comments_mergegen.cpp
+++ b/./output/art/bed1f3cd-96d09a10-f701823d/libnativeloader@library_namespaces.no_comments_merged.cpp
@@ -144 +144,3 @@ Result<NativeLoaderNamespace*> LibraryNamespaces::Create(JNIEnv* env,
-} else if (api_domain == API_DOMAIN_PRODUCT) { unbundled_app_domain = API_DOMAIN_PRODUCT; api_domain_msg = "unbundled product apk";
+    } else if (api_domain == API_DOMAIN_PRODUCT) {
+      unbundled_app_domain = API_DOMAIN_PRODUCT;
+      api_domain_msg = "unbundled product apk";
@@ -259 +261,4 @@ Result<NativeLoaderNamespace*> LibraryNamespaces::Create(JNIEnv* env,
- 44<NativeLoaderNamespace> target_ns = system_ns; if (is_product_treblelized()) { target_ns = NativeLoaderNamespace::GetExportedNamespace(kProductNamespaceName, is_bridged); }
+    Result<NativeLoaderNamespace> target_ns = system_ns;
+    if (is_product_treblelized()) {
+      target_ns = NativeLoaderNamespace::GetExportedNamespace(kProductNamespaceName, is_bridged);
+    }
