diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
index 7613939..49e3bdc 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/35918fe8-4aa3f70e-63728a5c/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
@@ -50,3 +49,0 @@ using namespace std::chrono_literals;
-using ::aidl::android::media::IAidlBufferSource;
-using ::aidl::android::media::IAidlNode;
-using android::base::StringPrintf;
@@ -53,0 +51 @@ using ::android::hardware::graphics::bufferqueue::V1_0::utils::H2BGraphicBufferP
+using android::base::StringPrintf;
@@ -54,0 +53,2 @@ using ::android::hardware::media::c2::V1_0::IInputSurface;
+using ::aidl::android::media::IAidlBufferSource;
+using ::aidl::android::media::IAidlNode;
@@ -78 +78 @@ class CCodecWatchdog : public AHandler {
-    ~CCodecWatchdog()
+    ~CCodecWatchdog() = default;
@@ -125,3 +125,5 @@ class C2InputSurfaceWrapper : public InputSurfaceWrapper {
-    explicit C2InputSurfaceWrapper(const std::shared_ptr<Codec2Client::InputSurface>& surface)
-        : mSurface(surface){}
-          () = delete;
+    explicit C2InputSurfaceWrapper(
+            const std::shared_ptr<Codec2Client::InputSurface> &surface) :
+        mSurface(surface) {
+    }
+    ~C2InputSurfaceWrapper() override = default;
@@ -162,2 +164,5 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    HGraphicBufferSourceWrapper(const sp<HGraphicBufferSource>& source, uint32_t width,
-                                uint32_t height, uint64_t usage)
+    HGraphicBufferSourceWrapper(
+            const sp<HGraphicBufferSource> &source,
+            uint32_t width,
+            uint32_t height,
+            uint64_t usage)
@@ -168 +173 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    () = delete;
+    ~HGraphicBufferSourceWrapper() override = default;
@@ -175 +180,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        status_t res = (*node)->setParameter((OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits64,
+        status_t res = (*node)->setParameter(
+                (OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits64,
@@ -179,2 +185,3 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            (void)(*node)->setParameter((OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits, &usage,
-                                        sizeof(usage));
+            (void)(*node)->setParameter(
+                    (OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits,
+                    &usage, sizeof(usage));
@@ -221 +228,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        status_t err = (*node)->getParameter(OMX_IndexParamPortDefinition, &param, sizeof(param));
+        status_t err = (*node)->getParameter(OMX_IndexParamPortDefinition,
+                                           &param, sizeof(param));
@@ -261,2 +269,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1)) &&
-            config.mMaxFps != mConfig.mMaxFps) {
+        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1))
+                && config.mMaxFps != mConfig.mMaxFps) {
@@ -291,2 +299,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if (config.mStartAtUs != mConfig.mStartAtUs ||
-            (config.mStopped != mConfig.mStopped && !config.mStopped)) {
+        if (config.mStartAtUs != mConfig.mStartAtUs
+                || (config.mStopped != mConfig.mStopped && !config.mStopped)) {
@@ -304,2 +312,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            status << " " << (config.mSuspended ? "suspend" : "resume") << " at "
-                   << config.mSuspendAtUs << "us";
+            status << " " << (config.mSuspended ? "suspend" : "resume")
+                    << " at " << config.mSuspendAtUs << "us";
@@ -321,2 +329,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-                hardware::Return<void> trans =
-                        mSource->getStopTimeOffsetUs([&res, &delayUs = config.mInputDelayUs](
+                hardware::Return<void> trans = mSource->getStopTimeOffsetUs(
+                        [&res, &delayUs = config.mInputDelayUs](
@@ -391,2 +399,5 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    AGraphicBufferSourceWrapper(const std::shared_ptr<AGraphicBufferSource>& source, uint32_t width,
-                                uint32_t height, uint64_t usage)
+    AGraphicBufferSourceWrapper(
+            const std::shared_ptr<AGraphicBufferSource> &source,
+            uint32_t width,
+            uint32_t height,
+            uint64_t usage)
@@ -397 +408 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    () = delete;
+    ~AGraphicBufferSourceWrapper() override = default;
@@ -409,10 +420,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-<<<<<<< HEAD
-                (*node),
-                static_cast<::aidl::android::hardware::graphics::common::Dataspace>(dataspace)));
-||||||| 63728a5cdd
-                mNode,
-                static_cast<::aidl::android::hardware::graphics::common::Dataspace>(mDataSpace)));
-=======
-                (*node),
-                static_cast<::aidl::android::hardware::graphics::common::Dataspace>(mDataSpace)));
->>>>>>> 4aa3f70e45fd04e9a77762dd991d3f31ec0a66eb
+                (*node), static_cast<::aidl::android::hardware::graphics::common::Dataspace>(
+                        dataspace)));
@@ -481,2 +484,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1)) &&
-            config.mMaxFps != mConfig.mMaxFps) {
+        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1))
+                && config.mMaxFps != mConfig.mMaxFps) {
@@ -501,2 +504,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            status_t res = fromAidlStatus(
-                    mSource->setTimeLapseConfig(config.mCodedFps, config.mCaptureFps));
+            status_t res =
+                fromAidlStatus(mSource->setTimeLapseConfig(config.mCodedFps, config.mCaptureFps));
@@ -511,2 +514,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if (config.mStartAtUs != mConfig.mStartAtUs ||
-            (config.mStopped != mConfig.mStopped && !config.mStopped)) {
+        if (config.mStartAtUs != mConfig.mStartAtUs
+                || (config.mStopped != mConfig.mStopped && !config.mStopped)) {
@@ -523,4 +526,4 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            status_t res =
-                    fromAidlStatus(mSource->setSuspend(config.mSuspended, config.mSuspendAtUs));
-            status << " " << (config.mSuspended ? "suspend" : "resume") << " at "
-                   << config.mSuspendAtUs << "us";
+            status_t res = fromAidlStatus(mSource->setSuspend(
+                    config.mSuspended, config.mSuspendAtUs));
+            status << " " << (config.mSuspended ? "suspend" : "resume")
+                    << " at " << config.mSuspendAtUs << "us";
@@ -604,4 +607,5 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    Codec2ClientInterfaceWrapper(std::shared_ptr<Codec2Client> client) : mClient(client) {}
-    virtual ~Codec2ClientInterfaceWrapper()
-            virtual c2_status_t
-            config_sm(const std::vector<C2Param*>& params,
+    Codec2ClientInterfaceWrapper(std::shared_ptr<Codec2Client> client)
+        : mClient(client) { }
+    virtual ~Codec2ClientInterfaceWrapper() = default;
+    virtual c2_status_t config_sm(
+            const std::vector<C2Param *> &params,
@@ -610,2 +614,3 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    }
-    virtual c2_status_t copyBuffer(std::shared_ptr<C2GraphicBuffer>,
+    };
+    virtual c2_status_t copyBuffer(
+            std::shared_ptr<C2GraphicBuffer>,
@@ -615 +620,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual c2_status_t createComponent(C2String, std::shared_ptr<C2Component>* const component) {
+    virtual c2_status_t createComponent(
+            C2String, std::shared_ptr<C2Component> *const component) {
@@ -619,2 +625,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual c2_status_t createInterface(C2String,
-                                        std::shared_ptr<C2ComponentInterface>* const interface) {
+    virtual c2_status_t createInterface(
+            C2String, std::shared_ptr<C2ComponentInterface> *const interface) {
@@ -624 +630,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual c2_status_t query_sm(const std::vector<C2Param*>& stackParams,
+    virtual c2_status_t query_sm(
+            const std::vector<C2Param *> &stackParams,
@@ -637 +644,3 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual C2String getName() const { return mClient->getName(); }
+    virtual C2String getName() const {
+        return mClient->getName();
+    }
@@ -645 +654,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-void RevertOutputFormatIfNeeded(const sp<AMessage>& oldFormat, sp<AMessage>& currentFormat) {
+void RevertOutputFormatIfNeeded(
+        const sp<AMessage> &oldFormat, sp<AMessage> &currentFormat) {
@@ -670,2 +680,2 @@ void RevertOutputFormatIfNeeded(const sp<AMessage>& oldFormat, sp<AMessage>& cur
-void AmendOutputFormatWithCodecSpecificData(const uint8_t* data, size_t size,
-                                            const std::string& mediaType,
+void AmendOutputFormatWithCodecSpecificData(
+        const uint8_t *data, size_t size, const std::string &mediaType,
@@ -681 +691,2 @@ void AmendOutputFormatWithCodecSpecificData(const uint8_t* data, size_t size,
-            outputFormat->setBuffer(AStringPrintf("csd-%u", csdIndex).c_str(), csd);
+            outputFormat->setBuffer(
+                    AStringPrintf("csd-%u", csdIndex).c_str(), csd);
@@ -685 +696,2 @@ void AmendOutputFormatWithCodecSpecificData(const uint8_t* data, size_t size,
-            ALOGW("Expected two NAL units from AVC codec config, but %u found", csdIndex);
+            ALOGW("Expected two NAL units from AVC codec config, but %u found",
+                    csdIndex);
@@ -697 +709,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onWorkDone(const std::weak_ptr<Codec2Client::Component>& component,
+    virtual void onWorkDone(
+            const std::weak_ptr<Codec2Client::Component>& component,
@@ -708 +721,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-            const std::vector<std::shared_ptr<C2SettingResult>>& settingResult) override {
+            const std::vector<std::shared_ptr<C2SettingResult>>& settingResult
+            ) override {
@@ -712 +726,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onError(const std::weak_ptr<Codec2Client::Component>& component,
+    virtual void onError(
+            const std::weak_ptr<Codec2Client::Component>& component,
@@ -726 +741,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-        codec->mCallback->onError(errorCode == C2_NO_MEMORY ? NO_MEMORY : UNKNOWN_ERROR,
+        codec->mCallback->onError(
+                errorCode == C2_NO_MEMORY ? NO_MEMORY : UNKNOWN_ERROR,
@@ -729 +745,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onDeath(const std::weak_ptr<Codec2Client::Component>& component) override {
+    virtual void onDeath(
+            const std::weak_ptr<Codec2Client::Component>& component) override {
@@ -744 +761,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onFrameRendered(uint64_t bufferQueueId, int32_t slotId,
+    virtual void onFrameRendered(uint64_t bufferQueueId,
+                                 int32_t slotId,
@@ -750 +768,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onInputBufferDone(uint64_t frameIndex, size_t arrayIndex) override {
+    virtual void onInputBufferDone(
+            uint64_t frameIndex, size_t arrayIndex) override {
@@ -761,2 +780,2 @@ class CCodecCallbackImpl : public CCodecCallback {
-    explicit CCodecCallbackImpl(CCodec* codec) : mCodec(codec){}() = delete;
-    () = delete;
+    explicit CCodecCallbackImpl(CCodec *codec) : mCodec(codec) {}
+    ~CCodecCallbackImpl() override = default;
@@ -769,2 +788,6 @@ class CCodecCallbackImpl : public CCodecCallback {
-    void onOutputBuffersChanged() override { mCodec->mCallback->onOutputBuffersChanged(); }
-    void onFirstTunnelFrameReady() override { mCodec->mCallback->onFirstTunnelFrameReady(); }
+    void onOutputBuffersChanged() override {
+        mCodec->mCallback->onOutputBuffersChanged();
+    }
+    void onFirstTunnelFrameReady() override {
+        mCodec->mCallback->onFirstTunnelFrameReady();
+    }
@@ -776,2 +799,4 @@ CCodec::CCodec()
-      mConfig(new CCodecConfig) {}
-CCodec::~CCodec() {}
+      mConfig(new CCodecConfig) {
+}
+CCodec::~CCodec() {
+}
@@ -818 +843,2 @@ void CCodec::allocate(const sp<MediaCodecInfo>& codecInfo) {
-        SetPreferredCodec2ComponentStore(std::make_shared<Codec2ClientInterfaceWrapper>(client));
+        SetPreferredCodec2ComponentStore(
+                std::make_shared<Codec2ClientInterfaceWrapper>(client));
@@ -821,2 +847,5 @@ void CCodec::allocate(const sp<MediaCodecInfo>& codecInfo) {
-    c2_status_t status = Codec2Client::CreateComponentByName(componentName.c_str(), mClientListener,
-                                                             &comp, &client);
+    c2_status_t status = Codec2Client::CreateComponentByName(
+            componentName.c_str(),
+            mClientListener,
+            &comp,
+            &client);
@@ -900,2 +929,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if (encoder && msg->findInt32("android._input-metadata-buffer-type", &storeMeta) &&
-            storeMeta != kMetadataBufferTypeInvalid) {
+        if (encoder
+                && msg->findInt32("android._input-metadata-buffer-type", &storeMeta)
+                && storeMeta != kMetadataBufferTypeInvalid) {
@@ -918,2 +948,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if ((config->mDomain & Config::IS_DECODER) &&
-                    (config->mDomain & Config::IS_VIDEO)) {
+                if ((config->mDomain & Config::IS_DECODER)
+                        && (config->mDomain & Config::IS_VIDEO)) {
@@ -950,2 +980,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        ALOGD("[%s] buffers are %sbound to CCodec for this session", comp->getName().c_str(),
-              config->mBuffersBoundToCodec ? "" : "not ");
+        ALOGD("[%s] buffers are %sbound to CCodec for this session",
+              comp->getName().c_str(), config->mBuffersBoundToCodec ? "" : "not ");
@@ -963,3 +993,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            if ((config->mDomain & Config::IS_ENCODER) &&
-                !mime.equalsIgnoreCase(MEDIA_MIMETYPE_AUDIO_FLAC) &&
-                !msg->findInt32(KEY_BIT_RATE, &i32) && !msg->findFloat(KEY_BIT_RATE, &flt)) {
+            if ((config->mDomain & Config::IS_ENCODER)
+                    && !mime.equalsIgnoreCase(MEDIA_MIMETYPE_AUDIO_FLAC)
+                    && !msg->findInt32(KEY_BIT_RATE, &i32)
+                    && !msg->findFloat(KEY_BIT_RATE, &flt)) {
@@ -989,2 +1020,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (!msg->findInt32(KEY_BIT_RATE, &i32) &&
-                        !msg->findFloat(KEY_BIT_RATE, &flt)) {
+                    if (!msg->findInt32(KEY_BIT_RATE, &i32)
+                            && !msg->findFloat(KEY_BIT_RATE, &flt)) {
@@ -995,2 +1026,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if (!msg->findInt32(KEY_I_FRAME_INTERVAL, &i32) &&
-                    !msg->findFloat(KEY_I_FRAME_INTERVAL, &flt)) {
+                if (!msg->findInt32(KEY_I_FRAME_INTERVAL, &i32)
+                        && !msg->findFloat(KEY_I_FRAME_INTERVAL, &flt)) {
@@ -1000,2 +1031,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if (!msg->findInt32(KEY_FRAME_RATE, &i32) &&
-                    !msg->findFloat(KEY_FRAME_RATE, &flt)) {
+                if (!msg->findInt32(KEY_FRAME_RATE, &i32)
+                        && !msg->findFloat(KEY_FRAME_RATE, &flt)) {
@@ -1007,2 +1038,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE)) &&
-            (config->mDomain & Config::IS_ENCODER)) {
+        if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE))
+                && (config->mDomain & Config::IS_ENCODER)) {
@@ -1016 +1047,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if (!msg->findFloat(KEY_MAX_FPS_TO_ENCODER, &config->mISConfig->mMaxFps)) {
+                if (!msg->findFloat(
+                        KEY_MAX_FPS_TO_ENCODER, &config->mISConfig->mMaxFps)) {
@@ -1061,7 +1093,10 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            int vendorSdkVersion = base::GetIntProperty("ro.vendor.build.version.sdk",
-                                                        android_get_device_api_level());
-            if (mClient->query({}, {C2StoreFlexiblePixelFormatDescriptorsInfo::PARAM_TYPE},
-                               C2_MAY_BLOCK, &heapParams) == C2_OK &&
-                heapParams.size() == 1u) {
-                pixelFormatInfo =
-                        C2StoreFlexiblePixelFormatDescriptorsInfo::From(heapParams[0].get());
+            int vendorSdkVersion = base::GetIntProperty(
+                    "ro.vendor.build.version.sdk", android_get_device_api_level());
+            if (mClient->query(
+                        {},
+                        {C2StoreFlexiblePixelFormatDescriptorsInfo::PARAM_TYPE},
+                        C2_MAY_BLOCK,
+                        &heapParams) == C2_OK
+                    && heapParams.size() == 1u) {
+                pixelFormatInfo = C2StoreFlexiblePixelFormatDescriptorsInfo::From(
+                        heapParams[0].get());
@@ -1085,2 +1120,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (desc.layout == C2Color::PLANAR_PACKED &&
-                        flexPlanarPixelFormat.count(desc.bitDepth) == 0) {
+                    if (desc.layout == C2Color::PLANAR_PACKED
+                            && flexPlanarPixelFormat.count(desc.bitDepth) == 0) {
@@ -1089,2 +1124,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (desc.layout == C2Color::SEMIPLANAR_PACKED &&
-                        flexSemiPlanarPixelFormat.count(desc.bitDepth) == 0) {
+                    if (desc.layout == C2Color::SEMIPLANAR_PACKED
+                            && flexSemiPlanarPixelFormat.count(desc.bitDepth) == 0) {
@@ -1106,2 +1141 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                              "using default %splanar color format",
-                              prefix);
+                                "using default %splanar color format", prefix);
@@ -1166 +1200,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            if (msg->findInt32("width", &width) && msg->findInt32("height", &height)) {
+            if (msg->findInt32("width", &width)
+                    && msg->findInt32("height", &height)) {
@@ -1182,2 +1217,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if (android::media::codec::provider_->region_of_interest() &&
-            android::media::codec::provider_->region_of_interest_support()) {
+        if (android::media::codec::provider_->region_of_interest()
+            && android::media::codec::provider_->region_of_interest_support()) {
@@ -1202,2 +1237,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        err = config->getConfigUpdateFromSdkParams(comp, sdkParams, Config::IS_CONFIG,
-                                                   C2_DONT_BLOCK, &configUpdate);
+        err = config->getConfigUpdateFromSdkParams(
+                comp, sdkParams, Config::IS_CONFIG, C2_DONT_BLOCK, &configUpdate);
@@ -1208,2 +1243,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_ENCODER) && (config->mDomain & Config::IS_VIDEO) &&
-            sdkParams->findInt32(KEY_MAX_B_FRAMES, &maxBframes) && maxBframes > 0) {
+        if ((config->mDomain & Config::IS_ENCODER)
+                && (config->mDomain & Config::IS_VIDEO)
+                && sdkParams->findInt32(KEY_MAX_B_FRAMES, &maxBframes)
+                && maxBframes > 0) {
@@ -1213 +1250,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            gop->m.values[1] = {C2Config::picture_type_t(P_FRAME | B_FRAME), uint32_t(maxBframes)};
+            gop->m.values[1] = {
+                C2Config::picture_type_t(P_FRAME | B_FRAME),
+                uint32_t(maxBframes)
+            };
@@ -1216 +1256,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_ENCODER) && (config->mDomain & Config::IS_VIDEO)) {
+        if ((config->mDomain & Config::IS_ENCODER)
+                && (config->mDomain & Config::IS_VIDEO)) {
@@ -1246,2 +1287,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_VIDEO) &&
-            msg->findInt32("android._background-mode", &background) && background) {
+        if ((config->mDomain & Config::IS_VIDEO)
+                && msg->findInt32("android._background-mode", &background)
+                && background) {
@@ -1268,2 +1310,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if (config->mDomain & (Config::IS_IMAGE | Config::IS_VIDEO) &&
-            !sdkParams->findInt32("color-transfer-request", &colorTransferRequest)) {
+        if (config->mDomain & (Config::IS_IMAGE | Config::IS_VIDEO)
+                && !sdkParams->findInt32("color-transfer-request", &colorTransferRequest)) {
@@ -1274 +1316,5 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            c2err = comp->query({&usage, &maxInputSize, &prepend}, indices, C2_DONT_BLOCK, &params);
+            c2err = comp->query(
+                    { &usage, &maxInputSize, &prepend },
+                    indices,
+                    C2_DONT_BLOCK,
+                    &params);
@@ -1276 +1322,5 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            c2err = comp->query({&usage, &maxInputSize, &prepend}, {}, C2_DONT_BLOCK, &params);
+            c2err = comp->query(
+                    { &usage, &maxInputSize, &prepend },
+                    {},
+                    C2_DONT_BLOCK,
+                    &params);
@@ -1315,2 +1365 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                          "recommendation.",
-                          clientInputSize, maxInputSize.value);
+                          "recommendation.", clientInputSize, maxInputSize.value);
@@ -1331,3 +1380,5 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_VIDEO) && (config->mDomain & Config::IS_ENCODER) &&
-            msg->findInt32(KEY_PREPEND_HEADER_TO_SYNC_FRAMES, &clientPrepend) && clientPrepend &&
-            (!prepend || prepend.value != PREPEND_HEADER_TO_ALL_SYNC)) {
+        if ((config->mDomain & Config::IS_VIDEO)
+                && (config->mDomain & Config::IS_ENCODER)
+                && msg->findInt32(KEY_PREPEND_HEADER_TO_SYNC_FRAMES, &clientPrepend)
+                && clientPrepend
+                && (!prepend || prepend.value != PREPEND_HEADER_TO_ALL_SYNC)) {
@@ -1340,2 +1391,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            if (msg->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo) &&
-                !config->mOutputFormat->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo)) {
+            if (msg->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo)
+                    && !config->mOutputFormat->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo)) {
@@ -1409,3 +1460,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                std::shared_ptr<C2GraphicBlock> block =
-                        FetchGraphicBlock(align(width, 2), align(height, 2), componentColorFormat,
-                                          blockUsage, {comp->getName()});
+                std::shared_ptr<C2GraphicBlock> block = FetchGraphicBlock(
+                        align(width, 2), align(height, 2), componentColorFormat, blockUsage,
+                        {comp->getName()});
@@ -1415 +1466,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                            config->mInputFormat, block,
+                            config->mInputFormat,
+                            block,
@@ -1425,2 +1477,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (imageData && imageData->data() &&
-                        imageData->size() >= sizeof(MediaImage2)) {
+                    if (imageData && imageData->data()
+                            && imageData->size() >= sizeof(MediaImage2)) {
@@ -1433,2 +1485,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                            int64_t offsetDelta = (int64_t)img->mPlane[1].mOffset -
-                                                  (int64_t)img->mPlane[0].mOffset;
+                            int64_t offsetDelta =
+                                (int64_t)img->mPlane[1].mOffset - (int64_t)img->mPlane[0].mOffset;
@@ -1452,2 +1504,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_ENCODER) && (config->mDomain & Config::IS_VIDEO) &&
-            msg->findInt32(KEY_VIDEO_ENCODING_STATISTICS_LEVEL, &encodingStatisticsLevel)) {
+        if ((config->mDomain & Config::IS_ENCODER)
+            && (config->mDomain & Config::IS_VIDEO)
+            && msg->findInt32(KEY_VIDEO_ENCODING_STATISTICS_LEVEL, &encodingStatisticsLevel)) {
@@ -1468,2 +1521,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        ALOGD("setup formats input: %s", config->mInputFormat->debugString().c_str());
-        ALOGD("setup formats output: %s", config->mOutputFormat->debugString().c_str());
+        ALOGD("setup formats input: %s",
+                config->mInputFormat->debugString().c_str());
+        ALOGD("setup formats output: %s",
+                config->mOutputFormat->debugString().c_str());
@@ -1508,3 +1563,4 @@ sp<PersistentSurface> CCodec::CreateOmxInputSurface() {
-    typedef android::hardware::graphics::bufferqueue::V1_0::IGraphicBufferProducer
-            HGraphicBufferProducer;
-    typedef android::hardware::media::omx::V1_0::IGraphicBufferSource HGraphicBufferSource;
+    typedef android::hardware::graphics::bufferqueue::V1_0::
+            IGraphicBufferProducer HGraphicBufferProducer;
+    typedef android::hardware::media::omx::V1_0::
+            IGraphicBufferSource HGraphicBufferSource;
@@ -1516 +1572,3 @@ sp<PersistentSurface> CCodec::CreateOmxInputSurface() {
-            [&s, &gbp, &gbs](OmxStatus status, const android::sp<HGraphicBufferProducer>& producer,
+            [&s, &gbp, &gbs](
+                    OmxStatus status,
+                    const android::sp<HGraphicBufferProducer>& producer,
@@ -1554,2 +1612,2 @@ void CCodec::createInputSurface() {
-            err = setupInputSurface(
-                    std::make_shared<AGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            err = setupInputSurface(std::make_shared<AGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1569 +1627,2 @@ void CCodec::createInputSurface() {
-            err = setupInputSurface(std::make_shared<C2InputSurfaceWrapper>(inputSurface));
+            err = setupInputSurface(std::make_shared<C2InputSurfaceWrapper>(
+                    inputSurface));
@@ -1576,2 +1635,2 @@ void CCodec::createInputSurface() {
-            err = setupInputSurface(
-                    std::make_shared<HGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            err = setupInputSurface(std::make_shared<HGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1597 +1656,3 @@ void CCodec::createInputSurface() {
-    mCallback->onInputSurfaceCreated(inputFormat, outputFormat,
+    mCallback->onInputSurfaceCreated(
+            inputFormat,
+            outputFormat,
@@ -1621,2 +1682,2 @@ status_t CCodec::setupInputSurface(const std::shared_ptr<InputSurfaceWrapper>& s
-    android_dataspace dataSpace =
-            getDataSpaceForColorAspects(colorAspects, false);
+    android_dataspace dataSpace = getDataSpaceForColorAspects(
+            colorAspects, false);
@@ -1626 +1687,2 @@ status_t CCodec::setupInputSurface(const std::shared_ptr<InputSurfaceWrapper>& s
-    ALOGD("input format %s to %s", inputFormatChanged ? "changed" : "unchanged",
+    ALOGD("input format %s to %s",
+            inputFormatChanged ? "changed" : "unchanged",
@@ -1664,2 +1726,2 @@ void CCodec::setInputSurface(const sp<PersistentSurface>& surface) {
-            status_t err = setupInputSurface(
-                    std::make_shared<AGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            status_t err = setupInputSurface(std::make_shared<AGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1693,2 +1755,2 @@ void CCodec::setInputSurface(const sp<PersistentSurface>& surface) {
-            status_t err = setupInputSurface(
-                    std::make_shared<HGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            status_t err = setupInputSurface(std::make_shared<HGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1744 +1806,2 @@ void CCodec::start() {
-        mCallback->onError(toStatusT(err, C2_OPERATION_Component_start), ACTION_CODE_FATAL);
+        mCallback->onError(toStatusT(err, C2_OPERATION_Component_start),
+                           ACTION_CODE_FATAL);
@@ -1803,2 +1866,4 @@ void CCodec::initiateStop() {
-        if (state->get() == ALLOCATED || state->get() == RELEASED || state->get() == STOPPING ||
-            state->get() == RELEASING) {
+        if (state->get() == ALLOCATED
+                || state->get() == RELEASED
+                || state->get() == STOPPING
+                || state->get() == RELEASING) {
@@ -1835 +1900,2 @@ void CCodec::stop(bool pushBlankBuffer) {
-    bool stopHalBeforeSurface = Codec2Client::IsAidlSelected() ||
+    bool stopHalBeforeSurface =
+            Codec2Client::IsAidlSelected() ||
@@ -1886 +1952,3 @@ void CCodec::initiateRelease(bool sendCallback ) {
-        if (state->get() == STARTING || state->get() == RUNNING || state->get() == STOPPING) {
+        if (state->get() == STARTING
+                || state->get() == RUNNING
+                || state->get() == STOPPING) {
@@ -1903,3 +1971,2 @@ void CCodec::initiateRelease(bool sendCallback ) {
-    std::thread([thiz, sendCallback, pushBlankBuffer] {
-        thiz->release(sendCallback, pushBlankBuffer);
-    }).detach();
+    std::thread([thiz, sendCallback, pushBlankBuffer]
+                { thiz->release(sendCallback, pushBlankBuffer); }).detach();
@@ -1921 +1988,2 @@ void CCodec::release(bool sendCallback, bool pushBlankBuffer) {
-    bool stopHalBeforeSurface = Codec2Client::IsAidlSelected() ||
+    bool stopHalBeforeSurface =
+            Codec2Client::IsAidlSelected() ||
@@ -2092,2 +2160,2 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-    if (params->findInt32("audio-hw-sync", &syncId) ||
-        params->findInt32("hw-av-sync-id", &syncId)) {
+    if (params->findInt32("audio-hw-sync", &syncId)
+            || params->findInt32("hw-av-sync-id", &syncId)) {
@@ -2098,2 +2166,3 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-    if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE)) &&
-        (config->mDomain & Config::IS_ENCODER) && config->mInputSurface && config->mISConfig) {
+    if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE))
+            && (config->mDomain & Config::IS_ENCODER)
+            && config->mInputSurface && config->mISConfig) {
@@ -2114,2 +2183,2 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-            config->mInputFormat->setInt64("android._stop-time-offset-us",
-                                           config->mISConfig->mInputDelayUs);
+            config->mInputFormat->setInt64(
+                    "android._stop-time-offset-us", config->mISConfig->mInputDelayUs);
@@ -2118,2 +2187,2 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-    if (android::media::codec::provider_->region_of_interest() &&
-        android::media::codec::provider_->region_of_interest_support()) {
+    if (android::media::codec::provider_->region_of_interest()
+            && android::media::codec::provider_->region_of_interest_support()) {
@@ -2121,3 +2190,3 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-        if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE)) &&
-            (config->mDomain & Config::IS_ENCODER) &&
-            params->findBuffer(PARAMETER_KEY_QP_OFFSET_MAP, &qpOffsetMap)) {
+        if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE))
+                && (config->mDomain & Config::IS_ENCODER)
+                && params->findBuffer(PARAMETER_KEY_QP_OFFSET_MAP, &qpOffsetMap)) {
@@ -2157,5 +2226,5 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-    (void)config->getConfigUpdateFromSdkParams(comp, params, Config::IS_PARAM, C2_MAY_BLOCK,
-                                               &configUpdate);
-    if (config->mInputSurface == nullptr &&
-        (property_get_bool("debug.stagefright.ccodec_delayed_params", false) ||
-         comp->getName().find("c2.android.") == 0)) {
+    (void)config->getConfigUpdateFromSdkParams(
+            comp, params, Config::IS_PARAM, C2_MAY_BLOCK, &configUpdate);
+    if (config->mInputSurface == nullptr
+            && (property_get_bool("debug.stagefright.ccodec_delayed_params", false)
+                    || comp->getName().find("c2.android.") == 0)) {
@@ -2195 +2264,2 @@ void CCodec::signalRequestIDRFrame() {
-    params.push_back(std::make_unique<C2StreamRequestSyncFrameTuning::output>(0u, true));
+    params.push_back(
+            std::make_unique<C2StreamRequestSyncFrameTuning::output>(0u, true));
@@ -2203 +2273,2 @@ status_t CCodec::querySupportedParameters(std::vector<std::string>* names) {
-status_t CCodec::describeParameter(const std::string& name, CodecParameterDescriptor* desc) {
+status_t CCodec::describeParameter(
+        const std::string &name, CodecParameterDescriptor *desc) {
@@ -2324,2 +2395,3 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                if (!work->worklets.empty() &&
-                    (work->worklets.front()->output.flags & C2FrameData::FLAG_DISCARD_FRAME) == 0) {
+                if (!work->worklets.empty()
+                        && (work->worklets.front()->output.flags
+                                & C2FrameData::FLAG_DISCARD_FRAME) == 0) {
@@ -2327,2 +2399,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                    for (const std::unique_ptr<C2Param>& param :
-                         work->worklets.front()->output.configUpdate) {
+                    for (const std::unique_ptr<C2Param> &param
+                            : work->worklets.front()->output.configUpdate) {
@@ -2342,2 +2414,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                            updates.emplace_back(
-                                    new C2StreamCropRectInfo::output(stream, block.crop()));
+                            updates.emplace_back(new C2StreamCropRectInfo::output(
+                                    stream, block.crop()));
@@ -2359 +2431,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                            C2StreamSurfaceScalingInfo::output::PARAM_TYPE};
+                        C2StreamSurfaceScalingInfo::output::PARAM_TYPE
+                    };
@@ -2377,2 +2450,4 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                    if (work->worklets.empty() || !work->worklets.back() ||
-                        (work->worklets.back()->output.flags & C2FrameData::FLAG_INCOMPLETE) == 0) {
+                    if (work->worklets.empty()
+                           || !work->worklets.back()
+                           || (work->worklets.back()->output.flags
+                                  & C2FrameData::FLAG_INCOMPLETE) == 0) {
@@ -2384,2 +2459,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                    AmendOutputFormatWithCodecSpecificData(initData->m.value, initData->flexCount(),
-                                                           config->mCodingMediaType,
+                    AmendOutputFormatWithCodecSpecificData(
+                            initData->m.value, initData->flexCount(), config->mCodingMediaType,
@@ -2390,2 +2465,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-            mChannel->onWorkDone(std::move(work), outputFormat,
-                                 initData ? initData.get() : nullptr);
+            mChannel->onWorkDone(
+                    std::move(work), outputFormat, initData ? initData.get() : nullptr);
@@ -2418 +2493,3 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-void CCodec::setDeadline(const TimePoint& now, const std::chrono::milliseconds& timeout,
+void CCodec::setDeadline(
+        const TimePoint &now,
+        const std::chrono::milliseconds &timeout,
@@ -2424 +2501,2 @@ void CCodec::setDeadline(const TimePoint& now, const std::chrono::milliseconds&
-status_t CCodec::configureTunneledVideoPlayback(std::shared_ptr<Codec2Client::Component> comp,
+status_t CCodec::configureTunneledVideoPlayback(
+        std::shared_ptr<Codec2Client::Component> comp,
@@ -2430,2 +2508,4 @@ status_t CCodec::configureTunneledVideoPlayback(std::shared_ptr<Codec2Client::Co
-                    1, C2PortTunneledModeTuning::Struct::SIDEBAND,
-                    C2PortTunneledModeTuning::Struct::REALTIME, 0);
+            1,
+            C2PortTunneledModeTuning::Struct::SIDEBAND,
+            C2PortTunneledModeTuning::Struct::REALTIME,
+            0);
@@ -2487,2 +2567 @@ void CCodec::initiateReleaseIfStuck() {
-                  "and the component is already released",
-                  name.c_str());
+                  "and the component is already released", name.c_str());
@@ -2500 +2579,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-    std::shared_ptr<Codec2Client::InputSurface> inputSurface = Codec2Client::CreateInputSurface();
+    std::shared_ptr<Codec2Client::InputSurface> inputSurface =
+            Codec2Client::CreateInputSurface();
@@ -2514 +2594,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-                return new PersistentSurface(gbs->getIGraphicBufferProducer(), wrapper->asBinder());
+                return new PersistentSurface(
+                      gbs->getIGraphicBufferProducer(), wrapper->asBinder());
@@ -2524,2 +2605,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-                return new PersistentSurface(gbs->getIGraphicBufferProducer(),
-                                             new TWGraphicBufferSource(gbs));
+                return new PersistentSurface(
+                        gbs->getIGraphicBufferProducer(), new TWGraphicBufferSource(gbs));
@@ -2533 +2614,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-            static_cast<sp<android::hidl::base::V1_0::IBase>>(inputSurface->getHalInterface()));
+            static_cast<sp<android::hidl::base::V1_0::IBase>>(
+            inputSurface->getHalInterface()));
@@ -2537 +2619 @@ class IntfCache {
-    IntfCache()
+    IntfCache() = default;
@@ -2547,2 +2629,2 @@ class IntfCache {
-        mFields.push_back(
-                C2FieldSupportedValuesQuery::Possible(C2ParamField{&sUsage, &sUsage.value}));
+        mFields.push_back(C2FieldSupportedValuesQuery::Possible(
+                C2ParamField{&sUsage, &sUsage.value}));
@@ -2551,2 +2633,2 @@ class IntfCache {
-            ALOGW("IntfCache [%s]: failed to query usage supported value (err=%d)", name.c_str(),
-                  err);
+            ALOGW("IntfCache [%s]: failed to query usage supported value (err=%d)",
+                    name.c_str(), err);
@@ -2556,4 +2638,8 @@ class IntfCache {
-        err = intf->query({&mApiFeatures},
-                          {C2StreamBufferTypeSetting::input::PARAM_TYPE,
-                           C2PortAllocatorsTuning::input::PARAM_TYPE},
-                          C2_MAY_BLOCK, &params);
+        err = intf->query(
+                {&mApiFeatures},
+                {
+                    C2StreamBufferTypeSetting::input::PARAM_TYPE,
+                    C2PortAllocatorsTuning::input::PARAM_TYPE
+                },
+                C2_MAY_BLOCK,
+                &params);
@@ -2561 +2647,2 @@ class IntfCache {
-            ALOGW("IntfCache [%s]: failed to query api features (err=%d)", name.c_str(), err);
+            ALOGW("IntfCache [%s]: failed to query api features (err=%d)",
+                    name.c_str(), err);
@@ -2570 +2657,2 @@ class IntfCache {
-                mInputStreamFormat.reset(C2StreamBufferTypeSetting::input::From(param));
+                mInputStreamFormat.reset(
+                        C2StreamBufferTypeSetting::input::From(param));
@@ -2572 +2660,2 @@ class IntfCache {
-                mInputAllocators.reset(C2PortAllocatorsTuning::input::From(param));
+                mInputAllocators.reset(
+                        C2PortAllocatorsTuning::input::From(param));
@@ -2583 +2672,3 @@ class IntfCache {
-    const C2ApiFeaturesSetting& getApiFeatures() const { return mApiFeatures; }
+    const C2ApiFeaturesSetting &getApiFeatures() const {
+        return mApiFeatures;
+    }
@@ -2627,2 +2718,4 @@ static const IntfCache& GetIntfCache(const std::string& name) {
-static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-                                      C2Allocator::type_t type, std::set<C2Allocator::id_t>* ids) {
+static status_t GetCommonAllocatorIds(
+        const std::vector<std::string> &names,
+        C2Allocator::type_t type,
+        std::set<C2Allocator::id_t> *ids) {
@@ -2646,2 +2739,2 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-            if (streamFormat.value == C2BufferData::GRAPHIC ||
-                streamFormat.value == C2BufferData::GRAPHIC_CHUNKS) {
+            if (streamFormat.value == C2BufferData::GRAPHIC
+                    || streamFormat.value == C2BufferData::GRAPHIC_CHUNKS) {
@@ -2654,2 +2747 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-                      " uses default allocator id(%d)",
-                      name.c_str(), type, defaultAllocatorId);
+                        " uses default allocator id(%d)", name.c_str(), type, defaultAllocatorId);
@@ -2663 +2755,2 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-                ids->insert(allocators.m.values, allocators.m.values + allocators.flexCount());
+                ids->insert(allocators.m.values,
+                            allocators.m.values + allocators.flexCount());
@@ -2707,2 +2800,2 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-static status_t CalculateMinMaxUsage(const std::vector<std::string>& names, uint64_t* minUsage,
-                                     uint64_t* maxUsage) {
+static status_t CalculateMinMaxUsage(
+        const std::vector<std::string> &names, uint64_t *minUsage, uint64_t *maxUsage) {
@@ -2743,2 +2836,2 @@ static status_t CalculateMinMaxUsage(const std::vector<std::string>& names, uint
-status_t CCodec::CanFetchLinearBlock(const std::vector<std::string>& names,
-                                     const C2MemoryUsage& usage, bool* isCompatible) {
+status_t CCodec::CanFetchLinearBlock(
+        const std::vector<std::string> &names, const C2MemoryUsage &usage, bool *isCompatible) {
@@ -2787,2 +2880,2 @@ static std::shared_ptr<C2BlockPool> GetPool(C2Allocator::id_t allocId) {
-std::shared_ptr<C2LinearBlock> CCodec::FetchLinearBlock(size_t capacity, const C2MemoryUsage& usage,
-                                                        const std::vector<std::string>& names) {
+std::shared_ptr<C2LinearBlock> CCodec::FetchLinearBlock(
+        size_t capacity, const C2MemoryUsage &usage, const std::vector<std::string> &names) {
@@ -2817 +2910,2 @@ std::shared_ptr<C2LinearBlock> CCodec::FetchLinearBlock(size_t capacity, const C
-status_t CCodec::CanFetchGraphicBlock(const std::vector<std::string>& names, bool* isCompatible) {
+status_t CCodec::CanFetchGraphicBlock(
+        const std::vector<std::string> &names, bool *isCompatible) {
@@ -2830,2 +2924,5 @@ status_t CCodec::CanFetchGraphicBlock(const std::vector<std::string>& names, boo
-std::shared_ptr<C2GraphicBlock> CCodec::FetchGraphicBlock(int32_t width, int32_t height,
-                                                          int32_t format, uint64_t usage,
+std::shared_ptr<C2GraphicBlock> CCodec::FetchGraphicBlock(
+        int32_t width,
+        int32_t height,
+        int32_t format,
+        uint64_t usage,
@@ -2858,2 +2955,2 @@ std::shared_ptr<C2GraphicBlock> CCodec::FetchGraphicBlock(int32_t width, int32_t
-        err = pool->fetchGraphicBlock(width, height, halPixelFormat, C2MemoryUsage{minUsage},
-                                      &block);
+        err = pool->fetchGraphicBlock(
+                width, height, halPixelFormat, C2MemoryUsage{minUsage}, &block);
