--- a/home/whalien/codebase/python/mergebot-eva/output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_merged.h
@@ -42,2 +42 @@
-#define REDIS_REQUEST_MAX_SIZE \
-  (1024 * 1024 * 256)
+#define REDIS_REQUEST_MAX_SIZE (1024*1024*256)
@@ -46,2 +45 @@
-#define REDIS_MAX_LOGMSG_LEN \
-  1024
+#define REDIS_MAX_LOGMSG_LEN 1024
@@ -83,8 +80,0 @@
-#define REDIS_IO_LOAD 1
-#define REDIS_IO_SAVE 2
-#define REDIS_IO_LOADINPROG 4
-#define REDIS_IO_SAVEINPROG 8
-#define REDIS_IO_ONLYLOADS 1
-#define REDIS_IO_ASAP 2
-#define REDIS_MAX_COMPLETED_JOBS_PROCESSED 1
-#define REDIS_THREAD_STACK_SIZE (1024 * 1024 * 4)
@@ -96 +85,0 @@
-#define REDIS_IO_WAIT 32
@@ -99,2 +88,2 @@
-#define REDIS_UNBLOCKED \
-  256
+#define REDIS_UNBLOCKED 256
+#define REDIS_LUA_CLIENT 512
@@ -109,4 +98,2 @@
-#define REDIS_REPL_WAIT_BGSAVE_START \
-  3
-#define REDIS_REPL_WAIT_BGSAVE_END \
-  4
+#define REDIS_REPL_WAIT_BGSAVE_START 3
+#define REDIS_REPL_WAIT_BGSAVE_END 4
@@ -125,0 +113 @@
+#define REDIS_NOTUSED(V) ((void) V)
@@ -147,0 +136,2 @@
+#define redisAssert(_e) ((_e)?(void)0 : (_redisAssert(#_e,__FILE__,__LINE__),_exit(1)))
+#define redisPanic(_e) _redisPanic(#_e,__FILE__,__LINE__),_exit(1)
@@ -167,0 +158,6 @@ typedef struct vmPointer {
+#define initStaticStringObject(_var,_ptr) do { \
+    _var.refcount = 1; \
+    _var.type = REDIS_STRING; \
+    _var.encoding = REDIS_ENCODING_RAW; \
+    _var.ptr = _ptr; \
+} while(0);
@@ -271,4 +269 @@ typedef struct clusterLink {
-#define REDIS_NODE_NULL_NAME \
-  "\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000" \
-  "\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000" \
-  "\000\000\000\000"
+#define REDIS_NODE_NULL_NAME "\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000"
@@ -895,0 +876,6 @@ void evalShaCommand(redisClient *c);
+#if defined(__GNUC__)
+void *calloc(size_t count, size_t size) __attribute__ ((deprecated));
+void free(void *ptr) __attribute__ ((deprecated));
+void *malloc(size_t size) __attribute__ ((deprecated));
+void *realloc(void *ptr, size_t size) __attribute__ ((deprecated));
+#endif
