--- a/./output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_conflict.h
+++ b/./output/redis/45ec3243-d5b36c51-55937b79/src@redis.no_comments_merged.h
@@ -135,5 +134,0 @@
-<<<<<<< HEAD
-#define REDIS_BGSAVE_THREAD_UNACTIVE 0
-#define REDIS_BGSAVE_THREAD_ACTIVE 1
-#define REDIS_BGSAVE_THREAD_DONE_OK 2
-#define REDIS_BGSAVE_THREAD_DONE_ERR 3
@@ -141,7 +135,0 @@
-||||||| 55937b797
-#define REDIS_BGSAVE_THREAD_UNACTIVE 0
-#define REDIS_BGSAVE_THREAD_ACTIVE 1
-#define REDIS_BGSAVE_THREAD_DONE_OK 2
-#define REDIS_BGSAVE_THREAD_DONE_ERR 3
-=======
->>>>>>> d5b36c51
