--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/6a2f32b4-ff135b34-1c8f7c1f/format.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/6a2f32b4-ff135b34-1c8f7c1f/format.no_comments_merged.c
@@ -15 +15 @@
-struct format_entry
+struct format_entry;
@@ -81,0 +88 @@ static int format_entry_cmp(struct format_entry *, struct format_entry *);
+RB_GENERATE_STATIC(format_entry_tree, format_entry, entry, format_entry_cmp);
@@ -166 +177,4 @@ static void printflike(3, 4)
-static void format_job_update(struct job *job) {
+#define format_log(ft,fmt,...) format_log1(ft, __func__, fmt, ##__VA_ARGS__)
+static void
+format_job_update(struct job *job)
+{
@@ -498,6 +579,9 @@ static voidformat_cb_current_path(struct format_tree *ft,
-  if (wp == NULL) return;
-  cwd = get_proc_cwd(wp->fd);
-  if (cwd != NULL) fe->value = xstrdup(cwd);
-}
-static void format_cb_history_bytes(struct format_tree *ft,
-                                    struct format_entry *fe) {
+ if (wp == NULL)
+  return;
+ cwd = osdep_get_cwd(wp->fd);
+ if (cwd != NULL)
+  fe->value = xstrdup(cwd);
+}
+static void
+format_cb_history_bytes(struct format_tree *ft, struct format_entry *fe)
+{
