diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_merged.h
index 538139d..6730e88 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odr_config.no_comments_merged.h
@@ -46 +46,2 @@ class OdrConfig final {
-        artifact_dir_(GetApexDataDalvikCacheDirectory(InstructionSet::kNone)) {}
+      artifact_dir_(GetApexDataDalvikCacheDirectory(InstructionSet::kNone)) {
+  }
@@ -91,4 +92,12 @@ class OdrConfig final {
-  bool HasPartialCompilation() const { return partial_compilation_.has_value(); }
-  bool GetPartialCompilation() const { return partial_compilation_.value_or(true); }
-  bool GetRefresh() const { return refresh_.value_or(true); }
-  const std::string& GetSystemServerClasspath() const { return system_server_classpath_; }
+  bool HasPartialCompilation() const {
+    return partial_compilation_.has_value();
+  }
+  bool GetPartialCompilation() const {
+    return partial_compilation_.value_or(true);
+  }
+  bool GetRefresh() const {
+    return refresh_.value_or(true);
+  }
+  const std::string& GetSystemServerClasspath() const {
+    return system_server_classpath_;
+  }
@@ -98 +107,3 @@ class OdrConfig final {
-  const std::string& GetStagingDir() const { return staging_dir_; }
+  const std::string& GetStagingDir() const {
+    return staging_dir_;
+  }
@@ -106 +117,3 @@ class OdrConfig final {
-  void SetArtifactDirectory(const std::string& artifact_dir) { artifact_dir_ = artifact_dir; }
+  void SetArtifactDirectory(const std::string& artifact_dir) {
+    artifact_dir_ = artifact_dir;
+  }
@@ -108,2 +121,6 @@ class OdrConfig final {
-  void SetPartialCompilation(bool value) { partial_compilation_ = value; }
-  void SetRefresh(bool value) { refresh_ = value; }
+  void SetPartialCompilation(bool value) {
+    partial_compilation_ = value;
+  }
+  void SetRefresh(bool value) {
+    refresh_ = value;
+  }
@@ -120 +137,3 @@ class OdrConfig final {
-  void SetStagingDir(const std::string& staging_dir) { staging_dir_ = staging_dir; }
+  void SetStagingDir(const std::string& staging_dir) {
+    staging_dir_ = staging_dir;
+  }
@@ -145 +164,3 @@ class OdrConfig final {
-  OdrConfig() OdrConfig(const OdrConfig&) OdrConfig& operator=(const OdrConfig&) = delete
+  OdrConfig() = delete;
+  OdrConfig(const OdrConfig&) = delete;
+  OdrConfig& operator=(const OdrConfig&) = delete;
