--- a/./output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odrefresh.no_comments_mergegen.cc
+++ b/./output/art/9395cbb4-2c4b444b-c4be0fca/odrefresh@odrefresh.no_comments_merged.cc
@@ -27 +27 @@
-#include <regex> #include <set>
+#include <set>
@@ -73,0 +77 @@ constexpr const char* kCacheInfoFile = "cache-info.xml";
+constexpr time_t kMaximumExecutionSeconds = 300;
@@ -126,0 +131,5 @@ bool MoveOrEraseFiles(const std::vector<std::unique_ptr<File>>& files,
+std::optional<apex::ApexInfo> GetArtApexInfo(const std::vector<apex::ApexInfo>& info_list) {
+  auto it = std::find_if(info_list.begin(), info_list.end(), [](const apex::ApexInfo& info) {
+    return info.getModuleName() == "com.android.art";
+  });
+  return it != info_list.end() ? std::make_optional(*it) : std::nullopt;
@@ -1264 +1273,4 @@ WARN_UNUSED ExitCode OnDeviceRefresh::Compile(OdrMetrics& metrics,
-if (config_.GetRefresh()) { Result<void> result = RefreshExistingArtifacts(); if (!result.ok()) { LOG(ERROR) << "Failed to refresh existing artifacts: " << result.error(); return ExitCode::kCleanupFailed;
+  if (config_.GetRefresh()) {
+    Result<void> result = RefreshExistingArtifacts();
+    if (!result.ok()) {
+      LOG(ERROR) << "Failed to refresh existing artifacts: " << result.error();
