diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_merged.cc
index 1386091..7d7db8e 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-std::string Version() < < < < < < < HEAD { return "1.34.0-dev"; }
-|||||||
-{ return "1.33.0-dev"; }
-=======
-{ return "1.33.2"; }
->>>>>>> ee5b762f33a42170144834f5ab7efda9d76c480b
+std::string Version() { return "1.34.0-dev"; }
