diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@objective-c@tests@version.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@objective-c@tests@version.no_comments_merged.h
index e24c145..3fb9aab 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@objective-c@tests@version.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/20c350fd-ee5b762f-63369300/src@objective-c@tests@version.no_comments_merged.h
@@ -1 +0,0 @@
-#define GRPC_OBJC_VERSION_STRING @"1.33.2"
@@ -4,2 +2,0 @@
-#define GRPC_OBJC_VERSION_STRING @"1.33.0-dev"
-#define GRPC_C_VERSION_STRING @"12.0.0"
