--- a/./output/grpc/19294066-be58863a-172f61a3/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_mergegen.cc
+++ b/./output/grpc/19294066-be58863a-172f61a3/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_merged.cc
@@ -916 +916,3 @@ void grpc_chttp2_add_incoming_goaway(grpc_chttp2_transport* t,
-absl::Status status = grpc_error_to_absl_status(t->goaway_error); lie: use transient failure from the transport to indicate goaway has been
+  absl::Status status = grpc_error_to_absl_status(t->goaway_error);
+  connectivity_state_set(t, GRPC_CHANNEL_TRANSIENT_FAILURE, status,
+                         "got_goaway");
