diff --git a/./output/grpc/19294066-be58863a-172f61a3/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_conflict.cc b/./output/grpc/19294066-be58863a-172f61a3/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_merged.cc
index f1cfa3e..8f06c14 100644
--- a/./output/grpc/19294066-be58863a-172f61a3/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_conflict.cc
+++ b/./output/grpc/19294066-be58863a-172f61a3/src@core@ext@transport@chttp2@transport@chttp2_transport.no_comments_merged.cc
@@ -916,5 +915,0 @@ void grpc_chttp2_add_incoming_goaway(grpc_chttp2_transport* t,
-<<<<<<< HEAD
-  connectivity_state_set(t, GRPC_CHANNEL_TRANSIENT_FAILURE, "got_goaway");
-||||||| 172f61a3a2
-  connectivity_state_set(t, GRPC_CHANNEL_TRANSIENT_FAILURE, "got_goaway");
-=======
@@ -924 +918,0 @@ void grpc_chttp2_add_incoming_goaway(grpc_chttp2_transport* t,
->>>>>>> be58863a
