--- a/./output/tmux/191a8365-f3931497-674ec410/format.no_comments_conflict.c
+++ b/./output/tmux/191a8365-f3931497-674ec410/format.no_comments_merged.c
@@ -584 +583,0 @@ format_cb_current_path(struct format_tree *ft)
-<<<<<<< HEAD
@@ -586,8 +584,0 @@ format_cb_current_path(struct format_tree *ft)
- if (cwd != NULL)
-  fe->value = xstrdup(cwd);
-||||||| 674ec410
- cwd = get_proc_cwd(wp->fd);
- if (cwd != NULL)
-  fe->value = xstrdup(cwd);
-=======
- cwd = get_proc_cwd(wp->fd);
@@ -597 +587,0 @@ format_cb_current_path(struct format_tree *ft)
->>>>>>> f3931497
