diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_merged.h
index 9e3f719..42fdafd 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_merged.h
@@ -91,2 +91,2 @@ class DBImpl : public DB {
-  virtual const Options& GetOptions(
-      const ColumnFamilyHandle& column_family) const;
+  virtual const Options& GetOptions(const ColumnFamilyHandle& column_family)
+      const;
@@ -111 +111,3 @@ class DBImpl : public DB {
-  Status TEST_CompactRange(int level, const Slice* begin, const Slice* end);
+  Status TEST_CompactRange(int level,
+                           const Slice* begin,
+                           const Slice* end);
@@ -121,4 +123,3 @@ class DBImpl : public DB {
-  void TEST_SetDefaultTimeToCheck(
-      uint64_t default_interval_to_delete_obsolete_WAL) {
-    default_interval_to_delete_obsolete_WAL_ =
-        default_interval_to_delete_obsolete_WAL;
+  void TEST_SetDefaultTimeToCheck(uint64_t default_interval_to_delete_obsolete_WAL)
+  {
+    default_interval_to_delete_obsolete_WAL_ = default_interval_to_delete_obsolete_WAL;
@@ -128 +129,2 @@ class DBImpl : public DB {
-      return all_files.size() || sst_delete_files.size() ||
+      return all_files.size() ||
+        sst_delete_files.size() ||
@@ -156 +158,2 @@ class DBImpl : public DB {
-  void FindObsoleteFiles(DeletionState& deletion_state, bool force,
+  void FindObsoleteFiles(DeletionState& deletion_state,
+                         bool force,
@@ -169 +171,0 @@ class DBImpl : public DB {
-  SuperVersion* GetSuperVersion() { return super_version_; }
@@ -205,2 +207 @@ class DBImpl : public DB {
-  Status BackgroundCompaction(bool* madeProgress,
-                              DeletionState& deletion_state);
+  Status BackgroundCompaction(bool* madeProgress,DeletionState& deletion_state);
@@ -218 +219,2 @@ class DBImpl : public DB {
-                                VectorLogPtr& log_files, WalFileType type);
+                                VectorLogPtr& log_files,
+                                WalFileType type);
@@ -232 +234,2 @@ class DBImpl : public DB {
-      const ReadOptions& options, uint64_t* superversion_number);
+    const ReadOptions& options,
+    uint64_t* superversion_number);
@@ -286 +289,2 @@ class DBImpl : public DB {
-      SequenceNumber in, std::vector<SequenceNumber>& snapshots,
+    SequenceNumber in,
+    std::vector<SequenceNumber>& snapshots,
