diff --git a/./output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_conflict.h b/./output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_merged.h
index e0eee4b..42fdafd 100644
--- a/./output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_conflict.h
+++ b/./output/rocksdb/29bacb2e-5b3b6549-30a70065/db@db_impl.no_comments_merged.h
@@ -171 +170,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
@@ -173,9 +171,0 @@ class DBImpl : public DB {
-||||||| 30a700657
-  MemTable* GetMemTable() {
-    return mem_;
-  }
-=======
-  SuperVersion* GetSuperVersion() {
-    return super_version_;
-  }
->>>>>>> 5b3b6549
