diff --git a/./output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_mergegen.h b/./output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_merged.h
index 2636f9a..5d6b5c1 100644
--- a/./output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_mergegen.h
+++ b/./output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_merged.h
@@ -51 +51,5 @@ class TransactionLogIteratorImpl : public TransactionLogIterator {
- TransactionLogIteratorImpl( const std::string& dir, const DBOptions* options, const TransactionLogIterator::ReadOptions& read_options, const EnvOptions& soptions, const SequenceNumber seqNum, std::unique_ptr<VectorLogPtr> files, DBImpl const* const dbimpl);
+  TransactionLogIteratorImpl(
+      const std::string& dir, const DBOptions* options,
+      const TransactionLogIterator::ReadOptions& read_options,
+      const EnvOptions& soptions, const SequenceNumber seqNum,
+      std::unique_ptr<VectorLogPtr> files, DBImpl const* const dbimpl);
@@ -58 +62,2 @@ class TransactionLogIteratorImpl : public TransactionLogIterator {
-const DBOptions* options_; const TransactionLogIterator::ReadOptions read_options_;
+  const DBOptions* options_;
+  const TransactionLogIterator::ReadOptions read_options_;
