diff --git a/./output/frameworks_av/c986050f-be2bb331-c5e03bef/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_conflict.h b/./output/frameworks_av/c986050f-be2bb331-c5e03bef/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_merged.h
index 21a3933..0cec1c2 100644
--- a/./output/frameworks_av/c986050f-be2bb331-c5e03bef/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_conflict.h
+++ b/./output/frameworks_av/c986050f-be2bb331-c5e03bef/services@camera@libcameraservice@utils@SessionConfigurationUtils.no_comments_merged.h
@@ -115 +114,0 @@ void filterParameters(const CameraMetadata& src, const CameraMetadata& deviceInf
-<<<<<<< HEAD
@@ -119,6 +117,0 @@ status_t overrideDefaultRequestKeys(CameraMetadata *request);
-||||||| merged common ancestors
-<<<<<<<<< Temporary merge branch 1
-        metadata_vendor_id_t vendorTagId, CameraMetadata& dst);
-=======
-        metadata_vendor_id_t vendorTagId, CameraMetadata& dst);
->>>>>>> be2bb331
