--- a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@envoy@api@v2@route@route_components.upb.no_comments_conflict.h
+++ b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@envoy@api@v2@route@route_components.upb.no_comments_merged.h
@@ -211,13 +210,0 @@ UPB_INLINE const envoy_api_v2_route_CorsPolicy* envoy_api_v2_route_VirtualHost_c
-<<<<<<< HEAD
-UPB_INLINE const struct envoy_api_v2_core_HeaderValueOption* const* envoy_api_v2_route_VirtualHost_response_headers_to_add(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (const struct envoy_api_v2_core_HeaderValueOption* const*)_upb_array_accessor(msg, UPB_SIZE(56, 104), len); }
-UPB_INLINE upb_strview const* envoy_api_v2_route_VirtualHost_response_headers_to_remove(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (upb_strview const*)_upb_array_accessor(msg, UPB_SIZE(60, 112), len); }
-UPB_INLINE size_t envoy_api_v2_route_VirtualHost_per_filter_config_size(const envoy_api_v2_route_VirtualHost *msg) {return _upb_msg_map_size(msg, UPB_SIZE(64, 120)); }
-UPB_INLINE bool envoy_api_v2_route_VirtualHost_per_filter_config_get(const envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Struct* *val) { return _upb_msg_map_get(msg, UPB_SIZE(64, 120), &key, 0, val, sizeof(*val)); }
-UPB_INLINE const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* envoy_api_v2_route_VirtualHost_per_filter_config_next(const envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(64, 120), iter); }
-UPB_INLINE upb_strview const* envoy_api_v2_route_VirtualHost_request_headers_to_remove(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (upb_strview const*)_upb_array_accessor(msg, UPB_SIZE(68, 128), len); }
-||||||| 694f491e06
-UPB_INLINE const struct envoy_api_v2_core_HeaderValueOption* const* envoy_api_v2_route_VirtualHost_response_headers_to_add(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (const struct envoy_api_v2_core_HeaderValueOption* const*)_upb_array_accessor(msg, UPB_SIZE(56, 104), len); }
-UPB_INLINE upb_strview const* envoy_api_v2_route_VirtualHost_response_headers_to_remove(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (upb_strview const*)_upb_array_accessor(msg, UPB_SIZE(60, 112), len); }
-UPB_INLINE const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* const* envoy_api_v2_route_VirtualHost_per_filter_config(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* const*)_upb_array_accessor(msg, UPB_SIZE(64, 120), len); }
-UPB_INLINE upb_strview const* envoy_api_v2_route_VirtualHost_request_headers_to_remove(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (upb_strview const*)_upb_array_accessor(msg, UPB_SIZE(68, 128), len); }
-=======
@@ -226 +213,3 @@ UPB_INLINE upb_strview const* envoy_api_v2_route_VirtualHost_response_headers_to
-UPB_INLINE const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* const* envoy_api_v2_route_VirtualHost_per_filter_config(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* const*)_upb_array_accessor(msg, UPB_SIZE(68, 128), len); }
+UPB_INLINE size_t envoy_api_v2_route_VirtualHost_per_filter_config_size(const envoy_api_v2_route_VirtualHost *msg) {return _upb_msg_map_size(msg, UPB_SIZE(68, 128)); }
+UPB_INLINE bool envoy_api_v2_route_VirtualHost_per_filter_config_get(const envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Struct* *val) { return _upb_msg_map_get(msg, UPB_SIZE(68, 128), &key, 0, val, sizeof(*val)); }
+UPB_INLINE const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* envoy_api_v2_route_VirtualHost_per_filter_config_next(const envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (const envoy_api_v2_route_VirtualHost_PerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(68, 128), iter); }
@@ -228 +216,0 @@ UPB_INLINE upb_strview const* envoy_api_v2_route_VirtualHost_request_headers_to_
->>>>>>> 729af3a4
@@ -230,9 +218,3 @@ UPB_INLINE bool envoy_api_v2_route_VirtualHost_include_request_attempt_count(con
-<<<<<<< HEAD
-UPB_INLINE size_t envoy_api_v2_route_VirtualHost_typed_per_filter_config_size(const envoy_api_v2_route_VirtualHost *msg) {return _upb_msg_map_size(msg, UPB_SIZE(72, 136)); }
-UPB_INLINE bool envoy_api_v2_route_VirtualHost_typed_per_filter_config_get(const envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Any* *val) { return _upb_msg_map_get(msg, UPB_SIZE(72, 136), &key, 0, val, sizeof(*val)); }
-UPB_INLINE const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* envoy_api_v2_route_VirtualHost_typed_per_filter_config_next(const envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(72, 136), iter); }
-||||||| 694f491e06
-UPB_INLINE const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* const* envoy_api_v2_route_VirtualHost_typed_per_filter_config(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* const*)_upb_array_accessor(msg, UPB_SIZE(72, 136), len); }
-=======
-UPB_INLINE const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* const* envoy_api_v2_route_VirtualHost_typed_per_filter_config(const envoy_api_v2_route_VirtualHost *msg, size_t *len) { return (const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* const*)_upb_array_accessor(msg, UPB_SIZE(76, 144), len); }
->>>>>>> 729af3a4
+UPB_INLINE size_t envoy_api_v2_route_VirtualHost_typed_per_filter_config_size(const envoy_api_v2_route_VirtualHost *msg) {return _upb_msg_map_size(msg, UPB_SIZE(76, 144)); }
+UPB_INLINE bool envoy_api_v2_route_VirtualHost_typed_per_filter_config_get(const envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Any* *val) { return _upb_msg_map_get(msg, UPB_SIZE(76, 144), &key, 0, val, sizeof(*val)); }
+UPB_INLINE const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* envoy_api_v2_route_VirtualHost_typed_per_filter_config_next(const envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (const envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(76, 144), iter); }
@@ -251,7 +233 @@ UPB_INLINE upb_strview* envoy_api_v2_route_VirtualHost_resize_domains(envoy_api_
-<<<<<<< HEAD
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(36, 64), len, UPB_TYPE_STRING, arena);
-||||||| 694f491e06
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(36, 64), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
-=======
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(40, 72), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
->>>>>>> 729af3a4
+  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(40, 72), len, UPB_TYPE_STRING, arena);
@@ -260,2 +236 @@ UPB_INLINE bool envoy_api_v2_route_VirtualHost_add_domains(envoy_api_v2_route_Vi
-<<<<<<< HEAD
-  return _upb_array_append_accessor(msg, UPB_SIZE(36, 64), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
+  return _upb_array_append_accessor(msg, UPB_SIZE(40, 72), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
@@ -263,7 +237,0 @@ UPB_INLINE bool envoy_api_v2_route_VirtualHost_add_domains(envoy_api_v2_route_Vi
-||||||| 694f491e06
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(36, 64), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
-=======
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(40, 72), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
->>>>>>> 729af3a4
@@ -275,7 +243 @@ UPB_INLINE envoy_api_v2_route_Route** envoy_api_v2_route_VirtualHost_resize_rout
-<<<<<<< HEAD
-  return (envoy_api_v2_route_Route**)_upb_array_resize_accessor(msg, UPB_SIZE(40, 72), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (envoy_api_v2_route_Route**)_upb_array_resize_accessor(msg, UPB_SIZE(40, 72), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (envoy_api_v2_route_Route**)_upb_array_resize_accessor(msg, UPB_SIZE(44, 80), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (envoy_api_v2_route_Route**)_upb_array_resize_accessor(msg, UPB_SIZE(44, 80), len, UPB_TYPE_MESSAGE, arena);
@@ -297,7 +259 @@ UPB_INLINE envoy_api_v2_route_VirtualCluster** envoy_api_v2_route_VirtualHost_re
-<<<<<<< HEAD
-  return (envoy_api_v2_route_VirtualCluster**)_upb_array_resize_accessor(msg, UPB_SIZE(44, 80), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (envoy_api_v2_route_VirtualCluster**)_upb_array_resize_accessor(msg, UPB_SIZE(44, 80), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (envoy_api_v2_route_VirtualCluster**)_upb_array_resize_accessor(msg, UPB_SIZE(48, 88), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (envoy_api_v2_route_VirtualCluster**)_upb_array_resize_accessor(msg, UPB_SIZE(48, 88), len, UPB_TYPE_MESSAGE, arena);
@@ -308,30 +263,0 @@ UPB_INLINE struct envoy_api_v2_route_VirtualCluster* envoy_api_v2_route_VirtualH
-<<<<<<< HEAD
-      msg, UPB_SIZE(44, 80), UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, &sub, arena);
-  if (!ok) return NULL;
-  return sub;
-}
-UPB_INLINE envoy_api_v2_route_RateLimit** envoy_api_v2_route_VirtualHost_mutable_rate_limits(envoy_api_v2_route_VirtualHost *msg, size_t *len) {
-  return (envoy_api_v2_route_RateLimit**)_upb_array_mutable_accessor(msg, UPB_SIZE(48, 88), len);
-}
-UPB_INLINE envoy_api_v2_route_RateLimit** envoy_api_v2_route_VirtualHost_resize_rate_limits(envoy_api_v2_route_VirtualHost *msg, size_t len, upb_arena *arena) {
-  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(48, 88), len, UPB_TYPE_MESSAGE, arena);
-}
-UPB_INLINE struct envoy_api_v2_route_RateLimit* envoy_api_v2_route_VirtualHost_add_rate_limits(envoy_api_v2_route_VirtualHost *msg, upb_arena *arena) {
-  struct envoy_api_v2_route_RateLimit* sub = (struct envoy_api_v2_route_RateLimit*)_upb_msg_new(&envoy_api_v2_route_RateLimit_msginit, arena);
-  bool ok = _upb_array_append_accessor(
-||||||| 694f491e06
-      msg, UPB_SIZE(44, 80), UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, &sub, arena);
-  if (!ok) return NULL;
-  return sub;
-}
-UPB_INLINE envoy_api_v2_route_RateLimit** envoy_api_v2_route_VirtualHost_mutable_rate_limits(envoy_api_v2_route_VirtualHost *msg, size_t *len) {
-  return (envoy_api_v2_route_RateLimit**)_upb_array_mutable_accessor(msg, UPB_SIZE(48, 88), len);
-}
-UPB_INLINE envoy_api_v2_route_RateLimit** envoy_api_v2_route_VirtualHost_resize_rate_limits(envoy_api_v2_route_VirtualHost *msg, size_t len, upb_arena *arena) {
-  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(48, 88), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-}
-UPB_INLINE struct envoy_api_v2_route_RateLimit* envoy_api_v2_route_VirtualHost_add_rate_limits(envoy_api_v2_route_VirtualHost *msg, upb_arena *arena) {
-  struct envoy_api_v2_route_RateLimit* sub = (struct envoy_api_v2_route_RateLimit*)upb_msg_new(&envoy_api_v2_route_RateLimit_msginit, arena);
-  bool ok = _upb_array_append_accessor(
-=======
->>>>>>> 729af3a4
@@ -346 +272 @@ UPB_INLINE envoy_api_v2_route_RateLimit** envoy_api_v2_route_VirtualHost_resize_
-  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(52, 96), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
+  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(52, 96), len, UPB_TYPE_MESSAGE, arena);
@@ -349 +275 @@ UPB_INLINE struct envoy_api_v2_route_RateLimit* envoy_api_v2_route_VirtualHost_a
-  struct envoy_api_v2_route_RateLimit* sub = (struct envoy_api_v2_route_RateLimit*)upb_msg_new(&envoy_api_v2_route_RateLimit_msginit, arena);
+  struct envoy_api_v2_route_RateLimit* sub = (struct envoy_api_v2_route_RateLimit*)_upb_msg_new(&envoy_api_v2_route_RateLimit_msginit, arena);
@@ -359,7 +285 @@ UPB_INLINE struct envoy_api_v2_core_HeaderValueOption** envoy_api_v2_route_Virtu
-<<<<<<< HEAD
-  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(52, 96), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(52, 96), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(56, 104), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(56, 104), len, UPB_TYPE_MESSAGE, arena);
@@ -390,7 +310 @@ UPB_INLINE struct envoy_api_v2_core_HeaderValueOption** envoy_api_v2_route_Virtu
-<<<<<<< HEAD
-  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(56, 104), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(56, 104), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(60, 112), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (struct envoy_api_v2_core_HeaderValueOption**)_upb_array_resize_accessor(msg, UPB_SIZE(60, 112), len, UPB_TYPE_MESSAGE, arena);
@@ -409,7 +323 @@ UPB_INLINE upb_strview* envoy_api_v2_route_VirtualHost_resize_response_headers_t
-<<<<<<< HEAD
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(60, 112), len, UPB_TYPE_STRING, arena);
-||||||| 694f491e06
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(60, 112), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
-=======
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(64, 120), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
->>>>>>> 729af3a4
+  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(64, 120), len, UPB_TYPE_STRING, arena);
@@ -418,2 +326 @@ UPB_INLINE bool envoy_api_v2_route_VirtualHost_add_response_headers_to_remove(en
-<<<<<<< HEAD
-  return _upb_array_append_accessor(msg, UPB_SIZE(60, 112), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
+  return _upb_array_append_accessor(msg, UPB_SIZE(64, 120), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
@@ -421,9 +327,0 @@ UPB_INLINE bool envoy_api_v2_route_VirtualHost_add_response_headers_to_remove(en
-||||||| 694f491e06
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(60, 112), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
-}
-UPB_INLINE envoy_api_v2_route_VirtualHost_PerFilterConfigEntry** envoy_api_v2_route_VirtualHost_mutable_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t *len) {
-  return (envoy_api_v2_route_VirtualHost_PerFilterConfigEntry**)_upb_array_mutable_accessor(msg, UPB_SIZE(64, 120), len);
-}
-UPB_INLINE envoy_api_v2_route_VirtualHost_PerFilterConfigEntry** envoy_api_v2_route_VirtualHost_resize_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t len, upb_arena *arena) {
-  return (envoy_api_v2_route_VirtualHost_PerFilterConfigEntry**)_upb_array_resize_accessor(msg, UPB_SIZE(64, 120), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
@@ -431,28 +329,4 @@ UPB_INLINE envoy_api_v2_route_VirtualHost_PerFilterConfigEntry** envoy_api_v2_ro
-UPB_INLINE struct envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* envoy_api_v2_route_VirtualHost_add_per_filter_config(envoy_api_v2_route_VirtualHost *msg, upb_arena *arena) {
-  struct envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* sub = (struct envoy_api_v2_route_VirtualHost_PerFilterConfigEntry*)upb_msg_new(&envoy_api_v2_route_VirtualHost_PerFilterConfigEntry_msginit, arena);
-  bool ok = _upb_array_append_accessor(
-      msg, UPB_SIZE(64, 120), UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, &sub, arena);
-  if (!ok) return NULL;
-  return sub;
-=======
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(64, 120), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
-}
-UPB_INLINE envoy_api_v2_route_VirtualHost_PerFilterConfigEntry** envoy_api_v2_route_VirtualHost_mutable_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t *len) {
-  return (envoy_api_v2_route_VirtualHost_PerFilterConfigEntry**)_upb_array_mutable_accessor(msg, UPB_SIZE(68, 128), len);
-}
-UPB_INLINE envoy_api_v2_route_VirtualHost_PerFilterConfigEntry** envoy_api_v2_route_VirtualHost_resize_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t len, upb_arena *arena) {
-  return (envoy_api_v2_route_VirtualHost_PerFilterConfigEntry**)_upb_array_resize_accessor(msg, UPB_SIZE(68, 128), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-}
-UPB_INLINE struct envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* envoy_api_v2_route_VirtualHost_add_per_filter_config(envoy_api_v2_route_VirtualHost *msg, upb_arena *arena) {
-  struct envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* sub = (struct envoy_api_v2_route_VirtualHost_PerFilterConfigEntry*)upb_msg_new(&envoy_api_v2_route_VirtualHost_PerFilterConfigEntry_msginit, arena);
-  bool ok = _upb_array_append_accessor(
-      msg, UPB_SIZE(68, 128), UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, &sub, arena);
-  if (!ok) return NULL;
-  return sub;
->>>>>>> 729af3a4
-}
-UPB_INLINE void envoy_api_v2_route_VirtualHost_per_filter_config_clear(envoy_api_v2_route_VirtualHost *msg) { _upb_msg_map_clear(msg, UPB_SIZE(64, 120)); }
-UPB_INLINE bool envoy_api_v2_route_VirtualHost_per_filter_config_set(envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Struct* val, upb_arena *a) { return _upb_msg_map_set(msg, UPB_SIZE(64, 120), &key, 0, &val, sizeof(val), a); }
-UPB_INLINE bool envoy_api_v2_route_VirtualHost_per_filter_config_delete(envoy_api_v2_route_VirtualHost *msg, upb_strview key) { return _upb_msg_map_delete(msg, UPB_SIZE(64, 120), &key, 0); }
-UPB_INLINE envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* envoy_api_v2_route_VirtualHost_per_filter_config_nextmutable(envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (envoy_api_v2_route_VirtualHost_PerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(64, 120), iter); }
+UPB_INLINE void envoy_api_v2_route_VirtualHost_per_filter_config_clear(envoy_api_v2_route_VirtualHost *msg) { _upb_msg_map_clear(msg, UPB_SIZE(68, 128)); }
+UPB_INLINE bool envoy_api_v2_route_VirtualHost_per_filter_config_set(envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Struct* val, upb_arena *a) { return _upb_msg_map_set(msg, UPB_SIZE(68, 128), &key, 0, &val, sizeof(val), a); }
+UPB_INLINE bool envoy_api_v2_route_VirtualHost_per_filter_config_delete(envoy_api_v2_route_VirtualHost *msg, upb_strview key) { return _upb_msg_map_delete(msg, UPB_SIZE(68, 128), &key, 0); }
+UPB_INLINE envoy_api_v2_route_VirtualHost_PerFilterConfigEntry* envoy_api_v2_route_VirtualHost_per_filter_config_nextmutable(envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (envoy_api_v2_route_VirtualHost_PerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(68, 128), iter); }
@@ -463,7 +337 @@ UPB_INLINE upb_strview* envoy_api_v2_route_VirtualHost_resize_request_headers_to
-<<<<<<< HEAD
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(68, 128), len, UPB_TYPE_STRING, arena);
-||||||| 694f491e06
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(68, 128), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
-=======
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(72, 136), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
->>>>>>> 729af3a4
+  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(72, 136), len, UPB_TYPE_STRING, arena);
@@ -472,2 +340 @@ UPB_INLINE bool envoy_api_v2_route_VirtualHost_add_request_headers_to_remove(env
-<<<<<<< HEAD
-  return _upb_array_append_accessor(msg, UPB_SIZE(68, 128), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
+  return _upb_array_append_accessor(msg, UPB_SIZE(72, 136), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
@@ -475,7 +341,0 @@ UPB_INLINE bool envoy_api_v2_route_VirtualHost_add_request_headers_to_remove(env
-||||||| 694f491e06
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(68, 128), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
-=======
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(72, 136), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
->>>>>>> 729af3a4
@@ -486,34 +346,4 @@ UPB_INLINE void envoy_api_v2_route_VirtualHost_set_include_request_attempt_count
-<<<<<<< HEAD
-UPB_INLINE void envoy_api_v2_route_VirtualHost_typed_per_filter_config_clear(envoy_api_v2_route_VirtualHost *msg) { _upb_msg_map_clear(msg, UPB_SIZE(72, 136)); }
-UPB_INLINE bool envoy_api_v2_route_VirtualHost_typed_per_filter_config_set(envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Any* val, upb_arena *a) { return _upb_msg_map_set(msg, UPB_SIZE(72, 136), &key, 0, &val, sizeof(val), a); }
-UPB_INLINE bool envoy_api_v2_route_VirtualHost_typed_per_filter_config_delete(envoy_api_v2_route_VirtualHost *msg, upb_strview key) { return _upb_msg_map_delete(msg, UPB_SIZE(72, 136), &key, 0); }
-UPB_INLINE envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* envoy_api_v2_route_VirtualHost_typed_per_filter_config_nextmutable(envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(72, 136), iter); }
-||||||| 694f491e06
-UPB_INLINE envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry** envoy_api_v2_route_VirtualHost_mutable_typed_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t *len) {
-  return (envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry**)_upb_array_mutable_accessor(msg, UPB_SIZE(72, 136), len);
-}
-UPB_INLINE envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry** envoy_api_v2_route_VirtualHost_resize_typed_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t len, upb_arena *arena) {
-  return (envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry**)_upb_array_resize_accessor(msg, UPB_SIZE(72, 136), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-}
-UPB_INLINE struct envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* envoy_api_v2_route_VirtualHost_add_typed_per_filter_config(envoy_api_v2_route_VirtualHost *msg, upb_arena *arena) {
-  struct envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* sub = (struct envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry*)upb_msg_new(&envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry_msginit, arena);
-  bool ok = _upb_array_append_accessor(
-      msg, UPB_SIZE(72, 136), UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, &sub, arena);
-  if (!ok) return NULL;
-  return sub;
-}
-=======
-UPB_INLINE envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry** envoy_api_v2_route_VirtualHost_mutable_typed_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t *len) {
-  return (envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry**)_upb_array_mutable_accessor(msg, UPB_SIZE(76, 144), len);
-}
-UPB_INLINE envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry** envoy_api_v2_route_VirtualHost_resize_typed_per_filter_config(envoy_api_v2_route_VirtualHost *msg, size_t len, upb_arena *arena) {
-  return (envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry**)_upb_array_resize_accessor(msg, UPB_SIZE(76, 144), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-}
-UPB_INLINE struct envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* envoy_api_v2_route_VirtualHost_add_typed_per_filter_config(envoy_api_v2_route_VirtualHost *msg, upb_arena *arena) {
-  struct envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* sub = (struct envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry*)upb_msg_new(&envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry_msginit, arena);
-  bool ok = _upb_array_append_accessor(
-      msg, UPB_SIZE(76, 144), UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, &sub, arena);
-  if (!ok) return NULL;
-  return sub;
-}
->>>>>>> 729af3a4
+UPB_INLINE void envoy_api_v2_route_VirtualHost_typed_per_filter_config_clear(envoy_api_v2_route_VirtualHost *msg) { _upb_msg_map_clear(msg, UPB_SIZE(76, 144)); }
+UPB_INLINE bool envoy_api_v2_route_VirtualHost_typed_per_filter_config_set(envoy_api_v2_route_VirtualHost *msg, upb_strview key, struct google_protobuf_Any* val, upb_arena *a) { return _upb_msg_map_set(msg, UPB_SIZE(76, 144), &key, 0, &val, sizeof(val), a); }
+UPB_INLINE bool envoy_api_v2_route_VirtualHost_typed_per_filter_config_delete(envoy_api_v2_route_VirtualHost *msg, upb_strview key) { return _upb_msg_map_delete(msg, UPB_SIZE(76, 144), &key, 0); }
+UPB_INLINE envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry* envoy_api_v2_route_VirtualHost_typed_per_filter_config_nextmutable(envoy_api_v2_route_VirtualHost *msg, size_t* iter) { return (envoy_api_v2_route_VirtualHost_TypedPerFilterConfigEntry*)_upb_msg_map_next(msg, UPB_SIZE(76, 144), iter); }
@@ -565 +395 @@ UPB_INLINE struct google_protobuf_Any* envoy_api_v2_route_VirtualHost_mutable_re
-    sub = (struct google_protobuf_Any*)upb_msg_new(&google_protobuf_Any_msginit, arena);
+    sub = (struct google_protobuf_Any*)_upb_msg_new(&google_protobuf_Any_msginit, arena);
@@ -1476,7 +1306 @@ UPB_INLINE envoy_api_v2_route_RateLimit** envoy_api_v2_route_RouteAction_resize_
-<<<<<<< HEAD
-  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(76, 128), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(76, 128), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(84, 144), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (envoy_api_v2_route_RateLimit**)_upb_array_resize_accessor(msg, UPB_SIZE(84, 144), len, UPB_TYPE_MESSAGE, arena);
@@ -1507,7 +1331 @@ UPB_INLINE envoy_api_v2_route_RouteAction_HashPolicy** envoy_api_v2_route_RouteA
-<<<<<<< HEAD
-  return (envoy_api_v2_route_RouteAction_HashPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(80, 136), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (envoy_api_v2_route_RouteAction_HashPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(80, 136), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (envoy_api_v2_route_RouteAction_HashPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(88, 152), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (envoy_api_v2_route_RouteAction_HashPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(88, 152), len, UPB_TYPE_MESSAGE, arena);
@@ -1565,7 +1383 @@ UPB_INLINE envoy_api_v2_route_RouteAction_UpgradeConfig** envoy_api_v2_route_Rou
-<<<<<<< HEAD
-  return (envoy_api_v2_route_RouteAction_UpgradeConfig**)_upb_array_resize_accessor(msg, UPB_SIZE(84, 144), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (envoy_api_v2_route_RouteAction_UpgradeConfig**)_upb_array_resize_accessor(msg, UPB_SIZE(84, 144), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (envoy_api_v2_route_RouteAction_UpgradeConfig**)_upb_array_resize_accessor(msg, UPB_SIZE(92, 160), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (envoy_api_v2_route_RouteAction_UpgradeConfig**)_upb_array_resize_accessor(msg, UPB_SIZE(92, 160), len, UPB_TYPE_MESSAGE, arena);
@@ -1614,7 +1426 @@ UPB_INLINE envoy_api_v2_route_RouteAction_RequestMirrorPolicy** envoy_api_v2_rou
-<<<<<<< HEAD
-  return (envoy_api_v2_route_RouteAction_RequestMirrorPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(88, 152), len, UPB_TYPE_MESSAGE, arena);
-||||||| 694f491e06
-  return (envoy_api_v2_route_RouteAction_RequestMirrorPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(88, 152), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
-=======
-  return (envoy_api_v2_route_RouteAction_RequestMirrorPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(96, 168), len, UPB_SIZE(4, 8), UPB_TYPE_MESSAGE, arena);
->>>>>>> 729af3a4
+  return (envoy_api_v2_route_RouteAction_RequestMirrorPolicy**)_upb_array_resize_accessor(msg, UPB_SIZE(96, 168), len, UPB_TYPE_MESSAGE, arena);
@@ -1647 +1453 @@ UPB_INLINE struct envoy_type_matcher_RegexMatchAndSubstitute* envoy_api_v2_route
-    sub = (struct envoy_type_matcher_RegexMatchAndSubstitute*)upb_msg_new(&envoy_type_matcher_RegexMatchAndSubstitute_msginit, arena);
+    sub = (struct envoy_type_matcher_RegexMatchAndSubstitute*)_upb_msg_new(&envoy_type_matcher_RegexMatchAndSubstitute_msginit, arena);
@@ -1659 +1465 @@ UPB_INLINE struct google_protobuf_Any* envoy_api_v2_route_RouteAction_mutable_re
-    sub = (struct google_protobuf_Any*)upb_msg_new(&google_protobuf_Any_msginit, arena);
+    sub = (struct google_protobuf_Any*)_upb_msg_new(&google_protobuf_Any_msginit, arena);
@@ -1704 +1510 @@ UPB_INLINE struct google_protobuf_BoolValue* envoy_api_v2_route_RouteAction_Requ
-    sub = (struct google_protobuf_BoolValue*)upb_msg_new(&google_protobuf_BoolValue_msginit, arena);
+    sub = (struct google_protobuf_BoolValue*)_upb_msg_new(&google_protobuf_BoolValue_msginit, arena);
@@ -1798 +1604 @@ UPB_INLINE struct envoy_api_v2_route_RouteAction_HashPolicy_FilterState* envoy_a
-    sub = (struct envoy_api_v2_route_RouteAction_HashPolicy_FilterState*)upb_msg_new(&envoy_api_v2_route_RouteAction_HashPolicy_FilterState_msginit, arena);
+    sub = (struct envoy_api_v2_route_RouteAction_HashPolicy_FilterState*)_upb_msg_new(&envoy_api_v2_route_RouteAction_HashPolicy_FilterState_msginit, arena);
@@ -1882 +1688 @@ UPB_INLINE envoy_api_v2_route_RouteAction_HashPolicy_FilterState *envoy_api_v2_r
-  return (envoy_api_v2_route_RouteAction_HashPolicy_FilterState *)upb_msg_new(&envoy_api_v2_route_RouteAction_HashPolicy_FilterState_msginit, arena);
+  return (envoy_api_v2_route_RouteAction_HashPolicy_FilterState *)_upb_msg_new(&envoy_api_v2_route_RouteAction_HashPolicy_FilterState_msginit, arena);
