diff --git a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@envoy@config@filter@network@http_connection_manager@v2@http_connection_manager.upb.no_comments_mergegen.h b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@envoy@config@filter@network@http_connection_manager@v2@http_connection_manager.upb.no_comments_merged.h
index c568e9e..2036c82 100644
--- a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@envoy@config@filter@network@http_connection_manager@v2@http_connection_manager.upb.no_comments_mergegen.h
+++ b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@envoy@config@filter@network@http_connection_manager@v2@http_connection_manager.upb.no_comments_merged.h
@@ -472 +472 @@ UPB_INLINE struct envoy_config_filter_network_http_connection_manager_v2_Request
-    sub = (struct envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension*)upb_msg_new(&envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension_msginit, arena);
+    sub = (struct envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension*)_upb_msg_new(&envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension_msginit, arena);
@@ -508 +508,2 @@ UPB_INLINE bool envoy_config_filter_network_http_connection_manager_v2_HttpConne
-return _upb_array_append_accessor(msg, UPB_SIZE(32, 56), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
+  return _upb_array_append_accessor(msg, UPB_SIZE(32, 56), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
+      arena);
@@ -580 +581 @@ UPB_INLINE struct envoy_config_trace_v2_Tracing_Http* envoy_config_filter_networ
-    sub = (struct envoy_config_trace_v2_Tracing_Http*)upb_msg_new(&envoy_config_trace_v2_Tracing_Http_msginit, arena);
+    sub = (struct envoy_config_trace_v2_Tracing_Http*)_upb_msg_new(&envoy_config_trace_v2_Tracing_Http_msginit, arena);
@@ -1003 +1004 @@ UPB_INLINE envoy_config_filter_network_http_connection_manager_v2_RequestIDExten
-  return (envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension *)upb_msg_new(&envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension_msginit, arena);
+  return (envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension *)_upb_msg_new(&envoy_config_filter_network_http_connection_manager_v2_RequestIDExtension_msginit, arena);
@@ -1020 +1021 @@ UPB_INLINE struct google_protobuf_Any* envoy_config_filter_network_http_connecti
-    sub = (struct google_protobuf_Any*)upb_msg_new(&google_protobuf_Any_msginit, arena);
+    sub = (struct google_protobuf_Any*)_upb_msg_new(&google_protobuf_Any_msginit, arena);
