--- a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_mergegen.cc
+++ b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@filters@client_channel@backend_metric.no_comments_merged.cc
@@ -15,2 +15,7 @@ std::map<absl::string_view, double, StringLess> ParseMap(
-while (true) { const auto* entry = entry_func(msg, &i); if (entry == nullptr) break; upb_strview key_view = key_func(entry);
-    char* key = static_cast<char*>(arena->Alloc(key_view.size + 1));
+  std::map<absl::string_view, double, StringLess> result;
+  size_t i = UPB_MAP_BEGIN;
+  while (true) {
+    const auto* entry = entry_func(msg, &i);
+    if (entry == nullptr) break;
+    upb_strview key_view = key_func(entry);
+    char* key = static_cast<char*>(arena->Alloc(key_view.size));
@@ -18 +23 @@ while (true) { const auto* entry = entry_func(msg, &i); if (entry == nullptr) br
-char*>(arena->Alloc(key_view.size)); memcpy(key, key_view.data, key_view.size); result[absl::string_view(key, key_view.size)] = value_func(entry);
+    result[absl::string_view(key, key_view.size)] = value_func(entry);
