diff --git a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@validate@validate.upb.no_comments_conflict.h b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@validate@validate.upb.no_comments_merged.h
index 3f3e097..9e5b206 100644
--- a/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@validate@validate.upb.no_comments_conflict.h
+++ b/./output/grpc/d7311d7d-729af3a4-694f491e/src@core@ext@upb-generated@validate@validate.upb.no_comments_merged.h
@@ -1319,7 +1319 @@ UPB_INLINE upb_strview* validate_StringRules_resize_in(validate_StringRules *msg
-<<<<<<< HEAD
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(104, 152), len, UPB_TYPE_STRING, arena);
-||||||| 694f491e06
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(104, 152), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
-=======
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(108, 160), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
->>>>>>> 729af3a4
+  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(108, 160), len, UPB_TYPE_STRING, arena);
@@ -1328,2 +1322 @@ UPB_INLINE bool validate_StringRules_add_in(validate_StringRules *msg, upb_strvi
-<<<<<<< HEAD
-  return _upb_array_append_accessor(msg, UPB_SIZE(104, 152), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
+  return _upb_array_append_accessor(msg, UPB_SIZE(108, 160), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
@@ -1331,7 +1323,0 @@ UPB_INLINE bool validate_StringRules_add_in(validate_StringRules *msg, upb_strvi
-||||||| 694f491e06
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(104, 152), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
-=======
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(108, 160), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
->>>>>>> 729af3a4
@@ -1343,7 +1329 @@ UPB_INLINE upb_strview* validate_StringRules_resize_not_in(validate_StringRules
-<<<<<<< HEAD
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(108, 160), len, UPB_TYPE_STRING, arena);
-||||||| 694f491e06
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(108, 160), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
-=======
-  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(112, 168), len, UPB_SIZE(8, 16), UPB_TYPE_STRING, arena);
->>>>>>> 729af3a4
+  return (upb_strview*)_upb_array_resize_accessor(msg, UPB_SIZE(112, 168), len, UPB_TYPE_STRING, arena);
@@ -1352,2 +1332 @@ UPB_INLINE bool validate_StringRules_add_not_in(validate_StringRules *msg, upb_s
-<<<<<<< HEAD
-  return _upb_array_append_accessor(msg, UPB_SIZE(108, 160), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
+  return _upb_array_append_accessor(msg, UPB_SIZE(112, 168), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val,
@@ -1355,7 +1333,0 @@ UPB_INLINE bool validate_StringRules_add_not_in(validate_StringRules *msg, upb_s
-||||||| 694f491e06
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(108, 160), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
-=======
-  return _upb_array_append_accessor(
-      msg, UPB_SIZE(112, 168), UPB_SIZE(8, 16), UPB_TYPE_STRING, &val, arena);
->>>>>>> 729af3a4
