diff --git a/home/whalien/codebase/python/mergebot-eva/output/flameshot/93b03ee8-91ba28c3-cb597578/src@config@geneneralconf.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/flameshot/93b03ee8-91ba28c3-cb597578/src@config@geneneralconf.no_comments_merged.h
index 4096de7..2b046c3 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/flameshot/93b03ee8-91ba28c3-cb597578/src@config@geneneralconf.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/flameshot/93b03ee8-91ba28c3-cb597578/src@config@geneneralconf.no_comments_merged.h
@@ -3 +3 @@
-class QVBoxLayout
+class QVBoxLayout;
@@ -12 +12 @@ class GeneneralConf : public QWidget {
- public:
+public slots:
@@ -14 +14 @@ class GeneneralConf : public QWidget {
- private:
+private slots:
@@ -24,0 +25 @@ class GeneneralConf : public QWidget {
+private:
@@ -44 +44,0 @@ class GeneneralConf : public QWidget {
-  void initSaveAfterCopy();
@@ -45,0 +46 @@ class GeneneralConf : public QWidget {
+    void initSaveAfterCopy();
