--- a/home/whalien/codebase/python/mergebot-eva/output/flameshot/93b03ee8-91ba28c3-cb597578/src@utils@confighandler.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/flameshot/93b03ee8-91ba28c3-cb597578/src@utils@confighandler.no_comments_merged.h
@@ -38,0 +39,2 @@ class ConfigHandler {
+    bool copyAndCloseAfterUploadEnabled();
+    void setCopyAndCloseAfterUploadEnabled(const bool);
@@ -43,2 +44,0 @@ class ConfigHandler {
-  bool copyAndCloseAfterUploadEnabled();
-  void setCopyAndCloseAfterUploadEnabled(const bool);
