diff --git a/./output/flameshot/93b03ee8-91ba28c3-cb597578/src@utils@confighandler.no_comments_mergegen.h b/./output/flameshot/93b03ee8-91ba28c3-cb597578/src@utils@confighandler.no_comments_merged.h
index 655cc76..e7379d5 100644
--- a/./output/flameshot/93b03ee8-91ba28c3-cb597578/src@utils@confighandler.no_comments_mergegen.h
+++ b/./output/flameshot/93b03ee8-91ba28c3-cb597578/src@utils@confighandler.no_comments_merged.h
@@ -39 +39,6 @@ public:
-bool copyAndCloseAfterUploadEnabled(); void setCopyAndCloseAfterUploadEnabled(const bool); bool saveAfterCopyValue(); void setSaveAfterCopy(const bool); QString saveAfterCopyPathValue(); void setSaveAfterCopyPath(const QString &);
+    bool copyAndCloseAfterUploadEnabled();
+    void setCopyAndCloseAfterUploadEnabled(const bool);
+    bool saveAfterCopyValue();
+    void setSaveAfterCopy(const bool);
+    QString saveAfterCopyPathValue();
+    void setSaveAfterCopyPath(const QString &);
