--- a/./output/frameworks_av/44f156ab-5f074002-03d96f08/services@audioflinger@Effects.no_comments_conflict.cpp
+++ b/./output/frameworks_av/44f156ab-5f074002-03d96f08/services@audioflinger@Effects.no_comments_merged.cpp
@@ -1964 +1963,0 @@ status_t EffectChain::addEffect_l(const sp<IAfEffectModule>& effect)
-<<<<<<< HEAD
@@ -1966,5 +1964,0 @@ status_t EffectChain::addEffect_l(const sp<IAfEffectModule>& effect)
-||||||| 03d96f08b9
-        ssize_t idx_insert = getInsertIndex_ll(desc);
-=======
-        ssize_t idx_insert = getInsertIndex_l(desc);
->>>>>>> 5f074002
