--- a/./output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_mergegen.c
+++ b/./output/tmux/98a18d06-31e3f2d5-6bde1c18/spawn.no_comments_merged.c
@@ -296 +296,9 @@ complete:
-#ifdef HAVE_UTEMPTER if (~new_wp->flags & PANE_EMPTY) { xasprintf(&cp, "tmux(%lu).%%%u", (long)getpid(), new_wp->id); utempter_add_record(new_wp->fd, cp); kill(getpid(), SIGCHLD); free(cp); }
+#ifdef HAVE_UTEMPTER
+ if (~new_wp->flags & PANE_EMPTY) {
+  xasprintf(&cp, "tmux(%lu).%%%u", (long)getpid(), new_wp->id);
+  utempter_add_record(new_wp->fd, cp);
+  kill(getpid(), SIGCHLD);
+  free(cp);
+ }
+#endif
+ new_wp->pipe_off = 0;
