--- a/./output/redis/83e87bac-fddc4757-4b2579a0/src@server.no_comments_mergegen.h
+++ b/./output/redis/83e87bac-fddc4757-4b2579a0/src@server.no_comments_merged.h
@@ -1078 +1078,4 @@ void moduleCallCommandFilters(client *c);
-void ModuleForkDoneHandler(int exitcode, int bysignal); void TerminateModuleForkChild(int wait);
+void ModuleForkDoneHandler(int exitcode, int bysignal);
+void TerminateModuleForkChild(int wait);
+ssize_t rdbSaveModulesAux(rio *rdb, int when);
+int moduleAllDatatypesHandleErrors();
