diff --git a/./output/redis/61853ad8-a9628142-bc5cb168/src@server.no_comments_mergegen.h b/./output/redis/61853ad8-a9628142-bc5cb168/src@server.no_comments_merged.h
index aa7b227..dd42b78 100644
--- a/./output/redis/61853ad8-a9628142-bc5cb168/src@server.no_comments_mergegen.h
+++ b/./output/redis/61853ad8-a9628142-bc5cb168/src@server.no_comments_merged.h
@@ -1079 +1079,5 @@ void moduleCallCommandFilters(client *c);
-void ModuleForkDoneHandler(int exitcode, int bysignal); int TerminateModuleForkChild(int child_pid, int wait); ssize_t rdbSaveModulesAux(rio *rdb, int when); int moduleAllDatatypesHandleErrors();sds modulesCollectInfo(sds info, sds section, int for_crash_report, int sections);
+void ModuleForkDoneHandler(int exitcode, int bysignal);
+int TerminateModuleForkChild(int child_pid, int wait);
+ssize_t rdbSaveModulesAux(rio *rdb, int when);
+int moduleAllDatatypesHandleErrors();
+sds modulesCollectInfo(sds info, sds section, int for_crash_report, int sections);
