--- a/./output/redis/61853ad8-a9628142-bc5cb168/src@module.no_comments_mergegen.c
+++ b/./output/redis/61853ad8-a9628142-bc5cb168/src@module.no_comments_merged.c
@@ -27 +27,2 @@ struct RedisModule {
-int options; int in_call; int options;
+    int options;
+    RedisModuleInfoFunc info_cb;
@@ -3244 +3245,12 @@ void moduleRegisterCoreAPI(void) {
-REGISTER_API(RegisterInfoFunc); REGISTER_API(InfoAddSection); REGISTER_API(InfoBeginDictField); REGISTER_API(InfoEndDictField); REGISTER_API(InfoAddFieldString); REGISTER_API(InfoAddFieldCString); REGISTER_API(InfoAddFieldDouble); REGISTER_API(InfoAddFieldLongLong); REGISTER_API(InfoAddFieldULongLong); REGISTER_API(Fork); REGISTER_API(ExitFromChild); REGISTER_API(KillForkChild);
+    REGISTER_API(Fork);
+    REGISTER_API(ExitFromChild);
+    REGISTER_API(KillForkChild);
+    REGISTER_API(RegisterInfoFunc);
+    REGISTER_API(InfoAddSection);
+    REGISTER_API(InfoBeginDictField);
+    REGISTER_API(InfoEndDictField);
+    REGISTER_API(InfoAddFieldString);
+    REGISTER_API(InfoAddFieldCString);
+    REGISTER_API(InfoAddFieldDouble);
+    REGISTER_API(InfoAddFieldLongLong);
+    REGISTER_API(InfoAddFieldULongLong);
