diff --git a/./output/redis/61853ad8-a9628142-bc5cb168/src@redismodule.no_comments_mergegen.h b/./output/redis/61853ad8-a9628142-bc5cb168/src@redismodule.no_comments_merged.h
index 5d0529d..86c1d30 100644
--- a/./output/redis/61853ad8-a9628142-bc5cb168/src@redismodule.no_comments_mergegen.h
+++ b/./output/redis/61853ad8-a9628142-bc5cb168/src@redismodule.no_comments_merged.h
@@ -113 +113,2 @@ typedef void (*RedisModuleCommandFilterFunc) (RedisModuleCommandFilterCtx *filte
-typedef void (*RedisModuleInfoFunc)(RedisModuleCommandFilterCtx *filter);typedef void (*RedisModuleForkDoneHandler) (int exitcode, int bysignal, void *user_data);
+typedef void (*RedisModuleForkDoneHandler) (int exitcode, int bysignal, void *user_data);
+typedef void (*RedisModuleInfoFunc)(RedisModuleInfoCtx *ctx, int for_crash_report);
