diff --git a/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@parallel@pipeline.no_comments_mergegen.hpp b/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@parallel@pipeline.no_comments_merged.hpp
index 31911ae..2b6e11d 100644
--- a/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@parallel@pipeline.no_comments_mergegen.hpp
+++ b/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@parallel@pipeline.no_comments_merged.hpp
@@ -3 +3,2 @@
-#include "duckdb/execution/physical_operator.hpp"
+#include "duckdb/common/unordered_set.hpp"
+#include "duckdb/common/set.hpp"
