--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@parallel@pipeline.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@parallel@pipeline.no_comments_merged.hpp
@@ -15,0 +16 @@ public:
+public:
@@ -16,0 +18 @@ public:
+public:
@@ -35,0 +38 @@ public:
+public:
@@ -71,0 +75 @@ private:
+private:
