diff --git a/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_mergegen.hpp b/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_merged.hpp
index 7b079f9..ebfdb80 100644
--- a/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_mergegen.hpp
+++ b/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_merged.hpp
@@ -9 +9,3 @@
-#include "duckdb/common/optional_idx.hpp" #include "duckdb/execution/physical_operator_states.hpp" #include "duckdb/common/enums/order_preservation_type.hpp"
+#include "duckdb/common/optional_idx.hpp"
+#include "duckdb/execution/physical_operator_states.hpp"
+#include "duckdb/common/enums/order_preservation_type.hpp"
@@ -17 +18,0 @@ class MetaPipeline;
-struct SourcePartitionInfo {
