diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_mergebot.hpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_merged.hpp
index 09332e7..ebfdb80 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@include@duckdb@execution@physical_operator.no_comments_merged.hpp
@@ -19,4 +18,0 @@ class MetaPipeline;
-struct SourcePartitionInfo {
- optional_idx batch_index;
- optional_idx min_batch_index;
-};
@@ -38,0 +35 @@ public:
+public:
@@ -52,0 +50 @@ public:
+public:
@@ -67,0 +66 @@ public:
+public:
@@ -87,0 +87 @@ public:
+public:
@@ -107,0 +108 @@ public:
+public:
@@ -110,0 +112 @@ public:
+public:
@@ -128,4 +130,2 @@ public:
- () = delete;
- {}
- () = delete;
- {}
+ ~CachingOperatorState() override {
+ }
@@ -142,0 +143 @@ public:
+public:
