diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@helper@physical_batch_collector.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@helper@physical_batch_collector.no_comments_merged.cpp
index 3800cb5..c0d4106 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@helper@physical_batch_collector.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@helper@physical_batch_collector.no_comments_merged.cpp
@@ -24,7 +23,0 @@ SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, GlobalSin
-<<<<<<< HEAD
- auto &state = (BatchCollectorLocalState &)lstate_p;
- state.data.Append(input, state.partition_info.batch_index.GetIndex());
-|||||||
- auto &state = (BatchCollectorLocalState &)lstate_p;
- state.data.Append(input, state.batch_index);
-=======
@@ -32,2 +25 @@ SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, GlobalSin
- state.data.Append(input, state.batch_index);
->>>>>>> c5737e4a948a72b20c39cee054a65a524542d832
+ state.data.Append(input, state.partition_info.batch_index.GetIndex());
