--- a/./output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_conflict.cc
+++ b/./output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_merged.cc
@@ -179,5 +178,0 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
-<<<<<<< HEAD
-  ChannelArguments args;
-||||||| 37d5d93983
-  grpc::ChannelArguments args;
-=======
@@ -191 +185,0 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
->>>>>>> b4031fa4
