diff --git a/./output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_mergegen.cc b/./output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_merged.cc
index e2658b8..a4aeee8 100644
--- a/./output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_mergegen.cc
+++ b/./output/grpc/88a180a5-b4031fa4-37d5d939/src@cpp@server@server_builder.no_comments_merged.cc
@@ -179 +179,7 @@ std::unique_ptr<grpc::Server> ServerBuilder::BuildAndStart() {
-if (max_receive_message_size_ >= -1) { args.SetInt(GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH, max_receive_message_size_); } if (max_send_message_size_ >= -1) { args.SetInt(GRPC_ARG_MAX_SEND_MESSAGE_LENGTH, max_send_message_size_); }
+  grpc::ChannelArguments args;
+  if (max_receive_message_size_ >= -1) {
+    args.SetInt(GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH, max_receive_message_size_);
+  }
+  if (max_send_message_size_ >= -1) {
+    args.SetInt(GRPC_ARG_MAX_SEND_MESSAGE_LENGTH, max_send_message_size_);
+  }
