--- a/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@compaction_picker.no_comments_mergegen.h
+++ b/./output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@compaction_picker.no_comments_merged.h
@@ -54 +54,5 @@ class UniversalCompactionPicker : public CompactionPicker {
- percentalCompactionPicker(const ColumnFamilyOptions* options, const InternalKeyComparator* icmp, Logger* logger) : CompactionPicker(options, icmp, logger) {} virtual Compaction* PickCompaction(Version* version, LogBuffer* log_buffer) override;
+  UniversalCompactionPicker(const ColumnFamilyOptions* options,
+                            const InternalKeyComparator* icmp, Logger* logger)
+      : CompactionPicker(options, icmp, logger) {}
+  virtual Compaction* PickCompaction(Version* version,
+                                     LogBuffer* log_buffer) override;
@@ -65 +69,5 @@ class LevelCompactionPicker : public CompactionPicker {
-streamsactionPicker(const ColumnFamilyOptions* options, const InternalKeyComparator* icmp, Logger* logger) : CompactionPicker(options, icmp, logger) {} virtual Compaction* PickCompaction(Version* version, LogBuffer* log_buffer) override;
+  LevelCompactionPicker(const ColumnFamilyOptions* options,
+                        const InternalKeyComparator* icmp, Logger* logger)
+      : CompactionPicker(options, icmp, logger) {}
+  virtual Compaction* PickCompaction(Version* version,
+                                     LogBuffer* log_buffer) override;
