diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_merged.h
index bc4f9a6..4da7523 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/db@version_set.no_comments_merged.h
@@ -18,3 +18 @@ namespace rocksdb {
-namespace log {
-class Writer;
-}
+namespace log { class Writer; }
@@ -32,2 +30,4 @@ extern int FindFile(const InternalKeyComparator& icmp,
-                    const std::vector<FileMetaData*>& files, const Slice& key);
-extern bool SomeFileOverlapsRange(const InternalKeyComparator& icmp,
+                    const std::vector<FileMetaData*>& files,
+                    const Slice& key);
+extern bool SomeFileOverlapsRange(
+    const InternalKeyComparator& icmp,
@@ -47,2 +47,3 @@ class Version {
-           Status* status, MergeContext* merge_context, GetStats* stats,
-           const Options& db_option, bool* value_found = nullptr);
+           Status* status, MergeContext* merge_context,
+           GetStats* stats, const Options& db_option,
+           bool* value_found = nullptr);
@@ -75,3 +76,3 @@ class Version {
-      unsigned int
-          index);
-  bool OverlapInLevel(int level, const Slice* smallest_user_key,
+      unsigned int index);
+  bool OverlapInLevel(int level,
+                      const Slice* smallest_user_key,
@@ -148,4 +149,8 @@ class VersionSet {
-  Status LogAndApply(
-      ColumnFamilyData* column_family_data, VersionEdit* edit, port::Mutex* mu,
-      Directory* db_directory = nullptr, bool new_descriptor_log = false,
-      const ColumnFamilyOptions* column_family_options = nullptr);
+  Status LogAndApply(ColumnFamilyData* column_family_data, VersionEdit* edit,
+                     port::Mutex* mu, Directory* db_directory = nullptr,
+                     bool new_descriptor_log = false,
+                     const ColumnFamilyOptions* column_family_options =
+                         nullptr);
+  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
+  static Status ListColumnFamilies(std::vector<std::string>* column_families,
+                                   const std::string& dbname, Env* env);
@@ -181 +185,0 @@ class VersionSet {
-  Compaction* PickCompaction(LogBuffer* log_buffer);
@@ -191 +195,2 @@ class VersionSet {
-  void GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata);
+  void GetLiveFilesMetaData(
+    std::vector<LiveFileMetaData> *metadata);
@@ -197 +201,0 @@ class VersionSet {
-  friend class Compaction;
@@ -229,4 +232,0 @@ class VersionSet {
- public:
-  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
-  static Status ListColumnFamilies(std::vector<std::string>* column_families,
-                                   const std::string& dbname, Env* env);
