--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e2dd148a-abeee9f2-8ca30bd5/include@rocksdb@options.no_comments_merged.h
@@ -121,0 +118 @@ struct DBOptions {
+  InfoLogLevel info_log_level;
@@ -157 +159,3 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
-  InfoLogLevel info_log_level;
+  Options() :
+    DBOptions(),
+    ColumnFamilyOptions() {}
