--- a/./output/tmux/70ff8cfe-09afc6c8-7e497c7f/utf8.no_comments_conflict.c
+++ b/./output/tmux/70ff8cfe-09afc6c8-7e497c7f/utf8.no_comments_merged.c
@@ -172,14 +171,0 @@ utf8_width(struct utf8_data *ud, int *width)
-<<<<<<< HEAD
-#endif
- if (*width >= 0 && *width <= 0xff)
-  return (UTF8_DONE);
- log_debug("UTF-8 %.*s, wcwidth() %d", (int)ud->size, ud->data, *width);
- return (UTF8_ERROR);
-||||||| 7e497c7f
- if (*width < 0 || *width > 0xff) {
-  log_debug("UTF-8 %.*s, wcwidth() %d", (int)ud->size, ud->data,
-      *width);
-  return (UTF8_ERROR);
- }
- return (UTF8_DONE);
-=======
@@ -191 +176,0 @@ utf8_width(struct utf8_data *ud, int *width)
->>>>>>> 09afc6c8
