diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/4445969f-5a4e1a51-d0c19465/src@main@database.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/4445969f-5a4e1a51-d0c19465/src@main@database.no_comments_merged.cpp
index e1b806f..ce7e3c4 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/4445969f-5a4e1a51-d0c19465/src@main@database.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/4445969f-5a4e1a51-d0c19465/src@main@database.no_comments_merged.cpp
@@ -160,13 +159,0 @@ void DatabaseInstance::Initialize(const char *database_path, DBConfig *user_conf
-<<<<<<< HEAD
- db_manager = make_unique<DatabaseManager>(*this);
- buffer_manager = make_unique<StandardBufferManager>(*this, config.options.temporary_directory);
- scheduler = make_unique<TaskScheduler>(*this);
- object_cache = make_unique<ObjectCache>();
- connection_manager = make_unique<ConnectionManager>();
-|||||||
- db_manager = make_unique<DatabaseManager>(*this);
- buffer_manager = make_unique<BufferManager>(*this, config.options.temporary_directory);
- scheduler = make_unique<TaskScheduler>(*this);
- object_cache = make_unique<ObjectCache>();
- connection_manager = make_unique<ConnectionManager>();
-=======
@@ -174 +161 @@ void DatabaseInstance::Initialize(const char *database_path, DBConfig *user_conf
- buffer_manager = make_uniq<BufferManager>(*this, config.options.temporary_directory);
+ buffer_manager = make_uniq<StandardBufferManager>(*this, config.options.temporary_directory);
@@ -178 +164,0 @@ void DatabaseInstance::Initialize(const char *database_path, DBConfig *user_conf
->>>>>>> 5a4e1a513e3ada1c37f9d6c70962139438035d4f
