--- a/./output/frameworks_av/2ef972fb-043bd860-b4efe510/services@audiopolicy@service@AudioPolicyEffects.no_comments_conflict.h
+++ b/./output/frameworks_av/2ef972fb-043bd860-b4efe510/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
@@ -66 +65,0 @@ private:
-<<<<<<< HEAD
@@ -68,5 +66,0 @@ private:
-||||||| b4efe510d2
-        EffectDesc(const char *name, const effect_uuid_t& uuid) :
-=======
-        EffectDesc(const char *name, const effect_uuid_t& uuid) :
->>>>>>> 043bd860
@@ -101 +94,0 @@ private:
-<<<<<<< HEAD
@@ -104,7 +96,0 @@ private:
-||||||| b4efe510d2
-        int mRefCount;
-        Vector< sp<AudioEffect> >mEffects;
-=======
-        int mRefCount = 0;
-        Vector< sp<AudioEffect> >mEffects;
->>>>>>> 043bd860
