diff --git a/./output/art/727fa8fb-9da5bba7-8af6a618/libnativeloader@native_loader_test.no_comments_mergegen.cpp b/./output/art/727fa8fb-9da5bba7-8af6a618/libnativeloader@native_loader_test.no_comments_merged.cpp
index c523b8f..a9a4e1c 100644
--- a/./output/art/727fa8fb-9da5bba7-8af6a618/libnativeloader@native_loader_test.no_comments_mergegen.cpp
+++ b/./output/art/727fa8fb-9da5bba7-8af6a618/libnativeloader@native_loader_test.no_comments_merged.cpp
@@ -296 +296,12 @@ TEST_P(NativeLoaderTest_Create, UnbundledProductApp) {
-if (is_product_treblelized()) { expected_namespace_prefix = "product-clns"; expected_library_path = expected_library_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR; expected_permitted_path = expected_permitted_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR; expected_shared_libs_to_platform_ns = append_extended_libraries(default_public_libraries() + ":" + llndk_libraries_product()); if (android::base::GetProperty("ro.product.vndk.version", "") != "") { expected_link_with_vndk_product_ns = true; } }
+  if (is_product_treblelized()) {
+    expected_namespace_prefix = "product-clns";
+    expected_library_path =
+        expected_library_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR;
+    expected_permitted_path =
+        expected_permitted_path + ":/product/" LIB_DIR ":/system/product/" LIB_DIR;
+    expected_shared_libs_to_platform_ns =
+        append_extended_libraries(default_public_libraries() + ":" + llndk_libraries_product());
+    if (android::base::GetProperty("ro.product.vndk.version", "") != "") {
+      expected_link_with_vndk_product_ns = true;
+    }
+  }
