--- a/./output/art/727fa8fb-9da5bba7-8af6a618/libnativeloader@public_libraries.no_comments_conflict.cpp
+++ b/./output/art/727fa8fb-9da5bba7-8af6a618/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -154 +153,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -158,7 +156,0 @@ static std::string InitProductPublicLibraries() {
-||||||| 8af6a618ff
-  if (is_product_vndk_version_defined()) {
-    ReadExtensionLibraries("/product/etc", &sonames);
-  }
-=======
-  ReadExtensionLibraries("/product/etc", &sonames);
->>>>>>> 9da5bba7
@@ -173 +164,0 @@ static std::string InitExtendedPublicLibraries() {
-<<<<<<< HEAD
@@ -177,6 +167,0 @@ static std::string InitExtendedPublicLibraries() {
-||||||| 8af6a618ff
-  if (!is_product_vndk_version_defined()) {
-    ReadExtensionLibraries("/product/etc", &sonames);
-  }
-=======
->>>>>>> 9da5bba7
@@ -219 +203,0 @@ static std::string InitLlndkLibrariesProduct() {
-<<<<<<< HEAD
@@ -224,7 +207,0 @@ static std::string InitLlndkLibrariesProduct() {
-||||||| 8af6a618ff
-  if (!is_product_vndk_version_defined()) {
-    ALOGD("InitLlndkLibrariesProduct: No product VNDK version defined");
-    return "";
-  }
-=======
->>>>>>> 9da5bba7
@@ -370 +346,0 @@ const std::map<std::string, std::string>& apex_public_libraries() {
-<<<<<<< HEAD
@@ -381,10 +356,0 @@ bool is_product_treblelized() {
-||||||| 8af6a618ff
-bool is_product_vndk_version_defined() {
-#if defined(ART_TARGET_ANDROID)
-  return android::sysprop::VndkProperties::product_vndk_version().has_value();
-#else
-  return false;
-#endif
-}
-=======
->>>>>>> 9da5bba7
