--- a/./output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_conflict.h
+++ b/./output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_merged.h
@@ -10 +9,0 @@
-#include <android/permission/PermissionChecker.h>
@@ -409,18 +407,0 @@ private:
-    bool checkPermission(const std::string& cameraId, const std::string& permission,
-            const content::AttributionSourceState& attributionSource, const std::string& message,
-            int32_t attributedOpCode) const;
-    bool hasPermissionsForSystemCamera(const std::string& cameraId, int callingPid, int callingUid)
-            const;
-    bool hasPermissionsForCameraHeadlessSystemUser(const std::string& cameraId, int callingPid,
-            int callingUid) const;
-    bool hasCameraPermissions() const;
-    bool hasPermissionsForCameraPrivacyAllowlist(int callingPid, int callingUid) const;
-    bool checkPermission(const std::string& cameraId, const std::string& permission,
-            const content::AttributionSourceState& attributionSource, const std::string& message,
-            int32_t attributedOpCode) const;
-    bool hasPermissionsForSystemCamera(const std::string& cameraId, int callingPid, int callingUid)
-            const;
-    bool hasPermissionsForCameraHeadlessSystemUser(const std::string& cameraId, int callingPid,
-            int callingUid) const;
-    bool hasCameraPermissions() const;
-    bool hasPermissionsForCameraPrivacyAllowlist(int callingPid, int callingUid) const;
@@ -796,2 +776,0 @@ private:
-    std::unique_ptr<permission::PermissionChecker> mPermissionChecker =
-            std::make_unique<permission::PermissionChecker>();
