diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_merged.h
index 0ab39e1..e11affa 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/5827691a-71a5eb28-c03ada6c/services@camera@libcameraservice@CameraService.no_comments_merged.h
@@ -10 +9,0 @@
-#include <android/permission/PermissionChecker.h>
@@ -50 +49 @@ namespace android {
-volatile int32_t gLogLevel = 0;
+extern volatile int32_t gLogLevel;
@@ -53 +52,2 @@ class MediaPlayer;
-class CameraService : public BinderService<CameraService>,
+class CameraService :
+    public BinderService<CameraService>,
@@ -58 +58,2 @@ class CameraService : public BinderService<CameraService>,
-                      public AttributionAndPermissionUtilsEncapsulator {
+    public AttributionAndPermissionUtilsEncapsulator
+{
@@ -62,125 +63,2 @@ class CameraService : public BinderService<CameraService>,
-    class Client : public hardware::BnCamera, public BasicClient {
-      public:
-        typedef hardware::ICameraClient TCamCallbacks;
-        virtual binder::Status disconnect();
-        virtual status_t connect(const sp<hardware::ICameraClient>& client) = 0;
-        virtual status_t lock() = 0;
-        virtual status_t unlock() = 0;
-        virtual status_t setPreviewTarget(const sp<IGraphicBufferProducer>& bufferProducer) = 0;
-        virtual void setPreviewCallbackFlag(int flag) = 0;
-        virtual status_t setPreviewCallbackTarget(
-                const sp<IGraphicBufferProducer>& callbackProducer) = 0;
-        virtual status_t startPreview() = 0;
-        virtual void stopPreview() = 0;
-        virtual bool previewEnabled() = 0;
-        virtual status_t setVideoBufferMode(int32_t videoBufferMode) = 0;
-        virtual status_t startRecording() = 0;
-        virtual void stopRecording() = 0;
-        virtual bool recordingEnabled() = 0;
-        virtual void releaseRecordingFrame(const sp<IMemory>& mem) = 0;
-        virtual status_t autoFocus() = 0;
-        virtual status_t cancelAutoFocus() = 0;
-        virtual status_t takePicture(int msgType) = 0;
-        virtual status_t setParameters(const String8& params) = 0;
-        virtual String8 getParameters() const = 0;
-        virtual status_t sendCommand(int32_t cmd, int32_t arg1, int32_t arg2) = 0;
-        virtual status_t setVideoTarget(const sp<IGraphicBufferProducer>& bufferProducer) = 0;
-        Client(const sp<CameraService>& cameraService,
-               const sp<hardware::ICameraClient>& cameraClient,
-               const std::string& clientPackageName, bool systemNativeClient,
-               const std::optional<std::string>& clientFeatureId, const std::string& cameraIdStr,
-               int api1CameraId, int cameraFacing, int sensorOrientation, int clientPid,
-               uid_t clientUid, int servicePid, bool overrideToPortrait);
-        ~Client();
-        const sp<hardware::ICameraClient>& getRemoteCallback() { return mRemoteCallback; }
-        virtual sp<IBinder> asBinderWrapper() { return asBinder(this); }
-        virtual void notifyError(int32_t errorCode, const CaptureResultExtras& resultExtras);
-        virtual bool canCastToApiClient(apiLevel level) const;
-        void setImageDumpMask(int ) {}
-      protected:
-        sp<hardware::ICameraClient> mRemoteCallback;
-        int mCameraId;
-    };
-    class BasicClient : public virtual RefBase {
-        friend class CameraService;
-      public:
-        virtual status_t initialize(sp<CameraProviderManager> manager,
-                                    const std::string& monitorTags) = 0;
-        virtual binder::Status disconnect();
-        virtual sp<IBinder> asBinderWrapper() = 0;
-        sp<IBinder> getRemote() { return mRemoteBinder; }
-        bool getOverrideToPortrait() const { return mOverrideToPortrait; }
-        virtual status_t dump(int fd, const Vector<String16>& args);
-        virtual status_t dumpClient(int fd, const Vector<String16>& args) = 0;
-        virtual status_t startWatchingTags(const std::string& tags, int outFd);
-        virtual status_t stopWatchingTags(int outFd);
-        virtual status_t dumpWatchedEventsToVector(std::vector<std::string>& out);
-        virtual std::string getPackageName() const;
-        virtual int getCameraFacing() const;
-        virtual int getCameraOrientation() const;
-        virtual void notifyError(int32_t errorCode, const CaptureResultExtras& resultExtras) = 0;
-        virtual uid_t getClientUid() const;
-        virtual int getClientPid() const;
-        virtual bool canCastToApiClient(apiLevel level) const;
-        virtual void block();
-        virtual status_t setAudioRestriction(int32_t mode);
-        virtual int32_t getServiceAudioRestriction() const;
-        virtual int32_t getAudioRestriction() const;
-        static bool isValidAudioRestriction(int32_t mode);
-        virtual status_t setRotateAndCropOverride(uint8_t rotateAndCrop, bool fromHal = false) = 0;
-        virtual status_t setAutoframingOverride(uint8_t autoframingValue) = 0;
-        virtual bool supportsCameraMute() = 0;
-        virtual status_t setCameraMute(bool enabled) = 0;
-        virtual status_t setCameraServiceWatchdog(bool enabled) = 0;
-        virtual void setStreamUseCaseOverrides(const std::vector<int64_t>& useCaseOverrides) = 0;
-        virtual void clearStreamUseCaseOverrides() = 0;
-        virtual bool supportsZoomOverride() = 0;
-        virtual status_t setZoomOverride(int32_t zoomOverride) = 0;
-        virtual status_t injectCamera(const std::string& injectedCamId,
-                                      sp<CameraProviderManager> manager) = 0;
-        virtual status_t stopInjection() = 0;
-        virtual status_t injectSessionParams(
-                const hardware::camera2::impl::CameraMetadataNative& sessionParams) = 0;
-      protected:
-        BasicClient(const sp<CameraService>& cameraService, const sp<IBinder>& remoteCallback,
-                    const std::string& clientPackageName, bool nativeClient,
-                    const std::optional<std::string>& clientFeatureId,
-                    const std::string& cameraIdStr, int cameraFacing, int sensorOrientation,
-                    int clientPid, uid_t clientUid, int servicePid, bool overrideToPortrait);
-        virtual ~BasicClient();
-        bool mDestructionStarted;
-        static sp<CameraService> sCameraService;
-        const std::string mCameraIdStr;
-        const int mCameraFacing;
-        const int mOrientation;
-        std::string mClientPackageName;
-        bool mSystemNativeClient;
-        std::optional<std::string> mClientFeatureId;
-        pid_t mClientPid;
-        const uid_t mClientUid;
-        const pid_t mServicePid;
-        bool mDisconnected;
-        bool mUidIsTrusted;
-        bool mOverrideToPortrait;
-        mutable Mutex mAudioRestrictionLock;
-        int32_t mAudioRestriction;
-        sp<IBinder> mRemoteBinder;
-        virtual status_t startCameraOps();
-        virtual status_t startCameraStreamingOps();
-        virtual status_t finishCameraStreamingOps();
-        virtual status_t finishCameraOps();
-        virtual status_t handleAppOpMode(int32_t mode);
-        virtual status_t noteAppOp();
-        std::unique_ptr<AppOpsManager> mAppOpsManager = nullptr;
-        class OpsCallback : public BnAppOpsCallback {
-          public:
-            explicit OpsCallback(wp<BasicClient> client);
-            virtual void opChanged(int32_t op, const String16& packageName);
-          private:
-            wp<BasicClient> mClient;
-        };
-        sp<OpsCallback> mOpsCallback;
-        bool mOpsActive;
-        bool mOpsStreaming;
-        virtual void opChanged(int32_t op, const String16& packageName);
-    };
+    class Client;
+    class BasicClient;
@@ -188 +66,4 @@ class CameraService : public BinderService<CameraService>,
-    enum apiLevel { API_1 = 1, API_2 = 2 };
+    enum apiLevel {
+        API_1 = 1,
+        API_2 = 2
+    };
@@ -197,3 +78,4 @@ class CameraService : public BinderService<CameraService>,
-    CameraService(
-            std::shared_ptr<CameraServiceProxyWrapper> cameraServiceProxyWrapper = nullptr,
-            std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils = nullptr);
+                        CameraService(std::shared_ptr<CameraServiceProxyWrapper>
+                                cameraServiceProxyWrapper = nullptr,
+                                std::shared_ptr<AttributionAndPermissionUtils>
+                                attributionAndPermissionUtils = nullptr);
@@ -208 +90,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual void onTorchStatusChanged(const std::string& cameraId, TorchModeStatus newStatus,
+    virtual void onTorchStatusChanged(const std::string& cameraId,
+            TorchModeStatus newStatus,
@@ -211,2 +94,4 @@ class CameraService : public BinderService<CameraService>,
-    virtual binder::Status getNumberOfCameras(int32_t type, int32_t deviceId, int32_t devicePolicy,
-                                              int32_t* numCameras);
+    virtual binder::Status getNumberOfCameras(int32_t type, int32_t deviceId,
+            int32_t devicePolicy, int32_t* numCameras);
+    virtual binder::Status getCameraInfo(int cameraId, bool overrideToPortrait,
+            int32_t deviceId, int32_t devicePolicy, hardware::CameraInfo* cameraInfo) override;
@@ -214,3 +99,2 @@ class CameraService : public BinderService<CameraService>,
-                                                    int targetSdkVersion, bool overrideToPortrait,
-                                                    int32_t deviceId, int32_t devicePolicy,
-                                                    CameraMetadata* cameraInfo) override;
+            int targetSdkVersion, bool overrideToPortrait, int32_t deviceId,
+            int32_t devicePolicy, CameraMetadata* cameraInfo) override;
@@ -224,3 +108,2 @@ class CameraService : public BinderService<CameraService>,
-                                   bool overrideToPortrait, bool forceSlowJpegMode,
-                                   int32_t deviceId, int32_t devicePolicy,
-                                           sp<hardware::ICamera>* device) override;
+            bool overrideToPortrait, bool forceSlowJpegMode, int32_t deviceId,
+            int32_t devicePolicy, sp<hardware::ICamera>* device) override;
@@ -229,3 +112,4 @@ class CameraService : public BinderService<CameraService>,
-            const std::string& cameraId, const std::string& clientPackageName,
-            const std::optional<std::string>& clientFeatureId, int32_t clientUid, int scoreOffset,
-            int targetSdkVersion, bool overrideToPortrait, int32_t deviceId, int32_t devicePolicy,
+            const std::string& cameraId,
+            const std::string& clientPackageName, const std::optional<std::string>& clientFeatureId,
+            int32_t clientUid, int scoreOffset, int targetSdkVersion, bool overrideToPortrait,
+            int32_t deviceId, int32_t devicePolicy,
@@ -235 +119,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual binder::Status removeListener(const sp<hardware::ICameraServiceListener>& listener);
+    virtual binder::Status removeListener(
+            const sp<hardware::ICameraServiceListener>& listener);
@@ -237,2 +122 @@ class CameraService : public BinderService<CameraService>,
-            std::vector<hardware::camera2::utils::ConcurrentCameraIdCombination>*
-                    concurrentCameraIds);
+        std::vector<hardware::camera2::utils::ConcurrentCameraIdCombination>* concurrentCameraIds);
@@ -242 +126,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual binder::Status getLegacyParameters(int32_t cameraId,
+    virtual binder::Status getLegacyParameters(
+            int32_t cameraId,
@@ -245,2 +130 @@ class CameraService : public BinderService<CameraService>,
-                                        const sp<IBinder>& clientBinder, int32_t deviceId,
-                                        int32_t devicePolicy);
+            const sp<IBinder>& clientBinder, int32_t deviceId, int32_t devicePolicy);
@@ -248,3 +132,2 @@ class CameraService : public BinderService<CameraService>,
-                                                        int32_t torchStrength,
-                                                        const sp<IBinder>& clientBinder,
-                                                        int32_t deviceId, int32_t devicePolicy);
+            int32_t torchStrength, const sp<IBinder>& clientBinder, int32_t deviceId,
+            int32_t devicePolicy);
@@ -253 +136,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual binder::Status notifySystemEvent(int32_t eventId, const std::vector<int32_t>& args);
+    virtual binder::Status notifySystemEvent(int32_t eventId,
+            const std::vector<int32_t>& args);
@@ -256 +140,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual binder::Status supportsCameraApi(const std::string& cameraId, int32_t apiVersion,
+    virtual binder::Status supportsCameraApi(
+            const std::string& cameraId, int32_t apiVersion,
@@ -258 +143,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual binder::Status isHiddenPhysicalCamera(const std::string& cameraId,
+    virtual binder::Status isHiddenPhysicalCamera(
+            const std::string& cameraId,
@@ -267,3 +153,2 @@ class CameraService : public BinderService<CameraService>,
-  private:
-    void remapCameraIds(const TCameraIdRemapping& cameraIdRemapping);
-  public:
+    virtual binder::Status remapCameraIds(const hardware::CameraIdRemapping&
+            cameraIdRemapping);
@@ -273,2 +158,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual binder::Status createDefaultRequest(
-            const std::string& cameraId, int templateId, int32_t deviceId, int32_t devicePolicy,
+    virtual binder::Status createDefaultRequest(const std::string& cameraId, int templateId,
+            int32_t deviceId, int32_t devicePolicy,
@@ -277 +162,2 @@ class CameraService : public BinderService<CameraService>,
-            const std::string& cameraId, const SessionConfiguration& sessionConfiguration,
+            const std::string& cameraId,
+            const SessionConfiguration& sessionConfiguration,
@@ -284 +170,2 @@ class CameraService : public BinderService<CameraService>,
-    virtual status_t onTransact(uint32_t code, const Parcel& data, Parcel* reply, uint32_t flags);
+    virtual status_t onTransact(uint32_t code, const Parcel& data,
+                                   Parcel* reply, uint32_t flags);
@@ -288,2 +175,2 @@ class CameraService : public BinderService<CameraService>,
-                                     std::vector<hardware::CameraStatus>* cameraStatuses,
-                                     bool isVendor = false, bool isProcessLocalTest = false);
+            std::vector<hardware::CameraStatus>* cameraStatuses, bool isVendor = false,
+            bool isProcessLocalTest = false);
@@ -306,2 +193 @@ class CameraService : public BinderService<CameraService>,
-                                                  int* facing = nullptr,
-                                                  int* orientation = nullptr);
+            int* facing = nullptr, int* orientation = nullptr);
@@ -312 +198,4 @@ class CameraService : public BinderService<CameraService>,
-    class BasicClient : public virtual RefBase {
+    bool isAutomotiveExteriorSystemCamera(const std::string& cameraId) const;
+    class BasicClient :
+        public virtual RefBase,
+        public AttributionAndPermissionUtilsEncapsulator {
@@ -319,2 +208,6 @@ class CameraService : public BinderService<CameraService>,
-        sp<IBinder> getRemote() { return mRemoteBinder; }
-        bool getOverrideToPortrait() const { return mOverrideToPortrait; }
+        sp<IBinder> getRemote() {
+            return mRemoteBinder;
+        }
+        bool getOverrideToPortrait() const {
+            return mOverrideToPortrait;
+        }
@@ -329 +222,2 @@ class CameraService : public BinderService<CameraService>,
-        virtual void notifyError(int32_t errorCode, const CaptureResultExtras& resultExtras) = 0;
+        virtual void notifyError(int32_t errorCode,
+                const CaptureResultExtras& resultExtras) = 0;
@@ -343 +237,2 @@ class CameraService : public BinderService<CameraService>,
-        virtual void setStreamUseCaseOverrides(const std::vector<int64_t>& useCaseOverrides) = 0;
+        virtual void setStreamUseCaseOverrides(
+                const std::vector<int64_t>& useCaseOverrides) = 0;
@@ -353,2 +248,5 @@ class CameraService : public BinderService<CameraService>,
-        BasicClient(const sp<CameraService>& cameraService, const sp<IBinder>& remoteCallback,
-                    const std::string& clientPackageName, bool nativeClient,
+        BasicClient(const sp<CameraService>& cameraService,
+                const sp<IBinder>& remoteCallback,
+                std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
+                const std::string& clientPackageName,
+                bool nativeClient,
@@ -356,2 +254,7 @@ class CameraService : public BinderService<CameraService>,
-                    const std::string& cameraIdStr, int cameraFacing, int sensorOrientation,
-                    int clientPid, uid_t clientUid, int servicePid, bool overrideToPortrait);
+                const std::string& cameraIdStr,
+                int cameraFacing,
+                int sensorOrientation,
+                int clientPid,
+                uid_t clientUid,
+                int servicePid,
+                bool overrideToPortrait);
@@ -395 +298,2 @@ class CameraService : public BinderService<CameraService>,
-    class Client : public hardware::BnCamera, public BasicClient {
+    class Client : public hardware::BnCamera, public BasicClient
+    {
@@ -423,4 +327,12 @@ class CameraService : public BinderService<CameraService>,
-               const std::string& clientPackageName, bool systemNativeClient,
-               const std::optional<std::string>& clientFeatureId, const std::string& cameraIdStr,
-               int api1CameraId, int cameraFacing, int sensorOrientation, int clientPid,
-               uid_t clientUid, int servicePid, bool overrideToPortrait);
+                std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils,
+                const std::string& clientPackageName,
+                bool systemNativeClient,
+                const std::optional<std::string>& clientFeatureId,
+                const std::string& cameraIdStr,
+                int api1CameraId,
+                int cameraFacing,
+                int sensorOrientation,
+                int clientPid,
+                uid_t clientUid,
+                int servicePid,
+                bool overrideToPortrait);
@@ -428,3 +340,8 @@ class CameraService : public BinderService<CameraService>,
-        const sp<hardware::ICameraClient>& getRemoteCallback() { return mRemoteCallback; }
-        virtual sp<IBinder> asBinderWrapper() { return asBinder(this); }
-        virtual void notifyError(int32_t errorCode, const CaptureResultExtras& resultExtras);
+        const sp<hardware::ICameraClient>& getRemoteCallback() {
+            return mRemoteCallback;
+        }
+        virtual sp<IBinder> asBinderWrapper() {
+            return asBinder(this);
+        }
+        virtual void notifyError(int32_t errorCode,
+                                         const CaptureResultExtras& resultExtras);
@@ -439,4 +356,4 @@ class CameraService : public BinderService<CameraService>,
-        void onClientAdded(const resource_policy::ClientDescriptor<
-                           std::string, sp<CameraService::BasicClient>>& descriptor);
-        void onClientRemoved(const resource_policy::ClientDescriptor<
-                             std::string, sp<CameraService::BasicClient>>& descriptor);
+        void onClientAdded(const resource_policy::ClientDescriptor<std::string,
+                sp<CameraService::BasicClient>>& descriptor);
+        void onClientRemoved(const resource_policy::ClientDescriptor<std::string,
+                sp<CameraService::BasicClient>>& descriptor);
@@ -444,6 +361,4 @@ class CameraService : public BinderService<CameraService>,
-    typedef std::shared_ptr<
-            resource_policy::ClientDescriptor<std::string, sp<CameraService::BasicClient>>>
-            DescriptorPtr;
-    class CameraClientManager
-        : public resource_policy::ClientManager<std::string, sp<CameraService::BasicClient>,
-                                                ClientEventListener> {
+    typedef std::shared_ptr<resource_policy::ClientDescriptor<std::string,
+            sp<CameraService::BasicClient>>> DescriptorPtr;
+    class CameraClientManager : public resource_policy::ClientManager<std::string,
+            sp<CameraService::BasicClient>, ClientEventListener> {
@@ -457,3 +372,2 @@ class CameraService : public BinderService<CameraService>,
-                                                  const std::set<std::string>& conflictingKeys,
-                                                  int32_t score, int32_t ownerId, int32_t state,
-                                                  int oomScoreOffset, bool systemNativeClient);
+                const std::set<std::string>& conflictingKeys, int32_t score,
+                int32_t ownerId, int32_t state, int oomScoreOffset, bool systemNativeClient);
@@ -461,2 +375,2 @@ class CameraService : public BinderService<CameraService>,
-                                                  const CameraService::DescriptorPtr& partial,
-                                                  int oomScoreOffset, bool systemNativeClient);
+                const CameraService::DescriptorPtr& partial, int oomScoreOffset,
+                bool systemNativeClient);
@@ -467 +380,0 @@ class CameraService : public BinderService<CameraService>,
-    bool isAutomotiveExteriorSystemCamera(const std::string& cameraId) const;
@@ -473,2 +386 @@ class CameraService : public BinderService<CameraService>,
-                return interface_cast<IActivityManager>(
-                        sm->checkService(String16(kActivityService)));
+                 return interface_cast<IActivityManager>(sm->checkService(String16(kActivityService)));
@@ -500,2 +412 @@ class CameraService : public BinderService<CameraService>,
-        NOT_AVAILABLE =
-                static_cast<int32_t>(hardware::ICameraServiceListener::STATUS_NOT_AVAILABLE),
+        NOT_AVAILABLE = static_cast<int32_t>(hardware::ICameraServiceListener::STATUS_NOT_AVAILABLE),
@@ -512 +423,2 @@ class CameraService : public BinderService<CameraService>,
-        void updateStatus(StatusInternal status, const std::string& cameraId,
+        void updateStatus(StatusInternal status,
+                const std::string& cameraId,
@@ -538 +450,2 @@ class CameraService : public BinderService<CameraService>,
-    class UidPolicy : public BnUidObserver,
+    class UidPolicy :
+        public BnUidObserver,
@@ -542 +455,2 @@ class CameraService : public BinderService<CameraService>,
-        explicit UidPolicy(sp<CameraService> service) : mRegistered(false), mService(service) {}
+        explicit UidPolicy(sp<CameraService> service)
+                : mRegistered(false), mService(service) {}
@@ -586,2 +500 @@ class CameraService : public BinderService<CameraService>,
-        explicit SensorPrivacyPolicy(
-                wp<CameraService> service,
+            explicit SensorPrivacyPolicy(wp<CameraService> service,
@@ -592,2 +505 @@ class CameraService : public BinderService<CameraService>,
-              mCameraPrivacyState(SensorPrivacyManager::DISABLED),
-              mRegistered(false) {}
+                    mCameraPrivacyState(SensorPrivacyManager::DISABLED), mRegistered(false) {}
@@ -600 +512,2 @@ class CameraService : public BinderService<CameraService>,
-        binder::Status onSensorPrivacyChanged(int toggleType, int sensor, bool enabled);
+            binder::Status onSensorPrivacyChanged(int toggleType, int sensor,
+                                                  bool enabled);
@@ -623,2 +536 @@ class CameraService : public BinderService<CameraService>,
-                                                   int& clientUid, int& clientPid,
-                                                 int& originalClientPid) const;
+                     int& clientUid, int& clientPid, int& originalClientPid) const;
@@ -626,3 +538 @@ class CameraService : public BinderService<CameraService>,
-                                                   const std::string& clientName,
-                                                             int& clientUid,
-                                                             int& clientPid,
+            const std::string& clientName, int& clientUid, int& clientPid,
@@ -632,4 +542,3 @@ class CameraService : public BinderService<CameraService>,
-    status_t handleEvictionsLocked(
-            const std::string& cameraId, int clientPid, apiLevel effectiveApiLevel,
-            const sp<IBinder>& remoteCallback, const std::string& packageName, int scoreOffset,
-            bool systemNativeClient,
+    status_t handleEvictionsLocked(const std::string& cameraId, int clientPid,
+        apiLevel effectiveApiLevel, const sp<IBinder>& remoteCallback,
+        const std::string& packageName, int scoreOffset, bool systemNativeClient,
@@ -637,2 +546 @@ class CameraService : public BinderService<CameraService>,
-            std::shared_ptr<resource_policy::ClientDescriptor<std::string, sp<BasicClient>>>*
-                    partial);
+        std::shared_ptr<resource_policy::ClientDescriptor<std::string, sp<BasicClient>>>* partial);
@@ -644,0 +553,7 @@ class CameraService : public BinderService<CameraService>,
+    template<class CALLBACK, class CLIENT>
+    binder::Status connectHelper(const sp<CALLBACK>& cameraCb, const std::string& cameraId,
+            int api1CameraId, const std::string& clientPackageNameMaybe, bool systemNativeClient,
+            const std::optional<std::string>& clientFeatureId, int clientUid, int clientPid,
+            apiLevel effectiveApiLevel, bool shimUpdateOnly, int scoreOffset, int targetSdkVersion,
+            bool overrideToPortrait, bool forceSlowJpegMode, const std::string& originalCameraId,
+                   sp<CLIENT>& device);
@@ -657 +572,2 @@ class CameraService : public BinderService<CameraService>,
-    binder::Status parseCameraIdRemapping(const hardware::CameraIdRemapping& cameraIdRemapping,
+    binder::Status parseCameraIdRemapping(
+            const hardware::CameraIdRemapping& cameraIdRemapping,
@@ -659 +575,3 @@ class CameraService : public BinderService<CameraService>,
-    std::string resolveCameraId(const std::string& inputCameraId, int clientUid,
+    std::string resolveCameraId(
+            const std::string& inputCameraId,
+            int clientUid,
@@ -661,2 +579,5 @@ class CameraService : public BinderService<CameraService>,
-    std::optional<std::string> resolveCameraId(const std::string& inputCameraId, int32_t deviceId,
-                                               int32_t devicePolicy, int clientUid,
+    std::optional<std::string> resolveCameraId(
+            const std::string& inputCameraId,
+            int32_t deviceId,
+            int32_t devicePolicy,
+            int clientUid,
@@ -665,2 +586,2 @@ class CameraService : public BinderService<CameraService>,
-    std::vector<std::string> findOriginalIdsForRemappedCameraId(const std::string& inputCameraId,
-                                                                int clientUid);
+    std::vector<std::string> findOriginalIdsForRemappedCameraId(
+        const std::string& inputCameraId, int clientUid);
@@ -693 +614,2 @@ class CameraService : public BinderService<CameraService>,
-    void logUserSwitch(const std::set<userid_t>& oldUserIds, const std::set<userid_t>& newUserIds);
+    void logUserSwitch(const std::set<userid_t>& oldUserIds,
+        const std::set<userid_t>& newUserIds);
@@ -718,4 +640 @@ class CameraService : public BinderService<CameraService>,
-            : mParent(parent),
-              mListener(listener),
-              mListenerUid(uid),
-              mListenerPid(pid),
+                    : mParent(parent), mListener(listener), mListenerUid(uid), mListenerPid(pid),
@@ -731 +650,2 @@ class CameraService : public BinderService<CameraService>,
-        void handleBinderStatus(const binder::Status& ret, const char* logOnError, args_t... args) {
+            void handleBinderStatus(const binder::Status &ret, const char *logOnError,
+                    args_t... args) {
@@ -733,2 +653,2 @@ class CameraService : public BinderService<CameraService>,
-                (ret.exceptionCode() != binder::Status::Exception::EX_TRANSACTION_FAILED ||
-                 !mLastTransactFailed)) {
+                        (ret.exceptionCode() != binder::Status::Exception::EX_TRANSACTION_FAILED
+                        || !mLastTransactFailed)) {
@@ -769,3 +689,6 @@ class CameraService : public BinderService<CameraService>,
-    void updateStatus(StatusInternal status, const std::string& cameraId,
-                      std::initializer_list<StatusInternal> rejectedSourceStates);
-    void updateStatus(StatusInternal status, const std::string& cameraId);
+    void updateStatus(StatusInternal status,
+            const std::string& cameraId,
+            std::initializer_list<StatusInternal>
+                rejectedSourceStates);
+    void updateStatus(StatusInternal status,
+            const std::string& cameraId);
@@ -778 +701,2 @@ class CameraService : public BinderService<CameraService>,
-    KeyedVector<std::string, TorchModeStatus> mTorchStatusMap;
+    KeyedVector<std::string, TorchModeStatus>
+            mTorchStatusMap;
@@ -782 +706,2 @@ class CameraService : public BinderService<CameraService>,
-    void onTorchStatusChangedLocked(const std::string& cameraId, TorchModeStatus newStatus,
+    void onTorchStatusChangedLocked(const std::string& cameraId,
+            TorchModeStatus newStatus,
@@ -784,2 +709,4 @@ class CameraService : public BinderService<CameraService>,
-    status_t getTorchStatusLocked(const std::string& cameraId, TorchModeStatus* status) const;
-    status_t setTorchStatusLocked(const std::string& cameraId, TorchModeStatus status);
+    status_t getTorchStatusLocked(const std::string &cameraId,
+             TorchModeStatus *status) const;
+    status_t setTorchStatusLocked(const std::string &cameraId,
+            TorchModeStatus status);
@@ -787,2 +714,2 @@ class CameraService : public BinderService<CameraService>,
-                                          const std::list<std::string>& logicalCameraIds,
-                                          SystemCameraKind deviceKind, int32_t virtualDeviceId);
+            const std::list<std::string>& logicalCameraIds, SystemCameraKind deviceKind,
+            int32_t virtualDeviceId);
@@ -822,8 +749,8 @@ class CameraService : public BinderService<CameraService>,
-    static binder::Status makeClient(
-            const sp<CameraService>& cameraService, const sp<IInterface>& cameraCb,
-            const std::string& packageName, bool systemNativeClient,
-            const std::optional<std::string>& featureId, const std::string& cameraId,
-            int api1CameraId, int facing, int sensorOrientation, int clientPid, uid_t clientUid,
-            int servicePid, std::pair<int, IPCTransport> deviceVersionAndIPCTransport,
-            apiLevel effectiveApiLevel, bool overrideForPerfClass, bool overrideToPortrait,
-            bool forceSlowJpegMode, const std::string& originalCameraId,
+    static binder::Status makeClient(const sp<CameraService>& cameraService,
+            const sp<IInterface>& cameraCb, const std::string& packageName,
+            bool systemNativeClient, const std::optional<std::string>& featureId,
+            const std::string& cameraId, int api1CameraId, int facing, int sensorOrientation,
+            int clientPid, uid_t clientUid, int servicePid,
+            std::pair<int, IPCTransport> deviceVersionAndIPCTransport, apiLevel effectiveApiLevel,
+            bool overrideForPerfClass, bool overrideToPortrait, bool forceSlowJpegMode,
+            const std::string& originalCameraId,
@@ -835,2 +762,2 @@ class CameraService : public BinderService<CameraService>,
-    void broadcastTorchModeStatus(const std::string& cameraId, TorchModeStatus status,
-                                  SystemCameraKind systemCameraKind);
+    void broadcastTorchModeStatus(const std::string& cameraId,
+            TorchModeStatus status, SystemCameraKind systemCameraKind);
@@ -850,2 +776,0 @@ class CameraService : public BinderService<CameraService>,
-    std::unique_ptr<permission::PermissionChecker> mPermissionChecker =
-            std::make_unique<permission::PermissionChecker>();
@@ -884,4 +808,0 @@ class CameraService : public BinderService<CameraService>,
-  public:
-    virtual binder::Status getCameraInfo(int cameraId, bool overrideToPortrait, int32_t deviceId,
-                                         int32_t devicePolicy,
-                                         hardware::CameraInfo* cameraInfo) override;
