diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_merged.cpp
index 7e951b0..67dabd1 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_merged.cpp
@@ -24 +23,0 @@ PhysicalBatchInsert::PhysicalBatchInsert(LogicalOperator &op, SchemaCatalogEntry
-{ PhysicalInsert::GetInsertInfo(*info, insert_types, bound_defaults); }
@@ -30,0 +30 @@ public:
+public:
@@ -179,3 +179,2 @@ public:
- void AddCollection(ClientContext &context, idx_t batch_index, idx_t min_batch_index,
-                    unique_ptr<RowGroupCollection> current_collection, OptimisticDataWriter *writer = nullptr,
-                    bool *written_to_disk = nullptr, optional_ptr<OptimisticDataWriter> writer = nullptr,
+ void AddCollection(ClientContext &context, idx_t batch_index, idx_t min_batch_index, unique_ptr<RowGroupCollection> current_collection,
+                    optional_ptr<OptimisticDataWriter> writer = nullptr,
@@ -236,8 +234,0 @@ public:
- void CreateNewCollection(DuckTableEntry *table, const vector<LogicalType> &insert_types) {
-  auto &table_info = table->GetStorage().info;
-  auto &block_manager = TableIOManager::Get(table->GetStorage()).GetBlockManagerForRowData();
-  current_collection = make_uniq<RowGroupCollection>(table_info, block_manager, insert_types, MAX_ROW_ID);
-  current_collection->InitializeEmpty();
-  current_collection->InitializeAppend(current_append_state);
-  written_to_disk = false;
- }
@@ -284,3 +274,0 @@ SinkResultType PhysicalBatchInsert::Sink(ExecutionContext &context, GlobalSinkSt
-|||||||
-  lstate.writer = gstate.table->GetStorage().CreateOptimisticWriter(context.client);
- } else if (lstate.current_index != lstate.batch_index) {
@@ -290 +278 @@ SinkResultType PhysicalBatchInsert::Sink(ExecutionContext &context, GlobalSinkSt
->>>>>>> c5737e4a948a72b20c39cee054a65a524542d832
+>>>>>>> master
