diff --git a/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_conflict.cpp b/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_merged.cpp
index 0a4755f..67dabd1 100644
--- a/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_conflict.cpp
+++ b/./output/duckdb/3b8b934f-c5737e4a-05ec2cc6/src@execution@operator@persistent@physical_batch_insert.no_comments_merged.cpp
@@ -179,10 +179 @@ public:
-<<<<<<< HEAD
- void AddCollection(ClientContext &context, idx_t batch_index, idx_t min_batch_index,
-                    unique_ptr<RowGroupCollection> current_collection, OptimisticDataWriter *writer = nullptr,
-                    bool *written_to_disk = nullptr) {
-  optional_idx merged_batch_index;
-||||||| 05ec2cc675
- void AddCollection(ClientContext &context, idx_t batch_index, unique_ptr<RowGroupCollection> current_collection,
-                    OptimisticDataWriter *writer = nullptr, bool *written_to_disk = nullptr) {
-=======
- void AddCollection(ClientContext &context, idx_t batch_index, unique_ptr<RowGroupCollection> current_collection,
+ void AddCollection(ClientContext &context, idx_t batch_index, idx_t min_batch_index, unique_ptr<RowGroupCollection> current_collection,
@@ -191 +182 @@ public:
->>>>>>> c5737e4a
+  optional_idx merged_batch_index;
@@ -284,3 +274,0 @@ SinkResultType PhysicalBatchInsert::Sink(ExecutionContext &context, GlobalSinkSt
-||||||| 05ec2cc675
-  lstate.writer = gstate.table->GetStorage().CreateOptimisticWriter(context.client);
- } else if (lstate.current_index != lstate.batch_index) {
@@ -290 +278 @@ SinkResultType PhysicalBatchInsert::Sink(ExecutionContext &context, GlobalSinkSt
->>>>>>> c5737e4a
+>>>>>>> master
