--- a/./output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_conflict.cpp
+++ b/./output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_merged.cpp
@@ -34,6 +33,0 @@ extern "C" {
-<<<<<<< HEAD
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, char *null_value, int columnar,
-                            char thousands, char decimal_sep);
-||||||| 8644661676
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, char *null_value, int columnar);
-=======
@@ -41,2 +35 @@ char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_wid
-                            int columnar);
->>>>>>> 059ac75f
+                            int columnar, char thousands, char decimal_sep);
@@ -201,6 +193,0 @@ int sqlite3_prepare_v2(sqlite3 *db,
-<<<<<<< HEAD
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, char *null_value, int columnar,
-                            char thousand_separator, char decimal_separator) {
-||||||| 8644661676
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, char *null_value, int columnar) {
-=======
@@ -208,2 +195 @@ char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_wid
-                            int columnar) {
->>>>>>> 059ac75f
+                            int columnar, char thousand_separator, char decimal_separator) {
