diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_merged.cpp
index 9d124e2..6501e16 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_merged.cpp
@@ -34,7 +33,0 @@ extern "C" {
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, char *null_value, int columnar,
-                            char thousands, char decimal_sep);
-}
-extern "C" {
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, char *null_value, int columnar);
-}
-extern "C" {
@@ -42 +35 @@ char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_wid
-                            int columnar);
+                            int columnar, char thousands, char decimal_sep);
@@ -202 +195 @@ char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_wid
-                            char *null_value, int columnar, char thousand_separator, char decimal_separator) {
+                            int columnar, char thousand_separator, char decimal_separator) {
