--- a/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@host_init_stubs.no_comments_mergegen.h
+++ b/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@host_init_stubs.no_comments_merged.h
@@ -16,0 +17,4 @@ uint32_t HandlePropertySet(const std::string& name, const std::string& value,
+inline void SetFatalRebootTarget() {}
+inline void __attribute__((noreturn)) InitFatalReboot() {
+    abort();
+}
