--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@init.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@init.no_comments_merged.cpp
@@ -23 +22,0 @@
-#include <cutils/android_reboot.h>
@@ -65 +64 @@ static char qemu[32];
-extern std::string default_console;
+std::string default_console = "/dev/console";
@@ -550 +488,0 @@ int SecondStageMain(int argc, char** argv) {
-<<<<<<< HEAD
@@ -554,6 +491,0 @@ int SecondStageMain(int argc, char** argv) {
-|||||||
-  InitKernelLogging(argv, InitAborter);
-=======
-  SetStdioToDevNull(argv);
-  InitKernelLogging(argv);
->>>>>>> d0ed53e5e702d5fbea82d86a096a489bd18dcceb
