diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@host_init_stubs.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@host_init_stubs.no_comments_merged.h
index d4d1767..337fc8b 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@host_init_stubs.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@host_init_stubs.no_comments_merged.h
@@ -13,4 +13,2 @@ extern std::string default_console;
-bool CanReadProperty(const std::string& source_context,
-                     const std::string& name);
-extern uint32_t (*property_set)(const std::string& name,
-                                const std::string& value);
+bool CanReadProperty(const std::string& source_context, const std::string& name);
+extern uint32_t (*property_set)(const std::string& name, const std::string& value);
@@ -18,2 +16 @@ uint32_t HandlePropertySet(const std::string& name, const std::string& value,
-                           const std::string& source_context, const ucred& cr,
-                           std::string* error);
+                           const std::string& source_context, const ucred& cr, std::string* error);
@@ -21 +18,3 @@ inline void SetFatalRebootTarget() {}
-inline void __attribute__((noreturn)) InitFatalReboot() { abort(); }
+inline void __attribute__((noreturn)) InitFatalReboot() {
+    abort();
+}
@@ -24,2 +23 @@ void SelabelInitialize();
-bool SelabelLookupFileContext(const std::string& key, int type,
-                              std::string* result);
+bool SelabelLookupFileContext(const std::string& key, int type, std::string* result);
