--- a/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@init.no_comments_mergegen.cpp
+++ b/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@init.no_comments_merged.cpp
@@ -489 +489,3 @@ int SecondStageMain(int argc, char** argv) {
- boot_clock::time_point start_time = boot_clock::now(); SetStdioToDevNull(argv);
+    boot_clock::time_point start_time = boot_clock::now();
+    SetStdioToDevNull(argv);
+    InitKernelLogging(argv);
