diff --git a/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@util.no_comments_mergegen.cpp b/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@util.no_comments_merged.cpp
index 3782613..2639646 100644
--- a/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@util.no_comments_mergegen.cpp
+++ b/./output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@util.no_comments_merged.cpp
@@ -23 +23,2 @@
-#include "reboot_utils.h" #include "selabel.h"
+#include "reboot_utils.h"
+#include "selabel.h"
@@ -334 +335,8 @@ void SetStdioToDevNull(char** argv) {
-fd2(fd, STDIN_FILENO); dup2(fd, STDOUT_FILENO); dup2(fd, STDERR_FILENO); if (fd > STDERR_FILENO) close(fd); } void InitKernelLogging(char** argv) { SetFatalRebootTarget(); android::base::InitLogging(argv, &android::base::KernelLogger, InitAborter);
+    dup2(fd, STDIN_FILENO);
+    dup2(fd, STDOUT_FILENO);
+    dup2(fd, STDERR_FILENO);
+    if (fd > STDERR_FILENO) close(fd);
+}
+void InitKernelLogging(char** argv) {
+    SetFatalRebootTarget();
+    android::base::InitLogging(argv, &android::base::KernelLogger, InitAborter);
