diff --git a/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_mergegen.h b/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_merged.h
index f90ac07..e078581 100644
--- a/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_mergegen.h
+++ b/./output/grpc/67667eb7-3c1cb4f3-249c2614/include@grpcpp@impl@codegen@completion_queue.no_comments_merged.h
@@ -124 +124,2 @@ class CompletionQueue : private ::grpc::GrpcLibraryCodegen {
-template <::grpc::StatusCode code> friend class ::grpc::internal::ErrorMethodHandler; friend class ::grpc::ServerContextBase; friend class ::grpc::ServerInterface;
+  friend class ::grpc::internal::ErrorMethodHandler;
+  friend class ::grpc::ServerContextBase;
