diff --git a/./output/grpc/67667eb7-3c1cb4f3-249c2614/src@compiler@cpp_generator.no_comments_mergegen.cc b/./output/grpc/67667eb7-3c1cb4f3-249c2614/src@compiler@cpp_generator.no_comments_merged.cc
index 7a3056f..5ea6384 100644
--- a/./output/grpc/67667eb7-3c1cb4f3-249c2614/src@compiler@cpp_generator.no_comments_mergegen.cc
+++ b/./output/grpc/67667eb7-3c1cb4f3-249c2614/src@compiler@cpp_generator.no_comments_merged.cc
@@ -1250 +1250,2 @@ void PrintHeaderServerMethodStreamedUnary(
-" [this](::grpc::ServerContext* context,\n" " ::grpc::ServerUnaryStreamer<\n" " $Request$, $Response$>* streamer) {\n"
+                   "        [this](::grpc::ServerContext* context,\n"
+                   "               ::grpc::ServerUnaryStreamer<\n"
@@ -1302 +1303,2 @@ void PrintHeaderServerMethodSplitStreaming(
-" [this](::grpc::ServerContext* context,\n" " ::grpc::ServerSplitStreamer<\n" " $Request$, $Response$>* streamer) {\n"
+                   "        [this](::grpc::ServerContext* context,\n"
+                   "               ::grpc::ServerSplitStreamer<\n"
@@ -1891,2 +1893 @@ void PrintSourceClientMethod(grpc_generator::Printer* printer,
-      printer->Print(
-          *vars,
+      printer->Print(*vars,
@@ -1918,2 +1919 @@ void PrintSourceClientMethod(grpc_generator::Printer* printer,
-    printer->Print(
-        *vars,
+    printer->Print(*vars,
@@ -2090 +2090,2 @@ void PrintSourceService(grpc_generator::Printer* printer,
-" ::grpc::ServerContext* ctx,\n" " ::grpc::ServerReader<$Request$>* reader,\n" " $Response$* resp) {\n"
+          "           ::grpc::ServerContext* ctx,\n"
+          "           ::grpc::ServerReader<$Request$>* reader,\n"
@@ -2116 +2117,2 @@ void PrintSourceService(grpc_generator::Printer* printer,
-" ::grpc::ServerContext* ctx,\n" " ::grpc::ServerReaderWriter<$Response$,\n" " $Request$>* stream) {\n"
+                     "           ::grpc::ServerContext* ctx,\n"
+                     "           ::grpc::ServerReaderWriter<$Response$,\n"
