--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/bc2e0cf7-392b381d-6f03e49e/server-client.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/bc2e0cf7-392b381d-6f03e49e/server-client.no_comments_merged.c
@@ -7,2 +6,0 @@
-#include <imsg.h>
-#include <paths.h>
@@ -41 +46,4 @@ static int server_client_window_cmp(struct client_window *cw1,
-u_int server_client_how_many(void) {
+RB_GENERATE(client_windows, client_window, entry, server_client_window_cmp);
+u_int
+server_client_how_many(void)
+{
@@ -1421 +1758,0 @@ static void server_client_dispatch_identify(struct client *c,
-<<<<<<< HEAD
@@ -1425,8 +1761,0 @@ static void server_client_dispatch_identify(struct client *c,
-  if (c->flags & CLIENT_CONTROL) {
-    close(c->fd);
-    c->fd = -1;
-|||||||
-  if (c->flags & CLIENT_CONTROL) {
-    close(c->fd);
-    c->fd = -1;
-=======
@@ -1434 +1762,0 @@ static void server_client_dispatch_identify(struct client *c,
->>>>>>> 392b381d1cec6d63c4baaa709243f760ff6c3403
