diff --git a/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@objective-c@tests@version.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@objective-c@tests@version.no_comments_merged.h
index a48c0e4..8e5479d 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@objective-c@tests@version.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/fc9e5092-7d89dbb3-e3c57771/src@objective-c@tests@version.no_comments_merged.h
@@ -1 +0,0 @@
-#define GRPC_OBJC_VERSION_STRING @"1.29.0-dev"
@@ -3 +1,0 @@
-#define GRPC_OBJC_VERSION_STRING @"1.29.1"
