--- a/./output/art/9eadb229-b5a9e3a4-fbd7b4da/runtime@runtime.no_comments_mergegen.h
+++ b/./output/art/9eadb229-b5a9e3a4-fbd7b4da/runtime@runtime.no_comments_merged.h
@@ -802,0 +803,10 @@ class Runtime {
+  static std::string GetApexVersions(ArrayRef<const std::string> boot_class_path_locations);
+  bool AllowInMemoryCompilation() const { return allow_in_memory_compilation_; }
+  void SetOutOfMemoryErrorHook(void (*hook)()) {
+    out_of_memory_error_hook_ = hook;
+  }
+  void OutOfMemoryErrorHook() {
+    if (out_of_memory_error_hook_ != nullptr) {
+      out_of_memory_error_hook_();
+    }
+  }
