--- a/./output/tmux/52e3d960-c20eb0c0-26f5dfbe/tmux.no_comments_conflict.c
+++ b/./output/tmux/52e3d960-c20eb0c0-26f5dfbe/tmux.no_comments_merged.c
@@ -164 +163,0 @@ getversion(void)
-<<<<<<< HEAD
@@ -166,19 +164,0 @@ getversion(void)
-||||||| 26f5dfbe
- static char *version;
- struct utsname u;
- if (version == NULL) {
-  if (uname(&u) < 0)
-   fatalx("uname failed");
-  xasprintf(&version, "openbsd-%s", u.release);
- }
- return version;
-=======
- static char *version;
- struct utsname u;
- if (version == NULL) {
-  if (uname(&u) < 0)
-   fatalx("uname failed");
-  xasprintf(&version, "openbsd-%s", u.release);
- }
- return (version);
->>>>>>> c20eb0c0
