--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/52e3d960-c20eb0c0-26f5dfbe/tmux.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/52e3d960-c20eb0c0-26f5dfbe/tmux.no_comments_merged.c
@@ -4 +3,0 @@
-#include <err.h>
@@ -8 +6,0 @@
-#include <paths.h>
@@ -12 +9,0 @@
-#include <util.h>
@@ -132,2 +161,3 @@ const char *find_home(void) {
-const char *getversion(void) {
-<<<<<<< HEAD
+const char *
+getversion(void)
+{
@@ -135,14 +164,0 @@ const char *getversion(void) {
-|||||||
-  static char *version;
-  struct utsname u;
-  if (version == NULL) {
-    if (uname(&u) < 0) fatalx("uname failed");
-    xasprintf(&version, "openbsd-%s", u.release);
-  }
-  return version;
-=======
-  static char *version;
-  struct utsname u;
-  if (version == NULL) {
-    if (uname(&u) < 0) fatalx("uname failed");
-    xasprintf(&version, "openbsd-%s", u.release);
@@ -150,4 +166,3 @@ const char *getversion(void) {
-  return (version);
->>>>>>> c20eb0c0ae3347c768894a6355adfd7ebae6f2f3
-}
-int main(int argc, char **argv) {
+int
+main(int argc, char **argv)
+{
