diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/fefde6e3-be899b82-f847dd3a/src@debug.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/fefde6e3-be899b82-f847dd3a/src@debug.no_comments_merged.c
index 33bf260..a41204b 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/fefde6e3-be899b82-f847dd3a/src@debug.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/fefde6e3-be899b82-f847dd3a/src@debug.no_comments_merged.c
@@ -26 +26,2 @@ void xorDigest(unsigned char *digest, void *ptr, size_t len) {
-  for (j = 0; j < 20; j++) digest[j] ^= hash[j];
+    for (j = 0; j < 20; j++)
+        digest[j] ^= hash[j];
@@ -163 +164,2 @@ void computeDatasetDigest(unsigned char *final) {
-            streamIteratorGetField(&si, &field, &value, &field_len, &value_len);
+                        streamIteratorGetField(&si,&field,&value,
+                                                   &field_len,&value_len);
@@ -191 +192,0 @@ void debugCommand(client *c) {
-<<<<<<< HEAD
@@ -193,4 +194,2 @@ void debugCommand(client *c) {
-        "CHANGE-REPL-ID -- Change the replication IDs of the instance. "
-        "Dangerous, should be used only for testing the replication subsystem.",
-        "CRASH-and-recover <milliseconds> -- Hard crash and restart after "
-        "<milliseconds> delay.",
+"CHANGE-REPL-ID -- Change the replication IDs of the instance. Dangerous, should be used only for testing the replication subsystem.",
+"CRASH-AND-RECOVER <milliseconds> -- Hard crash and restart after <milliseconds> delay.",
@@ -198,9 +197,5 @@ void debugCommand(client *c) {
-        "ERROR <string> -- Return a Redis protocol error with <string> as "
-        "message. Useful for clients unit tests to simulate Redis errors.",
-        "HTSTATS <dbid> -- Return hash table statistics of the specified Redis "
-        "database.",
-        "LOADAOF -- Flush the AOF buffers on disk and reload the AOF in "
-        "memory.",
-        "LUA-ALWAYS-REPLICATE-COMMANDS (0|1) -- Setting it to 1 makes Lua "
-        "replication defaulting to replicating single commands, without the "
-        "script having to enable effects replication.",
+"ERROR <string> -- Return a Redis protocol error with <string> as message. Useful for clients unit tests to simulate Redis errors.",
+"HTSTATS <dbid> -- Return hash table statistics of the specified Redis database.",
+"HTSTATS-KEY <key> -- Like htstats but for the hash table stored as key's value.",
+"LOADAOF -- Flush the AOF buffers on disk and reload the AOF in memory.",
+"LUA-ALWAYS-REPLICATE-COMMANDS <0|1> -- Setting it to 1 makes Lua replication defaulting to replicating single commands, without the script having to enable effects replication.",
@@ -209,3 +204 @@ void debugCommand(client *c) {
-        "POPULATE <count> [prefix] [size] -- Create <count> string keys named "
-        "key:<num>. If a prefix is specified is used instead of the 'key' "
-        "prefix.",
+"POPULATE <count> [prefix] [size] -- Create <count> string keys named key:<num>. If a prefix is specified is used instead of the 'key' prefix.",
@@ -214,2 +207 @@ void debugCommand(client *c) {
-        "SDSLEN <key> -- Show low level SDS string info representing key and "
-        "value.",
+"SDSLEN <key> -- Show low level SDS string info representing key and value.",
@@ -217,3 +209 @@ void debugCommand(client *c) {
-        "SET-ACTIVE-EXPIRE (0|1) -- Setting it to 0 disables expiring keys in "
-        "background when they are not accessed (otherwise the Redis behavior). "
-        "Setting it to 1 reenables back the default.",
+"SET-ACTIVE-EXPIRE <0|1> -- Setting it to 0 disables expiring keys in background when they are not accessed (otherwise the Redis behavior). Setting it to 1 reenables back the default.",
@@ -223,70 +213,2 @@ void debugCommand(client *c) {
-|||||||
-        "assert -- Crash by assertion failed.",
-        "change-repl-id -- Change the replication IDs of the instance. "
-        "Dangerous, should be used only for testing the replication subsystem.",
-        "crash-and-recover <milliseconds> -- Hard crash and restart after "
-        "<milliseconds> delay.",
-        "digest -- Outputs an hex signature representing the current DB "
-        "content.",
-        "htstats <dbid> -- Return hash table statistics of the specified Redis "
-        "database.",
-        "loadaof -- Flush the AOF buffers on disk and reload the AOF in "
-        "memory.",
-        "lua-always-replicate-commands (0|1) -- Setting it to 1 makes Lua "
-        "replication defaulting to replicating single commands, without the "
-        "script having to enable effects replication.",
-        "object <key> -- Show low level info about key and associated value.",
-        "panic -- Crash the server simulating a panic.",
-        "populate <count> [prefix] [size] -- Create <count> string keys named "
-        "key:<num>. If a prefix is specified is used instead of the 'key' "
-        "prefix.",
-        "reload -- Save the RDB on disk and reload it back in memory.",
-        "restart -- Graceful restart: save config, db, restart.",
-        "sdslen <key> -- Show low level SDS string info representing key and "
-        "value.",
-        "segfault -- Crash the server with sigsegv.",
-        "set-active-expire (0|1) -- Setting it to 0 disables expiring keys in "
-        "background when they are not accessed (otherwise the Redis behavior). "
-        "Setting it to 1 reenables back the default.",
-        "sleep <seconds> -- Stop the server for <seconds>. Decimals allowed.",
-        "structsize -- Return the size of different Redis core C structures.",
-        "ziplist <key> -- Show low level info about the ziplist encoding.",
-        "error <string> -- Return a Redis protocol error with <string> as "
-        "message. Useful for clients unit tests to simulate Redis errors.",
-=======
-        "assert -- Crash by assertion failed.",
-        "change-repl-id -- Change the replication IDs of the instance. "
-        "Dangerous, should be used only for testing the replication subsystem.",
-        "crash-and-recover <milliseconds> -- Hard crash and restart after "
-        "<milliseconds> delay.",
-        "digest -- Outputs an hex signature representing the current DB "
-        "content.",
-        "htstats <dbid> -- Return hash table statistics of the specified Redis "
-        "database.",
-        "htstats-key <key> -- Like htstats but for the hash table stored as "
-        "key's value.",
-        "loadaof -- Flush the AOF buffers on disk and reload the AOF in "
-        "memory.",
-        "lua-always-replicate-commands (0|1) -- Setting it to 1 makes Lua "
-        "replication defaulting to replicating single commands, without the "
-        "script having to enable effects replication.",
-        "object <key> -- Show low level info about key and associated value.",
-        "panic -- Crash the server simulating a panic.",
-        "populate <count> [prefix] [size] -- Create <count> string keys named "
-        "key:<num>. If a prefix is specified is used instead of the 'key' "
-        "prefix.",
-        "reload -- Save the RDB on disk and reload it back in memory.",
-        "restart -- Graceful restart: save config, db, restart.",
-        "sdslen <key> -- Show low level SDS string info representing key and "
-        "value.",
-        "segfault -- Crash the server with sigsegv.",
-        "set-active-expire (0|1) -- Setting it to 0 disables expiring keys in "
-        "background when they are not accessed (otherwise the Redis behavior). "
-        "Setting it to 1 reenables back the default.",
-        "sleep <seconds> -- Stop the server for <seconds>. Decimals allowed.",
-        "structsize -- Return the size of different Redis core C structures.",
-        "ziplist <key> -- Show low level info about the ziplist encoding.",
-        "error <string> -- Return a Redis protocol error with <string> as "
-        "message. Useful for clients unit tests to simulate Redis errors.",
->>>>>>> be899b824edd312d4e3a1998c56626d66fae3b61
-        NULL};
+NULL
+        };
@@ -299 +221,2 @@ void debugCommand(client *c) {
-             !strcasecmp(c->argv[1]->ptr, "crash-and-recover")) {
+               !strcasecmp(c->argv[1]->ptr,"crash-and-recover"))
+    {
@@ -302,2 +225,2 @@ void debugCommand(client *c) {
-      if (getLongLongFromObjectOrReply(c, c->argv[2], &delay, NULL) != C_OK)
-        return;
+            if (getLongLongFromObjectOrReply(c, c->argv[2], &delay, NULL)
+                != C_OK) return;
@@ -306,4 +229,3 @@ void debugCommand(client *c) {
-    int flags =
-        !strcasecmp(c->argv[1]->ptr, "restart")
-            ? (RESTART_SERVER_GRACEFULLY | RESTART_SERVER_CONFIG_REWRITE)
-            : RESTART_SERVER_NONE;
+        int flags = !strcasecmp(c->argv[1]->ptr,"restart") ?
+            (RESTART_SERVER_GRACEFULLY|RESTART_SERVER_CONFIG_REWRITE) :
+             RESTART_SERVER_NONE;
@@ -384,3 +306,3 @@ void debugCommand(client *c) {
-                         (void *)val, val->refcount, strenc,
-                         rdbSavedObjectLen(val), val->lru,
-                         estimateObjectIdleTime(val) / 1000, extra);
+            (void*)val, val->refcount,
+            strenc, rdbSavedObjectLen(val),
+            val->lru, estimateObjectIdleTime(val)/1000, extra);
@@ -400,2 +322 @@ void debugCommand(client *c) {
-      addReplyStatusFormat(
-          c,
+            addReplyStatusFormat(c,
@@ -404,2 +325,4 @@ void debugCommand(client *c) {
-          (long long)sdslen(key), (long long)sdsavail(key),
-          (long long)sdsZmallocSize(key), (long long)sdslen(val->ptr),
+                (long long) sdslen(key),
+                (long long) sdsavail(key),
+                (long long) sdsZmallocSize(key),
+                (long long) sdslen(val->ptr),
@@ -411,3 +334,2 @@ void debugCommand(client *c) {
-    if ((o = objectCommandLookupOrReply(c, c->argv[2], shared.nokeyerr)) ==
-        NULL)
-      return;
+        if ((o = objectCommandLookupOrReply(c,c->argv[2],shared.nokeyerr))
+                == NULL) return;
@@ -420,2 +342,2 @@ void debugCommand(client *c) {
-  } else if (!strcasecmp(c->argv[1]->ptr, "populate") && c->argc >= 3 &&
-             c->argc <= 5) {
+    } else if (!strcasecmp(c->argv[1]->ptr,"populate") &&
+               c->argc >= 3 && c->argc <= 5) {
@@ -425 +347,2 @@ void debugCommand(client *c) {
-    if (getLongFromObjectOrReply(c, c->argv[2], &keys, NULL) != C_OK) return;
+        if (getLongFromObjectOrReply(c, c->argv[2], &keys, NULL) != C_OK)
+            return;
@@ -457 +380,2 @@ void debugCommand(client *c) {
-    for (j = 0; j < 20; j++) d = sdscatprintf(d, "%02x", digest[j]);
+        for (j = 0; j < 20; j++)
+            d = sdscatprintf(d, "%02x",digest[j]);
@@ -469 +393,2 @@ void debugCommand(client *c) {
-             c->argc == 3) {
+               c->argc == 3)
+    {
@@ -473 +398,2 @@ void debugCommand(client *c) {
-             c->argc == 3) {
+               c->argc == 3)
+    {
@@ -497 +423,2 @@ void debugCommand(client *c) {
-    if (getLongFromObjectOrReply(c, c->argv[2], &dbid, NULL) != C_OK) return;
+        if (getLongFromObjectOrReply(c, c->argv[2], &dbid, NULL) != C_OK)
+            return;
@@ -512,3 +439,2 @@ void debugCommand(client *c) {
-    if ((o = objectCommandLookupOrReply(c, c->argv[2], shared.nokeyerr)) ==
-        NULL)
-      return;
+        if ((o = objectCommandLookupOrReply(c,c->argv[2],shared.nokeyerr))
+                == NULL) return;
@@ -516 +442,2 @@ void debugCommand(client *c) {
-      case OBJ_ENCODING_SKIPLIST: {
+        case OBJ_ENCODING_SKIPLIST:
+            {
@@ -519 +446,2 @@ void debugCommand(client *c) {
-      } break;
+            }
+            break;
@@ -525,2 +453 @@ void debugCommand(client *c) {
-      addReplyError(c,
-                    "The value stored at the specified key is not "
+            addReplyError(c,"The value stored at the specified key is not "
@@ -534,2 +461 @@ void debugCommand(client *c) {
-    serverLog(LL_WARNING,
-              "Changing replication IDs after receiving DEBUG change-repl-id");
+        serverLog(LL_WARNING,"Changing replication IDs after receiving DEBUG change-repl-id");
@@ -573,2 +499,2 @@ void _serverAssertPrintClientInfo(const client *c) {
-    serverLog(LL_WARNING, "client->argv[%d] = \"%s\" (refcount: %d)", j, arg,
-              c->argv[j]->refcount);
+        serverLog(LL_WARNING,"client->argv[%d] = \"%s\" (refcount: %d)",
+            j, arg, c->argv[j]->refcount);
@@ -597,2 +523 @@ void serverLogObjectDebugInfo(const robj *o) {
-      serverLog(LL_WARNING, "Skiplist level: %d",
-                (int)((const zset *)o->ptr)->zsl->level);
+            serverLog(LL_WARNING,"Skiplist level: %d", (int) ((const zset*)o->ptr)->zsl->level);
@@ -606,2 +531 @@ void _serverAssertPrintObject(const robj *o) {
-void _serverAssertWithInfo(const client *c, const robj *o, const char *estr,
-                           const char *file, int line) {
+void _serverAssertWithInfo(const client *c, const robj *o, const char *estr, const char *file, int line) {
@@ -620,2 +544 @@ void _serverPanic(const char *file, int line, const char *msg, ...) {
-  serverLog(LL_WARNING,
-            "!!! Software Failure. Press left mouse button to continue");
+    serverLog(LL_WARNING,"!!! Software Failure. Press left mouse button to continue");
@@ -631,2 +554 @@ void bugReportStart(void) {
-    serverLogRaw(
-        LL_WARNING | LL_RAW,
+        serverLogRaw(LL_WARNING|LL_RAW,
@@ -711 +633,2 @@ void logRegisters(ucontext_t *uc) {
-            (unsigned long)uc->uc_mcontext->__ss.__gs);
+        (unsigned long) uc->uc_mcontext->__ss.__gs
+    );
@@ -735 +658,2 @@ void logRegisters(ucontext_t *uc) {
-            (unsigned long)uc->uc_mcontext->__ss.__gs);
+        (unsigned long) uc->uc_mcontext->__ss.__gs
+    );
@@ -761 +685,2 @@ void logRegisters(ucontext_t *uc) {
-            (unsigned long)uc->uc_mcontext.gregs[0]);
+        (unsigned long) uc->uc_mcontext.gregs[0]
+    );
@@ -789 +714,2 @@ void logRegisters(ucontext_t *uc) {
-            (unsigned long)uc->uc_mcontext.gregs[18]);
+        (unsigned long) uc->uc_mcontext.gregs[18]
+    );
@@ -799,3 +725,3 @@ int openDirectLogFiledes(void) {
-  int fd = log_to_stdout
-               ? STDOUT_FILENO
-               : open(server.logfile, O_APPEND | O_CREAT | O_WRONLY, 0644);
+    int fd = log_to_stdout ?
+        STDOUT_FILENO :
+        open(server.logfile, O_APPEND|O_CREAT|O_WRONLY, 0644);
@@ -816,2 +742 @@ void logStackTrace(ucontext_t *uc) {
-    if (write(fd, msg1, strlen(msg1)) == -1) {
-    };
+        if (write(fd,msg1,strlen(msg1)) == -1) { };
@@ -820,2 +745 @@ void logStackTrace(ucontext_t *uc) {
-    if (write(fd, msg2, strlen(msg2)) == -1) {
-    };
+        if (write(fd,msg2,strlen(msg2)) == -1) { };
@@ -838 +762,2 @@ void logCurrentClient(void) {
-    serverLog(LL_WARNING | LL_RAW, "argv[%d]: '%s'\n", j, (char *)decoded->ptr);
+        serverLog(LL_WARNING|LL_RAW,"argv[%d]: '%s'\n", j,
+            (char*)decoded->ptr);
@@ -848,3 +773 @@ void logCurrentClient(void) {
-      serverLog(LL_WARNING,
-                "key '%s' found in DB containing the following object:",
-                (char *)key->ptr);
+            serverLog(LL_WARNING,"key '%s' found in DB containing the following object:", (char*)key->ptr);
@@ -880,2 +803,3 @@ int memtest_test_linux_anonymous_maps(void) {
-    if (strstr(p, "stack") || strstr(p, "vdso") || strstr(p, "vsyscall"))
-      continue;
+        if (strstr(p,"stack") ||
+            strstr(p,"vdso") ||
+            strstr(p,"vsyscall")) continue;
@@ -893,2 +817 @@ int memtest_test_linux_anonymous_maps(void) {
-    if (write(fd, logbuf, strlen(logbuf)) == -1) {
-    }
+        if (write(fd,logbuf,strlen(logbuf)) == -1) { }
@@ -899,2 +822 @@ int memtest_test_linux_anonymous_maps(void) {
-    if (write(fd, ".", 1) == -1) {
-    }
+        if (write(fd,".",1) == -1) { }
@@ -902,4 +824 @@ int memtest_test_linux_anonymous_maps(void) {
-    if (write(fd, errors ? "E" : "O", 1) == -1) {
-    }
-  }
-  if (write(fd, "\n", 1) == -1) {
+        if (write(fd, errors ? "E" : "O",1) == -1) { }
@@ -906,0 +826 @@ int memtest_test_linux_anonymous_maps(void) {
+    if (write(fd,"\n",1) == -1) { }
@@ -938 +858,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-  serverLog(LL_WARNING, "Redis %s crashed by signal: %d", REDIS_VERSION, sig);
+    serverLog(LL_WARNING,
+        "Redis %s crashed by signal: %d", REDIS_VERSION, sig);
@@ -940 +861,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-    serverLog(LL_WARNING, "Crashed running the instruction at: %p", eip);
+        serverLog(LL_WARNING,
+        "Crashed running the instruction at: %p", eip);
@@ -943 +865,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-    serverLog(LL_WARNING, "Accessing address: %p", (void *)info->si_addr);
+        serverLog(LL_WARNING,
+        "Accessing address: %p", (void*)info->si_addr);
@@ -945 +868,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-  serverLog(LL_WARNING, "Failed assertion: %s (%s:%d)", server.assert_failed,
+    serverLog(LL_WARNING,
+        "Failed assertion: %s (%s:%d)", server.assert_failed,
@@ -966,4 +890,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-    serverLogRaw(
-        LL_WARNING | LL_RAW,
-        "Fast memory test PASSED, however your memory can still be broken. "
-        "Please run a memory test for several hours if possible.\n");
+        serverLogRaw(LL_WARNING|LL_RAW,
+            "Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.\n");
@@ -980,2 +902 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-                "$ objdump --adjust-vma=%p -D -b binary -m i386:x86-64 "
-                "/tmp/dump.bin\n"
+                "$ objdump --adjust-vma=%p -D -b binary -m i386:x86-64 /tmp/dump.bin\n"
@@ -992 +913,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-        serverLogHexDump(LL_WARNING, "dump of function", info.dli_saddr, len);
+                serverLogHexDump(LL_WARNING, "dump of function",
+                    info.dli_saddr ,len);
@@ -997,4 +919,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-  serverLogRaw(
-      LL_WARNING | LL_RAW,
-      "\n=== REDIS BUG REPORT END. Make sure to include from START to END. "
-      "===\n\n"
+    serverLogRaw(LL_WARNING|LL_RAW,
+"\n=== REDIS BUG REPORT END. Make sure to include from START to END. ===\n\n"
@@ -1003 +923,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-      "  Suspect RAM error? Use redis-server --test-memory to verify it.\n\n");
+"  Suspect RAM error? Use redis-server --test-memory to verify it.\n\n"
+);
@@ -1031,0 +953 @@ void serverLogHexDump(int level, char *descr, void *value, size_t len) {
+#include <sys/time.h>
