--- a/home/whalien/codebase/python/mergebot-eva/output/redis/fefde6e3-be899b82-f847dd3a/src@debug.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/fefde6e3-be899b82-f847dd3a/src@debug.no_comments_merged.c
@@ -191 +192,0 @@ void debugCommand(client *c) {
-<<<<<<< HEAD
@@ -193,4 +194,2 @@ void debugCommand(client *c) {
-        "CHANGE-REPL-ID -- Change the replication IDs of the instance. "
-        "Dangerous, should be used only for testing the replication subsystem.",
-        "CRASH-and-recover <milliseconds> -- Hard crash and restart after "
-        "<milliseconds> delay.",
+"CHANGE-REPL-ID -- Change the replication IDs of the instance. Dangerous, should be used only for testing the replication subsystem.",
+"CRASH-AND-RECOVER <milliseconds> -- Hard crash and restart after <milliseconds> delay.",
@@ -198,9 +197,5 @@ void debugCommand(client *c) {
-        "ERROR <string> -- Return a Redis protocol error with <string> as "
-        "message. Useful for clients unit tests to simulate Redis errors.",
-        "HTSTATS <dbid> -- Return hash table statistics of the specified Redis "
-        "database.",
-        "LOADAOF -- Flush the AOF buffers on disk and reload the AOF in "
-        "memory.",
-        "LUA-ALWAYS-REPLICATE-COMMANDS (0|1) -- Setting it to 1 makes Lua "
-        "replication defaulting to replicating single commands, without the "
-        "script having to enable effects replication.",
+"ERROR <string> -- Return a Redis protocol error with <string> as message. Useful for clients unit tests to simulate Redis errors.",
+"HTSTATS <dbid> -- Return hash table statistics of the specified Redis database.",
+"HTSTATS-KEY <key> -- Like htstats but for the hash table stored as key's value.",
+"LOADAOF -- Flush the AOF buffers on disk and reload the AOF in memory.",
+"LUA-ALWAYS-REPLICATE-COMMANDS <0|1> -- Setting it to 1 makes Lua replication defaulting to replicating single commands, without the script having to enable effects replication.",
@@ -209,3 +204 @@ void debugCommand(client *c) {
-        "POPULATE <count> [prefix] [size] -- Create <count> string keys named "
-        "key:<num>. If a prefix is specified is used instead of the 'key' "
-        "prefix.",
+"POPULATE <count> [prefix] [size] -- Create <count> string keys named key:<num>. If a prefix is specified is used instead of the 'key' prefix.",
@@ -214,2 +207 @@ void debugCommand(client *c) {
-        "SDSLEN <key> -- Show low level SDS string info representing key and "
-        "value.",
+"SDSLEN <key> -- Show low level SDS string info representing key and value.",
@@ -217,3 +209 @@ void debugCommand(client *c) {
-        "SET-ACTIVE-EXPIRE (0|1) -- Setting it to 0 disables expiring keys in "
-        "background when they are not accessed (otherwise the Redis behavior). "
-        "Setting it to 1 reenables back the default.",
+"SET-ACTIVE-EXPIRE <0|1> -- Setting it to 0 disables expiring keys in background when they are not accessed (otherwise the Redis behavior). Setting it to 1 reenables back the default.",
@@ -223,70 +213,2 @@ void debugCommand(client *c) {
-|||||||
-        "assert -- Crash by assertion failed.",
-        "change-repl-id -- Change the replication IDs of the instance. "
-        "Dangerous, should be used only for testing the replication subsystem.",
-        "crash-and-recover <milliseconds> -- Hard crash and restart after "
-        "<milliseconds> delay.",
-        "digest -- Outputs an hex signature representing the current DB "
-        "content.",
-        "htstats <dbid> -- Return hash table statistics of the specified Redis "
-        "database.",
-        "loadaof -- Flush the AOF buffers on disk and reload the AOF in "
-        "memory.",
-        "lua-always-replicate-commands (0|1) -- Setting it to 1 makes Lua "
-        "replication defaulting to replicating single commands, without the "
-        "script having to enable effects replication.",
-        "object <key> -- Show low level info about key and associated value.",
-        "panic -- Crash the server simulating a panic.",
-        "populate <count> [prefix] [size] -- Create <count> string keys named "
-        "key:<num>. If a prefix is specified is used instead of the 'key' "
-        "prefix.",
-        "reload -- Save the RDB on disk and reload it back in memory.",
-        "restart -- Graceful restart: save config, db, restart.",
-        "sdslen <key> -- Show low level SDS string info representing key and "
-        "value.",
-        "segfault -- Crash the server with sigsegv.",
-        "set-active-expire (0|1) -- Setting it to 0 disables expiring keys in "
-        "background when they are not accessed (otherwise the Redis behavior). "
-        "Setting it to 1 reenables back the default.",
-        "sleep <seconds> -- Stop the server for <seconds>. Decimals allowed.",
-        "structsize -- Return the size of different Redis core C structures.",
-        "ziplist <key> -- Show low level info about the ziplist encoding.",
-        "error <string> -- Return a Redis protocol error with <string> as "
-        "message. Useful for clients unit tests to simulate Redis errors.",
-=======
-        "assert -- Crash by assertion failed.",
-        "change-repl-id -- Change the replication IDs of the instance. "
-        "Dangerous, should be used only for testing the replication subsystem.",
-        "crash-and-recover <milliseconds> -- Hard crash and restart after "
-        "<milliseconds> delay.",
-        "digest -- Outputs an hex signature representing the current DB "
-        "content.",
-        "htstats <dbid> -- Return hash table statistics of the specified Redis "
-        "database.",
-        "htstats-key <key> -- Like htstats but for the hash table stored as "
-        "key's value.",
-        "loadaof -- Flush the AOF buffers on disk and reload the AOF in "
-        "memory.",
-        "lua-always-replicate-commands (0|1) -- Setting it to 1 makes Lua "
-        "replication defaulting to replicating single commands, without the "
-        "script having to enable effects replication.",
-        "object <key> -- Show low level info about key and associated value.",
-        "panic -- Crash the server simulating a panic.",
-        "populate <count> [prefix] [size] -- Create <count> string keys named "
-        "key:<num>. If a prefix is specified is used instead of the 'key' "
-        "prefix.",
-        "reload -- Save the RDB on disk and reload it back in memory.",
-        "restart -- Graceful restart: save config, db, restart.",
-        "sdslen <key> -- Show low level SDS string info representing key and "
-        "value.",
-        "segfault -- Crash the server with sigsegv.",
-        "set-active-expire (0|1) -- Setting it to 0 disables expiring keys in "
-        "background when they are not accessed (otherwise the Redis behavior). "
-        "Setting it to 1 reenables back the default.",
-        "sleep <seconds> -- Stop the server for <seconds>. Decimals allowed.",
-        "structsize -- Return the size of different Redis core C structures.",
-        "ziplist <key> -- Show low level info about the ziplist encoding.",
-        "error <string> -- Return a Redis protocol error with <string> as "
-        "message. Useful for clients unit tests to simulate Redis errors.",
->>>>>>> be899b824edd312d4e3a1998c56626d66fae3b61
-        NULL};
+NULL
+        };
@@ -902,4 +824 @@ int memtest_test_linux_anonymous_maps(void) {
-    if (write(fd, errors ? "E" : "O", 1) == -1) {
-    }
-  }
-  if (write(fd, "\n", 1) == -1) {
+        if (write(fd, errors ? "E" : "O",1) == -1) { }
@@ -906,0 +826 @@ int memtest_test_linux_anonymous_maps(void) {
+    if (write(fd,"\n",1) == -1) { }
@@ -966,4 +890,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-    serverLogRaw(
-        LL_WARNING | LL_RAW,
-        "Fast memory test PASSED, however your memory can still be broken. "
-        "Please run a memory test for several hours if possible.\n");
+        serverLogRaw(LL_WARNING|LL_RAW,
+            "Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.\n");
@@ -980,2 +902 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-                "$ objdump --adjust-vma=%p -D -b binary -m i386:x86-64 "
-                "/tmp/dump.bin\n"
+                "$ objdump --adjust-vma=%p -D -b binary -m i386:x86-64 /tmp/dump.bin\n"
@@ -997,4 +919,2 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
-  serverLogRaw(
-      LL_WARNING | LL_RAW,
-      "\n=== REDIS BUG REPORT END. Make sure to include from START to END. "
-      "===\n\n"
+    serverLogRaw(LL_WARNING|LL_RAW,
+"\n=== REDIS BUG REPORT END. Make sure to include from START to END. ===\n\n"
@@ -1031,0 +953 @@ void serverLogHexDump(int level, char *descr, void *value, size_t len) {
+#include <sys/time.h>
