diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_merged.h
index 70fa2fb..04089fa 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/libs@renderengine@skia@SkiaGLRenderEngine.no_comments_merged.h
@@ -24 +24 @@
-class SkData
+class SkData;
@@ -30,7 +29,0 @@ class SkiaGLRenderEngine : public skia::SkiaRenderEngine {
-protected:
-    virtual SkiaRenderEngine::Contexts createDirectContexts(const GrContextOptions& options);
-    bool supportsProtectedContentImpl() const override;
-    bool useProtectedContextImpl(GrProtected isProtected) override;
-    void waitFence(GrDirectContext* grContext, base::borrowed_fd fenceFd) override;
-    base::unique_fd flushAndSubmit(GrDirectContext* context) override;
-    void appendBackendSpecificInfoToDump(std::string& result) override;
@@ -43,0 +37,7 @@ public:
+protected:
+    virtual SkiaRenderEngine::Contexts createDirectContexts(const GrContextOptions& options);
+    bool supportsProtectedContentImpl() const override;
+    bool useProtectedContextImpl(GrProtected isProtected) override;
+    void waitFence(GrDirectContext* grContext, base::borrowed_fd fenceFd) override;
+    base::unique_fd flushAndSubmit(GrDirectContext* context) override;
+    void appendBackendSpecificInfoToDump(std::string& result) override;
@@ -44,0 +45,2 @@ private:
+    bool waitGpuFence(base::borrowed_fd fenceFd);
+    base::unique_fd flush();
@@ -52,4 +54,2 @@ private:
-    static EGLSurface createPlaceholderEglPbufferSurface(EGLDisplay display, EGLConfig config,
-                                                         int hwcFormat, Protection protection);
-    base::unique_fd flush();
-    bool waitGpuFence(base::borrowed_fd fenceFd);
+    static EGLSurface createPlaceholderEglPbufferSurface(
+            EGLDisplay display, EGLConfig config, int hwcFormat, Protection protection);
