--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/99b1002b-b22850c1-b16c01a3/services@surfaceflinger@Layer.no_comments_merged.h
@@ -5 +4,0 @@
-#include <gui/ISurfaceComposerClient.h>
@@ -31,2 +29,0 @@
-#include "ClientCache.h"
-#include "DisplayHardware/ComposerHal.h"
@@ -34,2 +30,0 @@
-#include "MonitoredProducer.h"
-#include "RenderArea.h"
@@ -53 +47,0 @@ class SurfaceFlinger;
-class LayerDebugInfo;
@@ -58 +52,3 @@ struct LayerFECompositionState;
-namespace gui {}
+namespace gui {
+class LayerDebugInfo;
+}
@@ -67,2 +62,0 @@ class Layer : public virtual RefBase {
-private:
-    friend class SlotGenerationTest;
@@ -162 +155,0 @@ public:
-    ;
@@ -179 +172 @@ public:
-    virtual bool isDimmingEnabled() const { return getDrawingState().dimmingEnabled; }
+    virtual bool isDimmingEnabled() const { return getDrawingState().dimmingEnabled; };
@@ -191 +183,0 @@ public:
-    ;
@@ -246,4 +238 @@ public:
-    bool hasRoundedCorners() const {
-        return getRoundedCornerState().hasRoundedCorners() {
-            return getRoundedCornerState().hasRoundedCorners();
-        }
+    bool hasRoundedCorners() const { return getRoundedCornerState().hasRoundedCorners(); }
@@ -251,4 +239,0 @@ public:
-        bool hasRoundedCorners() const override {
-            return getRoundedCornerState().hasRoundedCorners() {
-                return getRoundedCornerState().hasRoundedCorners();
-            }
@@ -459,9 +401,0 @@ public:
-        private:
-            bool getAutoRefresh() const {
-                return mDrawingState.autoRefresh;
-            }
-            bool getSidebandStreamChanged() const {
-                return mSidebandStreamChanged;
-            }
-            std::atomic<bool> mSidebandStreamChanged{false};
-        protected:
@@ -534,3 +461,7 @@ public:
-            virtual bool canDrawShadows() const {
-                return true;
-            }
+    friend class SlotGenerationTest;
+    friend class TransactionFrameTracerTest;
+    friend class TransactionSurfaceFrameTest;
+    bool getAutoRefresh() const { return mDrawingState.autoRefresh; }
+    bool getSidebandStreamChanged() const { return mSidebandStreamChanged; }
+    std::atomic<bool> mSidebandStreamChanged{false};
+    virtual bool canDrawShadows() const { return true; }
@@ -575,5 +504 @@ public:
-                return ((mDrawingState.sidebandStream != nullptr) ||
-                        (mDrawingState.buffer != nullptr));
-            }
-            bool hasSomethingToDraw() const {
-                return hasEffect() || hasBufferOrSidebandStream();
+        return ((mDrawingState.sidebandStream != nullptr) || (mDrawingState.buffer != nullptr));
@@ -580,0 +506 @@ public:
+    bool hasSomethingToDraw() const { return hasEffect() || hasBufferOrSidebandStream(); }
@@ -622 +547 @@ public:
-            LayerSnapshotGuard& operator=(const LayerSnapshotGuard&) = delete
+    LayerSnapshotGuard& operator=(const LayerSnapshotGuard&) = delete;
