diff --git a/./output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_mergegen.c b/./output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_merged.c
index 0b7ee1f..933755e 100644
--- a/./output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_mergegen.c
+++ b/./output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_merged.c
@@ -2 +2,2 @@
-#include <errno.h> #include <fcntl.h>
+#include <errno.h>
+#include <fcntl.h>
