--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_merged.c
@@ -4 +3,0 @@
-#include <paths.h>
@@ -12,7 +11,4 @@ const struct cmd_entry cmd_split_window_entry = {
-    "split-window",
-    "splitw",
-    "c:dF:l:hp:Pt:v",
-    0,
-    1,
-    "[-dhvP] [-c start-directory] [-F format] [-p percentage|-l "
-    "size] " CMD_TARGET_PANE_USAGE " [command]",
+ "split-window", "splitw",
+ "c:dF:l:hp:Pt:v", 0, 1,
+ "[-dhvP] [-c start-directory] [-F format] [-p percentage|-l size] "
+ CMD_TARGET_PANE_USAGE " [command]",
