--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_merged.c
@@ -20,8 +20,0 @@ enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
-<<<<<<< HEAD
-    if (args->argc != 1) {
-      cmdq_error(cmdq, "missing key");
-      return (CMD_RETURN_ERROR);
-    }
-    return (CMD_RETURN_ERROR);
-|||||||
-=======
@@ -32 +24,0 @@ enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
->>>>>>> d518067be6220757a9101ca27fff14d5f599c410
