--- a/./output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_mergegen.c
+++ b/./output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_merged.c
@@ -21 +21,4 @@ cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq)
-if (args->argc != 1) { cmdq_error(cmdq, "missing key"); return (CMD_RETURN_ERROR); } return (CMD_RETURN_ERROR);
+  if (args->argc != 1) {
+   cmdq_error(cmdq, "missing key");
+   return (CMD_RETURN_ERROR);
+  }
