diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_merged.c
index 5a048b6..3fea150 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-unbind-key.no_comments_merged.c
@@ -6,5 +6,3 @@ enum cmd_retval cmd_unbind_key_table(struct cmd *, struct cmd_q *, int);
-const struct cmd_entry cmd_unbind_key_entry = {"unbind-key",
-                                               "unbind",
-                                               "acnt:",
-                                               0,
-                                               1,
+const struct cmd_entry cmd_unbind_key_entry = {
+ "unbind-key", "unbind",
+ "acnt:", 0, 1,
@@ -14,2 +12,5 @@ const struct cmd_entry cmd_unbind_key_entry = {"unbind-key",
-                                               cmd_unbind_key_exec};
-enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
+ cmd_unbind_key_exec
+};
+enum cmd_retval
+cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq)
+{
@@ -20,8 +20,0 @@ enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
-<<<<<<< HEAD
-    if (args->argc != 1) {
-      cmdq_error(cmdq, "missing key");
-      return (CMD_RETURN_ERROR);
-    }
-    return (CMD_RETURN_ERROR);
-|||||||
-=======
@@ -32 +24,0 @@ enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
->>>>>>> d518067be6220757a9101ca27fff14d5f599c410
@@ -45 +37,2 @@ enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (args_has(args, 't')) return (cmd_unbind_key_table(self, cmdq, key));
+ if (args_has(args, 't'))
+  return (cmd_unbind_key_table(self, cmdq, key));
@@ -53 +46,2 @@ enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (!args_has(args, 'n')) key |= KEYC_PREFIX;
+ if (!args_has(args, 'n'))
+  key |= KEYC_PREFIX;
@@ -57,2 +51,3 @@ enum cmd_retval cmd_unbind_key_exec(struct cmd *self, struct cmd_q *cmdq) {
-enum cmd_retval cmd_unbind_key_table(struct cmd *self, struct cmd_q *cmdq,
-                                     int key) {
+enum cmd_retval
+cmd_unbind_key_table(struct cmd *self, struct cmd_q *cmdq, int key)
+{
