--- a/./output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_mergegen.c
+++ b/./output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_merged.c
@@ -390,0 +407 @@ format_window_pane(struct format_tree *ft, struct window_pane *wp)
+ const char *cmd;
@@ -417,0 +435,3 @@ format_window_pane(struct format_tree *ft, struct window_pane *wp)
+ if ((cwd = osdep_get_cwd(wp->fd)) != NULL)
+  format_add(ft, "pane_current_path", "%s", cwd);
+ if ((cmd = format_get_command(wp)) != NULL) {
