diff --git a/./output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_mergegen.c b/./output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_merged.c
index b8df074..6b4b414 100644
--- a/./output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_mergegen.c
+++ b/./output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_merged.c
@@ -270 +270,17 @@ format_expand(struct format_tree *ft, const char *fmt)
-                                   char * format_get_command(struct window_pane *wp) { char *cmd, *out; cmd = osdep_get_name(wp->fd, wp->tty); if (cmd == NULL || *cmd == '\0') { free(cmd); cmd = xstrdup(wp->cmd); if (cmd == NULL || *cmd == '\0') { free(cmd); cmd = xstrdup(wp->shell); } } out = parse_window_name(cmd); free(cmd); return (out); }
+char *
+format_get_command(struct window_pane *wp)
+{
+ char *cmd, *out;
+ cmd = osdep_get_name(wp->fd, wp->tty);
+ if (cmd == NULL || *cmd == '\0') {
+  free(cmd);
+  cmd = xstrdup(wp->cmd);
+  if (cmd == NULL || *cmd == '\0') {
+   free(cmd);
+   cmd = xstrdup(wp->shell);
+  }
+ }
+ out = parse_window_name(cmd);
+ free(cmd);
+ return (out);
+}
@@ -390,0 +407 @@ format_window_pane(struct format_tree *ft, struct window_pane *wp)
+ const char *cmd;
@@ -417,0 +435,3 @@ format_window_pane(struct format_tree *ft, struct window_pane *wp)
+ if ((cwd = osdep_get_cwd(wp->fd)) != NULL)
+  format_add(ft, "pane_current_path", "%s", cwd);
+ if ((cmd = format_get_command(wp)) != NULL) {
