diff --git a/./output/redis/e45239e3-c697edf4-a15a5d70/src@module.no_comments_mergegen.c b/./output/redis/e45239e3-c697edf4-a15a5d70/src@module.no_comments_merged.c
index 5d1d269..4ce30c3 100644
--- a/./output/redis/e45239e3-c697edf4-a15a5d70/src@module.no_comments_mergegen.c
+++ b/./output/redis/e45239e3-c697edf4-a15a5d70/src@module.no_comments_merged.c
@@ -3913 +3913,7 @@ void moduleRegisterCoreAPI(void) {
-REGISTER_API(GetUsedMemoryRatio); REGISTER_API(MallocSize); REGISTER_API(ScanCursorCreate); REGISTER_API(ScanCursorDestroy); REGISTER_API(ScanCursorRestart); REGISTER_API(Scan); REGISTER_API(ScanKey);
+    REGISTER_API(GetUsedMemoryRatio);
+    REGISTER_API(MallocSize);
+    REGISTER_API(ScanCursorCreate);
+    REGISTER_API(ScanCursorDestroy);
+    REGISTER_API(ScanCursorRestart);
+    REGISTER_API(Scan);
+    REGISTER_API(ScanKey);
