diff --git a/./output/tmux/0526d074-d768fc25-32186950/proc.no_comments_mergegen.c b/./output/tmux/0526d074-d768fc25-32186950/proc.no_comments_merged.c
index 426a877..e4057a0 100644
--- a/./output/tmux/0526d074-d768fc25-32186950/proc.no_comments_mergegen.c
+++ b/./output/tmux/0526d074-d768fc25-32186950/proc.no_comments_merged.c
@@ -2 +2,2 @@
-#include <sys/queue.h> #include <sys/socket.h>
+#include <sys/queue.h>
+#include <sys/socket.h>
