--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/9f15092e-e5fa4944-abeee9f2/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/9f15092e-e5fa4944-abeee9f2/db@db_impl.no_comments_merged.h
@@ -108,0 +109,3 @@ class DBImpl : public DB {
+  Status RunManualCompaction(ColumnFamilyData* cfd, int input_level,
+                             int output_level, const Slice* begin,
+                             const Slice* end);
@@ -226,0 +232,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -290,9 +299,0 @@ class DBImpl : public DB {
- public:
-  Status RunManualCompaction(ColumnFamilyData* cfd, int input_level,
-                             int output_level, const Slice* begin,
-                             const Slice* end);
- private:
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
