diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/9f15092e-e5fa4944-abeee9f2/db@db_impl.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/9f15092e-e5fa4944-abeee9f2/db@db_impl.no_comments_merged.h
index 9552182..0454b3a 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/9f15092e-e5fa4944-abeee9f2/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/9f15092e-e5fa4944-abeee9f2/db@db_impl.no_comments_merged.h
@@ -104,2 +104,2 @@ class DBImpl : public DB {
-      const TransactionLogIterator::ReadOptions& read_options =
-          TransactionLogIterator::ReadOptions());
+      const TransactionLogIterator::ReadOptions&
+          read_options = TransactionLogIterator::ReadOptions());
@@ -108,0 +109,3 @@ class DBImpl : public DB {
+  Status RunManualCompaction(ColumnFamilyData* cfd, int input_level,
+                             int output_level, const Slice* begin,
+                             const Slice* end);
@@ -114,4 +117,4 @@ class DBImpl : public DB {
-  Iterator* TEST_NewInternalIterator(
-      ColumnFamilyHandle* column_family = nullptr);
-  int64_t TEST_MaxNextLevelOverlappingBytes(
-      ColumnFamilyHandle* column_family = nullptr);
+  Iterator* TEST_NewInternalIterator(ColumnFamilyHandle* column_family =
+                                         nullptr);
+  int64_t TEST_MaxNextLevelOverlappingBytes(ColumnFamilyHandle* column_family =
+                                                nullptr);
@@ -122,4 +125,3 @@ class DBImpl : public DB {
-  void TEST_SetDefaultTimeToCheck(
-      uint64_t default_interval_to_delete_obsolete_WAL) {
-    default_interval_to_delete_obsolete_WAL_ =
-        default_interval_to_delete_obsolete_WAL;
+  void TEST_SetDefaultTimeToCheck(uint64_t default_interval_to_delete_obsolete_WAL)
+  {
+    default_interval_to_delete_obsolete_WAL_ = default_interval_to_delete_obsolete_WAL;
@@ -131 +133,2 @@ class DBImpl : public DB {
-      return candidate_files.size() || sst_delete_files.size() ||
+      return candidate_files.size() ||
+        sst_delete_files.size() ||
@@ -158 +161,2 @@ class DBImpl : public DB {
-  void FindObsoleteFiles(DeletionState& deletion_state, bool force,
+  void FindObsoleteFiles(DeletionState& deletion_state,
+                         bool force,
@@ -216 +220,2 @@ class DBImpl : public DB {
-                                VectorLogPtr& log_files, WalFileType type);
+                                VectorLogPtr& log_files,
+                                WalFileType type);
@@ -226,0 +232,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -279 +288,2 @@ class DBImpl : public DB {
-      SequenceNumber in, std::vector<SequenceNumber>& snapshots,
+    SequenceNumber in,
+    std::vector<SequenceNumber>& snapshots,
@@ -284,3 +294,3 @@ class DBImpl : public DB {
-  virtual Status GetPropertiesOfAllTables(
-      ColumnFamilyHandle* column_family,
-      TablePropertiesCollection* props) override;
+  virtual Status GetPropertiesOfAllTables(ColumnFamilyHandle* column_family,
+                                          TablePropertiesCollection* props)
+      override;
@@ -290,9 +299,0 @@ class DBImpl : public DB {
- public:
-  Status RunManualCompaction(ColumnFamilyData* cfd, int input_level,
-                             int output_level, const Slice* begin,
-                             const Slice* end);
- private:
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
