diff --git a/./output/tmux/87521214-1d4296f1-33f9b08b/server.no_comments_mergegen.c b/./output/tmux/87521214-1d4296f1-33f9b08b/server.no_comments_merged.c
index 4997dd0..5f9e6c8 100644
--- a/./output/tmux/87521214-1d4296f1-33f9b08b/server.no_comments_mergegen.c
+++ b/./output/tmux/87521214-1d4296f1-33f9b08b/server.no_comments_merged.c
@@ -113 +113,5 @@ server_tidy_event(__unused int fd, __unused short events, __unused void *data)
-log_debug("%s: took %llu milliseconds", __func__, (unsigned long long)(get_timer() - t));
+#ifdef HAVE_MALLOC_TRIM
+    malloc_trim(0);
+#endif
+    log_debug("%s: took %llu milliseconds", __func__,
+        (unsigned long long)(get_timer() - t));
