--- a/./output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_conflict.cpp
+++ b/./output/frameworks_native/c37fbef9-b5f3b9b8-419db573/services@inputflinger@PointerChoreographer.no_comments_merged.cpp
@@ -330 +329,0 @@ PointerChoreographer::PointerDisplayChange PointerChoreographer::updatePointerCo
-<<<<<<< HEAD
@@ -333,6 +331,0 @@ PointerChoreographer::PointerDisplayChange PointerChoreographer::updatePointerCo
-||||||| 419db573a7
-        if (isFromSource(sources, AINPUT_SOURCE_MOUSE) ||
-            isFromSource(sources, AINPUT_SOURCE_MOUSE_RELATIVE)) {
-=======
-        if (isMouseOrTouchpad(sources)) {
->>>>>>> b5f3b9b8
