diff --git a/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_conflict.cpp b/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_merged.cpp
index a371ca0..ea0645f 100644
--- a/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_conflict.cpp
+++ b/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_merged.cpp
@@ -64 +63,0 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-<<<<<<< HEAD
@@ -71,10 +69,0 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-||||||| 88127a1114
-    std::shared_ptr<const android::vintf::RuntimeInfo> runtimeInfo =
-            android::vintf::VintfObject::GetInstance()->getRuntimeInfo(
-                    vintf::RuntimeInfo::FetchFlag::CONFIG_GZ);
-    LOG_ALWAYS_FATAL_IF(runtimeInfo == nullptr, "Kernel configs could not be fetched");
-=======
-    std::map<std::string, std::string> kernelConfigs;
-    const status_t result = android::kernelconfigs::LoadKernelConfigs(&kernelConfigs);
-    LOG_ALWAYS_FATAL_IF(result != OK, "Kernel configs could not be fetched");
->>>>>>> bc7936f8
