--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_merged.cpp
@@ -13,4 +12,0 @@
-#include <vintf/RuntimeInfo.h>
-#include <vintf/VintfObject.h>
-#endif
-#if defined(__ANDROID__)
@@ -68 +64 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-    <<<<<<< HEAD if (configs.empty()) {
+    if (configs.empty()) {
@@ -74,10 +69,0 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-|||||||
-    std::shared_ptr<const android::vintf::RuntimeInfo> runtimeInfo =
-            android::vintf::VintfObject::GetInstance()->getRuntimeInfo(
-                    vintf::RuntimeInfo::FetchFlag::CONFIG_GZ);
-    LOG_ALWAYS_FATAL_IF(runtimeInfo == nullptr, "Kernel configs could not be fetched");
-=======
-    std::map<std::string, std::string> kernelConfigs;
-    const status_t result = android::kernelconfigs::LoadKernelConfigs(&kernelConfigs);
-    LOG_ALWAYS_FATAL_IF(result != OK, "Kernel configs could not be fetched");
->>>>>>> bc7936f870b420a5368cac869b129d60254e31fb
@@ -104,3 +90,4 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-KeyLayoutMap::KeyLayoutMap() KeyLayoutMap::~KeyLayoutMap()
-        base::Result<std::shared_ptr<KeyLayoutMap>> KeyLayoutMap::loadContents(
-                const std::string& filename, const char* contents) {
+KeyLayoutMap::KeyLayoutMap() = default;
+KeyLayoutMap::~KeyLayoutMap() = default;
+base::Result<std::shared_ptr<KeyLayoutMap>> KeyLayoutMap::loadContents(const std::string& filename,
+                                                                       const char* contents) {
