diff --git a/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_mergegen.cpp b/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_merged.cpp
index f1c59b9..ea0645f 100644
--- a/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/8af3ec98-bc7936f8-88127a11/libs@input@KeyLayoutMap.no_comments_merged.cpp
@@ -64 +64,6 @@ bool kernelConfigsArePresent(const std::set<std::string>& configs) {
-if (configs.empty()) { return true; } std::map<std::string, std::string> kernelConfigs; const status_t result = android::kernelconfigs::LoadKernelConfigs(&kernelConfigs); LOG_ALWAYS_FATAL_IF(result != OK, "Kernel configs could not be fetched");
+    if (configs.empty()) {
+        return true;
+    }
+    std::map<std::string, std::string> kernelConfigs;
+    const status_t result = android::kernelconfigs::LoadKernelConfigs(&kernelConfigs);
+    LOG_ALWAYS_FATAL_IF(result != OK, "Kernel configs could not be fetched");
