--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/102265d1-f24e4b3a-7f8cca5d/adb@client@incremental.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/102265d1-f24e4b3a-7f8cca5d/adb@client@incremental.no_comments_merged.cpp
@@ -82,2 +77 @@ static std::pair<unique_fd, std::string> read_and_encode_signature(
-            "Verity tree size mismatch in signature file: %s [was %lld, "
-            "expected %lld].\n",
+                "Verity tree size mismatch in signature file: %s [was %lld, expected %lld].\n",
@@ -151 +141,0 @@ std::optional<Process> install(std::vector<std::string> files) {
-  <<<<<<< HEAD
@@ -154,14 +144 @@ std::optional<Process> install(std::vector<std::string> files) {
-    fprintf(stderr,
-            "Failed to create socket pair for child to print to parent\n");
-    return {};
-  }
-  auto [pipe_read_fd, pipe_write_fd] = print_fds;
-  auto pipe_write_fd_param =
-      std::to_string(intptr_t(adb_get_os_handle(pipe_write_fd)));
-  close_on_exec(pipe_read_fd);
-|||||||
-=======
-  int print_fds[2];
-  if (adb_socketpair(print_fds) != 0) {
-    fprintf(stderr,
-            "Failed to create socket pair for child to print to parent\n");
+        fprintf(stderr, "Failed to create socket pair for child to print to parent\n");
@@ -171 +148 @@ std::optional<Process> install(std::vector<std::string> files) {
-  auto pipe_write_fd_param = std::to_string(pipe_write_fd);
+    auto pipe_write_fd_param = std::to_string(intptr_t(adb_get_os_handle(pipe_write_fd)));
@@ -173 +149,0 @@ std::optional<Process> install(std::vector<std::string> files) {
->>>>>>> f24e4b3ae4d5a391a1f86136589b59a27dde55d0
