--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/1caa878b-0b55c341-c02c6800/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/1caa878b-0b55c341-c02c6800/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-std::string Version() < < < < < < < HEAD { return "1.33.0-dev"; }
-|||||||
-{ return "1.32.0-dev"; }
-=======
-{ return "1.32.0"; }
->>>>>>> 0b55c341b0596c1076de8580ef5babba17e2e08b
+std::string Version() { return "1.33.0-dev"; }
