diff --git a/./output/frameworks_native/ee799cc9-d4433288-7869c733/include@android@surface_control.no_comments_conflict.h b/./output/frameworks_native/ee799cc9-d4433288-7869c733/include@android@surface_control.no_comments_merged.h
index 94f5de5..3f16f61 100644
--- a/./output/frameworks_native/ee799cc9-d4433288-7869c733/include@android@surface_control.no_comments_conflict.h
+++ b/./output/frameworks_native/ee799cc9-d4433288-7869c733/include@android@surface_control.no_comments_merged.h
@@ -123 +122,0 @@ void ASurfaceTransaction_setHdrMetadata_cta861_3(ASurfaceTransaction* _Nonnull t
-<<<<<<< HEAD
@@ -132,11 +130,0 @@ void ASurfaceTransaction_setDesiredHdrHeadroom(ASurfaceTransaction* _Nonnull tra
-||||||| 7869c73351
-void ASurfaceTransaction_setExtendedRangeBrightness(ASurfaceTransaction* transaction,
-                                            ASurfaceControl* surface_control,
-                                            float currentBufferRatio,
-                                            float desiredRatio) __INTRODUCED_IN(__ANDROID_API_U__);
-=======
-void ASurfaceTransaction_setExtendedRangeBrightness(ASurfaceTransaction* _Nonnull transaction,
-                                                    ASurfaceControl* _Nonnull surface_control,
-                                                    float currentBufferRatio, float desiredRatio)
-                                                    __INTRODUCED_IN(__ANDROID_API_U__);
->>>>>>> d4433288
