--- a/./output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_conflict.h
+++ b/./output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_merged.h
@@ -791 +790,0 @@ private:
-<<<<<<< HEAD
@@ -800,5 +798,0 @@ private:
-||||||| 5b03292d00
-=======
-        bool checkHapticCompatibilityOnSpatializerOutput(const audio_config_t* config,
-                                                         audio_session_t sessionId) const;
->>>>>>> 6e247f9f
