diff --git a/./output/redis/ca5ccc58-16435e02-0e9b5adb/src@networking.no_comments_mergegen.c b/./output/redis/ca5ccc58-16435e02-0e9b5adb/src@networking.no_comments_merged.c
index dff269f..ff33929 100644
--- a/./output/redis/ca5ccc58-16435e02-0e9b5adb/src@networking.no_comments_mergegen.c
+++ b/./output/redis/ca5ccc58-16435e02-0e9b5adb/src@networking.no_comments_merged.c
@@ -1613,0 +1614,2 @@ void flushSlavesOutputBuffers(void) {
+        if (slave->replstate == SLAVE_STATE_ONLINE &&
+            !slave->repl_put_online_on_ack &&
