--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/ca5ccc58-16435e02-0e9b5adb/src@networking.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/ca5ccc58-16435e02-0e9b5adb/src@networking.no_comments_merged.c
@@ -1614,10 +1613,0 @@ void flushSlavesOutputBuffers(void) {
-        events = aeGetFileEvents(server.el,slave->fd);
-        if (events & AE_WRITABLE &&
-            slave->replstate == SLAVE_STATE_ONLINE &&
-|||||||
-         * writes before the first ACK. */
-        events = aeGetFileEvents(server.el,slave->fd);
-        if (events & AE_WRITABLE &&
-            slave->replstate == SLAVE_STATE_ONLINE &&
-=======
-         * writes before the first ACK. */
@@ -1626 +1615,0 @@ void flushSlavesOutputBuffers(void) {
->>>>>>> 16435e02bfa80ff7254e8b062256494cb77ad708
