diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_merged.cc
index e7fac1c..971695c 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_merged.cc
@@ -35 +35,3 @@ class BackupEngineImpl : public BackupEngine {
-  void StopBackup() { stop_backup_.store(true, std::memory_order_release); }
+  void StopBackup() {
+    stop_backup_.store(true, std::memory_order_release);
+  }
@@ -37,6 +39,8 @@ class BackupEngineImpl : public BackupEngine {
-  Status RestoreDBFromBackup(
-      BackupID backup_id, const std::string& db_dir, const std::string& wal_dir,
-      const RestoreOptions& restore_options = RestoreOptions());
-  Status RestoreDBFromLatestBackup(
-      const std::string& db_dir, const std::string& wal_dir,
-      const RestoreOptions& restore_options = RestoreOptions()) {
+  Status RestoreDBFromBackup(BackupID backup_id, const std::string& db_dir,
+                             const std::string& wal_dir,
+                             const RestoreOptions& restore_options =
+                                 RestoreOptions());
+  Status RestoreDBFromLatestBackup(const std::string& db_dir,
+                                   const std::string& wal_dir,
+                                   const RestoreOptions& restore_options =
+                                       RestoreOptions()) {
@@ -60,5 +64,2 @@ class BackupEngineImpl : public BackupEngine {
-        : timestamp_(0),
-          size_(0),
-          meta_filename_(meta_filename),
-          file_infos_(file_infos),
-          env_(env) {}
+      : timestamp_(0), size_(0), meta_filename_(meta_filename),
+        file_infos_(file_infos), env_(env) {}
@@ -66,3 +67,9 @@ class BackupEngineImpl : public BackupEngine {
-    void RecordTimestamp() { env_->GetCurrentTime(&timestamp_); }
-    int64_t GetTimestamp() const { return timestamp_; }
-    uint64_t GetSize() const { return size_; }
+    void RecordTimestamp() {
+      env_->GetCurrentTime(&timestamp_);
+    }
+    int64_t GetTimestamp() const {
+      return timestamp_;
+    }
+    uint64_t GetSize() const {
+      return size_;
+    }
@@ -72 +79,3 @@ class BackupEngineImpl : public BackupEngine {
-    uint64_t GetSequenceNumber() { return sequence_number_; }
+    uint64_t GetSequenceNumber() {
+      return sequence_number_;
+    }
@@ -75,2 +84,6 @@ class BackupEngineImpl : public BackupEngine {
-    bool Empty() { return files_.empty(); }
-    const std::vector<std::string>& GetFiles() { return files_; }
+    bool Empty() {
+      return files_.empty();
+    }
+    const std::vector<std::string>& GetFiles() {
+      return files_;
+    }
@@ -94,2 +107,5 @@ class BackupEngineImpl : public BackupEngine {
-  inline std::string GetPrivateDirRel() const { return "private"; }
-  inline std::string GetPrivateFileRel(BackupID backup_id, bool tmp = false,
+  inline std::string GetPrivateDirRel() const {
+    return "private";
+  }
+  inline std::string GetPrivateFileRel(BackupID backup_id,
+                                       bool tmp = false,
@@ -117,4 +133,11 @@ class BackupEngineImpl : public BackupEngine {
-  Status CopyFile(const std::string& src, const std::string& dst, Env* src_env,
-                  Env* dst_env, bool sync, uint64_t* size = nullptr,
-                  uint32_t* checksum_value = nullptr, uint64_t size_limit = 0);
-  Status BackupFile(BackupID backup_id, BackupMeta* backup, bool shared,
+  Status CopyFile(const std::string& src,
+                  const std::string& dst,
+                  Env* src_env,
+                  Env* dst_env,
+                  bool sync,
+                  uint64_t* size = nullptr,
+                  uint32_t* checksum_value = nullptr,
+                  uint64_t size_limit = 0);
+  Status BackupFile(BackupID backup_id,
+                    BackupMeta* backup,
+                    bool shared,
@@ -124,2 +147,4 @@ class BackupEngineImpl : public BackupEngine {
-  Status CalculateChecksum(const std::string& src, Env* src_env,
-                           uint64_t size_limit, uint32_t* checksum_value);
+  Status CalculateChecksum(const std::string& src,
+                           Env* src_env,
+                           uint64_t size_limit,
+                           uint32_t* checksum_value);
@@ -200 +225,2 @@ BackupEngineImpl::BackupEngineImpl(Env* db_env,
-    if (!s.ok() || backups_.find(latest_backup_id_) == backups_.end()) {
+    if (!s.ok() ||
+        backups_.find(latest_backup_id_) == backups_.end()) {
@@ -213 +239,2 @@ BackupEngineImpl::BackupEngineImpl(Env* db_env,
-  Log(options_.info_log, "Initialized BackupEngine, the latest backup is %u.",
+  Log(options_.info_log,
+      "Initialized BackupEngine, the latest backup is %u.",
@@ -227,5 +253,0 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
-<<<<<<< HEAD
-  if (s.ok()) {
-||||||| 9b8a2b52d
-  if (s.ok() && !flush_before_backup) {
-=======
@@ -233 +254,0 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
->>>>>>> 5601bc4619402676900539c4ef828cedfa998a32
@@ -263 +284,2 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
-    s = BackupFile(new_backup_id, &new_backup,
+    s = BackupFile(new_backup_id,
+                   &new_backup,
@@ -271,2 +293,5 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
-      s = BackupFile(new_backup_id, &new_backup, false,
-                     db->GetOptions().wal_dir, live_wal_files[i]->PathName());
+      s = BackupFile(new_backup_id,
+                     &new_backup,
+                     false,
+                     db->GetOptions().wal_dir,
+                     live_wal_files[i]->PathName());
@@ -403 +428,2 @@ Status BackupEngineImpl::RestoreDBFromBackup(
-    dst = ((type == kLogFile) ? wal_dir : db_dir) + "/" + dst;
+    dst = ((type == kLogFile) ? wal_dir : db_dir) +
+      "/" + dst;
@@ -424 +450,2 @@ Status BackupEngineImpl::GetLatestBackupFileContents(uint32_t* latest_backup) {
-  s = backup_env_->NewSequentialFile(GetLatestBackupFile(), &file,
+  s = backup_env_->NewSequentialFile(GetLatestBackupFile(),
+                                     &file,
@@ -448 +475,2 @@ Status BackupEngineImpl::PutLatestBackupFileContents(uint32_t latest_backup) {
-  s = backup_env_->NewWritableFile(GetLatestBackupFile(true), &file,
+  s = backup_env_->NewWritableFile(GetLatestBackupFile(true),
+                                   &file,
@@ -502,3 +530,2 @@ Status BackupEngineImpl::CopyFile(const std::string& src,
-    size_t buffer_to_read = (copy_file_buffer_size_ < size_limit)
-                                ? copy_file_buffer_size_
-                                : size_limit;
+    size_t buffer_to_read = (copy_file_buffer_size_ < size_limit) ?
+      copy_file_buffer_size_ : size_limit;
@@ -514,2 +541,2 @@ Status BackupEngineImpl::CopyFile(const std::string& src,
-      *checksum_value =
-          crc32c::Extend(*checksum_value, data.data(), data.size());
+      *checksum_value = crc32c::Extend(*checksum_value, data.data(),
+                                       data.size());
@@ -547 +574,3 @@ Status BackupEngineImpl::BackupFile(BackupID backup_id, BackupMeta* backup,
-    s = CalculateChecksum(src_dir + src_fname, db_env_, size_limit,
+    s = CalculateChecksum(src_dir + src_fname,
+                          db_env_,
+                          size_limit,
@@ -551,2 +580,8 @@ Status BackupEngineImpl::BackupFile(BackupID backup_id, BackupMeta* backup,
-    s = CopyFile(src_dir + src_fname, dst_path_tmp, db_env_, backup_env_,
-                 options_.sync, &size, &checksum_value, size_limit);
+    s = CopyFile(src_dir + src_fname,
+                 dst_path_tmp,
+                 db_env_,
+                 backup_env_,
+                 options_.sync,
+                 &size,
+                 &checksum_value,
+                 size_limit);
@@ -583,3 +618,2 @@ Status BackupEngineImpl::CalculateChecksum(const std::string& src, Env* src_env,
-    size_t buffer_to_read = (copy_file_buffer_size_ < size_limit)
-                                ? copy_file_buffer_size_
-                                : size_limit;
+    size_t buffer_to_read = (copy_file_buffer_size_ < size_limit) ?
+      copy_file_buffer_size_ : size_limit;
@@ -802,2 +836,3 @@ BackupableDB::BackupableDB(DB* db, const BackupableDBOptions& options)
-BackupableDB::~BackupableDB() { delete backup_engine_; }
-{ delete backup_engine_; }
+BackupableDB::~BackupableDB() {
+  delete backup_engine_;
+}
@@ -816 +851,3 @@ Status BackupableDB::DeleteBackup(BackupID backup_id) {
-void BackupableDB::StopBackup() { backup_engine_->StopBackup(); }
+void BackupableDB::StopBackup() {
+  backup_engine_->StopBackup();
+}
@@ -820,3 +857,5 @@ RestoreBackupableDB::RestoreBackupableDB(Env* db_env,
-RestoreBackupableDB::~RestoreBackupableDB() { delete backup_engine_; }
-{ delete backup_engine_; }
-void RestoreBackupableDB::GetBackupInfo(std::vector<BackupInfo>* backup_info) {
+RestoreBackupableDB::~RestoreBackupableDB() {
+  delete backup_engine_;
+}
+void
+RestoreBackupableDB::GetBackupInfo(std::vector<BackupInfo>* backup_info) {
