diff --git a/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_conflict.cc b/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_merged.cc
index c3bc5b4..971695c 100644
--- a/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_conflict.cc
+++ b/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_merged.cc
@@ -254,5 +253,0 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
-<<<<<<< HEAD
-  if (s.ok()) {
-||||||| 9b8a2b52d
-  if (s.ok() && !flush_before_backup) {
-=======
@@ -260 +254,0 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
->>>>>>> 5601bc46
