diff --git a/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_mergegen.cc b/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_merged.cc
index a351db1..971695c 100644
--- a/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_mergegen.cc
+++ b/./output/rocksdb/e0c12115-5601bc46-9b8a2b52/utilities@backupable@backupable_db.no_comments_merged.cc
@@ -254 +254 @@ Status BackupEngineImpl::CreateNewBackup(DB* db, bool flush_before_backup) {
-if (s.ok()) {
+  if (s.ok() && !flush_before_backup && options_.backup_log_files) {
