diff --git a/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_conflict.cc b/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_merged.cc
index f23c251..f12e77f 100644
--- a/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_conflict.cc
+++ b/./output/art/cdd3206f-4d45b861-403cf5cb/runtime@trace.no_comments_merged.cc
@@ -639 +638,0 @@ TraceWriter::TraceWriter(File* trace_file,
-<<<<<<< HEAD
@@ -642,8 +640,0 @@ TraceWriter::TraceWriter(File* trace_file,
-||||||| 403cf5cb88
-  CHECK_IMPLIES(trace_file == nullptr, output_mode == TraceOutputMode::kDDMS);
-  uint16_t trace_version = GetTraceVersion(clock_source_);
-=======
-  CHECK_IMPLIES(trace_file == nullptr, output_mode == TraceOutputMode::kDDMS);
-  uint16_t trace_version = GetTraceVersion(clock_source_);
-  uint64_t start_time_monotonic = start_time_ + (MicroTime() - GetMicroTime(GetTimestamp()));
->>>>>>> 4d45b861
@@ -653,24 +643,0 @@ TraceWriter::TraceWriter(File* trace_file,
-<<<<<<< HEAD
-||||||| 403cf5cb88
-  memset(buf_.get(), 0, kTraceHeaderLength);
-  Append4LE(buf_.get(), kTraceMagicValue);
-  Append2LE(buf_.get() + 4, trace_version);
-  Append2LE(buf_.get() + 6, kTraceHeaderLength);
-  Append8LE(buf_.get() + 8, start_time_);
-  if (trace_version >= kTraceVersionDualClock) {
-    uint16_t record_size = GetRecordSize(clock_source_);
-    Append2LE(buf_.get() + 16, record_size);
-  }
-  static_assert(18 <= kMinBufSize, "Minimum buffer size not large enough for trace header");
-=======
-  memset(buf_.get(), 0, kTraceHeaderLength);
-  Append4LE(buf_.get(), kTraceMagicValue);
-  Append2LE(buf_.get() + 4, trace_version);
-  Append2LE(buf_.get() + 6, kTraceHeaderLength);
-  Append8LE(buf_.get() + 8, start_time_monotonic);
-  if (trace_version >= kTraceVersionDualClock) {
-    uint16_t record_size = GetRecordSize(clock_source_);
-    Append2LE(buf_.get() + 16, record_size);
-  }
-  static_assert(18 <= kMinBufSize, "Minimum buffer size not large enough for trace header");
->>>>>>> 4d45b861
