diff --git a/./output/tmux/27126f87-5c49e1d0-66b5477c/format.no_comments_mergegen.c b/./output/tmux/27126f87-5c49e1d0-66b5477c/format.no_comments_merged.c
index 01fbec3..734caa0 100644
--- a/./output/tmux/27126f87-5c49e1d0-66b5477c/format.no_comments_mergegen.c
+++ b/./output/tmux/27126f87-5c49e1d0-66b5477c/format.no_comments_merged.c
@@ -19 +19,18 @@ static void format_job_timer(int, short, void *);
-static void format_cb_host(struct format_tree *, struct format_entry *); static void format_cb_host_short(struct format_tree *, struct format_entry *); static void format_cb_pid(struct format_tree *, struct format_entry *); static void format_cb_session_alerts(struct format_tree *, struct format_entry *); static void format_cb_window_layout(struct format_tree *, struct format_entry *); static void format_cb_window_visible_layout(struct format_tree *, struct format_entry *); static void format_cb_start_command(struct format_tree *, struct format_entry *); static void format_cb_current_command(struct format_tree *, struct format_entry *); static void format_cb_current_path(struct format_tree *, struct format_entry *); static void format_cb_history_bytes(struct format_tree *, struct format_entry *); static void format_cb_pane_tabs(struct format_tree *, struct format_entry *); static
+static void format_cb_host(struct format_tree *, struct format_entry *);
+static void format_cb_host_short(struct format_tree *,
+       struct format_entry *);
+static void format_cb_pid(struct format_tree *, struct format_entry *);
+static void format_cb_session_alerts(struct format_tree *,
+       struct format_entry *);
+static void format_cb_window_layout(struct format_tree *,
+       struct format_entry *);
+static void format_cb_window_visible_layout(struct format_tree *,
+       struct format_entry *);
+static void format_cb_start_command(struct format_tree *,
+       struct format_entry *);
+static void format_cb_current_command(struct format_tree *,
+       struct format_entry *);
+static void format_cb_history_bytes(struct format_tree *,
+       struct format_entry *);
+static void format_cb_pane_tabs(struct format_tree *,
+       struct format_entry *);
