--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/27126f87-5c49e1d0-66b5477c/format.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/27126f87-5c49e1d0-66b5477c/format.no_comments_merged.c
@@ -14 +14 @@
-struct format_entry
+struct format_entry;
@@ -32 +32,0 @@ static void format_cb_current_command(struct format_tree *,
-void format_cb_current_path(struct format_tree *, struct format_entry *);
@@ -55 +58,5 @@ static int format_job_cmp(struct format_job *, struct format_job *);
-format_job_cmp(struct format_job *fj1, struct format_job *fj2) {
+static RB_HEAD(format_job_tree, format_job) format_jobs = RB_INITIALIZER();
+RB_GENERATE_STATIC(format_job_tree, format_job, entry, format_job_cmp);
+static int
+format_job_cmp(struct format_job *fj1, struct format_job *fj2)
+{
@@ -77 +84,4 @@ static int format_entry_cmp(struct format_entry *, struct format_entry *);
-static intformat_entry_cmp(struct format_entry *fe1, struct format_entry *fe2) {
+RB_GENERATE_STATIC(format_entry_tree, format_entry, entry, format_entry_cmp);
+static int
+format_entry_cmp(struct format_entry *fe1, struct format_entry *fe2)
+{
