diff --git a/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@include@duckdb@main@attached_database.no_comments_conflict.hpp b/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@include@duckdb@main@attached_database.no_comments_merged.hpp
index 0e2158e..caf2cae 100644
--- a/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@include@duckdb@main@attached_database.no_comments_conflict.hpp
+++ b/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@include@duckdb@main@attached_database.no_comments_merged.hpp
@@ -27,6 +26,0 @@ struct AttachOptions {
-<<<<<<< HEAD
- string unrecognized_option;
- QualifiedName default_table;
-||||||| 0a75499c43
- string unrecognized_option;
-=======
@@ -34 +28 @@ struct AttachOptions {
->>>>>>> 56a1d037
+ QualifiedName default_table;
