diff --git a/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_mergegen.cpp b/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_merged.cpp
index ef69ebe..d1bea15 100644
--- a/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_mergegen.cpp
+++ b/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_merged.cpp
@@ -53 +53,5 @@ SourceResultType PhysicalAttach::GetData(ExecutionContext &context, DataChunk &c
-const auto storage_options = info->GetStorageOptions(); attached_db->Initialize(storage_options); if (!options.default_table.name.empty()) { attached_db->GetCatalog().SetDefaultTable(options.default_table.schema, options.default_table.name); }
+ const auto storage_options = info->GetStorageOptions();
+ attached_db->Initialize(storage_options);
+ if (!options.default_table.name.empty()) {
+  attached_db->GetCatalog().SetDefaultTable(options.default_table.schema, options.default_table.name);
+ }
