diff --git a/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_conflict.cpp b/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_merged.cpp
index 35221c4..d1bea15 100644
--- a/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_conflict.cpp
+++ b/./output/duckdb/ddf8a49a-56a1d037-0a75499c/src@execution@operator@schema@physical_attach.no_comments_merged.cpp
@@ -53,3 +53,2 @@ SourceResultType PhysicalAttach::GetData(ExecutionContext &context, DataChunk &c
-<<<<<<< HEAD
- const auto block_alloc_size = info->GetBlockAllocSize();
- attached_db->Initialize(block_alloc_size);
+ const auto storage_options = info->GetStorageOptions();
+ attached_db->Initialize(storage_options);
@@ -59,7 +57,0 @@ SourceResultType PhysicalAttach::GetData(ExecutionContext &context, DataChunk &c
-||||||| 0a75499c43
- const auto block_alloc_size = info->GetBlockAllocSize();
- attached_db->Initialize(block_alloc_size);
-=======
- const auto storage_options = info->GetStorageOptions();
- attached_db->Initialize(storage_options);
->>>>>>> 56a1d037
