diff --git a/./output/duckdb/c79657e5-10c42435-8d3f8f81/src@execution@window_executor.no_comments_conflict.cpp b/./output/duckdb/c79657e5-10c42435-8d3f8f81/src@execution@window_executor.no_comments_merged.cpp
index 244fcfc..cd35761 100644
--- a/./output/duckdb/c79657e5-10c42435-8d3f8f81/src@execution@window_executor.no_comments_conflict.cpp
+++ b/./output/duckdb/c79657e5-10c42435-8d3f8f81/src@execution@window_executor.no_comments_merged.cpp
@@ -1290,2 +1289,0 @@ void WindowLeadLagExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-<<<<<<< HEAD
-    const auto target_limit = MinValue(partition_end[i], row_end) - row_idx;
@@ -1304,14 +1301,0 @@ void WindowLeadLagExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-||||||| 8d3f8f8195
-    const auto target_limit = MinValue(partition_end[i], row_end) - row_idx;
-    const auto width = MinValue(source_limit, target_limit);
-    auto &source = payload_collection.data[0];
-    VectorOperations::Copy(source, result, index + width, index, i);
-    i += width;
-    row_idx += width;
-=======
-    const auto width = MinValue(source_limit, target_limit);
-    auto &source = payload_collection.data[0];
-    VectorOperations::Copy(source, result, index + width, index, i);
-    i += width;
-    row_idx += width;
->>>>>>> 10c42435
