--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/fec26383-30d444e4-5c0a1ed7/tools@pythonpkg@src@include@duckdb_python@pyfilesystem.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/fec26383-30d444e4-5c0a1ed7/tools@pythonpkg@src@include@duckdb_python@pyfilesystem.no_comments_merged.hpp
@@ -7,3 +6,0 @@
-#include "duckdb_python/pybind_wrapper.hpp"
-#include "duckdb_python/python_object_container.hpp"
-#include <vector>
@@ -15,0 +13 @@ public:
+public:
@@ -22,0 +21 @@ public:
+public:
