--- a/./output/duckdb/fec26383-30d444e4-5c0a1ed7/tools@pythonpkg@src@include@duckdb_python@pyfilesystem.no_comments_mergegen.hpp
+++ b/./output/duckdb/fec26383-30d444e4-5c0a1ed7/tools@pythonpkg@src@include@duckdb_python@pyfilesystem.no_comments_merged.hpp
@@ -4 +4,3 @@
-#include "duckdb/common/vector.hpp" #include "duckdb_python/pybind11/pybind_wrapper.hpp" #include "duckdb_python/pybind11/gil_wrapper.hpp"
+#include "duckdb_python/pybind11/pybind_wrapper.hpp"
+#include "duckdb_python/pybind11/gil_wrapper.hpp"
+#include "duckdb/common/vector.hpp"
