--- a/./output/duckdb/ea6567fe-d84e329b-9de6e816/tools@pythonpkg@src@pyconnection.no_comments_mergegen.cpp
+++ b/./output/duckdb/ea6567fe-d84e329b-9de6e816/tools@pythonpkg@src@pyconnection.no_comments_merged.cpp
@@ -390 +390 @@ void DuckDBPyConnection::RegisterArrowObject(const py::object &arrow_object, con
-     make_unique<PythonTableArrowArrayStreamFactory>(arrow_object.ptr(), connection->context->config);
+     make_uniq<PythonTableArrowArrayStreamFactory>(arrow_object.ptr(), connection->context->config);
@@ -404 +404 @@ void DuckDBPyConnection::RegisterArrowObject(const py::object &arrow_object, con
-     make_shared<PythonDependencies>(make_unique<RegisteredArrow>(std::move(stream_factory), arrow_object)));
+     make_shared<PythonDependencies>(make_uniq<RegisteredArrow>(std::move(stream_factory), arrow_object)));
@@ -424 +424,5 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
-(python_object), make_uniq<RegisteredObject>(new_df))); connection->context->external_dependencies[name] = std::move(dependencies); } } else if (IsAcceptedArrowObject(python_object) || IsPolarsDataframe(python_object)) {
+   vector<shared_ptr<ExternalDependency>> dependencies;
+   dependencies.push_back(make_shared<PythonDependencies>(make_uniq<RegisteredObject>(python_object),
+                                                          make_uniq<RegisteredObject>(new_df)));
+   connection->context->external_dependencies[name] = std::move(dependencies);
+  }
@@ -439 +443 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
-shapeArrowObject(arrow_object, name); } else if (DuckDBPyRelation::IsRelation(python_object)) {
+  RegisterArrowObject(arrow_object, name);
@@ -1027 +1031,11 @@ static unique_ptr<TableRef> TryReplacement(py::dict &dict, py::str &table_name,
-if (DuckDBPyConnection::IsPandasDataframe(entry)) { if (IsArrowBackedDataFrame(entry)) { auto table = ArrowTableFromDataframe(entry); CreateArrowScan(table, *table_function, children, config); } else { string name = "df_" + StringUtil::GenerateRandomName(); auto new_df = PandasScanFunction::PandasReplaceCopiedNames(entry); children.push_back(make_uniq<ConstantExpression>(Value::POINTER((uintptr_t)new_df.ptr()))); table_function->function = make_uniq<FunctionExpression>("pandas_scan", std::move(children)); table_function->external_dependency = make_uniq<PythonDependencies>( make_uniq<RegisteredObject>(entry), make_uniq<RegisteredObject>(new_df)); }
+  if (IsArrowBackedDataFrame(entry)) {
+   auto table = ArrowTableFromDataframe(entry);
+   CreateArrowScan(table, *table_function, children, config);
+  } else {
+   string name = "df_" + StringUtil::GenerateRandomName();
+   auto new_df = PandasScanFunction::PandasReplaceCopiedNames(entry);
+   children.push_back(make_uniq<ConstantExpression>(Value::POINTER((uintptr_t)new_df.ptr())));
+   table_function->function = make_uniq<FunctionExpression>("pandas_scan", std::move(children));
+   table_function->external_dependency =
+       make_uniq<PythonDependencies>(make_uniq<RegisteredObject>(entry), make_uniq<RegisteredObject>(new_df));
+  }
