diff --git a/./output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_mergegen.cpp b/./output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_merged.cpp
index a43ca05..f5813e0 100644
--- a/./output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_merged.cpp
@@ -413,0 +414 @@ Loader::driver_t* Loader::attempt_to_load_angle(egl_connection_t* cnx) {
+    android::GraphicsEnv::getInstance().setDriverToLoad(android::GpuStatsInfo::Driver::ANGLE);
