--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_merged.cpp
@@ -36,0 +37,3 @@ static const char* const VENDOR_LIB_EGL_DIR =
+#else
+        "/vendor/lib/egl";
+#endif
@@ -39,0 +43,3 @@ static const char* const SYSTEM_LIB_DIR =
+#else
+        "/system/lib";
+#endif
@@ -395,3 +413,0 @@ Loader::driver_t* Loader::attempt_to_load_angle(egl_connection_t* cnx) {
-<<<<<<< HEAD
-    android::GraphicsEnv::getInstance().setDriverToLoad(android::GpuStatsInfo::Driver::ANGLE);
-|||||||
@@ -399,2 +414,0 @@ Loader::driver_t* Loader::attempt_to_load_angle(egl_connection_t* cnx) {
-=======
->>>>>>> 826244e86737d09e29aa7e6af11d49e51d9117ce
