--- a/home/whalien/codebase/python/mergebot-eva/output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/abb731e5-c0b3d423-68254919/src@redis-cli.no_comments_merged.c
@@ -148 +137,0 @@ static int cliReadMultiBulkReply(int fd) {
-<<<<<<< HEAD
@@ -150,7 +138,0 @@ static int cliReadMultiBulkReply(int fd) {
-    if (cliReadReply(fd)) return 1;
-    if (elements) printf("%c", config.mb_sep);
-|||||||
-    printf("%d. ", c);
-    if (cliReadReply(fd)) return 1;
-=======
-    printf("%d. ", c);
@@ -158 +140 @@ static int cliReadMultiBulkReply(int fd) {
->>>>>>> c0b3d42372dbe67c6ef096372869e2b60d4a1cdc
+        if (elements) printf("%c",config.mb_sep);
@@ -282,4 +267,3 @@ static int parseOptions(int argc, char **argv) {
-              "Starting interactive mode using -i is deprecated. Interactive "
-              "mode is started\n"
-              "by default when redis-cli is executed without a command to "
-              "execute.\n");
+"Starting interactive mode using -i is deprecated. Interactive mode is started\n"
+"by default when redis-cli is executed without a command to execute.\n"
+            );
@@ -313,9 +296,3 @@ static void usage() {
-  fprintf(stderr,
-          "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r "
-          "repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-a authpw] "
-          "[-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
-  fprintf(stderr,
-          "\nIf a pipe from standard input is detected this data is used as "
-          "last argument.\n\n");
+    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
+    fprintf(stderr, "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
+    fprintf(stderr, "\nIf a pipe from standard input is detected this data is used as last argument.\n\n");
@@ -325,3 +302 @@ static void usage() {
-  fprintf(stderr,
-          "\nRun in interactive mode: redis-cli -i or just don't pass any "
-          "command\n");
+    fprintf(stderr, "\nRun in interactive mode: redis-cli -i or just don't pass any command\n");
@@ -403 +380,0 @@ int main(int argc, char **argv) {
-<<<<<<< HEAD
@@ -405,6 +381,0 @@ int main(int argc, char **argv) {
-|||||||
-  if (argc == 0 || config.interactive == 1) repl();
-=======
-  if (argc == 0) config.interactive = 1;
-  if (config.interactive) repl();
->>>>>>> c0b3d42372dbe67c6ef096372869e2b60d4a1cdc
