diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_merged.cc
index 94bfadc..539f939 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/cfd05080-1cdb2324-172f61a3/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-std::string Version() < < < < < < < HEAD { return "1.32.0-dev"; }
-|||||||
-{ return "1.31.0-dev"; }
-=======
-{ return "1.31.0"; }
->>>>>>> 1cdb23240d6f5bdc38d3992b8caa30c104c20929
+std::string Version() { return "1.32.0-dev"; }
