--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/191a8365-f3931497-674ec410/format.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/191a8365-f3931497-674ec410/format.no_comments_merged.c
@@ -15,3 +15 @@
-struct format_entry
-    static char *
-    format_job_get(struct format_tree *, const char *);
+static char *format_job_get(struct format_tree *, const char *);
@@ -79,0 +83 @@ static int format_entry_cmp(struct format_entry *, struct format_entry *);
+RB_GENERATE_STATIC(format_entry_tree, format_entry, entry, format_entry_cmp);
@@ -164 +172,4 @@ static void printflike(3, 4)
-static void format_job_update(struct job *job) {
+#define format_log(ft,fmt,...) format_log1(ft, __func__, fmt, ##__VA_ARGS__)
+static void
+format_job_update(struct job *job)
+{
@@ -487,2 +582,2 @@ static char *format_cb_current_path(struct format_tree *ft) {
-  if (wp == NULL) return (NULL);
-<<<<<<< HEAD
+ if (wp == NULL)
+  return (NULL);
@@ -490,7 +585,2 @@ static char *format_cb_current_path(struct format_tree *ft) {
-  if (cwd != NULL) fe->value = xstrdup(cwd);
-|||||||
-  cwd = get_proc_cwd(wp->fd);
-  if (cwd != NULL) fe->value = xstrdup(cwd);
-=======
-  cwd = get_proc_cwd(wp->fd);
-  if (cwd == NULL) return (NULL);
+ if (cwd == NULL)
+  return (NULL);
@@ -498 +587,0 @@ static char *format_cb_current_path(struct format_tree *ft) {
->>>>>>> f3931497f8aee291bce132fb106cedb55c5b3fa9
