diff --git a/./output/redis/04233097-f1f259de-fdaea2a7/src@redismodule.no_comments_conflict.h b/./output/redis/04233097-f1f259de-fdaea2a7/src@redismodule.no_comments_merged.h
index 2112add..f4699ed 100644
--- a/./output/redis/04233097-f1f259de-fdaea2a7/src@redismodule.no_comments_conflict.h
+++ b/./output/redis/04233097-f1f259de-fdaea2a7/src@redismodule.no_comments_merged.h
@@ -250 +249,0 @@ typedef struct RedisModuleInfoCtx RedisModuleInfoCtx;
-<<<<<<< HEAD
@@ -252,2 +250,0 @@ typedef struct RedisModuleServerInfoData RedisModuleServerInfoData;
-||||||| fdaea2a7a
-=======
@@ -255 +251,0 @@ typedef struct RedisModuleScanCursor RedisModuleScanCursor;
->>>>>>> f1f259de
