--- a/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_conflict.h
+++ b/./output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_merged.h
@@ -172 +171,0 @@ class SnapshotManager final : public ISnapshotManager {
-<<<<<<< HEAD
@@ -183,9 +181,0 @@ class SnapshotManager final : public ISnapshotManager {
-||||||| 6834fe66d
-=======
-    using MergeConsistencyChecker =
-            std::function<MergeFailureCode(const std::string& name, const SnapshotStatus& status)>;
-    void set_merge_consistency_checker(MergeConsistencyChecker checker) {
-        merge_consistency_checker_ = checker;
-    }
-    MergeConsistencyChecker merge_consistency_checker() const { return merge_consistency_checker_; }
->>>>>>> 739f4f5f
@@ -402 +391,0 @@ class SnapshotManager final : public ISnapshotManager {
-<<<<<<< HEAD
@@ -405,4 +393,0 @@ class SnapshotManager final : public ISnapshotManager {
-||||||| 6834fe66d
-=======
-    MergeConsistencyChecker merge_consistency_checker_;
->>>>>>> 739f4f5f
