--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/d7d520c9-739f4f5f-6834fe66/fs_mgr@libsnapshot@include@libsnapshot@snapshot.no_comments_merged.h
@@ -23 +22,0 @@
-#include <libsnapshot/snapuserd_client.h>
@@ -88 +83 @@ class ISnapshotManager {
-  virtual ~ISnapshotManager()
+    virtual ~ISnapshotManager() = default;
@@ -139,2 +126,0 @@ class SnapshotManager final : public ISnapshotManager {
- private:
-  FRIEND_TEST(SnapshotUpdateTest, SpaceSwapUpdate);
@@ -203,0 +183,24 @@ class SnapshotManager final : public ISnapshotManager {
+    FRIEND_TEST(SnapshotTest, CleanFirstStageMount);
+    FRIEND_TEST(SnapshotTest, CreateSnapshot);
+    FRIEND_TEST(SnapshotTest, FirstStageMountAfterRollback);
+    FRIEND_TEST(SnapshotTest, FirstStageMountAndMerge);
+    FRIEND_TEST(SnapshotTest, FlashSuperDuringMerge);
+    FRIEND_TEST(SnapshotTest, FlashSuperDuringUpdate);
+    FRIEND_TEST(SnapshotTest, MapPartialSnapshot);
+    FRIEND_TEST(SnapshotTest, MapSnapshot);
+    FRIEND_TEST(SnapshotTest, Merge);
+    FRIEND_TEST(SnapshotTest, MergeFailureCode);
+    FRIEND_TEST(SnapshotTest, NoMergeBeforeReboot);
+    FRIEND_TEST(SnapshotTest, UpdateBootControlHal);
+    FRIEND_TEST(SnapshotUpdateTest, AddPartition);
+    FRIEND_TEST(SnapshotUpdateTest, ConsistencyCheckResume);
+    FRIEND_TEST(SnapshotUpdateTest, DaemonTransition);
+    FRIEND_TEST(SnapshotUpdateTest, DataWipeAfterRollback);
+    FRIEND_TEST(SnapshotUpdateTest, DataWipeRollbackInRecovery);
+    FRIEND_TEST(SnapshotUpdateTest, DataWipeWithStaleSnapshots);
+    FRIEND_TEST(SnapshotUpdateTest, FullUpdateFlow);
+    FRIEND_TEST(SnapshotUpdateTest, MergeCannotRemoveCow);
+    FRIEND_TEST(SnapshotUpdateTest, MergeInRecovery);
+    FRIEND_TEST(SnapshotUpdateTest, QueryStatusError);
+    FRIEND_TEST(SnapshotUpdateTest, SnapshotStatusFileWithoutCow);
+    FRIEND_TEST(SnapshotUpdateTest, SpaceSwapUpdate);
@@ -411 +383,0 @@ class SnapshotManager final : public ISnapshotManager {
-  std::string metadata_dir_;
@@ -412,0 +385 @@ class SnapshotManager final : public ISnapshotManager {
+    std::string metadata_dir_;
