diff --git a/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_mergegen.h b/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_merged.h
index d9bb600..971bc66 100644
--- a/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_mergegen.h
+++ b/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_merged.h
@@ -205,0 +206,2 @@ private:
+    std::optional<std::vector<uint32_t>> mReturnedVolume;
+    std::string mEffectInterfaceDebug;
