diff --git a/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_conflict.h b/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_merged.h
index dd9253d..971bc66 100644
--- a/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_conflict.h
+++ b/./output/frameworks_av/937b85a4-12f41a9a-eda0965e/services@audioflinger@Effects.no_comments_merged.h
@@ -206 +205,0 @@ private:
-<<<<<<< HEAD
@@ -209,4 +207,0 @@ private:
-||||||| eda0965e84
-=======
-    std::optional<std::vector<uint32_t>> mReturnedVolume;
->>>>>>> 12f41a9a
