--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_mergegen.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_merged.cpp
@@ -318 +318,2 @@ void ReplayState::ReplayCreateIndex() {
-case IndexType::ART: { index = make_uniq<ART>(info->column_ids, TableIOManager::Get(data_table), expressions, info->constraint_type, data_table.db);
+  index = make_uniq<ART>(info->column_ids, TableIOManager::Get(data_table), expressions, info->constraint_type,
+                           data_table.db);
