--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@execution@operator@schema@physical_create_index.no_comments_mergegen.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@execution@operator@schema@physical_create_index.no_comments_merged.cpp
@@ -38 +38,2 @@ unique_ptr<GlobalSinkState> PhysicalCreateIndex::GetGlobalSinkState(ClientContex
-auto &storage = table.GetStorage(); state->global_index = make_uniq<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions, info->constraint_type, storage.db);
+  state->global_index = make_uniq<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
+                                         info->constraint_type, storage.db);
@@ -51 +52,2 @@ unique_ptr<LocalSinkState> PhysicalCreateIndex::GetLocalSinkState(ExecutionConte
-auto &storage = table.GetStorage(); state->local_index = make_uniq<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions, info->constraint_type, storage.db);
+  state->local_index = make_uniq<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
+                                        info->constraint_type, storage.db);
