diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_merged.cpp
index 90f1169..d5490c4 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_merged.cpp
@@ -99,7 +98,0 @@ LocalTableStorage::LocalTableStorage(DataTable &table)
-<<<<<<< HEAD
-   indexes.AddIndex(make_unique<ART>(art.column_ids, art.table_io_manager, std::move(unbound_expressions),
-                                     art.constraint_type, art.db));
-|||||||
-   indexes.AddIndex(make_unique<ART>(art.column_ids, art.table_io_manager, std::move(unbound_expressions),
-                                     art.constraint_type, art.db, true));
-=======
@@ -107,2 +100 @@ LocalTableStorage::LocalTableStorage(DataTable &table)
-                                   art.constraint_type, art.db, true));
->>>>>>> d84e329b281c1246646f5e7c5388bdf11df1e78a
+                                     art.constraint_type, art.db));
@@ -133 +124,0 @@ LocalTableStorage::LocalTableStorage(ClientContext &context, DataTable &new_dt,
- row_groups {
@@ -169,2 +160,2 @@ LocalTableStorage::LocalTableStorage(ClientContext &context, DataTable &new_dt,
-                                                   TableIndexList & index_list,
-                                                   const vector<LogicalType> &table_types, row_t &start_row) {
+                                                  TableIndexList &index_list, const vector<LogicalType> &table_types,
+                                                  row_t &start_row) {
@@ -359,2 +350 @@ LocalTableStorage::LocalTableStorage(ClientContext &context, DataTable &new_dt,
-   auto error =
-       storage->AppendToIndexes(transaction, collection, storage->indexes, table->GetTypes(), base_id);
+  auto error = storage->AppendToIndexes(transaction, collection, storage->indexes, table->GetTypes(), base_id);
@@ -474,3 +464,2 @@ LocalTableStorage::LocalTableStorage(ClientContext &context, DataTable &new_dt,
- void LocalStorage::ChangeType(DataTable * old_dt, DataTable * new_dt, idx_t changed_idx,
-                               const LogicalType &target_type, const vector<column_t> &bound_columns,
-                               Expression &cast_expr) {
+void LocalStorage::ChangeType(DataTable *old_dt, DataTable *new_dt, idx_t changed_idx, const LogicalType &target_type,
+                              const vector<column_t> &bound_columns, Expression &cast_expr) {
@@ -481,2 +470,2 @@ LocalTableStorage::LocalTableStorage(ClientContext &context, DataTable &new_dt,
-  auto new_storage = make_shared<LocalTableStorage>(context, *new_dt, *storage, changed_idx, target_type,
-                                                    bound_columns, cast_expr);
+ auto new_storage =
+     make_shared<LocalTableStorage>(context, *new_dt, *storage, changed_idx, target_type, bound_columns, cast_expr);
