diff --git a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_conflict.cpp b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_merged.cpp
index 6ed23b8..d5490c4 100644
--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_conflict.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@local_storage.no_comments_merged.cpp
@@ -99,7 +98,0 @@ LocalTableStorage::LocalTableStorage(DataTable &table)
-<<<<<<< HEAD
-   indexes.AddIndex(make_unique<ART>(art.column_ids, art.table_io_manager, std::move(unbound_expressions),
-                                     art.constraint_type, art.db));
-||||||| 3dcba6d0b5
-   indexes.AddIndex(make_unique<ART>(art.column_ids, art.table_io_manager, std::move(unbound_expressions),
-                                     art.constraint_type, art.db, true));
-=======
@@ -107,2 +100 @@ LocalTableStorage::LocalTableStorage(DataTable &table)
-                                   art.constraint_type, art.db, true));
->>>>>>> d84e329b
+                                     art.constraint_type, art.db));
