diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_merged.cpp
index 52a4c38..3365b87 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@storage@wal_replay.no_comments_merged.cpp
@@ -318,7 +317,0 @@ void ReplayState::ReplayCreateIndex() {
-<<<<<<< HEAD
-  index = make_unique<ART>(info->column_ids, TableIOManager::Get(data_table), expressions, info->constraint_type,
-                           data_table.db);
-|||||||
-  index = make_unique<ART>(info->column_ids, TableIOManager::Get(data_table), expressions, info->constraint_type,
-                           data_table.db, true);
-=======
@@ -326,2 +319 @@ void ReplayState::ReplayCreateIndex() {
-                         data_table.db, true);
->>>>>>> d84e329b281c1246646f5e7c5388bdf11df1e78a
+                           data_table.db);
