--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@execution@operator@schema@physical_create_index.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/467d79b9-d84e329b-3dcba6d0/src@execution@operator@schema@physical_create_index.no_comments_merged.cpp
@@ -26,2 +26 @@ public:
- explicit CreateIndexLocalSinkState(ClientContext &context) : arena_allocator(Allocator::Get(context)) {
- }
+ explicit CreateIndexLocalSinkState(ClientContext &context) : arena_allocator(Allocator::Get(context)) {};
@@ -39,7 +37,0 @@ unique_ptr<GlobalSinkState> PhysicalCreateIndex::GetGlobalSinkState(ClientContex
-<<<<<<< HEAD
-  state->global_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                         info->constraint_type, storage.db);
-|||||||
-  state->global_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                         info->constraint_type, storage.db, true);
-=======
@@ -47,2 +39 @@ unique_ptr<GlobalSinkState> PhysicalCreateIndex::GetGlobalSinkState(ClientContex
-                                       info->constraint_type, storage.db, true);
->>>>>>> d84e329b281c1246646f5e7c5388bdf11df1e78a
+                                         info->constraint_type, storage.db);
@@ -61,7 +51,0 @@ unique_ptr<LocalSinkState> PhysicalCreateIndex::GetLocalSinkState(ExecutionConte
-<<<<<<< HEAD
-  state->local_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                        info->constraint_type, storage.db);
-|||||||
-  state->local_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                        info->constraint_type, storage.db, false);
-=======
@@ -69,2 +53 @@ unique_ptr<LocalSinkState> PhysicalCreateIndex::GetLocalSinkState(ExecutionConte
-                                      info->constraint_type, storage.db, false);
->>>>>>> d84e329b281c1246646f5e7c5388bdf11df1e78a
+                                        info->constraint_type, storage.db);
@@ -92,9 +74,0 @@ SinkResultType PhysicalCreateIndex::Sink(ExecutionContext &context, GlobalSinkSt
-<<<<<<< HEAD
- auto art =
-     make_unique<ART>(lstate.local_index->column_ids, lstate.local_index->table_io_manager,
-                      lstate.local_index->unbound_expressions, lstate.local_index->constraint_type, storage.db);
-|||||||
- auto art = make_unique<ART>(lstate.local_index->column_ids, lstate.local_index->table_io_manager,
-                             lstate.local_index->unbound_expressions, lstate.local_index->constraint_type,
-                             storage.db, false);
-=======
@@ -103,2 +77 @@ SinkResultType PhysicalCreateIndex::Sink(ExecutionContext &context, GlobalSinkSt
-                    lstate.local_index->unbound_expressions, lstate.local_index->constraint_type, storage.db, false);
->>>>>>> d84e329b281c1246646f5e7c5388bdf11df1e78a
+                      lstate.local_index->unbound_expressions, lstate.local_index->constraint_type, storage.db);
