--- a/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@execution@operator@schema@physical_create_index.no_comments_conflict.cpp
+++ b/./output/duckdb/467d79b9-d84e329b-3dcba6d0/src@execution@operator@schema@physical_create_index.no_comments_merged.cpp
@@ -38,7 +37,0 @@ unique_ptr<GlobalSinkState> PhysicalCreateIndex::GetGlobalSinkState(ClientContex
-<<<<<<< HEAD
-  state->global_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                         info->constraint_type, storage.db);
-||||||| 3dcba6d0b5
-  state->global_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                         info->constraint_type, storage.db, true);
-=======
@@ -46,2 +39 @@ unique_ptr<GlobalSinkState> PhysicalCreateIndex::GetGlobalSinkState(ClientContex
-                                       info->constraint_type, storage.db, true);
->>>>>>> d84e329b
+                                         info->constraint_type, storage.db);
@@ -60,7 +51,0 @@ unique_ptr<LocalSinkState> PhysicalCreateIndex::GetLocalSinkState(ExecutionConte
-<<<<<<< HEAD
-  state->local_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                        info->constraint_type, storage.db);
-||||||| 3dcba6d0b5
-  state->local_index = make_unique<ART>(storage_ids, TableIOManager::Get(storage), unbound_expressions,
-                                        info->constraint_type, storage.db, false);
-=======
@@ -68,2 +53 @@ unique_ptr<LocalSinkState> PhysicalCreateIndex::GetLocalSinkState(ExecutionConte
-                                      info->constraint_type, storage.db, false);
->>>>>>> d84e329b
+                                        info->constraint_type, storage.db);
@@ -91,9 +74,0 @@ SinkResultType PhysicalCreateIndex::Sink(ExecutionContext &context, GlobalSinkSt
-<<<<<<< HEAD
- auto art =
-     make_unique<ART>(lstate.local_index->column_ids, lstate.local_index->table_io_manager,
-                      lstate.local_index->unbound_expressions, lstate.local_index->constraint_type, storage.db);
-||||||| 3dcba6d0b5
- auto art = make_unique<ART>(lstate.local_index->column_ids, lstate.local_index->table_io_manager,
-                             lstate.local_index->unbound_expressions, lstate.local_index->constraint_type,
-                             storage.db, false);
-=======
@@ -102,2 +77 @@ SinkResultType PhysicalCreateIndex::Sink(ExecutionContext &context, GlobalSinkSt
-                    lstate.local_index->unbound_expressions, lstate.local_index->constraint_type, storage.db, false);
->>>>>>> d84e329b
+                      lstate.local_index->unbound_expressions, lstate.local_index->constraint_type, storage.db);
