diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_merged.c
index e042354..facb583 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_merged.c
@@ -7,2 +6,0 @@
-#include <imsg.h>
-#include <paths.h>
@@ -31,7 +29,14 @@ static void server_client_dispatch_shell(struct client *);
-static void server_client_dispatch_write_ready(struct client *, struct imsg *);
-static void server_client_dispatch_read_data(struct client *, struct imsg *);
-static void server_client_dispatch_read_done(struct client *, struct imsg *);
-static int server_client_window_cmp(struct client_window *cw1,
-                                    struct client_window *cw2) {
-  if (cw1->window < cw2->window) return (-1);
-  if (cw1->window > cw2->window) return (1);
+static void server_client_dispatch_write_ready(struct client *,
+      struct imsg *);
+static void server_client_dispatch_read_data(struct client *,
+      struct imsg *);
+static void server_client_dispatch_read_done(struct client *,
+      struct imsg *);
+static int
+server_client_window_cmp(struct client_window *cw1,
+    struct client_window *cw2)
+{
+ if (cw1->window < cw2->window)
+  return (-1);
+ if (cw1->window > cw2->window)
+  return (1);
@@ -40 +45,4 @@ static int server_client_window_cmp(struct client_window *cw1,
-u_int server_client_how_many(void) {
+RB_GENERATE(client_windows, client_window, entry, server_client_window_cmp);
+u_int
+server_client_how_many(void)
+{
@@ -45 +53,2 @@ u_int server_client_how_many(void) {
-    if (c->session != NULL && (~c->flags & CLIENT_UNATTACHEDFLAGS)) n++;
+  if (c->session != NULL && (~c->flags & CLIENT_UNATTACHEDFLAGS))
+   n++;
@@ -49,2 +58,3 @@ u_int server_client_how_many(void) {
-static void server_client_overlay_timer(__unused int fd, __unused short events,
-                                        void *data) {
+static void
+server_client_overlay_timer(__unused int fd, __unused short events, void *data)
+{
@@ -53 +63,2 @@ static void server_client_overlay_timer(__unused int fd, __unused short events,
-void server_client_set_overlay(struct client *c, u_int delay,
+void
+server_client_set_overlay(struct client *c, u_int delay,
@@ -55,2 +66,3 @@ void server_client_set_overlay(struct client *c, u_int delay,
-                               overlay_draw_cb drawcb, overlay_key_cb keycb,
-                               overlay_free_cb freecb, void *data) {
+    overlay_draw_cb drawcb, overlay_key_cb keycb, overlay_free_cb freecb,
+    void *data)
+{
@@ -58 +70,2 @@ void server_client_set_overlay(struct client *c, u_int delay,
-  if (c->overlay_draw != NULL) server_client_clear_overlay(c);
+ if (c->overlay_draw != NULL)
+  server_client_clear_overlay(c);
@@ -61 +74,2 @@ void server_client_set_overlay(struct client *c, u_int delay,
-  if (event_initialized(&c->overlay_timer)) evtimer_del(&c->overlay_timer);
+ if (event_initialized(&c->overlay_timer))
+  evtimer_del(&c->overlay_timer);
@@ -63 +77,2 @@ void server_client_set_overlay(struct client *c, u_int delay,
-  if (delay != 0) evtimer_add(&c->overlay_timer, &tv);
+ if (delay != 0)
+  evtimer_add(&c->overlay_timer, &tv);
@@ -71 +86,2 @@ void server_client_set_overlay(struct client *c, u_int delay,
-  if (c->overlay_mode == NULL) c->tty.flags |= TTY_NOCURSOR;
+ if (c->overlay_mode == NULL)
+  c->tty.flags |= TTY_NOCURSOR;
@@ -74,4 +90,9 @@ void server_client_set_overlay(struct client *c, u_int delay,
-void server_client_clear_overlay(struct client *c) {
-  if (c->overlay_draw == NULL) return;
-  if (event_initialized(&c->overlay_timer)) evtimer_del(&c->overlay_timer);
-  if (c->overlay_free != NULL) c->overlay_free(c);
+void
+server_client_clear_overlay(struct client *c)
+{
+ if (c->overlay_draw == NULL)
+  return;
+ if (event_initialized(&c->overlay_timer))
+  evtimer_del(&c->overlay_timer);
+ if (c->overlay_free != NULL)
+  c->overlay_free(c);
@@ -87 +108,3 @@ void server_client_clear_overlay(struct client *c) {
-int server_client_check_nested(struct client *c) {
+int
+server_client_check_nested(struct client *c)
+{
@@ -91 +114,2 @@ int server_client_check_nested(struct client *c) {
-  if (envent == NULL || *envent->value == '\0') return (0);
+ if (envent == NULL || *envent->value == '\0')
+  return (0);
@@ -93 +117,2 @@ int server_client_check_nested(struct client *c) {
-    if (strcmp(wp->tty, c->ttyname) == 0) return (1);
+  if (strcmp(wp->tty, c->ttyname) == 0)
+   return (1);
@@ -97,2 +122,5 @@ int server_client_check_nested(struct client *c) {
-void server_client_set_key_table(struct client *c, const char *name) {
-  if (name == NULL) name = server_client_get_key_table(c);
+void
+server_client_set_key_table(struct client *c, const char *name)
+{
+ if (name == NULL)
+  name = server_client_get_key_table(c);
@@ -103 +131,3 @@ void server_client_set_key_table(struct client *c, const char *name) {
-const char *server_client_get_key_table(struct client *c) {
+const char *
+server_client_get_key_table(struct client *c)
+{
@@ -106 +136,2 @@ const char *server_client_get_key_table(struct client *c) {
-  if (s == NULL) return ("root");
+ if (s == NULL)
+  return ("root");
@@ -108 +139,2 @@ const char *server_client_get_key_table(struct client *c) {
-  if (*name == '\0') return ("root");
+ if (*name == '\0')
+  return ("root");
@@ -111,2 +143,3 @@ const char *server_client_get_key_table(struct client *c) {
-static int server_client_is_default_key_table(struct client *c,
-                                              struct key_table *table) {
+static int
+server_client_is_default_key_table(struct client *c, struct key_table *table)
+{
@@ -115 +148,3 @@ static int server_client_is_default_key_table(struct client *c,
-struct client *server_client_create(int fd) {
+struct client *
+server_client_create(int fd)
+{
@@ -121 +156,2 @@ struct client *server_client_create(int fd) {
-  if (gettimeofday(&c->creation_time, NULL) != 0) fatal("gettimeofday failed");
+ if (gettimeofday(&c->creation_time, NULL) != 0)
+  fatal("gettimeofday failed");
@@ -141 +177,3 @@ struct client *server_client_create(int fd) {
-int server_client_open(struct client *c, char **cause) {
+int
+server_client_open(struct client *c, char **cause)
+{
@@ -143 +181,2 @@ int server_client_open(struct client *c, char **cause) {
-  if (c->flags & CLIENT_CONTROL) return (0);
+ if (c->flags & CLIENT_CONTROL)
+  return (0);
@@ -145 +184,2 @@ int server_client_open(struct client *c, char **cause) {
-      ((isatty(STDIN_FILENO) && (ttynam = ttyname(STDIN_FILENO)) != NULL &&
+     ((isatty(STDIN_FILENO) &&
+     (ttynam = ttyname(STDIN_FILENO)) != NULL &&
@@ -147 +187,2 @@ int server_client_open(struct client *c, char **cause) {
-       (isatty(STDOUT_FILENO) && (ttynam = ttyname(STDOUT_FILENO)) != NULL &&
+     (isatty(STDOUT_FILENO) &&
+     (ttynam = ttyname(STDOUT_FILENO)) != NULL &&
@@ -149 +190,2 @@ int server_client_open(struct client *c, char **cause) {
-       (isatty(STDERR_FILENO) && (ttynam = ttyname(STDERR_FILENO)) != NULL &&
+     (isatty(STDERR_FILENO) &&
+     (ttynam = ttyname(STDERR_FILENO)) != NULL &&
@@ -158 +200,2 @@ int server_client_open(struct client *c, char **cause) {
-  if (tty_open(&c->tty, cause) != 0) return (-1);
+ if (tty_open(&c->tty, cause) != 0)
+  return (-1);
@@ -161 +204,3 @@ int server_client_open(struct client *c, char **cause) {
-void server_client_lost(struct client *c) {
+void
+server_client_lost(struct client *c)
+{
@@ -178,2 +223,4 @@ void server_client_lost(struct client *c) {
-  if (c->flags & CLIENT_CONTROL) control_stop(c);
-  if (c->flags & CLIENT_TERMINAL) tty_free(&c->tty);
+ if (c->flags & CLIENT_CONTROL)
+  control_stop(c);
+ if (c->flags & CLIENT_TERMINAL)
+  tty_free(&c->tty);
@@ -190 +237,2 @@ void server_client_lost(struct client *c) {
-  if (event_initialized(&c->message_timer)) evtimer_del(&c->message_timer);
+ if (event_initialized(&c->message_timer))
+  evtimer_del(&c->message_timer);
@@ -198 +246,2 @@ void server_client_lost(struct client *c) {
-  if (c->out_fd != -1) close(c->out_fd);
+ if (c->out_fd != -1)
+  close(c->out_fd);
@@ -209 +258,3 @@ void server_client_lost(struct client *c) {
-void server_client_unref(struct client *c) {
+void
+server_client_unref(struct client *c)
+{
@@ -215,2 +266,3 @@ void server_client_unref(struct client *c) {
-static void server_client_free(__unused int fd, __unused short events,
-                               void *arg) {
+static void
+server_client_free(__unused int fd, __unused short events, void *arg)
+{
@@ -225 +277,3 @@ static void server_client_free(__unused int fd, __unused short events,
-void server_client_suspend(struct client *c) {
+void
+server_client_suspend(struct client *c)
+{
@@ -227 +281,2 @@ void server_client_suspend(struct client *c) {
-  if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS)) return;
+ if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS))
+  return;
@@ -232 +287,3 @@ void server_client_suspend(struct client *c) {
-void server_client_detach(struct client *c, enum msgtype msgtype) {
+void
+server_client_detach(struct client *c, enum msgtype msgtype)
+{
@@ -234 +291,2 @@ void server_client_detach(struct client *c, enum msgtype msgtype) {
-  if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS)) return;
+ if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS))
+  return;
@@ -240 +298,3 @@ void server_client_detach(struct client *c, enum msgtype msgtype) {
-void server_client_exec(struct client *c, const char *cmd) {
+void
+server_client_exec(struct client *c, const char *cmd)
+{
@@ -245 +305,2 @@ void server_client_exec(struct client *c, const char *cmd) {
-  if (*cmd == '\0') return;
+ if (*cmd == '\0')
+  return;
@@ -251 +312,2 @@ void server_client_exec(struct client *c, const char *cmd) {
-  if (!checkshell(shell)) shell = _PATH_BSHELL;
+ if (!checkshell(shell))
+  shell = _PATH_BSHELL;
@@ -259,2 +321,3 @@ void server_client_exec(struct client *c, const char *cmd) {
-static key_code server_client_check_mouse(struct client *c,
-                                          struct key_event *event) {
+static key_code
+server_client_check_mouse(struct client *c, struct key_event *event)
+{
@@ -270,2 +333 @@ static key_code server_client_check_mouse(struct client *c,
-  enum {
-    NOTYPE,
+ enum { NOTYPE,
@@ -279,4 +341,2 @@ static key_code server_client_check_mouse(struct client *c,
-    TRIPLE
-  } type = NOTYPE;
-  enum {
-    NOWHERE,
+        TRIPLE } type = NOTYPE;
+ enum { NOWHERE,
@@ -288,4 +348,3 @@ static key_code server_client_check_mouse(struct client *c,
-    BORDER
-  } where = NOWHERE;
-  log_debug("%s mouse %02x at %u,%u (last %u,%u) (%d)", c->name, m->b, m->x,
-            m->y, m->lx, m->ly, c->tty.mouse_drag_flag);
+        BORDER } where = NOWHERE;
+ log_debug("%s mouse %02x at %u,%u (last %u,%u) (%d)", c->name, m->b,
+     m->x, m->y, m->lx, m->ly, c->tty.mouse_drag_flag);
@@ -297 +356,2 @@ static key_code server_client_check_mouse(struct client *c,
-  } else if ((m->sgr_type != ' ' && MOUSE_DRAG(m->sgr_b) &&
+ } else if ((m->sgr_type != ' ' &&
+     MOUSE_DRAG(m->sgr_b) &&
@@ -299,2 +359,4 @@ static key_code server_client_check_mouse(struct client *c,
-             (m->sgr_type == ' ' && MOUSE_DRAG(m->b) &&
-              MOUSE_BUTTONS(m->b) == 3 && MOUSE_BUTTONS(m->lb) == 3)) {
+     (m->sgr_type == ' ' &&
+     MOUSE_DRAG(m->b) &&
+     MOUSE_BUTTONS(m->b) == 3 &&
+     MOUSE_BUTTONS(m->lb) == 3)) {
@@ -308 +370,2 @@ static key_code server_client_check_mouse(struct client *c,
-      if (x == m->lx && y == m->ly) return (KEYC_UNKNOWN);
+   if (x == m->lx && y == m->ly)
+    return (KEYC_UNKNOWN);
@@ -358 +421,2 @@ have_event:
-  if (type == NOTYPE) return (KEYC_UNKNOWN);
+ if (type == NOTYPE)
+  return (KEYC_UNKNOWN);
@@ -364 +428,2 @@ have_event:
-  if (m->statusat != -1 && y >= (u_int)m->statusat &&
+ if (m->statusat != -1 &&
+     y >= (u_int)m->statusat &&
@@ -380,2 +445,4 @@ have_event:
-          wl = winlink_find_by_index(&s->windows, sr->argument);
-          if (wl == NULL) return (KEYC_UNKNOWN);
+    wl = winlink_find_by_index(&s->windows,
+        sr->argument);
+    if (wl == NULL)
+     return (KEYC_UNKNOWN);
@@ -397,3 +464,4 @@ have_event:
-    log_debug("mouse window @%u at %u,%u (%ux%u)", s->curw->window->id, m->ox,
-              m->oy, sx, sy);
-    if (px > sx || py > sy) return (KEYC_UNKNOWN);
+  log_debug("mouse window @%u at %u,%u (%ux%u)",
+      s->curw->window->id, m->ox, m->oy, sx, sy);
+  if (px > sx || py > sy)
+   return (KEYC_UNKNOWN);
@@ -404 +472,2 @@ have_event:
-        if ((wp->xoff + wp->sx == px && wp->yoff <= 1 + py &&
+    if ((wp->xoff + wp->sx == px &&
+        wp->yoff <= 1 + py &&
@@ -406 +475,2 @@ have_event:
-            (wp->yoff + wp->sy == py && wp->xoff <= 1 + px &&
+        (wp->yoff + wp->sy == py &&
+        wp->xoff <= 1 + px &&
@@ -410 +480,2 @@ have_event:
-      if (wp != NULL) where = BORDER;
+   if (wp != NULL)
+    where = BORDER;
@@ -428 +499,2 @@ have_event:
-    if (c->tty.mouse_drag_release != NULL) c->tty.mouse_drag_release(c, m);
+  if (c->tty.mouse_drag_release != NULL)
+   c->tty.mouse_drag_release(c, m);
@@ -433,6 +505,12 @@ have_event:
-        if (where == PANE) key = KEYC_MOUSEDRAGEND1_PANE;
-        if (where == STATUS) key = KEYC_MOUSEDRAGEND1_STATUS;
-        if (where == STATUS_LEFT) key = KEYC_MOUSEDRAGEND1_STATUS_LEFT;
-        if (where == STATUS_RIGHT) key = KEYC_MOUSEDRAGEND1_STATUS_RIGHT;
-        if (where == STATUS_DEFAULT) key = KEYC_MOUSEDRAGEND1_STATUS_DEFAULT;
-        if (where == BORDER) key = KEYC_MOUSEDRAGEND1_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEDRAGEND1_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEDRAGEND1_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEDRAGEND1_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEDRAGEND1_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEDRAGEND1_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEDRAGEND1_BORDER;
@@ -441,6 +519,12 @@ have_event:
-        if (where == PANE) key = KEYC_MOUSEDRAGEND2_PANE;
-        if (where == STATUS) key = KEYC_MOUSEDRAGEND2_STATUS;
-        if (where == STATUS_LEFT) key = KEYC_MOUSEDRAGEND2_STATUS_LEFT;
-        if (where == STATUS_RIGHT) key = KEYC_MOUSEDRAGEND2_STATUS_RIGHT;
-        if (where == STATUS_DEFAULT) key = KEYC_MOUSEDRAGEND2_STATUS_DEFAULT;
-        if (where == BORDER) key = KEYC_MOUSEDRAGEND2_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEDRAGEND2_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEDRAGEND2_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEDRAGEND2_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEDRAGEND2_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEDRAGEND2_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEDRAGEND2_BORDER;
@@ -449,6 +533,12 @@ have_event:
-        if (where == PANE) key = KEYC_MOUSEDRAGEND3_PANE;
-        if (where == STATUS) key = KEYC_MOUSEDRAGEND3_STATUS;
-        if (where == STATUS_LEFT) key = KEYC_MOUSEDRAGEND3_STATUS_LEFT;
-        if (where == STATUS_RIGHT) key = KEYC_MOUSEDRAGEND3_STATUS_RIGHT;
-        if (where == STATUS_DEFAULT) key = KEYC_MOUSEDRAGEND3_STATUS_DEFAULT;
-        if (where == BORDER) key = KEYC_MOUSEDRAGEND3_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEDRAGEND3_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEDRAGEND3_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEDRAGEND3_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEDRAGEND3_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEDRAGEND3_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEDRAGEND3_BORDER;
@@ -468,6 +558,12 @@ have_event:
-      if (where == PANE) key = KEYC_MOUSEMOVE_PANE;
-      if (where == STATUS) key = KEYC_MOUSEMOVE_STATUS;
-      if (where == STATUS_LEFT) key = KEYC_MOUSEMOVE_STATUS_LEFT;
-      if (where == STATUS_RIGHT) key = KEYC_MOUSEMOVE_STATUS_RIGHT;
-      if (where == STATUS_DEFAULT) key = KEYC_MOUSEMOVE_STATUS_DEFAULT;
-      if (where == BORDER) key = KEYC_MOUSEMOVE_BORDER;
+  if (where == PANE)
+   key = KEYC_MOUSEMOVE_PANE;
+  if (where == STATUS)
+   key = KEYC_MOUSEMOVE_STATUS;
+  if (where == STATUS_LEFT)
+   key = KEYC_MOUSEMOVE_STATUS_LEFT;
+  if (where == STATUS_RIGHT)
+   key = KEYC_MOUSEMOVE_STATUS_RIGHT;
+  if (where == STATUS_DEFAULT)
+   key = KEYC_MOUSEMOVE_STATUS_DEFAULT;
+  if (where == BORDER)
+   key = KEYC_MOUSEMOVE_BORDER;
@@ -481,6 +577,12 @@ have_event:
-            if (where == PANE) key = KEYC_MOUSEDRAG1_PANE;
-            if (where == STATUS) key = KEYC_MOUSEDRAG1_STATUS;
-            if (where == STATUS_LEFT) key = KEYC_MOUSEDRAG1_STATUS_LEFT;
-            if (where == STATUS_RIGHT) key = KEYC_MOUSEDRAG1_STATUS_RIGHT;
-            if (where == STATUS_DEFAULT) key = KEYC_MOUSEDRAG1_STATUS_DEFAULT;
-            if (where == BORDER) key = KEYC_MOUSEDRAG1_BORDER;
+    if (where == PANE)
+     key = KEYC_MOUSEDRAG1_PANE;
+    if (where == STATUS)
+     key = KEYC_MOUSEDRAG1_STATUS;
+    if (where == STATUS_LEFT)
+     key = KEYC_MOUSEDRAG1_STATUS_LEFT;
+    if (where == STATUS_RIGHT)
+     key = KEYC_MOUSEDRAG1_STATUS_RIGHT;
+    if (where == STATUS_DEFAULT)
+     key = KEYC_MOUSEDRAG1_STATUS_DEFAULT;
+    if (where == BORDER)
+     key = KEYC_MOUSEDRAG1_BORDER;
@@ -489,6 +591,12 @@ have_event:
-            if (where == PANE) key = KEYC_MOUSEDRAG2_PANE;
-            if (where == STATUS) key = KEYC_MOUSEDRAG2_STATUS;
-            if (where == STATUS_LEFT) key = KEYC_MOUSEDRAG2_STATUS_LEFT;
-            if (where == STATUS_RIGHT) key = KEYC_MOUSEDRAG2_STATUS_RIGHT;
-            if (where == STATUS_DEFAULT) key = KEYC_MOUSEDRAG2_STATUS_DEFAULT;
-            if (where == BORDER) key = KEYC_MOUSEDRAG2_BORDER;
+    if (where == PANE)
+     key = KEYC_MOUSEDRAG2_PANE;
+    if (where == STATUS)
+     key = KEYC_MOUSEDRAG2_STATUS;
+    if (where == STATUS_LEFT)
+     key = KEYC_MOUSEDRAG2_STATUS_LEFT;
+    if (where == STATUS_RIGHT)
+     key = KEYC_MOUSEDRAG2_STATUS_RIGHT;
+    if (where == STATUS_DEFAULT)
+     key = KEYC_MOUSEDRAG2_STATUS_DEFAULT;
+    if (where == BORDER)
+     key = KEYC_MOUSEDRAG2_BORDER;
@@ -497,6 +605,12 @@ have_event:
-            if (where == PANE) key = KEYC_MOUSEDRAG3_PANE;
-            if (where == STATUS) key = KEYC_MOUSEDRAG3_STATUS;
-            if (where == STATUS_LEFT) key = KEYC_MOUSEDRAG3_STATUS_LEFT;
-            if (where == STATUS_RIGHT) key = KEYC_MOUSEDRAG3_STATUS_RIGHT;
-            if (where == STATUS_DEFAULT) key = KEYC_MOUSEDRAG3_STATUS_DEFAULT;
-            if (where == BORDER) key = KEYC_MOUSEDRAG3_BORDER;
+    if (where == PANE)
+     key = KEYC_MOUSEDRAG3_PANE;
+    if (where == STATUS)
+     key = KEYC_MOUSEDRAG3_STATUS;
+    if (where == STATUS_LEFT)
+     key = KEYC_MOUSEDRAG3_STATUS_LEFT;
+    if (where == STATUS_RIGHT)
+     key = KEYC_MOUSEDRAG3_STATUS_RIGHT;
+    if (where == STATUS_DEFAULT)
+     key = KEYC_MOUSEDRAG3_STATUS_DEFAULT;
+    if (where == BORDER)
+     key = KEYC_MOUSEDRAG3_BORDER;
@@ -510,6 +624,12 @@ have_event:
-        if (where == PANE) key = KEYC_WHEELUP_PANE;
-        if (where == STATUS) key = KEYC_WHEELUP_STATUS;
-        if (where == STATUS_LEFT) key = KEYC_WHEELUP_STATUS_LEFT;
-        if (where == STATUS_RIGHT) key = KEYC_WHEELUP_STATUS_RIGHT;
-        if (where == STATUS_DEFAULT) key = KEYC_WHEELUP_STATUS_DEFAULT;
-        if (where == BORDER) key = KEYC_WHEELUP_BORDER;
+   if (where == PANE)
+    key = KEYC_WHEELUP_PANE;
+   if (where == STATUS)
+    key = KEYC_WHEELUP_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_WHEELUP_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_WHEELUP_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_WHEELUP_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_WHEELUP_BORDER;
@@ -517,6 +637,12 @@ have_event:
-        if (where == PANE) key = KEYC_WHEELDOWN_PANE;
-        if (where == STATUS) key = KEYC_WHEELDOWN_STATUS;
-        if (where == STATUS_LEFT) key = KEYC_WHEELDOWN_STATUS_LEFT;
-        if (where == STATUS_RIGHT) key = KEYC_WHEELDOWN_STATUS_RIGHT;
-        if (where == STATUS_DEFAULT) key = KEYC_WHEELDOWN_STATUS_DEFAULT;
-        if (where == BORDER) key = KEYC_WHEELDOWN_BORDER;
+   if (where == PANE)
+    key = KEYC_WHEELDOWN_PANE;
+   if (where == STATUS)
+    key = KEYC_WHEELDOWN_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_WHEELDOWN_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_WHEELDOWN_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_WHEELDOWN_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_WHEELDOWN_BORDER;
@@ -528,6 +654,12 @@ have_event:
-          if (where == PANE) key = KEYC_MOUSEUP1_PANE;
-          if (where == STATUS) key = KEYC_MOUSEUP1_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_MOUSEUP1_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_MOUSEUP1_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_MOUSEUP1_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_MOUSEUP1_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEUP1_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEUP1_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEUP1_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEUP1_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEUP1_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEUP1_BORDER;
@@ -536,6 +668,12 @@ have_event:
-          if (where == PANE) key = KEYC_MOUSEUP2_PANE;
-          if (where == STATUS) key = KEYC_MOUSEUP2_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_MOUSEUP2_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_MOUSEUP2_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_MOUSEUP2_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_MOUSEUP2_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEUP2_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEUP2_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEUP2_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEUP2_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEUP2_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEUP2_BORDER;
@@ -544,6 +682,12 @@ have_event:
-          if (where == PANE) key = KEYC_MOUSEUP3_PANE;
-          if (where == STATUS) key = KEYC_MOUSEUP3_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_MOUSEUP3_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_MOUSEUP3_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_MOUSEUP3_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_MOUSEUP3_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEUP3_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEUP3_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEUP3_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEUP3_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEUP3_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEUP3_BORDER;
@@ -556,6 +700,12 @@ have_event:
-          if (where == PANE) key = KEYC_MOUSEDOWN1_PANE;
-          if (where == STATUS) key = KEYC_MOUSEDOWN1_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_MOUSEDOWN1_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_MOUSEDOWN1_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_MOUSEDOWN1_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_MOUSEDOWN1_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEDOWN1_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEDOWN1_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEDOWN1_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEDOWN1_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEDOWN1_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEDOWN1_BORDER;
@@ -564,6 +714,12 @@ have_event:
-          if (where == PANE) key = KEYC_MOUSEDOWN2_PANE;
-          if (where == STATUS) key = KEYC_MOUSEDOWN2_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_MOUSEDOWN2_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_MOUSEDOWN2_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_MOUSEDOWN2_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_MOUSEDOWN2_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEDOWN2_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEDOWN2_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEDOWN2_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEDOWN2_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEDOWN2_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEDOWN2_BORDER;
@@ -572,6 +728,12 @@ have_event:
-          if (where == PANE) key = KEYC_MOUSEDOWN3_PANE;
-          if (where == STATUS) key = KEYC_MOUSEDOWN3_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_MOUSEDOWN3_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_MOUSEDOWN3_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_MOUSEDOWN3_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_MOUSEDOWN3_BORDER;
+   if (where == PANE)
+    key = KEYC_MOUSEDOWN3_PANE;
+   if (where == STATUS)
+    key = KEYC_MOUSEDOWN3_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_MOUSEDOWN3_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_MOUSEDOWN3_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_MOUSEDOWN3_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_MOUSEDOWN3_BORDER;
@@ -584,6 +746,12 @@ have_event:
-          if (where == PANE) key = KEYC_SECONDCLICK1_PANE;
-          if (where == STATUS) key = KEYC_SECONDCLICK1_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_SECONDCLICK1_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_SECONDCLICK1_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_SECONDCLICK1_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_SECONDCLICK1_BORDER;
+   if (where == PANE)
+    key = KEYC_SECONDCLICK1_PANE;
+   if (where == STATUS)
+    key = KEYC_SECONDCLICK1_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_SECONDCLICK1_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_SECONDCLICK1_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_SECONDCLICK1_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_SECONDCLICK1_BORDER;
@@ -592,6 +760,12 @@ have_event:
-          if (where == PANE) key = KEYC_SECONDCLICK2_PANE;
-          if (where == STATUS) key = KEYC_SECONDCLICK2_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_SECONDCLICK2_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_SECONDCLICK2_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_SECONDCLICK2_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_SECONDCLICK2_BORDER;
+   if (where == PANE)
+    key = KEYC_SECONDCLICK2_PANE;
+   if (where == STATUS)
+    key = KEYC_SECONDCLICK2_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_SECONDCLICK2_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_SECONDCLICK2_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_SECONDCLICK2_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_SECONDCLICK2_BORDER;
@@ -600,6 +774,12 @@ have_event:
-          if (where == PANE) key = KEYC_SECONDCLICK3_PANE;
-          if (where == STATUS) key = KEYC_SECONDCLICK3_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_SECONDCLICK3_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_SECONDCLICK3_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_SECONDCLICK3_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_SECONDCLICK3_BORDER;
+   if (where == PANE)
+    key = KEYC_SECONDCLICK3_PANE;
+   if (where == STATUS)
+    key = KEYC_SECONDCLICK3_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_SECONDCLICK3_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_SECONDCLICK3_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_SECONDCLICK3_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_SECONDCLICK3_BORDER;
@@ -612,6 +792,12 @@ have_event:
-          if (where == PANE) key = KEYC_DOUBLECLICK1_PANE;
-          if (where == STATUS) key = KEYC_DOUBLECLICK1_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_DOUBLECLICK1_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_DOUBLECLICK1_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_DOUBLECLICK1_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_DOUBLECLICK1_BORDER;
+   if (where == PANE)
+    key = KEYC_DOUBLECLICK1_PANE;
+   if (where == STATUS)
+    key = KEYC_DOUBLECLICK1_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_DOUBLECLICK1_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_DOUBLECLICK1_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_DOUBLECLICK1_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_DOUBLECLICK1_BORDER;
@@ -620,6 +806,12 @@ have_event:
-          if (where == PANE) key = KEYC_DOUBLECLICK2_PANE;
-          if (where == STATUS) key = KEYC_DOUBLECLICK2_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_DOUBLECLICK2_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_DOUBLECLICK2_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_DOUBLECLICK2_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_DOUBLECLICK2_BORDER;
+   if (where == PANE)
+    key = KEYC_DOUBLECLICK2_PANE;
+   if (where == STATUS)
+    key = KEYC_DOUBLECLICK2_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_DOUBLECLICK2_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_DOUBLECLICK2_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_DOUBLECLICK2_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_DOUBLECLICK2_BORDER;
@@ -628,6 +820,12 @@ have_event:
-          if (where == PANE) key = KEYC_DOUBLECLICK3_PANE;
-          if (where == STATUS) key = KEYC_DOUBLECLICK3_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_DOUBLECLICK3_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_DOUBLECLICK3_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_DOUBLECLICK3_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_DOUBLECLICK3_BORDER;
+   if (where == PANE)
+    key = KEYC_DOUBLECLICK3_PANE;
+   if (where == STATUS)
+    key = KEYC_DOUBLECLICK3_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_DOUBLECLICK3_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_DOUBLECLICK3_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_DOUBLECLICK3_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_DOUBLECLICK3_BORDER;
@@ -640,6 +838,12 @@ have_event:
-          if (where == PANE) key = KEYC_TRIPLECLICK1_PANE;
-          if (where == STATUS) key = KEYC_TRIPLECLICK1_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_TRIPLECLICK1_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_TRIPLECLICK1_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_TRIPLECLICK1_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_TRIPLECLICK1_BORDER;
+   if (where == PANE)
+    key = KEYC_TRIPLECLICK1_PANE;
+   if (where == STATUS)
+    key = KEYC_TRIPLECLICK1_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_TRIPLECLICK1_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_TRIPLECLICK1_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_TRIPLECLICK1_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_TRIPLECLICK1_BORDER;
@@ -648,6 +852,12 @@ have_event:
-          if (where == PANE) key = KEYC_TRIPLECLICK2_PANE;
-          if (where == STATUS) key = KEYC_TRIPLECLICK2_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_TRIPLECLICK2_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_TRIPLECLICK2_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_TRIPLECLICK2_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_TRIPLECLICK2_BORDER;
+   if (where == PANE)
+    key = KEYC_TRIPLECLICK2_PANE;
+   if (where == STATUS)
+    key = KEYC_TRIPLECLICK2_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_TRIPLECLICK2_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_TRIPLECLICK2_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_TRIPLECLICK2_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_TRIPLECLICK2_BORDER;
@@ -656,6 +866,12 @@ have_event:
-          if (where == PANE) key = KEYC_TRIPLECLICK3_PANE;
-          if (where == STATUS) key = KEYC_TRIPLECLICK3_STATUS;
-          if (where == STATUS_LEFT) key = KEYC_TRIPLECLICK3_STATUS_LEFT;
-          if (where == STATUS_RIGHT) key = KEYC_TRIPLECLICK3_STATUS_RIGHT;
-          if (where == STATUS_DEFAULT) key = KEYC_TRIPLECLICK3_STATUS_DEFAULT;
-          if (where == BORDER) key = KEYC_TRIPLECLICK3_BORDER;
+   if (where == PANE)
+    key = KEYC_TRIPLECLICK3_PANE;
+   if (where == STATUS)
+    key = KEYC_TRIPLECLICK3_STATUS;
+   if (where == STATUS_LEFT)
+    key = KEYC_TRIPLECLICK3_STATUS_LEFT;
+   if (where == STATUS_RIGHT)
+    key = KEYC_TRIPLECLICK3_STATUS_RIGHT;
+   if (where == STATUS_DEFAULT)
+    key = KEYC_TRIPLECLICK3_STATUS_DEFAULT;
+   if (where == BORDER)
+    key = KEYC_TRIPLECLICK3_BORDER;
@@ -666 +882,2 @@ have_event:
-  if (key == KEYC_UNKNOWN) return (KEYC_UNKNOWN);
+ if (key == KEYC_UNKNOWN)
+  return (KEYC_UNKNOWN);
@@ -668,3 +885,6 @@ out:
-  if (b & MOUSE_MASK_META) key |= KEYC_META;
-  if (b & MOUSE_MASK_CTRL) key |= KEYC_CTRL;
-  if (b & MOUSE_MASK_SHIFT) key |= KEYC_SHIFT;
+ if (b & MOUSE_MASK_META)
+  key |= KEYC_META;
+ if (b & MOUSE_MASK_CTRL)
+  key |= KEYC_CTRL;
+ if (b & MOUSE_MASK_SHIFT)
+  key |= KEYC_SHIFT;
@@ -675 +895,3 @@ out:
-static int server_client_assume_paste(struct session *s) {
+static int
+server_client_assume_paste(struct session *s)
+{
@@ -684 +906,2 @@ static int server_client_assume_paste(struct session *s) {
-    if (s->flags & SESSION_PASTING) return (1);
+  if (s->flags & SESSION_PASTING)
+   return (1);
@@ -692 +915,3 @@ static int server_client_assume_paste(struct session *s) {
-static void server_client_update_latest(struct client *c) {
+static void
+server_client_update_latest(struct client *c)
+{
@@ -694 +919,2 @@ static void server_client_update_latest(struct client *c) {
-  if (c->session == NULL) return;
+ if (c->session == NULL)
+  return;
@@ -696 +922,2 @@ static void server_client_update_latest(struct client *c) {
-  if (w->latest == c) return;
+ if (w->latest == c)
+  return;
@@ -701,2 +928,3 @@ static void server_client_update_latest(struct client *c) {
-static enum cmd_retval server_client_key_callback(struct cmdq_item *item,
-                                                  void *data) {
+static enum cmd_retval
+server_client_key_callback(struct cmdq_item *item, void *data)
+{
@@ -717 +945,2 @@ static enum cmd_retval server_client_key_callback(struct cmdq_item *item,
-  if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS)) goto out;
+ if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS))
+  goto out;
@@ -719 +948,2 @@ static enum cmd_retval server_client_key_callback(struct cmdq_item *item,
-  if (gettimeofday(&c->activity_time, NULL) != 0) fatal("gettimeofday failed");
+ if (gettimeofday(&c->activity_time, NULL) != 0)
+  fatal("gettimeofday failed");
@@ -723 +953,2 @@ static enum cmd_retval server_client_key_callback(struct cmdq_item *item,
-    if (c->flags & CLIENT_READONLY) goto out;
+  if (c->flags & CLIENT_READONLY)
+   goto out;
@@ -725 +956,2 @@ static enum cmd_retval server_client_key_callback(struct cmdq_item *item,
-    if (key == KEYC_UNKNOWN) goto out;
+  if (key == KEYC_UNKNOWN)
+   goto out;
@@ -739,3 +971,6 @@ static enum cmd_retval server_client_key_callback(struct cmdq_item *item,
-  if (!KEYC_IS_MOUSE(key) && server_client_assume_paste(s)) goto forward_key;
-  if (server_client_is_default_key_table(c, c->keytable) && wp != NULL &&
-      (wme = TAILQ_FIRST(&wp->modes)) != NULL && wme->mode->key_table != NULL)
+ if (!KEYC_IS_MOUSE(key) && server_client_assume_paste(s))
+  goto forward_key;
+ if (server_client_is_default_key_table(c, c->keytable) &&
+     wp != NULL &&
+     (wme = TAILQ_FIRST(&wp->modes)) != NULL &&
+     wme->mode->key_table != NULL)
@@ -761 +996,2 @@ try_again:
-  if (c->flags & CLIENT_REPEAT) log_debug("currently repeating");
+ if (c->flags & CLIENT_REPEAT)
+  log_debug("currently repeating");
@@ -764,2 +1000,4 @@ try_again:
-    if ((c->flags & CLIENT_REPEAT) && (~bd->flags & KEY_BINDING_REPEAT)) {
-      log_debug("found in key table %s (not repeating)", table->name);
+  if ((c->flags & CLIENT_REPEAT) &&
+      (~bd->flags & KEY_BINDING_REPEAT)) {
+   log_debug("found in key table %s (not repeating)",
+       table->name);
@@ -800 +1038,2 @@ try_again:
-    if (c->flags & CLIENT_REPEAT) first = table;
+  if (c->flags & CLIENT_REPEAT)
+   first = table;
@@ -811,2 +1050,4 @@ forward_key:
-  if (c->flags & CLIENT_READONLY) goto out;
-  if (wp != NULL) window_pane_key(wp, c, s, wl, key, m);
+ if (c->flags & CLIENT_READONLY)
+  goto out;
+ if (wp != NULL)
+  window_pane_key(wp, c, s, wl, key, m);
@@ -814 +1055,2 @@ out:
-  if (s != NULL && key != KEYC_FOCUS_OUT) server_client_update_latest(c);
+ if (s != NULL && key != KEYC_FOCUS_OUT)
+  server_client_update_latest(c);
@@ -818 +1060,3 @@ out:
-int server_client_handle_key(struct client *c, struct key_event *event) {
+int
+server_client_handle_key(struct client *c, struct key_event *event)
+{
@@ -821 +1065,2 @@ int server_client_handle_key(struct client *c, struct key_event *event) {
-  if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS)) return (0);
+ if (s == NULL || (c->flags & CLIENT_UNATTACHEDFLAGS))
+  return (0);
@@ -835 +1080,2 @@ int server_client_handle_key(struct client *c, struct key_event *event) {
-      if (status_prompt_key(c, event->key) == 0) return (0);
+   if (status_prompt_key(c, event->key) == 0)
+    return (0);
@@ -842 +1088,3 @@ int server_client_handle_key(struct client *c, struct key_event *event) {
-void server_client_loop(void) {
+void
+server_client_loop(void)
+{
@@ -860 +1108,2 @@ void server_client_loop(void) {
-        if (focus) server_client_check_pane_focus(wp);
+    if (focus)
+     server_client_check_pane_focus(wp);
@@ -869 +1118,3 @@ void server_client_loop(void) {
-static void server_client_check_window_resize(struct window *w) {
+static void
+server_client_check_window_resize(struct window *w)
+{
@@ -871 +1122,2 @@ static void server_client_check_window_resize(struct window *w) {
-  if (~w->flags & WINDOW_RESIZE) return;
+ if (~w->flags & WINDOW_RESIZE)
+  return;
@@ -873 +1125,2 @@ static void server_client_check_window_resize(struct window *w) {
-    if (wl->session->attached != 0 && wl->session->curw == wl) break;
+  if (wl->session->attached != 0 && wl->session->curw == wl)
+   break;
@@ -875 +1128,2 @@ static void server_client_check_window_resize(struct window *w) {
-  if (wl == NULL) return;
+ if (wl == NULL)
+  return;
@@ -879,2 +1133,3 @@ static void server_client_check_window_resize(struct window *w) {
-static void server_client_resize_timer(__unused int fd, __unused short events,
-                                       void *data) {
+static void
+server_client_resize_timer(__unused int fd, __unused short events, void *data)
+{
@@ -885 +1140,3 @@ static void server_client_resize_timer(__unused int fd, __unused short events,
-static void server_client_start_resize_timer(struct window_pane *wp) {
+static void
+server_client_start_resize_timer(struct window_pane *wp)
+{
@@ -890,2 +1147,3 @@ static void server_client_start_resize_timer(struct window_pane *wp) {
-static void server_client_force_timer(__unused int fd, __unused short events,
-                                      void *data) {
+static void
+server_client_force_timer(__unused int fd, __unused short events, void *data)
+{
@@ -897 +1155,3 @@ static void server_client_force_timer(__unused int fd, __unused short events,
-static void server_client_start_force_timer(struct window_pane *wp) {
+static void
+server_client_start_force_timer(struct window_pane *wp)
+{
@@ -902 +1162,3 @@ static void server_client_start_force_timer(struct window_pane *wp) {
-static void server_client_check_pane_resize(struct window_pane *wp) {
+static void
+server_client_check_pane_resize(struct window_pane *wp)
+{
@@ -907 +1169,2 @@ static void server_client_check_pane_resize(struct window_pane *wp) {
-  if (~wp->flags & PANE_RESIZE) return;
+ if (~wp->flags & PANE_RESIZE)
+  return;
@@ -920 +1183,2 @@ static void server_client_check_pane_resize(struct window_pane *wp) {
-      log_debug("%s: resizing %%%u after forced resize", __func__, wp->id);
+   log_debug("%s: resizing %%%u after forced resize",
+       __func__, wp->id);
@@ -924 +1188,2 @@ static void server_client_check_pane_resize(struct window_pane *wp) {
-      log_debug("%s: forcing resize of %%%u", __func__, wp->id);
+   log_debug("%s: forcing resize of %%%u", __func__,
+       wp->id);
@@ -930 +1195,3 @@ static void server_client_check_pane_resize(struct window_pane *wp) {
-static void server_client_check_pane_buffer(struct window_pane *wp) {
+static void
+server_client_check_pane_buffer(struct window_pane *wp)
+{
@@ -942 +1209,2 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-    if (c->session == NULL) continue;
+  if (c->session == NULL)
+   continue;
@@ -953 +1221,2 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-    if (!flag) off = 0;
+  if (!flag)
+   off = 0;
@@ -955,3 +1224,5 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-    log_debug("%s: %s has %zu bytes used and %zu left for %%%u", __func__,
-              c->name, wpo->used - wp->base_offset, new_size, wp->id);
-    if (wpo->used < minimum) minimum = wpo->used;
+  log_debug("%s: %s has %zu bytes used and %zu left for %%%u",
+      __func__, c->name, wpo->used - wp->base_offset, new_size,
+      wp->id);
+  if (wpo->used < minimum)
+   minimum = wpo->used;
@@ -959 +1230,2 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-  if (attached_clients == 0) off = 0;
+ if (attached_clients == 0)
+  off = 0;
@@ -961,3 +1233,4 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-  if (minimum == 0) goto out;
-  log_debug("%s: %%%u has %zu minimum (of %zu) bytes used", __func__, wp->id,
-            minimum, EVBUFFER_LENGTH(evb));
+ if (minimum == 0)
+  goto out;
+ log_debug("%s: %%%u has %zu minimum (of %zu) bytes used", __func__,
+     wp->id, minimum, EVBUFFER_LENGTH(evb));
@@ -968 +1241,2 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-    if (wp->pipe_fd != -1) wp->pipe_offset.used -= wp->base_offset;
+  if (wp->pipe_fd != -1)
+   wp->pipe_offset.used -= wp->base_offset;
@@ -970 +1244,2 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-      if (c->session == NULL || (~c->flags & CLIENT_CONTROL)) continue;
+   if (c->session == NULL || (~c->flags & CLIENT_CONTROL))
+    continue;
@@ -972 +1247,2 @@ static void server_client_check_pane_buffer(struct window_pane *wp) {
-      if (wpo != NULL && !flag) wpo->used -= wp->base_offset;
+   if (wpo != NULL && !flag)
+    wpo->used -= wp->base_offset;
@@ -984 +1260,3 @@ out:
-static void server_client_check_pane_focus(struct window_pane *wp) {
+static void
+server_client_check_pane_focus(struct window_pane *wp)
+{
@@ -989,2 +1267,4 @@ static void server_client_check_pane_focus(struct window_pane *wp) {
-  if (wp->window->active != wp) goto not_focused;
-  if (wp->screen != &wp->base) goto not_focused;
+ if (wp->window->active != wp)
+  goto not_focused;
+ if (wp->screen != &wp->base)
+  goto not_focused;
@@ -992,3 +1272,6 @@ static void server_client_check_pane_focus(struct window_pane *wp) {
-    if (c->session == NULL || !(c->flags & CLIENT_FOCUSED)) continue;
-    if (c->session->attached == 0) continue;
-    if (c->session->curw->window == wp->window) goto focused;
+  if (c->session == NULL || !(c->flags & CLIENT_FOCUSED))
+   continue;
+  if (c->session->attached == 0)
+   continue;
+  if (c->session->curw->window == wp->window)
+   goto focused;
@@ -998 +1281,2 @@ not_focused:
-    if (wp->base.mode & MODE_FOCUSON) bufferevent_write(wp->event, "\033[O", 3);
+  if (wp->base.mode & MODE_FOCUSON)
+   bufferevent_write(wp->event, "\033[O", 3);
@@ -1005 +1289,2 @@ focused:
-    if (wp->base.mode & MODE_FOCUSON) bufferevent_write(wp->event, "\033[I", 3);
+  if (wp->base.mode & MODE_FOCUSON)
+   bufferevent_write(wp->event, "\033[I", 3);
@@ -1011 +1296,3 @@ focused:
-static void server_client_reset_state(struct client *c) {
+static void
+server_client_reset_state(struct client *c)
+{
@@ -1019 +1306,2 @@ static void server_client_reset_state(struct client *c) {
-  if (c->flags & (CLIENT_CONTROL | CLIENT_SUSPENDED)) return;
+ if (c->flags & (CLIENT_CONTROL|CLIENT_SUSPENDED))
+  return;
@@ -1023 +1311,2 @@ static void server_client_reset_state(struct client *c) {
-    if (c->overlay_mode != NULL) s = c->overlay_mode(c, &cx, &cy);
+  if (c->overlay_mode != NULL)
+   s = c->overlay_mode(c, &cx, &cy);
@@ -1026 +1315,2 @@ static void server_client_reset_state(struct client *c) {
-  if (s != NULL) mode = s->mode;
+ if (s != NULL)
+  mode = s->mode;
@@ -1038 +1328,2 @@ static void server_client_reset_state(struct client *c) {
-      if (status_at_line(c) == 0) cy += status_line_size(c);
+   if (status_at_line(c) == 0)
+    cy += status_line_size(c);
@@ -1040 +1331,2 @@ static void server_client_reset_state(struct client *c) {
-    if (!cursor) mode &= ~MODE_CURSOR;
+  if (!cursor)
+   mode &= ~MODE_CURSOR;
@@ -1048 +1340,2 @@ static void server_client_reset_state(struct client *c) {
-        if (loop->screen->mode & MODE_MOUSE_ALL) mode |= MODE_MOUSE_ALL;
+    if (loop->screen->mode & MODE_MOUSE_ALL)
+     mode |= MODE_MOUSE_ALL;
@@ -1051 +1344,2 @@ static void server_client_reset_state(struct client *c) {
-    if (~mode & MODE_MOUSE_ALL) mode |= MODE_MOUSE_BUTTON;
+  if (~mode & MODE_MOUSE_ALL)
+   mode |= MODE_MOUSE_BUTTON;
@@ -1060,2 +1354,3 @@ static void server_client_reset_state(struct client *c) {
-static void server_client_repeat_timer(__unused int fd, __unused short events,
-                                       void *data) {
+static void
+server_client_repeat_timer(__unused int fd, __unused short events, void *data)
+{
@@ -1069,2 +1364,3 @@ static void server_client_repeat_timer(__unused int fd, __unused short events,
-static void server_client_click_timer(__unused int fd, __unused short events,
-                                      void *data) {
+static void
+server_client_click_timer(__unused int fd, __unused short events, void *data)
+{
@@ -1078 +1374,2 @@ static void server_client_click_timer(__unused int fd, __unused short events,
-    if (!server_client_handle_key(c, event)) free(event);
+  if (!server_client_handle_key(c, event))
+   free(event);
@@ -1082 +1379,3 @@ static void server_client_click_timer(__unused int fd, __unused short events,
-static void server_client_check_exit(struct client *c) {
+static void
+server_client_check_exit(struct client *c)
+{
@@ -1087,2 +1386,4 @@ static void server_client_check_exit(struct client *c) {
-  if (c->flags & (CLIENT_DEAD | CLIENT_EXITED)) return;
-  if (~c->flags & CLIENT_EXIT) return;
+ if (c->flags & (CLIENT_DEAD|CLIENT_EXITED))
+  return;
+ if (~c->flags & CLIENT_EXIT)
+  return;
@@ -1091 +1392,2 @@ static void server_client_check_exit(struct client *c) {
-    if (!control_all_done(c)) return;
+  if (!control_all_done(c))
+   return;
@@ -1094 +1396,2 @@ static void server_client_check_exit(struct client *c) {
-    if (EVBUFFER_LENGTH(cf->buffer) != 0) return;
+  if (EVBUFFER_LENGTH(cf->buffer) != 0)
+   return;
@@ -1096 +1399,2 @@ static void server_client_check_exit(struct client *c) {
-  if (c->flags & CLIENT_ATTACHED) notify_client("client-detached", c);
+ if (c->flags & CLIENT_ATTACHED)
+  notify_client("client-detached", c);
@@ -1122,2 +1426,4 @@ static void server_client_check_exit(struct client *c) {
-static void server_client_redraw_timer(__unused int fd, __unused short events,
-                                       __unused void *data) {
+static void
+server_client_redraw_timer(__unused int fd, __unused short events,
+    __unused void *data)
+{
@@ -1126 +1432,3 @@ static void server_client_redraw_timer(__unused int fd, __unused short events,
-static void server_client_check_redraw(struct client *c) {
+static void
+server_client_check_redraw(struct client *c)
+{
@@ -1137 +1445,2 @@ static void server_client_check_redraw(struct client *c) {
-  if (c->flags & (CLIENT_CONTROL | CLIENT_SUSPENDED)) return;
+ if (c->flags & (CLIENT_CONTROL|CLIENT_SUSPENDED))
+  return;
@@ -1156 +1465,2 @@ static void server_client_check_redraw(struct client *c) {
-    if (needed) new_flags |= CLIENT_REDRAWPANES;
+  if (needed)
+   new_flags |= CLIENT_REDRAWPANES;
@@ -1169 +1479,2 @@ static void server_client_check_redraw(struct client *c) {
-          log_debug("%s: pane %%%u needs redraw", c->name, wp->id);
+     log_debug("%s: pane %%%u needs redraw",
+         c->name, wp->id);
@@ -1178 +1489,2 @@ static void server_client_check_redraw(struct client *c) {
-      if (c->redraw_panes != 0) c->flags |= CLIENT_REDRAWPANES;
+   if (c->redraw_panes != 0)
+    c->flags |= CLIENT_REDRAWPANES;
@@ -1194 +1506,2 @@ static void server_client_check_redraw(struct client *c) {
-      if (!redraw) continue;
+   if (!redraw)
+    continue;
@@ -1215 +1528,3 @@ static void server_client_check_redraw(struct client *c) {
-static void server_client_set_title(struct client *c) {
+static void
+server_client_set_title(struct client *c)
+{
@@ -1232 +1547,3 @@ static void server_client_set_title(struct client *c) {
-static void server_client_dispatch(struct imsg *imsg, void *arg) {
+static void
+server_client_dispatch(struct imsg *imsg, void *arg)
+{
@@ -1236 +1553,2 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-  if (c->flags & CLIENT_DEAD) return;
+ if (c->flags & CLIENT_DEAD)
+  return;
@@ -1260,2 +1578,4 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-      if (datalen != 0) fatalx("bad MSG_RESIZE size");
-      if (c->flags & CLIENT_CONTROL) break;
+  if (datalen != 0)
+   fatalx("bad MSG_RESIZE size");
+  if (c->flags & CLIENT_CONTROL)
+   break;
@@ -1267 +1587,2 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-      if (c->session != NULL) notify_client("client-resized", c);
+  if (c->session != NULL)
+   notify_client("client-resized", c);
@@ -1270 +1591,2 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-      if (datalen != 0) fatalx("bad MSG_EXITING size");
+  if (datalen != 0)
+   fatalx("bad MSG_EXITING size");
@@ -1277,2 +1599,4 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-      if (datalen != 0) fatalx("bad MSG_WAKEUP size");
-      if (!(c->flags & CLIENT_SUSPENDED)) break;
+  if (datalen != 0)
+   fatalx("bad MSG_WAKEUP size");
+  if (!(c->flags & CLIENT_SUSPENDED))
+   break;
@@ -1288 +1612,2 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-      if (s != NULL) session_update_activity(s, &c->activity_time);
+  if (s != NULL)
+   session_update_activity(s, &c->activity_time);
@@ -1291 +1616,2 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-      if (datalen != 0) fatalx("bad MSG_SHELL size");
+  if (datalen != 0)
+   fatalx("bad MSG_SHELL size");
@@ -1305,2 +1631,3 @@ static void server_client_dispatch(struct imsg *imsg, void *arg) {
-static enum cmd_retval server_client_command_done(struct cmdq_item *item,
-                                                  __unused void *data) {
+static enum cmd_retval
+server_client_command_done(struct cmdq_item *item, __unused void *data)
+{
@@ -1314,2 +1641,3 @@ static enum cmd_retval server_client_command_done(struct cmdq_item *item,
-static void server_client_dispatch_command(struct client *c,
-                                           struct imsg *imsg) {
+static void
+server_client_dispatch_command(struct client *c, struct imsg *imsg)
+{
@@ -1322 +1650,2 @@ static void server_client_dispatch_command(struct client *c,
-  if (c->flags & CLIENT_EXIT) return;
+ if (c->flags & CLIENT_EXIT)
+  return;
@@ -1328 +1657,2 @@ static void server_client_dispatch_command(struct client *c,
-  if (len > 0 && buf[len - 1] != '\0') fatalx("bad MSG_COMMAND string");
+ if (len > 0 && buf[len - 1] != '\0')
+  fatalx("bad MSG_COMMAND string");
@@ -1361,2 +1691,3 @@ error:
-static void server_client_dispatch_identify(struct client *c,
-                                            struct imsg *imsg) {
+static void
+server_client_dispatch_identify(struct client *c, struct imsg *imsg)
+{
@@ -1368 +1699,2 @@ static void server_client_dispatch_identify(struct client *c,
-  if (c->flags & CLIENT_IDENTIFIED) fatalx("out-of-order identify message");
+ if (c->flags & CLIENT_IDENTIFIED)
+  fatalx("out-of-order identify message");
@@ -1373 +1705,2 @@ static void server_client_dispatch_identify(struct client *c,
-      if (datalen != sizeof feat) fatalx("bad MSG_IDENTIFY_FEATURES size");
+  if (datalen != sizeof feat)
+   fatalx("bad MSG_IDENTIFY_FEATURES size");
@@ -1376 +1709,2 @@ static void server_client_dispatch_identify(struct client *c,
-      log_debug("client %p IDENTIFY_FEATURES %s", c, tty_get_features(feat));
+  log_debug("client %p IDENTIFY_FEATURES %s", c,
+      tty_get_features(feat));
@@ -1379 +1713,2 @@ static void server_client_dispatch_identify(struct client *c,
-      if (datalen != sizeof flags) fatalx("bad MSG_IDENTIFY_FLAGS size");
+  if (datalen != sizeof flags)
+   fatalx("bad MSG_IDENTIFY_FLAGS size");
@@ -1419 +1754,2 @@ static void server_client_dispatch_identify(struct client *c,
-      if (datalen != 0) fatalx("bad MSG_IDENTIFY_STDIN size");
+  if (datalen != 0)
+   fatalx("bad MSG_IDENTIFY_STDIN size");
@@ -1424 +1760,2 @@ static void server_client_dispatch_identify(struct client *c,
-      if (datalen != 0) fatalx("bad MSG_IDENTIFY_STDOUT size");
+  if (datalen != 0)
+   fatalx("bad MSG_IDENTIFY_STDOUT size");
@@ -1431 +1768,2 @@ static void server_client_dispatch_identify(struct client *c,
-      if (strchr(data, '=') != NULL) environ_put(c->environ, data, 0);
+  if (strchr(data, '=') != NULL)
+   environ_put(c->environ, data, 0);
@@ -1435 +1773,2 @@ static void server_client_dispatch_identify(struct client *c,
-      if (datalen != sizeof c->pid) fatalx("bad MSG_IDENTIFY_CLIENTPID size");
+  if (datalen != sizeof c->pid)
+   fatalx("bad MSG_IDENTIFY_CLIENTPID size");
@@ -1442 +1781,2 @@ static void server_client_dispatch_identify(struct client *c,
-  if (imsg->hdr.type != MSG_IDENTIFY_DONE) return;
+ if (imsg->hdr.type != MSG_IDENTIFY_DONE)
+  return;
@@ -1466 +1806,3 @@ static void server_client_dispatch_identify(struct client *c,
-  if ((~c->flags & CLIENT_EXIT) && !cfg_finished && c == TAILQ_FIRST(&clients))
+ if ((~c->flags & CLIENT_EXIT) &&
+      !cfg_finished &&
+      c == TAILQ_FIRST(&clients))
@@ -1469 +1811,3 @@ static void server_client_dispatch_identify(struct client *c,
-static void server_client_dispatch_shell(struct client *c) {
+static void
+server_client_dispatch_shell(struct client *c)
+{
@@ -1472 +1816,2 @@ static void server_client_dispatch_shell(struct client *c) {
-  if (!checkshell(shell)) shell = _PATH_BSHELL;
+ if (!checkshell(shell))
+  shell = _PATH_BSHELL;
@@ -1476,2 +1821,3 @@ static void server_client_dispatch_shell(struct client *c) {
-static void server_client_dispatch_write_ready(struct client *c,
-                                               struct imsg *imsg) {
+static void
+server_client_dispatch_write_ready(struct client *c, struct imsg *imsg)
+{
@@ -1481 +1827,2 @@ static void server_client_dispatch_write_ready(struct client *c,
-  if (msglen != sizeof *msg) fatalx("bad MSG_WRITE_READY size");
+ if (msglen != sizeof *msg)
+  fatalx("bad MSG_WRITE_READY size");
@@ -1483 +1830,2 @@ static void server_client_dispatch_write_ready(struct client *c,
-  if ((cf = RB_FIND(client_files, &c->files, &find)) == NULL) return;
+ if ((cf = RB_FIND(client_files, &c->files, &find)) == NULL)
+  return;
@@ -1490,2 +1838,3 @@ static void server_client_dispatch_write_ready(struct client *c,
-static void server_client_dispatch_read_data(struct client *c,
-                                             struct imsg *imsg) {
+static void
+server_client_dispatch_read_data(struct client *c, struct imsg *imsg)
+{
@@ -1497 +1846,2 @@ static void server_client_dispatch_read_data(struct client *c,
-  if (msglen < sizeof *msg) fatalx("bad MSG_READ_DATA size");
+ if (msglen < sizeof *msg)
+  fatalx("bad MSG_READ_DATA size");
@@ -1499 +1849,2 @@ static void server_client_dispatch_read_data(struct client *c,
-  if ((cf = RB_FIND(client_files, &c->files, &find)) == NULL) return;
+ if ((cf = RB_FIND(client_files, &c->files, &find)) == NULL)
+  return;
@@ -1509,2 +1860,3 @@ static void server_client_dispatch_read_data(struct client *c,
-static void server_client_dispatch_read_done(struct client *c,
-                                             struct imsg *imsg) {
+static void
+server_client_dispatch_read_done(struct client *c, struct imsg *imsg)
+{
@@ -1514 +1866,2 @@ static void server_client_dispatch_read_done(struct client *c,
-  if (msglen != sizeof *msg) fatalx("bad MSG_READ_DONE size");
+ if (msglen != sizeof *msg)
+  fatalx("bad MSG_READ_DONE size");
@@ -1516 +1869,2 @@ static void server_client_dispatch_read_done(struct client *c,
-  if ((cf = RB_FIND(client_files, &c->files, &find)) == NULL) return;
+ if ((cf = RB_FIND(client_files, &c->files, &find)) == NULL)
+  return;
@@ -1521 +1875,3 @@ static void server_client_dispatch_read_done(struct client *c,
-const char *server_client_get_cwd(struct client *c, struct session *s) {
+const char *
+server_client_get_cwd(struct client *c, struct session *s)
+{
@@ -1523,5 +1879,10 @@ const char *server_client_get_cwd(struct client *c, struct session *s) {
-  if (!cfg_finished && cfg_client != NULL) return (cfg_client->cwd);
-  if (c != NULL && c->session == NULL && c->cwd != NULL) return (c->cwd);
-  if (s != NULL && s->cwd != NULL) return (s->cwd);
-  if (c != NULL && (s = c->session) != NULL && s->cwd != NULL) return (s->cwd);
-  if ((home = find_home()) != NULL) return (home);
+ if (!cfg_finished && cfg_client != NULL)
+  return (cfg_client->cwd);
+ if (c != NULL && c->session == NULL && c->cwd != NULL)
+  return (c->cwd);
+ if (s != NULL && s->cwd != NULL)
+  return (s->cwd);
+ if (c != NULL && (s = c->session) != NULL && s->cwd != NULL)
+  return (s->cwd);
+ if ((home = find_home()) != NULL)
+  return (home);
@@ -1530,2 +1891,3 @@ const char *server_client_get_cwd(struct client *c, struct session *s) {
-static uint64_t server_client_control_flags(struct client *c,
-                                            const char *next) {
+static uint64_t
+server_client_control_flags(struct client *c, const char *next)
+{
@@ -1540,2 +1902,4 @@ static uint64_t server_client_control_flags(struct client *c,
-  if (strcmp(next, "no-output") == 0) return (CLIENT_CONTROL_NOOUTPUT);
-  if (strcmp(next, "wait-exit") == 0) return (CLIENT_CONTROL_WAITEXIT);
+ if (strcmp(next, "no-output") == 0)
+  return (CLIENT_CONTROL_NOOUTPUT);
+ if (strcmp(next, "wait-exit") == 0)
+  return (CLIENT_CONTROL_WAITEXIT);
@@ -1544 +1908,3 @@ static uint64_t server_client_control_flags(struct client *c,
-void server_client_set_flags(struct client *c, const char *flags) {
+void
+server_client_set_flags(struct client *c, const char *flags)
+{
@@ -1551 +1917,2 @@ void server_client_set_flags(struct client *c, const char *flags) {
-    if (not ) next++;
+  if (not)
+   next++;
@@ -1562 +1929,2 @@ void server_client_set_flags(struct client *c, const char *flags) {
-    if (flag == 0) continue;
+  if (flag == 0)
+   continue;
@@ -1568 +1936,2 @@ void server_client_set_flags(struct client *c, const char *flags) {
-    if (flag == CLIENT_CONTROL_NOOUTPUT) control_reset_offsets(c);
+  if (flag == CLIENT_CONTROL_NOOUTPUT)
+   control_reset_offsets(c);
@@ -1573 +1942,3 @@ void server_client_set_flags(struct client *c, const char *flags) {
-const char *server_client_get_flags(struct client *c) {
+const char *
+server_client_get_flags(struct client *c)
+{
@@ -1577,5 +1948,10 @@ const char *server_client_get_flags(struct client *c) {
-  if (c->flags & CLIENT_ATTACHED) strlcat(s, "attached,", sizeof s);
-  if (c->flags & CLIENT_CONTROL) strlcat(s, "control-mode,", sizeof s);
-  if (c->flags & CLIENT_IGNORESIZE) strlcat(s, "ignore-size,", sizeof s);
-  if (c->flags & CLIENT_CONTROL_NOOUTPUT) strlcat(s, "no-output,", sizeof s);
-  if (c->flags & CLIENT_CONTROL_WAITEXIT) strlcat(s, "wait-exit,", sizeof s);
+ if (c->flags & CLIENT_ATTACHED)
+  strlcat(s, "attached,", sizeof s);
+ if (c->flags & CLIENT_CONTROL)
+  strlcat(s, "control-mode,", sizeof s);
+ if (c->flags & CLIENT_IGNORESIZE)
+  strlcat(s, "ignore-size,", sizeof s);
+ if (c->flags & CLIENT_CONTROL_NOOUTPUT)
+  strlcat(s, "no-output,", sizeof s);
+ if (c->flags & CLIENT_CONTROL_WAITEXIT)
+  strlcat(s, "wait-exit,", sizeof s);
@@ -1583 +1959,2 @@ const char *server_client_get_flags(struct client *c) {
-    xsnprintf(tmp, sizeof tmp, "pause-after=%u,", c->pause_age / 1000);
+  xsnprintf(tmp, sizeof tmp, "pause-after=%u,",
+      c->pause_age / 1000);
@@ -1586,5 +1963,10 @@ const char *server_client_get_flags(struct client *c) {
-  if (c->flags & CLIENT_READONLY) strlcat(s, "read-only,", sizeof s);
-  if (c->flags & CLIENT_ACTIVEPANE) strlcat(s, "active-pane,", sizeof s);
-  if (c->flags & CLIENT_SUSPENDED) strlcat(s, "suspended,", sizeof s);
-  if (c->flags & CLIENT_UTF8) strlcat(s, "UTF-8,", sizeof s);
-  if (*s != '\0') s[strlen(s) - 1] = '\0';
+ if (c->flags & CLIENT_READONLY)
+  strlcat(s, "read-only,", sizeof s);
+ if (c->flags & CLIENT_ACTIVEPANE)
+  strlcat(s, "active-pane,", sizeof s);
+ if (c->flags & CLIENT_SUSPENDED)
+  strlcat(s, "suspended,", sizeof s);
+ if (c->flags & CLIENT_UTF8)
+  strlcat(s, "UTF-8,", sizeof s);
+ if (*s != '\0')
+  s[strlen(s) - 1] = '\0';
@@ -1593,2 +1975,3 @@ const char *server_client_get_flags(struct client *c) {
-static struct client_window *server_client_get_client_window(struct client *c,
-                                                             u_int id) {
+static struct client_window *
+server_client_get_client_window(struct client *c, u_int id)
+{
@@ -1598 +1981,3 @@ static struct client_window *server_client_get_client_window(struct client *c,
-struct window_pane *server_client_get_pane(struct client *c) {
+struct window_pane *
+server_client_get_pane(struct client *c)
+{
@@ -1601,2 +1986,4 @@ struct window_pane *server_client_get_pane(struct client *c) {
-  if (s == NULL) return (NULL);
-  if (~c->flags & CLIENT_ACTIVEPANE) return (s->curw->window->active);
+ if (s == NULL)
+  return (NULL);
+ if (~c->flags & CLIENT_ACTIVEPANE)
+  return (s->curw->window->active);
@@ -1604 +1991,2 @@ struct window_pane *server_client_get_pane(struct client *c) {
-  if (cw == NULL) return (s->curw->window->active);
+ if (cw == NULL)
+  return (s->curw->window->active);
@@ -1607 +1995,3 @@ struct window_pane *server_client_get_pane(struct client *c) {
-void server_client_set_pane(struct client *c, struct window_pane *wp) {
+void
+server_client_set_pane(struct client *c, struct window_pane *wp)
+{
@@ -1610 +2000,2 @@ void server_client_set_pane(struct client *c, struct window_pane *wp) {
-  if (s == NULL) return;
+ if (s == NULL)
+  return;
@@ -1620 +2011,3 @@ void server_client_set_pane(struct client *c, struct window_pane *wp) {
-void server_client_remove_pane(struct window_pane *wp) {
+void
+server_client_remove_pane(struct window_pane *wp)
+{
