--- a/./output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_conflict.c
+++ b/./output/tmux/20e89605-b33a3022-31ed29e5/server-client.no_comments_merged.c
@@ -1790 +1789,0 @@ server_client_dispatch_identify(struct client *c, struct imsg *imsg)
-<<<<<<< HEAD
@@ -1795,5 +1793,0 @@ server_client_dispatch_identify(struct client *c, struct imsg *imsg)
-||||||| 31ed29e5
-  if (c->flags & CLIENT_CONTROL)
-=======
- if (c->flags & CLIENT_CONTROL)
->>>>>>> b33a3022
