--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/42b926a7-fec8c1ff-abdc490f/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/42b926a7-fec8c1ff-abdc490f/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-std::string Version() < < < < < < < HEAD { return "1.35.0-dev"; }
-|||||||
-{ return "1.34.0-dev"; }
-=======
-{ return "1.34.0"; }
->>>>>>> fec8c1ff57fd60305799c9811b5f166d40b4a195
+std::string Version() { return "1.35.0-dev"; }
