diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/42b926a7-fec8c1ff-abdc490f/src@php@ext@grpc@version.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/42b926a7-fec8c1ff-abdc490f/src@php@ext@grpc@version.no_comments_merged.h
index ebbc6f0..b5b8379 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/42b926a7-fec8c1ff-abdc490f/src@php@ext@grpc@version.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/42b926a7-fec8c1ff-abdc490f/src@php@ext@grpc@version.no_comments_merged.h
@@ -4,2 +3,0 @@
-#define PHP_GRPC_VERSION "1.34.0dev"
-#define PHP_GRPC_VERSION "1.34.0"
