diff --git a/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_mergegen.h b/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_merged.h
index 61c9795..59166be 100644
--- a/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_mergegen.h
+++ b/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_edit.no_comments_merged.h
@@ -126,0 +127,8 @@ class VersionEdit {
+  std::vector<std::pair<int, FileMetaData>> new_files_;
+  uint32_t column_family_;
+  bool is_column_family_drop_;
+  bool is_column_family_add_;
+  std::string column_family_name_;
+  enum {
+    kManifestVersion = 1
+  };
