diff --git a/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_set.no_comments_mergegen.cc b/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_set.no_comments_merged.cc
index 51f856b..ddeb080 100644
--- a/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_set.no_comments_mergegen.cc
+++ b/./output/rocksdb/bcea9c12-63cef900-5601bc46/db@version_set.no_comments_merged.cc
@@ -1210 +1210,7 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
-pending_manifest_file_number_ = manifest_file_number_);
+  } else {
+    pending_manifest_file_number_ = manifest_file_number_;
+  }
+  if (new_descriptor_log) {
+    if (column_family_set_->GetMaxColumnFamily() > 0) {
+      edit->SetMaxColumnFamily(column_family_set_->GetMaxColumnFamily());
+    }
@@ -1296,0 +1303,8 @@ if (edit->is_column_family_add_) {
+      assert(batch_edits.size() == 1);
+      assert(options != nullptr);
+      CreateColumnFamily(*options, edit);
+    } else if (edit->is_column_family_drop_) {
+      assert(batch_edits.size() == 1);
+      column_family_data->SetDropped();
+      if (column_family_data->Unref()) {
+        delete column_family_data;
@@ -1298 +1312,17 @@ if (edit->is_column_family_add_) {
-return m; } manifest_file_size_ = new_manifest_file_size; prev_log_number_ = edit->prev_log_number_;
+    } else {
+      uint64_t max_log_number_in_batch = 0;
+      for (auto& e : batch_edits) {
+        if (e->has_log_number_) {
+          max_log_number_in_batch =
+              std::max(max_log_number_in_batch, e->log_number_);
+        }
+      }
+      if (max_log_number_in_batch != 0) {
+        assert(column_family_data->GetLogNumber() < max_log_number_in_batch);
+        column_family_data->SetLogNumber(max_log_number_in_batch);
+      }
+      AppendVersion(column_family_data, v);
+    }
+    manifest_file_number_ = pending_manifest_file_number_;
+    manifest_file_size_ = new_manifest_file_size;
+    prev_log_number_ = edit->prev_log_number_;
@@ -1430 +1460,5 @@ Status VersionSet::Recover(
-if (edit.is_column_family_add_) { if (cf_in_builders || cf_in_not_found) { s = Status::Corruption( "Manifest adding the same column family twice"); break; } auto cf_options = cf_name_to_options.find(edit.column_family_name_); if (cf_options == cf_name_to_options.end()) { column_families_not_found.insert(edit.column_family_); } else { cfd = CreateColumnFamily(cf_options->second, &edit); builders.insert({edit.column_family_, new Builder(cfd)}); } } else if (edit.is_column_family_drop_) { if (cf_in_builders) { auto builder = builders.find(edit.column_family_); assert(builder != builders.end()); delete builder->second; builders.erase(builder); cfd = column_family_set_->GetColumnFamily(edit.column_family_); if (cfd->Unref()) { delete cfd; cfd = nullptr; } else {
+      if (edit.is_column_family_add_) {
+        if (cf_in_builders || cf_in_not_found) {
+          s = Status::Corruption(
+              "Manifest adding the same column family twice");
+          break;
@@ -1431,0 +1466,63 @@ if (edit.is_column_family_add_) { if (cf_in_builders || cf_in_not_found) { s = S
+        auto cf_options = cf_name_to_options.find(edit.column_family_name_);
+        if (cf_options == cf_name_to_options.end()) {
+          column_families_not_found.insert(edit.column_family_);
+        } else {
+          cfd = CreateColumnFamily(cf_options->second, &edit);
+          builders.insert({edit.column_family_, new Builder(cfd)});
+        }
+      } else if (edit.is_column_family_drop_) {
+        if (cf_in_builders) {
+          auto builder = builders.find(edit.column_family_);
+          assert(builder != builders.end());
+          delete builder->second;
+          builders.erase(builder);
+          cfd = column_family_set_->GetColumnFamily(edit.column_family_);
+          if (cfd->Unref()) {
+            delete cfd;
+            cfd = nullptr;
+          } else {
+            assert(false);
+          }
+        } else if (cf_in_not_found) {
+          column_families_not_found.erase(edit.column_family_);
+        } else {
+          s = Status::Corruption(
+              "Manifest - dropping non-existing column family");
+          break;
+        }
+      } else if (!cf_in_not_found) {
+        if (!cf_in_builders) {
+          s = Status::Corruption(
+              "Manifest record referencing unknown column family");
+          break;
+        }
+        cfd = column_family_set_->GetColumnFamily(edit.column_family_);
+        assert(cfd != nullptr);
+        if (edit.max_level_ >= cfd->current()->NumberLevels()) {
+          s = Status::InvalidArgument(
+              "db has more levels than options.num_levels");
+          break;
+        }
+        auto builder = builders.find(edit.column_family_);
+        assert(builder != builders.end());
+        builder->second->Apply(&edit);
+      }
+      if (cfd != nullptr) {
+        if (edit.has_log_number_) {
+          if (cfd->GetLogNumber() > edit.log_number_) {
+            log_number_decrease = true;
+          } else {
+            cfd->SetLogNumber(edit.log_number_);
+            have_log_number = true;
+          }
+        }
+        if (edit.has_comparator_ &&
+            edit.comparator_ != cfd->user_comparator()->Name()) {
+          s = Status::InvalidArgument(
+              cfd->user_comparator()->Name(),
+              "does not match existing comparator " + edit.comparator_);
+          break;
+        }
+      }
+      if (edit.has_version_number_) {
+        have_version_number = true;
@@ -1451 +1548,3 @@ if (edit.is_column_family_add_) { if (cf_in_builders || cf_in_not_found) { s = S
-        s = Status::Corruption("log number decreases");
+        s = Status::Corruption(
+            "MANIFEST corruption - Log numbers in records NOT "
+            "monotonically increasing");
@@ -1453,2 +1552,3 @@ if (edit.is_column_family_add_) { if (cf_in_builders || cf_in_not_found) { s = S
-        Log(options_->info_log, "decreasing of log_number is detected "
-            "in MANIFEST\n");
+        Log(options_->info_log,
+            "MANIFEST corruption detected, but ignored - Log numbers in "
+            "records NOT monotonically increasing");
@@ -1799,0 +1900 @@ Status VersionSet::WriteSnapshot(log::Writer* log) {
+  bool first_record = false;
@@ -1802,0 +1904,4 @@ Status VersionSet::WriteSnapshot(log::Writer* log) {
+      if (first_record) {
+        edit.SetVersionNumber();
+        first_record = false;
+      }
