--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a99e1cd6-8bf4487a-7c46c7a7/libs@ui@GraphicBufferMapper.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a99e1cd6-8bf4487a-7c46c7a7/libs@ui@GraphicBufferMapper.no_comments_merged.cpp
@@ -20,0 +21 @@ using LockResult = GraphicBufferMapper::LockResult;
+ANDROID_SINGLETON_STATIC_INSTANCE( GraphicBufferMapper )
@@ -154,13 +156,0 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
-<<<<<<< HEAD
-    ATRACE_CALL();
-    const uint64_t usage = static_cast<uint64_t>(ANDROID_NATIVE_UNSIGNED_CAST(
-            android_convertGralloc1To0Usage(producerUsage, consumerUsage)));
-    return mMapper->lock(handle, usage, bounds, fenceFd, vaddr, outBytesPerPixel,
-                         outBytesPerStride);
-|||||||
-    ATRACE_CALL();
-    const uint64_t usage =
-            static_cast<uint64_t>(android_convertGralloc1To0Usage(producerUsage, consumerUsage));
-    return mMapper->lock(handle, usage, bounds, fenceFd, vaddr, outBytesPerPixel,
-                         outBytesPerStride);
-=======
@@ -169 +158,0 @@ status_t GraphicBufferMapper::lockAsync(buffer_handle_t handle, uint64_t produce
->>>>>>> 8bf4487ae52ce8d1797ea58dd0b1109255518ceb
@@ -306 +295 @@ status_t GraphicBufferMapper::setSmpte2094_10(buffer_handle_t bufferHandle,
-}
+};
