diff --git a/./output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_mergegen.cc b/./output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_merged.cc
index 5bc4d93..e8b5ef8 100644
--- a/./output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_mergegen.cc
+++ b/./output/grpc/998bde89-c6947b74-25518e5b/src@core@ext@xds@xds_api.no_comments_merged.cc
@@ -1097,0 +1098,4 @@ grpc_error* RdsResponseParse(
+    XdsApi::RdsUpdate& rds_update =
+        (*rds_update_map)[std::move(route_config_name)];
+    grpc_error* error =
+        RouteConfigParse(client, tracer, symtab, route_config, &rds_update);
@@ -1388 +1392,3 @@ XdsApi::AdsParseResult XdsApi::ParseAdsResponse(
-result.lds_update_map, arena.ptr());
+        LdsResponseParse(client_, tracer_, symtab_.ptr(), response,
+                         expected_listener_names, &result.lds_update_map,
+                         arena.ptr());
@@ -1390 +1396,4 @@ result.lds_update_map, arena.ptr());
-} else if (IsCds(result.type_url)) {
+    result.parse_error =
+        RdsResponseParse(client_, tracer_, symtab_.ptr(), response,
+                         expected_route_configuration_names,
+                         &result.rds_update_map, arena.ptr());
