--- a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@row_operations@row_aggregate.no_comments_mergegen.cpp
+++ b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@common@row_operations@row_aggregate.no_comments_merged.cpp
@@ -24 +24 @@ void RowOperations::InitializeStates(TupleDataLayout &layout, Vector &addresses,
-void RowOperations::DestroyStates(TupleDataLayout &layout, Vector &addresses, idx_t count) {
+void RowOperations::DestroyStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, idx_t count) {
@@ -53 +53,2 @@ void RowOperations::UpdateFilteredStates(RowOperationsState &state, AggregateFil
-void RowOperations::CombineStates(TupleDataLayout &layout, Vector &sources, Vector &targets, idx_t count) {
+void RowOperations::CombineStates(RowOperationsState &state, TupleDataLayout &layout, Vector &sources, Vector &targets,
+                                  idx_t count) {
@@ -67 +68,2 @@ void RowOperations::CombineStates(TupleDataLayout &layout, Vector &sources, Vect
-void RowOperations::FinalizeStates(TupleDataLayout &layout, Vector &addresses, DataChunk &result, idx_t aggr_idx) {
+void RowOperations::FinalizeStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses,
+                                   DataChunk &result, idx_t aggr_idx) {
