diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_mergebot.hpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_merged.hpp
index 248546d..1b28665 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_merged.hpp
@@ -22 +22,2 @@ struct RowOperations {
- static void DestroyStates(RowOperationsState &state, RowLayout &layout, Vector &addresses, idx_t count);
+ static void InitializeStates(TupleDataLayout &layout, Vector &addresses, const SelectionVector &sel, idx_t count);
+ static void DestroyStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, idx_t count);
@@ -27,6 +28 @@ struct RowOperations {
-<<<<<<< HEAD
- static void CombineStates(TupleDataLayout &layout, Vector &sources, Vector &targets, idx_t count);
-|||||||
- static void CombineStates(RowLayout &layout, Vector &sources, Vector &targets, idx_t count);
-=======
- static void CombineStates(RowOperationsState &state, RowLayout &layout, Vector &sources, Vector &targets,
+ static void CombineStates(RowOperationsState &state, TupleDataLayout &layout, Vector &sources, Vector &targets,
@@ -34,7 +30 @@ struct RowOperations {
->>>>>>> 30d444e45d4f6ad691baba9a173042892ed4547a
-<<<<<<< HEAD
- static void FinalizeStates(TupleDataLayout &layout, Vector &addresses, DataChunk &result, idx_t aggr_idx);
-|||||||
- static void FinalizeStates(RowLayout &layout, Vector &addresses, DataChunk &result, idx_t aggr_idx);
-=======
- static void FinalizeStates(RowOperationsState &state, RowLayout &layout, Vector &addresses, DataChunk &result,
+ static void FinalizeStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, DataChunk &result,
@@ -42 +31,0 @@ struct RowOperations {
->>>>>>> 30d444e45d4f6ad691baba9a173042892ed4547a
@@ -76,2 +64,0 @@ struct RowOperations {
- static void InitializeStates(TupleDataLayout &layout, Vector &addresses, const SelectionVector &sel, idx_t count);
- static void DestroyStates(TupleDataLayout &layout, Vector &addresses, idx_t count);
