--- a/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_mergegen.hpp
+++ b/./output/duckdb/0b56ce79-30d444e4-5c0a1ed7/src@include@duckdb@common@row_operations@row_operations.no_comments_merged.hpp
@@ -22,0 +23 @@ struct RowOperations {
+ static void DestroyStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, idx_t count);
@@ -26,0 +28,4 @@ struct RowOperations {
+ static void CombineStates(RowOperationsState &state, TupleDataLayout &layout, Vector &sources, Vector &targets,
+                           idx_t count);
+ static void FinalizeStates(RowOperationsState &state, TupleDataLayout &layout, Vector &addresses, DataChunk &result,
+                            idx_t aggr_idx);
