diff --git a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@runtime.no_comments_mergegen.cc b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@runtime.no_comments_merged.cc
index ccc6679..f4cc81c 100644
--- a/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@runtime.no_comments_mergegen.cc
+++ b/./output/art/183ef2e9-9f9413b0-4e97dd87/runtime@runtime.no_comments_merged.cc
@@ -2587 +2587,2 @@ bool Runtime::IsSystemServerProfiled() const {
-bool Runtime::GetOatFilesExecutable() const { return !IsAotCompiler() && !IsSystemServerProfiled();
+bool Runtime::GetOatFilesExecutable() const {
+  return !IsAotCompiler() && !IsSystemServerProfiled();
