--- a/home/whalien/codebase/python/mergebot-eva/output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_merged.c
@@ -41,6 +40,0 @@ static int cliConnect(int force) {
-<<<<<<< HEAD
-  if (fd == ANET_ERR) {
-    if (config.hostsocket == NULL) {
-|||||||
-  if (fd == ANET_ERR) {
-=======
@@ -49 +43 @@ static int cliConnect(int force) {
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
+        if (config.hostsocket == NULL) {
@@ -53,5 +46,0 @@ static int cliConnect(int force) {
-      if (fd == ANET_ERR) {
-        fprintf(stderr, "Could not connect to Redis at %s: %s",
-                config.hostsocket, err);
-        return -1;
-      }
@@ -317,4 +299,3 @@ static int parseOptions(int argc, char **argv) {
-              "Starting interactive mode using -i is deprecated. Interactive "
-              "mode is started\n"
-              "by default when redis-cli is executed without a command to "
-              "execute.\n");
+"Starting interactive mode using -i is deprecated. Interactive mode is started\n"
+"by default when redis-cli is executed without a command to execute.\n"
+            );
@@ -352,31 +332,2 @@ static void usage() {
-<<<<<<< HEAD
-  fprintf(
-      stderr,
-      "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a "
-      "authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-s "
-          "/path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 "
-          "arg2 ... arg(N-1)\n");
-  fprintf(stderr,
-          "\nIf a pipe from standard input is detected this data is used as "
-          "last argument.\n\n");
-  fprintf(stderr, "example: cat /etc/passwd | redis-cli set my_passwd\n");
-|||||||
-  fprintf(stderr,
-          "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r "
-          "repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-a authpw] "
-          "[-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
-  fprintf(stderr,
-          "\nIf a pipe from standard input is detected this data is used as "
-          "last argument.\n\n");
-  fprintf(stderr, "example: cat /etc/passwd | redis-cli set my_passwd\n");
-=======
-  fprintf(stderr,
-          "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r "
-          "repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... "
-          "arg(N-1)\n\n");
+    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
+    fprintf(stderr, "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... arg(N-1)\n\n");
@@ -384 +334,0 @@ static void usage() {
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
@@ -387,3 +337 @@ static void usage() {
-  fprintf(stderr,
-          "\nRun in interactive mode: redis-cli -i or just don't pass any "
-          "command\n");
+    fprintf(stderr, "\nRun in interactive mode: redis-cli -i or just don't pass any command\n");
