--- a/./output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_conflict.c
+++ b/./output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_merged.c
@@ -41,2 +41,2 @@ static int cliConnect(int force) {
-<<<<<<< HEAD
-    if (fd == ANET_ERR) {
+    if (fd == ANET_ERR || force) {
+        if (force) close(fd);
@@ -47,4 +46,0 @@ static int cliConnect(int force) {
-            if (fd == ANET_ERR) {
-                fprintf(stderr, "Could not connect to Redis at %s: %s", config.hostsocket, err);
-                return -1;
-            }
@@ -52,8 +47,0 @@ static int cliConnect(int force) {
-||||||| 0c7a9dec6
-    if (fd == ANET_ERR) {
-        fd = anetTcpConnect(err,config.hostip,config.hostport);
-=======
-    if (fd == ANET_ERR || force) {
-        if (force) close(fd);
-        fd = anetTcpConnect(err,config.hostip,config.hostport);
->>>>>>> b4f2e412
@@ -344 +331,0 @@ static void usage() {
-<<<<<<< HEAD
@@ -346,10 +332,0 @@ static void usage() {
-    fprintf(stderr, "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-s /path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
-    fprintf(stderr, "\nIf a pipe from standard input is detected this data is used as last argument.\n\n");
-    fprintf(stderr, "example: cat /etc/passwd | redis-cli set my_passwd\n");
-||||||| 0c7a9dec6
-    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-    fprintf(stderr, "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
-    fprintf(stderr, "\nIf a pipe from standard input is detected this data is used as last argument.\n\n");
-    fprintf(stderr, "example: cat /etc/passwd | redis-cli set my_passwd\n");
-=======
-    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
@@ -358 +334,0 @@ static void usage() {
->>>>>>> b4f2e412
