diff --git a/./output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_mergegen.cc b/./output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_merged.cc
index 03666c7..7d7db8e 100644
--- a/./output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_mergegen.cc
+++ b/./output/grpc/20c350fd-ee5b762f-63369300/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3 +3 @@ namespace grpc {
-namespace grpc {std::string Version() { return "1.34.2"; }
+std::string Version() { return "1.34.0-dev"; }
