diff --git a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_conflict.hpp b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_merged.hpp
index 7fcd96b..499f9f3 100644
--- a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_conflict.hpp
+++ b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_merged.hpp
@@ -5,29 +4,0 @@ enum class CSVState : uint8_t {
-<<<<<<< HEAD
- STANDARD = 0,
- DELIMITER = 1,
- RECORD_SEPARATOR = 2,
- CARRIAGE_RETURN = 3,
- QUOTED = 4,
- UNQUOTED = 5,
- ESCAPE = 6,
- INVALID = 7,
- NOT_SET = 8,
- QUOTED_NEW_LINE = 9,
- EMPTY_SPACE = 10,
- COMMENT = 11,
- UNQUOTED_ESCAPE = 12,
- ESCAPED_RETURN = 13
-||||||| d4c7e729ac
- STANDARD = 0,
- DELIMITER = 1,
- RECORD_SEPARATOR = 2,
- CARRIAGE_RETURN = 3,
- QUOTED = 4,
- UNQUOTED = 5,
- ESCAPE = 6,
- INVALID = 7,
- NOT_SET = 8,
- QUOTED_NEW_LINE = 9,
- EMPTY_SPACE = 10,
- COMMENT = 11
-=======
@@ -47 +18,2 @@ enum class CSVState : uint8_t {
->>>>>>> 8ce36237
+ UNQUOTED_ESCAPE = 13,
+ ESCAPED_RETURN = 14
