diff --git a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_conflict.cpp b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_merged.cpp
index 4debab1..4ff9244 100644
--- a/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_conflict.cpp
+++ b/./output/duckdb/ceb52a65-8ce36237-d4c7e729/src@execution@operator@csv_scanner@scanner@string_value_scanner.no_comments_merged.cpp
@@ -1188,2 +1188 @@ void StringValueScanner::ProcessOverBufferValue() {
-<<<<<<< HEAD
-   value = string_t(overbuffer_string.c_str(), UnsafeNumericCast<uint32_t>(overbuffer_string.size()));
+   value = string_t(over_buffer_string.c_str(), UnsafeNumericCast<uint32_t>(over_buffer_string.size()));
@@ -1191 +1190 @@ void StringValueScanner::ProcessOverBufferValue() {
-    value = RemoveEscape(overbuffer_string.c_str(), overbuffer_string.size(),
+    value = RemoveEscape(over_buffer_string.c_str(), over_buffer_string.size(),
@@ -1195,5 +1193,0 @@ void StringValueScanner::ProcessOverBufferValue() {
-||||||| d4c7e729ac
-   value = string_t(overbuffer_string.c_str(), UnsafeNumericCast<uint32_t>(overbuffer_string.size()));
-=======
-   value = string_t(over_buffer_string.c_str(), UnsafeNumericCast<uint32_t>(over_buffer_string.size()));
->>>>>>> 8ce36237
