diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_mergebot.hpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_merged.hpp
index 7c24665..499f9f3 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/ceb52a65-8ce36237-d4c7e729/src@include@duckdb@execution@operator@csv_scanner@csv_state.no_comments_merged.hpp
@@ -17,2 +16,0 @@ enum class CSVState : uint8_t {
- UNQUOTED_ESCAPE = 12,
- ESCAPED_RETURN = 13
@@ -19,0 +18,2 @@ enum class CSVState : uint8_t {
+ UNQUOTED_ESCAPE = 13,
+ ESCAPED_RETURN = 14
