--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/ceb52a65-8ce36237-d4c7e729/src@common@enum_util.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/ceb52a65-8ce36237-d4c7e729/src@common@enum_util.no_comments_merged.cpp
@@ -70 +69,0 @@
-#include "duckdb/core_functions/aggregate/quantile_enum.hpp"
@@ -80 +78,0 @@
-#include "duckdb/function/scalar/compressed_materialization_utils.hpp"
@@ -82 +80 @@
-#include "duckdb/function/scalar/compressed_materialization_functions.hpp"
+#include "duckdb/function/scalar/compressed_materialization_utils.hpp"
@@ -131 +128,0 @@
-#include "duckdb/storage/temporary_file_manager.hpp"
@@ -133,0 +131 @@
+#include "duckdb/storage/temporary_file_manager.hpp"
@@ -499 +500,4 @@ const StringUtil::EnumStringLiteral *GetCSVStateValues() {
-     {static_cast<uint32_t>(CSVState::STANDARD_NEWLINE), "STANDARD_NEWLINE"}};
+  { static_cast<uint32_t>(CSVState::STANDARD_NEWLINE), "STANDARD_NEWLINE" },
+    { static_cast<uint32_t>(CSVState::UNQUOTED_ESCAPE), "UNQUOTED_ESCAPE" },
+    { static_cast<uint32_t>(CSVState::ESCAPED_RETURN), "ESCAPED_RETURN" }
+ };
