diff --git a/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_mergegen.h b/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_merged.h
index 17592be..285f887 100644
--- a/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_mergegen.h
+++ b/./output/grpc/fdd44961-006c8158-7f36ff03/src@core@ext@filters@client_channel@xds@xds_client.no_comments_merged.h
@@ -140 +140,2 @@ class XdsClient : public InternallyRefCounted<XdsClient> {
-const bool xds_routing_enabled_; std::shared_ptr<WorkSerializer> work_serializer_;
+  const bool xds_routing_enabled_;
+  std::shared_ptr<WorkSerializer> work_serializer_;
