diff --git a/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_conflict.cpp b/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_merged.cpp
index 7022477..b8df972 100644
--- a/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_conflict.cpp
+++ b/./output/system_core/796162d7-50e44460-5d1d32dc/init@init.no_comments_merged.cpp
@@ -36 +35,0 @@
-<<<<<<< HEAD
@@ -41,4 +39,0 @@
-||||||| merged common ancestors
-=======
-#include "boringssl_self_test.h"
->>>>>>> 50e44460
