--- a/./output/rocksdb/0c59691d-4e07c99a-0be50ed1/db@db_test.no_comments_conflict.cc
+++ b/./output/rocksdb/0c59691d-4e07c99a-0be50ed1/db@db_test.no_comments_merged.cc
@@ -7725 +7724,0 @@ TEST_F(DBTest, PauseBackgroundWorkTest) {
-<<<<<<< HEAD
@@ -7968,2 +7966,0 @@ TEST_F(DBTest, WalFilterTestWithChangeBatch) {
-||||||| 0be50ed12
-=======
@@ -8004 +8000,0 @@ class BloomStatsTestWithParam
->>>>>>> 4e07c99a
