--- a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@Scheduler@EventThread.no_comments_conflict.h
+++ b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@Scheduler@EventThread.no_comments_merged.h
@@ -46 +45,0 @@ public:
-<<<<<<< HEAD
@@ -48,6 +46,0 @@ public:
-||||||| 7c11ca12cc
-    const ResyncCallback resyncCallback;
-=======
-    binder::Status getSchedulingPolicy(gui::SchedulingPolicy* outPolicy) override;
-    const ResyncCallback resyncCallback;
->>>>>>> 01759208
