--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@main_surfaceflinger.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@main_surfaceflinger.no_comments_merged.cpp
@@ -42 +41,0 @@ static void startDisplayService() {
-    using android::frameworks::displayservice::V1_0::IDisplayService;
@@ -43,0 +43 @@ static void startDisplayService() {
+    using android::frameworks::displayservice::V1_0::IDisplayService;
@@ -87 +87,0 @@ int main(int, char**) {
-<<<<<<< HEAD
@@ -91,4 +90,0 @@ int main(int, char**) {
-|||||||
-=======
-    composerAIDL->setMinSchedulerPolicy(SCHED_FIFO, newPriority);
->>>>>>> 01759208ae46249f5b0a11802af03731f375de95
@@ -103,0 +100 @@ int main(int, char**) {
+#pragma clang diagnostic pop
