diff --git a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_mergegen.cpp b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
index b429935..605b640 100644
--- a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
@@ -7443 +7443,4 @@ binder::Status SurfaceComposerAIDL::createConnection(sp<gui::ISurfaceComposerCli
-if (FlagManager::getInstance().misc1()) { const int policy = SCHED_FIFO; client->setMinSchedulerPolicy(policy, sched_get_priority_min(policy)); }
+        if (FlagManager::getInstance().misc1()) {
+            const int policy = SCHED_FIFO;
+            client->setMinSchedulerPolicy(policy, sched_get_priority_min(policy));
+        }
