diff --git a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@Scheduler@EventThread.no_comments_mergegen.h b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@Scheduler@EventThread.no_comments_merged.h
index 5232a88..ce9fac8 100644
--- a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@Scheduler@EventThread.no_comments_mergegen.h
+++ b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@Scheduler@EventThread.no_comments_merged.h
@@ -46 +46 @@ public:
-binder::Status getLatestVsyncEventData(ParcelableVsyncEventData* outVsyncEventData) override;
+    binder::Status getSchedulingPolicy(gui::SchedulingPolicy* outPolicy) override;
