diff --git a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_conflict.cpp b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
index a00ca8e..605b640 100644
--- a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_conflict.cpp
+++ b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
@@ -7443 +7442,0 @@ binder::Status SurfaceComposerAIDL::createConnection(sp<gui::ISurfaceComposerCli
-<<<<<<< HEAD
@@ -7448,5 +7446,0 @@ binder::Status SurfaceComposerAIDL::createConnection(sp<gui::ISurfaceComposerCli
-||||||| 7c11ca12cc
-=======
-        const int policy = SCHED_FIFO;
-        client->setMinSchedulerPolicy(policy, sched_get_priority_min(policy));
->>>>>>> 01759208
