--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@util.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/a2e43c72-d0ed53e5-ca0d4ffb/init@util.no_comments_merged.cpp
@@ -24,11 +23,0 @@
-#include "selinux.h"
-#else
-#include "host_init_stubs.h"
-#endif
-#if defined(__ANDROID__)
-#include "selinux.h"
-#else
-#include "host_init_stubs.h"
-#endif
-#if defined(__ANDROID__)
-#include "reboot_utils.h"
@@ -40,2 +29 @@
-#error \
-    "Do not include init.h in files used by ueventd; it will expose init's globals"
+#error "Do not include init.h in files used by ueventd; it will expose init's globals"
@@ -361,0 +341 @@ void InitKernelLogging(char** argv) {
+    SetFatalRebootTarget();
