diff --git a/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_conflict.cpp b/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
index ce4ff52..86b80eb 100644
--- a/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_conflict.cpp
+++ b/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
@@ -30 +29,0 @@
-<<<<<<< HEAD
@@ -32,2 +30,0 @@
-||||||| c220f7bc2f
-=======
@@ -35 +31,0 @@
->>>>>>> 18132018
