--- a/./output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_mergegen.cpp
+++ b/./output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_merged.cpp
@@ -18 +17,0 @@ using std::stringstream;
-void ReorderTableEntries(catalog_entry_vector_t &tables);
@@ -196 +195,6 @@ SourceResultType PhysicalExport::GetData(ExecutionContext &context, DataChunk &c
-if (catalog.IsDuckCatalog()) { auto &duck_catalog = catalog.Cast<DuckCatalog>(); auto &dependency_manager = duck_catalog.GetDependencyManager(); dependency_manager.ReorderEntries(catalog_entries, ccontext);
+ catalog_entry_vector_t catalog_entries;
+ catalog_entries = GetNaiveExportOrder(context.client, catalog);
+ if (catalog.IsDuckCatalog()) {
+  auto &duck_catalog = catalog.Cast<DuckCatalog>();
+  auto &dependency_manager = duck_catalog.GetDependencyManager();
+  dependency_manager.ReorderEntries(catalog_entries, ccontext);
