diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_merged.cpp
index b847b81..a3136c6 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/8656b2cc-18132018-c220f7bc/src@execution@operator@persistent@physical_export.no_comments_merged.cpp
@@ -16 +15,0 @@ namespace duckdb {
-using std::stringstream;
@@ -17,0 +17 @@ void ReorderTableEntries(catalog_entry_vector_t &tables);
+using std::stringstream;
@@ -195 +194,0 @@ SourceResultType PhysicalExport::GetData(ExecutionContext &context, DataChunk &c
-<<<<<<< HEAD
@@ -202,13 +200,0 @@ SourceResultType PhysicalExport::GetData(ExecutionContext &context, DataChunk &c
-|||||||
- auto schema_list = Catalog::GetSchemas(ccontext, info->catalog);
- ExtractEntries(context.client, schema_list, entries);
- entries.tables.clear();
- for (idx_t i = 0; i < exported_tables.data.size(); i++) {
-  entries.tables.push_back(exported_tables.data[i].entry);
-=======
- auto schema_list = Catalog::GetSchemas(ccontext, info->catalog);
- ExtractEntries(context.client, schema_list, entries);
- entries.tables.clear();
- for (idx_t i = 0; i < exported_tables->data.size(); i++) {
-  entries.tables.push_back(exported_tables->data[i].entry);
->>>>>>> 181320182c4d7c07f0b91a6e5602bd75717f6e99
