diff --git a/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_mergegen.cpp b/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
index 02d7146..86b80eb 100644
--- a/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_mergegen.cpp
+++ b/./output/duckdb/8656b2cc-18132018-c220f7bc/src@storage@serialization@serialize_nodes.no_comments_merged.cpp
@@ -30 +30,2 @@
-#include "duckdb/parser/qualified_name.hpp" #include "duckdb/parser/parsed_data/exported_table_data.hpp"
+#include "duckdb/parser/qualified_name.hpp"
+#include "duckdb/parser/parsed_data/exported_table_data.hpp"
