--- a/./output/art/fd522049-c15aee68-e82f25f3/artd@artd.no_comments_conflict.h
+++ b/./output/art/fd522049-c15aee68-e82f25f3/artd@artd.no_comments_merged.h
@@ -121 +120,0 @@ class Artd : public aidl::com::android::server::art::BnArtd {
-<<<<<<< HEAD
@@ -128,6 +126,0 @@ class Artd : public aidl::com::android::server::art::BnArtd {
-||||||| e82f25f37b
-  ndk::ScopedAStatus isIncrementalFsPath(const std::string& in_dexFile,
-                                         bool* _aidl_return) override;
-=======
-  ndk::ScopedAStatus isInDalvikCache(const std::string& in_dexFile, bool* _aidl_return) override;
->>>>>>> c15aee68
