diff --git a/./output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_mergegen.cpp b/./output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
index 5e8b221..2d29cab 100644
--- a/./output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
@@ -40 +40,2 @@
-#include "C2AidlNode.h" #include "C2OMXNode.h"
+#include "C2AidlNode.h"
+#include "C2OMXNode.h"
