diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
index f264faf..2d29cab 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/c97c07c8-b8e06f04-5a21203a/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
@@ -40 +39,0 @@
-#include "C2NodeImpl.h"
@@ -50 +48,0 @@ using namespace std::chrono_literals;
-using android::base::StringPrintf;
@@ -51,0 +50 @@ using ::android::hardware::graphics::bufferqueue::V1_0::utils::H2BGraphicBufferP
+using android::base::StringPrintf;
@@ -75 +74 @@ class CCodecWatchdog : public AHandler {
-    ~CCodecWatchdog()
+    ~CCodecWatchdog() = default;
@@ -122,3 +121,5 @@ class C2InputSurfaceWrapper : public InputSurfaceWrapper {
-    explicit C2InputSurfaceWrapper(const std::shared_ptr<Codec2Client::InputSurface>& surface)
-        : mSurface(surface){}
-          () = delete;
+    explicit C2InputSurfaceWrapper(
+            const std::shared_ptr<Codec2Client::InputSurface> &surface) :
+        mSurface(surface) {
+    }
+    ~C2InputSurfaceWrapper() override = default;
@@ -157,7 +157,0 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-  private:
-    sp<HGraphicBufferSource> mSource;
-    sp<C2OMXNode> mNode;
-    sp<hardware::media::omx::V1_0::IOmxNode> mOmxNode;
-    uint32_t mWidth;
-    uint32_t mHeight;
-    Config mConfig;
@@ -166,2 +160,5 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    HGraphicBufferSourceWrapper(const sp<HGraphicBufferSource>& source, uint32_t width,
-                                uint32_t height, uint64_t usage)
+    HGraphicBufferSourceWrapper(
+            const sp<HGraphicBufferSource> &source,
+            uint32_t width,
+            uint32_t height,
+            uint64_t usage)
@@ -172 +169 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    () = delete;
+    ~HGraphicBufferSourceWrapper() override = default;
@@ -178 +175,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        status_t res = mNode->setParameter((OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits64,
+        status_t res = mNode->setParameter(
+                (OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits64,
@@ -182,2 +180,3 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            (void)mNode->setParameter((OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits, &usage,
-                                      sizeof(usage));
+            (void)mNode->setParameter(
+                    (OMX_INDEXTYPE)OMX_IndexParamConsumerUsageBits,
+                    &usage, sizeof(usage));
@@ -187,0 +187,22 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
+    void disconnect() override {
+        if (mNode == nullptr) {
+            return;
+        }
+        sp<IOMXBufferSource> source = mNode->getSource();
+        if (source == nullptr) {
+            ALOGD("GBSWrapper::disconnect: node is not configured with OMXBufferSource.");
+            return;
+        }
+        source->onOmxIdle();
+        source->onOmxLoaded();
+        mNode.clear();
+        mOmxNode.clear();
+    }
+    status_t GetStatus(hardware::Return<OmxStatus> &&status) {
+        if (status.isOk()) {
+            return static_cast<status_t>(status.withDefault(OmxStatus::UNKNOWN_ERROR));
+        } else if (status.isDeadObject()) {
+            return DEAD_OBJECT;
+        }
+        return UNKNOWN_ERROR;
+    }
@@ -197 +218,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        status_t err = mNode->getParameter(OMX_IndexParamPortDefinition, &param, sizeof(param));
+        status_t err = mNode->getParameter(OMX_IndexParamPortDefinition,
+                                           &param, sizeof(param));
@@ -206,0 +229,3 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
+    status_t signalEndOfInputStream() override {
+        return GetStatus(mSource->signalEndOfInputStream());
+    }
@@ -233,2 +258,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1)) &&
-            config.mMaxFps != mConfig.mMaxFps) {
+        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1))
+                && config.mMaxFps != mConfig.mMaxFps) {
@@ -263,2 +288,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if (config.mStartAtUs != mConfig.mStartAtUs ||
-            (config.mStopped != mConfig.mStopped && !config.mStopped)) {
+        if (config.mStartAtUs != mConfig.mStartAtUs
+                || (config.mStopped != mConfig.mStopped && !config.mStopped)) {
@@ -276,2 +301,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            status << " " << (config.mSuspended ? "suspend" : "resume") << " at "
-                   << config.mSuspendAtUs << "us";
+            status << " " << (config.mSuspended ? "suspend" : "resume")
+                    << " at " << config.mSuspendAtUs << "us";
@@ -293,2 +318,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-                hardware::Return<void> trans =
-                        mSource->getStopTimeOffsetUs([&res, &delayUs = config.mInputDelayUs](
+                hardware::Return<void> trans = mSource->getStopTimeOffsetUs(
+                        [&res, &delayUs = config.mInputDelayUs](
@@ -325,3 +350,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    void disconnect() override {
-        if (mNode == nullptr) {
-            return;
+    void onInputBufferDone(c2_cntr64_t index) override {
+        mNode->onInputBufferDone(index);
@@ -329,4 +353,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        sp<IOMXBufferSource> source = mNode->getSource();
-        if (source == nullptr) {
-            ALOGD("GBSWrapper::disconnect: node is not configured with OMXBufferSource.");
-            return;
+    android_dataspace getDataspace() override {
+        return mNode->getDataspace();
@@ -334,15 +356,2 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        source->onOmxIdle();
-        source->onOmxLoaded();
-        mNode.clear();
-        mOmxNode.clear();
-    }
-    status_t GetStatus(hardware::Return<OmxStatus>&& status) {
-        if (status.isOk()) {
-            return static_cast<status_t>(status.withDefault(OmxStatus::UNKNOWN_ERROR));
-        } else if (status.isDeadObject()) {
-            return DEAD_OBJECT;
-        }
-        return UNKNOWN_ERROR;
-    }
-    status_t signalEndOfInputStream() override {
-        return GetStatus(mSource->signalEndOfInputStream());
+    uint32_t getPixelFormat() override {
+        return mNode->getPixelFormat();
@@ -350,3 +359,7 @@ class HGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    void onInputBufferDone(c2_cntr64_t index) override { mNode->onInputBufferDone(index); }
-    android_dataspace getDataspace() override { return mNode->getDataspace(); }
-    uint32_t getPixelFormat() override { return mNode->getPixelFormat(); }
+private:
+    sp<HGraphicBufferSource> mSource;
+    sp<C2OMXNode> mNode;
+    sp<hardware::media::omx::V1_0::IOmxNode> mOmxNode;
+    uint32_t mWidth;
+    uint32_t mHeight;
+    Config mConfig;
@@ -356,2 +369,5 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    AGraphicBufferSourceWrapper(const std::shared_ptr<AGraphicBufferSource>& source, uint32_t width,
-                                uint32_t height, uint64_t usage)
+    AGraphicBufferSourceWrapper(
+            const std::shared_ptr<AGraphicBufferSource> &source,
+            uint32_t width,
+            uint32_t height,
+            uint64_t usage)
@@ -362 +378 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    () = delete;
+    ~AGraphicBufferSourceWrapper() override = default;
@@ -369,2 +385,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-                mNode,
-                static_cast<::aidl::android::hardware::graphics::common::Dataspace>(mDataSpace)));
+                mNode, static_cast<::aidl::android::hardware::graphics::common::Dataspace>(
+                        mDataSpace)));
@@ -427,2 +443,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1)) &&
-            config.mMaxFps != mConfig.mMaxFps) {
+        if ((config.mMaxFps > 0 || (config.mFixedAdjustedFps > 0 && config.mMaxFps == -1))
+                && config.mMaxFps != mConfig.mMaxFps) {
@@ -447,2 +463,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            status_t res = fromAidlStatus(
-                    mSource->setTimeLapseConfig(config.mCodedFps, config.mCaptureFps));
+            status_t res =
+                fromAidlStatus(mSource->setTimeLapseConfig(config.mCodedFps, config.mCaptureFps));
@@ -457,2 +473,2 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-        if (config.mStartAtUs != mConfig.mStartAtUs ||
-            (config.mStopped != mConfig.mStopped && !config.mStopped)) {
+        if (config.mStartAtUs != mConfig.mStartAtUs
+                || (config.mStopped != mConfig.mStopped && !config.mStopped)) {
@@ -469,4 +485,4 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-            status_t res =
-                    fromAidlStatus(mSource->setSuspend(config.mSuspended, config.mSuspendAtUs));
-            status << " " << (config.mSuspended ? "suspend" : "resume") << " at "
-                   << config.mSuspendAtUs << "us";
+            status_t res = fromAidlStatus(mSource->setSuspend(
+                    config.mSuspended, config.mSuspendAtUs));
+            status << " " << (config.mSuspended ? "suspend" : "resume")
+                    << " at " << config.mSuspendAtUs << "us";
@@ -512,3 +528,9 @@ class AGraphicBufferSourceWrapper : public InputSurfaceWrapper {
-    void onInputBufferDone(c2_cntr64_t index) override { mNode->onInputBufferDone(index); }
-    android_dataspace getDataspace() override { return mNode->getDataspace(); }
-    uint32_t getPixelFormat() override { return mNode->getPixelFormat(); }
+    void onInputBufferDone(c2_cntr64_t index) override {
+        mNode->onInputBufferDone(index);
+    }
+    android_dataspace getDataspace() override {
+        return mNode->getDataspace();
+    }
+    uint32_t getPixelFormat() override {
+        return mNode->getPixelFormat();
+    }
@@ -525,4 +547,5 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    Codec2ClientInterfaceWrapper(std::shared_ptr<Codec2Client> client) : mClient(client) {}
-    virtual ~Codec2ClientInterfaceWrapper()
-            virtual c2_status_t
-            config_sm(const std::vector<C2Param*>& params,
+    Codec2ClientInterfaceWrapper(std::shared_ptr<Codec2Client> client)
+        : mClient(client) { }
+    virtual ~Codec2ClientInterfaceWrapper() = default;
+    virtual c2_status_t config_sm(
+            const std::vector<C2Param *> &params,
@@ -531,2 +554,3 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    }
-    virtual c2_status_t copyBuffer(std::shared_ptr<C2GraphicBuffer>,
+    };
+    virtual c2_status_t copyBuffer(
+            std::shared_ptr<C2GraphicBuffer>,
@@ -536 +560,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual c2_status_t createComponent(C2String, std::shared_ptr<C2Component>* const component) {
+    virtual c2_status_t createComponent(
+            C2String, std::shared_ptr<C2Component> *const component) {
@@ -540,2 +565,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual c2_status_t createInterface(C2String,
-                                        std::shared_ptr<C2ComponentInterface>* const interface) {
+    virtual c2_status_t createInterface(
+            C2String, std::shared_ptr<C2ComponentInterface> *const interface) {
@@ -545 +570,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual c2_status_t query_sm(const std::vector<C2Param*>& stackParams,
+    virtual c2_status_t query_sm(
+            const std::vector<C2Param *> &stackParams,
@@ -558 +584,3 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-    virtual C2String getName() const { return mClient->getName(); }
+    virtual C2String getName() const {
+        return mClient->getName();
+    }
@@ -566 +594,2 @@ class Codec2ClientInterfaceWrapper : public C2ComponentStore {
-void RevertOutputFormatIfNeeded(const sp<AMessage>& oldFormat, sp<AMessage>& currentFormat) {
+void RevertOutputFormatIfNeeded(
+        const sp<AMessage> &oldFormat, sp<AMessage> &currentFormat) {
@@ -591,2 +620,2 @@ void RevertOutputFormatIfNeeded(const sp<AMessage>& oldFormat, sp<AMessage>& cur
-void AmendOutputFormatWithCodecSpecificData(const uint8_t* data, size_t size,
-                                            const std::string& mediaType,
+void AmendOutputFormatWithCodecSpecificData(
+        const uint8_t *data, size_t size, const std::string &mediaType,
@@ -602 +631,2 @@ void AmendOutputFormatWithCodecSpecificData(const uint8_t* data, size_t size,
-            outputFormat->setBuffer(AStringPrintf("csd-%u", csdIndex).c_str(), csd);
+            outputFormat->setBuffer(
+                    AStringPrintf("csd-%u", csdIndex).c_str(), csd);
@@ -606 +636,2 @@ void AmendOutputFormatWithCodecSpecificData(const uint8_t* data, size_t size,
-            ALOGW("Expected two NAL units from AVC codec config, but %u found", csdIndex);
+            ALOGW("Expected two NAL units from AVC codec config, but %u found",
+                    csdIndex);
@@ -618 +649,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onWorkDone(const std::weak_ptr<Codec2Client::Component>& component,
+    virtual void onWorkDone(
+            const std::weak_ptr<Codec2Client::Component>& component,
@@ -629 +661,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-            const std::vector<std::shared_ptr<C2SettingResult>>& settingResult) override {
+            const std::vector<std::shared_ptr<C2SettingResult>>& settingResult
+            ) override {
@@ -633 +666,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onError(const std::weak_ptr<Codec2Client::Component>& component,
+    virtual void onError(
+            const std::weak_ptr<Codec2Client::Component>& component,
@@ -647 +681,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-        codec->mCallback->onError(errorCode == C2_NO_MEMORY ? NO_MEMORY : UNKNOWN_ERROR,
+        codec->mCallback->onError(
+                errorCode == C2_NO_MEMORY ? NO_MEMORY : UNKNOWN_ERROR,
@@ -650 +685,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onDeath(const std::weak_ptr<Codec2Client::Component>& component) override {
+    virtual void onDeath(
+            const std::weak_ptr<Codec2Client::Component>& component) override {
@@ -665 +701,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onFrameRendered(uint64_t bufferQueueId, int32_t slotId,
+    virtual void onFrameRendered(uint64_t bufferQueueId,
+                                 int32_t slotId,
@@ -671 +708,2 @@ struct CCodec::ClientListener : public Codec2Client::Listener {
-    virtual void onInputBufferDone(uint64_t frameIndex, size_t arrayIndex) override {
+    virtual void onInputBufferDone(
+            uint64_t frameIndex, size_t arrayIndex) override {
@@ -682,2 +720,2 @@ class CCodecCallbackImpl : public CCodecCallback {
-    explicit CCodecCallbackImpl(CCodec* codec) : mCodec(codec){}() = delete;
-    () = delete;
+    explicit CCodecCallbackImpl(CCodec *codec) : mCodec(codec) {}
+    ~CCodecCallbackImpl() override = default;
@@ -690,2 +728,6 @@ class CCodecCallbackImpl : public CCodecCallback {
-    void onOutputBuffersChanged() override { mCodec->mCallback->onOutputBuffersChanged(); }
-    void onFirstTunnelFrameReady() override { mCodec->mCallback->onFirstTunnelFrameReady(); }
+    void onOutputBuffersChanged() override {
+        mCodec->mCallback->onOutputBuffersChanged();
+    }
+    void onFirstTunnelFrameReady() override {
+        mCodec->mCallback->onFirstTunnelFrameReady();
+    }
@@ -697,2 +739,4 @@ CCodec::CCodec()
-      mConfig(new CCodecConfig) {}
-CCodec::~CCodec() {}
+      mConfig(new CCodecConfig) {
+}
+CCodec::~CCodec() {
+}
@@ -739 +783,2 @@ void CCodec::allocate(const sp<MediaCodecInfo>& codecInfo) {
-        SetPreferredCodec2ComponentStore(std::make_shared<Codec2ClientInterfaceWrapper>(client));
+        SetPreferredCodec2ComponentStore(
+                std::make_shared<Codec2ClientInterfaceWrapper>(client));
@@ -742,2 +787,5 @@ void CCodec::allocate(const sp<MediaCodecInfo>& codecInfo) {
-    c2_status_t status = Codec2Client::CreateComponentByName(componentName.c_str(), mClientListener,
-                                                             &comp, &client);
+    c2_status_t status = Codec2Client::CreateComponentByName(
+            componentName.c_str(),
+            mClientListener,
+            &comp,
+            &client);
@@ -821,2 +869,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if (encoder && msg->findInt32("android._input-metadata-buffer-type", &storeMeta) &&
-            storeMeta != kMetadataBufferTypeInvalid) {
+        if (encoder
+                && msg->findInt32("android._input-metadata-buffer-type", &storeMeta)
+                && storeMeta != kMetadataBufferTypeInvalid) {
@@ -839,2 +888,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if ((config->mDomain & Config::IS_DECODER) &&
-                    (config->mDomain & Config::IS_VIDEO)) {
+                if ((config->mDomain & Config::IS_DECODER)
+                        && (config->mDomain & Config::IS_VIDEO)) {
@@ -871,2 +920,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        ALOGD("[%s] buffers are %sbound to CCodec for this session", comp->getName().c_str(),
-              config->mBuffersBoundToCodec ? "" : "not ");
+        ALOGD("[%s] buffers are %sbound to CCodec for this session",
+              comp->getName().c_str(), config->mBuffersBoundToCodec ? "" : "not ");
@@ -884,3 +933,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            if ((config->mDomain & Config::IS_ENCODER) &&
-                !mime.equalsIgnoreCase(MEDIA_MIMETYPE_AUDIO_FLAC) &&
-                !msg->findInt32(KEY_BIT_RATE, &i32) && !msg->findFloat(KEY_BIT_RATE, &flt)) {
+            if ((config->mDomain & Config::IS_ENCODER)
+                    && !mime.equalsIgnoreCase(MEDIA_MIMETYPE_AUDIO_FLAC)
+                    && !msg->findInt32(KEY_BIT_RATE, &i32)
+                    && !msg->findFloat(KEY_BIT_RATE, &flt)) {
@@ -910,2 +960,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (!msg->findInt32(KEY_BIT_RATE, &i32) &&
-                        !msg->findFloat(KEY_BIT_RATE, &flt)) {
+                    if (!msg->findInt32(KEY_BIT_RATE, &i32)
+                            && !msg->findFloat(KEY_BIT_RATE, &flt)) {
@@ -916,2 +966,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if (!msg->findInt32(KEY_I_FRAME_INTERVAL, &i32) &&
-                    !msg->findFloat(KEY_I_FRAME_INTERVAL, &flt)) {
+                if (!msg->findInt32(KEY_I_FRAME_INTERVAL, &i32)
+                        && !msg->findFloat(KEY_I_FRAME_INTERVAL, &flt)) {
@@ -921,2 +971,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if (!msg->findInt32(KEY_FRAME_RATE, &i32) &&
-                    !msg->findFloat(KEY_FRAME_RATE, &flt)) {
+                if (!msg->findInt32(KEY_FRAME_RATE, &i32)
+                        && !msg->findFloat(KEY_FRAME_RATE, &flt)) {
@@ -928,2 +978,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE)) &&
-            (config->mDomain & Config::IS_ENCODER)) {
+        if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE))
+                && (config->mDomain & Config::IS_ENCODER)) {
@@ -937 +987,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                if (!msg->findFloat(KEY_MAX_FPS_TO_ENCODER, &config->mISConfig->mMaxFps)) {
+                if (!msg->findFloat(
+                        KEY_MAX_FPS_TO_ENCODER, &config->mISConfig->mMaxFps)) {
@@ -982,7 +1033,10 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            int vendorSdkVersion = base::GetIntProperty("ro.vendor.build.version.sdk",
-                                                        android_get_device_api_level());
-            if (mClient->query({}, {C2StoreFlexiblePixelFormatDescriptorsInfo::PARAM_TYPE},
-                               C2_MAY_BLOCK, &heapParams) == C2_OK &&
-                heapParams.size() == 1u) {
-                pixelFormatInfo =
-                        C2StoreFlexiblePixelFormatDescriptorsInfo::From(heapParams[0].get());
+            int vendorSdkVersion = base::GetIntProperty(
+                    "ro.vendor.build.version.sdk", android_get_device_api_level());
+            if (mClient->query(
+                        {},
+                        {C2StoreFlexiblePixelFormatDescriptorsInfo::PARAM_TYPE},
+                        C2_MAY_BLOCK,
+                        &heapParams) == C2_OK
+                    && heapParams.size() == 1u) {
+                pixelFormatInfo = C2StoreFlexiblePixelFormatDescriptorsInfo::From(
+                        heapParams[0].get());
@@ -1006,2 +1060,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (desc.layout == C2Color::PLANAR_PACKED &&
-                        flexPlanarPixelFormat.count(desc.bitDepth) == 0) {
+                    if (desc.layout == C2Color::PLANAR_PACKED
+                            && flexPlanarPixelFormat.count(desc.bitDepth) == 0) {
@@ -1010,2 +1064,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (desc.layout == C2Color::SEMIPLANAR_PACKED &&
-                        flexSemiPlanarPixelFormat.count(desc.bitDepth) == 0) {
+                    if (desc.layout == C2Color::SEMIPLANAR_PACKED
+                            && flexSemiPlanarPixelFormat.count(desc.bitDepth) == 0) {
@@ -1027,2 +1081 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                              "using default %splanar color format",
-                              prefix);
+                                "using default %splanar color format", prefix);
@@ -1087 +1140,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            if (msg->findInt32("width", &width) && msg->findInt32("height", &height)) {
+            if (msg->findInt32("width", &width)
+                    && msg->findInt32("height", &height)) {
@@ -1110,2 +1164,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        err = config->getConfigUpdateFromSdkParams(comp, sdkParams, Config::IS_CONFIG,
-                                                   C2_DONT_BLOCK, &configUpdate);
+        err = config->getConfigUpdateFromSdkParams(
+                comp, sdkParams, Config::IS_CONFIG, C2_DONT_BLOCK, &configUpdate);
@@ -1116,2 +1170,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_ENCODER) && (config->mDomain & Config::IS_VIDEO) &&
-            sdkParams->findInt32(KEY_MAX_B_FRAMES, &maxBframes) && maxBframes > 0) {
+        if ((config->mDomain & Config::IS_ENCODER)
+                && (config->mDomain & Config::IS_VIDEO)
+                && sdkParams->findInt32(KEY_MAX_B_FRAMES, &maxBframes)
+                && maxBframes > 0) {
@@ -1121 +1177,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            gop->m.values[1] = {C2Config::picture_type_t(P_FRAME | B_FRAME), uint32_t(maxBframes)};
+            gop->m.values[1] = {
+                C2Config::picture_type_t(P_FRAME | B_FRAME),
+                uint32_t(maxBframes)
+            };
@@ -1124 +1183,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_ENCODER) && (config->mDomain & Config::IS_VIDEO)) {
+        if ((config->mDomain & Config::IS_ENCODER)
+                && (config->mDomain & Config::IS_VIDEO)) {
@@ -1154,2 +1214,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_VIDEO) &&
-            msg->findInt32("android._background-mode", &background) && background) {
+        if ((config->mDomain & Config::IS_VIDEO)
+                && msg->findInt32("android._background-mode", &background)
+                && background) {
@@ -1176,2 +1237,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if (config->mDomain & (Config::IS_IMAGE | Config::IS_VIDEO) &&
-            !sdkParams->findInt32("color-transfer-request", &colorTransferRequest)) {
+        if (config->mDomain & (Config::IS_IMAGE | Config::IS_VIDEO)
+                && !sdkParams->findInt32("color-transfer-request", &colorTransferRequest)) {
@@ -1182 +1243,5 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            c2err = comp->query({&usage, &maxInputSize, &prepend}, indices, C2_DONT_BLOCK, &params);
+            c2err = comp->query(
+                    { &usage, &maxInputSize, &prepend },
+                    indices,
+                    C2_DONT_BLOCK,
+                    &params);
@@ -1184 +1249,5 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            c2err = comp->query({&usage, &maxInputSize, &prepend}, {}, C2_DONT_BLOCK, &params);
+            c2err = comp->query(
+                    { &usage, &maxInputSize, &prepend },
+                    {},
+                    C2_DONT_BLOCK,
+                    &params);
@@ -1223,2 +1292 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                          "recommendation.",
-                          clientInputSize, maxInputSize.value);
+                          "recommendation.", clientInputSize, maxInputSize.value);
@@ -1239,3 +1307,5 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_VIDEO) && (config->mDomain & Config::IS_ENCODER) &&
-            msg->findInt32(KEY_PREPEND_HEADER_TO_SYNC_FRAMES, &clientPrepend) && clientPrepend &&
-            (!prepend || prepend.value != PREPEND_HEADER_TO_ALL_SYNC)) {
+        if ((config->mDomain & Config::IS_VIDEO)
+                && (config->mDomain & Config::IS_ENCODER)
+                && msg->findInt32(KEY_PREPEND_HEADER_TO_SYNC_FRAMES, &clientPrepend)
+                && clientPrepend
+                && (!prepend || prepend.value != PREPEND_HEADER_TO_ALL_SYNC)) {
@@ -1248,2 +1318,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-            if (msg->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo) &&
-                !config->mOutputFormat->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo)) {
+            if (msg->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo)
+                    && !config->mOutputFormat->findBuffer(KEY_HDR_STATIC_INFO, &hdrInfo)) {
@@ -1317,3 +1387,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                std::shared_ptr<C2GraphicBlock> block =
-                        FetchGraphicBlock(align(width, 2), align(height, 2), componentColorFormat,
-                                          blockUsage, {comp->getName()});
+                std::shared_ptr<C2GraphicBlock> block = FetchGraphicBlock(
+                        align(width, 2), align(height, 2), componentColorFormat, blockUsage,
+                        {comp->getName()});
@@ -1323 +1393,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                            config->mInputFormat, block,
+                            config->mInputFormat,
+                            block,
@@ -1333,2 +1404,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                    if (imageData && imageData->data() &&
-                        imageData->size() >= sizeof(MediaImage2)) {
+                    if (imageData && imageData->data()
+                            && imageData->size() >= sizeof(MediaImage2)) {
@@ -1341,2 +1412,2 @@ void CCodec::configure(const sp<AMessage>& msg) {
-                            int64_t offsetDelta = (int64_t)img->mPlane[1].mOffset -
-                                                  (int64_t)img->mPlane[0].mOffset;
+                            int64_t offsetDelta =
+                                (int64_t)img->mPlane[1].mOffset - (int64_t)img->mPlane[0].mOffset;
@@ -1360,2 +1431,3 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        if ((config->mDomain & Config::IS_ENCODER) && (config->mDomain & Config::IS_VIDEO) &&
-            msg->findInt32(KEY_VIDEO_ENCODING_STATISTICS_LEVEL, &encodingStatisticsLevel)) {
+        if ((config->mDomain & Config::IS_ENCODER)
+            && (config->mDomain & Config::IS_VIDEO)
+            && msg->findInt32(KEY_VIDEO_ENCODING_STATISTICS_LEVEL, &encodingStatisticsLevel)) {
@@ -1376,2 +1448,4 @@ void CCodec::configure(const sp<AMessage>& msg) {
-        ALOGD("setup formats input: %s", config->mInputFormat->debugString().c_str());
-        ALOGD("setup formats output: %s", config->mOutputFormat->debugString().c_str());
+        ALOGD("setup formats input: %s",
+                config->mInputFormat->debugString().c_str());
+        ALOGD("setup formats output: %s",
+                config->mOutputFormat->debugString().c_str());
@@ -1416,3 +1490,4 @@ sp<PersistentSurface> CCodec::CreateOmxInputSurface() {
-    typedef android::hardware::graphics::bufferqueue::V1_0::IGraphicBufferProducer
-            HGraphicBufferProducer;
-    typedef android::hardware::media::omx::V1_0::IGraphicBufferSource HGraphicBufferSource;
+    typedef android::hardware::graphics::bufferqueue::V1_0::
+            IGraphicBufferProducer HGraphicBufferProducer;
+    typedef android::hardware::media::omx::V1_0::
+            IGraphicBufferSource HGraphicBufferSource;
@@ -1424 +1499,3 @@ sp<PersistentSurface> CCodec::CreateOmxInputSurface() {
-            [&s, &gbp, &gbs](OmxStatus status, const android::sp<HGraphicBufferProducer>& producer,
+            [&s, &gbp, &gbs](
+                    OmxStatus status,
+                    const android::sp<HGraphicBufferProducer>& producer,
@@ -1462,2 +1539,2 @@ void CCodec::createInputSurface() {
-            err = setupInputSurface(
-                    std::make_shared<AGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            err = setupInputSurface(std::make_shared<AGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1477 +1554,2 @@ void CCodec::createInputSurface() {
-            err = setupInputSurface(std::make_shared<C2InputSurfaceWrapper>(inputSurface));
+            err = setupInputSurface(std::make_shared<C2InputSurfaceWrapper>(
+                    inputSurface));
@@ -1484,2 +1562,2 @@ void CCodec::createInputSurface() {
-            err = setupInputSurface(
-                    std::make_shared<HGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            err = setupInputSurface(std::make_shared<HGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1505 +1583,3 @@ void CCodec::createInputSurface() {
-    mCallback->onInputSurfaceCreated(inputFormat, outputFormat,
+    mCallback->onInputSurfaceCreated(
+            inputFormat,
+            outputFormat,
@@ -1529,2 +1609,2 @@ status_t CCodec::setupInputSurface(const std::shared_ptr<InputSurfaceWrapper>& s
-    android_dataspace dataSpace =
-            getDataSpaceForColorAspects(colorAspects, false);
+    android_dataspace dataSpace = getDataSpaceForColorAspects(
+            colorAspects, false);
@@ -1534 +1614,2 @@ status_t CCodec::setupInputSurface(const std::shared_ptr<InputSurfaceWrapper>& s
-    ALOGD("input format %s to %s", inputFormatChanged ? "changed" : "unchanged",
+    ALOGD("input format %s to %s",
+            inputFormatChanged ? "changed" : "unchanged",
@@ -1572,2 +1653,2 @@ void CCodec::setInputSurface(const sp<PersistentSurface>& surface) {
-            status_t err = setupInputSurface(
-                    std::make_shared<AGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            status_t err = setupInputSurface(std::make_shared<AGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1601,2 +1682,2 @@ void CCodec::setInputSurface(const sp<PersistentSurface>& surface) {
-            status_t err = setupInputSurface(
-                    std::make_shared<HGraphicBufferSourceWrapper>(gbs, width, height, usage));
+            status_t err = setupInputSurface(std::make_shared<HGraphicBufferSourceWrapper>(
+                    gbs, width, height, usage));
@@ -1652 +1733,2 @@ void CCodec::start() {
-        mCallback->onError(toStatusT(err, C2_OPERATION_Component_start), ACTION_CODE_FATAL);
+        mCallback->onError(toStatusT(err, C2_OPERATION_Component_start),
+                           ACTION_CODE_FATAL);
@@ -1711,2 +1793,4 @@ void CCodec::initiateStop() {
-        if (state->get() == ALLOCATED || state->get() == RELEASED || state->get() == STOPPING ||
-            state->get() == RELEASING) {
+        if (state->get() == ALLOCATED
+                || state->get() == RELEASED
+                || state->get() == STOPPING
+                || state->get() == RELEASING) {
@@ -1786 +1870,3 @@ void CCodec::initiateRelease(bool sendCallback ) {
-        if (state->get() == STARTING || state->get() == RUNNING || state->get() == STOPPING) {
+        if (state->get() == STARTING
+                || state->get() == RUNNING
+                || state->get() == STOPPING) {
@@ -1803,3 +1889,2 @@ void CCodec::initiateRelease(bool sendCallback ) {
-    std::thread([thiz, sendCallback, pushBlankBuffer] {
-        thiz->release(sendCallback, pushBlankBuffer);
-    }).detach();
+    std::thread([thiz, sendCallback, pushBlankBuffer]
+                { thiz->release(sendCallback, pushBlankBuffer); }).detach();
@@ -1985,2 +2070,2 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-    if (params->findInt32("audio-hw-sync", &syncId) ||
-        params->findInt32("hw-av-sync-id", &syncId)) {
+    if (params->findInt32("audio-hw-sync", &syncId)
+            || params->findInt32("hw-av-sync-id", &syncId)) {
@@ -1991,2 +2076,3 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-    if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE)) &&
-        (config->mDomain & Config::IS_ENCODER) && config->mInputSurface && config->mISConfig) {
+    if ((config->mDomain & (Config::IS_VIDEO | Config::IS_IMAGE))
+            && (config->mDomain & Config::IS_ENCODER)
+            && config->mInputSurface && config->mISConfig) {
@@ -2007,2 +2093,2 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-            config->mInputFormat->setInt64("android._stop-time-offset-us",
-                                           config->mISConfig->mInputDelayUs);
+            config->mInputFormat->setInt64(
+                    "android._stop-time-offset-us", config->mISConfig->mInputDelayUs);
@@ -2012,5 +2098,5 @@ void CCodec::signalSetParameters(const sp<AMessage>& msg) {
-    (void)config->getConfigUpdateFromSdkParams(comp, params, Config::IS_PARAM, C2_MAY_BLOCK,
-                                               &configUpdate);
-    if (config->mInputSurface == nullptr &&
-        (property_get_bool("debug.stagefright.ccodec_delayed_params", false) ||
-         comp->getName().find("c2.android.") == 0)) {
+    (void)config->getConfigUpdateFromSdkParams(
+            comp, params, Config::IS_PARAM, C2_MAY_BLOCK, &configUpdate);
+    if (config->mInputSurface == nullptr
+            && (property_get_bool("debug.stagefright.ccodec_delayed_params", false)
+                    || comp->getName().find("c2.android.") == 0)) {
@@ -2041 +2127,2 @@ void CCodec::signalRequestIDRFrame() {
-    params.push_back(std::make_unique<C2StreamRequestSyncFrameTuning::output>(0u, true));
+    params.push_back(
+            std::make_unique<C2StreamRequestSyncFrameTuning::output>(0u, true));
@@ -2049 +2136,2 @@ status_t CCodec::querySupportedParameters(std::vector<std::string>* names) {
-status_t CCodec::describeParameter(const std::string& name, CodecParameterDescriptor* desc) {
+status_t CCodec::describeParameter(
+        const std::string &name, CodecParameterDescriptor *desc) {
@@ -2170,2 +2258,3 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                if (!work->worklets.empty() &&
-                    (work->worklets.front()->output.flags & C2FrameData::FLAG_DISCARD_FRAME) == 0) {
+                if (!work->worklets.empty()
+                        && (work->worklets.front()->output.flags
+                                & C2FrameData::FLAG_DISCARD_FRAME) == 0) {
@@ -2173,2 +2262,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                    for (const std::unique_ptr<C2Param>& param :
-                         work->worklets.front()->output.configUpdate) {
+                    for (const std::unique_ptr<C2Param> &param
+                            : work->worklets.front()->output.configUpdate) {
@@ -2188,2 +2277,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                            updates.emplace_back(
-                                    new C2StreamCropRectInfo::output(stream, block.crop()));
+                            updates.emplace_back(new C2StreamCropRectInfo::output(
+                                    stream, block.crop()));
@@ -2205 +2294,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                            C2StreamSurfaceScalingInfo::output::PARAM_TYPE};
+                        C2StreamSurfaceScalingInfo::output::PARAM_TYPE
+                    };
@@ -2223,2 +2313,4 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                    if (work->worklets.empty() || !work->worklets.back() ||
-                        (work->worklets.back()->output.flags & C2FrameData::FLAG_INCOMPLETE) == 0) {
+                    if (work->worklets.empty()
+                           || !work->worklets.back()
+                           || (work->worklets.back()->output.flags
+                                  & C2FrameData::FLAG_INCOMPLETE) == 0) {
@@ -2230,2 +2322,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-                    AmendOutputFormatWithCodecSpecificData(initData->m.value, initData->flexCount(),
-                                                           config->mCodingMediaType,
+                    AmendOutputFormatWithCodecSpecificData(
+                            initData->m.value, initData->flexCount(), config->mCodingMediaType,
@@ -2236,2 +2328,2 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-            mChannel->onWorkDone(std::move(work), outputFormat,
-                                 initData ? initData.get() : nullptr);
+            mChannel->onWorkDone(
+                    std::move(work), outputFormat, initData ? initData.get() : nullptr);
@@ -2264 +2356,3 @@ void CCodec::onMessageReceived(const sp<AMessage>& msg) {
-void CCodec::setDeadline(const TimePoint& now, const std::chrono::milliseconds& timeout,
+void CCodec::setDeadline(
+        const TimePoint &now,
+        const std::chrono::milliseconds &timeout,
@@ -2270 +2364,2 @@ void CCodec::setDeadline(const TimePoint& now, const std::chrono::milliseconds&
-status_t CCodec::configureTunneledVideoPlayback(std::shared_ptr<Codec2Client::Component> comp,
+status_t CCodec::configureTunneledVideoPlayback(
+        std::shared_ptr<Codec2Client::Component> comp,
@@ -2276,2 +2371,4 @@ status_t CCodec::configureTunneledVideoPlayback(std::shared_ptr<Codec2Client::Co
-                    1, C2PortTunneledModeTuning::Struct::SIDEBAND,
-                    C2PortTunneledModeTuning::Struct::REALTIME, 0);
+            1,
+            C2PortTunneledModeTuning::Struct::SIDEBAND,
+            C2PortTunneledModeTuning::Struct::REALTIME,
+            0);
@@ -2333,2 +2430 @@ void CCodec::initiateReleaseIfStuck() {
-                  "and the component is already released",
-                  name.c_str());
+                  "and the component is already released", name.c_str());
@@ -2346 +2442,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-    std::shared_ptr<Codec2Client::InputSurface> inputSurface = Codec2Client::CreateInputSurface();
+    std::shared_ptr<Codec2Client::InputSurface> inputSurface =
+            Codec2Client::CreateInputSurface();
@@ -2360 +2457,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-                return new PersistentSurface(gbs->getIGraphicBufferProducer(), wrapper->asBinder());
+                return new PersistentSurface(
+                      gbs->getIGraphicBufferProducer(), wrapper->asBinder());
@@ -2370,2 +2468,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-                return new PersistentSurface(gbs->getIGraphicBufferProducer(),
-                                             new TWGraphicBufferSource(gbs));
+                return new PersistentSurface(
+                        gbs->getIGraphicBufferProducer(), new TWGraphicBufferSource(gbs));
@@ -2379 +2477,2 @@ PersistentSurface* CCodec::CreateInputSurface() {
-            static_cast<sp<android::hidl::base::V1_0::IBase>>(inputSurface->getHalInterface()));
+            static_cast<sp<android::hidl::base::V1_0::IBase>>(
+            inputSurface->getHalInterface()));
@@ -2383 +2482 @@ class IntfCache {
-    IntfCache()
+    IntfCache() = default;
@@ -2393,2 +2492,2 @@ class IntfCache {
-        mFields.push_back(
-                C2FieldSupportedValuesQuery::Possible(C2ParamField{&sUsage, &sUsage.value}));
+        mFields.push_back(C2FieldSupportedValuesQuery::Possible(
+                C2ParamField{&sUsage, &sUsage.value}));
@@ -2397,2 +2496,2 @@ class IntfCache {
-            ALOGW("IntfCache [%s]: failed to query usage supported value (err=%d)", name.c_str(),
-                  err);
+            ALOGW("IntfCache [%s]: failed to query usage supported value (err=%d)",
+                    name.c_str(), err);
@@ -2402,4 +2501,8 @@ class IntfCache {
-        err = intf->query({&mApiFeatures},
-                          {C2StreamBufferTypeSetting::input::PARAM_TYPE,
-                           C2PortAllocatorsTuning::input::PARAM_TYPE},
-                          C2_MAY_BLOCK, &params);
+        err = intf->query(
+                {&mApiFeatures},
+                {
+                    C2StreamBufferTypeSetting::input::PARAM_TYPE,
+                    C2PortAllocatorsTuning::input::PARAM_TYPE
+                },
+                C2_MAY_BLOCK,
+                &params);
@@ -2407 +2510,2 @@ class IntfCache {
-            ALOGW("IntfCache [%s]: failed to query api features (err=%d)", name.c_str(), err);
+            ALOGW("IntfCache [%s]: failed to query api features (err=%d)",
+                    name.c_str(), err);
@@ -2416 +2520,2 @@ class IntfCache {
-                mInputStreamFormat.reset(C2StreamBufferTypeSetting::input::From(param));
+                mInputStreamFormat.reset(
+                        C2StreamBufferTypeSetting::input::From(param));
@@ -2418 +2523,2 @@ class IntfCache {
-                mInputAllocators.reset(C2PortAllocatorsTuning::input::From(param));
+                mInputAllocators.reset(
+                        C2PortAllocatorsTuning::input::From(param));
@@ -2429 +2535,3 @@ class IntfCache {
-    const C2ApiFeaturesSetting& getApiFeatures() const { return mApiFeatures; }
+    const C2ApiFeaturesSetting &getApiFeatures() const {
+        return mApiFeatures;
+    }
@@ -2473,2 +2581,4 @@ static const IntfCache& GetIntfCache(const std::string& name) {
-static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-                                      C2Allocator::type_t type, std::set<C2Allocator::id_t>* ids) {
+static status_t GetCommonAllocatorIds(
+        const std::vector<std::string> &names,
+        C2Allocator::type_t type,
+        std::set<C2Allocator::id_t> *ids) {
@@ -2492,2 +2602,2 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-            if (streamFormat.value == C2BufferData::GRAPHIC ||
-                streamFormat.value == C2BufferData::GRAPHIC_CHUNKS) {
+            if (streamFormat.value == C2BufferData::GRAPHIC
+                    || streamFormat.value == C2BufferData::GRAPHIC_CHUNKS) {
@@ -2500,2 +2610 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-                      " uses default allocator id(%d)",
-                      name.c_str(), type, defaultAllocatorId);
+                        " uses default allocator id(%d)", name.c_str(), type, defaultAllocatorId);
@@ -2509 +2618,2 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-                ids->insert(allocators.m.values, allocators.m.values + allocators.flexCount());
+                ids->insert(allocators.m.values,
+                            allocators.m.values + allocators.flexCount());
@@ -2553,2 +2663,2 @@ static status_t GetCommonAllocatorIds(const std::vector<std::string>& names,
-static status_t CalculateMinMaxUsage(const std::vector<std::string>& names, uint64_t* minUsage,
-                                     uint64_t* maxUsage) {
+static status_t CalculateMinMaxUsage(
+        const std::vector<std::string> &names, uint64_t *minUsage, uint64_t *maxUsage) {
@@ -2589,2 +2699,2 @@ static status_t CalculateMinMaxUsage(const std::vector<std::string>& names, uint
-status_t CCodec::CanFetchLinearBlock(const std::vector<std::string>& names,
-                                     const C2MemoryUsage& usage, bool* isCompatible) {
+status_t CCodec::CanFetchLinearBlock(
+        const std::vector<std::string> &names, const C2MemoryUsage &usage, bool *isCompatible) {
@@ -2633,2 +2743,2 @@ static std::shared_ptr<C2BlockPool> GetPool(C2Allocator::id_t allocId) {
-std::shared_ptr<C2LinearBlock> CCodec::FetchLinearBlock(size_t capacity, const C2MemoryUsage& usage,
-                                                        const std::vector<std::string>& names) {
+std::shared_ptr<C2LinearBlock> CCodec::FetchLinearBlock(
+        size_t capacity, const C2MemoryUsage &usage, const std::vector<std::string> &names) {
@@ -2663 +2773,2 @@ std::shared_ptr<C2LinearBlock> CCodec::FetchLinearBlock(size_t capacity, const C
-status_t CCodec::CanFetchGraphicBlock(const std::vector<std::string>& names, bool* isCompatible) {
+status_t CCodec::CanFetchGraphicBlock(
+        const std::vector<std::string> &names, bool *isCompatible) {
@@ -2676,2 +2787,5 @@ status_t CCodec::CanFetchGraphicBlock(const std::vector<std::string>& names, boo
-std::shared_ptr<C2GraphicBlock> CCodec::FetchGraphicBlock(int32_t width, int32_t height,
-                                                          int32_t format, uint64_t usage,
+std::shared_ptr<C2GraphicBlock> CCodec::FetchGraphicBlock(
+        int32_t width,
+        int32_t height,
+        int32_t format,
+        uint64_t usage,
@@ -2704,2 +2818,2 @@ std::shared_ptr<C2GraphicBlock> CCodec::FetchGraphicBlock(int32_t width, int32_t
-        err = pool->fetchGraphicBlock(width, height, halPixelFormat, C2MemoryUsage{minUsage},
-                                      &block);
+        err = pool->fetchGraphicBlock(
+                width, height, halPixelFormat, C2MemoryUsage{minUsage}, &block);
