--- a/./output/tmux/7b189f1a-253f1395-42272dfb/cmd.no_comments_mergegen.c
+++ b/./output/tmux/7b189f1a-253f1395-42272dfb/cmd.no_comments_merged.c
@@ -965,4 +965 @@ cmd_get_default_path(struct cmd_ctx *ctx, const char *cwd)
-out = osdep_get_cwd(s->curw->window->active->fd);
-=======
-   root = osdep_get_cwd(s->curw->window->active->pid);
->>>>>>> 253f1395
+   root = osdep_get_cwd(s->curw->window->active->fd);
