--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7b189f1a-253f1395-42272dfb/cmd.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7b189f1a-253f1395-42272dfb/cmd.no_comments_merged.c
@@ -4 +3,0 @@
-#include <paths.h>
@@ -816 +964,0 @@ const char *cmd_get_default_path(struct cmd_ctx *ctx, const char *cwd) {
-<<<<<<< HEAD
@@ -818,5 +965,0 @@ const char *cmd_get_default_path(struct cmd_ctx *ctx, const char *cwd) {
-|||||||
-      root = get_proc_cwd(s->curw->window->active->fd);
-=======
-      root = osdep_get_cwd(s->curw->window->active->pid);
->>>>>>> 253f1395a03b7b3371799055a7e3a442a8fa7ba6
