diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_merged.c
index 561743d..8988cdf 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/7b189f1a-253f1395-42272dfb/format.no_comments_merged.c
@@ -9,3 +9,6 @@
-int format_replace(struct format_tree *, const char *, size_t, char **,
-                   size_t *, size_t *);
-int format_cmp(struct format_entry *fe1, struct format_entry *fe2) {
+int format_replace(struct format_tree *,
+     const char *, size_t, char **, size_t *, size_t *);
+RB_GENERATE(format_tree, format_entry, entry, format_cmp);
+int
+format_cmp(struct format_entry *fe1, struct format_entry *fe2)
+{
@@ -42 +45,3 @@ const char *format_aliases[26] = {
-struct format_tree *format_create(void) {
+struct format_tree *
+format_create(void)
+{
@@ -47 +52,2 @@ struct format_tree *format_create(void) {
-  if (gethostname(host, sizeof host) == 0) format_add(ft, "host", "%s", host);
+ if (gethostname(host, sizeof host) == 0)
+  format_add(ft, "host", "%s", host);
@@ -50 +56,3 @@ struct format_tree *format_create(void) {
-void format_free(struct format_tree *ft) {
+void
+format_free(struct format_tree *ft)
+{
@@ -63 +71,3 @@ void format_free(struct format_tree *ft) {
-void format_add(struct format_tree *ft, const char *key, const char *fmt, ...) {
+void
+format_add(struct format_tree *ft, const char *key, const char *fmt, ...)
+{
@@ -73 +83,3 @@ void format_add(struct format_tree *ft, const char *key, const char *fmt, ...) {
-const char *format_find(struct format_tree *ft, const char *key) {
+const char *
+format_find(struct format_tree *ft, const char *key)
+{
@@ -77 +89,2 @@ const char *format_find(struct format_tree *ft, const char *key) {
-  if (fe == NULL) return (NULL);
+ if (fe == NULL)
+  return (NULL);
@@ -80,2 +93,4 @@ const char *format_find(struct format_tree *ft, const char *key) {
-int format_replace(struct format_tree *ft, const char *key, size_t keylen,
-                   char **buf, size_t *len, size_t *off) {
+int
+format_replace(struct format_tree *ft,
+    const char *key, size_t keylen, char **buf, size_t *len, size_t *off)
+{
@@ -90 +105,2 @@ int format_replace(struct format_tree *ft, const char *key, size_t keylen,
-    if (ptr == NULL) goto fail;
+  if (ptr == NULL)
+   goto fail;
@@ -96 +112,2 @@ int format_replace(struct format_tree *ft, const char *key, size_t keylen,
-      if (ptr == NULL) goto fail;
+   if (ptr == NULL)
+    goto fail;
@@ -100 +117,2 @@ int format_replace(struct format_tree *ft, const char *key, size_t keylen,
-      if (ptr == NULL) goto fail;
+   if (ptr == NULL)
+    goto fail;
@@ -105 +123,2 @@ int format_replace(struct format_tree *ft, const char *key, size_t keylen,
-    if (value == NULL) value = "";
+  if (value == NULL)
+   value = "";
@@ -120 +139,3 @@ fail:
-char *format_expand(struct format_tree *ft, const char *fmt) {
+char *
+format_expand(struct format_tree *ft, const char *fmt)
+{
@@ -142 +163,2 @@ char *format_expand(struct format_tree *ft, const char *fmt) {
-        if (ptr == NULL) break;
+   if (ptr == NULL)
+    break;
@@ -144 +166,2 @@ char *format_expand(struct format_tree *ft, const char *fmt) {
-        if (format_replace(ft, fmt, n, &buf, &len, &off) != 0) break;
+   if (format_replace(ft, fmt, n, &buf, &len, &off) != 0)
+    break;
@@ -152 +175,3 @@ char *format_expand(struct format_tree *ft, const char *fmt) {
-            if (format_replace(ft, s, n, &buf, &len, &off) != 0) break;
+     if (format_replace (
+         ft, s, n, &buf, &len, &off) != 0)
+      break;
@@ -168 +193,3 @@ char *format_expand(struct format_tree *ft, const char *fmt) {
-void format_session(struct format_tree *ft, struct session *s) {
+void
+format_session(struct format_tree *ft, struct session *s)
+{
@@ -190 +217,3 @@ void format_session(struct format_tree *ft, struct session *s) {
-void format_client(struct format_tree *ft, struct client *c) {
+void
+format_client(struct format_tree *ft, struct client *c)
+{
@@ -217,2 +246,3 @@ void format_client(struct format_tree *ft, struct client *c) {
-void format_winlink(struct format_tree *ft, struct session *s,
-                    struct winlink *wl) {
+void
+format_winlink(struct format_tree *ft, struct session *s, struct winlink *wl)
+{
@@ -235 +265,3 @@ void format_winlink(struct format_tree *ft, struct session *s,
-void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
+void
+format_window_pane(struct format_tree *ft, struct window_pane *wp)
+{
@@ -248 +280,2 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
-  if (window_pane_index(wp, &idx) != 0) fatalx("index not found");
+ if (window_pane_index(wp, &idx) != 0)
+  fatalx("index not found");
@@ -259,3 +292,4 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
-  if (wp->cmd != NULL) format_add(ft, "pane_start_command", "%s", wp->cmd);
-  if (wp->cwd != NULL) format_add(ft, "pane_start_path", "%s", wp->cwd);
-<<<<<<< HEAD
+ if (wp->cmd != NULL)
+  format_add(ft, "pane_start_command", "%s", wp->cmd);
+ if (wp->cwd != NULL)
+  format_add(ft, "pane_start_path", "%s", wp->cwd);
@@ -263,5 +296,0 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
-|||||||
-  format_add(ft, "pane_current_path", "%s", get_proc_cwd(wp->fd));
-=======
-  format_add(ft, "pane_current_path", "%s", osdep_get_cwd(wp->pid));
->>>>>>> 253f1395a03b7b3371799055a7e3a442a8fa7ba6
@@ -271 +300,3 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
-void format_paste_buffer(struct format_tree *ft, struct paste_buffer *pb) {
+void
+format_paste_buffer(struct format_tree *ft, struct paste_buffer *pb)
+{
