diff --git a/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_conflict.h b/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
index d5f19cd..df48b44 100644
--- a/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_conflict.h
+++ b/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -32 +31,0 @@ public:
-<<<<<<< HEAD
@@ -35,7 +33,0 @@ public:
-||||||| 5101ebae20
-    void setAngleInfo(const std::string& path, const std::string& packageName,
-                      const std::string& devOptIn, const std::vector<std::string> eglFeatures);
-=======
-    void setAngleInfo(const std::string& path, const bool useSystemAngle,
-                      const std::string& packageName, const std::vector<std::string> eglFeatures);
->>>>>>> 58e0e142
@@ -69 +60,0 @@ private:
-<<<<<<< HEAD
@@ -73,6 +63,0 @@ private:
-||||||| 5101ebae20
-    UseAngle mShouldUseAngle = UNKNOWN;
-=======
-    bool mShouldUseAngle = false;
-    bool mUseSystemAngle = false;
->>>>>>> 58e0e142
