--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -7 +7 @@
-struct android_namespace_t
+struct android_namespace_t;
@@ -33 +31,0 @@ struct android_namespace_t
-<<<<<<< HEAD
@@ -35,10 +33 @@ struct android_namespace_t
-                          const std::string& packageName,
-                          const std::vector<std::string> eglFeatures);
-|||||||
-        void setAngleInfo(const std::string& path, const std::string& packageName,
-                          const std::string& devOptIn, const std::vector<std::string> eglFeatures);
-=======
-        void setAngleInfo(const std::string& path, const bool useSystemAngle,
-                          const std::string& packageName,
-                          const std::vector<std::string> eglFeatures);
->>>>>>> 58e0e1422d717b7317d6ef18f2508f1c7a01d214
+                      const std::string& packageName, const std::vector<std::string> eglFeatures);
@@ -63,3 +52,2 @@ struct android_namespace_t
-        void sendGpuStatsLocked(GpuStatsInfo::Api api, bool isDriverLoaded,
-                                int64_t driverLoadingTime);
-        GraphicsEnv()
+    void sendGpuStatsLocked(GpuStatsInfo::Api api, bool isDriverLoaded, int64_t driverLoadingTime);
+    GraphicsEnv() = default;
@@ -76 +63,0 @@ struct android_namespace_t
-        bool mUseSystemAngle = false;
