--- a/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergegen.h
+++ b/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -31,0 +32,2 @@ public:
+    void setAngleInfo(const std::string& path, const bool shouldUseNativeDriver,
+                      const std::string& packageName, const std::vector<std::string> eglFeatures);
@@ -58,0 +61,3 @@ private:
+    bool mShouldUseAngle = false;
+    bool mShouldUseSystemAngle = false;
+    bool mShouldUseNativeDriver = false;
