diff --git a/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@GraphicsEnv.no_comments_mergegen.cpp b/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
index 9912ae6..4a8d3c1 100644
--- a/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@GraphicsEnv.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/53e36f03-58e0e142-5101ebae/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
@@ -401,0 +402,2 @@ bool GraphicsEnv::shouldUseAngle() {
+void GraphicsEnv::setAngleInfo(const std::string& path, const bool shouldUseNativeDriver,
+                               const std::string& packageName,
@@ -411 +413,8 @@ bool GraphicsEnv::shouldUseAngle() {
-mPackageName = std::move(packageName); if (mAnglePath == "system") { mShouldUseSystemAngle = true; } if (!mAnglePath.empty()) { mShouldUseAngle = true; } mShouldUseNativeDriver = shouldUseNativeDriver;
+    mPackageName = std::move(packageName);
+    if (mAnglePath == "system") {
+        mShouldUseSystemAngle = true;
+    }
+    if (!mAnglePath.empty()) {
+        mShouldUseAngle = true;
+    }
+    mShouldUseNativeDriver = shouldUseNativeDriver;
