--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/071849f8-71d453f1-8636848e/input.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/071849f8-71d453f1-8636848e/input.no_comments_merged.c
@@ -207,160 +240,19 @@ struct input_state {
-      {0x1a, 0x1a, input_c0_dispatch, &input_state_ground}, { \
-    0x1b, 0x1b, NULL, &input_state_esc_enter \
-  }
-static const struct input_transition input_state_ground_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x7e, input_print, NULL},
-    {0x7f, 0x7f, NULL, NULL},
-    {0x80, 0xff, input_top_bit_set, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_esc_enter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_esc_intermediate},
-    {0x30, 0x4f, input_esc_dispatch, &input_state_ground},
-    {0x50, 0x50, NULL, &input_state_dcs_enter},
-    {0x51, 0x57, input_esc_dispatch, &input_state_ground},
-    {0x58, 0x58, NULL, &input_state_consume_st},
-    {0x59, 0x59, input_esc_dispatch, &input_state_ground},
-    {0x5a, 0x5a, input_esc_dispatch, &input_state_ground},
-    {0x5b, 0x5b, NULL, &input_state_csi_enter},
-    {0x5c, 0x5c, input_esc_dispatch, &input_state_ground},
-    {0x5d, 0x5d, NULL, &input_state_osc_string},
-    {0x5e, 0x5e, NULL, &input_state_consume_st},
-    {0x5f, 0x5f, NULL, &input_state_apc_string},
-    {0x60, 0x6a, input_esc_dispatch, &input_state_ground},
-    {0x6b, 0x6b, NULL, &input_state_rename_string},
-    {0x6c, 0x7e, input_esc_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_esc_intermediate_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, NULL},
-    {0x30, 0x7e, input_esc_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_enter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_csi_intermediate},
-    {0x30, 0x39, input_parameter, &input_state_csi_parameter},
-    {0x3a, 0x3a, input_parameter, &input_state_csi_parameter},
-    {0x3b, 0x3b, input_parameter, &input_state_csi_parameter},
-    {0x3c, 0x3f, input_intermediate, &input_state_csi_parameter},
-    {0x40, 0x7e, input_csi_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_parameter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_csi_intermediate},
-    {0x30, 0x39, input_parameter, NULL},
-    {0x3a, 0x3a, input_parameter, NULL},
-    {0x3b, 0x3b, input_parameter, NULL},
-    {0x3c, 0x3f, NULL, &input_state_csi_ignore},
-    {0x40, 0x7e, input_csi_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_intermediate_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x2f, input_intermediate, NULL},
-    {0x30, 0x3f, NULL, &input_state_csi_ignore},
-    {0x40, 0x7e, input_csi_dispatch, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_csi_ignore_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, input_c0_dispatch, NULL},
-    {0x19, 0x19, input_c0_dispatch, NULL},
-    {0x1c, 0x1f, input_c0_dispatch, NULL},
-    {0x20, 0x3f, NULL, NULL},
-    {0x40, 0x7e, NULL, &input_state_ground},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_enter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL},
-    {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_dcs_intermediate},
-    {0x30, 0x39, input_parameter, &input_state_dcs_parameter},
-    {0x3a, 0x3a, NULL, &input_state_dcs_ignore},
-    {0x3b, 0x3b, input_parameter, &input_state_dcs_parameter},
-    {0x3c, 0x3f, input_intermediate, &input_state_dcs_parameter},
-    {0x40, 0x7e, input_input, &input_state_dcs_handler},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_parameter_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL},
-    {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL},
-    {0x20, 0x2f, input_intermediate, &input_state_dcs_intermediate},
-    {0x30, 0x39, input_parameter, NULL},
-    {0x3a, 0x3a, NULL, &input_state_dcs_ignore},
-    {0x3b, 0x3b, input_parameter, NULL},
-    {0x3c, 0x3f, NULL, &input_state_dcs_ignore},
-    {0x40, 0x7e, input_input, &input_state_dcs_handler},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_intermediate_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL},
-    {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL},
-    {0x20, 0x2f, input_intermediate, NULL},
-    {0x30, 0x3f, NULL, &input_state_dcs_ignore},
-    {0x40, 0x7e, input_input, &input_state_dcs_handler},
-    {0x7f, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_handler_table[] = {
-    {0x00, 0x1a, input_input, NULL},
-    {0x1b, 0x1b, NULL, &input_state_dcs_escape},
-    {0x1c, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_escape_table[] = {
-    {0x00, 0x5b, input_input, &input_state_dcs_handler},
-    {0x5c, 0x5c, input_dcs_dispatch, &input_state_ground},
-    {0x5d, 0xff, input_input, &input_state_dcs_handler},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_dcs_ignore_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_osc_string_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x06, NULL, NULL}, {0x07, 0x07, input_end_bel, &input_state_ground},
-    {0x08, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_apc_string_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_rename_string_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, input_input, NULL},
-    {-1, -1, NULL, NULL}};
-static const struct input_transition input_state_consume_st_table[] = {
-    INPUT_STATE_ANYWHERE,
-    {0x00, 0x17, NULL, NULL}, {0x19, 0x19, NULL, NULL},
-    {0x1c, 0x1f, NULL, NULL}, {0x20, 0xff, NULL, NULL},
-    {-1, -1, NULL, NULL}};
+ { 0x1a, 0x1a, input_c0_dispatch, &input_state_ground }, \
+ { 0x1b, 0x1b, NULL, &input_state_esc_enter }
+static const struct input_transition input_state_ground_table[];
+static const struct input_transition input_state_esc_enter_table[];
+static const struct input_transition input_state_esc_intermediate_table[];
+static const struct input_transition input_state_csi_enter_table[];
+static const struct input_transition input_state_csi_parameter_table[];
+static const struct input_transition input_state_csi_intermediate_table[];
+static const struct input_transition input_state_csi_ignore_table[];
+static const struct input_transition input_state_dcs_enter_table[];
+static const struct input_transition input_state_dcs_parameter_table[];
+static const struct input_transition input_state_dcs_intermediate_table[];
+static const struct input_transition input_state_dcs_handler_table[];
+static const struct input_transition input_state_dcs_escape_table[];
+static const struct input_transition input_state_dcs_ignore_table[];
+static const struct input_transition input_state_osc_string_table[];
+static const struct input_transition input_state_apc_string_table[];
+static const struct input_transition input_state_rename_string_table[];
+static const struct input_transition input_state_consume_st_table[];
@@ -1734 +1852,0 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-<<<<<<< HEAD
@@ -1742,8 +1861,2 @@ static int input_dcs_dispatch(struct input_ctx *ictx) {
-|||||||
-    allow_passthrough = options_get_number(wp->options, "allow-passthrough");
-=======
-  }
-  log_debug("%s: %zu bytes", __func__, len);
-  allow_passthrough = options_get_number(wp->options, "allow-passthrough");
->>>>>>> 71d453f169ffac345db3e4c82f4409fc41091002
-    if (!allow_passthrough) return (0);
+ if (!allow_passthrough)
+  return (0);
