diff --git a/./output/tmux/071849f8-71d453f1-8636848e/input.no_comments_conflict.c b/./output/tmux/071849f8-71d453f1-8636848e/input.no_comments_merged.c
index c877ab3..9742c44 100644
--- a/./output/tmux/071849f8-71d453f1-8636848e/input.no_comments_conflict.c
+++ b/./output/tmux/071849f8-71d453f1-8636848e/input.no_comments_merged.c
@@ -1853 +1852,0 @@ input_dcs_dispatch(struct input_ctx *ictx)
-<<<<<<< HEAD
@@ -1862,8 +1860,0 @@ input_dcs_dispatch(struct input_ctx *ictx)
-||||||| 8636848e
- allow_passthrough = options_get_number(wp->options,
-     "allow-passthrough");
-=======
- }
- log_debug("%s: %zu bytes", __func__, len);
- allow_passthrough = options_get_number(wp->options, "allow-passthrough");
->>>>>>> 71d453f1
