diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_merged.cpp
index 85dc5f3..0bc1f1c 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/07e1df1c-d4e7c63c-d40df83d/src@capture@capturewidget.no_comments_merged.cpp
@@ -5,2 +4,0 @@
-#include "button.h"
-#include <src/utils/confighandler.h>
@@ -9 +7,2 @@
-#include <QGuiApplication>
+#include "src/utils/confighandler.h"
+#include <QScreen>
@@ -10,0 +10 @@
+#include <QGuiApplication>
@@ -18 +17,0 @@
-#include <QSettings>
@@ -21 +19,0 @@
-#include <QDebug>
@@ -26,10 +24,6 @@ const int HANDLE_SIZE = 9;
-CaptureWidget::CaptureWidget(bool enableSaveWindow, QWidget *parent)
-    : QWidget(parent),
-      m_mouseOverHandle(0),
-      m_mouseIsClicked(false),
-      m_rightClick(false),
-      m_newSelection(false),
-      m_grabbing(false),
-      m_onButton(false),
-      m_enableSaveWindow(enableSaveWindow),
-      m_state(CaptureButton::TYPE_MOVESELECTION) {
+CaptureWidget::CaptureWidget(bool enableSaveWindow, QWidget *parent) :
+    QWidget(parent), m_mouseOverHandle(0), m_mouseIsClicked(false),
+    m_rightClick(false), m_newSelection(false), m_grabbing(false),
+    m_onButton(false), m_enableSaveWindow(enableSaveWindow),
+    m_state(CaptureButton::TYPE_MOVESELECTION)
+{
@@ -39,8 +33,4 @@ CaptureWidget::CaptureWidget(bool enableSaveWindow, QWidget *parent)
-  m_TLHandle = baseRect;
-  m_TRHandle = baseRect;
-  m_BLHandle = baseRect;
-  m_BRHandle = baseRect;
-  m_LHandle = baseRect;
-  m_THandle = baseRect;
-  m_RHandle = baseRect;
-  m_BHandle = baseRect;
+    m_TLHandle = baseRect; m_TRHandle = baseRect;
+    m_BLHandle = baseRect; m_BRHandle = baseRect;
+    m_LHandle = baseRect; m_THandle = baseRect;
+    m_RHandle = baseRect; m_BHandle = baseRect;
@@ -49,2 +39,4 @@ CaptureWidget::CaptureWidget(bool enableSaveWindow, QWidget *parent)
-  setWindowFlags(Qt::BypassWindowManagerHint | Qt::WindowStaysOnTopHint |
-                 Qt::FramelessWindowHint | Qt::Tool);
+    setWindowFlags(Qt::BypassWindowManagerHint
+                   | Qt::WindowStaysOnTopHint
+                   | Qt::FramelessWindowHint
+                   | Qt::Tool);
@@ -64 +56,2 @@ CaptureWidget::CaptureWidget(bool enableSaveWindow, QWidget *parent)
-CaptureWidget::~CaptureWidget() {}
+CaptureWidget::~CaptureWidget() {
+}
@@ -80,2 +73,2 @@ void CaptureWidget::updateButtons() {
-    connect(b->getTool(), &CaptureTool::requestAction, this,
-            &CaptureWidget::handleButtonSignal);
+        connect(b->getTool(), &CaptureTool::requestAction,
+                this, &CaptureWidget::handleButtonSignal);
@@ -93,2 +86,2 @@ void CaptureWidget::paintEvent(QPaintEvent *) {
-    painter.drawPixmap(
-        0, 0, m_screenshot->paintTemporalModification(m_modifications.last()));
+        painter.drawPixmap(0, 0, m_screenshot->paintTemporalModification(
+                               m_modifications.last()));
@@ -108,2 +101 @@ void CaptureWidget::paintEvent(QPaintEvent *) {
-    QString helpTxt =
-        tr("Select an area with the mouse, or press Esc to exit."
+        QString helpTxt = tr("Select an area with the mouse, or press Esc to exit."
@@ -114,2 +106,2 @@ void CaptureWidget::paintEvent(QPaintEvent *) {
-    QColor textColor(
-        (CaptureButton::iconIsWhiteByColor(rectColor) ? Qt::white : Qt::black));
+        QColor textColor((CaptureButton::iconIsWhiteByColor(rectColor) ?
+                              Qt::white : Qt::black));
@@ -151,2 +143,5 @@ void CaptureWidget::mousePressEvent(QMouseEvent *e) {
-      m_modifications.append(new CaptureModification(
-          m_state, e->pos(), m_colorPicker->getDrawColor(), this));
+            m_modifications.append(
+                        new CaptureModification(m_state, e->pos(),
+                                                m_colorPicker->getDrawColor(),
+                                                this)
+                        );
@@ -172,3 +167,2 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      m_selection =
-          QRect(m_dragStartPoint, limitPointToRect(m_mousePos, rect()))
-              .normalized();
+            m_selection = QRect(m_dragStartPoint, limitPointToRect(
+                                    m_mousePos, rect())).normalized();
@@ -186,3 +180,2 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      if (m_mouseOverHandle == &m_TLHandle || m_mouseOverHandle == &m_THandle ||
-          m_mouseOverHandle ==
-              &m_TRHandle) {
+            if (m_mouseOverHandle == &m_TLHandle || m_mouseOverHandle == &m_THandle
+                    || m_mouseOverHandle == &m_TRHandle) {
@@ -194,3 +187,2 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      if (m_mouseOverHandle == &m_TLHandle || m_mouseOverHandle == &m_LHandle ||
-          m_mouseOverHandle ==
-              &m_BLHandle) {
+            if (m_mouseOverHandle == &m_TLHandle || m_mouseOverHandle == &m_LHandle
+                    || m_mouseOverHandle == &m_BLHandle) {
@@ -202,3 +194,2 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      if (m_mouseOverHandle == &m_BLHandle || m_mouseOverHandle == &m_BHandle ||
-          m_mouseOverHandle ==
-              &m_BRHandle) {
+            if (m_mouseOverHandle == &m_BLHandle || m_mouseOverHandle == &m_BHandle
+                    || m_mouseOverHandle == &m_BRHandle) {
@@ -210,3 +201,2 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      if (m_mouseOverHandle == &m_TRHandle || m_mouseOverHandle == &m_RHandle ||
-          m_mouseOverHandle ==
-              &m_BRHandle) {
+            if (m_mouseOverHandle == &m_TRHandle || m_mouseOverHandle == &m_RHandle
+                    || m_mouseOverHandle == &m_BRHandle) {
@@ -252,2 +242 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      if (m_mouseOverHandle == &m_TLHandle ||
-          m_mouseOverHandle == &m_BRHandle) {
+            if (m_mouseOverHandle == &m_TLHandle || m_mouseOverHandle == &m_BRHandle) {
@@ -255,2 +244 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      } else if (m_mouseOverHandle == &m_TRHandle ||
-                 m_mouseOverHandle == &m_BLHandle) {
+            } else if (m_mouseOverHandle == &m_TRHandle || m_mouseOverHandle == &m_BLHandle) {
@@ -258,2 +246 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      } else if (m_mouseOverHandle == &m_LHandle ||
-                 m_mouseOverHandle == &m_RHandle) {
+            } else if (m_mouseOverHandle == &m_LHandle || m_mouseOverHandle == &m_RHandle) {
@@ -261,2 +248 @@ void CaptureWidget::mouseMoveEvent(QMouseEvent *e) {
-      } else if (m_mouseOverHandle == &m_THandle ||
-                 m_mouseOverHandle == &m_BHandle) {
+            } else if (m_mouseOverHandle == &m_THandle || m_mouseOverHandle == &m_BHandle) {
@@ -297 +283,2 @@ void CaptureWidget::keyPressEvent(QKeyEvent *e) {
-  } else if (e->key() == Qt::Key_Up && m_selection.top() > rect().top()) {
+    } else if (e->key() == Qt::Key_Up
+               && m_selection.top() > rect().top()) {
@@ -299,2 +286,2 @@ void CaptureWidget::keyPressEvent(QKeyEvent *e) {
-  } else if (e->key() == Qt::Key_Down &&
-             m_selection.bottom() < rect().bottom()) {
+    } else if (e->key() == Qt::Key_Down
+               && m_selection.bottom() < rect().bottom()) {
@@ -302 +289,2 @@ void CaptureWidget::keyPressEvent(QKeyEvent *e) {
-  } else if (e->key() == Qt::Key_Left && m_selection.left() > rect().left()) {
+    } else if (e->key() == Qt::Key_Left
+               && m_selection.left() > rect().left()) {
@@ -304,2 +292,2 @@ void CaptureWidget::keyPressEvent(QKeyEvent *e) {
-  } else if (e->key() == Qt::Key_Right &&
-             m_selection.right() < rect().right()) {
+    } else if (e->key() == Qt::Key_Right
+               && m_selection.right() < rect().right()) {
@@ -349,2 +337,2 @@ void CaptureWidget::copyScreenshot() {
-    QApplication::clipboard()->setPixmap(
-        m_screenshot->getScreenshot().copy(getExtendedSelection()));
+        QApplication::clipboard()->setPixmap(m_screenshot->getScreenshot()
+                                             .copy(getExtendedSelection()));
@@ -362 +350,2 @@ void CaptureWidget::openURL(QNetworkReply *reply) {
-          new QMessageBox(QMessageBox::Warning, QObject::tr("Resource Error"),
+                    new QMessageBox(QMessageBox::Warning,
+                                    QObject::tr("Resource Error"),
@@ -370,2 +359,3 @@ void CaptureWidget::openURL(QNetworkReply *reply) {
-    QMessageBox *netErrBox = new QMessageBox(
-        QMessageBox::Warning, "Network Error", reply->errorString());
+        QMessageBox *netErrBox =
+                new QMessageBox(QMessageBox::Warning, "Network Error",
+                                reply->errorString());
@@ -381 +371,2 @@ void CaptureWidget::uploadScreenshot() {
-  connect(am, &QNetworkAccessManager::finished, this, &CaptureWidget::openURL);
+    connect(am, &QNetworkAccessManager::finished, this,
+            &CaptureWidget::openURL);
@@ -451,2 +442 @@ void CaptureWidget::handleButtonSignal(CaptureTool::Request r) {
-      while (undo())
-        ;
+        while(undo());
@@ -466,2 +456,3 @@ void CaptureWidget::handleButtonSignal(CaptureTool::Request r) {
-      m_enableSaveWindow ? saveScreenshot()
-                         : saveScreenshot(ConfigHandler().getSavePath());
+        m_enableSaveWindow ?
+                    saveScreenshot() :
+                    saveScreenshot(ConfigHandler().getSavePath());
@@ -493,2 +484,6 @@ void CaptureWidget::handleButtonSignal(CaptureTool::Request r) {
-void CaptureWidget::leaveButton() { m_onButton = false; }
-void CaptureWidget::enterButton() { m_onButton = true; }
+void CaptureWidget::leaveButton() {
+    m_onButton = false;
+}
+void CaptureWidget::enterButton() {
+    m_onButton = true;
+}
@@ -497,4 +492,2 @@ void CaptureWidget::initShortcuts() {
-  new QShortcut(QKeySequence(Qt::CTRL + Qt::Key_S), this,
-                SLOT(saveScreenshot()));
-  new QShortcut(QKeySequence(Qt::CTRL + Qt::Key_C), this,
-                SLOT(copyScreenshot()));
+    new QShortcut(QKeySequence(Qt::CTRL + Qt::Key_S), this, SLOT(saveScreenshot()));
+    new QShortcut(QKeySequence(Qt::CTRL + Qt::Key_C), this, SLOT(copyScreenshot()));
@@ -502,4 +495,2 @@ void CaptureWidget::initShortcuts() {
-  new QShortcut(QKeySequence(Qt::SHIFT + Qt::Key_Right), this,
-                SLOT(rightResize()));
-  new QShortcut(QKeySequence(Qt::SHIFT + Qt::Key_Left), this,
-                SLOT(leftResize()));
+    new QShortcut(QKeySequence(Qt::SHIFT + Qt::Key_Right), this, SLOT(rightResize()));
+    new QShortcut(QKeySequence(Qt::SHIFT + Qt::Key_Left), this, SLOT(leftResize()));
@@ -507,2 +498 @@ void CaptureWidget::initShortcuts() {
-  new QShortcut(QKeySequence(Qt::SHIFT + Qt::Key_Down), this,
-                SLOT(downResize()));
+    new QShortcut(QKeySequence(Qt::SHIFT + Qt::Key_Down), this, SLOT(downResize()));
