--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@aggregate@physical_window.no_comments_merged.cpp
@@ -4,0 +5 @@
+#include "duckdb/common/radix_partitioning.hpp"
@@ -6,0 +8 @@
+#include "duckdb/common/types/chunk_collection.hpp"
@@ -9,3 +10,0 @@
-#include "duckdb/common/types/chunk_collection.hpp"
-#include "duckdb/common/types/column_data_consumer.hpp"
-#include "duckdb/common/types/row_data_collection_scanner.hpp"
@@ -22 +20,0 @@
-#include "duckdb/common/radix_partitioning.hpp"
@@ -874,0 +873 @@ public:
+public:
@@ -1046 +1043,0 @@ SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &c
- auto &lpsource = lsource.partition_source;
@@ -1048 +1044,0 @@ SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &c
- auto &gpsource = gsource.partition_source;
@@ -1053,6 +1049,6 @@ SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &c
-  while (!lpsource.scanner || !lpsource.scanner->Remaining()) {
-   lpsource.scanner.reset();
-   lpsource.rows.reset();
-   lpsource.heap.reset();
-   lpsource.hash_group.reset();
-   auto hash_bin = gpsource.next_bin++;
+  while (!lsource.scanner || !lsource.scanner->Remaining()) {
+   lsource.scanner.reset();
+   lsource.rows.reset();
+   lsource.heap.reset();
+   lsource.hash_group.reset();
+   auto hash_bin = gsource.next_bin++;
@@ -1062 +1058 @@ SourceResultType PhysicalWindow::GetData(ExecutionContext &context, DataChunk &c
-   for (; hash_bin < hash_groups.size(); hash_bin = gpsource.next_bin++) {
+   for (; hash_bin < hash_groups.size(); hash_bin = gsource.next_bin++) {
