diff --git a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_mergegen.hpp b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_merged.hpp
index 5db5ecc..e481974 100644
--- a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_mergegen.hpp
+++ b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@include@duckdb@main@client_config.no_comments_merged.hpp
@@ -35,0 +36,2 @@ struct ClientConfig {
+ bool force_asof_iejoin = false;
+ bool force_async_pipelines = true;
