diff --git a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_mergegen.cpp b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_merged.cpp
index d7c82db..86f2d7e 100644
--- a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_mergegen.cpp
+++ b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_merged.cpp
@@ -22 +22,2 @@ unique_ptr<GlobalSourceState> PhysicalColumnDataScan::GetGlobalSourceState(Clien
-jResultType PhysicalColumnDataScan::GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const { auto &state = input.global_state.Cast<PhysicalColumnDataScanState>();
+SourceResultType PhysicalColumnDataScan::GetData(ExecutionContext &context, DataChunk &chunk, OperatorSourceInput &input) const {
+ auto &state = input.global_state.Cast<PhysicalColumnDataScanState>();
