diff --git a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_conflict.cpp b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_merged.cpp
index a016fcc..86f2d7e 100644
--- a/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_conflict.cpp
+++ b/./output/duckdb/8e5b98eb-65cfcd58-e8610c85/src@execution@operator@scan@physical_column_data_scan.no_comments_merged.cpp
@@ -22 +21,0 @@ unique_ptr<GlobalSourceState> PhysicalColumnDataScan::GetGlobalSourceState(Clien
-<<<<<<< HEAD
@@ -25,11 +23,0 @@ SourceResultType PhysicalColumnDataScan::GetData(ExecutionContext &context, Data
- D_ASSERT(collection);
-||||||| e8610c85fb
-void PhysicalColumnDataScan::GetData(ExecutionContext &context, DataChunk &chunk, GlobalSourceState &gstate,
-                                     LocalSourceState &lstate) const {
- auto &state = gstate.Cast<PhysicalColumnDataScanState>();
- D_ASSERT(collection);
-=======
-void PhysicalColumnDataScan::GetData(ExecutionContext &context, DataChunk &chunk, GlobalSourceState &gstate,
-                                     LocalSourceState &lstate) const {
- auto &state = gstate.Cast<PhysicalColumnDataScanState>();
->>>>>>> 65cfcd58
