diff --git a/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_conflict.cpp b/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_merged.cpp
index 91b9867..3a14e6f 100644
--- a/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_conflict.cpp
+++ b/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_merged.cpp
@@ -5,0 +6 @@
+#include "duckdb/common/http_stats.hpp"
@@ -7 +8 @@
-#include "duckdb/common/progress_bar.hpp"
+#include "duckdb/common/progress_bar/progress_bar.hpp"
@@ -15,0 +17 @@
+#include "duckdb/main/attached_database.hpp"
@@ -18,0 +21 @@
+#include "duckdb/main/database_manager.hpp"
@@ -42 +44,0 @@
-<<<<<<< HEAD
@@ -43,0 +46 @@
+#include "duckdb/transaction/meta_transaction.hpp"
@@ -46,29 +48,0 @@
-||||||| merged common ancestors
-#include "duckdb/common/file_system.hpp"
-#include "duckdb/execution/column_binding_resolver.hpp"
-#include "duckdb/execution/operator/helper/physical_result_collector.hpp"
-#include "duckdb/parser/query_node/select_node.hpp"
-#include "duckdb/parser/parsed_expression_iterator.hpp"
-#include "duckdb/parser/statement/prepare_statement.hpp"
-#include "duckdb/parser/statement/execute_statement.hpp"
-#include "duckdb/common/types/column_data_collection.hpp"
-#include "duckdb/common/preserved_error.hpp"
-#include "duckdb/common/progress_bar.hpp"
-#include "duckdb/main/error_manager.hpp"
-=======
-#include "duckdb/common/file_system.hpp"
-#include "duckdb/execution/column_binding_resolver.hpp"
-#include "duckdb/execution/operator/helper/physical_result_collector.hpp"
-#include "duckdb/parser/query_node/select_node.hpp"
-#include "duckdb/parser/parsed_expression_iterator.hpp"
-#include "duckdb/parser/statement/prepare_statement.hpp"
-#include "duckdb/parser/statement/execute_statement.hpp"
-#include "duckdb/common/types/column_data_collection.hpp"
-#include "duckdb/common/preserved_error.hpp"
-#include "duckdb/common/progress_bar/progress_bar.hpp"
-#include "duckdb/main/error_manager.hpp"
-#include "duckdb/main/database_manager.hpp"
-#include "duckdb/transaction/meta_transaction.hpp"
-#include "duckdb/common/http_stats.hpp"
-#include "duckdb/main/attached_database.hpp"
->>>>>>> b3f6a8f1
