diff --git a/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_mergegen.cpp b/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_merged.cpp
index 28e1ab8..3a14e6f 100644
--- a/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_mergegen.cpp
+++ b/./output/duckdb/8ba017a5-b3f6a8f1-6f67ea9e/src@main@client_context.no_comments_merged.cpp
@@ -5,0 +6 @@
+#include "duckdb/common/http_stats.hpp"
@@ -7 +8 @@
-#include "duckdb/common/progress_bar.hpp"
+#include "duckdb/common/progress_bar/progress_bar.hpp"
@@ -15,0 +17 @@
+#include "duckdb/main/attached_database.hpp"
@@ -18,0 +21 @@
+#include "duckdb/main/database_manager.hpp"
@@ -42 +45,4 @@
-#include "duckdb/storage/data_table.hpp" #include "duckdb/transaction/transaction.hpp" #include "duckdb/transaction/transaction_manager.hpp" #include "duckdb/main/database_manager.hpp" #include "duckdb/transaction/meta_transaction.hpp" #include "duckdb/common/http_stats.hpp" #include "duckdb/main/attached_database.hpp"
+#include "duckdb/storage/data_table.hpp"
+#include "duckdb/transaction/meta_transaction.hpp"
+#include "duckdb/transaction/transaction.hpp"
+#include "duckdb/transaction/transaction_manager.hpp"
