diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_merged.c
index 61a9eac..3563d0e 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/cmd-split-window.no_comments_merged.c
@@ -3 +2,0 @@
-#include <string.h>
@@ -6,0 +6 @@
+#include <string.h>
@@ -12,7 +12,4 @@ const struct cmd_entry cmd_split_window_entry = {
-    "split-window",
-    "splitw",
-    "c:dF:l:hp:Pt:v",
-    0,
-    1,
-    "[-dhvP] [-c start-directory] [-F format] [-p percentage|-l "
-    "size] " CMD_TARGET_PANE_USAGE " [command]",
+ "split-window", "splitw",
+ "c:dF:l:hp:Pt:v", 0, 1,
+ "[-dhvP] [-c start-directory] [-F format] [-p percentage|-l size] "
+ CMD_TARGET_PANE_USAGE " [command]",
@@ -21,3 +18,5 @@ const struct cmd_entry cmd_split_window_entry = {
-    NULL,
-    cmd_split_window_exec};
-void cmd_split_window_key_binding(struct cmd *self, int key) {
+ cmd_split_window_exec
+};
+void
+cmd_split_window_key_binding(struct cmd *self, int key)
+{
@@ -25 +24,2 @@ void cmd_split_window_key_binding(struct cmd *self, int key) {
-  if (key == '%') args_set(self->args, 'h', NULL);
+ if (key == '%')
+  args_set(self->args, 'h', NULL);
@@ -27 +27,3 @@ void cmd_split_window_key_binding(struct cmd *self, int key) {
-enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
+enum cmd_retval
+cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq)
+{
@@ -56 +58,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL) format_client(ft, c);
+  if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL)
+   format_client(ft, c);
@@ -65 +68,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-      cmdq_error(cmdq, "bad working directory: %s", strerror(errno));
+   cmdq_error(cmdq, "bad working directory: %s",
+       strerror(errno));
@@ -74 +78,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (args_has(args, 'h')) type = LAYOUT_LEFTRIGHT;
+ if (args_has(args, 'h'))
+  type = LAYOUT_LEFTRIGHT;
@@ -99 +104,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (*shell == '\0' || areshell(shell)) shell = _PATH_BSHELL;
+ if (*shell == '\0' || areshell(shell))
+  shell = _PATH_BSHELL;
@@ -105 +111,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (window_pane_spawn(new_wp, cmd, shell, cwd, &env, s->tio, &cause) != 0)
+ if (window_pane_spawn(
+     new_wp, cmd, shell, cwd, &env, s->tio, &cause) != 0)
@@ -120 +127,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL) format_client(ft, c);
+  if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL)
+   format_client(ft, c);
@@ -130 +138,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (fd != -1) close(fd);
+ if (fd != -1)
+  close(fd);
@@ -134 +143,2 @@ error:
-  if (new_wp != NULL) window_remove_pane(w, new_wp);
+ if (new_wp != NULL)
+  window_remove_pane(w, new_wp);
@@ -137 +147,2 @@ error:
-  if (fd != -1) close(fd);
+ if (fd != -1)
+  close(fd);
