--- a/./output/tmux/c190c732-4901d9dd-a9ebb62d/format.no_comments_mergegen.c
+++ b/./output/tmux/c190c732-4901d9dd-a9ebb62d/format.no_comments_merged.c
@@ -434 +434 @@ format_window_pane(struct format_tree *ft, struct window_pane *wp)
-if ((cmd = osdep_get_cwd(wp)) != NULL) format_add(ft, "pane_start_path", "%s", wp->cwd); if ((cwd = get_proc_cwd(wp->fd)) != NULL) format_add(ft, "pane_current_path", "%s", cwd);
+ if ((cmd = format_get_command(wp)) != NULL) {
