diff --git a/./output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_mergegen.c b/./output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_merged.c
index e589388..16a7452 100644
--- a/./output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_mergegen.c
+++ b/./output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_merged.c
@@ -32 +32,3 @@ window_name_callback(unused int fd, unused short events, void *data)
-if (w->active->screen != &w->active->base) name = NULL; else name = osdep_get_name(w->active->fd, w->active->tty); if (name == NULL) wname = default_window_name(w); else { if (w->active->cmd != NULL && *w->active->cmd == '\0' && name = format_window_name(w); if (strcmp(name, w->name) != 0) { window_set_name(w, name);
+ name = format_window_name(w);
+ if (strcmp(name, w->name) != 0) {
+  window_set_name(w, name);
