diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_merged.c
index 6104488..16a7452 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/c190c732-4901d9dd-a9ebb62d/names.no_comments_merged.c
@@ -6 +5,0 @@
-#include <unistd.h>
@@ -9 +8,3 @@ void window_name_callback(unused int, unused short, void *);
-void queue_window_name(struct window *w) {
+void
+queue_window_name(struct window *w)
+{
@@ -13 +14,2 @@ void queue_window_name(struct window *w) {
-  if (event_initialized(&w->name_timer)) evtimer_del(&w->name_timer);
+ if (event_initialized(&w->name_timer))
+  evtimer_del(&w->name_timer);
@@ -17 +19,3 @@ void queue_window_name(struct window *w) {
-void window_name_callback(unused int fd, unused short events, void *data) {
+void
+window_name_callback(unused int fd, unused short events, void *data)
+{
@@ -20 +24,2 @@ void window_name_callback(unused int fd, unused short events, void *data) {
-  if (w->active == NULL) return;
+ if (w->active == NULL)
+  return;
@@ -22 +27,2 @@ void window_name_callback(unused int fd, unused short events, void *data) {
-    if (event_initialized(&w->name_timer)) event_del(&w->name_timer);
+  if (event_initialized(&w->name_timer))
+   event_del(&w->name_timer);
@@ -26,45 +31,0 @@ void window_name_callback(unused int fd, unused short events, void *data) {
-<<<<<<< HEAD
-  if (w->active->screen != &w->active->base)
-    name = NULL;
-  else
-    name = osdep_get_name(w->active->fd, w->active->tty);
-  if (name == NULL)
-    wname = default_window_name(w);
-  else {
-    if (w->active->cmd != NULL && *w->active->cmd == '\0' && name != NULL &&
-        name[0] == '-' && name[1] != '\0')
-      wname = parse_window_name(name + 1);
-    else
-      wname = parse_window_name(name);
-    free(name);
-  }
-  if (w->active->fd == -1) {
-    xasprintf(&name, "%s[dead]", wname);
-    free(wname);
-    wname = name;
-  }
-  if (strcmp(wname, w->name)) {
-    window_set_name(w, wname);
-|||||||
-  if (w->active->screen != &w->active->base)
-    name = NULL;
-  else
-    name = get_proc_name(w->active->fd, w->active->tty);
-  if (name == NULL)
-    wname = default_window_name(w);
-  else {
-    if (w->active->cmd != NULL && *w->active->cmd == '\0' && name != NULL &&
-        name[0] == '-' && name[1] != '\0')
-      wname = parse_window_name(name + 1);
-    else
-      wname = parse_window_name(name);
-    free(name);
-  }
-  if (w->active->fd == -1) {
-    xasprintf(&name, "%s[dead]", wname);
-    free(wname);
-    wname = name;
-  }
-  if (strcmp(wname, w->name)) {
-    window_set_name(w, wname);
-=======
@@ -74 +34,0 @@ void window_name_callback(unused int fd, unused short events, void *data) {
->>>>>>> 4901d9ddc8d8c33ecdca363dcb67e66482745fa5
@@ -79 +39,3 @@ void window_name_callback(unused int fd, unused short events, void *data) {
-char *default_window_name(struct window *w) {
+char *
+default_window_name(struct window *w)
+{
@@ -84 +46,3 @@ char *default_window_name(struct window *w) {
-char *format_window_name(struct window *w) {
+char *
+format_window_name(struct window *w)
+{
@@ -95 +59,3 @@ char *format_window_name(struct window *w) {
-char *parse_window_name(const char *in) {
+char *
+parse_window_name(const char *in)
+{
@@ -100,2 +66,4 @@ char *parse_window_name(const char *in) {
-  while (*name == ' ' || *name == '-') name++;
-  if ((ptr = strchr(name, ' ')) != NULL) *ptr = '\0';
+ while (*name == ' ' || *name == '-')
+  name++;
+ if ((ptr = strchr(name, ' ')) != NULL)
+  *ptr = '\0';
@@ -104 +72,2 @@ char *parse_window_name(const char *in) {
-    while (ptr > name && !isalnum((u_char)*ptr)) *ptr-- = '\0';
+  while (ptr > name && !isalnum((u_char)*ptr))
+   *ptr-- = '\0';
@@ -106 +75,2 @@ char *parse_window_name(const char *in) {
-  if (*name == '/') name = basename(name);
+ if (*name == '/')
+  name = basename(name);
