diff --git a/./output/tmux/c190c732-4901d9dd-a9ebb62d/server-client.no_comments_mergegen.c b/./output/tmux/c190c732-4901d9dd-a9ebb62d/server-client.no_comments_merged.c
index 18ef1cd..1ed321d 100644
--- a/./output/tmux/c190c732-4901d9dd-a9ebb62d/server-client.no_comments_mergegen.c
+++ b/./output/tmux/c190c732-4901d9dd-a9ebb62d/server-client.no_comments_merged.c
@@ -588 +588,2 @@ server_client_msg_dispatch(struct client *c)
-if (datalen != sizeof identifydata) fatalx("bad MSG_IDENTIFY size"); memcpy(&identifydata, imsg.data, sizeof identifydata); #ifdef __CYGWIN__ imsg.fd = open(identifydata.ttyname, O_RDWR|O_NOCTTY); #endif
+  case MSG_COMMAND:
+   server_client_msg_command(c, &imsg);
