--- a/./output/tmux/c190c732-4901d9dd-a9ebb62d/tmux.no_comments_mergegen.h
+++ b/./output/tmux/c190c732-4901d9dd-a9ebb62d/tmux.no_comments_merged.h
@@ -353,0 +354,2 @@ struct msg_command_data {
+ int argc;
+};
@@ -1792 +1794 @@ u_int utf8_split2(u_int, u_char *);
-char *osdep_get_name(int, char *); char *osdep_get_cwd(int); struct event_base *osdep_event_init(void);
+char *get_proc_name(int, char *);
