--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/a68c19df-e916058f-576a7b8c/tests@modules@misc.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/a68c19df-e916058f-576a7b8c/tests@modules@misc.no_comments_merged.c
@@ -175,3 +179 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-<<<<<<< HEAD
-  if (RedisModule_CreateCommand(ctx, "test.ld_conversion", test_ld_conv, "", 0,
-                                0, 0) == REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.ld_conversion", test_ld_conv, "",0,0,0) == REDISMODULE_ERR)
@@ -179,4 +181 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-|||||||
-=======
-  if (RedisModule_CreateCommand(ctx, "test.flushall", test_flushall, "", 0, 0,
-                                0) == REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.flushall", test_flushall,"",0,0,0) == REDISMODULE_ERR)
@@ -202 +194,0 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
->>>>>>> e916058f0ba59e964f5de3dee17f46ae08f1d385
