diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/redis/a68c19df-e916058f-576a7b8c/tests@modules@misc.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/redis/a68c19df-e916058f-576a7b8c/tests@modules@misc.no_comments_merged.c
index 7c90bdb..0ced00b 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/redis/a68c19df-e916058f-576a7b8c/tests@modules@misc.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/redis/a68c19df-e916058f-576a7b8c/tests@modules@misc.no_comments_merged.c
@@ -8 +8,2 @@
-int test_call_generic(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_call_generic(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -14,2 +15 @@ int test_call_generic(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  RedisModuleCallReply *reply =
-      RedisModule_Call(ctx, cmdname, "v", argv + 2, argc - 2);
+    RedisModuleCallReply *reply = RedisModule_Call(ctx, cmdname, "v", argv+2, argc-2);
@@ -24 +24,2 @@ int test_call_generic(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-int test_call_info(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_call_info(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -44 +45,2 @@ int test_ld_conv(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  RedisModuleString *s2 = RedisModule_CreateString(ctx, ldstr, strlen(ldstr));
+    RedisModuleString *s2 =
+        RedisModule_CreateString(ctx, ldstr, strlen(ldstr));
@@ -47,3 +49,4 @@ int test_ld_conv(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-    snprintf(
-        err, 4096, "Failed to convert long double to string ('%s' != '%s')",
-        RedisModule_StringPtrLen(s1, NULL), RedisModule_StringPtrLen(s2, NULL));
+        snprintf(err, 4096,
+            "Failed to convert long double to string ('%s' != '%s')",
+            RedisModule_StringPtrLen(s1, NULL),
+            RedisModule_StringPtrLen(s2, NULL));
@@ -55 +58,2 @@ int test_ld_conv(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-    RedisModule_ReplyWithError(ctx, "Failed to convert string to long double");
+        RedisModule_ReplyWithError(ctx,
+            "Failed to convert string to long double");
@@ -61 +65,2 @@ int test_ld_conv(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-             "Failed to convert string to long double (%.40Lf != %.40Lf)", ld2,
+            "Failed to convert string to long double (%.40Lf != %.40Lf)",
+            ld2,
@@ -72 +77,2 @@ final:
-int test_flushall(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_flushall(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -79 +85,2 @@ int test_flushall(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-int test_dbsize(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_dbsize(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -86 +93,2 @@ int test_dbsize(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-int test_randomkey(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_randomkey(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -94,2 +102 @@ int test_randomkey(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-RedisModuleKey *open_key_or_reply(RedisModuleCtx *ctx,
-                                  RedisModuleString *keyname, int mode) {
+RedisModuleKey *open_key_or_reply(RedisModuleCtx *ctx, RedisModuleString *keyname, int mode) {
@@ -103 +110,2 @@ RedisModuleKey *open_key_or_reply(RedisModuleCtx *ctx,
-int test_getlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_getlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -108,2 +116 @@ int test_getlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  RedisModuleKey *key = open_key_or_reply(
-      ctx, argv[1], REDISMODULE_READ | REDISMODULE_OPEN_KEY_NOTOUCH);
+    RedisModuleKey *key = open_key_or_reply(ctx, argv[1], REDISMODULE_READ|REDISMODULE_OPEN_KEY_NOTOUCH);
@@ -116 +123,2 @@ int test_getlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-int test_setlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_setlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -121,2 +129 @@ int test_setlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  RedisModuleKey *key = open_key_or_reply(
-      ctx, argv[1], REDISMODULE_READ | REDISMODULE_OPEN_KEY_NOTOUCH);
+    RedisModuleKey *key = open_key_or_reply(ctx, argv[1], REDISMODULE_READ|REDISMODULE_OPEN_KEY_NOTOUCH);
@@ -133 +140,2 @@ int test_setlru(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-int test_getlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_getlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -138,2 +146 @@ int test_getlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  RedisModuleKey *key = open_key_or_reply(
-      ctx, argv[1], REDISMODULE_READ | REDISMODULE_OPEN_KEY_NOTOUCH);
+    RedisModuleKey *key = open_key_or_reply(ctx, argv[1], REDISMODULE_READ|REDISMODULE_OPEN_KEY_NOTOUCH);
@@ -146 +153,2 @@ int test_getlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-int test_setlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+int test_setlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)
+{
@@ -151,2 +159 @@ int test_setlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  RedisModuleKey *key = open_key_or_reply(
-      ctx, argv[1], REDISMODULE_READ | REDISMODULE_OPEN_KEY_NOTOUCH);
+    RedisModuleKey *key = open_key_or_reply(ctx, argv[1], REDISMODULE_READ|REDISMODULE_OPEN_KEY_NOTOUCH);
@@ -163,2 +170 @@ int test_setlfu(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-                       int argc) {
+int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
@@ -169,2 +175 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.call_generic", test_call_generic, "",
-                                0, 0, 0) == REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.call_generic", test_call_generic,"",0,0,0) == REDISMODULE_ERR)
@@ -172,2 +177 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.call_info", test_call_info, "", 0, 0,
-                                0) == REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.call_info", test_call_info,"",0,0,0) == REDISMODULE_ERR)
@@ -175,3 +179 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-<<<<<<< HEAD
-  if (RedisModule_CreateCommand(ctx, "test.ld_conversion", test_ld_conv, "", 0,
-                                0, 0) == REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.ld_conversion", test_ld_conv, "",0,0,0) == REDISMODULE_ERR)
@@ -179,4 +181 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-|||||||
-=======
-  if (RedisModule_CreateCommand(ctx, "test.flushall", test_flushall, "", 0, 0,
-                                0) == REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.flushall", test_flushall,"",0,0,0) == REDISMODULE_ERR)
@@ -184,2 +183 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.dbsize", test_dbsize, "", 0, 0, 0) ==
-      REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.dbsize", test_dbsize,"",0,0,0) == REDISMODULE_ERR)
@@ -187,2 +185 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.randomkey", test_randomkey, "", 0, 0,
-                                0) == REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.randomkey", test_randomkey,"",0,0,0) == REDISMODULE_ERR)
@@ -190,2 +187 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.setlru", test_setlru, "", 0, 0, 0) ==
-      REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.setlru", test_setlru,"",0,0,0) == REDISMODULE_ERR)
@@ -193,2 +189 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.getlru", test_getlru, "", 0, 0, 0) ==
-      REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.getlru", test_getlru,"",0,0,0) == REDISMODULE_ERR)
@@ -196,2 +191 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.setlfu", test_setlfu, "", 0, 0, 0) ==
-      REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.setlfu", test_setlfu,"",0,0,0) == REDISMODULE_ERR)
@@ -199,2 +193 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
-  if (RedisModule_CreateCommand(ctx, "test.getlfu", test_getlfu, "", 0, 0, 0) ==
-      REDISMODULE_ERR)
+    if (RedisModule_CreateCommand(ctx,"test.getlfu", test_getlfu,"",0,0,0) == REDISMODULE_ERR)
@@ -202 +194,0 @@ int RedisModule_OnLoad(RedisModuleCtx *ctx, RedisModuleString **argv,
->>>>>>> e916058f0ba59e964f5de3dee17f46ae08f1d385
