--- a/./output/flameshot/c84b6df7-b1ba67ac-d1274add/src@config@geneneralconf.no_comments_mergegen.cpp
+++ b/./output/flameshot/c84b6df7-b1ba67ac-d1274add/src@config@geneneralconf.no_comments_merged.cpp
@@ -35,0 +36,6 @@ void GeneneralConf::updateComponents()
+    if (!config.saveAfterCopyPathValue().isEmpty()) {
+        m_savePath->setText(config.saveAfterCopyPathValue());
+    } else {
+        ConfigHandler().setSaveAfterCopyPath(
+          QStandardPaths::writableLocation(QStandardPaths::PicturesLocation));
+    }
