--- a/./output/flameshot/c84b6df7-b1ba67ac-d1274add/src@widgets@panel@utilitypanel.no_comments_mergegen.cpp
+++ b/./output/flameshot/c84b6df7-b1ba67ac-d1274add/src@widgets@panel@utilitypanel.no_comments_merged.cpp
@@ -24 +24,4 @@ UtilityPanel::UtilityPanel(QWidget* parent)
-#if (defined(Q_OS_WIN) || defined(Q_OS_MAC) || defined(Q_OS_MAC64) || \ defined(Q_OS_MACOS) || defined(Q_OS_MACX)) move(0, 0); #endif hide();
+#if (defined(Q_OS_WIN) || defined(Q_OS_MAC) || defined(Q_OS_MAC64) || \
+     defined(Q_OS_MACOS) || defined(Q_OS_MACX))
+    move(0, 0);
+#endif
@@ -102,0 +106,5 @@ void UtilityPanel::initInternalPanel()
+    m_hide = new QPushButton();
+    m_hide->setText(tr("Hide"));
+    m_upLayout->addWidget(m_hide);
+    connect(m_hide, SIGNAL(clicked()), this, SLOT(slotHidePanel()));
+    m_internalPanel->hide();
