--- a/./output/flameshot/c84b6df7-b1ba67ac-d1274add/src@config@shortcutswidget.no_comments_mergegen.cpp
+++ b/./output/flameshot/c84b6df7-b1ba67ac-d1274add/src@config@shortcutswidget.no_comments_merged.cpp
@@ -62 +62,7 @@ void ShortcutsWidget::initInfoTable()
-const auto key_sequence = identifier.isEmpty() ? default_key_sequence : m_config.shortcut(identifier); QTableWidgetItem* item = new QTableWidgetItem(key_sequence);
+#if (defined(Q_OS_MAC) || defined(Q_OS_MAC64) || defined(Q_OS_MACOS) || \
+     defined(Q_OS_MACX))
+        QTableWidgetItem* item =
+          new QTableWidgetItem(nativeOSHotKeyText(m_shortcuts.at(i).at(2)));
+#else
+        QTableWidgetItem* item = new QTableWidgetItem(m_shortcuts.at(i).at(2));
+#endif
