diff --git a/./output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/libs@input@KeyLayoutMap.no_comments_mergegen.cpp b/./output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/libs@input@KeyLayoutMap.no_comments_merged.cpp
index 434a73c..57457c9 100644
--- a/./output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/libs@input@KeyLayoutMap.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/f2c5f522-ef1a90de-0d4ac562/libs@input@KeyLayoutMap.no_comments_merged.cpp
@@ -184 +184,5 @@ std::vector<int32_t> KeyLayoutMap::findScanCodesForKey(int32_t keyCode) const {
-std::optional<AxisInfo> KeyLayoutMap::mapAxis(int32_t scanCode) const { auto it = mAxes.find(scanCode); if (it == mAxes.end()) { ALOGD_IF(DEBUG_MAPPING, "mapAxis: scanCode=%d ~ Failed.", scanCode); return std::nullopt;
+std::optional<AxisInfo> KeyLayoutMap::mapAxis(int32_t scanCode) const {
+    auto it = mAxes.find(scanCode);
+    if (it == mAxes.end()) {
+        ALOGD_IF(DEBUG_MAPPING, "mapAxis: scanCode=%d ~ Failed.", scanCode);
+        return std::nullopt;
@@ -186 +190,7 @@ std::optional<AxisInfo> KeyLayoutMap::mapAxis(int32_t scanCode) const { auto it
-const AxisInfo& axisInfo = it->second; ALOGD_IF(DEBUG_MAPPING, "mapAxis: scanCode=%d ~ Result mode=%d, axis=%d, highAxis=%d, " "splitValue=%d, flatOverride=%d.", scanCode, axisInfo.mode, axisInfo.axis, axisInfo.highAxis, axisInfo.splitValue, axisInfo.flatOverride); return axisInfo;
+    const AxisInfo& axisInfo = it->second;
+    ALOGD_IF(DEBUG_MAPPING,
+             "mapAxis: scanCode=%d ~ Result mode=%d, axis=%d, highAxis=%d, "
+             "splitValue=%d, flatOverride=%d.",
+             scanCode, axisInfo.mode, axisInfo.axis, axisInfo.highAxis, axisInfo.splitValue,
+             axisInfo.flatOverride);
+    return axisInfo;
@@ -188 +198,5 @@ const AxisInfo& axisInfo = it->second; ALOGD_IF(DEBUG_MAPPING, "mapAxis: scanCod
-std::optional<int32_t> KeyLayoutMap::findScanCodeForLed(int32_t ledCode) const { for (const auto& [scanCode, led] : mLedsByScanCode) { if (led.ledCode == ledCode) { ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d, scanCode=%d.", __func__, ledCode, scanCode); return scanCode;
+std::optional<int32_t> KeyLayoutMap::findScanCodeForLed(int32_t ledCode) const {
+    for (const auto& [scanCode, led] : mLedsByScanCode) {
+        if (led.ledCode == ledCode) {
+            ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d, scanCode=%d.", __func__, ledCode, scanCode);
+            return scanCode;
@@ -191 +205,2 @@ std::optional<int32_t> KeyLayoutMap::findScanCodeForLed(int32_t ledCode) const {
-ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d ~ Not found.", __func__, ledCode); return std::nullopt;
+    ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d ~ Not found.", __func__, ledCode);
+    return std::nullopt;
@@ -193 +208,5 @@ ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d ~ Not found.", __func__, ledCode); retur
-std::optional<int32_t> KeyLayoutMap::findUsageCodeForLed(int32_t ledCode) const { for (const auto& [usageCode, led] : mLedsByUsageCode) { if (led.ledCode == ledCode) { ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d, usage=%x.", __func__, ledCode, usageCode); return usageCode;
+std::optional<int32_t> KeyLayoutMap::findUsageCodeForLed(int32_t ledCode) const {
+    for (const auto& [usageCode, led] : mLedsByUsageCode) {
+        if (led.ledCode == ledCode) {
+            ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d, usage=%x.", __func__, ledCode, usageCode);
+            return usageCode;
@@ -196 +215,2 @@ std::optional<int32_t> KeyLayoutMap::findUsageCodeForLed(int32_t ledCode) const
-ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d ~ Not found.", __func__, ledCode); return std::nullopt;
+    ALOGD_IF(DEBUG_MAPPING, "%s: ledCode=%d ~ Not found.", __func__, ledCode);
+    return std::nullopt;
@@ -385 +405,6 @@ status_t KeyLayoutMap::Parser::parseAxis() {
-ALOGD_IF(DEBUG_PARSER, "Parsed axis: scanCode=%d, mode=%d, axis=%d, highAxis=%d, " "splitValue=%d, flatOverride=%d.", scanCode, axisInfo.mode, axisInfo.axis, axisInfo.highAxis, axisInfo.splitValue, axisInfo.flatOverride); mMap->mAxes.insert({scanCode, axisInfo});
+    ALOGD_IF(DEBUG_PARSER,
+             "Parsed axis: scanCode=%d, mode=%d, axis=%d, highAxis=%d, "
+             "splitValue=%d, flatOverride=%d.",
+             scanCode, axisInfo.mode, axisInfo.axis, axisInfo.highAxis, axisInfo.splitValue,
+             axisInfo.flatOverride);
+    mMap->mAxes.insert({scanCode, axisInfo});
