--- a/./output/frameworks_native/71425310-cb7de996-9bb81c04/services@inputflinger@reader@mapper@accumulator@MultiTouchMotionAccumulator.no_comments_conflict.cpp
+++ b/./output/frameworks_native/71425310-cb7de996-9bb81c04/services@inputflinger@reader@mapper@accumulator@MultiTouchMotionAccumulator.no_comments_merged.cpp
@@ -93 +92,0 @@ void MultiTouchMotionAccumulator::warnIfNotInUse(const RawEvent& event, const Sl
-<<<<<<< HEAD
@@ -115,17 +113,0 @@ void MultiTouchMotionAccumulator::populateCurrentSlot(
-||||||| 9bb81c04ef
-=======
-void MultiTouchMotionAccumulator::populateCurrentSlot(
-        const android::InputDeviceContext& deviceContext) {
-    if (!mUsingSlotsProtocol) {
-        return;
-    }
-    int32_t initialSlot;
-    if (const auto status = deviceContext.getAbsoluteAxisValue(ABS_MT_SLOT, &initialSlot);
-        status == OK) {
-        mCurrentSlot = initialSlot;
-    } else {
-        ALOGE("Could not retrieve current multi-touch slot index. status=%s",
-              statusToString(status).c_str());
-    }
-}
->>>>>>> cb7de996
