--- a/./output/frameworks_native/71425310-cb7de996-9bb81c04/services@inputflinger@tests@TouchpadInputMapper_test.no_comments_conflict.cpp
+++ b/./output/frameworks_native/71425310-cb7de996-9bb81c04/services@inputflinger@tests@TouchpadInputMapper_test.no_comments_merged.cpp
@@ -75 +74,0 @@ protected:
-<<<<<<< HEAD
@@ -81,7 +79,0 @@ protected:
-||||||| 9bb81c04ef
-=======
-        EXPECT_CALL(mMockEventHub, getMtSlotValues(EVENTHUB_ID, testing::_, testing::_))
-                .WillRepeatedly([]() -> base::Result<std::vector<int32_t>> {
-                    return base::ResultError("Axis not supported", NAME_NOT_FOUND);
-                });
->>>>>>> cb7de996
