--- a/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_conflict.cpp
+++ b/./output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@wal_replay.no_comments_merged.cpp
@@ -503 +502,0 @@ void WriteAheadLogDeserializer::ReplayCreateIndex() {
-<<<<<<< HEAD
@@ -505,5 +503,0 @@ void WriteAheadLogDeserializer::ReplayCreateIndex() {
-||||||| 9afef29d90
- vector<column_t> column_ids;
-=======
- vector<column_t> column_ids;
->>>>>>> eebe9bce
