diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_merged.cpp
index 9abe506..3e4be52 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/0051fbac-eebe9bce-9afef29d/src@storage@table@row_group_collection.no_comments_merged.cpp
@@ -914,2 +914,2 @@ void RowGroupCollection::VerifyNewConstraint(DataTable &parent, const BoundConst
- vector<column_t> column_ids;
- column_ids.push_back(physical_index);
+ vector<StorageIndex> column_ids;
+ column_ids.emplace_back(physical_index);
@@ -917,11 +916,0 @@ void RowGroupCollection::VerifyNewConstraint(DataTable &parent, const BoundConst
-<<<<<<< HEAD
- vector<StorageIndex> cids;
- cids.emplace_back(physical_index);
- state.Initialize(cids, nullptr);
- InitializeScan(state.table_state, cids, nullptr);
-|||||||
- vector<column_t> cids;
- cids.push_back(physical_index);
- state.Initialize(cids, nullptr);
- InitializeScan(state.table_state, cids, nullptr);
-=======
@@ -931 +919,0 @@ void RowGroupCollection::VerifyNewConstraint(DataTable &parent, const BoundConst
->>>>>>> eebe9bceb34a3ed24f02abd9fdec23408122c0a6
