diff --git a/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_mergegen.cpp b/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_merged.cpp
index 3a599ff..aaf2dab 100644
--- a/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_merged.cpp
@@ -7 +7,4 @@
-#include <perfetto/trace/android/winscope_extensions.pbzero.h> #include <perfetto/trace/android/winscope_extensions_impl.pbzero.h> #include <private/android_filesystem_config.h> #include <utils/String16.h>
+#include <perfetto/trace/android/winscope_extensions.pbzero.h>
+#include <perfetto/trace/android/winscope_extensions_impl.pbzero.h>
+#include <private/android_filesystem_config.h>
+#include <utils/String16.h>
@@ -164 +167,6 @@ void PerfettoBackend::traceMotionEvent(const TracedMotionEvent& event,
-auto* winscopeExtensions = static_cast<perfetto::protos::pbzero::WinscopeExtensionsImpl*>( tracePacket->set_winscope_extensions()); auto* inputEvent = winscopeExtensions->set_android_input_event(); auto* dispatchEvent = isRedacted ? inputEvent->set_dispatcher_window_dispatch_event_redacted() : inputEvent->set_dispatcher_window_dispatch_event(); AndroidInputEventProtoConverter::toProtoWindowDispatchEvent(dispatchArgs, *dispatchArgs, isRedacted);
+        auto* winscopeExtensions = static_cast<perfetto::protos::pbzero::WinscopeExtensionsImpl*>(
+                tracePacket->set_winscope_extensions());
+        auto* inputEvent = winscopeExtensions->set_android_input_event();
+        auto* dispatchMotion = isRedacted ? inputEvent->set_dispatcher_motion_event_redacted()
+                                          : inputEvent->set_dispatcher_motion_event();
+        AndroidInputEventProtoConverter::toProtoMotionEvent(event, *dispatchMotion, isRedacted);
@@ -183,0 +192,6 @@ void PerfettoBackend::traceKeyEvent(const TracedKeyEvent& event,
+        auto* winscopeExtensions = static_cast<perfetto::protos::pbzero::WinscopeExtensionsImpl*>(
+                tracePacket->set_winscope_extensions());
+        auto* inputEvent = winscopeExtensions->set_android_input_event();
+        auto* dispatchKey = isRedacted ? inputEvent->set_dispatcher_key_event_redacted()
+                                       : inputEvent->set_dispatcher_key_event();
+        AndroidInputEventProtoConverter::toProtoKeyEvent(event, *dispatchKey, isRedacted);
@@ -202,0 +217,8 @@ void PerfettoBackend::traceWindowDispatch(const WindowDispatchArgs& dispatchArgs
+        auto* winscopeExtensions = static_cast<perfetto::protos::pbzero::WinscopeExtensionsImpl*>(
+                tracePacket->set_winscope_extensions());
+        auto* inputEvent = winscopeExtensions->set_android_input_event();
+        auto* dispatchEvent = isRedacted
+                ? inputEvent->set_dispatcher_window_dispatch_event_redacted()
+                : inputEvent->set_dispatcher_window_dispatch_event();
+        AndroidInputEventProtoConverter::toProtoWindowDispatchEvent(dispatchArgs, *dispatchEvent,
+                                                                    isRedacted);
