--- a/./output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_conflict.h
+++ b/./output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_merged.h
@@ -3,5 +2,0 @@
-<<<<<<< HEAD
-#include "opencensus/trace/span.h"
-||||||| cca89309a2
-#include "grpcpp/opencensus_impl.h"
-=======
@@ -12 +6,0 @@ class ServerContext;
->>>>>>> c483ece5
@@ -14,21 +7,0 @@ namespace grpc {
-<<<<<<< HEAD
-class ServerContext;
-||||||| cca89309a2
-=======
->>>>>>> c483ece5
-<<<<<<< HEAD
-void RegisterOpenCensusPlugin();
-void RegisterOpenCensusViewsForExport();
-::opencensus::trace::Span GetSpanFromServerContext(ServerContext* context);
-||||||| cca89309a2
-static inline void RegisterOpenCensusPlugin() {
-  ::grpc_impl::RegisterOpenCensusPlugin();
-}
-static inline void RegisterOpenCensusViewsForExport() {
-  ::grpc_impl::RegisterOpenCensusViewsForExport();
-}
-static inline ::opencensus::trace::Span GetSpanFromServerContext(
-    ::grpc_impl::ServerContext* context) {
-  return ::grpc_impl::GetSpanFromServerContext(context);
-}
-=======
@@ -39 +11,0 @@ void RegisterOpenCensusViewsForExport();
->>>>>>> c483ece5
