--- a/./output/system_core/763fa64d-d7846a2c-1fa02519/fs_mgr@fs_mgr_avb.no_comments_mergegen.cpp
+++ b/./output/system_core/763fa64d-d7846a2c-1fa02519/fs_mgr@fs_mgr_avb.no_comments_merged.cpp
@@ -1 +1 @@
-#include "fs_mgr_avb.h" #include "fs_mgr_avb.h"
+#include "fs_mgr_avb.h"
@@ -367 +367,5 @@ FsManagerAvbUniquePtr FsManagerAvbHandle::DoOpen(FsManagerAvbOps* avb_ops) {
-AvbSlotVerifyFlags flags = avb_verifier->IsDeviceUnlocked() ? AVB_SLOT_VERIFY_FLAGS_ALLOW_VERIFICATION_ERROR : AVB_SLOT_VERIFY_FLAGS_NONE; AvbSlotVerifyResult verify_result = avb_ops->AvbSlotVerify(fs_mgr_get_slot_suffix(), flags,
+    AvbSlotVerifyFlags flags = avb_verifier->IsDeviceUnlocked()
+                                   ? AVB_SLOT_VERIFY_FLAGS_ALLOW_VERIFICATION_ERROR
+                                   : AVB_SLOT_VERIFY_FLAGS_NONE;
+    AvbSlotVerifyResult verify_result =
+        avb_ops->AvbSlotVerify(fs_mgr_get_slot_suffix(), flags, &avb_handle->avb_slot_data_);
