--- a/./output/frameworks_av/e0c3d791-d7227f1c-10d0f99e/media@libmediaplayerservice@nuplayer@NuPlayerDriver.no_comments_conflict.h
+++ b/./output/frameworks_av/e0c3d791-d7227f1c-10d0f99e/media@libmediaplayerservice@nuplayer@NuPlayerDriver.no_comments_merged.h
@@ -98 +97,0 @@ private:
-<<<<<<< HEAD
@@ -101,6 +99,0 @@ private:
-||||||| 10d0f99ec7
-    MediaAnalyticsItem *mAnalyticsItem;
-=======
-    MediaAnalyticsItem *mAnalyticsItem;
-    mutable Mutex mMetricsLock;
->>>>>>> d7227f1c
