diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_merged.h
index 3663215..266b6d3 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_merged.h
@@ -13,2 +13,2 @@ const std::unordered_map<std::string, int32_t>& getSpatializerModeMap();
-std::vector<int32_t> vectorFromMap(const std::string& str,
-                                   const std::unordered_map<std::string, int32_t>& map);
+std::vector<int32_t> vectorFromMap(
+        const std::string &str, const std::unordered_map<std::string, int32_t>& map);
@@ -19,2 +19 @@ enum DeviceConnectionResult : int32_t {
-    DEVICE_CONNECTION_RESULT_JAVA_SERVICE_CANCEL =
-            2,
+    DEVICE_CONNECTION_RESULT_JAVA_SERVICE_CANCEL = 2,
@@ -43,0 +43,2 @@ enum AudioEnumCategory {
+template <AudioEnumCategory C, typename T, typename S>
+T lookup(const S &str);
