--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioAnalytics.no_comments_merged.h
@@ -148,0 +156 @@ class AudioAnalytics {
+        static constexpr int64_t kBootDurationThreshold = 120 * 1e9;
@@ -149,0 +158 @@ class AudioAnalytics {
+        int64_t mFirstCreateTimeNs GUARDED_BY(mLock) = 0;
