--- a/./output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_conflict.h
+++ b/./output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_merged.h
@@ -10 +9,0 @@ const std::unordered_map<std::string, int32_t>& getAudioTrackTraitsMap();
-<<<<<<< HEAD
@@ -17,8 +15,0 @@ std::vector<int64_t> channelMaskVectorFromString(const std::string &s);
-||||||| 3be023edb8
-=======
-const std::unordered_map<std::string, int32_t>& getHeadTrackingModeMap();
-const std::unordered_map<std::string, int32_t>& getSpatializerLevelMap();
-const std::unordered_map<std::string, int32_t>& getSpatializerModeMap();
-std::vector<int32_t> vectorFromMap(
-        const std::string &str, const std::unordered_map<std::string, int32_t>& map);
->>>>>>> 4591aab2
