--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6bf4d09e-4591aab2-3be023ed/services@mediametrics@include@mediametricsservice@AudioTypes.no_comments_merged.h
@@ -43,0 +43,2 @@ enum AudioEnumCategory {
+template <AudioEnumCategory C, typename T, typename S>
+T lookup(const S &str);
