diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_merged.cpp
index 7f3bd45..272fa14 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_batch_collector.no_comments_merged.cpp
@@ -25 +25 @@ SinkResultType PhysicalBatchCollector::Sink(ExecutionContext &context, DataChunk
- state.data.Append(chunk, state.batch_index);
+ state.data.Append(chunk, state.partition_info.batch_index.GetIndex());
