--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@parallel@pipeline_executor.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@parallel@pipeline_executor.no_comments_merged.cpp
@@ -360,2 +360,3 @@ SourceResultType PipelineExecutor::FetchFromSource(DataChunk &result) {
-<<<<<<< HEAD
- pipeline.source->GetData(context, result, *pipeline.source_state, *local_source_state);
+ OperatorSourceInput source_input = {*pipeline.source_state, *local_source_state, interrupt_state};
+ auto res = GetData(result, source_input);
+ D_ASSERT(res != SourceResultType::BLOCKED || result.size() == 0);
@@ -368,11 +368,0 @@ SourceResultType PipelineExecutor::FetchFromSource(DataChunk &result) {
-|||||||
- pipeline.source->GetData(context, result, *pipeline.source_state, *local_source_state);
- if (result.size() != 0 && requires_batch_index) {
-  auto next_batch_index =
-=======
- OperatorSourceInput source_input = {*pipeline.source_state, *local_source_state, interrupt_state};
- auto res = GetData(result, source_input);
- D_ASSERT(res != SourceResultType::BLOCKED || result.size() == 0);
- if (result.size() != 0 && requires_batch_index) {
-  auto next_batch_index =
->>>>>>> 6255d3078116b01d90c98248cd0c4c58159db366
