diff --git a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@persistent@physical_batch_insert.no_comments_conflict.cpp b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@persistent@physical_batch_insert.no_comments_merged.cpp
index f0c87d2..4f79a19 100644
--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@persistent@physical_batch_insert.no_comments_conflict.cpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@persistent@physical_batch_insert.no_comments_merged.cpp
@@ -255 +254,0 @@ unique_ptr<LocalSinkState> PhysicalBatchInsert::GetLocalSinkState(ExecutionConte
-<<<<<<< HEAD
@@ -273,10 +271,0 @@ void PhysicalBatchInsert::NextBatch(ExecutionContext &context, GlobalSinkState &
-SinkResultType PhysicalBatchInsert::Sink(ExecutionContext &context, GlobalSinkState &state, LocalSinkState &lstate_p,
-                                         DataChunk &chunk) const {
- auto &gstate = state.Cast<BatchInsertGlobalState>();
- auto &lstate = lstate_p.Cast<BatchInsertLocalState>();
-||||||| fc797c18cf
-SinkResultType PhysicalBatchInsert::Sink(ExecutionContext &context, GlobalSinkState &state, LocalSinkState &lstate_p,
-                                         DataChunk &chunk) const {
- auto &gstate = state.Cast<BatchInsertGlobalState>();
- auto &lstate = lstate_p.Cast<BatchInsertLocalState>();
-=======
@@ -286 +274,0 @@ SinkResultType PhysicalBatchInsert::Sink(ExecutionContext &context, DataChunk &c
->>>>>>> 6255d307
