diff --git a/./output/duckdb/1a69272b-6255d307-fc797c18/src@parallel@pipeline_executor.no_comments_conflict.cpp b/./output/duckdb/1a69272b-6255d307-fc797c18/src@parallel@pipeline_executor.no_comments_merged.cpp
index 40a0d02..04ca7fb 100644
--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@parallel@pipeline_executor.no_comments_conflict.cpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@parallel@pipeline_executor.no_comments_merged.cpp
@@ -360,2 +360,3 @@ SourceResultType PipelineExecutor::FetchFromSource(DataChunk &result) {
-<<<<<<< HEAD
- pipeline.source->GetData(context, result, *pipeline.source_state, *local_source_state);
+ OperatorSourceInput source_input = {*pipeline.source_state, *local_source_state, interrupt_state};
+ auto res = GetData(result, source_input);
+ D_ASSERT(res != SourceResultType::BLOCKED || result.size() == 0);
@@ -383,21 +383,0 @@ SourceResultType PipelineExecutor::FetchFromSource(DataChunk &result) {
-||||||| fc797c18cf
- pipeline.source->GetData(context, result, *pipeline.source_state, *local_source_state);
- if (result.size() != 0 && requires_batch_index) {
-  auto next_batch_index =
-      pipeline.source->GetBatchIndex(context, result, *pipeline.source_state, *local_source_state);
-  next_batch_index += pipeline.base_batch_index;
-  D_ASSERT(local_sink_state->batch_index <= next_batch_index ||
-           local_sink_state->batch_index == DConstants::INVALID_INDEX);
-  local_sink_state->batch_index = next_batch_index;
-=======
- OperatorSourceInput source_input = {*pipeline.source_state, *local_source_state, interrupt_state};
- auto res = GetData(result, source_input);
- D_ASSERT(res != SourceResultType::BLOCKED || result.size() == 0);
- if (result.size() != 0 && requires_batch_index) {
-  auto next_batch_index =
-      pipeline.source->GetBatchIndex(context, result, *pipeline.source_state, *local_source_state);
-  next_batch_index += pipeline.base_batch_index;
-  D_ASSERT(local_sink_state->batch_index <= next_batch_index ||
-           local_sink_state->batch_index == DConstants::INVALID_INDEX);
-  local_sink_state->batch_index = next_batch_index;
->>>>>>> 6255d307
