--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_conflict.cpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_merged.cpp
@@ -92,8 +92 @@ SinkResultType PhysicalLimit::Sink(ExecutionContext &context, DataChunk &chunk,
-<<<<<<< HEAD
- state.data.Append(input, lstate.partition_info.batch_index.GetIndex());
- state.current_offset += input.size();
-||||||| fc797c18cf
- state.data.Append(input, lstate.batch_index);
- state.current_offset += input.size();
-=======
- state.data.Append(chunk, input.local_state.batch_index);
+ state.data.Append(chunk, state.partition_info.batch_index.GetIndex());
@@ -101 +93,0 @@ SinkResultType PhysicalLimit::Sink(ExecutionContext &context, DataChunk &chunk,
->>>>>>> 6255d307
