diff --git a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_mergegen.cpp b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_merged.cpp
index 8f1aaa2..99f2086 100644
--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_mergegen.cpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_merged.cpp
@@ -92 +92,2 @@ SinkResultType PhysicalLimit::Sink(ExecutionContext &context, DataChunk &chunk,
-state.data.Append(chunk, input.local_state.partition_info.batch_index.GetIndex());
+ state.data.Append(chunk, state.partition_info.batch_index.GetIndex());
+ state.current_offset += chunk.size();
