diff --git a/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_conflict.hpp b/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_merged.hpp
index 8d510b2..5142084 100644
--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_conflict.hpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_merged.hpp
@@ -28 +27,0 @@ public:
-<<<<<<< HEAD
@@ -30,6 +28,0 @@ public:
- SinkResultType Sink(ExecutionContext &context, GlobalSinkState &state, LocalSinkState &lstate,
-                     DataChunk &input) const override;
-||||||| fc797c18cf
- SinkResultType Sink(ExecutionContext &context, GlobalSinkState &state, LocalSinkState &lstate,
-                     DataChunk &input) const override;
-=======
@@ -37 +29,0 @@ public:
->>>>>>> 6255d307
