--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/1a69272b-6255d307-fc797c18/src@execution@operator@helper@physical_limit.no_comments_merged.cpp
@@ -92 +92 @@ SinkResultType PhysicalLimit::Sink(ExecutionContext &context, DataChunk &chunk,
- state.data.Append(chunk, input.local_state.batch_index);
+ state.data.Append(chunk, state.partition_info.batch_index.GetIndex());
