diff --git a/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_mergegen.hpp b/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_merged.hpp
index 7c71f99..5142084 100644
--- a/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_mergegen.hpp
+++ b/./output/duckdb/1a69272b-6255d307-fc797c18/src@include@duckdb@execution@operator@persistent@physical_batch_insert.no_comments_merged.hpp
@@ -28 +28,2 @@ public:
-void NextBatch(ExecutionContext &context, GlobalSinkState &state, LocalSinkState &lstate_p) const override; SinkResultType Sink(ExecutionContext &context, DataChunk &chunk, OperatorSinkInput &input) const override; void Combine(ExecutionContext &context, GlobalSinkState &gstate, LocalSinkState &lstate) const override;
+ void NextBatch(ExecutionContext &context, GlobalSinkState &state, LocalSinkState &lstate_p) const override;
+ SinkResultType Sink(ExecutionContext &context, DataChunk &chunk, OperatorSinkInput &input) const override;
