--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_merged.h
@@ -4 +3,0 @@
-#include "grpcpp/opencensus_impl.h"
@@ -9 +7,0 @@ namespace grpc {
-class ServerContext;
@@ -12 +9,0 @@ void RegisterOpenCensusViewsForExport();
-::opencensus::trace::Span GetSpanFromServerContext(ServerContext* context);
