diff --git a/./output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_mergegen.h b/./output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_merged.h
index 6700c14..c151f2e 100644
--- a/./output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_mergegen.h
+++ b/./output/grpc/e55267a6-c483ece5-cca89309/include@grpcpp@opencensus.no_comments_merged.h
@@ -3 +3,4 @@
-#include "opencensus/trace/span.h" namespace grpc_impl { class ServerContext; }
+#include "opencensus/trace/span.h"
+namespace grpc_impl {
+class ServerContext;
+}
@@ -4,0 +8,4 @@ namespace grpc {
+void RegisterOpenCensusPlugin();
+void RegisterOpenCensusViewsForExport();
+::opencensus::trace::Span GetSpanFromServerContext(
+    ::grpc_impl::ServerContext* context);
