--- a/./output/grpc/e55267a6-c483ece5-cca89309/src@cpp@ext@filters@census@grpc_plugin.no_comments_conflict.cc
+++ b/./output/grpc/e55267a6-c483ece5-cca89309/src@cpp@ext@filters@census@grpc_plugin.no_comments_merged.cc
@@ -11 +10,0 @@ namespace grpc {
-<<<<<<< HEAD
@@ -13,2 +12 @@ void RegisterOpenCensusPlugin() {
-  RegisterChannelFilter<grpc::CensusChannelData,
-                              grpc::CensusClientCallData>(
+  RegisterChannelFilter<CensusChannelData, CensusClientCallData>(
@@ -17,2 +15 @@ void RegisterOpenCensusPlugin() {
-  RegisterChannelFilter<grpc::CensusChannelData,
-                              grpc::CensusServerCallData>(
+  RegisterChannelFilter<CensusChannelData, CensusServerCallData>(
@@ -38,26 +34,0 @@ void RegisterOpenCensusPlugin() {
-||||||| cca89309a2
-=======
-void RegisterOpenCensusPlugin() {
-  RegisterChannelFilter<CensusChannelData, CensusClientCallData>(
-      "opencensus_client", GRPC_CLIENT_CHANNEL, INT_MAX ,
-      nullptr );
-  RegisterChannelFilter<CensusChannelData, CensusServerCallData>(
-      "opencensus_server", GRPC_SERVER_CHANNEL, INT_MAX ,
-      nullptr );
-  RpcClientSentBytesPerRpc();
-  RpcClientReceivedBytesPerRpc();
-  RpcClientRoundtripLatency();
-  RpcClientServerLatency();
-  RpcClientSentMessagesPerRpc();
-  RpcClientReceivedMessagesPerRpc();
-  RpcServerSentBytesPerRpc();
-  RpcServerReceivedBytesPerRpc();
-  RpcServerServerLatency();
-  RpcServerSentMessagesPerRpc();
-  RpcServerReceivedMessagesPerRpc();
-}
-::opencensus::trace::Span GetSpanFromServerContext(ServerContext* context) {
-  return reinterpret_cast<const CensusContext*>(context->census_context())
-      ->Span();
-}
->>>>>>> c483ece5
