--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@HwModule.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@HwModule.no_comments_merged.h
@@ -66,2 +63,3 @@ class HwModule : public RefBase {
-  sp<AudioPort> findPortByTagName(const std::string &tagName) const {
-    return mPorts.findByTagName(tagName);
+    sp<PolicyAudioPort> findPortByTagName(const std::string &tagName) const
+    {
+        return findByTagName(mPorts, tagName);
