diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
index 63f28ef..4fdab07 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
@@ -2 +1,0 @@
-#include "AudioPort.h"
@@ -14,2 +13,2 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-                         public PolicyAudioPort,
-                         public PolicyAudioPortConfig {
+                         public PolicyAudioPort, public PolicyAudioPortConfig
+{
@@ -17 +15,0 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-<<<<<<< HEAD
@@ -19,7 +17,6 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-||||||| 5af8fef591
-  explicit DeviceDescriptor(audio_devices_t type,
-                            const String8 &tagName = String8(""));
-=======
-  explicit DeviceDescriptor(audio_devices_t type,
-                            const std::string &tagName = "");
->>>>>>> f1e48f349fcc5313e8956b07176ac975b7a0f196
+    DeviceDescriptor(audio_devices_t type, const std::string &tagName,
+            const FormatVector &encodedFormats = FormatVector{});
+    DeviceDescriptor(audio_devices_t type, const std::string &tagName,
+            const std::string &address, const FormatVector &encodedFormats = FormatVector{});
+    DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr, const std::string &tagName = "",
+            const FormatVector &encodedFormats = FormatVector{});
@@ -42,2 +39 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-  virtual status_t applyAudioPortConfig(
-      const struct audio_port_config *config,
+    virtual status_t applyAudioPortConfig(const struct audio_port_config *config,
@@ -45,2 +41 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-  virtual void toAudioPortConfig(
-      struct audio_port_config *dstConfig,
+    virtual void toAudioPortConfig(struct audio_port_config *dstConfig,
@@ -58,2 +53 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-  std::string
-      mTagName;
+    std::string mTagName;
@@ -62,9 +55,0 @@ class DeviceDescriptor : public DeviceDescriptorBase,
- public:
-  DeviceDescriptor(audio_devices_t type, const std::string &tagName,
-                   const FormatVector &encodedFormats = FormatVector{});
-  DeviceDescriptor(audio_devices_t type, const std::string &tagName,
-                   const std::string &address,
-                   const FormatVector &encodedFormats = FormatVector{});
-  DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr,
-                   const std::string &tagName = "",
-                   const FormatVector &encodedFormats = FormatVector{});
@@ -72 +57,2 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
+class DeviceVector : public SortedVector<sp<DeviceDescriptor> >
+{
@@ -75 +61,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  explicit DeviceVector(const sp<DeviceDescriptor> &item) : DeviceVector() {
+    explicit DeviceVector(const sp<DeviceDescriptor>& item) : DeviceVector()
+    {
@@ -93,4 +80,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  DeviceVector getFirstDevicesFromTypes(
-      std::vector<audio_devices_t> orderedTypes) const;
-  sp<DeviceDescriptor> getFirstExistingDevice(
-      std::vector<audio_devices_t> orderedTypes) const;
+    DeviceVector getFirstDevicesFromTypes(std::vector<audio_devices_t> orderedTypes) const;
+    sp<DeviceDescriptor> getFirstExistingDevice(std::vector<audio_devices_t> orderedTypes) const;
@@ -98,2 +83 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  void replaceDevicesByType(audio_devices_t typeToRemove,
-                            const DeviceVector &devicesToAdd);
+    void replaceDevicesByType(audio_devices_t typeToRemove, const DeviceVector &devicesToAdd);
@@ -109,3 +93 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  bool contains(const sp<DeviceDescriptor> &item) const {
-    return indexOf(item) >= 0;
-  }
+    bool contains(const sp<DeviceDescriptor>& item) const { return indexOf(item) >= 0; }
@@ -116 +98,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  ssize_t merge(const DeviceVector &devices) {
+    ssize_t merge(const DeviceVector &devices)
+    {
@@ -123 +106,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  bool operator==(const DeviceVector &right) const {
+    bool operator==(const DeviceVector &right) const
+    {
@@ -134 +118,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  bool operator!=(const DeviceVector &right) const {
+    bool operator!=(const DeviceVector &right) const
+    {
@@ -137 +122,2 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  String8 getFirstValidAddress() const {
+    String8 getFirstValidAddress() const
+    {
@@ -146,2 +132 @@ class DeviceVector : public SortedVector<sp<DeviceDescriptor> > {
-  void dump(String8 *dst, const String8 &tag, int spaces = 0,
-            bool verbose = true) const;
+    void dump(String8 *dst, const String8 &tag, int spaces = 0, bool verbose = true) const;
