diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@HwModule.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@HwModule.no_comments_merged.h
index 0da59d6..672541c 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@HwModule.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@HwModule.no_comments_merged.h
@@ -18 +18,2 @@ typedef Vector<sp<IOProfile> > IOProfileCollection;
-class HwModule : public RefBase {
+class HwModule : public RefBase
+{
@@ -20,2 +21 @@ class HwModule : public RefBase {
-  explicit HwModule(const char *name, uint32_t halVersionMajor = 0,
-                    uint32_t halVersionMinor = 0);
+    explicit HwModule(const char *name, uint32_t halVersionMajor = 0, uint32_t halVersionMinor = 0);
@@ -26 +26,2 @@ class HwModule : public RefBase {
-  DeviceVector getAllDevices() const {
+    DeviceVector getAllDevices() const
+    {
@@ -31 +32,2 @@ class HwModule : public RefBase {
-  void addDynamicDevice(const sp<DeviceDescriptor> &device) {
+    void addDynamicDevice(const sp<DeviceDescriptor> &device)
+    {
@@ -34 +36,2 @@ class HwModule : public RefBase {
-  bool removeDynamicDevice(const sp<DeviceDescriptor> &device) {
+    bool removeDynamicDevice(const sp<DeviceDescriptor> &device)
+    {
@@ -38,6 +41,2 @@ class HwModule : public RefBase {
-  const InputProfileCollection &getInputProfiles() const {
-    return mInputProfiles;
-  }
-  const OutputProfileCollection &getOutputProfiles() const {
-    return mOutputProfiles;
-  }
+    const InputProfileCollection &getInputProfiles() const { return mInputProfiles; }
+    const OutputProfileCollection &getOutputProfiles() const { return mOutputProfiles; }
@@ -56,2 +55 @@ class HwModule : public RefBase {
-  status_t addOutputProfile(const std::string &name,
-                            const audio_config_t *config,
+    status_t addOutputProfile(const std::string& name, const audio_config_t *config,
@@ -60,3 +58,2 @@ class HwModule : public RefBase {
-  status_t addInputProfile(const std::string &name,
-                           const audio_config_t *config, audio_devices_t device,
-                           const String8 &address);
+    status_t addInputProfile(const std::string& name, const audio_config_t *config,
+            audio_devices_t device, const String8& address);
@@ -66,2 +63,3 @@ class HwModule : public RefBase {
-  sp<AudioPort> findPortByTagName(const std::string &tagName) const {
-    return mPorts.findByTagName(tagName);
+    sp<PolicyAudioPort> findPortByTagName(const std::string &tagName) const
+    {
+        return findByTagName(mPorts, tagName);
@@ -76,4 +74,2 @@ class HwModule : public RefBase {
-  OutputProfileCollection
-      mOutputProfiles;
-  InputProfileCollection
-      mInputProfiles;
+    OutputProfileCollection mOutputProfiles;
+    InputProfileCollection mInputProfiles;
@@ -81,4 +77,2 @@ class HwModule : public RefBase {
-  DeviceVector
-      mDeclaredDevices;
-  DeviceVector
-      mDynamicDevices;
+    DeviceVector mDeclaredDevices;
+    DeviceVector mDynamicDevices;
@@ -88 +82,2 @@ class HwModule : public RefBase {
-class HwModuleCollection : public Vector<sp<HwModule> > {
+class HwModuleCollection : public Vector<sp<HwModule> >
+{
@@ -95,2 +90,2 @@ class HwModuleCollection : public Vector<sp<HwModule> > {
-  DeviceVector getAvailableDevicesFromModuleName(
-      const char *name, const DeviceVector &availableDevices) const;
+    DeviceVector getAvailableDevicesFromModuleName(const char *name,
+                                                   const DeviceVector &availableDevices) const;
@@ -104 +99,2 @@ class HwModuleCollection : public Vector<sp<HwModule> > {
-                                    const char *address, const char *name,
+                                      const char *address,
+                                      const char *name,
