--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/c1dbe47f-f1e48f34-5af8fef5/services@audiopolicy@common@managerdefinitions@include@DeviceDescriptor.no_comments_merged.h
@@ -2 +1,0 @@
-#include "AudioPort.h"
@@ -17 +15,0 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-<<<<<<< HEAD
@@ -19,7 +17,6 @@ class DeviceDescriptor : public DeviceDescriptorBase,
-||||||| 5af8fef591
-  explicit DeviceDescriptor(audio_devices_t type,
-                            const String8 &tagName = String8(""));
-=======
-  explicit DeviceDescriptor(audio_devices_t type,
-                            const std::string &tagName = "");
->>>>>>> f1e48f349fcc5313e8956b07176ac975b7a0f196
+    DeviceDescriptor(audio_devices_t type, const std::string &tagName,
+            const FormatVector &encodedFormats = FormatVector{});
+    DeviceDescriptor(audio_devices_t type, const std::string &tagName,
+            const std::string &address, const FormatVector &encodedFormats = FormatVector{});
+    DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr, const std::string &tagName = "",
+            const FormatVector &encodedFormats = FormatVector{});
@@ -62,9 +55,0 @@ class DeviceDescriptor : public DeviceDescriptorBase,
- public:
-  DeviceDescriptor(audio_devices_t type, const std::string &tagName,
-                   const FormatVector &encodedFormats = FormatVector{});
-  DeviceDescriptor(audio_devices_t type, const std::string &tagName,
-                   const std::string &address,
-                   const FormatVector &encodedFormats = FormatVector{});
-  DeviceDescriptor(const AudioDeviceTypeAddr &deviceTypeAddr,
-                   const std::string &tagName = "",
-                   const FormatVector &encodedFormats = FormatVector{});
