diff --git a/home/whalien/codebase/python/mergebot-eva/output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_merged.c
index 4873beb..6c3e6e0 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/0ff991b2-8f345047-a9ac6146/tty-term.no_comments_merged.c
@@ -2 +1,0 @@
-#include <curses.h>
@@ -10 +8,0 @@
-#include <vis.h>
@@ -265,3 +263,10 @@ static const struct tty_term_code_entry tty_term_codes[] = {
-    [TTYC_XT] = {TTYCODE_FLAG, "XT"}};
-u_int tty_term_ncodes(void) { return (nitems(tty_term_codes)); }
-static char *tty_term_strip(const char *s) {
+ [TTYC_XT] = { TTYCODE_FLAG, "XT" }
+};
+u_int
+tty_term_ncodes(void)
+{
+ return (nitems(tty_term_codes));
+}
+static char *
+tty_term_strip(const char *s)
+{
@@ -271 +276,2 @@ static char *tty_term_strip(const char *s) {
-  if (strchr(s, '$') == NULL) return (xstrdup(s));
+ if (strchr(s, '$') == NULL)
+  return (xstrdup(s));
@@ -275,3 +281,6 @@ static char *tty_term_strip(const char *s) {
-      while (*ptr != '\0' && *ptr != '>') ptr++;
-      if (*ptr == '>') ptr++;
-      if (*ptr == '\0') break;
+   while (*ptr != '\0' && *ptr != '>')
+    ptr++;
+   if (*ptr == '>')
+    ptr++;
+   if (*ptr == '\0')
+    break;
@@ -280 +289,2 @@ static char *tty_term_strip(const char *s) {
-    if (len == (sizeof buf) - 1) break;
+  if (len == (sizeof buf) - 1)
+   break;
@@ -285 +295,3 @@ static char *tty_term_strip(const char *s) {
-static char *tty_term_override_next(const char *s, size_t *offset) {
+static char *
+tty_term_override_next(const char *s, size_t *offset)
+{
@@ -288 +300,2 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-  if (s[at] == '\0') return (NULL);
+ if (s[at] == '\0')
+  return (NULL);
@@ -300 +313,2 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-    if (n == (sizeof value) - 1) return (NULL);
+  if (n == (sizeof value) - 1)
+   return (NULL);
@@ -309,2 +323,3 @@ static char *tty_term_override_next(const char *s, size_t *offset) {
-void tty_term_apply(struct tty_term *term, const char *capabilities,
-                    int quiet) {
+void
+tty_term_apply(struct tty_term *term, const char *capabilities, int quiet)
+{
@@ -319 +334,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-    if (*s == '\0') continue;
+  if (*s == '\0')
+   continue;
@@ -344 +360,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-      if (strcmp(s, ent->name) != 0) continue;
+   if (strcmp(s, ent->name) != 0)
+    continue;
@@ -354 +371,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-          if (code->type == TTYCODE_STRING) free(code->value.string);
+    if (code->type == TTYCODE_STRING)
+     free(code->value.string);
@@ -360 +378,2 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-          if (errstr != NULL) break;
+    if (errstr != NULL)
+     break;
@@ -373 +392,3 @@ void tty_term_apply(struct tty_term *term, const char *capabilities,
-void tty_term_apply_overrides(struct tty_term *term) {
+void
+tty_term_apply_overrides(struct tty_term *term)
+{
@@ -392 +413,2 @@ void tty_term_apply_overrides(struct tty_term *term) {
-  if (tty_term_has(term, TTYC_SETRGBF) && tty_term_has(term, TTYC_SETRGBB))
+ if (tty_term_has(term, TTYC_SETRGBF) &&
+     tty_term_has(term, TTYC_SETRGBB))
@@ -420,2 +442,4 @@ void tty_term_apply_overrides(struct tty_term *term) {
-struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-                                 u_int ncaps, int *feat, char **cause) {
+struct tty_term *
+tty_term_create(struct tty *tty, char *name, char **caps, u_int ncaps,
+    int *feat, char **cause)
+{
@@ -440 +464,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-    if (namelen == 0) continue;
+  if (namelen == 0)
+   continue;
@@ -444,2 +469,4 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-      if (strncmp(ent->name, caps[i], namelen) != 0) continue;
-      if (ent->name[namelen] != '\0') continue;
+   if (strncmp(ent->name, caps[i], namelen) != 0)
+    continue;
+   if (ent->name[namelen] != '\0')
+    continue;
@@ -496 +523,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-      (!tty_term_has(term, TTYC_SETRGBF) || !tty_term_has(term, TTYC_SETRGBB)))
+     (!tty_term_has(term, TTYC_SETRGBF) ||
+     !tty_term_has(term, TTYC_SETRGBB)))
@@ -498 +526,2 @@ struct tty_term *tty_term_create(struct tty *tty, char *name, char **caps,
-  if (tty_apply_features(term, *feat)) tty_term_apply_overrides(term);
+ if (tty_apply_features(term, *feat))
+  tty_term_apply_overrides(term);
@@ -506 +535,3 @@ error:
-void tty_term_free(struct tty_term *term) {
+void
+tty_term_free(struct tty_term *term)
+{
@@ -518,2 +549,4 @@ void tty_term_free(struct tty_term *term) {
-int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-                       char **cause) {
+int
+tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
+    char **cause)
+{
@@ -528 +561,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        xasprintf(cause, "can't use hardcopy terminal: %s", name);
+   xasprintf(cause, "can't use hardcopy terminal: %s",
+       name);
@@ -531 +565,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        xasprintf(cause, "missing or unsuitable terminal: %s", name);
+   xasprintf(cause, "missing or unsuitable terminal: %s",
+       name);
@@ -551 +586,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (s == NULL || s == (char *)-1) continue;
+   if (s == NULL || s == (char *)-1)
+    continue;
@@ -555 +591,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (n == -1 || n == -2) continue;
+   if (n == -1 || n == -2)
+    continue;
@@ -561 +598,2 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-        if (n == -1) continue;
+   if (n == -1)
+    continue;
@@ -578 +616,3 @@ int tty_term_read_list(const char *name, int fd, char ***caps, u_int *ncaps,
-void tty_term_free_list(char **caps, u_int ncaps) {
+void
+tty_term_free_list(char **caps, u_int ncaps)
+{
@@ -580 +620,2 @@ void tty_term_free_list(char **caps, u_int ncaps) {
-  for (i = 0; i < ncaps; i++) free(caps[i]);
+ for (i = 0; i < ncaps; i++)
+  free(caps[i]);
@@ -583 +624,3 @@ void tty_term_free_list(char **caps, u_int ncaps) {
-int tty_term_has(struct tty_term *term, enum tty_code_code code) {
+int
+tty_term_has(struct tty_term *term, enum tty_code_code code)
+{
@@ -586,2 +629,5 @@ int tty_term_has(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_string(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return ("");
+const char *
+tty_term_string(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return ("");
@@ -592,2 +638,3 @@ const char *tty_term_string(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-                              int a) {
+const char *
+tty_term_string_i(struct tty_term *term, enum tty_code_code code, int a)
+{
@@ -596 +643,2 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
@@ -599,8 +647,3 @@ const char *tty_term_string_i(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-                               int a, int b) {
-<<<<<<< HEAD
-  return (
-      tparm((char *)tty_term_string(term, code), a, b, 0, 0, 0, 0, 0, 0, 0));
-|||||||
-  return (tparm((char *)tty_term_string(term, code), a, b));
-=======
+const char *
+tty_term_string_ii(struct tty_term *term, enum tty_code_code code, int a, int b)
+{
@@ -609 +652,2 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
@@ -611 +654,0 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
->>>>>>> 8f34504736cf3547992c4ba948c1e65f3813715c
@@ -613,2 +656,4 @@ const char *tty_term_string_ii(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-                                int a, int b, int c) {
+const char *
+tty_term_string_iii(struct tty_term *term, enum tty_code_code code, int a, int b,
+    int c)
+{
@@ -617 +662,2 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
@@ -620,2 +666,3 @@ const char *tty_term_string_iii(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-                              const char *a) {
+const char *
+tty_term_string_s(struct tty_term *term, enum tty_code_code code, const char *a)
+{
@@ -624 +671,2 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
@@ -627,8 +675,4 @@ const char *tty_term_string_s(struct tty_term *term, enum tty_code_code code,
-const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-                               const char *a, const char *b) {
-<<<<<<< HEAD
-  return (tparm((char *)tty_term_string(term, code), (long)a, (long)b, 0, 0, 0,
-                0, 0, 0, 0));
-|||||||
-  return (tparm((char *)tty_term_string(term, code), a, b));
-=======
+const char *
+tty_term_string_ss(struct tty_term *term, enum tty_code_code code, const char *a,
+    const char *b)
+{
@@ -637 +681,2 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-  if (s == NULL) fatalx("could not expand %s", tty_term_codes[code].name);
+ if (s == NULL)
+  fatalx("could not expand %s", tty_term_codes[code].name);
@@ -639 +683,0 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
->>>>>>> 8f34504736cf3547992c4ba948c1e65f3813715c
@@ -641,2 +685,5 @@ const char *tty_term_string_ss(struct tty_term *term, enum tty_code_code code,
-int tty_term_number(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return (0);
+int
+tty_term_number(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return (0);
@@ -647,3 +694,7 @@ int tty_term_number(struct tty_term *term, enum tty_code_code code) {
-int tty_term_flag(struct tty_term *term, enum tty_code_code code) {
-  if (!tty_term_has(term, code)) return (0);
-  if (term->codes[code].type != TTYCODE_FLAG) fatalx("not a flag: %d", code);
+int
+tty_term_flag(struct tty_term *term, enum tty_code_code code)
+{
+ if (!tty_term_has(term, code))
+  return (0);
+ if (term->codes[code].type != TTYCODE_FLAG)
+  fatalx("not a flag: %d", code);
@@ -652 +703,3 @@ int tty_term_flag(struct tty_term *term, enum tty_code_code code) {
-const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
+const char *
+tty_term_describe(struct tty_term *term, enum tty_code_code code)
+{
@@ -657,2 +710,2 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: [missing]", code,
-                tty_term_codes[code].name);
+  xsnprintf(s, sizeof s, "%4u: %s: [missing]",
+      code, tty_term_codes[code].name);
@@ -663,2 +716,3 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (string) %s", code,
-                tty_term_codes[code].name, out);
+  xsnprintf(s, sizeof s, "%4u: %s: (string) %s",
+      code, tty_term_codes[code].name,
+      out);
@@ -667,2 +721,3 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (number) %d", code,
-                tty_term_codes[code].name, term->codes[code].value.number);
+  xsnprintf(s, sizeof s, "%4u: %s: (number) %d",
+      code, tty_term_codes[code].name,
+      term->codes[code].value.number);
@@ -671,2 +726,2 @@ const char *tty_term_describe(struct tty_term *term, enum tty_code_code code) {
-      xsnprintf(s, sizeof s, "%4u: %s: (flag) %s", code,
-                tty_term_codes[code].name,
+  xsnprintf(s, sizeof s, "%4u: %s: (flag) %s",
+      code, tty_term_codes[code].name,
