--- a/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@common@radix_partitioning.no_comments_mergegen.cpp
+++ b/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@common@radix_partitioning.no_comments_merged.cpp
@@ -55,0 +56 @@ idx_t RadixPartitioning::Select(Vector &hashes, const SelectionVector *sel, idx_
+struct ComputePartitionIndicesFunctor {
@@ -89 +90 @@ void RadixPartitionedColumnData::InitializeAppendStateInternal(PartitionedColumn
-for (idx_t i = 0; i < num_partitions; i++) {
+  state.partition_append_states.emplace_back(make_uniq<ColumnDataAppendState>());
@@ -129 +130 @@ void RadixPartitionedTupleData::InitializeAppendStateInternal(PartitionedTupleDa
-  state.partition_pin_states.emplace_back(make_unique<TupleDataPinState>());
+  state.partition_pin_states.emplace_back(make_uniq<TupleDataPinState>());
