diff --git a/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@execution@aggregate_hashtable.no_comments_conflict.cpp b/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@execution@aggregate_hashtable.no_comments_merged.cpp
index 4beb9d9..d5626c1 100644
--- a/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@execution@aggregate_hashtable.no_comments_conflict.cpp
+++ b/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@execution@aggregate_hashtable.no_comments_merged.cpp
@@ -45 +45 @@ GroupedAggregateHashTable::GroupedAggregateHashTable(ClientContext &context, All
- data_collection = make_unique<TupleDataCollection>(buffer_manager, layout);
+ data_collection = make_uniq<TupleDataCollection>(buffer_manager, layout);
@@ -64,6 +63,0 @@ GroupedAggregateHashTable::GroupedAggregateHashTable(ClientContext &context, All
-<<<<<<< HEAD
-||||||| e8b7132504
- string_heap = make_unique<RowDataCollection>(buffer_manager, (idx_t)Storage::BLOCK_SIZE, 1, true);
-=======
- string_heap = make_uniq<RowDataCollection>(buffer_manager, (idx_t)Storage::BLOCK_SIZE, 1, true);
->>>>>>> 5b9bcdfb
@@ -478 +472 @@ void GroupedAggregateHashTable::Partition(vector<GroupedAggregateHashTable *> &p
-     make_unique<RadixPartitionedTupleData>(buffer_manager, layout, radix_bits, layout.ColumnCount() - 1);
+     make_uniq<RadixPartitionedTupleData>(buffer_manager, layout, radix_bits, layout.ColumnCount() - 1);
