diff --git a/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@common@radix_partitioning.no_comments_conflict.cpp b/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@common@radix_partitioning.no_comments_merged.cpp
index 2c6a604..fbd9b86 100644
--- a/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@common@radix_partitioning.no_comments_conflict.cpp
+++ b/./output/duckdb/b2ae2d3a-5b9bcdfb-e8b71325/src@common@radix_partitioning.no_comments_merged.cpp
@@ -56 +55,0 @@ idx_t RadixPartitioning::Select(Vector &hashes, const SelectionVector *sel, idx_
-<<<<<<< HEAD
@@ -58,37 +56,0 @@ struct ComputePartitionIndicesFunctor {
-||||||| e8b7132504
-template <idx_t radix_bits>
-static void InitPartitions(BufferManager &buffer_manager, vector<unique_ptr<RowDataCollection>> &partition_collections,
-                           RowDataBlock *partition_blocks[], vector<BufferHandle> &partition_handles,
-                           data_ptr_t partition_ptrs[], idx_t block_capacity, idx_t row_width) {
- using CONSTANTS = RadixPartitioningConstants<radix_bits>;
- partition_collections.reserve(CONSTANTS::NUM_PARTITIONS);
- partition_handles.reserve(CONSTANTS::NUM_PARTITIONS);
- for (idx_t i = 0; i < CONSTANTS::NUM_PARTITIONS; i++) {
-  partition_collections.push_back(make_unique<RowDataCollection>(buffer_manager, block_capacity, row_width));
-  partition_blocks[i] = &partition_collections[i]->CreateBlock();
-  partition_handles.push_back(buffer_manager.Pin(partition_blocks[i]->block));
-  if (partition_ptrs) {
-   partition_ptrs[i] = partition_handles[i].Ptr();
-  }
- }
-}
-struct PartitionFunctor {
-=======
-template <idx_t radix_bits>
-static void InitPartitions(BufferManager &buffer_manager, vector<unique_ptr<RowDataCollection>> &partition_collections,
-                           RowDataBlock *partition_blocks[], vector<BufferHandle> &partition_handles,
-                           data_ptr_t partition_ptrs[], idx_t block_capacity, idx_t row_width) {
- using CONSTANTS = RadixPartitioningConstants<radix_bits>;
- partition_collections.reserve(CONSTANTS::NUM_PARTITIONS);
- partition_handles.reserve(CONSTANTS::NUM_PARTITIONS);
- for (idx_t i = 0; i < CONSTANTS::NUM_PARTITIONS; i++) {
-  partition_collections.push_back(make_uniq<RowDataCollection>(buffer_manager, block_capacity, row_width));
-  partition_blocks[i] = &partition_collections[i]->CreateBlock();
-  partition_handles.push_back(buffer_manager.Pin(partition_blocks[i]->block));
-  if (partition_ptrs) {
-   partition_ptrs[i] = partition_handles[i].Ptr();
-  }
- }
-}
-struct PartitionFunctor {
->>>>>>> 5b9bcdfb
@@ -128,5 +89,0 @@ void RadixPartitionedColumnData::InitializeAppendStateInternal(PartitionedColumn
-<<<<<<< HEAD
-  state.partition_append_states.emplace_back(make_unique<ColumnDataAppendState>());
-||||||| e8b7132504
-  state.partition_append_states.emplace_back(make_unique<ColumnDataAppendState>());
-=======
@@ -134 +90,0 @@ void RadixPartitionedColumnData::InitializeAppendStateInternal(PartitionedColumn
->>>>>>> 5b9bcdfb
@@ -174 +130 @@ void RadixPartitionedTupleData::InitializeAppendStateInternal(PartitionedTupleDa
-  state.partition_pin_states.emplace_back(make_unique<TupleDataPinState>());
+  state.partition_pin_states.emplace_back(make_uniq<TupleDataPinState>());
