diff --git a/./output/art/40c2e498-29579cc5-38b68d85/libnativeloader@library_namespaces.no_comments_conflict.cpp b/./output/art/40c2e498-29579cc5-38b68d85/libnativeloader@library_namespaces.no_comments_merged.cpp
index 2eb11fc..9602318 100644
--- a/./output/art/40c2e498-29579cc5-38b68d85/libnativeloader@library_namespaces.no_comments_conflict.cpp
+++ b/./output/art/40c2e498-29579cc5-38b68d85/libnativeloader@library_namespaces.no_comments_merged.cpp
@@ -134 +133,0 @@ Result<NativeLoaderNamespace*> LibraryNamespaces::Create(JNIEnv* env, uint32_t t
-<<<<<<< HEAD
@@ -136,5 +134,0 @@ Result<NativeLoaderNamespace*> LibraryNamespaces::Create(JNIEnv* env, uint32_t t
-||||||| 38b68d8557
-    } else if (apk_origin == APK_ORIGIN_PRODUCT && is_product_vndk_version_defined()) {
-=======
-    } else if (apk_origin == APK_ORIGIN_PRODUCT) {
->>>>>>> 29579cc5
@@ -247 +240,0 @@ Result<NativeLoaderNamespace*> LibraryNamespaces::Create(JNIEnv* env, uint32_t t
-<<<<<<< HEAD
@@ -252,8 +244,0 @@ Result<NativeLoaderNamespace*> LibraryNamespaces::Create(JNIEnv* env, uint32_t t
-||||||| 38b68d8557
-    auto target_ns = system_ns;
-    if (is_product_vndk_version_defined()) {
-      target_ns = NativeLoaderNamespace::GetExportedNamespace(kProductNamespaceName, is_bridged);
-    }
-=======
-    auto target_ns = NativeLoaderNamespace::GetExportedNamespace(kProductNamespaceName, is_bridged);
->>>>>>> 29579cc5
