diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/40c2e498-29579cc5-38b68d85/libnativeloader@public_libraries.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/art/40c2e498-29579cc5-38b68d85/libnativeloader@public_libraries.no_comments_merged.cpp
index f92e7f7..bd6dee4 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/40c2e498-29579cc5-38b68d85/libnativeloader@public_libraries.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/40c2e498-29579cc5-38b68d85/libnativeloader@public_libraries.no_comments_merged.cpp
@@ -23 +22,0 @@ using internal::ConfigEntry;
-using internal::ParseApexLibrariesConfig;
@@ -24,0 +24 @@ using internal::ParseConfig;
+using internal::ParseApexLibrariesConfig;
@@ -105,2 +105 @@ void ReadExtensionLibraries(const char* dirname, std::vector<std::string>* sonam
-                           config_file_path.c_str(),
-                           ret.error().message().c_str());
+                           config_file_path.c_str(), ret.error().message().c_str());
@@ -124,2 +123 @@ static std::string InitDefaultPublicLibraries(bool for_preload) {
-                     config_file.c_str(),
-                     sonames.error().message().c_str());
+                     config_file.c_str(), sonames.error().message().c_str());
@@ -156 +153,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -158,4 +154,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-  if (is_product_vndk_version_defined()) {
-=======
->>>>>>> 29579cc5f300a30eafd2b0ea39b7181b5cf31dce
@@ -162,0 +156 @@ static std::string InitProductPublicLibraries() {
+  }
@@ -171 +164,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -175,6 +167,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-    if (!is_product_vndk_version_defined()) {
-      ReadExtensionLibraries("/product/etc", &sonames);
-    }
-=======
->>>>>>> 29579cc5f300a30eafd2b0ea39b7181b5cf31dce
@@ -217 +203,0 @@ static std::string InitProductPublicLibraries() {
-<<<<<<< HEAD
@@ -222,7 +207,0 @@ static std::string InitProductPublicLibraries() {
-|||||||
-    if (!is_product_vndk_version_defined()) {
-      ALOGD("InitLlndkLibrariesProduct: No product VNDK version defined");
-      return "";
-    }
-=======
->>>>>>> 29579cc5f300a30eafd2b0ea39b7181b5cf31dce
@@ -321,2 +300 @@ static std::string InitProductPublicLibraries() {
-    return ApexLibrariesConfigLine{
-        std::move(tokens[0]), std::move(tokens[1]), std::move(tokens[2])};
+  return ApexLibrariesConfigLine{std::move(tokens[0]), std::move(tokens[1]), std::move(tokens[2])};
@@ -425,2 +403 @@ Result<std::vector<std::string>> ParseConfig(
-    if (entry.bitness == ONLY_32)
-      continue;
+    if (entry.bitness == ONLY_32) continue;
@@ -428,2 +405 @@ Result<std::vector<std::string>> ParseConfig(
-    if (entry.bitness == ONLY_64)
-      continue;
+    if (entry.bitness == ONLY_64) continue;
@@ -432,2 +408 @@ Result<std::vector<std::string>> ParseConfig(
-    if (entry.soname == "libRS.so")
-      continue;
+    if (entry.soname == "libRS.so") continue;
@@ -445,2 +420 @@ Result<std::vector<std::string>> ParseConfig(
-Result<std::map<std::string, std::string>> ParseApexLibrariesConfig(const std::string& file_content,
-                                                                    const std::string& tag) {
+Result<std::map<std::string, std::string>> ParseApexLibrariesConfig(const std::string& file_content, const std::string& tag) {
