diff --git a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@main_surfaceflinger.no_comments_mergegen.cpp b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@main_surfaceflinger.no_comments_merged.cpp
index ba684d7..858d510 100644
--- a/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@main_surfaceflinger.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/a9c51987-01759208-7c11ca12/services@surfaceflinger@main_surfaceflinger.no_comments_merged.cpp
@@ -88 +88,3 @@ int main(int, char**) {
-if (FlagManager::getInstance().misc1()) { composerAIDL->setMinSchedulerPolicy(SCHED_FIFO, newPriority); }
+    if (FlagManager::getInstance().misc1()) {
+        composerAIDL->setMinSchedulerPolicy(SCHED_FIFO, newPriority);
+    }
