diff --git a/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/media@utils@ServiceUtilities.no_comments_conflict.cpp b/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/media@utils@ServiceUtilities.no_comments_merged.cpp
index 7dd68ea..cc9ec2c 100644
--- a/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/media@utils@ServiceUtilities.no_comments_conflict.cpp
+++ b/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/media@utils@ServiceUtilities.no_comments_merged.cpp
@@ -20,0 +21 @@ static const String16 sCallAudioInterception("android.permission.CALL_AUDIO_INTE
+static const String16 sAndroidPermissionBluetoothConnect("android.permission.BLUETOOTH_CONNECT");
@@ -286 +286,0 @@ status_t checkIMemory(const sp<IMemory>& iMemory)
-<<<<<<< HEAD
@@ -310,26 +309,0 @@ void anonymizeBluetoothAddress(char *address) {
-||||||| 92765b5cf0
-bool mustAnonymizeBluetoothAddress(
-        const AttributionSourceState& attributionSource, const String16& caller) {
-    uid_t uid = VALUE_OR_FATAL(aidl2legacy_int32_t_uid_t(attributionSource.uid));
-    if (isAudioServerOrSystemServerUid(uid)) {
-        return false;
-    }
-    const std::optional<AttributionSourceState> resolvedAttributionSource =
-            resolveAttributionSource(attributionSource);
-    if (!resolvedAttributionSource.has_value()) {
-        return true;
-    }
-    permission::PermissionChecker permissionChecker;
-    return permissionChecker.checkPermissionForPreflightFromDatasource(
-            sAndroidPermissionBluetoothConnect, resolvedAttributionSource.value(), caller,
-            AppOpsManager::OP_BLUETOOTH_CONNECT)
-                != permission::PermissionChecker::PERMISSION_GRANTED;
-}
-void anonymizeBluetoothAddress(char *address) {
-    if (address == nullptr || strlen(address) != strlen("AA:BB:CC:DD:EE:FF")) {
-        return;
-    }
-    memcpy(address, "XX:XX:XX:XX", strlen("XX:XX:XX:XX"));
-}
-=======
->>>>>>> 6ddf1d0a
