--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
@@ -14,2 +14 @@
-    ({ \
-        auto _tmp = (x); \
+    ({ auto _tmp = (x); \
@@ -17,2 +16 @@
-        std::move(_tmp.value()); \
-    })
+       std::move(_tmp.value()); })
@@ -28 +26 @@ namespace android {
-using aidl_utils::binderStatusFromStatusT;
+namespace audiopolicy_flags = android::media::audiopolicy;
@@ -29,0 +28 @@ using binder::Status;
+using aidl_utils::binderStatusFromStatusT;
@@ -1252,0 +1315,13 @@ Status AudioPolicyService::isDirectOutputSupported(
+template <typename Port>
+void anonymizePortBluetoothAddress(Port *port) {
+    if (port->type != AUDIO_PORT_TYPE_DEVICE) {
+        return;
+    }
+    if (!(audio_is_a2dp_device(port->ext.device.type)
+            || audio_is_ble_device(port->ext.device.type)
+            || audio_is_bluetooth_sco_device(port->ext.device.type)
+            || audio_is_hearing_aid_out_device(port->ext.device.type))) {
+        return;
+    }
+    anonymizeBluetoothAddress(port->ext.device.address);
+}
@@ -1269,7 +1343,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-<<<<<<< HEAD
-    const AttributionSourceState attributionSource = getCallingAttributionSource();
-||||||| 92765b5cf0
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
@@ -1277,6 +1344,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-=======
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
->>>>>>> 6ddf1d0a261b924e08e6de7d404230e1ae4ec4ac
@@ -1292 +1354,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-<<<<<<< HEAD
@@ -1299,8 +1360,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-||||||| 92765b5cf0
-        if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-            for (size_t i = 0; i < numPortsReq; ++i) {
-                anonymizePortBluetoothAddress(&ports[i]);
-            }
-        }
-=======
->>>>>>> 6ddf1d0a261b924e08e6de7d404230e1ae4ec4ac
@@ -1326,7 +1380,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-<<<<<<< HEAD
-    const AttributionSourceState attributionSource = getCallingAttributionSource();
-||||||| 92765b5cf0
-        Mutex::Autolock _l(mLock);
-        if (mAudioPolicyManager == NULL) {
-            return binderStatusFromStatusT(NO_INIT);
-        }
@@ -1334,6 +1381,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-=======
-        Mutex::Autolock _l(mLock);
-        if (mAudioPolicyManager == NULL) {
-            return binderStatusFromStatusT(NO_INIT);
-        }
->>>>>>> 6ddf1d0a261b924e08e6de7d404230e1ae4ec4ac
@@ -1347,4 +1388,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-<<<<<<< HEAD
-    }
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        anonymizePortBluetoothAddress(&port);
@@ -1352 +1389,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-||||||| 92765b5cf0
@@ -1356,2 +1392,0 @@ Status AudioPolicyService::getAudioPort(int portId, media::AudioPortFw* _aidl_re
-=======
->>>>>>> 6ddf1d0a261b924e08e6de7d404230e1ae4ec4ac
@@ -1406 +1444,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-<<<<<<< HEAD
@@ -1408,12 +1445,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-||||||| 92765b5cf0
-            Mutex::Autolock _l(mLock);
-            if (mAudioPolicyManager == NULL) {
-                return binderStatusFromStatusT(NO_INIT);
-            }
-            const AttributionSourceState attributionSource = getCallingAttributionSource();
-=======
-            Mutex::Autolock _l(mLock);
-            if (mAudioPolicyManager == NULL) {
-                return binderStatusFromStatusT(NO_INIT);
-            }
->>>>>>> 6ddf1d0a261b924e08e6de7d404230e1ae4ec4ac
@@ -1429,11 +1454,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-<<<<<<< HEAD
-    }
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        for (size_t i = 0; i < numPatchesReq; ++i) {
-            for (size_t j = 0; j < patches[i].num_sources; ++j) {
-                anonymizePortBluetoothAddress(&patches[i].sources[j]);
-            }
-            for (size_t j = 0; j < patches[i].num_sinks; ++j) {
-                anonymizePortBluetoothAddress(&patches[i].sinks[j]);
-            }
-        }
@@ -1441 +1455,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-||||||| 92765b5cf0
@@ -1452,2 +1465,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-=======
->>>>>>> 6ddf1d0a261b924e08e6de7d404230e1ae4ec4ac
