diff --git a/./output/art/3557b865-344f4e7e-f5107e92/runtime@gc@heap.no_comments_conflict.cc b/./output/art/3557b865-344f4e7e-f5107e92/runtime@gc@heap.no_comments_merged.cc
index f6df308..7ae9f87 100644
--- a/./output/art/3557b865-344f4e7e-f5107e92/runtime@gc@heap.no_comments_conflict.cc
+++ b/./output/art/3557b865-344f4e7e-f5107e92/runtime@gc@heap.no_comments_merged.cc
@@ -3172,22 +3171,0 @@ void Heap::RequestCollectorTransition(CollectorType desired_collector_type, uint
-<<<<<<< HEAD
-||||||| f5107e929f
-    size_t num_bytes_allocated_since_gc = GetBytesAllocated() - num_bytes_alive_after_gc_;
-    if (num_bytes_allocated_since_gc <
-        (UnsignedDifference(target_footprint_.load(std::memory_order_relaxed),
-                            num_bytes_alive_after_gc_)/4)
-        && !kStressCollectorTransition
-        && !IsLowMemoryMode()) {
-      return;
-    }
-=======
-  }
-  if (collector_type_ == kCollectorTypeCC || collector_type_ == kCollectorTypeCMC) {
-    size_t num_bytes_allocated_since_gc = GetBytesAllocated() - num_bytes_alive_after_gc_;
-    if (num_bytes_allocated_since_gc <
-        (UnsignedDifference(target_footprint_.load(std::memory_order_relaxed),
-                            num_bytes_alive_after_gc_)/4)
-        && !kStressCollectorTransition
-        && !IsLowMemoryMode()) {
-      return;
-    }
->>>>>>> 344f4e7e
