--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -7 +7 @@
-struct android_namespace_t
+struct android_namespace_t;
@@ -54,3 +52,2 @@ struct android_namespace_t
-        void sendGpuStatsLocked(GpuStatsInfo::Api api, bool isDriverLoaded,
-                                int64_t driverLoadingTime);
-        GraphicsEnv()
+    void sendGpuStatsLocked(GpuStatsInfo::Api api, bool isDriverLoaded, int64_t driverLoadingTime);
+    GraphicsEnv() = default;
