diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_merged.cpp
index 74d4242..8b89877 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_merged.cpp
@@ -38 +38,2 @@ static int do_android_unload_sphal_library(void* dso) {
-Loader::driver_t::driver_t(void* gles) {
+Loader::driver_t::driver_t(void* gles)
+{
@@ -40 +41,2 @@ Loader::driver_t::driver_t(void* gles) {
-    for (size_t i = 1; i < NELEM(dso); i++) dso[i] = nullptr;
+    for (size_t i=1 ; i<NELEM(dso) ; i++)
+        dso[i] = nullptr;
@@ -42 +44,2 @@ Loader::driver_t::driver_t(void* gles) {
-Loader::driver_t::~driver_t() {
+Loader::driver_t::~driver_t()
+{
@@ -50 +53,2 @@ Loader::driver_t::~driver_t() {
-int Loader::driver_t::set(void* hnd, int32_t api) {
+int Loader::driver_t::set(void* hnd, int32_t api)
+{
@@ -66,2 +70,6 @@ int Loader::driver_t::set(void* hnd, int32_t api) {
-Loader::Loader() : getProcAddress(nullptr) {}
-Loader::~Loader() {}
+Loader::Loader()
+    : getProcAddress(nullptr)
+{
+}
+Loader::~Loader() {
+}
@@ -94 +101,0 @@ static bool should_unload_system_driver(egl_connection_t* cnx) {
-<<<<<<< HEAD
@@ -100,4 +106,0 @@ static bool should_unload_system_driver(egl_connection_t* cnx) {
-|||||||
-=======
-        if (!(cnx->angleLoaded && android::GraphicsEnv::getInstance().shouldUseSystemAngle())) {
->>>>>>> 87425707b37cce4759e29fc9c77369f771ec4319
@@ -106 +108,0 @@ static bool should_unload_system_driver(egl_connection_t* cnx) {
-}
@@ -222 +224,2 @@ void* Loader::open(egl_connection_t* cnx) {
-    LOG_ALWAYS_FATAL_IF(!cnx->libEgl, "couldn't load system EGL wrapper libraries");
+    LOG_ALWAYS_FATAL_IF(!cnx->libEgl,
+            "couldn't load system EGL wrapper libraries");
@@ -229 +232,2 @@ void* Loader::open(egl_connection_t* cnx) {
-void Loader::close(egl_connection_t* cnx) {
+void Loader::close(egl_connection_t* cnx)
+{
@@ -235 +239,3 @@ void Loader::close(egl_connection_t* cnx) {
-void Loader::init_api(void* dso, char const* const* api, char const* const* ref_api,
+void Loader::init_api(void* dso,
+        char const * const * api,
+        char const * const * ref_api,
@@ -237 +243,2 @@ void Loader::init_api(void* dso, char const* const* api, char const* const* ref_
-                      getProcAddressType getProcAddress) {
+        getProcAddressType getProcAddress)
+{
@@ -309,2 +316,2 @@ static void* load_system_driver(const char* kind, const char* suffix, const bool
-        static bool find(std::string& result, const std::string& pattern, const char* const search,
-                         bool exact) {
+        static bool find(std::string& result,
+                const std::string& pattern, const char* const search, bool exact) {
@@ -353 +360,2 @@ static void* load_system_driver(const char* kind, const char* suffix, const bool
-    void* dso = do_android_load_sphal_library(driver_absolute_path, RTLD_NOW | RTLD_LOCAL);
+    void* dso = do_android_load_sphal_library(driver_absolute_path,
+                                              RTLD_NOW | RTLD_LOCAL);
@@ -484 +492,2 @@ void Loader::initialize_api(void* dso, egl_connection_t* cnx, uint32_t mask) {
-        ALOGE_IF(!getProcAddress, "can't find eglGetProcAddress() in EGL driver library");
+        ALOGE_IF(!getProcAddress,
+                "can't find eglGetProcAddress() in EGL driver library");
@@ -505,3 +514,2 @@ void Loader::initialize_api(void* dso, egl_connection_t* cnx, uint32_t mask) {
-                 (__eglMustCastToProperFunctionPointerType*)&cnx
-                         ->hooks[egl_connection_t::GLESv1_INDEX]
-                         ->gl,
+            (__eglMustCastToProperFunctionPointerType*)
+                &cnx->hooks[egl_connection_t::GLESv1_INDEX]->gl,
@@ -512,3 +520,2 @@ void Loader::initialize_api(void* dso, egl_connection_t* cnx, uint32_t mask) {
-                 (__eglMustCastToProperFunctionPointerType*)&cnx
-                         ->hooks[egl_connection_t::GLESv2_INDEX]
-                         ->gl,
+            (__eglMustCastToProperFunctionPointerType*)
+                &cnx->hooks[egl_connection_t::GLESv2_INDEX]->gl,
