--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_unsecure_plugin_registry.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_unsecure_plugin_registry.no_comments_merged.cc
@@ -34,0 +35,2 @@ void grpc_lb_policy_weighted_target_shutdown(void);
+void grpc_lb_policy_xds_routing_init(void);
+void grpc_lb_policy_xds_routing_shutdown(void);
@@ -54,3 +58,14 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_client_channel_init, grpc_client_channel_shutdown);
-  grpc_register_plugin(grpc_inproc_plugin_init, grpc_inproc_plugin_shutdown);
-  grpc_register_plugin(grpc_resolver_fake_init, grpc_resolver_fake_shutdown);
+  grpc_register_plugin(grpc_client_channel_init,
+                       grpc_client_channel_shutdown);
+  grpc_register_plugin(grpc_inproc_plugin_init,
+                       grpc_inproc_plugin_shutdown);
+  grpc_register_plugin(grpc_resolver_dns_ares_init,
+                       grpc_resolver_dns_ares_shutdown);
+  grpc_register_plugin(grpc_resolver_dns_native_init,
+                       grpc_resolver_dns_native_shutdown);
+  grpc_register_plugin(grpc_resolver_sockaddr_init,
+                       grpc_resolver_sockaddr_shutdown);
+  grpc_register_plugin(grpc_resolver_fake_init,
+                       grpc_resolver_fake_shutdown);
+  grpc_register_plugin(grpc_resolver_xds_init,
+                       grpc_resolver_xds_shutdown);
@@ -59,10 +74,6 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_lb_policy_cds_init, grpc_lb_policy_cds_shutdown);
-<<<<<<< HEAD
-  grpc_register_plugin(grpc_lb_policy_xds_init, grpc_lb_policy_xds_shutdown);
-  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
-                       grpc_lb_policy_xds_routing_shutdown);
-|||||||
-  grpc_register_plugin(grpc_lb_policy_xds_init, grpc_lb_policy_xds_shutdown);
-=======
-  grpc_register_plugin(grpc_lb_policy_eds_init, grpc_lb_policy_eds_shutdown);
-  grpc_register_plugin(grpc_lb_policy_lrs_init, grpc_lb_policy_lrs_shutdown);
+  grpc_register_plugin(grpc_lb_policy_cds_init,
+                       grpc_lb_policy_cds_shutdown);
+  grpc_register_plugin(grpc_lb_policy_eds_init,
+                       grpc_lb_policy_eds_shutdown);
+  grpc_register_plugin(grpc_lb_policy_lrs_init,
+                       grpc_lb_policy_lrs_shutdown);
@@ -73 +84,2 @@ void grpc_register_built_in_plugins(void) {
->>>>>>> 9e82d59cd770e2c5c2b899bf44d943bb6b81cfbd
+  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
+                       grpc_lb_policy_xds_routing_shutdown);
@@ -78,7 +89,0 @@ void grpc_register_built_in_plugins(void) {
-  grpc_register_plugin(grpc_resolver_dns_ares_init,
-                       grpc_resolver_dns_ares_shutdown);
-  grpc_register_plugin(grpc_resolver_dns_native_init,
-                       grpc_resolver_dns_native_shutdown);
-  grpc_register_plugin(grpc_resolver_sockaddr_init,
-                       grpc_resolver_sockaddr_shutdown);
-  grpc_register_plugin(grpc_resolver_xds_init, grpc_resolver_xds_shutdown);
