--- a/./output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_mergegen.cc
+++ b/./output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_merged.cc
@@ -19 +19,10 @@ void grpc_lb_policy_cds_shutdown(void);
-void grpc_lb_policy_xds_routing_init(void); void grpc_lb_policy_xds_routing_shutdown(void); void grpc_lb_policy_lrs_init(void); void grpc_lb_policy_lrs_shutdown(void); void grpc_lb_policy_priority_init(void); void grpc_lb_policy_priority_shutdown(void); void grpc_lb_policy_weighted_target_init(void); void grpc_lb_policy_weighted_target_shutdown(void);
+void grpc_lb_policy_eds_init(void);
+void grpc_lb_policy_eds_shutdown(void);
+void grpc_lb_policy_lrs_init(void);
+void grpc_lb_policy_lrs_shutdown(void);
+void grpc_lb_policy_priority_init(void);
+void grpc_lb_policy_priority_shutdown(void);
+void grpc_lb_policy_weighted_target_init(void);
+void grpc_lb_policy_weighted_target_shutdown(void);
+void grpc_lb_policy_xds_routing_init(void);
+void grpc_lb_policy_xds_routing_shutdown(void);
@@ -59 +68,10 @@ void grpc_register_built_in_plugins(void) {
-grpc_register_plugin(grpc_lb_policy_xds_routing_init, grpc_lb_policy_xds_routing_shutdown); grpc_register_plugin(grpc_lb_policy_lrs_init, grpc_lb_policy_lrs_shutdown); grpc_register_plugin(grpc_lb_policy_priority_init, grpc_lb_policy_priority_shutdown); grpc_register_plugin(grpc_lb_policy_weighted_target_init, grpc_lb_policy_weighted_target_shutdown);
+  grpc_register_plugin(grpc_lb_policy_eds_init,
+                       grpc_lb_policy_eds_shutdown);
+  grpc_register_plugin(grpc_lb_policy_lrs_init,
+                       grpc_lb_policy_lrs_shutdown);
+  grpc_register_plugin(grpc_lb_policy_priority_init,
+                       grpc_lb_policy_priority_shutdown);
+  grpc_register_plugin(grpc_lb_policy_weighted_target_init,
+                       grpc_lb_policy_weighted_target_shutdown);
+  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
+                       grpc_lb_policy_xds_routing_shutdown);
