--- a/./output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_conflict.cc
+++ b/./output/grpc/d9611cf1-9e82d59c-7126ffe0/src@core@plugin_registry@grpc_plugin_registry.no_comments_merged.cc
@@ -19,9 +18,0 @@ void grpc_lb_policy_cds_shutdown(void);
-<<<<<<< HEAD
-void grpc_lb_policy_xds_init(void);
-void grpc_lb_policy_xds_shutdown(void);
-void grpc_lb_policy_xds_routing_init(void);
-void grpc_lb_policy_xds_routing_shutdown(void);
-||||||| 7126ffe0eb
-void grpc_lb_policy_xds_init(void);
-void grpc_lb_policy_xds_shutdown(void);
-=======
@@ -36 +27,2 @@ void grpc_lb_policy_weighted_target_shutdown(void);
->>>>>>> 9e82d59c
+void grpc_lb_policy_xds_routing_init(void);
+void grpc_lb_policy_xds_routing_shutdown(void);
@@ -76,9 +67,0 @@ void grpc_register_built_in_plugins(void) {
-<<<<<<< HEAD
-  grpc_register_plugin(grpc_lb_policy_xds_init,
-                       grpc_lb_policy_xds_shutdown);
-  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
-                       grpc_lb_policy_xds_routing_shutdown);
-||||||| 7126ffe0eb
-  grpc_register_plugin(grpc_lb_policy_xds_init,
-                       grpc_lb_policy_xds_shutdown);
-=======
@@ -93 +76,2 @@ void grpc_register_built_in_plugins(void) {
->>>>>>> 9e82d59c
+  grpc_register_plugin(grpc_lb_policy_xds_routing_init,
+                       grpc_lb_policy_xds_routing_shutdown);
