diff --git a/./output/frameworks_av/e79cb1c1-5a21203a-7b046f88/media@codec2@sfplugin@CCodec.no_comments_conflict.cpp b/./output/frameworks_av/e79cb1c1-5a21203a-7b046f88/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
index eb52354..2d29cab 100644
--- a/./output/frameworks_av/e79cb1c1-5a21203a-7b046f88/media@codec2@sfplugin@CCodec.no_comments_conflict.cpp
+++ b/./output/frameworks_av/e79cb1c1-5a21203a-7b046f88/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
@@ -40 +39,0 @@
-<<<<<<< HEAD
@@ -43,5 +41,0 @@
-||||||| 7b046f88f5
-#include "C2OMXNode.h"
-=======
-#include "C2NodeImpl.h"
->>>>>>> 5a21203a
