--- a/./output/art/645fa0ca-bf517232-d5a49147/runtime@runtime.no_comments_conflict.cc
+++ b/./output/art/645fa0ca-bf517232-d5a49147/runtime@runtime.no_comments_merged.cc
@@ -584 +583,0 @@ static void WaitUntilSingleThreaded() {
-<<<<<<< HEAD
@@ -587,6 +585,0 @@ static void WaitUntilSingleThreaded() {
-||||||| d5a4914776
-  ssize_t bytes_read = -1;
-=======
-  ssize_t bytes_read = -1;
-  uint64_t millis = 0;
->>>>>>> bf517232
@@ -616 +608,0 @@ static void WaitUntilSingleThreaded() {
-<<<<<<< HEAD
@@ -625,7 +616,0 @@ static void WaitUntilSingleThreaded() {
-||||||| d5a4914776
-  LOG(FATAL) << "Failed to reach single-threaded state: bytes_read = " << bytes_read
-             << " stat contents = \"" << buf << "...\"";
-=======
-  LOG(FATAL) << "Failed to reach single-threaded state: bytes_read = " << bytes_read
-             << " stat contents = \"" << buf << "...\" wait_time = " << MilliTime() - millis;
->>>>>>> bf517232
