diff --git a/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_conflict.cpp b/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_merged.cpp
index 3a4eace..aaf2dab 100644
--- a/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_conflict.cpp
+++ b/./output/frameworks_native/746ea4a5-38ed5444-53fb1692/services@inputflinger@dispatcher@trace@InputTracingPerfettoBackend.no_comments_merged.cpp
@@ -7 +6,0 @@
-<<<<<<< HEAD
@@ -12,5 +10,0 @@
-||||||| 53fb169272
-=======
-#include <perfetto/trace/android/winscope_extensions.pbzero.h>
-#include <perfetto/trace/android/winscope_extensions_impl.pbzero.h>
->>>>>>> 38ed5444
@@ -173 +166,0 @@ void PerfettoBackend::traceMotionEvent(const TracedMotionEvent& event,
-<<<<<<< HEAD
@@ -180,11 +172,0 @@ void PerfettoBackend::traceMotionEvent(const TracedMotionEvent& event,
-||||||| 53fb169272
-        auto* inputEvent = tracePacket->set_android_input_event();
-        auto* dispatchMotion = inputEvent->set_dispatcher_motion_event();
-        AndroidInputEventProtoConverter::toProtoMotionEvent(event, *dispatchMotion);
-=======
-        auto* winscopeExtensions = static_cast<perfetto::protos::pbzero::WinscopeExtensionsImpl*>(
-                tracePacket->set_winscope_extensions());
-        auto* inputEvent = winscopeExtensions->set_android_input_event();
-        auto* dispatchMotion = inputEvent->set_dispatcher_motion_event();
-        AndroidInputEventProtoConverter::toProtoMotionEvent(event, *dispatchMotion);
->>>>>>> 38ed5444
@@ -210 +191,0 @@ void PerfettoBackend::traceKeyEvent(const TracedKeyEvent& event,
-<<<<<<< HEAD
@@ -217,11 +197,0 @@ void PerfettoBackend::traceKeyEvent(const TracedKeyEvent& event,
-||||||| 53fb169272
-        auto* inputEvent = tracePacket->set_android_input_event();
-        auto* dispatchKey = inputEvent->set_dispatcher_key_event();
-        AndroidInputEventProtoConverter::toProtoKeyEvent(event, *dispatchKey);
-=======
-        auto* winscopeExtensions = static_cast<perfetto::protos::pbzero::WinscopeExtensionsImpl*>(
-                tracePacket->set_winscope_extensions());
-        auto* inputEvent = winscopeExtensions->set_android_input_event();
-        auto* dispatchKey = inputEvent->set_dispatcher_key_event();
-        AndroidInputEventProtoConverter::toProtoKeyEvent(event, *dispatchKey);
->>>>>>> 38ed5444
@@ -247 +216,0 @@ void PerfettoBackend::traceWindowDispatch(const WindowDispatchArgs& dispatchArgs
-<<<<<<< HEAD
@@ -256,13 +224,0 @@ void PerfettoBackend::traceWindowDispatch(const WindowDispatchArgs& dispatchArgs
-||||||| 53fb169272
-        auto* inputEventProto = tracePacket->set_android_input_event();
-        auto* dispatchEventProto = inputEventProto->set_dispatcher_window_dispatch_event();
-        AndroidInputEventProtoConverter::toProtoWindowDispatchEvent(dispatchArgs,
-                                                                    *dispatchEventProto);
-=======
-        auto* winscopeExtensions = static_cast<perfetto::protos::pbzero::WinscopeExtensionsImpl*>(
-                tracePacket->set_winscope_extensions());
-        auto* inputEventProto = winscopeExtensions->set_android_input_event();
-        auto* dispatchEventProto = inputEventProto->set_dispatcher_window_dispatch_event();
-        AndroidInputEventProtoConverter::toProtoWindowDispatchEvent(dispatchArgs,
-                                                                    *dispatchEventProto);
->>>>>>> 38ed5444
