diff --git a/./output/redis/e978bdf9-327cb45e-764b420f/src@module.no_comments_mergegen.c b/./output/redis/e978bdf9-327cb45e-764b420f/src@module.no_comments_merged.c
index 4519055..a1fcaa0 100644
--- a/./output/redis/e978bdf9-327cb45e-764b420f/src@module.no_comments_mergegen.c
+++ b/./output/redis/e978bdf9-327cb45e-764b420f/src@module.no_comments_merged.c
@@ -3661 +3661,5 @@ void moduleRegisterCoreAPI(void) {
-REGISTER_API(SetLRUOrLFU); REGISTER_API(GetLRUOrLFU); REGISTER_API(BlockClientOnKeys); REGISTER_API(SignalKeyAsReady); REGISTER_API(GetBlockedClientReadyKey);
+    REGISTER_API(SetLRUOrLFU);
+    REGISTER_API(GetLRUOrLFU);
+    REGISTER_API(BlockClientOnKeys);
+    REGISTER_API(SignalKeyAsReady);
+    REGISTER_API(GetBlockedClientReadyKey);
