diff --git a/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@execution@operator@csv_scanner@sniffer@dialect_detection.no_comments_conflict.cpp b/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@execution@operator@csv_scanner@sniffer@dialect_detection.no_comments_merged.cpp
index 4ad1113..2ae6097 100644
--- a/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@execution@operator@csv_scanner@sniffer@dialect_detection.no_comments_conflict.cpp
+++ b/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@execution@operator@csv_scanner@sniffer@dialect_detection.no_comments_merged.cpp
@@ -270 +269,0 @@ void CSVSniffer::AnalyzeDialectCandidate(unique_ptr<ColumnCountScanner> scanner,
-<<<<<<< HEAD
@@ -272,8 +270,0 @@ void CSVSniffer::AnalyzeDialectCandidate(unique_ptr<ColumnCountScanner> scanner,
-     (single_column_before || (more_values && !require_more_padding) ||
-      (more_than_one_column && require_less_padding) || quoted) &&
-||||||| 39f4ff4108
- if (rows_consistent &&
-     (single_column_before || (more_values && !require_more_padding) ||
-      (more_than_one_column && require_less_padding)) &&
-=======
- if (rows_consistent &&
@@ -281,2 +272 @@ void CSVSniffer::AnalyzeDialectCandidate(unique_ptr<ColumnCountScanner> scanner,
-      (more_than_one_column && require_less_padding)) &&
->>>>>>> 104cfacc
+      (more_than_one_column && require_less_padding) || quoted) &&
