diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_merged.h
index 5bb08d6..7251f1d 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_merged.h
@@ -105,0 +106,2 @@ class OnDeviceRefresh final {
+  CheckPreconditionForSystem(const std::vector<com::android::apex::ApexInfo>& apex_info_list) const;
+  WARN_UNUSED PreconditionCheckResult
@@ -140 +141,0 @@ class OnDeviceRefresh final {
-  CheckPreconditionForSystem(const std::vector<com::android::apex::ApexInfo>& apex_info_list) const;
