diff --git a/./output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_conflict.h b/./output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_merged.h
index 8d84398..7251f1d 100644
--- a/./output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_conflict.h
+++ b/./output/art/5eec36fc-3e30f7a4-9f9413b0/odrefresh@odrefresh.no_comments_merged.h
@@ -104 +103,0 @@ class OnDeviceRefresh final {
-<<<<<<< HEAD
@@ -106,8 +104,0 @@ class OnDeviceRefresh final {
-||||||| 9f9413b04b
-  WARN_UNUSED bool BootClasspathArtifactsOnSystemUsable(
-      const com::android::apex::ApexInfo& art_apex_info) const;
-=======
-  WARN_UNUSED bool CheckBuildUserfaultFdGc() const;
-  WARN_UNUSED bool BootClasspathArtifactsOnSystemUsable(
-      const com::android::apex::ApexInfo& art_apex_info) const;
->>>>>>> 3e30f7a4
