diff --git a/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergegen.h b/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
index cc0570d..df48b44 100644
--- a/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_mergegen.h
+++ b/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@include@graphicsenv@GraphicsEnv.no_comments_merged.h
@@ -38 +38,2 @@ public:
-bool shouldUseSystemAngle(); bool shouldUseNativeDriver();
+    bool shouldUseSystemAngle();
+    bool shouldUseNativeDriver();
@@ -61,0 +63 @@ bool mShouldUseSystemAngle = false;
+    bool mShouldUseNativeDriver = false;
