diff --git a/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@GraphicsEnv.no_comments_mergegen.cpp b/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
index 4def91c..4a8d3c1 100644
--- a/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@GraphicsEnv.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/a266fe84-87425707-58e0e142/libs@graphicsenv@GraphicsEnv.no_comments_merged.cpp
@@ -401,0 +402 @@ bool GraphicsEnv::shouldUseAngle() {
+void GraphicsEnv::setAngleInfo(const std::string& path, const bool shouldUseNativeDriver,
@@ -413 +414,2 @@ bool GraphicsEnv::shouldUseAngle() {
-if (mAnglePath == "system") { mShouldUseSystemAngle = true;
+    if (mAnglePath == "system") {
+        mShouldUseSystemAngle = true;
@@ -473 +475,34 @@ void GraphicsEnv::nativeToggleAngleAsSystemDriver(bool enabled) {
-bool GraphicsEnv::shouldUseSystemAngle() { return mShouldUseSystemAngle; } bool GraphicsEnv::shouldUseNativeDriver() { return mShouldUseNativeDriver; } void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace, const std::string& layerPaths) { if (mLayerPaths.empty()) { mLayerPaths = layerPaths; mAppNamespace = appNamespace; } else { ALOGV("Vulkan layer search path already set, not clobbering with '%s' for namespace %p'", layerPaths.c_str(), appNamespace); } } NativeLoaderNamespace* GraphicsEnv::getAppNamespace() { return mAppNamespace; } const std::string& GraphicsEnv::getLayerPaths() { return mLayerPaths; } const std::string& GraphicsEnv::getDebugLayers() { return mDebugLayers; } const std::string& GraphicsEnv::getDebugLayersGLES() { return mDebugLayersGLES; } void GraphicsEnv::setDebugLayers(const std::string& layers) { mDebugLayers = layers; } void GraphicsEnv::setDebugLayersGLES(const std::string& layers) { mDebugLayersGLES = layers; }
+bool GraphicsEnv::shouldUseSystemAngle() {
+    return mShouldUseSystemAngle;
+}
+bool GraphicsEnv::shouldUseNativeDriver() {
+    return mShouldUseNativeDriver;
+}
+void GraphicsEnv::setLayerPaths(NativeLoaderNamespace* appNamespace,
+                                const std::string& layerPaths) {
+    if (mLayerPaths.empty()) {
+        mLayerPaths = layerPaths;
+        mAppNamespace = appNamespace;
+    } else {
+        ALOGV("Vulkan layer search path already set, not clobbering with '%s' for namespace %p'",
+              layerPaths.c_str(), appNamespace);
+    }
+}
+NativeLoaderNamespace* GraphicsEnv::getAppNamespace() {
+    return mAppNamespace;
+}
+const std::string& GraphicsEnv::getLayerPaths() {
+    return mLayerPaths;
+}
+const std::string& GraphicsEnv::getDebugLayers() {
+    return mDebugLayers;
+}
+const std::string& GraphicsEnv::getDebugLayersGLES() {
+    return mDebugLayersGLES;
+}
+void GraphicsEnv::setDebugLayers(const std::string& layers) {
+    mDebugLayers = layers;
+}
+void GraphicsEnv::setDebugLayersGLES(const std::string& layers) {
+    mDebugLayersGLES = layers;
+}
