diff --git a/./output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_mergegen.cpp b/./output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_merged.cpp
index 59f42ee..8b89877 100644
--- a/./output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_merged.cpp
@@ -101,0 +102,6 @@ static bool should_unload_system_driver(egl_connection_t* cnx) {
+        if (!(cnx->angleLoaded && android::GraphicsEnv::getInstance().shouldUseSystemAngle())) {
+            return true;
+        }
+    }
+    if (android::GraphicsEnv::getInstance().shouldUseNativeDriver() && cnx->angleLoaded) {
+        return true;
