--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/a266fe84-87425707-58e0e142/opengl@libs@EGL@Loader.no_comments_merged.cpp
@@ -94 +101,0 @@ static bool should_unload_system_driver(egl_connection_t* cnx) {
-<<<<<<< HEAD
@@ -100,4 +106,0 @@ static bool should_unload_system_driver(egl_connection_t* cnx) {
-|||||||
-=======
-        if (!(cnx->angleLoaded && android::GraphicsEnv::getInstance().shouldUseSystemAngle())) {
->>>>>>> 87425707b37cce4759e29fc9c77369f771ec4319
@@ -106 +108,0 @@ static bool should_unload_system_driver(egl_connection_t* cnx) {
-}
