--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/fdfb9a01-5d1d32dc-83cda925/fs_mgr@fs_mgr_fstab.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/fdfb9a01-5d1d32dc-83cda925/fs_mgr@fs_mgr_fstab.no_comments_merged.cpp
@@ -23 +22,0 @@ using android::base::StartsWith;
-#define EM_AES_256_HEH 4
@@ -89,2 +87 @@ void ParseFileEncryption(const std::string& arg, FstabEntry* entry) {
-    LWARNING << "fileencryption= flag malformed, file contents encryption mode "
-                "not found: "
+        LWARNING << "fileencryption= flag malformed, file contents encryption mode not found: "
@@ -96,5 +93,3 @@ void ParseFileEncryption(const std::string& arg, FstabEntry* entry) {
-    if (std::find(kFileNamesEncryptionMode.begin(),
-                  kFileNamesEncryptionMode.end(),
-                  parts[1]) == kFileNamesEncryptionMode.end()) {
-      LWARNING << "fileencryption= flag malformed, file names encryption mode "
-                  "not found: "
+        if (std::find(kFileNamesEncryptionMode.begin(), kFileNamesEncryptionMode.end(), parts[1]) ==
+            kFileNamesEncryptionMode.end()) {
+            LWARNING << "fileencryption= flag malformed, file names encryption mode not found: "
