--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/fdfb9a01-5d1d32dc-83cda925/lmkd@include@lmkd.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/fdfb9a01-5d1d32dc-83cda925/lmkd@include@lmkd.no_comments_merged.h
@@ -11 +10,0 @@ enum lmk_cmd {
-<<<<<<< HEAD
@@ -14,4 +12,0 @@ enum lmk_cmd {
-|||||||
-=======
-  LMK_PROCPURGE,
->>>>>>> 5d1d32dcbffaede882c140ce5f00502135cd3f7e
@@ -79,4 +70,0 @@ static inline size_t lmkd_pack_set_procremove(LMKD_CTRL_PACKET packet,
-inline size_t lmkd_pack_set_procpurge(LMKD_CTRL_PACKET packet) {
-  packet[0] = htonl(LMK_PROCPURGE);
-  return sizeof(int);
-}
@@ -108,0 +96 @@ static inline size_t lmkd_pack_set_getkillcnt_repl(LMKD_CTRL_PACKET packet,
+__END_DECLS
