diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/49075cc3-6640ecb0-af54b6ec/services@surfaceflinger@WindowInfosListenerInvoker.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/49075cc3-6640ecb0-af54b6ec/services@surfaceflinger@WindowInfosListenerInvoker.no_comments_merged.h
index 472fc14..ead6193 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/49075cc3-6640ecb0-af54b6ec/services@surfaceflinger@WindowInfosListenerInvoker.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_native/49075cc3-6640ecb0-af54b6ec/services@surfaceflinger@WindowInfosListenerInvoker.no_comments_merged.h
@@ -23,2 +22,0 @@ public:
-    void windowInfosChanged(std::vector<gui::WindowInfo>, std::vector<gui::DisplayInfo>,
-                            bool shouldSync, bool forceImmediateCall);
@@ -31 +29 @@ private:
-            mWindowInfosListeners GUARDED_BY(mMessagesMutex);
+            mWindowInfosListeners GUARDED_BY(mListenersMutex);
@@ -33,3 +31,2 @@ private:
-    uint32_t mActiveMessageCount GUARDED_BY(mMessagesMutex);
-    std::function<void(bool)> mWindowInfosChangedDelayed GUARDED_BY(mMessagesMutex);
-    bool mShouldSyncDelayed;
+    uint32_t mActiveMessageCount GUARDED_BY(mMessagesMutex) = 0;
+    std::function<void()> mWindowInfosChangedDelayed GUARDED_BY(mMessagesMutex);
