--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/418ba990-58bb6f8c-8903c1f1/cmd-server-info.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/418ba990-58bb6f8c-8903c1f1/cmd-server-info.no_comments_merged.c
@@ -7 +6,0 @@
-#include <vis.h>
@@ -33,7 +39,0 @@ enum cmd_retval cmd_server_info_exec(unused struct cmd *self,
-<<<<<<< HEAD
-  ctx->print(ctx, "tmux " VERSION ", pid %ld, started %s", (long)getpid(), tim);
-  ctx->print(ctx, "socket path %s, debug level %d", socket_path, debug_level);
-|||||||
-  ctx->print(ctx, "pid %ld, started %s", (long)getpid(), tim);
-  ctx->print(ctx, "socket path %s, debug level %d", socket_path, debug_level);
-=======
@@ -41,2 +41,2 @@ enum cmd_retval cmd_server_info_exec(unused struct cmd *self,
-  cmdq_print(cmdq, "socket path %s, debug level %d", socket_path, debug_level);
->>>>>>> 58bb6f8c5650d496fb3b872766c0278aa024631d
+ cmdq_print(cmdq, "socket path %s, debug level %d", socket_path,
+     debug_level);
