--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/418ba990-58bb6f8c-8903c1f1/window.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/418ba990-58bb6f8c-8903c1f1/window.no_comments_merged.c
@@ -6,2 +5,0 @@
-#include <paths.h>
-#include <stdint.h>
@@ -10 +7,0 @@
-#include <util.h>
@@ -23 +20,4 @@ void window_pane_error_callback(struct bufferevent *, short, void *);
-int winlink_cmp(struct winlink *wl1, struct winlink *wl2) {
+RB_GENERATE(winlinks, winlink, entry, winlink_cmp);
+int
+winlink_cmp(struct winlink *wl1, struct winlink *wl2)
+{
@@ -26 +26,4 @@ int winlink_cmp(struct winlink *wl1, struct winlink *wl2) {
-int window_pane_cmp(struct window_pane *wp1, struct window_pane *wp2) {
+RB_GENERATE(window_pane_tree, window_pane, tree_entry, window_pane_cmp);
+int
+window_pane_cmp(struct window_pane *wp1, struct window_pane *wp2)
+{
@@ -576,11 +708,2 @@ void window_pane_resize(struct window_pane *wp, u_int sx, u_int sy) {
-  if (wp->mode != NULL) wp->mode->resize(wp, sx, sy);
-<<<<<<< HEAD
-  if (wp->fd != -1 && ioctl(wp->fd, TIOCSWINSZ, &ws) == -1)
-#ifdef __sun
-    if (errno != EINVAL)
-#endif
-      fatal("ioctl failed");
-|||||||
-  if (wp->fd != -1 && ioctl(wp->fd, TIOCSWINSZ, &ws) == -1)
-    fatal("ioctl failed");
-=======
+ if (wp->mode != NULL)
+  wp->mode->resize(wp, sx, sy);
@@ -588 +710,0 @@ void window_pane_resize(struct window_pane *wp, u_int sx, u_int sy) {
->>>>>>> 58bb6f8c5650d496fb3b872766c0278aa024631d
