diff --git a/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_conflict.h b/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_merged.h
index 3f76f7b..fa0e320 100644
--- a/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_conflict.h
+++ b/./output/rocksdb/b06840aa-e6b3e3b4-4159a284/db@version_set.no_comments_merged.h
@@ -25 +24,0 @@ class MergeContext;
-<<<<<<< HEAD
@@ -28,2 +26,0 @@ class ColumnFamilySet;
-||||||| 4159a284c
-=======
@@ -31 +27,0 @@ class LookupKey;
->>>>>>> e6b3e3b4
