diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_verity.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_verity.no_comments_merged.cpp
index 99aa4e7..bf37647 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_verity.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/9291de89-34543c03-6f831490/fs_mgr@fs_mgr_verity.no_comments_merged.cpp
@@ -17 +16,0 @@
-#include <android-base/unique_fd.h>
@@ -18,0 +18 @@
+#include <android-base/unique_fd.h>
@@ -20,3 +19,0 @@
-#include "mincrypt/rsa.h"
-#include "mincrypt/sha.h"
-#include "mincrypt/sha256.h"
@@ -45 +41,0 @@
-#define READ_BUF_SIZE 4096
@@ -52,0 +49 @@
+#define READ_BUF_SIZE 4096
@@ -61 +58,2 @@ extern struct fs_info info;
-static RSA *load_key(const char *path) {
+static RSA *load_key(const char *path)
+{
@@ -82 +80,2 @@ static int verify_table(const uint8_t *signature, size_t signature_size,
-                        const char *table, uint32_t table_length) {
+        const char *table, uint32_t table_length)
+{
@@ -102,3 +101,4 @@ out:
-static int verify_verity_signature(const struct fec_verity_metadata &verity) {
-  if (verify_table(verity.signature, sizeof(verity.signature), verity.table,
-                   verity.table_length) == 0 ||
+static int verify_verity_signature(const struct fec_verity_metadata& verity)
+{
+    if (verify_table(verity.signature, sizeof(verity.signature),
+            verity.table, verity.table_length) == 0 ||
@@ -111 +111,2 @@ static int verify_verity_signature(const struct fec_verity_metadata &verity) {
-static int invalidate_table(char *table, size_t table_length) {
+static int invalidate_table(char *table, size_t table_length)
+{
@@ -131,2 +132,2 @@ static int invalidate_table(char *table, size_t table_length) {
-static void verity_ioctl_init(struct dm_ioctl *io, const char *name,
-                              unsigned flags) {
+static void verity_ioctl_init(struct dm_ioctl *io, const char *name, unsigned flags)
+{
@@ -144 +145,2 @@ static void verity_ioctl_init(struct dm_ioctl *io, const char *name,
-static int create_verity_device(struct dm_ioctl *io, char *name, int fd) {
+static int create_verity_device(struct dm_ioctl *io, char *name, int fd)
+{
@@ -152 +154,2 @@ static int create_verity_device(struct dm_ioctl *io, char *name, int fd) {
-static int destroy_verity_device(struct dm_ioctl *io, char *name, int fd) {
+static int destroy_verity_device(struct dm_ioctl *io, char *name, int fd)
+{
@@ -160,2 +163,2 @@ static int destroy_verity_device(struct dm_ioctl *io, char *name, int fd) {
-static int get_verity_device_name(struct dm_ioctl *io, char *name, int fd,
-                                  char **dev_name) {
+static int get_verity_device_name(struct dm_ioctl *io, char *name, int fd, char **dev_name)
+{
@@ -180,2 +183,2 @@ struct verity_table_params {
-typedef bool (*format_verity_table_func)(
-    char *buf, const size_t bufsize, const struct verity_table_params *params);
+typedef bool (*format_verity_table_func)(char *buf, const size_t bufsize,
+        const struct verity_table_params *params);
@@ -183 +186,2 @@ static bool format_verity_table(char *buf, const size_t bufsize,
-                                const struct verity_table_params *params) {
+        const struct verity_table_params *params)
+{
@@ -193,4 +197,4 @@ static bool format_verity_table(char *buf, const size_t bufsize,
-      res = snprintf(buf, bufsize, "%s %u %s " VERITY_TABLE_OPT_FEC_FORMAT,
-                     params->table, 1 + VERITY_TABLE_OPT_FEC_ARGS, mode_flag,
-                     params->ecc_dev, params->ecc.start / FEC_BLOCKSIZE,
-                     params->ecc.blocks, params->ecc.roots);
+            res = snprintf(buf, bufsize,
+                    "%s %u %s " VERITY_TABLE_OPT_FEC_FORMAT,
+                    params->table, 1 + VERITY_TABLE_OPT_FEC_ARGS, mode_flag, params->ecc_dev,
+                    params->ecc.start / FEC_BLOCKSIZE, params->ecc.blocks, params->ecc.roots);
@@ -198 +202,2 @@ static bool format_verity_table(char *buf, const size_t bufsize,
-      res = snprintf(buf, bufsize, "%s %u " VERITY_TABLE_OPT_FEC_FORMAT,
+            res = snprintf(buf, bufsize,
+                    "%s %u " VERITY_TABLE_OPT_FEC_FORMAT,
@@ -200,2 +205 @@ static bool format_verity_table(char *buf, const size_t bufsize,
-                     params->ecc.start / FEC_BLOCKSIZE, params->ecc.blocks,
-                     params->ecc.roots);
+                    params->ecc.start / FEC_BLOCKSIZE, params->ecc.blocks, params->ecc.roots);
@@ -204,2 +208,2 @@ static bool format_verity_table(char *buf, const size_t bufsize,
-    res = snprintf(buf, bufsize, "%s 2 " VERITY_TABLE_OPT_IGNZERO " %s",
-                   params->table, mode_flag);
+        res = snprintf(buf, bufsize, "%s 2 " VERITY_TABLE_OPT_IGNZERO " %s", params->table,
+                    mode_flag);
@@ -207,2 +211 @@ static bool format_verity_table(char *buf, const size_t bufsize,
-    res =
-        snprintf(buf, bufsize, "%s 1 " VERITY_TABLE_OPT_IGNZERO, params->table);
+        res = snprintf(buf, bufsize, "%s 1 " VERITY_TABLE_OPT_IGNZERO, params->table);
@@ -216,2 +219,3 @@ static bool format_verity_table(char *buf, const size_t bufsize,
-static bool format_legacy_verity_table(
-    char *buf, const size_t bufsize, const struct verity_table_params *params) {
+static bool format_legacy_verity_table(char *buf, const size_t bufsize,
+        const struct verity_table_params *params)
+{
@@ -230,4 +234,3 @@ static bool format_legacy_verity_table(
-static int load_verity_table(struct dm_ioctl *io, char *name,
-                             uint64_t device_size, int fd,
-                             const struct verity_table_params *params,
-                             format_verity_table_func format) {
+static int load_verity_table(struct dm_ioctl *io, char *name, uint64_t device_size, int fd,
+        const struct verity_table_params *params, format_verity_table_func format)
+{
@@ -238,2 +241 @@ static int load_verity_table(struct dm_ioctl *io, char *name,
-  struct dm_target_spec *tgt =
-      (struct dm_target_spec *)&buffer[sizeof(struct dm_ioctl)];
+    struct dm_target_spec *tgt = (struct dm_target_spec *) &buffer[sizeof(struct dm_ioctl)];
@@ -245,2 +247 @@ static int load_verity_table(struct dm_ioctl *io, char *name,
-  verity_params =
-      buffer + sizeof(struct dm_ioctl) + sizeof(struct dm_target_spec);
+    verity_params = buffer + sizeof(struct dm_ioctl) + sizeof(struct dm_target_spec);
@@ -262 +263,2 @@ static int load_verity_table(struct dm_ioctl *io, char *name,
-static int resume_verity_table(struct dm_ioctl *io, char *name, int fd) {
+static int resume_verity_table(struct dm_ioctl *io, char *name, int fd)
+{
@@ -280 +282,2 @@ static int test_access(char *device) {
-static int check_verity_restart(const char *fname) {
+static int check_verity_restart(const char *fname)
+{
@@ -321,3 +324,7 @@ out:
-static int was_verity_restart() {
-  static const char *files[] = {"/sys/fs/pstore/console-ramoops",
-                                "/proc/last_kmsg", NULL};
+static int was_verity_restart()
+{
+    static const char *files[] = {
+        "/sys/fs/pstore/console-ramoops",
+        "/proc/last_kmsg",
+        NULL
+    };
@@ -333 +340,2 @@ static int metadata_add(FILE *fp, long start, const char *tag,
-                        unsigned int length, off64_t *offset) {
+        unsigned int length, off64_t *offset)
+{
@@ -339 +347,2 @@ static int metadata_add(FILE *fp, long start, const char *tag,
-  if (fseek(fp, length, SEEK_CUR) < 0 || fprintf(fp, METADATA_EOD " 0\n") < 0) {
+    if (fseek(fp, length, SEEK_CUR) < 0 ||
+        fprintf(fp, METADATA_EOD " 0\n") < 0) {
@@ -345 +354,2 @@ static int metadata_find(const char *fname, const char *stag,
-                         unsigned int slength, off64_t *offset) {
+        unsigned int slength, off64_t *offset)
+{
@@ -381,2 +391,2 @@ static int metadata_find(const char *fname, const char *stag,
-    n = fscanf(fp, "%" STRINGIFY(METADATA_TAG_MAX_LENGTH) "s %u\n", tag,
-               &length);
+        n = fscanf(fp, "%" STRINGIFY(METADATA_TAG_MAX_LENGTH) "s %u\n",
+                tag, &length);
@@ -398 +408,2 @@ static int metadata_find(const char *fname, const char *stag,
-        ERROR("Failed to write metadata to %s: %s\n", fname, strerror(errno));
+                ERROR("Failed to write metadata to %s: %s\n", fname,
+                    strerror(errno));
@@ -410 +421,2 @@ out:
-static int write_verity_state(const char *fname, off64_t offset, int32_t mode) {
+static int write_verity_state(const char *fname, off64_t offset, int32_t mode)
+{
@@ -431 +443,2 @@ out:
-static int read_verity_state(const char *fname, off64_t offset, int *mode) {
+static int read_verity_state(const char *fname, off64_t offset, int *mode)
+{
@@ -454 +467,2 @@ static int read_verity_state(const char *fname, off64_t offset, int *mode) {
-  if (s.mode < VERITY_MODE_EIO || s.mode > VERITY_MODE_LAST) {
+    if (s.mode < VERITY_MODE_EIO ||
+        s.mode > VERITY_MODE_LAST) {
@@ -466 +480,2 @@ out:
-static int read_partition(const char *path, uint64_t size) {
+static int read_partition(const char *path, uint64_t size)
+{
@@ -469,2 +484 @@ static int read_partition(const char *path, uint64_t size) {
-  android::base::unique_fd fd(
-      TEMP_FAILURE_RETRY(open(path, O_RDONLY | O_CLOEXEC)));
+    android::base::unique_fd fd(TEMP_FAILURE_RETRY(open(path, O_RDONLY | O_CLOEXEC)));
@@ -478 +492,2 @@ static int read_partition(const char *path, uint64_t size) {
-      ERROR("Error in reading partition %s: %s\n", path, strerror(errno));
+            ERROR("Error in reading partition %s: %s\n", path,
+                  strerror(errno));
@@ -485 +500,2 @@ static int read_partition(const char *path, uint64_t size) {
-static int compare_last_signature(struct fstab_rec *fstab, int *match) {
+static int compare_last_signature(struct fstab_rec *fstab, int *match)
+{
@@ -497 +513,2 @@ static int compare_last_signature(struct fstab_rec *fstab, int *match) {
-    ERROR("Failed to open '%s' (%s)\n", fstab->blk_device, strerror(errno));
+        ERROR("Failed to open '%s' (%s)\n", fstab->blk_device,
+            strerror(errno));
@@ -511,2 +528,2 @@ static int compare_last_signature(struct fstab_rec *fstab, int *match) {
-  if (metadata_find(fstab->verity_loc, tag, SHA256_DIGEST_LENGTH, &offset) <
-      0) {
+    if (metadata_find(fstab->verity_loc, tag, SHA256_DIGEST_LENGTH,
+            &offset) < 0) {
@@ -520,2 +537,2 @@ static int compare_last_signature(struct fstab_rec *fstab, int *match) {
-  if (TEMP_FAILURE_RETRY(pread64(fd, prev, sizeof(prev), offset)) !=
-      sizeof(prev)) {
+    if (TEMP_FAILURE_RETRY(pread64(fd, prev, sizeof(prev),
+            offset)) != sizeof(prev)) {
@@ -528,2 +545,2 @@ static int compare_last_signature(struct fstab_rec *fstab, int *match) {
-    if (TEMP_FAILURE_RETRY(pwrite64(fd, curr, sizeof(curr), offset)) !=
-        sizeof(curr)) {
+        if (TEMP_FAILURE_RETRY(pwrite64(fd, curr, sizeof(curr),
+                offset)) != sizeof(curr)) {
@@ -540 +557,2 @@ out:
-static int get_verity_state_offset(struct fstab_rec *fstab, off64_t *offset) {
+static int get_verity_state_offset(struct fstab_rec *fstab, off64_t *offset)
+{
@@ -550 +568,2 @@ static int get_verity_state_offset(struct fstab_rec *fstab, off64_t *offset) {
-static int load_verity_state(struct fstab_rec *fstab, int *mode) {
+static int load_verity_state(struct fstab_rec *fstab, int *mode)
+{
@@ -574 +593,2 @@ static int load_verity_state(struct fstab_rec *fstab, int *mode) {
-int fs_mgr_load_verity_state(int *mode) {
+int fs_mgr_load_verity_state(int *mode)
+{
@@ -609 +629,2 @@ out:
-int fs_mgr_update_verity_state(fs_mgr_verity_state_callback callback) {
+int fs_mgr_update_verity_state(fs_mgr_verity_state_callback callback)
+{
@@ -672 +693,2 @@ out:
-static void update_verity_table_blk_device(char *blk_device, char **table) {
+static void update_verity_table_blk_device(char *blk_device, char **table)
+{
@@ -694 +716,2 @@ static void update_verity_table_blk_device(char *blk_device, char **table) {
-int fs_mgr_setup_verity(struct fstab_rec *fstab) {
+int fs_mgr_setup_verity(struct fstab_rec *fstab)
+{
@@ -701 +723,0 @@ int fs_mgr_setup_verity(struct fstab_rec *fstab) {
-  bool verified_at_boot = false;
@@ -704,0 +727 @@ int fs_mgr_setup_verity(struct fstab_rec *fstab) {
+    bool verified_at_boot = false;
@@ -707 +730,2 @@ int fs_mgr_setup_verity(struct fstab_rec *fstab) {
-    ERROR("Failed to open '%s' (%s)\n", fstab->blk_device, strerror(errno));
+        ERROR("Failed to open '%s' (%s)\n", fstab->blk_device,
+            strerror(errno));
