diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_merged.h
index 1ee3a20..d5f782a 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_merged.h
@@ -19 +18,0 @@
-#include "memtablelist.h"
@@ -36 +35,2 @@ class DBImpl : public DB {
-  virtual Status Get(const ReadOptions& options, const Slice& key,
+  virtual Status Get(const ReadOptions& options,
+                     const Slice& key,
@@ -41,2 +41,4 @@ class DBImpl : public DB {
-  virtual bool KeyMayExist(const ReadOptions& options, const Slice& key,
-                           std::string* value, bool* value_found = nullptr);
+  virtual bool KeyMayExist(const ReadOptions& options,
+                           const Slice& key,
+                           std::string* value,
+                           bool* value_found = nullptr);
@@ -67 +69,2 @@ class DBImpl : public DB {
-  virtual void GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata);
+  virtual void GetLiveFilesMetaData(
+    std::vector<LiveFileMetaData> *metadata);
@@ -69,3 +72,7 @@ class DBImpl : public DB {
-  Status RunManualCompaction(int input_level, int output_level,
-                             const Slice* begin, const Slice* end);
-  Status TEST_CompactRange(int level, const Slice* begin, const Slice* end);
+  Status RunManualCompaction(int input_level,
+                             int output_level,
+                             const Slice* begin,
+                             const Slice* end);
+  Status TEST_CompactRange(int level,
+                           const Slice* begin,
+                           const Slice* end);
@@ -81,4 +88,3 @@ class DBImpl : public DB {
-  void TEST_SetDefaultTimeToCheck(
-      uint64_t default_interval_to_delete_obsolete_WAL) {
-    default_interval_to_delete_obsolete_WAL_ =
-        default_interval_to_delete_obsolete_WAL;
+  void TEST_SetDefaultTimeToCheck(uint64_t default_interval_to_delete_obsolete_WAL)
+  {
+    default_interval_to_delete_obsolete_WAL_ = default_interval_to_delete_obsolete_WAL;
@@ -102 +108,2 @@ class DBImpl : public DB {
-      return candidate_files.size() || sst_delete_files.size() ||
+      return candidate_files.size() ||
+        sst_delete_files.size() ||
@@ -118 +125,2 @@ class DBImpl : public DB {
-      new_superversion = create_superversion ? new SuperVersion() : nullptr;
+      new_superversion =
+          create_superversion ? new SuperVersion() : nullptr;
@@ -128 +136,2 @@ class DBImpl : public DB {
-  void FindObsoleteFiles(DeletionState& deletion_state, bool force,
+  void FindObsoleteFiles(DeletionState& deletion_state,
+                         bool force,
@@ -140 +149,3 @@ class DBImpl : public DB {
-  MemTable* GetMemTable() { return mem_; }
+  MemTable* GetMemTable() {
+    return mem_;
+  }
@@ -175,2 +186 @@ class DBImpl : public DB {
-  Status BackgroundCompaction(bool* madeProgress,
-                              DeletionState& deletion_state);
+  Status BackgroundCompaction(bool* madeProgress,DeletionState& deletion_state);
@@ -188 +198,2 @@ class DBImpl : public DB {
-                                VectorLogPtr& log_files, WalFileType type);
+                                VectorLogPtr& log_files,
+                                WalFileType type);
@@ -202 +213,2 @@ class DBImpl : public DB {
-      const ReadOptions& options, uint64_t* superversion_number);
+    const ReadOptions& options,
+    uint64_t* superversion_number);
@@ -257 +269,2 @@ class DBImpl : public DB {
-      SequenceNumber in, std::vector<SequenceNumber>& snapshots,
+    SequenceNumber in,
+    std::vector<SequenceNumber>& snapshots,
@@ -261,2 +274,4 @@ class DBImpl : public DB {
-  Status GetImpl(const ReadOptions& options, const Slice& key,
-                 std::string* value, bool* value_found = nullptr);
+  Status GetImpl(const ReadOptions& options,
+                 const Slice& key,
+                 std::string* value,
+                 bool* value_found = nullptr);
