diff --git a/./output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_mergegen.h b/./output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_merged.h
index c0a0e6f..d5f782a 100644
--- a/./output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_mergegen.h
+++ b/./output/rocksdb/a46ac921-3c0dcf0e-9dc29414/db@db_impl.no_comments_merged.h
@@ -18 +18,2 @@
-#include "util/stats_logger.h" #include "db/internal_stats.h" #include "util/stats_logger.h"
+#include "util/stats_logger.h"
+#include "db/internal_stats.h"
