--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@db_impl.no_comments_merged.h
@@ -132,9 +130,0 @@ class DBImpl : public DB {
-  static void SuperVersionUnrefHandle(void* ptr) {
-    DBImpl::SuperVersion* sv = static_cast<DBImpl::SuperVersion*>(ptr);
-    if (sv->Unref()) {
-      sv->db->mutex_.Lock();
-      sv->Cleanup();
-      sv->db->mutex_.Unlock();
-      delete sv;
-    }
-  }
@@ -158,7 +148,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
-      superversion_to_free = nullptr;
-      new_superversion = create_superversion ? new SuperVersion() : nullptr;
-||||||| b2795b799
-      superversion_to_free = nullptr;
-      new_superversion = create_superversion ? new SuperVersion() : nullptr;
-=======
@@ -166 +149,0 @@ class DBImpl : public DB {
->>>>>>> 5142b37000ab748433bdb5060a856663987067fb
@@ -246,0 +231,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -257 +244,0 @@ class DBImpl : public DB {
-  ThreadLocalPtr* local_sv_;
@@ -302,8 +289,0 @@ class DBImpl : public DB {
-  void ResetThreadLocalSuperVersions(DeletionState* deletion_state);
-  Status GetImpl(const ReadOptions& options, ColumnFamilyHandle* column_family,
-                 const Slice& key, std::string* value,
-                 bool* value_found = nullptr);
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
@@ -313,3 +293,6 @@ class DBImpl : public DB {
-  virtual Status GetPropertiesOfAllTables(
-      ColumnFamilyHandle* column_family,
-      TablePropertiesCollection* props) override;
+  virtual Status GetPropertiesOfAllTables(ColumnFamilyHandle* column_family,
+                                          TablePropertiesCollection* props)
+      override;
+  Status GetImpl(const ReadOptions& options, ColumnFamilyHandle* column_family,
+                 const Slice& key, std::string* value,
+                 bool* value_found = nullptr);
