--- a/./output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@db_impl.no_comments_mergegen.h
+++ b/./output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@db_impl.no_comments_merged.h
@@ -129,9 +129,2 @@ class DBImpl : public DB {
-  static void SuperVersionUnrefHandle(void* ptr) {
-    DBImpl::SuperVersion* sv = static_cast<DBImpl::SuperVersion*>(ptr);
-    if (sv->Unref()) {
-      sv->db->mutex_.Lock();
-      sv->Cleanup();
-      sv->db->mutex_.Unlock();
-      delete sv;
-    }
-  }
+  void TEST_GetFilesMetaData(ColumnFamilyHandle* column_family,
+                             std::vector<std::vector<FileMetaData>>* metadata);
@@ -149 +142 @@ class DBImpl : public DB {
-{ autovector<SuperVersion*> superversions_to_free; SuperVersion* new_superversion;
+    autovector<SuperVersion*> superversions_to_free;
@@ -248,0 +242,3 @@ new_superversion = create_superversion ? new SuperVersion() : nullptr;
+  ColumnFamilyHandleImpl* default_cf_handle_;
+  unique_ptr<ColumnFamilyMemTablesImpl> column_family_memtables_;
+  std::deque<uint64_t> alive_log_files_;
@@ -296 +292,3 @@ new_superversion = create_superversion ? new SuperVersion() : nullptr;
-void ResetThreadLocalSuperVersions(DeletionState* deletion_state); using DB::GetPropertiesOfAllTables;
+  using DB::GetPropertiesOfAllTables;
+  virtual Status GetPropertiesOfAllTables(ColumnFamilyHandle* column_family,
+                                          TablePropertiesCollection* props)
