diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_merged.h
index 68542a9..5d6b5c1 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_merged.h
@@ -17 +17,3 @@ struct LogReporter : public log::Reader::Reporter {
-  virtual void Info(const char* s) { Log(info_log, "%s", s); }
+  virtual void Info(const char* s) {
+    Log(info_log, "%s", s);
+  }
@@ -22,2 +24,2 @@ class LogFileImpl : public LogFile {
-              uint64_t sizeBytes)
-      : logNumber_(logNum),
+              uint64_t sizeBytes) :
+    logNumber_(logNum),
@@ -26 +28,2 @@ class LogFileImpl : public LogFile {
-        sizeFileBytes_(sizeBytes) {}
+    sizeFileBytes_(sizeBytes) {
+  }
@@ -49 +52 @@ class TransactionLogIteratorImpl : public TransactionLogIterator {
-      const std::string& dir, const Options* options,
+      const std::string& dir, const DBOptions* options,
