--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/db@transaction_log_impl.no_comments_merged.h
@@ -49 +52 @@ class TransactionLogIteratorImpl : public TransactionLogIterator {
-      const std::string& dir, const Options* options,
+      const std::string& dir, const DBOptions* options,
