--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/include@rocksdb@options.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/include@rocksdb@options.no_comments_merged.h
@@ -149,0 +151 @@ struct DBOptions {
+  bool allow_thread_local;
@@ -155,4 +157,3 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
-  void Dump(Logger* log) const;
-  Options* PrepareForBulkLoad();
-  bool allow_thread_local;
-  Options() : DBOptions(), ColumnFamilyOptions() {}
+  Options() :
+    DBOptions(),
+    ColumnFamilyOptions() {}
@@ -161,0 +163,2 @@ struct Options : public DBOptions, public ColumnFamilyOptions {
+  void Dump(Logger* log) const;
+  Options* PrepareForBulkLoad();
