diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/util@env.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/util@env.no_comments_merged.cc
index 36d9b8e..497e2be 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/util@env.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/f9b2f0ad-5142b370-b2795b79/util@env.no_comments_merged.cc
@@ -4,6 +4,12 @@ namespace rocksdb {
-Env::~Env() {}
-SequentialFile::~SequentialFile() {}
-RandomAccessFile::~RandomAccessFile() {}
-WritableFile::~WritableFile() {}
-Logger::~Logger() {}
-FileLock::~FileLock() {}
+Env::~Env() {
+}
+SequentialFile::~SequentialFile() {
+}
+RandomAccessFile::~RandomAccessFile() {
+}
+WritableFile::~WritableFile() {
+}
+Logger::~Logger() {
+}
+FileLock::~FileLock() {
+}
@@ -135 +141,2 @@ static Status DoWriteStringToFile(Env* env, const Slice& data,
-                                  const std::string& fname, bool should_sync) {
+                                  const std::string& fname,
+                                  bool should_sync) {
@@ -183 +190,2 @@ Status ReadFileToString(Env* env, const std::string& fname, std::string* data) {
-EnvWrapper::~EnvWrapper() {}
+EnvWrapper::~EnvWrapper() {
+}
