--- a/./output/rocksdb/136e4efb-1a8abe72-161d9e58/java@rocksjni@portal.no_comments_conflict.h
+++ b/./output/rocksdb/136e4efb-1a8abe72-161d9e58/java@rocksjni@portal.no_comments_merged.h
@@ -125,3 +124,0 @@ class HistogramDataJni {
-  }
-};
-||||||| 161d9e586
@@ -150,0 +148 @@ class BackupableDBOptionsJni {
+>>>>>>> 1a8abe72768b2b5cea800aa390c28e5ace6a552e
@@ -153 +150,0 @@ class BackupableDBOptionsJni {
->>>>>>> 1a8abe72
