diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_merged.cpp
index be1d854..c99ae52 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_merged.cpp
@@ -2,2 +2,2 @@
-#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
-#include "duckdb/common/types/column_data_collection.hpp"
+#include "duckdb/parallel/thread_context.hpp"
+#include "duckdb/catalog/catalog_entry/duck_table_entry.hpp"
@@ -6,0 +7 @@
+#include "duckdb/storage/data_table.hpp"
@@ -8,2 +8,0 @@
-#include "duckdb/catalog/catalog_entry/duck_table_entry.hpp"
-#include "duckdb/parallel/thread_context.hpp"
@@ -12 +10,0 @@
-#include "duckdb/storage/data_table.hpp"
@@ -56 +53,0 @@ PhysicalInsert::PhysicalInsert(LogicalOperator &op, SchemaCatalogEntry *schema,
-{ GetInsertInfo(*info, insert_types, bound_defaults); }
