--- a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_conflict.cpp
+++ b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_merged.cpp
@@ -2,8 +1,0 @@
-<<<<<<< HEAD
-#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
-#include "duckdb/common/types/column/column_data_collection.hpp"
-||||||| b3f6a8f16d
-#include "duckdb/parallel/thread_context.hpp"
-#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
-#include "duckdb/common/types/column_data_collection.hpp"
-=======
@@ -12,2 +4 @@
-#include "duckdb/common/types/column_data_collection.hpp"
->>>>>>> e4384e1f
+#include "duckdb/common/types/column/column_data_collection.hpp"
@@ -15,0 +7 @@
+#include "duckdb/storage/data_table.hpp"
@@ -17 +8,0 @@
-#include "duckdb/parallel/thread_context.hpp"
@@ -20 +10,0 @@
-#include "duckdb/storage/data_table.hpp"
