diff --git a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_conflict.cpp b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_merged.cpp
index 3b1b9a2..fb9e050 100644
--- a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_conflict.cpp
+++ b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_merged.cpp
@@ -2,2 +1,0 @@
-#include "duckdb/common/atomic.hpp"
-#include "duckdb/common/types/column/column_data_collection.hpp"
@@ -7,5 +4,0 @@
-<<<<<<< HEAD
-||||||| b3f6a8f16d
-#include "duckdb/common/types/column_data_collection.hpp"
-#include "duckdb/common/atomic.hpp"
-=======
@@ -13 +6 @@
-#include "duckdb/common/types/column_data_collection.hpp"
+#include "duckdb/common/types/column/column_data_collection.hpp"
@@ -15 +7,0 @@
->>>>>>> e4384e1f
