diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_mergebot.c b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_merged.c
index 78a12ef..933755e 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/cmd-split-window.no_comments_merged.c
@@ -4 +3,0 @@
-#include <paths.h>
@@ -12,7 +11,4 @@ const struct cmd_entry cmd_split_window_entry = {
-    "split-window",
-    "splitw",
-    "c:dF:l:hp:Pt:v",
-    0,
-    1,
-    "[-dhvP] [-c start-directory] [-F format] [-p percentage|-l "
-    "size] " CMD_TARGET_PANE_USAGE " [command]",
+ "split-window", "splitw",
+ "c:dF:l:hp:Pt:v", 0, 1,
+ "[-dhvP] [-c start-directory] [-F format] [-p percentage|-l size] "
+ CMD_TARGET_PANE_USAGE " [command]",
@@ -21,2 +17,5 @@ const struct cmd_entry cmd_split_window_entry = {
-    cmd_split_window_exec};
-void cmd_split_window_key_binding(struct cmd *self, int key) {
+ cmd_split_window_exec
+};
+void
+cmd_split_window_key_binding(struct cmd *self, int key)
+{
@@ -24 +23,2 @@ void cmd_split_window_key_binding(struct cmd *self, int key) {
-  if (key == '%') args_set(self->args, 'h', NULL);
+ if (key == '%')
+  args_set(self->args, 'h', NULL);
@@ -26 +26,3 @@ void cmd_split_window_key_binding(struct cmd *self, int key) {
-enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
+enum cmd_retval
+cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq)
+{
@@ -55 +57,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL) format_client(ft, c);
+  if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL)
+   format_client(ft, c);
@@ -64 +67,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-      cmdq_error(cmdq, "bad working directory: %s", strerror(errno));
+   cmdq_error(cmdq, "bad working directory: %s",
+       strerror(errno));
@@ -73 +77,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (args_has(args, 'h')) type = LAYOUT_LEFTRIGHT;
+ if (args_has(args, 'h'))
+  type = LAYOUT_LEFTRIGHT;
@@ -98 +103,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (*shell == '\0' || areshell(shell)) shell = _PATH_BSHELL;
+ if (*shell == '\0' || areshell(shell))
+  shell = _PATH_BSHELL;
@@ -104 +110,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (window_pane_spawn(new_wp, cmd, shell, cwd, &env, s->tio, &cause) != 0)
+ if (window_pane_spawn(
+     new_wp, cmd, shell, cwd, &env, s->tio, &cause) != 0)
@@ -119 +126,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-    if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL) format_client(ft, c);
+  if ((c = cmd_find_client(cmdq, NULL, 1)) != NULL)
+   format_client(ft, c);
@@ -129 +137,2 @@ enum cmd_retval cmd_split_window_exec(struct cmd *self, struct cmd_q *cmdq) {
-  if (fd != -1) close(fd);
+ if (fd != -1)
+  close(fd);
@@ -133 +142,2 @@ error:
-  if (new_wp != NULL) window_remove_pane(w, new_wp);
+ if (new_wp != NULL)
+  window_remove_pane(w, new_wp);
@@ -136 +146,2 @@ error:
-  if (fd != -1) close(fd);
+ if (fd != -1)
+  close(fd);
