--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/e588ddb5-d518067b-c190c732/format.no_comments_merged.c
@@ -15 +15,4 @@ void format_window_pane_tabs(struct format_tree *, struct window_pane *);
-int format_cmp(struct format_entry *fe1, struct format_entry *fe2) {
+RB_GENERATE(format_tree, format_entry, entry, format_cmp);
+int
+format_cmp(struct format_entry *fe1, struct format_entry *fe2)
+{
@@ -357,0 +407 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
+ const char *cmd;
@@ -381,7 +433,2 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
-  if (wp->cmd != NULL) format_add(ft, "pane_start_command", "%s", wp->cmd);
-<<<<<<< HEAD
-  if ((cwd = osdep_get_cwd(wp->fd)) != NULL)
-    format_add(ft, "pane_current_path", "%s", cwd);
-  if ((cmd = format_get_command(wp)) != NULL) {
-|||||||
-  if (wp->cwd != NULL) format_add(ft, "pane_start_path", "%s", wp->cwd);
+ if (wp->cmd != NULL)
+  format_add(ft, "pane_start_command", "%s", wp->cmd);
@@ -390,2 +436,0 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
-  if ((cmd = osdep_get_name(wp->fd, wp->tty)) != NULL) {
-=======
@@ -393 +437,0 @@ void format_window_pane(struct format_tree *ft, struct window_pane *wp) {
->>>>>>> d518067be6220757a9101ca27fff14d5f599c410
