diff --git a/./output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_mergegen.cpp b/./output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_merged.cpp
index 659f6a3..6501e16 100644
--- a/./output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_mergegen.cpp
+++ b/./output/duckdb/3444bc24-059ac75f-86446616/tools@sqlite3_api_wrapper@sqlite3_api_wrapper.no_comments_merged.cpp
@@ -34 +34,2 @@ extern "C" {
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, const char *null_value, int columnar, char thousands, char decimal_sep);
+char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, const char *null_value,
+                            int columnar, char thousands, char decimal_sep);
@@ -193 +194,2 @@ int sqlite3_prepare_v2(sqlite3 *db,
-char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, const char *null_value, int columnar, char thousand_separator, char decimal_separator) {
+char *sqlite3_print_duckbox(sqlite3_stmt *pStmt, size_t max_rows, size_t max_width, const char *null_value,
+                            int columnar, char thousand_separator, char decimal_separator) {
