--- a/./output/grpc/8bab28db-b6a7d550-cbf18222/src@core@ext@filters@client_channel@xds@xds_api.no_comments_conflict.cc
+++ b/./output/grpc/8bab28db-b6a7d550-cbf18222/src@core@ext@filters@client_channel@xds@xds_api.no_comments_merged.cc
@@ -847 +847,6 @@ grpc_error* RouteConfigParse(
-      if (prefix.size > 0) {
+      if (prefix.size == 1) {
+        if (prefix.data[0] != '/') {
+          return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
+              "Prefix is not empty and does starting with a /");
+        }
+      } else if (prefix.size > 1) {
@@ -911 +915,0 @@ grpc_error* RouteConfigParse(
-<<<<<<< HEAD
@@ -920,10 +923,0 @@ grpc_error* RouteConfigParse(
-||||||| cbf18222af
-  const upb_strview prefix = envoy_api_v2_route_RouteMatch_prefix(match);
-  if (!upb_strview_eql(prefix, upb_strview_makez(""))) {
-    return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Prefix is not empty string.");
-=======
-  const upb_strview prefix = envoy_api_v2_route_RouteMatch_prefix(match);
-  if (!upb_strview_eql(prefix, upb_strview_makez("")) &&
-      !upb_strview_eql(prefix, upb_strview_makez("/"))) {
-    return GRPC_ERROR_CREATE_FROM_STATIC_STRING("Prefix is not \"\" or \"/\".");
->>>>>>> b6a7d550
