--- a/home/whalien/codebase/python/mergebot-eva/output/grpc/8bab28db-b6a7d550-cbf18222/src@core@ext@filters@client_channel@xds@xds_api.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/grpc/8bab28db-b6a7d550-cbf18222/src@core@ext@filters@client_channel@xds@xds_api.no_comments_merged.cc
@@ -78,6 +78,4 @@ bool XdsApi::DropConfig::ShouldDrop(const std::string** category_name) const {
-const char* XdsApi::kEdsTypeUrl =
-    "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
-const char* XdsApi::kEdsTypeUrl =
-    "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
-const char* XdsApi::kEdsTypeUrl =
-    "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment";
+const char* XdsApi::kLdsTypeUrl = "type.googleapis.com/envoy.api.v2.Listener";
+const char* XdsApi::kRdsTypeUrl =
+    "type.googleapis.com/envoy.api.v2.RouteConfiguration";
+const char* XdsApi::kCdsTypeUrl = "type.googleapis.com/envoy.api.v2.Cluster";
@@ -849 +847,6 @@ grpc_error* RouteConfigParse(
-      if (prefix.size > 0) {
+      if (prefix.size == 1) {
+        if (prefix.data[0] != '/') {
+          return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
+              "Prefix is not empty and does starting with a /");
+        }
+      } else if (prefix.size > 1) {
