diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/1f133521-49fdce79-1cdb2324/src@cpp@common@version_cc.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/1f133521-49fdce79-1cdb2324/src@cpp@common@version_cc.no_comments_merged.cc
index 1f5762f..539f939 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/1f133521-49fdce79-1cdb2324/src@cpp@common@version_cc.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/1f133521-49fdce79-1cdb2324/src@cpp@common@version_cc.no_comments_merged.cc
@@ -3,6 +3 @@ namespace grpc {
-std::string Version() < < < < < < < HEAD { return "1.32.0-dev"; }
-|||||||
-{ return "1.31.0"; }
-=======
-{ return "1.31.1"; }
->>>>>>> 49fdce79c30651e201c7b4e8d253e1a9d5b99670
+std::string Version() { return "1.32.0-dev"; }
