diff --git a/./output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@reporter.no_comments_conflict.cc b/./output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@reporter.no_comments_merged.cc
index 97de328..450c1a2 100644
--- a/./output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@reporter.no_comments_conflict.cc
+++ b/./output/art/1bc767f0-37c19554-c7c9497d/runtime@metrics@reporter.no_comments_merged.cc
@@ -156 +155,0 @@ void MetricsReporter::ReportMetrics() {
-<<<<<<< HEAD
@@ -158,7 +156,0 @@ void MetricsReporter::ReportMetrics() {
-||||||| c7c9497dd4
-  for (auto& backend : backends_) {
-    metrics->ReportAllMetrics(backend.get());
-  }
-=======
-  metrics->ReportAllMetricsAndResetValueMetrics(ToRawPointers(backends_));
->>>>>>> 37c19554
