diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_merged.h
index 03a5f62..1bb1274 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_merged.h
@@ -104,2 +104,2 @@ class DBImpl : public DB {
-      const TransactionLogIterator::ReadOptions& read_options =
-          TransactionLogIterator::ReadOptions());
+      const TransactionLogIterator::ReadOptions&
+          read_options = TransactionLogIterator::ReadOptions());
@@ -117,4 +117,4 @@ class DBImpl : public DB {
-  Iterator* TEST_NewInternalIterator(
-      ColumnFamilyHandle* column_family = nullptr);
-  int64_t TEST_MaxNextLevelOverlappingBytes(
-      ColumnFamilyHandle* column_family = nullptr);
+  Iterator* TEST_NewInternalIterator(ColumnFamilyHandle* column_family =
+                                         nullptr);
+  int64_t TEST_MaxNextLevelOverlappingBytes(ColumnFamilyHandle* column_family =
+                                                nullptr);
@@ -125,4 +125,3 @@ class DBImpl : public DB {
-  void TEST_SetDefaultTimeToCheck(
-      uint64_t default_interval_to_delete_obsolete_WAL) {
-    default_interval_to_delete_obsolete_WAL_ =
-        default_interval_to_delete_obsolete_WAL;
+  void TEST_SetDefaultTimeToCheck(uint64_t default_interval_to_delete_obsolete_WAL)
+  {
+    default_interval_to_delete_obsolete_WAL_ = default_interval_to_delete_obsolete_WAL;
@@ -134 +133,2 @@ class DBImpl : public DB {
-      return candidate_files.size() || sst_delete_files.size() ||
+      return candidate_files.size() ||
+        sst_delete_files.size() ||
@@ -161 +161,2 @@ class DBImpl : public DB {
-  void FindObsoleteFiles(DeletionState& deletion_state, bool force,
+  void FindObsoleteFiles(DeletionState& deletion_state,
+                         bool force,
@@ -185,2 +185,0 @@ class DBImpl : public DB {
-                                   DeletionState& deletion_state);
-  Status FlushMemTableToOutputFile(bool* madeProgress,
@@ -193 +191,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
@@ -195,8 +193,2 @@ class DBImpl : public DB {
-                          VersionEdit* edit, uint64_t* filenumber);
-||||||| e5fa4944f
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber);
-=======
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber, LogBuffer* log_buffer);
->>>>>>> d5de22dc099311b59a9e3d735702b6b9f18c7855
+                          VersionEdit* edit, uint64_t* filenumber,
+                          LogBuffer* log_buffer);
@@ -224 +216,2 @@ class DBImpl : public DB {
-                          DeletionState& deletion_state, LogBuffer* log_buffer);
+                          DeletionState& deletion_state,
+                          LogBuffer* log_buffer);
@@ -232 +225,2 @@ class DBImpl : public DB {
-                                VectorLogPtr& log_files, WalFileType type);
+                                VectorLogPtr& log_files,
+                                WalFileType type);
@@ -242,0 +237,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -295 +293,2 @@ class DBImpl : public DB {
-      SequenceNumber in, std::vector<SequenceNumber>& snapshots,
+    SequenceNumber in,
+    std::vector<SequenceNumber>& snapshots,
@@ -297,7 +295,0 @@ class DBImpl : public DB {
-  Status GetImpl(const ReadOptions& options, ColumnFamilyHandle* column_family,
-                 const Slice& key, std::string* value,
-                 bool* value_found = nullptr);
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
@@ -307,3 +299,6 @@ class DBImpl : public DB {
-  virtual Status GetPropertiesOfAllTables(
-      ColumnFamilyHandle* column_family,
-      TablePropertiesCollection* props) override;
+  virtual Status GetPropertiesOfAllTables(ColumnFamilyHandle* column_family,
+                                          TablePropertiesCollection* props)
+      override;
+  Status GetImpl(const ReadOptions& options, ColumnFamilyHandle* column_family,
+                 const Slice& key, std::string* value,
+                 bool* value_found = nullptr);
