diff --git a/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_mergegen.h b/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_merged.h
index dd24ecc..1bb1274 100644
--- a/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_mergegen.h
+++ b/./output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_merged.h
@@ -185 +185,3 @@ class DBImpl : public DB {
-Status FlushMemTableToOutputFile(ColumnFamilyData* cfd, bool* madeProgress, DeletionState& deletion_state, LogBuffer* log_buffer);
+  Status FlushMemTableToOutputFile(ColumnFamilyData* cfd, bool* madeProgress,
+                                   DeletionState& deletion_state,
+                                   LogBuffer* log_buffer);
@@ -188 +190,5 @@ Status FlushMemTableToOutputFile(ColumnFamilyData* cfd, bool* madeProgress, Dele
-offset because it holds the mutex
+  Status WriteLevel0TableForRecovery(ColumnFamilyData* cfd, MemTable* mem,
+                                     VersionEdit* edit);
+  Status WriteLevel0Table(ColumnFamilyData* cfd, autovector<MemTable*>& mems,
+                          VersionEdit* edit, uint64_t* filenumber,
+                          LogBuffer* log_buffer);
