--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@db_impl.no_comments_merged.h
@@ -185,2 +185,0 @@ class DBImpl : public DB {
-                                   DeletionState& deletion_state);
-  Status FlushMemTableToOutputFile(bool* madeProgress,
@@ -193 +191,0 @@ class DBImpl : public DB {
-<<<<<<< HEAD
@@ -195,8 +193,2 @@ class DBImpl : public DB {
-                          VersionEdit* edit, uint64_t* filenumber);
-||||||| e5fa4944f
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber);
-=======
-  Status WriteLevel0Table(autovector<MemTable*>& mems, VersionEdit* edit,
-                          uint64_t* filenumber, LogBuffer* log_buffer);
->>>>>>> d5de22dc099311b59a9e3d735702b6b9f18c7855
+                          VersionEdit* edit, uint64_t* filenumber,
+                          LogBuffer* log_buffer);
@@ -242,0 +237,4 @@ class DBImpl : public DB {
+  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
+  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
+      const ReadOptions& options, ColumnFamilyData* cfd,
+      uint64_t* superversion_number);
@@ -297,7 +295,0 @@ class DBImpl : public DB {
-  Status GetImpl(const ReadOptions& options, ColumnFamilyHandle* column_family,
-                 const Slice& key, std::string* value,
-                 bool* value_found = nullptr);
-  Status ReFitLevel(ColumnFamilyData* cfd, int level, int target_level = -1);
-  std::pair<Iterator*, Iterator*> GetTailingIteratorPair(
-      const ReadOptions& options, ColumnFamilyData* cfd,
-      uint64_t* superversion_number);
@@ -307,3 +299,6 @@ class DBImpl : public DB {
-  virtual Status GetPropertiesOfAllTables(
-      ColumnFamilyHandle* column_family,
-      TablePropertiesCollection* props) override;
+  virtual Status GetPropertiesOfAllTables(ColumnFamilyHandle* column_family,
+                                          TablePropertiesCollection* props)
+      override;
+  Status GetImpl(const ReadOptions& options, ColumnFamilyHandle* column_family,
+                 const Slice& key, std::string* value,
+                 bool* value_found = nullptr);
