diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@version_set.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@version_set.no_comments_merged.h
index b391209..b2b394c 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@version_set.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/d4f2c610-d5de22dc-e5fa4944/db@version_set.no_comments_merged.h
@@ -18,3 +18 @@ namespace rocksdb {
-namespace log {
-class Writer;
-}
+namespace log { class Writer; }
@@ -26,0 +25,2 @@ class MemTable;
+class Version;
+class VersionSet;
@@ -31,2 +30,0 @@ class TableCache;
-class Version;
-class VersionSet;
@@ -34,2 +32,4 @@ extern int FindFile(const InternalKeyComparator& icmp,
-                    const std::vector<FileMetaData*>& files, const Slice& key);
-extern bool SomeFileOverlapsRange(const InternalKeyComparator& icmp,
+                    const std::vector<FileMetaData*>& files,
+                    const Slice& key);
+extern bool SomeFileOverlapsRange(
+    const InternalKeyComparator& icmp,
@@ -49,2 +49,3 @@ class Version {
-           Status* status, MergeContext* merge_context, GetStats* stats,
-           const Options& db_option, bool* value_found = nullptr);
+           Status* status, MergeContext* merge_context,
+           GetStats* stats, const Options& db_option,
+           bool* value_found = nullptr);
@@ -77,3 +78,3 @@ class Version {
-      unsigned int
-          index);
-  bool OverlapInLevel(int level, const Slice* smallest_user_key,
+      unsigned int index);
+  bool OverlapInLevel(int level,
+                      const Slice* smallest_user_key,
@@ -150,4 +151,8 @@ class VersionSet {
-  Status LogAndApply(
-      ColumnFamilyData* column_family_data, VersionEdit* edit, port::Mutex* mu,
-      Directory* db_directory = nullptr, bool new_descriptor_log = false,
-      const ColumnFamilyOptions* column_family_options = nullptr);
+  Status LogAndApply(ColumnFamilyData* column_family_data, VersionEdit* edit,
+                     port::Mutex* mu, Directory* db_directory = nullptr,
+                     bool new_descriptor_log = false,
+                     const ColumnFamilyOptions* column_family_options =
+                         nullptr);
+  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
+  static Status ListColumnFamilies(std::vector<std::string>* column_families,
+                                   const std::string& dbname, Env* env);
@@ -192 +197,2 @@ class VersionSet {
-  void GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata);
+  void GetLiveFilesMetaData(
+    std::vector<LiveFileMetaData> *metadata);
@@ -198 +203,0 @@ class VersionSet {
-  friend class Compaction;
@@ -230,4 +234,0 @@ class VersionSet {
- public:
-  Status Recover(const std::vector<ColumnFamilyDescriptor>& column_families);
-  static Status ListColumnFamilies(std::vector<std::string>* column_families,
-                                   const std::string& dbname, Env* env);
