--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/d785c37e-84612a81-b2a9f411/cmd.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/d785c37e-84612a81-b2a9f411/cmd.no_comments_merged.c
@@ -4 +3,0 @@
-#include <paths.h>
@@ -816 +964,0 @@ const char *cmd_get_default_path(struct cmd_ctx *ctx, const char *cwd) {
-<<<<<<< HEAD
@@ -818,5 +965,0 @@ const char *cmd_get_default_path(struct cmd_ctx *ctx, const char *cwd) {
-|||||||
-      root = get_proc_cwd(s->curw->window->active->pid);
-=======
-      root = get_proc_cwd(s->curw->window->active->fd);
->>>>>>> 84612a8196341839c5a18362b68047aa1b011a50
