diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_merged.cpp
index 7a0daf7..f1f4f59 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_merged.cpp
@@ -64 +63,0 @@ void logFatalTexture(const char* msg, const GrBackendTexture& tex, ui::Dataspace
-<<<<<<< HEAD
@@ -92,19 +90,0 @@ void logFatalTexture(const char* msg, const GrBackendTexture& tex, ui::Dataspace
-|||||||
-                             "\n\tGrBackendTexture: (%i x %i) hasMipmaps: %i isProtected: %i "
-                             "texType: %i"
-                             "\n\t\tGrGLTextureInfo: success: %i fTarget: %u fFormat: %u colorType "
-                             "%i",
-                             msg, tex.isValid(), dataspace, tex.width(), tex.height(),
-                             tex.hasMipmaps(), tex.isProtected(),
-                             static_cast<int>(tex.textureType()), retrievedTextureInfo,
-                             textureInfo.fTarget, textureInfo.fFormat, colorType);
-=======
-                             "\n\tGrBackendTexture: (%i x %i) hasMipmaps: %i isProtected: %i "
-                             "texType: %i"
-                             "\n\t\tGrGLTextureInfo: success: %i fTarget: %u fFormat: %u colorType "
-                             "%i",
-                             msg, tex.isValid(), static_cast<int32_t>(dataspace), tex.width(),
-                             tex.height(), tex.hasMipmaps(), tex.isProtected(),
-                             static_cast<int>(tex.textureType()), retrievedTextureInfo,
-                             textureInfo.fTarget, textureInfo.fFormat, colorType);
->>>>>>> 906306cacf0dad2d1822ab0eda36fe95ce0e37cf
