diff --git a/./output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_conflict.cpp b/./output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_merged.cpp
index 36bc636..f1f4f59 100644
--- a/./output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_conflict.cpp
+++ b/./output/frameworks_native/18e1c0d6-906306ca-443973f5/libs@renderengine@skia@AutoBackendTexture.no_comments_merged.cpp
@@ -59 +58,0 @@ void logFatalTexture(const char* msg, const GrBackendTexture& tex, ui::Dataspace
-<<<<<<< HEAD
@@ -92,19 +90,0 @@ void logFatalTexture(const char* msg, const GrBackendTexture& tex, ui::Dataspace
-||||||| 443973f505
-    GrGLTextureInfo textureInfo;
-    bool retrievedTextureInfo = tex.getGLTextureInfo(&textureInfo);
-    LOG_ALWAYS_FATAL("%s isTextureValid:%d dataspace:%d"
-                     "\n\tGrBackendTexture: (%i x %i) hasMipmaps: %i isProtected: %i texType: %i"
-                     "\n\t\tGrGLTextureInfo: success: %i fTarget: %u fFormat: %u colorType %i",
-                     msg, tex.isValid(), dataspace, tex.width(), tex.height(), tex.hasMipmaps(),
-                     tex.isProtected(), static_cast<int>(tex.textureType()), retrievedTextureInfo,
-                     textureInfo.fTarget, textureInfo.fFormat, colorType);
-=======
-    GrGLTextureInfo textureInfo;
-    bool retrievedTextureInfo = tex.getGLTextureInfo(&textureInfo);
-    LOG_ALWAYS_FATAL("%s isTextureValid:%d dataspace:%d"
-                     "\n\tGrBackendTexture: (%i x %i) hasMipmaps: %i isProtected: %i texType: %i"
-                     "\n\t\tGrGLTextureInfo: success: %i fTarget: %u fFormat: %u colorType %i",
-                     msg, tex.isValid(), static_cast<int32_t>(dataspace), tex.width(), tex.height(),
-                     tex.hasMipmaps(), tex.isProtected(), static_cast<int>(tex.textureType()),
-                     retrievedTextureInfo, textureInfo.fTarget, textureInfo.fFormat, colorType);
->>>>>>> 906306ca
