diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_merged.c
index 3be5c0a..e3ae183 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_merged.c
@@ -41,6 +40,0 @@ static int cliConnect(int force) {
-<<<<<<< HEAD
-  if (fd == ANET_ERR) {
-    if (config.hostsocket == NULL) {
-|||||||
-  if (fd == ANET_ERR) {
-=======
@@ -49 +43 @@ static int cliConnect(int force) {
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
+        if (config.hostsocket == NULL) {
@@ -53,5 +46,0 @@ static int cliConnect(int force) {
-      if (fd == ANET_ERR) {
-        fprintf(stderr, "Could not connect to Redis at %s: %s",
-                config.hostsocket, err);
-        return -1;
-      }
@@ -91 +80,2 @@ static int cliReadSingleLineReply(int fd, int quiet) {
-  if (!quiet) printf("%s", reply);
+    if (!quiet)
+        printf("%s", reply);
@@ -103,15 +93,5 @@ static void printStringRepr(char *s, int len) {
-      case '\n':
-        printf("\\n");
-        break;
-      case '\r':
-        printf("\\r");
-        break;
-      case '\t':
-        printf("\\t");
-        break;
-      case '\a':
-        printf("\\a");
-        break;
-      case '\b':
-        printf("\\b");
-        break;
+        case '\n': printf("\\n"); break;
+        case '\r': printf("\\r"); break;
+        case '\t': printf("\\t"); break;
+        case '\a': printf("\\a"); break;
+        case '\b': printf("\\b"); break;
@@ -182 +162,2 @@ static int cliReadReply(int fd) {
-        (nread == 0 || (nread == -1 && errno == ECONNRESET))) {
+            (nread == 0 || (nread == -1 && errno == ECONNRESET)))
+        {
@@ -212 +193,2 @@ static int selectDb(int fd) {
-  if (config.dbnum == 0) return 0;
+    if (config.dbnum == 0)
+        return 0;
@@ -227,2 +209 @@ static void showInteractiveHelp(void) {
-      "Welcome to redis-cli " REDIS_VERSION
-      "!\n"
+    "Welcome to redis-cli " REDIS_VERSION "!\n"
@@ -251,2 +232,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-  if (!strcasecmp(command, "subscribe") || !strcasecmp(command, "psubscribe"))
-    config.pubsub_mode = 1;
+    if (!strcasecmp(command,"subscribe") ||
+        !strcasecmp(command,"psubscribe")) config.pubsub_mode = 1;
@@ -261 +242,2 @@ static int cliSendCommand(int argc, char **argv, int repeat) {
-    cmd = sdscatprintf(cmd, "$%lu\r\n", (unsigned long)sdslen(argv[j]));
+        cmd = sdscatprintf(cmd,"$%lu\r\n",
+            (unsigned long)sdslen(argv[j]));
@@ -317,4 +299,3 @@ static int parseOptions(int argc, char **argv) {
-              "Starting interactive mode using -i is deprecated. Interactive "
-              "mode is started\n"
-              "by default when redis-cli is executed without a command to "
-              "execute.\n");
+"Starting interactive mode using -i is deprecated. Interactive mode is started\n"
+"by default when redis-cli is executed without a command to execute.\n"
+            );
@@ -322,2 +303 @@ static int parseOptions(int argc, char **argv) {
-      fprintf(
-          stderr,
+            fprintf(stderr,
@@ -326 +306,2 @@ static int parseOptions(int argc, char **argv) {
-          "automatically used as last argument.\n");
+"automatically used as last argument.\n"
+            );
@@ -341,2 +322 @@ static sds readArgFromStdin(void) {
-    if (nread == 0)
-      break;
+        if (nread == 0) break;
@@ -352,31 +332,2 @@ static void usage() {
-<<<<<<< HEAD
-  fprintf(
-      stderr,
-      "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a "
-      "authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-s "
-          "/path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 "
-          "arg2 ... arg(N-1)\n");
-  fprintf(stderr,
-          "\nIf a pipe from standard input is detected this data is used as "
-          "last argument.\n\n");
-  fprintf(stderr, "example: cat /etc/passwd | redis-cli set my_passwd\n");
-|||||||
-  fprintf(stderr,
-          "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r "
-          "repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -c [-h host] [-p port] [-a authpw] "
-          "[-r repeat_times] [-n db_num] cmd arg1 arg2 ... arg(N-1)\n");
-  fprintf(stderr,
-          "\nIf a pipe from standard input is detected this data is used as "
-          "last argument.\n\n");
-  fprintf(stderr, "example: cat /etc/passwd | redis-cli set my_passwd\n");
-=======
-  fprintf(stderr,
-          "usage: redis-cli [-iv] [-h host] [-p port] [-a authpw] [-r "
-          "repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
-  fprintf(stderr,
-          "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... "
-          "arg(N-1)\n\n");
+    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
+    fprintf(stderr, "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... arg(N-1)\n\n");
@@ -384 +334,0 @@ static void usage() {
->>>>>>> b4f2e412d087bae0a523fe6ea40fcad30fe74b5b
@@ -387,3 +337 @@ static void usage() {
-  fprintf(stderr,
-          "\nRun in interactive mode: redis-cli -i or just don't pass any "
-          "command\n");
+    fprintf(stderr, "\nRun in interactive mode: redis-cli -i or just don't pass any command\n");
@@ -395 +343,2 @@ static char **convertToSds(int count, char **args) {
-  for (j = 0; j < count; j++) sds[j] = sdsnew(args[j]);
+  for(j = 0; j < count; j++)
+    sds[j] = sdsnew(args[j]);
@@ -414 +363,2 @@ static void repl() {
-            strcasecmp(argv[0], "exit") == 0) {
+                    strcasecmp(argv[0],"exit") == 0)
+                {
@@ -429 +379,2 @@ static void repl() {
-      for (j = 0; j < argc; j++) sdsfree(argv[j]);
+            for (j = 0; j < argc; j++)
+                sdsfree(argv[j]);
