diff --git a/./output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_mergegen.c b/./output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_merged.c
index 03ab3fa..e3ae183 100644
--- a/./output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_mergegen.c
+++ b/./output/redis/4fe83b55-b4f2e412-0c7a9dec/src@redis-cli.no_comments_merged.c
@@ -41 +41,7 @@ static int cliConnect(int force) {
-if (fd == ANET_ERR || force) { if (force) close(fd); if (config.hostsocket == NULL) { fd = anetTcpConnect(err,config.hostip,config.hostport); } else { fd = anetUnixConnect(err,config.hostsocket); }
+    if (fd == ANET_ERR || force) {
+        if (force) close(fd);
+        if (config.hostsocket == NULL) {
+            fd = anetTcpConnect(err,config.hostip,config.hostport);
+        } else {
+            fd = anetUnixConnect(err,config.hostsocket);
+        }
@@ -326 +332,3 @@ static void usage() {
-fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n"); fprintf(stderr, "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... arg(N-1)\n\n"); fprintf(stderr, "example: cat /etc/passwd | redis-cli -x set my_passwd\n");
+    fprintf(stderr, "usage: redis-cli [-iv] [-h host] [-p port] [-s /path/to/socket] [-a authpw] [-r repeat_times] [-n db_num] cmd arg1 arg2 arg3 ... argN\n");
+    fprintf(stderr, "usage: echo \"argN\" | redis-cli -x [options] cmd arg1 arg2 ... arg(N-1)\n\n");
+    fprintf(stderr, "example: cat /etc/passwd | redis-cli -x set my_passwd\n");
