--- a/home/whalien/codebase/cpp/mergebot/eva/output/grpc/dbec2415-eaa67051-79c12947/include@grpcpp@security@server_credentials.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/grpc/dbec2415-eaa67051-79c12947/include@grpcpp@security@server_credentials.no_comments_merged.h
@@ -9,2 +9 @@
-struct grpc_server
-#include <grpcpp/security/server_credentials_impl.h>
+struct grpc_server;
@@ -15,17 +14,16 @@ namespace grpc {
-std::shared_ptr<ServerCredentials> SslServerCredentials(
-    const SslServerCredentialsOptions& options);
-std::shared_ptr<ServerCredentials> InsecureServerCredentials();
-namespace experimental {
-std::shared_ptr<ServerCredentials> AltsServerCredentials(
-    const AltsServerCredentialsOptions& options);
-std::shared_ptr<ServerCredentials> LocalServerCredentials(
-    grpc_local_connect_type type);
-std::shared_ptr<ServerCredentials> TlsServerCredentials(
-    const ::grpc_impl::experimental::TlsCredentialsOptions& options);
-std::shared_ptr<ServerCredentials> TlsServerCredentials(
-    const ::grpc::experimental::TlsCredentialsOptions& options);
-struct AltsServerCredentialsOptions {};
-std::shared_ptr<ServerCredentials> AltsServerCredentials(
-    const AltsServerCredentialsOptions& options);
-}
-struct SslServerCredentialsOptions;
+struct SslServerCredentialsOptions {
+  SslServerCredentialsOptions()
+      : force_client_auth(false),
+        client_certificate_request(GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE) {}
+  SslServerCredentialsOptions(
+      grpc_ssl_client_certificate_request_type request_type)
+      : force_client_auth(false), client_certificate_request(request_type) {}
+  struct PemKeyCertPair {
+    std::string private_key;
+    std::string cert_chain;
+  };
+  std::string pem_root_certs;
+  std::vector<PemKeyCertPair> pem_key_cert_pairs;
+  bool force_client_auth;
+  grpc_ssl_client_certificate_request_type client_certificate_request;
+};
@@ -40,0 +39,15 @@ class ServerCredentials {
+std::shared_ptr<ServerCredentials> SslServerCredentials(
+    const grpc::SslServerCredentialsOptions& options);
+std::shared_ptr<ServerCredentials> InsecureServerCredentials();
+namespace experimental {
+struct AltsServerCredentialsOptions {
+};
+std::shared_ptr<ServerCredentials> AltsServerCredentials(
+    const AltsServerCredentialsOptions& options);
+std::shared_ptr<ServerCredentials> AltsServerCredentials(
+    const AltsServerCredentialsOptions& options);
+std::shared_ptr<ServerCredentials> LocalServerCredentials(
+    grpc_local_connect_type type);
+std::shared_ptr<ServerCredentials> TlsServerCredentials(
+    const experimental::TlsCredentialsOptions& options);
+}
