diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_mergebot.h b/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_merged.h
index c50686f..710959c 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@util.no_comments_merged.h
@@ -11 +10,0 @@
-using namespace std::chrono_literals;
@@ -15,2 +14,2 @@ using namespace std::chrono_literals;
-int create_socket(const char* name, int type, mode_t perm, uid_t uid, gid_t gid,
-                  const char* socketcon);
+int create_socket(const char *name, int type, mode_t perm,
+                  uid_t uid, gid_t gid, const char *socketcon);
