diff --git a/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@init.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@init.no_comments_merged.cpp
index 99821a7..7947f95 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@init.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/09e55b91-e2e0d9ca-6e531991/init@init.no_comments_merged.cpp
@@ -37 +36,0 @@
-#include <set>
@@ -41,2 +39,0 @@
-#include "fs_mgr.h"
-#include "fs_mgr_avb.h"
@@ -82 +79,2 @@ void register_epoll_handler(int fd, void (*fn)()) {
-int add_environment(const char* key, const char* val) {
+int add_environment(const char *key, const char *val)
+{
@@ -103 +101,2 @@ int add_environment(const char* key, const char* val) {
-bool start_waiting_for_property(const char* name, const char* value) {
+bool start_waiting_for_property(const char *name, const char *value)
+{
@@ -112,2 +111,2 @@ bool start_waiting_for_property(const char* name, const char* value) {
-        LOG(INFO) << "start_waiting_for_property(\"" << name << "\", \"" << value
-                  << "\"): already set";
+        LOG(INFO) << "start_waiting_for_property(\""
+                  << name << "\", \"" << value << "\"): already set";
@@ -129 +128,2 @@ void property_changed(const std::string& name, const std::string& value) {
-static void restart_processes() {
+static void restart_processes()
+{
@@ -131,2 +131,3 @@ static void restart_processes() {
-    ServiceManager::GetInstance().ForEachServiceWithFlags(
-        SVC_RESTARTING, [](Service* s) { s->RestartIfNeeded(&process_needs_restart_at); });
+    ServiceManager::GetInstance().ForEachServiceWithFlags(SVC_RESTARTING, [](Service* s) {
+        s->RestartIfNeeded(&process_needs_restart_at);
+    });
@@ -160 +161,2 @@ static int wait_for_coldboot_done_action(const std::vector<std::string>& args) {
-static int mix_hwrng_into_linux_rng_action(const std::vector<std::string>& args) {
+static int mix_hwrng_into_linux_rng_action(const std::vector<std::string>& args)
+{
@@ -167 +169,2 @@ static int mix_hwrng_into_linux_rng_action(const std::vector<std::string>& args)
-    hwrandom_fd = TEMP_FAILURE_RETRY(open("/dev/hw_random", O_RDONLY | O_NOFOLLOW | O_CLOEXEC));
+    hwrandom_fd = TEMP_FAILURE_RETRY(
+            open("/dev/hw_random", O_RDONLY | O_NOFOLLOW | O_CLOEXEC));
@@ -177 +180,2 @@ static int mix_hwrng_into_linux_rng_action(const std::vector<std::string>& args)
-    urandom_fd = TEMP_FAILURE_RETRY(open("/dev/urandom", O_WRONLY | O_NOFOLLOW | O_CLOEXEC));
+    urandom_fd = TEMP_FAILURE_RETRY(
+            open("/dev/urandom", O_WRONLY | O_NOFOLLOW | O_CLOEXEC));
@@ -183 +187,2 @@ static int mix_hwrng_into_linux_rng_action(const std::vector<std::string>& args)
-        chunk_size = TEMP_FAILURE_RETRY(read(hwrandom_fd, buf, sizeof(buf) - total_bytes_written));
+        chunk_size = TEMP_FAILURE_RETRY(
+                read(hwrandom_fd, buf, sizeof(buf) - total_bytes_written));
@@ -213 +218,2 @@ static void security_failure() {
-static bool set_highest_available_option_value(std::string path, int min, int max) {
+static bool set_highest_available_option_value(std::string path, int min, int max)
+{
@@ -255 +261,2 @@ static bool __attribute__((unused)) set_mmap_rnd_bits_min(int start, int min, bo
-static int set_mmap_rnd_bits_action(const std::vector<std::string>& args) {
+static int set_mmap_rnd_bits_action(const std::vector<std::string>& args)
+{
@@ -258 +265,2 @@ static int set_mmap_rnd_bits_action(const std::vector<std::string>& args) {
-    if (set_mmap_rnd_bits_min(33, 24, false) && set_mmap_rnd_bits_min(16, 16, true)) {
+    if (set_mmap_rnd_bits_min(33, 24, false)
+            && set_mmap_rnd_bits_min(16, 16, true)) {
@@ -262 +270,2 @@ static int set_mmap_rnd_bits_action(const std::vector<std::string>& args) {
-    if (set_mmap_rnd_bits_min(32, 32, false) && set_mmap_rnd_bits_min(16, 16, true)) {
+    if (set_mmap_rnd_bits_min(32, 32, false)
+            && set_mmap_rnd_bits_min(16, 16, true)) {
@@ -284 +293,2 @@ static int set_mmap_rnd_bits_action(const std::vector<std::string>& args) {
-static int set_kptr_restrict_action(const std::vector<std::string>& args) {
+static int set_kptr_restrict_action(const std::vector<std::string>& args)
+{
@@ -292 +302,2 @@ static int set_kptr_restrict_action(const std::vector<std::string>& args) {
-static int keychord_init_action(const std::vector<std::string>& args) {
+static int keychord_init_action(const std::vector<std::string>& args)
+{
@@ -296 +307,2 @@ static int keychord_init_action(const std::vector<std::string>& args) {
-static int console_init_action(const std::vector<std::string>& args) {
+static int console_init_action(const std::vector<std::string>& args)
+{
@@ -330,30 +342,6 @@ static void export_kernel_boot_props() {
-        {
-            "ro.boot.serialno",
-            "ro.serialno",
-            "",
-        },
-        {
-            "ro.boot.mode",
-            "ro.bootmode",
-            "unknown",
-        },
-        {
-            "ro.boot.baseband",
-            "ro.baseband",
-            "unknown",
-        },
-        {
-            "ro.boot.bootloader",
-            "ro.bootloader",
-            "unknown",
-        },
-        {
-            "ro.boot.hardware",
-            "ro.hardware",
-            "unknown",
-        },
-        {
-            "ro.boot.revision",
-            "ro.revision",
-            "0",
-        },
+        { "ro.boot.serialno", "ro.serialno", "", },
+        { "ro.boot.mode", "ro.bootmode", "unknown", },
+        { "ro.boot.baseband", "ro.baseband", "unknown", },
+        { "ro.boot.bootloader", "ro.bootloader", "unknown", },
+        { "ro.boot.hardware", "ro.hardware", "unknown", },
+        { "ro.boot.revision", "ro.revision", "0", },
@@ -363,2 +351 @@ static void export_kernel_boot_props() {
-        property_set(prop_map[i].dst_prop,
-                     (!value.empty()) ? value.c_str() : prop_map[i].default_value);
+        property_set(prop_map[i].dst_prop, (!value.empty()) ? value.c_str() : prop_map[i].default_value);
@@ -376,2 +363 @@ static void process_kernel_dt() {
-        if (dp->d_type != DT_REG || !strcmp(dp->d_name, "compatible") ||
-            !strcmp(dp->d_name, "name")) {
+        if (dp->d_type != DT_REG || !strcmp(dp->d_name, "compatible") || !strcmp(dp->d_name, "name")) {
@@ -391 +377,2 @@ static void process_kernel_cmdline() {
-static int property_enable_triggers_action(const std::vector<std::string>& args) {
+static int property_enable_triggers_action(const std::vector<std::string>& args)
+{
@@ -395,3 +382,3 @@ static int property_enable_triggers_action(const std::vector<std::string>& args)
-static int queue_property_triggers_action(const std::vector<std::string>& args) {
-    ActionManager::GetInstance().QueueBuiltinAction(property_enable_triggers_action,
-                                                    "enable_property_trigger");
+static int queue_property_triggers_action(const std::vector<std::string>& args)
+{
+    ActionManager::GetInstance().QueueBuiltinAction(property_enable_triggers_action, "enable_property_trigger");
@@ -401 +388,2 @@ static int queue_property_triggers_action(const std::vector<std::string>& args)
-static void selinux_init_all_handles(void) {
+static void selinux_init_all_handles(void)
+{
@@ -409,2 +397 @@ static selinux_enforcing_status selinux_status_from_cmdline() {
-    import_kernel_cmdline(false,
-                          [&](const std::string& key, const std::string& value, bool in_qemu) {
+    import_kernel_cmdline(false, [&](const std::string& key, const std::string& value, bool in_qemu) {
@@ -417 +404,2 @@ static selinux_enforcing_status selinux_status_from_cmdline() {
-static bool selinux_is_enforcing(void) {
+static bool selinux_is_enforcing(void)
+{
@@ -429,2 +417,2 @@ static int audit_callback(void* data, security_class_t , char* buf, size_t len)
-    snprintf(buf, len, "property=%s pid=%d uid=%d gid=%d", d->name, d->cr->pid, d->cr->uid,
-             d->cr->gid);
+    snprintf(buf, len, "property=%s pid=%d uid=%d gid=%d", d->name,
+            d->cr->pid, d->cr->uid, d->cr->gid);
@@ -545,3 +533 @@ static constexpr const char plat_policy_cil_file[] = "/system/etc/selinux/plat_s
-static bool selinux_is_split_policy_device() {
-    return access(plat_policy_cil_file, R_OK) != -1;
-}
+static bool selinux_is_split_policy_device() { return access(plat_policy_cil_file, R_OK) != -1; }
@@ -790,2 +776,4 @@ int main(int argc, char** argv) {
-        parser.set_is_system_etc_init_loaded(parser.ParseConfig("/system/etc/init"));
-        parser.set_is_vendor_etc_init_loaded(parser.ParseConfig("/vendor/etc/init"));
+        parser.set_is_system_etc_init_loaded(
+                parser.ParseConfig("/system/etc/init"));
+        parser.set_is_vendor_etc_init_loaded(
+                parser.ParseConfig("/vendor/etc/init"));
