diff --git a/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_mergegen.cpp b/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_merged.cpp
index 0351281..ff609d4 100644
--- a/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_mergegen.cpp
+++ b/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_merged.cpp
@@ -50 +50 @@ static unique_ptr<FunctionData> CSVSniffBind(ClientContext &context, TableFuncti
-result->options.FromNamedParameters(input.named_parameters, context); result->options.Verify();
+ result->options.Verify();
