diff --git a/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_conflict.cpp b/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_merged.cpp
index 100a9ce..ff609d4 100644
--- a/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_conflict.cpp
+++ b/./output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_merged.cpp
@@ -50,3 +49,0 @@ static unique_ptr<FunctionData> CSVSniffBind(ClientContext &context, TableFuncti
-<<<<<<< HEAD
-||||||| 39f4ff4108
-=======
@@ -54 +50,0 @@ static unique_ptr<FunctionData> CSVSniffBind(ClientContext &context, TableFuncti
->>>>>>> 104cfacc
