--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/7a06a608-104cfacc-39f4ff41/src@function@table@sniff_csv.no_comments_merged.cpp
@@ -50,3 +49,0 @@ static unique_ptr<FunctionData> CSVSniffBind(ClientContext &context, TableFuncti
-<<<<<<< HEAD
-|||||||
-=======
@@ -54 +50,0 @@ static unique_ptr<FunctionData> CSVSniffBind(ClientContext &context, TableFuncti
->>>>>>> 104cfacc371eb020431f5ef61ae599b8dc969f07
