--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_merged.h
@@ -166,3 +205,2 @@ class AudioPolicyManager : public AudioPolicyInterface, public AudioPolicyManage
-    virtual status_t setDevicesRoleForCapturePreset(audio_source_t audioSource, device_role_t role,
-                                                    const AudioDeviceTypeAddrVector& devices);
-    virtual status_t addDevicesRoleForCapturePreset(audio_source_t audioSource, device_role_t role,
+        virtual status_t setDevicesRoleForCapturePreset(audio_source_t audioSource,
+                                                        device_role_t role,
@@ -170 +208 @@ class AudioPolicyManager : public AudioPolicyInterface, public AudioPolicyManage
-    virtual status_t removeDevicesRoleForCapturePreset(audio_source_t audioSource,
+        virtual status_t addDevicesRoleForCapturePreset(audio_source_t audioSource,
@@ -172,0 +211,3 @@ class AudioPolicyManager : public AudioPolicyInterface, public AudioPolicyManage
+        virtual status_t removeDevicesRoleForCapturePreset(
+                audio_source_t audioSource, device_role_t role,
+                const AudioDeviceTypeAddrVector& devices);
@@ -261,7 +328,3 @@ class AudioPolicyManager : public AudioPolicyInterface, public AudioPolicyManage
-    std::vector<volume_group_t> getVolumeGroups() const { return mEngine->getVolumeGroups(); }
-    VolumeSource toVolumeSource(volume_group_t volumeGroup) const {
-        return static_cast<VolumeSource>(volumeGroup);
-    }
-    VolumeSource toVolumeSource(const audio_attributes_t& attributes,
-                                bool fallbackOnDefault = true) const {
-        return toVolumeSource(mEngine->getVolumeGroupForAttributes(attributes, fallbackOnDefault));
+        std::vector<volume_group_t> getVolumeGroups() const
+        {
+            return mEngine->getVolumeGroups();
@@ -269,2 +332,3 @@ class AudioPolicyManager : public AudioPolicyInterface, public AudioPolicyManage
-    VolumeSource toVolumeSource(audio_stream_type_t stream, bool fallbackOnDefault = true) const {
-        return toVolumeSource(mEngine->getVolumeGroupForStreamType(stream, fallbackOnDefault));
+        VolumeSource toVolumeSource(volume_group_t volumeGroup) const
+        {
+            return static_cast<VolumeSource>(volumeGroup);
@@ -272,3 +336,16 @@ class AudioPolicyManager : public AudioPolicyInterface, public AudioPolicyManage
-    IVolumeCurves& getVolumeCurves(VolumeSource volumeSource) {
-        auto* curves =
-                mEngine->getVolumeCurvesForVolumeGroup(static_cast<volume_group_t>(volumeSource));
+        VolumeSource toVolumeSource(
+            const audio_attributes_t &attributes, bool fallbackOnDefault = true) const
+        {
+            return toVolumeSource(mEngine->getVolumeGroupForAttributes(
+                attributes, fallbackOnDefault));
+        }
+        VolumeSource toVolumeSource(
+            audio_stream_type_t stream, bool fallbackOnDefault = true) const
+        {
+            return toVolumeSource(mEngine->getVolumeGroupForStreamType(
+                stream, fallbackOnDefault));
+        }
+        IVolumeCurves &getVolumeCurves(VolumeSource volumeSource)
+        {
+          auto *curves = mEngine->getVolumeCurvesForVolumeGroup(
+              static_cast<volume_group_t>(volumeSource));
@@ -619 +800 @@ class AudioPolicyManager : public AudioPolicyInterface, public AudioPolicyManage
-}
+};
