diff --git a/./output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_mergegen.h b/./output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_merged.h
index eed320a..af66d2c 100644
--- a/./output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_mergegen.h
+++ b/./output/frameworks_av/320b0958-6e247f9f-5b03292d/services@audiopolicy@managerdefault@AudioPolicyManager.no_comments_merged.h
@@ -791 +791,8 @@ private:
-bool checkHapticCompatibilityOnSpatializerOutput(const audio_config_t* config, audio_session_t sessionId) const; void updateClientsInternalMute(const sp<SwAudioOutputDescriptor>& desc); float adjustDeviceAttenuationForAbsVolume(IVolumeCurves &curves, VolumeSource volumeSource, int index, const DeviceTypeSet &deviceTypes);
+        bool checkHapticCompatibilityOnSpatializerOutput(const audio_config_t* config,
+                                                         audio_session_t sessionId) const;
+        void updateClientsInternalMute(const sp<SwAudioOutputDescriptor>& desc);
+        float adjustDeviceAttenuationForAbsVolume(IVolumeCurves &curves,
+                                                  VolumeSource volumeSource,
+                                                  int index,
+                                                  const DeviceTypeSet &deviceTypes);
+        std::unordered_map<audio_devices_t, audio_attributes_t> mAbsoluteVolumeDrivingStreams;
