diff --git a/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_mergegen.cpp b/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
index aef1e45..f0641c1 100644
--- a/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_mergegen.cpp
+++ b/./output/frameworks_native/926bb129-5c7935f8-f2b19cdc/services@surfaceflinger@SurfaceFlinger.no_comments_merged.cpp
@@ -3709 +3709,2 @@ void SurfaceFlinger::setPowerModeInternal(const sp<DisplayDevice>& display, hal:
-const auto refreshRate = display->refreshRateConfigs().getActiveMode().getFps(); if (*currentMode == hal::PowerMode::OFF) {
+    const auto refreshRate = display->refreshRateConfigs().getActiveMode().getFps();
+    if (*currentMode == hal::PowerMode::OFF) {
@@ -3742,0 +3744,4 @@ const auto refreshRate = display->refreshRateConfigs().getActiveMode().getFps();
+        if (isDisplayActiveLocked(display) && *currentMode == hal::PowerMode::DOZE_SUSPEND) {
+            ALOGI("Force repainting for DOZE_SUSPEND -> DOZE or ON.");
+            mVisibleRegionsDirty = true;
+            scheduleRepaint();
