--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@common@sort@sort_state.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@common@sort@sort_state.no_comments_merged.cpp
@@ -1,0 +2 @@
+#include "duckdb/common/radix.hpp"
@@ -6 +6,0 @@
-#include "duckdb/common/radix.hpp"
@@ -259 +259 @@ void LocalSortState::ReOrder(SortedData &sd, data_ptr_t sorting_ptr, RowDataColl
- ordered_data_block->block->SetSwizzling(sd.swizzled ? "LocalSortState::ReOrder.ordered_data" : nullptr);
+ ordered_data_block->block->SetSwizzling(sd.layout.AllConstant() ? nullptr : "LocalSortState::ReOrder.ordered_data");
