--- a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@common@sort@sort_state.no_comments_conflict.cpp
+++ b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@common@sort@sort_state.no_comments_merged.cpp
@@ -7,6 +6,0 @@
-<<<<<<< HEAD
-#include "duckdb/common/radix.hpp"
-||||||| b3f6a8f16d
-#include "duckdb/common/radix.hpp"
-=======
->>>>>>> e4384e1f
@@ -265 +259 @@ void LocalSortState::ReOrder(SortedData &sd, data_ptr_t sorting_ptr, RowDataColl
- ordered_data_block->block->SetSwizzling(sd.swizzled ? "LocalSortState::ReOrder.ordered_data" : nullptr);
+ ordered_data_block->block->SetSwizzling(sd.layout.AllConstant() ? nullptr : "LocalSortState::ReOrder.ordered_data");
