diff --git a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@aggregate@physical_window.no_comments_conflict.cpp b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@aggregate@physical_window.no_comments_merged.cpp
index 579677e..319fcb3 100644
--- a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@aggregate@physical_window.no_comments_conflict.cpp
+++ b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@aggregate@physical_window.no_comments_merged.cpp
@@ -7 +7 @@
-<<<<<<< HEAD
+#include "duckdb/common/types/column/column_data_consumer.hpp"
@@ -9,6 +8,0 @@
-||||||| b3f6a8f16d
-#include "duckdb/common/types/row_data_collection_scanner.hpp"
-=======
-#include "duckdb/common/types/column_data_consumer.hpp"
-#include "duckdb/common/types/row_data_collection_scanner.hpp"
->>>>>>> e4384e1f
