diff --git a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_mergegen.cpp b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_merged.cpp
index 9f500bd..c99ae52 100644
--- a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_mergegen.cpp
+++ b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_insert.no_comments_merged.cpp
@@ -2 +2,3 @@
-#include "duckdb/catalog/catalog_entry/duck_table_entry.hpp" #include "duckdb/catalog/catalog_entry/duck_table_entry.hpp" #include "duckdb/common/types/column/column_data_collection.hpp"
+#include "duckdb/parallel/thread_context.hpp"
+#include "duckdb/catalog/catalog_entry/duck_table_entry.hpp"
+#include "duckdb/common/types/column/column_data_collection.hpp"
@@ -4,0 +7 @@
+#include "duckdb/storage/data_table.hpp"
@@ -6 +8,0 @@
-#include "duckdb/parallel/thread_context.hpp"
@@ -9 +10,0 @@
-#include "duckdb/storage/data_table.hpp"
