--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_merged.cpp
@@ -2,2 +1,0 @@
-#include "duckdb/common/atomic.hpp"
-#include "duckdb/common/types/column/column_data_collection.hpp"
@@ -8 +6,2 @@
-#include "duckdb/common/types/column_data_collection.hpp"
+#include "duckdb/common/types/column/column_data_collection.hpp"
+#include "duckdb/common/atomic.hpp"
@@ -24 +22,0 @@ public:
- { delete_chunk.Initialize(allocator, table_types); }
