--- a/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_mergegen.cpp
+++ b/./output/duckdb/ef94cb25-e4384e1f-b3f6a8f1/src@execution@operator@persistent@physical_delete.no_comments_merged.cpp
@@ -2,2 +1,0 @@
-#include "duckdb/common/atomic.hpp"
-#include "duckdb/common/types/column/column_data_collection.hpp"
@@ -6,0 +5,3 @@
+#include "duckdb/transaction/duck_transaction.hpp"
+#include "duckdb/common/types/column/column_data_collection.hpp"
+#include "duckdb/common/atomic.hpp"
