--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_mergebot.h
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/97834be3-ef710a3a-eb83011e/libs@binder@ndk@include_cpp@android@persistable_bundle_aidl.no_comments_merged.h
@@ -17,3 +17,5 @@ namespace aidl::android::os {
-PersistableBundle& operator=(const PersistableBundle& other) {
-    if AT_LEAST_V_OR_202404 {
-        mPBundle = APersistableBundle_dup(other.mPBundle);
+class PersistableBundle {
+   public:
+    PersistableBundle() noexcept {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
+            mPBundle = APersistableBundle_new();
@@ -21,3 +22,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-    const() {
-        if (!mPBundle) {
-            return STATUS_BAD_VALUE;
@@ -24,0 +24,3 @@ PersistableBundle& operator=(const PersistableBundle& other) {
+    PersistableBundle(APersistableBundle* _Nonnull bundle) noexcept : mPBundle(bundle) {}
+    PersistableBundle(PersistableBundle&& other) noexcept : mPBundle(other.release()) {}
+    PersistableBundle(const PersistableBundle& other) {
@@ -26 +28 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-            return APersistableBundle_writeToParcel(mPBundle, parcel);
+            mPBundle = APersistableBundle_dup(other.mPBundle);
@@ -29,2 +31,3 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-    ~PersistableBundle() {
-        reset();
+    PersistableBundle& operator=(const PersistableBundle& other) {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
+            mPBundle = APersistableBundle_dup(other.mPBundle);
@@ -31,0 +35,3 @@ PersistableBundle& operator=(const PersistableBundle& other) {
+        return *this;
+    }
+    ~PersistableBundle() { reset(); }
@@ -34 +40 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -40 +46 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-    const() {
+    binder_status_t writeToParcel(AParcel* _Nonnull parcel) const {
@@ -44 +50 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -50,2 +56 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-    noexcept
-    () {
+    void reset(APersistableBundle* _Nullable pBundle = nullptr) noexcept {
@@ -53 +58 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-            if AT_LEAST_V_OR_202404 {
+            if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -61 +66 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -88,0 +82,5 @@ PersistableBundle& operator=(const PersistableBundle& other) {
+    [[nodiscard]] APersistableBundle* _Nullable release() noexcept {
+        APersistableBundle* _Nullable ret = mPBundle;
+        mPBundle = nullptr;
+        return ret;
+    }
@@ -92 +90 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        } else if AT_LEAST_V_OR_202404 {
+        } else if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -102 +100 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -109 +107 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -140,25 +137,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-    void putBoolean(const std::string& key, bool val) {
-        if AT_LEAST_V_OR_202404 {
-            APersistableBundle_putBoolean(mPBundle, key.c_str(), val);
-        }
-    }
-    void putInt(const std::string& key, int32_t val) {
-        if AT_LEAST_V_OR_202404 {
-            APersistableBundle_putInt(mPBundle, key.c_str(), val);
-        }
-    }
-    void putLong(const std::string& key, int64_t val) {
-        if AT_LEAST_V_OR_202404 {
-            APersistableBundle_putLong(mPBundle, key.c_str(), val);
-        }
-    }
-    void putDouble(const std::string& key, double val) {
-        if AT_LEAST_V_OR_202404 {
-            APersistableBundle_putDouble(mPBundle, key.c_str(), val);
-        }
-    }
-    void putString(const std::string& key, const std::string& val) {
-        if AT_LEAST_V_OR_202404 {
-            APersistableBundle_putString(mPBundle, key.c_str(), val.c_str());
-        }
-    }
@@ -166 +139 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -181 +154 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -189 +162 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -197 +170 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -205 +178 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -224,5 +196,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-    void putPersistableBundle(const std::string& key, const PersistableBundle& pBundle) {
-        if AT_LEAST_V_OR_202404 {
-            APersistableBundle_putPersistableBundle(mPBundle, key.c_str(), pBundle.mPBundle);
-        }
-    }
@@ -230 +198 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -237 +205 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -244 +212 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -251 +219 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -261 +229 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-        if AT_LEAST_V_OR_202404 {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -272,0 +241,24 @@ PersistableBundle& operator=(const PersistableBundle& other) {
+    template <typename T>
+    bool getVecInternal(int32_t (*_Nonnull getVec)(const APersistableBundle* _Nonnull,
+                                                   const char* _Nonnull, T* _Nullable, int32_t),
+                        const APersistableBundle* _Nonnull pBundle, const char* _Nonnull key,
+                        std::vector<T>* _Nonnull vec) {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
+            int32_t bytes = 0;
+            bytes = getVec(pBundle, key, nullptr, 0);
+            if (bytes > 0) {
+                T* newVec = (T*)malloc(bytes);
+                if (newVec) {
+                    bytes = getVec(pBundle, key, newVec, bytes);
+                    int32_t elements = bytes / sizeof(T);
+                    vec->clear();
+                    for (int32_t i = 0; i < elements; i++) {
+                        vec->push_back(newVec[i]);
+                    }
+                    free(newVec);
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
@@ -274 +265,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -276,5 +266,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -287 +272,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -289,5 +273,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -300 +279,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -302,5 +280,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -313 +286,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -315,5 +287,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -340 +307,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -342,5 +308,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -362 +323,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -364,5 +324,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -385,9 +341,13 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-            const APersistableBundle* _Nonnull pBundle) std::set<std::string>
-            getBooleanKeys() {
-<<<<<<< HEAD
-        if API_LEVEL_AT_LEAST (__ANDROID_API_V__, 202404) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
+            const APersistableBundle* _Nonnull pBundle) {
+        int32_t bytes = getTypedKeys(pBundle, nullptr, 0, &stringAllocator, nullptr);
+        if (bytes > 0) {
+            char** keys = (char**)malloc(bytes);
+            if (keys) {
+                bytes = getTypedKeys(pBundle, keys, bytes, &stringAllocator, nullptr);
+                return moveStringsInternal<std::set<std::string>>(keys, bytes);
+            }
+        }
+        return {};
+    }
+    std::set<std::string> getBooleanKeys() {
+        if API_LEVEL_AT_LEAST(__ANDROID_API_V__, 202404) {
@@ -400 +359,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -402,5 +360,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -413 +366,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -415,5 +367,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -426 +373,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -428,5 +374,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -439 +380,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -441,5 +381,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -452 +387,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -454,5 +388,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -465 +394,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -467,5 +395,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -478 +401,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -480,5 +402,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -491 +408,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -493,5 +409,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -504 +415,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -506,5 +416,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -517 +422,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-<<<<<<< HEAD
@@ -519,5 +423,0 @@ PersistableBundle& operator=(const PersistableBundle& other) {
-|||||||
-        if (__builtin_available(android __ANDROID_API_V__, *)) {
-=======
-        if AT_LEAST_V_OR_202404 {
->>>>>>> ef710a3ab7306d2db144ed678aa98e4c712ae52f
@@ -531,0 +432,5 @@ PersistableBundle& operator=(const PersistableBundle& other) {
+   private:
+    inline APersistableBundle* _Nullable get() const { return mPBundle; }
+    APersistableBundle* _Nullable mPBundle = nullptr;
+};
+}
