--- a/./output/tmux/d682ef88-9456258c-71d453f1/tmux.no_comments_mergegen.h
+++ b/./output/tmux/d682ef88-9456258c-71d453f1/tmux.no_comments_merged.h
@@ -2740 +2740,9 @@ int utf8_cstrhas(const char *, const struct utf8_data *);
-                char *osdep_get_name(int, char *); char *osdep_get_cwd(int); struct event_base *osdep_event_init(void);
+char *osdep_get_name(int, char *);
+char *osdep_get_cwd(int);
+struct event_base *osdep_event_init(void);
+void utf8_build_combined(void);
+int utf8_try_combined(const struct utf8_data *,
+       const struct utf8_data *, const struct utf8_data **,
+       u_int *width);
+char *get_proc_name(int, char *);
+char *get_proc_cwd(int);
