--- a/home/whalien/codebase/python/mergebot-eva/output/tmux/d682ef88-9456258c-71d453f1/utf8.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/tmux/d682ef88-9456258c-71d453f1/utf8.no_comments_merged.c
@@ -6 +5,0 @@
-#include <vis.h>
@@ -10,4 +9 @@ struct utf8_item {
-  data_entry;
-  data_entry;
-  index_entry;
-  data_entry;
+ RB_ENTRY(utf8_item) index_entry;
@@ -15,2 +11 @@ struct utf8_item {
-  data_entry;
-  data_entry;
+ RB_ENTRY(utf8_item) data_entry;
@@ -24,0 +24,2 @@ static int utf8_data_cmp(struct utf8_item *ui1, struct utf8_item *ui2) {
+RB_HEAD(utf8_data_tree, utf8_item);
+RB_GENERATE_STATIC(utf8_data_tree, utf8_item, data_entry, utf8_data_cmp);
@@ -30,0 +36,2 @@ static int utf8_index_cmp(struct utf8_item *ui1, struct utf8_item *ui2) {
+RB_HEAD(utf8_index_tree, utf8_item);
+RB_GENERATE_STATIC(utf8_index_tree, utf8_item, index_entry, utf8_index_cmp);
@@ -33 +40,7 @@ static u_int utf8_next_index;
-static struct utf8_item *utf8_item_by_data(const u_char *data, size_t size) {
+#define UTF8_GET_SIZE(uc) (((uc) >> 24) & 0x1f)
+#define UTF8_GET_WIDTH(uc) (((uc) >> 29) - 1)
+#define UTF8_SET_SIZE(size) (((utf8_char)(size)) << 24)
+#define UTF8_SET_WIDTH(width) ((((utf8_char)(width)) + 1) << 29)
+static struct utf8_item *
+utf8_item_by_data(const u_char *data, size_t size)
+{
@@ -132 +167,0 @@ static enum utf8_state utf8_width(struct utf8_data *ud, int *width) {
-<<<<<<< HEAD
@@ -138,4 +172,0 @@ static enum utf8_state utf8_width(struct utf8_data *ud, int *width) {
-  ||||||| log_debug("UTF-8 %.*s is %08X", (int)ud->size, ud->data, (u_int)wc);
-=======
-  log_debug("UTF-8 %.*s is %05X", (int)ud->size, ud->data, (u_int)wc);
->>>>>>> 9456258ccc03a1a959cfb7d020011d751b39bb1b
