diff --git a/./output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_conflict.cpp b/./output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_merged.cpp
index f28a05b..2ae7960 100644
--- a/./output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_conflict.cpp
+++ b/./output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_merged.cpp
@@ -1821 +1820,0 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
-<<<<<<< HEAD
@@ -1827,18 +1825,0 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
-||||||| 97c05f176e
-        if (!device->classes.any(InputDeviceClass::TOUCH | InputDeviceClass::TOUCH_MT)) {
-            for (int32_t keycode : STYLUS_BUTTON_KEYCODES) {
-                if (device->hasKeycodeLocked(keycode)) {
-                    device->classes |= InputDeviceClass::EXTERNAL_STYLUS;
-                    break;
-                }
-            }
-=======
-        if (!device->classes.any(InputDeviceClass::TOUCH | InputDeviceClass::TOUCH_MT) &&
-            !device->classes.any(InputDeviceClass::ALPHAKEY)) {
-            for (int32_t keycode : STYLUS_BUTTON_KEYCODES) {
-                if (device->hasKeycodeLocked(keycode)) {
-                    device->classes |= InputDeviceClass::EXTERNAL_STYLUS;
-                    break;
-                }
-            }
->>>>>>> cddfc361
