diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_merged.cpp
index 65e00b4..2ae7960 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/13964cc1-cddfc361-97c05f17/services@inputflinger@reader@EventHub.no_comments_merged.cpp
@@ -1821 +1820,0 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
-<<<<<<< HEAD
@@ -1826,10 +1824,0 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
-|||||||
-        if (!device->classes.any(InputDeviceClass::TOUCH | InputDeviceClass::TOUCH_MT)) {
-            for (int32_t keycode : STYLUS_BUTTON_KEYCODES) {
-                if (device->hasKeycodeLocked(keycode)) {
-=======
-        if (!device->classes.any(InputDeviceClass::TOUCH | InputDeviceClass::TOUCH_MT) &&
-            !device->classes.any(InputDeviceClass::ALPHAKEY)) {
-            for (int32_t keycode : STYLUS_BUTTON_KEYCODES) {
-                if (device->hasKeycodeLocked(keycode)) {
->>>>>>> cddfc361807b9966e2583208ff8da447054e806f
