--- a/./output/duckdb/71385710-31a2467e-07b75e3c/src@function@function.no_comments_conflict.cpp
+++ b/./output/duckdb/71385710-31a2467e-07b75e3c/src@function@function.no_comments_merged.cpp
@@ -76,7 +75,0 @@ void BuiltinFunctions::Initialize() {
-<<<<<<< HEAD
-||||||| 07b75e3c3a
- RegisterCompressedMaterializationFunctions();
- RegisterOperators();
-=======
- RegisterOperators();
->>>>>>> 31a2467e
