--- a/./output/duckdb/71385710-31a2467e-07b75e3c/src@include@duckdb@function@built_in_functions.no_comments_conflict.hpp
+++ b/./output/duckdb/71385710-31a2467e-07b75e3c/src@include@duckdb@function@built_in_functions.no_comments_merged.hpp
@@ -38,7 +37,0 @@ private:
-<<<<<<< HEAD
-||||||| 07b75e3c3a
- void RegisterCompressedMaterializationFunctions();
- void RegisterOperators();
-=======
- void RegisterOperators();
->>>>>>> 31a2467e
