diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@include@duckdb@function@built_in_functions.no_comments_mergebot.hpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@include@duckdb@function@built_in_functions.no_comments_merged.hpp
index 869b3ed..f1808ba 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@include@duckdb@function@built_in_functions.no_comments_mergebot.hpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@include@duckdb@function@built_in_functions.no_comments_merged.hpp
@@ -9,0 +10 @@ public:
+public:
@@ -24,0 +26 @@ private:
+private:
