diff --git a/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@function@function_list.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@function@function_list.no_comments_merged.cpp
index 19cbd45..dbfc513 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@function@function_list.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/duckdb/71385710-31a2467e-07b75e3c/src@function@function_list.no_comments_merged.cpp
@@ -12,0 +13,16 @@ namespace duckdb {
+#define DUCKDB_SCALAR_FUNCTION_BASE(_PARAM,_NAME) \
+ { _NAME, _PARAM::Parameters, _PARAM::Description, _PARAM::Example, _PARAM::GetFunction, nullptr, nullptr, nullptr }
+#define DUCKDB_SCALAR_FUNCTION(_PARAM) DUCKDB_SCALAR_FUNCTION_BASE(_PARAM, _PARAM::Name)
+#define DUCKDB_SCALAR_FUNCTION_ALIAS(_PARAM) DUCKDB_SCALAR_FUNCTION_BASE(_PARAM::ALIAS, _PARAM::Name)
+#define DUCKDB_SCALAR_FUNCTION_SET_BASE(_PARAM,_NAME) \
+ { _NAME, _PARAM::Parameters, _PARAM::Description, _PARAM::Example, nullptr, _PARAM::GetFunctions, nullptr, nullptr }
+#define DUCKDB_SCALAR_FUNCTION_SET(_PARAM) DUCKDB_SCALAR_FUNCTION_SET_BASE(_PARAM, _PARAM::Name)
+#define DUCKDB_SCALAR_FUNCTION_SET_ALIAS(_PARAM) DUCKDB_SCALAR_FUNCTION_SET_BASE(_PARAM::ALIAS, _PARAM::Name)
+#define DUCKDB_AGGREGATE_FUNCTION_BASE(_PARAM,_NAME) \
+ { _NAME, _PARAM::Parameters, _PARAM::Description, _PARAM::Example, nullptr, nullptr, _PARAM::GetFunction, nullptr }
+#define DUCKDB_AGGREGATE_FUNCTION(_PARAM) DUCKDB_AGGREGATE_FUNCTION_BASE(_PARAM, _PARAM::Name)
+#define DUCKDB_AGGREGATE_FUNCTION_ALIAS(_PARAM) DUCKDB_AGGREGATE_FUNCTION_BASE(_PARAM::ALIAS, _PARAM::Name)
+#define DUCKDB_AGGREGATE_FUNCTION_SET_BASE(_PARAM,_NAME) \
+ { _NAME, _PARAM::Parameters, _PARAM::Description, _PARAM::Example, nullptr, nullptr, nullptr, _PARAM::GetFunctions }
+#define DUCKDB_AGGREGATE_FUNCTION_SET(_PARAM) DUCKDB_AGGREGATE_FUNCTION_SET_BASE(_PARAM, _PARAM::Name)
+#define DUCKDB_AGGREGATE_FUNCTION_SET_ALIAS(_PARAM) DUCKDB_AGGREGATE_FUNCTION_SET_BASE(_PARAM::ALIAS, _PARAM::Name)
@@ -15 +31,2 @@ namespace duckdb {
-static const StaticFunctionDefinition function[] = {DUCKDB_SCALAR_FUNCTION(NotLikeFun),
+static const StaticFunctionDefinition function[] = {
+ DUCKDB_SCALAR_FUNCTION(NotLikeFun),
@@ -17 +33,0 @@ static const StaticFunctionDefinition function[] = {DUCKDB_SCALAR_FUNCTION(NotLi
-<<<<<<< HEAD
@@ -43,21 +58,0 @@ static const StaticFunctionDefinition function[] = {DUCKDB_SCALAR_FUNCTION(NotLi
-|||||||
-=======
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUbigintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUintegerFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUsmallintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalCompressIntegralUtinyintFun),
-                                                     DUCKDB_SCALAR_FUNCTION(InternalCompressStringHugeintFun),
-                                                     DUCKDB_SCALAR_FUNCTION(InternalCompressStringUbigintFun),
-                                                     DUCKDB_SCALAR_FUNCTION(InternalCompressStringUintegerFun),
-                                                     DUCKDB_SCALAR_FUNCTION(InternalCompressStringUsmallintFun),
-                                                     DUCKDB_SCALAR_FUNCTION(InternalCompressStringUtinyintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralBigintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralHugeintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralIntegerFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralSmallintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUbigintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUhugeintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUintegerFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressIntegralUsmallintFun),
-                                                     DUCKDB_SCALAR_FUNCTION_SET(InternalDecompressStringFun),
->>>>>>> 31a2467e30e767db06ea392726c2759d12d71218
@@ -148 +143,2 @@ static const StaticFunctionDefinition function[] = {DUCKDB_SCALAR_FUNCTION(NotLi
-                                                    FINAL_FUNCTION};
+ FINAL_FUNCTION
+};
