diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_merged.h
index 0fe15e6..4c144a7 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmHal.no_comments_merged.h
@@ -10,2 +9,0 @@
-#include <media/IResourceManagerService.h>
-#include <media/MediaAnalyticsItem.h>
@@ -18,3 +16 @@
-using ::android::hardware::hidl_vec;
-using ::android::hardware::Return;
-using ::android::hardware::Void;
+namespace drm = ::android::hardware::drm;
@@ -27,0 +24,3 @@ using drm::V1_2::OfflineLicenseState;
+using ::android::hardware::hidl_vec;
+using ::android::hardware::Return;
+using ::android::hardware::Void;
@@ -36,13 +35,3 @@ inline bool operator==(const Vector<uint8_t> &l, const Vector<uint8_t> &r) {
-struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  struct DrmSessionClient : public BnResourceManagerClient {
-    explicit DrmSessionClient(DrmHal *drm, const Vector<uint8_t> &sessionId)
-        : mSessionId(sessionId), mDrm(drm) {}
-    virtual bool reclaimResource();
-    virtual String8 getName();
-    const Vector<uint8_t> mSessionId;
-   protected:
-    virtual ~DrmSessionClient();
-   private:
-    wp<DrmHal> mDrm;
-    DISALLOW_EVIL_CONSTRUCTORS(DrmSessionClient);
-  };
+struct DrmHal : public IDrm,
+                public IDrmPluginListener_V1_2 {
+    struct DrmSessionClient;
@@ -62,2 +51,3 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  virtual status_t getKeyRequest(
-      Vector<uint8_t> const &sessionId, Vector<uint8_t> const &initData,
+    virtual status_t
+        getKeyRequest(Vector<uint8_t> const &sessionId,
+                      Vector<uint8_t> const &initData,
@@ -85,2 +75 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  virtual status_t getSecureStop(Vector<uint8_t> const &ssid,
-                                 Vector<uint8_t> &secureStop);
+    virtual status_t getSecureStop(Vector<uint8_t> const &ssid, Vector<uint8_t> &secureStop);
@@ -96,2 +85 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  virtual status_t getOfflineLicenseKeySetIds(
-      List<Vector<uint8_t>> &keySetIds) const;
+    virtual status_t getOfflineLicenseKeySetIds(List<Vector<uint8_t>> &keySetIds) const;
@@ -99,2 +87 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  virtual status_t getOfflineLicenseState(
-      Vector<uint8_t> const &keySetId,
+    virtual status_t getOfflineLicenseState(Vector<uint8_t> const &keySetId,
@@ -105,2 +92 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  virtual status_t setPropertyString(String8 const &name,
-                                     String8 const &value) const;
+    virtual status_t setPropertyString(String8 const &name, String8 const &value ) const;
@@ -117 +103,2 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-                           Vector<uint8_t> const &iv, Vector<uint8_t> &output);
+                             Vector<uint8_t> const &iv,
+                             Vector<uint8_t> &output);
@@ -121 +108,2 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-                           Vector<uint8_t> const &iv, Vector<uint8_t> &output);
+                             Vector<uint8_t> const &iv,
+                             Vector<uint8_t> &output);
@@ -129 +117,2 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-                          Vector<uint8_t> const &signature, bool &match);
+                            Vector<uint8_t> const &signature,
+                            bool &match);
@@ -137,2 +126 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-                         const hidl_vec<uint8_t> &sessionId,
-                         const hidl_vec<uint8_t> &data);
+            const hidl_vec<uint8_t>& sessionId, const hidl_vec<uint8_t>& data);
@@ -142,2 +130 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-                              const hidl_vec<KeyStatus_V1_0> &keyStatusList,
-                              bool hasNewUsableKey);
+            const hidl_vec<KeyStatus_V1_0>& keyStatusList, bool hasNewUsableKey);
@@ -145,2 +132 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-                                  const hidl_vec<KeyStatus> &keyStatusList,
-                                  bool hasNewUsableKey);
+            const hidl_vec<KeyStatus>& keyStatusList, bool hasNewUsableKey);
@@ -159 +144,0 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-<<<<<<< HEAD
@@ -161,5 +145,0 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-||||||| 35b28e5f80
-  Vector<Vector<uint8_t>> mOpenSessions;
-=======
-  Vector<sp<DrmSessionClient>> mOpenSessions;
->>>>>>> 77b8c802dbf3d63e19ff44e7bb2fe47b37e8d6b2
@@ -171,2 +151 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-                               const uint8_t uuid[16],
-                               const String8 &appPackageName);
+            const uint8_t uuid[16], const String8& appPackageName);
@@ -184 +163 @@ struct DrmHal : public IDrm, public IDrmPluginListener_V1_2 {
-  Mutex DrmHal::mLock;
+    DISALLOW_EVIL_CONSTRUCTORS(DrmHal);
