diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_merged.h
index 4dc34f2..751b8af 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/drm@libmediadrm@include@mediadrm@DrmSessionManager.no_comments_merged.h
@@ -11,2 +10,0 @@
-#include <binder/IBinder.h>
-#include <media/IResourceManagerService.h>
@@ -21,3 +19 @@ using aidl::android::media::IResourceManagerService;
-class IResourceManagerClient;
-bool isEqualSessionId(const Vector<uint8_t>& sessionId1,
-                      const Vector<uint8_t>& sessionId2);
+bool isEqualSessionId(const Vector<uint8_t> &sessionId1, const Vector<uint8_t> &sessionId2);
@@ -30 +26 @@ typedef std::map<std::vector<uint8_t>, SessionInfo> SessionInfoMap;
-struct DrmSessionManager : public IBinder::DeathRecipient {
+struct DrmSessionManager : public RefBase {
@@ -33,3 +29,3 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-  explicit DrmSessionManager(
-      const std::shared_ptr<IResourceManagerService>& service);
-  void addSession(int pid, const std::shared_ptr<IResourceManagerClient>& drm,
+    explicit DrmSessionManager(const std::shared_ptr<IResourceManagerService> &service);
+    void addSession(int pid,
+            const std::shared_ptr<IResourceManagerClient>& drm,
@@ -37 +32,0 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-  explicit DrmSessionManager(const sp<IResourceManagerService>& service);
@@ -44 +38,0 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-  virtual void binderDied(const wp<IBinder>& );
@@ -49 +43 @@ struct DrmSessionManager : public IBinder::DeathRecipient {
-  sp<IResourceManagerService> mService;
+    std::shared_ptr<IResourceManagerService> mService;
