--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@include@media@MediaResource.no_comments_conflict.h
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@include@media@MediaResource.no_comments_merged.h
@@ -5 +4,0 @@
-#include <vector>
@@ -12 +10,0 @@ public:
-<<<<<<< HEAD
@@ -15,20 +12,0 @@ public:
-||||||| 35b28e5f80
-    enum Type {
-        kUnspecified = 0,
-        kSecureCodec,
-        kNonSecureCodec,
-        kGraphicMemory,
-        kCpuBoost,
-        kBattery,
-    };
-=======
-    enum Type {
-        kUnspecified = 0,
-        kSecureCodec,
-        kNonSecureCodec,
-        kGraphicMemory,
-        kCpuBoost,
-        kBattery,
-        kDrmSession,
-    };
->>>>>>> 77b8c802
@@ -39 +16,0 @@ public:
-<<<<<<< HEAD
@@ -45,27 +21,0 @@ public:
-||||||| 35b28e5f80
-    MediaResource();
-    MediaResource(Type type, uint64_t value);
-    MediaResource(Type type, SubType subType, uint64_t value);
-    void readFromParcel(const Parcel &parcel);
-    void writeToParcel(Parcel *parcel) const;
-    String8 toString() const;
-    bool operator==(const MediaResource &other) const;
-    bool operator!=(const MediaResource &other) const;
-    Type mType;
-    SubType mSubType;
-    uint64_t mValue;
-=======
-    MediaResource();
-    MediaResource(Type type, uint64_t value);
-    MediaResource(Type type, SubType subType, uint64_t value);
-    MediaResource(Type type, const std::vector<uint8_t> &id, uint64_t value);
-    void readFromParcel(const Parcel &parcel);
-    void writeToParcel(Parcel *parcel) const;
-    String8 toString() const;
-    bool operator==(const MediaResource &other) const;
-    bool operator!=(const MediaResource &other) const;
-    Type mType;
-    SubType mSubType;
-    uint64_t mValue;
-    std::vector<uint8_t> mId;
->>>>>>> 77b8c802
@@ -75 +24,0 @@ inline static const char *asString(MediaResource::Type i, const char *def = "??"
-<<<<<<< HEAD
@@ -84,18 +32,0 @@ inline static const char *asString(MediaResource::Type i, const char *def = "??"
-||||||| 35b28e5f80
-        case MediaResource::kUnspecified: return "unspecified";
-        case MediaResource::kSecureCodec: return "secure-codec";
-        case MediaResource::kNonSecureCodec: return "non-secure-codec";
-        case MediaResource::kGraphicMemory: return "graphic-memory";
-        case MediaResource::kCpuBoost: return "cpu-boost";
-        case MediaResource::kBattery: return "battery";
-        default: return def;
-=======
-        case MediaResource::kUnspecified: return "unspecified";
-        case MediaResource::kSecureCodec: return "secure-codec";
-        case MediaResource::kNonSecureCodec: return "non-secure-codec";
-        case MediaResource::kGraphicMemory: return "graphic-memory";
-        case MediaResource::kCpuBoost: return "cpu-boost";
-        case MediaResource::kBattery: return "battery";
-        case MediaResource::kDrmSession: return "drm-session";
-        default: return def;
->>>>>>> 77b8c802
