--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@MediaResource.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@MediaResource.no_comments_merged.cpp
@@ -6 +5,0 @@ namespace android {
-MediaResource()
@@ -17,3 +16,6 @@ MediaResource::MediaResource(Type type, SubType subType, int64_t value) {
-MediaResource::MediaResource(Type type, const std::vector<uint8_t> &id,
-                             uint64_t value)
-    : mType(type), mSubType(kUnspecifiedSubType), mValue(value), mId(id) {}
+MediaResource::MediaResource(Type type, const std::vector<int8_t> &id, int64_t value) {
+    this->type = type;
+    this->subType = SubType::kUnspecifiedSubType;
+    this->id = id;
+    this->value = value;
+}
@@ -37 +40 @@ MediaResource MediaResource::DrmSessionResource(const std::vector<int8_t> &id,
-static String8 bytesToHexString(const std::vector<uint8_t> &bytes) {
+static String8 bytesToHexString(const std::vector<int8_t> &bytes) {
@@ -51 +55 @@ String8 toString(const MediaResourceParcel &resource) {
-}
+};
