diff --git a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@MediaResource.no_comments_conflict.cpp b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@MediaResource.no_comments_merged.cpp
index cfd870f..c8c1471 100644
--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@MediaResource.no_comments_conflict.cpp
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/media@libmedia@MediaResource.no_comments_merged.cpp
@@ -6 +5,0 @@ namespace android {
-<<<<<<< HEAD
@@ -11,41 +9,0 @@ MediaResource::MediaResource(Type type, int64_t value) {
-||||||| 35b28e5f80
-MediaResource::MediaResource()
-        : mType(kUnspecified),
-          mSubType(kUnspecifiedSubType),
-          mValue(0) {}
-MediaResource::MediaResource(Type type, uint64_t value)
-        : mType(type),
-          mSubType(kUnspecifiedSubType),
-          mValue(value) {}
-MediaResource::MediaResource(Type type, SubType subType, uint64_t value)
-        : mType(type),
-          mSubType(subType),
-          mValue(value) {}
-void MediaResource::readFromParcel(const Parcel &parcel) {
-    mType = static_cast<Type>(parcel.readInt32());
-    mSubType = static_cast<SubType>(parcel.readInt32());
-    mValue = parcel.readUint64();
-=======
-MediaResource::MediaResource()
-        : mType(kUnspecified),
-          mSubType(kUnspecifiedSubType),
-          mValue(0) {}
-MediaResource::MediaResource(Type type, uint64_t value)
-        : mType(type),
-          mSubType(kUnspecifiedSubType),
-          mValue(value) {}
-MediaResource::MediaResource(Type type, SubType subType, uint64_t value)
-        : mType(type),
-          mSubType(subType),
-          mValue(value) {}
-MediaResource::MediaResource(Type type, const std::vector<uint8_t> &id, uint64_t value)
-        : mType(type),
-          mSubType(kUnspecifiedSubType),
-          mValue(value),
-          mId(id) {}
-void MediaResource::readFromParcel(const Parcel &parcel) {
-    mType = static_cast<Type>(parcel.readInt32());
-    mSubType = static_cast<SubType>(parcel.readInt32());
-    mValue = parcel.readUint64();
-    parcel.readByteVector(&mId);
->>>>>>> 77b8c802
@@ -53 +10,0 @@ void MediaResource::readFromParcel(const Parcel &parcel) {
-<<<<<<< HEAD
@@ -58,19 +14,0 @@ MediaResource::MediaResource(Type type, SubType subType, int64_t value) {
-||||||| 35b28e5f80
-void MediaResource::writeToParcel(Parcel *parcel) const {
-    parcel->writeInt32(static_cast<int32_t>(mType));
-    parcel->writeInt32(static_cast<int32_t>(mSubType));
-    parcel->writeUint64(mValue);
-=======
-void MediaResource::writeToParcel(Parcel *parcel) const {
-    parcel->writeInt32(static_cast<int32_t>(mType));
-    parcel->writeInt32(static_cast<int32_t>(mSubType));
-    parcel->writeUint64(mValue);
-    parcel->writeByteVector(mId);
-}
-static String8 bytesToHexString(const std::vector<uint8_t> &bytes) {
-    String8 str;
-    for (auto &b : bytes) {
-        str.appendFormat("%02x", b);
-    }
-    return str;
->>>>>>> 77b8c802
@@ -104 +41,0 @@ static String8 bytesToHexString(const std::vector<int8_t> &bytes) {
-<<<<<<< HEAD
@@ -108,8 +44,0 @@ static String8 bytesToHexString(const std::vector<int8_t> &bytes) {
-||||||| 35b28e5f80
-    str.appendFormat("%s/%s:%llu", asString(mType), asString(mSubType), (unsigned long long)mValue);
-=======
-    str.appendFormat("%s/%s:[%s]:%llu",
-        asString(mType), asString(mSubType),
-        bytesToHexString(mId).c_str(),
-        (unsigned long long)mValue);
->>>>>>> 77b8c802
@@ -118 +46,0 @@ static String8 bytesToHexString(const std::vector<int8_t> &bytes) {
-<<<<<<< HEAD
@@ -121,12 +48,0 @@ String8 toString(const MediaResourceParcel& resource) {
-||||||| 35b28e5f80
-bool MediaResource::operator==(const MediaResource &other) const {
-    return (other.mType == mType) && (other.mSubType == mSubType) && (other.mValue == mValue);
-}
-=======
-bool MediaResource::operator==(const MediaResource &other) const {
-    return (other.mType == mType)
-      && (other.mSubType == mSubType)
-      && (other.mValue == mValue)
-      && (other.mId == mId);
-}
->>>>>>> 77b8c802
