diff --git a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_mergegen.h b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_merged.h
index e86d115..05fdc35 100644
--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_mergegen.h
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_merged.h
@@ -16 +16,8 @@ struct ProcessInfoInterface;
-using ::aidl::android::media::IResourceManagerClient; using ::aidl::android::media::BnResourceManagerService; using ::aidl::android::media::MediaResourceParcel; using ::aidl::android::media::MediaResourcePolicyParcel; typedef std::map<std::tuple< MediaResource::Type, MediaResource::SubType, std::vector<int8_t>>, MediaResourceParcel> ResourceList; struct ResourceInfo {
+using Status = ::ndk::ScopedAStatus;
+using ::aidl::android::media::IResourceManagerClient;
+using ::aidl::android::media::BnResourceManagerService;
+using ::aidl::android::media::MediaResourceParcel;
+using ::aidl::android::media::MediaResourcePolicyParcel;
+typedef std::map<std::tuple<
+        MediaResource::Type, MediaResource::SubType, std::vector<int8_t>>,
+        MediaResourceParcel> ResourceList;
@@ -87 +93,0 @@ private:
-    void mergeResources(MediaResource& r1, const MediaResource& r2);
