diff --git a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@test@ResourceManagerService_test.no_comments_conflict.cpp b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@test@ResourceManagerService_test.no_comments_merged.cpp
index 5602311..326d9c6 100644
--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@test@ResourceManagerService_test.no_comments_conflict.cpp
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@test@ResourceManagerService_test.no_comments_merged.cpp
@@ -138 +137,0 @@ protected:
-<<<<<<< HEAD
@@ -142,8 +140,0 @@ protected:
-||||||| 35b28e5f80
-            const auto resType = std::make_pair(resources1[i].mType, resources1[i].mSubType);
-            r1[resType] = resources1[i];
-=======
-            const auto &res = resources1[i];
-            const auto resType = std::tuple(res.mType, res.mSubType, res.mId);
-            r1[resType] = res;
->>>>>>> 77b8c802
