diff --git a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_conflict.h b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_merged.h
index 5a4fa5b..05fdc35 100644
--- a/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_conflict.h
+++ b/./output/frameworks_av/2475cf40-77b8c802-35b28e5f/services@mediaresourcemanager@ResourceManagerService.no_comments_merged.h
@@ -16 +15,0 @@ struct ProcessInfoInterface;
-<<<<<<< HEAD
@@ -25,5 +23,0 @@ typedef std::map<std::tuple<
-||||||| 35b28e5f80
-typedef std::map<std::pair<MediaResource::Type, MediaResource::SubType>, MediaResource> ResourceList;
-=======
-typedef std::map<std::tuple<MediaResource::Type, MediaResource::SubType, std::vector<uint8_t>>, MediaResource> ResourceList;
->>>>>>> 77b8c802
@@ -100 +93,0 @@ private:
-    void mergeResources(MediaResource& r1, const MediaResource& r2);
