diff --git a/home/whalien/codebase/python/mergebot-eva/output/redis/04a10b1a-0b7f6d09-c2f816b4/src@object.no_comments_mergebot.c b/home/whalien/codebase/python/mergebot-eva/output/redis/04a10b1a-0b7f6d09-c2f816b4/src@object.no_comments_merged.c
index 634476c..6d18b98 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/redis/04a10b1a-0b7f6d09-c2f816b4/src@object.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/04a10b1a-0b7f6d09-c2f816b4/src@object.no_comments_merged.c
@@ -18 +18,2 @@ robj *createStringObjectFromLongLong(long long value) {
-  if (value >= 0 && value < REDIS_SHARED_INTEGERS && !server.ds_enabled &&
+    if (value >= 0 && value < REDIS_SHARED_INTEGERS &&
+        !server.ds_enabled &&
@@ -70,3 +70,0 @@ robj *createZsetObject(void) {
-<<<<<<< HEAD
-|||||||
-=======
@@ -74 +71,0 @@ robj *createZsetObject(void) {
->>>>>>> 0b7f6d0913f96f2fe8280afc681c6e0a65ca81c9
@@ -145 +142,3 @@ void freeHashObject(robj *o) {
-void incrRefCount(robj *o) { o->refcount++; }
+void incrRefCount(robj *o) {
+    o->refcount++;
+}
@@ -151,18 +150,6 @@ void decrRefCount(void *obj) {
-      case REDIS_STRING:
-        freeStringObject(o);
-        break;
-      case REDIS_LIST:
-        freeListObject(o);
-        break;
-      case REDIS_SET:
-        freeSetObject(o);
-        break;
-      case REDIS_ZSET:
-        freeZsetObject(o);
-        break;
-      case REDIS_HASH:
-        freeHashObject(o);
-        break;
-      default:
-        redisPanic("Unknown object type");
-        break;
+        case REDIS_STRING: freeStringObject(o); break;
+        case REDIS_LIST: freeListObject(o); break;
+        case REDIS_SET: freeSetObject(o); break;
+        case REDIS_ZSET: freeZsetObject(o); break;
+        case REDIS_HASH: freeHashObject(o); break;
+        default: redisPanic("Unknown object type"); break;
@@ -184 +171,2 @@ robj *tryObjectEncoding(robj *o) {
-  if (o->encoding != REDIS_ENCODING_RAW) return o;
+    if (o->encoding != REDIS_ENCODING_RAW)
+        return o;
@@ -188,3 +176,4 @@ robj *tryObjectEncoding(robj *o) {
-  if (!server.ds_enabled && server.maxmemory == 0 && value >= 0 &&
-      value < REDIS_SHARED_INTEGERS &&
-      pthread_equal(pthread_self(), server.mainthread)) {
+    if (!server.ds_enabled &&
+        server.maxmemory == 0 && value >= 0 && value < REDIS_SHARED_INTEGERS &&
+        pthread_equal(pthread_self(),server.mainthread))
+    {
@@ -272,2 +261 @@ int getDoubleFromObject(robj *o, double *target) {
-int getDoubleFromObjectOrReply(redisClient *c, robj *o, double *target,
-                               const char *msg) {
+int getDoubleFromObjectOrReply(redisClient *c, robj *o, double *target, const char *msg) {
@@ -307,2 +295 @@ int getLongLongFromObject(robj *o, long long *target) {
-int getLongLongFromObjectOrReply(redisClient *c, robj *o, long long *target,
-                                 const char *msg) {
+int getLongLongFromObjectOrReply(redisClient *c, robj *o, long long *target, const char *msg) {
@@ -321,2 +308 @@ int getLongLongFromObjectOrReply(redisClient *c, robj *o, long long *target,
-int getLongFromObjectOrReply(redisClient *c, robj *o, long *target,
-                             const char *msg) {
+int getLongFromObjectOrReply(redisClient *c, robj *o, long *target, const char *msg) {
@@ -324,2 +310 @@ int getLongFromObjectOrReply(redisClient *c, robj *o, long *target,
-  if (getLongLongFromObjectOrReply(c, o, &value, msg) != REDIS_OK)
-    return REDIS_ERR;
+    if (getLongLongFromObjectOrReply(c, o, &value, msg) != REDIS_OK) return REDIS_ERR;
@@ -339,18 +324,9 @@ char *strEncoding(int encoding) {
-    case REDIS_ENCODING_RAW:
-      return "raw";
-    case REDIS_ENCODING_INT:
-      return "int";
-    case REDIS_ENCODING_HT:
-      return "hashtable";
-    case REDIS_ENCODING_ZIPMAP:
-      return "zipmap";
-    case REDIS_ENCODING_LINKEDLIST:
-      return "linkedlist";
-    case REDIS_ENCODING_ZIPLIST:
-      return "ziplist";
-    case REDIS_ENCODING_INTSET:
-      return "intset";
-    case REDIS_ENCODING_SKIPLIST:
-      return "skiplist";
-    default:
-      return "unknown";
+    case REDIS_ENCODING_RAW: return "raw";
+    case REDIS_ENCODING_INT: return "int";
+    case REDIS_ENCODING_HT: return "hashtable";
+    case REDIS_ENCODING_ZIPMAP: return "zipmap";
+    case REDIS_ENCODING_LINKEDLIST: return "linkedlist";
+    case REDIS_ENCODING_ZIPLIST: return "ziplist";
+    case REDIS_ENCODING_INTSET: return "intset";
+    case REDIS_ENCODING_SKIPLIST: return "skiplist";
+    default: return "unknown";
@@ -380,3 +356,2 @@ void objectCommand(redisClient *c) {
-    if ((o = objectCommandLookupOrReply(c, c->argv[2], shared.nullbulk)) ==
-        NULL)
-      return;
+        if ((o = objectCommandLookupOrReply(c,c->argv[2],shared.nullbulk))
+                == NULL) return;
@@ -385,3 +360,2 @@ void objectCommand(redisClient *c) {
-    if ((o = objectCommandLookupOrReply(c, c->argv[2], shared.nullbulk)) ==
-        NULL)
-      return;
+        if ((o = objectCommandLookupOrReply(c,c->argv[2],shared.nullbulk))
+                == NULL) return;
@@ -390,3 +364,2 @@ void objectCommand(redisClient *c) {
-    if ((o = objectCommandLookupOrReply(c, c->argv[2], shared.nullbulk)) ==
-        NULL)
-      return;
+        if ((o = objectCommandLookupOrReply(c,c->argv[2],shared.nullbulk))
+                == NULL) return;
