--- a/home/whalien/codebase/python/mergebot-eva/output/redis/04a10b1a-0b7f6d09-c2f816b4/src@object.no_comments_mergebot.c
+++ b/home/whalien/codebase/python/mergebot-eva/output/redis/04a10b1a-0b7f6d09-c2f816b4/src@object.no_comments_merged.c
@@ -70,3 +70,0 @@ robj *createZsetObject(void) {
-<<<<<<< HEAD
-|||||||
-=======
@@ -74 +71,0 @@ robj *createZsetObject(void) {
->>>>>>> 0b7f6d0913f96f2fe8280afc681c6e0a65ca81c9
