diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_mergebot.cc b/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_merged.cc
index daff5dd..3a87a55 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_mergebot.cc
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_merged.cc
@@ -147 +146,0 @@ int InitializeConfig(int argc, char** argv, OdrConfig* config) {
-<<<<<<< HEAD
@@ -150,6 +148,0 @@ int InitializeConfig(int argc, char** argv, OdrConfig* config) {
-|||||||
-    std::string filter = GetProperty("dalvik.vm.systemservercompilerfilter", "speed");
-=======
-    std::string filter = GetProperty("dalvik.vm.systemservercompilerfilter", "speed");
-    filter = GetProperty(kSystemPropertySystemServerCompilerFilterOverride, filter);
->>>>>>> af3127b30b1b93a728eb1461ce697d182f8a731a
