--- a/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odr_config.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odr_config.no_comments_merged.h
@@ -38 +37,0 @@ const android::base::NoDestructor<std::vector<SystemPropertyConfig>> kSystemProp
-<<<<<<< HEAD
@@ -43,9 +41,0 @@ const android::base::NoDestructor<std::vector<SystemPropertyConfig>> kSystemProp
-|||||||
-                          .default_value = "false"},
-     SystemPropertyConfig{.name = kPhDisableCompactDex, .default_value = "false"}}};
-=======
-                          .default_value = "false"},
-     SystemPropertyConfig{.name = kPhDisableCompactDex, .default_value = "false"},
-     SystemPropertyConfig{.name = kSystemPropertySystemServerCompilerFilterOverride,
-                          .default_value = ""}}};
->>>>>>> af3127b30b1b93a728eb1461ce697d182f8a731a
@@ -195 +206,3 @@ class OdrConfig final {
-  OdrConfig() OdrConfig(const OdrConfig&) OdrConfig& operator=(const OdrConfig&) = delete
+  OdrConfig() = delete;
+  OdrConfig(const OdrConfig&) = delete;
+  OdrConfig& operator=(const OdrConfig&) = delete;
