diff --git a/./output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_mergegen.cc b/./output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_merged.cc
index 9b63a62..3a87a55 100644
--- a/./output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_mergegen.cc
+++ b/./output/art/7a4c7a14-af3127b3-47458ed7/odrefresh@odrefresh_main.no_comments_merged.cc
@@ -147 +147,2 @@ int InitializeConfig(int argc, char** argv, OdrConfig* config) {
-std::string filter = GetProperty("dalvik.vm.systemservercompilerfilter", ""); filter = GetProperty(kSystemPropertySystemServerCompilerFilterOverride, filter);
+    std::string filter = GetProperty("dalvik.vm.systemservercompilerfilter", "");
+    filter = GetProperty(kSystemPropertySystemServerCompilerFilterOverride, filter);
