--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/d7a0c2b1-b0061268-b5f3b9b8/services@inputflinger@reader@InputDevice.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/d7a0c2b1-b0061268-b5f3b9b8/services@inputflinger@reader@InputDevice.no_comments_merged.cpp
@@ -392 +391,0 @@ std::vector<std::unique_ptr<InputMapper>> InputDevice::createMappers(
-<<<<<<< HEAD
@@ -394,14 +392,0 @@ std::vector<std::unique_ptr<InputMapper>> InputDevice::createMappers(
-|||||||
-    static const bool ENABLE_TOUCHPAD_GESTURES_LIBRARY =
-            sysprop::InputProperties::enable_touchpad_gestures_library().value_or(true);
-    const InputDeviceIdentifier identifier = contextPtr.getDeviceIdentifier();
-    const bool isSonyDualShock4Touchpad = identifier.vendor == 0x054c &&
-            (identifier.product == 0x05c4 || identifier.product == 0x09cc);
-    if (ENABLE_TOUCHPAD_GESTURES_LIBRARY && classes.test(InputDeviceClass::TOUCHPAD) &&
-        classes.test(InputDeviceClass::TOUCH_MT) && !isSonyDualShock4Touchpad) {
-=======
-    static const bool ENABLE_TOUCHPAD_GESTURES_LIBRARY =
-            sysprop::InputProperties::enable_touchpad_gestures_library().value_or(true);
-    if (ENABLE_TOUCHPAD_GESTURES_LIBRARY && classes.test(InputDeviceClass::TOUCHPAD) &&
-        classes.test(InputDeviceClass::TOUCH_MT)) {
->>>>>>> b00612688d861942b431e8d3132c70f99bc63f11
