diff --git a/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_mergegen.h b/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
index 5f6f91c..7b2201c 100644
--- a/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_mergegen.h
+++ b/./output/frameworks_av/8ccfaf94-8a06b7a8-c91de17e/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
@@ -137,4 +137,8 @@ private:
-std::map<audio_source_t, std::shared_ptr<EffectDescVector>> mInputSources GUARDED_BY(mMutex);
-std::map<audio_session_t, std::shared_ptr<EffectVector>> mInputSessions GUARDED_BY(mMutex);
-std::map<audio_stream_type_t, std::shared_ptr<EffectDescVector>> mOutputStreams GUARDED_BY(mMutex);
-std::map<audio_session_t, std::shared_ptr<EffectVector>> mOutputSessions GUARDED_BY(mMutex);
+    std::map<audio_source_t, std::shared_ptr<EffectDescVector>> mInputSources
+            GUARDED_BY(mMutex);
+    std::map<audio_session_t, std::shared_ptr<EffectVector>> mInputSessions
+            GUARDED_BY(mMutex);
+    std::map<audio_stream_type_t, std::shared_ptr<EffectDescVector>> mOutputStreams
+            GUARDED_BY(mMutex);
+    std::map<audio_session_t, std::shared_ptr<EffectVector>> mOutputSessions
+            GUARDED_BY(mMutex);
