diff --git a/./output/frameworks_av/2ef972fb-043bd860-b4efe510/services@audiopolicy@service@AudioPolicyEffects.no_comments_mergegen.h b/./output/frameworks_av/2ef972fb-043bd860-b4efe510/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
index 8241dc1..7b2201c 100644
--- a/./output/frameworks_av/2ef972fb-043bd860-b4efe510/services@audiopolicy@service@AudioPolicyEffects.no_comments_mergegen.h
+++ b/./output/frameworks_av/2ef972fb-043bd860-b4efe510/services@audiopolicy@service@AudioPolicyEffects.no_comments_merged.h
@@ -65,0 +66 @@ private:
+        EffectDesc(std::string_view name, const effect_uuid_t& uuid) :
@@ -94 +95,2 @@ private:
-int mRefCount = 0; std::vector<sp<AudioEffect>> mEffects;
+        int mRefCount = 0;
+        std::vector<sp<AudioEffect>> mEffects;
