--- a/./output/redis/eaeba1b2-39c70e72-cf76af6b/src@config.no_comments_conflict.c
+++ b/./output/redis/eaeba1b2-39c70e72-cf76af6b/src@config.no_comments_merged.c
@@ -764 +763,0 @@ void configSetCommand(client *c) {
-<<<<<<< HEAD
@@ -776,13 +774,0 @@ void configSetCommand(client *c) {
-||||||| cf76af6b9
-    } else if (!strcasecmp(c->argv[2]->ptr,"maxmemory")) {
-        if (getLongLongFromObject(o,&ll) == REDIS_ERR ||
-            ll < 0) goto badfmt;
-        server.maxmemory = ll;
-        if (server.maxmemory) {
-            if (server.maxmemory < zmalloc_used_memory()) {
-                redisLog(REDIS_WARNING,"WARNING: the new maxmemory value set via CONFIG SET is smaller than the current memory usage. This will result in keys eviction and/or inability to accept new write commands depending on the maxmemory-policy.");
-            }
-            freeMemoryIfNeeded();
-        }
-    } else if (!strcasecmp(c->argv[2]->ptr,"maxclients")) {
-=======
@@ -793 +778,0 @@ void configSetCommand(client *c) {
->>>>>>> 39c70e72
