diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_merged.cpp
index 6c5f6a5..e43033b 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_merged.cpp
@@ -28,2 +28 @@ public:
- explicit CreateIndexLocalSinkState(ClientContext &context) : arena_allocator(Allocator::Get(context)) {
- }
+ explicit CreateIndexLocalSinkState(ClientContext &context) : arena_allocator(Allocator::Get(context)) {};
@@ -103,15 +101,0 @@ SinkFinalizeType PhysicalCreateIndex::Finalize(Pipeline &pipeline, Event &event,
-<<<<<<< HEAD
- auto &schema = *table.schema;
- auto index_entry = (DuckIndexEntry *)schema.CreateIndex(context, info.get(), &table);
-|||||||
- state.global_index->Verify();
- if (state.global_index->track_memory) {
-  state.global_index->buffer_manager.IncreaseUsedMemory(state.global_index->memory_size);
- }
- auto &schema = *table.schema;
- auto index_entry = (DuckIndexEntry *)schema.CreateIndex(context, info.get(), &table);
-=======
- state.global_index->Verify();
- if (state.global_index->track_memory) {
-  state.global_index->buffer_manager.IncreaseUsedMemory(state.global_index->memory_size);
- }
@@ -120 +103,0 @@ SinkFinalizeType PhysicalCreateIndex::Finalize(Pipeline &pipeline, Event &event,
->>>>>>> 094226ff6959a9c5a5c3556b67f97337a6742800
