diff --git a/./output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_conflict.cpp b/./output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_merged.cpp
index 9050557..e43033b 100644
--- a/./output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_conflict.cpp
+++ b/./output/duckdb/ebb024b0-094226ff-f4295958/src@execution@operator@schema@physical_create_index.no_comments_merged.cpp
@@ -102,15 +101,0 @@ SinkFinalizeType PhysicalCreateIndex::Finalize(Pipeline &pipeline, Event &event,
-<<<<<<< HEAD
- auto &schema = *table.schema;
- auto index_entry = (DuckIndexEntry *)schema.CreateIndex(context, info.get(), &table);
-||||||| f429595834
- state.global_index->Verify();
- if (state.global_index->track_memory) {
-  state.global_index->buffer_manager.IncreaseUsedMemory(state.global_index->memory_size);
- }
- auto &schema = *table.schema;
- auto index_entry = (DuckIndexEntry *)schema.CreateIndex(context, info.get(), &table);
-=======
- state.global_index->Verify();
- if (state.global_index->track_memory) {
-  state.global_index->buffer_manager.IncreaseUsedMemory(state.global_index->memory_size);
- }
@@ -119 +103,0 @@ SinkFinalizeType PhysicalCreateIndex::Finalize(Pipeline &pipeline, Event &event,
->>>>>>> 094226ff
