--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@catalog@catalog_entry@duck_table_entry.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@catalog@catalog_entry@duck_table_entry.no_comments_merged.cpp
@@ -38 +38 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
-                       storage.db, true, index_block->block_id, index_block->offset);
+                       storage.db, index_block->block_id, index_block->offset);
@@ -41 +41 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
-                       storage.db, true);
+                       storage.db);
