--- a/./output/duckdb/ebb024b0-094226ff-f4295958/src@catalog@catalog_entry@duck_table_entry.no_comments_mergegen.cpp
+++ b/./output/duckdb/ebb024b0-094226ff-f4295958/src@catalog@catalog_entry@duck_table_entry.no_comments_merged.cpp
@@ -37 +37,2 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
-desc = make_uniq<ART>(column_ids, TableIOManager::Get(storage), std::move(unbound_expressions), constraint_type, storage.db, index_block->block_id, index_block->offset);
+  art = make_uniq<ART>(column_ids, TableIOManager::Get(storage), std::move(unbound_expressions), constraint_type,
+                       storage.db, index_block->block_id, index_block->offset);
@@ -39 +40,3 @@ desc = make_uniq<ART>(column_ids, TableIOManager::Get(storage), std::move(unboun
-if (!storage.IsRoot()) { throw TransactionException("Transaction conflict: cannot add an index to a table that has been altered!");
+  art = make_uniq<ART>(column_ids, TableIOManager::Get(storage), std::move(unbound_expressions), constraint_type,
+                       storage.db);
+  if (!storage.IsRoot()) {
