diff --git a/./output/duckdb/ebb024b0-094226ff-f4295958/src@storage@checkpoint_manager.no_comments_mergegen.cpp b/./output/duckdb/ebb024b0-094226ff-f4295958/src@storage@checkpoint_manager.no_comments_merged.cpp
index 6bf59f1..4e933c8 100644
--- a/./output/duckdb/ebb024b0-094226ff-f4295958/src@storage@checkpoint_manager.no_comments_mergegen.cpp
+++ b/./output/duckdb/ebb024b0-094226ff-f4295958/src@storage@checkpoint_manager.no_comments_merged.cpp
@@ -247,0 +248,5 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
+ auto &schema_catalog = catalog.GetSchema(context, info->schema);
+ auto &table_catalog = catalog.GetEntry(context, CatalogType::TABLE_ENTRY, info->schema, info->table->table_name)
+                           .Cast<DuckTableEntry>();
+ auto &index_catalog = schema_catalog.CreateIndex(context, *info, table_catalog)->Cast<DuckIndexEntry>();
+ index_catalog.info = table_catalog.GetStorage().info;
@@ -276 +281,2 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
-auto art = make_uniq<ART>(info->column_ids, TableIOManager::Get(storage), std::move(unbound_expressions), info->constraint_type, storage.db, root_block_id, root_offset); index_catalog.index = art.get();
+                            info->constraint_type, storage.db, root_block_id, root_offset);
+  index_catalog.index = art.get();
