diff --git a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_mergebot.cpp b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_merged.cpp
index e96134a..f5813e0 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/frameworks_native/dfe9f602-826244e8-315bc4e1/opengl@libs@EGL@Loader.no_comments_merged.cpp
@@ -36,0 +37,3 @@ static const char* const VENDOR_LIB_EGL_DIR =
+#else
+        "/vendor/lib/egl";
+#endif
@@ -39,0 +43,3 @@ static const char* const SYSTEM_LIB_DIR =
+#else
+        "/system/lib";
+#endif
@@ -65 +71,2 @@ Loader& Loader::getInstance() {
-Loader::driver_t::driver_t(void* gles) {
+Loader::driver_t::driver_t(void* gles)
+{
@@ -67 +74,2 @@ Loader::driver_t::driver_t(void* gles) {
-    for (size_t i = 1; i < NELEM(dso); i++) dso[i] = nullptr;
+    for (size_t i=1 ; i<NELEM(dso) ; i++)
+        dso[i] = nullptr;
@@ -69 +77,2 @@ Loader::driver_t::driver_t(void* gles) {
-Loader::driver_t::~driver_t() {
+Loader::driver_t::~driver_t()
+{
@@ -77 +86,2 @@ Loader::driver_t::~driver_t() {
-int Loader::driver_t::set(void* hnd, int32_t api) {
+int Loader::driver_t::set(void* hnd, int32_t api)
+{
@@ -93,2 +103,6 @@ int Loader::driver_t::set(void* hnd, int32_t api) {
-Loader::Loader() : getProcAddress(nullptr) {}
-Loader::~Loader() {}
+Loader::Loader()
+    : getProcAddress(nullptr)
+{
+}
+Loader::~Loader() {
+}
@@ -223 +237,2 @@ void* Loader::open(egl_connection_t* cnx) {
-    LOG_ALWAYS_FATAL_IF(!cnx->libEgl, "couldn't load system EGL wrapper libraries");
+    LOG_ALWAYS_FATAL_IF(!cnx->libEgl,
+            "couldn't load system EGL wrapper libraries");
@@ -230 +245,2 @@ void* Loader::open(egl_connection_t* cnx) {
-void Loader::close(egl_connection_t* cnx) {
+void Loader::close(egl_connection_t* cnx)
+{
@@ -236 +252,3 @@ void Loader::close(egl_connection_t* cnx) {
-void Loader::init_api(void* dso, char const* const* api, char const* const* ref_api,
+void Loader::init_api(void* dso,
+        char const * const * api,
+        char const * const * ref_api,
@@ -238 +256,2 @@ void Loader::init_api(void* dso, char const* const* api, char const* const* ref_
-                      getProcAddressType getProcAddress) {
+        getProcAddressType getProcAddress)
+{
@@ -395,3 +413,0 @@ Loader::driver_t* Loader::attempt_to_load_angle(egl_connection_t* cnx) {
-<<<<<<< HEAD
-    android::GraphicsEnv::getInstance().setDriverToLoad(android::GpuStatsInfo::Driver::ANGLE);
-|||||||
@@ -399,2 +414,0 @@ Loader::driver_t* Loader::attempt_to_load_angle(egl_connection_t* cnx) {
-=======
->>>>>>> 826244e86737d09e29aa7e6af11d49e51d9117ce
@@ -485 +499,2 @@ void Loader::initialize_api(void* dso, egl_connection_t* cnx, uint32_t mask) {
-        ALOGE_IF(!getProcAddress, "can't find eglGetProcAddress() in EGL driver library");
+        ALOGE_IF(!getProcAddress,
+                "can't find eglGetProcAddress() in EGL driver library");
@@ -506,3 +521,2 @@ void Loader::initialize_api(void* dso, egl_connection_t* cnx, uint32_t mask) {
-                 (__eglMustCastToProperFunctionPointerType*)&cnx
-                         ->hooks[egl_connection_t::GLESv1_INDEX]
-                         ->gl,
+            (__eglMustCastToProperFunctionPointerType*)
+                &cnx->hooks[egl_connection_t::GLESv1_INDEX]->gl,
@@ -513,3 +527,2 @@ void Loader::initialize_api(void* dso, egl_connection_t* cnx, uint32_t mask) {
-                 (__eglMustCastToProperFunctionPointerType*)&cnx
-                         ->hooks[egl_connection_t::GLESv2_INDEX]
-                         ->gl,
+            (__eglMustCastToProperFunctionPointerType*)
+                &cnx->hooks[egl_connection_t::GLESv2_INDEX]->gl,
