--- a/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@storage@checkpoint_manager.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/duckdb/ebb024b0-094226ff-f4295958/src@storage@checkpoint_manager.no_comments_merged.cpp
@@ -248,13 +247,0 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
-<<<<<<< HEAD
- auto schema_catalog = catalog.GetSchema(context, info->schema);
- auto table_catalog =
-     (DuckTableEntry *)catalog.GetEntry(context, CatalogType::TABLE_ENTRY, info->schema, info->table->table_name);
- auto index_catalog = (DuckIndexEntry *)schema_catalog->CreateIndex(context, info.get(), table_catalog);
- index_catalog->info = table_catalog->GetStorage().info;
-|||||||
- auto schema_catalog = catalog.GetSchema(context, info->schema);
- auto table_catalog =
-     (DuckTableEntry *)catalog.GetEntry(context, CatalogType::TABLE_ENTRY, info->schema, info->table->table_name);
- auto index_catalog = (DuckIndexEntry *)schema_catalog->CreateIndex(context, info.get(), table_catalog);
- index_catalog->info = table_catalog->GetStorage().info;
-=======
@@ -266 +252,0 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
->>>>>>> 094226ff6959a9c5a5c3556b67f97337a6742800
@@ -295 +280,0 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
-<<<<<<< HEAD
@@ -297,6 +281,0 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
-  index_catalog->index = art.get();
-|||||||
-                            info->constraint_type, storage.db, true, root_block_id, root_offset);
-  index_catalog->index = art.get();
-=======
-                            info->constraint_type, storage.db, true, root_block_id, root_offset);
@@ -304 +282,0 @@ void CheckpointReader::ReadIndex(ClientContext &context, MetaBlockReader &reader
->>>>>>> 094226ff6959a9c5a5c3556b67f97337a6742800
