--- a/./output/duckdb/ebb024b0-094226ff-f4295958/src@catalog@catalog_entry@duck_table_entry.no_comments_conflict.cpp
+++ b/./output/duckdb/ebb024b0-094226ff-f4295958/src@catalog@catalog_entry@duck_table_entry.no_comments_merged.cpp
@@ -37,7 +36,0 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
-<<<<<<< HEAD
-  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions), constraint_type,
-                       storage->db, index_block->block_id, index_block->offset);
-||||||| f429595834
-  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions), constraint_type,
-                       storage->db, true, index_block->block_id, index_block->offset);
-=======
@@ -45,2 +38 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
-                       storage.db, true, index_block->block_id, index_block->offset);
->>>>>>> 094226ff
+                       storage.db, index_block->block_id, index_block->offset);
@@ -48,9 +39,0 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
-<<<<<<< HEAD
-  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions), constraint_type,
-                       storage->db);
-  if (!storage->IsRoot()) {
-||||||| f429595834
-  art = make_uniq<ART>(column_ids, TableIOManager::Get(*storage), std::move(unbound_expressions), constraint_type,
-                       storage->db, true);
-  if (!storage->IsRoot()) {
-=======
@@ -58 +41 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
-                       storage.db, true);
+                       storage.db);
@@ -60 +42,0 @@ void AddDataTableIndex(DataTable &storage, const ColumnList &columns, const vect
->>>>>>> 094226ff
