diff --git a/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@runtime.no_comments_conflict.cc b/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@runtime.no_comments_merged.cc
index 222d941..f4cc81c 100644
--- a/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@runtime.no_comments_conflict.cc
+++ b/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@runtime.no_comments_merged.cc
@@ -1326 +1325,0 @@ bool Runtime::Init(RuntimeArgumentMap&& runtime_options_in) {
-<<<<<<< HEAD
@@ -1328,6 +1326,0 @@ bool Runtime::Init(RuntimeArgumentMap&& runtime_options_in) {
-||||||| 49a98ce7de
-    if (implicit_null_checks_ || implicit_so_checks_ || implicit_suspend_checks_) {
-      fault_manager.Init();
-=======
-    if (implicit_null_checks_ || implicit_so_checks_ || implicit_suspend_checks_) {
->>>>>>> f4a0906b
