diff --git a/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@gc@collector@mark_compact.no_comments_conflict.h b/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@gc@collector@mark_compact.no_comments_merged.h
index 5601c87..1777e57 100644
--- a/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@gc@collector@mark_compact.no_comments_conflict.h
+++ b/./output/art/28e11de3-f4a0906b-49a98ce7/runtime@gc@collector@mark_compact.no_comments_merged.h
@@ -48,2 +47,0 @@ class MarkCompact final : public GarbageCollector {
-  bool IsUsingSigbusFeature() const { return use_uffd_sigbus_; }
-  bool SigbusHandler(siginfo_t* info) REQUIRES(!lock_) NO_THREAD_SAFETY_ANALYSIS;
@@ -296 +293,0 @@ class MarkCompact final : public GarbageCollector {
-<<<<<<< HEAD
@@ -303,6 +299,0 @@ class MarkCompact final : public GarbageCollector {
-||||||| 49a98ce7de
-  MemMap compaction_buffers_map_;
-=======
-  void ZeropageIoctl(void* addr, bool tolerate_eexist, bool tolerate_enoent);
-  void CopyIoctl(void* dst, void* buffer);
->>>>>>> f4a0906b
@@ -311,5 +301,0 @@ class MarkCompact final : public GarbageCollector {
-<<<<<<< HEAD
-  Mutex lock_;
-||||||| 49a98ce7de
-  Mutex mark_stack_lock_;
-=======
@@ -317 +302,0 @@ class MarkCompact final : public GarbageCollector {
->>>>>>> f4a0906b
