diff --git a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_mergebot.cc b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_merged.cc
index b736949..bdc30b1 100644
--- a/home/whalien/codebase/python/mergebot-eva/output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_merged.cc
@@ -167 +167,3 @@ class SpecialEnv : public EnvWrapper {
-          : env_(env), base_(std::move(base)) {}
+          : env_(env),
+            base_(std::move(base)) {
+      }
@@ -288 +290,2 @@ class SpecialEnv : public EnvWrapper {
-          : target_(std::move(target)), counter_(counter) {}
+          : target_(std::move(target)), counter_(counter) {
+      }
@@ -400,3 +403,3 @@ class DBTest : public testing::Test {
-  DBTest()
-      : option_config_(kDefault),
-        mem_env_(!getenv("MEM_ENV") ? nullptr : new MockEnv(Env::Default())),
+  DBTest() : option_config_(kDefault),
+             mem_env_(!getenv("MEM_ENV") ? nullptr :
+                                           new MockEnv(Env::Default())),
@@ -445,2 +448 @@ class DBTest : public testing::Test {
-           option_config_ == kHashSkipList)) {
-        ;
+           option_config_ == kHashSkipList)) {;
@@ -533 +535,2 @@ class DBTest : public testing::Test {
-        options.memtable_factory.reset(NewHashSkipListRepFactory(16));
+        options.memtable_factory.reset(
+            NewHashSkipListRepFactory(16));
@@ -658 +661,3 @@ class DBTest : public testing::Test {
-  DBImpl* dbfull() { return reinterpret_cast<DBImpl*>(db_); }
+  DBImpl* dbfull() {
+    return reinterpret_cast<DBImpl*>(db_);
+  }
@@ -683 +688,2 @@ class DBTest : public testing::Test {
-  Status TryReopenWithColumnFamilies(const std::vector<std::string>& cfs,
+  Status TryReopenWithColumnFamilies(
+      const std::vector<std::string>& cfs,
@@ -700 +706,3 @@ class DBTest : public testing::Test {
-  void Reopen(const Options& options) { ASSERT_OK(TryReopen(options)); }
+  void Reopen(const Options& options) {
+    ASSERT_OK(TryReopen(options));
+  }
@@ -747 +755,3 @@ class DBTest : public testing::Test {
-  Status Delete(const std::string& k) { return db_->Delete(WriteOptions(), k); }
+  Status Delete(const std::string& k) {
+    return db_->Delete(WriteOptions(), k);
+  }
@@ -1001,3 +1011,3 @@ class DBTest : public testing::Test {
-    fprintf(
-        stderr, "maxoverlap: %lld\n",
-        static_cast<long long>(dbfull()->TEST_MaxNextLevelOverlappingBytes()));
+    fprintf(stderr, "maxoverlap: %lld\n",
+            static_cast<long long>(
+                dbfull()->TEST_MaxNextLevelOverlappingBytes()));
@@ -1087,3 +1097,3 @@ class DBTest : public testing::Test {
-  static UpdateStatus updateInPlaceSmallerSize(char* prevValue,
-                                               uint32_t* prevSize, Slice delta,
-                                               std::string* newValue) {
+  static UpdateStatus
+      updateInPlaceSmallerSize(char* prevValue, uint32_t* prevSize,
+                               Slice delta, std::string* newValue) {
@@ -1100,4 +1110,3 @@ class DBTest : public testing::Test {
-  static UpdateStatus updateInPlaceSmallerVarintSize(char* prevValue,
-                                                     uint32_t* prevSize,
-                                                     Slice delta,
-                                                     std::string* newValue) {
+  static UpdateStatus
+      updateInPlaceSmallerVarintSize(char* prevValue, uint32_t* prevSize,
+                                     Slice delta, std::string* newValue) {
@@ -1114,3 +1123,3 @@ class DBTest : public testing::Test {
-  static UpdateStatus updateInPlaceLargerSize(char* prevValue,
-                                              uint32_t* prevSize, Slice delta,
-                                              std::string* newValue) {
+  static UpdateStatus
+      updateInPlaceLargerSize(char* prevValue, uint32_t* prevSize,
+                              Slice delta, std::string* newValue) {
@@ -1120,3 +1129,3 @@ class DBTest : public testing::Test {
-  static UpdateStatus updateInPlaceNoAction(char* prevValue, uint32_t* prevSize,
-                                            Slice delta,
-                                            std::string* newValue) {
+  static UpdateStatus
+      updateInPlaceNoAction(char* prevValue, uint32_t* prevSize,
+                            Slice delta, std::string* newValue) {
@@ -1351,2 +1360 @@ TEST_F(DBTest, CompactedDB) {
-  std::vector<Status> status_list = dbfull()->MultiGet(
-      ReadOptions(),
+  std::vector<Status> status_list = dbfull()->MultiGet(ReadOptions(),
@@ -1470,2 +1478 @@ class CoutingUserTblPropCollector : public TablePropertiesCollector {
-        {"CoutingUserTblPropCollector", message_},
-        {"Count", encoded},
+        {"CoutingUserTblPropCollector", message_}, {"Count", encoded},
@@ -2590,2 +2597 @@ TEST_F(DBTest, NumImmutableMemTable) {
-    SetPerfLevel(kEnableTime);
-    ;
+    SetPerfLevel(kEnableTime);;
@@ -2864,2 +2870 @@ TEST_F(DBTest, FLUSH) {
-    SetPerfLevel(kEnableTime);
-    ;
+    SetPerfLevel(kEnableTime);;
@@ -3117,2 +3122,8 @@ TEST_F(DBTest, TrivialMoveNonOverlappingFiles) {
-      {100, 199}, {300, 399}, {0, 99}, {200, 299},
-      {600, 699}, {400, 499}, {500, 550}, {551, 599},
+    {100, 199},
+    {300, 399},
+    {0, 99},
+    {200, 299},
+    {600, 699},
+    {400, 499},
+    {500, 550},
+    {551, 599},
@@ -3148,2 +3159,7 @@ TEST_F(DBTest, TrivialMoveNonOverlappingFiles) {
-      {100, 199}, {300, 399}, {0, 99}, {200, 299},
-      {600, 699}, {400, 499}, {500, 560},
+    {100, 199},
+    {300, 399},
+    {0, 99},
+    {200, 299},
+    {600, 699},
+    {400, 499},
+    {500, 560},
@@ -3380,2 +3396,2 @@ class KeepFilter : public CompactionFilter {
-                      std::string* new_value,
-                      bool* value_changed) const override {
+                      std::string* new_value, bool* value_changed) const
+      override {
@@ -3390,2 +3406,2 @@ class DeleteFilter : public CompactionFilter {
-                      std::string* new_value,
-                      bool* value_changed) const override {
+                      std::string* new_value, bool* value_changed) const
+      override {
@@ -3427,2 +3443,2 @@ class ChangeFilter : public CompactionFilter {
-                      std::string* new_value,
-                      bool* value_changed) const override {
+                      std::string* new_value, bool* value_changed) const
+      override {
@@ -3648,0 +3665,36 @@ TEST_P(DBTestUniversalCompactionMultiLevels, UniversalCompactionMultiLevels) {
+TEST_P(DBTestUniversalCompactionMultiLevels, UniversalCompactionTrivialMove) {
+  int32_t trivial_move = 0;
+  int32_t non_trivial_move = 0;
+  rocksdb::SyncPoint::GetInstance()->SetCallBack(
+      "DBImpl::BackgroundCompaction:TrivialMove",
+      [&](void* arg) { trivial_move++; });
+  rocksdb::SyncPoint::GetInstance()->SetCallBack(
+      "DBImpl::BackgroundCompaction:NonTrivial",
+      [&](void* arg) { non_trivial_move++; });
+  rocksdb::SyncPoint::GetInstance()->EnableProcessing();
+  Options options;
+  options.compaction_style = kCompactionStyleUniversal;
+  options.compaction_options_universal.allow_trivial_move = true;
+  options.num_levels = 3;
+  options.write_buffer_size = 100 << 10;
+  options.level0_file_num_compaction_trigger = 3;
+  options.max_background_compactions = 1;
+  options.target_file_size_base = 32 * 1024;
+  options = CurrentOptions(options);
+  DestroyAndReopen(options);
+  CreateAndReopenWithCF({"pikachu"}, options);
+  options.compaction_options_universal.max_size_amplification_percent = 110;
+  options = CurrentOptions(options);
+  ReopenWithColumnFamilies({"default", "pikachu"}, options);
+  Random rnd(301);
+  int num_keys = 15000;
+  for (int i = 0; i < num_keys; i++) {
+    ASSERT_OK(Put(1, Key(i), Key(i)));
+  }
+  std::vector<std::string> values;
+  ASSERT_OK(Flush(1));
+  dbfull()->TEST_WaitForCompact();
+  ASSERT_GT(trivial_move, 0);
+  ASSERT_EQ(non_trivial_move, 0);
+  rocksdb::SyncPoint::GetInstance()->DisableProcessing();
+}
@@ -3669,2 +3721,2 @@ TEST_P(DBTestUniversalCompactionParallel, UniversalCompactionParallel) {
-  rocksdb::SyncPoint::GetInstance()->SetCallBack(
-      "CompactionJob::Run():Start", [&](void* arg) {
+  rocksdb::SyncPoint::GetInstance()->SetCallBack("CompactionJob::Run():Start",
+                                                 [&](void* arg) {
@@ -3831,2 +3883 @@ TEST_F(DBTest, CompressedCache) {
-    ReopenWithColumnFamilies(
-        {"default", "pikachu"},
+    ReopenWithColumnFamilies({"default", "pikachu"},
@@ -4456,2 +4507,4 @@ void MinLevelHelper(DBTest* self, Options& options) {
-  for (int num = 0; num < options.level0_file_num_compaction_trigger - 1;
-       num++) {
+  for (int num = 0;
+    num < options.level0_file_num_compaction_trigger - 1;
+    num++)
+  {
@@ -4477,4 +4530 @@ bool MinLevelToCompress(CompressionType& type, Options& options, int wbits,
-  fprintf(stderr,
-          "Test with compression options : window_bits = %d, level =  %d, "
-          "strategy = %d}\n",
-          wbits, lev, strategy);
+  fprintf(stderr, "Test with compression options : window_bits = %d, level =  %d, strategy = %d}\n", wbits, lev, strategy);
@@ -4607 +4657,2 @@ TEST_F(DBTest, InPlaceUpdateCallbackSmallerSize) {
-    options.inplace_callback = rocksdb::DBTest::updateInPlaceSmallerSize;
+    options.inplace_callback =
+      rocksdb::DBTest::updateInPlaceSmallerSize;
@@ -4627 +4678,2 @@ TEST_F(DBTest, InPlaceUpdateCallbackSmallerVarintSize) {
-    options.inplace_callback = rocksdb::DBTest::updateInPlaceSmallerVarintSize;
+    options.inplace_callback =
+      rocksdb::DBTest::updateInPlaceSmallerVarintSize;
@@ -4647 +4699,2 @@ TEST_F(DBTest, InPlaceUpdateCallbackLargeNewValue) {
-    options.inplace_callback = rocksdb::DBTest::updateInPlaceLargerSize;
+    options.inplace_callback =
+      rocksdb::DBTest::updateInPlaceLargerSize;
@@ -4665 +4718,2 @@ TEST_F(DBTest, InPlaceUpdateCallbackNoAction) {
-    options.inplace_callback = rocksdb::DBTest::updateInPlaceNoAction;
+    options.inplace_callback =
+      rocksdb::DBTest::updateInPlaceNoAction;
@@ -4807 +4861,2 @@ TEST_F(DBTest, CompactionFilterWithValueChange) {
-    options.compaction_filter_factory = std::make_shared<ChangeFilterFactory>();
+    options.compaction_filter_factory =
+      std::make_shared<ChangeFilterFactory>();
@@ -4946,2 +5001,3 @@ class KeepFilterV2 : public CompactionFilterV2 {
-  virtual std::vector<bool> Filter(
-      int level, const SliceVector& keys, const SliceVector& existing_values,
+  virtual std::vector<bool> Filter(int level,
+                                   const SliceVector& keys,
+                                   const SliceVector& existing_values,
@@ -4949 +5005,2 @@ class KeepFilterV2 : public CompactionFilterV2 {
-      std::vector<bool>* values_changed) const override {
+                                   std::vector<bool>* values_changed)
+    const override {
@@ -4960 +5017,3 @@ class KeepFilterV2 : public CompactionFilterV2 {
-  virtual const char* Name() const override { return "KeepFilterV2"; }
+  virtual const char* Name() const override {
+    return "KeepFilterV2";
+  }
@@ -4964,2 +5023,3 @@ class DeleteFilterV2 : public CompactionFilterV2 {
-  virtual std::vector<bool> Filter(
-      int level, const SliceVector& keys, const SliceVector& existing_values,
+  virtual std::vector<bool> Filter(int level,
+                                   const SliceVector& keys,
+                                   const SliceVector& existing_values,
@@ -4967 +5027,2 @@ class DeleteFilterV2 : public CompactionFilterV2 {
-      std::vector<bool>* values_changed) const override {
+                                   std::vector<bool>* values_changed)
+    const override {
@@ -4978 +5039,3 @@ class DeleteFilterV2 : public CompactionFilterV2 {
-  virtual const char* Name() const override { return "DeleteFilterV2"; }
+  virtual const char* Name() const override {
+    return "DeleteFilterV2";
+  }
@@ -4982,2 +5045,3 @@ class ChangeFilterV2 : public CompactionFilterV2 {
-  virtual std::vector<bool> Filter(
-      int level, const SliceVector& keys, const SliceVector& existing_values,
+  virtual std::vector<bool> Filter(int level,
+                                   const SliceVector& keys,
+                                   const SliceVector& existing_values,
@@ -4985 +5049,2 @@ class ChangeFilterV2 : public CompactionFilterV2 {
-      std::vector<bool>* values_changed) const override {
+                                   std::vector<bool>* values_changed)
+    const override {
@@ -4996 +5061,3 @@ class ChangeFilterV2 : public CompactionFilterV2 {
-  virtual const char* Name() const override { return "ChangeFilterV2"; }
+  virtual const char* Name() const override {
+    return "ChangeFilterV2";
+  }
@@ -5002 +5069,2 @@ class KeepFilterFactoryV2 : public CompactionFilterFactoryV2 {
-  virtual std::unique_ptr<CompactionFilterV2> CreateCompactionFilterV2(
+  virtual std::unique_ptr<CompactionFilterV2>
+  CreateCompactionFilterV2(
@@ -5006 +5074,3 @@ class KeepFilterFactoryV2 : public CompactionFilterFactoryV2 {
-  virtual const char* Name() const override { return "KeepFilterFactoryV2"; }
+  virtual const char* Name() const override {
+    return "KeepFilterFactoryV2";
+  }
@@ -5012 +5082,2 @@ class DeleteFilterFactoryV2 : public CompactionFilterFactoryV2 {
-  virtual std::unique_ptr<CompactionFilterV2> CreateCompactionFilterV2(
+  virtual std::unique_ptr<CompactionFilterV2>
+  CreateCompactionFilterV2(
@@ -5016 +5087,3 @@ class DeleteFilterFactoryV2 : public CompactionFilterFactoryV2 {
-  virtual const char* Name() const override { return "DeleteFilterFactoryV2"; }
+  virtual const char* Name() const override {
+    return "DeleteFilterFactoryV2";
+  }
@@ -5022 +5095,2 @@ class ChangeFilterFactoryV2 : public CompactionFilterFactoryV2 {
-  virtual std::unique_ptr<CompactionFilterV2> CreateCompactionFilterV2(
+  virtual std::unique_ptr<CompactionFilterV2>
+  CreateCompactionFilterV2(
@@ -5026 +5100,3 @@ class ChangeFilterFactoryV2 : public CompactionFilterFactoryV2 {
-  virtual const char* Name() const override { return "ChangeFilterFactoryV2"; }
+  virtual const char* Name() const override {
+    return "ChangeFilterFactoryV2";
+  }
@@ -5034,2 +5110,2 @@ TEST_F(DBTest, CompactionFilterV2) {
-  options.compaction_filter_factory_v2 =
-      std::make_shared<KeepFilterFactoryV2>(prefix_extractor.get());
+  options.compaction_filter_factory_v2
+    = std::make_shared<KeepFilterFactoryV2>(prefix_extractor.get());
@@ -5191 +5267,2 @@ static bool Between(uint64_t val, uint64_t low, uint64_t high) {
-            (unsigned long long)(val), (unsigned long long)(low),
+            (unsigned long long)(val),
+            (unsigned long long)(low),
@@ -5660,2 +5737,2 @@ TEST_F(DBTest, ComparatorCheck) {
-    Status s = TryReopenWithColumnFamilies(
-        {"default", "pikachu"}, std::vector<Options>({options, new_options}));
+    Status s = TryReopenWithColumnFamilies({"default", "pikachu"},
+        std::vector<Options>({options, new_options}));
@@ -6025 +6102,2 @@ TEST_F(DBTest, ManifestWriteError) {
-    std::atomic<bool>* error_type = (iter == 0) ? &env_->manifest_sync_error_
+    std::atomic<bool>* error_type = (iter == 0)
+        ? &env_->manifest_sync_error_
@@ -6195,2 +6273,3 @@ class WrappedBloom : public FilterPolicy {
-  explicit WrappedBloom(int bits_per_key)
-      : filter_(NewBloomFilterPolicy(bits_per_key)), counter_(0) {}
+  explicit WrappedBloom(int bits_per_key) :
+        filter_(NewBloomFilterPolicy(bits_per_key)),
+        counter_(0) {}
@@ -6199,2 +6278,2 @@ class WrappedBloom : public FilterPolicy {
-  void CreateFilter(const rocksdb::Slice* keys, int n,
-                    std::string* dst) const override {
+  void CreateFilter(const rocksdb::Slice* keys, int n, std::string* dst)
+      const override {
@@ -6207,2 +6286,2 @@ class WrappedBloom : public FilterPolicy {
-  bool KeyMayMatch(const rocksdb::Slice& key,
-                   const rocksdb::Slice& filter) const override {
+  bool KeyMayMatch(const rocksdb::Slice& key, const rocksdb::Slice& filter)
+      const override {
@@ -6216 +6295,3 @@ class WrappedBloom : public FilterPolicy {
-  rocksdb::Slice convertKey(const rocksdb::Slice& key) const { return key; }
+  rocksdb::Slice convertKey(const rocksdb::Slice& key) const {
+    return key;
+  }
@@ -6635 +6716,2 @@ namespace {
-SequenceNumber ReadRecords(std::unique_ptr<TransactionLogIterator>& iter,
+SequenceNumber ReadRecords(
+    std::unique_ptr<TransactionLogIterator>& iter,
@@ -6650 +6732,2 @@ SequenceNumber ReadRecords(std::unique_ptr<TransactionLogIterator>& iter,
-void ExpectRecords(const int expected_no_records,
+void ExpectRecords(
+    const int expected_no_records,
@@ -6689 +6772,2 @@ TEST_F(DBTest, TransactionLogIteratorRace) {
-      {"WalManager::GetSortedWalsOfType:1", "WalManager::PurgeObsoleteFiles:1",
+      {"WalManager::GetSortedWalsOfType:1",
+       "WalManager::PurgeObsoleteFiles:1",
@@ -6693,2 +6777,2 @@ TEST_F(DBTest, TransactionLogIteratorRace) {
-    rocksdb::SyncPoint::GetInstance()->LoadDependency({
-        {sync_points[test][0], sync_points[test][1]},
+    rocksdb::SyncPoint::GetInstance()->LoadDependency(
+      { { sync_points[test][0], sync_points[test][1] },
@@ -6784,6 +6868,35 @@ class RecoveryTestHelper {
-  static size_t FillData(DBTest* test, const Options& options) {
-    size_t count = 0;
-    test->DestroyAndReopen(options);
-    for (int i = 0; i < 1024; i++) {
-      test->Put("key" + ToString(i), test->DummyString(10));
-      ++count;
+  static const int kWALFilesCount = 10;
+  static const int kWALFileOffset = 10;
+  static const int kKeysPerWALFile = 1024;
+  static const int kValueSize = 10;
+  static void FillData(DBTest* test, Options& options,
+                       const size_t wal_count, size_t & count) {
+    DBOptions & db_options = options;
+    count = 0;
+    shared_ptr<Cache> table_cache = NewLRUCache(50000, 16);
+    EnvOptions env_options;
+    WriteBuffer write_buffer(db_options.db_write_buffer_size);
+    unique_ptr<VersionSet> versions;
+    unique_ptr<WalManager> wal_manager;
+    WriteController write_controller;
+    versions.reset(new VersionSet(test->dbname_, &db_options, env_options,
+                                  table_cache.get(), &write_buffer,
+                                  &write_controller));
+    wal_manager.reset(new WalManager(db_options, env_options));
+    std::unique_ptr<log::Writer> current_log_writer;
+    for (size_t j = kWALFileOffset; j < wal_count + kWALFileOffset; j++) {
+      uint64_t current_log_number = j;
+      std::string fname = LogFileName(test->dbname_, current_log_number);
+      unique_ptr<WritableFile> file;
+      ASSERT_OK(db_options.env->NewWritableFile(fname, &file, env_options));
+      current_log_writer.reset(new log::Writer(std::move(file)));
+      for (int i = 0; i < kKeysPerWALFile; i++) {
+        std::string key = "key" + ToString(count++);
+        std::string value = test->DummyString(kValueSize);
+        assert(current_log_writer.get() != nullptr);
+        uint64_t seq = versions->LastSequence() + 1;
+        WriteBatch batch;
+        batch.Put(key, value);
+        WriteBatchInternal::SetSequence(&batch, seq);
+        current_log_writer->AddRecord(WriteBatchInternal::Contents(&batch));
+        versions->SetLastSequence(seq);
@@ -6790,0 +6904,8 @@ class RecoveryTestHelper {
+    }
+  }
+  static size_t FillData(DBTest* test, Options& options) {
+    options.create_if_missing = true;
+    test->DestroyAndReopen(options);
+    test->Close();
+    size_t count = 0;
+    FillData(test, options, kWALFilesCount, count);
@@ -6795 +6916 @@ class RecoveryTestHelper {
-    for (size_t i = 0; i < 1024; i++) {
+    for (size_t i = 0; i < kWALFilesCount * kKeysPerWALFile; i++) {
@@ -6801,0 +6923,17 @@ class RecoveryTestHelper {
+  static void CorruptWAL(DBTest * test, Options& options,
+                         const double off, const double len,
+                         const int wal_file_id, const bool trunc = false) {
+    Env* env = options.env;
+    std::string fname = LogFileName(test->dbname_, wal_file_id);
+    uint64_t size;
+    ASSERT_OK(env->GetFileSize(fname, &size));
+    ASSERT_GT(size, 0);
+#ifdef OS_WIN
+    test->Close();
+#endif
+    if (trunc) {
+      ASSERT_EQ(0, truncate(fname.c_str(), size * off));
+    } else {
+      InduceCorruption(fname, size * off, size * len);
+    }
+  }
@@ -6813,17 +6950,0 @@ class RecoveryTestHelper {
-  static void CorruptWAL(DBTest* test, const double off, const double len,
-                         const bool trunc = false) {
-    rocksdb::VectorLogPtr wal_files;
-    ASSERT_OK(test->dbfull()->GetSortedWalFiles(wal_files));
-    ASSERT_EQ(wal_files.size(), 1);
-    const auto logfile_path =
-        test->dbname_ + "/" + wal_files.front()->PathName();
-    auto size = wal_files.front()->SizeFileBytes();
-#ifdef OS_WIN
-    test->Close();
-#endif
-    if (trunc) {
-      ASSERT_EQ(0, truncate(logfile_path.c_str(), size * off));
-    } else {
-      InduceCorruption(logfile_path, size * off, size * len);
-    }
-  }
@@ -6831,0 +6953,2 @@ TEST_F(DBTest, kTolerateCorruptedTailRecords) {
+  const int jstart = RecoveryTestHelper::kWALFileOffset;
+  const int jend = jstart + RecoveryTestHelper::kWALFilesCount;
@@ -6833,0 +6957 @@ TEST_F(DBTest, kTolerateCorruptedTailRecords) {
+      for (int j = jstart; j < jend; j++) {
@@ -6836 +6960,2 @@ TEST_F(DBTest, kTolerateCorruptedTailRecords) {
-      RecoveryTestHelper::CorruptWAL(this, i * .3, .1, trunc);
+        RecoveryTestHelper::CorruptWAL(this, options, i * .3,
+                                                 .1, j, trunc);
@@ -6839,0 +6965 @@ TEST_F(DBTest, kTolerateCorruptedTailRecords) {
+          options.create_if_missing = false;
@@ -6851,0 +6978 @@ TEST_F(DBTest, kTolerateCorruptedTailRecords) {
+}
@@ -6852,0 +6980,2 @@ TEST_F(DBTest, kAbsoluteConsistency) {
+  const int jstart = RecoveryTestHelper::kWALFileOffset;
+  const int jend = jstart + RecoveryTestHelper::kWALFilesCount;
@@ -6855,0 +6985 @@ TEST_F(DBTest, kAbsoluteConsistency) {
+  options.create_if_missing = false;
@@ -6863 +6993 @@ TEST_F(DBTest, kAbsoluteConsistency) {
-      options = CurrentOptions();
+      for (int j = jstart; j < jend; j++) {
@@ -6865 +6995,2 @@ TEST_F(DBTest, kAbsoluteConsistency) {
-      RecoveryTestHelper::CorruptWAL(this, i * .3, .1, trunc);
+        RecoveryTestHelper::CorruptWAL(this, options, i * .3,
+                                                .1, j, trunc);
@@ -6866,0 +6998 @@ TEST_F(DBTest, kAbsoluteConsistency) {
+        options.create_if_missing = false;
@@ -6870,0 +7003 @@ TEST_F(DBTest, kAbsoluteConsistency) {
+}
@@ -6871,0 +7005,4 @@ TEST_F(DBTest, kPointInTimeRecovery) {
+  const int jstart = RecoveryTestHelper::kWALFileOffset;
+  const int jend = jstart + RecoveryTestHelper::kWALFilesCount;
+  const int maxkeys = RecoveryTestHelper::kWALFilesCount *
+                        RecoveryTestHelper::kKeysPerWALFile;
@@ -6873,0 +7011 @@ TEST_F(DBTest, kPointInTimeRecovery) {
+      for (int j = jstart; j < jend; j++) {
@@ -6876 +7014,2 @@ TEST_F(DBTest, kPointInTimeRecovery) {
-      RecoveryTestHelper::CorruptWAL(this, i * .3, .1, trunc);
+        RecoveryTestHelper::CorruptWAL(this, options, i * .3,
+                                                .1, j, trunc);
@@ -6877,0 +7017 @@ TEST_F(DBTest, kPointInTimeRecovery) {
+        options.create_if_missing = false;
@@ -6882 +7022 @@ TEST_F(DBTest, kPointInTimeRecovery) {
-      for (size_t j = 0; j < 1024; ++j) {
+        for (size_t k = 0; k < maxkeys; ++k) {
@@ -6889,2 +7029,9 @@ TEST_F(DBTest, kPointInTimeRecovery) {
-      ASSERT_TRUE(i != 0 || recovered_row_count == 0);
-      ASSERT_TRUE(i != 1 || recovered_row_count < row_count / 2);
+        const size_t min = RecoveryTestHelper::kKeysPerWALFile *
+                              (j - RecoveryTestHelper::kWALFileOffset);
+        ASSERT_GE(recovered_row_count, min);
+        if (!trunc && i != 0) {
+          const size_t max = RecoveryTestHelper::kKeysPerWALFile *
+                                (j - RecoveryTestHelper::kWALFileOffset + 1);
+          ASSERT_LE(recovered_row_count, max);
+        }
+      }
@@ -6894,0 +7042,2 @@ TEST_F(DBTest, kSkipAnyCorruptedRecords) {
+  const int jstart = RecoveryTestHelper::kWALFileOffset;
+  const int jend = jstart + RecoveryTestHelper::kWALFilesCount;
@@ -6896,0 +7046 @@ TEST_F(DBTest, kSkipAnyCorruptedRecords) {
+      for (int j = jstart; j < jend; j++) {
@@ -6899 +7049,2 @@ TEST_F(DBTest, kSkipAnyCorruptedRecords) {
-      RecoveryTestHelper::CorruptWAL(this, i * .3, .1, trunc);
+        RecoveryTestHelper::CorruptWAL(this, options, i * .3,
+                                                .1, j, trunc);
@@ -6900,0 +7052 @@ TEST_F(DBTest, kSkipAnyCorruptedRecords) {
+        options.create_if_missing = false;
@@ -6909,0 +7062 @@ TEST_F(DBTest, kSkipAnyCorruptedRecords) {
+}
@@ -7040,2 +7193,2 @@ static void MTThreadBody(void* arg) {
-          ASSERT_EQ(5, sscanf(values[i].c_str(), "%d.%d.%d.%d.%d", &k, &w, &c,
-                              &cf, &u))
+          ASSERT_EQ(5, sscanf(values[i].c_str(), "%d.%d.%d.%d.%d", &k, &w,
+                              &c, &cf, &u))
@@ -7305 +7458,2 @@ class ModelDB : public DB {
-  virtual Status CompactFiles(const CompactionOptions& compact_options,
+  virtual Status CompactFiles(
+      const CompactionOptions& compact_options,
@@ -7308,2 +7462 @@ class ModelDB : public DB {
-                              const int output_level,
-                              const int output_path_id = -1) override {
+      const int output_level, const int output_path_id = -1) override {
@@ -7359,2 +7512,2 @@ class ModelDB : public DB {
-      const TransactionLogIterator::ReadOptions& read_options =
-          TransactionLogIterator::ReadOptions()) override {
+      const TransactionLogIterator::ReadOptions&
+          read_options = TransactionLogIterator::ReadOptions()) override {
@@ -7373 +7526,2 @@ class ModelDB : public DB {
-        : map_(map), owned_(owned), iter_(map_->end()) {}
+        : map_(map), owned_(owned), iter_(map_->end()) {
+    }
@@ -7418 +7572,3 @@ static std::string RandomKey(Random* rnd, int minimum = 0) {
-static bool CompareIterators(int step, DB* model, DB* db,
+static bool CompareIterators(int step,
+                             DB* model,
+                             DB* db,
@@ -7429 +7585,2 @@ static bool CompareIterators(int step, DB* model, DB* db,
-       ok && miter->Valid() && dbiter->Valid(); miter->Next(), dbiter->Next()) {
+       ok && miter->Valid() && dbiter->Valid();
+       miter->Next(), dbiter->Next()) {
@@ -7432 +7589,2 @@ static bool CompareIterators(int step, DB* model, DB* db,
-      fprintf(stderr, "step %d: Key mismatch: '%s' vs. '%s'\n", step,
+      fprintf(stderr, "step %d: Key mismatch: '%s' vs. '%s'\n",
+              step,
@@ -7440 +7598,2 @@ static bool CompareIterators(int step, DB* model, DB* db,
-              step, EscapeString(miter->key()).c_str(),
+              step,
+              EscapeString(miter->key()).c_str(),
@@ -7470 +7629,2 @@ TEST_F(DBTest, Randomized) {
-      if (option_config_ == kHashSkipList || option_config_ == kHashLinkList ||
+      if (option_config_ == kHashSkipList ||
+          option_config_ == kHashLinkList ||
@@ -7479,2 +7639,4 @@ TEST_F(DBTest, Randomized) {
-        v = RandomString(
-            &rnd, rnd.OneIn(20) ? 100 + rnd.Uniform(100) : rnd.Uniform(8));
+        v = RandomString(&rnd,
+                         rnd.OneIn(20)
+                         ? 100 + rnd.Uniform(100)
+                         : rnd.Uniform(8));
@@ -7609 +7771,2 @@ void PrefixScanInit(DBTest* dbtest) {
-    snprintf(buf, sizeof(buf), "%02d______:end", small_range_sstfiles + i + 1);
+    snprintf(buf, sizeof(buf), "%02d______:end",
+             small_range_sstfiles+i+1);
@@ -7749,2 +7912 @@ TEST_F(DBTest, TailingIteratorDeletes) {
-  for (; iter->Valid(); iter->Next(), ++count)
-    ;
+  for (; iter->Valid(); iter->Next(), ++count) ;
@@ -8202,2 +8364,2 @@ TEST_F(DBTest, RateLimitingTest) {
-    ASSERT_OK(
-        Put(RandomString(&rnd, 32), RandomString(&rnd, (1 << 10) + 1), wo));
+    ASSERT_OK(Put(RandomString(&rnd, 32),
+                  RandomString(&rnd, (1 << 10) + 1), wo));
@@ -8214,2 +8376,2 @@ TEST_F(DBTest, RateLimitingTest) {
-    ASSERT_OK(
-        Put(RandomString(&rnd, 32), RandomString(&rnd, (1 << 10) + 1), wo));
+    ASSERT_OK(Put(RandomString(&rnd, 32),
+                  RandomString(&rnd, (1 << 10) + 1), wo));
@@ -8230,2 +8392,2 @@ TEST_F(DBTest, RateLimitingTest) {
-    ASSERT_OK(
-        Put(RandomString(&rnd, 32), RandomString(&rnd, (1 << 10) + 1), wo));
+    ASSERT_OK(Put(RandomString(&rnd, 32),
+                  RandomString(&rnd, (1 << 10) + 1), wo));
@@ -8242,2 +8404,3 @@ namespace {
-bool HaveOverlappingKeyRanges(const Comparator* c, const SstFileMetaData& a,
-                              const SstFileMetaData& b) {
+  bool HaveOverlappingKeyRanges(
+      const Comparator* c,
+      const SstFileMetaData& a, const SstFileMetaData& b) {
@@ -8261,2 +8424,4 @@ void GetOverlappingFileNumbersForLevelCompaction(
-    const ColumnFamilyMetaData& cf_meta, const Comparator* comparator,
-    int min_level, int max_level, const SstFileMetaData* input_file_meta,
+      const ColumnFamilyMetaData& cf_meta,
+      const Comparator* comparator,
+      int min_level, int max_level,
+      const SstFileMetaData* input_file_meta,
@@ -8269 +8434,2 @@ void GetOverlappingFileNumbersForLevelCompaction(
-        if (HaveOverlappingKeyRanges(comparator, *included_file, file)) {
+          if (HaveOverlappingKeyRanges(
+                  comparator, *included_file, file)) {
@@ -8290,3 +8456,6 @@ void VerifyCompactionResult(
-const SstFileMetaData* PickFileRandomly(const ColumnFamilyMetaData& cf_meta,
-                                        Random* rand, int* level = nullptr) {
-  auto file_id = rand->Uniform(static_cast<int>(cf_meta.file_count)) + 1;
+  const SstFileMetaData* PickFileRandomly(
+      const ColumnFamilyMetaData& cf_meta,
+      Random* rand,
+      int* level = nullptr) {
+    auto file_id = rand->Uniform(static_cast<int>(
+        cf_meta.file_count)) + 1;
@@ -8340,2 +8509,2 @@ TEST_F(DBTest, DISABLED_CompactFilesOnLevelCompaction) {
-          cf_meta, options.comparator, level, output_level, file_meta,
-          &overlapping_file_names);
+          cf_meta, options.comparator, level, output_level,
+          file_meta, &overlapping_file_names);
@@ -8343 +8512,2 @@ TEST_F(DBTest, DISABLED_CompactFilesOnLevelCompaction) {
-    ASSERT_OK(dbfull()->CompactFiles(CompactionOptions(), handles_[1],
+    ASSERT_OK(dbfull()->CompactFiles(
+        CompactionOptions(), handles_[1],
@@ -8384 +8554,2 @@ TEST_F(DBTest, CompactFilesOnUniversalCompaction) {
-    compaction_input_file_names.push_back(cf_meta.levels[0].files[0].name);
+    compaction_input_file_names.push_back(
+        cf_meta.levels[0].files[0].name);
@@ -8386,5 +8557,5 @@ TEST_F(DBTest, CompactFilesOnUniversalCompaction) {
-  ASSERT_TRUE(!dbfull()
-                   ->CompactFiles(CompactionOptions(), handles_[1],
-                                  compaction_input_file_names, 1)
-                   .ok());
-  ASSERT_OK(dbfull()->CompactFiles(CompactionOptions(), handles_[1],
+  ASSERT_TRUE(!dbfull()->CompactFiles(
+      CompactionOptions(), handles_[1],
+      compaction_input_file_names, 1).ok());
+  ASSERT_OK(dbfull()->CompactFiles(
+      CompactionOptions(), handles_[1],
@@ -8394 +8565,2 @@ TEST_F(DBTest, CompactFilesOnUniversalCompaction) {
-      cf_meta, std::set<std::string>(compaction_input_file_names.begin(),
+      cf_meta,
+      std::set<std::string>(compaction_input_file_names.begin(),
@@ -8397 +8568,0 @@ TEST_F(DBTest, CompactFilesOnUniversalCompaction) {
-  compaction_input_file_names.push_back(cf_meta.levels[0].files[0].name);
@@ -8399,2 +8570,6 @@ TEST_F(DBTest, CompactFilesOnUniversalCompaction) {
-      cf_meta.levels[0].files[cf_meta.levels[0].files.size() - 1].name);
-  ASSERT_OK(dbfull()->CompactFiles(CompactionOptions(), handles_[1],
+      cf_meta.levels[0].files[0].name);
+  compaction_input_file_names.push_back(
+      cf_meta.levels[0].files[
+          cf_meta.levels[0].files.size() - 1].name);
+  ASSERT_OK(dbfull()->CompactFiles(
+      CompactionOptions(), handles_[1],
@@ -8725 +8900,2 @@ TEST_F(DBTest, GetThreadStatus) {
-      ASSERT_EQ(thread_type_counts[ThreadStatus::HIGH_PRIORITY] +
+      ASSERT_EQ(
+          thread_type_counts[ThreadStatus::HIGH_PRIORITY] +
@@ -8728 +8904,2 @@ TEST_F(DBTest, GetThreadStatus) {
-      ASSERT_EQ(thread_type_counts[ThreadStatus::HIGH_PRIORITY],
+      ASSERT_EQ(
+          thread_type_counts[ThreadStatus::HIGH_PRIORITY],
@@ -8730 +8907,2 @@ TEST_F(DBTest, GetThreadStatus) {
-      ASSERT_EQ(thread_type_counts[ThreadStatus::LOW_PRIORITY],
+      ASSERT_EQ(
+          thread_type_counts[ThreadStatus::LOW_PRIORITY],
@@ -8735,2 +8913,2 @@ TEST_F(DBTest, GetThreadStatus) {
-      env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(handles_,
-                                                                     true);
+      env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(
+          handles_, true);
@@ -8742,2 +8920,2 @@ TEST_F(DBTest, GetThreadStatus) {
-  env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(handles_,
-                                                                 true);
+  env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(
+      handles_, true);
@@ -8745,2 +8923,2 @@ TEST_F(DBTest, GetThreadStatus) {
-  env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(handles_,
-                                                                 true);
+  env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(
+      handles_, true);
@@ -8754,2 +8932,2 @@ TEST_F(DBTest, DisableThreadStatus) {
-  env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(handles_,
-                                                                 false);
+  env_->GetThreadStatusUpdater()->TEST_VerifyColumnFamilyInfoMap(
+      handles_, false);
@@ -8805 +8982,0 @@ TEST_F(DBTest, ThreadStatusSingleCompaction) {
-  rocksdb::SyncPoint::GetInstance()->EnableProcessing();
@@ -8807,0 +8985,2 @@ TEST_F(DBTest, ThreadStatusSingleCompaction) {
+    rocksdb::SyncPoint::GetInstance()->ClearTrace();
+    rocksdb::SyncPoint::GetInstance()->EnableProcessing();
@@ -8827 +9005,0 @@ TEST_F(DBTest, ThreadStatusSingleCompaction) {
-  }
@@ -8829,0 +9008 @@ TEST_F(DBTest, ThreadStatusSingleCompaction) {
+}
@@ -9399,2 +9578,5 @@ class OnFileDeletionListener : public EventListener {
-  OnFileDeletionListener() : matched_count_(0), expected_file_name_("") {}
-  void SetExpectedFileName(const std::string file_name) {
+  OnFileDeletionListener() :
+      matched_count_(0),
+      expected_file_name_("") {}
+  void SetExpectedFileName(
+      const std::string file_name) {
@@ -9406 +9588,2 @@ class OnFileDeletionListener : public EventListener {
-  void OnTableFileDeleted(const TableFileDeletionInfo& info) override {
+  void OnTableFileDeleted(
+      const TableFileDeletionInfo& info) override {
@@ -9526 +9709 @@ TEST_F(DBTest, DynamicLevelCompressionPerLevel2) {
-          ASSERT_TRUE(compaction->OutputCompressionType() == kLZ4Compression);
+          ASSERT_TRUE(compaction->output_compression() == kLZ4Compression);
@@ -9557 +9740 @@ TEST_F(DBTest, DynamicLevelCompressionPerLevel2) {
-          ASSERT_TRUE(compaction->OutputCompressionType() == kZlibCompression);
+          ASSERT_TRUE(compaction->output_compression() == kZlibCompression);
@@ -9560 +9743 @@ TEST_F(DBTest, DynamicLevelCompressionPerLevel2) {
-          ASSERT_TRUE(compaction->OutputCompressionType() == kLZ4Compression);
+          ASSERT_TRUE(compaction->output_compression() == kLZ4Compression);
@@ -9633,2 +9816,4 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_OK(dbfull()->SetOptions({{"level0_file_num_compaction_trigger", "2"},
-                                  {"target_file_size_base", ToString(k32KB)}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"level0_file_num_compaction_trigger", "2"},
+    {"target_file_size_base", ToString(k32KB) }
+  }));
@@ -9647,2 +9832,3 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_OK(
-      dbfull()->SetOptions({{"max_bytes_for_level_base", ToString(k1MB)}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"max_bytes_for_level_base", ToString(k1MB) }
+  }));
@@ -9657,3 +9843,4 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_OK(
-      dbfull()->SetOptions({{"max_bytes_for_level_multiplier", "2"},
-                            {"max_bytes_for_level_base", ToString(k128KB)}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"max_bytes_for_level_multiplier", "2"},
+    {"max_bytes_for_level_base", ToString(k128KB) }
+  }));
@@ -9664 +9851,2 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  uint64_t total_size = SizeAtLevel(1) + SizeAtLevel(2) + SizeAtLevel(3);
+  uint64_t total_size =
+    SizeAtLevel(1) + SizeAtLevel(2) + SizeAtLevel(3);
@@ -9682 +9870,3 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_OK(dbfull()->SetOptions({{"level0_stop_writes_trigger", "6"}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"level0_stop_writes_trigger", "6"}
+  }));
@@ -9696 +9886,3 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_OK(dbfull()->SetOptions({{"disable_auto_compactions", "true"}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"disable_auto_compactions", "true"}
+  }));
@@ -9705 +9897,3 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_OK(dbfull()->SetOptions({{"disable_auto_compactions", "false"}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"disable_auto_compactions", "false"}
+  }));
@@ -9726,2 +9920,5 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_TRUE(Put("max_mem_compaction_level_key", RandomString(&rnd, 8)).ok());
-  ASSERT_OK(dbfull()->SetOptions({{"max_mem_compaction_level", "1"}}));
+  ASSERT_TRUE(Put("max_mem_compaction_level_key",
+              RandomString(&rnd, 8)).ok());
+  ASSERT_OK(dbfull()->SetOptions({
+    {"max_mem_compaction_level", "1"}
+  }));
@@ -9732,2 +9929,5 @@ TEST_F(DBTest, DynamicCompactionOptions) {
-  ASSERT_TRUE(Put("max_mem_compaction_level_key", RandomString(&rnd, 8)).ok());
-  ASSERT_OK(dbfull()->SetOptions({{"max_mem_compaction_level", "0"}}));
+  ASSERT_TRUE(Put("max_mem_compaction_level_key",
+              RandomString(&rnd, 8)).ok());
+  ASSERT_OK(dbfull()->SetOptions({
+    {"max_mem_compaction_level", "0"}
+  }));
@@ -9794 +9994,2 @@ TEST_F(DBTest, PartialCompactionFailure) {
-      options.write_buffer_size * (options.max_write_buffer_number - 1);
+      options.write_buffer_size *
+      (options.max_write_buffer_number - 1);
@@ -9808 +10009,2 @@ TEST_F(DBTest, PartialCompactionFailure) {
-  const int kNumInsertedKeys = options.level0_file_num_compaction_trigger *
+  const int kNumInsertedKeys =
+      options.level0_file_num_compaction_trigger *
@@ -9867 +10069,3 @@ TEST_F(DBTest, DynamicMiscOptions) {
-  ASSERT_OK(dbfull()->SetOptions({{"max_sequential_skip_in_iterations", "4"}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"max_sequential_skip_in_iterations", "4"}
+  }));
@@ -9870,2 +10074,3 @@ TEST_F(DBTest, DynamicMiscOptions) {
-  ASSERT_OK(
-      dbfull()->SetOptions({{"max_sequential_skip_in_iterations", "16"}}));
+  ASSERT_OK(dbfull()->SetOptions({
+    {"max_sequential_skip_in_iterations", "16"}
+  }));
@@ -9885,0 +10091 @@ TEST_F(DBTest, DontDeletePendingOutputs) {
+    job_context.Clean();
@@ -10074,2 +10280,2 @@ TEST_F(DBTest, MutexWaitStats) {
-  ThreadStatusUtil::TEST_SetStateDelay(ThreadStatus::STATE_MUTEX_WAIT,
-                                       kMutexWaitDelay);
+  ThreadStatusUtil::TEST_SetStateDelay(
+      ThreadStatus::STATE_MUTEX_WAIT, kMutexWaitDelay);
@@ -10077,2 +10283,4 @@ TEST_F(DBTest, MutexWaitStats) {
-  ASSERT_GE(TestGetTickerCount(options, DB_MUTEX_WAIT_MICROS), kMutexWaitDelay);
-  ThreadStatusUtil::TEST_SetStateDelay(ThreadStatus::STATE_MUTEX_WAIT, 0);
+  ASSERT_GE(TestGetTickerCount(
+            options, DB_MUTEX_WAIT_MICROS), kMutexWaitDelay);
+  ThreadStatusUtil::TEST_SetStateDelay(
+      ThreadStatus::STATE_MUTEX_WAIT, 0);
@@ -10473,2 +10681 @@ TEST_F(DBTest, SuggestCompactRangeTest) {
-                 compaction_filter_factory)
-          ->saved_context.is_manual_compaction;
+                 compaction_filter_factory)->saved_context.is_manual_compaction;
@@ -10967 +11174 @@ TEST_F(DBTest, RowCache) {
-TEST_F(DBTest, PrevAfterMerge) {
+TEST_F(DBTest, DISABLED_PrevAfterMerge) {
@@ -10983,0 +11191,28 @@ TEST_F(DBTest, PrevAfterMerge) {
+TEST_F(DBTest, DeletingOldWalAfterDrop) {
+  rocksdb::SyncPoint::GetInstance()->LoadDependency(
+      { { "Test:AllowFlushes", "DBImpl::BGWorkFlush" },
+        { "DBImpl::BGWorkFlush:done", "Test:WaitForFlush"} });
+  rocksdb::SyncPoint::GetInstance()->ClearTrace();
+  rocksdb::SyncPoint::GetInstance()->DisableProcessing();
+  Options options = CurrentOptions();
+  options.max_total_wal_size = 8192;
+  options.compression = kNoCompression;
+  options.write_buffer_size = 1 << 20;
+  options.level0_file_num_compaction_trigger = (1<<30);
+  options.level0_slowdown_writes_trigger = (1<<30);
+  options.level0_stop_writes_trigger = (1<<30);
+  options.disable_auto_compactions = true;
+  DestroyAndReopen(options);
+  rocksdb::SyncPoint::GetInstance()->EnableProcessing();
+  CreateColumnFamilies({"cf1", "cf2"}, options);
+  ASSERT_OK(Put(0, "key1", DummyString(8192)));
+  ASSERT_OK(Put(0, "key2", DummyString(8192)));
+  ASSERT_OK(db_->DropColumnFamily(handles_[0]));
+  TEST_SYNC_POINT("Test:AllowFlushes");
+  TEST_SYNC_POINT("Test:WaitForFlush");
+  uint64_t lognum1 = dbfull()->TEST_LogfileNumber();
+  ASSERT_OK(Put(1, "key3", DummyString(8192)));
+  ASSERT_OK(Put(1, "key4", DummyString(8192)));
+  uint64_t lognum2 = dbfull()->TEST_LogfileNumber();
+  EXPECT_GT(lognum2, lognum1);
+}
@@ -10986,51 +11220,0 @@ TEST_F(DBTest, PrevAfterMerge) {
-#include <algorithm>
-#include <iostream>
-#include <set>
-#include <unistd.h>
-#include <thread>
-#include <unordered_set>
-#include <utility>
-#include <fcntl.h>
-#include "db/filename.h"
-#include "db/dbformat.h"
-#include "db/db_impl.h"
-#include "db/job_context.h"
-#include "db/version_set.h"
-#include "db/write_batch_internal.h"
-#include "port/stack_trace.h"
-#include "rocksdb/cache.h"
-#include "rocksdb/compaction_filter.h"
-#include "rocksdb/db.h"
-#include "rocksdb/env.h"
-#include "rocksdb/experimental.h"
-#include "rocksdb/filter_policy.h"
-#include "rocksdb/perf_context.h"
-#include "rocksdb/slice.h"
-#include "rocksdb/slice_transform.h"
-#include "rocksdb/table.h"
-#include "rocksdb/options.h"
-#include "rocksdb/table_properties.h"
-#include "rocksdb/thread_status.h"
-#include "rocksdb/utilities/write_batch_with_index.h"
-#include "rocksdb/utilities/checkpoint.h"
-#include "rocksdb/utilities/convenience.h"
-#include "rocksdb/utilities/optimistic_transaction_db.h"
-#include "table/block_based_table_factory.h"
-#include "table/mock_table.h"
-#include "table/plain_table_factory.h"
-#include "util/hash.h"
-#include "util/hash_linklist_rep.h"
-#include "utilities/merge_operators.h"
-#include "util/logging.h"
-#include "util/compression.h"
-#include "util/mutexlock.h"
-#include "util/rate_limiter.h"
-#include "util/statistics.h"
-#include "util/testharness.h"
-#include "util/scoped_arena_iterator.h"
-#include "util/sync_point.h"
-#include "util/testutil.h"
-#include "util/mock_env.h"
-#include "util/string_util.h"
-#include "util/thread_status_util.h"
-#include "util/xfunc.h"
