diff --git a/./output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_conflict.cc b/./output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_merged.cc
index b8cfe4c..bdc30b1 100644
--- a/./output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_conflict.cc
+++ b/./output/rocksdb/5c791323-4bed00a4-e70115e7/db@db_test.no_comments_merged.cc
@@ -6765,0 +6766 @@ TEST_F(DBTest, TransactionLogIterator) {
+#ifndef NDEBUG
@@ -6808,0 +6810 @@ TEST_F(DBTest, TransactionLogIteratorRace) {
+#endif
@@ -6928,0 +6931,3 @@ class RecoveryTestHelper {
+#ifdef OS_WIN
+    test->Close();
+#endif
@@ -6946,35 +6950,0 @@ class RecoveryTestHelper {
-<<<<<<< HEAD
-  static void CorruptWAL(DBTest* test, const double off, const double len,
-                         const bool trunc = false) {
-    rocksdb::VectorLogPtr wal_files;
-    ASSERT_OK(test->dbfull()->GetSortedWalFiles(wal_files));
-    ASSERT_EQ(wal_files.size(), 1);
-    const auto logfile_path =
-        test->dbname_ + "/" + wal_files.front()->PathName();
-    auto size = wal_files.front()->SizeFileBytes();
-#ifdef OS_WIN
-    test->Close();
-#endif
-    if (trunc) {
-      ASSERT_EQ(0, truncate(logfile_path.c_str(), size * off));
-    } else {
-      InduceCorruption(logfile_path, size * off, size * len);
-    }
-  }
-||||||| e70115e71
-  static void CorruptWAL(DBTest* test, const double off, const double len,
-                         const bool trunc = false) {
-    rocksdb::VectorLogPtr wal_files;
-    ASSERT_OK(test->dbfull()->GetSortedWalFiles(wal_files));
-    ASSERT_EQ(wal_files.size(), 1);
-    const auto logfile_path =
-        test->dbname_ + "/" + wal_files.front()->PathName();
-    auto size = wal_files.front()->SizeFileBytes();
-    if (trunc) {
-      ASSERT_EQ(0, truncate(logfile_path.c_str(), size * off));
-    } else {
-      InduceCorruption(logfile_path, size * off, size * len);
-    }
-  }
-=======
->>>>>>> 4bed00a4
