diff --git a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_mergegen.h b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_merged.h
index ef68529..532e251 100644
--- a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_mergegen.h
+++ b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_merged.h
@@ -2 +2,4 @@
-#include <vector> #include <binder/Parcel.h> #include <binder/Parcelable.h> #include <media/AudioContainers.h>
+#include <vector>
+#include <binder/Parcel.h>
+#include <binder/Parcelable.h>
+#include <media/AudioContainers.h>
