diff --git a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_conflict.h b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_merged.h
index d262691..532e251 100644
--- a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_conflict.h
+++ b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_merged.h
@@ -2 +1,0 @@
-<<<<<<< HEAD
@@ -7,5 +5,0 @@
-||||||| 3944a1b7f7
-=======
-#include <binder/Parcel.h>
-#include <binder/Parcelable.h>
->>>>>>> cb0fcde9
