diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_merged.h
index 61d7d9a..532e251 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_merged.h
@@ -13,3 +13,2 @@ namespace android {
-class DeviceDescriptorBase : public AudioPort, public AudioPortConfig {
- protected:
-  AudioDeviceTypeAddr mDeviceTypeAddr;
+class DeviceDescriptorBase : public AudioPort, public AudioPortConfig
+{
@@ -23,6 +22,2 @@ class DeviceDescriptorBase : public AudioPort, public AudioPortConfig {
-  void setAddress(const std::string& address) {
-    mDeviceTypeAddr.mAddress = address;
-  }
-  const AudioDeviceTypeAddr& getDeviceTypeAddr() const {
-    return mDeviceTypeAddr;
-  }
+    void setAddress(const std::string &address) { mDeviceTypeAddr.mAddress = address; }
+    const AudioDeviceTypeAddr& getDeviceTypeAddr() const { return mDeviceTypeAddr; }
@@ -32,2 +27 @@ class DeviceDescriptorBase : public AudioPort, public AudioPortConfig {
-  virtual void toAudioPortConfig(
-      struct audio_port_config* dstConfig,
+    virtual void toAudioPortConfig(struct audio_port_config *dstConfig,
@@ -42,0 +37,2 @@ class DeviceDescriptorBase : public AudioPort, public AudioPortConfig {
+protected:
+    AudioDeviceTypeAddr mDeviceTypeAddr;
@@ -46,2 +42 @@ std::string toString(const DeviceDescriptorBaseVector& devices);
-AudioDeviceTypeAddrVector deviceTypeAddrsFromDescriptors(
-    const DeviceDescriptorBaseVector& devices);
+AudioDeviceTypeAddrVector deviceTypeAddrsFromDescriptors(const DeviceDescriptorBaseVector& devices);
