--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/media@libaudiofoundation@include@media@DeviceDescriptorBase.no_comments_merged.h
@@ -13,3 +13,2 @@ namespace android {
-class DeviceDescriptorBase : public AudioPort, public AudioPortConfig {
- protected:
-  AudioDeviceTypeAddr mDeviceTypeAddr;
+class DeviceDescriptorBase : public AudioPort, public AudioPortConfig
+{
@@ -42,0 +37,2 @@ class DeviceDescriptorBase : public AudioPort, public AudioPortConfig {
+protected:
+    AudioDeviceTypeAddr mDeviceTypeAddr;
