diff --git a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_mergegen.cpp b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_merged.cpp
index a2f0edf..5cc14b5 100644
--- a/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_mergegen.cpp
+++ b/./output/frameworks_av/85ee9083-cb0fcde9-3944a1b7/services@audiopolicy@common@managerdefinitions@src@AudioOutputDescriptor.no_comments_merged.cpp
@@ -383 +383,5 @@ status_t SwAudioOutputDescriptor::open(const audio_config_t *config,
-LOG_ALWAYS_FATAL_IF(device == nullptr, "%s failed to get device DeviceTypeSet deviceTypes = devices, all device types: %s", __func__, dumpDeviceTypes(devices.types()).c_str());
+    sp<DeviceDescriptor> device = devices.getDeviceForOpening();
+    LOG_ALWAYS_FATAL_IF(device == nullptr,
+                        "%s failed to get device descriptor for opening "
+                        "with the requested devices, all device types: %s",
+                        __func__, dumpDeviceTypes(devices.types()).c_str());
@@ -408 +411,0 @@ LOG_ALWAYS_FATAL_IF(device == nullptr, "%s failed to get device DeviceTypeSet de
-    audio_devices_t deviceType = deviceTypesToBitMask(deviceTypes);
@@ -412 +415 @@ LOG_ALWAYS_FATAL_IF(device == nullptr, "%s failed to get device DeviceTypeSet de
-(mLatency, mFlags);
+                                                   device,
@@ -417 +420,4 @@ LOG_ALWAYS_FATAL_IF(device == nullptr, "%s failed to get device DeviceTypeSet de
-"selected device %s, " "selected device %s for opening", __FUNCTION__, *output, devices.toString().c_str(), device->toString().c_str());
+                            "%s openOutput returned output handle %d for device %s, "
+                            "selected device %s for opening",
+                            __FUNCTION__, *output, devices.toString().c_str(),
+                            device->toString().c_str());
