--- a/home/whalien/codebase/python/mergebot-eva/output/art/645fa0ca-bf517232-d5a49147/runtime@runtime.no_comments_mergebot.cc
+++ b/home/whalien/codebase/python/mergebot-eva/output/art/645fa0ca-bf517232-d5a49147/runtime@runtime.no_comments_merged.cc
@@ -23,2 +22,0 @@
-#include "android-base/strings.h"
-#include "aot_class_linker.h"
@@ -34 +31,0 @@
-#include "base/enums.h"
@@ -53 +49,0 @@
-#include "elf_file.h"
@@ -65 +60,0 @@
-#include "image-inl.h"
@@ -129,3 +123,0 @@
-#include "oat.h"
-#include "oat_file_manager.h"
-#include "oat_quick_method_header.h"
@@ -150 +141,0 @@
-#include "transaction.h"
@@ -179 +170 @@ namespace apex = com::android::apex;
-namespace {
+namespace art HIDDEN {
@@ -301 +290 @@ Runtime::Runtime()
-~Runtime() {
+Runtime::~Runtime() {
@@ -706 +698 @@ bool Runtime::ParseOptions(const RuntimeOptions& raw_options,
-NO_THREAD_SAFETY_ANALYSIS() {
+static bool IsSafeToCallAbort() NO_THREAD_SAFETY_ANALYSIS {
@@ -2291 +2308,5 @@ bool Runtime::IsVerificationEnabled() const {
-  return verify_ == verifier::VerifyMode::kEnable || verify_ == verifier::VerifyMode::kSoftFail;
+  return verify_ == verifier::VerifyMode::kEnable ||
+      verify_ == verifier::VerifyMode::kSoftFail;
+}
+bool Runtime::IsVerificationSoftFail() const {
+  return verify_ == verifier::VerifyMode::kSoftFail;
@@ -2293 +2313,0 @@ bool Runtime::IsVerificationEnabled() const {
-bool Runtime::IsVerificationSoftFail() const { return verify_ == verifier::VerifyMode::kSoftFail; }
@@ -2431 +2461,3 @@ Runtime::ScopedThreadPoolUsage::ScopedThreadPoolUsage()
-~ScopedThreadPoolUsage() { Runtime::Current()->ReleaseThreadPool(); }
+Runtime::ScopedThreadPoolUsage::~ScopedThreadPoolUsage() {
+  Runtime::Current()->ReleaseThreadPool();
+}
