--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/media@libaudioclient@include@media@IAudioPolicyService.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/media@libaudioclient@include@media@IAudioPolicyService.no_comments_merged.h
@@ -179 +186 @@ class BnAudioPolicyService : public BnInterface<IAudioPolicyService> {
-}
+};
