diff --git a/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/media@libaudioclient@include@media@IAudioPolicyService.no_comments_mergegen.h b/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/media@libaudioclient@include@media@IAudioPolicyService.no_comments_merged.h
index 4d7759e..fd00316 100644
--- a/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/media@libaudioclient@include@media@IAudioPolicyService.no_comments_mergegen.h
+++ b/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/media@libaudioclient@include@media@IAudioPolicyService.no_comments_merged.h
@@ -167 +167,6 @@ public:
-virtual bool isCallScreenModeSupported() = 0; virtual status_t setPreferredDeviceForStrategy(product_strategy_t strategy, const AudioDeviceTypeAddr &device) = 0; virtual status_t removePreferredDeviceForStrategy(product_strategy_t strategy) = 0; virtual status_t getPreferredDeviceForStrategy(product_strategy_t strategy, AudioDeviceTypeAddr &device) = 0;
+    virtual bool isCallScreenModeSupported() = 0;
+    virtual status_t setPreferredDeviceForStrategy(product_strategy_t strategy,
+                                                   const AudioDeviceTypeAddr &device) = 0;
+    virtual status_t removePreferredDeviceForStrategy(product_strategy_t strategy) = 0;
+    virtual status_t getPreferredDeviceForStrategy(product_strategy_t strategy,
+                                                   AudioDeviceTypeAddr &device) = 0;
