--- a/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
@@ -154,3 +170 @@ status_t AudioPolicyService::getOutputForAttr(
-          ALOGE(
-              "%s() permission denied: modify phone state not allowed for uid "
-              "%d",
+                ALOGE("%s() permission denied: modify phone state not allowed for uid %d",
@@ -163,3 +177 @@ status_t AudioPolicyService::getOutputForAttr(
-          ALOGE(
-              "%s() permission denied: modify audio routing not allowed for "
-              "uid %d",
+                ALOGE("%s() permission denied: modify audio routing not allowed for uid %d",
@@ -343,3 +367 @@ status_t AudioPolicyService::getInputForAttr(
-            ALOGE(
-                "getInputForAttr() permission denied: modify audio routing not "
-                "allowed");
+                    ALOGE("getInputForAttr() permission denied: modify audio routing not allowed");
