diff --git a/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/services@audiopolicy@AudioPolicyInterface.no_comments_conflict.h b/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/services@audiopolicy@AudioPolicyInterface.no_comments_merged.h
index 46afb95..2278f5a 100644
--- a/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/services@audiopolicy@AudioPolicyInterface.no_comments_conflict.h
+++ b/./output/frameworks_av/6a87ed87-0b39d3eb-fb33f9bc/services@audiopolicy@AudioPolicyInterface.no_comments_merged.h
@@ -160 +159,0 @@ public:
-<<<<<<< HEAD
@@ -167,8 +165,0 @@ public:
-||||||| fb33f9bc85
-=======
-    virtual status_t setPreferredDeviceForStrategy(product_strategy_t strategy,
-                                                   const AudioDeviceTypeAddr &device) = 0;
-    virtual status_t removePreferredDeviceForStrategy(product_strategy_t strategy) = 0;
-    virtual status_t getPreferredDeviceForStrategy(product_strategy_t strategy,
-                                                   AudioDeviceTypeAddr &device) = 0;
->>>>>>> 0b39d3eb
