--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/82096a63-0e43d4b3-875192b4/src@widgets@capture@capturewidget.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/82096a63-0e43d4b3-875192b4/src@widgets@capture@capturewidget.no_comments_merged.cpp
@@ -689,0 +690,7 @@ void CaptureWidget::moveEvent(QMoveEvent* e)
+void CaptureWidget::changeEvent(QEvent* e)
+{
+    if (e->type() == QEvent::ActivationChange) {
+        QPoint bottomRight = rect().bottomRight();
+        update(QRect(bottomRight - QPoint(1000, 200), bottomRight));
+    }
+}
@@ -707,7 +713,0 @@ void CaptureWidget::initContext(bool fullscreen, uint requestId)
-void CaptureWidget::changeEvent(QEvent* e)
-{
-    if (e->type() == QEvent::ActivationChange) {
-        QPoint bottomRight = rect().bottomRight();
-        update(QRect(bottomRight - QPoint(1000, 200), bottomRight));
-    }
-}
