diff --git a/./output/tmux/147b5ae5-44657bf9-9952201c/client.no_comments_mergegen.c b/./output/tmux/147b5ae5-44657bf9-9952201c/client.no_comments_merged.c
index 7635d1d..db13f26 100644
--- a/./output/tmux/147b5ae5-44657bf9-9952201c/client.no_comments_mergegen.c
+++ b/./output/tmux/147b5ae5-44657bf9-9952201c/client.no_comments_merged.c
@@ -10 +9,0 @@
-#include <imsg.h>
@@ -217 +215,0 @@ client_main(struct event_base *base, int argc, char **argv, int flags)
-                                                                 #ifdef HAVE_SETPROCTITLE setproctitle("client (%s)", socket_path); #endif logfile("client");
