--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/147b5ae5-44657bf9-9952201c/server.no_comments_merged.c
@@ -10 +9,0 @@
-#include <paths.h>
@@ -118,9 +137,0 @@ int server_start(struct event_base *base, int lockfd, char *lockfile) {
-<<<<<<< HEAD
-  log_debug("socket path %s", socket_path);
-#ifdef HAVE_SETPROCTITLE
-  setproctitle("server (%s)", socket_path);
-#endif
-  ||||||| log_debug("socket path %s", socket_path);
-  setproctitle("server (%s)", socket_path);
-=======
->>>>>>> 44657bf932b068aff5ce1019a4e8a2e7b00b5321
