--- a/./output/art/38145f10-cf589753-20e64859/runtime@gc@heap.no_comments_conflict.cc
+++ b/./output/art/38145f10-cf589753-20e64859/runtime@gc@heap.no_comments_merged.cc
@@ -1274 +1273,0 @@ void Heap::DoPendingCollectorTransition() {
-<<<<<<< HEAD
@@ -1277,8 +1275,0 @@ void Heap::DoPendingCollectorTransition() {
-||||||| 20e648599b
-  } else if (desired_collector_type == kCollectorTypeCCBackground) {
-    DCHECK(gUseReadBarrier);
-=======
-  } else if (desired_collector_type == kCollectorTypeCCBackground ||
-             desired_collector_type == kCollectorTypeCMC) {
-    DCHECK(gUseReadBarrier);
->>>>>>> cf589753
