diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/c67d48c8-be7e273d-e6b3e3b4/include@rocksdb@db.no_comments_mergebot.h b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/c67d48c8-be7e273d-e6b3e3b4/include@rocksdb@db.no_comments_merged.h
index 1e3ca7a..3d27313 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/c67d48c8-be7e273d-e6b3e3b4/include@rocksdb@db.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/c67d48c8-be7e273d-e6b3e3b4/include@rocksdb@db.no_comments_merged.h
@@ -7 +6,0 @@
-#include <unordered_map>
@@ -8,0 +8 @@
+#include <unordered_map>
@@ -17 +17 @@ class ColumnFamilyHandle {
-  ~ColumnFamilyHandle() {}
+  virtual ~ColumnFamilyHandle() {}
@@ -61 +61,2 @@ class DB {
-  static Status Open(const Options& options, const std::string& name,
+  static Status Open(const Options& options,
+                     const std::string& name,
@@ -63,2 +64,2 @@ class DB {
-  static Status OpenForReadOnly(const Options& options, const std::string& name,
-                                DB** dbptr,
+  static Status OpenForReadOnly(const Options& options,
+      const std::string& name, DB** dbptr,
@@ -112,3 +113,2 @@ class DB {
-    return MultiGet(
-        options,
-        std::vector<ColumnFamilyHandle*>(keys.size(), DefaultColumnFamily()),
+    return MultiGet(options, std::vector<ColumnFamilyHandle*>(
+                                 keys.size(), DefaultColumnFamily()),
@@ -149 +148,0 @@ class DB {
-    GetApproximateSizes(DefaultColumnFamily(), range, n, sizes) {
@@ -152,3 +151,4 @@ class DB {
-    virtual Status CompactRange(
-        ColumnFamilyHandle * column_family, const Slice* begin,
-        const Slice* end, bool reduce_level = false, int target_level = -1) = 0;
+  virtual Status CompactRange(ColumnFamilyHandle* column_family,
+                              const Slice* begin, const Slice* end,
+                              bool reduce_level = false,
+                              int target_level = -1) = 0;
@@ -161,4 +161 @@ class DB {
-    int NumberLevels() {
-      return NumberLevels(DefaultColumnFamily()) {
-        return NumberLevels(DefaultColumnFamily());
-      }
+  int NumberLevels() { return NumberLevels(DefaultColumnFamily()); }
@@ -169,2 +166 @@ class DB {
-      virtual int Level0StopWriteTrigger(ColumnFamilyHandle *
-                                         column_family) = 0;
+  virtual int Level0StopWriteTrigger(ColumnFamilyHandle* column_family) = 0;
@@ -193,2 +189 @@ class DB {
-      virtual Status GetUpdatesSince(
-          SequenceNumber seq_number,
+  virtual Status GetUpdatesSince(SequenceNumber seq_number,
@@ -197,2 +192 @@ class DB {
-      virtual void GetLiveFilesMetaData(std::vector<LiveFileMetaData> *
-                                        metadata) {}
+  virtual void GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata) {}
@@ -201,2 +195 @@ class DB {
-      virtual Status GetPropertiesOfAllTables(TablePropertiesCollection *
-                                              props) = 0;
+  virtual Status GetPropertiesOfAllTables(TablePropertiesCollection* props) = 0;
