--- a/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/c67d48c8-be7e273d-e6b3e3b4/include@rocksdb@db.no_comments_mergebot.h
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/rocksdb/c67d48c8-be7e273d-e6b3e3b4/include@rocksdb@db.no_comments_merged.h
@@ -7 +6,0 @@
-#include <unordered_map>
@@ -8,0 +8 @@
+#include <unordered_map>
@@ -17 +17 @@ class ColumnFamilyHandle {
-  ~ColumnFamilyHandle() {}
+  virtual ~ColumnFamilyHandle() {}
@@ -149 +148,0 @@ class DB {
-    GetApproximateSizes(DefaultColumnFamily(), range, n, sizes) {
@@ -161,4 +161 @@ class DB {
-    int NumberLevels() {
-      return NumberLevels(DefaultColumnFamily()) {
-        return NumberLevels(DefaultColumnFamily());
-      }
+  int NumberLevels() { return NumberLevels(DefaultColumnFamily()); }
