--- a/home/whalien/codebase/python/mergebot-eva/output/system_core/53c122bb-0d4f0508-698e065e/adb@commandline.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/python/mergebot-eva/output/system_core/53c122bb-0d4f0508-698e065e/adb@commandline.no_comments_merged.cpp
@@ -33,2 +32,0 @@
-#include "bugreport.h"
-#include "commandline.h"
@@ -53,2 +45,0 @@ static constexpr char BUGZ_FAIL_PREFIX[] = "FAIL:";
-DefaultStandardStreamsCallback DEFAULT_STANDARD_STREAMS_CALLBACK(nullptr,
-                                                                 nullptr);
@@ -67,16 +57,8 @@ static void help() {
-  fprintf(
-      stderr,
-      " -a                            - directs adb to listen on all "
-      "interfaces for a connection\n"
-      " -d                            - directs command to the only connected "
-      "USB device\n"
-      "                                 returns an error if more than one USB "
-      "device is present.\n"
-      " -e                            - directs command to the only running "
-      "emulator.\n"
-      "                                 returns an error if more than one "
-      "emulator is running.\n"
-      " -s <specific device>          - directs command to the device or "
-      "emulator with the given\n"
-      "                                 serial number or qualifier. Overrides "
-      "ANDROID_SERIAL\n"
+    fprintf(stderr,
+        " -a                            - directs adb to listen on all interfaces for a connection\n"
+        " -d                            - directs command to the only connected USB device\n"
+        "                                 returns an error if more than one USB device is present.\n"
+        " -e                            - directs command to the only running emulator.\n"
+        "                                 returns an error if more than one emulator is running.\n"
+        " -s <specific device>          - directs command to the device or emulator with the given\n"
+        "                                 serial number or qualifier. Overrides ANDROID_SERIAL\n"
@@ -86,6 +68,3 @@ static void help() {
-      "                                 out directory like "
-      "'out/target/product/sooner'.\n"
-      "                                 If -p is not specified, the "
-      "ANDROID_PRODUCT_OUT\n"
-      "                                 environment variable is used, which "
-      "must\n"
+        "                                 out directory like 'out/target/product/sooner'.\n"
+        "                                 If -p is not specified, the ANDROID_PRODUCT_OUT\n"
+        "                                 environment variable is used, which must\n"
@@ -93,2 +72 @@ static void help() {
-      " -H                            - Name of adb server host (default: "
-      "localhost)\n"
+        " -H                            - Name of adb server host (default: localhost)\n"
@@ -97,2 +75 @@ static void help() {
-      "                                 ('-l' will also list device "
-      "qualifiers)\n"
+        "                                 ('-l' will also list device qualifiers)\n"
@@ -100,2 +77 @@ static void help() {
-      "                                 Port 5555 is used by default if no "
-      "port number is specified.\n"
+        "                                 Port 5555 is used by default if no port number is specified.\n"
@@ -103,6 +79,3 @@ static void help() {
-      "                                 Port 5555 is used by default if no "
-      "port number is specified.\n"
-      "                                 Using this command with no additional "
-      "arguments\n"
-      "                                 will disconnect from all connected "
-      "TCP/IP devices.\n"
+        "                                 Port 5555 is used by default if no port number is specified.\n"
+        "                                 Using this command with no additional arguments\n"
+        "                                 will disconnect from all connected TCP/IP devices.\n"
@@ -115,2 +88 @@ static void help() {
-      "                                 (-a preserves file timestamp and "
-      "mode)\n"
+        "                                 (-a preserves file timestamp and mode)\n"
@@ -120,4 +92,2 @@ static void help() {
-      "                               - run remote shell command (interactive "
-      "shell if no command given)\n"
-      "                                 (-e: choose escape character, or "
-      "\"none\"; default '~')\n"
+        "                               - run remote shell command (interactive shell if no command given)\n"
+        "                                 (-e: choose escape character, or \"none\"; default '~')\n"
@@ -127,2 +97 @@ static void help() {
-      "                                 (-x: disable remote exit codes and "
-      "stdout/stderr separation)\n"
+        "                                 (-x: disable remote exit codes and stdout/stderr separation)\n"
@@ -132,4 +101,2 @@ static void help() {
-      "                                 the format is a list of lines with the "
-      "following format:\n"
-      "                                    <serial> \" \" <local> \" \" "
-      "<remote> \"\\n\"\n"
+        "                                 the format is a list of lines with the following format:\n"
+        "                                    <serial> \" \" <local> \" \" <remote> \"\\n\"\n"
@@ -138,8 +105,4 @@ static void help() {
-      "                                   tcp:<port> (<local> may be \"tcp:0\" "
-      "to pick any open port)\n"
-      "                                   localabstract:<unix domain socket "
-      "name>\n"
-      "                                   localreserved:<unix domain socket "
-      "name>\n"
-      "                                   localfilesystem:<unix domain socket "
-      "name>\n"
+        "                                   tcp:<port> (<local> may be \"tcp:0\" to pick any open port)\n"
+        "                                   localabstract:<unix domain socket name>\n"
+        "                                   localreserved:<unix domain socket name>\n"
+        "                                   localfilesystem:<unix domain socket name>\n"
@@ -149,2 +112 @@ static void help() {
-      "                               - same as 'adb forward <local> <remote>' "
-      "but fails\n"
+        "                               - same as 'adb forward <local> <remote>' but fails\n"
@@ -152,2 +114 @@ static void help() {
-      "  adb forward --remove <local> - remove a specific forward socket "
-      "connection\n"
+        "  adb forward --remove <local> - remove a specific forward socket connection\n"
@@ -155,2 +116 @@ static void help() {
-      "  adb reverse --list           - list all reverse socket connections "
-      "from device\n"
+        "  adb reverse --list           - list all reverse socket connections from device\n"
@@ -159,8 +119,4 @@ static void help() {
-      "                                   tcp:<port> (<remote> may be "
-      "\"tcp:0\" to pick any open port)\n"
-      "                                   localabstract:<unix domain socket "
-      "name>\n"
-      "                                   localreserved:<unix domain socket "
-      "name>\n"
-      "                                   localfilesystem:<unix domain socket "
-      "name>\n"
+        "                                   tcp:<port> (<remote> may be \"tcp:0\" to pick any open port)\n"
+        "                                   localabstract:<unix domain socket name>\n"
+        "                                   localreserved:<unix domain socket name>\n"
+        "                                   localfilesystem:<unix domain socket name>\n"
@@ -168,2 +124 @@ static void help() {
-      "                               - same as 'adb reverse <remote> <local>' "
-      "but fails\n"
+        "                               - same as 'adb reverse <remote> <local>' but fails\n"
@@ -172,6 +127,3 @@ static void help() {
-      "                               - remove a specific reversed socket "
-      "connection\n"
-      "  adb reverse --remove-all     - remove all reversed socket connections "
-      "from device\n"
-      "  adb jdwp                     - list PIDs of processes hosting a JDWP "
-      "transport\n"
+        "                               - remove a specific reversed socket connection\n"
+        "  adb reverse --remove-all     - remove all reversed socket connections from device\n"
+        "  adb jdwp                     - list PIDs of processes hosting a JDWP transport\n"
@@ -179,2 +131 @@ static void help() {
-      "                               - push this package file to the device "
-      "and install it\n"
+        "                               - push this package file to the device and install it\n"
@@ -185,2 +136 @@ static void help() {
-      "                                 (-d: allow version code downgrade "
-      "(debuggable packages only))\n"
+        "                                 (-d: allow version code downgrade (debuggable packages only))\n"
@@ -189,2 +139 @@ static void help() {
-      "                               - push this package file to the device "
-      "and install it\n"
+        "                               - push this package file to the device and install it\n"
@@ -195,2 +144 @@ static void help() {
-      "                                 (-d: allow version code downgrade "
-      "(debuggable packages only))\n"
+        "                                 (-d: allow version code downgrade (debuggable packages only))\n"
@@ -199,8 +147,4 @@ static void help() {
-      "  adb uninstall [-k] <package> - remove this app package from the "
-      "device\n"
-      "                                 ('-k' means keep the data and cache "
-      "directories)\n"
-      "  adb bugreport [<zip_file>]   - return all information from the "
-      "device\n"
-      "                                 that should be included in a bug "
-      "report.\n"
+        "  adb uninstall [-k] <package> - remove this app package from the device\n"
+        "                                 ('-k' means keep the data and cache directories)\n"
+        "  adb bugreport [<zip_file>]   - return all information from the device\n"
+        "                                 that should be included in a bug report.\n"
@@ -208,16 +152,8 @@ static void help() {
-      "  adb backup [-f <file>] [-apk|-noapk] [-obb|-noobb] "
-      "[-shared|-noshared] [-all] [-system|-nosystem] [<packages...>]\n"
-      "                               - write an archive of the device's data "
-      "to <file>.\n"
-      "                                 If no -f option is supplied then the "
-      "data is written\n"
-      "                                 to \"backup.ab\" in the current "
-      "directory.\n"
-      "                                 (-apk|-noapk enable/disable backup of "
-      "the .apks themselves\n"
-      "                                    in the archive; the default is "
-      "noapk.)\n"
-      "                                 (-obb|-noobb enable/disable backup of "
-      "any installed apk expansion\n"
-      "                                    (aka .obb) files associated with "
-      "each application; the default\n"
+        "  adb backup [-f <file>] [-apk|-noapk] [-obb|-noobb] [-shared|-noshared] [-all] [-system|-nosystem] [<packages...>]\n"
+        "                               - write an archive of the device's data to <file>.\n"
+        "                                 If no -f option is supplied then the data is written\n"
+        "                                 to \"backup.ab\" in the current directory.\n"
+        "                                 (-apk|-noapk enable/disable backup of the .apks themselves\n"
+        "                                    in the archive; the default is noapk.)\n"
+        "                                 (-obb|-noobb enable/disable backup of any installed apk expansion\n"
+        "                                    (aka .obb) files associated with each application; the default\n"
@@ -225,20 +161,10 @@ static void help() {
-      "                                 (-shared|-noshared enable/disable "
-      "backup of the device's\n"
-      "                                    shared storage / SD card contents; "
-      "the default is noshared.)\n"
-      "                                 (-all means to back up all installed "
-      "applications)\n"
-      "                                 (-system|-nosystem toggles whether "
-      "-all automatically includes\n"
-      "                                    system applications; the default is "
-      "to include system apps)\n"
-      "                                 (<packages...> is the list of "
-      "applications to be backed up.  If\n"
-      "                                    the -all or -shared flags are "
-      "passed, then the package\n"
-      "                                    list is optional.  Applications "
-      "explicitly given on the\n"
-      "                                    command line will be included even "
-      "if -nosystem would\n"
-      "                                    ordinarily cause them to be "
-      "omitted.)\n"
+        "                                 (-shared|-noshared enable/disable backup of the device's\n"
+        "                                    shared storage / SD card contents; the default is noshared.)\n"
+        "                                 (-all means to back up all installed applications)\n"
+        "                                 (-system|-nosystem toggles whether -all automatically includes\n"
+        "                                    system applications; the default is to include system apps)\n"
+        "                                 (<packages...> is the list of applications to be backed up.  If\n"
+        "                                    the -all or -shared flags are passed, then the package\n"
+        "                                    list is optional.  Applications explicitly given on the\n"
+        "                                    command line will be included even if -nosystem would\n"
+        "                                    ordinarily cause them to be omitted.)\n"
@@ -246,2 +172 @@ static void help() {
-      "  adb restore <file>           - restore device contents from the "
-      "<file> backup archive\n"
+        "  adb restore <file>           - restore device contents from the <file> backup archive\n"
@@ -249,8 +174,4 @@ static void help() {
-      "  adb disable-verity           - disable dm-verity checking on "
-      "USERDEBUG builds\n"
-      "  adb enable-verity            - re-enable dm-verity checking on "
-      "USERDEBUG builds\n"
-      "  adb keygen <file>            - generate adb public/private key. The "
-      "private key is stored in <file>,\n"
-      "                                 and the public key is stored in "
-      "<file>.pub. Any existing files\n"
+        "  adb disable-verity           - disable dm-verity checking on USERDEBUG builds\n"
+        "  adb enable-verity            - re-enable dm-verity checking on USERDEBUG builds\n"
+        "  adb keygen <file>            - generate adb public/private key. The private key is stored in <file>,\n"
+        "                                 and the public key is stored in <file>.pub. Any existing files\n"
@@ -263,6 +184,3 @@ static void help() {
-      "                               - wait for device to be in the given "
-      "state:\n"
-      "                                 device, recovery, sideload, or "
-      "bootloader\n"
-      "                                 Transport is: usb, local or any "
-      "[default=any]\n"
+        "                               - wait for device to be in the given state:\n"
+        "                                 device, recovery, sideload, or bootloader\n"
+        "                                 Transport is: usb, local or any [default=any]\n"
@@ -274,2 +192 @@ static void help() {
-      "  adb remount                  - remounts the /system, /vendor (if "
-      "present) and /oem (if present) partitions on the device read-write\n"
+        "  adb remount                  - remounts the /system, /vendor (if present) and /oem (if present) partitions on the device read-write\n"
@@ -277,4 +194,2 @@ static void help() {
-      "                               - reboots the device, optionally into "
-      "the bootloader or recovery program.\n"
-      "  adb reboot sideload          - reboots the device into the sideload "
-      "mode in recovery program (adb root required).\n"
+        "                               - reboots the device, optionally into the bootloader or recovery program.\n"
+        "  adb reboot sideload          - reboots the device into the sideload mode in recovery program (adb root required).\n"
@@ -282,2 +197 @@ static void help() {
-      "                               - reboots into the sideload mode, then "
-      "reboots automatically after the sideload regardless of the result.\n"
+        "                               - reboots into the sideload mode, then reboots automatically after the sideload regardless of the result.\n"
@@ -285,8 +199,4 @@ static void help() {
-      "  adb root                     - restarts the adbd daemon with root "
-      "permissions\n"
-      "  adb unroot                   - restarts the adbd daemon without root "
-      "permissions\n"
-      "  adb usb                      - restarts the adbd daemon listening on "
-      "USB\n"
-      "  adb tcpip <port>             - restarts the adbd daemon listening on "
-      "TCP on the specified port\n"
+        "  adb root                     - restarts the adbd daemon with root permissions\n"
+        "  adb unroot                   - restarts the adbd daemon without root permissions\n"
+        "  adb usb                      - restarts the adbd daemon listening on USB\n"
+        "  adb tcpip <port>             - restarts the adbd daemon listening on TCP on the specified port\n"
@@ -303,2 +213 @@ static void help() {
-      "  - If <directory> is not specified, /system, /vendor (if present), "
-      "/oem (if present) and /data partitions will be updated.\n"
+        "  - If <directory> is not specified, /system, /vendor (if present), /oem (if present) and /data partitions will be updated.\n"
@@ -306,2 +215 @@ static void help() {
-      "  - If it is \"system\", \"vendor\", \"oem\" or \"data\", only the "
-      "corresponding partition\n"
+        "  - If it is \"system\", \"vendor\", \"oem\" or \"data\", only the corresponding partition\n"
@@ -311,4 +219,2 @@ static void help() {
-      "  adb reconnect                  Kick current connection from host side "
-      "and make it reconnect.\n"
-      "  adb reconnect device           Kick current connection from device "
-      "side and make it reconnect.\n"
+        "  adb reconnect                  Kick current connection from host side and make it reconnect.\n"
+        "  adb reconnect device           Kick current connection from device side and make it reconnect.\n"
@@ -316,8 +222,5 @@ static void help() {
-      "  ADB_TRACE                    - Print debug information. A comma "
-      "separated list of the following values\n"
-      "                                 1 or all, adb, sockets, packets, rwx, "
-      "usb, sync, sysdeps, transport, jdwp\n"
-      "  ANDROID_SERIAL               - The serial number to connect to. -s "
-      "takes priority over this if given.\n"
-      "  ANDROID_LOG_TAGS             - When used with the logcat option, only "
-      "these debug tags are printed.\n");
+        "  ADB_TRACE                    - Print debug information. A comma separated list of the following values\n"
+        "                                 1 or all, adb, sockets, packets, rwx, usb, sync, sysdeps, transport, jdwp\n"
+        "  ANDROID_SERIAL               - The serial number to connect to. -s takes priority over this if given.\n"
+        "  ANDROID_LOG_TAGS             - When used with the logcat option, only these debug tags are printed.\n"
+        );
@@ -329,3 +232,20 @@ static int usage() {
-int read_and_dump(int fd, bool use_shell_protocol = false,
-                  StandardStreamsCallbackInterface* callback =
-                      &DEFAULT_STANDARD_STREAMS_CALLBACK) {
+#if defined(_WIN32)
+void stdin_raw_init();
+void stdin_raw_restore();
+#else
+static termios g_saved_terminal_state;
+static void stdin_raw_init() {
+    if (tcgetattr(STDIN_FILENO, &g_saved_terminal_state)) return;
+    termios tio;
+    if (tcgetattr(STDIN_FILENO, &tio)) return;
+    cfmakeraw(&tio);
+    tio.c_cc[VTIME] = 0;
+    tio.c_cc[VMIN] = 1;
+    tcsetattr(STDIN_FILENO, TCSAFLUSH, &tio);
+}
+static void stdin_raw_restore() {
+    tcsetattr(STDIN_FILENO, TCSAFLUSH, &g_saved_terminal_state);
+}
+#endif
+static int read_and_dump(int fd, bool use_shell_protocol=false, std::string* output=nullptr,
+                         std::string* err=nullptr) {
@@ -335,0 +256,2 @@ int read_and_dump(int fd, bool use_shell_protocol = false,
+    FILE* outfile = stdout;
+    std::string* outstring = output;
@@ -351 +272,0 @@ int read_and_dump(int fd, bool use_shell_protocol = false,
-      length = protocol->data_length();
@@ -354 +275,2 @@ int read_and_dump(int fd, bool use_shell_protocol = false,
-          callback->OnStdout(buffer_ptr, length);
+                    outfile = stdout;
+                    outstring = output;
@@ -357 +279,2 @@ int read_and_dump(int fd, bool use_shell_protocol = false,
-          callback->OnStderr(buffer_ptr, length);
+                    outfile = stderr;
+                    outstring = err;
@@ -373 +296,6 @@ int read_and_dump(int fd, bool use_shell_protocol = false,
-      callback->OnStdout(buffer_ptr, length);
+        }
+        if (outstring == nullptr) {
+            fwrite(buffer_ptr, 1, length, outfile);
+            fflush(outfile);
+        } else {
+            outstring->append(buffer_ptr, length);
@@ -376 +304 @@ int read_and_dump(int fd, bool use_shell_protocol = false,
-  return callback->Done(exit_code);
+    return exit_code;
@@ -702,2 +630 @@ static int adb_shell(int argc, const char** argv) {
-                "Remote PTY will not be allocated because stdin is not a "
-                "terminal.\n"
+                        "Remote PTY will not be allocated because stdin is not a terminal.\n"
@@ -994,6 +908,3 @@ static int bugreport(TransportType transport_type, const char* serial, int argc,
-  fprintf(stderr,
-          "Bugreport is in progress and it could take minutes to complete.\n"
-          "Please be patient and do not cancel or disconnect your device until "
-          "it completes.\n");
-  int status = send_shell_command(transport_type, serial, "bugreportz", false,
-                                  &output, nullptr);
+    fprintf(stderr, "Bugreport is in progress and it could take minutes to complete.\n"
+            "Please be patient and do not cancel or disconnect your device until it completes.\n");
+    int status = send_shell_command(transport_type, serial, "bugreportz", false, &output, nullptr);
@@ -1025,38 +931 @@ static int bugreport(TransportType transport_type, const char* serial, int argc,
-int send_shell_command(TransportType transport_type, const char* serial,
-                       const std::string& command, bool disable_shell_protocol,
-                       StandardStreamsCallbackInterface* callback) {
-  int fd;
-  bool use_shell_protocol = false;
-  while (true) {
-    bool attempt_connection = true;
-    if (!disable_shell_protocol) {
-      FeatureSet features;
-      std::string error;
-      if (adb_get_feature_set(&features, &error)) {
-        use_shell_protocol = CanUseFeature(features, kFeatureShell2);
-      } else {
-        attempt_connection = false;
-      }
-    }
-    if (attempt_connection) {
-      std::string error;
-      std::string service_string =
-          ShellServiceString(use_shell_protocol, "", command);
-      fd = adb_connect(service_string, &error);
-      if (fd >= 0) {
-        break;
-      }
-    }
-    fprintf(stderr, "- waiting for device -\n");
-    if (!wait_for_device("wait-for-device", transport_type, serial)) {
-      return 1;
-    }
-  }
-  int exit_code = read_and_dump(fd, use_shell_protocol, callback);
-  if (adb_close(fd) < 0) {
-    PLOG(ERROR) << "failure closing FD " << fd;
-  }
-  return exit_code;
-}
-static int logcat(TransportType transport, const char* serial, int argc,
-                  const char** argv) {
+static int logcat(TransportType transport, const char* serial, int argc, const char** argv) {
@@ -1266,2 +1129 @@ int adb_commandline(int argc, const char** argv) {
-              "adb: Env var ANDROID_ADB_SERVER_PORT must be a positive number "
-              "less than 65536. Got \"%s\"\n",
+                    "adb: Env var ANDROID_ADB_SERVER_PORT must be a positive number less than 65536. Got \"%s\"\n",
@@ -1344,2 +1206 @@ int adb_commandline(int argc, const char** argv) {
-                  "adb: port number must be a positive number less than 65536. "
-                  "Got \"%s\"\n",
+                            "adb: port number must be a positive number less than 65536. Got \"%s\"\n",
@@ -1351,2 +1212 @@ int adb_commandline(int argc, const char** argv) {
-                "adb: port number must be a positive number less than 65536. "
-                "Got empty string.\n");
+                "adb: port number must be a positive number less than 65536. Got empty string.\n");
@@ -1710,2 +1589 @@ static int uninstall_app(TransportType transport, const char* serial, int argc,
-          "The -k option uninstalls the application while retaining the "
-          "data/cache.\n"
+                "The -k option uninstalls the application while retaining the data/cache.\n"
@@ -1713,4 +1591,2 @@ static int uninstall_app(TransportType transport, const char* serial, int argc,
-          "You will have to reinstall the application with the same signature, "
-          "and fully uninstall it.\n"
-          "If you truly wish to continue, execute 'adb shell cmd package "
-          "uninstall -k'.\n");
+                "You will have to reinstall the application with the same signature, and fully uninstall it.\n"
+                "If you truly wish to continue, execute 'adb shell cmd package uninstall -k'.\n");
@@ -1892,2 +1764 @@ static int uninstall_app_legacy(TransportType transport, const char* serial,
-          "The -k option uninstalls the application while retaining the "
-          "data/cache.\n"
+                "The -k option uninstalls the application while retaining the data/cache.\n"
@@ -1895,4 +1766,2 @@ static int uninstall_app_legacy(TransportType transport, const char* serial,
-          "You will have to reinstall the application with the same signature, "
-          "and fully uninstall it.\n"
-          "If you truly wish to continue, execute 'adb shell pm uninstall "
-          "-k'\n.");
+                "You will have to reinstall the application with the same signature, and fully uninstall it.\n"
+                "If you truly wish to continue, execute 'adb shell pm uninstall -k'\n.");
