--- a/home/whalien/codebase/cpp/mergebot/eva/output/tmux/418ba990-58bb6f8c-8903c1f1/cmd.no_comments_mergebot.c
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/tmux/418ba990-58bb6f8c-8903c1f1/cmd.no_comments_merged.c
@@ -4 +3,0 @@
-#include <paths.h>
@@ -837,11 +991,0 @@ const char *cmd_get_default_path(struct cmd_q *cmdq, const char *cwd) {
-<<<<<<< HEAD
-    if (ctx->cmdclient != NULL && ctx->cmdclient->cwd != NULL)
-      root = ctx->cmdclient->cwd;
-    else if (ctx->curclient != NULL && s->curw != NULL)
-      root = osdep_get_cwd(s->curw->window->active->fd);
-|||||||
-    if (ctx->cmdclient != NULL && ctx->cmdclient->cwd != NULL)
-      root = ctx->cmdclient->cwd;
-    else if (ctx->curclient != NULL && s->curw != NULL)
-      root = get_proc_cwd(s->curw->window->active->fd);
-=======
@@ -852 +995,0 @@ const char *cmd_get_default_path(struct cmd_q *cmdq, const char *cwd) {
->>>>>>> 58bb6f8c5650d496fb3b872766c0278aa024631d
