diff --git a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@utils@confighandler.no_comments_mergebot.cpp b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@utils@confighandler.no_comments_merged.cpp
index 72e4fbd..76b2877 100644
--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@utils@confighandler.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@utils@confighandler.no_comments_merged.cpp
@@ -3 +2,0 @@
-#include "src/tools/storage/storagemanager.h"
@@ -351,0 +351,8 @@ void ConfigHandler::setCopyPathAfterSaveEnabled(const bool value)
+QString ConfigHandler::saveAfterCopyPathValue()
+{
+    return m_settings.value(QStringLiteral("saveAfterCopyPath")).toString();
+}
+void ConfigHandler::setSaveAfterCopyPath(const QString& path)
+{
+    m_settings.setValue(QStringLiteral("saveAfterCopyPath"), path);
+}
