--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@config@shortcutswidget.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@config@shortcutswidget.no_comments_merged.cpp
@@ -62,8 +61,0 @@ void ShortcutsWidget::initInfoTable()
-<<<<<<< HEAD
-        const auto key_sequence = identifier.isEmpty()
-                                    ? default_key_sequence
-                                    : m_config.shortcut(identifier);
-        QTableWidgetItem* item = new QTableWidgetItem(key_sequence);
-|||||||
-        QTableWidgetItem* item = new QTableWidgetItem(m_shortcuts.at(i).at(2));
-=======
@@ -77 +68,0 @@ void ShortcutsWidget::initInfoTable()
->>>>>>> b1ba67ac4d2641f137c0837a7908432f606298ac
@@ -131,0 +123,12 @@ void ShortcutsWidget::slotShortcutCellClicked(int row, int col)
+#if (defined(Q_OS_MAC) || defined(Q_OS_MAC64) || defined(Q_OS_MACOS) || \
+     defined(Q_OS_MACX))
+const QString& ShortcutsWidget::nativeOSHotKeyText(const QString& text)
+{
+    m_res = text;
+    m_res.replace("Ctrl+", "⌘");
+    m_res.replace("Alt+", "⌥");
+    m_res.replace("Meta+", "⌃");
+    m_res.replace("Shift+", "⇧");
+    return m_res;
+}
+#endif
