--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@widgets@capture@capturewidget.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@widgets@capture@capturewidget.no_comments_merged.cpp
@@ -2 +1,0 @@
-#include "src/tools/storage/storagemanager.h"
@@ -13,2 +11,0 @@
-#include "src/widgets/updatenotificationwidget.h"
-#include <QDateTime>
@@ -16,0 +14 @@
+#include "src/widgets/updatenotificationwidget.h"
@@ -17,0 +16 @@
+#include <QDateTime>
