--- a/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@widgets@panel@utilitypanel.no_comments_mergebot.cpp
+++ b/home/whalien/codebase/cpp/mergebot/eva/output/flameshot/c84b6df7-b1ba67ac-d1274add/src@widgets@panel@utilitypanel.no_comments_merged.cpp
@@ -24,4 +23,0 @@ UtilityPanel::UtilityPanel(QWidget* parent)
-<<<<<<< HEAD
-    hide();
-|||||||
-=======
@@ -32 +27,0 @@ UtilityPanel::UtilityPanel(QWidget* parent)
->>>>>>> b1ba67ac4d2641f137c0837a7908432f606298ac
@@ -111,7 +105,0 @@ void UtilityPanel::initInternalPanel()
-<<<<<<< HEAD
-    m_hide = new QPushButton();
-    m_hide->setText(tr("Hide"));
-    m_upLayout->addWidget(m_hide);
-    connect(m_hide, SIGNAL(clicked()), this, SLOT(slotHidePanel()));
-|||||||
-    m_internalPanel->hide();
@@ -122 +109,0 @@ void UtilityPanel::initInternalPanel()
-=======
@@ -124 +110,0 @@ void UtilityPanel::initInternalPanel()
->>>>>>> b1ba67ac4d2641f137c0837a7908432f606298ac
