diff --git a/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_mergegen.cpp b/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_merged.cpp
index 482dc15..10023a1 100644
--- a/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_mergegen.cpp
+++ b/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_merged.cpp
@@ -1351 +1351,2 @@ void WindowLeadLagExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-const auto width = MinValue(delta, target_limit); leadlag_default.CopyCell(result, i, width);
+    const auto width = MinValue(delta, target_limit);
+    leadlag_default.CopyCell(result, i, width);
