diff --git a/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_conflict.cpp b/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_merged.cpp
index 6d4eb11..10023a1 100644
--- a/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_conflict.cpp
+++ b/./output/duckdb/b4e61168-104cfacc-f162a9ae/src@execution@window_executor.no_comments_merged.cpp
@@ -1351,7 +1350,0 @@ void WindowLeadLagExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-<<<<<<< HEAD
-    const auto width = MinValue(delta, count - i);
-    leadlag_default.CopyCell(result, i, width);
-||||||| f162a9aeaf
-    const auto width = MinValue(delta, count - i);
-    llstate.leadlag_default.CopyCell(result, i, width);
-=======
@@ -1359,2 +1352 @@ void WindowLeadLagExecutor::EvaluateInternal(WindowExecutorGlobalState &gstate,
-    llstate.leadlag_default.CopyCell(result, i, width);
->>>>>>> 104cfacc
+    leadlag_default.CopyCell(result, i, width);
