diff --git a/./output/frameworks_av/192831ab-da52cc20-7f82d1a7/media@codec2@sfplugin@CCodec.no_comments_conflict.cpp b/./output/frameworks_av/192831ab-da52cc20-7f82d1a7/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
index 8bc6886..2a18c04 100644
--- a/./output/frameworks_av/192831ab-da52cc20-7f82d1a7/media@codec2@sfplugin@CCodec.no_comments_conflict.cpp
+++ b/./output/frameworks_av/192831ab-da52cc20-7f82d1a7/media@codec2@sfplugin@CCodec.no_comments_merged.cpp
@@ -1094 +1093,0 @@ void CCodec::start() {
-<<<<<<< HEAD
@@ -1097,6 +1095,0 @@ void CCodec::start() {
-||||||| 7f82d1a74e
-    err2 = mChannel->start(inputFormat, outputFormat);
-=======
-    (void)mSentConfigAfterResume.test_and_set();
-    err2 = mChannel->start(inputFormat, outputFormat);
->>>>>>> da52cc20
@@ -1319 +1311,0 @@ void CCodec::signalResume() {
-<<<<<<< HEAD
@@ -1331,10 +1322,0 @@ void CCodec::signalResume() {
-||||||| 7f82d1a74e
-    (void)mChannel->start(nullptr, nullptr);
-=======
-    mSentConfigAfterResume.clear();
-    {
-        Mutexed<Config>::Locked config(mConfig);
-        config->queryConfiguration(comp);
-    }
-    (void)mChannel->start(nullptr, nullptr);
->>>>>>> da52cc20
@@ -1505 +1486,0 @@ void CCodec::onMessageReceived(const sp<AMessage> &msg) {
-<<<<<<< HEAD
@@ -1509,7 +1489,0 @@ void CCodec::onMessageReceived(const sp<AMessage> &msg) {
-||||||| 7f82d1a74e
-            Mutexed<Config>::Locked config(mConfig);
-            bool changed = false;
-=======
-            Mutexed<Config>::Locked config(mConfig);
-            bool changed = !mSentConfigAfterResume.test_and_set();
->>>>>>> da52cc20
