diff --git a/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_conflict.cpp b/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
index 7ef9393..c6de7b8 100644
--- a/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_conflict.cpp
+++ b/./output/frameworks_av/2a4f1676-6ddf1d0a-92765b5c/services@audiopolicy@service@AudioPolicyInterfaceImpl.no_comments_merged.cpp
@@ -1314,0 +1315,13 @@ Status AudioPolicyService::isDirectOutputSupported(
+template <typename Port>
+void anonymizePortBluetoothAddress(Port *port) {
+    if (port->type != AUDIO_PORT_TYPE_DEVICE) {
+        return;
+    }
+    if (!(audio_is_a2dp_device(port->ext.device.type)
+            || audio_is_ble_device(port->ext.device.type)
+            || audio_is_bluetooth_sco_device(port->ext.device.type)
+            || audio_is_hearing_aid_out_device(port->ext.device.type))) {
+        return;
+    }
+    anonymizeBluetoothAddress(port->ext.device.address);
+}
@@ -1331,7 +1343,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-<<<<<<< HEAD
-    const AttributionSourceState attributionSource = getCallingAttributionSource();
-||||||| 92765b5cf0
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
@@ -1339,6 +1344,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-=======
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
->>>>>>> 6ddf1d0a
@@ -1346 +1345,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-<<<<<<< HEAD
@@ -1362,14 +1360,0 @@ Status AudioPolicyService::listAudioPorts(media::AudioPortRole roleAidl,
-||||||| 92765b5cf0
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
-            mAudioPolicyManager->listAudioPorts(role, type, &num_ports, ports.get(), &generation)));
-    numPortsReq = std::min(numPortsReq, num_ports);
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        for (size_t i = 0; i < numPortsReq; ++i) {
-            anonymizePortBluetoothAddress(&ports[i]);
-        }
-    }
-=======
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
-            mAudioPolicyManager->listAudioPorts(role, type, &num_ports, ports.get(), &generation)));
-    numPortsReq = std::min(numPortsReq, num_ports);
->>>>>>> 6ddf1d0a
@@ -1396 +1380,0 @@ Status AudioPolicyService::getAudioPort(int portId,
-<<<<<<< HEAD
@@ -1398,12 +1381,0 @@ Status AudioPolicyService::getAudioPort(int portId,
-||||||| 92765b5cf0
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
-    const AttributionSourceState attributionSource = getCallingAttributionSource();
-=======
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
->>>>>>> 6ddf1d0a
@@ -1411 +1382,0 @@ Status AudioPolicyService::getAudioPort(int portId,
-<<<<<<< HEAD
@@ -1422,8 +1392,0 @@ Status AudioPolicyService::getAudioPort(int portId,
-||||||| 92765b5cf0
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(mAudioPolicyManager->getAudioPort(&port)));
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        anonymizePortBluetoothAddress(&port);
-    }
-=======
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(mAudioPolicyManager->getAudioPort(&port)));
->>>>>>> 6ddf1d0a
@@ -1482,7 +1444,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-<<<<<<< HEAD
-    const AttributionSourceState attributionSource = getCallingAttributionSource();
-||||||| 92765b5cf0
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
@@ -1490,6 +1445,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-=======
-    Mutex::Autolock _l(mLock);
-    if (mAudioPolicyManager == NULL) {
-        return binderStatusFromStatusT(NO_INIT);
-    }
->>>>>>> 6ddf1d0a
@@ -1497 +1446,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-<<<<<<< HEAD
@@ -1517,19 +1465,0 @@ Status AudioPolicyService::listAudioPatches(Int* count,
-||||||| 92765b5cf0
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
-            mAudioPolicyManager->listAudioPatches(&num_patches, patches.get(), &generation)));
-    numPatchesReq = std::min(numPatchesReq, num_patches);
-    if (mustAnonymizeBluetoothAddress(attributionSource, String16(__func__))) {
-        for (size_t i = 0; i < numPatchesReq; ++i) {
-            for (size_t j = 0; j < patches[i].num_sources; ++j) {
-                anonymizePortBluetoothAddress(&patches[i].sources[j]);
-            }
-            for (size_t j = 0; j < patches[i].num_sinks; ++j) {
-                anonymizePortBluetoothAddress(&patches[i].sinks[j]);
-            }
-        }
-    }
-=======
-    RETURN_IF_BINDER_ERROR(binderStatusFromStatusT(
-            mAudioPolicyManager->listAudioPatches(&num_patches, patches.get(), &generation)));
-    numPatchesReq = std::min(numPatchesReq, num_patches);
->>>>>>> 6ddf1d0a
